diff -ruBbw a/extra/Python/setup.py b/extra/Python/setup.py
--- a/extra/Python/setup.py	2010-09-26 19:42:26.000000000 +0200
+++ b/extra/Python/setup.py	2025-08-06 18:45:54.895173657 +0200
@@ -11,11 +11,11 @@
   NUMPYPATH = p + '/Lib/site-packages/numpy/core/include'
 else:
   NUMPY = 'NONUMPY'
-print 'numpy: ' + NUMPY
+print('numpy: ' + NUMPY)
 windir = getenv('windir')
 if windir == None:
   WIN32 = 'NOWIN32'
-  LPSOLVE55 = '../../lpsolve55/bin/ux32'
+  LPSOLVE55 = '../../lpsolve55/bin/ux64'
 else:
   WIN32 = 'WIN32'
   LPSOLVE55 = '../../lpsolve55/bin/win32'
@@ -27,10 +27,10 @@
        url = "http://www.peno.be/",
        py_modules=['lp_solve', 'lp_maker'],
        ext_modules = [Extension("lpsolve55",
-				["lpsolve.c", "hash.c", "pythonmod.c"],
+				["lpsolve.c", "pythonmod.c"],
                                 define_macros=[('PYTHON', '1'), (WIN32, '1'), ('NODEBUG', '1'), ('DINLINE', 'static'), (NUMPY, '1'), ('_CRT_SECURE_NO_WARNINGS', '1')],
                                 include_dirs=['../..', NUMPYPATH],
                                 library_dirs=[LPSOLVE55],
-				libraries = ["lpsolve55"])
+				libraries = ["lpsolve55", "colamd"])
 		      ]
 )
