Link binaries with the shared antimony library.

Åke Sandgren, 2024-01-04
diff -ru antimony-2.14.orig/CMakeLists.txt antimony-2.14/CMakeLists.txt
--- antimony-2.14.orig/CMakeLists.txt	2023-12-19 21:11:13.000000000 +0100
+++ antimony-2.14/CMakeLists.txt	2024-01-04 10:33:10.353865750 +0100
@@ -657,41 +657,41 @@
 #
 
 add_executable(rehashantimony ${ANTIMONY_SRC_DIR}rehashantimony.cpp)
-add_dependencies(rehashantimony ${LIBANTIMONY_LIBRARY}-static)
-target_link_libraries(rehashantimony ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS})
+add_dependencies(rehashantimony ${LIBANTIMONY_LIBRARY})
+target_link_libraries(rehashantimony ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS})
 install(TARGETS rehashantimony DESTINATION bin)
 
 add_executable(testantimony ${ANTIMONY_SRC_DIR}testantimony.cpp)
-add_dependencies(testantimony ${LIBANTIMONY_LIBRARY}-static)
-target_link_libraries(testantimony ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS})
+add_dependencies(testantimony ${LIBANTIMONY_LIBRARY})
+target_link_libraries(testantimony ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS})
 install(TARGETS testantimony DESTINATION bin)
 
 add_executable(sbtranslate ${ANTIMONY_SRC_DIR}sbtranslate.cpp)
-add_dependencies(sbtranslate ${LIBANTIMONY_LIBRARY}-static)
-target_link_libraries(sbtranslate ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS})
+add_dependencies(sbtranslate ${LIBANTIMONY_LIBRARY})
+target_link_libraries(sbtranslate ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS})
 install(TARGETS sbtranslate DESTINATION bin)
 
 if(WITH_SBML)
   add_executable(antimony2sbml ${ANTIMONY_SRC_DIR}antimony2sbml.cpp)
-  add_dependencies(antimony2sbml ${LIBANTIMONY_LIBRARY}-static)
-  target_link_libraries(antimony2sbml ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS})
+  add_dependencies(antimony2sbml ${LIBANTIMONY_LIBRARY})
+  target_link_libraries(antimony2sbml ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS})
   install(TARGETS antimony2sbml DESTINATION bin)
 
   add_executable(sbml2antimony ${ANTIMONY_SRC_DIR}sbml2antimony.cpp)
-  add_dependencies(sbml2antimony ${LIBANTIMONY_LIBRARY}-static)
-  target_link_libraries(sbml2antimony ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS})
+  add_dependencies(sbml2antimony ${LIBANTIMONY_LIBRARY})
+  target_link_libraries(sbml2antimony ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS})
   install(TARGETS sbml2antimony DESTINATION bin)
 endif(WITH_SBML)
 
 if(WITH_CELLML)
   add_executable(antimony2cellml ${ANTIMONY_SRC_DIR}antimony2cellml.cpp)
-  add_dependencies(antimony2cellml ${LIBANTIMONY_LIBRARY}-static)
-  target_link_libraries(antimony2cellml ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS})
+  add_dependencies(antimony2cellml ${LIBANTIMONY_LIBRARY})
+  target_link_libraries(antimony2cellml ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS})
   install(TARGETS antimony2cellml DESTINATION bin)
 
   add_executable(cellml2antimony ${ANTIMONY_SRC_DIR}cellml2antimony.cpp)
-  add_dependencies(cellml2antimony ${LIBANTIMONY_LIBRARY}-static)
-  target_link_libraries(cellml2antimony ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS})
+  add_dependencies(cellml2antimony ${LIBANTIMONY_LIBRARY})
+  target_link_libraries(cellml2antimony ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS})
   install(TARGETS cellml2antimony DESTINATION bin)
 endif(WITH_CELLML)
 
@@ -810,7 +810,7 @@
         INCLUDE_DIRECTORIES(${QTANTIMONY_DIALOG_DIR})
         INCLUDE_DIRECTORIES(${Qt5PrintSupport_INCLUDE_DIRS})
         add_executable(QTAntimony WIN32 MACOSX_BUNDLE ${QTANTIMONY_SOURCES} ${QTANTIMONY_MOC_SRCS} ${QTANTIMONY_FORMS_HEADERS} ${QTANTIMONY_HEADERS} ${QTANTIMONY_RESOURCES})
-        target_link_libraries(QTAntimony Qt5::Widgets Qt5::PrintSupport ${LIBANTIMONY_LIBRARY}-static ${LIBANTIMONY_LIBS} )
+        target_link_libraries(QTAntimony Qt5::Widgets Qt5::PrintSupport ${LIBANTIMONY_LIBRARY} ${LIBANTIMONY_LIBS} )
         add_definitions(${QT_DEFINITIONS})
         set(MACOSX_BUNDLE_ICON_FILE ${QTANTIMONY_SRC_DIR}antimony.icns)
         install(TARGETS QTAntimony DESTINATION bin)
