--- ctffind-4.0.8/Makefile.in.orig	2014-12-12 12:08:28.159297567 +0100
+++ ctffind-4.0.8/Makefile.in	2014-12-12 12:08:44.935297601 +0100
@@ -353,7 +353,7 @@
 @FC_IS_INTEL_TRUE@AM_FCFLAGS = -fpp -assume realloc_lhs -traceback \
 @FC_IS_INTEL_TRUE@	-heap-arrays -warn all -warn \
 @FC_IS_INTEL_TRUE@	notruncated_source -gen-interfaces -fpe0 \
-@FC_IS_INTEL_TRUE@	-standard-semantics -assume nostd_value \
+@FC_IS_INTEL_TRUE@	-standard-semantics \
 @FC_IS_INTEL_TRUE@	-init=snan $(am__append_27) $(am__append_30) \
 @FC_IS_INTEL_TRUE@	$(am__append_32) $(am__append_35) \
 @FC_IS_INTEL_TRUE@	$(am__append_37)
@@ -391,7 +391,7 @@
 @FC_IS_INTEL_TRUE@AM_FFLAGS = -assume realloc_lhs -traceback \
 @FC_IS_INTEL_TRUE@	-heap-arrays -warn all -warn \
 @FC_IS_INTEL_TRUE@	notruncated_source -gen-interfaces -fpe0 \
-@FC_IS_INTEL_TRUE@	-standard-semantics -assume nostd_value \
+@FC_IS_INTEL_TRUE@	-standard-semantics \
 @FC_IS_INTEL_TRUE@	-init=snan $(am__append_28) $(am__append_31) \
 @FC_IS_INTEL_TRUE@	$(am__append_33) $(am__append_34) \
 @FC_IS_INTEL_TRUE@	$(am__append_38)
