diff -ru racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/bench/Makefile racon-v1.4.13/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/bench/Makefile
--- racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/bench/Makefile	2021-03-22 19:24:46.060685315 +0000
+++ racon-v1.4.13/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/bench/Makefile	2021-03-22 19:25:56.959028278 +0000
@@ -1,5 +1,5 @@
 CXX	?= g++
-CXXFLAGS	= -march=native -Wall -Wextra -pedantic -Wconversion -std=c++11 -pthread -I../include -fmax-errors=1
+CXXFLAGS	= -Wall -Wextra -pedantic -Wconversion -std=c++11 -pthread -I../include -fmax-errors=1
 CXX_RELEASE_FLAGS = -O3 -flto -Wl,--no-as-needed
 
 
diff -ru racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/example/Makefile racon-v1.4.13/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/example/Makefile
--- racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/example/Makefile	2021-03-22 19:24:46.044685237 +0000
+++ racon-v1.4.13/vendor/ClaraGenomicsAnalysis/3rdparty/spdlog/example/Makefile	2021-03-22 19:25:24.345870515 +0000
@@ -1,6 +1,6 @@
 CXX	?= g++
 CXX_FLAGS = -Wall -Wextra -pedantic -std=c++11 -pthread -I../include -fmax-errors=1 -Wconversion
-CXX_RELEASE_FLAGS = -O3 -march=native
+CXX_RELEASE_FLAGS = -O3 
 CXX_DEBUG_FLAGS= -g
 
 all:	example 
diff -ru racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/cudaaligner/CMakeLists.txt racon-v1.4.13/vendor/ClaraGenomicsAnalysis/cudaaligner/CMakeLists.txt
--- racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/cudaaligner/CMakeLists.txt	2021-03-22 19:24:46.229686132 +0000
+++ racon-v1.4.13/vendor/ClaraGenomicsAnalysis/cudaaligner/CMakeLists.txt	2021-03-22 19:26:03.967062179 +0000
@@ -39,7 +39,7 @@
 
 target_link_libraries(cudaaligner utils logging)
 
-target_compile_options(cudaaligner PRIVATE -Werror -Wall -Wextra -march=native)
+target_compile_options(cudaaligner PRIVATE -Werror -Wall -Wextra)
 
 target_include_directories(cudaaligner 
     PUBLIC 
diff -ru racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/cudapoa/CMakeLists.txt racon-v1.4.13/vendor/ClaraGenomicsAnalysis/cudapoa/CMakeLists.txt
--- racon-v1.4.13.orig/vendor/ClaraGenomicsAnalysis/cudapoa/CMakeLists.txt	2021-03-22 19:24:46.159685794 +0000
+++ racon-v1.4.13/vendor/ClaraGenomicsAnalysis/cudapoa/CMakeLists.txt	2021-03-22 19:25:17.882839251 +0000
@@ -27,7 +27,7 @@
 
 target_link_libraries(cudapoa utils logging)
 
-target_compile_options(cudapoa PRIVATE -Werror -Wall -Wextra -march=native)
+target_compile_options(cudapoa PRIVATE -Werror -Wall -Wextra)
 
 if(cga_enable_cudapoa_nw_print)
     target_compile_definitions(cudapoa PUBLIC NW_VERBOSE_PRINT)
diff -ru racon-v1.4.13.orig/vendor/spoa/CMakeLists.txt racon-v1.4.13/vendor/spoa/CMakeLists.txt
--- racon-v1.4.13.orig/vendor/spoa/CMakeLists.txt	2021-03-22 19:24:45.091680627 +0000
+++ racon-v1.4.13/vendor/spoa/CMakeLists.txt	2021-03-22 19:26:17.828129230 +0000
@@ -20,7 +20,7 @@
 if (spoa_optimize_for_portability)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.1")
 elseif (spoa_optimize_for_native)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ")
 endif()
 
 # build SPOA as a static library by default
