change hardcoded zlib dependency for bioparser

author: Pavel Grochal (INUITS)
diff -ru bioparser-0d66534d75b319f43ce5c59fe6548eee48f8f651.orig/CMakeLists.txt bioparser-0d66534d75b319f43ce5c59fe6548eee48f8f651/CMakeLists.txt
--- bioparser-0d66534d75b319f43ce5c59fe6548eee48f8f651.orig/CMakeLists.txt	2019-08-17 12:32:49.000000000 +0200
+++ bioparser-0d66534d75b319f43ce5c59fe6548eee48f8f651/CMakeLists.txt	2020-04-02 11:08:19.991863794 +0200
@@ -14,16 +14,12 @@

 add_library(bioparser INTERFACE)

+find_package(ZLIB REQUIRED)
 target_include_directories(bioparser INTERFACE
     ${PROJECT_SOURCE_DIR}/include
-    ${PROJECT_SOURCE_DIR}/vendor/zlib
-    ${PROJECT_BINARY_DIR}/vendor/zlib)
+    ${ZLIB_INCLUDE_DIRS})

-if (NOT TARGET zlib)
-    add_subdirectory(vendor/zlib EXCLUDE_FROM_ALL)
-endif()
-
-target_link_libraries(bioparser INTERFACE zlibstatic)
+target_link_libraries(bioparser INTERFACE z)

 if (bioparser_build_tests)
     set(bioparser_test_data_path ${PROJECT_SOURCE_DIR}/test/data/)
