--- ./ext/Errno/Errno_pm.PL.orig	2012-12-19 12:25:24.000000000 +0000
+++ ./ext/Errno/Errno_pm.PL	2018-03-09 17:25:07.876398350 +0000
@@ -144,8 +144,8 @@
 	     ) {
 	# Some Linuxes have weird errno.hs which generate
 	# no #file or #line directives
-	my $linux_errno_h = -e '/usr/include/errno.h' ?
-	    '/usr/include/errno.h' : '/usr/local/include/errno.h';
+       my $linux_errno_h = -e '/usr/include/errno.h' ?
+       '/usr/include/errno.h' : -e '/usr/local/include/errno.h' ? '/usr/local/include/errno.h' : '/cvmfs/soft.computecanada.ca/nix/var/nix/profiles/16.09/include/errno.h';
 	$file{$linux_errno_h} = 1;
     } elsif ($^O eq 'beos' || $^O eq 'haiku') {
 	# hidden in a special place
