* fix function declaration to match previous definition
author: Miguel Dias Costa (National University of Singapore)
--- toglDecls.h.orig	2020-01-17 14:57:44.000178028 +0800
+++ toglDecls.h	2020-01-17 14:58:07.046072088 +0800
@@ -117,7 +117,7 @@
 EXTERN Bool		Togl_UpdatePending _ANSI_ARGS_((const Togl * togl));
 /* 38 */
 EXTERN int		Togl_WriteObj _ANSI_ARGS_((const Togl * togl, 
-				const Tcl_Obj * toglfont, Tcl_Obj * obj));
+				const Tcl_Obj * toglfont, const Tcl_Obj * obj));
 /* 39 */
 EXTERN int		Togl_WriteChars _ANSI_ARGS_((const Togl * togl, 
 				const Tcl_Obj * toglfont, const char * str, 
