function not yet available in Torch 2.1.2
--- a/dorado/torch_utils/torch_utils.cpp
+++ b/dorado/torch_utils/torch_utils.cpp
@@ -25,5 +25,5 @@
-#if TORCH_VERSION_MAJOR >= 2 && TORCH_VERSION_MINOR > 0
+#if TORCH_VERSION_MAJOR >= 2 && TORCH_VERSION_MINOR > 1
     // We don't want empty tensors to be initialised with data since we always overwrite them.
     torch::globalContext().setDeterministicFillUninitializedMemory(false);
 #endif
