--- rmark/Makefile.in.orig	2023-09-06 13:55:10.000000000 +0000
+++ rmark/Makefile.in	2024-03-18 17:06:49.247205219 +0000
@@ -10,7 +10,7 @@
 CPPFLAGS       = @CPPFLAGS@
 LDFLAGS        = @LDFLAGS@
 DEFS           = @DEFS@
-LIBS           = -lhmmer -leasel @LIBS@ @PTHREAD_LIBS@ -lm
+LIBS           = -lhmmer -leasel @LIBS@ @PTHREAD_LIBS@ -lm -lgsl -lgslcblas
 IMPLDIR        = impl_@IMPL_CHOICE@
 
 SRCDIR   = ../src
