diff -ru virtualenv-20.13.0.orig/src/virtualenv/run/plugin/base.py virtualenv-20.13.0/src/virtualenv/run/plugin/base.py
--- virtualenv-20.13.0.orig/src/virtualenv/run/plugin/base.py	2022-02-15 21:01:37.101426157 +0000
+++ virtualenv-20.13.0/src/virtualenv/run/plugin/base.py	2022-02-15 21:01:56.882548797 +0000
@@ -3,7 +3,7 @@
 import sys
 from collections import OrderedDict
 
-if sys.version_info >= (3, 8):
+if sys.version_info >= (3, 10):
     from importlib.metadata import entry_points
 
     importlib_metadata_version = ()
