--- ./third-party/gasnet/gasnet-src/ofi-conduit/gasnet_core.c.orig	2021-09-20 13:29:12.000000000 -0700
+++ ./third-party/gasnet/gasnet-src/ofi-conduit/gasnet_core.c	2021-12-03 12:14:59.144649814 -0800
@@ -142,10 +142,6 @@
 
   if (! gasneti_mynode) {
     fflush(NULL);
-    fprintf(stderr,
-      " WARNING: ofi-conduit is experimental and should not be used for\n"
-      "          performance measurements.\n"
-      "          Please see `ofi-conduit/README` for more details.\n");
     fflush(NULL);
   }
 
