--- third-party/gasnet/Makefile.include.orig	2017-07-20 20:55:46.057199427 +0000
+++ third-party/gasnet/Makefile.include	2017-07-20 20:56:32.272800689 +0000
@@ -8,6 +8,7 @@
 GASNET_LIB_DIR = $(GASNET_INSTALL_DIR)/lib
 GASNET_INC_MAKEFILE = $(GASNET_INSTALL_DIR)/include/$(CHPL_MAKE_COMM_SUBSTRATE)-conduit/$(CHPL_MAKE_COMM_SUBSTRATE)-par.mak
 GASNET_SETUP_MAKEFILE = $(GASNET_DIR)/Makefile.setup
+CHPL_GASNET_CFG_OPTIONS += --with-pmi=/dev/shm/masao/slurm/17.02.5/
 
 # decide whether to do a debug or no-debug build
 ifdef CHPL_COMM_DEBUG
