diff -ruN cellpose-omni-0.6.5_orig/setup.py cellpose-omni-0.6.5/setup.py
--- cellpose-omni-0.6.5_orig/setup.py	2021-05-08 00:57:04.000000000 +0200
+++ cellpose-omni-0.6.5/setup.py	2023-07-14 12:09:18.942712815 +0200
@@ -4,7 +4,7 @@
 install_deps = ['numpy', 'scipy', 'natsort',
                 'tifffile', 'tqdm', 'numba', 
                 'torch>=1.6',
-                'opencv-python-headless']
+                'opencv-contrib-python']
 
 try:
     import torch
@@ -20,6 +20,7 @@
 
 setup(
     name="cellpose",
+    version='0.6.5',
     license="BSD",
     author="Marius Pachitariu and Carsen Stringer",
     author_email="stringerc@janelia.hhmi.org",
@@ -32,7 +33,7 @@
       'setuptools_scm',
     ],
     packages=setuptools.find_packages(),
-    use_scm_version=True,
+    use_scm_version=False,
     install_requires = install_deps,
     tests_require=[
       'pytest'