Default to BLAST+ as search engine

Åke Sandgren, 2024-10-16
diff -ru DAS_Tool-1.1.7.orig/src/DAS_Tool.R DAS_Tool-1.1.7/src/DAS_Tool.R
--- DAS_Tool-1.1.7.orig/src/DAS_Tool.R	2024-01-09 00:20:13.000000000 +0100
+++ DAS_Tool-1.1.7/src/DAS_Tool.R	2024-10-16 11:39:26.418829032 +0200
@@ -38,7 +38,7 @@
    -c --contigs=<contigs>                   Contigs in fasta format.
    -o --outputbasename=<outputbasename>     Basename of output files.
    -l --labels=<labels>                     Comma separated list of binning prediction names.
-   --search_engine=<search_engine>          Engine used for single copy gene identification (diamond/blastp/usearch) [default: diamond].
+   --search_engine=<search_engine>          Engine used for single copy gene identification (diamond/blastp/usearch) [default: blastp].
    -p --proteins=<proteins>                 Predicted proteins (optional) in prodigal fasta format (>contigID_geneNo).
                                             Gene prediction step will be skipped.
    --write_bin_evals                        Write evaluation of input bin sets.
@@ -324,8 +324,8 @@
 searchEngine <- tolower(arguments$search_engine)
 if(!searchEngine %in% c('diamond', 'usearch', 'blastp')){
    write.log(paste0('Unknown argument for --search_engine: ',arguments$search_engine,'\n',
-                    'Defaulting to diamond'),filename = logFile,append = T,write_to_file = T,type = 'warning')
-   searchEngine <- 'diamond'
+                    'Defaulting to blastp'),filename = logFile,append = T,write_to_file = T,type = 'warning')
+   searchEngine <- 'blastp'
 }
 
 # Check dependencies
