diff -Nru dftd3-lib-0.10.orig/prg/extras.f90 dftd3-lib-0.10/prg/extras.f90
--- dftd3-lib-0.10.orig/prg/extras.f90	2022-03-09 10:15:01.823373191 +0100
+++ dftd3-lib-0.10/prg/extras.f90	2022-03-09 10:21:47.048668456 +0100
@@ -287,7 +287,7 @@
     close(11)
 
     write(*,*) 'checksum (Edisp) ',check
-    if (abs(check-etot).gt.1.d-3)stop'something is weired in adisp'
+    if (abs(check-etot).gt.1.d-3) stop 'something is weired in adisp'
 
     inquire(file='fragment',exist=ex)
     if (.not.ex) return
@@ -1739,7 +1739,7 @@
     close(11)
 
     write(*,*) 'checksum (Edisp) ',check
-    if (abs(check-etot).gt.1.d-3)stop'something is weired in adisp'
+    if (abs(check-etot).gt.1.d-3) stop 'something is weired in adisp'
 
     deallocate(dist,li)
     return
