Fix syntax of extras.f90
Author: Alex Domingo (Vrije Universiteit Brussel)
--- prg/extras.f90.orig	2020-04-07 15:53:37.579346000 +0200
+++ prg/extras.f90	2020-04-07 15:53:47.270025090 +0200
@@ -285,7 +285,7 @@
     close(11)
 
     write(*,*) 'checksum (Edisp) ',check
-    if (abs(check-etot).gt.1.d-3)stop'something is weired in adisp'
+    if (abs(check-etot).gt.1.d-3) stop 'something is weired in adisp'
 
     inquire(file='fragment',exist=ex)
     if (.not.ex) return
@@ -1746,7 +1746,7 @@
     close(11)
 
     write(*,*) 'checksum (Edisp) ',check
-    if (abs(check-etot).gt.1.d-3)stop'something is weired in adisp'
+    if (abs(check-etot).gt.1.d-3) stop 'something is weired in adisp'
 
     deallocate(dist,li)
     return
