--- extras.f90	2022-03-08 15:29:22.336522856 +0100
+++ extras_new.f90	2022-03-08 15:30:50.750421061 +0100
@@ -287,7 +287,7 @@
     close(11)
 
     write(*,*) 'checksum (Edisp) ',check
-    if (abs(check-etot).gt.1.d-3)stop'something is weired in adisp'
+    if (abs(check-etot).gt.1.d-3) stop'something is weired in adisp'
 
     inquire(file='fragment',exist=ex)
     if (.not.ex) return
@@ -1739,7 +1739,7 @@
     close(11)
 
     write(*,*) 'checksum (Edisp) ',check
-    if (abs(check-etot).gt.1.d-3)stop'something is weired in adisp'
+    if (abs(check-etot).gt.1.d-3) stop'something is weired in adisp'
 
     deallocate(dist,li)
     return
