--- lib/vmalloc/malloc.c.orig	2013-10-04 14:55:15.718833040 +0200
+++ lib/vmalloc/malloc.c	2013-10-04 14:55:30.088915918 +0200
@@ -399,28 +399,6 @@
 }
 #endif
 
-#if _lib_mstats
-#if __STD_C
-struct mstats mstats(void)
-#else
-struct mstats mstats()
-#endif
-{
-    Vmstat_t sb;
-    struct mstats ms;
-
-    VMFLINIT();
-    memset(&ms, 0, sizeof(ms));
-    if (vmstat(Vmregion, &sb) >= 0) {
-	ms.bytes_total = sb.extent;
-	ms.chunks_used = sb.n_busy;
-	ms.bytes_used = sb.s_busy;
-	ms.chunks_free = sb.n_free;
-	ms.bytes_free = sb.s_free;
-    }
-    return ms;
-}
-#endif
 
 #endif/*_hdr_malloc*/
 
