# -fPIC flag is also needed for RISC-V builds
--- Makefile.fip.orig	2025-01-16 12:58:29.624119756 +0100
+++ Makefile.fip	2025-01-16 13:00:01.016862229 +0100
@@ -33,6 +33,11 @@
 	CXXFLAGS += -fPIC
 endif
 
+ifeq ($(shell sh -c 'uname -m 2>/dev/null || echo not'),riscv64)
+	CFLAGS += -fPIC
+	CXXFLAGS += -fPIC
+endif
+
 TARGET  = freeimageplus
 STATICLIB = lib$(TARGET).a
 SHAREDLIB = lib$(TARGET)-$(VER_MAJOR).$(VER_MINOR).so
