Add missing wchar.h include

author: micketeer@gmail.com
--- Source/LibJXR/jxrgluelib/JXRGlueJxr.c.orig	2025-10-04 13:45:25.101217200 +0000
+++ Source/LibJXR/jxrgluelib/JXRGlueJxr.c	2025-10-04 13:45:40.498990568 +0000
@@ -28,6 +28,7 @@
 //*@@@---@@@@******************************************************************
 #include <limits.h>
 #include <JXRGlue.h>
+#include <wchar.h>
 
 
 static const char szHDPhotoFormat[] = "<dc:format>image/vnd.ms-photo</dc:format>";
