Add missing unistd.h include

author: micketeer@gmail.com
--- Source/ZLib/gzwrite.c.orig	2025-10-02 23:57:01.370230707 +0000
+++ Source/ZLib/gzwrite.c	2025-10-02 23:57:11.692078929 +0000
@@ -4,6 +4,7 @@
  */
 
 #include "gzguts.h"
+#include <unistd.h>
 
 /* Local functions */
 local int gz_init OF((gz_statep));
--- Source/ZLib/gzlib.c.orig	2025-10-02 23:56:06.164042487 +0000
+++ Source/ZLib/gzlib.c	2025-10-02 23:56:22.804797793 +0000
@@ -4,6 +4,7 @@
  */
 
 #include "gzguts.h"
+#include <unistd.h>
 
 #if defined(_WIN32) && !defined(__BORLANDC__) && !defined(__MINGW32__)
 #  define LSEEK _lseeki64
--- Source/ZLib/gzread.c.orig	2025-10-02 23:56:35.215615298 +0000
+++ Source/ZLib/gzread.c	2025-10-02 23:56:49.416406482 +0000
@@ -4,6 +4,7 @@
  */
 
 #include "gzguts.h"
+#include <unistd.h>
 
 /* Local functions */
 local int gz_load OF((gz_statep, unsigned char *, unsigned, unsigned *));
