# -fPIC flag is also needed for RISC-V builds
--- Makefile.gnu.orig	2025-01-15 17:26:47.616603010 +0100
+++ Makefile.gnu	2025-01-15 17:28:09.158140964 +0100
@@ -33,6 +33,11 @@
 	CXXFLAGS += -fPIC
 endif
 
+ifeq ($(shell sh -c 'uname -m 2>/dev/null || echo not'),riscv64)
+	CFLAGS += -fPIC
+	CXXFLAGS += -fPIC
+endif
+
 TARGET  = freeimage
 STATICLIB = lib$(TARGET).a
 SHAREDLIB = lib$(TARGET)-$(VER_MAJOR).$(VER_MINOR).so
