diff -ru baypass_2.1.orig/sources/Makefile baypass_2.1/sources/Makefile
--- baypass_2.1.orig/sources/Makefile	2018-11-14 22:48:07.692963690 +0000
+++ baypass_2.1/sources/Makefile	2018-11-14 22:48:40.525662592 +0000
@@ -14,7 +14,7 @@
  endif
 else
  FFLAGS :=  -O3 -r8 -traceback
- FFLAGS2 := -O3 -r8 -openmp -nowarn -traceback
+ FFLAGS2 := -O3 -r8 -fopenmp -nowarn -traceback
  LDFLAGS := -static
  LDFLAGS2 := 
  EXEC := i_baypass
