use FlexBlas
author: John Dey Fred Hutch 
diff -ruN spams-python-main.orig/setup.py spams-python-main/setup.py
--- spams-python-main.orig/setup.py	2022-04-01 07:48:20.000000000 -0700
+++ spams-python-main/setup.py	2023-05-23 12:14:10.610242000 -0700
@@ -126,7 +126,7 @@
                 libdirs.append(_)
         libs.extend(['mkl_rt'])
     else:
-        libs.extend(['blas', 'lapack'])
+        libs.extend(['flexiblas', 'lapack'])
 
     # openMP
     if check_openmp() == 0:
