--- bin/beast.orig	2018-11-16 18:34:48.520496791 +0000
+++ bin/beast	2018-11-16 18:39:59.384662623 +0000
@@ -40,6 +40,15 @@
 	fi
 fi
 
+# Explicitly add "${EBROOTBEAGLEMINLIB}/lib" to library search path to ensure 
+# beast continues to find beagle when installed here.  (This is
+# necessary due to known problems with certain JREs.)
+if [ -z "$LD_LIBRARY_PATH" ]; then
+    export LD_LIBRARY_PATH="${EBROOTBEAGLEMINLIB}/lib"
+else
+    export LD_LIBRARY_PATH="$LD_LIBRARY_PATH":"${EBROOTBEAGLEMINLIB}/lib"
+fi
+
 # use BEAST_EXTRA_LIBS variable to load BEAGLE and other libraries from non-default locations 
 # this assumes that the library path contains all these libraries (or are set through LD_LIBRARY_PATH)
 if [ -n "$BEAST_EXTRA_LIBS" ]; then 
