FDS Build: Remove static links for linux

From upstream https://github.com/firemodels/fds/commit/6ec0fbf23faaf
diff --git a/Build/makefile b/Build/makefile
index 66b2067255a..17760a5787c 100644
--- a/Build/makefile
+++ b/Build/makefile
@@ -183,9 +183,9 @@ impi_intel_win_64_db : setup_win $(objwin_mpi)
 # Intel Compiler for Linux
 impi_intel_linux_64 : FFLAGS = -m64 -O2 -ipo -g -traceback -no-wrap-margin $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_linux_64 : LFLAGSMKL = $(LFLAGSMKL_INTEL)
-impi_intel_linux_64 : LFLAGS = -static-intel
+impi_intel_linux_64 : LFLAGS =
 impi_intel_linux_64 : FCOMPL = mpiifort
-impi_intel_linux_64 : FOPENMPFLAGS = -qopenmp -qopenmp-link static -liomp5
+impi_intel_linux_64 : FOPENMPFLAGS = -qopenmp
 impi_intel_linux_64 : obj = fds_impi_intel_linux_64
 impi_intel_linux_64 : setup $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) $(LFLAGS) $(FOPENMPFLAGS) -o $(obj) $(obj_mpi) $(LFLAGSMKL)
