fix hardcoded version to ensure 'fds' reports correct version
author: Kenneth Hoste (HPC-UGent)
--- fds-FDS6.6.0/Source/main.f90.orig	2018-03-09 14:42:46.913683000 +0100
+++ fds-FDS6.6.0/Source/main.f90	2018-03-09 14:42:52.631185105 +0100
@@ -100,7 +100,7 @@
 
 ! Assign a compilation date (All Nodes)
 
-WRITE(VERSION_STRING,'(A)') 'FDS 6.5.3'
+WRITE(VERSION_STRING,'(A)') 'FDS 6.6.0'
 
 CALL GET_INFO (REVISION,REVISION_DATE,COMPILE_DATE)
 
