FDS Build: Remove static links for linux

From upstream https://github.com/firemodels/fds/commit/6ec0fbf23faaf
diff -ru fds-FDS6.7.6.orig/Build/makefile fds-FDS6.7.6/Build/makefile
--- fds-FDS6.7.6.orig/Build/makefile	2021-09-30 18:55:52.209902338 +0000
+++ fds-FDS6.7.6/Build/makefile	2021-09-30 18:56:47.670254768 +0000
@@ -238,8 +238,8 @@
 # OSX
 mpi_intel_osx_64 : FFLAGS = -m64 -O2 -ipo -stand:f18 -no-wrap-margin $(GITINFO) $(OPENMPI_COMPINFO) $(FFLAGSMKL_CUSTOM)
 mpi_intel_osx_64 : LFLAGSMKL = $(LFLAGSMKL_CUSTOM)
-mpi_intel_osx_64 : LFLAGS = -static-intel
-mpi_intel_osx_64 : FOPENMPFLAGS = -qopenmp -qopenmp-link static
+mpi_intel_osx_64 : LFLAGS = 
+mpi_intel_osx_64 : FOPENMPFLAGS = -qopenmp 
 mpi_intel_osx_64 : FCOMPL = mpifort
 mpi_intel_osx_64 : obj = fds_mpi_intel_osx_64
 mpi_intel_osx_64 : setup $(obj_mpi)
