--- lib/Xm/XmString.c
+++ lib/Xm/XmString.c
@@ -7707,7 +7707,7 @@
     case XmMULTIBYTE_TEXT:
       /* Non-NULL values (except _MOTIF_DEFAULT_LOCALE)
          are not accepted in Motif 2.0. */
-      if (tag != NULL && !(tag == _MOTIF_DEFAULT_LOCALE ||
+      if (tag != NULL && !(/* tag == _MOTIF_DEFAULT_LOCALE || */
 			   strcmp(tag, _MOTIF_DEFAULT_LOCALE) == 0))
       {
 	_XmProcessUnlock();
@@ -8317,7 +8317,7 @@
 	_XmProcessUnlock();
 	return NULL;
       }
-      if ((value != _MOTIF_DEFAULT_LOCALE) &&
+      if (/* (value != _MOTIF_DEFAULT_LOCALE) && */
 	  (strcmp((char*) value, _MOTIF_DEFAULT_LOCALE) != 0)) {
 	_XmProcessUnlock();
 	return NULL;
