--- demos/programs/ButtonBox/bboxdemo.c
+++ demos/programs/ButtonBox/bboxdemo.c
@@ -668,4 +668,5 @@
     XtRealizeWidget(top);
 
     XtAppMainLoop(app);
+    return 0;
 }
--- demos/programs/ColorSel/bxutil-c.c
+++ demos/programs/ColorSel/bxutil-c.c
@@ -3471,6 +3471,7 @@
     default:
         return -1;
     }
+    abort();
 }
 
 
@@ -3494,6 +3495,7 @@
     default:
         return -1;
     }
+    abort();
 }
 
 /*
--- demos/programs/FontSel/fontsel.c
+++ demos/programs/FontSel/fontsel.c
@@ -68,6 +68,7 @@
  * Returns:       nothing
  *
  */
+void
 InitializeDemoForm(Widget form)
 {
     Widget w;
@@ -119,5 +120,6 @@
     XtRealizeWidget(top);
 
     XtAppMainLoop(app);
+    return 0;
 }
 
--- demos/programs/MultiList/bxutil-c.c
+++ demos/programs/MultiList/bxutil-c.c
@@ -3469,6 +3469,7 @@
         }
         return (c);
     }
+    abort();
 }
 
 
@@ -3490,6 +3491,7 @@
             mdata->InsideString = !mdata->InsideString;
         return (ungetc(c, mdata->stream.file));
     }
+    abort();
 }
 
 /*
--- demos/programs/Tree/bxutil-c.c
+++ demos/programs/Tree/bxutil-c.c
@@ -3471,6 +3471,7 @@
     default:
     	return -1;
     }
+    abort();
 }
 
 
@@ -3494,6 +3495,7 @@
     default:
 	return -1;
     }
+    abort();
 }
 
 /*
--- tools/wml/wmlresolve.c
+++ tools/wml/wmlresolve.c
@@ -69,6 +69,7 @@
 
 #if defined(__STDC__)
 #include <stdlib.h>
+#include <string.h>
 #endif
 #include <stdio.h>
 
