--- third_party/gpus/cuda_configure.bzl	2021-11-22 16:09:32.000000000 -0600
+++ third_party/gpus/cuda_configure.bzl	2021-11-22 16:28:00.000000000 -0600
@@ -1173,10 +1173,11 @@
         out_dir = "cuda/bin",
     ))
 
-    if [int(x) for x in cuda_config.cudnn_version.split(".")] < [8, 0]:
+    if int(cuda_config.cudnn_version.split(".")[0]) < 8:
       cudnn_headers = ["cudnn.h"]
     else:
-      cudnn_headers = ["cudnn_adv_infer.h",
+      cudnn_headers = ["cudnn_backend.h",
+                       "cudnn_adv_infer.h",
                        "cudnn_adv_train.h",
                        "cudnn_cnn_infer.h",
                        "cudnn_cnn_train.h",
