--- tensorflow/tools/pip_package/setup.py.orig	2025-04-10 15:08:15.144018000 +0200
+++ tensorflow/tools/pip_package/setup.py	2025-04-10 15:11:38.059538000 +0200
@@ -82,8 +82,7 @@
     'gast >=0.2.1,!=0.5.0,!=0.5.1,!=0.5.2',
     'google_pasta >= 0.1.1',
     'h5py >= 3.10.0',
-    'libclang >= 13.0.0',
-    'ml_dtypes >= 0.3.1, < 0.5.0',
+    'ml_dtypes >= 0.3.1',
     # TODO(b/304751256): Adjust the numpy pin to a single version, when ready
     'numpy >= 1.23.5, < 2.0.0 ; python_version <= "3.11"',
     'numpy >= 1.26.0, < 2.0.0 ; python_version >= "3.12"',
@@ -99,8 +98,6 @@
     'termcolor >= 1.1.0',
     'typing_extensions >= 3.6.6',
     'wrapt >= 1.11.0',
-    # TODO(b/305196096): Remove the <3.12 condition once the pkg is updated
-    'tensorflow-io-gcs-filesystem >= 0.23.1 ; python_version < "3.12"',
     # grpcio does not build correctly on big-endian machines due to lack of
     # BoringSSL support.
     # See https://github.com/tensorflow/tensorflow/issues/17882.
