diff --git a/third_party/systemlibs/pybind11.BUILD b/third_party/systemlibs/pybind11.BUILD
index 79a483d7b5d..463dd1a8ec7 100644
--- a/third_party/systemlibs/pybind11.BUILD
+++ b/third_party/systemlibs/pybind11.BUILD
@@ -6,3 +6,9 @@ cc_library(
         "@org_tensorflow//third_party/python_runtime:headers",
     ],
 )
+
+# Needed by pybind11_bazel.
+config_setting(
+    name = "osx",
+    constraint_values = ["@platforms//os:osx"],
+)
