# Required on CentOS6
# https://github.com/tensorflow/tensorflow/issues/15129
--- tensorflow-1.13.1/tensorflow/tensorflow.bzl.orig	2019-03-01 10:57:59.056530606 +0100
+++ tensorflow-1.13.1/tensorflow/tensorflow.bzl	2019-03-01 11:03:28.555868154 +0100
@@ -377,7 +377,7 @@
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ["-lrt"],
         framework_so = tf_binary_additional_srcs(),
         kernels = [],
         **kwargs):
@@ -413,7 +413,7 @@
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ["-lrt"],
         copts = tf_copts(),
         kernels = [],
         **kwargs):
@@ -470,7 +470,7 @@
     tf_cc_binary(
         name = tool,
         copts = tf_copts(),
-        linkopts = if_not_windows(["-lm", "-Wl,-ldl"]),
+        linkopts = if_not_windows(["-lm", "-Wl,-ldl", "-lrt"]),
         linkstatic = 1,  # Faster to link this one-time-use binary dynamically
         deps = [op_gen] + deps,
     )
@@ -638,7 +638,7 @@
         hidden_file = None,
         generated_target_name = None,
         op_whitelist = [],
-        cc_linkopts = [],
+        cc_linkopts = ["-lrt"],
         api_def_srcs = []):
     if (hidden or hidden_file) and op_whitelist:
         fail("Cannot pass specify both hidden and op_whitelist.")
@@ -650,7 +650,7 @@
     tf_cc_binary(
         name = tool_name,
         copts = tf_copts(),
-        linkopts = if_not_windows(["-lm", "-Wl,-ldl"]) + cc_linkopts,
+        linkopts = if_not_windows(["-lm", "-Wl,-ldl", "-lrt"]) + cc_linkopts,
         linkstatic = 1,  # Faster to link this one-time-use binary dynamically
         visibility = [clean_dep("//tensorflow:internal")],
         deps = ([
@@ -924,7 +924,7 @@
         tags = [],
         size = "medium",
         args = None,
-        linkopts = [],
+        linkopts = ["-lrt"],
         kernels = [],
         nocopts = None):
     for src in srcs:
