From ef1f781ae856fbdc7fc42da0284d61e7e5ed7713 Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Thu, 8 Apr 2021 01:22:21 -0500
Subject: [PATCH] Fix cub BUILD file for local_config_cuda reference

---
 third_party/cub.BUILD | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/cub.BUILD b/third_party/cub.BUILD
index 581d8dd9421ac..c446f0517d3da 100644
--- a/third_party/cub.BUILD
+++ b/third_party/cub.BUILD
@@ -11,5 +11,5 @@ exports_files(["LICENSE.TXT"])
 cc_library(
     name = "cub",
     hdrs = glob(["cub/**"]),
-    deps = ["@local_cuda//:cuda_headers"],
+    deps = ["@local_config_cuda//cuda:cuda_headers"],
 )
