diff -ru tensorflow-1.5.0.orig/tensorflow/tools/pip_package/setup.py tensorflow-1.5.0/tensorflow/tools/pip_package/setup.py
--- tensorflow-1.5.0.orig/tensorflow/tools/pip_package/setup.py	2018-01-25 23:22:10.000000000 +0100
+++ tensorflow-1.5.0/tensorflow/tools/pip_package/setup.py	2018-01-31 20:15:47.257969549 +0100
@@ -46,6 +46,11 @@
   sys.argv.remove('--project_name')
   sys.argv.pop(project_name_idx)
 
+# enum34 is only required for Python 3.4 or older
+# see https://bitbucket.org/stoneleaf/enum34/issues/19
+if not (sys.version_info.major >= 3 and sys.version_info.minor >= 4):
+    REQUIRED_PACKAGES.append('enum34 >= 1.1.6')
+
 # python3 requires wheel 0.26
 if sys.version_info.major == 3:
   REQUIRED_PACKAGES.append('wheel >= 0.26')
