# Required on CentOS6
# https://github.com/tensorflow/tensorflow/issues/15129
--- tensorflow/tensorflow.bzl.old	2018-03-28 13:36:45.640203970 +0200
+++ tensorflow/tensorflow.bzl	2018-03-28 13:37:17.867282309 +0200
@@ -283,7 +283,7 @@
     name,
     srcs=[],
     deps=[],
-    linkopts=[],
+    linkopts=["-lrt"],
     framework_so=tf_binary_additional_srcs(),
     **kwargs):
   native.cc_binary(
@@ -1292,7 +1292,7 @@
 )
 
 def tf_extension_linkopts():
-  return []  # No extension link opts
+  return ["-lrt"]  # No extension link opts
 
 def tf_extension_copts():
   return []  # No extension c opts
