# Required on CentOS6
# https://github.com/tensorflow/tensorflow/issues/15129
--- tensorflow-1.12.0/tensorflow/tensorflow.bzl.orig	2018-11-13 20:27:28.959012202 +0100
+++ tensorflow-1.12.0/tensorflow/tensorflow.bzl	2018-11-13 20:32:39.045225504 +0100
@@ -367,7 +367,7 @@
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ["-lrt"],
         framework_so = tf_binary_additional_srcs(),
         kernels = [],
         **kwargs):
@@ -403,7 +403,7 @@
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ["-lrt"],
         copts = tf_copts(),
         kernels = [],
         **kwargs):
