diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index fc5fd364c47..17ba9dc3323 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -88,7 +88,6 @@ REQUIRED_PACKAGES = [
     'gast >=0.2.1,!=0.5.0,!=0.5.1,!=0.5.2',
     'google_pasta >= 0.1.1',
     'h5py >= 2.9.0',
-    'libclang >= 13.0.0',
     'ml_dtypes ~= 0.3.1',
     'numpy >= 1.23.5, < 2.0.0',
     'opt_einsum >= 2.3.2',
