Fix assetion string in testIsAbstract of Bijector test
author: Alex Domingo (Vrije Universiteit Brussel)
--- a/tensorflow/python/kernel_tests/distributions/bijector_test.py	2022-02-05 00:42:51.288983000 +0100
+++ b/tensorflow/python/kernel_tests/distributions/bijector_test.py	2022-02-05 00:43:02.729512435 +0100
@@ -38,7 +38,7 @@
   def testIsAbstract(self):
     with self.assertRaisesRegex(TypeError,
                                 ("Can't instantiate abstract class Bijector "
-                                 "with abstract methods __init__")):
+                                 "with abstract method __init__")):
       bijector.Bijector()  # pylint: disable=abstract-class-instantiated
 
   def testDefaults(self):
