diff --git a/third_party/systemlibs/jsoncpp.BUILD b/third_party/systemlibs/jsoncpp.BUILD
index 7d54f9289b..ba921bdfb0 100644
--- a/third_party/systemlibs/jsoncpp.BUILD
+++ b/third_party/systemlibs/jsoncpp.BUILD
@@ -32,8 +32,8 @@ genrule(
 
 cc_library(
     name = "jsoncpp",
-    hdrs = HEADERS,
-    includes = ["."],
+    #hdrs = HEADERS,
+    #includes = ["."],
     linkopts = ["-ljsoncpp"],
     visibility = ["//visibility:public"],
 )
