Add a missing target for the absl-py SYSTEMLIB
From https://github.com/tensorflow/tensorflow/pull/60636

Author: Alexander Grund (TU Dresden)

diff --git a/third_party/systemlibs/absl_py.absl.flags.BUILD b/third_party/systemlibs/absl_py.absl.flags.BUILD
index 4049989ae2fda..d92f4949df1a5 100644
--- a/third_party/systemlibs/absl_py.absl.flags.BUILD
+++ b/third_party/systemlibs/absl_py.absl.flags.BUILD
@@ -5,3 +5,7 @@ package(default_visibility = ["//visibility:public"])
 py_library(
     name = "flags",
 )
+
+py_library(
+    name = "argparse_flags",
+)
