--- CONFIGURE/IFORT-AMD64-MPI.orig	2017-07-07 17:50:55.740058851 +0000
+++ CONFIGURE/IFORT-AMD64-MPI	2017-07-07 21:57:07.062811373 +0000
@@ -12,20 +12,19 @@
 #INFO# NOTE: CPMD cannot be compiled with the GNU Fortran compiler.
 #INFO#
      IRAT=2
-     CFLAGS='-O2 -Wall -m64'
-     CPP='/lib/cpp -P -C -traditional'
-     CPPFLAGS='-D__Linux -D__PGI -DFFT_DEFAULT -DPOINTER8 -DLINUX_IFC \
-	-DPARALLEL -DMYRINET'
-     FFLAGS='-pc64  -tpp6 -O2 -unroll'
-     LFLAGS=' -L. -latlas_x86_64'
+     CFLAGS='-O2 -Wall '
+     CPP='fpp -P -C'
+     CPPFLAGS='-D__Linux -DFFT_FFTW3 -DPOINTER8 -DMALLOC8 -DPARALLEL -DMYRINET -DLINUX_IFC -DHAS_BF_STREAM_IO'
+     FFLAGS='-O2 -unroll -I$(MKLROOT)/include/fftw -axAVX'
+     LFLAGS='-mkl=sequential -L$(MKLROOT)/include/fftw -axAVX'
      FFLAGS_GROMOS='-Dgood_luck $(FFLAGS)'
      if [ $debug ]; then
        FC='mpif77 -c -g'
        CC='mpicc -g -Wall -m64'
        LD='mpif77 -g'
      else
-       FC='mpif77 -c '
-       CC='mpicc'
-       LD='mpif77 -i-static '
+       FC='mpifort -c '
+       CC='icc'
+       LD='mpifort '
      fi
  
