Use wxWidgets version 3.0.3 instead of 3.0.2 (version 3.0.2 reports as 3.0.1)
author: Sam Moors, Vrije Universiteit Brussel (VUB)
diff -ur cistem-1.0.0-beta.orig/configure cistem-1.0.0-beta/configure
--- cistem-1.0.0-beta.orig/configure	2017-12-03 02:10:52.000000000 +0100
+++ cistem-1.0.0-beta/configure	2018-05-24 13:41:28.749291371 +0200
@@ -15184,7 +15184,7 @@
 
 # Verify minimus requires
 vers=`echo $wxversion | $AWK 'BEGIN { FS = "."; } { printf "% d", ($1 * 1000 + $2) * 1000 + $3;}'`
-if test -n "$vers" && test "$vers" -eq 3000002; then
+if test -n "$vers" && test "$vers" -eq 3000003; then
         WX_CPPFLAGS="`$WXCONFIG --cppflags`"
         WX_CXXFLAGS="`$WXCONFIG --cxxflags | sed -e 's/-fno-exceptions//'`"
         WX_LIBS="`$WXCONFIG --libs richtext,std,aui`"
