diff -ur a/lib/Makefile b/lib/Makefile
--- a/lib/Makefile	2023-03-16 01:47:22.000000000 +0000
+++ b/lib/Makefile	2023-10-19 18:21:44.974789869 +0000
@@ -4,7 +4,7 @@
 FPIC = -fPIC -shared
 HTSDIR = ./htslib/
 LIB_OBJECT = nextcorrect.so ovlseq.so ctg_cns.so ckit.so
-CTG_CNS_CFLAGS = -D_SVID_SOURCE -DLGS_CORRECT -I $(HTSDIR) -lm -lpthread -lz
+CTG_CNS_CFLAGS = -D_SVID_SOURCE -DLGS_CORRECT -I $(HTSDIR) -lcurl -lcrypto -lm -lpthread -lz $(HTSLIB_CFLAGS) -ldeflate
 
 all: $(LIB_OBJECT)
 
diff -ur a/util/Makefile b/util/Makefile
--- a/util/Makefile	2023-03-16 01:47:22.000000000 +0000
+++ b/util/Makefile	2023-10-19 18:21:44.974789869 +0000
@@ -27,7 +27,7 @@
 	$(CC) $(CFLAGS) -std=c99 $(INCLUDES) -o $@ $^ -lz
 
 bam_sort: bam_sort.c ../lib/bsort.c $(HTSDIR)/libhts.a
-		$(CC) $(CFLAGS) -o $@ $^ -I $(HTSDIR) $(INCLUDES) -lpthread -lm -lz
+		$(CC) $(CFLAGS) -o $@ $^ -I $(HTSDIR) $(INCLUDES) -lpthread -lm -lz -lcrypto -ldeflate -lcurl
 
 # align: realign.c ../lib/ovl.c ../lib/bseq.c ../lib/align.c
 # 	$(CC) $(CFLAGS) -Wno-unused-function -o $@ $^ -lz
