diff -ru cupy-12.3.0/install/cupy_builder/_compiler.py cupy-12.3.0_use_nvflags/install/cupy_builder/_compiler.py
--- cupy-12.3.0/install/cupy_builder/_compiler.py	2023-12-06 08:29:04.000000000 +0100
+++ cupy-12.3.0_use_nvflags/install/cupy_builder/_compiler.py	2023-12-11 17:24:26.148232507 +0100
@@ -13,7 +13,7 @@
 
 from cupy_builder._context import Context
 import cupy_builder.install_build as build
-
+_nvflags=os.getenv("NVFLAGS")
 
 def _nvcc_gencode_options(cuda_version: int) -> List[str]:
     """Returns NVCC GPU code generation options."""
@@ -222,7 +222,7 @@
             postargs += [f'-t{num_threads}']
         else:
             postargs += ['--std=c++11']
-        postargs += ['-Xcompiler=-fno-gnu-unique']
+        postargs += ['-Xcompiler=-fno-gnu-unique']; postargs += ['-Xcompiler=-Wno-error=deprecated-declarations']
         print('NVCC options:', postargs)
         self.spawn(compiler_so + base_opts + cc_args + [src, '-o', obj] +
                    postargs)
Only in cupy-12.3.0_use_nvflags/install/cupy_builder: ._compiler.py.swp
