diff -Nru spark-2.2.0.orig/build/mvn spark-2.2.0/build/mvn
--- spark-2.2.0.orig/build/mvn	2017-06-30 22:56:29.000000000 +0000
+++ spark-2.2.0/build/mvn	2017-07-18 16:57:29.300838631 +0000
@@ -131,27 +131,16 @@
 fi
 
 # Install the proper version of Scala, Zinc and Maven for the build
-install_zinc
 install_scala
 install_mvn
 
 # Reset the current working directory
 cd "${_CALLING_DIR}"
 
-# Now that zinc is ensured to be installed, check its status and, if its
-# not running or just installed, start it
-if [ -n "${ZINC_INSTALL_FLAG}" -o -z "`"${ZINC_BIN}" -status -port ${ZINC_PORT}`" ]; then
-  export ZINC_OPTS=${ZINC_OPTS:-"$_COMPILE_JVM_OPTS"}
-  "${ZINC_BIN}" -shutdown -port ${ZINC_PORT}
-  "${ZINC_BIN}" -start -port ${ZINC_PORT} \
-    -scala-compiler "${SCALA_COMPILER}" \
-    -scala-library "${SCALA_LIBRARY}" &>/dev/null
-fi
-
 # Set any `mvn` options if not already present
 export MAVEN_OPTS=${MAVEN_OPTS:-"$_COMPILE_JVM_OPTS"}
 
 echo "Using \`mvn\` from path: $MVN_BIN" 1>&2
 
 # Last, call the `mvn` command as usual
-${MVN_BIN} -DzincPort=${ZINC_PORT} "$@"
+${MVN_BIN} "$@"
diff -Nru spark-2.2.0.orig/conf/spark-env.sh spark-2.2.0/conf/spark-env.sh
--- spark-2.2.0.orig/conf/spark-env.sh	1970-01-01 00:00:00.000000000 +0000
+++ spark-2.2.0/conf/spark-env.sh	2017-07-18 16:59:26.335726333 +0000
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+export SPARK_LOG_DIR="$HOME/.spark/2.2.0/log"
+export SPARK_WORKER_DIR="$SCRATCH/spark/2.2.0/$(hostname)"
+if [[ -z "$SPARK_CONF_DIR" ]]; then
+    export SPARK_CONF_DIR="$HOME/.spark/2.2.0/conf"
+    if [[ -f $SPARK_CONF_DIR/spark-env.sh ]]; then
+        . $SPARK_CONF_DIR/spark-env.sh
+    fi
+fi
+
+# MLlib uses netlib-java which is linked on MKL
+export MKL_NUM_THREADS=1
diff -Nru spark-2.2.0.orig/dev/make-distribution.sh spark-2.2.0/dev/make-distribution.sh
--- spark-2.2.0.orig/dev/make-distribution.sh	2017-06-30 22:56:30.000000000 +0000
+++ spark-2.2.0/dev/make-distribution.sh	2017-07-18 17:00:12.887289417 +0000
@@ -237,9 +237,8 @@
 fi
 
 # Copy other things
-mkdir "$DISTDIR/conf"
-cp "$SPARK_HOME"/conf/*.template "$DISTDIR/conf"
 cp "$SPARK_HOME/README.md" "$DISTDIR"
+cp -r "$SPARK_HOME"/conf "$DISTDIR"/conf
 cp -r "$SPARK_HOME/bin" "$DISTDIR"
 cp -r "$SPARK_HOME/python" "$DISTDIR"
 
diff -Nru spark-2.2.0.orig/pom.xml spark-2.2.0/pom.xml
--- spark-2.2.0.orig/pom.xml	2017-06-30 22:56:30.000000000 +0000
+++ spark-2.2.0/pom.xml	2017-07-18 16:56:06.699623673 +0000
@@ -1950,8 +1950,6 @@
           </executions>
           <configuration>
             <scalaVersion>${scala.version}</scalaVersion>
-            <recompileMode>incremental</recompileMode>
-            <useZincServer>true</useZincServer>
             <args>
               <arg>-unchecked</arg>
               <arg>-deprecation</arg>
