diff -Nru spark-2.3.0.orig/conf/spark-env.sh spark-2.3.0/conf/spark-env.sh
--- spark-2.3.0.orig/conf/spark-env.sh	1970-01-01 00:00:00.000000000 +0000
+++ spark-2.3.0/conf/spark-env.sh	2018-03-12 14:30:25.201036857 +0000
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+export SPARK_LOG_DIR="$HOME/.spark/2.3.0/log"
+export SPARK_WORKER_DIR="$SCRATCH/spark/2.3.0/$(hostname)"
+if [[ -z "$SPARK_CONF_DIR" ]]; then
+    export SPARK_CONF_DIR="$HOME/.spark/2.3.0/conf"
+    if [[ -f $SPARK_CONF_DIR/spark-env.sh ]]; then
+        . $SPARK_CONF_DIR/spark-env.sh
+    fi
+fi
+
+# MLlib uses netlib-java which is linked on MKL
+export MKL_NUM_THREADS=1
diff -Nru spark-2.3.0.orig/dev/make-distribution.sh spark-2.3.0/dev/make-distribution.sh
--- spark-2.3.0.orig/dev/make-distribution.sh	2018-02-22 19:16:38.000000000 +0000
+++ spark-2.3.0/dev/make-distribution.sh	2018-03-12 14:33:08.013451391 +0000
@@ -243,10 +243,9 @@
 fi
 
 # Copy other things
-mkdir "$DISTDIR/conf"
-cp "$SPARK_HOME"/conf/*.template "$DISTDIR/conf"
 cp "$SPARK_HOME/README.md" "$DISTDIR"
 cp -r "$SPARK_HOME/bin" "$DISTDIR"
+cp -r "$SPARK_HOME"/conf "$DISTDIR"/conf
 cp -r "$SPARK_HOME/python" "$DISTDIR"
 
 # Remove the python distribution from dist/ if we built it
