diff -ru ChronQC-1.0.4.original/setup.py ChronQC-1.0.4.patched/setup.py
--- ChronQC-1.0.4.original/setup.py	2017-12-21 06:53:55.000000000 +0100
+++ ChronQC-1.0.4.patched/setup.py	2018-04-05 16:16:00.000000000 +0200
@@ -16,7 +16,7 @@
     history = history_file.read().replace('.. :changelog:', '')
 
 requirements = [
-     'numpy', 'pandas>=0.20.2', 'matplotlib', 'bottle'
+     'numpy', 'pandas>=0.20.2', 'matplotlib', 'bottle==0.12.13'
 ]
 
 test_requirements = [
