commit 36d4c60c4aa2dec96d1623f658a951a8134c2895
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Mon Jul 13 14:12:10 2020 +0200

    Fix mkl include path to match EasyBuild
    
    Åke Sandgren, 20170517

diff --git a/AmberTools/src/configure2 b/AmberTools/src/configure2
index 2d1473e..40280d5 100755
--- a/AmberTools/src/configure2
+++ b/AmberTools/src/configure2
@@ -3042,7 +3042,7 @@ if [ "$compiler" = "intel" ]; then
         if [ -n "$MKL_HOME" ]; then
             echo "     MKL_HOME set to" "$MKL_HOME"
             echo `mkdir -p $amberprefix/include`
-            echo `cp $MKL_HOME/include/fftw/fftw3.f $amberprefix/include`
+            echo `cp $MKL_HOME/mkl/include/fftw/fftw3.f $amberprefix/include`
         fi
         if [ "$intelmpi" = "yes" ]; then
             pmemd_fpp_flags="$pmemd_fppflags -DFFTW_FFT -DMKL_FFTW_FFT "
