--- src/pmemd/src/xray/src/xray_non_bulk_impl_cpu.F90.orig	2025-06-01 15:25:13.577818000 +0000
+++ src/pmemd/src/xray/src/xray_non_bulk_impl_cpu.F90	2025-06-01 22:16:14.229057000 +0000
@@ -78,7 +78,8 @@
     real(real_kind), intent(in) :: frac(:, :)
     ! locals
     integer :: ihkl
-    
+    integer :: i
+
     call check_precondition(associated(hkl))
     call check_precondition(associated(mSS4))
     call check_precondition(allocated(atomic_scatter_factor))
@@ -127,4 +128,4 @@
   end subroutine calc_f_non_bulk
 
 
-end module xray_non_bulk_impl_cpu_module
\ No newline at end of file
+end module xray_non_bulk_impl_cpu_module
