--- AmberTools/src/amberlite/Makefile.orig	2021-03-25 20:37:31.481650438 +0000
+++ AmberTools/src/amberlite/Makefile	2021-03-25 20:38:32.099977174 +0000
@@ -16,13 +16,13 @@
 	cd ../nab && $(MAKE) $@
 
 $(BINDIR)/ffgbsa$(SFX): ffgbsa.nab
-	$(BINDIR)/nab$(SFX) -o $(BINDIR)/ffgbsa$(SFX) ffgbsa.nab
+	$(BINDIR)/nab$(SFX) -o $(BINDIR)/ffgbsa$(SFX) ffgbsa.nab $(CUSTOMBUILDFLAGS)
 
 $(BINDIR)/minab$(SFX): minab.nab
-	$(BINDIR)/nab$(SFX) -o $(BINDIR)/minab$(SFX) minab.nab
+	$(BINDIR)/nab$(SFX) -o $(BINDIR)/minab$(SFX) minab.nab $(CUSTOMBUILDFLAGS)
 
 $(BINDIR)/mdnab$(SFX): mdnab.nab
-	$(BINDIR)/nab$(SFX) -o $(BINDIR)/mdnab$(SFX) mdnab.nab
+	$(BINDIR)/nab$(SFX) -o $(BINDIR)/mdnab$(SFX) mdnab.nab $(CUSTOMBUILDFLAGS)
 
 clean:
 	/bin/rm -f *.c
--- AmberTools/src/etc/Makefile.orig	2021-03-25 20:39:56.310431073 +0000
+++ AmberTools/src/etc/Makefile	2021-03-25 20:40:30.773616769 +0000
@@ -74,7 +74,7 @@
 
 molsurf$(SFX):  molsurf.nab
 	@echo "[ETC]  NAB $@"
-	$(VB)$(BINDIR)/nab$(SFX) -o molsurf$(SFX) molsurf.nab
+	$(VB)$(BINDIR)/nab$(SFX) -o molsurf$(SFX) molsurf.nab $(CUSTOMBUILDFLAGS)
 
 resp$(SFX):	lapack.o resp.o
 	@echo "[ETC]  FC $@"
@@ -83,7 +83,7 @@
 
 tinker_to_amber$(SFX): tinker_to_amber.o cspline.o
 	@echo "[ETC]  FC $@"
-	$(VB)$(FC) -o tinker_to_amber$(SFX) tinker_to_amber.o cspline.o
+	$(VB)$(FC) -o tinker_to_amber$(SFX) tinker_to_amber.o cspline.o $(CUSTOMBUILDFLAGS)
 
 gwh$(SFX):		gwh.o syslib
 	@echo "[ETC]  FC $@"
--- AmberTools/src/mm_pbsa/Makefile.orig	2021-03-25 20:41:34.516960224 +0000
+++ AmberTools/src/mm_pbsa/Makefile	2021-03-25 20:41:54.792069469 +0000
@@ -21,7 +21,7 @@
 
 #Note dependency on nab from AMBERTools here.
 $(BINDIR)/mm_pbsa_nabnmode$(SFX): mm_pbsa_nabnmode.nab
-	$(BINDIR)/nab$(SFX) $(NABFLAGS) -o $(BINDIR)/mm_pbsa_nabnmode$(SFX) mm_pbsa_nabnmode.nab
+	$(BINDIR)/nab$(SFX) $(NABFLAGS) -o $(BINDIR)/mm_pbsa_nabnmode$(SFX) mm_pbsa_nabnmode.nab $(CUSTOMBUILDFLAGS)
 
 ../lib/amopen.o: ../lib/amopen.F
 	cd ../lib; $(MAKE) amopen.o
--- AmberTools/src/nss/Makefile.orig	2021-03-25 20:44:00.152744926 +0000
+++ AmberTools/src/nss/Makefile	2021-03-25 20:44:58.120057261 +0000
@@ -54,28 +54,28 @@
 	-ranlib $(LIBDIR)/libnab.a
 
 matextract$(SFX): matextract.o
-	$(NAB) -o matextract$(SFX) matextract.o
+	$(NAB) -o matextract$(SFX) matextract.o $(CUSTOMBUILDFLAGS)
 
 matgen$(SFX):	matgen.o
-	$(NAB) -o matgen$(SFX) matgen.o
+	$(NAB) -o matgen$(SFX) matgen.o $(CUSTOMBUILDFLAGS)
 
 matmerge$(SFX):	matmerge.o
-	$(NAB) -o matmerge$(SFX) matmerge.o
+	$(NAB) -o matmerge$(SFX) matmerge.o $(CUSTOMBUILDFLAGS)
 
 matmul$(SFX):	matmul.o
-	$(NAB) -o matmul$(SFX) matmul.o
+	$(NAB) -o matmul$(SFX) matmul.o $(CUSTOMBUILDFLAGS)
 
 transform$(SFX):	transform.o
-	$(NAB) -o transform$(SFX) transform.o
+	$(NAB) -o transform$(SFX) transform.o $(CUSTOMBUILDFLAGS)
 
 tss_init$(SFX):	tss_init.o
-	$(NAB) -o tss_init$(SFX) tss_init.o
+	$(NAB) -o tss_init$(SFX) tss_init.o $(CUSTOMBUILDFLAGS)
 
 tss_main$(SFX):	tss_main.o
-	$(NAB) -o tss_main$(SFX) tss_main.o
+	$(NAB) -o tss_main$(SFX) tss_main.o $(CUSTOMBUILDFLAGS)
 
 tss_next$(SFX):	tss_next.o
-	$(NAB) -o tss_next$(SFX) tss_next.o
+	$(NAB) -o tss_next$(SFX) tss_next.o $(CUSTOMBUILDFLAGS)
 
 clean:
 	/bin/rm -f	\
--- AmberTools/src/rism/Makefile.orig	2021-03-24 00:25:02.218180253 +0000
+++ AmberTools/src/rism/Makefile	2021-03-25 20:47:42.345942125 +0000
@@ -116,12 +116,12 @@
 rism3d.snglpnt$(SFX): $(SPOBJ)
 	@echo "[RISM]  CC $@"
 	$(VB)$(CC) $(CFLAGS) $^ -L$(LIBDIR) $(FLIBS) $(LM) \
-       $(LDFLAGS) -o $(BINDIR)/rism3d.snglpnt$(SFX)
+       $(LDFLAGS) -o $(BINDIR)/rism3d.snglpnt$(SFX) $(CUSTOMBUILDFLAGS)
 
 rism3d.snglpnt.MPI$(SFX): $(SPOBJ)
 	@echo "[RISM]  CC $@"
 	$(VB)$(CC) $(CFLAGS) $^ -L$(LIBDIR) $(FLIBS) $(LM) \
-         $(LDFLAGS) -o $(BINDIR)/rism3d.snglpnt.MPI$(SFX)
+         $(LDFLAGS) -o $(BINDIR)/rism3d.snglpnt.MPI$(SFX) $(CUSTOMBUILDFLAGS)
 # --------------------------------------------------------------------------
 
 yes: install
--- AmberTools/src/mmpbsa_py/Makefile.orig	2021-03-29 07:44:38.954485335 +0000
+++ AmberTools/src/mmpbsa_py/Makefile	2021-03-29 07:46:38.750051341 +0000
@@ -9,10 +9,10 @@
 	$(PYTHON) setup.py install -f $(PYTHON_INSTALL) --install-scripts=$(BINDIR)
 
 $(BINDIR)/mmpbsa_py_nabnmode$(SFX): mmpbsa_entropy.nab
-	$(BINDIR)/nab$(SFX) -o $(BINDIR)/mmpbsa_py_nabnmode$(SFX) mmpbsa_entropy.nab
+	$(BINDIR)/nab$(SFX) -o $(BINDIR)/mmpbsa_py_nabnmode$(SFX) mmpbsa_entropy.nab $(CUSTOMBUILDFLAGS)
 
 $(BINDIR)/mmpbsa_py_energy$(SFX): mmpbsa_energy.nab
-	$(BINDIR)/nab$(SFX) -o $(BINDIR)/mmpbsa_py_energy$(SFX) mmpbsa_energy.nab
+	$(BINDIR)/nab$(SFX) -o $(BINDIR)/mmpbsa_py_energy$(SFX) mmpbsa_energy.nab $(CUSTOMBUILDFLAGS)
 
 serial: install
 
