Use EasyBuild Boost

Åke Sandgren, 20181030
diff -ru amber18.orig/AmberTools/src/packmol_memgen/packmol_memgen/lib/memembed/Makefile amber18/AmberTools/src/packmol_memgen/packmol_memgen/lib/memembed/Makefile
--- amber18.orig/AmberTools/src/packmol_memgen/packmol_memgen/lib/memembed/Makefile	2018-03-26 04:57:12.000000000 +0200
+++ amber18/AmberTools/src/packmol_memgen/packmol_memgen/lib/memembed/Makefile	2018-10-30 19:50:18.368208184 +0100
@@ -1,8 +1,8 @@
 CPP = g++
 CFLAGS = -Wall -Wextra -Werror -O3 -fPIC -std=c++11
-INC = -Isrc -I../boost/1.66/
+INC = -Isrc
 DIR = ${CURDIR}
-LIBS = -Wl,-rpath=$(DIR)/../boost/1.66/stage/lib/ -L$(DIR)/../boost/1.66/stage/lib/ -lboost_thread -lboost_system
+LIBS = -lboost_thread -lboost_system -lpthread
 
 all: memembed
 
