--- AmberTools/src/sander/Makefile.orig	2021-03-24 00:34:28.886473380 +0000
+++ AmberTools/src/sander/Makefile	2021-03-25 19:27:26.129841011 +0000
@@ -349,7 +349,7 @@
 	      -lFpbsa ../lib/nxtsec.o $(EMILLIB) \
 	      $(SEBOMDLIB) $(FBLIBS) \
 	      ../lib/sys.a $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LOCALFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -364,7 +364,7 @@
 	      -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	      $(SEBOMDLIB) $(FBLIBS) $(XRAY_OBJS) \
 	      ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -379,7 +379,7 @@
 		$(LSCIVROBJ) -L$(LIBDIR) -lsqm -lFpbsa \
 	        $(SEBOMDLIB) $(FBLIBS) $(XRAY_OBJS) \
 		../lib/nxtsec.o $(EMILLIB) ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	    	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	    	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(PUPILLIBS) $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -393,7 +393,7 @@
 		$(XRAY_OBJS) -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	        $(SEBOMDLIB) $(FBLIBS) \
 		../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
