diff -ru gofbc-2020.1.403.114.orig/cmake/CudaConfig.cmake gofbc-2020.1.403.114/cmake/CudaConfig.cmake
--- gofbc-2020.1.403.114.orig/cmake/CudaConfig.cmake	2021-11-15 17:56:40.371201270 +0000
+++ gofbc-2020.1.403.114/cmake/CudaConfig.cmake	2021-11-15 18:12:02.910511227 +0000
@@ -21,8 +21,11 @@
 
 		#Note at present we do not include SM3.5 or SM3.7 since they sometimes show performance
 		#regressions over just using SM3.0.
+		set(SM70FLAGS -gencode arch=compute_70,code=sm_70)
+		set(SM75FLAGS -gencode arch=compute_75,code=sm_75)
         #SM8.0 = A100
         set(SM80FLAGS -gencode arch=compute_80,code=sm_80)
+        set(SM86FLAGS -gencode arch=compute_80,code=sm_80)
         #SM7.5 = RTX20xx, RTX Titan, T4 and Quadro RTX
 		set(SM75FLAGS -gencode arch=compute_60,code=sm_75)        
 		#SM7.0 = V100 and Volta Geforce / GTX Ampere?
@@ -66,12 +69,14 @@
 			list(APPEND CUDA_NVCC_FLAGS ${SM30FLAGS} ${SM50FLAGS} ${SM52FLAGS} ${SM53FLAGS} ${SM60FLAGS} ${SM61FLAGS} ${SM70FLAGS} ${SM75FLAGS} -Wno-deprecated-gpu-targets -Wno-deprecated-declarations)	
 		elseif(${CUDA_VERSION} VERSION_GREATER_EQUAL 11.0)
                         # Implement the standard compilation rather than a warp-synchronous one, which is deprecated as of CUDA 11
-		        set(SM70FLAGS -gencode arch=compute_70,code=sm_70)
-			set(SM75FLAGS -gencode arch=compute_75,code=sm_75)
             message(STATUS "Configuring for SM3.5, SM5.0, SM5.2, SM5.3, SM6.0, SM6.1, SM7.0, SM7.5 and SM8.0")
             list(APPEND CUDA_NVCC_FLAGS ${SM35FLAGS} ${SM50FLAGS} ${SM52FLAGS} ${SM53FLAGS} ${SM60FLAGS} ${SM61FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS} -Wno-deprecated-gpu-targets -Wno-deprecated-declarations)                                      
+		elseif(${CUDA_VERSION} VERSION_GREATER_EQUAL 11.4)
+                        # Implement the standard compilation rather than a warp-synchronous one, which is deprecated as of CUDA 11
+			message(STATUS "Configuring for SM3.5, SM5.0, SM5.2, SM5.3, SM6.0, SM6.1, SM7.0, SM7.5, SM8.0 SM8.6")
+			list(APPEND CUDA_NVCC_FLAGS ${SM35FLAGS} ${SM50FLAGS} ${SM52FLAGS} ${SM53FLAGS} ${SM60FLAGS} ${SM61FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS} ${SM86FLAGS} -Wno-deprecated-gpu-targets -Wno-deprecated-declarations)                                      
 		else()
-			message(FATAL_ERROR "Error: Unsupported CUDA version. AMBER requires CUDA version >= 7.5 and <= 11.0.
+			message(FATAL_ERROR "Error: Unsupported CUDA version. AMBER requires CUDA version >= 7.5 and <= 11.4.
 				Please upgrade your CUDA installation or disable building with CUDA.")
 		endif()
 						
