diff -ru gofbc-2020.1.403.114.orig/cmake/CudaConfig.cmake gofbc-2020.1.403.114/cmake/CudaConfig.cmake
--- gofbc-2020.1.403.114.orig/cmake/CudaConfig.cmake	2021-11-15 14:53:59.632250115 +0000
+++ gofbc-2020.1.403.114/cmake/CudaConfig.cmake	2021-11-15 14:56:28.719240644 +0000
@@ -64,6 +64,9 @@
 		elseif((${CUDA_VERSION} VERSION_GREATER_EQUAL 10.0) AND (${CUDA_VERSION} VERSION_LESS 11.0))
 			message(STATUS "Configuring for SM3.0, SM3.5, SM3.7, SM5.0, SM5.2, SM5.3, SM6.0, SM6.1, SM7.0 and SM7.5")
 			list(APPEND CUDA_NVCC_FLAGS ${SM30FLAGS} ${SM50FLAGS} ${SM52FLAGS} ${SM53FLAGS} ${SM60FLAGS} ${SM61FLAGS} ${SM70FLAGS} ${SM75FLAGS} -Wno-deprecated-gpu-targets -Wno-deprecated-declarations)	
+		elseif((${CUDA_VERSION} VERSION_GREATER_EQUAL 11.1) AND (${CUDA_VERSION} VERSION_LESS_EQUAL 11.4))
+			message(STATUS "Configuring for SM3.5, SM5.0, SM6.0, SM7.0, SM7.5, SM8.0 and SM8.6")
+			list(APPEND CUDA_NVCC_FLAGS ${SM35FLAGS} ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS} ${SM86FLAGS} -Wno-deprecated-gpu-targets -Wno-deprecated-declarations)
 		elseif(${CUDA_VERSION} VERSION_EQUAL 11.0)
                         # Implement the standard compilation rather than a warp-synchronous one, which is deprecated as of CUDA 11
 		        set(SM70FLAGS -gencode arch=compute_70,code=sm_70)
