Must not override EasyBuilds PYTHONPATH

Åke Sandgren, 20181030
diff -ru amber18.orig/AmberTools/src/configure2 amber18/AmberTools/src/configure2
--- amber18.orig/AmberTools/src/configure2	2018-10-30 10:51:32.247644248 +0100
+++ amber18/AmberTools/src/configure2	2018-10-30 19:12:22.810002328 +0100
@@ -3968,7 +3968,7 @@
 PYTHON=$python
 PYTHON_INSTALL=$python_install_string
 SKIP_PYTHON=$skippython
-PYTHONPATH=\$(AMBER_PREFIX)/lib/python$python_ver/site-packages
+#PYTHONPATH=\$(AMBER_PREFIX)/lib/python$python_ver/site-packages
 
 PYSANDER=$pysander
 PYTRAJ=$pytraj
