Fix link of external FFTW3 for sander

Åke Sandgren, 20170517
diff -ru amber18.orig/AmberTools/src/sander/Makefile amber18/AmberTools/src/sander/Makefile
--- amber18.orig/AmberTools/src/sander/Makefile	2018-03-12 18:11:30.000000000 +0100
+++ amber18/AmberTools/src/sander/Makefile	2018-10-30 09:59:36.821771677 +0100
@@ -329,7 +329,7 @@
 	      -lFpbsa ../lib/nxtsec.o $(EMILLIB) \
 	      $(SEBOMDLIB) \
 	      ../lib/sys.a $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -343,7 +343,7 @@
 	      -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	      $(SEBOMDLIB) $(XRAY_OBJS) \
 	      ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -357,7 +357,7 @@
 		$(LSCIVROBJ) -L$(LIBDIR) -lsqm -lFpbsa \
 	        $(SEBOMDLIB) $(XRAY_OBJS) \
 		../lib/nxtsec.o $(EMILLIB) ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	    	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	    	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(PUPILLIBS) $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -370,7 +370,7 @@
 		$(XRAY_OBJS) -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	        $(SEBOMDLIB) \
 		../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -382,7 +382,7 @@
               $(PARTPIMDOBJ) $(LSCIVROBJ) $(XRAY_OBJS) \
               -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) $(SEBOMDLIB) \
               ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-              $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+              $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
               $(LDFLAGS) $(AMBERLDFLAGS) $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -397,7 +397,7 @@
               -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	        $(SEBOMDLIB) $(XRAY_OBJS) \
               ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
               $(LDFLAGS) $(AMBERLDFLAGS) $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
