commit 72191821fbd59207045c67ea914f1e90f80c1aef
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Mon Jul 13 14:25:20 2020 +0200

    Fix link of external FFTW3 for sander
    
    Åke Sandgren, 20170517

diff --git a/AmberTools/src/sander/Makefile b/AmberTools/src/sander/Makefile
index 36f5881..3d28bd0 100644
--- a/AmberTools/src/sander/Makefile
+++ b/AmberTools/src/sander/Makefile
@@ -343,7 +343,7 @@ $(BINDIR)/sander$(SFX): configured_serial libsqm $(MMOBJ) $(QMOBJ) \
 	      -lFpbsa ../lib/nxtsec.o $(EMILLIB) \
 	      $(SEBOMDLIB) $(FBLIBS) \
 	      ../lib/sys.a $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -358,7 +358,7 @@ $(BINDIR)/sander.MPI$(SFX): configured_parallel libsqm $(MMOBJ) $(QMOBJ) \
 	      -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	      $(SEBOMDLIB) $(FBLIBS) $(XRAY_OBJS) \
 	      ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -373,7 +373,7 @@ $(BINDIR)/sander.PUPIL$(SFX): configured_serial libsqm $(PUPILOBJ) $(QMOBJ) \
 		$(LSCIVROBJ) -L$(LIBDIR) -lsqm -lFpbsa \
 	        $(SEBOMDLIB) $(FBLIBS) $(XRAY_OBJS) \
 		../lib/nxtsec.o $(EMILLIB) ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	    	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	    	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(PUPILLIBS) $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -387,7 +387,7 @@ $(BINDIR)/sander.LES$(SFX): configured_serial libsqm $(LESOBJ) $(PARTPIMDOBJ) \
 		$(XRAY_OBJS) -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	        $(SEBOMDLIB) $(FBLIBS) \
 		../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -400,7 +400,7 @@ $(BINDIR)/sander.LES.MPI$(SFX): configured_parallel libsqm $(LESOBJ) $(EVBPIMD)
               $(PARTPIMDOBJ) $(LSCIVROBJ) $(XRAY_OBJS) \
               -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) $(SEBOMDLIB) $(FBLIBS) \
               ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-              $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+              $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
               $(LDFLAGS) $(AMBERLDFLAGS) $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -416,7 +416,7 @@ $(BINDIR)/sander.APBS$(SFX): configured_serial libsqm $(APBSOBJ) $(QMOBJ) \
               -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	        $(SEBOMDLIB) $(FBLIBS) $(XRAY_OBJS) \
               ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
               $(LDFLAGS) $(AMBERLDFLAGS) $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
