--- AmberTools/src/configure2.orig	2021-03-25 20:53:16.719743770 +0000
+++ AmberTools/src/configure2	2021-03-25 23:08:25.142653893 +0000
@@ -1453,6 +1453,10 @@
             exit 1
          fi
     fi
+    # -ipo (multi-file Interprocedural Optimizations optimizations) causes issues with
+    #  netCDF+HDF5 linking, just do single-file IPO for the moment. ÅS
+    pmemd_coptflags=`echo $pmemd_coptflags | sed -e 's/ipo/ip/g'`
+    pmemd_foptflags=`echo $pmemd_foptflags | sed -e 's/ipo/ip/g'`
 
     #CUDA Specifics
     if [ "$cuda" = 'yes' ]; then
