Fix incorrect intel mpi compiler version detection.

Åke Sandgren, 20170517
diff -ru amber16.orig/AmberTools/src/configure2 amber16/AmberTools/src/configure2
--- amber16.orig/AmberTools/src/configure2	2017-04-13 15:00:53.000000000 +0200
+++ amber16/AmberTools/src/configure2	2017-05-17 10:34:26.841606227 +0200
@@ -304,7 +304,8 @@
                   | sed -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
     else
       cc_version=`$cc $1 2>&1 | grep -E "$cc |[vV]ersion " \
-                  | sed -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
+                  | sed -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//' \
+		  | grep -v '^Intel(R)'`
     fi
     if [ -z "$cc_version" ] ; then
        echo "Error: $cc is not well formed or produces unusual version details!"
@@ -354,7 +355,8 @@
                   -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
     else
       fc_version=`$fc $1 2>&1 | grep -E "$fc |$cc |[vV]ersion " | sed -e "s@$fc @@" \
-                  -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
+                  -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//' \
+		  | grep -v '^Intel(R)'`
     fi
     if [ -z "$fc_version" ] ; then
        echo "Error: $fc is not well formed or produces unusual version details!"
