diff -ru 18.orig/AmberTools/src/configure2 18/AmberTools/src/configure2
--- 18.orig/AmberTools/src/configure2	2018-07-26 21:25:32.615935557 +0000
+++ 18/AmberTools/src/configure2	2018-07-26 21:28:26.616785452 +0000
@@ -563,6 +563,7 @@
 mpinab=''
 mpi='no'
 netcdf_dir=''
+netcdf_fort_dir=''
 netcdf_flag=''
 netcdfstatic='no'
 pmemd_gem='no'
@@ -652,6 +653,7 @@
         --skip-python)  skippython='yes' ;;
         --with-python)  shift; python="$1";;
         --with-netcdf)  shift; netcdf_dir="$1";;
+        --with-netcdf-fort)  shift; netcdf_fort_dir="$1";;
         --with-pnetcdf) shift; pnetcdf_dir="$1" ;;
         --python-install) shift; python_install="$1";;
         -netcdfstatic)  netcdfstatic='yes' ;;
@@ -925,7 +927,7 @@
 flibsf="-larpack -llapack -lblas"
 # only used when the user requests a static build or when a static build is
 # automatically set, eg, windows:
-staticflag='-static'
+staticflag='-static-intel'
 omp_flag=
 mpi_flag=
 fp_flags=
@@ -1493,7 +1495,7 @@
         nvcc="$nvcc -use_fast_math -O3 "
       fi
       if [ "$mpi" = 'yes' ]; then
-        mpi_inc=`(mpicc -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
+        mpi_inc=`(${CC} -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
         pmemd_cu_includes="$pmemd_cu_includes $mpi_inc"
         pmemd_cu_defines="$pmemd_cu_defines -DMPI -DMPICH_IGNORE_CXX_SEEK"
         pmemd_coptflags="$coptflags -DMPICH_IGNORE_CXX_SEEK"
@@ -1878,7 +1880,7 @@
       fi
 
       if [ "$mpi" = 'yes' ]; then
-          mpi_inc=`(mpicc -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
+          mpi_inc=`(${CC} -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
         pmemd_cu_includes="$pmemd_cu_includes $mpi_inc"
         pmemd_cu_defines="$pmemd_cu_defines -DMPI -DMPICH_IGNORE_CXX_SEEK"
         pmemd_coptflags="$pmemd_coptflags -DMPICH_IGNORE_CXX_SEEK"
@@ -2217,7 +2219,7 @@
         nvcc="$nvcc -use_fast_math -O3 "
       fi
       if [ "$mpi" = 'yes' ]; then
-        mpi_inc=`(mpicc -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
+        mpi_inc=`(${CC} -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
         pmemd_cu_includes="$pmemd_cu_includes $mpi_inc"
         pmemd_cu_defines="$pmemd_cu_defines -DMPI -DMPICH_IGNORE_CXX_SEEK"
         pmemd_coptflags="$coptflags -DMPICH_IGNORE_CXX_SEEK"
@@ -2354,7 +2356,7 @@
         nvcc="$nvcc -use_fast_math -O3 "
       fi
       if [ "$mpi" = 'yes' ]; then
-        mpi_inc=`(mpicc -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
+        mpi_inc=`(${CC} -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
         pmemd_cu_includes="$pmemd_cu_includes $mpi_inc"
         pmemd_cu_defines="$pmemd_cu_defines -DMPI -DMPICH_IGNORE_CXX_SEEK"
         pmemd_coptflags="$coptflags -DMPICH_IGNORE_CXX_SEEK"
@@ -2427,17 +2429,17 @@
     blas=skip
     flibs="-larpack "
     flibsf="-larpack "
-    mkll="$MKL_HOME/lib/32"
+    mkll="$MKL_HOME/mkl/lib/32"
     mkl_processor="32"
     mkl_procstring="ia32"
     mklinterfacelayer='libmkl_intel.a'
     if [ "$x86_64" = 'yes' ]; then
-        if [ -d "$MKL_HOME/lib/em64t" ]; then
-            mkll="$MKL_HOME/lib/em64t"
+        if [ -d "$MKL_HOME/mkl/lib/em64t" ]; then
+            mkll="$MKL_HOME/mkl/lib/em64t"
             mkl_processor="em64t"
             mkl_procstring="em64t"
         else
-            mkll="$MKL_HOME/lib/intel64"
+            mkll="$MKL_HOME/mkl/lib/intel64"
             mkl_processor="intel64"
             mkl_procstring="intel64"
         fi
@@ -2508,7 +2510,7 @@
 #  Support platforms without a C <complex.h> by building c9x-complex.
 #--------------------------------------------------------------------------
 
-if [ -r /usr/include/complex.h ]; then
+if [ -r $NIXUSER_PROFILE/include/complex.h ]; then
     c9xcomplex='skip'
 elif [ "$cygwin" = 'yes' -o "$windows" = 'yes' ]; then
     c9xcomplex='skip'
@@ -2523,7 +2525,7 @@
 #  Support platforms without <sys/dir.h> but with <dirent.h>
 #  usually a non-Linux Unix with a non-native compiler.
 #--------------------------------------------------------------------------
-if [ ! -r /usr/include/sys/dir.h  -a -r /usr/include/dirent.h ]; then
+if [ ! -r $NIXUSER_PROFILE/include/sys/dir.h  -a -r $NIXUSER_PROFILE/include/dirent.h ]; then
     if [ "$cygwin" != 'yes' ]; then
        cflags="$cflags -DSYSV"
        pmemd_coptflags="$pmemd_coptflags -DSYSV"
@@ -2863,11 +2865,16 @@
   else
     # A NetCDF directory was specified. Check that library exists and compiles
     printf "\tUsing external NetCDF in '$netcdf_dir'\n"
-     netcdfinc="-I"$netcdf_dir"/include"
+    # Support separate NetCDF-Fortran installation with --with-netcdf-fort
+    if [ ! -e "$netcdf_fort_dir" ]; then
+      netcdf_fort_dir="$netcdf_dir"
+    fi
+    printf "\tUsing external NetCDF-Fortran in '$netcdf_fort_dir'\n"
+    netcdfinc="-I"$netcdf_dir"/include -I"$netcdf_fort_dir"/include"
     if [ "${netcdf_dir}" != '/usr' -a "$netcdf_dir" != '/usr/' ]; then
         netcdf_flag="-L${netcdf_dir}/lib $netcdf_flag"
     fi
-    netcdf=$netcdf_dir"/include/netcdf.mod"
+    netcdf=$netcdf_fort_dir"/include/netcdf.mod"
     if [ "$netcdfstatic" = 'no' ] ; then
       if [ "${netcdf_dir}" != '/usr' -a "${netcdf_dir}" != '/usr/' ]; then
           netcdfflagc="-L${netcdf_dir}/lib -lnetcdf"
@@ -2883,7 +2891,7 @@
         echo "Error: '$netcdfflagc' not found."
         exit 1
       fi
-      netcdfflagf=$netcdf_dir"/lib/libnetcdff.a"
+      netcdfflagf=$netcdf_fort_dir"/lib/libnetcdff.a"
       if [ ! -e "$netcdfflagf" ]; then
         echo "Error: '$netcdfflagf' not found."
         exit 1
Only in 18/AmberTools/src: configure2.orig
