--- AmberTools/src/configure2.orig	2021-03-23 21:04:56.841065261 +0000
+++ AmberTools/src/configure2	2021-03-23 21:11:49.908665819 +0000
@@ -256,7 +256,8 @@
                   | sed -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
     else
       cc_version=`$cc $1 2>&1 | grep -E "$basecc |[vV]ersion " \
-                  | sed -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
+                  | sed -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//' \
+		  | grep -v '^Intel(R)'`
     fi
     if [ -z "$cc_version" ] ; then
        echo "Error: $cc is not well formed or produces unusual version details!"
@@ -307,7 +308,8 @@
                   -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
     else
       fc_version=`$fc $1 2>&1 | grep -E "$basefc |$basecc |[vV]ersion " | sed -e "s@$basefc @@" \
-                  -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//'`
+                  -e 's/Open64//' -e 's/^[a-zA-Z :]* //' -e 's/ .*//' \
+		  | grep -v '^Intel(R)'`
     fi
     if [ -z "$fc_version" ] ; then
       # DRR - Last ditch; compiler name may not be in version string so just
