diff -ru amber18.orig/Makefile amber18/Makefile
--- amber18.orig/Makefile	2018-03-26 04:46:35.000000000 +0200
+++ amber18/Makefile	2018-10-30 10:34:35.482737196 +0100
@@ -96,6 +96,10 @@
 	fi ;\
 	)
 
+test.cuda: is_amberhome_defined
+        # For backwards compatibility in the easyblock
+	-(cd test && $(MAKE) test.cuda.serial)
+
 test.cuda_serial: is_amberhome_defined
 	-(cd test && $(MAKE) test.cuda.serial)
 
