--- AmberTools/src/configure2.orig	2021-03-25 19:32:26.536427337 +0000
+++ AmberTools/src/configure2	2021-03-25 19:32:33.412465227 +0000
@@ -3577,7 +3577,7 @@
 if [ "$optimise" = 'no' ]  ; then CPPTRAJOPTS="$CPPTRAJOPTS -noopt" ; fi
 if [ -z "$zlib" ]          ; then CPPTRAJOPTS="$CPPTRAJOPTS -nozlib" ; fi
 if [ -z "$bzlib" ]         ; then CPPTRAJOPTS="$CPPTRAJOPTS -nobzlib" ; fi
-if [ "$static" = 'yes' ]   ; then CPPTRAJOPTS="$CPPTRAJOPTS -static" ; fi
+#if [ "$static" = 'yes' ]   ; then CPPTRAJOPTS="$CPPTRAJOPTS -static" ; fi
 if [ ! -z "$pnetcdf_dir" ] ; then CPPTRAJOPTS="$CPPTRAJOPTS --with-pnetcdf=$pnetcdf_dir" ; fi
 if [ "$has_fftw3" = 'yes' ]; then
   if [ "$mkl_type" != 'mkl' -o "$mkl_fftw" = 'no' ] ; then
