--- AmberTools/src/packmol_memgen/packmol_memgen/lib/memembed/Makefile.orig	2021-03-25 20:53:41.381876651 +0000
+++ AmberTools/src/packmol_memgen/packmol_memgen/lib/memembed/Makefile	2021-03-25 20:54:51.827256218 +0000
@@ -10,7 +10,7 @@
 LIBS = -lboost_thread -lboost_system -lpthread
 else
 INC = -Isrc -I$(INCDIR)
-LIBS = -L$(LIBDIR) -lboost_thread -lboost_system -lpthread
+LIBS = -lboost_thread -lboost_system -lpthread
 endif
 
 all: memembed
