--- AmberTools/src/configure2.orig	2021-03-25 20:51:07.632048230 +0000
+++ AmberTools/src/configure2	2021-03-25 20:52:32.764506934 +0000
@@ -4034,7 +4034,7 @@
 PYTHON=$python
 PYTHON_INSTALL=$python_install_string
 SKIP_PYTHON=$skippython
-PYTHONPATH=\$(AMBER_PREFIX)/lib/python$python_ver/site-packages
+#PYTHONPATH=\$(AMBER_PREFIX)/lib/python$python_ver/site-packages
 PYTHONLOG=$pythonlog
 
 PYSANDER=$pysander
