Fix link of external FFTW3 for sander

Åke Sandgren, 20170517
diff -ru amber16.orig/AmberTools/src/sander/Makefile amber16/AmberTools/src/sander/Makefile
--- amber16.orig/AmberTools/src/sander/Makefile	2017-04-13 15:00:54.000000000 +0200
+++ amber16/AmberTools/src/sander/Makefile	2017-05-17 11:04:29.961465834 +0200
@@ -317,7 +317,7 @@
 	      -lFpbsa ../lib/nxtsec.o $(EMILLIB) \
 	      $(SEBOMDLIB) \
 	      ../lib/sys.a $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -331,7 +331,7 @@
 	      -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	      $(SEBOMDLIB) $(XRAY_OBJS) \
 	      ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 	      $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -345,7 +345,7 @@
 		$(LSCIVROBJ) -L$(LIBDIR) -lsqm -lFpbsa \
 	        $(SEBOMDLIB) $(XRAY_OBJS) \
 		../lib/nxtsec.o $(EMILLIB) ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	    	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	    	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(PUPILLIBS) $(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 
@@ -358,7 +358,7 @@
 		$(XRAY_OBJS) -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	        $(SEBOMDLIB) \
 		../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      	$(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      	$(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
 		$(LDFLAGS) $(AMBERLDFLAGS)  $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -370,7 +370,7 @@
               $(PARTPIMDOBJ) $(LSCIVROBJ) $(XRAY_OBJS) \
               -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) $(SEBOMDLIB) \
               ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-              $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+              $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
               $(LDFLAGS) $(AMBERLDFLAGS) $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
@@ -385,7 +385,7 @@
               -L$(LIBDIR) -lsqm -lFpbsa $(EMILLIB) \
 	        $(SEBOMDLIB) $(XRAY_OBJS) \
               ../lib/nxtsec.o ../lib/sys.a $(NFE_OBJECTS) $(NETCDFLIBF) \
-	      $(FLIBS_RISMSANDER) $(FFTW3) $(FLIBSF) \
+	      $(FLIBS_RISMSANDER) $(FLIBS_FFTW3) $(FLIBSF) \
               $(LDFLAGS) $(AMBERLDFLAGS) $(LIOLIBS) $(PLUMED_LOAD)
 
 #---------------------------------------------------------------------------
