Fix incorrect OMP directive in RISM

Åke Sandgren, 20171010
diff -ru amber16.orig/AmberTools/src/rism/rism3d_c.F90 amber16/AmberTools/src/rism/rism3d_c.F90
--- amber16.orig/AmberTools/src/rism/rism3d_c.F90	2017-04-13 15:00:54.000000000 +0200
+++ amber16/AmberTools/src/rism/rism3d_c.F90	2017-10-10 15:39:22.259467974 +0200
@@ -4061,7 +4061,7 @@
     numElectronsAtGridCenter = totalSolventElectrons - electronRDFSum
 
 ! #pragma omp parallel for schedule(dynamic, 10) shared (dx, conc, result, stepx, stepy, stepz, center_grid)
-!$omp parallel do private(local_equal), shared(this, UNITS, electronMap, numSmearGridPoints, numElectronsAtGridCenter)
+!$omp parallel do shared(this, electronMap, numSmearGridPoints, numElectronsAtGridCenter)
     do igzCenter = 0, this%grid%globalDimsR(3) - 1
        rzCenter = igzCenter * this%grid%voxelVectorsR(3, :)
        do igyCenter = 0, this%grid%globalDimsR(2) - 1
