diff -ur unpatched/AmberTools/src/configure2 amber12/AmberTools/src/configure2
--- unpatched/AmberTools/src/configure2	2013-04-13 00:04:14.000000000 +1200
+++ amber12/AmberTools/src/configure2	2014-10-06 12:58:25.929458307 +1300
@@ -804,7 +804,7 @@
     else
       # following appears to work on Intel compilers version 9,10,11; the
       # "-L$fl" part is not needed for version 11, but doesn't seem to hurt
-      fl=`which ifort | sed -e 's@\(.*\)bin@\1lib@'`
+      fl=`which ifort | sed -e 's@\(.*\)bin@\1compiler/lib@'`
       flibs_arch="-L`dirname $fl` -lifport -lifcore"
 
       flibsf_arch=
@@ -817,7 +817,7 @@
       fc=ifort
       fflags=
       freeformat_flag='-FR'
-      staticflag='-static'
+      staticflag='-static-intel'
     
       # If -noopt has been requested, force lack of optimisation;
       # otherwise, use the default levels. Since cflags, cxxflags
@@ -870,7 +870,10 @@
       if [ "$icc_version" -ge 11 ] ; then
 #       cxxflags="-std=c++0x $cxxflags"
 #       ambercxxflags="-std=c++0x $ambercxxflags"
-        ldflags="-shared-intel "
+        if [ "${static}" = "no" ]
+        then
+          ldflags="-shared-intel "
+        fi
       fi
       
       #RISM and PBSA FFT solver require ISO_C_BINDING support
@@ -948,7 +951,7 @@
       fi
    
       if [ "$mpi" = 'yes' ]; then
-         ld=' mpif90 '
+         ld=' mpiifort '
       else
          ld=' ifort '
       fi
@@ -1029,7 +1032,7 @@
         fi
 
         if [ "$mpi" = 'yes' ]; then
-          mpi_inc=`(mpicc -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
+          mpi_inc=`(${CC} -show 2>&1) | awk 'BEGIN{i=0} {while (i < NF) {if ( substr($i, 1, 2) == "-I" ) {printf("%s ", $i);}; i++;}}'`
           pmemd_cu_includes="$pmemd_cu_includes $mpi_inc"
           pmemd_cu_defines="$pmemd_cu_defines -DMPI -DMPICH_IGNORE_CXX_SEEK"
           pmemd_coptflags="$pmemd_coptflags -DMPICH_IGNORE_CXX_SEEK"
@@ -1581,17 +1584,17 @@
     blas=skip
     flibs="-larpack "
     flibsf="-larpack "
-    mkll="$MKL_HOME/lib/32"
+    mkll="$MKL_HOME/mkl/lib/32"
     mkl_processor="32"
     mkl_procstring="ia32"
     mklinterfacelayer='libmkl_intel.a'
     if [ "$x86_64" = 'yes' ]; then
-        if [ -d "$MKL_HOME/lib/em64t" ]; then
-            mkll="$MKL_HOME/lib/em64t"
+        if [ -d "$MKL_HOME/mkl/lib/em64t" ]; then
+            mkll="$MKL_HOME/mkl/lib/em64t"
             mkl_processor="em64t"
             mkl_procstring="em64t"
         else
-            mkll="$MKL_HOME/lib/intel64"
+            mkll="$MKL_HOME/mkl/lib/intel64"
             mkl_processor="intel64"
             mkl_procstring="intel64"
         fi
@@ -1599,7 +1602,7 @@
         mklinterfacelayer='libmkl_intel_lp64.a'
     fi
     if [ "$processor" = "ia64" ]; then
-        mkll="$MKL_HOME/lib/64"
+        mkll="$MKL_HOME/mkl/lib/64"
         mkl_processor="64"
         mkl_procstring="ia64"
         mklinterfacelayer='libmkl_intel_lp64.a'
@@ -2354,12 +2357,11 @@
   if [ "$crayxt5" = 'yes' -o "$compiler" = 'cray' ]; then
       cc="cc"
       fc="ftn"
-      mpi_flag="-DMPI "
   else
-      cc="mpicc"
-      fc="mpif90"
-      mpi_flag="-DMPI "
+      cc=${CC}
+      fc=${F90}
   fi
+  mpi_flag="-DMPI "
 fi
 
 #------------------------------------------------------------------------------
diff -ur unpatched/AmberTools/src/cpptraj/configure amber12/AmberTools/src/cpptraj/configure
--- unpatched/AmberTools/src/cpptraj/configure	2013-04-13 00:04:14.000000000 +1200
+++ amber12/AmberTools/src/cpptraj/configure	2014-10-03 16:34:21.961264423 +1300
@@ -35,7 +35,7 @@
   echo "  EXPERIMENTAL OPTIONS:"
   echo "    -profile   : Use Gnu compiler profiling (>= V4.5)"
   echo "    -gprofile  : Use Gnu compiler GLIBC profiling (>= V4.5)"
-  echo "    -mpi       : Use mpicc/mpicxx to compile."
+  echo "    -mpi       : Use MPI compilers to compile."
   echo "    NOTE: -openmp and -mpi are mutually exclusive, as are -profile and -gprofile."
 #  echo "   --with-hdf5=<DIR>"
 }
@@ -560,9 +560,9 @@
   echo "Using MPI"
   DIRECTIVES="$DIRECTIVES -DMPI"
   if [[ $USECRAY -eq 0 ]] ; then
-    CC=mpicc
-    CXX=mpicxx
-    FC=mpif90
+    CC=${CC}
+    CXX=${CXX}
+    FC=${F90}
   fi
 fi
 
