Fix incorrect OMP directive in RISM

Åke Sandgren, 20171010
diff -ru amber18.orig/AmberTools/src/rism/rism3d_c.F90 amber18/AmberTools/src/rism/rism3d_c.F90
--- amber18.orig/AmberTools/src/rism/rism3d_c.F90	2017-12-22 15:01:28.000000000 +0100
+++ amber18/AmberTools/src/rism/rism3d_c.F90	2018-10-30 10:07:59.258364538 +0100
@@ -4097,7 +4097,7 @@
 
 #ifdef _OPENMP_
 ! #pragma omp parallel for schedule(dynamic, 10) shared (dx, conc, result, stepx, stepy, stepz, center_grid)
-!$omp parallel do private(local_equal), shared(this, UNITS, electronMap, numSmearGridPoints, numElectronsAtGridCenter)
+!$omp parallel do shared(this, electronMap, numSmearGridPoints, numElectronsAtGridCenter)
 #endif
     do igzCenter = 0, this%grid%globalDimsR(3) - 1
        rzCenter = igzCenter * this%grid%voxelVectorsR(3, :)
