--- AmberTools/src/configure2.orig	2021-03-25 19:47:47.314501312 +0000
+++ AmberTools/src/configure2	2021-03-25 20:50:18.970786038 +0000
@@ -916,38 +916,6 @@
 if [ "$noX11" = "true" ]; then
     make_xleap="skip_xleap"
     xhome=''
-else
-    if [ -d /usr/X11R6/lib ]; then
-        xhome='/usr/X11R6'
-    elif [ -d /usr/X11/lib ]; then    # location for MacOSX 10.11
-        xhome='/usr/X11'
-    elif [ -d /usr/lib/x86_64-linux-gnu ]; then
-        xhome='/usr'
-    elif [ -f /usr/lib/i386-linux-gnu/libX11.a ]; then
-        xhome='/usr'
-    elif [ -f /usr/lib/libX11.a -o -f /usr/lib/libX11.so \
-           -o -f /usr/lib/libX11.dll.a \
-           -o -f /usr/lib64/libX11.a -o -f /usr/lib64/libX11.so ]; then
-        xhome='/usr'
-    elif [ -f /opt/local/lib/libX11.a -o -f /opt/local/lib/libX11.dylib ]; then
-        xhome='/opt/local'
-    else
-        echo "Could not find the X11 libraries; you may need to edit config.h"
-        echo "   to set the XHOME and XLIBS variables."
-    fi
-
-    if [ "$xhome" != "/usr" ]; then
-        # Do not add -L/usr/lib to linker. This is always in the standard path
-        # and could cause issues trying to build MPI when /usr has an MPI
-        # installed that you *don't* want to use.
-        xlibs="-L$xhome/lib"
-        if [ "$x86_64" = 'yes' ]; then
-            xlibs="-L$xhome/lib64 $xlibs"
-        fi
-    fi
-    if [ -d /usr/lib/x86_64-linux-gnu ]; then
-        xlibs="-L/usr/lib/x86_64-linux-gnu $xlibs"
-    fi
 fi
 #--------------------------------------------------------------------------
 #  Check if the X11 library files for XLEaP are present:
