--- AmberTools/src/configure2.orig	2021-03-25 23:09:30.887996697 +0000
+++ AmberTools/src/configure2	2021-03-25 23:15:37.735910185 +0000
@@ -2461,7 +2461,7 @@
 #  Support platforms without <sys/dir.h> but with <dirent.h>
 #  usually a non-Linux Unix with a non-native compiler.
 #--------------------------------------------------------------------------
-if [ ! -r /usr/include/sys/dir.h  -a -r /usr/include/dirent.h ]; then
+if [ ! -r $EBROOTGENTOO/include/sys/dir.h  -a -r $EBROOTGENTOO/include/dirent.h ]; then
     if [ "$cygwin" != 'yes' ]; then
        cflags="$cflags -DSYSV"
        pmemd_coptflags="$pmemd_coptflags -DSYSV"
