--- AmberTools/src/configure2.orig	2021-03-24 00:30:29.875111516 +0000
+++ AmberTools/src/configure2	2021-03-24 00:33:27.237123883 +0000
@@ -4028,7 +4028,7 @@
 PMEMD_FOPTFLAGS=$pmemd_foptflags \$(AMBERBUILDFLAGS)
 PMEMD_CC=$cc
 PMEMD_COPTFLAGS=$pmemd_coptflags $mpi_flag \$(AMBERBUILDFLAGS)
-PMEMD_FLIBSF=$pmemd_flibsf $flibs_mkl $flibsf_arch $win_mpilibs $emillib
+PMEMD_FLIBSF=$pmemd_flibsf $flibs_mkl $flibsf_arch $win_mpilibs $emillib $flibs_fftw3
 PMEMD_LD=$ld \$(AMBERBUILDFLAGS)
 LDOUT=$ldout
 PMEMD_GNU_BUG303=$pmemd_gnu_bug303
