commit f4cd6171962deae6a269da970143902f68fab946
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Mon Jul 13 14:32:10 2020 +0200

    Fix missing openmp flag at link.
    
    Åke Sandgren, 20171013

diff --git a/AmberTools/src/ucpp-1.3/Makefile b/AmberTools/src/ucpp-1.3/Makefile
index 96a1b7b..9d737fd 100644
--- a/AmberTools/src/ucpp-1.3/Makefile
+++ b/AmberTools/src/ucpp-1.3/Makefile
@@ -44,7 +44,7 @@ clean:
 
 ucpp$(SFX): $(COBJ)
 	@echo "[UCPP]  CC $@"
-	$(VB)$(CC) $(LDFLAGS) -o ucpp$(SFX) $(COBJ) $(LIBS)
+	$(VB)$(CC) $(LDFLAGS) $(CFLAGS) -o ucpp$(SFX) $(COBJ) $(LIBS)
 
 assert.o: tune.h ucppi.h cpp.h nhash.h mem.h
 cpp.o: tune.h ucppi.h cpp.h nhash.h mem.h
