Fix hardcoding of complex.h, dir.h, dirent.h

Maxime Boissonneault, 20190121
diff -ru amber18.orig/AmberTools/src/configure2 amber18/AmberTools/src/configure2
--- amber18.orig/AmberTools/src/configure2	2018-10-30 09:35:16.787250576 +0100
+++ amber18/AmberTools/src/configure2	2018-04-16 01:39:34.000000000 +0200
@@ -2440,7 +2440,7 @@
 #  Support platforms without a C <complex.h> by building c9x-complex.
 #--------------------------------------------------------------------------

-if [ -r /usr/include/complex.h ]; then
+if [ -r $NIXUSER_PROFILE/include/complex.h ]; then
     c9xcomplex='skip'
 elif [ "$cygwin" = 'yes' -o "$windows" = 'yes' ]; then
     c9xcomplex='skip'
@@ -2455,7 +2455,7 @@
 #  Support platforms without <sys/dir.h> but with <dirent.h>
 #  usually a non-Linux Unix with a non-native compiler.
 #--------------------------------------------------------------------------
-if [ ! -r /usr/include/sys/dir.h  -a -r /usr/include/dirent.h ]; then
+if [ ! -r $NIXUSER_PROFILE/include/sys/dir.h  -a -r $NIXUSER_PROFILE/include/dirent.h ]; then
     if [ "$cygwin" != 'yes' ]; then
        cflags="$cflags -DSYSV"
        pmemd_coptflags="$pmemd_coptflags -DSYSV"
