Fix FFTW3 link flags for rism.

Åke Sandgren, 20180403
diff -ru amber16.orig/AmberTools/src/rism/Makefile amber16/AmberTools/src/rism/Makefile
--- amber16.orig/AmberTools/src/rism/Makefile	2018-04-03 21:52:52.811347758 +0200
+++ amber16/AmberTools/src/rism/Makefile	2018-04-03 21:51:47.999962945 +0200
@@ -92,7 +92,7 @@
 # ------ Single-Point information ------------------------------------------
 SPSRC=rism3d.snglpnt.nab
 rism3d.snglpnt$(SFX): $(SPSRC)
-	$(BINDIR)/nab $(SPSRC) -lfftw3 -o $(BINDIR)/rism3d.snglpnt$(SFX) $(CUSTOMBUILDFLAGS)
+	$(BINDIR)/nab $(SPSRC) $(FLIBS_FFTW3) -o $(BINDIR)/rism3d.snglpnt$(SFX) $(CUSTOMBUILDFLAGS)
 rism3d.snglpnt.MPI$(SFX): $(SPSRC)
 	$(BINDIR)/mpinab -v $(SPSRC) -o $(BINDIR)/rism3d.snglpnt.MPI$(SFX) $(CUSTOMBUILDFLAGS)
 # --------------------------------------------------------------------------
