Allow use of all CUDA 9 versions

Åke Sandgren, 20180727
diff -ru amber16.orig/AmberTools/src/configure2 amber16/AmberTools/src/configure2
--- amber16.orig/AmberTools/src/configure2	2018-07-27 10:27:34.745642223 +0200
+++ amber16/AmberTools/src/configure2	2018-07-27 10:30:05.442542975 +0200
@@ -1203,7 +1203,8 @@
     #SM2.0 = All GF variants = C2050, 2075, M2090, GTX480, GTX580 etc.
     sm20flags='-gencode arch=compute_20,code=sm_20'
     cudaversion=`$nvcc --version | grep 'release' | cut -d' ' -f5 | cut -d',' -f1`
-    if [ "$cudaversion" = "9.0" ]; then
+    cudamajversion=`echo $cudaversion | cut -d. -f1`
+    if [ $cudamajversion -ge 9 ]; then
       echo "CUDA Version $cudaversion detected"
       echo "Configuring for SM3.0, SM5.0, SM5.2, SM5.3, SM6.0, SM6.1 and SM7.0"
       nvccflags="$sm30flags $sm50flags $sm52flags $sm53flags $sm60flags $sm61flags $sm70flags -Wno-deprecated-declarations"
