Fix ModuleNotFoundError for simtk.openmm.
Author: Cintia Willemyns (Vrije Universiteit Brussel)
diff -Naru deepfold-c91701d1d92db341e85fba687644046d416b1f19.orig/deepfold/relax/amber_minimize.py deepfold-c91701d1d92db341e85fba687644046d416b1f19/deepfold/relax/amber_minimize.py
--- deepfold-c91701d1d92db341e85fba687644046d416b1f19.orig/deepfold/relax/amber_minimize.py	2024-03-22 16:58:35.884771000 +0100
+++ deepfold-c91701d1d92db341e85fba687644046d416b1f19/deepfold/relax/amber_minimize.py	2024-03-22 16:38:55.688749393 +0100
@@ -28,8 +28,8 @@
 import numpy as np
 from simtk import openmm
 from simtk import unit
-from simtk.openmm import app as openmm_app
-from simtk.openmm.app.internal.pdbstructure import PdbStructure
+from openmm import app as openmm_app
+from openmm.app.internal.pdbstructure import PdbStructure
 
 
 ENERGY = unit.kilocalories_per_mole
diff -Naru deepfold-c91701d1d92db341e85fba687644046d416b1f19.orig/deepfold/relax/cleanup.py deepfold-c91701d1d92db341e85fba687644046d416b1f19/deepfold/relax/cleanup.py
--- deepfold-c91701d1d92db341e85fba687644046d416b1f19.orig/deepfold/relax/cleanup.py	2024-03-22 16:58:35.884999000 +0100
+++ deepfold-c91701d1d92db341e85fba687644046d416b1f19/deepfold/relax/cleanup.py	2024-03-29 14:23:44.288779000 +0100
@@ -19,8 +19,8 @@
 import io
 
 import pdbfixer
-from simtk.openmm import app
-from simtk.openmm.app import element
+from openmm import app
+from openmm.app import element
 
 
 def fix_pdb(pdbfile, alterations_info):
diff -Naru deepfold-c91701d1d92db341e85fba687644046d416b1f19.orig/deepfold/relax/utils.py deepfold-c91701d1d92db341e85fba687644046d416b1f19/deepfold/relax/utils.py
--- deepfold-c91701d1d92db341e85fba687644046d416b1f19.orig/deepfold/relax/utils.py	2024-03-22 16:58:35.887221923 +0100
+++ deepfold-c91701d1d92db341e85fba687644046d416b1f19/deepfold/relax/utils.py	2024-03-22 17:00:41.078566002 +0100
@@ -17,8 +17,8 @@
 from deepfold.common import residue_constants
 from Bio import PDB
 import numpy as np
-from simtk.openmm import app as openmm_app
-from simtk.openmm.app.internal.pdbstructure import PdbStructure
+from openmm import app as openmm_app
+from openmm.app.internal.pdbstructure import PdbStructure
 
 
 def overwrite_pdb_coordinates(pdb_str: str, pos) -> str:
