# Add external linking to lcms2
# Author: micketeer@gmail.com
--- third_party/CMakeLists.txt.orig	2021-07-27 23:27:44.236251072 +0200
+++ third_party/CMakeLists.txt	2021-07-27 23:28:34.219135611 +0200
@@ -195,14 +195,14 @@
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/skcms/LICENSE"
                  ${PROJECT_BINARY_DIR}/LICENSE.skcms COPYONLY)
 endif ()
-if (JPEGXL_ENABLE_VIEWERS OR NOT JPEGXL_ENABLE_SKCMS)
-  if( NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/lcms/.git" )
-    message(SEND_ERROR "Please run git submodule update --init")
-  endif()
-  include(lcms2.cmake)
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/lcms/COPYING"
-                 ${PROJECT_BINARY_DIR}/LICENSE.lcms COPYONLY)
-endif()
+#if (JPEGXL_ENABLE_VIEWERS OR NOT JPEGXL_ENABLE_SKCMS)
+#  if( NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/lcms/.git" )
+#    message(SEND_ERROR "Please run git submodule update --init")
+#  endif()
+#  include(lcms2.cmake)
+#  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/lcms/COPYING"
+#                 ${PROJECT_BINARY_DIR}/LICENSE.lcms COPYONLY)
+#endif()
 
 # sjpeg
 if (JPEGXL_ENABLE_SJPEG)
--- lib/jxl.cmake.orig	2021-07-27 23:29:41.173980148 +0200
+++ lib/jxl.cmake	2021-07-30 13:36:44.745536958 +0200
@@ -352,12 +352,6 @@
     $<TARGET_PROPERTY:skcms,INCLUDE_DIRECTORIES>
   )
 else ()
-  target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
-  )
-  target_include_directories(jxl_dec-obj PRIVATE
-    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
-  )
 endif ()
