From ce7a3265019e4d66198c1581d9e8c859c34e8ef1 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Sat, 19 Oct 2019 09:32:41 +0200
Subject: [PATCH] configure: define HAVE_LIBBSD when libbsd was found

---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index e497325..d4d8aa4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,6 +53,9 @@ AC_CHECK_HEADERS([endian.h poll.h sys/poll.h])
 # Checks for library functions.
 AC_CHECK_FUNCS([poll readlink])
 AC_SEARCH_LIBS([strlcat], [bsd])
+if test "$ac_cv_search_strlcat" = "-lbsd"; then
+  AC_DEFINE(HAVE_LIBBSD,1,[Has libbsd])
+fi
 AC_CONFIG_LIBOBJ_DIR([src/util])
 AC_REPLACE_FUNCS([reallocarray realpath strlcat strlcpy])
 
-- 
GitLab

