diff -ru prepareGenFromBeagle4/makefile prepareGenFromBeagle4_correct/makefile
--- prepareGenFromBeagle4/makefile	2014-04-16 16:36:47.000000000 +0200
+++ prepareGenFromBeagle4_correct/makefile	2016-12-02 15:13:28.000000000 +0100
@@ -31,7 +31,7 @@
 #include path
 ISTDP= -Isrc
 IBAMP= -Ilib
-ICL3P= -I/users/delaneau/BOOST/include
+ICL3P= -I${BOOST_ROOT}include/
 
 #library path
 LSTDP= -Llib
@@ -41,7 +41,7 @@
 OBOST=
 
 #default
-all: dynamic
+all: cluster
 
 #dynamic release
 dynamic: CFLAG=$(COPTI) $(CWARN) $(CSHV1)
@@ -65,7 +65,8 @@
 cluster: CFLAG=$(COPTI) $(CWARN) $(CSHV1)
 cluster: LFLAG=$(LOPTI) $(LCL3S)
 cluster: IFLAG=$(ISTDP) $(ICL3P)
-cluster: OBOST=~/BOOST/lib/libboost_iostreams.a ~/BOOST/lib/libboost_program_options.a
+#cluster: OBOST=${BOOST_ROOT}/lib/libboost_iostreams.a ${BOOST_ROOT}/lib/libboost_program_options.a
+OBOST=-lz ${BOOST_ROOT}/lib/libboost_iostreams.a
 cluster: $(EFILE)
 
 $(EFILE): $(OFILE)
@@ -84,4 +85,4 @@
 	cp $(EFILE) ~/bin/.
 	
 install:
-	cp $(EFILE) /usr/local/bin/.
+	cp $(EFILE) ./tmp/
