diff -Nru rmats-turbo-4.1.1.orig/rMATS_C/Makefile rmats-turbo-4.1.1/rMATS_C/Makefile
--- rmats-turbo-4.1.1.orig/rMATS_C/Makefile	2021-04-30 14:25:18.316325000 +0200
+++ rmats-turbo-4.1.1/rMATS_C/Makefile	2021-04-30 14:25:43.474876000 +0200
@@ -10,7 +10,7 @@
 endif
 
 FFLAGS := -c
-LDFLAGS := -lm -lgfortran -lgsl -lgslcblas -lgomp -lblas -llapack
+LDFLAGS := -lm -lgfortran -lgsl -lgslcblas -lgomp -lopenblas -lscalapack
 CFLAGS := -Wall -O2 -msse2 -funroll-loops -fopenmp
 
 # Allow gsl flags to be set by environment variable
