--- Makefile.in.orig	2014-06-04 18:24:52.000000000 +0200
+++ Makefile.in	2015-09-19 23:55:31.924081000 +0200
@@ -17,7 +17,10 @@
 
 #install: pythonlib sara_install
 install: pythonlib
-	python ./setup.py install --prefix $(DESTDIR)/$(exec_prefix) ;
+	export PYTHONLIBDIR=`python -c "import distutils.sysconfig; print(distutils.sysconfig.get_python_lib(plat_specific=False, prefix=\" $(DESTDIR)/$(exec_prefix)\"))"`
+	mkdir -p ${PYTHONLIBDIR}
+	export PYTHONPATH=${PYTHONLIBDIR}:${PYTHONPATH}
+	easy_install --zip-ok --prefix $(DESTDIR)/$(exec_prefix);
 	install -D LICENSE.openpbs $(DESTDIR)/$(PREFIX)/$(DOCDIR)/LICENSE.openpbs
 	install -D COPYING $(DESTDIR)/$(PREFIX)/$(DOCDIR)/COPYING
 	install -D COPYING.LESSER $(DESTDIR)/$(PREFIX)/$(DOCDIR)/COPYING.LESSER
