diff -uN pbs_python-4.3.5.orig/setup.py.in pbs_python-4.3.5/setup.py.in
--- pbs_python-4.3.5.orig/setup.py.in   2013-12-05 19:49:17.601962399 +0100
+++ pbs_python-4.3.5/setup.py.in    2013-12-05 19:49:45.713469496 +0100
@@ -7,8 +7,10 @@

 import sys
 import os
+import glob

-from distutils.core import setup, Extension 
+from distutils.core import setup, Extension
+from distutils.version import LooseVersion

 # Strip off the -L it is also added by the python setup tools
 #
@@ -25,13 +27,6 @@
 ### Make symlinks to right torque version
 #
 VERSION = "@pbs_version@"
-tmp = VERSION.split('.')
-
-major_version = int(tmp[0])
-minor_version = int(tmp[1])
-#build_version = int(tmp[2])
-build_version = int(tmp[2].split('-')[0])
-

 os.chdir('src')

@@ -43,7 +38,23 @@
 except OSError:
     pass

-if major_version  >= 2 and minor_version >= 4 and build_version >= 7:
+if LooseVersion(VERSION) >= LooseVersion('4.2'):
+    inc=os.environ.get('PBS_PYTHON_INCLUDEDIR','/usr/include/torque')
+    os.environ['CC']=os.environ['CXX']
+    os.environ['CFLAGS']=' '.join(os.environ.get('CFLAGS','').split(' ')+['-I%s'%inc])
+
+    if not os.path.exists(os.path.join(inc,'log.h')):
+        print 'Failed to find log.h in inlcude dir %s. (Set incude dir via PBS_PYTHON_INCLUDEDIR variable)'%inc
+        sys.exit(2)
+
+    for fn in glob.glob('*.h'):
+        os.remove(fn)
+
+    os.symlink('pbs_wrap_4.2.c', 'pbs_wrap.c')
+    os.symlink('pbs_2.4.py', 'pbs.py')
+
+    TORQUE_VERSION='TORQUE_4_2'
+elif LooseVersion(VERSION) >= LooseVersion('2.4.7'):
     os.symlink('pbs_wrap_2.4.c', 'pbs_wrap.c')
     os.symlink('pbs_2.4.py', 'pbs.py')
     TORQUE_VERSION='TORQUE_2_4'

