diff -ur src/bin/detci/tpool.cc src/bin/detci/tpool.cc
--- src/bin/detci/tpool.cc	2013-07-01 23:11:05.000000000 +0200
+++ src/bin/detci/tpool.cc	2013-09-24 16:00:34.584896769 +0200
@@ -78,7 +78,7 @@
   tpool->queue_tail = NULL;
   tpool->queue_closed = 0;  
   tpool->shutdown = 0; 
-  tpool->threads_awake = 0;
+  tpool->threads_awake = num_worker_threads;
   if ((rtn = pthread_mutex_init(&(tpool->queue_lock), NULL)) != 0){
     str = "pthread_mutex_init ";
     str += strerror(rtn);
@@ -115,7 +115,6 @@
       str += boost::lexical_cast<std::string>( rtn) ;
       throw PsiException(str,__FILE__,__LINE__);
   }
-    tpool->threads_awake++;
   }
 
   *tpoolp = tpool;
@@ -263,10 +262,22 @@
 }
 void tpool_queue_open(tpool_t tpool)
 {
-  pthread_mutex_lock(&tpool->queue_lock);
+  int rtn;
+  std::string str;
+
+  if ((rtn = pthread_mutex_lock(&(tpool->queue_lock))) != 0){
+      str = "pthread_mutex_lock ";
+      str += static_cast<std::ostringstream*>( &(std::ostringstream() << rtn) )->str();
+      throw PsiException(str,__FILE__,__LINE__);
+  }
+
   tpool->queue_closed = 0;
-  tpool->threads_awake = 0;
-  pthread_mutex_unlock(&tpool->queue_lock);
+
+  if ((rtn = pthread_mutex_unlock(&(tpool->queue_lock))) != 0){
+      str = "pthread_mutex_unlock ";
+      str += static_cast<std::ostringstream*>( &(std::ostringstream() << rtn) )->str();
+      throw PsiException(str,__FILE__,__LINE__);
+  }
 }
   
 void tpool_queue_close(tpool_t tpool, int finish)
@@ -274,7 +285,12 @@
   std::string str;
   int rtn;
   
-  pthread_mutex_lock(&tpool->queue_lock);
+  if ((rtn = pthread_mutex_lock(&(tpool->queue_lock))) != 0){
+      str = "pthread_mutex_lock ";
+      str += static_cast<std::ostringstream*>( &(std::ostringstream() << rtn) )->str();
+      throw PsiException(str,__FILE__,__LINE__);
+  }
+
   tpool->queue_closed = 1;
   
   if (finish) {
@@ -288,7 +304,11 @@
     
     }
   
-  pthread_mutex_unlock(&tpool->queue_lock);
+  if ((rtn = pthread_mutex_unlock(&(tpool->queue_lock))) != 0){
+      str = "pthread_mutex_unlock ";
+      str += static_cast<std::ostringstream*>( &(std::ostringstream() << rtn) )->str();
+      throw PsiException(str,__FILE__,__LINE__);
+  }
   
 }
   
