--- psi4.0b4/src/lib/libparallel/mpi_wrapper.h.orig 2013-10-02 14:24:01.948223891 +0200
+++ psi4.0b4/src/lib/libparallel/mpi_wrapper.h  2013-10-02 14:31:02.282094030 +0200
@@ -1,5 +1,6 @@
 #if defined(HAVE_MPI)
 
+#include <cstring>
 #include <mpi.h>
 
 namespace psi {
--- psi4.0b4/src/lib/libmints/wavefunction.h.orig	2013-05-07 18:35:18.813690853 +0200
+++ psi4.0b4/src/lib/libmints/wavefunction.h	2013-05-07 18:35:43.113781216 +0200
@@ -6,7 +6,8 @@
 
 #include "typedefs.h"
 #include <libparallel/parallel.h>
+#include <liboptions/liboptions.h>
 
 #define MAX_IOFF 30000
 extern size_t ioff[MAX_IOFF];
