diff -ur psi4.0b5.orig/tests/Makefile.in psi4.0b5/tests/Makefile.in
--- psi4.0b5.orig/tests/Makefile.in	2013-07-01 23:11:05.000000000 +0200
+++ psi4.0b5/tests/Makefile.in	2013-09-11 16:40:22.672796277 +0200
@@ -33,7 +33,7 @@
 
 opt_subdirs = opt1 opt1-fd opt2 opt2-fd opt3 opt4 opt5 opt6 opt7 pywrap-opt-sowreap
 
-freq_subdirs = fd-gradient fd-freq-energy fd-freq-gradient dft-freq gibbs pywrap-freq-e-sowreap
+freq_subdirs = fd-gradient fd-freq-gradient dft-freq gibbs pywrap-freq-e-sowreap
 
 mp2_subdirs = mp2-1 omp2-1 omp2-2 omp2-3 omp2-4 omp2-5 omp3-1 omp3-2 omp3-3 omp3-4 omp3-5 ocepa1 ocepa2 ocepa3 omp2_5-1 omp2_5-2 omp2-grad1 omp2-grad2 omp3-grad1 omp3-grad2 omp2_5-grad1 omp2_5-grad2 ocepa-grad1 ocepa-grad2 mp2-grad1 mp2-grad2 mp3-grad1 mp3-grad2 mp2_5-grad1 mp2_5-grad2 cepa0-grad1 cepa0-grad2 ocepa-freq1
 
