--- iRODS/scripts/find_irods_home.sh.org	2015-05-05 13:00:25.677556552 +0300
+++ iRODS/scripts/find_irods_home.sh	2015-05-05 13:05:22.845957385 +0300
@@ -9,6 +9,6 @@
 if [ "$1" = "runinplace" ] ; then
     echo `pwd`
 else
-    CFG=`find . -name "irods.config"`
-    grep "IRODS_HOME =" $CFG | awk -F\' '{print $2}'
+#    No config as of https://github.com/irods/irods/commits/master/iRODS/scripts/find_irods_home.sh
+    echo $EBROOTIRODS/lib/
 fi
