--- packaging/build.sh.org	2015-05-04 16:52:11.177058901 +0300
+++ packaging/build.sh	2015-05-05 10:18:08.775566247 +0300
@@ -695,26 +695,8 @@
     fi
 fi
 
-# needed for boost, of all things...
-PYTHONDEV=`find /usr -name Python.h 2> /dev/null`
-if [[ "$PYTHONDEV" == "" ]] ; then
-    if [ "$DETECTEDOS" == "Ubuntu" -o "$DETECTEDOS" == "Debian" ] ; then
-        PREFLIGHT="$PREFLIGHT python-dev"
-    fi
-else
-    echo "Detected Python.h [$PYTHONDEV]"
-fi
-
 # needed for rpmbuild
 if [[ "$DETECTEDOS" == "RedHatCompatible" || "$DETECTEDOS" == "SuSE" ]] ; then
-    PYTHONDEV=`find /usr -name Python.h 2> /dev/null`
-    if [[ "$PYTHONDEV" == "" ]] ; then
-        if [ "$DETECTEDOS" == "RedHatCompatible" ] ; then
-            PREFLIGHT="$PREFLIGHT python-devel"
-        elif [ "$DETECTEDOS" == "SuSE" ] ; then
-            PREFLIGHT="$PREFLIGHT python-devel"
-        fi
-    fi
     RPMBUILD=`which rpmbuild`
     if [[ "$?" != "0" || `echo $RPMBUILD | awk '{print $1}'` == "no" ]] ; then
         if [ "$DETECTEDOS" == "RedHatCompatible" ] ; then
@@ -829,39 +811,7 @@
     echo "Detected libfuse library [$LIBFUSEDEV]"
 fi
 
-LIBCURLDEV=`find /usr -name curl.h 2> /dev/null`
-if [ "$LIBCURLDEV" == "" ] ; then
-    if [ "$DETECTEDOS" == "Ubuntu" -o "$DETECTEDOS" == "Debian" ] ; then
-        PREFLIGHT="$PREFLIGHT libcurl4-gnutls-dev"
-    elif [ "$DETECTEDOS" == "RedHatCompatible" ] ; then
-        PREFLIGHT="$PREFLIGHT curl-devel"
-    elif [ "$DETECTEDOS" == "SuSE" ] ; then
-        PREFLIGHT="$PREFLIGHT libcurl-devel"
-    elif [ "$DETECTEDOS" == "Solaris" ] ; then
-        PREFLIGHT="$PREFLIGHT curl_devel"
-    else
-        PREFLIGHTDOWNLOAD=$'\n'"$PREFLIGHTDOWNLOAD      :: download from: http://curl.haxx.se/download.html"
-    fi
-else
-    echo "Detected libcurl library [$LIBCURLDEV]"
-fi
-
-BZIP2DEV=`find /usr -name bzlib.h 2> /dev/null`
-if [ "$BZIP2DEV" == "" ] ; then
-    if [ "$DETECTEDOS" == "Ubuntu" -o "$DETECTEDOS" == "Debian" ] ; then
-        PREFLIGHT="$PREFLIGHT libbz2-dev"
-    elif [ "$DETECTEDOS" == "RedHatCompatible" ] ; then
-        PREFLIGHT="$PREFLIGHT bzip2-devel"
-    elif [ "$DETECTEDOS" == "SuSE" ] ; then
-        PREFLIGHT="$PREFLIGHT libbz2-devel"
-    elif [ "$DETECTEDOS" == "Solaris" ] ; then
-        PREFLIGHT="$PREFLIGHT libbz2_dev"
-    else
-        PREFLIGHTDOWNLOAD=$'\n'"$PREFLIGHTDOWNLOAD      :: download from: http://www.bzip.org/downloads.html"
-    fi
-else
-    echo "Detected bzip2 library [$BZIP2DEV]"
-fi
+## Removed python-devel, curl-devel and bzip2-devel system/OS dependency check - this is handled by Easybuild.
 
 ZLIBDEV=`find /usr/include -name zlib.h 2> /dev/null`
 if [ "$ZLIBDEV" == "" ] ; then
