--- Makefile.orig	2021-07-21 18:12:38.834139301 +0000
+++ Makefile	2021-07-21 18:12:56.899267573 +0000
@@ -458,9 +458,9 @@
 # Depending on PIO version, libraries may be libpio.a, or libpiof.a and libpioc.a
 # Keep open the possibility of shared libraries in future with, e.g., .so suffix
 #
-ifneq ($(wildcard $(PIO_LIB)/libpio\.*), )
-	LIBS += -lpio
-endif
+#ifneq ($(wildcard $(PIO_LIB)/libpio\.*), )
+#	LIBS += -lpio
+#endif
 ifneq ($(wildcard $(PIO_LIB)/libpiof\.*), )
 	LIBS += -lpiof
 endif
