Small patch to add a missing include statement
Author: J. Saßmannshausen (Imperial College London/UK)
diff --git a/wpebackend-fdo-1.16.1.orig/src/ws-egl.cpp b/wpebackend-fdo-1.16.1/src/ws-egl.cpp
index 76d6194..cc7a8f2 100644
--- a/wpebackend-fdo-1.16.1.orig/src/ws-egl.cpp
+++ b/wpebackend-fdo-1.16.1/src/ws-egl.cpp
@@ -33,6 +33,7 @@
 #include <vector>
 #include <sys/mman.h>
 #include <sys/stat.h>
+#include <unistd.h>
 
 #ifndef EGL_WL_bind_wayland_display
 #define EGL_WAYLAND_BUFFER_WL 0x31D5
