#Patch to set the correct netCDF Fortran library
--- configure.orig	2014-02-13 15:25:53.464250437 +0200
+++ configure	2014-02-13 15:26:24.909255244 +0200
@@ -1307,9 +1307,9 @@
 fi
 
 
-as_ac_File=`echo "ac_cv_file_${NC_LIB}/libnetcdf.a" | $as_tr_sh`
-echo "$as_me:$LINENO: checking for ${NC_LIB}/libnetcdf.a" >&5
-echo $ECHO_N "checking for ${NC_LIB}/libnetcdf.a... $ECHO_C" >&6
+as_ac_File=`echo "ac_cv_file_${NC_LIB}/libnetcdff.a" | $as_tr_sh`
+echo "$as_me:$LINENO: checking for ${NC_LIB}/libnetcdff.a" >&5
+echo $ECHO_N "checking for ${NC_LIB}/libnetcdff.a... $ECHO_C" >&6
 if eval "test \"\${$as_ac_File+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -1317,7 +1317,7 @@
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
-if test -r "${NC_LIB}/libnetcdf.a"; then
+if test -r "${NC_LIB}/libnetcdff.a"; then
   eval "$as_ac_File=yes"
 else
   eval "$as_ac_File=no"
