--- podman-4.5.0/pkg/specgen/generate/config_linux.go~	2023-04-14 13:28:20.000000000 +0000
+++ podman-4.5.0/pkg/specgen/generate/config_linux.go	2023-04-20 12:43:13.967663880 +0000
@@ -23,6 +23,7 @@
 	if isCDIDevice(devicePath) {
 		registry := cdi.GetRegistry(
 			cdi.WithAutoRefresh(false),
+			cdi.WithSpecDirs([]string{"/etc/cdi", "/var/run/cdi", "/var/tmp/cdi"}...),
 		)
 		if err := registry.Refresh(); err != nil {
 			logrus.Debugf("The following error was triggered when refreshing the CDI registry: %v", err)
