fix typo that breaks compilation with Intel compilers
author: Kenneth Hoste (HPC-UGent)
--- gapfiller-2.1.1/src/data_structures/Reads.cpp.orig	2017-04-06 13:30:10.978641815 +0200
+++ gapfiller-2.1.1/src/data_structures/Reads.cpp	2017-04-06 13:38:39.941387943 +0200
@@ -154,7 +154,7 @@
 			case 1: out.append("C"); break;
 			case 2: out.append("G"); break;
 			case 3: out.append("T"); break;
-			default: cout << "strange "<< cout << "\n";
+			default: cout << "strange " << out << "\n";
 		}
 	}
 
