diff -ur OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/c OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/c
--- OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/c	2014-02-11 05:59:42.000000000 -0600
+++ OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/c	2015-03-03 13:42:47.586118000 -0600
@@ -2,7 +2,7 @@
 
 cWARN        = -Wall
 
-cc          = gcc -m64 -mcpu=power5+
+cc          = $(MPICC)
 
 include $(RULES)/c$(WM_COMPILE_OPTION)
 
diff -ur OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/c++ OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/c++
--- OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/c++	2014-02-11 05:59:42.000000000 -0600
+++ OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/c++	2015-03-03 13:42:41.395421000 -0600
@@ -2,7 +2,7 @@
 
 c++WARN     = -Wall -Wextra -Wno-unused-parameter -Wold-style-cast
 
-CC          = g++ -m64 -mcpu=power5+
+CC          = $(MPICXX)
 
 include $(RULES)/c++$(WM_COMPILE_OPTION)
 
diff -ur OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/cOpt OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/cOpt
--- OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/cOpt	2014-02-11 05:59:42.000000000 -0600
+++ OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/cOpt	2015-03-03 13:43:24.491257000 -0600
@@ -1,2 +1,2 @@
 cDBUG       =
-cOPT        = -O3
+cOPT        = $(CFLAGS)
diff -ur OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/c++Opt OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/c++Opt
--- OpenFOAM-2.3.0.ORIG/wmake/rules/linuxPPC64Gcc/c++Opt	2014-02-11 05:59:42.000000000 -0600
+++ OpenFOAM-2.3.0/wmake/rules/linuxPPC64Gcc/c++Opt	2015-03-03 13:43:39.185361000 -0600
@@ -1,2 +1,2 @@
 c++DBUG     =
-c++OPT      = -O3
+c++OPT      = $(CXXFLAGS)
