diff -ru ThirdParty-2.1.1.ORIG/Allwmake ThirdParty-2.1.1/Allwmake
--- ThirdParty-2.1.1.ORIG/Allwmake	2011-08-15 13:30:50.862605278 +0200
+++ ThirdParty-2.1.1/Allwmake	2011-08-15 13:31:09.902605278 +0200
@@ -109,12 +109,12 @@
     ;;
 
 MPICH)
-    if [ -r $MPI_ARCH_PATH/lib/libmpich.so ]
+    if [ -r $MPI_ARCH_PATH/lib64/libmpich.so ]
     then
         echo "    have $WM_MPLIB shared library ($FOAM_MPI)"
         echo
         echo
-    elif [ -r $MPI_ARCH_PATH/lib/libmpich.a ]
+    elif [ -r $MPI_ARCH_PATH/lib64/libmpich.a ]
     then
         echo "    have $WM_MPLIB static library ($FOAM_MPI)"
         echo
