diff -ru OpenFOAM-7-version-7.ebpatched/applications/utilities/mesh/manipulation/setSet/Allwmake OpenFOAM-7-version-7/applications/utilities/mesh/manipulation/setSet/Allwmake
--- OpenFOAM-7-version-7.ebpatched/applications/utilities/mesh/manipulation/setSet/Allwmake	2019-07-11 20:59:22.721333916 +0000
+++ OpenFOAM-7-version-7/applications/utilities/mesh/manipulation/setSet/Allwmake	2019-07-11 21:01:57.580139777 +0000
@@ -7,12 +7,12 @@
 unset COMP_FLAGS LINK_FLAGS
 
 # Use readline if available
-if [ -f $EBROOTLIBREADLINE/include/readline/readline.h ]
+if [ -f $EBROOTGENTOO/include/readline/readline.h ]
 then
     echo "    found <readline/readline.h>  --  enabling readline support."
     export COMP_FLAGS="-DHAS_READLINE"
 
-    export LINK_FLAGS="-L$EBROOTLIBREADLINE/lib -lreadline -L$EBROOTNCURSES -lncurses"
+    export LINK_FLAGS="-lreadline -lncurses"
 fi
 
 wmake $targetType
diff -ru OpenFOAM-7-version-7.ebpatched/etc/config.sh/gperftools OpenFOAM-7-version-7/etc/config.sh/gperftools
--- OpenFOAM-7-version-7.ebpatched/etc/config.sh/gperftools	2019-07-11 20:59:22.940335056 +0000
+++ OpenFOAM-7-version-7/etc/config.sh/gperftools	2019-07-11 21:01:30.657999679 +0000
@@ -29,7 +29,7 @@
 #
 #------------------------------------------------------------------------------
 
-GPERFTOOLS_VERSION=gperftools-$EBVERSIONGPERFTOOLS
-GPERFTOOLS_ARCH_PATH=$EBROOTGPERFTOOLS
+GPERFTOOLS_VERSION=gperftools-2.4
+GPERFTOOLS_ARCH_PATH=$EBROOTGENTOO
 
 #------------------------------------------------------------------------------
