--- OpenFOAM-5.0/src/OSspecific/POSIX/signals/sigFpe.C.orig	2017-04-11 13:25:34.090422771 +0000
+++ OpenFOAM-5.0/src/OSspecific/POSIX/signals/sigFpe.C	2017-04-11 13:26:13.356034446 +0000
@@ -59,7 +59,7 @@
     extern void* __libc_malloc(size_t size);
 
     // Override the GLIBC malloc to support mallocNan
-    void* malloc(size_t size)
+    void* malloc(size_t size) throw()
     {
         if (Foam::sigFpe::mallocNanActive_)
         {
