diff -ru OpenFOAM-v2006.orig/wmake/scripts/have_readline OpenFOAM-v2006/wmake/scripts/have_readline
--- OpenFOAM-v2006.orig/wmake/scripts/have_readline	2021-02-15 17:10:37.325268967 +0000
+++ OpenFOAM-v2006/wmake/scripts/have_readline	2021-02-15 17:11:04.656434376 +0000
@@ -108,7 +108,7 @@
 # Output as per search_* function
 have_readline()
 {
-    search_readline "$EBROOTLIBREADLINE"
+    search_readline "$EBROOTGENTOO"
 }
 
 
