diff -ru OpenFOAM-v1906.ebpatched/etc/config.sh/gperftools OpenFOAM-v1906/etc/config.sh/gperftools
--- OpenFOAM-v1906.ebpatched/etc/config.sh/gperftools	2019-07-11 20:02:40.192131248 +0000
+++ OpenFOAM-v1906/etc/config.sh/gperftools	2019-07-11 20:04:37.717723876 +0000
@@ -38,7 +38,7 @@
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
 ##EasyBuild## settings
-GPERFTOOLS_ARCH_PATH="$EBROOTGPERFTOOLS"
+GPERFTOOLS_ARCH_PATH="$NIXUSER_PROFILE"
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff -ru OpenFOAM-v1906.ebpatched/wmake/scripts/have_readline OpenFOAM-v1906/wmake/scripts/have_readline
--- OpenFOAM-v1906.ebpatched/wmake/scripts/have_readline	2019-07-11 20:02:40.950135070 +0000
+++ OpenFOAM-v1906/wmake/scripts/have_readline	2019-07-11 20:05:44.928062787 +0000
@@ -56,7 +56,7 @@
     # warn="==> skip readline"
 
     # Expected location, include/library names
-    prefix="$EBROOTLIBREADLINE"
+    prefix="$NIXUSER_PROFILE"
     incName="readline/readline.h"
     libName="libreadline"
 
