#Use external Paraview installed in $EBROOTPARAVIEW directory
# B. Hajgato Oct 12 2015
--- OpenFOAM-2.4.0/etc/config/paraview.csh.orig	2015-05-22 11:50:30.000000000 +0200
+++ OpenFOAM-2.4.0/etc/config/paraview.csh	2015-10-12 11:10:56.839014057 +0200
@@ -51,7 +51,7 @@
 #- ParaView version, automatically determine major version:
 #setenv ParaView_VERSION 3.12.0
 #setenv ParaView_VERSION 4.0.1
-setenv ParaView_VERSION 4.1.0
+setenv ParaView_VERSION 4.4.0
 setenv ParaView_MAJOR detect
 
 
@@ -85,7 +85,7 @@
 set paraviewInstDir=$WM_THIRD_PARTY_DIR/ParaView-${ParaView_VERSION}
 set paraviewArchName=ParaView-$ParaView_VERSION
 
-setenv ParaView_DIR $WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$paraviewArchName
+setenv ParaView_DIR $EBROOTPARAVIEW
 
 # set paths if binaries or source are present
 if ( -r $ParaView_DIR || -r $paraviewInstDir ) then
--- OpenFOAM-2.4.0/etc/config/paraview.sh.orig	2015-05-22 11:50:30.000000000 +0200
+++ OpenFOAM-2.4.0/etc/config/paraview.sh	2015-10-12 11:10:56.839014057 +0200
@@ -53,7 +53,7 @@
 #- ParaView version, automatically determine major version
 #export ParaView_VERSION=3.12.0
 #export ParaView_VERSION=4.0.1
-export ParaView_VERSION=4.1.0
+export ParaView_VERSION=4.4.0
 export ParaView_MAJOR=detect
 
 
@@ -93,7 +93,7 @@
 paraviewInstDir=$WM_THIRD_PARTY_DIR/ParaView-$ParaView_VERSION
 paraviewArchName=ParaView-$ParaView_VERSION
 
-export ParaView_DIR=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$paraviewArchName
+export ParaView_DIR=$EBROOTPARAVIEW
 
 # set paths if binaries or source are present
 if [ -r $ParaView_DIR -o -r $paraviewInstDir ]
--- OpenFOAM-2.4.0/applications/utilities/postProcessing/graphics/PV4Readers/Allwmake.orig	2015-05-22 11:50:30.000000000 +0200
+++ OpenFOAM-2.4.0/applications/utilities/postProcessing/graphics/PV4Readers/Allwmake	2015-10-12 11:10:56.839014057 +0200
@@ -12,7 +12,6 @@
         }
 
         # ensure CMake gets the correct C++ compiler
-        [ -n "$WM_CXX" ] && export CXX="$WM_CXX"
 
         wmake libso vtkPV4Readers
         PV4blockMeshReader/Allwmake
