#Use external Paraview installed in $EBROOTPARAVIEW directory
# B. Hajgato Oct 28 2014
--- OpenFOAM-2.3.1.ORIG/etc/config/paraview.csh	2014-02-11 12:59:42.000000000 +0100
+++ OpenFOAM-2.3.1/etc/config/paraview.csh	2014-10-27 13:42:19.661573027 +0100
@@ -51,7 +51,7 @@
 #- ParaView version, automatically determine major version:
 #setenv ParaView_VERSION 3.12.0
 #setenv ParaView_VERSION 4.0.1
-setenv ParaView_VERSION 4.1.0
+setenv ParaView_VERSION 4.3.1
 setenv ParaView_MAJOR detect
 
 
@@ -90,7 +90,7 @@
     set paraviewArchName=paraview-$ParaView_VERSION
 endif
 
-setenv ParaView_DIR $WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$paraviewArchName
+setenv ParaView_DIR $EBROOTPARAVIEW
 
 # set paths if binaries or source are present
 if ( -r $ParaView_DIR || -r $paraviewInstDir ) then
--- OpenFOAM-2.3.1.ORIG/etc/config/paraview.sh	2014-10-27 13:38:59.227573046 +0100
+++ OpenFOAM-2.3.1/etc/config/paraview.sh	2014-10-27 13:41:03.145573040 +0100
@@ -53,7 +53,7 @@
 #- ParaView version, automatically determine major version
 #export ParaView_VERSION=3.12.0
 #export ParaView_VERSION=4.0.1
-export ParaView_VERSION=4.1.0
+export ParaView_VERSION=4.3.1
 export ParaView_MAJOR=detect
 
 
@@ -99,7 +99,7 @@
     paraviewArchName=paraview-$ParaView_VERSION
 fi
 
-export ParaView_DIR=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$paraviewArchName
+export ParaView_DIR=$EBROOTPARAVIEW
 
 # set paths if binaries or source are present
 if [ -r $ParaView_DIR -o -r $paraviewInstDir ]
--- OpenFOAM-2.3.1.ORIG/applications/utilities/postProcessing/graphics/PV4Readers/Allwmake	2014-02-11 12:59:42.000000000 +0100
+++ OpenFOAM-2.3.1/applications/utilities/postProcessing/graphics/PV4Readers/Allwmake	2014-10-28 11:44:47.452566004 +0100
@@ -12,7 +12,6 @@
         }
 
         # ensure CMake gets the correct C++ compiler
-        [ -n "$WM_CXX" ] && export CXX="$WM_CXX"
 
         wmake libso vtkPV4Readers
         PV4blockMeshReader/Allwmake
