diff -ru OpenFOAM-2.0.1.ORIG/src/thermophysicalModels/specie/reaction/reactions/makeReactionThermoReactions.C OpenFOAM-2.0.1/src/thermophysicalModels/specie/reaction/reactions/makeReactionThermoReactions.C
--- OpenFOAM-2.0.1.ORIG/src/thermophysicalModels/specie/reaction/reactions/makeReactionThermoReactions.C	2011-08-03 11:42:19.000000000 +0200
+++ OpenFOAM-2.0.1/src/thermophysicalModels/specie/reaction/reactions/makeReactionThermoReactions.C	2014-09-28 21:56:36.651356163 +0200
@@ -52,6 +52,10 @@
 defineTemplateRunTimeSelectionTable(icoPoly8Reaction, Istream);
 defineTemplateRunTimeSelectionTable(icoPoly8Reaction, dictionary);
 
+defineTemplateTypeNameAndDebug(constGasReaction, 0);
+defineTemplateRunTimeSelectionTable(constGasReaction, Istream);
+defineTemplateRunTimeSelectionTable(constGasReaction, dictionary);
+
 
 // * * * * * * * * * * * * * Make CHEMKIN reactions  * * * * * * * * * * * * //
 
