--- ThirdParty-3.0.0/Allwmake.orig	2015-11-02 20:55:34.663135234 +0100
+++ ThirdParty-3.0.0/Allwmake	2015-11-16 11:33:12.114263189 +0100
@@ -108,12 +108,12 @@
     ;;
 
 MPICH)
-    if [ -r $MPI_ARCH_PATH/lib/libmpich.so ]
+    if [ -r $MPI_ARCH_PATH/lib64/libmpich.so ]
     then
         echo "    have $WM_MPLIB shared library ($FOAM_MPI)"
         echo
         echo
-    elif [ -r $MPI_ARCH_PATH/lib/libmpich.a ]
+    elif [ -r $MPI_ARCH_PATH/lib64/libmpich.a ]
     then
         echo "    have $WM_MPLIB static library ($FOAM_MPI)"
         echo
@@ -197,6 +197,13 @@
     echo "    scotch header in $SCOTCH_ARCH_PATH/include"
     echo "    scotch libs   in $FOAM_EXT_LIBBIN"
     echo
+elif [ -f $SCOTCH_ARCH_PATH/include/scotch.h \
+  -a \( -r $SCOTCH_ARCH_PATH/lib/libscotch.so        -o -r $SCOTCH_ARCH_PATH/lib/libscotch.a        \) \
+  -a \( -r $SCOTCH_ARCH_PATH/lib/libscotcherrexit.so -o -r $SCOTCH_ARCH_PATH/lib/libscotcherrexit.a \) ]
+then
+    echo "    scotch header in $SCOTCH_ARCH_PATH/include"
+    echo "    scotch libs   in $SCOTCH_ARCH_PATH/lib"
+    echo
 else
 (
     set -x
@@ -258,6 +265,13 @@
         echo "    ptscotch header in $SCOTCH_ARCH_PATH/include/$FOAM_MPI"
         echo "    ptscotch libs   in $FOAM_EXT_LIBBIN/$FOAM_MPI"
         echo
+    elif [ -f $SCOTCH_ARCH_PATH/include/ptscotch.h \
+      -a \( -r $SCOTCH_ARCH_PATH/lib/libptscotch.so        -o -r $SCOTCH_ARCH_PATH/lib/libptscotch.a        \) \
+      -a \( -r $SCOTCH_ARCH_PATH/lib/libptscotcherrexit.so -o -r $SCOTCH_ARCH_PATH/lib/libptscotcherrexit.a \) ]
+    then
+        echo "    ptscotch header in $SCOTCH_ARCH_PATH/include"
+        echo "    ptscotch libs   in $SCOTCH_ARCH_PATH/lib"
+        echo
     else
     (
         set -x
@@ -298,7 +312,7 @@
     fi
 
     # verify existence of scotch include
-    [ -f $SCOTCH_ARCH_PATH/include/$FOAM_MPI/ptscotch.h ] || {
+    [ -f $SCOTCH_ARCH_PATH/include/$FOAM_MPI/ptscotch.h -o -f $SCOTCH_ARCH_PATH/include/ptscotch.h ] || {
         echo
         echo "    WARNING: required include file 'ptscotch.h' not found!"
         echo
