diff -ru python-parakeet-0.4.7/setup.py python-parakeet-0.4.7_relax_requirements/setup.py
--- python-parakeet-0.4.7/setup.py	2023-10-24 17:22:22.000000000 +0200
+++ python-parakeet-0.4.7_relax_requirements/setup.py	2023-11-23 10:38:54.117546972 +0100
@@ -64,11 +64,11 @@
             "h5py",
             "maptools",
             "mrcfile",
-            "numpy==1.23",  # Until scikit-image updates
+            "numpy>=1.22",  # Until scikit-image updates
             "pandas",
             "pillow",
             "profet",
-            "pydantic==1.10.10",
+            "pydantic>=1.10",
             "python-multem",
             "pyyaml",
             "scipy",
Only in python-parakeet-0.4.7_relax_requirements/: .setup.py.swp
