skip building of docs
author: Kenneth Hoste (HPC-UGent)
--- Makefile.in.orig	2014-03-26 17:18:53.474997093 +0100
+++ Makefile.in	2014-03-26 17:19:18.844596386 +0100
@@ -291,7 +291,7 @@
 AM_CPPFLAGS = $(GNL_CPPFLAGS)
 
 #SUBDIRS = External doc sophia dint src 
-SUBDIRS = External sophia sibyll dint src  doc
+SUBDIRS = External sophia sibyll dint src # doc
 EXTRA_DIST = sophia/src/sophia.h sibyll/sibyll.h src/Interactions/proton_f77 src/Interactions/proton_sophia dint/src/tables sysdep.h get_externals.sh GetPDCrossSections.sh HalfLifeTable doc/Doxyfile COPYRIGHT doc/UserGuide.tex doc/precision_traj.ps doc/PhotonDensityCollection2.eps doc/crpdetector.eps doc/AllIRBzEvolutionModelsCan.eps IRBzRedshiftEvol_Kneiske examples/GettingStarted Plot
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
