diff -ru mpfr-3.1.0.orig/configure mpfr-3.1.0/configure
--- mpfr-3.1.0.orig/configure	2011-10-03 10:17:33.000000000 +0200
+++ mpfr-3.1.0/configure	2012-10-31 12:07:01.554917561 +0100
@@ -4751,7 +4751,7 @@
 
  { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
- CFLAGS="-fp_port -mp -wd1572 -wd265 -wd186 -wd239 $CFLAGS"
+ CFLAGS="-fp_port -wd1572 -wd265 -wd186 -wd239 $CFLAGS"
 
 else
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
