diff -ruN simdjson-4.6.4.orig/CMakeLists.txt simdjson-4.6.4/CMakeLists.txt
--- simdjson-4.6.4.orig/CMakeLists.txt	2026-05-06 23:29:56.000000000 +0200
+++ simdjson-4.6.4/CMakeLists.txt	2026-05-07 23:44:46.680706223 +0200
@@ -309,6 +309,13 @@
         FILES "${CMAKE_CURRENT_BINARY_DIR}/simdjson.pc"
         DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
     )
+    if(SIMDJSON_BUILD_STATIC_LIB)
+        configure_file("simdjson_static.pc.in" "simdjson_static.pc" @ONLY)
+        install(
+            FILES "${CMAKE_CURRENT_BINARY_DIR}/simdjson_static.pc"
+            DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
+        )
+    endif()
 endif()
 
 #
diff -ruN simdjson-4.6.4.orig/simdjson_static.pc.in simdjson-4.6.4/simdjson_static.pc.in
--- simdjson-4.6.4.orig/simdjson_static.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ simdjson-4.6.4/simdjson_static.pc.in	2026-05-07 23:44:54.282712236 +0200
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+includedir=@PKGCONFIG_INCLUDEDIR@
+libdir=@PKGCONFIG_LIBDIR@
+
+Name: @PROJECT_NAME@
+Description: @PROJECT_DESCRIPTION@
+URL: @PROJECT_HOMEPAGE_URL@
+Version: @PROJECT_VERSION@
+Cflags: -I${includedir} @PKGCONFIG_CFLAGS@
+Libs: -L${libdir} -l@PROJECT_NAME@_static
+@PKGCONFIG_LIBS_PRIVATE@
