Remove includes of non-existing headers
author: Alex Domingo (Vrije Universiteit Brussel)
--- src/simde/x86/avx512/fmadd.h.orig	2023-02-22 01:14:34.516573000 +0100
+++ src/simde/x86/avx512/fmadd.h	2023-02-22 01:14:41.974985000 +0100
@@ -30,7 +30,6 @@
 
 #include "types.h"
 #include "mov.h"
-#include "../fma.h"
 
 HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
--- src/simde/x86/avx512/fmsub.h.orig	2023-02-22 01:28:37.589836000 +0100
+++ src/simde/x86/avx512/fmsub.h	2023-02-22 01:28:46.136128842 +0100
@@ -30,7 +30,6 @@
 
 #include "types.h"
 #include "mov.h"
-#include "../fma.h"
 
 HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
--- src/simde/x86/avx512/fnmadd.h.orig	2023-02-22 01:35:50.825564000 +0100
+++ src/simde/x86/avx512/fnmadd.h	2023-02-15 15:36:00.000000000 +0100
@@ -30,7 +30,6 @@
 
 #include "types.h"
 #include "mov.h"
-#include "../fma.h"
 
 HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
--- src/simde/x86/avx512/fnmsub.h.orig	2023-02-22 01:42:35.610381000 +0100
+++ src/simde/x86/avx512/fnmsub.h	2023-02-22 01:42:43.828398000 +0100
@@ -30,7 +30,6 @@
 
 #include "types.h"
 #include "mov.h"
-#include "../fma.h"
 
 HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
--- src/simde/x86/avx512/scalef.h.orig	2023-02-22 01:48:02.614308000 +0100
+++ src/simde/x86/avx512/scalef.h	2023-02-22 01:48:09.999588167 +0100
@@ -3,7 +3,6 @@
 
 #include "types.h"
 #include "flushsubnormal.h"
-#include "../svml.h"
 
 HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
