Fix typecast syntax
author: Alex Domingo (Vrije Universiteit Brussel)
--- src/select64-common.h.orig	2023-02-22 02:29:16.028806000 +0100
+++ src/select64-common.h	2023-02-22 02:28:24.058561000 +0100
@@ -67,7 +67,7 @@
 	return place + kSelectInByte[((x >> place) & 0xFF) | (byteRank << 8)];
 #elif defined(__GNUC__) || defined(__clang__)
 	// GCC and Clang won't inline the intrinsics.
-	uint64_t result = uint64_t(1) << k;
+	uint64_t result = (uint64_t)(1) << k;
 
 	asm("pdep %1, %0, %0\n\t"
 		"tzcnt %0, %0"
