# use 256bit version of call
# wpoely86@gmail.com
diff -ur gmap-2019-03-15.orig/src/merge-uint8.c gmap-2019-03-15/src/merge-uint8.c
--- gmap-2019-03-15.orig/src/merge-uint8.c	2018-10-07 18:38:05.000000000 +0200
+++ gmap-2019-03-15/src/merge-uint8.c	2019-04-09 12:10:14.232828157 +0200
@@ -69,7 +69,7 @@
   /* print_vector(vTmp,"Tmp 4"); */
   /* print_vector(vMin,"Min 4"); */
 
-  *vMergedB = _mm_max_epu64(vTmp, vMax);
+  *vMergedB = _mm256_max_epu64(vTmp, vMax);
   *vMergedA = _mm256_alignr_epi64(vMin, vMin, 1);
 
   return;
