diff -ru Multiwfn_3.6_dev_src_Linux.orig/Makefile Multiwfn_3.6_dev_src_Linux/Makefile
--- Multiwfn_3.6_dev_src_Linux.orig/Makefile	2018-11-16 15:59:47.346309775 +0000
+++ Multiwfn_3.6_dev_src_Linux/Makefile	2018-11-16 16:00:17.680033511 +0000
@@ -1,6 +1,6 @@
-OPT = -O1 -openmp -diag-disable 8290,8291,6371 -threads -openmp-link=static -multiple-processes=4
+OPT = -O1 -qopenmp -diag-disable 8290,8291,6371 -threads -qopenmp-link=static -multiple-processes=4
 #Options in the next line is for debugging purpose
-#OPT = -O0 -openmp -diag-disable 8290,8291,6371 -threads -openmp-link=static -debug all -g -traceback -check all -fstack-protector -multiple-processes=4
+#OPT = -O0 -qopenmp -diag-disable 8290,8291,6371 -threads -qopenmp-link=static -debug all -g -traceback -check all -fstack-protector -multiple-processes=4
 
 #LIB = ./lapack.a ./blas.a ./dislin_d-11.0.a -lXm -lGL #Works for CentOS 6.1
 LIB = ./lapack.a ./blas.a ./dislin_d-11.0.a -lXm -lXt -lX11 -lGL #Works for CentOS 7.2
diff -ru Multiwfn_3.6_dev_src_Linux.orig/Makefile_noGUI Multiwfn_3.6_dev_src_Linux/Makefile_noGUI
--- Multiwfn_3.6_dev_src_Linux.orig/Makefile_noGUI	2018-11-16 15:59:47.346309775 +0000
+++ Multiwfn_3.6_dev_src_Linux/Makefile_noGUI	2018-11-16 16:00:17.680033511 +0000
@@ -1,4 +1,4 @@
-OPT = -O1 -openmp -diag-disable 8290,8291,6371 -threads -openmp-link=static -multiple-processes=4
+OPT = -O1 -qopenmp -diag-disable 8290,8291,6371 -threads -qopenmp-link=static -multiple-processes=4
 LIB = ./lapack.a ./blas.a
 FC = ifort
 EXE = Multiwfn
