diff -ru deepTools-3.1.2/bin/estimateScaleFactor deepTools-3.1.2.fixed/bin/estimateScaleFactor
--- deepTools-3.1.2/bin/estimateScaleFactor	2018-08-21 08:21:03.000000000 +0000
+++ deepTools-3.1.2.fixed/bin/estimateScaleFactor	2018-08-29 16:23:53.229940000 +0000
@@ -16,7 +16,7 @@
     try:
          open( string, 'r').close()
     except IOError as detail:
-         print detail
+         print(detail)
          exit()
     return string
 
@@ -103,7 +103,7 @@
 
     """
     if len(args.bamfiles) > 2:
-        print "SES method to stimate scale factors only works for two samples"
+        print("SES method to stimate scale factors only works for two samples")
         exit(0)
 
     sys.stderr.write("{:,} number of samples will be computed.\n".format(args.numberOfSamples))
@@ -115,7 +115,7 @@
                                           verbose=args.verbose)
 
     for k, v in sizeFactorsDict.iteritems():
-         print "{}: {}".format(k, v)
+         print("{}: {}".format(k, v))
 
 
 if __name__ == "__main__":

