diff -ru mpiblast-1.6.0.orig/ncbi/make/makedis.csh mpiblast-1.6.0/ncbi/make/makedis.csh
--- mpiblast-1.6.0.orig/ncbi/make/makedis.csh	2010-06-29 03:04:36.000000000 +0200
+++ mpiblast-1.6.0/ncbi/make/makedis.csh	2013-01-23 15:32:34.201394361 +0100
@@ -355,8 +355,8 @@
 #
 
 #uncomment two following lines to don't build X11 apps
-#set HAVE_OGL=0
-#set HAVE_MOTIF=0
+set HAVE_OGL=0
+set HAVE_MOTIF=0
 
 # if $OPENGL_TARGETS (in <platform>.ncbi.mk) is defined, 
 # then add the appropriate flags, libraries, and binaries for OpenGL apps
