--- ocaml-4.01.0/configure.orig	2014-08-19 12:26:23.008974288 +0200
+++ ocaml-4.01.0/configure	2014-08-19 12:27:15.800298562 +0200
@@ -879,12 +879,12 @@
 
 # For the Sys module
 
-if sh ./hasgot getrusage; then
+if cc="$cc -wd165" sh ./hasgot getrusage; then
   echo "getrusage() found."
   echo "#define HAS_GETRUSAGE" >> s.h
 fi
 
-if sh ./hasgot times; then
+if cc="$cc -wd165" sh ./hasgot times; then
   echo "times() found."
   echo "#define HAS_TIMES" >> s.h
 fi
@@ -967,7 +967,7 @@
   echo "#define HAS_MKFIFO" >> s.h
 fi
 
-if sh ./hasgot getcwd; then
+if cc="$cc -wd165" sh ./hasgot getcwd; then
   echo "getcwd() found."
   echo "#define HAS_GETCWD" >> s.h
 fi
@@ -982,7 +982,7 @@
   echo "#define HAS_GETPRIORITY" >> s.h
 fi
 
-if sh ./hasgot -i sys/types.h -i utime.h && sh ./hasgot utime; then
+if cc="$cc -wd165" sh ./hasgot -i sys/types.h -i utime.h && cc="$cc -wd165" sh ./hasgot utime; then
   echo "utime() found."
   echo "#define HAS_UTIME" >> s.h
 fi
@@ -992,7 +992,7 @@
   echo "#define HAS_UTIMES" >> s.h
 fi
 
-if sh ./hasgot dup2; then
+if cc="$cc -wd165" sh ./hasgot dup2; then
   echo "dup2() found."
   echo "#define HAS_DUP2" >> s.h
 fi
@@ -1002,7 +1002,7 @@
   echo "#define HAS_FCHMOD" >> s.h
 fi
 
-if sh ./hasgot truncate ftruncate; then
+if cc="$cc -wd165" sh ./hasgot truncate ftruncate; then
   echo "truncate() found."
   echo "#define HAS_TRUNCATE" >> s.h
 fi
@@ -1015,7 +1015,7 @@
 fi
 
 has_select=no
-if sh ./hasgot select && \
+if cc="$cc -wd165" sh ./hasgot select && \
    sh ./hasgot -i sys/types.h $select_include -t fd_set ; then
   echo "select() found."
   echo "#define HAS_SELECT" >> s.h
@@ -1028,7 +1028,7 @@
 fi
 
 has_wait=no
-if sh ./hasgot waitpid;  then
+if cc="$cc -wd165" sh ./hasgot waitpid;  then
   echo "waitpid() found."
   echo "#define HAS_WAITPID" >> s.h
   has_wait=yes
@@ -1084,26 +1084,26 @@
 fi
 
 has_gettimeofday=no
-if sh ./hasgot gettimeofday; then
+if cc="$cc -wd165" sh ./hasgot gettimeofday; then
   echo "gettimeofday() found."
   echo "#define HAS_GETTIMEOFDAY" >> s.h
   has_gettimeofday="yes"
 fi
 
-if sh ./hasgot mktime; then
+if cc="$cc -wd165" sh ./hasgot mktime; then
   echo "mktime() found."
   echo "#define HAS_MKTIME" >> s.h
 fi
 
 case "$host" in
   *-*-cygwin*) ;;  # setsid emulation under Cygwin breaks the debugger
-  *) if sh ./hasgot setsid; then
+  *) if cc="$cc -wd165" sh ./hasgot setsid; then
        echo "setsid() found."
        echo "#define HAS_SETSID" >> s.h
      fi;;
 esac
 
-if sh ./hasgot putenv; then
+if cc="$cc -wd165" sh ./hasgot putenv; then
   echo "putenv() found."
   echo "#define HAS_PUTENV" >> s.h
 fi
@@ -1131,7 +1131,7 @@
   fi
 fi
 
-if sh ./hasgot -i sys/types.h -i sys/mman.h && sh ./hasgot mmap munmap; then
+if cc="$cc -wd165" sh ./hasgot -i sys/types.h -i sys/mman.h && cc="$cc -wd165" sh ./hasgot mmap munmap; then
   echo "mmap() found."
   echo "#define HAS_MMAP" >> s.h
 fi
