--- julia-1.1.0/base/Makefile.orig	2019-01-13 00:25:11.000000000 +0000
+++ julia-1.1.0/base/Makefile	2019-02-22 13:24:54.111566565 +0000
@@ -208,7 +208,7 @@
 LLVM_CONFIG_HOST_LIBS := $(shell $(LLVM_CONFIG_HOST) --libdir)/libLLVM.$(SHLIB_EXT)
 endif
 $(build_private_libdir)/libLLVM.dylib:
-	REALPATH=$(LLVM_CONFIG_HOST_LIBS) && \
+	REALPATH=`$(LLVM_CONFIG_HOST) --libdir`/libLLVM.$(SHLIB_EXT) && \
 	$(call resolve_path,REALPATH) && \
 	[ -e "$$REALPATH" ] && \
 	([ ! -e "$@" ] || rm "$@") && \
