diff -ur julia-1.5.2.orig/deps/blas.mk julia-1.5.2/deps/blas.mk
--- julia-1.5.2.orig/deps/blas.mk	2020-06-26 20:32:24.000000000 -0000
+++ julia-1.5.2/deps/blas.mk	2020-10-07 02:51:39.382511592 -0000
@@ -23,7 +23,7 @@
 OPENBLAS_BUILD_OPTS += NUM_THREADS=16
 else
 # On linux, try to provision for the largest possible machine currently
-OPENBLAS_BUILD_OPTS += NUM_THREADS=16
+OPENBLAS_BUILD_OPTS += NUM_THREADS=64
 endif
 else
 OPENBLAS_BUILD_OPTS += USE_THREAD=0
