Seulement dans phana: clapack.h
Seulement dans phana: f2c.h
diff -ru phana.orig/Makefile phana/Makefile
--- phana.orig/Makefile	2018-04-20 14:13:02.277272267 +0000
+++ phana/Makefile	2018-04-20 14:18:44.750990302 +0000
@@ -1,16 +1,18 @@
 .SUFFIXES : .o .cpp
 # compiler and flags
 CC     = g++ -Wno-unused-result
-LINK   = $(CC) -static
-CFLAGS = -O3 $(DEBUG) $(UFLAG)
+LINK   = $(CC) -fopenmp 
+CFLAGS = -O3 -fopenmp $(DEBUG) $(UFLAG)
 #
 OFLAGS = -O3 $(DEBUG)
 INC    = $(LPKINC) $(TCINC) $(SPGINC)
 LIB    = $(LPKLIB) $(TCLIB) $(SPGLIB)
 #
 # cLapack library needed
-LPKINC = -I/opt/libs/clapack/3.2.1/include
-LPKLIB = -L/opt/libs/clapack/3.2.1/lib -lclapack -lblas -lf2c #-lm
+#LPKINC = -I/opt/libs/clapack/3.2.1/include
+#LPKLIB = -L/opt/libs/clapack/3.2.1/lib -lclapack -lblas -lf2c #-lm
+LPKINC = 
+LPKLIB =  -L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_intel_ilp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
 #
 # Tricubic library needed
 TCINC = -I/opt/libs/tricubic/1.0/include
