fix import statement for pandas and Jinja2
author: Denis Kristak (INUITS) + Kenneth Hoste (HPC-UGent)
diff -ru vispr-0.4.14.orig/vispr/results/rna.py vispr-0.4.14/vispr/results/rna.py
--- vispr-0.4.14.orig/vispr/results/rna.py	2016-08-11 09:16:23.000000000 +0200
+++ vispr-0.4.14/vispr/results/rna.py	2023-04-10 19:19:07.488661966 +0200
@@ -11,7 +11,7 @@
 
 from flask import render_template
 import pandas as pd
-from pandas.io.common import EmptyDataError
+from pandas.errors import EmptyDataError
 import numpy as np
 from sklearn.decomposition import PCA
 from scipy.cluster.hierarchy import average, leaves_list, dendrogram
diff -ru vispr-0.4.14.orig/vispr/server.py vispr-0.4.14/vispr/server.py
--- vispr-0.4.14.orig/vispr/server.py	2015-11-08 00:47:23.000000000 +0100
+++ vispr-0.4.14/vispr/server.py	2023-04-10 19:25:06.417148288 +0200
@@ -10,7 +10,7 @@
 
 import numpy as np
 from flask import Flask, render_template, request, session, abort
-from jinja2 import Markup
+from markupsafe import Markup
 import yaml
 
 from vispr import __version__
