--- configure.orig	2011-03-16 22:52:37.277341979 +0100
+++ configure	2011-03-16 22:53:46.787359175 +0100
@@ -3228,7 +3228,7 @@
 
     # At this time, libfftw3.a should always be prefered over .so,
     # to avoid forcing users to install fftw3.
-    check_library "libfftw3 libfftw3-3" "no" "$fftw3libdir" \
+    check_library "libfftw3 libfftw3-3 libmkl_core" "no" "$fftw3libdir" \
         $FFTW3 ${FFTW3:+$FFTW3/lib} ${FFTW3:+$FFTW3/.libs} \
         ${finkdir:+$finkdir/lib} \
         /usr/local/lib /usr/lib /opt/fftw3/lib
