diff -ru root.orig/net/rpdutils/src/rpdutils.cxx root/net/rpdutils/src/rpdutils.cxx
--- root.orig/net/rpdutils/src/rpdutils.cxx	2017-09-01 17:22:20.446335818 +0000
+++ root/net/rpdutils/src/rpdutils.cxx	2017-09-01 17:23:32.544661570 +0000
@@ -226,8 +226,8 @@
 #else
 #if defined(linux) && !defined(R__HAS_SETRESUID)
 extern "C" {
-   int setresgid(gid_t r, gid_t e, gid_t s);
-   int setresuid(uid_t r, uid_t e, uid_t s);
+   int setresgid(gid_t r, gid_t e, gid_t s) throw ();
+   int setresuid(uid_t r, uid_t e, uid_t s) throw ();
 }
 #endif
 #endif
