diff -Nru root-6.10.08.orig/cmake/modules/CheckCompiler.cmake root-6.10.08/cmake/modules/CheckCompiler.cmake
--- root-6.10.08.orig/cmake/modules/CheckCompiler.cmake	2018-12-21 12:27:59.355248710 +0100
+++ root-6.10.08/cmake/modules/CheckCompiler.cmake	2018-12-21 13:49:10.884122500 +0100
@@ -17,15 +17,6 @@
   set(CMAKE_Fortran_COMPILER CMAKE_Fortran_COMPILER-NOTFOUND)
 endif()
 
-#----Get the compiler file name (to ensure re-location)---------------------------------------------
-get_filename_component(_compiler_name ${CMAKE_CXX_COMPILER} NAME)
-get_filename_component(_compiler_path ${CMAKE_CXX_COMPILER} PATH)
-if("$ENV{PATH}" MATCHES ${_compiler_path})
-  set(CXX ${_compiler_name})
-else()
-  set(CXX ${CMAKE_CXX_COMPILER})
-endif()
-
 #----Test if clang setup works----------------------------------------------------------------------
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   exec_program(${CMAKE_CXX_COMPILER} ARGS "--version 2>&1 | grep version" OUTPUT_VARIABLE _clang_version_info)
diff -Nru root-6.10.08.orig/cmake/modules/RootConfiguration.cmake root-6.10.08/cmake/modules/RootConfiguration.cmake
--- root-6.10.08.orig/cmake/modules/RootConfiguration.cmake	2018-12-21 12:27:59.274249410 +0100
+++ root-6.10.08/cmake/modules/RootConfiguration.cmake	2018-12-21 14:39:42.189750685 +0100
@@ -672,7 +672,8 @@
   # We cannot use the compiledata.sh script for windows
   configure_file(${CMAKE_SOURCE_DIR}/cmake/scripts/compiledata.win32.in ${CMAKE_BINARY_DIR}/include/compiledata.h NEWLINE_STYLE UNIX)
 else()
-  execute_process(COMMAND ${CMAKE_SOURCE_DIR}/build/unix/compiledata.sh ${CMAKE_BINARY_DIR}/include/compiledata.h "${CXX}"
+  execute_process(COMMAND ${CMAKE_SOURCE_DIR}/build/unix/compiledata.sh 
+    ${CMAKE_BINARY_DIR}/include/compiledata.h "${CMAKE_CXX_COMPILER}"
         "${CMAKE_CXX_FLAGS_RELEASE}" "${CMAKE_CXX_FLAGS_DEBUG}" "${CMAKE_CXX_FLAGS}"
         "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS}" "${CMAKE_EXE_FLAGS}"
         "${LibSuffix}" "${SYSLIBS}"
