diff -ru trilinos-10.12.2-Source.orig/packages/ml/src/Coarsen/ml_agg_ParMETIS.c trilinos-10.12.2-Source/packages/ml/src/Coarsen/ml_agg_ParMETIS.c
--- trilinos-10.12.2-Source.orig/packages/ml/src/Coarsen/ml_agg_ParMETIS.c	2012-08-02 19:44:06.000000000 +0200
+++ trilinos-10.12.2-Source/packages/ml/src/Coarsen/ml_agg_ParMETIS.c	2012-08-12 08:20:41.812840502 +0200
@@ -105,6 +105,8 @@
 #define idxtype int
 #endif
 
+typedef idx_t idxtype;
+
 int OPTIMAL_VALUE = 27*27; /* don't ask me why */
 
 int PARMETIS_DEBUG_LEVEL = 0;
