diff -ru trilinos-10.12.2-Source.orig/packages/zoltan/src/zz/murmur3.c trilinos-10.12.2-Source/packages/zoltan/src/zz/murmur3.c
--- trilinos-10.12.2-Source.orig/packages/zoltan/src/zz/murmur3.c	2012-08-02 19:41:42.000000000 +0200
+++ trilinos-10.12.2-Source/packages/zoltan/src/zz/murmur3.c	2013-03-28 11:14:00.430206123 +0100
@@ -12,7 +12,7 @@
 /*----------------------------------------------------------------------------*/
 /* Platform-specific functions and macros */
 
-#define FORCE_INLINE __attribute__((always_inline))
+#define FORCE_INLINE inline __attribute__((always_inline))
 
 /* KDDKDD 
 inline uint32_t rotl32 ( uint32_t x, int8_t r )
