Adding missing include statement
Author: J. Saßmannshausen (Imperial College London/UK)
diff --git a/Trilinos-trilinos-release-16-1-0.orig/packages/krino/krino/geometry/Akri_Triangle.hpp b/Trilinos-trilinos-release-16-1-0/packages/krino/krino/geometry/Akri_Triangle.hpp
index a40ba79..e6a7d5c 100644
--- a/Trilinos-trilinos-release-16-1-0.orig/packages/krino/krino/geometry/Akri_Triangle.hpp
+++ b/Trilinos-trilinos-release-16-1-0/packages/krino/krino/geometry/Akri_Triangle.hpp
@@ -15,6 +15,7 @@
 #include <array>
 
 #include <stk_math/StkVector.hpp>
+#include <tuple>
 
 namespace krino {
 
