--- Trilinos-trilinos-release-13-4-1/packages/seacas/libraries/exodus/src/ex_utils.c~	2022-11-10 17:02:37.000000000 -0500
+++ Trilinos-trilinos-release-13-4-1/packages/seacas/libraries/exodus/src/ex_utils.c	2025-08-23 11:40:44.408198839 -0400
@@ -1742,7 +1742,7 @@
         */
 
         /* const int NC_SZIP_EC = 4; */ /* Selects entropy coding method for szip. */
-        const int NC_SZIP_NN = 32;      /* Selects nearest neighbor coding method for szip. */
+        /* const int NC_SZIP_NN = 32; */ /* Selects nearest neighbor coding method for szip. */
         /* Even and between 4 and 32; typical values are 8, 10, 16, 32 */
         const int SZIP_PIXELS_PER_BLOCK =
             file->compression_level == 0 ? 32 : file->compression_level;
--- Trilinos-trilinos-release-13-4-1/packages/seacas/applications/blot/mscomd.f~	2022-11-10 17:02:37.000000000 -0500
+++ Trilinos-trilinos-release-13-4-1/packages/seacas/applications/blot/mscomd.f	2025-08-23 12:56:51.195840808 -0400
@@ -482,8 +482,8 @@
             CALL INIINT (3, 1, LTYP)
             CALL SETMSH (2, 'UNDEFORM', 'NONE', MSHSEL, LTYP,
      &         0, IDUM, 0, IDUM, 'WIREFRAM', ' ', ISSNPS, ISSESS)
-            CALL SCOLOR (.TRUE., CDUM, IDUM, IDUM, RDUM, IDUM,
-     *        CDUM, SHDCOL, ISHDCL, IDELB)
+            CALL SCOLOR (.TRUE., INLINE, IFLD, INTYP,
+     &         RFIELD, IFIELD, CFIELD, SHDCOL, ISHDCL, IDELB)
 
 C         --Set display options
 
--- Trilinos-trilinos-release-13-4-1/packages/PyTrilinos/src/numpy_include.hpp~	2022-11-10 17:02:37.000000000 -0500
+++ Trilinos-trilinos-release-13-4-1/packages/PyTrilinos/src/numpy_include.hpp	2025-08-23 21:45:37.201104575 -0400
@@ -60,6 +60,7 @@
 // import_array().
 
 #include <Python.h>
+#define NPY_API_SYMBOL_ATTRIBUTE
 #define PY_ARRAY_UNIQUE_SYMBOL PyTrilinos_NumPy
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>
