diff -ru rkcommon-1.10.0.orig/cmake/rkcommon_macros.cmake rkcommon-1.10.0/cmake/rkcommon_macros.cmake
--- rkcommon-1.10.0.orig/cmake/rkcommon_macros.cmake	2023-01-31 20:15:30.506753620 +0000
+++ rkcommon-1.10.0/cmake/rkcommon_macros.cmake	2023-01-31 20:15:50.274908544 +0000
@@ -183,7 +183,7 @@
     endif()
 
     # Try getting TBB via config first
-    find_package(TBB 2021.1 QUIET COMPONENTS tbb tbbmalloc CONFIG ${TBB_FIND_PACKAGE_OPTION})
+    find_package(TBB 2020...2021.1 QUIET COMPONENTS tbb tbbmalloc CONFIG ${TBB_FIND_PACKAGE_OPTION})
     if (TBB_FOUND)
       list(APPEND RKCOMMON_TASKING_LIBS TBB::tbb TBB::tbbmalloc)
       set(RKCOMMON_TASKING_DEFINITIONS RKCOMMON_TASKING_TBB)
