Fixes a copy-paste typo of spambind
Author: Petr Král (INUITS)
--- src/spambind/mesh/src/connectivityMatrix.cpp.orig	2025-12-15 20:13:57.899663788 +0100
+++ src/spambind/mesh/src/connectivityMatrix.cpp	2025-12-15 20:18:18.035151554 +0100
@@ -198,7 +198,7 @@
                                 numTetrahedra_alpha[0] = minAlpha;
                         }
                         as.set_alpha ( numTetrahedra_alpha[0] );
-                        CGAL_assertion(as.number_of_vertices() == numTetrahedra_numVertices);
+                        CGAL_assertion(as.number_of_vertices() == connectivityMatrix_numVertices);
                         int numAlpha = 0; int i= 0;
                         RTriangulation::Finite_cells_iterator cell_end = as.finite_cells_end();
                         for ( RTriangulation::Finite_cells_iterator cell = as.finite_cells_begin(); cell != cell_end; cell++ ){
@@ -212,7 +212,7 @@
 
                 }else{
                         // std::cout << "running in standard mode" << std::endl;
-                        CGAL_assertion(T.number_of_vertices() == numTetrahedra_numVertices);
+                        CGAL_assertion(T.number_of_vertices() == connectivityMatrix_numVertices);
                 }
 
                 RTriangulation::Finite_cells_iterator cit; uint32_t tetNumber = 0; uint32_t cellNumber = 0;
