Add missing include of cstdint

Åke Sandgren, 2024-12-13
diff -ru Mash-2.3.orig/src/mash/Command.cpp Mash-2.3/src/mash/Command.cpp
--- Mash-2.3.orig/src/mash/Command.cpp	2021-02-27 00:13:33.000000000 +0100
+++ Mash-2.3/src/mash/Command.cpp	2024-12-13 08:08:30.586895593 +0100
@@ -8,6 +8,7 @@
 #include <sys/ioctl.h>
 #include <sstream>
 #include <fstream>
+#include <cstdint>
 
 #include "Command.h"
 #include "version.h"
