diff -ru model-angelo-1.0.14/setup.py model-angelo-1.0.14_requirements/setup.py
--- model-angelo-1.0.14/setup.py	2025-05-08 13:15:47.000000000 +0200
+++ model-angelo-1.0.14_requirements/setup.py	2025-05-23 11:16:34.285077409 +0200
@@ -33,9 +33,9 @@
         "matplotlib",
         "mrcfile",
         "pandas",
-        "fair-esm==1.0.3",
-        "pyhmmer==0.7.1",
+        "fair-esm<=2.0.0",
+        "pyhmmer<=1.10.15",
         "loguru",
-        "numpy<2.0,
+        "numpy<=1.25.1",
     ],
 )
