--- grace-5.99.0/src/Makefile	2004-06-25 21:45:18.000000000 +0000
+++ grace-5.99.0/src/Makefile	2018-11-12 16:36:45.288958943 +0000
@@ -41,7 +41,7 @@
 
 include Make.dep
 
-Make.dep : $(GRSRCS) $(GUISRCS) $(PARS_C) buildinfo.c
+Make.dep : $(GRSRCS) $(GUISRCS) $(PARS_C) buildinfo.c buildinfo.h
 	echo "# Generated automatically by \`make depend'" > $@
 	$(CC) $(CFLAGS) -MM $(GRSRCS) $(GUISRCS) $(PARS_C) buildinfo.c \
 		| sed 's/\.o:/\$$(O)\ :/g' >> $@
