diff -ru libxcb-1.8.orig/configure.ac libxcb-1.8/configure.ac
--- libxcb-1.8.orig/configure.ac	2012-01-11 18:23:46.000000000 +0100
+++ libxcb-1.8/configure.ac	2012-09-26 15:07:45.758493553 +0200
@@ -35,7 +35,7 @@
 
 # Checks for pkg-config packages
 PKG_CHECK_MODULES(XCBPROTO, xcb-proto >= 1.6)
-NEEDED="pthread-stubs xau >= 0.99.2"
+NEEDED="xau >= 0.99.2"
 PKG_CHECK_MODULES(NEEDED, $NEEDED)
 
 have_xdmcp="no"
diff -ru libxcb-1.8.orig/configure libxcb-1.8/configure
--- libxcb-1.8.orig/configure	2012-01-11 18:27:53.000000000 +0100
+++ libxcb-1.8/configure	2012-09-26 15:11:58.399805814 +0200
@@ -13088,7 +13088,7 @@
 $as_echo "yes" >&6; }
 
 fi
-NEEDED="pthread-stubs xau >= 0.99.2"
+NEEDED="xau >= 0.99.2"
 
 pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for NEEDED" >&5
