index 06c587f..9c95a11 100644
--- Makefile.in
+++ Makefile.in
@@ -71,7 +71,7 @@ install_mpich2:
 	@rmdir $(prefix)/sbin || echo ok
 # Hack: use lib*.so.3 when app needs lib*.so.1.1 or 1.2
 	set -x; \
-	for l in $(prefix)/lib/lib*.so.3; do \
+	for l in $(prefix)/lib*/lib*.so.3; do \
 	    test -f "$$l" || { echo "Error: Missing $$l!"; exit 1; }; \
 	    ln -s "$${l##*/}" "$${l%.3}.1.1"; \
 	    ln -s "$${l##*/}" "$${l%.3}.1.2"; \
