--- a/Makefile.in
+++ b/Makefile.in
@@ -70,8 +70,7 @@ install_mpich2:
 	$(RM) $(prefix)/sbin/mpetestlink
 	@rmdir $(prefix)/sbin || echo ok
 # Hack: use lib*.so.3 when app needs lib*.so.1.1 or 1.2
-	set -x; \
-	for l in $(prefix)/lib/lib*.so.3; do \
+	for l in $(prefix)/lib*/lib*.so.3; do \
 	    test -f "$$l" || { echo "Error: Missing $$l!"; exit 1; }; \
 	    ln -sf "$${l##*/}" "$${l%.3}.1.1"; \
 	    ln -sf "$${l##*/}" "$${l%.3}.1.2"; \
