# Patch numpy version to build with 2020b toolchains
# S.D.Pinches, 2022.08.31
diff -Nru muscle3-0.5.0-orig/docs/source/examples/python/requirements.txt muscle3-0.5.0/docs/source/examples/python/requirements.txt
--- muscle3-0.5.0-orig/docs/source/examples/python/requirements.txt	2022-07-15 16:58:45.000000000 +0200
+++ muscle3-0.5.0/docs/source/examples/python/requirements.txt	2022-08-31 12:00:19.778089113 +0200
@@ -1,7 +1,7 @@
 matplotlib>=3,<4
 numpy==1.19.5; python_version=='3.6'
 numpy<1.22; python_version=='3.7'
-numpy>=1.22,<=1.25; python_version>='3.8'
+numpy>=1.19,<=1.25; python_version>='3.8'
 sobol_seq==0.2.0
 yatiml==0.9.0
 ymmsl>=0.12.0,<0.13
diff -Nru muscle3-0.5.0-orig/setup.py muscle3-0.5.0/setup.py
--- muscle3-0.5.0-orig/setup.py	2022-07-15 16:58:45.000000000 +0200
+++ muscle3-0.5.0/setup.py	2022-08-31 12:00:23.812601438 +0200
@@ -52,7 +52,7 @@
         'netifaces==0.11.0',
         "numpy==1.19.5; python_version=='3.6'",
         "numpy<1.22; python_version=='3.7'",
-        "numpy>=1.22,<=1.25; python_version>='3.8'",
+        "numpy>=1.19,<=1.25; python_version>='3.8'",
         'qcg-pilotjob==0.13.1',
         'typing_extensions<4',
         'ymmsl>=0.12.0,<0.13'          # Also in CI, update there as well
