# Correct libray paths for running tests
# S.D.Pinches, 2023.01.24
diff -Nru muscle3-0.6.0-orig/libmuscle/fortran/build/libmuscle/tests/Makefile muscle3-0.6.0/libmuscle/fortran/build/libmuscle/tests/Makefile
--- muscle3-0.6.0-orig/libmuscle/fortran/build/libmuscle/tests/Makefile	2023-01-17 23:11:43.000000000 +0100
+++ muscle3-0.6.0/libmuscle/fortran/build/libmuscle/tests/Makefile	2023-01-24 16:00:54.524139000 +0100
@@ -47,6 +47,10 @@
 
 CPP_BUILD_DIR := $(CURDIR)/../../../../cpp/build
 
+LDFLAGS += -L../../ymmsl
+LDFLAGS += -L../../../../cpp/build/libmuscle
+LDFLAGS += -L../../../../cpp/build/ymmsl
+
 LDFLAGS += -pthread $(CURDIR)/../libmuscle_fortran.a $(CURDIR)/../../ymmsl/libymmsl_fortran.a
 LDFLAGS += $(CPP_BUILD_DIR)/libmuscle/libmuscle_d.a $(CPP_BUILD_DIR)/ymmsl/libymmsl_d.a
 
