diff -Nru muscle3-0.7.1-orig/setup.py muscle3-0.7.1/setup.py
--- muscle3-0.7.1-orig/setup.py	2023-10-01 18:50:49.000000000 +0200
+++ muscle3-0.7.1/setup.py	2024-01-11 11:22:30.494825000 +0100
@@ -57,7 +57,7 @@
         'msgpack>=1,<2',
         'netifaces==0.11.0',
         "numpy<1.22; python_version=='3.7'",
-        "numpy>=1.22,<=1.25; python_version>='3.8'",
+        "numpy>=1.22; python_version>='3.8'",
         'qcg-pilotjob==0.13.1',
         'typing_extensions>=4.4.0,<5',
         'ymmsl>=0.13.0,<0.14'          # Also in CI, update there as well
