# Correct libray paths for running tests
# S.D.Pinches, 2022.08.31
diff -Nru muscle3-0.5.0-orig/docs/source/examples/python/requirements.txt muscle3-0.5.0/docs/source/examples/python/requirements.txt
--- muscle3-0.5.0-orig/docs/source/examples/python/requirements.txt	2022-07-15 16:58:45.000000000 +0200
+++ muscle3-0.5.0/docs/source/examples/python/requirements.txt	2022-08-31 12:00:19.778089113 +0200
@@ -1,7 +1,7 @@
 matplotlib>=3,<4
 numpy==1.19.5; python_version=='3.6'
 numpy<1.22; python_version=='3.7'
-numpy>=1.22,<=1.25; python_version>='3.8'
+numpy>=1.19,<=1.25; python_version>='3.8'
 sobol_seq==0.2.0
 yatiml==0.9.0
 ymmsl>=0.12.0,<0.13
diff -Nru muscle3-0.5.0-orig/libmuscle/fortran/build/libmuscle/tests/Makefile muscle3-0.5.0/libmuscle/fortran/build/libmuscle/tests/Makefile
--- muscle3-0.5.0-orig/libmuscle/fortran/build/libmuscle/tests/Makefile	2022-07-15 16:58:45.000000000 +0200
+++ muscle3-0.5.0/libmuscle/fortran/build/libmuscle/tests/Makefile	2022-08-31 18:49:40.620578022 +0200
@@ -41,6 +41,10 @@
 
 CPP_BUILD_DIR := $(CURDIR)/../../../../cpp/build
 
+LDFLAGS += -L../../ymmsl
+LDFLAGS += -L../../../../cpp/build/libmuscle
+LDFLAGS += -L../../../../cpp/build/ymmsl
+
 LDFLAGS += -pthread $(CURDIR)/../libmuscle_fortran.a $(CURDIR)/../../ymmsl/libymmsl_fortran.a
 LDFLAGS += $(CPP_BUILD_DIR)/libmuscle/libmuscle_d.a $(CPP_BUILD_DIR)/ymmsl/libymmsl_d.a
 
