# S.D.Pinches,2023.01.18
# Patch to use numpy version available in 2020b toolchains
diff -Nru muscle3-0.6.0-orig/docs/source/examples/python/requirements.txt muscle3-0.6.0/docs/source/examples/python/requirements.txt
--- muscle3-0.6.0-orig/docs/source/examples/python/requirements.txt	2023-01-17 23:11:43.000000000 +0100
+++ muscle3-0.6.0/docs/source/examples/python/requirements.txt	2023-01-18 11:21:03.338771000 +0100
@@ -1,6 +1,6 @@
 matplotlib>=3,<4
 numpy<1.22; python_version=='3.7'
-numpy>=1.22,<=1.25; python_version>='3.8'
+numpy>=1.19,<=1.25; python_version>='3.8'
 sobol_seq==0.2.0
 yatiml>=0.10.0,<0.11
 ymmsl>=0.13,<0.14
diff -Nru muscle3-0.6.0-orig/setup.py muscle3-0.6.0/setup.py
--- muscle3-0.6.0-orig/setup.py	2023-01-17 23:11:43.000000000 +0100
+++ muscle3-0.6.0/setup.py	2023-01-18 11:53:23.568013000 +0100
@@ -54,7 +54,7 @@
         'msgpack>=1,<2',
         'netifaces==0.11.0',
         "numpy<1.22; python_version=='3.7'",
-        "numpy>=1.22,<=1.25; python_version>='3.8'",
+        "numpy>=1.19,<=1.25; python_version>='3.8'",
         'qcg-pilotjob==0.13.1',
         'typing_extensions<4',
         'ymmsl>=0.13.0,<0.14'          # Also in CI, update there as well
