diff -Nru yatiml-0.10.0-orig/setup.py yatiml-0.10.0/setup.py
--- yatiml-0.10.0-orig/setup.py	2022-12-19 15:16:27.000000000 +0100
+++ yatiml-0.10.0/setup.py	2024-01-11 11:21:08.019950000 +0100
@@ -33,7 +33,7 @@
         'Programming Language :: Python :: 3.11'
     ],
     install_requires=[
-        'ruamel.yaml>=0.15.71,<=0.17.21,!=0.17.5',
+        'ruamel.yaml',
         'typing_extensions'
     ]
 )
