# Use more recent version of ruamel.yaml
# S.D.Pinches
diff -Nru yatiml-0.9.0-orig/setup.py yatiml-0.9.0/setup.py
--- yatiml-0.9.0-orig/setup.py	2022-03-30 11:43:36.000000000 +0200
+++ yatiml-0.9.0/setup.py	2022-08-31 16:49:26.849791000 +0200
@@ -33,7 +33,7 @@
         'Programming Language :: Python :: 3.10'
     ],
     install_requires=[
-        'ruamel.yaml>=0.15.71,<0.17',
+        'ruamel.yaml>=0.15.71,<=0.17.21',
         'typing_extensions'
     ]
 )
