diff -Nru ymmsl-0.13.0-orig/setup.py ymmsl-0.13.0/setup.py
--- ymmsl-0.13.0-orig/setup.py	2023-01-17 22:00:21.000000000 +0100
+++ ymmsl-0.13.0/setup.py	2024-01-11 11:32:59.149099000 +0100
@@ -37,7 +37,7 @@
     python_requires='>=3.7, <4',
     test_suite='tests',
     install_requires=[
-        'ruamel.yaml<=0.17.21,!=0.17.5',
+        'ruamel.yaml',
         'yatiml>=0.10.0,<0.11.0'
     ],
 )
