# Use more recent version of ruamel.yaml
# S.D.Pinches
diff -Nru ymmsl-0.12.0-orig/setup.py ymmsl-0.12.0/setup.py
--- ymmsl-0.12.0-orig/setup.py	2022-07-15 15:15:14.000000000 +0200
+++ ymmsl-0.12.0/setup.py	2022-08-31 16:39:03.225926000 +0200
@@ -37,7 +37,7 @@
     ],
     test_suite='tests',
     install_requires=[
-        'ruamel.yaml<=0.16.12',
+        'ruamel.yaml<=0.17.21',
         'yatiml>=0.9.0,<0.10.0'
     ],
 )
