diff -ru ray-v3.0.0.orig/code/GeneOntology/GeneOntology.cpp ray-v3.0.0/code/GeneOntology/GeneOntology.cpp
--- ray-v3.0.0.orig/code/GeneOntology/GeneOntology.cpp	2018-11-19 21:01:31.611254498 +0000
+++ ray-v3.0.0/code/GeneOntology/GeneOntology.cpp	2018-11-19 21:02:07.383922963 +0000
@@ -218,7 +218,7 @@
 		cout<<m_annotationFileName<<endl;
 
 		ifstream test1(m_ontologyFileName.c_str());
-		bool test1Result=test1;
+		bool test1Result=test1.good();
 		test1.close();
 
 		if(!test1Result){
@@ -227,7 +227,7 @@
 		}
 
 		ifstream test2(m_annotationFileName.c_str());
-		bool test2Result=test2;
+		bool test2Result=test2.good();
 		test2.close();
 
 		if(!test2Result){
diff -ru ray-v3.0.0.orig/code/Mock/Parameters.cpp ray-v3.0.0/code/Mock/Parameters.cpp
--- ray-v3.0.0.orig/code/Mock/Parameters.cpp	2018-11-19 21:01:31.637254257 +0000
+++ ray-v3.0.0/code/Mock/Parameters.cpp	2018-11-19 21:06:21.152571082 +0000
@@ -1069,7 +1069,7 @@
 	bool hasCommandFile=false;
 	if(argc==2){
 		ifstream f(argv[1]);
-		hasCommandFile=f;
+		hasCommandFile=f.good();
 		f.close();
 	}
 
diff -ru ray-v3.0.0.orig/code/SequencesLoader/Loader.cpp ray-v3.0.0/code/SequencesLoader/Loader.cpp
--- ray-v3.0.0.orig/code/SequencesLoader/Loader.cpp	2018-11-19 21:01:31.603254572 +0000
+++ ray-v3.0.0/code/SequencesLoader/Loader.cpp	2018-11-19 21:09:08.223022704 +0000
@@ -47,7 +47,7 @@
 
 int Loader::load(string file,bool isGenome){
 	ifstream f(file.c_str());
-	bool exists=f;
+	bool exists=f.good();
 	f.close();
 	if(!exists){
 		cout<<"Ray: cannot access '"<<file<<"': No such file or directory"<<endl;
