--- ./models/atm/cam/src/physics/rrtmg/ext/rrtmg_lw/rrtmg_lw_rtrnmr.f90.orig	2017-10-30 18:40:20.452586848 +0000
+++ ./models/atm/cam/src/physics/rrtmg/ext/rrtmg_lw/rrtmg_lw_rtrnmr.f90	2017-10-30 18:39:20.118138507 +0000
@@ -245,6 +245,8 @@
            secdiff(ibnd) = 1.66_r8
          else
            secdiff(ibnd) = a0(ibnd) + a1(ibnd)*exp(a2(ibnd)*pwvcm)
+           if (secdiff(ibnd) .gt. 1.80_r8) secdiff(ibnd) = 1.80_r8
+           if (secdiff(ibnd) .lt. 1.50_r8) secdiff(ibnd) = 1.50_r8
          endif
       enddo
       if (pwvcm.lt.1.0) secdiff(6) = 1.80_r8
