<ctime> included
Author: J. Sassmannshausen (Imperial College London/UK)
diff --git a/openbabel-openbabel-3-1-1.orig/include/openbabel/obutil.h b/openbabel-openbabel-3-1-1/include/openbabel/obutil.h
index 233ab0b..4ff35f0 100644
--- a/openbabel-openbabel-3-1-1.orig/include/openbabel/obutil.h
+++ b/openbabel-openbabel-3-1-1/include/openbabel/obutil.h
@@ -37,6 +37,7 @@ GNU General Public License for more details.
 #endif
 
 #include <math.h>
+#include <ctime>
 
 #ifndef M_PI
 #define M_PI 3.14159265358979323846
