Fix python bindings installdir not respecting CMAKE_INSTALL_PREFIX
Author: Davide Grassano - CECAM EPFL (@Crivella)
diff --git a/scripts/CMakeLists.txt b/scripts/CMakeLists.txt
index bf5913c59..3b4182db9 100644
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -132,7 +132,7 @@ if (DO_PYTHON_BINDINGS)
 	  if( "${PYTHON_INSTDIR}" STREQUAL "" )
             execute_process(
               COMMAND
-             ${Python_EXECUTABLE} -c "import sysconfig; print(sysconfig.get_path('platlib', vars={'base': '${CMAKE_INSTALL_PREFIX}'}))"
+             ${Python_EXECUTABLE} -c "import sysconfig; print(sysconfig.get_path('platlib', vars={'base': '${CMAKE_INSTALL_PREFIX}', 'platbase': '${CMAKE_INSTALL_PREFIX}' }))"
               OUTPUT_VARIABLE PYTHON_INSTDIR
               OUTPUT_STRIP_TRAILING_WHITESPACE
             )
