diff -ru amber20_src.orig/AmberTools/src/quick/quick-cmake/QUICKCudaConfig.cmake amber20_src/AmberTools/src/quick/quick-cmake/QUICKCudaConfig.cmake
--- amber20_src.orig/AmberTools/src/quick/quick-cmake/QUICKCudaConfig.cmake	2021-11-12 18:38:33.444933045 +0000
+++ amber20_src/AmberTools/src/quick/quick-cmake/QUICKCudaConfig.cmake	2021-11-12 18:43:15.381662155 +0000
@@ -69,7 +69,7 @@
             list(APPEND CUDA_NVCC_FLAGS ${SM35FLAGS} ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS})
             set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
 
-	elseif((${CUDA_VERSION} VERSION_GREATER_EQUAL 11.1) AND (${CUDA_VERSION} VERSION_LESS_EQUAL 11.2))
+	elseif((${CUDA_VERSION} VERSION_GREATER_EQUAL 11.1) AND (${CUDA_VERSION} VERSION_LESS_EQUAL 11.4))
 	    message(STATUS "Configuring QUICK for SM3.5, SM5.0, SM6.0, SM7.0, SM7.5, SM8.0 and SM8.6")
             list(APPEND CUDA_NVCC_FLAGS ${SM35FLAGS} ${SM50FLAGS} ${SM60FLAGS} ${SM70FLAGS} ${SM75FLAGS} ${SM80FLAGS} ${SM86FLAGS})
             set(DISABLE_OPTIMIZER_CONSTANTS TRUE)
