Fix incorrect parameter to dvout

Åke Sandgren, 20220623
diff --git a/AmberTools/src/arpack/dnaitr.f b/AmberTools/src/arpack/dnaitr.f
index 64b008617f..5d3c268a95 100644
--- a/AmberTools/src/arpack/dnaitr.f
+++ b/AmberTools/src/arpack/dnaitr.f
@@ -373,7 +373,7 @@ c
          if (msglvl .gt. 1) then
             call ivout (logfil, 1, [j], ndigit, 
      &                  '_naitr: generating Arnoldi vector number')
-            call dvout (logfil, 1, rnorm, ndigit, 
+            call dvout (logfil, 1, [rnorm], ndigit, 
      &                  '_naitr: B-norm of the current residual is')
          end if
 c 
diff --git a/AmberTools/src/arpack/dsaitr.f b/AmberTools/src/arpack/dsaitr.f
index f725e23de9..5a29f153e6 100644
--- a/AmberTools/src/arpack/dsaitr.f
+++ b/AmberTools/src/arpack/dsaitr.f
@@ -366,7 +366,7 @@ c
          if (msglvl .gt. 2) then
             call ivout (logfil, 1, [j], ndigit, 
      &                  '_saitr: generating Arnoldi vector no.')
-            call dvout (logfil, 1, rnorm, ndigit, 
+            call dvout (logfil, 1, [rnorm], ndigit, 
      &                  '_saitr: B-norm of the current residual =')
          end if
 c 
