Fix potential use before init.

Åke Sandgren, 20220623
diff --git a/AmberTools/src/nmr_aux/prepare_input/makeDIST_RST.c b/AmberTools/src/nmr_aux/prepare_input/makeDIST_RST.c
index a455a8bc15..3d6f5baec6 100644
--- a/AmberTools/src/nmr_aux/prepare_input/makeDIST_RST.c
+++ b/AmberTools/src/nmr_aux/prepare_input/makeDIST_RST.c
@@ -714,6 +714,7 @@ static int read_maplib( FILE *fp )
 	printf( "%s: begin;\n", module );
 #endif
 	nm=0;
+        npseudo=0;
  	while ( getline ) {
 		lineno++;
 
