--- cmake/PMEMDCompilerFlags.cmake.orig	2025-05-27 12:00:31.493682843 +0000
+++ cmake/PMEMDCompilerFlags.cmake	2025-05-27 12:42:38.132237970 +0000
@@ -281,9 +281,6 @@
 	endif()
 
 	option(NCCL "Use NCCL for inter-GPU communications." FALSE)
-	if(NCCL AND NOT nccl_ENABLED)
-		message(FATAL_ERROR "NCCL is selected for inter-GPU communications but was not found.")
-	endif()
 
 	if(NCCL)
 		list(APPEND PMEMD_CUDA_DEFINES -DNCCL)

