diff -ru amber24_src.orig/test/test_amber_clean.sh amber24_src/test/test_amber_clean.sh 
--- amber24_src.orig/test/test_amber_clean.sh	2024-04-29 14:47:49
+++ amber24_src/test/test_amber_clean.sh	2024-10-01 19:22:14
@@ -1273,7 +1273,9 @@
 echo $delete_list | xargs /bin/rm -f
 
 # Remove cuda remd test output
-cd cuda/remd && make clean && cd ../..
+if [ -d "cuda/remd" ]; then
+    cd cuda/remd && make clean && cd ../..
+fi
 
 # Remove hip remd test output
 cd hip/remd && make clean && cd ../..
