Fix incorrect parameter to mexit

Åke Sandgren, 20220623
diff --git a/AmberTools/src/sander/fastwt.F90 b/AmberTools/src/sander/fastwt.F90
index 9863c12e27..f7e2bf1df4 100644
--- a/AmberTools/src/sander/fastwt.F90
+++ b/AmberTools/src/sander/fastwt.F90
@@ -1058,13 +1058,13 @@ subroutine getwds(igraph    ,nres      ,ipres     ,lbres     , &
 #ifndef API
          write(iout,1001)
 #endif
-         call mexit(1)
+         call mexit(6,1)
       end if
    else if (abs(roh1-roh2) > small) then
 #ifndef API
       write(iout,1002)
 #endif
-      call mexit(1)
+      call mexit(6,1)
    else
       roh = roh1
    end if
