--- AmberTools/src/quick/cmake/CudaConfig.cmake.orig	2025-05-23 16:30:29.891312953 +0000
+++ AmberTools/src/quick/cmake/CudaConfig.cmake	2025-05-24 14:50:21.626994417 +0000
@@ -127,15 +127,6 @@
 		) )
 			message(STATUS "Checking CUDA and GNU versions -- compatibility unknown")
 			message(STATUS "    See https://stackoverflow.com/questions/6622454/cuda-incompatible-with-my-gcc-version")
-		elseif ( "${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" )
-			message(STATUS "")
-			message("************************************************************")
-			message("Error: Incompatible CUDA and GNU versions!")
-			message("  GNU version is ${CMAKE_CXX_COMPILER_VERSION}.")
-			message("  See https://stackoverflow.com/questions/6622454/cuda-incompatible-with-my-gcc-version")
-			message("************************************************************")
-			message(STATUS "")
-			message(FATAL_ERROR)
 		else()
 			message(STATUS "Checking CUDA and compiler versions -- compatibility unknown")
 		endif()
