--- cmake/CudaConfig.cmake.orig	2025-05-24 14:47:10.645996163 +0000
+++ cmake/CudaConfig.cmake	2025-05-27 13:13:17.707398807 +0000
@@ -130,16 +130,6 @@
 		) )
 			message(STATUS "Checking CUDA and GNU versions -- compatibility unknown")
 			message(STATUS "    See https://stackoverflow.com/questions/6622454/cuda-incompatible-with-my-gcc-version")
-		elseif ( "${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" )
-			message(STATUS "")
-			message("************************************************************")
-			message("Error: Incompatible CUDA and GNU versions")
-			message(" ${CMAKE_CXX_COMPILER_VERSION}")
-			message(" ${CMAKE_CXX_COMPILER_VERSION_MAJOR}")
-			message("See https://stackoverflow.com/questions/6622454/cuda-incompatible-with-my-gcc-version")
-			message("************************************************************")
-			message(STATUS "")
-			message(FATAL_ERROR)
 		else()
 			message(STATUS "Checking CUDA and compiler versions -- compatibility unknown")
 		endif()
