--- AmberTools/src/pysander/CMakeLists.txt.orig	2025-05-20 11:43:47.157120070 +0000
+++ AmberTools/src/pysander/CMakeLists.txt	2025-05-23 13:28:00.218517952 +0000
@@ -43,11 +43,11 @@
 install(CODE "
 	${FIX_BACKSLASHES_CMD}
 	execute_process(
-    	COMMAND \"${CMAKE_COMMAND}\" -E env --unset=AMBERHOME \"${PYTHON_EXECUTABLE}\" 
-    	\"${CMAKE_CURRENT_SOURCE_DIR}/setup.py\"
-	    -I${AMBERTOOLS_INC_DIR}
-	    -I${CMAKE_SOURCE_DIR}/AmberTools/src/sander
-        -L${CMAKE_INSTALL_PREFIX}/lib
-        build -b ${BUILD_DIR}
-    	 install -f ${PYTHON_PREFIX_ARG} --single-version-externally-managed --root /
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})" COMPONENT Python)
\ No newline at end of file
+        COMMAND \"${CMAKE_COMMAND}\" -E env --unset=AMBERHOME \"${PYTHON_EXECUTABLE}\" 
+        -m pip install 
+        --global-option=-I${AMBERTOOLS_INC_DIR} 
+        --global-option=-I${CMAKE_SOURCE_DIR}/AmberTools/src/sander 
+        --global-option=-L${CMAKE_INSTALL_PREFIX}/lib 
+        ${PYTHON_PREFIX_ARG} 
+        ./
+       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})" COMPONENT Python)
