--- AmberTools/src/quick/quick-cmake/QUICKCudaConfig.cmake.orig	2025-05-27 13:24:24.707721389 +0000
+++ AmberTools/src/quick/quick-cmake/QUICKCudaConfig.cmake	2025-05-27 13:25:11.078952380 +0000
@@ -254,15 +254,6 @@
     ) )
         message(STATUS "Checking CUDA and GNU versions -- compatibility unknown")
         message(STATUS "    See https://stackoverflow.com/questions/6622454/cuda-incompatible-with-my-gcc-version")
-    elseif ( "${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" )
-        message(STATUS "")
-        message("************************************************************")
-        message("Error: Incompatible CUDA and GNU versions")
-        message("  GNU version is ${CMAKE_CXX_COMPILER_VERSION}.")
-        message("See https://stackoverflow.com/questions/6622454/cuda-incompatible-with-my-gcc-version")
-        message("************************************************************")
-        message(STATUS "")
-        message(FATAL_ERROR)
     else()
         message(STATUS "Checking CUDA and compiler versions -- compatibility unknown")
     endif()

