Fix numpy syntax to be compatible with v2
--- AmberTools/src/pysander/sander/__init__.py.orig	2025-08-08 11:21:39.084040000 +0200
+++ AmberTools/src/pysander/sander/__init__.py	2025-08-08 11:22:44.758277000 +0200
@@ -83,7 +83,7 @@
     # numpy arrays to solve this quickly, but in cases where the coordinates
     # are given as a list (or tuple) of Vec3's (or tuples), this requires
     # separate handling
-    positions = _np.array(positions, copy=False, subok=True)
+    positions = _np.asarray(positions)
     positions = positions.flatten()
     natom = _pys.natom()
     if len(positions) != natom * 3:
@@ -260,7 +260,7 @@
                 box = rst.box
 
         # Convert from numpy arrays to regular arrays
-        coordinates = _np.array(coordinates, copy=False, subok=True)
+        coordinates = _np.asarray(coordinates)
         coordinates = coordinates.flatten().tolist()
         if box is None or box is False:
             box = _np.zeros(6)
