--- AmberTools/src/Makefile.orig	2025-05-01 13:06:08.904270721 +0000
+++ AmberTools/src/Makefile	2025-05-01 13:27:10.023841965 +0000
@@ -6,7 +6,7 @@
 MIC_TOKEN=mmic
 MIC2_TOKEN=DMIC2
 MIC_OFFLOAD_TOKEN=DMIC_offload
-PYINSTALL=PYTHONPATH=$(PYTHONPATH) $(PYTHON) setup.py install -f $(PYTHON_INSTALL)
+PYINSTALL=PYTHONPATH=$(PYTHONPATH) $(PYTHON) -m pip install --force-reinstall --no-deps $(PYTHON_INSTALL) .
 
 include $(CONFIG_FILE)
 
@@ -626,6 +626,6 @@
 	# Install development version so you can edit file in source folder, run code without install again.
 	# Note: if you install develop version, do not use "pip uninstall", you must use
 	# make pdb4amber.develop_uninstall
-	(cd pdb4amber && PYTHONPATH=$(PYTHONPATH) $(PYTHON) setup.py develop $(PYTHON_INSTALL))
+	(cd pdb4amber && PYTHONPATH=$(PYTHONPATH) $(PYTHON) -m pip -e $(PYTHON_INSTALL) .)
 pdb4amber.develop_uninstall:
-	(cd pdb4amber && PYTHONPATH=$(PYTHONPATH) $(PYTHON) setup.py develop $(PYTHON_INSTALL) --uninstall)
+	(cd pdb4amber && PYTHONPATH=$(PYTHONPATH) $(PYTHON) -m pip uninstall -y pdb4amber.develop $(PYTHON_INSTALL)) 
