Fix DGESVD workspace query 

Åke Sandgren, 20220623
diff --git a/AmberTools/src/sqm/qm2_scf.F90 b/AmberTools/src/sqm/qm2_scf.F90
index 6185d93dba..93eb7f60e8 100644
--- a/AmberTools/src/sqm/qm2_scf.F90
+++ b/AmberTools/src/sqm/qm2_scf.F90
@@ -2254,7 +2254,7 @@ SUBROUTINE SvdInvert_SymMat(n,a,ainv,thresh) ! THRESH,ERR
   _REAL_ :: S(n)
   _REAL_ :: U(n,n)
   _REAL_ :: VT(n,n)
-  _REAL_ :: twork
+  _REAL_ :: twork(1)
   _REAL_,ALLOCATABLE :: WORK(:)
   INTEGER :: LWORK
 
@@ -2269,7 +2269,7 @@ SUBROUTINE SvdInvert_SymMat(n,a,ainv,thresh) ! THRESH,ERR
   LWORK = -1
   CALL DGESVD("A","A",n,n, ainv,n, S,U,n, VT,n, twork,LWORK,my_err)
   
-  LWORK = NINT(twork)
+  LWORK = NINT(twork(1))
   ALLOCATE( WORK(LWORK) )
   WORK = 0.d0
   
