--- cmake/PythonBuildConfig.cmake.orig	2025-05-01 11:17:04.752478129 +0000
+++ cmake/PythonBuildConfig.cmake	2025-05-01 11:19:49.820192096 +0000
@@ -175,7 +175,7 @@
         execute_process(
 		    COMMAND \"${CMAKE_COMMAND}\" -E env
 		     \"${PYTHON_EXECUTABLE}\"
-		    ./setup.py install -f ${PYTHON_PREFIX_ARG} --single-version-externally-managed --root /
+		    -m pip install --force-reinstall --no-deps --root / ${PYTHON_PREFIX_ARG} .
 		    ${SCRIPT_ARGS_SPC}
 		    WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")"
 		    COMPONENT Python)
