--- ./tcltk/tclreadline/configure.orig	2018-01-02 20:12:29.985657688 +0000
+++ ./tcltk/tclreadline/configure	2018-01-02 20:13:33.619091827 +0000
@@ -2124,7 +2124,7 @@
 echo $ac_n "checking where to look for readline.h""... $ac_c" 1>&6
 echo "configure:2126: checking where to look for readline.h" >&5
 READLINE_INCLUDE_DIR=""
-for dir in $rl_includes /usr/local/include/readline /usr/local/include /usr/include/readline /usr/include ; do
+for dir in $rl_includes ${EBROOTGENTOO}/include/readline /usr/local/include/readline /usr/local/include /usr/include/readline /usr/include ; do
     if test -r $dir/readline.h; then
         READLINE_INCLUDE_DIR=$dir
         break
@@ -2133,7 +2133,7 @@
 
 if test -z "$READLINE_INCLUDE_DIR"; then
     { echo "configure: error: 
-  Can't find readline.h.
+  Cannot find readline.h.
   Use --with-readline-includes to specify the directory
   containing readline.h on your system." 1>&2; exit 1; }
 else
