Create custom Makefile

author: Pavel Grochal (INUITS)
--- Makefile.orig	2020-03-30 17:12:39.099451802 +0200
+++ Makefile	2020-03-30 17:59:38.570459472 +0200
@@ -0,0 +1,23 @@
+CC ?= gcc
+CXX ?= g++
+CFLAGS ?= -g -O2
+CXXFLAGS ?= -g -O2
+
+SRC_DIR = src
+BIN_DIR = bin
+
+cpp_executables = ATCGmapMerge CGmapSelectByRegion CGmapMethInBed CGmapMethInFragReg
+c_executables = CGmapFromBAM CGmapToCGbz CGbzToCGmap ATCGmapToATCGbz ATCGbzToATCGmap CGbzFetchRegion ATCGbzFetchRegion
+
+all: mk_bin_dir $(cpp_executables) $(c_executables)
+
+mk_bin_dir:
+	mkdir -p $(BIN_DIR)
+
+$(c_executables):
+	@echo "Building $@"
+	$(CC) $(CFLAGS) -o $(BIN_DIR)/$@ $(SRC_DIR)/$@.c $(LDFLAGS) -lz -lbam -lpthread
+
+$(cpp_executables):
+	@echo "Building $@"
+	$(CXX) $(CXXFLAGS) -o $(BIN_DIR)/$@ $(SRC_DIR)/$@.cpp $(LDFLAGS) -lz
