--- source/makefile.orig	2017-09-19 19:23:33.887557237 +0000
+++ source/makefile	2017-09-19 19:30:41.217708433 +0000
@@ -7,7 +7,7 @@
 CODE3 = statAl.cpp
 NAME3 = statal
 
-CC    = g++
+#CC    = g++
 FLAGS = -Wall -O2
 LIBM  = -lm
 DUGB  =
@@ -17,34 +17,34 @@
 preview:		alignment.o statisticsGaps.o utils.o similarityMatrix.o statisticsConservation.o sequencesMatrix.o compareFiles.o
 
 trimal:			preview
-			$(CC) $(FLAGS) $(DUGB) -o $(NAME) $(CODE) $(LIBM) alignment.o statisticsGaps.o utils.o similarityMatrix.o statisticsConservation.o sequencesMatrix.o compareFiles.o
+			$(CXX) $(FLAGS) $(DUGB) -o $(NAME) $(CODE) $(LIBM) alignment.o statisticsGaps.o utils.o similarityMatrix.o statisticsConservation.o sequencesMatrix.o compareFiles.o
 
 readal:			preview
-			$(CC) $(FLAGS) $(DUGB) -o $(NAME2) $(CODE2) $(LIBM) alignment.o statisticsGaps.o utils.o similarityMatrix.o statisticsConservation.o sequencesMatrix.o compareFiles.o
+			$(CXX) $(FLAGS) $(DUGB) -o $(NAME2) $(CODE2) $(LIBM) alignment.o statisticsGaps.o utils.o similarityMatrix.o statisticsConservation.o sequencesMatrix.o compareFiles.o
 
 statal:			preview
-			$(CC) $(FLAGS) $(DUGB) -o $(NAME3) $(CODE3) $(LIBM) alignment.o statisticsGaps.o utils.o similarityMatrix.o statisticsConservation.o sequencesMatrix.o compareFiles.o
+			$(CXX) $(FLAGS) $(DUGB) -o $(NAME3) $(CODE3) $(LIBM) alignment.o statisticsGaps.o utils.o similarityMatrix.o statisticsConservation.o sequencesMatrix.o compareFiles.o
 
 alignment.o:		alignment.cpp rwAlignment.cpp autAlignment.cpp alignment.h values.h defines.h
-			$(CC) $(FLAGS) $(DUGB) -c alignment.cpp rwAlignment.cpp autAlignment.cpp
+			$(CXX) $(FLAGS) $(DUGB) -c alignment.cpp rwAlignment.cpp autAlignment.cpp
 
 statisticsGaps.o:	statisticsGaps.cpp statisticsGaps.h
-			$(CC) $(FLAGS) $(DUGB) -c statisticsGaps.cpp
+			$(CXX) $(FLAGS) $(DUGB) -c statisticsGaps.cpp
 
 sequencesMatrix.o:	sequencesMatrix.cpp sequencesMatrix.h
-			$(CC) $(FLAGS) $(DUGB) -c sequencesMatrix.cpp
+			$(CXX) $(FLAGS) $(DUGB) -c sequencesMatrix.cpp
 
 statisticsConservation.o: statisticsConservation.cpp statisticsConservation.h
-			$(CC) $(FLAGS) $(DUGB) -c statisticsConservation.cpp
+			$(CXX) $(FLAGS) $(DUGB) -c statisticsConservation.cpp
 
 similarityMatrix.o:	similarityMatrix.cpp similarityMatrix.h values.h
-			$(CC) $(FLAGS) $(DUGB) -c similarityMatrix.cpp
+			$(CXX) $(FLAGS) $(DUGB) -c similarityMatrix.cpp
 
 utils.o:		utils.cpp utils.h values.h  defines.h
-			$(CC) $(FLAGS) $(DUGB) -c utils.cpp
+			$(CXX) $(FLAGS) $(DUGB) -c utils.cpp
 
 compareFiles.o:		compareFiles.cpp compareFiles.h
-			$(CC) $(FLAGS) $(DUGB) -c compareFiles.cpp
+			$(CXX) $(FLAGS) $(DUGB) -c compareFiles.cpp
 
 clean:
 			-rm -f $(NAME) $(NAME2)  $(NAME3) *.o *~
