diff -ru libbeef-0.1.1.orig/configure libbeef-0.1.1/configure
--- libbeef-0.1.1.orig/configure	2021-03-23 13:31:23.549664440 +0000
+++ libbeef-0.1.1/configure	2021-03-23 13:36:10.161049891 +0000
@@ -4393,7 +4393,7 @@
 else
 
   ax_check_save_flags=$CFLAGS
-  CFLAGS="$CFLAGS  -no-prec-div -xHost"
+  CFLAGS="$CFLAGS  -no-prec-div "
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -4416,7 +4416,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ax_cv_check_cflags___no_prec_div__xHost" >&5
 $as_echo "$ax_cv_check_cflags___no_prec_div__xHost" >&6; }
 if test x"$ax_cv_check_cflags___no_prec_div__xHost" = xyes; then :
-  CFLAGS="$CFLAGS -no-prec-div -xHost"
+  CFLAGS="$CFLAGS -no-prec-div "
 else
   :
 fi
