--- vampire-6.0/makefile.orig	2024-04-16 09:59:14.730461427 +0200
+++ vampire-6.0/makefile	2024-04-16 14:12:23.399037000 +0200
@@ -321,17 +321,14 @@
 	$(MAKE) -C util/vdc/
 
 install:
-	echo "Preparing installation package"
-	rm -rf vampire.pkg
-	mkdir vampire.pkg
-	mkdir vampire.pkg/bin
-	cp vampire-* vampire.pkg/bin/
-	cp util/vdc/vdc vampire.pkg/bin/
-	mkdir vampire.pkg/examples
-	cp input vampire.pkg/examples/
-	cp Co.mat vampire.pkg/examples/
-	sudo mv -f vampire.pkg /opt/vampire
-	sudo echo "/opt/vampire/bin" > /etc/paths.d/vampire_path
+	mkdir $(PREFIX)/bin
+	install -m 0755 vampire-serial $(PREFIX)/bin/ 
+	install -m 0755 vampire-parallel $(PREFIX)/bin/ 
+	install -m 0755 util/vdc/vdc $(PREFIX)/bin/
+
+	mkdir $(PREFIX)/share
+	install -m 0644 input $(PREFIX)/share/
+	install -m 0644 Co.mat $(PREFIX)/share/
 
 uninstall:
 	rm -rf /opt/vampire
