From 0db42e859462a3b04c14a0155253b0ca4b9b64c9 Mon Sep 17 00:00:00 2001
From: Viktor Rehnberg <viktor.rehnberg@gmail.com>
Date: Tue, 3 Dec 2024 09:54:22 +0000
Subject: [PATCH] Migrate to Bazel v6

---
 bazel/BUILD | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bazel/BUILD b/bazel/BUILD
index 05458d5..7075c8f 100644
--- a/bazel/BUILD
+++ b/bazel/BUILD
@@ -24,15 +24,15 @@ licenses(["notice"])
 
 config_setting(
     name = "linux",
-    constraint_values = ["@bazel_tools//platforms:linux"],
+    constraint_values = ["@platforms//os:linux"],
 )
 
 config_setting(
     name = "apple",
-    constraint_values = ["@bazel_tools//platforms:osx"],
+    constraint_values = ["@platforms//os:osx"],
 )
 
 config_setting(
     name = "windows",
-    constraint_values = ["@bazel_tools//platforms:windows"],
+    constraint_values = ["@platforms//os:windows"],
 )
-- 
2.39.3

