diff -ru lua-5.1.4.5.orig/lua/src/Makefile.in lua-5.1.4.5/lua/src/Makefile.in
--- lua-5.1.4.5.orig/lua/src/Makefile.in    2011-02-18 19:49:01.000000000 +0100
+++ lua-5.1.4.5/lua/src/Makefile.in 2013-06-02 13:22:28.601496000 +0200
@@ -98,7 +98,7 @@
	$(MAKE) all MYCFLAGS=
 
 linux:
-	$(MAKE) all MYCFLAGS="$(MYCFLAGS) -DLUA_USE_LINUX" MYLIBS="-Wl,-E -ldl @LIBS@ -lncurses"
+	$(MAKE) all MYCFLAGS="$(MYCFLAGS) -DLUA_USE_LINUX" MYLIBS="-Wl,-E -ldl -Wl,-Bstatic @LIBS@ -lncurses -Wl,-Bdynamic"
 
 macosx:
 #	$(MAKE) all MYCFLAGS=-DLUA_USE_MACOSX
