diff -ru libpng-1.6.29.original/libpng.pc.in libpng-1.6.29.patched/libpng.pc.in
--- libpng-1.6.29.original/libpng.pc.in	2017-03-16 13:21:02.000000000 +0100
+++ libpng-1.6.29.patched/libpng.pc.in	2017-06-29 11:04:17.000000000 +0200
@@ -6,6 +6,7 @@
 Name: libpng
 Description: Loads and saves PNG files
 Version: @PNGLIB_VERSION@
+Requires: zlib
 Libs: -L${libdir} -lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@
 Libs.private: @LIBS@
 Cflags: -I${includedir}
