# Replace zmq with pyzmq
# Author: maxim-masterov (SURF)
diff -Nru QCG-PilotJob-0.12.3.orig/components/core/setup.py QCG-PilotJob-0.12.3/components/core/setup.py
--- QCG-PilotJob-0.12.3.orig/components/core/setup.py	2021-12-02 13:33:02.564287000 +0100
+++ QCG-PilotJob-0.12.3/components/core/setup.py	2021-12-02 15:00:08.997343000 +0100
@@ -27,7 +27,7 @@
 	long_description_content_type="text/markdown",
 
 	install_requires=[
-		"zmq",
+		"pyzmq",
 		"click",
                 "psutil",
 		],
