diff -ruN 1.1.7.tgz.orig/src/View.cpp 1.1.7.tgz/src/View.cpp
--- 1.1.7.tgz.orig/src/View.cpp	2020-06-29 01:19:43.000000000 -0700
+++ 1.1.7.tgz/src/View.cpp	2022-06-06 18:05:10.650577000 -0700
@@ -177,7 +177,7 @@
 
 			// get file size
 			{
-				std::ios::streampos origin = m_stream->tellg() ;
+				std::streampos origin = m_stream->tellg() ;
 				m_stream->seekg( 0, std::ios::end ) ;
 				m_file_metadata.size = m_stream->tellg() - origin ;
 				m_stream->seekg( 0, std::ios::beg ) ;
