diff -Nru rstudio-1.1.447.orig/src/cpp/core/system/PosixSystem.cpp rstudio-1.1.447/src/cpp/core/system/PosixSystem.cpp
--- rstudio-1.1.447.orig/src/cpp/core/system/PosixSystem.cpp	2018-04-17 20:08:08.000000000 +0000
+++ rstudio-1.1.447/src/cpp/core/system/PosixSystem.cpp	2018-04-24 20:55:40.624873664 +0000
@@ -1765,6 +1765,7 @@
 
       // setup environment
       core::system::Options env;
+      core::system::environment(&env);
       copyEnvironmentVar("PATH", &env);
       copyEnvironmentVar("MANPATH", &env);
       copyEnvironmentVar("LANG", &env);
