diff -ru BOLT-LMM_v2.3.4.orig/src/Makefile BOLT-LMM_v2.3.4/src/Makefile
--- BOLT-LMM_v2.3.4.orig/src/Makefile	2021-07-08 16:36:15.529381246 +0200
+++ BOLT-LMM_v2.3.4/src/Makefile	2021-07-13 09:55:00.700045096 +0200
@@ -1,19 +1,17 @@
+# The Makefile is orginally written for Intel. This patch file updates the 
+# Makefile arguments for the foss toolchain.
 ### modify these paths to local Boost and NLopt install directories
-BOOST_INSTALL_DIR = /home/pl88/boost_1_58_0/install
-NLOPT_INSTALL_DIR = /n/groups/price/poru/HSPH_SVN/src/BOLT-LMM/nlopt-2.4.2
-
-INTELROOT = /n/groups/price/poru/external_software/intel_mkl_2019u4
-MKLROOT = ${INTELROOT}/mkl
-ZLIB_STATIC_DIR = /n/groups/price/poru/external_software/zlib/zlib-1.2.11 # probably unnecessary on most systems
-LIBSTDCXX_STATIC_DIR = /n/groups/price/poru/external_software/libstdc++/usr/lib/gcc/x86_64-redhat-linux/4.8.5/
-GLIBC_STATIC_DIR = /home/pl88/glibc-static/usr/lib64
+BOOST_INSTALL_DIR = ${EBROOTBOOST}
+NLOPT_INSTALL_DIR = ${EBROOTNLOPT}
+ZLIB_STATIC_DIR = ${EBROOTZLIB}
+LIBSTDCXX_STATIC_DIR = /usr/lib64
+GLIBC_STATIC_DIR = /usr/lib64
 
 ifeq ($(strip ${linking}),)
 	linking = dynamic
 endif
 
-# CC = g++
-CC = icpc
+CC = g++
 
 ifeq (${debug},true)
 	CFLAGS += -g
@@ -68,7 +66,7 @@
 ifeq (${CC},g++)
 	CFLAGS += -fopenmp
 	LFLAGS += -fopenmp
-	LLAPACK = -llapack -lgfortran
+	LLAPACK = -lopenblas -llapack -lgfortran
 	ifeq (${linking},static)
 		LFLAGS += -static
 		LPATHS += -L${GLIBC_STATIC_DIR} -L${ZLIB_STATIC_DIR}
