diff -ru ParMGridGen-1.0.orig/MGridGen/IMlib/IMlib.h ParMGridGen-1.0/MGridGen/IMlib/IMlib.h
--- ParMGridGen-1.0.orig/MGridGen/IMlib/IMlib.h	2013-09-18 21:51:04.000000000 +0200
+++ ParMGridGen-1.0/MGridGen/IMlib/IMlib.h	2013-09-18 21:51:14.000000000 +0200
@@ -43,7 +43,7 @@
 #ifdef DMALLOC
 #include <dmalloc.h>
 #else
-#include <malloc.h>
+//#include <malloc.h>
 #endif
 
 /*************************************************************************
diff -ru ParMGridGen-1.0.orig/MGridGen/Lib/mgridgen.h ParMGridGen-1.0/MGridGen/Lib/mgridgen.h
--- ParMGridGen-1.0.orig/MGridGen/Lib/mgridgen.h	2013-09-18 21:52:14.000000000 +0200
+++ ParMGridGen-1.0/MGridGen/Lib/mgridgen.h	2013-09-18 21:52:00.000000000 +0200
@@ -26,7 +26,7 @@
 #ifdef DMALLOC
 #include <dmalloc.h>
 #else
-#include <malloc.h>
+//#include <malloc.h>
 #endif
 
 #include "defs.h"
diff -ru ParMGridGen-1.0.orig/ParMGridGen/IMParMetis-2.0/ParMETISLib/parmetis.h ParMGridGen-1.0/ParMGridGen/IMParMetis-2.0/ParMETISLib/parmetis.h
--- ParMGridGen-1.0.orig/ParMGridGen/IMParMetis-2.0/ParMETISLib/parmetis.h	2013-09-18 21:57:20.000000000 +0200
+++ ParMGridGen-1.0/ParMGridGen/IMParMetis-2.0/ParMETISLib/parmetis.h	2013-09-18 21:57:30.000000000 +0200
@@ -21,7 +21,7 @@
 #ifdef DMALLOC
 #include <dmalloc.h>
 #else
-#include <malloc.h>
+//#include <malloc.h>
 #endif
 
 #include "rename.h"
diff -ru ParMGridGen-1.0.orig/ParMGridGen/ParLib/parmgridgen.h ParMGridGen-1.0/ParMGridGen/ParLib/parmgridgen.h
--- ParMGridGen-1.0.orig/ParMGridGen/ParLib/parmgridgen.h	2013-09-18 21:57:57.000000000 +0200
+++ ParMGridGen-1.0/ParMGridGen/ParLib/parmgridgen.h	2013-09-18 21:58:06.000000000 +0200
@@ -21,7 +21,7 @@
 #ifdef DMALLOC
 #include <dmalloc.h>
 #else
-#include <malloc.h>
+//#include <malloc.h>
 #endif
 
 #include "IMlib.h"
