don't try to link binaries in /usr/local/bin
diff -Nru augustus.orig/Makefile augustus/Makefile
--- augustus.orig/Makefile	2017-07-11 15:00:33.000000000 +0200
+++ augustus/Makefile	2018-04-13 17:00:38.239949441 +0200
@@ -17,13 +17,13 @@
 install:
 	install -d $(INSTALLDIR)
 	cp -a config bin scripts $(INSTALLDIR)
-	ln -sf $(INSTALLDIR)/bin/augustus /usr/local/bin/augustus
-	ln -sf $(INSTALLDIR)/bin/etraining /usr/local/bin/etraining
-	ln -sf $(INSTALLDIR)/bin/prepareAlign /usr/local/bin/prepareAlign
-	ln -sf $(INSTALLDIR)/bin/fastBlockSearch /usr/local/bin/fastBlockSearch
-	ln -sf $(INSTALLDIR)/bin/load2db /usr/local/bin/load2db
-	ln -sf $(INSTALLDIR)/bin/getSeq /usr/local/bin/getSeq
-	ln -sf $(INSTALLDIR)/bin/espoca /usr/local/bin/espoca
+	#ln -sf $(INSTALLDIR)/bin/augustus /usr/local/bin/augustus
+	#ln -sf $(INSTALLDIR)/bin/etraining /usr/local/bin/etraining
+	#ln -sf $(INSTALLDIR)/bin/prepareAlign /usr/local/bin/prepareAlign
+	#ln -sf $(INSTALLDIR)/bin/fastBlockSearch /usr/local/bin/fastBlockSearch
+	#ln -sf $(INSTALLDIR)/bin/load2db /usr/local/bin/load2db
+	#ln -sf $(INSTALLDIR)/bin/getSeq /usr/local/bin/getSeq
+	#ln -sf $(INSTALLDIR)/bin/espoca /usr/local/bin/espoca
 
 # for internal purposes:
 release:
