diff -Nru augustus.orig/common.mk augustus/common.mk
--- augustus.orig/common.mk	2017-07-11 12:24:37.000000000 +0200
+++ augustus/common.mk	2018-04-12 11:21:52.085710047 +0200
@@ -7,7 +7,7 @@
 ZIPINPUT = true
 
 # uncomment this line to enable comparative gene finding (requires compiler which supports C++11 standard)
-# COMPGENEPRED = true
+COMPGENEPRED = true
 
 # uncomment this line when you need MySQL access to sequences (most users don't)
 #MYSQL = true

