Patch to fix license field in pyproject.toml
Author: C. Comito (FZJ JSC, Germany)
--- pyproject.toml.orig	2025-09-10 17:39:16
+++ pyproject.toml	2025-09-10 17:40:07
@@ -27,8 +27,7 @@
     { name = "Hakan Akdag", email = "hakan.akdag@dlr.de"},
     { name = "Berkant Palazoglu", email = "b.palazoglu@fz-juelich.de"}
 ]
-license = "MIT"
-license-files = ["LICENSE"]
+license = { file = "LICENSE" }
 keywords=["data", "analytics", "tensors", "distributed", "gpu"]
 classifiers=[
     "Development Status :: 5 - Production/Stable",
