diff -ur eccodes-1.3.2-orig/gribapi/bindings.py eccodes-1.3.2/gribapi/bindings.py
--- eccodes-1.3.2-orig/gribapi/bindings.py	2021-04-22 19:30:53.505518656 +0000
+++ eccodes-1.3.2/gribapi/bindings.py	2021-04-22 19:33:55.500607592 +0000
@@ -15,6 +15,8 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+from os import environ
+
 import logging
 import pkgutil
 
@@ -24,12 +26,7 @@
 
 LOG = logging.getLogger(__name__)
 
-try:
-    import ecmwflibs as findlibs
-except ImportError:
-    import findlibs
-
-library_path = findlibs.find("eccodes")
+library_path = environ["EBROOTECCODES"] + "/lib64/libeccodes.so"
 if library_path is None:
     raise RuntimeError("Cannot find the ecCodes library")
 
