From: https://src.fedoraproject.org/rpms/eccodes/raw/epel8/f/eccodes-soversion.patch
Add SOVERSION to CMakeLists library targets
--- eccodes-2.35.0-Source.unchanged/fortran/CMakeLists.txt	2024-04-11 12:13:21.000000000 +0200
+++ eccodes-2.35.0-Source/fortran/CMakeLists.txt	2024-05-04 16:18:42.732436699 +0200
@@ -46,7 +46,8 @@ if( HAVE_FORTRAN )
                          GENERATED       grib_f90.f90 eccodes_f90.f90
                          PUBLIC_INCLUDES $<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>
                                          $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
-                         PRIVATE_LIBS    eccodes ${ECCODES_PTHREADS_LIBRARIES} )
+                         PRIVATE_LIBS    eccodes ${ECCODES_PTHREADS_LIBRARIES}
+                         SOVERSION       ${ECCODES_SOVERSION_F90})
 
     if( DEFINED ecbuild_VERSION AND NOT ${ecbuild_VERSION} VERSION_LESS 3.1 )
         # Installed module directory is not in the PUBLIC INCLUDES!
--- eccodes-2.35.0-Source.unchanged/src/CMakeLists.txt	2024-04-11 12:13:21.000000000 +0200
+++ eccodes-2.35.0-Source/src/CMakeLists.txt	2024-05-04 16:18:42.733436675 +0200
@@ -416,6 +416,8 @@ ecbuild_add_library( TARGET    eccodes
                      PRIVATE_LIBS ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${AEC_LIBRARIES} ${PNG_LIBRARIES}
                      PUBLIC_LIBS  ${CMATH_LIBRARIES} ${THREADS_LIBRARIES}
                      TEMPLATES ${eccodes_extra_src_files}
+                     SOVERSION ${ECCODES_SOVERSION}
+
                      INSTALL_HEADERS_LIST
                               grib_api.h
                               eccodes.h
