diff -ru openvdb-10.0.1.orig/cmake/config/OpenVDBVersions.cmake openvdb-10.0.1/cmake/config/OpenVDBVersions.cmake
--- openvdb-10.0.1.orig/cmake/config/OpenVDBVersions.cmake	2023-01-30 18:27:38.543768510 +0000
+++ openvdb-10.0.1/cmake/config/OpenVDBVersions.cmake	2023-01-30 18:28:51.893465623 +0000
@@ -30,7 +30,7 @@
   # What's usually important is that the version of libstdc++ and glibc in use
   # matches. Compilers other than GCC should provide options to ensure this
   # targetting e.g. --gcc-toolchain and -fgnuc-version with Clang.
-  set(MINIMUM_GCC_VERSION 9.3.1)
+  set(MINIMUM_GCC_VERSION 9.3.0)
   set(MINIMUM_CLANG_VERSION 5.0)
   set(MINIMUM_ICC_VERSION 19)
   set(MINIMUM_MSVC_VERSION 19.10) # 1910 (Visual Studio 2017 15.0)
