# What: Add riscv64 as a 64 bit cpu
# Author: julianmorillo (BSC)
--- qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc.gni.orig	2025-02-03 16:03:22.374870122 +0100
+++ qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc.gni	2025-02-03 16:04:16.691723988 +0100
@@ -14,7 +14,7 @@
 if (is_nacl) {
   # NaCl targets don't use 64-bit pointers.
   has_64_bit_pointers = false
-} else if (current_cpu == "x64" || current_cpu == "arm64") {
+} else if (current_cpu == "x64" || current_cpu == "arm64" || current_cpu == "riscv64") {
   has_64_bit_pointers = true
 } else if (current_cpu == "x86" || current_cpu == "arm") {
   has_64_bit_pointers = false
