# What: Add riscv64 as a 64 bit cpu
# Author: julianmorillo (BSC)
--- qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc.gni.orig	2025-02-03 16:03:22.374870122 +0100
+++ qtwebengine/src/3rdparty/chromium/base/allocator/partition_allocator/partition_alloc.gni	2025-02-03 16:04:16.691723988 +0100
@@ -14,7 +14,7 @@
 if (is_nacl) {
   # NaCl targets don't use 64-bit pointers.
   has_64_bit_pointers = false
-} else if (current_cpu == "x64" || current_cpu == "arm64") {
+} else if (current_cpu == "x64" || current_cpu == "arm64" || current_cpu == "riscv64") {
   has_64_bit_pointers = true
 } else if (current_cpu == "x86" || current_cpu == "arm") {
   has_64_bit_pointers = false
# What: Allow current_cpu=riscv64 to pass the check to use 64 bits
# Author: julianmorillo (BSC)
--- qtwebengine/src/3rdparty/chromium/third_party/angle/gni/angle.gni.orig	2025-02-03 17:36:58.966944504 +0100
+++ qtwebengine/src/3rdparty/chromium/third_party/angle/gni/angle.gni	2025-02-03 17:42:15.602107628 +0100
@@ -104,7 +104,7 @@
 
   if (current_cpu == "arm64" || current_cpu == "x64" ||
       current_cpu == "mips64el" || current_cpu == "s390x" ||
-      current_cpu == "ppc64" || current_cpu == "loong64") {
+      current_cpu == "ppc64" || current_cpu == "loong64" || current_cpu == "riscv64") {
     angle_64bit_current_cpu = true
   } else if (current_cpu == "arm" || current_cpu == "x86" ||
              current_cpu == "mipsel" || current_cpu == "s390" ||
# What: Add riscv64 to the list of VALID_ARCHS
# Author: julianmorillo (BSC)
--- qtwebengine/src/3rdparty/chromium/build/linux/sysroot_scripts/install-sysroot.py.orig	2025-02-05 17:34:54.362920556 +0100
+++ qtwebengine/src/3rdparty/chromium/build/linux/sysroot_scripts/install-sysroot.py	2025-02-05 17:35:54.855760536 +0100
@@ -44,7 +44,7 @@
 URL_PREFIX = 'https://commondatastorage.googleapis.com'
 URL_PATH = 'chrome-linux-sysroot/toolchain'
 
-VALID_ARCHS = ('arm', 'arm64', 'i386', 'amd64', 'mips', 'mips64el')
+VALID_ARCHS = ('arm', 'arm64', 'i386', 'amd64', 'mips', 'mips64el', 'riscv64')
 
 ARCH_TRANSLATIONS = {
     'x64': 'amd64',
