--- AutoDock-Vina-1.2.5.orig/src/main/main.cpp	2023-06-16 07:46:43.000000000 +0200
+++ AutoDock-Vina-1.2.5/src/main/main.cpp	2024-04-12 20:20:28.455768279 +0200
@@ -67,7 +67,7 @@
 
 int main(int argc, char* argv[]) {
 	using namespace boost::program_options;
-	const std::string git_version = VERSION;
+	const std::string git_version = "1.2.5";
 	const std::string version_string = "AutoDock Vina " + git_version;
 	const std::string error_message = "\n\n\
 Please report bugs through the Issue Tracker on GitHub \n\
