--- AutoDock-Vina-1.2.5.orig/src/split/split.cpp	2024-04-12 20:27:11.218902248 +0200
+++ AutoDock-Vina-1.2.5/src//split/split.cpp	2024-04-12 20:27:32.995268409 +0200
@@ -133,7 +133,7 @@
 int main(int argc, char* argv[]) {
 	using namespace boost::program_options;
 	const bool advanced = false;
-	const std::string git_version = VERSION;
+	const std::string git_version = "1.2.5";
 	const std::string version_string = "AutoDock Vina PDBQT Split " + git_version;
 
 	const std::string error_message = "\n\n\
