--- AutoDock-Vina-1.2.5.orig/build/python/setup.py	2023-06-16 07:46:43.000000000 +0200
+++ AutoDock-Vina-1.2.5/build/python/setup.py	2024-04-12 21:30:36.979080848 +0200
@@ -314,6 +314,7 @@
         self.compiler.compiler_so.append("-pedantic")
         # Source: https://stackoverflow.com/questions/9723793/undefined-reference-to-boostsystemsystem-category-when-compiling
         self.compiler.compiler_so.append('-DBOOST_ERROR_CODE_HEADER_ONLY')
+        self.compiler.compiler_so.append('-DBOOST_TIMER_ENABLE_DEPRECATED')
 
         print('- compiler options: %s' % self.compiler.compiler_so)
         build_ext.build_extensions(self)
