Support libbfd API change introduced with binutils 2.34.  Patch backported
from upstream repo; fix will be included in releases after v6.0.
--- a/src/services/unwinding/scorep_unwinding_unify.c
+++ b/src/services/unwinding/scorep_unwinding_unify.c
@@ -28,6 +28,19 @@
 
 #if HAVE( LIBBFD )
 #include <bfd.h>
+
+#ifndef bfd_get_section_size
+#define bfd_get_section_size( asection ) bfd_section_size( asection )
+#endif
+
+#ifndef bfd_get_section_vma
+#define bfd_get_section_vma( abfd, asection ) bfd_section_vma( asection )
+#endif
+
+#ifndef bfd_get_section_flags
+#define bfd_get_section_flags( abfd, asection ) bfd_section_flags( asection )
+#endif
+
 #endif
 
 #include "scorep_unwinding_region.h"
