--- Make.inc.orig	2014-10-14 16:48:12.744906000 +1300
+++ Make.inc	2014-10-21 13:12:53.039528000 +1300
@@ -12,7 +12,8 @@
 # for solaris
 #
 # CC = gcc
-  CC = /usr/lang-4.0/bin/cc
+# CC = /usr/lang-4.0/bin/cc
+# CC = mpiicc
 #
 # for sgi
 #
@@ -28,14 +29,14 @@
 # 
 # OPTLEVEL = 
 # OPTLEVEL = -g -v
-  OPTLEVEL = -O
+#  OPTLEVEL = -O
 # OPTLEVEL = -xO5 -v
 # OPTLEVEL = -O3
 # OPTLEVEL = -O4
 # CFLAGS = -Wall -g
 # CFLAGS = -Wall -pg
 # CFLAGS = $(OPTLEVEL) -D_POSIX_C_SOURCE=199506L
-  CFLAGS = $(OPTLEVEL)
+#  CFLAGS = $(CFLAGS)
 # CFLAGS = -Wall $(OPTLEVEL)
 #
 #----------------------------------------
@@ -51,7 +52,7 @@
 #
 # THREAD_LIBS =
 # THREAD_LIBS = -D_REENTRANT=199506L -lpthread
-  THREAD_LIBS = -D_POSIX_C_SOURCE=199506L -lpthread
+#  THREAD_LIBS = -D_POSIX_C_SOURCE=199506L -lpthread
 # THREAD_LIBS = -lpthread
 #
 #---------------------------------------------------------------------
@@ -71,8 +72,8 @@
 #
 #  set the archive flags
 #
-  AR = ar
-  ARFLAGS = rv
+#  AR = ar
+#  ARFLAGS = rv
 #
 #---------------------------------------------------------------------
 #
@@ -103,7 +104,7 @@
 #  MPI install library
 #
 # MPI_INSTALL_DIR = 
-  MPI_INSTALL_DIR = /usr/local/mpich-1.0.13
+#  MPI_INSTALL_DIR = /usr/local/mpich-1.0.13
 #
 #---------------------------------------------------------------------
 #
@@ -115,7 +116,7 @@
 #
 # for solaris
 #
-  MPI_LIB_PATH = -L$(MPI_INSTALL_DIR)/lib/solaris/ch_p4
+#  MPI_LIB_PATH = -L$(MPI_INSTALL_DIR)/lib/solaris/ch_p4
 #
 # for hp
 #
@@ -127,7 +128,7 @@
 #
 # for solaris
 #
-  MPI_LIBS = $(MPI_LIB_PATH) -D_REENTRANT -lmpi -lsocket -lnsl -lthread
+#  MPI_LIBS = $(MPI_LIB_PATH) -D_REENTRANT -lmpi -lsocket -lnsl -lthread
 #
 # for sgi
 #
@@ -142,6 +143,6 @@
 #  MPI include path
 #
 # MPI_INCLUDE_DIR = 
-  MPI_INCLUDE_DIR = -I$(MPI_INSTALL_DIR)/include
+#  MPI_INCLUDE_DIR = -I$(MPI_INSTALL_DIR)/include
 #
 #---------------------------------------------------------------------
