Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Description: build multithreaded, MPI targets by default
diff -Nru spooles-2.2.orig/MPI/makefile spooles-2.2.p1/MPI/makefile
--- spooles-2.2.orig/MPI/makefile	1998-12-17 15:47:44.000000000 +0000
+++ spooles-2.2.p1/MPI/makefile	2022-11-25 16:07:34.000000000 +0000
@@ -2,7 +2,7 @@
 	cd drivers ; make drivers
 
 lib :
-	cd src ; make spoolesMPI.a
+	cd src ; make makeLib
 
 clean :
 	cd src ; make clean
diff -Nru spooles-2.2.orig/MPI/src/makefile spooles-2.2.p1/MPI/src/makefile
--- spooles-2.2.orig/MPI/src/makefile	1998-12-16 21:54:41.000000000 +0000
+++ spooles-2.2.p1/MPI/src/makefile	2022-11-25 16:08:18.000000000 +0000
@@ -42,3 +42,8 @@
 
 clean :
 	- rm -f *.a *.o
+
+makeLib:
+	perl ../../makeLib > makeG
+	make -f makeG
+	rm -f makeG
diff -Nru spooles-2.2.orig/MT/makefile spooles-2.2.p1/MT/makefile
--- spooles-2.2.orig/MT/makefile	1998-12-17 15:47:48.000000000 +0000
+++ spooles-2.2.p1/MT/makefile	2022-11-25 16:08:55.000000000 +0000
@@ -2,7 +2,7 @@
 	cd drivers ; make drivers
 
 lib :
-	cd src ; make spoolesMT.a
+	cd src ; make makeLib
 
 clean :
 	cd src ; make clean
