Author: William Grzybowski <william@grzy.org>
Subject: Fix build for debug and hardening, do not install symlinks.
Last-Update: 2019-07-10
--- pbzip2.orig/Makefile	2019-07-08 18:46:10.382725910 -0300
+++ pbzip2/Makefile	2019-07-08 18:46:24.886865907 -0300
@@ -19,3 +19,3 @@
 # Optimization flags
-CXXFLAGS = -O2
+CXXFLAGS += $(CPPFLAGS)

@@ -49,9 +49,6 @@
 CXXFLAGS += $(CXXFLAGS_PTHREAD)
 
 
-# Linker flags
-LDFLAGS =
-
 
 # External libraries
 LDLIBS = -lbz2
@@ -81,8 +81,6 @@
 	if ( test ! -d $(DESTDIR)$(PREFIX)/share/man/man1 ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 ; fi
 	cp -f pbzip2 $(DESTDIR)$(PREFIX)/bin/pbzip2
 	chmod a+x $(DESTDIR)$(PREFIX)/bin/pbzip2
-	ln -s -f pbzip2 $(DESTDIR)$(PREFIX)/bin/pbunzip2
-	ln -s -f pbzip2 $(DESTDIR)$(PREFIX)/bin/pbzcat
 	cp -f pbzip2.1 $(DESTDIR)$(PREFIX)/share/man/man1
 	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/pbzip2.1
 
