--- ./platform.pl.orig	2019-05-29 10:35:05.000000000 +0000
+++ platform.pl	2022-05-27 15:49:04.724333960 +0000
@@ -212,12 +212,12 @@
   if ( $compiler eq "ifort" )
   {
     print OUTFILE "##############################################################################\n";
-    print OUTFILE "# IA32_Intel/x86-64_Intel:	Intel Pentium with Linux using Intel compiler 17.\n";
+    print OUTFILE "# x86-64_Intel:	Intel with Linux using Intel compiler 21.\n";
     print OUTFILE "##############################################################################\n";
     print OUTFILE "F90_SER = ifort\n";
     print OUTFILE "F90_OMP = ifort\n";
     print OUTFILE "# if appropriate, use mpiifort of Intel instead\n";
-    print OUTFILE "F90_MPI = mpif90\n";
+    print OUTFILE "F90_MPI = mpiifort\n";
 #    if ($cpu =~ /i686/i) {
 #      print OUTFILE "FLAGS_OPT = -O2 -xN -mp1\n";
 #    }
@@ -225,7 +225,7 @@
 #      print OUTFILE "FLAGS_OPT = -O2 -ipo -xW -mp1\n";
 #    }
 #    else {
-      print OUTFILE "FLAGS_OPT = -O2\n";
+      print OUTFILE "FLAGS_OPT = \$(FFLAGS)\n";
 #    }
 #    print OUTFILE "FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable remark\n";
     print OUTFILE "FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293\n";
@@ -235,13 +235,13 @@
     print OUTFILE "FLAGS_OMP = -qopenmp\n";
     print OUTFILE "FLAGS_MPI =\n";
     print OUTFILE "NETCDFROOT =\n";
-    print OUTFILE "ifneq (\$(NETCDFROOT),)\n";
-    print OUTFILE "  INCS_SER = -I\$(NETCDFROOT)/include\n";
-    print OUTFILE "  INCS_OMP = -I\$(NETCDFROOT)/include\n";
-    print OUTFILE "  INCS_MPI = -I\$(NETCDFROOT)/include\n";
-    print OUTFILE "  LIBS_SER = -L\$(NETCDFROOT)/lib -lnetcdf -lnetcdff\n";
-    print OUTFILE "  LIBS_OMP = -L\$(NETCDFROOT)/lib -lnetcdf -lnetcdff\n";
-    print OUTFILE "  LIBS_MPI = -L\$(NETCDFROOT)/lib -lnetcdf -lnetcdff\n";
+    print OUTFILE "ifneq (\$(EBROOTNETCDFMINFORTRAN),)\n";
+    print OUTFILE "  INCS_SER = \n";
+    print OUTFILE "  INCS_OMP = \n";
+    print OUTFILE "  INCS_MPI = \n";
+    print OUTFILE "  LIBS_SER = -lnetcdf -lnetcdff\n";
+    print OUTFILE "  LIBS_OMP = -lnetcdf -lnetcdff\n";
+    print OUTFILE "  LIBS_MPI = -lnetcdf -lnetcdff\n";
     print OUTFILE "  NCF_OBJS = nctablemd.o agioncmd.o swn_outnc.o\n";
     print OUTFILE "else\n";
     print OUTFILE "  INCS_SER =\n";
@@ -257,7 +257,7 @@
     print OUTFILE "EXTO = o\n";
     print OUTFILE "MAKE = make\n";
     print OUTFILE "RM = rm -f\n";
-    print OUTFILE "ifneq (\$(NETCDFROOT),)\n";
+    print OUTFILE "ifneq (\$(EBROOTNETCDFMINFORTRAN),)\n";
     print OUTFILE "  swch = -unix -impi -netcdf\n";
     print OUTFILE "else\n";
     print OUTFILE "  swch = -unix -impi\n";
