--- Chrysalis/base/CommandLineParser.h.orig	2011-09-26 15:16:12.477156046 +0200
+++ Chrysalis/base/CommandLineParser.h	2011-09-26 15:16:17.238244105 +0200
@@ -49,8 +49,8 @@
     string GetName() const { return mName; }
     string GetDescription() { return mDesc; }
 
-    friend bool operator< (const commandArg<argType> &lhs,
-                           const commandArg<argType> &rhs)
+    friend bool operator< (commandArg<argType> &lhs,
+                           commandArg<argType> &rhs)
     {
         if ( lhs.GetName() < rhs.GetName() )
             return true;

