--- Makefile.orig	2014-04-24 03:59:20.000000000 +1200
+++ Makefile	2014-06-03 16:25:26.143650000 +1200
@@ -27,7 +27,7 @@
 	cd Chrysalis && $(MAKE) UNSUPPORTED=yes $(CHRYSALIS_MAKE_FLAGS)
 
 jellyfish:
-	cd trinity-plugins/jellyfish && ./configure CC=gcc CXX=g++ --prefix=`pwd` && $(MAKE) AM_CPPFLAGS="-Wall -Wnon-virtual-dtor -I"`pwd`
+	#cd trinity-plugins/jellyfish && ./configure CC=gcc CXX=g++ --prefix=`pwd` && $(MAKE) AM_CPPFLAGS="-Wall -Wnon-virtual-dtor -I"`pwd`
 
 fastool:
 	cd trinity-plugins/fastool && $(MAKE)
