--- trinityrnaseq_r2012-10-05/Makefile.orig.eb	2012-10-05 16:42:25.000000000 +0200
+++ trinityrnaseq_r2012-10-05/Makefile	2013-03-27 17:52:05.872442000 +0100
@@ -20,7 +20,7 @@
                 && ./configure --prefix=`pwd` $(INCHWORM_CONFIGURE_FLAGS) && $(MAKE) install
 	cd Chrysalis && $(MAKE) UNSUPPORTED=yes $(CHRYSALIS_MAKE_FLAGS)
 	cd trinity-plugins/rsem && $(MAKE) 
-	cd trinity-plugins/jellyfish && ./configure --prefix=`pwd` && $(MAKE)
+	cd trinity-plugins/jellyfish && ./configure --prefix=`pwd` && $(MAKE) CXXFLAGS="$(CXXFLAGS) -Wno-narrowing"
 	cd trinity-plugins/fastool && $(MAKE)
 	cd trinity-plugins/slclust && $(MAKE)
 	cd trinity-plugins/coreutils && ./build_parallel_sort.sh
