--- trinityrnaseq_r2013-02-25/Makefile.orig	2013-02-25 14:24:19.000000000 +0100
+++ trinityrnaseq_r2013-02-25/Makefile	2013-03-28 17:04:49.999074000 +0100
@@ -20,7 +20,7 @@
                 && ./configure --prefix=`pwd` $(INCHWORM_CONFIGURE_FLAGS) && $(MAKE) install
 	cd Chrysalis && $(MAKE) UNSUPPORTED=yes $(CHRYSALIS_MAKE_FLAGS)
 	cd trinity-plugins/rsem && $(MAKE) 
-	cd trinity-plugins/jellyfish && ./configure CC=gcc CXX=g++ --prefix=`pwd` && $(MAKE) AM_CPPFLAGS="-Wall -Wnon-virtual-dtor -I"`pwd`
+	# cd trinity-plugins/jellyfish && ./configure CC=gcc CXX=g++ --prefix=`pwd` && $(MAKE) AM_CPPFLAGS="-Wall -Wnon-virtual-dtor -I"`pwd`
 	cd trinity-plugins/fastool && $(MAKE)
 	cd trinity-plugins/parafly && ./configure --prefix=`pwd` && $(MAKE) install
 	cd trinity-plugins/slclust && $(MAKE) install
