diff -ru esmf-8.4.0.orig/src/addon/esmpy/pyproject.toml esmf-8.4.0/src/addon/esmpy/pyproject.toml
--- esmf-8.4.0.orig/src/addon/esmpy/pyproject.toml	2023-02-02 18:59:29.183281388 +0000
+++ esmf-8.4.0/src/addon/esmpy/pyproject.toml	2023-02-02 19:02:26.367569185 +0000
@@ -6,7 +6,7 @@
 name = "esmpy"
 description = "ESMF Python interface"
 # readme = "README.md"
-maintainers = [{author = "ESMF Core Team", email = "esmf_support@ucar.edu"}]
+maintainers = [{name = "ESMF Core Team", email = "esmf_support@ucar.edu"}]
 requires-python = ">=3.7"
 license = {text = "University of Illinois-NCSA"}
 dependencies = [
