diff -Nru esmf-8.8.0.orig/src/addon/esmpy/pyproject.toml esmf-8.8.0/src/addon/esmpy/pyproject.toml
--- esmf-8.8.0.orig/src/addon/esmpy/pyproject.toml	2025-03-11 19:24:08.480962617 -0400
+++ esmf-8.8.0/src/addon/esmpy/pyproject.toml	2025-03-11 19:26:27.292730989 -0400
@@ -16,13 +16,6 @@
     # compatibility:
     "numpy >= 1.19, < 3",
     'importlib-metadata; python_version < "3.8"',
-    # setuptools-git-versioning shouldn't be needed here, but is
-    # included as a workaround for problems with the build-time
-    # installation of this package with python 3.10 (given by the
-    # build-system section above). By including it here, we at least
-    # ensure that this package will be available for a second or
-    # subsequent pip install of esmpy.
-    'setuptools-git-versioning; python_version >= "3.10"',
 ]
 dynamic = [ "version" ]
 
