diff -Nru esmf.orig/build_config/Linux.intel.default/build_rules.mk esmf/build_config/Linux.intel.default/build_rules.mk
--- esmf.orig/build_config/Linux.intel.default/build_rules.mk	2018-02-05 17:50:29.000000000 +0100
+++ esmf/build_config/Linux.intel.default/build_rules.mk	2018-02-05 17:51:34.000000000 +0100
@@ -186,10 +186,10 @@
 ############################################################
 # OpenMP compiler and linker flags
 #
-ESMF_OPENMP_F90COMPILEOPTS += -openmp
-ESMF_OPENMP_CXXCOMPILEOPTS += -openmp
-ESMF_OPENMP_F90LINKOPTS    += -openmp
-ESMF_OPENMP_CXXLINKOPTS    += -openmp
+ESMF_OPENMP_F90COMPILEOPTS += -qopenmp
+ESMF_OPENMP_CXXCOMPILEOPTS += -qopenmp
+ESMF_OPENMP_F90LINKOPTS    += -qopenmp
+ESMF_OPENMP_CXXLINKOPTS    += -qopenmp
 
 ############################################################
 # Set rpath syntax
