diff -ru pindel-0.2.5a7.orig/bam2pindel.pl pindel-0.2.5a7/bam2pindel.pl
--- pindel-0.2.5a7.orig/bam2pindel.pl	2014-09-16 20:53:13.000000000 +0200
+++ pindel-0.2.5a7/bam2pindel.pl	2015-06-03 21:12:57.775900885 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/env perl
 
 ########################################################################################################
 #                                                                                                      #
