diff -Nru apache-arrow-0.17.1.orig/cpp/CMakeLists.txt apache-arrow-0.17.1/cpp/CMakeLists.txt
--- apache-arrow-0.17.1.orig/cpp/CMakeLists.txt	2021-05-20 17:57:53.155371000 +0100
+++ apache-arrow-0.17.1/cpp/CMakeLists.txt	2021-05-20 17:58:42.354451915 +0100
@@ -149,7 +149,7 @@
   find_package(PythonInterp)
 else()
   # Use the first Python installation on PATH, not the newest one
-  set(Python3_FIND_STRATEGY "LOCATION")
+  # set(Python3_FIND_STRATEGY "LOCATION")
   # On Windows, use registry last, not first
   set(Python3_FIND_REGISTRY "LAST")
   # On macOS, use framework last, not first
diff -Nru apache-arrow-0.17.1.orig/python/CMakeLists.txt apache-arrow-0.17.1/python/CMakeLists.txt
--- apache-arrow-0.17.1.orig/python/CMakeLists.txt	2021-05-20 17:57:41.766056000 +0100
+++ apache-arrow-0.17.1/python/CMakeLists.txt	2021-05-20 18:42:24.229518213 +0100
@@ -42,7 +42,7 @@
 endif()
 
 # Use the first Python installation on PATH, not the newest one
-set(Python3_FIND_STRATEGY "LOCATION")
+# set(Python3_FIND_STRATEGY "LOCATION")
 # On Windows, use registry last, not first
 set(Python3_FIND_REGISTRY "LAST")
 # On macOS, use framework last, not first
