Fix lsp-types error 
Author: Cintia Willemyns (Vrije Universiteit)
diff -Narub ruff-0.7.0.orig/Cargo.lock ruff-0.7.0/Cargo.lock
--- ruff-0.7.0.orig/Cargo.lock	2024-10-17 18:14:21.000000000 +0200
+++ ruff-0.7.0/Cargo.lock	2025-07-23 14:40:41.728066303 +0200
@@ -1439,7 +1439,7 @@
 [[package]]
 name = "lsp-types"
 version = "0.95.1"
-source = "git+https://github.com/astral-sh/lsp-types.git?rev=3512a9f#3512a9f33eadc5402cfab1b8f7340824c8ca1439"
+source = "git+https://github.com/astral-sh/lsp-types.git?rev=3512a9f33eadc5402cfab1b8f7340824c8ca1439#3512a9f33eadc5402cfab1b8f7340824c8ca1439"
 dependencies = [
  "bitflags 1.3.2",
  "serde",
diff -Narub ruff-0.7.0.orig/Cargo.toml ruff-0.7.0/Cargo.toml
--- ruff-0.7.0.orig/Cargo.toml	2025-07-23 10:21:42.187804160 +0200
+++ ruff-0.7.0/Cargo.toml	2025-07-23 15:33:36.625826512 +0200
@@ -93,7 +93,7 @@
 libcst = { version = "1.1.0", default-features = false }
 log = { version = "0.4.17" }
 lsp-server = { version = "0.7.6" }
-lsp-types = { git = "https://github.com/astral-sh/lsp-types.git", rev = "3512a9f", features = [
+lsp-types = { git = "https://github.com/astral-sh/lsp-types.git", rev = "3512a9f33eadc5402cfab1b8f7340824c8ca1439", features = [
     "proposed",
 ] }
 matchit = { version = "0.8.1" }
