Use build environment defined by EasyBuild
author: Nicholas Cheng (@nicholascheng94)
author: Alex Domingo (Vrije Universiteit Brussel)
--- input.h.orig	2022-11-22 15:28:24.141800278 +0100
+++ input.h	2022-11-22 15:28:56.761244080 +0100
@@ -18,7 +18,7 @@
 #include "SpinQuantum.h"
 #include "timer.h"
 #include "couplingCoeffs.h"
-#include <boost/tr1/unordered_map.hpp>
+#include <boost/unordered_map.hpp>
 #include "IntegralMatrix.h"
 
 
--- makefile.orig	2022-11-22 16:40:08.227179000 +0100
+++ makefile	2022-11-22 16:41:34.467330533 +0100
@@ -6,28 +6,28 @@
 
 ######### GENERAL OPTIONS FOR USER #########
 #specify boost include file
-BOOSTINCLUDE = /opt/local/include
+BOOSTINCLUDE = ${EBROOTBOOST}/include
 
 #specify boost and lapack-blas library locations
-BOOSTLIB = -L/opt/local/lib  -lboost_system -lboost_filesystem -lboost_serialization
+BOOSTLIB = -L${EBROOTBOOST}/lib -lboost_system -lboost_filesystem -lboost_serialization
 #BOOSTLIB = -lboost_serialization -lboost_system -lboost_filesystem
-LAPACKBLAS =    /usr/lib/liblapack.dylib /usr/lib/libblas.dylib
+LAPACKBLAS = ${LIBLAPACK}
 
 # set if using BOOST version >= 1.56.0
-USE_BOOST56 = no
+USE_BOOST56 = yes
 ifeq ($(USE_BOOST56), yes)
 	B56 = -DBOOST_1_56_0
 endif
 
 #use these variable to set if we will use mpi or not 
 USE_MPI = no
 USE_MKL = no
 
 # change to icpc for Intel
-CXX =  icpc
-MPICXX = mpiicpc
-export CXX
-export MPICXX
+#CXX =  icpc
+#MPICXX = mpiicpc
+#export CXX
+#export MPICXX
 
 # use this variable to set if we will use integer size of 8 or not.
 # molpro compilation requires I8, since their integers are long
@@ -40,7 +40,7 @@
 # molcas compilation w/ -64 option requires I8 as well
 MOLCAS = no
 
-OPENMP = no
+OPENMP = yes
 
 ifeq ($(USE_MKL), yes)
 MKLLIB = /opt/intel/compilers_and_libraries/linux/mkl/lib/intel64
@@ -101,7 +101,7 @@
       OPENMP_FLAGS= -openmp -D_OPENMP 
    endif
 # Intel compiler
-   OPT = -DNDEBUG -O3 -funroll-loops -Werror
+   OPT = -DNDEBUG -funroll-loops $(OPTFLAGS)
 #  OPT = -g -fPIC
    ifeq ($(USE_MPI), no) 
       CXX = icc
@@ -113,7 +113,7 @@
    ifeq ($(OPENMP), yes)
       OPENMP_FLAGS= -fopenmp #-D_OPENMP 
    endif
-   OPT = -DNDEBUG -O3 -g -funroll-loops -Werror
+   OPT = -DNDEBUG -g -funroll-loops $(OPTFLAGS)
 #   OPT = -g -fPIC
 endif
 
@@ -122,7 +122,7 @@
       OPENMP_FLAGS= -fopenmp #-D_OPENMP 
    endif
 
-   OPT = -DNDEBUG -g -Werror
+   OPT = -DNDEBUG -g $(OPTFLAGS)
 endif
 
 ifeq ($(DOPROF),yes)
