--- pyproject.toml.orig	1970-01-01 01:00:00.000000000 +0100
+++ pyproject.toml	2026-01-21 17:09:37.176254000 +0100
@@ -43,6 +43,24 @@
 black = "*"
 mypy = "*"
 
+[tool.poetry]
+name = "sentieon_cli"
+version = "1.5.0"
+description = "entry point for sentieon command-line tools"
+authors = ["Don Freed <don.freed@sentieon.com>", "Brent <bpederse@gmail.com>"]
+readme = "README.md"
+packages = [{ include = "sentieon_cli" }]
+include = ["LICENSE"]
+
+[tool.poetry.dependencies]
+python = ">=3.8"
+colorlog = ">=6.7.0,<7.0.0"
+importlib-resources = ">=5.0,<6.0"
+packaging = ">=23.2,<26.0"
+
+[tool.poetry.scripts]
+sentieon-cli = "sentieon_cli:main"
+
 [build-system]
 requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
