include location of PyQt .sip files in SIPCMD macro

author: Kenneth Hoste (HPC-UGent)
author: Pavel Grochal (INUITS)
--- qgis-3.4.12/cmake/SIPMacros.cmake.orig	2019-10-04 13:33:43.751421147 +0200
+++ qgis-3.4.12/cmake/SIPMacros.cmake	2019-10-04 13:35:00.297687978 +0200
@@ -111,7 +111,7 @@
     ADD_DEFINITIONS( /bigobj )
   ENDIF(MSVC)
 
-  SET(SIPCMD ${SIP_BINARY_PATH} ${_sip_tags} -w -e ${_sip_x} ${SIP_EXTRA_OPTIONS} -j ${SIP_CONCAT_PARTS} -c ${CMAKE_CURRENT_BINARY_DIR}/${_module_path} -I ${CMAKE_CURRENT_BINARY_DIR}/${_module_path} ${_sip_includes} ${_configured_module_sip})
+  SET(SIPCMD ${SIP_BINARY_PATH} ${_sip_tags} -w -e ${_sip_x} ${SIP_EXTRA_OPTIONS} -j ${SIP_CONCAT_PARTS} -c ${CMAKE_CURRENT_BINARY_DIR}/${_module_path} -I ${CMAKE_CURRENT_BINARY_DIR}/${_module_path} ${_sip_includes} -I$ENV{EBROOTPYQT5}/share/sip/PyQt5 ${_configured_module_sip})
   ADD_CUSTOM_COMMAND(
     OUTPUT ${_sip_output_files}
     COMMAND ${CMAKE_COMMAND} -E echo ${message}
