include location of PyQt .sip files in SIPCMD macro

author: Kenneth Hoste (HPC-UGent)
--- qgis-2.18.4/cmake/SIPMacros.cmake.orig	2017-03-08 17:44:20.698576501 +0100
+++ qgis-2.18.4/cmake/SIPMacros.cmake	2017-03-08 17:46:39.110172584 +0100
@@ -96,7 +96,7 @@
     ADD_DEFINITIONS( /bigobj )
   ENDIF(MSVC)
 
-  SET(SIPCMD ${SIP_BINARY_PATH} ${_sip_tags} -w -e ${_sip_x} ${SIP_EXTRA_OPTIONS} -j ${SIP_CONCAT_PARTS} -c ${CMAKE_CURRENT_BINARY_DIR}/${_module_path} ${_sip_includes} ${_abs_module_sip})
+  SET(SIPCMD ${SIP_BINARY_PATH} ${_sip_tags} -w -e ${_sip_x} ${SIP_EXTRA_OPTIONS} -j ${SIP_CONCAT_PARTS} -c ${CMAKE_CURRENT_BINARY_DIR}/${_module_path} ${_sip_includes} -I$ENV{EBROOTPYQT}/share/sip/PyQt4 ${_abs_module_sip})
   SET(SUPPRESS_SIP_WARNINGS FALSE CACHE BOOL "Hide SIP warnings")
   MARK_AS_ADVANCED(SUPPRESS_SIP_WARNINGS)
   IF(SUPPRESS_SIP_WARNINGS)
