diff -Nru g2lib-1.4.0-orig/enc_jpeg2000.c g2lib-1.4.0/enc_jpeg2000.c
--- g2lib-1.4.0-orig/enc_jpeg2000.c	2014-07-08 20:36:50.000000000 +0200
+++ g2lib-1.4.0/enc_jpeg2000.c	2018-07-06 09:08:10.000000000 +0200
@@ -139,7 +139,7 @@
     image.clrspc_=JAS_CLRSPC_SGRAY;         /* grayscale Image */
     image.cmprof_=0; 
 #endif
-    image.inmem_=1;
+    // image.inmem_=1;
 
     cmpt.tlx_=0;
     cmpt.tly_=0;
