fix order of compilation in makefile
Author: Samuel Moors, Vrije Universiteit Brussel (VUB)
diff -ur g2lib-3.1.0.orig/makefile g2lib-3.1.0/makefile
--- g2lib-3.1.0.orig/makefile	2017-06-12 21:39:43.000000000 +0200
+++ g2lib-3.1.0/makefile	2019-03-28 14:50:24.070862781 +0100
@@ -134,6 +134,7 @@
 .SUFFIXES: .a .f .F .c
 
 $(LIB):	$(LIB)(gridtemplates.o) \
+	$(LIB)(intmath.o) \
 	$(LIB)(pdstemplates.o) \
 	$(LIB)(drstemplates.o) \
 	$(LIB)(gribmod.o) \
@@ -196,7 +197,6 @@
 	$(LIB)(params.o) \
         $(LIB)(params_ecmwf.o) \
 	$(LIB)(getidx.o) \
-        $(LIB)(intmath.o) \
 	$(LIB)(gdt2gds.o)
 
 .F.f:
