fix build order in makefile, simpack.o required intmath.mod
author: Kenneth Hoste (HPC-UGent)
--- g2lib-3.1.0/makefile.orig	2018-08-14 10:37:57.049429915 +0200
+++ g2lib-3.1.0/makefile	2018-08-14 10:38:05.149505287 +0200
@@ -161,6 +161,7 @@
 	$(LIB)(gribinfo.o) \
 	$(LIB)(mkieee.o) \
 	$(LIB)(rdieee.o) \
+        $(LIB)(intmath.o) \
 	$(LIB)(simpack.o) \
 	$(LIB)(simunpack.o) \
 	$(LIB)(cmplxpack.o) \
@@ -196,7 +197,6 @@
 	$(LIB)(params.o) \
         $(LIB)(params_ecmwf.o) \
 	$(LIB)(getidx.o) \
-        $(LIB)(intmath.o) \
 	$(LIB)(gdt2gds.o)
 
 .F.f:
