# Fix kind of arguments used in iand
# S.D. Pinches, 30.08.2020
diff -Nru g2lib-3.1.0-orig/intmath.f g2lib-3.1.0/intmath.f
--- g2lib-3.1.0-orig/intmath.f	2017-06-12 21:38:08.000000000 +0200
+++ g2lib-3.1.0/intmath.f	2020-08-30 18:24:35.724869000 +0200
@@ -84,7 +84,7 @@
       ilog2_8=0
       i=i_in
       if(i<=0) return
-      if(iand(i,i-1)/=0) then
+      if(iand(i,i-1_8)/=0) then
          !write(0,*) 'iand i-1'
          ilog2_8=1
       endif
@@ -129,7 +129,7 @@
       ilog2_4=0
       i=i_in
       if(i<=0) return
-      if(iand(i,i-1)/=0) then
+      if(iand(i,i-1_4)/=0) then
          !write(0,*) 'iand i-1'
          ilog2_4=1
       endif
@@ -169,7 +169,7 @@
       ilog2_2=0
       i=i_in
       if(i<=0) return
-      if(iand(i,i-1)/=0) then
+      if(iand(i,i-1_2)/=0) then
          !write(0,*) 'iand i-1'
          ilog2_2=1
       endif
@@ -204,7 +204,7 @@
       ilog2_1=0
       i=i_in
       if(i<=0) return
-      if(iand(i,i-1)/=0) then
+      if(iand(i,i-1_1)/=0) then
          !write(0,*) 'iand i-1'
          ilog2_1=1
       endif
