allow setting SAGE_ROOT and SAGE_LOCAL independently
author: Miguel Dias Costa (National University of Singapore)
--- sage-conf-10.1/setup.py.orig	2023-11-12 12:41:19.314126027 +0800
+++ sage-conf-10.1/setup.py	2023-11-12 12:47:01.900150343 +0800
@@ -26,8 +26,8 @@
         machine = platform.machine()
         arch_tag = f'{system}-{machine}'
         # TODO: These two should be user-configurable with options passed to "setup.py install"
-        SAGE_ROOT = os.path.join(DOT_SAGE, f'sage-{sage_version}-{arch_tag}')
-        SAGE_LOCAL = os.path.join(SAGE_ROOT, 'local')
+        SAGE_ROOT = os.environ.get('SAGE_ROOT', os.path.join(DOT_SAGE, f'sage-{sage_version}-{arch_tag}'))
+        SAGE_LOCAL = os.environ.get('SAGE_LOCAL', os.path.join(SAGE_ROOT, 'local'))
         if os.path.exists(os.path.join(SAGE_ROOT, 'config.status')):
             print(f'Reusing SAGE_ROOT={SAGE_ROOT}')
         else:
