diff -ru sage-9.3.orig/build/bin/sage-bootstrap-python sage-9.3/build/bin/sage-bootstrap-python
--- sage-9.3.orig/build/bin/sage-bootstrap-python	2021-07-16 15:39:55.144271065 +0000
+++ sage-9.3/build/bin/sage-bootstrap-python	2021-07-16 15:43:19.969672280 +0000
@@ -34,7 +34,7 @@
 # operation in Python 3.0.x-3.6.x by setting LC_ALL=C or similar.
 
 if [ "$LC_ALL" = "C" -o "$LANG" = "C" -o "$LC_CTYPE" = "C" ]; then
-    LC_ALL=$(locale -a | grep -E -i '^(c|en_us)[-.]utf-?8$' | head -n 1)
+    LC_ALL=$(locale -a | grep -E -i '^(en_ca|c|en_us)[-.]utf-?8$' | head -n 1)
     LANG=$LC_ALL
     export LC_ALL
     export LANG
diff -ru sage-9.3.orig/build/bin/sage-pip-install sage-9.3/build/bin/sage-pip-install
--- sage-9.3.orig/build/bin/sage-pip-install	2021-07-16 15:39:55.145271071 +0000
+++ sage-9.3/build/bin/sage-pip-install	2021-07-16 15:50:52.761763005 +0000
@@ -12,7 +12,7 @@
 #                      This also disables pip's version self-check.
 # --isolated         : Don't read configuration files such as
 #                      ~/.pydistutils.cfg
-pip_install_flags="--ignore-installed --verbose --no-deps --no-index --isolated"
+pip_install_flags="--ignore-installed --verbose --no-deps --no-index "
 
 # Note: "sage-pip-install" is meant to be run after $(PYTHON) has
 # been installed (either as an spkg or as a venv over a system python3).
