SciPy-bundle provides numpy and requires pybind11 2.11.
Author: Georgios Kafanas (University of Luxembourg)
diff -ruN hoomd-5.1.0.orig/CMake/hoomd/HOOMDPythonSetup.cmake hoomd-5.1.0/CMake/hoomd/HOOMDPythonSetup.cmake
--- hoomd-5.1.0.orig/CMake/hoomd/HOOMDPythonSetup.cmake	2025-02-25 00:48:03.924127000 +0100
+++ hoomd-5.1.0/CMake/hoomd/HOOMDPythonSetup.cmake	2025-02-21 15:41:45.000000000 +0100
@@ -12,7 +12,7 @@
     endif()
 endif()
 
-find_package(pybind11 2.12 CONFIG REQUIRED)
+find_package(pybind11 2.11 CONFIG REQUIRED)
 
 if (pybind11_FOUND)
     find_package_message(pybind11 "Found pybind11: ${pybind11_DIR} ${pybind11_INCLUDE_DIR} (version ${pybind11_VERSION})" "[${pybind11_DIR}][${pybind11_INCLUDE_DIR}]")
