Remove fixed version for some Python modules
Author: J. Sassmannshausen (Imperial College London/UK)
diff --git a/aqme-1.7.3.orig/setup.py b/aqme-1.7.3/setup.py
index 4bb2c0a..a6584ff 100755
--- a/aqme-1.7.3.orig/setup.py
+++ b/aqme-1.7.3/setup.py
@@ -36,11 +36,11 @@ setup(
     ],
     install_requires=[
         "PyYAML",
-        "pandas==2.2.3",
+        "pandas",
         "progress",
-        "numpy==1.26.4",
-        "rdkit==2024.3.3",
-        "cclib==1.7.2",
+        "numpy",
+        "rdkit",
+        "cclib",
         "cffi",
         "morfeus-ml==0.7.2"
     ],
