fix linking with OpenBLAS (rather than using "-lblas -llapack")

author: Kenneth Hoste (HPC-UGent)

--- python-igraph-0.9.0/setup.py.orig	2021-02-25 20:12:21.000000000 +0100
+++ python-igraph-0.9.0/setup.py	2021-03-25 20:59:10.525415847 +0100
@@ -523,7 +523,7 @@
         if exit_code > 0 or len(line) == 0:
             return False
 
-        opts = line.strip().split()
+        opts = line.strip().replace('-lblas', '-lopenblas').replace('-llapack', '').split()
         self.libraries = [opt[2:] for opt in opts if opt.startswith("-l")]
         self.library_dirs = [opt[2:] for opt in opts if opt.startswith("-L")]
         self.include_dirs = [opt[2:] for opt in opts if opt.startswith("-I")]
