diff -Nru VTK-9.0.1.orig/ThirdParty/vtkm/vtkvtkm/vtk-m/CMake/VTKmDeviceAdapters.cmake VTK-9.0.1/ThirdParty/vtkm/vtkvtkm/vtk-m/CMake/VTKmDeviceAdapters.cmake
--- VTK-9.0.1.orig/ThirdParty/vtkm/vtkvtkm/vtk-m/CMake/VTKmDeviceAdapters.cmake	2021-06-13 20:37:57.262973000 +0100
+++ VTK-9.0.1/ThirdParty/vtkm/vtkvtkm/vtk-m/CMake/VTKmDeviceAdapters.cmake	2021-06-13 22:56:14.260066318 +0100
@@ -185,7 +185,7 @@
 
     #specify the property
     set(VTKm_CUDA_Architecture "native" CACHE STRING "Which GPU Architecture(s) to compile for")
-    set_property(CACHE VTKm_CUDA_Architecture PROPERTY STRINGS native fermi kepler maxwell pascal volta turing all none)
+    set_property(CACHE VTKm_CUDA_Architecture PROPERTY STRINGS native fermi kepler maxwell pascal volta turing ampere all none)
 
     #detect what the property is set too
     if(VTKm_CUDA_Architecture STREQUAL "native")
@@ -239,13 +239,15 @@
       set(arch_flags --generate-code=arch=compute_70,code=sm_70)
     elseif(VTKm_CUDA_Architecture STREQUAL "turing")
       set(arch_flags --generate-code=arch=compute_75,code=sm_75)
+    elseif(VTKm_CUDA_Architecture STREQUAL "ampere")
+      set(arch_flags --generate-code=arch=compute_80,code=sm_80)
     elseif(VTKm_CUDA_Architecture STREQUAL "all")
-      set(arch_flags --generate-code=arch=compute_30,code=sm_30
-                     --generate-code=arch=compute_35,code=sm_35
+      set(arch_flags --generate-code=arch=compute_35,code=sm_35
                      --generate-code=arch=compute_50,code=sm_50
                      --generate-code=arch=compute_60,code=sm_60
                      --generate-code=arch=compute_70,code=sm_70
-                     --generate-code=arch=compute_75,code=sm_75)
+                     --generate-code=arch=compute_75,code=sm_75
+                     --generate-code=arch=compute_80,code=sm_80)
     endif()
 
     string(REPLACE ";" " " arch_flags "${arch_flags}")
