diff -ruN DeepCell_Toolbox-0.11.2.orig/setup.py DeepCell_Toolbox-0.11.2/setup.py
--- DeepCell_Toolbox-0.11.2.orig/setup.py	2022-05-17 10:14:38.000000000 -0700
+++ DeepCell_Toolbox-0.11.2/setup.py	2022-05-25 12:32:53.012704000 -0700
@@ -96,8 +96,8 @@
       cmdclass={'build_ext': BuildExtension},
       install_requires=[
           'cython>=0.28.0',
-          'opencv-python-headless<=4.2.0.32; python_version < "3"',
-          'opencv-python-headless; python_version >= "3"',
+          #'opencv-python-headless<=4.2.0.32; python_version < "3"',
+          #'opencv-python-headless; python_version >= "3"',
           'pandas',
           'networkx>=2.1',
           'numpy>=1.16.6',
