# Patch to allow Symengine repo to detect CMake/4.0.3
--- setup.py.orig	2026-02-04 11:24:01.844143000 +0100
+++ setup.py	2026-02-04 11:25:12.923643376 +0100
@@ -42,7 +42,8 @@
     from distutils.command.build import build as _build
 
 cmake_opts = [("PYTHON_BIN", sys.executable),
-              ("CMAKE_INSTALL_RPATH_USE_LINK_PATH", "yes")]
+              ("CMAKE_INSTALL_RPATH_USE_LINK_PATH", "yes"),
+              ("CMAKE_POLICY_VERSION_MINIMUM", "3.5")]
 cmake_generator = [None]
 cmake_build_type = ["Release"]
 
