See https://aur.archlinux.org/packages/med

diff --git a/config/cmake_files/medMacros.cmake b/config/cmake_files/medMacros.cmake
index 992103e..b60ef52 100644
--- a/config/cmake_files/medMacros.cmake
+++ b/config/cmake_files/medMacros.cmake
@@ -447,7 +447,7 @@ MACRO(MED_FIND_HDF5)
     ##
     ## Requires 1.12.x version
     ##
-    IF (NOT HDF_VERSION_MAJOR_REF EQUAL 1 OR NOT HDF_VERSION_MINOR_REF EQUAL 12 OR NOT HDF_VERSION_RELEASE_REF GREATER 0)
+    IF (HDF5_VERSION VERSION_LESS 1.12.1)
         MESSAGE(FATAL_ERROR "HDF5 version is ${HDF_VERSION_REF}. Only versions >= 1.12.1 are supported.")
     ENDIF()
     ##
diff --git a/src/ci/MEDfileCompatibility.c b/src/ci/MEDfileCompatibility.c
index b0f7c59..2c1bb8e 100644
--- a/src/ci/MEDfileCompatibility.c
+++ b/src/ci/MEDfileCompatibility.c
@@ -116,7 +116,7 @@ MEDfileCompatibility(const char* const filename,
 #if MED_NUM_MAJEUR != 5
 #error "Don't forget to update the test version here when you change the major version of the library !"
 #endif
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to check the compatibility version of the library, depending on the internal hdf model choice !"
 #error "Cf. _MEDfileCreate ..."
 #endif
diff --git a/src/hdfi/_MEDfileCreate.c b/src/hdfi/_MEDfileCreate.c
index 8958f57..c27967b 100644
--- a/src/hdfi/_MEDfileCreate.c
+++ b/src/hdfi/_MEDfileCreate.c
@@ -189,7 +189,7 @@ med_idt _MEDfileCreate(const char * const filename, const med_access_mode access
    * Cette ligne est censée obliger HDF à ne pas utiliser un modèle interne différent de 1.10.z
    * Un test autoconf permet de fixer un intervalle de version HDF à MED.
    */
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif
    
diff --git a/src/hdfi/_MEDfileOpen.c b/src/hdfi/_MEDfileOpen.c
index 9ef55a0..5500776 100644
--- a/src/hdfi/_MEDfileOpen.c
+++ b/src/hdfi/_MEDfileOpen.c
@@ -113,7 +113,7 @@ med_idt _MEDfileOpen(const char * const filename,const med_access_mode accessmod
        has been set in the group creation property list (see H5Pset_link_creation_order).
   */
 
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif
 /* L'avantage de bloquer le modèle interne HDF5
diff --git a/src/hdfi/_MEDfileOpenForImport.c b/src/hdfi/_MEDfileOpenForImport.c
index 938c55d..497e1e3 100644
--- a/src/hdfi/_MEDfileOpenForImport.c
+++ b/src/hdfi/_MEDfileOpenForImport.c
@@ -53,7 +53,7 @@ med_idt  _MEDfileOpenForImport(const char * const filename,const med_access_mode
   }
 
 
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif
 /* L'avantage de bloquer le modèle interne HDF5
diff --git a/src/hdfi/_MEDmemFileOpen.c b/src/hdfi/_MEDmemFileOpen.c
index fbf059c..516e101 100644
--- a/src/hdfi/_MEDmemFileOpen.c
+++ b/src/hdfi/_MEDmemFileOpen.c
@@ -439,7 +439,7 @@ med_idt _MEDmemFileOpen(const char * const filename, med_memfile * const memfile
     goto ERROR;
   }
 
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif
   if ( H5Pset_libver_bounds( _fapl, H5F_LIBVER_V112, H5F_LIBVER_V112 ) ) {
@@ -506,7 +506,7 @@ med_idt _MEDmemFileOpen(const char * const filename, med_memfile * const memfile
 	goto ERROR;
       }
       _fversionMM = 100*_fmajor+10*_fminor;
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif
       if ( _fversionMM < 500 ) { /*100*MED_NUM_MAJEUR+10*MED_NUM_MINEUR*/
diff --git a/src/hdfi/_MEDparFileCreate.c b/src/hdfi/_MEDparFileCreate.c
index 2d4e164..2a8ca7d 100644
--- a/src/hdfi/_MEDparFileCreate.c
+++ b/src/hdfi/_MEDparFileCreate.c
@@ -64,7 +64,7 @@ med_idt _MEDparFileCreate(const char * const filename, const med_access_mode acc
    * En HDF5-1.10.0p1 cela n'a aucun effet ! 
    * Un test autoconf permet de fixer un intervalle de version HDF à MED.
    */
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif
    
diff --git a/src/hdfi/_MEDparFileOpen.c b/src/hdfi/_MEDparFileOpen.c
index d072f18..a226683 100644
--- a/src/hdfi/_MEDparFileOpen.c
+++ b/src/hdfi/_MEDparFileOpen.c
@@ -86,7 +86,7 @@ med_idt _MEDparFileOpen(const char * const filename,const med_access_mode access
   }
   _fversionMM = 100*_fmajor+10*_fminor;
 
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif 
   if ( _fversionMM < 500 ) { /*100*MED_NUM_MAJEUR+10*MED_NUM_MINEUR*/
diff --git a/src/misc/MEDversionedApi3C.c b/src/misc/MEDversionedApi3C.c
index acc044f..e04078b 100644
--- a/src/misc/MEDversionedApi3C.c
+++ b/src/misc/MEDversionedApi3C.c
@@ -114,7 +114,7 @@ MedFuncType _MEDversionedApi3( const char * const key,
   /*      (_fversionMM <= (100*MED_NUM_MAJEUR+10*MED_NUM_MINEUR) ) */
   /*      ) { */
 
-#if H5_VERS_MINOR > 12
+#if H5_VERS_MINOR > 14
 #error "Don't forget to change the compatibility version of the library !"
 #endif
   
