diff -ur openmpi-1.6.5.orig/ompi/contrib/vt/vt/vtlib/vt_cudart.c openmpi-1.6.5/ompi/contrib/vt/vt/vtlib/vt_cudart.c
--- openmpi-1.6.5.orig/ompi/contrib/vt/vt/vtlib/vt_cudart.c	2013-05-29 18:54:34.000000000 +0200
+++ openmpi-1.6.5/ompi/contrib/vt/vt/vtlib/vt_cudart.c	2013-09-24 17:12:56.581152936 +0200
@@ -2722,13 +2722,12 @@
           /*  get an already created unused event */
           kernel->evt = vtDev->evtbuf_pos;
 
+#if defined(VT_CUPTI_EVENTS)
           if(!vt_cupti_events_enabled){
             /* increment buffers */
             vtDev->evtbuf_pos++;
             vtDev->buf_pos += sizeof(VTCUDAKernel);
           }
-
-#if defined(VT_CUPTI_EVENTS)
         
         /* zero CUPTI counter */
         if(vt_cupti_events_enabled){
