--- openmpi-2.1.1//opal/mca/memory/patcher/memory_patcher_component.c.orig	2018-08-30 17:25:08.562957211 +0000
+++ openmpi-2.1.1/opal/mca/memory/patcher/memory_patcher_component.c	2018-08-30 17:25:48.348597788 +0000
@@ -42,9 +42,7 @@
 #include <dlfcn.h>
 #include <assert.h>
 #include <sys/time.h>
-#if defined(HAVE_SYS_SYSCALL_H)
 #include <sys/syscall.h>
-#endif
 #if defined(HAVE_LINUX_MMAN_H)
 #include <linux/mman.h>
 #endif
@@ -162,7 +160,7 @@
     opal_mem_hooks_release_hook (start, length, true);
 
     if (!original_munmap) {
-        result = memory_patcher_syscall(SYS_munmap, start, length);
+        result = memory_patcher_syscall(__NR_munmap, start, length);
     } else {
         result = original_munmap (start, length);
     }
