diff -Naur OpenSees-3.5.0.orig/SRC/Makefile OpenSees-3.5.0/SRC/Makefile
--- OpenSees-3.5.0.orig/SRC/Makefile	2023-07-06 14:03:14.860284679 +0000
+++ OpenSees-3.5.0/SRC/Makefile	2023-07-06 14:04:19.010933469 +0000
@@ -181,7 +181,7 @@
 	$(FE)/material/uniaxial/snap/TclSnapMaterialCommand.o \
 	$(FE)/material/uniaxial/fedeas/TclFedeasMaterialCommand.o \
 	$(FE)/material/uniaxial/drain/TclDrainMaterialCommand.o \
-	$(FE)/api/elementAPI_TCL.o
+	$(FE)/api/elementAPI_Dummy.o
 
 
 skypack:
