Fix SuperLU_DIST options beeing renamed in 5.x.x version

author: Pavel Grochal (INUITS)

diff -rub --new-file OpenSees-3.2.0.after/SRC/system_of_eqn/linearSOE/sparseGEN/DistributedSuperLU.cpp OpenSees-3.2.0/SRC/system_of_eqn/linearSOE/sparseGEN/DistributedSuperLU.cpp
--- OpenSees-3.2.0.after/SRC/system_of_eqn/linearSOE/sparseGEN/DistributedSuperLU.cpp	2020-03-09 18:02:19.000000000 +0100
+++ OpenSees-3.2.0/SRC/system_of_eqn/linearSOE/sparseGEN/DistributedSuperLU.cpp	2020-11-12 15:08:44.344103344 +0100
@@ -41,7 +41,7 @@
 
 
 
-superlu_options_t options;
+superlu_dist_options_t options;
 SuperLUStat_t stat;
 SuperMatrix A;
 ScalePermstruct_t ScalePermstruct;
@@ -366,14 +366,3 @@
   opserr << "DistributedSuperLU::recvSelf(int cTag, Channel &theChannel) - END\n";
   return 0;
 }
-
-
-
-
-
-
-
-
-
-
-
