--- MAKES/Makefile.def.EC2-REDHAT-ENTERPRISE	2018-11-02 18:16:24.000000000 +0000
+++ MAKES/Makefile.def.EC2-REDHAT-ENTERPRISE	2019-02-12 17:20:57.927366071 +0000
@@ -50,7 +50,7 @@
 OPERATING_SYSTEM = LINUX
 GRAPHICS = NONE
 GRAPHIC_FLAG = -D_NOGRAPHICS
-PROGRAMMING_MODE = SEQUENTIAL
+PROGRAMMING_MODE = DISTRIBUTED_MPI
 DEBUG_MODE = NO_DEBUG
 RELIABILITY = NO_RELIABILITY
 
@@ -63,13 +63,16 @@
 # any of the libraries already leave the directory location blank AND
 # remove the directory from DIRS.
 
-BASE		= /usr/local
+BASE		= $(HOME)/local
 FE		= $(HOME)/OpenSees/SRC
 
 AMDdir       = $(HOME)/OpenSees/OTHER/AMD
-BLASdir      = $(HOME)/OpenSees/OTHER/BLAS
-CBLASdir     = $(HOME)/OpenSees/OTHER/CBLAS
-LAPACKdir    = $(HOME)/OpenSees/OTHER/LAPACK
+BLASdir      = 
+#$(HOME)/OpenSees/OTHER/BLAS
+CBLASdir     = 
+#$(HOME)/OpenSees/OTHER/CBLAS
+LAPACKdir    = 
+#$(HOME)/OpenSees/OTHER/LAPACK
 SUPERLUdir   = $(HOME)/OpenSees/OTHER/SuperLU_5.1.1/SRC
 ARPACKdir    = $(HOME)/OpenSees/OTHER/ARPACK
 UMFPACKdir   = $(HOME)/OpenSees/OTHER/UMFPACK
@@ -100,16 +103,17 @@
 NDARRAY_LIBRARY = $(HOME)/lib/libndarray.a # BJ_UCD jeremic@ucdavis.edu
 MATMOD_LIBRARY  = $(HOME)/lib/libmatmod.a  # BJ_UCD jeremic@ucdavis.edu
 BJMISC_LIBRARY  = $(HOME)/lib/libBJmisc.a  # BJ_UCD jeremic@ucdavis.edu
-LAPACK_LIBRARY  = $(HOME)/lib/libLapack.a
-BLAS_LIBRARY    = $(HOME)/lib/libBlas.a
+LAPACK_LIBRARY  = $(EBROOTOPENBLAS)/lib/libopenblas.a
+BLAS_LIBRARY    = $(EBROOTOPENBLAS)/lib/libopenblas.a -lpthread -lgomp
 SUPERLU_LIBRARY = $(HOME)/lib/libSuperLU.a
-CBLAS_LIBRARY   = $(HOME)/lib/libCBlas.a
-CSPARSE_LIBRARY   = $(HOME)/lib/libCBlas.a
+CBLAS_LIBRARY   = 
+#$(HOME)/lib/libCBlas.a
+CSPARSE_LIBRARY   = $(HOME)/lib/libCsparse.a
 ARPACK_LIBRARY  = $(HOME)/lib/libArpack.a
 AMD_LIBRARY  = $(HOME)/lib/libAMD.a
 UMFPACK_LIBRARY = $(HOME)/lib/libUmfpack.a
 METIS_LIBRARY   = $(HOME)/lib/libMetis.a
-TCL_LIBRARY = /usr/lib64/libtcl8.5.so
+TCL_LIBRARY = $(TCL_PATH)/lib/libtcl8.6.so #$(EBROOTTCL)/lib/libtcl8.5.so 
 BLITZ_LIBRARY = $(HOME)/blitz/lib/libblitz.a
 GRAPHIC_LIBRARY     = 
 
@@ -132,9 +136,9 @@
 # %---------------------------------------------------------%
 
 # Compilers
-CC++	= /usr/bin/g++
-CC      = /usr/bin/gcc
-FC	= /usr/bin/gfortran
+CC++	= mpicxx 
+CC      = mpicc
+FC	= gfortran
 
 AR		= ar 
 ARFLAGS		= cqls
@@ -174,7 +178,7 @@
 
 # Linker
 LINKER          = $(CC++)
-LINKFLAGS       = -rdynamic -Wl
+LINKFLAGS       = -rdynamic -fopenmp
 
 endif
 
