diff -ru ncbi-blast-2.2.28+-src.orig/c++/src/build-system/configure ncbi-blast-2.2.28+-src/c++/src/build-system/configure
--- ncbi-blast-2.2.28+-src.orig/c++/src/build-system/configure	2012-08-08 17:36:55.000000000 +0200
+++ ncbi-blast-2.2.28+-src/c++/src/build-system/configure	2013-01-18 13:53:21.006190873 +0100
@@ -5546,6 +5546,8 @@
     mips-sgi-irix*:KCC )
       ARCH_CFLAGS="-64"
       ;;
+    *:ICC )
+      ;;
     * )
       { { echo "$as_me:$LINENO: error: Do not know how to compile 64-bit with compiler $CXX $compiler_ver $host:$compiler" >&5
 echo "$as_me: error: Do not know how to compile 64-bit with compiler $CXX $compiler_ver $host:$compiler" >&2;}
