--- src/build-system/Makefile.in.top.orig	2017-02-17 09:14:45.000000000 -0700
+++ src/build-system/Makefile.in.top	2017-02-17 09:15:26.000000000 -0700
@@ -1,4 +1,4 @@
-# $Id: Makefile.in.top 451817 2014-11-12 16:41:55Z ucko $
+# $Id$
 # Top-level meta-makefile that simplifies building even further.
 
 # include @builddir@/Makefile.mk
@@ -49,9 +49,10 @@
 	    for x in *.a; do \
 	        $(LN_S) "$$x" "`$(BASENAME) \"$$x\" .a`-static.a"; \
 	    done
-	cd $(includedir0) && find * -name CVS -prune -o -print |\
-            cpio -pd $(pincludedir)
-	$(INSTALL) -m 644 $(incdir)/* $(pincludedir)
+	for d in $(includedir0) $(incdir); do \
+	    cd $$d && find * -name .svn -prune -o -print | \
+                cpio -pd $(pincludedir) ; \
+	done
 ## set up appropriate build and status directories somewhere under $(libdir)?
 
 install-gbench:
