Disabling optional widget, as it requires `yarn`
Author: Denis Kristak (Inuits)
diff -ruN catboost-1.2_pypi_orig/setup.py catboost-1.2_pypi/setup.py
--- catboost-1.2_pypi_orig/setup.py	2023-08-08 13:41:49.501481901 +0200
+++ catboost-1.2_pypi/setup.py	2023-08-08 13:43:39.744105439 +0200
@@ -245,7 +245,7 @@
 
     @staticmethod
     def initialize_options(command):
-        command.no_widget = False
+        command.no_widget = True
         command.prebuilt_widget = False
 
     @staticmethod
