diff -Naur bgcdist_orig/Makefile bgcdist/Makefile
--- bgcdist_orig/Makefile	1970-01-01 00:00:00.000000000 +0000
+++ bgcdist/Makefile	2019-03-19 16:03:55.360304477 +0000
@@ -0,0 +1,12 @@
+
+bgc_dependencies = bgc_func_readdata.C bgc_func_initialize.C \
+		   bgc_func_mcmc.C bgc_func_write.C bgc_func_linkage.C \
+		   bgc_func_ngs.C bgc_func_hdf5.C mvrandist.c
+
+bgc: $(bgc_dependencies)
+	h5c++ -Wall -O2 -o bgc bgc_main.C $(bgc_dependencies) -lgsl -lgslcblas
+
+clean:
+	rm -f bgc *.o
+
+
