--- Makefile.in.orig	2020-06-17 04:00:20.272566719 +0000
+++ Makefile.in	2020-06-17 04:29:41.929667302 +0000
@@ -143,7 +143,9 @@
 		${CC} ${LDFLAGS_BIN} ${DEFS} -o $@ ${@}.o -L. -l${PACKAGE} -lm
 
 testprog:	testprog.o lib${PACKAGE}.a ${OBJECTS}
-		${CC} ${LDFLAGS_BIN} ${DEFS} -o $@ ${@}.o -L. -l${PACKAGE} -lm ${LIBS}
+		${CC} ${LDFLAGS_BIN} ${DEFS} -o $@ ${@}.o -L. -l${PACKAGE} -lm ${LIBS} -Wl,-rpath,. 
+		./testprog > testprog.out.here
+		diff testprog.out testprog.out.here
 
 fpack:		fpack.o fpackutil.o lib${PACKAGE}.a ${OBJECTS}
 		${CC} ${LDFLAGS_BIN} ${DEFS} -o $@ ${@}.o fpackutil.o -L. -l${PACKAGE} -lm ${LIBS}
