use pyvcf3 instead of pyvcf which would need Python 2

author: Petr Král (INUITS)
--- cuteSV-cuteSV-v2.0.3/setup.py.orig	2023-05-12 04:58:30.000000000 +0200
+++ cuteSV-cuteSV-v2.0.3/setup.py	2023-06-12 09:48:08.251609584 +0200
@@ -21,5 +21,5 @@
     long_description = readme,
     long_description_content_type = 'text/markdown',
     zip_safe = False,
-    install_requires = ['pysam', 'Biopython', 'Cigar', 'numpy', 'pyvcf']
+    install_requires = ['pysam', 'Biopython', 'Cigar', 'numpy', 'pyvcf3']
 )
