diff -Nru alf-2.1.orig/src/Makefile alf-2.1/src/Makefile
--- alf-2.1.orig/src/Makefile   2021-03-25 15:18:13.239565000 +1100
+++ alf-2.1/src/Makefile        2021-03-25 15:24:39.392558000 +1100
@@ -16,7 +16,7 @@
 #  (Uncomment only one)
 #========================
 # GNU
-F90FLAGS = -O3 -fno-strict-overflow -cpp
+F90FLAGS = -O3 -fno-strict-overflow -cpp -mcmodel=medium
 #F90FLAGS = -O3 -fno-strict-overflow  #for odyssey
 #---------------------
 # Intel 
