# see https://github.com/easybuilders/easybuild-easyblocks/pull/1975#issuecomment-591218724
diff -ru lammps-stable_7Aug2019.org/lib/kokkos/cmake/kokkos_options.cmake lammps-stable_7Aug2019/lib/kokkos/cmake/kokkos_options.cmake
--- lammps-stable_7Aug2019.org/lib/kokkos/cmake/kokkos_options.cmake	2019-08-06 17:17:40.000000000 +0200
+++ lammps-stable_7Aug2019/lib/kokkos/cmake/kokkos_options.cmake	2020-09-23 11:24:09.301136347 +0200
@@ -78,6 +78,7 @@
 list(APPEND KOKKOS_ARCH_LIST
      None            # No architecture optimization
      AMDAVX          # (HOST) AMD chip
+     EPYC            # (HOST) AMD EPYC Zen-Core CPU
      ARMv80          # (HOST) ARMv8.0 Compatible CPU
      ARMv81          # (HOST) ARMv8.1 Compatible CPU
      ARMv8-ThunderX  # (HOST) ARMv8 Cavium ThunderX CPU
