--- lammps-stable_29Sep2021/src/KOKKOS/kokkos_old.cpp	2022-01-04 12:53:30.797462629 +0000
+++ lammps-stable_29Sep2021/src/KOKKOS/kokkos.cpp	2022-01-04 12:53:30.797462629 +0000
@@ -276,9 +276,9 @@
 
     if (gpu_aware_flag == 1 && have_gpu_aware == 0) {
       if (me == 0)
-        error->warning(FLERR,"Turning off GPU-aware MPI since it is not detected, "
-                       "use '-pk kokkos gpu/aware on' to override");
-      gpu_aware_flag = 0;
+        error->warning(FLERR,"GPU-aware MPI is not detected, "
+                       "use '-pk kokkos gpu/aware off'  in case of segfaults");
+      //gpu_aware_flag = 0;
     } else if (have_gpu_aware == -1) { // maybe we are dealing with MPICH, MVAPICH2 or some derivative?
     // MVAPICH2
 #if defined(MPICH) && defined(MVAPICH2_VERSION)
