diff -Naur lammps-stable_31Mar2017_orig/lib/atc/Makefile.lammps.installed lammps-stable_31Mar2017/lib/atc/Makefile.lammps.installed
--- lammps-stable_31Mar2017_orig/lib/atc/Makefile.lammps.installed	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/lib/atc/Makefile.lammps.installed	2017-04-29 21:13:34.885005212 +0000
@@ -1,5 +1,5 @@
 # Settings that the LAMMPS build will import when this package library is used
 
-user-atc_SYSINC =
-user-atc_SYSLIB = -lblas -llapack
-user-atc_SYSPATH =
+user-atc_SYSINC = -I${MKLROOT}/include                                 
+user-atc_SYSLIB = -L${MKLROOT}/lib/intel64 -lmkl_rt -ldl -lpthread -lm      
+user-atc_SYSPATH = -L${MKLROOT}/lib/intel64 
diff -Naur lammps-stable_31Mar2017_orig/lib/awpmd/Makefile.lammps.installed lammps-stable_31Mar2017/lib/awpmd/Makefile.lammps.installed
--- lammps-stable_31Mar2017_orig/lib/awpmd/Makefile.lammps.installed	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/lib/awpmd/Makefile.lammps.installed	2017-04-29 21:35:36.195388876 +0000
@@ -1,5 +1,5 @@
 # Settings that the LAMMPS build will import when this package library is used
 
-user-awpmd_SYSINC =
-user-awpmd_SYSLIB = -lblas -llapack
-user-awpmd_SYSPATH =
+user-awpmd_SYSINC =                   
+user-awpmd_SYSLIB = -lmkl_rt -ldl -lpthread -lm      
+user-awpmd_SYSPATH = 
diff -Naur lammps-stable_31Mar2017_orig/lib/kim/Makefile.lammps lammps-stable_31Mar2017/lib/kim/Makefile.lammps
--- lammps-stable_31Mar2017_orig/lib/kim/Makefile.lammps	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/lib/kim/Makefile.lammps	2017-07-02 19:13:25.352688405 +0000
@@ -16,7 +16,7 @@
 # Settings that the LAMMPS build will import when this package is installed
 
 
-KIM_CONFIG_HELPER = kim-api-build-config
+KIM_CONFIG_HELPER = ${EBROOTKIM}/bin/kim-api-v1-build-config
 ifeq ($(shell $(KIM_CONFIG_HELPER) --version 2> /dev/null),)
   $(error $(KIM_CONFIG_HELPER) utility is not available.  Something is wrong with your KIM API package setup)
 endif
diff -Naur lammps-stable_31Mar2017_orig/lib/quip/Makefile.lammps lammps-stable_31Mar2017/lib/quip/Makefile.lammps
--- lammps-stable_31Mar2017_orig/lib/quip/Makefile.lammps	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/lib/quip/Makefile.lammps	2017-07-31 17:49:54.088629497 +0000
@@ -2,7 +2,10 @@
 
 # include ${QUIP_ROOT}/Makefiles/Makefile.${QUIP_ARCH}
 
-F95=$(shell egrep 'F95[ ]*=' ${QUIP_ROOT}/arch/Makefile.${QUIP_ARCH} | sed 's/.*F95[ ]*=[ ]*//')
+QUIP_ROOT=${EBROOTQUIP}
+QUIP_ARCH=linux_x86_64_ifort_icc
+
+FC=ifort
 
 ifeq (${QUIP_ROOT},)
 $(error Environment variable QUIP_ROOT must be set.)
@@ -19,12 +22,12 @@
 quip_SYSLIB += ${NETCDF_SYSLIBS}
 quip_SYSLIB += ${MATH_LINKOPTS}
 
-ifeq (${F95},gfortran)
+ifeq (${FC},gfortran)
 	quip_SYSLIB += -lgfortran
-else ifeq (${F95},ifort)
+else ifeq (${FC},ifort)
 	quip_SYSLIB += -lifcore -lifport
 else
-$(error fortran compiler >>${F95}<< not recognised. Edit lib/quip/Makefile.lammps to specify the fortran library your linker should link to)
+$(error fortran compiler >>${FC}<< not recognised. Edit lib/quip/Makefile.lammps to specify the fortran library your linker should link to)
 endif
 
 quip_SYSPATH = -L${QUIP_ROOT}/build/${QUIP_ARCH}
diff -Naur lammps-stable_31Mar2017_orig/lib/voronoi/Makefile.lammps lammps-stable_31Mar2017/lib/voronoi/Makefile.lammps
--- lammps-stable_31Mar2017_orig/lib/voronoi/Makefile.lammps	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/lib/voronoi/Makefile.lammps	2017-08-02 16:30:44.494425398 +0000
@@ -1,5 +1,5 @@
 # Settings that the LAMMPS build will import when this package library is used
 
-voronoi_SYSINC = 
-voronoi_SYSLIB = 
-voronoi_SYSPATH = 
+voronoi_SYSINC = ${EBROOTVOROPLUSPLUS}/include 
+voronoi_SYSLIB = -lvoro++ 
+voronoi_SYSPATH = ${EBROOTVOROPLUSPLUS}/lib
diff -Naur lammps-stable_31Mar2017_orig/lib/vtk/Makefile.lammps lammps-stable_31Mar2017/lib/vtk/Makefile.lammps
--- lammps-stable_31Mar2017_orig/lib/vtk/Makefile.lammps	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/lib/vtk/Makefile.lammps	2017-07-27 17:01:30.177955244 +0000
@@ -1,13 +1,7 @@
 # Settings that the LAMMPS build will import when this package library is used
 #
-# settings for VTK-5.8.0 on RHEL/CentOS 6.x
- vtk_SYSINC = -I/usr/include/vtk
- vtk_SYSLIB = -lvtkCommon -lvtkIO 
- vtk_SYSPATH = -L/usr/lib64/vtk
-#
-# settings for VTK 6.2.0 on Fedora 23
-#vtk_SYSINC = -I/usr/include/vtk
-#vtk_SYSLIB = -lvtkCommonCore -lvtkIOCore -lvtkCommonDataModel -lvtkIOXML -lvtkIOLegacy -lvtkIOParallelXML
-#vtk_SYSPATH = -L/usr/lib64/vtk
+vtk_SYSINC = -I${EBROOTVTK}/include/vtk-8.0
+vtk_SYSLIB = -lvtkCommonCore-8.0 -lvtkIOCore-8.0 -lvtkIOXML-8.0 -lvtkIOLegacy-8.0 -lvtkCommonDataModel-8.0  
+vtk_SYSPATH = -L${EBROOTVTK}/lib
 #
 
diff -Naur lammps-stable_31Mar2017_orig/src/MAKE/Makefile.mpi lammps-stable_31Mar2017/src/MAKE/Makefile.mpi
--- lammps-stable_31Mar2017_orig/src/MAKE/Makefile.mpi	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/src/MAKE/Makefile.mpi	2017-07-02 16:39:44.231278285 +0000
@@ -28,7 +28,7 @@
 # LAMMPS ifdef settings
 # see possible settings in Section 2.2 (step 4) of manual
 
-LMP_INC =	-DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64
+LMP_INC = -DFFTW -DLAMMPS_EXCEPTIONS -DPACK_POINTER -DLAMMPS_PNG -DLAMMPS_JPEG -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64 -DLAMMPS_SMALLBIG
 
 # MPI library
 # see discussion in Section 2.2 (step 5) of manual
@@ -61,9 +61,9 @@
 # PATH = path(s) for JPEG library and/or PNG library
 # LIB = name(s) of JPEG library and/or PNG library
 
-JPG_INC =       
-JPG_PATH = 	
-JPG_LIB =	
+JPG_INC = -DLAMMPS_JPEG -DLAMMPS_PNG       
+JPG_PATH = -L/usr/lib64	
+JPG_LIB = -ljpeg -lpng 
 
 # ---------------------------------------------------------------------
 # build rules and dependencies
diff -Naur lammps-stable_31Mar2017_orig/src/MAKE/OPTIONS/Makefile.icc_openmpi lammps-stable_31Mar2017/src/MAKE/OPTIONS/Makefile.icc_openmpi
--- lammps-stable_31Mar2017_orig/src/MAKE/OPTIONS/Makefile.icc_openmpi	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/src/MAKE/OPTIONS/Makefile.icc_openmpi	2017-07-17 15:38:44.796580184 +0000
@@ -29,7 +29,7 @@
 # LAMMPS ifdef settings
 # see possible settings in Section 2.2 (step 4) of manual
 
-LMP_INC =	-DLAMMPS_GZIP
+LMP_INC = -DFFTW -DLAMMPS_EXCEPTIONS -DPACK_POINTER -DLAMMPS_PNG -DLAMMPS_JPEG -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64 -DLAMMPS_SMALLBIG
 
 # MPI library
 # see discussion in Section 2.2 (step 5) of manual
@@ -62,9 +62,9 @@
 # PATH = path(s) for JPEG library and/or PNG library
 # LIB = name(s) of JPEG library and/or PNG library
 
-JPG_INC =       
-JPG_PATH = 	
-JPG_LIB =	
+JPG_INC = -DLAMMPS_JPEG -DLAMMPS_PNG       
+JPG_PATH = -L/usr/lib64	
+JPG_LIB = -ljpeg -lpng	
 
 # ---------------------------------------------------------------------
 # build rules and dependencies
diff -Naur lammps-stable_31Mar2017_orig/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi lammps-stable_31Mar2017/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi
--- lammps-stable_31Mar2017_orig/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi	2017-07-17 15:41:54.440770294 +0000
@@ -31,7 +31,7 @@
 # LAMMPS ifdef settings
 # see possible settings in Section 2.2 (step 4) of manual
 
-LMP_INC =	-DLAMMPS_GZIP
+LMP_INC =	-DFFTW -DLAMMPS_EXCEPTIONS -DPACK_POINTER -DLAMMPS_PNG -DLAMMPS_JPEG -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64 -DLAMMPS_SMALLBIG
 
 # MPI library
 # see discussion in Section 2.2 (step 5) of manual
@@ -64,9 +64,9 @@
 # PATH = path(s) for JPEG library and/or PNG library
 # LIB = name(s) of JPEG library and/or PNG library
 
-JPG_INC =       
-JPG_PATH = 	
-JPG_LIB =	
+JPG_INC = -DLAMMPS_JPEG -DLAMMPS_PNG      
+JPG_PATH = -L/usr/lib64 	
+JPG_LIB = -ljpeg -lpng	
 
 # ---------------------------------------------------------------------
 # build rules and dependencies
diff -Naur lammps-stable_31Mar2017_orig/src/MAKE/OPTIONS/Makefile.kokkos_omp lammps-stable_31Mar2017/src/MAKE/OPTIONS/Makefile.kokkos_omp
--- lammps-stable_31Mar2017_orig/src/MAKE/OPTIONS/Makefile.kokkos_omp	2017-03-30 16:07:27.000000000 +0000
+++ lammps-stable_31Mar2017/src/MAKE/OPTIONS/Makefile.kokkos_omp	2017-07-17 15:44:31.582273964 +0000
@@ -29,7 +29,7 @@
 # LAMMPS ifdef settings
 # see possible settings in Section 2.2 (step 4) of manual
 
-LMP_INC =	-DLAMMPS_GZIP
+LMP_INC = -DFFTW -DLAMMPS_EXCEPTIONS -DPACK_POINTER -DLAMMPS_PNG -DLAMMPS_JPEG -DLAMMPS_GZIP -DLAMMPS_MEMALIGN=64 -DLAMMPS_SMALLBIG
 
 # MPI library
 # see discussion in Section 2.2 (step 5) of manual
@@ -62,9 +62,9 @@
 # PATH = path(s) for JPEG library and/or PNG library
 # LIB = name(s) of JPEG library and/or PNG library
 
-JPG_INC =       
-JPG_PATH = 	
-JPG_LIB =	
+JPG_INC = -DLAMMPS_JPEG -DLAMMPS_PNG
+JPG_PATH = -L/usr/lib64
+JPG_LIB = -ljpeg -lpng
 
 # ---------------------------------------------------------------------
 # build rules and dependencies
 # ---------------------------------------------------------------------
