change the cuda aware definition
--- lammps-stable_29Sep2021_org/src/KOKKOS/kokkos.cpp	2022-07-21 11:53:51.022923873 +0100
+++ lammps-stable_29Sep2021/src/KOKKOS/kokkos.cpp	2022-07-21 14:35:00.199045080 +0100
@@ -47,10 +47,8 @@
 
 #include <mpi-ext.h>
 
-#if defined(MPIX_CUDA_AWARE_SUPPORT) && MPIX_CUDA_AWARE_SUPPORT
-static int have_gpu_aware = 1;
-#elif defined(MPIX_CUDA_AWARE_SUPPORT) && !MPIX_CUDA_AWARE_SUPPORT
-static int have_gpu_aware = 0;
+#if defined(MPIX_CUDA_AWARE_SUPPORT)
+static int have_gpu_aware = MPIX_Query_cuda_support();
 #else
 GPU_AWARE_UNKNOWN
 #endif // defined(MPIX_CUDA_AWARE_SUPPORT)
