diff -ru lammps-stable_29Oct2020.orig/cmake/CMakeLists.txt lammps-stable_29Oct2020/cmake/CMakeLists.txt
--- lammps-stable_29Oct2020.orig/cmake/CMakeLists.txt	2021-03-23 21:21:09.152815888 +0000
+++ lammps-stable_29Oct2020/cmake/CMakeLists.txt	2021-03-23 21:23:31.439621286 +0000
@@ -65,7 +65,7 @@
   if(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 17.3 OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 17.4)
     set(CMAKE_TUNE_DEFAULT "-xCOMMON-AVX512")
   else()
-    set(CMAKE_TUNE_DEFAULT "-xHost")
+    set(CMAKE_TUNE_DEFAULT "")
   endif()
 endif()
 
diff -ru lammps-stable_29Oct2020.orig/cmake/Modules/Packages/USER-INTEL.cmake lammps-stable_29Oct2020/cmake/Modules/Packages/USER-INTEL.cmake
--- lammps-stable_29Oct2020.orig/cmake/Modules/Packages/USER-INTEL.cmake	2021-03-23 21:21:09.154815899 +0000
+++ lammps-stable_29Oct2020/cmake/Modules/Packages/USER-INTEL.cmake	2021-03-23 21:23:57.845757572 +0000
@@ -67,7 +67,7 @@
   if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     message(FATAL_ERROR "Must use Intel compiler with USER-INTEL for KNL architecture")
   endif()
-  set(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -xHost -qopenmp -qoffload")
+  set(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -qopenmp -qoffload")
   set(MIC_OPTIONS "-qoffload-option,mic,compiler,\"-fp-model fast=2 -mGLOB_default_function_attrs=\\\"gather_scatter_loop_unroll=4\\\"\"")
   target_compile_options(lammps PRIVATE -xMIC-AVX512 -qoffload -fno-alias -ansi-alias -restrict -qoverride-limits ${MIC_OPTIONS})
   target_compile_definitions(lammps PRIVATE -DLMP_INTEL_OFFLOAD)
diff -ru lammps-stable_29Oct2020.orig/doc/src/Build_extras.rst lammps-stable_29Oct2020/doc/src/Build_extras.rst
--- lammps-stable_29Oct2020.orig/doc/src/Build_extras.rst	2021-03-23 21:21:09.800819546 +0000
+++ lammps-stable_29Oct2020/doc/src/Build_extras.rst	2021-03-23 21:25:14.798142944 +0000
@@ -1349,7 +1349,7 @@
 
       .. code-block:: make
 
-         OPTFLAGS =      -xHost -O2 -fp-model fast=2 -no-prec-div -qoverride-limits -qopt-zmm-usage=high
+         OPTFLAGS =      -O2 -fp-model fast=2 -no-prec-div -qoverride-limits -qopt-zmm-usage=high
          CCFLAGS =       -g -qopenmp -DLAMMPS_MEMALIGN=64 -no-offload -fno-alias -ansi-alias -restrict $(OPTFLAGS)
          LINKFLAGS =     -g -qopenmp $(OPTFLAGS)
          LIB =           -ltbbmalloc
diff -ru lammps-stable_29Oct2020.orig/doc/src/Howto_cmake.rst lammps-stable_29Oct2020/doc/src/Howto_cmake.rst
--- lammps-stable_29Oct2020.orig/doc/src/Howto_cmake.rst	2021-03-23 21:21:09.534818044 +0000
+++ lammps-stable_29Oct2020/doc/src/Howto_cmake.rst	2021-03-23 21:22:13.945183176 +0000
@@ -129,8 +129,8 @@
    -- Found FFMPEG: /usr/bin/ffmpeg
    -- Performing Test COMPILER_SUPPORTS-ffast-math
    -- Performing Test COMPILER_SUPPORTS-ffast-math - Success
-   -- Performing Test COMPILER_SUPPORTS-march=native
-   -- Performing Test COMPILER_SUPPORTS-march=native - Success
+   -- Performing Test COMPILER_SUPPORTS
+   -- Performing Test COMPILER_SUPPORTS - Success
    -- Looking for C++ include cmath
    -- Looking for C++ include cmath - found
    -- Generating style_angle.h...
@@ -154,7 +154,7 @@
          Version:       8.2.0
          C++ Flags:     -O2 -g -DNDEBUG
          Defines:       LAMMPS_SMALLBIG;LAMMPS_MEMALIGN=64;LAMMPS_JPEG;LAMMPS_PNG;LAMMPS_GZIP;LAMMPS_FFMPEG
-         Options:       -ffast-math;-march=native
+         Options:       -ffast-math;
    -- <<< Linker flags: >>>
    -- Executable name:  lmp
    -- Static library flags:
diff -ru lammps-stable_29Oct2020.orig/src/MAKE/MACHINES/Makefile.aarch64_g++_openmpi_armpl lammps-stable_29Oct2020/src/MAKE/MACHINES/Makefile.aarch64_g++_openmpi_armpl
--- lammps-stable_29Oct2020.orig/src/MAKE/MACHINES/Makefile.aarch64_g++_openmpi_armpl	2021-03-23 21:21:09.250816441 +0000
+++ lammps-stable_29Oct2020/src/MAKE/MACHINES/Makefile.aarch64_g++_openmpi_armpl	2021-03-23 21:22:13.946183181 +0000
@@ -8,7 +8,7 @@
 
 export OMPI_CXX = g++
 CC =		mpicxx
-CCFLAGS =	-O3 -march=native -mcpu=native
+CCFLAGS =	-O3  -mcpu=native
 SHFLAGS =	-fPIC
 DEPFLAGS =	-M
 
diff -ru lammps-stable_29Oct2020.orig/src/MAKE/MACHINES/Makefile.aarch64_g++_serial_armpl lammps-stable_29Oct2020/src/MAKE/MACHINES/Makefile.aarch64_g++_serial_armpl
--- lammps-stable_29Oct2020.orig/src/MAKE/MACHINES/Makefile.aarch64_g++_serial_armpl	2021-03-23 21:21:09.250816441 +0000
+++ lammps-stable_29Oct2020/src/MAKE/MACHINES/Makefile.aarch64_g++_serial_armpl	2021-03-23 21:22:13.946183181 +0000
@@ -7,7 +7,7 @@
 # specify flags and libraries needed for your compiler
 
 CC =		g++
-CCFLAGS =	-O3 -march=native -mcpu=native
+CCFLAGS =	-O3  -mcpu=native
 SHFLAGS =	-fPIC
 DEPFLAGS =	-M
 
diff -ru lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_coprocessor lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_coprocessor
--- lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_coprocessor	2021-03-23 21:21:09.252816452 +0000
+++ lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_coprocessor	2021-03-23 21:26:00.308384519 +0000
@@ -8,7 +8,7 @@
 
 CC =		mpiicpc -std=c++11
 MIC_OPT =       -qoffload-option,mic,compiler,"-fp-model fast=2 -mGLOB_default_function_attrs=\"gather_scatter_loop_unroll=4\""
-OPTFLAGS =      -xHost -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
+OPTFLAGS =      -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
                 -qopt-zmm-usage=high $(MIC_OPT)
 CCFLAGS =	-qopenmp -qoffload -ansi-alias -restrict \
                 -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG -DLMP_INTEL_OFFLOAD \
@@ -17,7 +17,7 @@
 DEPFLAGS =	-M
 
 LINK =		mpiicpc -std=c++11
-LINKFLAGS =	-O2 -xHost -qopenmp -qoffload -L$(MKLROOT)/lib/intel64/
+LINKFLAGS =	-O2 -qopenmp -qoffload -L$(MKLROOT)/lib/intel64/
 LIB =           -ltbbmalloc -lmkl_intel_ilp64 -lmkl_sequential -lmkl_core
 SIZE =		size
 
diff -ru lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_cpu_intelmpi lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_cpu_intelmpi
--- lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_cpu_intelmpi	2021-03-23 21:21:09.253816458 +0000
+++ lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_cpu_intelmpi	2021-03-23 21:26:22.533504493 +0000
@@ -7,7 +7,7 @@
 # specify flags and libraries needed for your compiler
 
 CC =		mpiicpc -std=c++11
-OPTFLAGS =      -xHost -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
+OPTFLAGS =      -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
                 -qopt-zmm-usage=high
 CCFLAGS =	-qopenmp -qno-offload -ansi-alias -restrict \
                 -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG $(OPTFLAGS) \
diff -ru lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_cpu_mpich lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_cpu_mpich
--- lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_cpu_mpich	2021-03-23 21:21:09.253816458 +0000
+++ lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_cpu_mpich	2021-03-23 21:24:46.916995558 +0000
@@ -7,7 +7,7 @@
 # specify flags and libraries needed for your compiler
 
 CC =		mpicxx -cxx=icc -std=c++11
-OPTFLAGS =      -xHost -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
+OPTFLAGS =      -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
                 -qopt-zmm-usage=high
 CCFLAGS =	-qopenmp -qno-offload -ansi-alias -restrict \
                 -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG $(OPTFLAGS) \
diff -ru lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi
--- lammps-stable_29Oct2020.orig/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi	2021-03-23 21:21:09.253816458 +0000
+++ lammps-stable_29Oct2020/src/MAKE/OPTIONS/Makefile.intel_cpu_openmpi	2021-03-23 21:24:25.981893897 +0000
@@ -8,7 +8,7 @@
 
 export OMPI_CXX = icc
 CC =		mpicxx -std=c++11
-OPTFLAGS =      -xHost -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
+OPTFLAGS =      -O2 -fp-model fast=2 -no-prec-div -qoverride-limits \
                 -qopt-zmm-usage=high
 CCFLAGS =	-qopenmp -qno-offload -ansi-alias -restrict \
                 -DLMP_INTEL_USELRT -DLMP_USE_MKL_RNG $(OPTFLAGS) \
