--- lammps-10Feb15/lib/meam/Makefile.lammps.ifort_orig	2015-04-22 08:12:15.171696000 +0200
+++ lammps-10Feb15/lib/meam/Makefile.lammps.ifort	2015-04-22 08:18:57.638595000 +0200
@@ -1,5 +1,5 @@
 # Settings that the LAMMPS build will import when this package library is used
 
 meam_SYSINC =
-meam_SYSLIB = -lifcore -lsvml -lompstub -limf
-meam_SYSPATH = -L/opt/intel/fce/10.0.023/lib
+meam_SYSLIB = -lifcore -lsvml -liompstubs5 -limf
+meam_SYSPATH =
