Remove single test with circular dependency on PySCF.
Author: micketeer@gmail.com
--- testsuite/test_3c2e.py.orig	2021-06-14 13:15:11.316287035 +0200
+++ testsuite/test_3c2e.py	2021-06-14 13:11:51.777769636 +0200
@@ -307,9 +307,10 @@
              ):
         test_int3c2e_sph(*f)
     if "--quick" not in sys.argv:
-        for f in (('cint3c2e', 'cint3c2e_sph', 4412.363002589966, 1, 10),
-                 ):
-            test_int3c2e_spinor(*f)
+#        for f in (('cint3c2e', 'cint3c2e_sph', 4412.363002589966, 1, 10),
+#                 ):
+#            test_int3c2e_spinor(*f)
+         pass
 
 #    for f in (('cint2c2e_sph', 'cint2e_sph', 782.3104849606677, 1, 10),
 #              ('cint2c2e_ip1_sph', 'cint2e_ip1_sph', 394.6515972715189, 3, 10),
