What: Fix incorrect path to the shared library
Author: maxim-mnasterov (SURF)

diff -Nru libcint-6.1.2.orig/testsuite/test_3c2e.py libcint-6.1.2/testsuite/test_3c2e.py
--- libcint-6.1.2.orig/testsuite/test_3c2e.py	2024-10-04 16:09:36.042124000 +0200
+++ libcint-6.1.2/testsuite/test_3c2e.py	2024-10-04 16:12:57.158040824 +0200
@@ -13,7 +13,7 @@
 import ctypes
 import numpy
 
-_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../build/libcint.so')))
+_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../libcint.so')))
 
 PTR_LIGHT_SPEED    = 0
 PTR_COMMON_ORIG    = 1
diff -Nru libcint-6.1.2.orig/testsuite/test_c2s.py libcint-6.1.2/testsuite/test_c2s.py
--- libcint-6.1.2.orig/testsuite/test_c2s.py	2024-10-04 16:09:36.042595000 +0200
+++ libcint-6.1.2/testsuite/test_c2s.py	2024-10-04 16:13:11.143154981 +0200
@@ -3,7 +3,7 @@
 import ctypes
 import numpy
 
-_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
+_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../..')))
 
 
 PTR_EXPCUTOFF      = 0
diff -Nru libcint-6.1.2.orig/testsuite/test_cart2sph.py libcint-6.1.2/testsuite/test_cart2sph.py
--- libcint-6.1.2.orig/testsuite/test_cart2sph.py	2024-10-04 16:09:36.043003000 +0200
+++ libcint-6.1.2/testsuite/test_cart2sph.py	2024-10-04 16:13:35.057998480 +0200
@@ -10,7 +10,7 @@
 sys.path.insert(0, os.path.abspath(os.path.join(__file__, '../../scripts')))
 import cart2sph
 
-_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../build/libcint.so')))
+_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../libcint.so')))
 
 pauli = np.array([[[0., 1.],
                    [1., 0.]],  # x
diff -Nru libcint-6.1.2.orig/testsuite/test_cint4c1e.py libcint-6.1.2/testsuite/test_cint4c1e.py
--- libcint-6.1.2.orig/testsuite/test_cint4c1e.py	2024-10-04 16:09:36.043792000 +0200
+++ libcint-6.1.2/testsuite/test_cint4c1e.py	2024-10-04 16:13:48.171695000 +0200
@@ -13,7 +13,7 @@
 import ctypes
 import numpy
 
-_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
+_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../..')))
 
 
 PTR_LIGHT_SPEED    = 0
diff -Nru libcint-6.1.2.orig/testsuite/test_cint.py libcint-6.1.2/testsuite/test_cint.py
--- libcint-6.1.2.orig/testsuite/test_cint.py	2024-10-04 16:09:36.043395000 +0200
+++ libcint-6.1.2/testsuite/test_cint.py	2024-10-04 16:12:23.988960299 +0200
@@ -13,7 +13,7 @@
 import ctypes
 import numpy
 
-_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../build/libcint.so')))
+_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../libcint.so')))
 
 
 PTR_EXPCUTOFF      = 0
diff -Nru libcint-6.1.2.orig/testsuite/test_int1e_grids.py libcint-6.1.2/testsuite/test_int1e_grids.py
--- libcint-6.1.2.orig/testsuite/test_int1e_grids.py	2024-10-04 16:09:36.045513000 +0200
+++ libcint-6.1.2/testsuite/test_int1e_grids.py	2024-10-04 16:14:20.427552000 +0200
@@ -13,7 +13,7 @@
 import ctypes
 import numpy
 
-_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../build/libcint.so')))
+_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../libcint.so')))
 
 PTR_EXPCUTOFF      = 0
 PTR_COMMON_ORIG    = 1
diff -Nru libcint-6.1.2.orig/testsuite/test_int1e.py libcint-6.1.2/testsuite/test_int1e.py
--- libcint-6.1.2.orig/testsuite/test_int1e.py	2024-10-04 16:09:36.045015000 +0200
+++ libcint-6.1.2/testsuite/test_int1e.py	2024-10-04 16:14:31.649911000 +0200
@@ -5,7 +5,7 @@
 import ctypes
 import numpy
 
-_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
+_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../..')))
 #_cint4 = ctypes.cdll.LoadLibrary('libcint.so.4')
 
 from pyscf import gto, lib
diff -Nru libcint-6.1.2.orig/testsuite/test_int2c2e.py libcint-6.1.2/testsuite/test_int2c2e.py
--- libcint-6.1.2.orig/testsuite/test_int2c2e.py	2024-10-04 16:09:36.045952547 +0200
+++ libcint-6.1.2/testsuite/test_int2c2e.py	2024-10-04 16:14:45.424744884 +0200
@@ -3,7 +3,7 @@
 import ctypes
 import numpy
 
-_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../build/libcint.so')))
+_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../libcint.so')))
 
 from pyscf import gto, lib
 
diff -Nru libcint-6.1.2.orig/testsuite/test_int2e_f12_etc.py libcint-6.1.2/testsuite/test_int2e_f12_etc.py
--- libcint-6.1.2.orig/testsuite/test_int2e_f12_etc.py	2024-10-04 16:09:36.046726088 +0200
+++ libcint-6.1.2/testsuite/test_int2e_f12_etc.py	2024-10-04 16:14:57.223888132 +0200
@@ -3,7 +3,7 @@
 import ctypes
 import numpy
 
-_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../build/libcint.so')))
+_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../libcint.so')))
 
 from pyscf import gto, lib
 
diff -Nru libcint-6.1.2.orig/testsuite/test_int2e.py libcint-6.1.2/testsuite/test_int2e.py
--- libcint-6.1.2.orig/testsuite/test_int2e.py	2024-10-04 16:09:36.046362000 +0200
+++ libcint-6.1.2/testsuite/test_int2e.py	2024-10-04 16:15:10.386953000 +0200
@@ -5,7 +5,7 @@
 import ctypes
 import numpy
 
-_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
+_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../..')))
 #_cint4 = ctypes.cdll.LoadLibrary('libcint.so.4')
 
 from pyscf import gto, lib
diff -Nru libcint-6.1.2.orig/testsuite/test_int3c1e.py libcint-6.1.2/testsuite/test_int3c1e.py
--- libcint-6.1.2.orig/testsuite/test_int3c1e.py	2024-10-04 16:09:36.047153000 +0200
+++ libcint-6.1.2/testsuite/test_int3c1e.py	2024-10-04 16:15:23.148032000 +0200
@@ -3,7 +3,7 @@
 import ctypes
 import numpy
 
-_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../../build')))
+_cint = numpy.ctypeslib.load_library('libcint', os.path.abspath(os.path.join(__file__, '../..')))
 #_cint4 = ctypes.cdll.LoadLibrary('libcint.so.4')
 
 from pyscf import gto, lib
diff -Nru libcint-6.1.2.orig/testsuite/test_int3c2e.py libcint-6.1.2/testsuite/test_int3c2e.py
--- libcint-6.1.2.orig/testsuite/test_int3c2e.py	2024-10-04 16:09:36.047561000 +0200
+++ libcint-6.1.2/testsuite/test_int3c2e.py	2024-10-04 16:15:33.932008000 +0200
@@ -3,7 +3,7 @@
 import ctypes
 import numpy
 
-_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../build/libcint.so')))
+_cint = ctypes.CDLL(os.path.abspath(os.path.join(__file__, '../../libcint.so')))
 
 from pyscf import gto, lib
 
