From: John Dey Fred Hutch
boost file extension API needs updating
diff -ruN shapemapper2-2.3.orig/internals/cpp-src/src/MutationCounter.cpp shapemapper2-2.3/internals/cpp-src/src/MutationCounter.cpp
--- shapemapper2-2.3.orig/internals/cpp-src/src/MutationCounter.cpp	2024-11-22 13:54:45.000000000 -0800
+++ shapemapper2-2.3/internals/cpp-src/src/MutationCounter.cpp	2025-05-20 13:00:32.181181000 -0700
@@ -112,7 +112,7 @@
                 throw std::runtime_error("ERROR: Could not open input file " + filename +
                                          " - unknown error.\nCheck file and folder permissions.");
             }
-            if (BF::extension(BF::path(filename)) == ".gz") {
+            if (BF::path(filename).extension() == ".gz") {
                 // decompress gzip if file looks compressed
                 files.back()->push(BI::gzip_decompressor());
             }
@@ -142,7 +142,7 @@
                         "ERROR: Could not open output file " + out_names[i] + "\nCheck file and folder permissions.");
             }
             std::unique_ptr <BI::filtering_ostream> out(new BI::filtering_ostream);
-            if (BF::extension(BF::path(out_names[i])) == ".gz") {
+            if (BF::path(out_names[i]).extension() == ".gz") {
                 // compress using gzip if requested
                 out->push(BI::gzip_compressor());
             }
diff -ruN shapemapper2-2.3.orig/internals/cpp-src/src/MutationParser.cpp shapemapper2-2.3/internals/cpp-src/src/MutationParser.cpp
--- shapemapper2-2.3.orig/internals/cpp-src/src/MutationParser.cpp	2024-11-22 13:54:45.000000000 -0800
+++ shapemapper2-2.3/internals/cpp-src/src/MutationParser.cpp	2025-05-20 12:42:42.821813000 -0700
@@ -891,7 +891,7 @@
         BI::filtering_istream in;
         // universal newline support filter
         in.push(BI::newline_filter(BI::newline::posix));
-        if (BF::extension(BF::path(filename)) == ".gz") {
+        if (BF::path(filename).extension() == ".gz") {
             // decompress gzip if file looks compressed
             in.push(BI::gzip_decompressor());
         }
@@ -904,7 +904,7 @@
         }
 
         BI::filtering_ostream out;
-        if (BF::extension(BF::path(outname)) == ".gz") {
+        if (BF::path(outname).extension() == ".gz") {
             // compress using gzip if requested
             out.push(BI::gzip_compressor());
         }
diff -ruN shapemapper2-2.3.orig/internals/cpp-src/src/ReadTrimmer.cpp shapemapper2-2.3/internals/cpp-src/src/ReadTrimmer.cpp
--- shapemapper2-2.3.orig/internals/cpp-src/src/ReadTrimmer.cpp	2024-11-22 13:54:45.000000000 -0800
+++ shapemapper2-2.3/internals/cpp-src/src/ReadTrimmer.cpp	2025-05-20 12:43:39.447289000 -0700
@@ -109,7 +109,7 @@
         BI::filtering_istream in;
         // universal newline support filter
         in.push(BI::newline_filter(BI::newline::posix));
-        if (BF::extension(BF::path(filename)) == ".gz") {
+        if (BF::path(filename).extension() == ".gz") {
             // decompress gzip if file looks compressed
             in.push(BI::gzip_decompressor());
         }
@@ -128,7 +128,7 @@
                     "ERROR: Could not open output file " + outname + "\nCheck file and folder permissions.");
         }
         BI::filtering_ostream out;
-        if (BF::extension(BF::path(outname)) == ".gz") {
+        if (BF::path(outname).extension() == ".gz") {
             // compress using gzip if requested
             out.push(BI::gzip_compressor());
         }
