diff -ur kaiju-1.5.0.orig/src/bwt/Makefile kaiju-1.5.0/src/bwt/Makefile
--- kaiju-1.5.0.orig/src/bwt/Makefile	2017-02-20 16:22:04.000000000 +0100
+++ kaiju-1.5.0/src/bwt/Makefile	2017-03-15 11:20:22.174636137 +0100
@@ -1,7 +1,7 @@
-CC = gcc
+CC ?= gcc
 #CFLAGS  = -g
-CFLAGS = -O3 -g -Wno-unused-result
-LDLIBS = -lpthread -lm
+CFLAGS ?= -O3 -g -Wno-unused-result
+LDLIBS ?= -lpthread -lm
 
 all: mkbwt mkfmi Makefile
 
diff -ur kaiju-1.5.0.orig/src/makefile kaiju-1.5.0/src/makefile
--- kaiju-1.5.0.orig/src/makefile	2017-02-20 16:22:04.000000000 +0100
+++ kaiju-1.5.0/src/makefile	2017-03-15 11:20:10.710515853 +0100
@@ -1,8 +1,8 @@
-CC = gcc
-CXX = g++
-CFLAGS = -g -O3 -DNDEBUG -Wall -Wno-uninitialized
-CXXFLAGS = -ansi -pedantic -O3 -pthread -std=c++11 -g -DNDEBUG -Wall -Wconversion -Wno-unused-function
-LDLIBS = -lpthread
+CC ?= gcc
+CXX ?= g++
+CFLAGS ?= -g -O3 -DNDEBUG -Wall -Wno-uninitialized
+CXXFLAGS ?= -ansi -pedantic -O3 -pthread -std=c++11 -g -DNDEBUG -Wall -Wconversion -Wno-unused-function
+LDLIBS ?= -lpthread
 INCLUDES	= -I./include/ProducerConsumerQueue/src -I./include/ncbi-blast+
 
 BLASTOBJS = include/ncbi-blast+/algo/blast/core/pattern.o \
