Use EasyBuild provided dependencies
Author: Cintia Willemyns (Vrije Universiteit Brussel)
--- mujoco-3.1.4.orig/mujoco/CMakeLists.txt	2024-04-17 15:42:21.449694000 +0200
+++ mujoco-3.1.4/mujoco/CMakeLists.txt	2024-04-17 15:43:56.347482395 +0200
@@ -132,7 +132,7 @@
 set(MUJOCO_PYTHON_ABSL_TARGETS absl::core_headers absl::flat_hash_map absl::span)
 findorfetch(
   USE_SYSTEM_PACKAGE
-  OFF
+  ON
   PACKAGE_NAME
   absl
   LIBRARY_NAME
@@ -165,7 +165,7 @@
 add_compile_definitions(EIGEN_MPL2_ONLY)
 findorfetch(
   USE_SYSTEM_PACKAGE
-  OFF
+  ON
   PACKAGE_NAME
   Eigen3
   LIBRARY_NAME
@@ -183,6 +183,7 @@
 option(MUJOCO_PYTHON_USE_SYSTEM_PYBIND11 "Use installed pybind11 version." OFF)
 findorfetch(
   USE_SYSTEM_PACKAGE
+  ON
   MUJOCO_PYTHON_USE_SYSTEM_PYBIND11
   PACKAGE_NAME
   pybind11
