diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclAdjustFasta orthomclSoftware-v2.0.8/bin/orthomclAdjustFasta
--- orthomclSoftware-v2.0.8.orig/bin/orthomclAdjustFasta	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclAdjustFasta	2013-05-16 18:29:23.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclBlastParser orthomclSoftware-v2.0.8/bin/orthomclBlastParser
--- orthomclSoftware-v2.0.8.orig/bin/orthomclBlastParser	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclBlastParser	2013-05-16 18:29:27.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclDropSchema orthomclSoftware-v2.0.8/bin/orthomclDropSchema
--- orthomclSoftware-v2.0.8.orig/bin/orthomclDropSchema	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclDropSchema	2013-05-16 18:29:35.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use DBI;
 use FindBin;
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclDumpPairsFiles orthomclSoftware-v2.0.8/bin/orthomclDumpPairsFiles
--- orthomclSoftware-v2.0.8.orig/bin/orthomclDumpPairsFiles	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclDumpPairsFiles	2013-05-16 18:29:37.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use DBI;
 use FindBin;
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclExtractProteinIdsFromGroupsFile orthomclSoftware-v2.0.8/bin/orthomclExtractProteinIdsFromGroupsFile
--- orthomclSoftware-v2.0.8.orig/bin/orthomclExtractProteinIdsFromGroupsFile	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclExtractProteinIdsFromGroupsFile	2013-05-16 18:29:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 use strict;
 
 usage() unless scalar(@ARGV) == 1;
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclExtractProteinPairsFromGroupsFile orthomclSoftware-v2.0.8/bin/orthomclExtractProteinPairsFromGroupsFile
--- orthomclSoftware-v2.0.8.orig/bin/orthomclExtractProteinPairsFromGroupsFile	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclExtractProteinPairsFromGroupsFile	2013-05-16 18:29:44.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 use strict;
 
 usage() unless scalar(@ARGV) == 1;
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclFilterFasta orthomclSoftware-v2.0.8/bin/orthomclFilterFasta
--- orthomclSoftware-v2.0.8.orig/bin/orthomclFilterFasta	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclFilterFasta	2013-05-16 18:29:48.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclInstallSchema orthomclSoftware-v2.0.8/bin/orthomclInstallSchema
--- orthomclSoftware-v2.0.8.orig/bin/orthomclInstallSchema	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclInstallSchema	2013-05-16 18:29:51.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use DBI;
 use FindBin;
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclLoadBlast orthomclSoftware-v2.0.8/bin/orthomclLoadBlast
--- orthomclSoftware-v2.0.8.orig/bin/orthomclLoadBlast	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclLoadBlast	2013-05-16 18:29:57.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use DBI;
 use FindBin;
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclMclToGroups orthomclSoftware-v2.0.8/bin/orthomclMclToGroups
--- orthomclSoftware-v2.0.8.orig/bin/orthomclMclToGroups	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclMclToGroups	2013-05-16 18:30:02.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 my ($prefix, $startId) = @ARGV;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclPairs orthomclSoftware-v2.0.8/bin/orthomclPairs
--- orthomclSoftware-v2.0.8.orig/bin/orthomclPairs	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclPairs	2013-05-16 18:30:07.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use DBI;
 use FindBin;
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclReduceFasta orthomclSoftware-v2.0.8/bin/orthomclReduceFasta
--- orthomclSoftware-v2.0.8.orig/bin/orthomclReduceFasta	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclReduceFasta	2013-05-16 18:30:10.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclReduceGroups orthomclSoftware-v2.0.8/bin/orthomclReduceGroups
--- orthomclSoftware-v2.0.8.orig/bin/orthomclReduceGroups	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclReduceGroups	2013-05-16 18:30:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclRemoveIdenticalGroups orthomclSoftware-v2.0.8/bin/orthomclRemoveIdenticalGroups
--- orthomclSoftware-v2.0.8.orig/bin/orthomclRemoveIdenticalGroups	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclRemoveIdenticalGroups	2013-05-16 18:30:16.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclSingletons orthomclSoftware-v2.0.8/bin/orthomclSingletons
--- orthomclSoftware-v2.0.8.orig/bin/orthomclSingletons	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclSingletons	2013-05-16 18:30:19.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclSortGroupMembersByScore orthomclSoftware-v2.0.8/bin/orthomclSortGroupMembersByScore
--- orthomclSoftware-v2.0.8.orig/bin/orthomclSortGroupMembersByScore	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclSortGroupMembersByScore	2013-05-16 18:30:22.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
diff -ru orthomclSoftware-v2.0.8.orig/bin/orthomclSortGroupsFile orthomclSoftware-v2.0.8/bin/orthomclSortGroupsFile
--- orthomclSoftware-v2.0.8.orig/bin/orthomclSortGroupsFile	2013-05-06 22:19:19.000000000 +0200
+++ orthomclSoftware-v2.0.8/bin/orthomclSortGroupsFile	2013-05-16 18:30:24.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 use strict;
 
