diff -ru Flye.orig/src/Makefile Flye/src/Makefile
--- Flye.orig/src/Makefile	2021-06-14 16:02:46.709190000 +0200
+++ Flye/src/Makefile	2021-06-14 16:03:31.146916000 +0200
@@ -1,6 +1,6 @@
 .PHONY: all clean debug profile
 
-CXXFLAGS += -Wall -Wextra -pthread -std=c++11 -g
+CXXFLAGS += -Wall -Wextra -pthread -std=c++11 -g -fsigned-char
 LDFLAGS += -pthread -std=c++11 -rdynamic
 
 MODULES_BIN := ${BIN_DIR}/flye-modules
