diff -ruN typing_extensions-4.15.0-orig/pyproject.toml typing_extensions-4.15.0/pyproject.toml
--- typing_extensions-4.15.0-orig/pyproject.toml	2025-10-21 19:25:17.987178000 +0200
+++ typing_extensions-4.15.0/pyproject.toml	2025-10-21 19:26:34.308773000 +0200
@@ -10,7 +10,7 @@
 description = "Backported and Experimental Type Hints for Python 3.9+"
 readme = "README.md"
 requires-python = ">=3.9"
-license = "PSF-2.0"
+license = { text = "PSF-2.0" }
 license-files = ["LICENSE"]
 keywords = [
     "annotations",
@@ -58,7 +58,7 @@
 email = "levkivskyi@gmail.com"
 
 [tool.flit.sdist]
-include = ["CHANGELOG.md", "README.md", "tox.ini", "*/*test*.py"]
+include = ["LICENSE", "CHANGELOG.md", "README.md", "tox.ini", "*/*test*.py"]
 exclude = []
 
 [tool.ruff]
