Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Description: Include string.h before poisoning {c,m}alloc
diff -ruN libwpe-1.13.3/src/alloc-private.h libwpe-1.13.3.p1/src/alloc-private.h
--- libwpe-1.13.3/src/alloc-private.h	2022-08-11 12:59:47.000000000 +0100
+++ libwpe-1.13.3.p1/src/alloc-private.h	2022-12-22 12:01:51.000000000 +0000
@@ -27,6 +27,7 @@
 #ifndef wpe_alloc_private_h
 #define wpe_alloc_private_h
 
+#include <string.h>
 #include <stdlib.h>
 
 #if defined(__has_attribute) && __has_attribute(noreturn)
