diff -ur ncbi_cxx--9_0_0.orig/src/build-system/configure ncbi_cxx--9_0_0/src/build-system/configure
--- ncbi_cxx--9_0_0.orig/src/build-system/configure	2012-03-28 22:49:37.000000000 +0200
+++ ncbi_cxx--9_0_0/src/build-system/configure	2013-01-18 13:50:58.301058922 +0100
@@ -4621,7 +4621,7 @@
    compiler_ver="$icc_ver"
    compiler="ICC"
    ncbi_compiler="ICC"
-   ncbi_compiler_ver="`echo $icc_ver | sed 's%.*Version \([0-9.]*\).*%\1%'`"
+   ncbi_compiler_ver="`echo $icc_ver | sed 's%.*Version \([0-9.]*\).*%\1%' | sed 's/\.//g'`"
              WithFeatures="$WithFeatures${WithFeaturesSep}ICC"; WithFeaturesSep=" "
 elif test "$VAC" = "yes" ; then
    compiler_ver="$vac_ver"
@@ -5444,6 +5444,8 @@
     mips-sgi-irix*:KCC )
       ARCH_CFLAGS="-64"
       ;;
+    *:ICC)
+      ;;
     * )
       { { echo "$as_me:$LINENO: error: Do not know how to compile 64-bit with compiler $CXX $compiler_ver $host:$compiler" >&5
 echo "$as_me: error: Do not know how to compile 64-bit with compiler $CXX $compiler_ver $host:$compiler" >&2;}
