diff -ur ncbi_cxx--9_0_0.orig/src/build-system/configure ncbi_cxx--9_0_0/src/build-system/configure
--- ncbi_cxx--9_0_0.orig/src/build-system/configure	2012-03-28 22:49:37.000000000 +0200
+++ ncbi_cxx--9_0_0/src/build-system/configure	2013-01-18 11:17:35.302229258 +0100
@@ -4621,7 +4621,7 @@
    compiler_ver="$icc_ver"
    compiler="ICC"
    ncbi_compiler="ICC"
-   ncbi_compiler_ver="`echo $icc_ver | sed 's%.*Version \([0-9.]*\).*%\1%'`"
+   ncbi_compiler_ver="`echo $icc_ver | sed 's%.*Version \([0-9.]*\).*%\1%' | sed 's/\.//g'`"
              WithFeatures="$WithFeatures${WithFeaturesSep}ICC"; WithFeaturesSep=" "
 elif test "$VAC" = "yes" ; then
    compiler_ver="$vac_ver"
