diff -ru alphafold3-3.0.0/pyproject.toml alphafold3-3.0.0_relax_requirements/pyproject.toml
--- alphafold3-3.0.0/pyproject.toml	2024-11-11 11:42:47.000000000 +0100
+++ alphafold3-3.0.0_relax_requirements/pyproject.toml	2024-12-19 17:09:46.784161491 +0100
@@ -19,12 +19,12 @@
     "chex",
     "dm-haiku==0.0.13",
     "dm-tree",
-    "jax==0.4.34",
-    "jax[cuda12]==0.4.34",
+    "jax>=0.4.34",
+    #"jax[cuda12]==0.4.34",
     "jax-triton==0.2.0",
     "jaxtyping",
     "numpy",
-    "rdkit==2024.3.5",
+    "rdkit",
     "triton==3.1.0",
     "tqdm",
     "zstandard",
