# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2025/01
diff -ru alphafold3-ea040346e10db1759170e723ef263316e64aa768/pyproject.toml alphafold3-ea040346e10db1759170e723ef263316e64aa768_relax_requirements/pyproject.toml
--- alphafold3-ea040346e10db1759170e723ef263316e64aa768/pyproject.toml	2025-01-09 13:59:41.000000000 +0100
+++ alphafold3-ea040346e10db1759170e723ef263316e64aa768_relax_requirements/pyproject.toml	2025-01-15 10:54:51.110182402 +0100
@@ -17,17 +17,17 @@
 dependencies = [
     "absl-py",
     "chex",
-    "dm-haiku==0.0.13",
+    "dm-haiku>=0.0.13",
     "dm-tree",
-    "jax==0.4.34",
-    "jax[cuda12]==0.4.34",
-    "jax-triton==0.2.0",
-    "jaxtyping==0.2.34",
+    "jax>=0.4.34",
+    #"jax[cuda12]==0.4.34",
+    "jax-triton>=0.2.0",
+    "jaxtyping>=0.2.34",
     "numpy",
-    "rdkit==2024.3.5",
-    "triton==3.1.0",
+    "rdkit>=2024.3.5",
+    "triton>=3.1.0",
     "tqdm",
-    "typeguard==2.13.3",
+    "typeguard>=2.13.3",
     "zstandard",
 ]
 
