diff -ru wtdbg2-2.3.orig/Makefile wtdbg2-2.3/Makefile
--- wtdbg2-2.3.orig/Makefile	2019-01-21 02:35:00.000000000 +0100
+++ wtdbg2-2.3/Makefile	2019-02-15 13:12:39.813903927 +0100
@@ -1,16 +1,17 @@
 VERSION=2.3
 
-CC  := gcc
-BIN := /usr/local/bin
+CC  ?= gcc
+PREFIX ?= /usr/local
+BIN := ${PREFIX}/bin
 
 ifeq (0, ${MAKELEVEL})
 TIMESTAMP=$(shell date)
 endif
 
 ifeq (1, ${DEBUG})
-CFLAGS=-g3 -W -Wall -Wno-unused-but-set-variable -O0 -DDEBUG=1 -DTIMESTAMP="$(TIMESTAMP)" -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -mpopcnt -msse4.2
+CFLAGS+=-g3 -W -Wall -Wno-unused-but-set-variable -O0 -DDEBUG=1 -DTIMESTAMP="$(TIMESTAMP)" -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
 else
-CFLAGS=-g3 -W -Wall -Wno-unused-but-set-variable -O4 -DTIMESTAMP="$(TIMESTAMP)" -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -mpopcnt -msse4.2
+CFLAGS+=-g3 -W -Wall -Wno-unused-but-set-variable -O4 -DTIMESTAMP="$(TIMESTAMP)" -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
 endif
 
 GLIBS=-lm -lrt -lpthread -lz
