diff -ru sextractor-2.19.5.orig/acx_mkl.m4 sextractor-2.19.5/acx_mkl.m4
--- sextractor-2.19.5.orig/acx_mkl.m4	2018-11-16 21:38:12.575649193 +0000
+++ sextractor-2.19.5/acx_mkl.m4	2018-11-16 21:38:50.365304798 +0000
@@ -97,7 +97,7 @@
 dnl INTEL compiler uses Intel64 architecture
     if test x$2 = xyes; then
 dnl 64 bit pointers
-      AC_SUBST(MKL_CFLAGS, "-openmp -DMKL_ILP64 -I$mklroot/include")
+      AC_SUBST(MKL_CFLAGS, "-qopenmp -DMKL_ILP64 -I$mklroot/include")
       if test x$3 = xyes; then
 dnl Static linking
         AC_SUBST(MKL_LIBS, ["$mklroot/lib/libmkl_intel_ilp64.a \
@@ -110,7 +110,7 @@
       fi
     else
 dnl 32 bit pointers
-      AC_SUBST(MKL_CFLAGS, "-openmp -I$mklroot/include")
+      AC_SUBST(MKL_CFLAGS, "-qopenmp -I$mklroot/include")
       if test x$3 = xyes; then
 dnl Static linking
         AC_SUBST(MKL_LIBS, ["$mklroot/lib/libmkl_intel_lp64.a \
@@ -124,7 +124,7 @@
     fi
   else
 dnl INTEL compiler uses IA32 architecture
-    AC_SUBST(MKL_CFLAGS, "-openmp -I$mklroot/include")
+    AC_SUBST(MKL_CFLAGS, "-qopenmp -I$mklroot/include")
     if test x$3 = xyes; then
 dnl Static linking
     AC_SUBST(MKL_LIBS, ["$mklroot/lib/libmkl_intel.a \
@@ -142,7 +142,7 @@
 dnl INTEL compiler uses Intel64 architecture
     if test x$2 = xyes; then
 dnl 64 bit pointers
-      AC_SUBST(MKL_CFLAGS, "-openmp -DMKL_ILP64 -I$mklroot/include")
+      AC_SUBST(MKL_CFLAGS, "-qopenmp -DMKL_ILP64 -I$mklroot/include")
       if test x$3 = xyes; then
 dnl Static linking
       AC_SUBST(MKL_LIBS,
@@ -156,7 +156,7 @@
       fi
     else
 dnl 32 bit pointers
-      AC_SUBST(MKL_CFLAGS, "-openmp -I$mklroot/include")
+      AC_SUBST(MKL_CFLAGS, "-qopenmp -I$mklroot/include")
       if test x$3 = xyes; then
 dnl Static linking
         AC_SUBST(MKL_LIBS,
@@ -171,7 +171,7 @@
     fi
   else
 dnl INTEL compiler uses IA32 architecture
-    AC_SUBST(MKL_CFLAGS, "-openmp -I$mklroot/include")
+    AC_SUBST(MKL_CFLAGS, "-qopenmp -I$mklroot/include")
     if test x$3 = xyes; then
 dnl Static linking
       AC_SUBST(MKL_LIBS, ["$startgroup,$mklroot/lib/ia32/libmkl_intel.a,\
diff -ru sextractor-2.19.5.orig/configure sextractor-2.19.5/configure
--- sextractor-2.19.5.orig/configure	2018-11-16 21:38:12.578649165 +0000
+++ sextractor-2.19.5/configure	2018-11-16 21:38:50.364304807 +0000
@@ -13390,7 +13390,7 @@
 if test x$flagosx = xyes; then
   if test x$flag64 = xyes; then
     if test x = xyes; then
-      MKL_CFLAGS="-openmp -DMKL_ILP64 -I$mklroot/include"
+      MKL_CFLAGS="-qopenmp -DMKL_ILP64 -I$mklroot/include"
 
       if test x$enable_best_link = xyes; then
         MKL_LIBS="$mklroot/lib/libmkl_intel_ilp64.a \
@@ -13403,7 +13403,7 @@
 
       fi
     else
-      MKL_CFLAGS="-openmp -I$mklroot/include"
+      MKL_CFLAGS="-qopenmp -I$mklroot/include"
 
       if test x$enable_best_link = xyes; then
         MKL_LIBS="$mklroot/lib/libmkl_intel_lp64.a \
@@ -13417,7 +13417,7 @@
       fi
     fi
   else
-    MKL_CFLAGS="-openmp -I$mklroot/include"
+    MKL_CFLAGS="-qopenmp -I$mklroot/include"
 
     if test x$enable_best_link = xyes; then
     MKL_LIBS="$mklroot/lib/libmkl_intel.a \
@@ -13433,7 +13433,7 @@
 else
   if test x$flag64 = xyes; then
     if test x = xyes; then
-      MKL_CFLAGS="-openmp -DMKL_ILP64 -I$mklroot/include"
+      MKL_CFLAGS="-qopenmp -DMKL_ILP64 -I$mklroot/include"
 
       if test x$enable_best_link = xyes; then
       MKL_LIBS="$startgroup,$mklroot/lib/intel64/libmkl_intel_ilp64.a,\
@@ -13446,7 +13446,7 @@
 
       fi
     else
-      MKL_CFLAGS="-openmp -I$mklroot/include"
+      MKL_CFLAGS="-qopenmp -I$mklroot/include"
 
       if test x$enable_best_link = xyes; then
         MKL_LIBS="$startgroup,$mklroot/lib/intel64/libmkl_intel_lp64.a,\
@@ -13460,7 +13460,7 @@
       fi
     fi
   else
-    MKL_CFLAGS="-openmp -I$mklroot/include"
+    MKL_CFLAGS="-qopenmp -I$mklroot/include"
 
     if test x$enable_best_link = xyes; then
       MKL_LIBS="$startgroup,$mklroot/lib/ia32/libmkl_intel.a,\
@@ -15218,7 +15218,7 @@
 # "Best" linking option
 if test x$enable_best_link = xyes; then
   if test x$enable_icc = xyes; then
-    AM_LDFLAGS="-static-intel -openmp-link static -shared-libgcc \
+    AM_LDFLAGS="-static-intel -qopenmp-link static -shared-libgcc \
 	-static-libtool-libs -no-intel-extensions -avoid-version $AM_LDFLAGS"
   else
     AM_LDFLAGS="-shared-libgcc -static-libtool-libs -avoid-version $AM_LDFLAGS"
