Only in ligateHAPLOTYPES/: bin
diff -ru ligateHAPLOTYPES/makefile ligateHAPLOTYPES_correct/makefile
--- ligateHAPLOTYPES/makefile	2013-08-05 17:23:30.000000000 +0200
+++ ligateHAPLOTYPES_correct/makefile	2016-12-02 14:12:17.000000000 +0100
@@ -31,7 +31,8 @@
 #include path
 ISTDP= -Isrc
 IBAMP= -Ilib
-ICL3P= -I/users/delaneau/BOOST/include
+#ICL3P= -I/users/delaneau/BOOST/include
+ICL3P= -I${BOOST_ROOT}/include
 
 #library path
 LSTDP= -Llib
@@ -59,7 +60,8 @@
 cluster: CFLAG=$(COPTI) $(CWARN) $(CSHV1)
 cluster: LFLAG=$(LOPTI) $(LCL3S)
 cluster: IFLAG=$(ISTDP) $(ICL3P)
-cluster: OBOST=~/BOOST/lib/libboost_iostreams.a ~/BOOST/lib/libboost_program_options.a
+#cluster: OBOST=~/BOOST/lib/libboost_iostreams.a ~/BOOST/lib/libboost_program_options.a
+cluster: OBOST=${BOOST_ROOT}/lib/libboost_iostreams.a ${BOOST_ROOT}/lib/libboost_program_options.a
 cluster: $(EFILE)
 
 $(EFILE): $(OFILE)
