diff -ruN JuliaCall_orig/inst/julia/JuliaObject.jl JuliaCall/inst/julia/JuliaObject.jl
--- JuliaCall_orig/inst/julia/JuliaObject.jl	2023-10-06 15:50:37.057320040 +0200
+++ JuliaCall/inst/julia/JuliaObject.jl	2023-10-09 16:21:42.293384419 +0200
@@ -74,14 +74,14 @@
 ## we should use JuliaObject for general AbstractArray
 @static if julia07
     @suppress_err begin
-        JuliaCall.sexpclass(x :: AbstractArray{T}) where {T} = RClass{:JuliaObject}
+        @eval JuliaCall.sexpclass(x :: AbstractArray{T}) where {T} = RClass{:JuliaObject}
     end
 
     ## AbstractArray{Any} should be converted to R List
     sexpclass(x :: AbstractArray{Any}) = RClass{:list}
 else
     @suppress_err begin
-        JuliaCall.sexp(x :: AbstractArray{T}) where {T} = sexp(JuliaObject(x))
+        @eval JuliaCall.sexp(x :: AbstractArray{T}) where {T} = sexp(JuliaObject(x))
     end
 
     ## AbstractArray{Any} should be converted to R List
