--- cufflinks-2.0.2/src/lemon/bits/base_extender.h.orig	2013-03-27 23:00:37.579508561 +0100
+++ cufflinks-2.0.2/src/lemon/bits/base_extender.h	2013-03-27 23:01:09.439604981 +0100
@@ -359,10 +359,10 @@
 		}
 		
 		Node source(const UEdge& edge) const {
-			return aNode(edge);
+			return this->aNode(edge);
 		}
 		Node target(const UEdge& edge) const {
-			return bNode(edge);
+			return this->bNode(edge);
 		}
 		
 		void firstInc(UEdge& edge, bool& dir, const Node& node) const {
