fix missing return value
author: Paul Jähne
--- a/src/lemon/error.h
+++ b/src/lemon/error.h
@@ -67,9 +67,9 @@ namespace lemon {
     }
 
     ExceptionMember& operator=(const ExceptionMember& copy) {
-      if (ptr.get() == 0) return;
+      if (ptr.get() == 0) return NULL;
       try {
-	if (!copy.valid()) return;
+	if (!copy.valid()) return NULL;
  	*ptr = copy.get();
       } catch (...) {}
     }
