fix for:
error: AM_INIT_AUTOMAKE expanded multiple times
author: Kenneth Hoste (HPC-UGent)
--- cufflinks-dc3b0cb72a4ac2b6bbc887099e71fc0c21e107b7/configure.ac.orig	2019-07-06 18:28:01.000000000 +0200
+++ cufflinks-dc3b0cb72a4ac2b6bbc887099e71fc0c21e107b7/configure.ac	2024-09-27 13:39:13.512597490 +0200
@@ -14,7 +14,7 @@
 AC_CONFIG_SRCDIR([config.h.in])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_AUX_DIR([build-aux])
-AM_INIT_AUTOMAKE
+#AM_INIT_AUTOMAKE
 
 #AM_PATH_CPPUNIT(1.10.2)
 
