--- install.orig	2018-08-30 14:34:57.034483477 +0000
+++ install	2018-08-30 15:05:07.754063177 +0000
@@ -84,7 +84,7 @@
 more $licensefile
 echo
 echo -n 'Do you accept these terms? (accept,decline) '
-read x
+x=$PGI_ACCEPT_EULA
 ans=`echo $x | awk '{print $1}'`
 if test "$ans" != "accept" ; then
   quit
@@ -104,7 +104,7 @@
  
 EOF_NETWORK
 echo -n 'Please choose install option: ' 
-read x
+x=1
 ans=`echo $x | awk '{print $1}'`
 if test -n "$ans" ; then
   case $ans in
@@ -175,7 +175,7 @@
 EOF_INSTALL_DIR
  
 echo -n "Installation directory? [$INSTALL_DIR] "
-read ans
+ans=$PGI_INSTALL_DIR
 if test -n "$ans" ; then
   case $ans in
   ~* )
@@ -287,7 +287,7 @@
 
 EOF_ACML
 echo -n 'Install the ACML? (y/n) '
-read x
+x="no"
 ans=`echo $x | awk '{print $1}'`
 
 if test "$ans" = "y" -o "$ans" = "yes" ; then
@@ -333,7 +333,7 @@
 
 EOF_CUDA
 echo -n 'Install CUDA Toolkit Components? (y/n) '
-read x
+x=$PGI_INSTALL_NVIDIA
 ans=`echo $x | awk '{print $1}'`
 
 if test "$ans" = "y" -o "$ans" = "yes" ; then
@@ -343,7 +343,7 @@
   more $SRC/common/LICENSE_NVIDIA_CUDA.txt
   echo
   echo -n 'Do you accept these terms? (accept,decline) '
-  read x
+  x=$PGI_ACCEPT_EULA
   ans=`echo $x | awk '{print $1}'`
   if test "$ans" != "accept" ; then
     install_nvidia_cuda=0
@@ -365,7 +365,7 @@
 EOF_JAVA_INST
      ans="no"
      echo -n "Overwrite the installed JAVA JRE `ls -1 $INSTALL_DIR/$target/$REL_VERSION/java` ? [$ans] "
-     read x 
+     x=$PGI_INSTALL_JAVA
      if test -n "$x" ; then
        ans=`echo $x | awk '{print $1}'`
      fi
@@ -386,7 +386,7 @@
 EOF_JAVA
      ans="yes"
      echo -n "Install JAVA JRE [$ans] "
-     read x 
+     x=$PGI_INSTALL_JAVA 
      if test -n "$x" ; then
        ans=`echo $x | awk '{print $1}'`
      fi
@@ -399,7 +399,7 @@
   more $SRC/common/LICENSE_JRE_6U21.txt
   echo
   echo -n 'Do you accept these terms? (accept,decline) '
-  read x
+  x=$PGI_ACCEPT_EULA
   ans=`echo $x | awk '{print $1}'`
   if test "$ans" != "accept" ; then
     install_java_jre=0
@@ -419,7 +419,7 @@
 
 EOF_AMD
 echo -n 'Install beta components for AMD GPUs? (y/n) '
-read x
+x=$PGI_INSTALL_AMD
 ans=`echo $x | awk '{print $1}'`
 
 if test "$ans" = "y" -o "$ans" = "yes" ; then
@@ -429,7 +429,7 @@
   more $SRC/common/LICENSE_AMD_BETA.txt
   echo
   echo -n 'Do you accept these terms? (accept,decline) '
-  read x
+  x=$PGI_ACCEPT_EULA
   ans=`echo $x | awk '{print $1}'`
   if test "$ans" != "accept" ; then
     install_amd_beta=0
@@ -562,7 +562,7 @@
 echo "update the links in that directory to point to the $release directory"
 echo " "
 echo -n "Do you wish to update/create links in the $REL_VERSION directory? (y/n) "
-read x
+x="yes"
 ans=`echo $x | awk '{print $1}'`
 if test "$ans" = "y" -o "$ans" = "yes" ; then
 
@@ -724,7 +724,7 @@
 
 if test -e $SRC/install_mpich; then
   echo -n 'Do you wish to install MPICH1? (y/n) '
-  read x
+  x=$PGI_INSTALL_MPI
   ans=`echo $x | awk '{print $1}'`
   if test "$ans" = "y" -o "$ans" = "yes" ; then
     $SRC/install_mpich "$INSTALL_DIR" "$REL_VERSION" "$release"
@@ -791,7 +791,7 @@
 
 echo
 echo -n 'Do you want the files in the install directory to be read-only? (y/n) '
-read x
+x="no"
 ans=`echo $x | awk '{print $1}'`
 if test "$ans" = "y" -o "$ans" = "yes" ; then
   /bin/chmod -Rf a-w $INSTALL_DIR/linux86*/$release
