Use old license syntax for older setuptools support

author: micketeer@gmail.com
--- pyproject.toml.orig 2025-06-23 19:28:32.372399449 +0000
+++ pyproject.toml      2025-06-23 19:28:57.818592635 +0000
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools >= 78", "setuptools_scm[toml] >= 6.2", "Cython >= 0.29.20"]
+requires = ["setuptools", "setuptools_scm[toml] >= 6.2", "Cython >= 0.29.20"]
 build-backend = "setuptools.build_meta"
 
 [project]
@@ -9,7 +9,7 @@
 ]
 description = "Adapter trimming and other preprocessing of high-throughput sequencing reads"
 readme = "README.rst"
-license = "MIT"
+license = {text = "MIT"}
 classifiers = [
     "Development Status :: 5 - Production/Stable",
     "Environment :: Console",

