# making AlphaPulldown's dependencies less strict
diff -ruN alphapulldown-0.30.4_orig/setup.cfg alphapulldown-0.30.4/setup.cfg
--- alphapulldown-0.30.4_orig/setup.cfg	2023-04-05 13:59:25.000000000 +0100
+++ alphapulldown-0.30.4/setup.cfg	2023-04-06 14:10:45.424577468 +0100
@@ -17,19 +17,19 @@
 include_package_data = True
 python_requires = ==3.8.*
 install_requires = 
-	absl-py >= 0.13.0
-	dm-haiku==0.0.9
+	absl-py >= 0.10.0
+	dm-haiku==0.0.4
 	dm-tree==0.1.6
-	h5py==3.1.0
+	h5py==2.10.0
 	matplotlib==3.3.3
-	ml-collections==0.1.0
-	numpy==1.21.6
-	pandas==1.3.4
-	tensorflow==2.9.0
-	importlib-resources==5.8.0
+	ml-collections==0.1.1
+	numpy==1.19.4
+	pandas==1.1.4
+	tensorflow==2.4.1
+	importlib-resources==5.12.0
 	biopython==1.78
-	nbformat==5.4.0
-	py3Dmol==2.0.1
+	nbformat==5.0.7
+	py3Dmol==2.0.1.post1
 	ipython
 	tqdm
 	appdirs
