making dependencies less strict
author: Denis Kristak (Inuits)
--- setup.cfg_orig	2023-05-05 15:41:50.159415210 +0200
+++ setup.cfg	2023-05-05 15:41:28.563066879 +0200
@@ -17,19 +17,19 @@
 include_package_data = True
 python_requires = ==3.8.*
 install_requires = 
-	absl-py >= 0.13.0
-	dm-haiku==0.0.9
-	dm-tree==0.1.6
-	h5py==3.1.0
+	absl-py >= 0.10.0
+	dm-haiku==0.0.4
+ 	dm-tree==0.1.5
+	h5py==2.10.0
 	matplotlib==3.3.3
-	ml-collections==0.1.0
-	numpy==1.21.6
-	pandas==1.3.4
-	tensorflow==2.9.0
-	importlib-resources==5.8.0
+	ml-collections==0.1.1
+	numpy==1.19.4
+	pandas==1.1.4
+	tensorflow==2.4.1
+	importlib-resources==5.12.0
 	biopython==1.78
-	nbformat==5.4.0
-	py3Dmol==2.0.1
+	nbformat==5.0.7
+	py3Dmol==2.0.1.post1
 	ipython
 	tqdm
 	appdirs
