diff -Nru DualSPHysics_v5.0.orig/src/source/Makefile DualSPHysics_v5.0/src/source/Makefile
--- DualSPHysics_v5.0.orig/src/source/Makefile	2022-05-13 17:20:45.946668000 +0100
+++ DualSPHysics_v5.0/src/source/Makefile	2022-05-14 18:18:06.476594543 +0100
@@ -1,10 +1,10 @@
 #DualSPHysics GPU/CPU v5.0.164 21-11-2020
 
 #=============== Compilation Options (YES/NO) ===============
-USE_GCC5=NO
+USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
@@ -26,7 +26,9 @@
     CCFLAGS+= -ffast-math
   endif
   ifeq ($(USE_NATIVE_CPU_OPTIMIZATIONS), YES)
-    CCFLAGS+= -march=native
+	ifeq ($(shell uname -p), x86_64)
+    	CCFLAGS+= -march=native
+	endif
   endif
 endif
 CC=g++
@@ -56,7 +58,7 @@
 endif
 
 #=============== CUDA selection ===============
-CUDAVER=92
+CUDAVER=110
 
 #=============== CUDA toolkit directory (make appropriate for local CUDA installation) ===============
 ifeq ($(CUDAVER),00)
@@ -71,6 +73,9 @@
 ifeq ($(CUDAVER),92)
   DIRTOOLKIT=/exports/opt/NVIDIA/cuda-9.2
 endif
+ifeq ($(CUDAVER),110)
+  DIRTOOLKIT=$(EBROOTCUDA)
+endif
 
 #=============== Select GPU architectures ===============
 ifeq ($(CUDAVER),00)
@@ -102,6 +107,13 @@
   GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
   GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
 endif
+ifeq ($(CUDAVER),110)
+  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_75,code=\"sm_75,compute_75\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_80,code=\"sm_80,compute_80\"
+endif
 
 
 #=============== Files to compile ===============
diff -Nru DualSPHysics_v5.0.orig/src/source/Makefile_cpu DualSPHysics_v5.0/src/source/Makefile_cpu
--- DualSPHysics_v5.0.orig/src/source/Makefile_cpu	2022-05-13 17:20:45.946887000 +0100
+++ DualSPHysics_v5.0/src/source/Makefile_cpu	2022-05-14 18:18:50.708985000 +0100
@@ -1,10 +1,10 @@
 #DualSPHysics CPU v5.0.164 21-11-2020
 
 #=============== Compilation Options (YES/NO) ===============
-USE_GCC5=NO
+USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
@@ -26,7 +26,9 @@
     CCFLAGS+= -ffast-math
   endif
   ifeq ($(USE_NATIVE_CPU_OPTIMIZATIONS), YES)
-    CCFLAGS+= -march=native
+	ifeq ($(shell uname -p), x86_64)
+    	CCFLAGS+= -march=native
+	endif
   endif
 endif
 CC=g++
diff -Nru DualSPHysics_v5.0.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile DualSPHysics_v5.0/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile
--- DualSPHysics_v5.0.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile	2022-05-13 17:20:52.812641000 +0100
+++ DualSPHysics_v5.0/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile	2022-05-14 17:48:08.676008253 +0100
@@ -1,10 +1,10 @@
 #DualSPHysics NNewtonian GPU/CPU v5.0.164 21-11-2020
 
 #=============== Compilation Options (YES/NO) ===============
-USE_GCC5=NO
+USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
@@ -26,7 +26,9 @@
     CCFLAGS+= -ffast-math
   endif
   ifeq ($(USE_NATIVE_CPU_OPTIMIZATIONS), YES)
-    CCFLAGS+= -march=native
+	ifeq ($(shell uname -p), x86_64)
+    	CCFLAGS+= -march=native
+	endif
   endif
 endif
 CC=g++
@@ -56,7 +58,7 @@
 endif
 
 #=============== CUDA selection ===============
-CUDAVER=92
+CUDAVER=110
 
 #=============== CUDA toolkit directory (make appropriate for local CUDA installation) ===============
 ifeq ($(CUDAVER),00)
@@ -71,6 +73,9 @@
 ifeq ($(CUDAVER),92)
   DIRTOOLKIT=/exports/opt/NVIDIA/cuda-9.2
 endif
+ifeq ($(CUDAVER),110)
+  DIRTOOLKIT=$(EBROOTCUDA)
+endif
 
 #=============== Select GPU architectures ===============
 ifeq ($(CUDAVER),00)
@@ -102,6 +107,13 @@
   GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
   GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
 endif
+ifeq ($(CUDAVER),110)
+  GENCODE:=$(GENCODE) -gencode=arch=compute_52,code=\"sm_52,compute_52\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_61,code=\"sm_61,compute_61\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_70,code=\"sm_70,compute_70\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_75,code=\"sm_75,compute_75\"
+  GENCODE:=$(GENCODE) -gencode=arch=compute_80,code=\"sm_80,compute_80\"
+endif
 
 
 #=============== Files to compile ===============
diff -Nru DualSPHysics_v5.0.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu DualSPHysics_v5.0/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu
--- DualSPHysics_v5.0.orig/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu	2022-05-13 17:20:52.813212000 +0100
+++ DualSPHysics_v5.0/src_mphase/DSPH_v5.0_NNewtonian/source/Makefile_cpu	2022-05-14 18:15:56.459456000 +0100
@@ -1,10 +1,10 @@
 #DualSPHysics NNewtonian CPU v5.0.164 21-11-2020
 
 #=============== Compilation Options (YES/NO) ===============
-USE_GCC5=NO
+USE_GCC5=YES
 USE_DEBUG=NO
 USE_FAST_MATH=YES
-USE_NATIVE_CPU_OPTIMIZATIONS=NO
+USE_NATIVE_CPU_OPTIMIZATIONS=YES
 COMPILE_VTKLIB=YES
 COMPILE_NUMEXLIB=YES
 COMPILE_CHRONO=YES
@@ -26,7 +26,9 @@
     CCFLAGS+= -ffast-math
   endif
   ifeq ($(USE_NATIVE_CPU_OPTIMIZATIONS), YES)
-    CCFLAGS+= -march=native
+	ifeq ($(shell uname -p), x86_64)
+    	CCFLAGS+= -march=native
+	endif
   endif
 endif
 CC=g++
