diff -Naur selscan-2.0.3.orig/src/Makefile selscan-2.0.3/src/Makefile
--- selscan-2.0.3.orig/src/Makefile	2024-09-05 20:28:10.658209806 +0000
+++ selscan-2.0.3/src/Makefile	2024-09-05 20:42:21.091261499 +0000
@@ -6,20 +6,20 @@
 #LINK_OPTS = -L$(L_PATH) -lgsl -lgslcblas
 
 #for osx/macos-arm systems
-CC = g++
-G++FLAG = -O3 -m64 
+#CC = g++
+#G++FLAG = -O3 -m64 
 #-mmmx -msse -msse2
-LINK_OPTS2 = -lpthread -lz
-I_PATH = -I../include
-L_PATH = ../lib/macos-arm
+#LINK_OPTS2 = -lpthread -lz
+#I_PATH = -I../include
+#L_PATH = ../lib/macos-arm
 #L_PATH = ../lib/osx
 
 #for linux systems
 #CC = g++
 #G++FLAG = -O3 -m64 -mmmx -msse -msse2
-#LINK_OPTS2 = -pthread -lz
-#I_PATH = -I../include
-#L_PATH = ../lib/linux
+LINK_OPTS2 = -pthread -lz -lgsl  -lgslcblas
+I_PATH = -I../include
+L_PATH = ../lib/linux
 
 #for windows, using MinGW build environment
 #CC = g++.exe
@@ -34,40 +34,40 @@
 all : selscan norm
 
 selscan : selscan-main.o selscan-maintools.o selscan-data.o selscan-cli.o binom.o selscan-pbar.o param_t.o gzstream.o hamming_t.o
-	$(CC) $(G++FLAG) -o selscan selscan-main.o selscan-maintools.o selscan-data.o selscan-cli.o binom.o selscan-pbar.o param_t.o gzstream.o hamming_t.o $(LINK_OPTS2)
+	$(CXX) $(G++FLAG) -o selscan selscan-main.o selscan-maintools.o selscan-data.o selscan-cli.o binom.o selscan-pbar.o param_t.o gzstream.o hamming_t.o $(LINK_OPTS2)
 
 selscan-main.o : selscan-main.cpp
-	$(CC) $(G++FLAG) -c selscan-main.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c selscan-main.cpp $(I_PATH)
 
 selscan-maintools.o : selscan-maintools.cpp selscan-maintools.h
-	$(CC) $(G++FLAG) -c selscan-maintools.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c selscan-maintools.cpp $(I_PATH)
 
 selscan-data.o : selscan-data.cpp selscan-data.h
-	$(CC) $(G++FLAG) -c selscan-data.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c selscan-data.cpp $(I_PATH)
 
 selscan-cli.o : selscan-cli.cpp selscan-cli.h
-	$(CC) $(G++FLAG) -c selscan-cli.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c selscan-cli.cpp $(I_PATH)
 
 selscan-pbar.o : selscan-pbar.cpp selscan-pbar.h
-	$(CC) $(G++FLAG) -c selscan-pbar.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c selscan-pbar.cpp $(I_PATH)
 
 binom.o : binom.cpp binom.h
-	$(CC) $(G++FLAG) -c binom.cpp
+	$(CXX) $(G++FLAG) -c binom.cpp
 
 norm : norm.o param_t.o
-	$(CC) $(G++FLAG) -o norm norm.o param_t.o $(LINK_OPTS)
+	$(CXX) $(G++FLAG) -o norm norm.o param_t.o $(LINK_OPTS)
 
 norm.o : norm.cpp
-	$(CC) $(G++FLAG) -c norm.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c norm.cpp $(I_PATH)
 
 param_t.o : param_t.cpp param_t.h
-	$(CC) $(G++FLAG) -c param_t.cpp
+	$(CXX) $(G++FLAG) -c param_t.cpp
 
 gzstream.o : gzstream.cpp gzstream.h
-	$(CC) $(G++FLAG) -c gzstream.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c gzstream.cpp $(I_PATH)
 
 hamming_t.o : hamming_t.cpp hamming_t.h
-	$(CC) $(G++FLAG) -c hamming_t.cpp $(I_PATH)
+	$(CXX) $(G++FLAG) -c hamming_t.cpp $(I_PATH)
 
 clean :
 	rm *.o
