diff -ruN svaba.orig/SeqLib/bwa/rle.h svaba/SeqLib/bwa/rle.h
--- svaba.orig/SeqLib/bwa/rle.h	2021-02-16 14:55:18.256248000 -0800
+++ svaba/SeqLib/bwa/rle.h	2021-02-16 15:10:36.205167000 -0800
@@ -30,7 +30,7 @@
  *** 43+3 codec ***
  ******************/
 
-const uint8_t rle_auxtab[8];
+extern const uint8_t rle_auxtab[8];
 
 #define RLE_MIN_SPACE 18
 #define rle_nptr(block) ((uint16_t*)(block))
diff -ruN svaba.orig/SeqLib/fermi-lite/rle.h svaba/SeqLib/fermi-lite/rle.h
--- svaba.orig/SeqLib/fermi-lite/rle.h	2021-02-16 14:55:18.529229000 -0800
+++ svaba/SeqLib/fermi-lite/rle.h	2021-02-16 15:18:40.067150000 -0800
@@ -30,7 +30,7 @@
  *** 43+3 codec ***
  ******************/
 
-const uint8_t rle_auxtab[8];
+extern const uint8_t rle_auxtab[8];
 
 #define RLE_MIN_SPACE 18
 #define rle_nptr(block) ((uint16_t*)(block))
