diff -ru SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.original/EM/Makefile SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.patched/EM/Makefile
--- SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.original/EM/Makefile	2018-04-23 08:53:32.000000000 +0200
+++ SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.patched/EM/Makefile	2018-05-01 15:49:18.000000000 +0200
@@ -20,15 +20,15 @@
 OMPFLAG = -fopenmp
 
 # Static build
-STFLAG = -static
+#STFLAG = -static
 
 # Include directories 
 INCL = -I./Ipopt/include
 
 # Library
 # LDFLAGS = -L./Ipopt/lib64 -L/usr/lib64/gcc/x86_64-suse-linux/4.7 -L/usr/lib64/gcc/x86_64-suse-linux/4.7/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib64/gcc/x86_64-suse-linux/4.7/../../../../x86_64-suse-linux/lib -L/usr/lib64/gcc/x86_64-suse-linux/4.7/../../.. -lipopt -llapack -ldl -lcoinmumps -lfblas -lpthread -lgfortran -lm -lquadmath
-LDFLAGS = -L./Ipopt/lib64 -L/usr/lib/gcc/x86_64-linux-gnu/5 -L/usr/lib/ -L/lib/ -L/lib64/ -L/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lipopt -llapack -ldl -lcoinmumps -lfblas -lpthread -lgfortran -lm -lquadmath
-
+#LDFLAGS = -L./Ipopt/lib64 -L/usr/lib/gcc/x86_64-linux-gnu/5 -L/usr/lib/ -L/lib/ -L/lib64/ -L/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lipopt -llapack -ldl -lcoinmumps -lfblas -lpthread -lgfortran -lm -lquadmath
+LDFLAGS = -L${EBROOTGCC}/lib64 -L./Ipopt/lib64 -lipopt -llapack -ldl -lcoinmumps -lfblas -lpthread -lgfortran -lm -lquadmath
 
 #########################################################
 # Please do NOT change below Makefile variables         #
diff -ru SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.original/setup.py SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.patched/setup.py
--- SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.original/setup.py	2018-04-23 08:53:32.000000000 +0200
+++ SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.patched/setup.py	2018-05-01 15:52:07.000000000 +0200
@@ -38,17 +38,31 @@
     },
     keywords=['bioinformatics', 'microbial', 'metagenomics'], 
     package_data={'SPARSE':[
-        'EM/*',
+#        'EM/*',
+        'EM/*.*',
+        'EM/Makefile',
+        'EM/solve-model',
         'bin/*',
         'docs/*',
         'modules/*',
-        'EM/Ipopt/include/*',
-        'EM/Ipopt/lib64/*',
-        'EM/Ipopt/*',
+#        'EM/Ipopt/include/*',
+        'EM/Ipopt/include/*.*',
+        'EM/Ipopt/include/ThirdParty/*.*',
+#        'EM/Ipopt/lib64/*',
+        'EM/Ipopt/lib64/*.*',
+        'EM/Ipopt/lib64/pkgconfig/*.*',
+#        'EM/Ipopt/*',
     ]},
     packages = ['SPARSE'], 
     package_dir = {'SPARSE':'.'},
-    install_requires=['pycapnp', 'numpy', 'pandas', 'Cython', 'scipy', 'msgpack'],
+    install_requires=[
+        'pycapnp==0.6.1',
+        'numpy',
+        'pandas',
+        'Cython',
+        'scipy',
+        'msgpack==0.5.6',
+    ],
     include_package_data=True,
     project_urls={ 
         'Bug Reports': 'https://github.com/zheminzhou/SPARSE/issues',
