diff -ru SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.original/setup.py SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.patched/setup.py
--- SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.original/setup.py	2018-04-23 08:53:32.000000000 +0200
+++ SPARSE-f97098e68c7e41bed159e9d7a14d25b76f7d96e3.patched/setup.py	2018-04-25 23:45:42.000000000 +0200
@@ -44,11 +44,18 @@
         'modules/*',
         'EM/Ipopt/include/*',
         'EM/Ipopt/lib64/*',
-        'EM/Ipopt/*',
+#        'EM/Ipopt/*',
     ]},
     packages = ['SPARSE'], 
     package_dir = {'SPARSE':'.'},
-    install_requires=['pycapnp', 'numpy', 'pandas', 'Cython', 'scipy', 'msgpack'],
+    install_requires=[
+        'pycapnp==0.6.1',
+        'numpy',
+        'pandas',
+        'Cython',
+        'scipy',
+        'msgpack==0.5.6',
+    ],
     include_package_data=True,
     project_urls={ 
         'Bug Reports': 'https://github.com/zheminzhou/SPARSE/issues',
