diff --git a/config/m4/rocm.m4 b/config/m4/rocm.m4
index c7c4ca14..cc89dc47 100644
--- a/config/m4/rocm.m4
+++ b/config/m4/rocm.m4
@@ -28,8 +28,8 @@ done])
 # Parse value of ARG into appropriate LIBS, LDFLAGS, and
 # CPPFLAGS variables.
 AC_DEFUN([ROCM_BUILD_FLAGS],
-    $4="-I$1/include/hsa -I$1/include"
-    $3="-L$1/lib -L$1/lib64 -L$1/hsa/lib"
+    $4="-I$EBROOTROCMMINLLVM/include/hsa -I$1/include"
+    $3="-L$EBROOTROCMMINLLVM/lib -L$1/lib -L$1/lib64"
     $2="-lhsa-runtime64 -lhsakmt"
     $5="$1"
 )
@@ -39,8 +39,8 @@ AC_DEFUN([ROCM_BUILD_FLAGS],
 # Parse value of ARG into appropriate LIBS, LDFLAGS, and
 # CPPFLAGS variables.
 AC_DEFUN([HIP_BUILD_FLAGS],
-    $4="-D__HIP_PLATFORM_AMD__ -I$1/include/hip -I$1/include -I$1/llvm/include"
-    $3="-L$1/lib -L$1/llvm/lib"
+    $4="-D__HIP_PLATFORM_AMD__ -I$1/include/hip -I$1/include -I$EBROOTROCMMINLLVM/include"
+    $3="-L$1/lib -L$EBROOTROCMMINLLVM/lib"
     $2="-lamdhip64"
 )
 
diff --git a/src/components/mc/rocm/Makefile.am b/src/components/mc/rocm/Makefile.am
index b97ef90f..2aff0e4a 100644
--- a/src/components/mc/rocm/Makefile.am
+++ b/src/components/mc/rocm/Makefile.am
@@ -15,7 +15,7 @@ libucc_mc_rocm_la_CPPFLAGS = $(AM_CPPFLAGS) $(HIP_CPPFLAGS) $(ROCM_CPPFLAGS) $(B
 libucc_mc_rocm_la_CFLAGS   = $(BASE_CFLAGS)
 libucc_mc_rocm_la_LDFLAGS  = -version-info $(SOVERSION) --as-needed $(HIP_LDFLAGS) $(ROCM_LDFLAGS)
 libucc_mc_rocm_la_LIBADD   = $(HIP_LIBS) $(ROCM_LIBS) \
-                             $(UCC_TOP_BUILDDIR)/src/libucc.la
+                             -lucc
 
 include $(top_srcdir)/config/module.am
 endif
diff --git a/src/components/tl/rccl/Makefile.am b/src/components/tl/rccl/Makefile.am
index f3da3fb6..8a3888e9 100644
--- a/src/components/tl/rccl/Makefile.am
+++ b/src/components/tl/rccl/Makefile.am
@@ -23,7 +23,7 @@ libucc_tl_rccl_la_SOURCES  = $(sources)
 libucc_tl_rccl_la_CPPFLAGS = $(AM_CPPFLAGS) $(BASE_CPPFLAGS) $(ROCM_CPPFLAGS) $(HIP_CPPFLAGS) $(RCCL_CPPFLAGS)
 libucc_tl_rccl_la_CFLAGS   = $(BASE_CFLAGS)
 libucc_tl_rccl_la_LDFLAGS  = -version-info $(SOVERSION) --as-needed $(ROCM_LDFLAGS) $(HIP_LDFLAGS) $(RCCL_LDFLAGS)
-libucc_tl_rccl_la_LIBADD   = $(ROCM_LIBS) $(HIP_LIBS) $(RCCL_LIBADD) $(UCC_TOP_BUILDDIR)/src/libucc.la
+libucc_tl_rccl_la_LIBADD   = $(ROCM_LIBS) $(HIP_LIBS) $(RCCL_LIBADD) -lucc
 
 include $(top_srcdir)/config/module.am
 
