be verbose about not having a license token available
author: Benjamin Roberts (NeSI)
--- molpro-mpp-2010.1.23.Linux_x86_64.sh	2015-09-02 12:57:08.239724674 +1200
+++ molpro-mpp-2010.1.23.Linux_x86_64.sh	2015-09-02 13:01:18.998587159 +1200
@@ -156,7 +156,12 @@
  eval ${download} "https://www.molpro.net/licensee/token" 1>/dev/null 2>&1 || rm -f ${INSTLIB}/.token
 fi
 
-if [ -f ${INSTLIB}/.token ] ; then
+if [ ! -f ${INSTLIB}/.token ] ; then
+ echo ""
+ echo "Licence token could not be installed successfully."
+ echo "A licence token must be provided whenever running Molpro."
+ exit 0
+else
  chmod 0644 ${INSTLIB}/.token
  echo
  echo Token sucessfully downloaded and installed
