Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Replace hardcoded /tmp to ensure build respects $TMPDIR
diff -Nru bazel-3.7.2.orig/src/main/java/com/google/devtools/build/lib/BUILD bazel-3.7.2/src/main/java/com/google/devtools/build/lib/BUILD
--- bazel-3.7.2.orig/src/main/java/com/google/devtools/build/lib/BUILD	1980-01-01 00:00:00.000000000 +0000
+++ bazel-3.7.2/src/main/java/com/google/devtools/build/lib/BUILD	2022-03-22 11:02:18.000000000 +0000
@@ -499,7 +499,7 @@
     outs = ["command-line-reference.html"],
     cmd = (
         "cat $(location //site:command-line-reference-prefix.html) > $@ && " +
-        "TMP=`mktemp -d /tmp/tmp.XXXXXXXXXX` && " +
+        "TMP=$$(mktemp -d $${TMPDIR:-/tmp}/tmp.XXXXXXXXXX) && " +
         # TODO(#11179): Remove when the stub template fix is released.
         "export JARBIN=$(JAVABASE)/bin/jar && " +
         "$(location //src/main/java/com/google/devtools/build/lib/bazel:BazelServer) " +
