diff -ru ruby-2.7.1.orig/io.c ruby-2.7.1/io.c
--- ruby-2.7.1.orig/io.c	2021-03-03 19:20:07.396740888 +0000
+++ ruby-2.7.1/io.c	2021-03-03 19:21:18.160151902 +0000
@@ -10887,7 +10887,7 @@
     return 0;
 }
 
-#if defined HAVE_COPY_FILE_RANGE || (defined __linux__ && defined __NR_copy_file_range)
+#if 0   //defined HAVE_COPY_FILE_RANGE || (defined __linux__ && defined __NR_copy_file_range)
 #  define USE_COPY_FILE_RANGE
 #endif
 
