Removes unnecessary options for pytest that induce additional dependencies.

--- 0.32.0/foss-2024a/modin/modin-0.32.0/setup.cfg.orig	2024-10-17 15:56:55.245266649 +0200
+++ 0.32.0/foss-2024a/modin/modin-0.32.0/setup.cfg	2024-10-17 15:57:34.748841878 +0200
@@ -11,15 +11,6 @@
 tag_prefix =
 parentdir_prefix = modin-
 
-[tool:pytest]
-addopts = --cov-config=setup.cfg --cov=modin --cov-append --cov-report= -m "not exclude_by_default"
-xfail_strict=true
-markers =
-    exclude_in_sanity
-    exclude_by_default
-filterwarnings =
-    error:.*defaulting to pandas.*:UserWarning
-
 [isort]
 profile = black
 
