diff -ru SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config.mk SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk
--- SuiteSparse.orig/SuiteSparse_config/SuiteSparse_config.mk	2018-11-16 14:57:05.635841940 +0000
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk	2018-11-16 14:58:01.897325242 +0000
@@ -115,7 +115,7 @@
             CC = icc -D_GNU_SOURCE
             CXX = $(CC)
             CFOPENMP = -qopenmp -I$(MKLROOT)/include
-	    LDFLAGS += -openmp
+	    LDFLAGS += -fopenmp
         endif
         ifneq ($(shell which ifort 2>/dev/null),)
             # use the Intel ifort compiler for Fortran codes
