From 1e65f271e37b218bd8b40d1d47cf86ca10cfc267 Mon Sep 17 00:00:00 2001
From: zxkjack123 <zxkjack123@163.com>
Date: Tue, 29 Dec 2020 20:15:03 +0800
Subject: [PATCH] use except + in transmuters

---
 pyne/cpp_transmuters.pxd | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyne/cpp_transmuters.pxd b/pyne/cpp_transmuters.pxd
index 2cd911a5a..d76e0e347 100644
--- a/pyne/cpp_transmuters.pxd
+++ b/pyne/cpp_transmuters.pxd
@@ -5,6 +5,6 @@ from libcpp.vector cimport vector
 
 cdef extern from "transmuters.h" namespace "pyne::transmuters":
 
-    map[int, double] cram(vector[double], const map[int, double]) except +ValueError
-    map[int, double] cram(vector[double], const map[int, double], const int) except +ValueError
+    map[int, double] cram(vector[double], const map[int, double]) except +
+    map[int, double] cram(vector[double], const map[int, double], const int) except +
 
