diff --git a/src/gc.cpp b/src/gc.cpp
index 9295b9f..b48f76d 100644
--- a/src/gc.cpp
+++ b/src/gc.cpp
@@ -1,5 +1,6 @@
 // #include<iostream>
 // #include<algorithm>
+#include <cstdint>
 #include <string>
 #include <vector>
 #include <sstream>
diff --git a/src/gc.h b/src/gc.h
index 8a6e556..c2897c3 100644
--- a/src/gc.h
+++ b/src/gc.h
@@ -1,6 +1,7 @@
 #ifndef GC_H
 #define GC_H
 
+#include <cstdint>
 #include <string>
 #include <string.h>
 #include <vector>
diff --git a/src/global_parameter.h b/src/global_parameter.h
index 9298ad9..038cd4e 100644
--- a/src/global_parameter.h
+++ b/src/global_parameter.h
@@ -1,5 +1,6 @@
 #ifndef _GLOBAL_PARAMETER_H
 #define _GLOBAL_PARAMETER_H
+#include <cstdint>
 #include <string>
 #include <functional>
 #include <vector>
diff --git a/src/global_variable.cpp b/src/global_variable.cpp
index 812c459..d601101 100644
--- a/src/global_variable.cpp
+++ b/src/global_variable.cpp
@@ -1,5 +1,5 @@
 #include "global_variable.h"
-
+#include <cstdint>
 C_general_stat::C_general_stat()
 {
 	read_max_length = 0;
@@ -99,4 +99,4 @@ C_global_variable::C_global_variable(C_global_parameter &gp)
 	trim2_stat = C_fastq_file_stat(gp);
 	clean1_stat = C_fastq_file_stat(gp);
 	clean2_stat = C_fastq_file_stat(gp);
-}
\ No newline at end of file
+}
diff --git a/src/global_variable.h b/src/global_variable.h
index 32b5787..7eaaaec 100644
--- a/src/global_variable.h
+++ b/src/global_variable.h
@@ -1,6 +1,6 @@
 #ifndef _GLOBAL_VARIABLE_H
 #define _GLOBAL_VARIABLE_H
-
+#include <cstdint>
 #include <string>
 #include <set>
 #include "global_parameter.h"
diff --git a/src/peprocess.cpp b/src/peprocess.cpp
index cb6ea96..8afd49e 100644
--- a/src/peprocess.cpp
+++ b/src/peprocess.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include <iostream>
 #include <string>
 #include <stdlib.h>
diff --git a/src/peprocess.h b/src/peprocess.h
index 745f018..e734a84 100644
--- a/src/peprocess.h
+++ b/src/peprocess.h
@@ -1,6 +1,7 @@
 #ifndef _PEPROCESS_H
 #define _PEPROCESS_H
 
+#include <cstdint>
 #include <iostream>
 #include <string>
 #include <vector>
diff --git a/src/sequence.cpp b/src/sequence.cpp
index 71511d2..0d424bd 100644
--- a/src/sequence.cpp
+++ b/src/sequence.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include <stdlib.h>
 #include <iostream>
 #include <stdio.h>
@@ -401,4 +402,4 @@ int C_pe_fastq_filter::pe_dis(bool a,bool b){
 string C_fastq::toString() {
     string completeRead=seq_id+"\n"+sequence+"\n+\n"+qual_seq+"\n";
     return completeRead;
-}
\ No newline at end of file
+}
diff --git a/src/sequence.h b/src/sequence.h
index 9cf0bfc..30a126e 100644
--- a/src/sequence.h
+++ b/src/sequence.h
@@ -1,6 +1,7 @@
 #ifndef _SEQUENCE_H
 #define _SEQUENCE_H
 
+#include <cstdint>
 #include "global_parameter.h"
 #include "global_variable.h"
 #include "zlib.h"
