--- base/Makefile.orig	2019-08-20 00:05:52.000000000 +0000
+++ base/Makefile	2019-09-19 12:39:52.305736732 +0000
@@ -199,13 +199,13 @@
 endif # WINNT
 
 symlink_libLLVM: $(build_private_libdir)/libLLVM.$(SHLIB_EXT)
-ifneq ($(USE_SYSTEM_LLVM),0)
-LLVM_CONFIG_HOST_LIBS := $(shell $(LLVM_CONFIG_HOST) --libfiles)
+#ifneq ($(USE_SYSTEM_LLVM),0)
+#LLVM_CONFIG_HOST_LIBS := $(shell $(LLVM_CONFIG_HOST) --libfiles)
 # HACK: llvm-config doesn't correctly point to shared libs on all platforms
 #       https://github.com/JuliaLang/julia/issues/29981
-else
+#else
 LLVM_CONFIG_HOST_LIBS := $(shell $(LLVM_CONFIG_HOST) --libdir)/libLLVM.$(SHLIB_EXT)
-endif
+#endif
 $(build_private_libdir)/libLLVM.$(SHLIB_EXT):
 	REALPATH=$(LLVM_CONFIG_HOST_LIBS) && \
 	$(call resolve_path,REALPATH) && \
--- src/Makefile.orig	2019-08-20 00:05:52.000000000 +0000
+++ src/Makefile	2019-09-19 12:40:06.640802657 +0000
@@ -90,17 +90,17 @@
 PUBLIC_HEADER_TARGETS := $(addprefix $(build_includedir)/julia/,$(notdir $(PUBLIC_HEADERS)) $(UV_HEADERS))
 
 ifeq ($(JULIACODEGEN),LLVM)
-ifneq ($(USE_SYSTEM_LLVM),0)
-LLVMLINK += $(shell $(LLVM_CONFIG_HOST) --ldflags --libs --system-libs)
+#ifneq ($(USE_SYSTEM_LLVM),0)
+#LLVMLINK += $(shell $(LLVM_CONFIG_HOST) --ldflags --libs --system-libs)
 # HACK: llvm-config doesn't correctly point to shared libs on all platforms
 #       https://github.com/JuliaLang/julia/issues/29981
-else
+#else
 ifneq ($(USE_LLVM_SHLIB),1)
 LLVMLINK += $(shell $(LLVM_CONFIG_HOST) --ldflags) $(shell $(LLVM_CONFIG_HOST) --libs $(LLVM_LIBS)) $(shell $(LLVM_CONFIG_HOST) --ldflags) $(shell $(LLVM_CONFIG_HOST) --system-libs 2> /dev/null)
 else
 LLVMLINK += $(shell $(LLVM_CONFIG_HOST) --ldflags) -lLLVM
 endif
-endif
+#endif
 ifeq ($(USE_LLVM_SHLIB),1)
 FLAGS += -DLLVM_SHLIB
 endif # USE_LLVM_SHLIB == 1
