--- Log-Log4perl-1.46/t/026FileApp.t.orig	2015-07-10 13:35:24.546495930 +0200
+++ Log-Log4perl-1.46/t/026FileApp.t	2015-07-10 13:33:59.123806246 +0200
@@ -484,7 +484,11 @@
   
   my ($dev,$ino,$mode,$nlink,$uid,$gid,$rdev,$size, $atime,$mtime,$ctime,$blksize,$blocks) = stat("${testpath}_1");
   
-  is($mode & 07777,0750); #Win32 777
+  #
+  # Make mode check less strict to allow Set Group ID (SGID) bit on dirs using "chmod g+s"
+  # 
+  #is($mode & 07777,0750); #Win32 777
+  is($mode & 05777,0750); #Win32 777
   
    ($dev,$ino,$mode,$nlink,$uid,$gid,$rdev,$size, $atime,$mtime,$ctime,$blksize,$blocks) = stat($testmkpathfile);
   
