--- expat-2.1.0/xmlwf/unixfilemap.c.orig	2013-07-05 11:21:42.298328733 +0200
+++ expat-2.1.0/xmlwf/unixfilemap.c	2013-07-05 11:21:59.308382014 +0200
@@ -51,7 +51,7 @@
     close(fd);
     return 1;
   }
-  p = (void *)mmap((caddr_t)0, (size_t)nbytes, PROT_READ,
+  p = (void *)mmap((void *)0, (size_t)nbytes, PROT_READ,
                    MAP_FILE|MAP_PRIVATE, fd, (off_t)0);
   if (p == (void *)-1) {
     perror(name);
@@ -59,7 +59,7 @@
     return 0;
   }
   processor(p, nbytes, name, arg);
-  munmap((caddr_t)p, nbytes);
+  munmap((void *)p, nbytes);
   close(fd);
   return 1;
 }
