maximum numpy version updated to 2.3.2 
Author: J. Saßmannshausen (Imperial College London/UK)
diff --git a/opencv-python-4.12.0.88.orig/setup.py b/opencv-python-4.12.0.88/setup.py
index 8941218..e5f3351 100755
--- a/opencv-python-4.12.0.88.orig/setup.py
+++ b/opencv-python-4.12.0.88/setup.py
@@ -26,7 +26,7 @@ def main():
     # see https://numpy.org/doc/stable/release/2.3.0-notes.html#numpy-2-3-0-release-notes
     install_requires = [
         'numpy<2.0; python_version<"3.9"',
-        'numpy(>=2, <2.3.0); python_version>="3.9"',
+        'numpy(>=2, <2.3.2); python_version>="3.9"',
     ]
 
     python_version = cmaker.CMaker.get_python_version()
