Author: Denis Kristak (Inuits)
Fixing version of platformdirs to 2.4.1.
diff -ruN requests_cache-1.1.0_orig/PKG-INFO requests_cache-1.1.0/PKG-INFO
--- requests_cache-1.1.0_orig/PKG-INFO	1970-01-01 01:00:00.000000000 +0100
+++ requests_cache-1.1.0/PKG-INFO	2023-10-11 19:42:09.554496296 +0100
@@ -36,7 +36,7 @@
 Requires-Dist: itsdangerous (>=2.0) ; extra == "security" or extra == "all"
 Requires-Dist: linkify-it-py (>=2.0,<3.0) ; extra == "docs"
 Requires-Dist: myst-parser (>=1.0,<2.0) ; extra == "docs"
-Requires-Dist: platformdirs (>=2.5)
+Requires-Dist: platformdirs (>=2.4.1)
 Requires-Dist: pymongo (>=3) ; extra == "mongodb" or extra == "all"
 Requires-Dist: pyyaml (>=5.4) ; extra == "yaml" or extra == "all"
 Requires-Dist: redis (>=3) ; extra == "redis" or extra == "all"
diff -ruN requests_cache-1.1.0_orig/pyproject.toml requests_cache-1.1.0/pyproject.toml
--- requests_cache-1.1.0_orig/pyproject.toml	2023-07-01 00:03:54.000000000 +0100
+++ requests_cache-1.1.0/pyproject.toml	2023-10-11 19:42:14.574433472 +0100
@@ -36,7 +36,7 @@
 urllib3       = ">=1.25.5" # Use a slightly newer version than required by requests (for bugfixes)
 attrs         = ">=21.2"   # For response data models
 cattrs        = ">=22.2"   # For response serialization
-platformdirs  = ">=2.5"    # For features that use platform-specific system directories
+platformdirs  = ">=2.4.1"    # For features that use platform-specific system directories
 url-normalize = ">=1.4"    # For more accurate request matching
 
 # Optional backend dependencies
