--- adf2016.101/bin/start.orig	2016-03-14 14:38:31.000000000 +0100
+++ adf2016.101/bin/start	2016-03-14 13:24:40.000000000 +0100
@@ -261,7 +261,8 @@
 #  This has been tested with SLURM 2.5.2 and IMPI 4.0
 #  Important: change the path below to the correct libpmi.so library
     export I_MPI_PMI_LIBRARY=/usr/lib64/libpmi.so
-    mpirun -bootstrap slurm "$PROG" "$@" -DSCM_EXPORT="$SCM_EXPORT" -DSCM_DEBUG="$SCM_DEBUG" \
+#   mpirun -bootstrap slurm "$PROG" "$@" -DSCM_EXPORT="$SCM_EXPORT" -DSCM_DEBUG="$SCM_DEBUG" \
+    srun "$PROG" "$@" -DSCM_EXPORT="$SCM_EXPORT" -DSCM_DEBUG="$SCM_DEBUG" \
        -DSCM_VECTORLENGTH="$SCM_VECTORLENGTH" -DSCM_ALARMTIME="$SCM_ALARMTIME" -DSCM_NOMEMCHECK="$SCM_NOMEMCHECK" \
        -DSCM_IOBUFFERSIZE="$SCM_IOBUFFERSIZE" -DSCM_TMPDIR="$SCM_TMPDIR" -DSCM_WD="$SCM_WD" -DSCM_STDIN="$SCM_STDIN" \
        -DSCMLICENSE="$SCMLICENSE" -DSCM_DOMAINCHECK="$SCM_DOMAINCHECK" -DSCM_DIAG_EXCLUDE="$SCM_DIAG_EXCLUDE" \
