Remove duplicate linker flags from fox-config
author: Alex Domingo (Vrije Universiteit Brussel)
diff --git a/build/cmake_modules/FindFOX.cmake.orig b/build/cmake_modules/FindFOX.cmake
index 6d7b7b4..37f5934 100644
--- a/build/cmake_modules/FindFOX.cmake.orig
+++ b/build/cmake_modules/FindFOX.cmake
@@ -16,6 +16,10 @@ endif(FOX_CONFIG)
 set(FOX_FOUND 0)
 # if fox-config was executed successfully, fox was found
 if(FOX_LIBRARY AND FOX_CXX_FLAGS)
+    # remove duplicate link flags
+    separate_arguments(FOX_LIBRARY)
+    list(REMOVE_DUPLICATES FOX_LIBRARY)
+    string(REPLACE ";" " " FOX_LIBRARY "${FOX_LIBRARY}")
     SET(FOX_FOUND TRUE)
 else()
 	# In Windows system, find the fox directory using as hint the environment variable "FOX_INCLUDE_DIR"
