diff -ru trunk_nosvn.orig/contrib/eigen3/eigen/Eigen/src/SparseCore/SparseBlock.h trunk_nosvn/contrib/eigen3/eigen/Eigen/src/SparseCore/SparseBlock.h
--- trunk_nosvn.orig/contrib/eigen3/eigen/Eigen/src/SparseCore/SparseBlock.h	2012-07-31 13:55:44.000000000 +0200
+++ trunk_nosvn/contrib/eigen3/eigen/Eigen/src/SparseCore/SparseBlock.h	2012-08-28 12:24:32.139047212 +0200
@@ -279,7 +279,7 @@
 template<typename Derived>
 SparseInnerVectorSet<Derived,1> SparseMatrixBase<Derived>::row(Index i)
 {
-  EIGEN_STATIC_ASSERT(IsRowMajor,THIS_METHOD_IS_ONLY_FOR_ROW_MAJOR_MATRICES);
+  //EIGEN_STATIC_ASSERT(IsRowMajor,THIS_METHOD_IS_ONLY_FOR_ROW_MAJOR_MATRICES);
   return innerVector(i);
 }
 
