diff -ru trunk_nosvn.orig/contrib/make_klu.sh trunk_nosvn/contrib/make_klu.sh
--- trunk_nosvn.orig/contrib/make_klu.sh	2012-07-31 13:55:45.000000000 +0200
+++ trunk_nosvn/contrib/make_klu.sh	2012-08-28 09:00:27.538193670 +0200
@@ -19,25 +19,25 @@
 cd AMD
 make
 cd Lib
-gcc --shared -fPIC -Wl,-soname,libamd.so -o libamd.so *.o
+$CC --shared -fPIC -Wl,-soname,libamd.so -o libamd.so *.o
 cd ..
 cd ..
 cd BTF
 make
 cd Lib
-gcc --shared -fPIC -Wl,-soname,libbtf.so -o libbtf.so *.o
+$CC --shared -fPIC -Wl,-soname,libbtf.so -o libbtf.so *.o
 cd ..
 cd ..
 cd CAMD
 make
 cd Lib
-gcc --shared -fPIC -Wl,-soname,libcamd.so -o libcamd.so *.o
+$CC --shared -fPIC -Wl,-soname,libcamd.so -o libcamd.so *.o
 cd ..
 cd ..
 cd CCOLAMD
 cd Lib
 make # Very strange, had to add this line after re-installing to linux mint :s ??
-gcc --shared -fPIC -Wl,-soname,libccolamd.so -o libccolamd.so *.o
+$CC --shared -fPIC -Wl,-soname,libccolamd.so -o libccolamd.so *.o
 cd ..
 make
 cd ..
@@ -47,25 +47,25 @@
 cd CHOLMOD
 make
 cd Lib
-gcc --shared -fPIC -Wl,-soname,libcholmod.so -o libcholmod.so *.o
+$CC --shared -fPIC -Wl,-soname,libcholmod.so -o libcholmod.so *.o
 cd ..
 cd ..
 cd COLAMD
 cd Lib
-gcc --shared -fPIC -Wl,-soname,libcolamd.so -o libcolamd.so *.o
+$CC --shared -fPIC -Wl,-soname,libcolamd.so -o libcolamd.so *.o
 cd ..
 make
 cd ..
 cd KLU
 make
 cd Lib
-gcc --shared -fPIC -Wl,-soname,libklu.so -o libklu.so *.o
+$CC --shared -fPIC -Wl,-soname,libklu.so -o libklu.so *.o
 cd ..
 cd ..
 
 # Linux:
 #gcc --shared -fPIC -Wl,-soname,liballklu.so -o liballklu.so */Lib/*.o #Sometimes this is not enough. Use the line below (e.g. Linux Mint)
-gcc --shared -fPIC -Wl,-soname,liballklu.so -o liballklu.so */Lib/*.o /usr/lib/libblas/libblas.so /usr/lib/liblapack.so
+$CC --shared -fPIC -Wl,-soname,liballklu.so -o liballklu.so */Lib/*.o -L$BLAS_LIB_DIR $LIBBLAS_MT -L$LAPACK_LIB_DIR $LIBLAPACK_MT
 
 # Might be necessary to add this to the last command: "/usr/lib/libblas/libblas.so /usr/lib/liblapack.so"
 
diff -ru trunk_nosvn.orig/contrib/metis-4.0.3/Makefile.in trunk_nosvn/contrib/metis-4.0.3/Makefile.in
--- trunk_nosvn.orig/contrib/metis-4.0.3/Makefile.in	2012-07-31 13:55:44.000000000 +0200
+++ trunk_nosvn/contrib/metis-4.0.3/Makefile.in	2012-08-28 09:00:59.402513706 +0200
@@ -1,9 +1,9 @@
 
 # Which compiler to use
-CC = cc
+CC = $(CC)
 
 # What optimization level to use
-OPTFLAGS = -O2 
+OPTFLAGS = $(CFLAGS)
 
 # What options to be used by the compiler
 COPTIONS = 
