diff -ru trunk_nosvn.orig/src/utils/GraphUtils.cpp trunk_nosvn/src/utils/GraphUtils.cpp
--- trunk_nosvn.orig/src/utils/GraphUtils.cpp	2012-07-31 13:55:50.000000000 +0200
+++ trunk_nosvn/src/utils/GraphUtils.cpp	2012-08-28 09:34:49.519811370 +0200
@@ -230,7 +230,7 @@
 	    p.property("pos", boost::get(boost::vertex_centrality,gr));
 	    p.property("pin", boost::get(boost::vertex_priority, gr));
 	    p.property("node_id", boost::get(boost::vertex_name, gr));
-	boost::write_graphviz(f,gr,p);
+	boost::write_graphviz_dp(f,gr,p);
 	f.close();
 
 	/*std::system("dot2tex test.dot > test.tex");
