diff -ru trunk_nosvn.orig/distutils/setup.py trunk_nosvn/distutils/setup.py
--- trunk_nosvn.orig/distutils/setup.py	2012-07-31 13:55:50.000000000 +0200
+++ trunk_nosvn/distutils/setup.py	2012-08-28 14:44:03.319215105 +0200
@@ -69,7 +69,7 @@
     # On linux, we want to put the data_files directly in the dist-packages folder (or at least, that's where get_python_lib points to in a default python installation)
     # So we don't want to put them in the folder caphe as in Windows, because then python doesn't find them.
     # So don't do this: data_files_path = os.path.join(get_python_lib(), 'caphe')
-    data_files_path = get_python_lib()
+    data_files_path = os.path.sep.join(get_python_lib().split(os.path.sep)[-3:])
 
 setup(name='Caphe',
       version=version,
@@ -83,5 +83,5 @@
       #package_data={'caphe': [allklu_pos]}
       data_files=[(data_files_path, data_files)],
       #cmdclass=dict(install=my_install)
-      cmdclass = {"install_data": post_install}
-      )
\ No newline at end of file
+      #cmdclass = {"install_data": post_install}
+      )
