diff -ruN foam-extend-3.0.orig/src/decompositionMethods/parMetisDecomp/Make/options foam-extend-3.0/src/decompositionMethods/parMetisDecomp/Make/options
--- foam-extend-3.0.orig/src/decompositionMethods/parMetisDecomp/Make/options	2014-02-27 09:52:22.558463656 +0100
+++ foam-extend-3.0/src/decompositionMethods/parMetisDecomp/Make/options	2014-02-27 11:51:10.259206062 +0100
@@ -9,4 +9,4 @@
 
 LIB_LIBS = \
     -L$(FOAM_MPI_LIBBIN) \
-    -L$(PARMETIS_LIB_DIR) -lparmetis
+    -L$(PARMETIS_LIB_DIR) -lparmetis -lmetis
diff -ruN foam-extend-3.0.orig/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options foam-extend-3.0/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options
--- foam-extend-3.0.orig/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options	2014-02-27 09:52:22.860463667 +0100
+++ foam-extend-3.0/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options	2014-02-27 11:48:54.290303826 +0100
@@ -7,9 +7,8 @@
 
 EXE_INC = \
     -I$(LIB_SRC)/finiteVolume/lnInclude \
-    -I$(PARMGRIDGEN_INCLUDE_DIR)/Lib    \
-    -I$(PARMGRIDGEN_INCLUDE_DIR)/IMlib  \
+    -I$(PARMGRIDGEN_INCLUDE_DIR)    \
     $(TYPE_REAL)
 
 LIB_LIBS = \
-    -L$(PARMGRIDGEN_LIB_DIR) -lMGridGen
+    -L$(PARMGRIDGEN_LIB_DIR) -lmgrid
