Author: Louwrens van Dellen <lthvd@protonmail.com>
Date:   Wed Jul 10 18:56:02 2024 +0200

setup() keyword causes an assertion error in modern setuptools

diff --git a/setup.py b/setup.py
index b15e0fc8..49650cfd 100644
--- a/setup.py
+++ b/setup.py
@@ -248,7 +248,6 @@ setup(
     # https://packaging.python.org/en/latest/single_source_version.html
     # The version is stored only in the setup.py file and read from it (option
     # 1 in https://packaging.python.org/en/latest/single_source_version.html)
-    use_scm_version=False,

     # Description of what tofu does
     description="A python library for Tomography for Fusion",
