diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8e77749..44b5a004 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,12 +93,11 @@ rocm_setup_version(VERSION ${VERSION_STRING})
 
 # Workaround until llvm and hip CMake modules fix symlink logic in their config files
 list(APPEND CMAKE_PREFIX_PATH
+  $ENV{EBROOTROCMMINLLVM}
+  $ENV{EBROOTHIP}
   ${ROCM_PATH}
   ${ROCM_PATH}/llvm
   ${ROCM_PATH}/hip
-  /opt/rocm
-  /opt/rocm/llvm
-  /opt/rocm/hip
 )
 
 if(CMAKE_BUILD_TYPE STREQUAL "Debug")
@@ -199,9 +198,9 @@ set(AMDGPU_TARGETS "${AMDGPU_TARGETS_INIT}"
   CACHE STRING "List of specific machine types for library to target")
 
 # Find HIP dependencies
-find_package(hip REQUIRED CONFIG PATHS ${ROCM_PATH} /opt/rocm)
+find_package(hip REQUIRED CONFIG PATHS $ENV{EBROOTHIP})
 
-find_package(rocblas REQUIRED CONFIG PATHS ${ROCM_PATH})
+find_package(rocblas REQUIRED CONFIG PATHS ENV{EBROOTROCBLAS})
 get_imported_target_location(location roc::rocblas)
 message(STATUS "Found rocBLAS: ${location}")
 set(rocblas_minimum 4.4)
@@ -210,7 +209,7 @@ rocm_package_add_rpm_dependencies(STATIC_DEPENDS "rocblas-static-devel >= ${rocb
 rocm_package_add_deb_dependencies(STATIC_DEPENDS "rocblas-static-dev >= ${rocblas_minimum}")
 
 if(BUILD_WITH_SPARSE)
-  find_package(rocsparse REQUIRED CONFIG PATHS ${ROCM_PATH})
+  find_package(rocsparse REQUIRED CONFIG PATHS ENV{EBROOTROCSPARSE})
   get_imported_target_location(location roc::rocsparse)
   message(STATUS "Found rocSPARSE: ${location}")
   set(rocsparse_minimum 2.2)
@@ -222,7 +221,7 @@ else()
   list(APPEND CPACK_RPM_RUNTIME_PACKAGE_SUGGESTS "rocsparse")
 endif()
 
-find_package(rocprim REQUIRED CONFIG PATHS ${ROCM_PATH})
+find_package(rocprim REQUIRED CONFIG PATHS ENV{EBROOTROCPRIM})
 rocm_package_add_rpm_dependencies(STATIC_DEPENDS "rocprim-static-devel")
 rocm_package_add_deb_dependencies(STATIC_DEPENDS "rocprim-static-dev")
 
@@ -317,7 +316,7 @@ if(BUILD_CODE_COVERAGE)
     COMMAND mkdir -p lcoverage
     COMMAND echo "\\#!/bin/bash" > llvm-gcov.sh
     COMMAND echo "\\# THIS FILE HAS BEEN GENERATED" >> llvm-gcov.sh
-    COMMAND printf "exec /opt/rocm/llvm/bin/llvm-cov gcov $$\\@" >> llvm-gcov.sh
+    COMMAND printf "exec $ENV{EBROOTROCMMINLLVM}/bin/llvm-cov gcov $$\\@" >> llvm-gcov.sh
     COMMAND chmod +x llvm-gcov.sh
   )
 
