From c61f113e6c6904a1ce6b2f53805c84dc8a2ab5ef Mon Sep 17 00:00:00 2001
From: cunshunxia <cunshunxia@tencent.com>
Date: Thu, 24 Aug 2023 02:08:50 +0000
Subject: [PATCH] fix compiler error of address will never be NULL

---
 test/common/doctest.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test/common/doctest.h b/test/common/doctest.h
index 8714c5b260..413a5b3fe6 100644
--- a/test/common/doctest.h
+++ b/test/common/doctest.h
@@ -1555,10 +1555,13 @@ DOCTEST_CLANG_SUPPRESS_WARNING_WITH_PUSH("-Wunused-comparison")
                 , m_at(at) {}
 
         DOCTEST_NOINLINE operator Result() {
+// TODO: upstream the change to doctest : Work-around for the warning: 'address will never be NULL'
+DOCTEST_GCC_SUPPRESS_WARNING_WITH_PUSH("-Waddress")
 // this is needed only for MSVC 2015
 DOCTEST_MSVC_SUPPRESS_WARNING_WITH_PUSH(4800) // 'int': forcing value to bool
             bool res = static_cast<bool>(lhs);
 DOCTEST_MSVC_SUPPRESS_WARNING_POP
+DOCTEST_GCC_SUPPRESS_WARNING_POP
             if(m_at & assertType::is_false) { //!OCLINT bitwise operator in conditional
                 res = !res;
             }
