diff -ru msmc2-2.1.3.orig/Makefile msmc2-2.1.3/Makefile
--- msmc2-2.1.3.orig/Makefile	2022-08-31 13:46:10.621140934 +0000
+++ msmc2-2.1.3/Makefile	2022-08-31 13:46:36.547319977 +0000
@@ -1,6 +1,7 @@
 # Set this variable to your static gsl libraries
 GSLDIR=/usr/local/lib
 GSL=${GSLDIR}/libgsl.a ${GSLDIR}/libgslcblas.a
+DMD=dmd
 
 release : build/release/msmc2
 
@@ -14,18 +15,18 @@
 	build/debug/unittest
 
 build/debug/msmc2 : model/*.d powell.d brent.d maximization_step.d expectation_step.d msmc2.d logger.d
-	dmd -debug ${GSL} -odbuild/test -of$@ $^
+	${DMD} -debug ${GSL} -odbuild/test -of$@ $^
 
 build/release/msmc2 : model/*.d powell.d brent.d maximization_step.d expectation_step.d msmc2.d logger.d
-	dmd -O ${GSL} -odbuild/test -of$@ $^
+	${DMD} -O ${GSL} -odbuild/test -of$@ $^
 
 build/debug/unittest : model/*.d
-	dmd -unittest ${GSL} -main -odbuild/debug -ofbuild/debug/unittest $^
+	${DMD} -unittest ${GSL} -main -odbuild/debug -ofbuild/debug/unittest $^
 
 decode : build/decode
 
 build/decode : model/*.d decode.d 
-	dmd -O ${GSL} -odbuild -of$@ $^
+	${DMD} -O ${GSL} -odbuild -of$@ $^
 
 clean :
 	rm -rf build/debug/* build/release/*
