--- VTK/GUISupport/Qt/CMakeLists.txt.orig	2019-12-10 17:30:09.749965553 +0000
+++ VTK/GUISupport/Qt/CMakeLists.txt	2019-12-10 17:27:55.909689682 +0000
@@ -66,6 +66,7 @@
 
 find_package(Qt5 COMPONENTS Widgets ${qt_component_x11extras} REQUIRED QUIET)
 include_directories(${Qt5Widgets_INCLUDE_DIRS})
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 add_definitions(${Qt5Widgets_DEFINITIONS})
 
 if(qt_component_x11extras)
--- Qt/Components/CMakeLists.txt.orig	2019-12-10 23:08:34.774182231 +0000
+++ Qt/Components/CMakeLists.txt	2019-12-10 22:32:51.444672641 +0000
@@ -560,6 +560,7 @@
 )
 
 include(ParaViewQt)
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 pv_find_package_qt(qt_private_targets REQUIRED QUIET
   QT5_COMPONENTS Widgets Network)
 pv_qt_wrap_cpp(__Module_MOC_SRCS ${Module_MOC_HDRS})
--- Qt/ApplicationComponents/CMakeLists.txt.orig	2019-12-11 16:57:27.396847840 +0000
+++ Qt/ApplicationComponents/CMakeLists.txt	2019-12-11 16:57:20.579783405 +0000
@@ -554,6 +554,7 @@
 )
 
 include(ParaViewQt)
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 if (PARAVIEW_USE_QTHELP)
   pv_find_package_qt(qt_private_targets REQUIRED QUIET
     QT5_COMPONENTS Widgets Help)
--- Plugins/EmbossingRepresentations/CMakeLists.txt.orig	2018-06-15 19:09:45.000000000 +0000
+++ Plugins/EmbossingRepresentations/CMakeLists.txt	2019-12-11 23:27:54.092152255 +0000
@@ -11,6 +11,7 @@
   ${VTK_INCLUDE_DIRS}
 )
 
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 if (PARAVIEW_BUILD_QT_GUI)
   pv_find_package_qt(qt_targets REQUIRED QUIET
     QT5_COMPONENTS Widgets)
--- Plugins/LagrangianParticleTracker/CMakeLists.txt.orig	2019-12-12 04:56:03.405560113 +0000
+++ Plugins/LagrangianParticleTracker/CMakeLists.txt	2019-12-12 04:56:18.452702164 +0000
@@ -1,5 +1,5 @@
 include_directories(${VTK_INCLUDE_DIRS})
-
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 if (PARAVIEW_BUILD_QT_GUI)
   pv_find_package_qt(qt_targets REQUIRED QUIET
     QT5_COMPONENTS Widgets)
--- Plugins/NonOrthogonalSource/CMakeLists.txt.orig	2019-12-12 16:37:55.337046178 +0000
+++ Plugins/NonOrthogonalSource/CMakeLists.txt	2019-12-12 16:38:14.549227029 +0000
@@ -2,7 +2,7 @@
 SET(PLUGIN_VERSION "1.0")
 
 include_directories(${VTK_INCLUDE_DIRS})
-
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 set(sources)
 
 if(PARAVIEW_BUILD_QT_GUI)
--- Plugins/SLACTools/CMakeLists.txt.orig	2019-12-12 17:00:58.314061238 +0000
+++ Plugins/SLACTools/CMakeLists.txt	2019-12-12 17:01:15.636224225 +0000
@@ -7,7 +7,7 @@
 INCLUDE_DIRECTORIES(
   ${VTK_INCLUDE_DIRS}
   )
-
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 SET(SM_XML
   SLACToolsSM.xml
   )
--- Plugins/SierraPlotTools/CMakeLists.txt.orig	2019-12-12 17:24:28.467329701 +0000
+++ Plugins/SierraPlotTools/CMakeLists.txt	2019-12-12 17:25:05.262675927 +0000
@@ -37,6 +37,7 @@
   ${VTK_INCLUDE_DIRS}
   )
 
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 SET(UI_FILES
   # pqSierraPlotToolsActionHolder.ui -- container for plot tools actions
   pqSierraPlotToolsActionHolder.ui
--- Plugins/StreamLinesRepresentation/CMakeLists.txt.orig	2019-12-12 17:46:18.713653824 +0000
+++ Plugins/StreamLinesRepresentation/CMakeLists.txt	2019-12-12 17:46:43.930890975 +0000
@@ -11,7 +11,7 @@
   )
 
 include(vtkOpenGL)
-
+include_directories("$ENV{EBROOTGCCCORE}/include/c++/$ENV{EBVERSIONGCCCORE}/x86_64-pc-linux-gnu/")
 set(SRCS)
 
 if(PARAVIEW_BUILD_QT_GUI)
