to avoid the `terminal_columns': undefined method `winsize' for nil (NoMethodError)` error
Author: Petr Král (INUITS)
--- lib/transrate/cmdline.rb.orig	2016-06-06 15:10:26.000000000 +0200
+++ lib/transrate/cmdline.rb	2024-07-25 11:23:20.250369056 +0200
@@ -87,11 +87,6 @@
     end
   end
 
-  def terminal_columns
-    require 'io/console'
-    IO.console.winsize.last
-  end
-
   def help_message
   <<-EOS
 
@@ -117,19 +112,17 @@
   end
 
   def transrate_banner
-    if terminal_columns > 70
-      txp = '░▓▓▓^▓▓▓░'
-      toptxp = txp.green
-      midtxp = txp.yellow
-      bottxp = txp.red
-      puts <<-EOS
+    txp = '░▓▓▓^▓▓▓░'
+    toptxp = txp.green
+    midtxp = txp.yellow
+    bottxp = txp.red
+    puts <<-EOS
            _                                        _
           | |_  _ __  __ _  _ __   ___  _ __  __ _ | |_  ___
 #{toptxp} | __|| '__|/ _` || '_ \\ / __|| '__|/ _` || __|/ _ \\ #{toptxp}
 #{midtxp} | |_ | |  | (_| || | | |\\__ \\| |  | (_| || |_|  __/ #{midtxp}
 #{bottxp}  \\__||_|   \\__,_||_| |_||___/|_|   \\__,_| \\__|\\___| #{bottxp}
-      EOS
-    end
+    EOS
     ""
   end
 
