From ecb4d4c11f683ea178a15fc9bcf08a5fb53b1d8d Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Mon, 5 Aug 2019 22:10:28 +0200
Subject: [PATCH] Add glib and gobject libs for text2image

---
 src/training/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/training/CMakeLists.txt b/src/training/CMakeLists.txt
index 9296d6bd9..3ab89528c 100644
--- a/src/training/CMakeLists.txt
+++ b/src/training/CMakeLists.txt
@@ -271,6 +271,7 @@ pkg_check_modules(Cairo REQUIRED cairo)
 pkg_check_modules(PangoFt2 REQUIRED pangoft2)
 pkg_check_modules(PangoCairo REQUIRED pangocairo)
 pkg_check_modules(FontConfig REQUIRED fontconfig)
+pkg_check_modules(Glib REQUIRED glib-2.0 gobject-2.0)
 endif()
 
 set(text2image_src
@@ -306,6 +307,7 @@ target_link_libraries       (text2image
     ${PangoCairo_LIBRARIES}
     ${PangoFt2_LIBRARIES}
     ${FontConfig_LIBRARIES}
+    ${Glib_LIBRARIES}
 )
 endif()
 if (CPPAN_BUILD)
