specify FFmpeg install location in PyFFmpeg's setup.py
author: Kenneth Hoste (HPC-UGent)
--- pyffmpeg-2.1beta/setup.py.orig	2011-01-12 16:46:15.000000000 +0100
+++ pyffmpeg-2.1beta/setup.py	2016-02-17 10:52:57.454775826 +0100
@@ -33,7 +33,7 @@
     
 else:
     ffmpegpath = '/opt/ffmpeg'
-    for x in [ os.environ["HOME"]+"build/ffmpeg",  '/usr/local/ffmpeg',  '/opt/ffmpeg' ]:
+    for x in [os.environ['EBROOTFFMPEG'], os.environ["HOME"]+"build/ffmpeg",  '/usr/local/ffmpeg',  '/opt/ffmpeg' ]:
         try:
              os.stat(x)
              ffmpegpath = x
@@ -69,6 +69,7 @@
     map(lambda x: x not in result and result.append(x), deps)
     return result
 
+print "ffmpegpath: %s" % ffmpegpath
 
 libs = [ 'avformat', 'avcodec', 'avutil', 'swscale' ]
 incdir = [ path_join(ffmpegpath, 'include'), "/usr/include/ffmpeg" , "./include" ] 
