Associate the finalize script with the correct $HOME directory
author: Stefan Harfst (HPC-UniOldenburg)

--- HEN_HOUSE/scripts/finalize_egs_foruser.orig	2018-10-30 15:50:18.278569800 +0100
+++ HEN_HOUSE/scripts/finalize_egs_foruser	2018-10-30 15:53:04.048167614 +0100
@@ -98,7 +98,7 @@
 
 read response
 
-exec 3>$my_dir/finalize_egs_$USER.log
+exec 3>$HOME/finalize_egs_$USER.log
 cat >&3 <<EOF
 ###############################################################################
 EGSnrc configuration for the user $USER
