Put lockfile in a temp directory
author: Alex Domingo (Vrije Universiteit Brussel)
--- a/mujoco_py/builder.py	2022-02-09 01:20:09.349870000 +0100
+++ b/mujoco_py/builder.py	2022-02-09 01:20:17.548552000 +0100
@@ -89,7 +89,8 @@
     builder = Builder(mujoco_path)
     cext_so_path = builder.get_so_file_path()
 
-    lockpath = os.path.join(os.path.dirname(cext_so_path), 'mujocopy-buildlock')
+    tmpdir_path = os.getenv('TMPDIR', '/tmp')
+    lockpath = os.path.join(tmpdir_path, 'mujocopy-buildlock')
 
     with fasteners.InterProcessLock(lockpath):
         mod = None
