diff -Naur phyx-0.999_orig/src/Makefile.in phyx-0.999/src/Makefile.in
--- phyx-0.999_orig/src/Makefile.in	2018-10-23 22:03:34.183795129 +0000
+++ phyx-0.999/src/Makefile.in	2018-10-25 16:08:03.057968624 +0000
@@ -1,7 +1,7 @@
 RM := rm -f
 ## Use compiler from configure. May be specified by user with e.g.:
 ## ./configure CXX=/usr/local/bin/g++-7
-CXX := @CXX@
+CXX = icc 
 
 CPPFLAGS := @CPPFLAGS@
 LDFLAGS := @LDFLAGS@
@@ -11,7 +11,7 @@
 HARM := @HARM@
 HOMP := @HOMP@
 
-CPP_LIBS = -llapack -lblas -lpthread -lm
+CPP_LIBS = -mkl -lpthread -lm
 ifeq "$(HNLOPT)" "Y"
     CPP_LIBS += -lnlopt
 endif
@@ -29,7 +29,7 @@
 endif
 
 ifneq "$(CXX)" "icc"
-    OPT_FLAGS += -ffast-math -ftree-vectorize
+    OPT_FLAGS += -O2     #-ffast-math -ftree-vectorize
 endif
 
 prefix := @prefix@
@@ -39,7 +39,7 @@
     NLOPT_PROGRAMS += pxbdfit
 endif
 
-FFLAGS	= -O3
+FFLAGS	= -O2
 
 CONSEQ_SRC = main_consq.cpp
 CONSEQ_OBJ = main_consq.o
