--- Make.top.orig	2012-03-31 08:46:46.371409361 +0200
+++ Make.top	2012-03-31 08:47:04.165487445 +0200
@@ -293,10 +293,20 @@
 	cp $(LIBdir)/libcblas.a $(INSTdir)/.
 	cp $(LIBdir)/liblapack.a $(INSTdir)/.
 	chmod 0644 $(INSTdir)/libatlas.a  $(INSTdir)/liblapack.a \
-                   $(INSTdir)/libcblas.a 
+	               $(INSTdir)/libcblas.a
 	- cp $(LIBdir)/libf77blas.a $(INSTdir)/.
 	- chmod 0644 $(INSTdir)/libf77blas.a
 	- cp $(LIBdir)/libptcblas.a $(INSTdir)/.
 	- cp $(LIBdir)/libptf77blas.a $(INSTdir)/.
 	- chmod 0644 $(INSTdir)/libptcblas.a $(INSTdir)/libptf77blas.a
+	cp $(LIBdir)/libatlas.so $(INSTdir)/.
+	cp $(LIBdir)/libcblas.so $(INSTdir)/.
+	cp $(LIBdir)/liblapack.so $(INSTdir)/.
+	chmod 0644 $(INSTdir)/libatlas.so  $(INSTdir)/liblapack.so \
+	                $(INSTdir)/libcblas.so 
+	- cp $(LIBdir)/libf77blas.so $(INSTdir)/.
+	- chmod 0644 $(INSTdir)/libf77blas.so
+	- cp $(LIBdir)/libptcblas.so $(INSTdir)/.
+	- cp $(LIBdir)/libptf77blas.so $(INSTdir)/.
+	- chmod 0644 $(INSTdir)/libptcblas.so $(INSTdir)/libptf77blas.so
 
