Don’t install pyyaml with pip, use EasyBuild provided pyyaml
Author: Samuel Moors (Vrije Universiteit Brussel)
diff -ur hpc-container-wrapper-0.4.2.orig/install.sh hpc-container-wrapper-0.4.2/install.sh
--- hpc-container-wrapper-0.4.2.orig/install.sh	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/install.sh	2025-09-07 11:18:27.335172061 +0200
@@ -15,21 +15,9 @@
     echo "Could not find python3"
     exit 1 
 fi
-if which pip3 &> /dev/null;then
-    PIP=pip3
-elif which pip &> /dev/null;then
-    PIP=pip
-else
-    echo "Could not find pip3 or pip"
-    exit 1 
-fi
 
 mkdir -p default_config
 ln -sf ../configs/$1.yaml default_config/config.yaml
 echo "Using python3: $(which python3)"
-unset PYTHONUSERBASE
-export PYTHONUSERBASE=PyDeps
 echo "export PY_EXE=$(which python3)" > frontends/inst_vars.sh
-echo "Installing pyyaml with $PIP"
-$PIP install --user pyyaml
 
