Don’t install pyyaml with pip, use EasyBuild provided pyyaml
Author: Samuel Moors (Vrije Universiteit Brussel)

diff -ur hpc-container-wrapper-0.4.2.orig/construct.py hpc-container-wrapper-0.4.2/construct.py
--- hpc-container-wrapper-0.4.2.orig/construct.py	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/construct.py	2025-09-08 13:12:16.745116267 +0200
@@ -6,7 +6,6 @@
 import random
 curr_dir=pathlib.Path(__file__).parent.resolve()
 info=sys.version_info
-sys.path.insert(0,str(curr_dir)+"/PyDeps/lib/python{}.{}/site-packages".format(info[0],info[1]))
 sys.path.insert(0,str(curr_dir))
 import yaml
 from string import Template
diff -ur hpc-container-wrapper-0.4.2.orig/frontends/conda-containerize.py hpc-container-wrapper-0.4.2/frontends/conda-containerize.py
--- hpc-container-wrapper-0.4.2.orig/frontends/conda-containerize.py	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/frontends/conda-containerize.py	2025-09-08 13:12:16.748801710 +0200
@@ -6,7 +6,6 @@
 root_dir=pathlib.Path(curr_dir).parent.resolve()
 info=sys.version_info
 sys.path.insert(0,str(root_dir))
-sys.path.insert(0,str(root_dir)+"/PyDeps/lib/python{}.{}/site-packages".format(info[0],info[1]))
 import yaml
 from cw_common import *
 from script_shared import *
diff -ur hpc-container-wrapper-0.4.2.orig/frontends/pip-containerize.py hpc-container-wrapper-0.4.2/frontends/pip-containerize.py
--- hpc-container-wrapper-0.4.2.orig/frontends/pip-containerize.py	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/frontends/pip-containerize.py	2025-09-08 13:12:16.773422243 +0200
@@ -6,7 +6,6 @@
 root_dir=pathlib.Path(curr_dir).parent.resolve()
 info=sys.version_info
 sys.path.insert(0,str(root_dir))
-sys.path.insert(0,str(root_dir)+"/PyDeps/lib/python{}.{}/site-packages".format(info[0],info[1]))
 import yaml
 from cw_common import *
 from script_shared import *
diff -ur hpc-container-wrapper-0.4.2.orig/frontends/script_shared.py hpc-container-wrapper-0.4.2/frontends/script_shared.py
--- hpc-container-wrapper-0.4.2.orig/frontends/script_shared.py	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/frontends/script_shared.py	2025-09-08 13:12:16.776658768 +0200
@@ -5,7 +5,6 @@
 root_dir=pathlib.Path(curr_dir).parent.resolve()
 info=sys.version_info
 sys.path.insert(0,str(root_dir))
-sys.path.insert(0,str(root_dir)+"/PyDeps/lib/python{}.{}/site-packages".format(info[0],info[1]))
 import yaml
 from cw_common import *
 def is_valid_file(par,arg):
diff -ur hpc-container-wrapper-0.4.2.orig/frontends/wrap-container.py hpc-container-wrapper-0.4.2/frontends/wrap-container.py
--- hpc-container-wrapper-0.4.2.orig/frontends/wrap-container.py	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/frontends/wrap-container.py	2025-09-08 13:12:16.779227572 +0200
@@ -6,7 +6,6 @@
 root_dir=pathlib.Path(curr_dir).parent.resolve()
 info=sys.version_info
 sys.path.insert(0,str(root_dir))
-sys.path.insert(0,str(root_dir)+"/PyDeps/lib/python{}.{}/site-packages".format(info[0],info[1]))
 import yaml
 from cw_common import *
 from script_shared import *
diff -ur hpc-container-wrapper-0.4.2.orig/frontends/wrap-install.py hpc-container-wrapper-0.4.2/frontends/wrap-install.py
--- hpc-container-wrapper-0.4.2.orig/frontends/wrap-install.py	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/frontends/wrap-install.py	2025-09-08 13:12:16.781751750 +0200
@@ -6,7 +6,6 @@
 root_dir=pathlib.Path(curr_dir).parent.resolve()
 info=sys.version_info
 sys.path.insert(0,str(root_dir))
-sys.path.insert(0,str(root_dir)+"/PyDeps/lib/python{}.{}/site-packages".format(info[0],info[1]))
 import yaml
 from cw_common import *
 from script_shared import *
diff -ur hpc-container-wrapper-0.4.2.orig/install.sh hpc-container-wrapper-0.4.2/install.sh
--- hpc-container-wrapper-0.4.2.orig/install.sh	2025-03-03 09:28:57.000000000 +0100
+++ hpc-container-wrapper-0.4.2/install.sh	2025-09-07 11:18:27.335172061 +0200
@@ -15,21 +15,9 @@
     echo "Could not find python3"
     exit 1 
 fi
-if which pip3 &> /dev/null;then
-    PIP=pip3
-elif which pip &> /dev/null;then
-    PIP=pip
-else
-    echo "Could not find pip3 or pip"
-    exit 1 
-fi
 
 mkdir -p default_config
 ln -sf ../configs/$1.yaml default_config/config.yaml
 echo "Using python3: $(which python3)"
-unset PYTHONUSERBASE
-export PYTHONUSERBASE=PyDeps
 echo "export PY_EXE=$(which python3)" > frontends/inst_vars.sh
-echo "Installing pyyaml with $PIP"
-$PIP install --user pyyaml
 
