diff -ur netcdf-c-4.4.1.1.orig/ncdump/isnan.h netcdf-c-4.4.1.1/ncdump/isnan.h
--- netcdf-c-4.4.1.1.orig/ncdump/isnan.h	2016-11-21 18:27:08.000000000 +0000
+++ netcdf-c-4.4.1.1/ncdump/isnan.h	2017-10-19 12:12:22.463095334 +0000
@@ -11,6 +11,11 @@
 #endif /* NO_FLOAT_H */
 #include <math.h>
 
+/* work around icc 14.0.4 incompatibility */
+#undef isinf
+#  define isinf(x) \
+     (sizeof (x) == sizeof (float) ? __isinff (x) : __isinf (x))
+
 #if ! (defined(isinf) || HAVE_DECL_ISINF)
 #define isinf(x) (DBL_MAX/((double)(x))==0.0)
 #endif /* !HAVE_DECL_ISINF */
