diff -Nru netcdf-c-4.4.1.orig/include/CMakeLists.txt netcdf-c-4.4.1/include/CMakeLists.txt
--- netcdf-c-4.4.1.orig/include/CMakeLists.txt	2016-09-19 15:25:52.905845000 +1200
+++ netcdf-c-4.4.1/include/CMakeLists.txt	2016-09-19 15:26:10.333653000 +1200
@@ -14,7 +14,7 @@
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
   COMPONENT headers)
 
-IF(ENABLE_PNETCDF OR ENABLE_PARALLEL)
+IF(ENABLE_PNETCDF OR ENABLE_PARALLEL4)
   INSTALL(FILES ${netCDF_SOURCE_DIR}/include/netcdf_par.h
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     COMPONENT headers)
