--- platform_specific.mk.x86_64-linux.orig	2018-04-18 15:16:22.343598081 +0000
+++ platform_specific.mk.x86_64-linux	2018-04-18 15:18:52.955216411 +0000
@@ -14,6 +14,7 @@
 	MYINCLUDES	= -I. \
 			  -I$(READLINE_DIR)/include \
 			  -I$(NETCDF4_DIR)/include \
+			  -I$(EBROOTNETCDFMINFORTRAN)/include \
 			  -I$(DIR_PREFIX)/fer/common \
 			  -I$(DIR_PREFIX)/fmt/cmn \
 			  -I$(DIR_PREFIX)/ppl/tmap_inc \
@@ -23,6 +24,7 @@
 else
 	MYINCLUDES	= -I. \
 			  -I$(NETCDF4_DIR)/include \
+			  -I$(EBROOTNETCDFMINFORTRAN)/include \
 			  -I$(DIR_PREFIX)/fer/common \
 			  -I$(DIR_PREFIX)/fmt/cmn \
 			  -I$(DIR_PREFIX)/ppl/tmap_inc \
@@ -107,7 +109,7 @@
 			  -ldl \
 			  -lz \
 			  -Wl,-Bstatic -lgfortran -Wl,-Bdynamic \
-			  -lm
+			  -lm -lquadmath
 else
 	SYSLIB		= -lX11 \
 			  -lreadline \
@@ -116,14 +118,14 @@
 			  -ldl \
 			  -lz \
 			  -Wl,-Bstatic -lgfortran -Wl,-Bdynamic \
-			  -lm
+			  -lm -lquadmath
 endif
 
 # For netCDF4 using HDF5
-	CDFLIB		= $(NETCDF4_DIR)/lib64/libnetcdff.a \
-			  $(NETCDF4_DIR)/lib64/libnetcdf.a \
-			  $(HDF5_DIR)/lib64/libhdf5_hl.a \
-			  $(HDF5_DIR)/lib64/libhdf5.a
+	CDFLIB		= $(EBROOTNETCDFMINFORTRAN)/lib/libnetcdff.a \
+			  $(NETCDF4_DIR)/lib64/libnetcdf.a \
+			  $(HDF5_DIR)/lib/libhdf5_hl.a \
+			  $(HDF5_DIR)/lib/libhdf5.a
 
 # Directly compile the .F source files to the .o object files
 # since gfortran can handle the C compiler directives in Fortran code
