--- site_specific.mk.in.orig	2017-12-01 17:32:35.000000000 +0000
+++ site_specific.mk.in	2018-04-17 21:24:05.983256962 +0000
@@ -7,7 +7,7 @@
 ## Do not use $(shell pwd) since this is included in Makefiles in other directories.
 ##
 # DIR_PREFIX = $(HOME)/build/trunk/FERRET
-DIR_PREFIX = $(HOME)/git/Ferret
+DIR_PREFIX = BUILDDIR
 # DIR_PREFIX = $(HOME)/svn/ferret
 
 ##
@@ -26,7 +26,7 @@
 ##
 # INSTALL_FER_DIR = $(HOME)/ferret_distributions/rhel6_64
 # INSTALL_FER_DIR = $(HOME)/Ferret32
-INSTALL_FER_DIR = $(HOME)/Ferret64
+INSTALL_FER_DIR = INSTALLDIR
 # INSTALL_FER_DIR = $(FER_DIR)
 
 ##
@@ -36,7 +36,7 @@
 # HDF5_DIR	= /usr
 # HDF5_DIR	= /usr/local
 # HDF5_DIR	= /usr/local/hdf5-1.8.18
-HDF5_DIR	= /usr/local/hdf5-1.8.18-64
+HDF5_DIR	= ${EBROOTHDF5}
 # HDF5_DIR	= /usr/local/hdf5-1.8.18-32
 
 ##
@@ -46,7 +46,7 @@
 # NETCDF4_DIR	= /usr
 # NETCDF4_DIR	= /usr/local
 # NETCDF4_DIR	= /usr/local/netcdf-4.4.1.1
-NETCDF4_DIR	= /usr/local/netcdf-4.4.1.1-64
+NETCDF4_DIR	= ${EBROOTNETCDF}
 # NETCDF4_DIR	= /usr/local/netcdf-4.4.1.1-32
 
 ##
