--- fsl/etc/fslconf/fslpython_install.sh.orig	2019-04-29 22:26:48.355659475 +0000
+++ fsl/etc/fslconf/fslpython_install.sh	2019-05-02 16:29:20.007680473 +0000
@@ -59,7 +59,7 @@
 shift $((OPTIND-1))
 
 [ "$1" = "--" ] && shift
-
+fsl_dir=${HOME}
 if [ -z "${fsl_dir}" ]; then
     if [ -z "${FSLDIR}" ]; then
         echo "Error - FSLDIR not given as an argument and \$FSLDIR not set!" >&2
@@ -218,6 +218,30 @@
 else
   ln -sf "${fslpython_env_dir}/bin/python" "${fsl_dir}/bin/fslpythonw"
 fi
+if [ -e "${fslpython_env_dir}/bin/atlasq" ]; then
+    ln -sf "${fslpython_env_dir}/bin/atlasq" "${fsl_dir}/bin/atlasq"
+fi
+if [ -e "${fslpython_env_dir}/bin/atlasquery" ]; then
+    ln -sf "${fslpython_env_dir}/bin/atlasquery" "${fsl_dir}/bin/atlasquery"
+fi
+if [ -e "${fslpython_env_dir}/bin/eddy_quad" ]; then
+    ln -sf "${fslpython_env_dir}/bin/eddy_quad" "${fsl_dir}/bin/eddy_quad"
+fi
+if [ -e "${fslpython_env_dir}/bin/eddy_squad" ]; then
+    ln -sf "${fslpython_env_dir}/bin/eddy_squad" "${fsl_dir}/bin/eddy_squad"
+fi
+if [ -e "${fslpython_env_dir}/bin/fsl_ents" ]; then
+    ln -sf "${fslpython_env_dir}/bin/fsl_ents" "${fsl_dir}/bin/fsl_ents"
+fi
+if [ -e "${fslpython_env_dir}/bin/imcp" ]; then
+    ln -sf "${fslpython_env_dir}/bin/imcp" "${fsl_dir}/bin/imcp"
+fi
+if [ -e "${fslpython_env_dir}/bin/imglob" ]; then
+    ln -sf "${fslpython_env_dir}/bin/imglob" "${fsl_dir}/bin/imglob"
+fi
+if [ -e "${fslpython_env_dir}/bin/immv" ]; then
+    ln -sf "${fslpython_env_dir}/bin/immv" "${fsl_dir}/bin/immv"
+fi
 
 if [ ${debug} -ne 1 ]; then
     echo "${miniconda_install_log}"
