diff -ru fsl.patched/src/mist-clean/common/common.pri fsl/src/mist-clean/common/common.pri
--- fsl.patched/src/mist-clean/common/common.pri	2021-02-24 20:10:40.120663315 +0000
+++ fsl/src/mist-clean/common/common.pri	2021-02-24 20:12:34.762254522 +0000
@@ -30,7 +30,7 @@
 LIBS += -lboost_log$$BOOSTSUFFIX -lboost_log_setup$$BOOSTSUFFIX -lboost_thread$$BOOSTSUFFIX \
 	-lboost_filesystem$$BOOSTSUFFIX -lboost_date_time$$BOOSTSUFFIX -lboost_chrono$$BOOSTSUFFIX \
 	-lboost_system$$BOOSTSUFFIX -lboost_serialization$$BOOSTSUFFIX -lboost_regex$$BOOSTSUFFIX
-LIBS += -lrt -lpthread -ldl $$SQLITEDIR/lib/libsqlite3.a
+LIBS += -lrt -lpthread -ldl -lsqlite3
 
 # NOTE: This is not compiled as a library because boost::serialization seems to require that all serialisation code
 # is moved out of the public header files in that case ...
diff -ru fsl.patched/src/mist-clean/Makefile fsl/src/mist-clean/Makefile
--- fsl.patched/src/mist-clean/Makefile	2021-02-24 20:10:40.055662980 +0000
+++ fsl/src/mist-clean/Makefile	2021-02-24 20:12:45.027307459 +0000
@@ -26,7 +26,7 @@
 			-lboost_log -lboost_log_setup -lboost_thread -lboost_filesystem -lboost_date_time -lboost_chrono \
 			-lboost_system -lboost_serialization -lboost_regex -lwarpfns -lbasisfield -lnewimage -lmiscmaths \
 			-lnewmat -lgdc -lgd -lpng -lNewNifti -lutils -lznz -lm -lz -lnlopt -lpthread -ldl \
-			${SQLITE_LIB}/libsqlite3.a ${LIBRT}
+			-lsqlite3 ${LIBRT}
 
 HDRS = common/gibbsshapemodel.h common/mrfshapemodel.h common/mvnshapemodel.h common/plotting.h common/profilefilters.h common/profilemixtures.h common/profilemodel.h common/profilepriors.h common/serialisation.h common/shape.h common/shapemodel.h common/stats.h common/transformation.h mist/builddate.h
 
