CUDA must now be disabled explicitly.
Author: Henrik R. Nagel (NTNU)
diff -Nru fsl.orig/config/buildSettings.mk fsl/config/buildSettings.mk
--- fsl.orig/config/buildSettings.mk	2020-06-19 15:48:02.000000000 +0200
+++ fsl/config/buildSettings.mk	2021-02-25 18:27:52.000000000 +0100
@@ -182,8 +182,8 @@
 #$(info GCC PAT VER $(GCCPAT))
 #Project specific variables
 EDDYBUILDPARAMETERS = "cuda=1 CUDAVER=8.0" "cuda=1 CUDAVER=9.1" "cpu=1"
-fdt_MASTERBUILD     = COMPILE_GPU = 1
-ptx2_MASTERBUILD    = COMPILE_GPU = 1
+fdt_MASTERBUILD     = COMPILE_GPU = 0
+ptx2_MASTERBUILD    = COMPILE_GPU = 0
 define newline
 
 
