patch out hardcoded compiler names and options
author: Kenneth Hoste (HPC-UGent)
diff -ru fsl.orig/config/apple-darwin10-gcc4.2/systemvars.mk fsl/config/apple-darwin10-gcc4.2/systemvars.mk
--- fsl.orig/config/apple-darwin10-gcc4.2/systemvars.mk	2009-11-03 18:02:14.000000000 +0100
+++ fsl/config/apple-darwin10-gcc4.2/systemvars.mk	2015-10-20 12:34:45.897743000 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
@@ -15,7 +15,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations
+OPTFLAGS :=  ${CFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/apple-darwin10-gcc4.2: systemvars.mk.orig
diff -ru fsl.orig/config/apple-darwin11-gcc4.2/systemvars.mk fsl/config/apple-darwin11-gcc4.2/systemvars.mk
--- fsl.orig/config/apple-darwin11-gcc4.2/systemvars.mk	2012-03-21 12:30:53.000000000 +0100
+++ fsl/config/apple-darwin11-gcc4.2/systemvars.mk	2015-10-20 12:34:45.898430000 +0200
@@ -15,7 +15,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3
+OPTFLAGS :=  -O3 -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -ansi -pedantic
 ANSI_FLAGS = ${GNU_ANSI_FLAGS}
Only in fsl/config/apple-darwin11-gcc4.2: systemvars.mk.orig
diff -ru fsl.orig/config/apple-darwin12-gcc4.2/systemvars.mk fsl/config/apple-darwin12-gcc4.2/systemvars.mk
--- fsl.orig/config/apple-darwin12-gcc4.2/systemvars.mk	2013-04-09 18:00:32.000000000 +0200
+++ fsl/config/apple-darwin12-gcc4.2/systemvars.mk	2015-10-20 12:34:45.899424000 +0200
@@ -15,7 +15,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3
+OPTFLAGS :=  -O3 -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -ansi -pedantic
 ANSI_FLAGS = ${GNU_ANSI_FLAGS}
Only in fsl/config/apple-darwin12-gcc4.2: systemvars.mk.orig
diff -ru fsl.orig/config/apple-darwin7-gcc3.1/systemvars.mk fsl/config/apple-darwin7-gcc3.1/systemvars.mk
--- fsl.orig/config/apple-darwin7-gcc3.1/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/apple-darwin7-gcc3.1/systemvars.mk	2015-10-20 12:34:45.900348000 +0200
@@ -7,8 +7,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -traditional-cpp -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/apple-darwin7-gcc3.1: systemvars.mk.orig
diff -ru fsl.orig/config/apple-darwin7-gcc3.3/systemvars.mk fsl/config/apple-darwin7-gcc3.3/systemvars.mk
--- fsl.orig/config/apple-darwin7-gcc3.3/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/apple-darwin7-gcc3.3/systemvars.mk	2015-10-20 12:34:45.901244000 +0200
@@ -7,8 +7,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/apple-darwin7-gcc3.3: systemvars.mk.orig
diff -ru fsl.orig/config/apple-darwin8-gcc4.0/systemvars.mk fsl/config/apple-darwin8-gcc4.0/systemvars.mk
--- fsl.orig/config/apple-darwin8-gcc4.0/systemvars.mk	2007-12-19 15:40:57.000000000 +0100
+++ fsl/config/apple-darwin8-gcc4.0/systemvars.mk	2015-10-20 12:34:45.902105000 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations
+OPTFLAGS :=  ${CFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/apple-darwin8-gcc4.0: systemvars.mk.orig
diff -ru fsl.orig/config/apple-darwin9-gcc4.0/systemvars.mk fsl/config/apple-darwin9-gcc4.0/systemvars.mk
--- fsl.orig/config/apple-darwin9-gcc4.0/systemvars.mk	2007-12-19 15:33:53.000000000 +0100
+++ fsl/config/apple-darwin9-gcc4.0/systemvars.mk	2015-10-20 12:34:45.902937000 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations
+OPTFLAGS :=  ${CFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/apple-darwin9-gcc4.0: systemvars.mk.orig
diff -ru fsl.orig/config/generic/systemvars.mk fsl/config/generic/systemvars.mk
--- fsl.orig/config/generic/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/generic/systemvars.mk	2015-10-20 12:34:45.903753000 +0200
@@ -16,8 +16,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -25,7 +25,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS =
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/generic: systemvars.mk.orig
diff -ru fsl.orig/config/gnu_64-gcc4.4/systemvars.mk fsl/config/gnu_64-gcc4.4/systemvars.mk
--- fsl.orig/config/gnu_64-gcc4.4/systemvars.mk	2011-04-19 10:47:52.000000000 +0200
+++ fsl/config/gnu_64-gcc4.4/systemvars.mk	2015-10-20 12:34:45.904633000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/gnu_64-gcc4.4: systemvars.mk.orig
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.2/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2015-10-20 12:34:45.905556000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc-2
-CXX = c++-2
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-deprecated
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/i686-pc-cygwin-gcc3.2: systemvars.mk.orig
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.3/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2015-10-20 12:34:45.906415000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-deprecated
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/i686-pc-cygwin-gcc3.3: systemvars.mk.orig
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.4/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2015-10-20 12:34:45.907306000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-deprecated
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/i686-pc-cygwin-gcc3.4: systemvars.mk.orig
diff -ru fsl.orig/config/linux_32-gcc2.96/systemvars.mk fsl/config/linux_32-gcc2.96/systemvars.mk
--- fsl.orig/config/linux_32-gcc2.96/systemvars.mk	2007-07-25 17:21:07.000000000 +0200
+++ fsl/config/linux_32-gcc2.96/systemvars.mk	2015-10-20 12:34:45.908177000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_32-gcc2.96: systemvars.mk.orig
diff -ru fsl.orig/config/linux_32-gcc3.2/systemvars.mk fsl/config/linux_32-gcc3.2/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.2/systemvars.mk	2007-07-25 17:21:08.000000000 +0200
+++ fsl/config/linux_32-gcc3.2/systemvars.mk	2015-10-20 12:34:45.909064000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_32-gcc3.2: systemvars.mk.orig
diff -ru fsl.orig/config/linux_32-gcc3.3/systemvars.mk fsl/config/linux_32-gcc3.3/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.3/systemvars.mk	2007-07-25 17:21:09.000000000 +0200
+++ fsl/config/linux_32-gcc3.3/systemvars.mk	2015-10-20 12:34:45.909865000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_32-gcc3.3: systemvars.mk.orig
diff -ru fsl.orig/config/linux_32-gcc3.4/systemvars.mk fsl/config/linux_32-gcc3.4/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.4/systemvars.mk	2007-07-25 17:21:10.000000000 +0200
+++ fsl/config/linux_32-gcc3.4/systemvars.mk	2015-10-20 12:34:45.910677000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_32-gcc3.4: systemvars.mk.orig
diff -ru fsl.orig/config/linux_32-gcc4.0/systemvars.mk fsl/config/linux_32-gcc4.0/systemvars.mk
--- fsl.orig/config/linux_32-gcc4.0/systemvars.mk	2007-07-25 17:21:11.000000000 +0200
+++ fsl/config/linux_32-gcc4.0/systemvars.mk	2015-10-20 12:34:45.911490000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_32-gcc4.0: systemvars.mk.orig
diff -ru fsl.orig/config/linux_32-gcc4.1/systemvars.mk fsl/config/linux_32-gcc4.1/systemvars.mk
--- fsl.orig/config/linux_32-gcc4.1/systemvars.mk	2012-04-20 11:37:28.000000000 +0200
+++ fsl/config/linux_32-gcc4.1/systemvars.mk	2015-10-20 12:34:45.912299000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_32-gcc4.1: systemvars.mk.orig
diff -ru fsl.orig/config/linux_64-gcc3.4/systemvars.mk fsl/config/linux_64-gcc3.4/systemvars.mk
--- fsl.orig/config/linux_64-gcc3.4/systemvars.mk	2007-07-25 17:21:12.000000000 +0200
+++ fsl/config/linux_64-gcc3.4/systemvars.mk	2015-10-20 12:34:45.913571000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_64-gcc3.4: systemvars.mk.orig
diff -ru fsl.orig/config/linux_64-gcc4.0/systemvars.mk fsl/config/linux_64-gcc4.0/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.0/systemvars.mk	2007-07-25 17:21:13.000000000 +0200
+++ fsl/config/linux_64-gcc4.0/systemvars.mk	2015-10-20 12:34:45.914423000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_64-gcc4.0: systemvars.mk.orig
diff -ru fsl.orig/config/linux_64-gcc4.1/systemvars.mk fsl/config/linux_64-gcc4.1/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.1/systemvars.mk	2007-07-25 11:19:45.000000000 +0200
+++ fsl/config/linux_64-gcc4.1/systemvars.mk	2015-10-20 12:34:45.915271000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_64-gcc4.1: systemvars.mk.orig
diff -ru fsl.orig/config/linux_64-gcc4.2/systemvars.mk fsl/config/linux_64-gcc4.2/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.2/systemvars.mk	2008-06-26 15:25:42.000000000 +0200
+++ fsl/config/linux_64-gcc4.2/systemvars.mk	2015-10-20 12:34:45.916140000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/linux_64-gcc4.2: systemvars.mk.orig
diff -ru fsl.orig/config/linux_64-gcc4.4/systemvars.mk fsl/config/linux_64-gcc4.4/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.4/systemvars.mk	2015-09-07 12:55:55.000000000 +0200
+++ fsl/config/linux_64-gcc4.4/systemvars.mk	2015-10-20 12:37:57.786741000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
@@ -30,6 +30,7 @@
 DEPENDFLAGS = -MM
 
 OPTFLAGS = -g -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
@@ -39,4 +40,4 @@
 CUDA_INSTALLATION = /opt/cuda-6.0
 LIB_CUDA = ${CUDA_INSTALLATION}/lib64
 INC_CUDA = ${CUDA_INSTALLATION}/inc
-NVCC = ${CUDA_INSTALLATION}/bin/nvcc
\ No newline at end of file
+NVCC = ${CUDA_INSTALLATION}/bin/nvcc
Only in fsl/config/linux_64-gcc4.4: systemvars.mk.orig
Only in fsl/config/linux_64-gcc4.4: systemvars.mk.rej
diff -ru fsl.orig/config/sparc-solaris2.8-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk
--- fsl.orig/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2007-07-13 13:00:21.000000000 +0200
+++ fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2015-10-20 12:34:45.920016000 +0200
@@ -5,14 +5,14 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 
 ARCHFLAGS = -mv8 -ffast-math -fomit-frame-pointer 
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O6 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS =
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/sparc-solaris2.8-gcc2.95: systemvars.mk.orig
diff -ru fsl.orig/config/sparc-solaris2.9-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk
--- fsl.orig/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2007-07-13 13:00:21.000000000 +0200
+++ fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2015-10-20 12:34:45.920855000 +0200
@@ -5,14 +5,14 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 
 ARCHFLAGS = -mv8 -ffast-math -fomit-frame-pointer
 ARCHLDFLAGS = -static
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O6 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS := ${CFLAGS} ${ARCHFLAGS} -wd803
 MACHDBGFLAGS =
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
Only in fsl/config/sparc-solaris2.9-gcc2.95: systemvars.mk.orig
diff -ru fsl.orig/src/film/Makefile fsl/src/film/Makefile
--- fsl.orig/src/film/Makefile	2013-03-21 12:16:04.000000000 +0100
+++ fsl/src/film/Makefile	2015-10-20 12:34:45.888426000 +0200
@@ -26,7 +26,7 @@
 	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} ftoz.o ${LIBS}
 
 film_gls:${OBJS} film_gls.o
-	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} film_gls.o ${LIBS} -l giftiio
+	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} film_gls.o ${LIBS} -lgiftiio
 
 film_gls_res:${OBJS} film_gls_res.o
 	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} film_gls_res.o ${LIBS}
Only in fsl/src/film: Makefile.orig
diff -ru fsl.orig/src/libmeshutils/Makefile fsl/src/libmeshutils/Makefile
--- fsl.orig/src/libmeshutils/Makefile	2012-07-23 15:25:20.000000000 +0200
+++ fsl/src/libmeshutils/Makefile	2015-10-20 12:34:45.886713000 +0200
@@ -3,7 +3,7 @@
 
 PROJNAME = meshUtils
 
-LD_LIBRARY_PATH=${FSLDIR}/lib
+#LD_LIBRARY_PATH=${FSLDIR}/lib
 
 USRINCFLAGS = -I${INC_NEWMAT} -I${INC_ZLIB} -I${INC_PROB} -I${INC_BOOST} 
 USRLDFLAGS =  -L${LIB_PROB} -L${LIB_NEWMAT} -L${LIB_ZLIB}
Only in fsl/src/libmeshutils: Makefile.orig
diff -ru fsl.orig/src/melodic/Makefile fsl/src/melodic/Makefile
--- fsl.orig/src/melodic/Makefile	2013-07-05 15:32:37.000000000 +0200
+++ fsl/src/melodic/Makefile	2015-10-20 12:34:45.887608000 +0200
@@ -3,7 +3,7 @@
 include ${FSLCONFDIR}/default.mk
 
 OPTFLAGS = -O3 -Wno-deprecated -ggdb
-OPTFLAGS_alphaev6-dec-osf5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
+#OPTFLAGS_alphaev6-dec-osf5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
 
 PROJNAME = melodic
 
Only in fsl/src/melodic: Makefile.orig
