--- fsl/etc/fslconf/fslpython_install.sh.orig	2019-04-14 03:56:09.238441864 +0000
+++ fsl/etc/fslconf/fslpython_install.sh	2019-04-14 03:56:46.083503448 +0000
@@ -30,9 +30,9 @@
 dropprivileges=0
 
 # Have we been called by sudo?
-if [ ! -z "${SUDO_UID}" ]; then
-    dropprivileges=1
-fi
+#if [ ! -z "${SUDO_UID}" ]; then
+#    dropprivileges=1
+#fi
 
 function syntax {
     echo "fslpython_install.sh [-f <FSLDIR>] [-q]"
