CUDA must now be disabled explicitly.
Author: Henrik R. Nagel (NTNU), ported to FSL 6.0.5.1 by Kenneth Hoste (HPC-UGent)
--- fsl/config/buildSettings.mk.orig	2022-06-22 19:09:08.833601000 +0200
+++ fsl/config/buildSettings.mk	2022-06-22 19:13:27.969222000 +0200
@@ -181,9 +181,9 @@
 #$(info GCC MIN VER $(GCCMIN))
 #$(info GCC PAT VER $(GCCPAT))
 #Project specific variables
-EDDYBUILDPARAMETERS = "cuda=1 CUDAVER=8.0" "cuda=1 CUDAVER=9.1" "cuda=1 CUDAVER=10.2" "cpu=1"
-fdt_MASTERBUILD     = COMPILE_GPU = 1
-ptx2_MASTERBUILD    = COMPILE_GPU = 1
+EDDYBUILDPARAMETERS = "cpu=1"
+fdt_MASTERBUILD     = COMPILE_GPU = 0
+ptx2_MASTERBUILD    = COMPILE_GPU = 0
 define newline
 
 
