diff -ru fsl.orig/src/fdt/CUDA/diffmodels.cuh fsl/src/fdt/CUDA/diffmodels.cuh
--- fsl.orig/src/fdt/CUDA/diffmodels.cuh	2021-07-27 13:30:38.611946614 +0000
+++ fsl/src/fdt/CUDA/diffmodels.cuh	2021-07-27 20:09:37.522602259 +0000
@@ -66,7 +66,7 @@
     University, to negotiate a licence. Contact details are:
     fsl@innovation.ox.ac.uk quoting Reference Project 9564, FSL.*/
 
-#include <device_vector.h>
+#include <thrust/device_vector.h>
 
 void fit_PVM_single(	//INPUT
 			const vector<ColumnVector> 	datam_vec, 
Seulement dans fsl/src/fdt/CUDA: diffmodels.cuh.orig
diff -ru fsl.orig/src/fdt/CUDA/runmcmc.cu fsl/src/fdt/CUDA/runmcmc.cu
--- fsl.orig/src/fdt/CUDA/runmcmc.cu	2021-07-27 13:30:38.336944792 +0000
+++ fsl/src/fdt/CUDA/runmcmc.cu	2021-07-27 20:09:37.558602519 +0000
@@ -71,8 +71,8 @@
 #include "runmcmc_kernels.cu"
 #include "sync_check.h"
 
-#include <host_vector.h>
-#include <device_vector.h> 
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h> 
 
 #include <time.h>
 #include <sys/time.h>
diff -ru fsl.orig/src/fdt/CUDA/xfibres_gpu.cu fsl/src/fdt/CUDA/xfibres_gpu.cu
--- fsl.orig/src/fdt/CUDA/xfibres_gpu.cu	2021-07-27 13:30:38.506945918 +0000
+++ fsl/src/fdt/CUDA/xfibres_gpu.cu	2021-07-27 20:09:37.586602720 +0000
@@ -76,8 +76,8 @@
 #include "samples.h"
 #include "options.h"
 
-#include <host_vector.h>
-#include <device_vector.h> 
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h> 
 
 #include <time.h>
 #include <sys/time.h>
diff -ru fsl.orig/src/fdt/CUDA/xfibres_gpu.cuh fsl/src/fdt/CUDA/xfibres_gpu.cuh
--- fsl.orig/src/fdt/CUDA/xfibres_gpu.cuh	2021-07-27 13:30:38.576946383 +0000
+++ fsl/src/fdt/CUDA/xfibres_gpu.cuh	2021-07-27 20:09:37.603602841 +0000
@@ -67,8 +67,8 @@
     fsl@innovation.ox.ac.uk quoting Reference Project 9564, FSL.*/
 
 #include "newimage/newimageall.h"
-#include <host_vector.h>
-#include <device_vector.h> 
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h> 
 
 #include "fibre_gpu.h"
 #include <curand_kernel.h>
Seulement dans fsl/src/fdt/CUDA: xfibres_gpu.cuh.orig
diff -ru fsl.orig/src/fdt/Makefile fsl/src/fdt/Makefile
--- fsl.orig/src/fdt/Makefile	2021-07-27 13:30:37.598939903 +0000
+++ fsl/src/fdt/Makefile	2021-07-27 20:10:37.158030696 +0000
@@ -171,7 +171,7 @@
 		   ${CXX} ${CXXFLAGS} ${LDFLAGS} -o $@ ${EDDYCOMBINEOBJS} ${DLIBS}
 
 ${LIBBEDPOSTX_CUDA}:
-		${NVCC} --shared --compiler-options '-fPIC' -o CUDA/libbedpostx_cuda.so CUDA/init_gpu.cu CUDA/samples.cu CUDA/diffmodels.cu CUDA/runmcmc.cu  CUDA/xfibres_gpu.cu -O3 ${GENCODE_FLAGS} -lcudart -lcuda -lcurand -I. -L${LIB_CUDA} -L${LIB_CUDA}/stubs -ICUDA/options -I${INC_NEWMAT} -I${FSLDIR}/include -I${INC_BOOST} -I${INC_CUDA} -I${INC_CUDA}/thrust -maxrregcount=64
+		${NVCC} --shared --compiler-options '-fPIC' -o CUDA/libbedpostx_cuda.so CUDA/init_gpu.cu CUDA/samples.cu CUDA/diffmodels.cu CUDA/runmcmc.cu  CUDA/xfibres_gpu.cu -O3 ${GENCODE_FLAGS} -lcudart -lcuda -lcurand -I. -L${LIB_CUDA} -L${LIB_CUDA}/stubs -ICUDA/options -I${INC_NEWMAT} -I${FSLDIR}/include -I${INC_BOOST} -I${INC_CUDA} -maxrregcount=64
 		@if [ ! -d ${FSLDEVDIR}/lib/ ] ; then ${MKDIR} ${FSLDEVDIR}/lib ; fi
 		${CP} -rf CUDA/libbedpostx_cuda.so ${FSLDEVDIR}/lib
 
Seulement dans fsl/src/fdt: Makefile.orig
Seulement dans fsl/src/fdt: Makefile.rej
