fix for "cprob/libprob.h: No such file or directory"
author: Kenneth Hoste (HPC-UGent)
diff -ru fsl/src/bint/Makefile fsl.patched/src/bint/Makefile
--- fsl/src/bint/Makefile	2019-01-24 16:13:24.000000000 +0100
+++ fsl.patched/src/bint/Makefile	2022-06-24 15:34:10.551524000 +0200
@@ -2,8 +2,8 @@
 
 PROJNAME = bint
 
-USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_ZLIB}
-USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_ZLIB}
+USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_CPROB} -I${INC_ZLIB}
+USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_CPROB} -L${LIB_ZLIB}
 
 LIBS = -lutils -lnewimage -lmiscmaths -lm -lnewmat -lNewNifti -lznz -lprob -lz
 
diff -ru fsl/src/eddy/Makefile fsl.patched/src/eddy/Makefile
--- fsl/src/eddy/Makefile	2021-06-15 13:04:56.000000000 +0200
+++ fsl.patched/src/eddy/Makefile	2022-06-24 15:34:10.567880609 +0200
@@ -79,7 +79,7 @@
 	EXENAME=eddy${TMPNAME_1}${TMPNAME_2}
 endif
 USRLDFLAGS=-L${LIB_NEWMAT} -Wl,-rpath,/opt/fmrib/fsl/lib -rdynamic
-USRINCFLAGS=-I. -I${INC_NEWMAT} -I${INC_PROB} -I${INC_BOOST} -I${INC_CUDA} -I$(CBF_DIR) -I$(INC_BASISFIELD)
+USRINCFLAGS=-I. -I${INC_NEWMAT} -I${INC_PROB} -I${INC_CPROB} -I${INC_BOOST} -I${INC_CUDA} -I$(CBF_DIR) -I$(INC_BASISFIELD)
 
 OBJS=eddy.o CPUStackResampler.o BiasFieldEstimatorImpl.o MoveBySuscCF.o PostEddyAlignShellsFunctions.o EddyCommandLineOptions.o ECModels.o KMatrix.o HyParEstimator.o ECScanClasses.o EddyUtils.o EddyHelperClasses.o DiffusionGP.o b0Predictor.o
 TESTOBJS=test_s2v_model.o PostEddyAlignShellsFunctions.o EddyCommandLineOptions.o ECModels.o KMatrix.o HyParEstimator.o ECScanClasses.o EddyUtils.o EddyHelperClasses.o DiffusionGP.o b0Predictor.o 
diff -ru fsl/src/feat5/Makefile fsl.patched/src/feat5/Makefile
--- fsl/src/feat5/Makefile	2021-04-15 13:34:17.000000000 +0200
+++ fsl.patched/src/feat5/Makefile	2022-06-24 15:35:08.189816000 +0200
@@ -2,8 +2,8 @@
 
 PROJNAME = feat5
 
-USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_ZLIB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} 
-USRLDFLAGS =  -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_ZLIB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG}
+USRINCFLAGS = -I${INC_NEWMAT} -I${INC_CPROB} -I${INC_PROB} -I${INC_ZLIB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} 
+USRLDFLAGS =  -L${LIB_NEWMAT} -L${LIB_CPROB} -L${LIB_PROB} -L${LIB_ZLIB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG}
 
 IOLIBS = -lNewNifti -lznz
 LIBS = -lnewimage -lmiscplot -lmiscpic -lmiscmaths -lprob -lnewmat -lutils $(IOLIBS) -lm -lz  -lgdc -lgd -lpng
diff -ru fsl/src/filmbabe/Makefile fsl.patched/src/filmbabe/Makefile
--- fsl/src/filmbabe/Makefile	2018-04-23 14:10:04.000000000 +0200
+++ fsl.patched/src/filmbabe/Makefile	2022-06-24 15:35:08.191008000 +0200
@@ -2,8 +2,8 @@
 
 PROJNAME = filmbabe
 
-USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB}
-USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
+USRINCFLAGS = -I${INC_NEWMAT} -I${INC_CPROB} -I${INC_PROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB}
+USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_CPROB} -L${LIB_PROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
 
 LIBS = -lutils -lmiscplot -lmiscpic -lnewimage -lmiscmaths -lnewmat -lNewNifti -lprob -lgdc -lgd -lpng -lznz -lz -lm
 
diff -ru fsl/src/libmeshutils/Makefile fsl.patched/src/libmeshutils/Makefile
--- fsl/src/libmeshutils/Makefile	2014-12-16 15:20:18.000000000 +0100
+++ fsl.patched/src/libmeshutils/Makefile	2022-06-24 15:34:10.552399000 +0200
@@ -5,8 +5,8 @@
 
 LD_LIBRARY_PATH=${FSLDIR}/lib
 
-USRINCFLAGS = -I${INC_NEWMAT} -I${INC_ZLIB} -I${INC_PROB} -I${INC_BOOST} 
-USRLDFLAGS =  -L${LIB_PROB} -L${LIB_NEWMAT} -L${LIB_ZLIB}
+USRINCFLAGS = -I${INC_NEWMAT} -I${INC_ZLIB} -I${INC_PROB} -I${INC_CPROB} -I${INC_BOOST} 
+USRLDFLAGS =  -L${LIB_PROB} -L${LIB_CPROB} -L${LIB_NEWMAT} -L${LIB_ZLIB}
 
 BUILT_VTKCONV = $(shell [ -f ../../libfslvtkconv* ]; echo $$?)
 
diff -ru fsl/src/libvis/Makefile fsl.patched/src/libvis/Makefile
--- fsl/src/libvis/Makefile	2019-09-20 13:45:59.000000000 +0200
+++ fsl.patched/src/libvis/Makefile	2022-06-24 15:34:10.550654000 +0200
@@ -4,8 +4,8 @@
 
 PROJNAME = libvis
 
-USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB} -I${INC_BOOST}
-USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
+USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_CPROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB} -I${INC_BOOST}
+USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_CPROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
 
 LIBS = -lnewimage -lmiscplot -lmiscpic -lmiscmaths -lfslio -lniftiio -lznz -lnewmat -lprob -lm  -lgdc -lgd -lpng -lz
 CXXFLAGS += -std=c++98
--- fsl/src/miscvis/Makefile.orig	2022-06-24 15:42:23.119451000 +0200
+++ fsl/src/miscvis/Makefile	2022-06-24 15:42:44.790736000 +0200
@@ -2,8 +2,8 @@
 
 PROJNAME = miscvis
 
-USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB}
-USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
+USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_CPROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB}
+USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_CPROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
 
 LIBS = -lutils -lnewimage -lmiscplot -lmiscpic -lmiscmaths -lNewNifti -lznz -lnewmat -lcprob -lm  -lgdc -lgd -lpng -lz
 
--- fsl/src/mm/Makefile.orig	2022-06-24 15:43:04.736781000 +0200
+++ fsl/src/mm/Makefile	2022-06-24 15:43:23.950776000 +0200
@@ -2,8 +2,8 @@
 
 PROJNAME = mm
 
-USRINCFLAGS = -I${INC_NEWRAN} -I${INC_NEWMAT} -I${INC_PROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB}
-USRLDFLAGS = -L${LIB_NEWRAN} -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
+USRINCFLAGS = -I${INC_NEWRAN} -I${INC_NEWMAT} -I${INC_PROB} -I${INC_CPROB} -I${INC_GD} -I${INC_GDC} -I${INC_PNG} -I${INC_ZLIB}
+USRLDFLAGS = -L${LIB_NEWRAN} -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_CPROB} -L${LIB_GD} -L${LIB_GDC} -L${LIB_PNG} -L${LIB_ZLIB}
 
 LIBS = -lnewimage -lmiscmaths -lutils -lmiscplot -lmiscpic -lNewNifti -lznz -lm -lnewmat -lprob -lgdc -lgd -lpng -lz
 
--- fsl/src/possum/Makefile.orig	2022-06-24 15:43:47.443376000 +0200
+++ fsl/src/possum/Makefile	2022-06-24 15:44:13.001084000 +0200
@@ -9,8 +9,8 @@
 #with the next line it will not do the table version automatically
 #USRCXXFLAGS = -DNOTABLE
 
-USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_ZLIB}
-USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_ZLIB}
+USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_CPROB} -I${INC_ZLIB}
+USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_CPROB} -L${LIB_ZLIB}
 
 LIBS=-lnewimage -lmiscmaths -lnewmat -lutils -lprob -lNewNifti -lznz -lm -lz
 
