diff -ru fsl.orig/config/buildSettings.mk fsl/config/buildSettings.mk
--- fsl.orig/config/buildSettings.mk	2021-07-27 13:41:10.032129832 +0000
+++ fsl/config/buildSettings.mk	2021-07-27 13:45:44.468948188 +0000
@@ -95,7 +95,7 @@
 RANLIB = ranlib
 ###############   External Libs   #####################################
 # Armadillo library
-LIB_NEWMAT = ${FSLEXTLIB} -llapack -lblas
+LIB_NEWMAT = ${FSLEXTLIB} ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
 # ZLIB library, not build on modern OSX as already present
 LIB_ZLIB = ${FSLEXTLIB}
 INC_ZLIB = ${FSLEXTINC}
@@ -169,7 +169,7 @@
 VTKDIR_LIB = ${EBROOTVTK}/lib
 VTKSUFFIX = -`echo ${EBVERSIONVTK} | cut -f1-2 -d.`
 # openblas
-LIB_NEWMAT = ${FSLEXTLIB} -lopenblas
+LIB_NEWMAT = ${FSLEXTLIB} ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
 # get and then parse gcc version to run context specific builds
 #GCCVER := $(shell gcc -dumpversion)
 #GCCARR = $(subst ., ,$(GCCVER))
Seulement dans fsl/config: buildSettings.mk.orig
Seulement dans fsl/config: buildSettings.mk.rej
diff -ru fsl.orig/src/asl_mfree/Makefile fsl/src/asl_mfree/Makefile
--- fsl.orig/src/asl_mfree/Makefile	2021-07-27 13:30:25.505859784 +0000
+++ fsl/src/asl_mfree/Makefile	2021-07-27 13:45:10.141720581 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_core/examples/Makefile fsl/src/fabber_core/examples/Makefile
--- fsl.orig/src/fabber_core/examples/Makefile	2021-07-27 13:31:28.942280057 +0000
+++ fsl/src/fabber_core/examples/Makefile	2021-07-27 13:45:10.141720581 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_core/Makefile fsl/src/fabber_core/Makefile
--- fsl.orig/src/fabber_core/Makefile	2021-07-27 13:31:31.541297276 +0000
+++ fsl/src/fabber_core/Makefile	2021-07-27 13:45:10.141720581 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_asl/Makefile fsl/src/fabber_models_asl/Makefile
--- fsl.orig/src/fabber_models_asl/Makefile	2021-07-27 13:30:53.679046435 +0000
+++ fsl/src/fabber_models_asl/Makefile	2021-07-27 13:45:10.142720587 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_cest/Makefile fsl/src/fabber_models_cest/Makefile
--- fsl.orig/src/fabber_models_cest/Makefile	2021-07-27 13:30:51.026028859 +0000
+++ fsl/src/fabber_models_cest/Makefile	2021-07-27 13:45:10.142720587 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_dce/Makefile fsl/src/fabber_models_dce/Makefile
--- fsl.orig/src/fabber_models_dce/Makefile	2021-07-27 13:30:58.390077646 +0000
+++ fsl/src/fabber_models_dce/Makefile	2021-07-27 13:45:10.142720587 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_dsc/Makefile fsl/src/fabber_models_dsc/Makefile
--- fsl.orig/src/fabber_models_dsc/Makefile	2021-07-27 13:30:44.878988133 +0000
+++ fsl/src/fabber_models_dsc/Makefile	2021-07-27 13:45:10.142720587 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_dualecho/Makefile fsl/src/fabber_models_dualecho/Makefile
--- fsl.orig/src/fabber_models_dualecho/Makefile	2021-07-27 13:30:23.964849576 +0000
+++ fsl/src/fabber_models_dualecho/Makefile	2021-07-27 13:45:10.142720587 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_dwi/Makefile fsl/src/fabber_models_dwi/Makefile
--- fsl.orig/src/fabber_models_dwi/Makefile	2021-07-27 13:31:02.552105220 +0000
+++ fsl/src/fabber_models_dwi/Makefile	2021-07-27 13:45:10.143720594 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_qbold/Makefile fsl/src/fabber_models_qbold/Makefile
--- fsl.orig/src/fabber_models_qbold/Makefile	2021-07-27 13:31:34.024313726 +0000
+++ fsl/src/fabber_models_qbold/Makefile	2021-07-27 13:45:10.143720594 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/fabber_models_t1/Makefile fsl/src/fabber_models_t1/Makefile
--- fsl.orig/src/fabber_models_t1/Makefile	2021-07-27 13:30:33.452912434 +0000
+++ fsl/src/fabber_models_t1/Makefile	2021-07-27 13:45:10.143720594 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.orig/src/oxford_asl/Makefile fsl/src/oxford_asl/Makefile
--- fsl.orig/src/oxford_asl/Makefile	2021-07-27 13:31:24.874253106 +0000
+++ fsl/src/oxford_asl/Makefile	2021-07-27 13:45:10.143720594 +0000
@@ -13,7 +13,7 @@
   NIFTILIB = -lNewNifti
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
 endif
 
