diff -ru fsl.orig/fsl/config/apple-darwin10-gcc4.2/systemvars.mk fsl/config/apple-darwin10-gcc4.2/systemvars.mk
--- fsl.orig/fsl/config/apple-darwin10-gcc4.2/systemvars.mk	2013-05-15 00:14:37.710187000 +0200
+++ fsl/config/apple-darwin10-gcc4.2/systemvars.mk	2013-05-15 00:19:46.085324000 +0200
@@ -15,7 +15,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations
+OPTFLAGS =  -O3 -fexpensive-optimizations -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/apple-darwin11-gcc4.2/systemvars.mk fsl/config/apple-darwin11-gcc4.2/systemvars.mk
--- fsl.orig/fsl/config/apple-darwin11-gcc4.2/systemvars.mk	2012-03-21 12:30:53.000000000 +0100
+++ fsl/config/apple-darwin11-gcc4.2/systemvars.mk	2013-05-15 00:19:48.066557000 +0200
@@ -15,7 +15,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3
+OPTFLAGS =  -O3 -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -ansi -pedantic
 ANSI_FLAGS = ${GNU_ANSI_FLAGS}
diff -ru fsl.orig/fsl/config/apple-darwin12-gcc4.2/systemvars.mk fsl/config/apple-darwin12-gcc4.2/systemvars.mk
--- fsl.orig/fsl/config/apple-darwin12-gcc4.2/systemvars.mk	2013-04-09 18:00:32.000000000 +0200
+++ fsl/config/apple-darwin12-gcc4.2/systemvars.mk	2013-05-15 00:19:49.812861000 +0200
@@ -15,7 +15,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3
+OPTFLAGS =  -O3 -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -ansi -pedantic
 ANSI_FLAGS = ${GNU_ANSI_FLAGS}
diff -ru fsl.orig/fsl/config/apple-darwin7-gcc3.1/systemvars.mk fsl/config/apple-darwin7-gcc3.1/systemvars.mk
--- fsl.orig/fsl/config/apple-darwin7-gcc3.1/systemvars.mk	2013-05-15 00:14:37.712623000 +0200
+++ fsl/config/apple-darwin7-gcc3.1/systemvars.mk	2013-05-15 00:19:51.866756000 +0200
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -traditional-cpp -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/apple-darwin7-gcc3.3/systemvars.mk fsl/config/apple-darwin7-gcc3.3/systemvars.mk
--- fsl.orig/fsl/config/apple-darwin7-gcc3.3/systemvars.mk	2013-05-15 00:14:37.737794000 +0200
+++ fsl/config/apple-darwin7-gcc3.3/systemvars.mk	2013-05-15 00:19:53.537845000 +0200
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/apple-darwin8-gcc4.0/systemvars.mk fsl/config/apple-darwin8-gcc4.0/systemvars.mk
--- fsl.orig/fsl/config/apple-darwin8-gcc4.0/systemvars.mk	2013-05-15 00:14:37.745810000 +0200
+++ fsl/config/apple-darwin8-gcc4.0/systemvars.mk	2013-05-15 00:19:55.229757000 +0200
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations
+OPTFLAGS =  -O3 -fexpensive-optimizations -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/apple-darwin9-gcc4.0/systemvars.mk fsl/config/apple-darwin9-gcc4.0/systemvars.mk
--- fsl.orig/fsl/config/apple-darwin9-gcc4.0/systemvars.mk	2013-05-15 00:14:37.751696000 +0200
+++ fsl/config/apple-darwin9-gcc4.0/systemvars.mk	2013-05-15 00:19:57.417620000 +0200
@@ -17,7 +17,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations
+OPTFLAGS =  -O3 -fexpensive-optimizations -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -Wno-long-long -Wno-long-double -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/generic/systemvars.mk fsl/config/generic/systemvars.mk
--- fsl.orig/fsl/config/generic/systemvars.mk	2013-05-15 00:14:37.757155000 +0200
+++ fsl/config/generic/systemvars.mk	2013-05-15 00:19:59.724851000 +0200
@@ -25,7 +25,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS =  -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS =
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/gnu_64-gcc4.4/systemvars.mk fsl/config/gnu_64-gcc4.4/systemvars.mk
--- fsl.orig/fsl/config/gnu_64-gcc4.4/systemvars.mk	2011-04-19 10:47:52.000000000 +0200
+++ fsl/config/gnu_64-gcc4.4/systemvars.mk	2013-05-15 00:20:03.204842000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk
--- fsl.orig/fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2013-05-15 00:14:37.763457000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2013-05-15 00:20:05.347565000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-deprecated
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk
--- fsl.orig/fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2013-05-15 00:14:37.769108000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2013-05-15 00:20:06.913095000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-deprecated
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk
--- fsl.orig/fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2013-05-15 00:14:37.775138000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2013-05-15 00:20:08.480567000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-deprecated
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_32-gcc2.96/systemvars.mk fsl/config/linux_32-gcc2.96/systemvars.mk
--- fsl.orig/fsl/config/linux_32-gcc2.96/systemvars.mk	2013-05-15 00:14:37.781755000 +0200
+++ fsl/config/linux_32-gcc2.96/systemvars.mk	2013-05-15 00:20:10.043882000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_32-gcc3.2/systemvars.mk fsl/config/linux_32-gcc3.2/systemvars.mk
--- fsl.orig/fsl/config/linux_32-gcc3.2/systemvars.mk	2013-05-15 00:14:37.787130000 +0200
+++ fsl/config/linux_32-gcc3.2/systemvars.mk	2013-05-15 00:20:11.484066000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_32-gcc3.3/systemvars.mk fsl/config/linux_32-gcc3.3/systemvars.mk
--- fsl.orig/fsl/config/linux_32-gcc3.3/systemvars.mk	2013-05-15 00:14:37.794387000 +0200
+++ fsl/config/linux_32-gcc3.3/systemvars.mk	2013-05-15 00:20:12.921148000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_32-gcc3.4/systemvars.mk fsl/config/linux_32-gcc3.4/systemvars.mk
--- fsl.orig/fsl/config/linux_32-gcc3.4/systemvars.mk	2013-05-15 00:14:37.799247000 +0200
+++ fsl/config/linux_32-gcc3.4/systemvars.mk	2013-05-15 00:20:14.478184000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_32-gcc4.0/systemvars.mk fsl/config/linux_32-gcc4.0/systemvars.mk
--- fsl.orig/fsl/config/linux_32-gcc4.0/systemvars.mk	2013-05-15 00:14:37.805358000 +0200
+++ fsl/config/linux_32-gcc4.0/systemvars.mk	2013-05-15 00:20:16.819361000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_32-gcc4.1/systemvars.mk fsl/config/linux_32-gcc4.1/systemvars.mk
--- fsl.orig/fsl/config/linux_32-gcc4.1/systemvars.mk	2013-05-15 00:14:37.811876000 +0200
+++ fsl/config/linux_32-gcc4.1/systemvars.mk	2013-05-15 00:20:18.322853000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_64-gcc3.4/systemvars.mk fsl/config/linux_64-gcc3.4/systemvars.mk
--- fsl.orig/fsl/config/linux_64-gcc3.4/systemvars.mk	2013-05-15 00:14:37.817395000 +0200
+++ fsl/config/linux_64-gcc3.4/systemvars.mk	2013-05-15 00:20:20.465845000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_64-gcc4.0/systemvars.mk fsl/config/linux_64-gcc4.0/systemvars.mk
--- fsl.orig/fsl/config/linux_64-gcc4.0/systemvars.mk	2013-05-15 00:14:37.823453000 +0200
+++ fsl/config/linux_64-gcc4.0/systemvars.mk	2013-05-15 00:20:22.065237000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_64-gcc4.1/systemvars.mk fsl/config/linux_64-gcc4.1/systemvars.mk
--- fsl.orig/fsl/config/linux_64-gcc4.1/systemvars.mk	2013-05-15 00:14:37.829561000 +0200
+++ fsl/config/linux_64-gcc4.1/systemvars.mk	2013-05-15 00:20:23.601865000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_64-gcc4.2/systemvars.mk fsl/config/linux_64-gcc4.2/systemvars.mk
--- fsl.orig/fsl/config/linux_64-gcc4.2/systemvars.mk	2013-05-15 00:14:37.835868000 +0200
+++ fsl/config/linux_64-gcc4.2/systemvars.mk	2013-05-15 00:20:25.058756000 +0200
@@ -27,7 +27,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/linux_64-gcc4.4/systemvars.mk fsl/config/linux_64-gcc4.4/systemvars.mk
--- fsl.orig/fsl/config/linux_64-gcc4.4/systemvars.mk	2012-08-22 17:17:13.000000000 +0200
+++ fsl/config/linux_64-gcc4.4/systemvars.mk	2013-05-15 00:20:26.561348000 +0200
@@ -29,7 +29,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O3 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS = -g
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic -Wno-long-long
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk
--- fsl.orig/fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2013-05-15 00:14:37.841452000 +0200
+++ fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2013-05-15 00:20:28.072793000 +0200
@@ -12,7 +12,7 @@
 
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O6 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O6 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS =
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
diff -ru fsl.orig/fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk
--- fsl.orig/fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2013-05-15 00:14:37.847832000 +0200
+++ fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2013-05-15 00:20:29.505593000 +0200
@@ -12,7 +12,7 @@
 ARCHLDFLAGS = -static
 DEPENDFLAGS = -MM
 
-OPTFLAGS = -O6 -fexpensive-optimizations ${ARCHFLAGS}
+OPTFLAGS = -O6 -fexpensive-optimizations ${ARCHFLAGS} -wd803
 MACHDBGFLAGS =
 GNU_ANSI_FLAGS = -Wall -ansi -pedantic
 SGI_ANSI_FLAGS = -ansi -fullwarn
