diff -ru fsl.patched_upstream_mkl/config/buildSettings.mk fsl/config/buildSettings.mk
--- fsl.patched_upstream_mkl/config/buildSettings.mk	2021-02-24 16:54:30.216007857 +0000
+++ fsl/config/buildSettings.mk	2021-02-24 17:15:19.877618704 +0000
@@ -65,7 +65,7 @@
 LIB_FFTW3 = ${EBROOTFFTW}/lib
 INC_FFTW3 = ${EBROOTFFTW}/include
 # LIBXML2 library
-INC_XML2 = ${EBROOTLIBXML2}/include
+INC_XML2 = ${EBROOTLIBXML2}/include/libxml2
 # LIBXML++ library
 INC_XML++ = ${EBROOTLIBXMLPLUSPLUS}/include/libxml++-2.6
 INC_XML++CONF = ${EBROOTLIBXMLPLUSPLUS}/lib/libxml++-2.6/include
