diff -ru fsl.patched_upstream/config/buildSettings.mk fsl.patched_upstream_mkl/config/buildSettings.mk
--- fsl.patched_upstream/config/buildSettings.mk	2021-02-24 16:41:52.192137784 +0000
+++ fsl.patched_upstream_mkl/config/buildSettings.mk	2021-02-24 17:45:38.764395612 +0000
@@ -94,7 +94,7 @@
 RANLIB = ranlib
 ###############   External Libs   #####################################
 # Armadillo library
-LIB_NEWMAT = ${FSLEXTLIB} -llapack -lblas
+LIB_NEWMAT = ${FSLEXTLIB} ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
 # ZLIB library
 LIB_ZLIB = /usr/lib
 INC_ZLIB = /usr/include
@@ -159,7 +159,7 @@
 VTKDIR_LIB = ${EBROOTVTK}/lib
 VTKSUFFIX = -`echo ${EBVERSIONVTK} | cut -f1-2 -d.`
 # openblas
-LIB_NEWMAT = ${FSLEXTLIB} -lopenblas
+LIB_NEWMAT = ${FSLEXTLIB} ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
 # get and then parse gcc version to run context specific builds
 #GCCVER := $(shell gcc -dumpversion)
 #GCCARR = $(subst ., ,$(GCCVER))
diff -ru fsl.patched_upstream/src/asl_mfree/Makefile fsl.patched_upstream_mkl/src/asl_mfree/Makefile
--- fsl.patched_upstream/src/asl_mfree/Makefile	2021-02-24 16:41:09.943932979 +0000
+++ fsl.patched_upstream_mkl/src/asl_mfree/Makefile	2021-02-24 17:42:15.226160761 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_core/examples/Makefile fsl.patched_upstream_mkl/src/fabber_core/examples/Makefile
--- fsl.patched_upstream/src/fabber_core/examples/Makefile	2021-02-24 16:41:13.274949127 +0000
+++ fsl.patched_upstream_mkl/src/fabber_core/examples/Makefile	2021-02-24 17:42:15.227160767 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_core/Makefile fsl.patched_upstream_mkl/src/fabber_core/Makefile
--- fsl.patched_upstream/src/fabber_core/Makefile	2021-02-24 16:41:13.432949892 +0000
+++ fsl.patched_upstream_mkl/src/fabber_core/Makefile	2021-02-24 17:42:15.228160773 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_asl/Makefile fsl.patched_upstream_mkl/src/fabber_models_asl/Makefile
--- fsl.patched_upstream/src/fabber_models_asl/Makefile	2021-02-24 16:41:10.910937667 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_asl/Makefile	2021-02-24 17:42:15.227160767 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_cest/Makefile fsl.patched_upstream_mkl/src/fabber_models_cest/Makefile
--- fsl.patched_upstream/src/fabber_models_cest/Makefile	2021-02-24 16:41:11.269939407 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_cest/Makefile	2021-02-24 17:42:15.227160767 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_dce/Makefile fsl.patched_upstream_mkl/src/fabber_models_dce/Makefile
--- fsl.patched_upstream/src/fabber_models_dce/Makefile	2021-02-24 16:41:13.019947890 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_dce/Makefile	2021-02-24 17:42:15.227160767 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_dsc/Makefile fsl.patched_upstream_mkl/src/fabber_models_dsc/Makefile
--- fsl.patched_upstream/src/fabber_models_dsc/Makefile	2021-02-24 16:41:12.053943207 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_dsc/Makefile	2021-02-24 17:42:15.227160767 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_dualecho/Makefile fsl.patched_upstream_mkl/src/fabber_models_dualecho/Makefile
--- fsl.patched_upstream/src/fabber_models_dualecho/Makefile	2021-02-24 16:41:10.367935034 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_dualecho/Makefile	2021-02-24 17:42:15.226160761 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_dwi/Makefile fsl.patched_upstream_mkl/src/fabber_models_dwi/Makefile
--- fsl.patched_upstream/src/fabber_models_dwi/Makefile	2021-02-24 16:41:09.616931394 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_dwi/Makefile	2021-02-24 17:42:15.226160761 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_qbold/Makefile fsl.patched_upstream_mkl/src/fabber_models_qbold/Makefile
--- fsl.patched_upstream/src/fabber_models_qbold/Makefile	2021-02-24 16:41:12.708946383 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_qbold/Makefile	2021-02-24 17:42:15.227160767 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/fabber_models_t1/Makefile fsl.patched_upstream_mkl/src/fabber_models_t1/Makefile
--- fsl.patched_upstream/src/fabber_models_t1/Makefile	2021-02-24 16:41:11.460940333 +0000
+++ fsl.patched_upstream_mkl/src/fabber_models_t1/Makefile	2021-02-24 17:42:15.227160767 +0000
@@ -12,7 +12,7 @@
 else 
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
   NIFTILIB = -lNewNifti
 endif
diff -ru fsl.patched_upstream/src/oxford_asl/Makefile fsl.patched_upstream_mkl/src/oxford_asl/Makefile
--- fsl.patched_upstream/src/oxford_asl/Makefile	2021-02-24 16:41:10.707936683 +0000
+++ fsl.patched_upstream_mkl/src/oxford_asl/Makefile	2021-02-24 17:42:15.226160761 +0000
@@ -13,7 +13,7 @@
   NIFTILIB = -lNewNifti
   UNAME := $(shell uname -s)
   ifeq ($(UNAME), Linux)
-    MATLIB = -lopenblas
+    MATLIB = ${EBVARLIBLAPACK} ${EBVARLIBBLAS}
   endif
 endif
 
