diff -ru fsl.orig/src/fslsurface/fslsurface_structs.h fsl/src/fslsurface/fslsurface_structs.h
--- fsl.orig/src/fslsurface/fslsurface_structs.h	2013-05-09 18:22:38.000000000 +0200
+++ fsl/src/fslsurface/fslsurface_structs.h	2013-05-20 14:02:37.520621000 +0200
@@ -71,7 +71,7 @@
 #else
 #include <OpenGL/gl.h>
 #endif
-//#include <cmath>
+#include <cmath>
 
 namespace fslsurface_name{
 	enum MarchingCubesMode {EQUAL_TO, GREATER_THAN, GREATER_THAN_EQUAL_TO};
