--- fsl/src/newimage/newimageio.h.orig	2019-04-09 05:21:31.477323743 +0000
+++ fsl/src/newimage/newimageio.h	2019-04-09 06:16:14.310207063 +0000
@@ -287,7 +287,7 @@
 		     int64_t xskip, int64_t yskip, int64_t zskip, int64_t tskip)
 {
   read_volumeROI(target,filename,x0,y0,z0,t0,x1,y1,z1,t1);
-  if (xskip<1) xskip=1;    
+  if (xskip<1) int64_t xskip=1;    
   if (yskip<1) yskip=1;
   if (zskip<1) zskip=1;
   if (tskip<1) tskip=1;
