diff -ru fsl.orig/config/apple-darwin10-gcc4.2/systemvars.mk fsl/config/apple-darwin10-gcc4.2/systemvars.mk
--- fsl.orig/config/apple-darwin10-gcc4.2/systemvars.mk	2009-11-03 18:02:14.000000000 +0100
+++ fsl/config/apple-darwin10-gcc4.2/systemvars.mk	2013-05-14 22:43:15.344366000 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin7-gcc3.1/systemvars.mk fsl/config/apple-darwin7-gcc3.1/systemvars.mk
--- fsl.orig/config/apple-darwin7-gcc3.1/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/apple-darwin7-gcc3.1/systemvars.mk	2013-05-14 22:43:15.363590000 +0200
@@ -7,8 +7,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin7-gcc3.3/systemvars.mk fsl/config/apple-darwin7-gcc3.3/systemvars.mk
--- fsl.orig/config/apple-darwin7-gcc3.3/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/apple-darwin7-gcc3.3/systemvars.mk	2013-05-14 22:43:15.385466000 +0200
@@ -7,8 +7,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin8-gcc4.0/systemvars.mk fsl/config/apple-darwin8-gcc4.0/systemvars.mk
--- fsl.orig/config/apple-darwin8-gcc4.0/systemvars.mk	2007-12-19 15:40:57.000000000 +0100
+++ fsl/config/apple-darwin8-gcc4.0/systemvars.mk	2013-05-14 22:43:15.402328000 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin9-gcc4.0/systemvars.mk fsl/config/apple-darwin9-gcc4.0/systemvars.mk
--- fsl.orig/config/apple-darwin9-gcc4.0/systemvars.mk	2007-12-19 15:33:53.000000000 +0100
+++ fsl/config/apple-darwin9-gcc4.0/systemvars.mk	2013-05-14 22:43:15.415331000 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/generic/systemvars.mk fsl/config/generic/systemvars.mk
--- fsl.orig/config/generic/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/generic/systemvars.mk	2013-05-14 22:43:15.438919000 +0200
@@ -16,8 +16,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.2/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2013-05-14 22:43:15.455595000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc-2
-CXX = c++-2
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.3/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2013-05-14 22:43:15.473639000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.4/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2013-05-14 22:43:15.494398000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc2.96/systemvars.mk fsl/config/linux_32-gcc2.96/systemvars.mk
--- fsl.orig/config/linux_32-gcc2.96/systemvars.mk	2007-07-25 17:21:07.000000000 +0200
+++ fsl/config/linux_32-gcc2.96/systemvars.mk	2013-05-14 22:43:15.507210000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc3.2/systemvars.mk fsl/config/linux_32-gcc3.2/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.2/systemvars.mk	2007-07-25 17:21:08.000000000 +0200
+++ fsl/config/linux_32-gcc3.2/systemvars.mk	2013-05-14 22:43:15.524309000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc3.3/systemvars.mk fsl/config/linux_32-gcc3.3/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.3/systemvars.mk	2007-07-25 17:21:09.000000000 +0200
+++ fsl/config/linux_32-gcc3.3/systemvars.mk	2013-05-14 22:43:15.539643000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc3.4/systemvars.mk fsl/config/linux_32-gcc3.4/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.4/systemvars.mk	2007-07-25 17:21:10.000000000 +0200
+++ fsl/config/linux_32-gcc3.4/systemvars.mk	2013-05-14 22:43:15.559881000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc4.0/systemvars.mk fsl/config/linux_32-gcc4.0/systemvars.mk
--- fsl.orig/config/linux_32-gcc4.0/systemvars.mk	2007-07-25 17:21:11.000000000 +0200
+++ fsl/config/linux_32-gcc4.0/systemvars.mk	2013-05-14 22:43:15.573696000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc4.1/systemvars.mk fsl/config/linux_32-gcc4.1/systemvars.mk
--- fsl.orig/config/linux_32-gcc4.1/systemvars.mk	2012-04-20 11:37:28.000000000 +0200
+++ fsl/config/linux_32-gcc4.1/systemvars.mk	2013-05-14 22:43:15.590022000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc3.4/systemvars.mk fsl/config/linux_64-gcc3.4/systemvars.mk
--- fsl.orig/config/linux_64-gcc3.4/systemvars.mk	2007-07-25 17:21:12.000000000 +0200
+++ fsl/config/linux_64-gcc3.4/systemvars.mk	2013-05-14 22:43:15.605655000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc4.0/systemvars.mk fsl/config/linux_64-gcc4.0/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.0/systemvars.mk	2007-07-25 17:21:13.000000000 +0200
+++ fsl/config/linux_64-gcc4.0/systemvars.mk	2013-05-14 22:43:15.624046000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc4.1/systemvars.mk fsl/config/linux_64-gcc4.1/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.1/systemvars.mk	2007-07-25 11:19:45.000000000 +0200
+++ fsl/config/linux_64-gcc4.1/systemvars.mk	2013-05-14 22:43:15.635678000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc4.2/systemvars.mk fsl/config/linux_64-gcc4.2/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.2/systemvars.mk	2008-06-26 15:25:42.000000000 +0200
+++ fsl/config/linux_64-gcc4.2/systemvars.mk	2013-05-14 22:43:15.648658000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc4.4/systemvars.mk fsl/config/linux_64-gcc4.4/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.4/systemvars.mk	2012-08-22 17:17:13.000000000 +0200
+++ fsl/config/linux_64-gcc4.4/systemvars.mk	2014-06-14 23:23:58.456480000 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/sparc-solaris2.8-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk
--- fsl.orig/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2007-07-13 13:00:21.000000000 +0200
+++ fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2013-05-14 22:43:15.658171000 +0200
@@ -5,8 +5,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 
 ARCHFLAGS = -mv8 -ffast-math -fomit-frame-pointer 
 
diff -ru fsl.orig/config/sparc-solaris2.9-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk
--- fsl.orig/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2007-07-13 13:00:21.000000000 +0200
+++ fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2013-05-14 22:43:15.674176000 +0200
@@ -5,8 +5,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 
 ARCHFLAGS = -mv8 -ffast-math -fomit-frame-pointer
 ARCHLDFLAGS = -static
diff -ru fsl.orig/src/libmeshutils/Makefile fsl/src/libmeshutils/Makefile
--- fsl.orig/src/libmeshutils/Makefile	2012-07-23 15:25:20.000000000 +0200
+++ fsl/src/libmeshutils/Makefile	2013-05-14 22:43:15.320209000 +0200
@@ -3,7 +3,7 @@
 
 PROJNAME = meshUtils
 
-LD_LIBRARY_PATH=${FSLDIR}/lib
+#LD_LIBRARY_PATH=${FSLDIR}/lib
 
 USRINCFLAGS = -I${INC_NEWMAT} -I${INC_ZLIB} -I${INC_PROB} -I${INC_BOOST} 
 USRLDFLAGS =  -L${LIB_PROB} -L${LIB_NEWMAT} -L${LIB_ZLIB}
diff -ru fsl.orig/src/melodic/Makefile fsl/src/melodic/Makefile
--- fsl.orig/src/melodic/Makefile	2013-03-13 21:22:35.000000000 +0100
+++ fsl/src/melodic/Makefile	2013-05-14 22:44:13.016352000 +0200
@@ -3,7 +3,7 @@
 include ${FSLCONFDIR}/default.mk
 
 OPTFLAGS = -O3 -Wno-deprecated -ggdb
-OPTFLAGS_alphaev6-dec-osf5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
+#OPTFLAGS_alphaev6-dec-osf5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
 
 PROJNAME = melodic
 
--- fsl/src/film/Makefile.orig	2014-06-15 00:43:34.908901000 +0200
+++ fsl/src/film/Makefile	2014-06-15 00:43:52.303973000 +0200
@@ -26,7 +26,7 @@
 	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} ftoz.o ${LIBS}
 
 film_gls:${OBJS} film_gls.o
-	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} film_gls.o ${LIBS} -l giftiio
+	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} film_gls.o ${LIBS} -lgiftiio
 
 film_gls_res:${OBJS} film_gls_res.o
 	${CXX}  ${CXXFLAGS} ${LDFLAGS} -o $@ ${OBJS} film_gls_res.o ${LIBS}
