CUDA must now be disabled explicitly.
Author: Henrik R. Nagel (NTNU)
diff -Nru fsl/config/buildSettings.mk.orig fsl/config/buildSettings.mk
--- fsl/config/buildSettings.mk.orig	2021-12-10 10:29:19.843028000 +0000
+++ fsl/config/buildSettings.mk	2021-12-10 10:31:28.688335000 +0000
@@ -182,8 +182,8 @@
 #$(info GCC PAT VER $(GCCPAT))
 #Project specific variables
 EDDYBUILDPARAMETERS = "cuda=1 CUDAVER=8.0" "cuda=1 CUDAVER=9.1" "cuda=1 CUDAVER=10.2" "cpu=1"
-fdt_MASTERBUILD     = COMPILE_GPU = 1
-ptx2_MASTERBUILD    = COMPILE_GPU = 1
+fdt_MASTERBUILD     = COMPILE_GPU = 0
+ptx2_MASTERBUILD    = COMPILE_GPU = 0
 define newline
 
 
