--- fsl.orig/src/melodic/Makefile	2012-07-31 15:45:06.997491869 +0200
+++ fsl/src/melodic/Makefile	2012-07-31 15:58:39.172984156 +0200
@@ -3,7 +3,7 @@
 include ${FSLCONFDIR}/default.mk
 
 OPTFLAGS = -O3 -Wno-deprecated 
-OPTFLAGS_alphaev6-dec-osf 5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
+#OPTFLAGS_alphaev6-dec-osf 5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
 
 PROJNAME = melodic
 --- fsl.orig/src/libmeshutils/Makefile	2009-10-23 18:49:08.000000000 +0200
+++ fsl/src/libmeshutils/Makefile	2012-08-02 11:20:48.707842124 +0200
@@ -3,7 +3,7 @@
 
 PROJNAME = meshUtils
 
-LD_LIBRARY_PATH=${FSLDIR}/lib
+#LD_LIBRARY_PATH=${FSLDIR}/lib
 
 USRINCFLAGS = -I${INC_NEWMAT} -I${INC_ZLIB} -I${INC_PROB} -I${INC_BOOST} 
 USRLDFLAGS =  -L${LIB_PROB} -L${LIB_NEWMAT} -L${LIB_ZLIB}
