--- fsl.orig/src/melodic/Makefile	2012-07-31 15:45:06.997491869 +0200
+++ fsl/src/melodic/Makefile	2012-07-31 15:58:39.172984156 +0200
@@ -3,7 +3,7 @@
 include ${FSLCONFDIR}/default.mk
 
 OPTFLAGS = -O3 -Wno-deprecated 
-OPTFLAGS_alphaev6-dec-osf 5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
+#OPTFLAGS_alphaev6-dec-osf 5.0-gcc2.95.2 = -O3 -mieee -mfp-trap-mode=sui
 
 PROJNAME = melodic
 --- fsl.orig/src/libmeshutils/Makefile	2009-10-23 18:49:08.000000000 +0200
+++ fsl/src/libmeshutils/Makefile	2012-08-02 11:20:48.707842124 +0200
@@ -3,7 +3,7 @@
 
 PROJNAME = meshUtils
 
-LD_LIBRARY_PATH=${FSLDIR}/lib
+#LD_LIBRARY_PATH=${FSLDIR}/lib
 
 USRINCFLAGS = -I${INC_NEWMAT} -I${INC_ZLIB} -I${INC_PROB} -I${INC_BOOST} 
 USRLDFLAGS =  -L${LIB_PROB} -L${LIB_NEWMAT} -L${LIB_ZLIB}
diff -ru fsl.orig/config/apple-darwin10-gcc4.2/systemvars.mk fsl/config/apple-darwin10-gcc4.2/systemvars.mk
--- fsl.orig/config/apple-darwin10-gcc4.2/systemvars.mk	2009-11-03 18:02:14.000000000 +0100
+++ fsl/config/apple-darwin10-gcc4.2/systemvars.mk	2012-08-02 14:45:57.697749330 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin7-gcc3.1/systemvars.mk fsl/config/apple-darwin7-gcc3.1/systemvars.mk
--- fsl.orig/config/apple-darwin7-gcc3.1/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/apple-darwin7-gcc3.1/systemvars.mk	2012-08-02 14:45:57.734749873 +0200
@@ -7,8 +7,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin7-gcc3.3/systemvars.mk fsl/config/apple-darwin7-gcc3.3/systemvars.mk
--- fsl.orig/config/apple-darwin7-gcc3.3/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/apple-darwin7-gcc3.3/systemvars.mk	2012-08-02 14:45:57.767750355 +0200
@@ -7,8 +7,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin8-gcc4.0/systemvars.mk fsl/config/apple-darwin8-gcc4.0/systemvars.mk
--- fsl.orig/config/apple-darwin8-gcc4.0/systemvars.mk	2007-12-19 15:40:57.000000000 +0100
+++ fsl/config/apple-darwin8-gcc4.0/systemvars.mk	2012-08-02 14:45:57.801750851 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/apple-darwin9-gcc4.0/systemvars.mk fsl/config/apple-darwin9-gcc4.0/systemvars.mk
--- fsl.orig/config/apple-darwin9-gcc4.0/systemvars.mk	2007-12-19 15:33:53.000000000 +0100
+++ fsl/config/apple-darwin9-gcc4.0/systemvars.mk	2012-08-02 14:45:57.835751349 +0200
@@ -3,8 +3,8 @@
 
 # Compiler dependent variables
 
-CC = cc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = 
 CXXSTATICFLAGS = 
 
diff -ru fsl.orig/config/generic/systemvars.mk fsl/config/generic/systemvars.mk
--- fsl.orig/config/generic/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/generic/systemvars.mk	2012-08-02 14:45:57.869751846 +0200
@@ -16,8 +16,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.2/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.2/systemvars.mk	2012-08-02 14:45:57.903752343 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc-2
-CXX = c++-2
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.3/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.3/systemvars.mk	2012-08-02 14:45:57.936752826 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/i686-pc-cygwin-gcc3.4/systemvars.mk fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk
--- fsl.orig/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2007-07-13 13:00:20.000000000 +0200
+++ fsl/config/i686-pc-cygwin-gcc3.4/systemvars.mk	2012-08-02 14:45:57.979753456 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc2.96/systemvars.mk fsl/config/linux_32-gcc2.96/systemvars.mk
--- fsl.orig/config/linux_32-gcc2.96/systemvars.mk	2007-07-25 17:21:07.000000000 +0200
+++ fsl/config/linux_32-gcc2.96/systemvars.mk	2012-08-02 14:45:58.021754069 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc3.2/systemvars.mk fsl/config/linux_32-gcc3.2/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.2/systemvars.mk	2007-07-25 17:21:08.000000000 +0200
+++ fsl/config/linux_32-gcc3.2/systemvars.mk	2012-08-02 14:45:58.063754683 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc3.3/systemvars.mk fsl/config/linux_32-gcc3.3/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.3/systemvars.mk	2007-07-25 17:21:09.000000000 +0200
+++ fsl/config/linux_32-gcc3.3/systemvars.mk	2012-08-02 14:45:58.105755298 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc3.4/systemvars.mk fsl/config/linux_32-gcc3.4/systemvars.mk
--- fsl.orig/config/linux_32-gcc3.4/systemvars.mk	2007-07-25 17:21:10.000000000 +0200
+++ fsl/config/linux_32-gcc3.4/systemvars.mk	2012-08-02 14:45:58.147755913 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc4.0/systemvars.mk fsl/config/linux_32-gcc4.0/systemvars.mk
--- fsl.orig/config/linux_32-gcc4.0/systemvars.mk	2007-07-25 17:21:11.000000000 +0200
+++ fsl/config/linux_32-gcc4.0/systemvars.mk	2012-08-02 14:45:58.231757140 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_32-gcc4.1/systemvars.mk fsl/config/linux_32-gcc4.1/systemvars.mk
--- fsl.orig/config/linux_32-gcc4.1/systemvars.mk	2007-07-13 13:00:21.000000000 +0200
+++ fsl/config/linux_32-gcc4.1/systemvars.mk	2012-08-02 14:45:58.265757638 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc3.4/systemvars.mk fsl/config/linux_64-gcc3.4/systemvars.mk
--- fsl.orig/config/linux_64-gcc3.4/systemvars.mk	2007-07-25 17:21:12.000000000 +0200
+++ fsl/config/linux_64-gcc3.4/systemvars.mk	2012-08-02 14:45:58.299758136 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc4.0/systemvars.mk fsl/config/linux_64-gcc4.0/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.0/systemvars.mk	2007-07-25 17:21:13.000000000 +0200
+++ fsl/config/linux_64-gcc4.0/systemvars.mk	2012-08-02 14:45:58.341758749 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc4.1/systemvars.mk fsl/config/linux_64-gcc4.1/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.1/systemvars.mk	2007-07-25 11:19:45.000000000 +0200
+++ fsl/config/linux_64-gcc4.1/systemvars.mk	2012-08-02 14:45:58.375759246 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/linux_64-gcc4.2/systemvars.mk fsl/config/linux_64-gcc4.2/systemvars.mk
--- fsl.orig/config/linux_64-gcc4.2/systemvars.mk	2008-06-26 15:25:42.000000000 +0200
+++ fsl/config/linux_64-gcc4.2/systemvars.mk	2012-08-02 14:45:58.410759759 +0200
@@ -18,8 +18,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 CSTATICFLAGS = -static
 CXXSTATICFLAGS = -static
 
diff -ru fsl.orig/config/sparc-solaris2.8-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk
--- fsl.orig/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2007-07-13 13:00:21.000000000 +0200
+++ fsl/config/sparc-solaris2.8-gcc2.95/systemvars.mk	2012-08-02 14:45:58.443760241 +0200
@@ -5,8 +5,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 
 ARCHFLAGS = -mv8 -ffast-math -fomit-frame-pointer 
 
diff -ru fsl.orig/config/sparc-solaris2.9-gcc2.95/systemvars.mk fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk
--- fsl.orig/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2007-07-13 13:00:21.000000000 +0200
+++ fsl/config/sparc-solaris2.9-gcc2.95/systemvars.mk	2012-08-02 14:45:58.484760840 +0200
@@ -5,8 +5,8 @@
 
 # Compiler dependent variables
 
-CC = gcc
-CXX = c++
+CC := ${CC}
+CXX := ${CXX}
 
 ARCHFLAGS = -mv8 -ffast-math -fomit-frame-pointer
 ARCHLDFLAGS = -static
