--- caffe2.orig/contrib/gloo/common_world_ops.cc        2017-08-22 14:58:26.233092558 -0400
+++ caffe2/contrib/gloo/common_world_ops.cc     2017-08-22 15:19:14.162203197 -0400
@@ -17,6 +17,7 @@
   // thread when they are used from an algorithm.
   //
   ::gloo::transport::tcp::attr attr;
+  attr.iface="ib0";
   static auto sharedDevice = ::gloo::transport::tcp::CreateDevice(attr);
   return sharedDevice;
 }

--- caffe2.orig/contrib/gloo/common_world_ops_gpu.cc    2017-08-22 14:58:26.233092558 -0400
+++ caffe2/contrib/gloo/common_world_ops_gpu.cc 2017-08-22 15:19:14.162203197 -0400
@@ -23,6 +23,7 @@
   static std::shared_ptr<::gloo::transport::Device> device;
   std::call_once(once, [&]() {
     ::gloo::transport::tcp::attr attr;
+    attr.iface="ib0";
     device = ::gloo::transport::tcp::CreateDevice(attr);
 
     // This operator is the first time any Gloo code is executed
     
