Author: Denis Kristak (Inuits)
Resolving deprecated code.
diff -ruN pyfasta-0.5.2_orig/pyfasta/fasta.py pyfasta-0.5.2/pyfasta/fasta.py
--- pyfasta-0.5.2_orig/pyfasta/fasta.py	2023-09-15 13:28:26.551806575 +0100
+++ pyfasta-0.5.2/pyfasta/fasta.py	2023-09-15 13:45:59.899502653 +0100
@@ -1,11 +1,11 @@
 from __future__ import print_function
 import string
 import os.path
-from collections import Mapping
+from collections.abc import Mapping
 import sys
 import numpy as np
 
-from records import NpyFastaRecord
+from .records import NpyFastaRecord
 
 # string.maketrans is bytes.maketrans in Python 3, but
 # we want to deal with strings instead of bytes
diff -ruN pyfasta-0.5.2_orig/pyfasta/__init__.py pyfasta-0.5.2/pyfasta/__init__.py
--- pyfasta-0.5.2_orig/pyfasta/__init__.py	2023-09-15 13:28:26.551806575 +0100
+++ pyfasta-0.5.2/pyfasta/__init__.py	2023-09-15 13:53:48.149038934 +0100
@@ -1,8 +1,8 @@
 from __future__ import print_function
 import sys
-from fasta import Fasta, complement, DuplicateHeaderException
-from records import *
-from split_fasta import split
+from .fasta import Fasta, complement, DuplicateHeaderException
+from .records import *
+from .split_fasta import split
 import optparse
 
 def main():
diff -ruN pyfasta-0.5.2_orig/pyfasta/records.py pyfasta-0.5.2/pyfasta/records.py
--- pyfasta-0.5.2_orig/pyfasta/records.py	2023-09-15 13:28:26.551806575 +0100
+++ pyfasta-0.5.2/pyfasta/records.py	2023-09-15 13:49:45.216271226 +0100
@@ -1,4 +1,4 @@
-import cPickle
+import pickle
 import numpy as np
 import sys
 import os
diff -ruN pyfasta-0.5.2_orig/pyfasta/split_fasta.py pyfasta-0.5.2/pyfasta/split_fasta.py
--- pyfasta-0.5.2_orig/pyfasta/split_fasta.py	2023-09-15 13:28:26.551806575 +0100
+++ pyfasta-0.5.2/pyfasta/split_fasta.py	2023-09-15 13:56:46.530774317 +0100
@@ -5,7 +5,7 @@
 import string
 import sys
 import optparse
-from cStringIO import StringIO
+from io import StringIO
 
 
 def newnames(oldname, n, kmers=None, overlap=None, header=None):
