--- Makefile	2011-08-02 20:21:59.000000000 +0200
+++ ../Makefile	2013-12-05 12:23:41.941714435 +0100
@@ -19,7 +19,7 @@
 # below to the path of your fortran compiler. The parallel version
 # must be compiled with gfortran version 4.2 or greater. 
 #
-FORTRAN = /usr/bin/gfortran
+FORTRAN = /apps/gent/SL6/sandybridge/software/ifort/2011.13.367/bin/intel64/ifort
 #
 FORTRAN_PARALLEL = gfortran 
 #
@@ -35,6 +35,7 @@
 #
 # Change the flags of the compilation if you want:
 #
+ 
 SERIALFLAGS = -O3 -ffast-math 
 #
 # Flags for the compilation of the parallel version
@@ -49,6 +50,7 @@
 #
 # Get the default fortran compiler
 #
+ 
 ifeq ($(FORTRAN),AUTO)
 FORTRAN = $(FC)
 endif 
@@ -69,7 +71,6 @@
 	@echo " Flags: $(SERIALFLAGS) " 
 	@echo " ------------------------------------------------------ " 
 	@$(FORTRAN) $(SERIALFLAGS) $(srcall) $(srcserial) -o packmol
-	@chmod +x solvate.tcl 
 	@echo " Packmol succesfully built." 
 	@echo " ------------------------------------------------------ " 
 #
@@ -83,7 +84,6 @@
 	@$(FORTRAN_PARALLEL) $(PARALLELFLAGS) $(srcall) $(srcparallel)\
                            -o packmol
 	@chmod +x ppackmol
-	@chmod +x solvate.tcl 
 	@echo " ------------------------------------------------------ " 
 	@echo " Packmol succesfully built. Paralell version available. " 
 	@echo " ------------------------------------------------------ " 
@@ -96,7 +96,6 @@
 	@echo " Flags: -Wunused"
 	@echo " ------------------------------------------------------ "
 	@$(FORTRAN) -Wunused $(srcall) $(srcserial) -o packmol
-	@chmod +x solvate.tcl 
 	@echo " ------------------------------------------------------ " 
 	@echo " Packmol succesfully built. " 
 	@echo " ------------------------------------------------------ " 
