diff -ru amos-3.1.0.orig/src/Align/find-tandem.cc amos-3.1.0/src/Align/find-tandem.cc
--- amos-3.1.0.orig/src/Align/find-tandem.cc	2011-07-25 01:27:26.000000000 +0200
+++ amos-3.1.0/src/Align/find-tandem.cc	2013-03-27 13:51:23.024914923 +0100
@@ -7,6 +7,7 @@
 #include <vector>
 #include <ctime>
 #include <sys/time.h>
+#include <unistd.h>
 using namespace std;
 
 const int OFFSET_TABLE_SIZE = 100;
