diff -Nru detettore-2.0.2.orig/pyproject.toml detettore-2.0.2/pyproject.toml
--- detettore-2.0.2.orig/pyproject.toml	2021-10-27 14:19:08.998749500 +0200
+++ detettore-2.0.2/pyproject.toml	2022-01-14 10:41:10.225724000 +0100
@@ -10,11 +10,11 @@
 [tool.poetry.dependencies]
 python = ">=3.7,<3.10"
 biopython = "^1.79"
-numpy = "^1.21.0"
+numpy = "^1.20.0"
 numba = "^0.53.1"
-scipy = "^1.7.0"
+scipy = "^1.6.0"
 #mappy = {url = "https://github.com/lh3/minimap2/releases/download/v2.22/minimap2-2.22.tar.bz2"}
-pysam = "^0.17.0"
+pysam = "^0.16.0"
 
 [tool.poetry.scripts]
 detettore = "detettore:main"
diff -Nru detettore-2.0.2.orig/setup.py detettore-2.0.2/setup.py
--- detettore-2.0.2.orig/setup.py	2021-10-27 14:20:05.735790500 +0200
+++ detettore-2.0.2/setup.py	2022-01-14 10:33:52.841864000 +0100
@@ -12,9 +12,9 @@
 install_requires = \
 ['biopython>=1.79,<2.0',
  'numba>=0.53.1,<0.54.0',
- 'numpy>=1.21.0,<2.0.0',
- 'pysam>=0.17.0,<0.18.0',
- 'scipy>=1.7.0,<2.0.0']
+ 'numpy>=1.20.0,<2.0.0',
+ 'pysam>=0.16.0,<0.18.0',
+ 'scipy>=1.6.0,<2.0.0']
 
 entry_points = \
 {'console_scripts': ['bamstats = scripts.bamstats:main',
