diff -rupN pango-1.57.0_orig/pango/meson.build pango-1.57.0/pango/meson.build
--- pango-1.57.0_orig/pango/meson.build	2025-08-25 22:39:22.000000000 +0200
+++ pango-1.57.0/pango/meson.build	2026-02-17 12:43:39.608351693 +0100
@@ -566,7 +566,7 @@ if cairo_dep.found()
     sources: pangocairo_dep_sources,
   )
   meson.override_dependency('pangocairo', libpangocairo_dep)
-  pango_cairo_requires = [ 'pango', 'cairo' ]
+  pango_cairo_requires = [ 'pango', 'cairo', 'fontconfig' ]
 
   pkgconfig.generate(libpangocairo,
     name: 'Pango Cairo',
