diff --git a/annotation/ImageScopeRepository.cpp b/annotation/ImageScopeRepository.cpp
index 6916a58..c135cb3 100644
--- a/annotation/ImageScopeRepository.cpp
+++ b/annotation/ImageScopeRepository.cpp
@@ -9,6 +9,7 @@
 #include <stdio.h>
 #include "pugixml.hpp"
 #include <algorithm>
+#include <cmath>
 
 ImageScopeRepository::ImageScopeRepository(const std::shared_ptr<AnnotationList>& list) :
   Repository(list),
