Use FlexiBLAS when available

Åke Sandgren, 2023-05-03
diff -ru flint-2.8.4.orig/CMake/FindCBLAS.cmake flint-2.8.4/CMake/FindCBLAS.cmake
--- flint-2.8.4.orig/CMake/FindCBLAS.cmake	2021-11-17 18:43:50.000000000 +0100
+++ flint-2.8.4/CMake/FindCBLAS.cmake	2023-05-03 09:29:36.599062848 +0200
@@ -11,13 +11,13 @@
 find_path(CBLAS_INCLUDE_DIRS NAMES cblas.h
   HINTS CBLAS_ROOT ENV CBLAS_ROOT
   PATHS ${INCLUDE_INSTALL_DIR} ${CMAKE_INSTALL_PREFIX}/include
-  PATH_SUFFIXES openblas cblas blis
+  PATH_SUFFIXES flexiblas openblas cblas blis
   )
 
-find_library(CBLAS_LIBRARIES NAMES accelerate openblas cblas blas blis
+find_library(CBLAS_LIBRARIES NAMES accelerate flexiblas openblas cblas blas blis
   HINTS CBLAS_ROOT ENV CBLAS_ROOT
   PATHS ${LIB_INSTALL_DIR} ${CMAKE_INSTALL_PREFIX}/lib
-  PATH_SUFFIXES openblas cblas blis
+  PATH_SUFFIXES flexiblas openblas cblas blis
   )
 
 include(FindPackageHandleStandardArgs)
