--- CalculiX/ccx_2.7/src/Makefile.orig	2014-10-15 14:46:37.755959000 +1300
+++ CalculiX/ccx_2.7/src/Makefile	2014-10-16 09:29:07.357968000 +1300
@@ -1,9 +1,9 @@
 
-CFLAGS = -Wall -O3  -I ../../../SPOOLES.2.2 -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE
+CFLAGS = -Wall -O3  -I $(EBROOTSPOOLES) -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE
 FFLAGS = -Wall -O3
 
 CC=cc
-FC=gfortran
+FC=$(F90)
 
 .c.o :
 	$(CC) $(CFLAGS) -c $<
@@ -18,12 +18,12 @@
 OCCXC = $(SCCXC:.c=.o)
 OCCXMAIN = $(SCCXMAIN:.c=.o)
 
-DIR=../../../SPOOLES.2.2
+DIR=$(EBROOTSPOOLES)
 
 LIBS = \
        $(DIR)/spooles.a \
-	../../../ARPACK/libarpack_INTEL.a \
-       -lpthread -lm -lc
+	  $(EBROOTARPACK)/lib/libarpack_linux.a \
+       -lpthread -lm -lc -lg2c
 
 ccx_2.7: $(OCCXMAIN) ccx_2.7.a  $(LIBS)
 	./date.pl; $(CC) $(CFLAGS) -c ccx_2.7.c; $(FC) -Wall -O3 -o $@ $(OCCXMAIN) ccx_2.7.a $(LIBS)
