fix undefined references to types in Boost namespace
author: Kenneth Hoste (HPC-UGent)
--- graph-tool-2.26/src/graph/centrality/graph_eigentrust.hh.orig	2018-01-24 16:43:52.802758216 +0100
+++ graph-tool-2.26/src/graph/centrality/graph_eigentrust.hh	2018-01-24 17:44:12.029412031 +0100
@@ -92,7 +92,7 @@
                      t_temp[v] = 0;
                      for (const auto& e : in_or_out_edges_range(v, g))
                      {
-                         typename graph_traits<Graph>::vertex_descriptor s;
+                         typename boost::graph_traits<Graph>::vertex_descriptor s;
                          if (is_directed::apply<Graph>::type::value)
                              s = source(e, g);
                          else
--- graph-tool-2.26/src/graph/centrality/graph_closeness.hh.orig	2018-01-24 17:55:24.470006264 +0100
+++ graph-tool-2.26/src/graph/centrality/graph_closeness.hh	2018-01-24 17:55:32.370130766 +0100
@@ -70,7 +70,7 @@
             (g,
              [&](auto v)
              {
-                 unchecked_vector_property_map<val_type,VertexIndex>
+                 boost::unchecked_vector_property_map<val_type,VertexIndex>
                      dist_map(vertex_index, num_vertices(g));
 
                  for (auto u : vertices_range(g))
