Author: Jasper Grimm
Ensure that shared library correctly links dependencies
diff -Nru vcflib-1.0.3/CMakeLists.txt vcflib-1.0.3_p2/CMakeLists.txt
--- vcflib-1.0.3/CMakeLists.txt	2022-03-29 15:31:38.000000000 +0100
+++ vcflib-1.0.3_p2/CMakeLists.txt	2022-03-29 15:36:41.000000000 +0100
@@ -244,6 +244,7 @@
 set_target_properties(vcflib_static PROPERTIES OUTPUT_NAME vcflib)
 add_library(vcflib SHARED ${SOURCES})
 set_target_properties(vcflib PROPERTIES VERSION ${BUILD_NUMBER} SOVERSION 1)
+target_link_libraries(vcflib tabixpp fastahack sw fsom filevercmp ${LDFLAGS} ${LIBS})
 
 # ---- Build htslib
 #
