diff -ruN brunsli-0.1_orig/c/dec/brunsli_decode.cc brunsli-0.1/c/dec/brunsli_decode.cc
--- brunsli-0.1_orig/c/dec/brunsli_decode.cc	2021-09-29 15:46:08.292937193 +0100
+++ brunsli-0.1/c/dec/brunsli_decode.cc	2021-09-30 12:27:04.564828827 +0100
@@ -13,7 +13,7 @@
 #include <string>
 #include <vector>
 
-#include <brotli/decode.h>
+#include "/scratch/gent/vo/001/gvo00117/easybuild/CO7/skylake-eth/software/Brotli/1.0.9-GCCcore-10.2.0/include/brotli/decode.h"
 #include "../common/constants.h"
 #include "../common/context.h"
 #include "../common/lehmer_code.h"
diff -ruN brunsli-0.1_orig/Makefile brunsli-0.1/Makefile
--- brunsli-0.1_orig/Makefile	2021-09-29 15:46:08.292937193 +0100
+++ brunsli-0.1/Makefile	2021-09-29 15:47:40.451603687 +0100
@@ -1,8 +1,8 @@
 OS := $(shell uname)
 BINDIR = bin
 OBJDIR = $(BINDIR)/obj
-BROTLI_DIR = third_party/brotli
-BROTLI_INCLUDE = $(BROTLI_DIR)/c/include
+BROTLI_DIR = $EBROOTBROTLI
+BROTLI_INCLUDE = $(BROTLI_DIR)/include
 BRUNSLI_INCLUDE = c/include
 SOURCES = $(wildcard c/common/*.cc) $(wildcard c/dec/*.cc) \
           $(wildcard c/enc/*.cc)
diff -ruN brunsli-0.1_orig/third_party/CMakeLists.txt brunsli-0.1/third_party/CMakeLists.txt
--- brunsli-0.1_orig/third_party/CMakeLists.txt	2021-09-29 15:46:08.300937078 +0100
+++ brunsli-0.1/third_party/CMakeLists.txt	2021-09-29 15:48:26.650928556 +0100
@@ -5,13 +5,13 @@
 # https://opensource.org/licenses/MIT.
 
 # brotli
-if (NOT TARGET brotlicommon-static)
-  if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/brotli/c/include/brotli/decode.h")
-    message(SEND_ERROR "Brotli is required to build brunsli, run \"git submodule update --init\" to get it, then run cmake again")
-  endif ()
-  add_subdirectory(brotli EXCLUDE_FROM_ALL)
-endif ()
+# if (NOT TARGET brotlicommon-static)
+#   if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/brotli/c/include/brotli/decode.h")
+#     message(SEND_ERROR "Brotli is required to build brunsli, run \"git submodule update --init\" to get it, then run cmake again")
+#   endif ()
+#   add_subdirectory(brotli EXCLUDE_FROM_ALL)
+# endif ()
 
-if (NOT TARGET gtest_main)
-  add_subdirectory(googletest EXCLUDE_FROM_ALL)
-endif ()
+# if (NOT TARGET gtest_main)
+#   add_subdirectory(googletest EXCLUDE_FROM_ALL)
+# endif ()
diff -ruN brunsli-0.1_orig/WORKSPACE brunsli-0.1/WORKSPACE
--- brunsli-0.1_orig/WORKSPACE	2021-09-29 15:46:08.292937193 +0100
+++ brunsli-0.1/WORKSPACE	2021-09-30 12:23:24.610917528 +0100
@@ -7,7 +7,7 @@
 
 local_repository(
     name = "brotli",
-    path = "third_party/brotli",
+    path = "/scratch/gent/vo/001/gvo00117/easybuild/CO7/skylake-eth/software/Brotli/1.0.9-GCCcore-10.2.0/",
 )
 
 new_local_repository(
