--- src/fft3dlib.F.orig	2013-09-06 19:52:02.161358630 -0700
+++ src/fft3dlib.F	2013-09-06 20:01:05.970692557 -0700
@@ -106,8 +106,8 @@
 
       DIMENSION AR((N-1)*INCN+(L-1)*INCL+1),AI((N-1)*INCN+(L-1)*INCL+1)
       DIMENSION WORK(4*L*N),TRIGS(2*L),IFAC(19)
-      DATA IWOFF /0/, MAXOLD /0/
       SAVE IWOFF,MAXOLD
+      DATA IWOFF /0/, MAXOLD /0/
 ! no data return immediately
       IF (N.EQ.0) RETURN
 ! IFAC(1) contains the number of factors
@@ -1072,8 +1072,8 @@
 
       DIMENSION AR((N-1)*INCN+(LR+1)*INCLR+1)
       DIMENSION WORK(2*LR*N),TRIGS(2*LR),IFAC(19)
-      DATA IWOFF /0/, MAXOLD /0/
       SAVE IWOFF,MAXOLD
+      DATA IWOFF /0/, MAXOLD /0/
 ! no data return immediately
       IF (N.EQ.0) RETURN
 ! IFAC(1) contains the number of factors
