--- EMBOSS-6.5.7/Makefile.in.orig	2012-07-24 16:11:48.000000000 +0200
+++ EMBOSS-6.5.7/Makefile.in	2014-07-09 14:35:41.606305045 +0200
@@ -791,7 +791,7 @@
 	tar cBf - jemboss | ( cd $(distdir); tar xBf - ; find jemboss -name CVS | xargs rm -rf; find jemboss -name Makefile | xargs rm -rf; find jemboss -name .cvsignore | xargs rm -rf )
 
 install-exec-hook:
-	$(bindir)/embossupdate
+	echo "Skipping $(bindir)/embossupdate"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- EMBOSS-6.5.7/Makefile.am.orig	2012-07-22 13:16:46.000000000 +0200
+++ EMBOSS-6.5.7/Makefile.am	2014-07-09 14:35:45.946399118 +0200
@@ -34,4 +34,4 @@
 	tar cBf - jemboss | ( cd $(distdir); tar xBf - ; find jemboss -name CVS | xargs rm -rf; find jemboss -name Makefile | xargs rm -rf; find jemboss -name .cvsignore | xargs rm -rf )
 
 install-exec-hook:
-	$(bindir)/embossupdate
+	echo "Skipping $(bindir)/embossupdate"
