diff -ru stampy-1.0.23.orig/makefile stampy-1.0.23/makefile
--- stampy-1.0.23.orig/makefile	2013-12-14 16:30:57.000000000 +0100
+++ stampy-1.0.23/makefile	2014-11-21 10:35:58.236080000 +0100
@@ -41,7 +41,7 @@
 buildall: $(bdir)/readalign.o $(bdir)/frontend.o $(bdir)/algebras.o
 	cp -r build/$(python)/* .
 ifeq ($(platform),linux-x86_64)
-	g++ `$(python)-config --ldflags` -pthread -shared -Wl $(objs) -o maptools.so
+	g++ `$(python)-config --ldflags` -pthread -shared $(objs) -o maptools.so
 else
 	g++ `$(python)-config --ldflags` -pthread -dynamiclib -Wl $(objs) -o maptools.so
 endif
