Restore installation functionality which was removed in v0.9.9.6

Åke Sandgren, 20200519
diff -ru glm-0.9.9.8.orig/CMakeLists.txt glm-0.9.9.8/CMakeLists.txt
--- glm-0.9.9.8.orig/CMakeLists.txt	2020-04-13 19:41:16.000000000 +0200
+++ glm-0.9.9.8/CMakeLists.txt	2020-05-19 16:16:12.731259305 +0200
@@ -8,6 +8,8 @@
 add_subdirectory(glm)
 add_library(glm::glm ALIAS glm)
 
+install(DIRECTORY glm DESTINATION "include")
+
 if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
 
 add_subdirectory(test)
