use $EBROOTMDSPLUSMINJAVA environment variable to refer to installation directory

author: Simon Pinches

diff -Nru mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatcherIp.template mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatcherIp.template
--- mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatcherIp.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatcherIp.template	2019-09-26 16:26:07.000000000 +0200
@@ -2,7 +2,7 @@
 #!
 #
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/lib \
-SHLIB_PATH=$MDSPLUS_DIR/lib \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jDispatcher.jar:$MDSPLUS_DIR/java/classes: \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/lib \
+SHLIB_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDispatcher.jar:$EBROOTMDSPLUSMINJAVA/java/classes: \
 java jDispatcherIp $1 $2 $3 $4 $5
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatchMonitor.template mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatchMonitor.template
--- mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatchMonitor.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatchMonitor.template	2019-09-26 16:26:08.000000000 +0200
@@ -2,7 +2,7 @@
 #!
 #
 exec env \
-LD_LIBRARY_PATH=$MDSPLUS_DIR/lib \
-SHLIB_PATH=$MDSPLUS_DIR/lib \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jDispatcher.jar:$MDSPLUS_DIR/java/classes: \
+LD_LIBRARY_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+SHLIB_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDispatcher.jar:$EBROOTMDSPLUSMINJAVA/java/classes: \
 java jDispatchMonitor $1 $2 $3 $4 $5
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jServer.template mdsplus-stable_release-7-84-6/java/jdispatcher/jServer.template
--- mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jServer.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jdispatcher/jServer.template	2019-09-26 16:26:08.000000000 +0200
@@ -2,7 +2,7 @@
 #!
 #
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/lib \
-SHLIB_PATH=$MDSPLUS_DIR/lib \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jDispatcher.jar:$MDSPLUS_DIR/java/classes: \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/lib \
+SHLIB_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDispatcher.jar:$EBROOTMDSPLUSMINJAVA/java/classes: \
 java -Xss10M jServer $1
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jscope/jScope.template mdsplus-stable_release-7-84-6/java/jscope/jScope.template
--- mdsplus-stable_release-7-84-6.orig/java/jscope/jScope.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jscope/jScope.template	2019-09-26 16:26:08.000000000 +0200
@@ -4,17 +4,17 @@
 #
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib64 ] ; then
     lib=/lib64
   fi 
 else  
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib32 ]; then
      lib=lib32
   fi
 fi  
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes:$HOME:$MDSPLUS_DIR/java/classes/mdsobjects.jar \
-java -Xmx512M -Djava.library.path=$MDSPLUS_DIR/${lib}  jScope $1 $2 $3 $4 $5 $6 $7
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes:$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/mdsobjects.jar \
+java -Xmx512M -Djava.library.path=$EBROOTMDSPLUSMINJAVA/${lib}  jScope $1 $2 $3 $4 $5 $6 $7
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jtraverser/jTraverser.template mdsplus-stable_release-7-84-6/java/jtraverser/jTraverser.template
--- mdsplus-stable_release-7-84-6.orig/java/jtraverser/jTraverser.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jtraverser/jTraverser.template	2019-09-26 16:26:08.000000000 +0200
@@ -4,17 +4,17 @@
 #
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib64 ] ; then
     lib=/lib64
   fi
 else
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib32 ]; then
      lib=lib32
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-CLASSPATH=$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jDevices.jar \
-java -Xss5M -Djava.library.path=$MDSPLUS_DIR/${lib} jTraverser $1 $2 $3 $4 $5 $6 $7
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+CLASSPATH=$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDevices.jar \
+java -Xss5M -Djava.library.path=$EBROOTMDSPLUSMINJAVA/${lib} jTraverser $1 $2 $3 $4 $5 $6 $7
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jtraverser2/jTraverser2.template mdsplus-stable_release-7-84-6/java/jtraverser2/jTraverser2.template
--- mdsplus-stable_release-7-84-6.orig/java/jtraverser2/jTraverser2.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jtraverser2/jTraverser2.template	2019-09-26 16:26:08.000000000 +0200
@@ -1,16 +1,16 @@
 #!/bin/sh
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib64 ] ; then
     lib=/lib64
   fi
 else
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib32 ]; then
      lib=lib32
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-java -jar $MDSPLUS_DIR/java/classes/jTraverser2.jar "$@"
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+java -jar $EBROOTMDSPLUSMINJAVA/java/classes/jTraverser2.jar "$@"
diff -Nru mdsplus-stable_release-7-84-6.orig/Makefile.in mdsplus-stable_release-7-84-6/Makefile.in
--- mdsplus-stable_release-7-84-6.orig/Makefile.in	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/Makefile.in	2019-09-26 16:28:01.000000000 +0200
@@ -40,8 +40,8 @@
 D3D_PACKAGE = d3dshr
 
 ifeq "@JAVA_ONLY@" "yes"
-NON_JAR =
-PARTS = $(JAVA_JAR)
+NON_JAR = @JAVA_APS@
+PARTS = $(NON_JAR) $(JAVA_JAR)
 else
 NON_JAR = \
 	mdsshr \
