diff -Nru mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatcherIp.template mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatcherIp.template
--- mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatcherIp.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatcherIp.template	2019-09-26 16:26:07.000000000 +0200
@@ -2,7 +2,7 @@
 #!
 #
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/lib \
-SHLIB_PATH=$MDSPLUS_DIR/lib \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jDispatcher.jar:$MDSPLUS_DIR/java/classes: \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/lib \
+SHLIB_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDispatcher.jar:$EBROOTMDSPLUSMINJAVA/java/classes: \
 java jDispatcherIp $1 $2 $3 $4 $5
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatchMonitor.template mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatchMonitor.template
--- mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jDispatchMonitor.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jdispatcher/jDispatchMonitor.template	2019-09-26 16:26:08.000000000 +0200
@@ -2,7 +2,7 @@
 #!
 #
 exec env \
-LD_LIBRARY_PATH=$MDSPLUS_DIR/lib \
-SHLIB_PATH=$MDSPLUS_DIR/lib \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jDispatcher.jar:$MDSPLUS_DIR/java/classes: \
+LD_LIBRARY_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+SHLIB_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDispatcher.jar:$EBROOTMDSPLUSMINJAVA/java/classes: \
 java jDispatchMonitor $1 $2 $3 $4 $5
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jServer.template mdsplus-stable_release-7-84-6/java/jdispatcher/jServer.template
--- mdsplus-stable_release-7-84-6.orig/java/jdispatcher/jServer.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jdispatcher/jServer.template	2019-09-26 16:26:08.000000000 +0200
@@ -2,7 +2,7 @@
 #!
 #
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/lib \
-SHLIB_PATH=$MDSPLUS_DIR/lib \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jDispatcher.jar:$MDSPLUS_DIR/java/classes: \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/lib \
+SHLIB_PATH=$EBROOTMDSPLUSMINJAVA/lib \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDispatcher.jar:$EBROOTMDSPLUSMINJAVA/java/classes: \
 java -Xss10M jServer $1
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jscope/jScope.template mdsplus-stable_release-7-84-6/java/jscope/jScope.template
--- mdsplus-stable_release-7-84-6.orig/java/jscope/jScope.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jscope/jScope.template	2019-09-26 16:26:08.000000000 +0200
@@ -4,17 +4,17 @@
 #
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib64 ] ; then
     lib=/lib64
   fi 
 else  
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib32 ]; then
      lib=lib32
   fi
 fi  
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-CLASSPATH=$HOME:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes:$HOME:$MDSPLUS_DIR/java/classes/mdsobjects.jar \
-java -Xmx512M -Djava.library.path=$MDSPLUS_DIR/${lib}  jScope $1 $2 $3 $4 $5 $6 $7
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+CLASSPATH=$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes:$HOME:$EBROOTMDSPLUSMINJAVA/java/classes/mdsobjects.jar \
+java -Xmx512M -Djava.library.path=$EBROOTMDSPLUSMINJAVA/${lib}  jScope $1 $2 $3 $4 $5 $6 $7
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jtraverser/jTraverser.template mdsplus-stable_release-7-84-6/java/jtraverser/jTraverser.template
--- mdsplus-stable_release-7-84-6.orig/java/jtraverser/jTraverser.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jtraverser/jTraverser.template	2019-09-26 16:26:08.000000000 +0200
@@ -4,17 +4,17 @@
 #
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib64 ] ; then
     lib=/lib64
   fi
 else
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib32 ]; then
      lib=lib32
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-CLASSPATH=$MDSPLUS_DIR/java/classes/jTraverser.jar:$MDSPLUS_DIR/java/classes/jScope.jar:$MDSPLUS_DIR/java/classes/jDevices.jar \
-java -Xss5M -Djava.library.path=$MDSPLUS_DIR/${lib} jTraverser $1 $2 $3 $4 $5 $6 $7
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+CLASSPATH=$EBROOTMDSPLUSMINJAVA/java/classes/jTraverser.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jScope.jar:$EBROOTMDSPLUSMINJAVA/java/classes/jDevices.jar \
+java -Xss5M -Djava.library.path=$EBROOTMDSPLUSMINJAVA/${lib} jTraverser $1 $2 $3 $4 $5 $6 $7
diff -Nru mdsplus-stable_release-7-84-6.orig/java/jtraverser2/jTraverser2.template mdsplus-stable_release-7-84-6/java/jtraverser2/jTraverser2.template
--- mdsplus-stable_release-7-84-6.orig/java/jtraverser2/jTraverser2.template	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/java/jtraverser2/jTraverser2.template	2019-09-26 16:26:08.000000000 +0200
@@ -1,16 +1,16 @@
 #!/bin/sh
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib64 ] ; then
     lib=/lib64
   fi
 else
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUSMINJAVA/lib32 ]; then
      lib=lib32
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-java -jar $MDSPLUS_DIR/java/classes/jTraverser2.jar "$@"
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUSMINJAVA/${lib} \
+java -jar $EBROOTMDSPLUSMINJAVA/java/classes/jTraverser2.jar "$@"
diff -Nru mdsplus-stable_release-7-84-6.orig/Makefile.in mdsplus-stable_release-7-84-6/Makefile.in
--- mdsplus-stable_release-7-84-6.orig/Makefile.in	2019-09-02 18:02:07.000000000 +0200
+++ mdsplus-stable_release-7-84-6/Makefile.in	2019-09-26 16:28:01.000000000 +0200
@@ -40,8 +40,8 @@
 D3D_PACKAGE = d3dshr
 
 ifeq "@JAVA_ONLY@" "yes"
-NON_JAR =
-PARTS = $(JAVA_JAR)
+NON_JAR = @JAVA_APS@
+PARTS = $(NON_JAR) $(JAVA_JAR)
 else
 NON_JAR = \
 	mdsshr \
diff -Nru mdsplus-stable_release-7-84-6.orig/Makefile.in~ mdsplus-stable_release-7-84-6/Makefile.in~
--- mdsplus-stable_release-7-84-6.orig/Makefile.in~	1970-01-01 01:00:00.000000000 +0100
+++ mdsplus-stable_release-7-84-6/Makefile.in~	2019-09-02 18:02:07.000000000 +0200
@@ -0,0 +1,252 @@
+#@(#)$RCSfile$ $Revision$
+# @configure_input@
+
+include @top_builddir@/Makefile.inc
+
+srcdir=@srcdir@
+builddir=@builddir@ 
+VPATH=@srcdir@
+@AX_RECONFIGURE_TARGET@
+
+
+
+JAVA_JAR = \
+	   java/mdsplus-api \
+	   java/mdsplus-api/tests \
+	   java/jscope \
+	   java/jtraverser \
+	   java/jtraverser2 \
+	   java/jdispatcher \
+	   java/jdevices \
+	   java/mdsplus \
+	   java/mdsobjects
+
+JAVA_APS = \
+	   javamds \
+	   java/mdsplus-api/src/main/jni/mds/mdslib
+
+ifeq "@HAVE_MOTIF@" "yes"
+MOTIF_APS = xmdsshr \
+	    actions \
+	    dwscope \
+	    @IDLMDSWIDGETS@ \
+	    mitdevices \
+	    traverser
+else
+MOTIF_APS = actions
+endif
+HDF5_APS = hdf5
+
+D3D_PACKAGE = d3dshr
+
+ifeq "@JAVA_ONLY@" "yes"
+NON_JAR =
+PARTS = $(JAVA_JAR)
+else
+NON_JAR = \
+	mdsshr \
+	treeshr \
+	tdishr \
+	tditest \
+	xtreeshr \
+	@MDSTCPIP@ \
+	mdslibidl \
+	mdsdcl  \
+	@CAMSHR@ \
+	@REMCAM@ \
+	ccl \
+	servershr \
+	tcl \
+	@TDIC@ \
+	mdslib \
+	math \
+	wfevent \
+	setevent \
+	mdsmisc \
+	scripts \
+	rpm \
+	mdsobjects/cpp \
+	manpages \
+	@D3D_PACKAGE@ \
+	@JAVA_APS@ \
+	@MOTIF_APS@ \
+	@MDSSQL@ \
+	@IDLMDSEVENT@ \
+	@HDF5_APS@ \
+	@LV@ \
+	LabView \
+	@MACOSX@
+
+@ENABLE_TESTS_TRUE@ NON_JAR += testing
+
+PARTS = $(NON_JAR) @JAVA_JAR@
+endif
+
+.PHONY: all $(PARTS)
+all: ##@build build all active submodules
+all: ChangeLog $(PARTS)
+
+DIRECTORIES = $(sort @MAKEBINDIR@ @MAKELIBDIR@ @MAKESHLIBDIR@ @MAKEUIDDIR@)
+
+$(NON_JAR): $(DIRECTORIES)
+@MINGW_FALSE@	$(MAKE) -C $@ @MAKEFLAG_J@ @MAKEFLAG_O@
+@MINGW_TRUE@	$(MAKE) -C $@
+# override parallel flags for java apps
+$(JAVA_JAR): $(DIRECTORIES)
+	$(MAKE) -C $@
+
+.PHONY: docs
+docs:   ##@docs build documentation using doxygen (ref to docs directory)
+	$(MAKE) -C docs install
+
+
+
+clean_DIRS = $(addprefix clean_, $(PARTS))
+
+ChangeLog:
+	if (! git --git-dir=@top_srcdir@/.git log --decorate=full --no-merges > ChangeLog 2>/dev/null); then touch ChangeLog; fi
+
+.PHONY: clean
+clean: ##@build recusively clean all built objects in submodules
+clean: $(clean_DIRS) clean_TESTS clean_DOCS
+
+.PHONY: $(clean_DIRS)
+$(clean_DIRS):
+	$(MAKE) -C $(@:clean_%=%) clean
+
+clean_DOCS: ##@docs clean generated documentation
+	$(MAKE) -C docs clean
+
+.PHONY: distclean
+
+distclean: ##@build make source directory look like original sources
+	@cpwd=$$(pwd); \
+	cd @top_srcdir@; \
+	if [ -r .git ]; \
+	then \
+	  git clean -x -f -d -q; \
+	else \
+	  if [ "$$(pwd)" == "$$cpwd" ]; \
+	     then make full_clean > /dev/null; \
+	  fi; \
+	  rm -Rf ./python/MDSplus/build; \
+	fi; \
+	cd $$cpwd
+
+# Testing
+
+.PHONY: tests
+tests: ##@tests recursively perform tests in submodules
+@ENABLE_TESTS_TRUE@	$(MAKE) $(AM_MAKEFLAGS) -C testing all tests
+@ENABLE_TESTS_FALSE@	@echo "Tests disabled"
+
+.PHONY: tests-valgrind rebuild-tests
+tests-valgrind: ##@tests perform tests using valgrind tool
+rebuild-tests:  ##@tests rebuild all tests binaries
+tests-valgrind rebuild-tests:
+@ENABLE_TESTS_TRUE@	$(MAKE) $(AM_MAKEFLAGS) -C testing all $@
+@ENABLE_TESTS_FALSE@	@echo "Tests disabled"
+
+.PHONY: $(clean_TESTS)
+clean_TESTS: ##@tests clean all tests results and compiled tests objects
+@ENABLE_TESTS_TRUE@	$(MAKE) -C testing tests-clean
+
+
+.PHONY: full_clean
+full_clean: ##@build perform deep clean of build directories
+full_clean: clean
+	@rm -Rf bin bin64 bin32 bin_x86 bin_x86_64 etc \
+		lib lib64 lib32 java uid uid64 uid32 \
+		config.cache config.log config.status aclocal.m4 \
+		include/stamp-h1 autom4te.cache \
+		$$(find . -name '.deps') \
+		`find . -name '*.in' | \
+	      	$(AWK) '{print substr($$1,0,length($$1)-3)}'`
+
+.PHONY: depend
+depend:
+	cdir=`pwd`;\
+	set -e; for dir in $(PARTS) ; do\
+	  cd $${dir}; $${MAKE:=make} depend > /dev/null 2> /dev/null; cd $$cdir;\
+	done
+
+MISC_PARTS = tdi pydevices idl trees setup.sh setup.csh include desktop pixmaps matlab php epics xml nodejs python/MDSplus
+
+install_DIRS = $(addprefix install_, $(PARTS))
+$(install_DIRS):
+	$(MAKE) -C $(@:install_%=%) install
+
+.PHONY: install
+install: ##@build build and install active modules to prefix directory
+install: $(install_DIRS)
+	$(INSTALL) ChangeLog $(prefix)/
+	$(MKDIR_P) $(prefix)/local/tdi
+	$(MKDIR_P) $(prefix)/java
+	$(MKDIR_P) $(prefix)/python/MDSplus
+	- find $(prefix)/java -name '*.class' -delete #### No need to include class files
+	tar -C ${top_srcdir} \
+		--exclude='.gitignore' \
+		--exclude='*.pyc' \
+		--exclude=__pycache__ \
+		--exclude=python/MDSplus/build \
+		--exclude=python/MDSplus/docs \
+                --exclude=tdi/*Devices/build \
+		--exclude='*egg-info' \
+		--exclude='*\.in' \
+		--exclude='Makefile\.*' \
+		-cf  - $(MISC_PARTS) | (cd $(exec_prefix); tar -xf -)
+	tar -C ${top_builddir} -cf - include/mdsplus/mdsconfig.h $$(find . -name _version.py) | (cd $(exec_prefix); tar -xf -) ## TODO: remove this and config.h from headers ...
+	if [ ! -z "$$MDSPLUS_VERSION" ]; then echo "mdsplus_version='$$MDSPLUS_VERSION'" > $(exec_prefix)/python/MDSplus/mdsplus_version.py; fi
+	$(INSTALL) ${top_srcdir}/MDSplus-License.txt ${top_srcdir}/MDSplus-License.rtf $(exec_prefix)
+	(cd $(exec_prefix); chmod -R 755 $(MISC_PARTS))
+
+.PHONY: tests-envdump
+tests-envdump:
+	cd python/MDSobjects/tests&&(\
+	make test-envdump;\
+	cd ../../..;)
+
+test-env:
+	cd python/MDSplus/tests&&(\
+	make test-env;\
+	cd ../../..;)
+
+@HAVE_WINE_TRUE@wineconsole:
+@HAVE_WINE_TRUE@	cd python/MDSplus/tests&&(\
+@HAVE_WINE_TRUE@	make wineconsole;\
+@HAVE_WINE_TRUE@	cd ../../..;)
+
+# Interdependent directories:
+actions: mdsshr tdishr treeshr xmdsshr mdstcpip servershr
+camshr: mdsdcl
+ccl: camshr  mdsshr mdsdcl
+tcl: mdsshr treeshr tdishr mdstcpip servershr mdsdcl
+dwscope: xmdsshr
+java/jdevices: java/jtraverser
+java/jdispatcher: java/jscope java/jtraverser
+javamds: mdstcpip
+java/jtraverser: java/jscope
+java/jtraverser2: java/mdsplus-api
+java/mdsplus-api/tests: java/mdsplus-api
+math: mdsshr
+mdsdcl: mdsshr
+mdslib: mdstcpip
+mdslibidl: tdishr
+mdsmisc: tdishr xtreeshr
+mdsobjects/cpp: mdstcpip
+java/mdsobjects: java/jscope
+mdstcpip: tdishr
+mitdevices: mdsshr tdishr treeshr xmdsshr
+remcam: mdstcpip
+servershr: mdsdcl tdishr treeshr mdsshr mdstcpip
+setevent: mdsshr
+tdic: mdstcpip
+tdishr: mdsshr treeshr
+tditest: tdishr
+traverser: xmdsshr
+treeshr: mdsshr
+wfevent: tdishr mdsshr
+xmdsshr: tdishr
+xtreeshr: tdishr
+
+am--refresh:
