Only in statgen-xhmm-cc14e528d909.patched: build
diff -ru statgen-xhmm-cc14e528d909/Makefile statgen-xhmm-cc14e528d909.patched/Makefile
--- statgen-xhmm-cc14e528d909/Makefile	2016-01-04 18:10:47.000000000 +0100
+++ statgen-xhmm-cc14e528d909.patched/Makefile	2016-06-09 17:37:55.000000000 +0200
@@ -86,7 +86,8 @@
 ifeq ($(LAPACK_LIBS),)
 #LAPACK_LIBS=lapack f77blas cblas atlas gfortran
 #LAPACK_LIBS=acml gfortran
-LAPACK_LIBS=lapack
+#LAPACK_LIBS=lapack
+LAPACK_LIBS=scalapack openblas gfortran
 endif
 LAPACK_LIB_LINKS =$(LAPACK_LIBS:%=-l%)
 
Only in statgen-xhmm-cc14e528d909.patched/sources/hmm++: build
Only in statgen-xhmm-cc14e528d909.patched: xhmm
