use FlexiBLAS as BLAS/LAPACK library
author: Kenneth Hoste (HPC-UGent)
diff -ru dlib-19.24.6.orig/dlib/cmake_utils/find_blas.cmake dlib-19.24.6/dlib/cmake_utils/find_blas.cmake
--- dlib-19.24.6.orig/dlib/cmake_utils/find_blas.cmake	2024-02-18 14:43:31.000000000 +0100
+++ dlib-19.24.6/dlib/cmake_utils/find_blas.cmake	2024-09-20 20:35:35.927348475 +0200
@@ -66,17 +66,15 @@
    
    # First, search for libraries via pkg-config, which is the cleanest path
    find_package(PkgConfig)
-   pkg_check_modules(BLAS_REFERENCE cblas)
-   pkg_check_modules(LAPACK_REFERENCE lapack)
+   pkg_check_modules(BLAS_REFERENCE flexiblas)
    # Make sure the cblas found by pkgconfig actually has cblas symbols.
    SET(CMAKE_REQUIRED_LIBRARIES "${BLAS_REFERENCE_LDFLAGS}")   
    CHECK_FUNCTION_EXISTS(cblas_ddot PKGCFG_HAVE_CBLAS)
    if (BLAS_REFERENCE_FOUND AND LAPACK_REFERENCE_FOUND AND PKGCFG_HAVE_CBLAS)
       set(blas_libraries "${BLAS_REFERENCE_LDFLAGS}")
-      set(lapack_libraries "${LAPACK_REFERENCE_LDFLAGS}")
       set(blas_found 1)
       set(lapack_found 1)
-      set(REQUIRES_LIBS "${REQUIRES_LIBS} cblas lapack")
+      set(REQUIRES_LIBS "${REQUIRES_LIBS} flexiblas")
       message(STATUS "Found BLAS and LAPACK via pkg-config")
       return()
    endif()
