--- src/tools/stringx.h	2012-06-30 13:32:42.000000000 +1200
+++ src/tools/stringx.h	2014-07-14 16:15:07.971734000 +1200
@@ -164,16 +164,6 @@
     return svec;
 }
 
-template <class T>
-std::vector < string >  VecElemToString(vector <T> number, int width)
-{
-    std::vector < string > svec;
-    typename std::vector <T> :: iterator i;
-    for(i=number.begin(); i != number.end(); i++)
-        svec.push_back(Pretty(*i, width));
-    return svec;
-}
-
 // careful: was not able to link this function
 // when its name was DoubleVecFromString(..)
 bool FromString(const std::string & in, DoubleVec1d & out);
@@ -208,6 +198,16 @@
 std::string Pretty(unsigned long number, int width=DEFWIDTH);
 std::string Pretty(std::string str, int width=DEFWIDTH);
 
+template <class T>
+std::vector < string >  VecElemToString(vector <T> number, int width)
+{
+    std::vector < string > svec;
+    typename std::vector <T> :: iterator i;
+    for(i=number.begin(); i != number.end(); i++)
+        svec.push_back(Pretty(*i, width));
+    return svec;
+}
+
 bool   StringCompare(const std::string&, const char*, long, long);
 bool   StringCompare(const std::string&, const std::string&, long, long);
 
