diff -ru ea-utils.1.1.2-537.orig/Makefile ea-utils.1.1.2-537/Makefile
--- ea-utils.1.1.2-537.orig/Makefile	2013-01-09 22:27:54.000000000 +0100
+++ ea-utils.1.1.2-537/Makefile	2016-05-17 15:53:34.256217000 +0200
@@ -76,7 +76,7 @@
 
 # why the libbam.a doesn't work?  not sure... *.o works
 sam-stats: sam-stats.cpp samtools/libbam.a samtools/bam.h fastq-lib.h
-	$(CC) $(CFLAGS) samtools/*.o -lz -lpthread fastq-lib.cpp $< -o $@
+	$(CC) $(CFLAGS) samtools/*.o -L${EBROOTZLIB}/lib -lz -lpthread fastq-lib.cpp $< -o $@
 
 samtools/libbam.a: samtools/*.c samtools/*.h
 	cd samtools && make libbam.a
