--- configure.orig	2013-10-24 15:26:59.994151181 +0300
+++ configure	2013-10-24 15:29:31.293151084 +0300
@@ -8433,7 +8433,7 @@
         ;;
       # icc used to be incompatible with GCC.
       # ICC 10 doesn't accept -KPIC any more.
-      icc* | ifort*)
+      icc* | ifort* | mpif77*)
 	lt_prog_compiler_wl='-Wl,'
 	lt_prog_compiler_pic='-fPIC'
 	lt_prog_compiler_static='-static'
@@ -12052,7 +12052,7 @@
         ;;
       # icc used to be incompatible with GCC.
       # ICC 10 doesn't accept -KPIC any more.
-      icc* | ifort*)
+      icc* | ifort* | mpif77*)
 	lt_prog_compiler_wl_F77='-Wl,'
 	lt_prog_compiler_pic_F77='-fPIC'
 	lt_prog_compiler_static_F77='-static'
