--- ClonalFrame.orig/src/move_hidden.cpp
+++ ClonalFrame/src/move_hidden.cpp
@@ -166,7 +166,8 @@ namespace wb
                     sum+=gsl_matrix_get(f,j,i)*q[j][state][site-siteprev-1];
                 gsl_matrix_set(f,state,i+1,sum*gsl_matrix_get(e,state,msgs[site]));
             }
-            Util::normalize(&(gsl_matrix_column(f,i+1).vector));
+            gsl_vector_view view = gsl_matrix_column(f,i+1);
+            Util::normalize(&view.vector);
         }
         return f;
     }
@@ -420,7 +421,8 @@ namespace wb
                 gsl_matrix_set(e,i+1,2,(1.0-m2)*m3+(1.0-m3)*m2);
             }
 
-            Util::normalize(&(gsl_matrix_row(e,i+1).vector));
+            gsl_vector_view view = gsl_matrix_row(e,i+1);
+            Util::normalize(&view.vector);
         }
     }
