Fix missing variable declarations in SWAT+
author: Alex Domingo (Vrije Universiteit Brussel)
--- a/source_codes/hru_module.f90	2020-11-05 01:10:02.319592000 +0100
+++ b/source_codes/hru_module.f90	2020-11-05 00:17:18.062841634 +0100
@@ -381,4 +381,7 @@
       integer, dimension(:), allocatable :: tillage_days
       real, dimension(:), allocatable :: tillage_factor
 
-      end module hru_module
\ No newline at end of file
+!! sim_inityr variables
+      real, dimension (:), allocatable :: lai_yrmx  !none   |maximum leaf area index for the year in the HRU
+
+      end module hru_module
--- a/source_codes/ru_module.f90	2020-11-05 01:10:02.354478000 +0100
+++ b/source_codes/ru_module.f90	2020-11-05 00:05:38.659428000 +0100
@@ -8,6 +8,7 @@
       real, dimension (:), allocatable :: ru_n     !                | 
       !real, dimension (:), allocatable :: hyd_flo  !                |
       integer, dimension (:), allocatable :: itsb  !none            |end of loop
+      real, dimension (:,:), allocatable :: uhs    !none            |
    
       type ru_databases_char
         character(len=16) :: elem_def = ""
