diff -ruN unittest2-1.1.0_orig/setup.py unittest2-1.1.0/setup.py
--- unittest2-1.1.0_orig/setup.py	2021-09-07 14:45:56.638877809 +0200
+++ unittest2-1.1.0/setup.py	2021-09-07 14:52:28.782893924 +0200
@@ -57,7 +57,7 @@
 # Both install and setup requires - because we read VERSION from within the
 # package, and the package also exports all the APIs.
 # six for compat helpers
-REQUIRES = ['argparse', 'six>=1.4', 'traceback2'],
+REQUIRES = ['six>=1.4', 'traceback2'],
 
 params = dict(
     name=NAME,
diff -ruN unittest2-1.1.0_orig/unittest2.egg-info/requires.txt unittest2-1.1.0/unittest2.egg-info/requires.txt
--- unittest2-1.1.0_orig/unittest2.egg-info/requires.txt	2021-09-07 14:45:56.654877436 +0200
+++ unittest2-1.1.0/unittest2.egg-info/requires.txt	2021-09-07 14:47:15.013107770 +0200
@@ -1,3 +1,2 @@
-argparse
 six>=1.4
 traceback2
