diff --git a/cmake/cuda.cmake b/cmake/cuda.cmake
index c5ba334e9..11b621386 100644
--- a/cmake/cuda.cmake
+++ b/cmake/cuda.cmake
@@ -47,7 +47,7 @@ else()
     # CMake already puts /usr/lib/<arch>-linux-gnu/ after cuda own directory in the
     # `CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES`, so we always put NO_DEFAULT_PATH here.
     find_library(CUBLAS cublas
-        HINT ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES} NO_DEFAULT_PATH)
+        HINT ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES} )
     find_library(CUSPARSE cusparse
         HINT ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
     find_library(CURAND curand
