--- config/make.inc.gfort.dynlib.orig	2023-11-28 18:54:02.418406733 +0000
+++ config/make.inc.gfort.dynlib	2023-11-28 22:49:36.895697665 +0000
@@ -4,10 +4,10 @@
 F90 = gfortran
 
 #Add next two lines for parallel postw90. Choose the correct name of the mpi f90 wrapper
-#COMMS  = mpi
-#MPIF90 = mpgfortran #mpif90
+COMMS  = mpi
+MPIF90 = mpif90
 
-FCOPTS = -O3 -fPIC
+FCOPTS = -O3 -fPIC -fallow-argument-mismatch
 LDOPTS = -fPIC
 
 #Next two lines are good for debugging
@@ -34,4 +34,4 @@
 #   sudo apt-get install libblas-dev liblapack-dev
 # on Ubuntu
 #=======================
-LIBS = -llapack -lblas
+LIBS = -lflexiblas 
