--- config/make.inc.ifort.orig	2020-10-27 21:47:52.038033981 -0000
+++ config/make.inc.ifort	2020-10-27 21:48:26.130187955 -0000
@@ -4,7 +4,7 @@
 #=====================================================
 F90 = ifort
 COMMS=mpi
-MPIF90=mpiifort
+MPIF90=mpif90
 FCOPTS=-O2
 LDOPTS=-O2
 
@@ -12,8 +12,7 @@
 # Intel mkl libraries. Set LIBPATH if not in default path
 #========================================================
 
-LIBDIR = /opt/intel/mkl/lib/intel64
-LIBS   =  -L$(LIBDIR) -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lpthread
+LIBS   =   -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lpthread
 
 #=======================
 # ATLAS Blas and LAPACK
