--- config/make.inc.gfort.orig	2020-03-05 13:41:10.000000000 -0500
+++ config/make.inc.gfort	2022-06-02 09:22:27.141245863 -0400
@@ -4,7 +4,7 @@
 F90 = gfortran
 
 #Add next two lines for parallel postw90. Choose the correct name of the mpi f90 wrapper
-#COMMS  = mpi
+COMMS  = mpi
 #MPIF90 = mpgfortran #mpif90
 
 FCOPTS = -O3
@@ -34,4 +34,4 @@
 #   sudo apt-get install libblas-dev liblapack-dev
 # on Ubuntu
 #=======================
-LIBS = -llapack -lblas
+LIBS = $(LIBLAPACK) $(LIBBLAS)
