--- config/make.sys.ifort.orig	2019-01-25 09:58:52.000000000 -0800
+++ config/make.sys.ifort	2019-01-25 09:59:53.000000000 -0800
@@ -4,14 +4,14 @@
 #=================================================
 F90 = ifort
 FCOPTS = -O2
-LDOPTS = -O2 -Vaxlib 
+LDOPTS = -O2 
 
 #========================================================
 # Intel mkl libraries. Set LIBPATH if not in default path
 #========================================================
 
-LIBDIR = /opt/intel/mkl721/lib/32
-LIBS   =  -L$(LIBDIR) -lmkl_lapack  -lmkl_ia32 -lguide -lpthread
+#LIBDIR = /opt/intel/mkl721/lib/32
+LIBS   =  -lmkl_core -lmkl_intel_lp64 -lmkl_sequential -lpthread
 
 #=======================
 # ATLAS Blas and LAPACK
