diff -ru MultiQC-1.6.original/setup.py MultiQC-1.6.patched/setup.py
--- MultiQC-1.6.original/setup.py	2018-08-04 02:05:22.000000000 +0200
+++ MultiQC-1.6.patched/setup.py        2018-09-03 12:17:23.000000000 +0200
@@ -38,13 +38,22 @@
         'future>0.14.0',
         'lzstring',
         'jinja2>=2.9',
-        'matplotlib>=2.1.1',
+        'matplotlib==2.2.2',
         'markdown',
         'numpy',
         'pyyaml',
         'requests',
         'simplejson',
-        'spectra>=0.0.10'
+	'colormath==3.0.0',
+	'decorator==4.3.0',
+	'future==0.16.0',
+	'Jinja2==2.10',
+	'lzstring==1.0.4',
+	'Markdown==2.6.11',
+	'networkx==2.1',
+	'PyYAML==4.2b4',
+	'simplejson==3.16.0',
+	'spectra==0.0.11'
     ]

 setup(
