diff -ru multiqc-1.0.org/setup.py multiqc-1.0.patched/setup.py
--- multiqc-1.0.org/setup.py	2017-05-17 17:55:00.000000000 +0200
+++ multiqc-1.0.patched/setup.py	2018-03-15 13:37:39.636634698 +0100
@@ -48,15 +48,26 @@
     zip_safe = False,
     scripts = ['scripts/multiqc'],
     install_requires = [
-        'click',
-        'future>0.14.0',
-        'lzstring',
-        'jinja2>=2.9',
-        'matplotlib',
-        'numpy',
-        'pyyaml',
-        'simplejson',
-        'spectra',
+        'decorator==4.0.4',
+        'six==1.10.0',
+        'python-dateutil==2.4.2',
+        'pytz==2015.7',
+        'pyparsing==2.0.6',
+        'MarkupSafe==0.23',
+        'functools32==3.2.3-2',
+        'subprocess32==3.5.0rc1',
+        'cycler==0.10',
+        'colormath==2.1.1',
+        'networkx==1.11',
+        'click==6.7',
+        'future==0.16.0',
+        'lzstring==1.0.3',
+        'jinja2==2.9.6',
+        'matplotlib==2.0.2',
+        'numpy==1.10.1',
+        'pyyaml==3.12',
+        'simplejson==3.11.1',
+        'spectra==0.0.8',
     ],
     entry_points = {
         'multiqc.modules.v1': [
