diff -ru multiqc-1.0.org/setup.py multiqc-1.0.patched/setup.py
--- multiqc-1.0.org/setup.py	2017-05-17 17:55:00.000000000 +0200
+++ multiqc-1.0.patched/setup.py	2017-09-16 21:27:20.000000000 +0200
@@ -48,15 +48,16 @@
     zip_safe = False,
     scripts = ['scripts/multiqc'],
     install_requires = [
-        'click',
-        'future>0.14.0',
-        'lzstring',
-        'jinja2>=2.9',
-        'matplotlib',
+        'networkx==1.11',
+        'click==6.7',
+        'future==0.16.0',
+        'lzstring==1.0.3',
+        'jinja2==2.9.6',
+        'matplotlib==2.0.2',
         'numpy',
-        'pyyaml',
-        'simplejson',
-        'spectra',
+        'pyyaml==3.12',
+        'simplejson==3.11.1',
+        'spectra==0.0.8',
     ],
     entry_points = {
         'multiqc.modules.v1': [
