diff -Nru flann-1.9.1.orig/src/cpp/CMakeLists.txt flann-1.9.1/src/cpp/CMakeLists.txt
--- flann-1.9.1.orig/src/cpp/CMakeLists.txt	2016-08-04 23:20:50.000000000 +0000
+++ flann-1.9.1/src/cpp/CMakeLists.txt	2020-06-13 20:56:59.634970091 +0000
@@ -29,7 +29,7 @@
 endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
-    add_library(flann_cpp SHARED "")
+    add_library(flann_cpp SHARED empty.cpp)
     set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
     target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
 
@@ -83,7 +83,7 @@
     set_property(TARGET flann_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
 
     if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
-        add_library(flann SHARED "")
+        add_library(flann SHARED empty.cpp)
         set_target_properties(flann PROPERTIES LINKER_LANGUAGE CXX)
         target_link_libraries(flann -Wl,-whole-archive flann_s -Wl,-no-whole-archive)
     else()
