diff -ru CheMPS2-1.8.9.orig/CMakeLists.txt CheMPS2-1.8.9/CMakeLists.txt
--- CheMPS2-1.8.9.orig/CMakeLists.txt	2021-03-23 13:43:37.275230128 +0000
+++ CheMPS2-1.8.9/CMakeLists.txt	2021-03-23 13:44:19.387436761 +0000
@@ -71,9 +71,9 @@
     check_cxx_compiler_flag (-xHost HAS_XHOST)
     check_cxx_compiler_flag (-march=native HAS_MARCH_NATIVE)
     if (HAS_XHOST)
-        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -xHost")
+        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
     elseif (HAS_MARCH_NATIVE)
-        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+        set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
     endif()
 endif()
 
