FastTree binary is not lowercase.

Åke Sandgren, 2023-11-13
diff -ru BiG-SCAPE-1.1.5.orig/bigscape.py BiG-SCAPE-1.1.5/bigscape.py
--- BiG-SCAPE-1.1.5.orig/bigscape.py	2022-11-14 17:04:47.000000000 +0100
+++ BiG-SCAPE-1.1.5/bigscape.py	2023-11-13 07:52:23.064042200 +0100
@@ -1644,7 +1644,7 @@
             # make tree
             newick_file_path = os.path.join(gcf_trees_path, "GCF_c{:4.2f}_{:05d}.newick".format(cutoff,exemplar_idx))
             with open(newick_file_path, "w") as newick_file:
-                command = ["fasttree", "-nopr", "-quiet", alignment_file_path]
+                command = ["FastTree", "-nopr", "-quiet", alignment_file_path]
                 p = subprocess.Popen(command, stdout=newick_file, shell=False)
                 p.wait() # only with process has terminated will the file be ready
 
