diff -ru gate_v6.2.orig/source/digits_hits/include/GateMaps.hh gate_v6.2/source/digits_hits/include/GateMaps.hh
--- gate_v6.2.orig/source/digits_hits/include/GateMaps.hh	2012-08-08 13:38:18.000000000 +0200
+++ gate_v6.2/source/digits_hits/include/GateMaps.hh	2013-03-28 10:50:28.225482716 +0100
@@ -117,7 +117,7 @@
   for (size_t i=0; i<n ; i++){
     GateMap<Tkey,Tvalue>* mapElement = mapArray[i];
     for (iterator iter = mapElement->begin(); iter != mapElement->end(); iter++)
-      insert(*iter);
+      this->insert(*iter);
   }
 }
 
--- gate_v6.2/cluster_tools/jobsplitter/include/GateMacfileParser.hh.orig	2014-12-02 11:50:23.586463425 +0100
+++ gate_v6.2/cluster_tools/jobsplitter/include/GateMacfileParser.hh	2014-12-02 11:51:45.498508680 +0100
@@ -19,6 +19,7 @@
 #include <sstream> 
 #include <fstream> 
 #include <math.h>
+#include <unistd.h>
 
 using namespace std;
 
