--- gate_v7.0/cluster_tools/filemerger/Makefile.orig	2014-05-12 11:18:16.000000000 +0200
+++ gate_v7.0/cluster_tools/filemerger/Makefile	2015-03-04 10:54:55.461301152 +0100
@@ -16,6 +16,8 @@
 INCLUDE := -I./include `geant4-config --cflags` `root-config --cflags`
 LDFLAGS := `geant4-config --libs` `root-config --glibs`
 
+PREFIX := /usr/local
+
 TARGET := gjm
 
 .PHONY: all clean directories cleanall install uninstall
@@ -49,8 +51,9 @@
 
 install:
 	@echo Installing ...
-	@$(CP) $(TARGET) /usr/local/bin
+	mkdir -p $(PREFIX)/bin
+	@$(CP) $(TARGET) $(PREFIX)/bin
 
 uninstall:
 	@echo Uninstalling...
-	@$(RM) /usr/local/bin/$(TARGET)
+	@$(RM) $(PREFIX)/bin/$(TARGET)
--- gate_v7.0/cluster_tools/jobsplitter/Makefile.orig	2014-05-12 11:18:16.000000000 +0200
+++ gate_v7.0/cluster_tools/jobsplitter/Makefile	2015-03-04 10:54:55.461301152 +0100
@@ -16,6 +16,8 @@
 INCLUDE := -I./include `geant4-config --cflags`
 LDFLAGS := `geant4-config --libs`
 
+PREFIX := /usr/local
+
 TARGET := gjs
 
 .PHONY: all clean directories cleanall install uninstall
@@ -57,8 +59,9 @@
 
 install:
 	@echo Installing ...
-	@$(CP) $(TARGET) /usr/local/bin
+	mkdir -p $(PREFIX)/bin
+	@$(CP) $(TARGET) $(PREFIX)/bin
 
 uninstall:
 	@echo Uninstalling...
-	@$(RM) /usr/local/bin/$(TARGET)
+	@$(RM) $(PREFIX)/bin/$(TARGET)
