john dey - GCC 10.2 fix extern referances

diff -ruN tagdust-2.33.orig/src/nuc_code.c tagdust-2.33/src/nuc_code.c
--- tagdust-2.33.orig/src/nuc_code.c	2015-02-01 21:24:57.000000000 -0800
+++ tagdust-2.33/src/nuc_code.c	2021-04-27 17:55:29.444774000 -0700
@@ -36,9 +36,11 @@
 #include <stdlib.h>
 
 #include "tagdust2.h"
-#include "nuc_code.h"
 
 
+unsigned int nuc_code[256];
+unsigned int rev_nuc_code[5];
+
 /** \fn void init_nuc_code()
  \brief Initializes nucleotide conversion arrays. 
  
diff -ruN tagdust-2.33.orig/src/nuc_code.h tagdust-2.33/src/nuc_code.h
--- tagdust-2.33.orig/src/nuc_code.h	2014-12-14 17:26:55.000000000 -0800
+++ tagdust-2.33/src/nuc_code.h	2021-04-27 17:40:49.349731000 -0700
@@ -32,14 +32,14 @@
  *
  *
  */
-unsigned int nuc_code[256];
+extern unsigned int nuc_code[256];
 
 /**
  * @brief Converts 0-5 nucleotides into printable ASCII to.
  *
  *
  */
-unsigned int rev_nuc_code[5];
+extern unsigned int rev_nuc_code[5];
 
 int init_nuc_code(void);
 
