diff --git a/CPP/Windows/FileDir.cpp b/CPP/Windows/FileDir.cpp
index c27e9a1..de69e68 100644
--- a/CPP/Windows/FileDir.cpp
+++ b/CPP/Windows/FileDir.cpp
@@ -859,10 +859,11 @@ bool CDelayedSymLink::Create()
 {
   struct stat st;
 
-  if (_dev == 0) {
-    errno = EPERM;
-    return false;
-  }
+// this test incorrectly prevents the proper handling of symbolic links  on CephFS because _dev == 0
+//  if (_dev == 0) {
+//    errno = EPERM;
+//    return false;
+//  }
   if (lstat(_source, &st) != 0)
     return false;
   if (_dev != st.st_dev || _ino != st.st_ino) {
