removes the not-set include-directory for NumPy
Author: J. Sassmannshausen (Imperial College London/UK)
diff --git a/blosc2-2.0.0.orig/blosc2/CMakeLists.txt b/blosc2-2.0.0/blosc2/CMakeLists.txt
index 012bb1e..c4aabd3 100644
--- a/blosc2-2.0.0.orig/blosc2/CMakeLists.txt
+++ b/blosc2-2.0.0/blosc2/CMakeLists.txt
@@ -11,8 +11,6 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(BLOSC_INSTALL ON)
 add_subdirectory(c-blosc2)
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/c-blosc2/include")
-include_directories("${Python_NumPy_INCLUDE_DIRS}")
-
 
 add_cython_target(blosc2_ext blosc2_ext.pyx)
 add_library(blosc2_ext MODULE ${blosc2_ext})
