change copy by copyfile to avoid permission issues
author: Danilo Gonzalez (DoitNow)
--- setup.py.orig	2025-05-29 16:30:25.438309725 +0200
+++ setup.py	2025-05-29 16:31:20.932235168 +0200
@@ -962,7 +962,7 @@
                 )
                 platform_system = platform.system()
                 if platform_system == "Linux":
-                    shutil.copy(libdir / "libblosc2.so", ROOT / "tables")
+                    shutil.copyfile(libdir / "libblosc2.so", ROOT / "tables" / "libblosc2.so")
                     copy_libs += ["libblosc2.so"]
 
                     dll_dir = "/tmp/hdf5/lib"
