diff -ru faiss-1.6.2.orig/python/Makefile faiss-1.6.2/python/Makefile
--- faiss-1.6.2.orig/python/Makefile	2020-03-19 19:18:36.148053891 +0000
+++ faiss-1.6.2/python/Makefile	2020-03-19 19:22:55.694473227 +0000
@@ -30,7 +30,7 @@
 	$(PYTHON) setup.py build
 
 install: build
-	$(PYTHON) setup.py install
+	$(PYTHON) setup.py install --prefix=$(INSTALL_PREFIX)
 
 clean:
 	rm -f swigfaiss*.cpp swigfaiss*.o swigfaiss*.py _swigfaiss*.so
