Add standard headers that would otherwise be implicitly called.
Author: Davide Vanzo (Vanderbilt University)
diff -ru gdchart0.11.5dev.orig/gdc.h gdchart0.11.5dev/gdc.h
--- gdchart0.11.5dev.orig/gdc.h	2019-08-06 09:57:15.703746776 -0500
+++ gdchart0.11.5dev/gdc.h	2019-08-06 11:43:04.915922463 -0500
@@ -23,6 +23,8 @@
 
 #include <limits.h>
 #include <float.h>
+#include <stdlib.h>
+#include <string.h>
 #ifdef GDC_INCL
 #include "gd.h"
 #include "gdfonts.h"
