# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2023/09
diff -ru PyTom-1.1/pytom/__init__.py PyTom-1.1_fix_version/pytom/__init__.py
--- PyTom-1.1/pytom/__init__.py	2023-08-29 17:09:10.000000000 +0200
+++ PyTom-1.1_fix_version/pytom/__init__.py	2023-09-01 12:16:30.901056016 +0200
@@ -3,7 +3,7 @@
 __all__ = ["alignment", "angles", "basic", "bin", "classification", "gui", "gpu", "localization", "parallel", "pytomc",
            "plotting", "reconstruction", "simulation", 'agnostic', "tools", 'voltools']
 
-__version__ = "1.0"
+__version__ = "1.1"
 
 try:
     import pytom.lib.pytom_volume as pytom_volume
