#Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/03
diff -ru ori/pytomc/installFunctions.py fix_pytomGUI_wrapper.patch/pytomc/installFunctions.py
--- ori/pytomc/installFunctions.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_pytomGUI_wrapper.patch/pytomc/installFunctions.py	2021-03-11 18:16:58.220680257 +0100
@@ -217,7 +217,7 @@
 
 def generatePyTomGuiScript(pytomDirectory, python_version):
     pytomguiCommand = '# !/bin/bash\n'
-    pytomguiCommand += f'python{python_version} {pytomDirectory}/gui/pytomGUI.py $1\n'
+    pytomguiCommand += f'python $EBROOTPYTOM/lib/python{python_version}/site-packages/pytom/gui/pytomGUI.py $1\n'
 
     f = open(pytomDirectory + os.sep + 'bin' + os.sep + 'pytomGUI.py', 'w')
     f.write(pytomguiCommand)
