#Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/03
diff -ru ori/gui/pytomGUI.py fix_pytomGUI_motioncor2IMOD.patch/gui/pytomGUI.py
--- ori/gui/pytomGUI.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_pytomGUI_motioncor2IMOD.patch/gui/pytomGUI.py	2021-03-11 18:08:37.491900369 +0100
@@ -630,7 +630,7 @@
 if __name__ == '__main__':
 
 
-    for fname,module in [( 'motioncor2','motioncor2/1.2.1' ),('header','imod/4.10.25')]:
+    for fname,module in [( 'MotionCor2','MotionCor2/1.4.2' ),('header','IMOD/4.11.0')]:
         if 1:
             result = os.popen('which {}'.format(fname)).read()[:-1]
             if not result:
