#Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/03
diff -ru ori/classification/calculate_correlation_matrix.py fix_shebang.patch/classification/calculate_correlation_matrix.py
--- ori/classification/calculate_correlation_matrix.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/classification/calculate_correlation_matrix.py	2021-03-12 12:58:03.281982942 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 '''
 Created on May 7, 2012
 
diff -ru ori/frm/FRMAlignment.py fix_shebang.patch/frm/FRMAlignment.py
--- ori/frm/FRMAlignment.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/frm/FRMAlignment.py	2021-03-12 12:58:03.294983117 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 '''
 FRM alignment classes
 Created on Mar 5, 2012
diff -ru ori/frm/GFRMAlignment.py fix_shebang.patch/frm/GFRMAlignment.py
--- ori/frm/GFRMAlignment.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/frm/GFRMAlignment.py	2021-03-12 12:58:03.307983293 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 '''
 Created on Sep 3, 2012
 
diff -ru ori/frm/GWienerFilterAlignment.py fix_shebang.patch/frm/GWienerFilterAlignment.py
--- ori/frm/GWienerFilterAlignment.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/frm/GWienerFilterAlignment.py	2021-03-12 12:58:03.319983454 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 '''
 Gold standard Wiener Filter alignment
 Created on Nov 6, 2012
diff -ru ori/frm/WienerFilterAlignment.py fix_shebang.patch/frm/WienerFilterAlignment.py
--- ori/frm/WienerFilterAlignment.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/frm/WienerFilterAlignment.py	2021-03-12 12:58:03.381984291 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 '''
 Created on Apr 11, 2012
 
diff -ru ori/gui/additional/multi_tilt_alignment.py fix_shebang.patch/gui/additional/multi_tilt_alignment.py
--- ori/gui/additional/multi_tilt_alignment.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/gui/additional/multi_tilt_alignment.py	2021-03-12 12:58:03.332983630 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 # from multiprocessing import Process
 # import os
 # import sys
diff -ru ori/simulation/simulateProjections.py fix_shebang.patch/simulation/simulateProjections.py
--- ori/simulation/simulateProjections.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/simulation/simulateProjections.py	2021-03-12 12:58:03.368984115 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 """
 Gijs' tilt series simulation - used for Ilja's contest paper
 """
diff -ru ori/tools/plotAngleDistribution.py fix_shebang.patch/tools/plotAngleDistribution.py
--- ori/tools/plotAngleDistribution.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/tools/plotAngleDistribution.py	2021-03-12 12:58:03.343983778 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 '''
 Created on Oct 2, 2012
 
diff -ru ori/tools/script_helper.py fix_shebang.patch/tools/script_helper.py
--- ori/tools/script_helper.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_shebang.patch/tools/script_helper.py	2021-03-12 12:58:03.356983954 +0100
@@ -1,3 +1,4 @@
+#!/bin/env pytom
 '''
 Created on Jun 29, 2011
 
