diff -ru PyTom-1.1/pytom/pytomc/installFunctions.py PyTom-1.1_fix_wrappers/pytom/pytomc/installFunctions.py
--- PyTom-1.1/pytom/pytomc/installFunctions.py	2023-08-29 17:09:10.000000000 +0200
+++ PyTom-1.1_fix_wrappers/pytom/pytomc/installFunctions.py	2023-09-01 12:43:58.533438703 +0200
@@ -211,7 +211,7 @@
 def generatePyTomScript(pytomDirectory,python_version):
     
     pytomCommand = f"""#!/usr/bin/env bash
-cat {sys.prefix}/pytom_data/LICENSE.txt
+cat $EBROOTPYTOM/pytom_data/LICENSE.txt
 
 FID=0
 export PYTOM_GPU=-1
@@ -229,7 +229,7 @@
     fi
 done
 
-{sys.executable} -O $*
+python -O $*
 """
 
     f = open(pytomDirectory + os.sep + 'bin' + os.sep + 'pytom','w')
@@ -240,7 +240,7 @@
 
 def generatePyTomGuiScript(pytomDirectory, python_version):
     pytomguiCommand = '# !/bin/bash\n'
-    pytomguiCommand += f'{sys.executable} {pytomDirectory}/gui/pytomGUI.py $1\n'
+    pytomguiCommand += f'python {pytomDirectory}/gui/pytomGUI.py $1\n'
 
     f = open(pytomDirectory + os.sep + 'bin' + os.sep + 'pytomGUI', 'w')
     f.write(pytomguiCommand)
@@ -251,7 +251,7 @@
 def generateIPyTomScript(pytomDirectory):
 
     ipytomCommand = '#!/usr/bin/env bash\n'
-    ipytomCommand += f'cat {sys.prefix}/pytom_data/LICENSE.txt\n'
+    ipytomCommand += f'cat $EBROOTPYTOM/pytom_data/LICENSE.txt\n'
     ipytomCommand += 'ipython $* -i\n'
     
     f = open(pytomDirectory + os.sep + 'bin' + os.sep + 'ipytom','w')
