#Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/03
diff -ru ori/__init__.py fix_version.patch/__init__.py
--- ori/__init__.py	2021-01-18 22:40:34.000000000 +0100
+++ fix_version.patch/__init__.py	2021-03-11 17:47:06.638424896 +0100
@@ -2,7 +2,7 @@
 
 __all__ = ["alignment","angles","basic","classification","cluster","frm","frontend","image2D","localization","parallel",
            "plotting","reconstruction","score","simulation",'tompy', "tools", "unittests", "visualization", 'voltools']
-__version__ = "0.991"
+__version__ = "0.993"
 
 import pytom_volume
 import pytom_numpy
diff -ru ori/LICENSE.txt fix_version.patch/LICENSE.txt
--- ori/LICENSE.txt	2021-01-18 22:40:34.000000000 +0100
+++ fix_version.patch/LICENSE.txt	2021-03-11 17:46:55.756277571 +0100
@@ -1,4 +1,4 @@
-PyTom v0.991
+PyTom v0.993
 
 This license affects the software package PyTom and all the herein distributed source / data files.
 Authors: 
