#Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/03
diff -ru PyTom-0.994/__init__.py PyTom-0.994_version.patch/__init__.py
--- PyTom-0.994/__init__.py	2021-03-19 13:07:28.000000000 +0100
+++ PyTom-0.994_version.patch/__init__.py	2021-03-25 15:25:59.671492503 +0100
@@ -2,7 +2,7 @@
 
 __all__ = ["alignment","angles","basic","classification","cluster","frm","frontend","image2D","localization","parallel",
            "plotting","reconstruction","score","simulation",'tompy', "tools", "unittests", "visualization", 'voltools']
-__version__ = "0.993"
+__version__ = "0.994"
 
 import pytom_volume
 import pytom_numpy
diff -ru PyTom-0.994/LICENSE.txt PyTom-0.994_version.patch/LICENSE.txt
--- PyTom-0.994/LICENSE.txt	2021-03-19 13:07:28.000000000 +0100
+++ PyTom-0.994_version.patch/LICENSE.txt	2021-03-26 12:36:59.310465784 +0100
@@ -1,4 +1,4 @@
-PyTom v0.993
+PyTom v0.994
 
 This license affects the software package PyTom and all the herein distributed source / data files.
 Authors: 
@@ -9,7 +9,7 @@
 
 Copyright (c) 2021
 Utrecht University
-http://www.pytom.org
+https://pytom.sites.uu.nl/
 
 This program is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
