Description: Remove failing loadsave tests
Bug: https://github.com/cburstedde/p4est/issues/32
Author: Matthias Maier <tamiko+DEBIAN@kyomu.43-1.org>
Last-Update: 2016-11-05

diff --git a/test/Makefile.am b/test/Makefile.am
index afcb9ba..803655d 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -10,7 +10,7 @@ p4est_test_programs += \
         test/p4est_test_quadrants test/p4est_test_balance \
         test/p4est_test_partition test/p4est_test_coarsen \
         test/p4est_test_valid test/p4est_test_balance_type \
-        test/p4est_test_loadsave test/p4est_test_order \
+        test/p4est_test_order \
         test/p4est_test_load test/p4est_test_ghost \
         test/p4est_test_conn_transformation \
         test/p4est_test_iterate test/p4est_test_lnodes \
@@ -33,7 +33,7 @@ p4est_test_programs += \
         test/p8est_test_partition test/p8est_test_coarsen \
         test/p8est_test_valid test/p8est_test_balance_type \
         test/p8est_test_face_transform test/p8est_test_edge_face_corners \
-        test/p8est_test_periodic test/p8est_test_loadsave \
+        test/p8est_test_periodic \
         test/p8est_test_load test/p8est_test_ghost \
         test/p8est_test_conn_transformation \
         test/p8est_test_iterate test/p8est_test_lnodes \
@@ -74,7 +74,6 @@ test_p4est_test_order_SOURCES = test/test_order.c
 test_p4est_test_coarsen_SOURCES = test/test_coarsen2.c
 test_p4est_test_valid_SOURCES = test/test_valid2.c
 test_p4est_test_balance_type_SOURCES = test/test_balance_type2.c
-test_p4est_test_loadsave_SOURCES = test/test_loadsave2.c
 test_p4est_test_load_SOURCES = test/test_load2.c
 test_p4est_test_ghost_SOURCES = test/test_ghost2.c
 test_p4est_test_conn_transformation_SOURCES = test/test_conn_transformation2.c
@@ -111,7 +110,6 @@ test_p8est_test_balance_type_SOURCES = test/test_balance_type3.c
 test_p8est_test_face_transform_SOURCES = test/test_face_transform3.c
 test_p8est_test_edge_face_corners_SOURCES = test/test_edge_face_corners3.c
 test_p8est_test_periodic_SOURCES = test/test_periodic3.c
-test_p8est_test_loadsave_SOURCES = test/test_loadsave3.c
 test_p8est_test_load_SOURCES = test/test_load3.c
 test_p8est_test_ghost_SOURCES = test/test_ghost3.c
 test_p8est_test_conn_transformation_SOURCES = test/test_conn_transformation3.c
@@ -157,7 +155,6 @@ LINT_CSOURCES += \
         $(test_p4est_test_coarsen_SOURCES) \
         $(test_p4est_test_valid_SOURCES) \
         $(test_p4est_test_balance_type_SOURCES) \
-        $(test_p4est_test_loadsave_SOURCES) \
         $(test_p4est_test_load_SOURCES) \
         $(test_p4est_test_ghost_SOURCES) \
         $(test_p4est_test_conn_transformation_SOURCES) \
@@ -185,7 +182,6 @@ LINT_CSOURCES += \
         $(test_p8est_test_face_transform_SOURCES) \
         $(test_p8est_test_edge_face_corners_SOURCES) \
         $(test_p8est_test_periodic_SOURCES) \
-        $(test_p8est_test_loadsave_SOURCES) \
         $(test_p8est_test_load_SOURCES) \
         $(test_p8est_test_ghost_SOURCES) \
         $(test_p8est_test_conn_transformation_SOURCES) \
