# Correct pkgconfig file
# S.D.Pinches, 2022.08.23
diff -Nru blitz-1.0.2-orig/cmake/CreatePkgConfig.cmake blitz-1.0.2/cmake/CreatePkgConfig.cmake
--- blitz-1.0.2-orig/cmake/CreatePkgConfig.cmake	2019-10-02 00:50:17.000000000 +0200
+++ blitz-1.0.2/cmake/CreatePkgConfig.cmake	2022-08-23 16:24:41.509703369 +0200
@@ -9,7 +9,7 @@
 exec_prefix=\${prefix}
 libdir=\${prefix}/@CMAKE_INSTALL_LIBDIR@
 includedir=\${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
-confdir=\${prefix}/@CMAKE_INSTALL_LIBDIR@/@pkgconfig_package_name@
+confdir=\${prefix}/lib/cmake
 
 Name: @pkgconfig_package_name@
 Description: @pkgconfig_package_description@
