Remove hardcoded path to email.txt
author: Alex Domingo (Vrije Universiteit Brussel)
--- alpha/lib/email.py	2019-05-21 18:07:12.000000000 +0200
+++ alpha/lib/email.py	2019-06-14 11:35:04.129371000 +0200
@@ -1,10 +1,12 @@
 import os
 THIS_FILE_DIR = os.path.dirname(os.path.realpath(__file__))
+user_home_dir = os.path.expanduser("~")
+user_alpha_dir = os.path.join( user_home_dir, ".alpha")
 
 import sys
 
 
-def getEmail(filename=THIS_FILE_DIR+'/email.txt'):
+def getEmail(filename=user_alpha_dir+'/email.txt'):
   """
   Read an email address from the file.
   """
@@ -18,4 +20,4 @@
           else:
             sys.exit('ERROR: email "{}" format unrecognized.'.format(email))
 
-  sys.exit('ERROR: please put your email address in "lib/email.txt".')
+  sys.exit('ERROR: please put your email address in: ' + user_alpha_dir + '/email.txt')
