diff -ru scipion-1.2.orig/SConstruct scipion-1.2/SConstruct
--- scipion-1.2.orig/SConstruct	2018-04-29 10:12:21.000000000 +0200
+++ scipion-1.2/SConstruct	2018-04-29 10:13:36.000000000 +0200
@@ -464,7 +464,7 @@
     cxxflagsCopy = cxxflags + env['CXXFLAGS']
     linkflagsCopy = linkflags + env['LINKFLAGS']
     ldLibraryPathCopy = [env['LIBPATH']]
-    appendUnique(libPathsCopy, env.get('LIBPATH', ''))
+    appendUnique(libPathsCopy, env.get('LIBPATH', '').split(os.pathsep))
     env2 = Environment(ENV=os.environ)
 #    if mpi: 
 #        appendUnique(incsCopy, env['MPI_INCLUDE'])
