diff -ru scipion-1.2.orig/SConstruct scipion-1.2/SConstruct
--- scipion-1.2.orig/SConstruct	2018-04-29 18:03:03.000000000 +0200
+++ scipion-1.2/SConstruct	2018-04-29 18:04:16.000000000 +0200
@@ -412,7 +412,7 @@
         return ''
     installDir = installDir or 'java/lib'
     classPath = ":".join(glob(join(Dir(installDir).abspath,'*.jar')))
-    cmd = '%s -cp %s org.junit.runner.JUnitCore xmipp.test.%s' % (join(env['JAVA_BINDIR'], 'java'), classPath, name)
+    cmd = '%s -cp %s org.junit.runner.JUnitCore xmipp.test.%s' % ('java', classPath, name)
     runTest = env.Command(name, join(installDir, source), cmd)
     env.Alias('run_java_tests', runTest)
     env.Default(runTest)
@@ -596,7 +596,7 @@
 # Java related environment variables, probably the main one
 # that need to be modified is JAVA_HOME
 env['JAVA_HOME'] = os.environ['JAVA_HOME']
-env['JAVA_BINDIR'] = os.environ['JAVA_BINDIR']
+#env['JAVA_BINDIR'] = os.environ['JAVA_BINDIR']
 env['JAVAC'] = os.environ.get('JAVAC')
 env['JAR'] = os.environ.get('JAR')
 env['JNI_CPPPATH'] = os.environ.get('JNI_CPPPATH').split(':')
