diff -ru scipion-1.2.orig/pyworkflow/em/packages/eman2/eman2.py scipion-1.2.patched/pyworkflow/em/packages/eman2/eman2.py
--- scipion-1.2.orig/pyworkflow/em/packages/eman2/eman2.py	2018-04-03 13:57:35.000000000 +0200
+++ scipion-1.2.patched/pyworkflow/em/packages/eman2/eman2.py	2018-05-04 11:17:25.000000000 +0200
@@ -45,10 +45,7 @@
         pathList.append(os.environ['REMOTE_MESA_LIB'])
 
     environ.update({
-            'PATH': join(EMAN2DIR, 'bin'),
-            'LD_LIBRARY_PATH': os.pathsep.join(pathList),
-            'PYTHONPATH': os.pathsep.join(pathList),
-            'EMAN_PYTHON': os.path.join(EMAN2DIR, 'Python/bin/python')
+            'EMAN_PYTHON': 'python',
             }, position=Environ.REPLACE)
     return environ
 
@@ -77,7 +74,7 @@
 
 def getEmanProgram(program):
     if not 'EMAN_PYTHON' in os.environ:
-        os.environ['EMAN_PYTHON'] = os.path.join(os.environ['EMAN2DIR'], 'Python/bin/python')
+        os.environ['EMAN_PYTHON'] = 'python'
     # For EMAN2 python scripts, join the path to bin
     # Comment this out cause there are eman programs that do not start with e2
     #if program.startswith('e2'):
