diff -ru scipion-1.2.orig/pyworkflow/em/packages/motioncorr/convert.py scipion-1.2/pyworkflow/em/packages/motioncorr/convert.py
--- scipion-1.2.orig/pyworkflow/em/packages/motioncorr/convert.py	2018-04-03 13:57:35.000000000 +0200
+++ scipion-1.2/pyworkflow/em/packages/motioncorr/convert.py	2018-05-04 20:09:52.000000000 +0200
@@ -95,7 +95,7 @@
     if var == 'MOTIONCORR':
         return ['2.1']
     elif var == 'MOTIONCOR2':
-        return ['01302017', '1.0.0', '1.0.2', '1.0.4', '1.0.5']
+        return ['01302017', '1.0.0', '1.0.2', '1.0.4', '1.0.5', '1.1.0']
 
 
 def parseMovieAlignment(logFile):
