diff -ru scipion-1.2.orig/SConstruct scipion-1.2/SConstruct
--- scipion-1.2.orig/SConstruct	2018-04-29 10:14:33.000000000 +0200
+++ scipion-1.2/SConstruct	2018-04-29 17:46:55.000000000 +0200
@@ -578,19 +578,19 @@
 env['LINKFLAGS'] = os.environ.get('LINKFLAGS', '').split()
 
 
-for path in ['MPI_LIBDIR', 'MPI_INCLUDE', 'MPI_BINDIR',
-             'JAVA_HOME', 'JAVA_BINDIR']:
-    if not os.path.isdir(os.environ.get(path, '')):
-        Exit('Path to $%s (%s) should exist, but it does not. Stopping.\n'
-             'Please run "scipion config"' % (path, os.environ.get(path, '')))
+#for path in ['MPI_LIBDIR', 'MPI_INCLUDE', 'MPI_BINDIR',
+#             'JAVA_HOME', 'JAVA_BINDIR']:
+#    if not os.path.isdir(os.environ.get(path, '')):
+#        Exit('Path to $%s (%s) should exist, but it does not. Stopping.\n'
+#             'Please run "scipion config"' % (path, os.environ.get(path, '')))
 
 env['MPI_CC'] = os.environ.get('MPI_CC')
 env['MPI_CXX'] = os.environ.get('MPI_CXX')
 env['MPI_LINKERFORPROGRAMS'] = os.environ.get('MPI_LINKERFORPROGRAMS')
-env['MPI_LIB'] = os.environ.get('MPI_LIB')
-env['MPI_LIBDIR'] = os.environ['MPI_LIBDIR']
-env['MPI_INCLUDE'] = os.environ['MPI_INCLUDE']
-env['MPI_BINDIR'] = os.environ['MPI_BINDIR']
+#env['MPI_LIB'] = os.environ.get('MPI_LIB')
+#env['MPI_LIBDIR'] = os.environ['MPI_LIBDIR']
+#env['MPI_INCLUDE'] = os.environ['MPI_INCLUDE']
+#env['MPI_BINDIR'] = os.environ['MPI_BINDIR']
 env['MATLAB_DIR'] = os.environ.get('MATLAB_DIR')
 
 # Java related environment variables, probably the main one
