diff -ru scipion-1.2.orig/SConstruct scipion-1.2/SConstruct
--- scipion-1.2.orig/SConstruct	2018-04-29 10:07:48.000000000 +0200
+++ scipion-1.2/SConstruct	2018-04-29 10:10:33.000000000 +0200
@@ -229,9 +229,9 @@
 
     mpiArgs = {}
     if mpi:
-        _libpath.append(env['MPI_LIBDIR'])
-        _libs.append(env['MPI_LIB']) 
-        _incs.append(env['MPI_INCLUDE'])
+#        _libpath.append(env['MPI_LIBDIR'])
+#        _libs.append(env['MPI_LIB']) 
+#        _incs.append(env['MPI_INCLUDE'])
                
         mpiArgs = {'CC': env['MPI_CC'],
                    'CXX': env['MPI_CXX'],
@@ -243,7 +243,7 @@
 #             print >> sys.stderr, 'ERROR: MPI is not properly working. Exiting...'
 #             Exit(1)
 #         env = conf.Finish()
-        env2.PrependENVPath('PATH', env['MPI_BINDIR'])
+#        env2.PrependENVPath('PATH', env['MPI_BINDIR'])
     
 
     _incs.append(env['CPPPATH'])
@@ -466,11 +466,11 @@
     ldLibraryPathCopy = [env['LIBPATH']]
     appendUnique(libPathsCopy, env.get('LIBPATH', ''))
     env2 = Environment(ENV=os.environ)
-    if mpi: 
-        appendUnique(incsCopy, env['MPI_INCLUDE'])
-        appendUnique(libPathsCopy, env['MPI_LIBDIR'])
-        appendUnique(libsCopy, env['MPI_LIB'])
-        appendUnique(ldLibraryPathCopy, env['MPI_LIBDIR'])
+#    if mpi: 
+#        appendUnique(incsCopy, env['MPI_INCLUDE'])
+#        appendUnique(libPathsCopy, env['MPI_LIBDIR'])
+#        appendUnique(libsCopy, env['MPI_LIB'])
+#        appendUnique(ldLibraryPathCopy, env['MPI_LIBDIR'])
     env2['ENV']['LD_LIBRARY_PATH'] = env['ENV'].get('LD_LIBRARY_PATH', '')
     env2['ENV']['PATH'] = env['ENV']['PATH']
 
