diff -ru scipion-1.2.orig/pyworkflow/em/packages/eman2/eman2.py scipion-1.2/pyworkflow/em/packages/eman2/eman2.py
--- scipion-1.2.orig/pyworkflow/em/packages/eman2/eman2.py	2018-04-03 13:57:35.000000000 +0200
+++ scipion-1.2/pyworkflow/em/packages/eman2/eman2.py	2018-05-04 09:07:59.000000000 +0200
@@ -62,7 +62,7 @@
 
 
 def getSupportedVersions():
-    return ['2.11', '2.12']
+    return ['2.11', '2.12', '2.2', '2.21']
 
 
 def validateVersion(protocol, errors):
