diff -ru scipion-2.0.0.orig/pyworkflow/apps/pw_config.py scipion-2.0.0/pyworkflow/apps/pw_config.py
--- scipion-2.0.0.orig/pyworkflow/apps/pw_config.py	2019-05-07 10:40:13.000000000 +0200
+++ scipion-2.0.0/pyworkflow/apps/pw_config.py	2019-10-09 10:52:37.686510064 +0200
@@ -131,7 +131,9 @@
     """ Check if protocol statistics should be collected. """
     if notify:
         config.set('VARIABLES', 'SCIPION_NOTIFY', 'True')
-        return
+    else:
+        config.set('VARIABLES', 'SCIPION_NOTIFY', 'False')
+    return
     notifyOn = config.get('VARIABLES', 'SCIPION_NOTIFY')
     if notifyOn=='False':
         # This works for  Python 2.x. and Python 3
