diff -ru scipion-1.2.orig/SConstruct scipion-1.2/SConstruct
--- scipion-1.2.orig/SConstruct	2018-04-03 13:57:35.000000000 +0200
+++ scipion-1.2/SConstruct	2018-04-29 10:00:04.000000000 +0200
@@ -224,7 +224,7 @@
         Exit('No sources found for Library: %s. Exiting!!!' % name)
 
     # FIXME: There must be a key in env dictionary that breaks the compilation. Please find it to make it more beautiful
-    env2 = Environment()
+    env2 = Environment(ENV=os.environ)
     env2['ENV']['PATH'] = env['ENV']['PATH']
 
     mpiArgs = {}
@@ -465,7 +465,7 @@
     linkflagsCopy = linkflags + env['LINKFLAGS']
     ldLibraryPathCopy = [env['LIBPATH']]
     appendUnique(libPathsCopy, env.get('LIBPATH', ''))
-    env2 = Environment()
+    env2 = Environment(ENV=os.environ)
     if mpi: 
         appendUnique(incsCopy, env['MPI_INCLUDE'])
         appendUnique(libPathsCopy, env['MPI_LIBDIR'])
@@ -532,7 +532,7 @@
 
 def libraryTest(env, name, lang='c'):
     """Check the existence of a concrete C/C++ library."""
-    env2 = Environment(LIBS=env.get('LIBS',''))
+    env2 = Environment(ENV=os.environ,LIBS=env.get('LIBS',''))
     conf = Configure(env2)
     conf.CheckLib(name, language=lang)
     env2 = conf.Finish()
