diff -ru scipion-1.2.orig/SConstruct scipion-1.2.ake2/SConstruct
--- scipion-1.2.orig/SConstruct	2018-04-03 13:57:35.000000000 +0200
+++ scipion-1.2.ake2/SConstruct	2018-04-27 20:11:51.000000000 +0200
@@ -568,13 +568,13 @@
 env['LINKERFORPROGRAMS'] = os.environ.get('LINKERFORPROGRAMS')
 env['CCFLAGS'] = os.environ.get('CCFLAGS', '').split()
 cxxFlags = os.environ.get('CXXFLAGS', '') 
-if os.environ.get('DEBUG', '0') == 'True': #FIXME, use 1, true, yes...
-   cxxFlags += ' -g'
-else:
-    if cxxFlags.find("-O")==-1:
-        cxxFlags += " -O3"
+#if os.environ.get('DEBUG', '0') == 'True': #FIXME, use 1, true, yes...
+#   cxxFlags += ' -g'
+#else:
+#    if cxxFlags.find("-O")==-1:
+#        cxxFlags += " -O3"
 env['CXXFLAGS'] = cxxFlags.split()
-os.environ['CXXFLAGS'] = cxxFlags # FIXME use only env or os.environ in the rest of the code
+#os.environ['CXXFLAGS'] = cxxFlags # FIXME use only env or os.environ in the rest of the code
 env['LINKFLAGS'] = os.environ.get('LINKFLAGS', '').split()
 
 
