diff -ru scipion-1.2.orig/SConstruct scipion-1.2/SConstruct
--- scipion-1.2.orig/SConstruct	2018-04-29 18:03:03.000000000 +0200
+++ scipion-1.2/SConstruct	2018-05-02 12:49:45.000000000 +0200
@@ -216,6 +216,7 @@
     sources = []
 
     _libpath.append(Dir('#software/lib').abspath)
+    appendUnique(_libpath, env.get('LIBPATH', '').split(os.pathsep))
     
     for d, p in izip(dirs, patterns):
         sources += glob(join(env['PACKAGE']['SCONSCRIPT'], d, p))
