--- scipion/config/scipion.conf.orig	2016-03-14 15:07:30.000000000 +0100
+++ scipion/config/scipion.conf	2016-03-14 15:04:02.000000000 +0100
@@ -1,35 +1,35 @@
+[DIRS_GLOBAL]
+SCIPION_TESTS = data/tests
+SCIPION_SOFTWARE = software
+
 [REMOTE]
 SCIPION_URL = http://scipion.cnb.csic.es/downloads/scipion
-SCIPION_URL_TESTDATA = %(SCIPION_URL)s/data/tests
 SCIPION_URL_SOFTWARE = %(SCIPION_URL)s/software
-
-[DIRS_GLOBAL]
-SCIPION_SOFTWARE = software
-SCIPION_TESTS = data/tests
+SCIPION_URL_TESTDATA = %(SCIPION_URL)s/data/tests
 
 [BUILD]
+CC = gcc
+CXX = g++
 LINKERFORPROGRAMS = g++
+CCFLAGS = -std=c99
 CXXFLAGS = 
-GTEST = False
+LINKFLAGS = -L/software/software/CUDA/7.5.18/lib64
+MPI_CC = mpicc
+MPI_CXX = mpiCC
 MPI_LINKERFORPROGRAMS = mpiCC
 MPI_LIB = mpi
-MPI_BINDIR = /usr/lib64/openmpi/bin
-JNI_CPPPATH = %(JAVA_HOME)s/include:%(JAVA_HOME)s/include/linux
-OPENCV = True
-MPI_CC = mpicc
-CUDA = False
-JAVA_HOME = /usr/lib/jvm/java-1.6.0-openjdk-1.6.0.0.x86_64
-MATLAB = False
+MPI_LIBDIR = /software/software/OpenMPI/1.10.2-GCC-4.9.3-2.25/lib
+MPI_INCLUDE = /software/software/OpenMPI/1.10.2-GCC-4.9.3-2.25/include
+MPI_BINDIR = /software/software/OpenMPI/1.10.2-GCC-4.9.3-2.25/bin
+MATLAB_DIR = /software/software/MATLAB/2015b-GCC-4.8.4
+JAVA_HOME = /software/software/Java/1.8.0_74
 JAVA_BINDIR = %(JAVA_HOME)s/bin
-CC = gcc
+JAVAC = %(JAVA_BINDIR)s/javac
 JAR = %(JAVA_BINDIR)s/jar
-MPI_CXX = mpiCC
+JNI_CPPPATH = %(JAVA_HOME)s/include:%(JAVA_HOME)s/include/linux
+GTEST = False
+CUDA = True
 DEBUG = False
-MPI_LIBDIR = /usr/lib64/openmpi/lib
-JAVAC = %(JAVA_BINDIR)s/javac
-CXX = g++
-LINKFLAGS = 
-CCFLAGS = -std=c99
-MATLAB_DIR = /usr/local/MATLAB/R2011a
-MPI_INCLUDE = /usr/include/openmpi-x86_64
+MATLAB = True
+OPENCV = True
 
