diff -u pod5-file-format-0.1.8/c++/CMakeLists.txt.orig pod5-file-format-0.1.8/c++/CMakeLists.txt
--- pod5-file-format-0.1.8/c++/CMakeLists.txt.orig	2023-01-20 12:22:05.000000000 +0100
+++ pod5-file-format-0.1.8/c++/CMakeLists.txt	2023-01-23 10:11:12.097686513 +0100
@@ -21,6 +21,7 @@
 endif()
 
 find_package(Threads REQUIRED)
+find_package(pybind11 REQUIRED)
 
 find_program(
     FLATBUFFERS_FLATC_EXECUTABLE
diff -u pod5-file-format-0.1.8/CMakeLists.txt.orig pod5-file-format-0.1.8/CMakeLists.txt
--- pod5-file-format-0.1.8/CMakeLists.txt.orig	2023-01-20 12:22:05.000000000 +0100
+++ pod5-file-format-0.1.8/CMakeLists.txt	2023-01-23 10:34:44.860294168 +0100
@@ -92,7 +91,6 @@

 if (BUILD_PYTHON_WHEEL)
     find_package (Python ${PYTHON_VERSION} EXACT COMPONENTS Interpreter Development)
-    add_subdirectory(third_party/pybind11)
 endif()

 add_subdirectory(c++)
