# Ensure the EB pybind11 is used during build
#
diff -ur pod5-file-format-0.3.23/CMakeLists.txt pod5-file-format-0.3.23.new/CMakeLists.txt
--- pod5-file-format-0.3.23/CMakeLists.txt      2024-11-21 10:27:14.000000000 +0000
+++ pod5-file-format-0.3.23.new/CMakeLists.txt  2026-02-23 17:57:31.591833644 +0000
@@ -94,7 +94,7 @@
         set(PYTHON_VERSION "")
     endif()
     find_package (Python ${PYTHON_VERSION} EXACT COMPONENTS Interpreter Development)
-    add_subdirectory(third_party/pybind11)
+    find_package (pybind11 CONFIG REQUIRED)
 endif()

 add_subdirectory(c++)
