--- nwchem-6.1.1-src/src/GNUmakefile.orig	2012-12-10 17:23:43.236474825 +0100
+++ nwchem-6.1.1-src/src/GNUmakefile	2012-12-10 17:23:55.916623423 +0100
@@ -7,7 +7,7 @@
 
   SUBDIRS =  $(NWSUBDIRS)
 
-  LIB_DEFINES =  -DCOMPILATION_DATE="'`date +%a_%b_%d_%H:%M:%S_%Y`'" \
+  LIB_DEFINES =  -DCOMPILATION_DATE="'`date`'" \
                  -DCOMPILATION_DIR="'$(TOPDIR)'" \
                  -DNWCHEM_BRANCH="'$(CODE_BRANCH)'"
 ifeq ($(TARGET),FUJITSU_VPP)
