--- qt-everywhere-src-5.15.15/qtwebengine/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/fileutil.py.original	2024-10-04 17:51:07.673416078 +0200
+++ qt-everywhere-src-5.15.15/qtwebengine/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/fileutil.py	2024-10-04 17:51:12.976524066 +0200
@@ -3,7 +3,6 @@
 # found in the LICENSE file.
 
 import errno
-import imp
 import os.path
 import sys
 
--- qt-everywhere-src-5.15.15/qtwebengine/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/parse/lexer.py.original	2024-10-04 17:56:45.794305404 +0200
+++ qt-everywhere-src-5.15.15/qtwebengine/src/3rdparty/chromium/mojo/public/tools/mojom/mojom/parse/lexer.py	2024-10-04 17:56:51.367418971 +0200
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
 import os.path
 import sys
 
