diff -ur qt-everywhere-src-5.15.15-original/qtlocation/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro qt-everywhere-src-5.15.15/qtlocation/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
--- qt-everywhere-src-5.15.15-original/qtlocation/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro	2024-07-23 12:25:58.000000000 +0200
+++ qt-everywhere-src-5.15.15/qtlocation/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro	2024-10-04 15:35:58.476413771 +0200
@@ -2,8 +2,8 @@
 
 load(qt_helper_lib)
 
-CONFIG += qt c++14 exceptions warn_off staticlib object_parallel_to_source
-CONFIG -= c++1z
+CONFIG += qt c++17 exceptions warn_off staticlib object_parallel_to_source
+#CONFIG -= c++1z
 
 QT += network-private \
       gui-private \
