Allow specifying the installation directory in Makefile
Author: Steven Vandenbrande
--- Makefile.Linux.orig	2025-09-19 12:15:08.436914000 +0200
+++ Makefile.Linux	2025-09-19 12:06:01.407024000 +0200
@@ -33,8 +33,8 @@
 	antiword\
 	kantiword
 
-LOCAL_INSTALL_DIR = $(HOME)/bin
-LOCAL_RESOURCES_DIR = $(HOME)/.antiword
+LOCAL_INSTALL_DIR = $(INSTALLDIR)/bin
+LOCAL_RESOURCES_DIR = $(INSTALLDIR)/.antiword
 
 GLOBAL_INSTALL_DIR = /usr/local/bin
 GLOBAL_RESOURCES_DIR = /usr/share/antiword
