# removes hardcoded /usr prefix in install stage of makefile
--- makefile.orig	2021-11-18 09:49:39.332109000 +0100
+++ makefile	2021-11-18 09:51:34.635823416 +0100
@@ -185,11 +185,11 @@
 	$(CC) $(CPPFLAGS) $(OPTIONS) $(INCLUDE) $(CFLAGS) $(MTFLAGS) -c $<
 
 install: all
-	install -m 755 -d $(DESTDIR)/usr/bin
-	install -m 755 -d $(DESTDIR)/usr/share/man/man1
-	install -m 755  uftp $(DESTDIR)/usr/bin
-	install -m 755  uftpd $(DESTDIR)/usr/bin
-	install -m 755  uftpproxyd $(DESTDIR)/usr/bin
-	install -m 755  uftp_keymgt $(DESTDIR)/usr/bin
-	install -m 644  uftp.1 uftpd.1 uftpproxyd.1 uftp_keymgt.1 $(DESTDIR)/usr/share/man/man1
+	install -m 755 -d $(DESTDIR)/bin
+	install -m 755 -d $(DESTDIR)/share/man/man1
+	install -m 755  uftp $(DESTDIR)/bin
+	install -m 755  uftpd $(DESTDIR)/bin
+	install -m 755  uftpproxyd $(DESTDIR)/bin
+	install -m 755  uftp_keymgt $(DESTDIR)/bin
+	install -m 644  uftp.1 uftpd.1 uftpproxyd.1 uftp_keymgt.1 $(DESTDIR)/share/man/man1
 
