Add settings file for pyscf.fciqmc using EasyBuild environment
author: Alex Domingo (Vrije Universiteit Brussel)
diff -Nru pyscf.orig/fciqmcscf/settings.py pyscf/fciqmcscf/settings.py
--- pyscf.orig/fciqmcscf/settings.py	1970-01-01 01:00:00.000000000 +0100
+++ pyscf/fciqmcscf/settings.py	2022-11-24 21:16:33.554298000 +0100
@@ -0,0 +1,9 @@
+#
+# This file sets the environment using EasyBuild packages
+# The configs will be used in fciqmc.py
+#
+
+import os
+
+# Use NECI (FCIQMC) as the FCI sovler for CASSCF
+FCIQMCEXE = os.path.join(os.getenv('EBROOTNECI'), 'bin', 'neci')
