diff --no-dereference -Nru pyscf-2.0.1.orig/pyscf/lib/gto/ft_ao.c pyscf-2.0.1/pyscf/lib/gto/ft_ao.c
--- pyscf-2.0.1.orig/pyscf/lib/gto/ft_ao.c	2021-11-04 20:03:12.000000000 +0100
+++ pyscf-2.0.1/pyscf/lib/gto/ft_ao.c	2022-09-14 11:43:53.059143934 +0200
@@ -1008,7 +1008,7 @@
 
         *jempty = 1;
         for (jp = 0; jp < j_prim; jp++) {
-                envs->aj = aj[jp];
+                *envs->aj = aj[jp];
                 if (j_ctr == 1) {
                         fac1j = fac1 * cj[jp];
                 } else {
@@ -1016,7 +1016,7 @@
                         *iempty = 1;
                 }
                 for (ip = 0; ip < i_prim; ip++) {
-                        envs->ai = ai[ip];
+                        *envs->ai = ai[ip];
                         aij = ai[ip] + aj[jp];
                         eij = (ai[ip] * aj[jp] / aij) * rrij;
                         if (eij > EXP_CUTOFF) {
diff --no-dereference -Nru pyscf-2.0.1.orig/pyscf/lib/gto/ft_ao_deriv.c pyscf-2.0.1/pyscf/lib/gto/ft_ao_deriv.c
--- pyscf-2.0.1.orig/pyscf/lib/gto/ft_ao_deriv.c	2021-11-04 20:03:12.000000000 +0100
+++ pyscf-2.0.1/pyscf/lib/gto/ft_ao_deriv.c	2022-09-14 11:44:41.386141916 +0200
@@ -28,7 +28,7 @@
                     double *Gv, size_t NGv, CINTEnvVars *envs)
 {
         const int dj = envs->g_stride_j;
-        const double ai2 = -2 * envs->ai;
+        const double ai2 = -2 * (*envs->ai);
         const size_t g_size = envs->g_size * NGv;
         double complex *gx = g;
         double complex *gy = g + g_size;
@@ -64,7 +64,7 @@
                     double *Gv, size_t NGv, CINTEnvVars *envs)
 {
         const int dj = envs->g_stride_j;
-        const double aj2 = -2 * envs->aj;
+        const double aj2 = -2 * (*envs->aj);
         const size_t g_size = envs->g_size * NGv;
         double complex *gx = g;
         double complex *gy = g + g_size;
