diff -Nru bowtie-1.1.2.orig/processor_support.h bowtie-1.1.2/processor_support.h
--- bowtie-1.1.2.orig/processor_support.h	2016-08-31 15:28:19.873048824 +0200
+++ bowtie-1.1.2/processor_support.h	2016-08-31 15:22:12.756008244 +0200
@@ -44,8 +44,8 @@
 
     try {
 #if ( defined(USING_INTEL_COMPILER) || defined(USING_MSC_COMPILER) )
-        __cpuid((void *) &regs,0); // test if __cpuid() works, if not catch the exception
-        __cpuid((void *) &regs,0x1); // POPCNT bit is bit 23 in ECX
+//        __cpuid((void *) &regs,0); // test if __cpuid() works, if not catch the exception
+ //       __cpuid((void *) &regs,0x1); // POPCNT bit is bit 23 in ECX
 #elif defined(USING_GCC_COMPILER)
         __get_cpuid(0x1, &regs.EAX, &regs.EBX, &regs.ECX, &regs.EDX);
 #else
