--- pset/install.sh.orig	2010-11-25 15:52:11.335504876 +0100
+++ pset/install.sh	2010-11-25 15:53:48.107531194 +0100
@@ -1917,7 +1917,7 @@
     [[ -z "$dst_dir" ]] && dst_dir=$("$RPM_EXTR_TOOL" -qp --qf %{PREFIXES} "$rpm_path")
     dst_dir=$(echo "$dst_dir" | sed -e"s/\/\{1,\}/\//g")
     dst_dir=$(echo "$dst_dir" | sed -e"s/\/\{1,\}$//g")
-    local db="$INTEL_SDP_PRODUCTS_DB"
+    local db="/tmp/easybuild_impi/intel_sdp_products.db"
     local db_dir=$(dirname "$db") || DIE "Unexpected error."
     local cur_dir=$(pwd) || DIE "Unable to find current directory."
     local rpm_file=$(basename "$rpm_path") || DIE "Unexpected error."
