# Use EB values in Makefile
# wpoely86@gmail.com
diff -ur ploticus242/src/Makefile ploticus242.new/src/Makefile
--- ploticus242/src/Makefile	2013-06-07 19:05:43.000000000 +0200
+++ ploticus242.new/src/Makefile	2019-02-05 16:47:23.430213501 +0100
@@ -7,7 +7,7 @@
 
 
 # defaults..
-CC = gcc
+CC := $(CC)
 PLATFORM = UNIX
 RMCOM = rm -f
 NOX11 = -DNOX11
@@ -92,8 +92,8 @@
 
 ########### Option 1: use bundled GD16 (PNG only).  Requires libpng and zlib.
 exetarget: plpng 
-GD16LIBS = -lpng -lz
-GD16H =  -I/usr/local/include
+GD16LIBS = -L$(EBROOTLIBPNG)/lib -L$(EBROOTZLIB)/lib -lpng -lz
+GD16H =  -I$(EBROOTLIBPNG)/include -I$(EBROOTZLIB)/include
 ZFLAG = -DWZ
 ### These should remain commented unless your libs are not in usual places, then alter to suit...
 ### GD16LIBS = /home/scg/lib/libpng.a /home/scg/lib/libz.a
@@ -140,7 +140,7 @@
 
 
 #### If you do a "make install", where do you want the executable(s) to be moved to?
-INSTALLBIN = /usr/local/bin
+INSTALLBIN := $(INSTALLDIR)
 
 
 #### For LOCALE support (non-roman alphabets & collation), uncomment the following..
