--- linux64/makefile.orig	2013-11-13 10:20:19.196147162 +0100
+++ linux64/makefile	2013-11-13 10:20:48.486687987 +0100
@@ -8,7 +8,7 @@
 CPPFLAGS = -I.. -I../gwnum -DX86_64 -O2
 
 LFLAGS = -Wl,-M
-LIBS = ../gwnum/gwnum.a ../gwnum/gwnum.ld -lm -lpthread -Wl,-Bstatic $(shell pkg-config --static --libs libcurl) -lstdc++ -Wl,-Bdynamic -ldl
+LIBS = ../gwnum/gwnum.a ../gwnum/gwnum.ld -lm -lpthread $(shell pkg-config --static --libs libcurl) -lstdc++ -ldl
 
 FACTOROBJ = factor64.o
 LINUXOBJS = prime.o menu.o
@@ -25,7 +25,7 @@
 	[ ! -e ../secure5.c ] && touch ../secure5.c || true
 
 clean:
-	rm -f $(EXE) $(EXE2) $(LINUXOBJS) $(FACTOROBJ)
+	rm -f $(EXE) $(EXE2) $(LINUXOBJS)
 
 .c.o:
 	$(CC) $(CFLAGS) -c $<
