--- cmake/upcxx.cmake.orig	2019-03-13 20:51:17.139535447 +0000
+++ cmake/upcxx.cmake	2019-03-13 20:51:44.323576700 +0000
@@ -16,7 +16,7 @@
   add_library(upcxx STATIC IMPORTED)
   else()
 set(UPCXX_NAME upcxx)
-  set(UPCXX_REPO https://bitbucket.org/upcxx/upcxx.git)
+  set(UPCXX_REPO https://github.com/brycelelbach/upcxx.git)
 
   ExternalProject_Add(${UPCXX_NAME}
       DEPENDS ${GASNET_NAME}
--- cmake/gasnet.cmake.orig	2019-03-13 19:30:06.961150008 +0000
+++ cmake/gasnet.cmake	2019-03-13 19:30:25.271177695 +0000
@@ -25,7 +25,7 @@
 else()
 
   set(GASNET_NAME gasnet)
-  set(GASNET_URL https://gasnet.lbl.gov/)
+  set(GASNET_URL https://gasnet.lbl.gov/download)
   set(GASNET_GZ  GASNet-1.26.3.tar.gz)
   set(GASNET_MD5 "93ab985881ca19de319bb73583c385d2")
   set(GASNET_CFLAGS "${CMAKE_C_FLAGS} -DGASNETI_PSHM_BARRIER_HIER=0")
