Use F2003 functions for argument handling.

Åke Sandgren, 2021-04-29
diff -ru molden6.8.orig/src/molden.f molden6.8/src/molden.f
--- molden6.8.orig/src/molden.f	2021-01-20 17:54:58.000000000 +0100
+++ molden6.8/src/molden.f	2021-04-29 08:49:09.667816492 +0200
@@ -3812,15 +3812,15 @@
       endif
 
       iredir = 0
-      nargs = iargc()
-      call getarg(0,liris)
+      nargs = COMMAND_ARGUMENT_COUNT()
+      call GET_COMMAND_ARGUMENT(0,liris)
       call parsfn(liris,linlen(liris),20)
 
       ntargs = 0
       n = 0
       if (nargs.gt.0) then
 2013         n = n + 1
-             call getarg(n,liris)
+             call GET_COMMAND_ARGUMENT(n,liris)
              if (liris(1:1).eq.'-') then
                 if (liris(1:2).eq.'-5') ido5d = 1
                 if (liris(1:2).eq.'-7') ido7f = 1
