commit 26cea9f35118aae0f02a195863bbf3a538727e44
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Mon Jul 13 15:18:16 2020 +0200

    Must not override EasyBuilds PYTHONPATH
    
    Åke Sandgren, 20181030

diff --git a/AmberTools/src/configure2 b/AmberTools/src/configure2
index 7072ee0..89bf90d 100755
--- a/AmberTools/src/configure2
+++ b/AmberTools/src/configure2
@@ -3837,7 +3837,7 @@ PUPILLIBS=$pupillibs
 PYTHON=$python
 PYTHON_INSTALL=$python_install_string
 SKIP_PYTHON=$skippython
-PYTHONPATH=\$(AMBER_PREFIX)/lib/python$python_ver/site-packages
+#PYTHONPATH=\$(AMBER_PREFIX)/lib/python$python_ver/site-packages
 PYTHONLOG=$pythonlog
 
 PYSANDER=$pysander
