Fix missing openmp flag at link.

Åke Sandgren, 20171013
diff -ru amber16.orig/AmberTools/src/byacc/Makefile amber16/AmberTools/src/byacc/Makefile
--- amber16.orig/AmberTools/src/byacc/Makefile	2017-04-07 21:08:11.000000000 +0200
+++ amber16/AmberTools/src/byacc/Makefile	2017-10-13 22:17:23.837195086 +0200
@@ -31,7 +31,7 @@
 all:		$(PROGRAM)
 
 $(PROGRAM):     $(OBJS) $(LIBS)
-		@$(CC) $(LDFLAGS) -o $(PROGRAM) $(OBJS) $(LIBS)
+		@$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM) $(OBJS) $(LIBS)
 
 clean:
 		@rm -f $(OBJS)
diff -ru amber16.orig/AmberTools/src/ucpp-1.3/Makefile amber16/AmberTools/src/ucpp-1.3/Makefile
--- amber16.orig/AmberTools/src/ucpp-1.3/Makefile	2017-04-07 21:08:15.000000000 +0200
+++ amber16/AmberTools/src/ucpp-1.3/Makefile	2017-10-13 21:54:44.418148511 +0200
@@ -42,7 +42,7 @@
 	rm -f *.o ucpp core
 
 ucpp$(SFX): $(COBJ)
-	$(CC) $(LDFLAGS) -o ucpp$(SFX) $(COBJ) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o ucpp$(SFX) $(COBJ) $(LIBS)
 
 assert.o: tune.h ucppi.h cpp.h nhash.h mem.h
 cpp.o: tune.h ucppi.h cpp.h nhash.h mem.h
