Use external forbear and aed2 libraries.

Åke Sandgren, 2022-01-25
diff -ru Simstrat-3.01.orig/build/fobos Simstrat-3.01/build/fobos
--- Simstrat-3.01.orig/build/fobos	2021-04-23 15:35:32.000000000 +0200
+++ Simstrat-3.01/build/fobos	2022-01-25 09:54:07.726977196 +0100
@@ -7,35 +7,14 @@
 lflags=-static-libgcc -static-libgfortran -static
 build_dir=.
 src=../src/
-libs = ../lib/libaed2/obj/aed2_bio_utils.o
-	   ../lib/libaed2/obj/aed2_carbon.o
-	   ../lib/libaed2/obj/aed2_chlorophylla.o
-	   ../lib/libaed2/obj/aed2_common.o
-	   ../lib/libaed2/obj/aed2_core.o
-	   ../lib/libaed2/obj/aed2_csv_reader.o
-	   ../lib/libaed2/obj/aed2_dummy.o
-	   ../lib/libaed2/obj/aed2_nitrogen.o
-	   ../lib/libaed2/obj/aed2_noncohesive.o
-	   ../lib/libaed2/obj/aed2_organic_matter.o
-	   ../lib/libaed2/obj/aed2_oxygen.o
-	   ../lib/libaed2/obj/aed2_phosphorus.o
-	   ../lib/libaed2/obj/aed2_phytoplankton.o
-	   ../lib/libaed2/obj/aed2_sedflux.o
-	   ../lib/libaed2/obj/aed2_silica.o
-	   ../lib/libaed2/obj/aed2_totals.o
-	   ../lib/libaed2/obj/aed2_tracer.o
-	   ../lib/libaed2/obj/aed2_util.o
-	   ../lib/libaed2/obj/aed2_zoop_utils.o
-	   ../lib/libaed2/obj/aed2_zooplankton.o
-	   ../lib/libaed2/obj/ufz_oxygen.o
-include = ../lib/libaed2/mod/
+include = $EBROOTLIBAED2/include $EBROOTFORBEAR/include
+ext_libs = aed2 forbear
 colors=False
 quiet=False
 target=simstrat.f90
 output=simstrat
 dependon=../lib/csv_fortran/fobos:release-gnu((direct))
          ../lib/json_fortran/fobos:release-gnu((direct))
-         ../lib/forbear/fobos:static-gnu((direct))
 log=True
 
 [release-gnu-dynamic]
@@ -44,12 +23,13 @@
 lflags=-dynamic-libgcc -dynamic-libgfortran -dynamic
 build_dir=.
 src=../src/
+include = $EBROOTLIBAED2/include $EBROOTFORBEAR/include
+ext_libs = aed2 forbear
 colors=False
 quiet=False
 target=simstrat.f90
 dependon=../lib/csv_fortran/fobos:release-gnu((direct))
          ../lib/json_fortran/fobos:release-gnu((direct))
-         ../lib/forbear/fobos:static-gnu((direct))
 log=True
 
 [release-intel]
@@ -58,6 +38,8 @@
 modsw=-module=
 build_dir=.
 src=../src/
+include = $EBROOTLIBAED2/include $EBROOTFORBEAR/include
+ext_libs = aed2 forbear
 colors=False
 quiet=False
 target=simstrat.f90
@@ -72,13 +54,14 @@
 lflags=-static-libgcc -static-libgfortran -static
 build_dir=.
 src=../src/
+include = $EBROOTLIBAED2/include $EBROOTFORBEAR/include
+ext_libs = aed2 forbear
 colors=False
 quiet=False
 target=simstrat.f90
 output=simstrat
 dependon=../lib/csv_fortran/fobos:release-gnu((direct))
          ../lib/json_fortran/fobos:release-gnu((direct))
-         ../lib/forbear/fobos:static-gnu((direct))
 log=True
 
 [debug-gnu-dynamic]
@@ -87,12 +70,13 @@
 lflags=-dynamic-libgcc -dynamic-libgfortran -dynamic
 build_dir=.
 src=../src/
+include = $EBROOTLIBAED2/include $EBROOTFORBEAR/include
+ext_libs = aed2 forbear
 colors=False
 quiet=False
 target=simstrat.f90
 dependon=../lib/csv_fortran/fobos:release-gnu((direct))
          ../lib/json_fortran/fobos:release-gnu((direct))
-         ../lib/forbear/fobos:static-gnu((direct))
 log=True
 
 [rule-makedoc]
