diff -ru bowtie2-2.3.4.1.original/Makefile bowtie2-2.3.4.1.patched/Makefile
--- bowtie2-2.3.4.1.original/Makefile	2018-02-03 13:57:06.000000000 +0100
+++ bowtie2-2.3.4.1.patched/Makefile	2018-05-02 11:13:33.000000000 +0200
@@ -101,6 +101,7 @@
 
 #default is to use Intel TBB
 ifneq (1,$(NO_TBB))
+	LDFLAGS += -L$(TBBLIBDIR)
 	LDLIBS += $(PTHREAD_LIB) -ltbb
 	ifdef RELEASE_BUILD
 		LDLIBS += -ltbbmalloc
