#Add riscv to the list of non-intel architectures
--- sse_wrap.h.orig	2024-11-15 16:52:03.206493623 +0100
+++ sse_wrap.h	2024-11-15 16:56:30.616429424 +0100
@@ -112,7 +112,7 @@
 
 #else /* no SSE_AVX2 */
 
-#if defined(__aarch64__) || defined(__s390x__) || defined(__powerpc__)
+#if defined(__aarch64__) || defined(__s390x__) || defined(__powerpc__) || defined(__riscv)
 #include "simde/x86/sse2.h"
 #else
 #include <emmintrin.h>
