--- Makefile.orig	2024-11-15 16:28:04.928878070 +0100
+++ Makefile	2024-11-15 16:28:55.730305558 +0100
@@ -83,7 +83,7 @@
   endif
 
   POPCNT_CAPABILITY ?= 1
-else ifneq (,$(findstring $(ARCH), aarch64 arm64 s390x powerpc64 powerpc64le ppc64 ppc64le))
+else ifneq (,$(findstring $(ARCH), aarch64 arm64 s390x powerpc64 powerpc64le ppc64 ppc64le riscv64))
   BITS := 64
   SSE_FLAG :=
   CXXFLAGS += -fopenmp-simd
