--- charm-v6.10.0/src/conv-core/memoryaffinity.C.orig	2020-02-13 01:12:11.000000000 +0000
+++ charm-v6.10.0/src/conv-core/memoryaffinity.C	2020-02-14 13:53:31.990750858 +0000
@@ -182,7 +182,7 @@
         for (i=0; i<strlen((const char *)nodemap); i++) {
             if (nodemap[i]==',') nodemapArrSize++;
         }
-        nodemapArr = malloc(nodemapArrSize*sizeof(int));
+        nodemapArr = (int *)malloc(nodemapArrSize*sizeof(int));
         prevIntStart=j=0;
         for (i=0; i<strlen((const char *)nodemap); i++) {
             if (nodemap[i]==',') {
