* exclude bwidget download
Author: Kenneth Hoste (Ghent University), updated to 1.6.2 by Miguel Dias Costa (National University of Singapore)
--- Makefile.orig	2020-01-16 18:58:30.499898000 +0800
+++ Makefile	2020-01-16 18:58:49.835523505 +0800
@@ -18,7 +18,7 @@
 
 all: tcl tk mesa togl fftw xcrysden
 #  meschach 
-xcrysden: usage bwidget bindir src-C src-F src-Tcl
+xcrysden: usage bindir src-C src-F src-Tcl
 
 usage: $(TOPDIR)/docs/xcrysden.1
 	man $(TOPDIR)/docs/xcrysden.1 | awk 'BEGIN {lprint=0; print "## do not edit changes will be lost (file automatically generated)\n"; } /SYNOPSIS/ { lprint=1; } /SEE ALSO/ { lprint=0; } /a*/ { if (lprint) print; }' > usage
