diff -ru xcrysden-1.5.53.orig/Makefile xcrysden-1.5.53/Makefile
--- xcrysden-1.5.53.orig/Makefile	2012-03-12 21:08:34.000000000 +0100
+++ xcrysden-1.5.53/Makefile	2012-10-08 17:01:13.871096254 +0200
@@ -17,7 +17,7 @@
 # some varibales ...
 
 all: tcl tk mesa fftw meschach xcrysden 
-xcrysden: usage bwidget bindir src-C src-F src-Tcl
+xcrysden: usage bindir src-C src-F src-Tcl
 
 usage: $(TOPDIR)/docs/xcrysden.1
 	man $(TOPDIR)/docs/xcrysden.1 | awk 'BEGIN {lprint=0; print "## do not edit changes will be lost (file automatically generated)\n"; } /SYNOPSIS/ { lprint=1; } /SEE ALSO/ { lprint=0; } /a*/ { if (lprint) print; }' > usage
