--- Utilities/cif2cell/Makefile.orig	2025-01-16 18:23:54.627112826 +0000
+++ Utilities/cif2cell/Makefile	2025-01-16 18:52:41.501207448 +0000
@@ -10,4 +10,4 @@
 
 install:
 	mkdir -p "$(INSTALL_DIR)"
-	PYTHONPATH="$(PYTHONPATH):$(INSTALL_DIR)"  python setup.py install --install-scripts="$(INSTALL_DIR)" --install-lib="$(INSTALL_DIR)" --install-data="$(ROOTDIR)"
+	PYTHONPATH="$(PYTHONPATH):$(INSTALL_DIR)/../"  pip install . --prefix="$(INSTALL_DIR)/../" --no-index 
--- Utilities/CASTEPconv/Makefile.orig	2025-01-16 18:34:43.805548947 +0000
+++ Utilities/CASTEPconv/Makefile	2025-01-16 18:53:02.530415057 +0000
@@ -9,4 +9,4 @@
 	python setup.py clean --all
 
 install:
-	PYTHONPATH=$(INSTALL_DIR) python setup.py install --install-scripts="$(INSTALL_DIR)" --install-lib="$(INSTALL_DIR)" --install-data="$(ROOTDIR)"
+	PYTHONPATH=$(INSTALL_DIR)/../  pip install . --prefix="$(INSTALL_DIR)/../" --no-index 
--- Utilities/elastic_constants/Makefile.orig	2025-01-16 18:35:49.413197813 +0000
+++ Utilities/elastic_constants/Makefile	2025-01-16 18:53:22.224609485 +0000
@@ -9,4 +9,4 @@
 	python setup.py clean --all
 
 install:
-	PYTHONPATH=$(INSTALL_DIR) python setup.py install --install-scripts="$(INSTALL_DIR)" --install-lib="$(INSTALL_DIR)" --install-data="$(ROOTDIR)"
+	PYTHONPATH=$(INSTALL_DIR)../ pip install . --prefix="$(INSTALL_DIR)/../" --no-index
--- obj/platforms/linux_x86_64_ifort-XT.mk.orig	2025-01-17 19:13:29.744598510 +0000
+++ obj/platforms/linux_x86_64_ifort-XT.mk	2025-01-17 19:13:55.234856946 +0000
@@ -115,7 +115,7 @@
 # FlexiBLAS (Run-time dynamic interface to underlying BLAS implementations)
 #
 ifeq ($(MATHLIBS),flexiblas)
-MATH_LIBS = -lflexiblas
+MATH_LIBS = -lflexiblas_intel
 endif
 #
 # AOCL   AMD optimized libraries fro Zen architectures
--- obj/platforms/linux_x86_64_ifort.mk.orig	2025-01-17 19:12:15.707847869 +0000
+++ obj/platforms/linux_x86_64_ifort.mk	2025-01-17 19:12:23.535927235 +0000
@@ -125,7 +125,7 @@
 # FlexiBLAS (Run-time dynamic interface to underlying BLAS implementations)
 #
 ifeq ($(MATHLIBS),flexiblas)
-MATH_LIBS = -lflexiblas
+MATH_LIBS = -lflexiblas_intel
 endif
 #
 # AOCL   AMD optimized libraries fro Zen architectures
