diff -ru CASTEP-19.11.orig/Utilities/c2x/Makefile CASTEP-19.11.patched/Utilities/c2x/Makefile
--- CASTEP-19.11.orig/Utilities/c2x/Makefile	2020-04-28 18:01:18.801978921 +0000
+++ CASTEP-19.11.patched/Utilities/c2x/Makefile	2020-04-29 17:35:12.962717357 +0000
@@ -51,4 +51,6 @@
 
 clean:
 	rm c2x $(OBJS)
+install:
+	/bin/cp c2x "$(INSTALL_DIR)"
 
diff -ru CASTEP-19.11.orig/Utilities/mdtep/Makefile CASTEP-19.11.patched/Utilities/mdtep/Makefile
--- CASTEP-19.11.orig/Utilities/mdtep/Makefile	2020-04-28 18:01:18.730978547 +0000
+++ CASTEP-19.11.patched/Utilities/mdtep/Makefile	2020-04-29 15:14:17.559874853 +0000
@@ -2,8 +2,8 @@
 .PRECIOUS: %.$(OBJ_EXTN)
 .PHONY:  clean
 
-F90 = gfortran
-FFLAGS = -O3
+F90 := $(F90)
+FFLAGS := $(FFLAGS)
 OBJ_EXTN = o
 include ../../../obj/$(ARCH).mk
 
