diff --color -ru CASTEP-22.11.old/Makefile CASTEP-22.11/Makefile
--- CASTEP-22.11.old/Makefile	2022-09-13 11:03:18
+++ CASTEP-22.11/Makefile	2023-01-04 14:57:14
@@ -684,7 +684,7 @@
               $(error FFT should be one of "generic" "vendor", "fftw3", "default", "mkl", "acml" (not $(FFT))) )
 
 mathlib_choice:
-	$(if $(filter-out acml atlas cxml default essl generic openblas mkl scilib scs scsl sunperf vendor,$(MATHLIBS)),\
+	$(if $(filter-out acml atlas cxml default essl generic openblas mkl scilib scs scsl sunperf vendor flexiblas,$(MATHLIBS)),\
               $(error MATHLIBS should be one of "acml","atlas","cxml","default","essl","generic","openblas ","mkl","scilib","scs","scsl","sunperf","vendor" (not $(MATHLIBS))) )
 
 build_choice:
diff --color -ru CASTEP-22.11.old/obj/platforms/linux_x86_64_gfortran10.mk CASTEP-22.11/obj/platforms/linux_x86_64_gfortran10.mk
--- CASTEP-22.11.old/obj/platforms/linux_x86_64_gfortran10.mk	2022-05-23 13:31:02
+++ CASTEP-22.11/obj/platforms/linux_x86_64_gfortran10.mk	2023-01-04 15:00:52
@@ -99,6 +99,13 @@
 MATH_LIBS = -lopenblas
 endif
 #
+# FlexiBLAS
+#
+ifeq ($(MATHLIBS),flexiblas)
+MATH_LIBS = -lflexiblas
+endif
+
+#
 # ATLAS BLAS
 #
 ifeq ($(MATHLIBS),atlas)
