diff -ru MOFA2/configure MOFA2diff/configure
--- MOFA2/configure	2020-11-04 01:57:25.000000000 +0000
+++ MOFA2diff/configure	2021-03-31 10:50:37.285205000 +0100
@@ -1,3 +1,2 @@
 #!/bin/sh
 
-${R_HOME}/bin/Rscript -e "if (basilisk.utils::useSystemDir()) basilisk.utils::installMiniconda(installed=FALSE)"
diff -ru MOFA2/NAMESPACE MOFA2diff/NAMESPACE
--- MOFA2/NAMESPACE	2020-10-27 18:24:26.000000000 +0000
+++ MOFA2diff/NAMESPACE	2021-03-31 10:51:09.201746000 +0100
@@ -86,7 +86,6 @@
 exportMethods(samples_metadata)
 exportMethods(samples_names)
 exportMethods(views_names)
-import(basilisk)
 import(dplyr)
 import(ggplot2)
 import(grDevices)
@@ -97,7 +96,6 @@
 importFrom(HDF5Array,HDF5ArraySeed)
 importFrom(RColorBrewer,brewer.pal)
 importFrom(Rtsne,Rtsne)
-importFrom(basilisk,BasiliskEnvironment)
 importFrom(corrplot,corrplot)
 importFrom(cowplot,plot_grid)
 importFrom(dplyr,bind_rows)
Common subdirectories: MOFA2/build and MOFA2diff/build
diff MOFA2/configure MOFA2diff/configure
3c3
< ${R_HOME}/bin/Rscript -e "if (basilisk.utils::useSystemDir()) basilisk.utils::installMiniconda(installed=FALSE)"
---
> ${R_HOME}/bin/Rscript -e ""
Common subdirectories: MOFA2/inst and MOFA2diff/inst
Common subdirectories: MOFA2/man and MOFA2diff/man
diff MOFA2/NAMESPACE MOFA2diff/NAMESPACE
89d88
< import(basilisk)
100d98
< importFrom(basilisk,BasiliskEnvironment)
Common subdirectories: MOFA2/R and MOFA2diff/R
Common subdirectories: MOFA2/tests and MOFA2diff/tests
Common subdirectories: MOFA2/vignettes and MOFA2diff/vignettes
