--- SRC/MPI/Makefile.orig	2022-01-03 21:46:53.000000000 -0800
+++ SRC/MPI/Makefile	2022-01-03 21:52:58.000000000 -0800
@@ -180,19 +180,19 @@
 #  -------------------------------------
 #  Compile the (ouch!) fortran internals
 #  -------------------------------------
-bi_f77_init.o : mpif.h bi_f77_init.f
+bi_f77_init.o :  bi_f77_init.f
 	$(F77) -c $(F77FLAGS) $*.f
-bi_f77_get_constants.o : mpif.h bi_f77_get_constants.f
+bi_f77_get_constants.o :  bi_f77_get_constants.f
 	$(F77) -c $(F77FLAGS) $*.f
-bi_f77_mpi_attr_get.o : mpif.h bi_f77_mpi_attr_get.f
+bi_f77_mpi_attr_get.o : bi_f77_mpi_attr_get.f
 	$(F77) -c $(F77FLAGS) $*.f
-bi_f77_mpi_initialized.o : mpif.h bi_f77_mpi_initialized.f
+bi_f77_mpi_initialized.o : bi_f77_mpi_initialized.f
 	$(F77) -c $(F77FLAGS) $*.f
-bi_f77_mpi_op_create.o : mpif.h bi_f77_mpi_op_create.f
+bi_f77_mpi_op_create.o : bi_f77_mpi_op_create.f
 	$(F77) -c $(F77FLAGS) $*.f
-bi_f77_mpi_test.o : mpif.h bi_f77_mpi_test.f
+bi_f77_mpi_test.o : bi_f77_mpi_test.f
 	$(F77) -c $(F77FLAGS) $*.f
-bi_f77_mpi_testall.o : mpif.h bi_f77_mpi_testall.f
+bi_f77_mpi_testall.o : bi_f77_mpi_testall.f
 	$(F77) -c $(F77FLAGS) $*.f
 
 mpif.h : $(MPIINCdir)/mpif.h
--- INSTALL/Makefile.orig	2022-01-03 22:07:23.000000000 -0800
+++ INSTALL/Makefile	2022-01-03 22:07:38.000000000 -0800
@@ -28,7 +28,7 @@
 $(INSTdir)/xsyserrors : syserrors.o
 	$(CCLOADER) $(CCLOADFLAGS) -o $@ syserrors.o $(MPILIB)
 
-$(INSTdir)/xtc_CsameF77 : mpif.h tc_fCsameF77.o tc_cCsameF77.o
+$(INSTdir)/xtc_CsameF77 : tc_fCsameF77.o tc_cCsameF77.o
 	$(F77LOADER) $(F77LOADFLAGS) -o $@ tc_fCsameF77.o tc_cCsameF77.o $(MPILIB)
 
 $(INSTdir)/xtc_UseMpich : tc_UseMpich.o
@@ -37,7 +37,7 @@
 $(INSTdir)/xcmpi_sane : cmpi_sane.o
 	$(CCLOADER) $(CCLOADFLAGS) -o $@ cmpi_sane.o $(MPILIB)
 
-$(INSTdir)/xfmpi_sane : mpif.h fmpi_sane.o
+$(INSTdir)/xfmpi_sane : fmpi_sane.o
 	$(F77LOADER) $(F77LOADFLAGS) -o $@ fmpi_sane.o $(MPILIB)
 
 clean:
