--- CADD.sh	2018-12-03 17:00:24.000000000 +0100
+++ /groups/umcg-gcc/tmp04/CADD/software/CADD/v1.4/bin/CADD.sh	2018-11-24 21:27:27.000000000 +0100
@@ -94,7 +94,11 @@
 ### Pipeline
 
 # Loading the environment
-source activate cadd-env
+
+# <<< START Replace Conda by EasyBuild
+# source activate cadd-env 
+# <<< END   Replace Conda by EasyBuild
+
 
 # File preparation
 if [ "$FILEFORMAT" == "vcf" ]
@@ -127,6 +131,12 @@
 fi
 
 # Variant annotation
+# <<< START Add vep to PATH
+if [[ ! ${PATH} =~ ${EBROOTVEP} ]]; then
+    export PATH=${EBROOTVEP}:${PATH}
+fi
+# <<< END   Add vep to PATH
+
 cat $TMP_VCF \
 | vep --quiet --cache --buffer 1000 --no_stats --offline --vcf \
     --dir $CADD/data/annotations/$GENOMEBUILD/vep \
