diff -rup cvs-1.11.23.orig/doc/cvs.texinfo cvs-1.11.23/doc/cvs.texinfo
--- cvs-1.11.23.orig/doc/cvs.texinfo	2014-03-20 14:09:54.695168055 +0100
+++ cvs-1.11.23/doc/cvs.texinfo	2014-03-20 14:10:09.206280539 +0100
@@ -111,9 +111,9 @@ approved by the Free Software Foundation
 @sp 4
 @comment The title is printed in a large font.
 @center @titlefont{Version Management}
-@sp
+@sp 1
 @center @titlefont{with}
-@sp
+@sp 1
 @center @titlefont{CVS}
 @sp 2
 @center for @sc{cvs} @value{VERSION}
diff -rup cvs-1.11.23.orig/lib/getline.c cvs-1.11.23/lib/getline.c
--- cvs-1.11.23.orig/lib/getline.c	2014-03-20 14:09:54.682167954 +0100
+++ cvs-1.11.23/lib/getline.c	2014-03-20 14:10:09.175280299 +0100
@@ -155,7 +155,7 @@ getstr (lineptr, n, stream, terminator,
 }
 
 int
-getline (lineptr, n, stream)
+get_line (lineptr, n, stream)
      char **lineptr;
      size_t *n;
      FILE *stream;
diff -rup cvs-1.11.23.orig/lib/getline.h cvs-1.11.23/lib/getline.h
--- cvs-1.11.23.orig/lib/getline.h	2014-03-20 14:09:54.681167947 +0100
+++ cvs-1.11.23/lib/getline.h	2014-03-20 14:10:09.176280306 +0100
@@ -12,7 +12,7 @@
 #define GETLINE_NO_LIMIT -1
 
 int
-  getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
+  get_line __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
 int
   getline_safe __PROTO ((char **_lineptr, size_t *_n, FILE *_stream,
                          int limit));
