--- signalp.orig	2017-05-17 14:00:41.894544003 +0000
+++ signalp	2017-05-17 15:06:38.773855155 +0000
@@ -10,11 +10,11 @@
 
 # full path to the signalp-4.1 directory on your system (mandatory)
 BEGIN {
-    $ENV{SIGNALP} = '/usr/cbs/bio/src/signalp-4.1';
+    $ENV{SIGNALP} = "$ENV{'EBROOTSIGNALP'}";
 }
 
 # determine where to store temporary files (must be writable to all users)
-my $outputDir = "/var/tmp";
+my $outputDir = "/tmp";
 
 # max number of sequences per run (any number can be handled)
 my $MAX_ALLOWED_ENTRIES=10000;
