--- config/Makefile.linuxx8664icc.orig	2011-03-14 15:15:44.000000000 +0100
+++ config/Makefile.linuxx8664icc	2011-03-17 16:53:59.072216893 +0100
@@ -96,8 +96,8 @@
 ifeq ($(F77),ifort)
 F77           = ifort
 F77FLAGS      = -fPIC
 ifeq ($(ICC_MAJOR),12)
-F77LIBS      := -L$(shell which ifort | sed -e 's|bin/intel64/ifort|compiler/lib|') \
+F77LIBS      := -L$(shell which ifort | sed -e 's|bin/intel64/ifort|compiler/lib/intel64|') \
                 -lifcore
 F77FLAGS     += -m64
 else
