--- root/config/Makefile.linuxx8664icc.orig	2013-12-02 17:22:25.630583702 +0100
+++ root/config/Makefile.linuxx8664icc	2013-12-02 17:22:51.951053273 +0100
@@ -87,7 +87,7 @@
 F77           = ifort
 F77FLAGS      = -fPIC
 ifeq ($(findstring $(ICC_MAJOR),11 10 9),)
-F77LIBS      := -L$(shell which ifort | sed -e 's|bin/intel64/ifort|compiler/lib|') \
+F77LIBS      := -L$(shell which ifort | sed -e 's|bin/intel64/ifort|compiler/lib/intel64|') \
                 -lifcore
 F77FLAGS     += -m64
 else
