Fix linking to libcuda in FA3 C extension
author: Alex Domingo (Vrije Universiteit Brussel)
--- hopper/setup.py.orig	2026-02-20 11:31:29.082202000 +0100
+++ hopper/setup.py	2026-02-23 16:22:33.698635631 +0100
@@ -630,6 +630,7 @@
                 "nvcc": nvcc_threads_args() + nvcc_flags + cc_flag + feature_args,
             },
             include_dirs=include_dirs,
+            extra_link_args=['-Wl,--no-as-needed', '-lcuda'],
             py_limited_api=True,
         )
     )
