Add install target
author: micketeer@gmail.com
--- CMakeLists.txt.orig	2026-09-30 20:30:30.525994094 +0200
+++ CMakeLists.txt	2026-09-30 20:34:33.482368848 +0200
@@ -8,3 +8,30 @@
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} -O2 -DNDEBUG")
 
 add_subdirectory(src)
+
+install(TARGETS gx
+    RUNTIME DESTINATION build/src/
+)
+
+install(DIRECTORY
+    ${CMAKE_SOURCE_DIR}/scripts/
+    DESTINATION scripts
+    FILE_PERMISSIONS
+        OWNER_READ OWNER_WRITE OWNER_EXECUTE
+        GROUP_READ GROUP_EXECUTE
+        WORLD_READ WORLD_EXECUTE
+)
+
+install(DIRECTORY
+    ${CMAKE_SOURCE_DIR}/make_gxdb/
+    DESTINATION make_gxdb
+    FILE_PERMISSIONS
+        OWNER_READ OWNER_WRITE OWNER_EXECUTE
+        GROUP_READ GROUP_EXECUTE
+        WORLD_READ WORLD_EXECUTE
+)
+
+install(DIRECTORY
+    ${CMAKE_SOURCE_DIR}/dist/
+    DESTINATION bin/
+)
