Fix path to internal helper script
author: micketeer@gmail.com
--- src/CMakeLists.txt.orig	2026-09-30 19:10:15.300312910 +0200
+++ src/CMakeLists.txt	2026-09-30 19:10:35.485011030 +0200
@@ -31,7 +31,7 @@
 
 include_directories(${PROJECT_SOURCE_DIR}/include/)
 
-set_source_files_properties(main.cpp PROPERTIES COMPILE_FLAGS "-DGIT_REVISION=`../../build-support/git_describe.sh`")
+set_source_files_properties(main.cpp PROPERTIES COMPILE_FLAGS "-DGIT_REVISION=`${CMAKE_SOURCE_DIR}/build-support/git_describe.sh`")
 
 file(GLOB_RECURSE SOURCES *.cpp)
 add_executable(gx ${SOURCES})
