diff -Naur hmmer-3.1b1-linux-intel-x86_64.orig/easel/Makefile.in hmmer-3.1b1-linux-intel-x86_64.patched/easel/Makefile.in
--- hmmer-3.1b1-linux-intel-x86_64.orig/easel/Makefile.in	2013-05-26 01:56:43.000000000 +0000
+++ hmmer-3.1b1-linux-intel-x86_64.patched/easel/Makefile.in	2013-06-07 13:17:27.873110176 +0000
@@ -403,10 +403,10 @@
 	   else DFILE=${srcdir}/esl_$${BASENAME}.c ;\
         fi;\
 	if test ${V} ;\
-	   then echo "${CC} ${CFLAGS} ${SIMDFLAGS} ${DEFS} ${LDFLAGS} -o $@ -I. -I${srcdir} -L. -D$${DFLAG} $${DFILE} -leasel -lm" ;\
+	   then echo "${CC} ${CFLAGS} ${SIMDFLAGS} ${DEFS} ${LDFLAGS} -o $@ -I. -I${srcdir} -L. -D$${DFLAG} $${DFILE} -leasel -lm ${LIBS} " ;\
 	   else echo '    ' GEN $@ ;\
 	fi ;\
-	${CC} ${CFLAGS} ${SIMDFLAGS} ${DEFS} ${LDFLAGS} -o $@ -I. -I${srcdir} -L. -D$${DFLAG} $${DFILE} -leasel -lm
+	${CC} ${CFLAGS} ${SIMDFLAGS} ${DEFS} ${LDFLAGS} -o $@ -I. -I${srcdir} -L. -D$${DFLAG} $${DFILE} -leasel -lm ${LIBS}
 
 # Benchmark compilation:
 # Name construction much like unit tests.
