Ensure headers from FlexiBLAS can be picked up correctly
diff --git a/cmake/custom/math.cmake b/cmake/custom/math.cmake
index fa5f81ea8..c817a1f39 100644
--- a/cmake/custom/math.cmake
+++ b/cmake/custom/math.cmake
@@ -231,6 +231,8 @@ if(_check_cblas_and_lapacke)
     PATHS
       ${BLAS_ROOT}/include
       ${LAPACK_ROOT}/include
+    PATH_SUFFIXES
+      flexiblas
     )
   list(APPEND _la_include_dirs ${_include_dirs})
 
