Add install target.

Åke Sandgren, 20190502
diff -ru adda-1.3b4.orig/src/common.mk adda-1.3b4/src/common.mk
--- adda-1.3b4.orig/src/common.mk	2014-02-20 15:48:03.000000000 +0100
+++ adda-1.3b4/src/common.mk	2019-05-02 10:29:45.487297028 +0200
@@ -108,3 +108,7 @@
 	echo -n "$(subst ",\",$(CPPCMD))" > $@
 
 -include $(CDEPEND)
+
+install: $(PROG)
+	mkdir -p $(DESTDIR)/bin
+	cp $(PROG) $(DESTDIR)/bin
diff -ru adda-1.3b4.orig/src/Makefile adda-1.3b4/src/Makefile
--- adda-1.3b4.orig/src/Makefile	2014-02-20 15:48:03.000000000 +0100
+++ adda-1.3b4/src/Makefile	2019-05-02 10:44:48.222497406 +0200
@@ -467,6 +467,11 @@
 	@echo "Compiling OpenCL version of ADDA"
 	$(MAKE) -C $(OCL)
 
+install:
+	for i in $(TARGETS); do \
+	    make -C $$i install; \
+	done
+
 cleanfull: clean cleanruns
 
 clean: cleanseq cleanmpi cleanocl
