Remove xxOPTSFILE targets, they interfere with installation.

Åke Sandgren, 20190502
diff -ru adda-1.3b4.orig/src/common.mk adda-1.3b4/src/common.mk
--- adda-1.3b4.orig/src/common.mk	2019-05-02 10:53:25.301453425 +0200
+++ adda-1.3b4/src/common.mk	2019-05-02 10:55:31.580222293 +0200
@@ -43,19 +43,6 @@
 
 READ_FILE = $(shell if [ -f $(1) ]; then cat $(1); fi)
 
-ifneq ($(call READ_FILE,$(LDOPTSFILE)),$(LDCMD))
-  $(shell rm -f $(LDOPTSFILE))
-endif
-ifneq ($(call READ_FILE,$(COPTSFILE)),$(CCMD))
-  $(shell rm -f $(COPTSFILE))
-endif
-ifneq ($(call READ_FILE,$(FOPTSFILE)),$(FCMD))
-  $(shell rm -f $(FOPTSFILE))
-endif
-ifneq ($(call READ_FILE,$(CPPOPTSFILE)),$(CPPCMD))
-  $(shell rm -f $(CPPOPTSFILE))
-endif
-
 vpath %.c $(PARENT)
 vpath %.cpp $(PARENT)/$(CPPFOLDER)
 vpath %.h $(PARENT)
@@ -68,22 +55,22 @@
 .DELETE_ON_ERROR:
 
 # C linker is used, while Fortran and C++ is handled by addition of libraries for linking
-$(PROG): $(COBJECTS) $(FOBJECTS) $(F90OBJECTS) $(CPPOBJECTS) $(LDOPTSFILE)
+$(PROG): $(COBJECTS) $(FOBJECTS) $(F90OBJECTS) $(CPPOBJECTS)
 	@echo "Building $@"
 	$(MYCC) -o $@ $(COBJECTS) $(FOBJECTS) $(F90OBJECTS) $(CPPOBJECTS) $(LDFLAGS)
 
-$(COBJECTS): %.o: %.c $(COPTSFILE)
+$(COBJECTS): %.o: %.c
 	$(MYCC) -c $(CFLAGS) $(DEPFLAG) $<
 
 # Dependencies are only generated for C and C++ sources; we assume that each Fortran file is completely independent or
 # all of the files from dependent set are compiled at once.
-$(FOBJECTS): %.o: %.f $(FOPTSFILE)
+$(FOBJECTS): %.o: %.f
 	$(MYCF) -c $(FFLAGS) $<
 	
-$(F90OBJECTS): %.o: %.f90 $(FOPTSFILE)
+$(F90OBJECTS): %.o: %.f90
 	$(MYCF) -c $(FFLAGS) $<
 
-$(CPPOBJECTS): %.o: %.cpp $(CPPOPTSFILE)
+$(CPPOBJECTS): %.o: %.cpp
 	$(MYCCPP) -c $(CPPFLAGS) $(DEPFLAG) $<
 
 # Special rule for generation of stringified CL source. Used only for ocl. All relevant variables are defined in
@@ -91,22 +78,6 @@
 $(addprefix $(CLSPREFIX),$(CLSTRING)): $(CLSPREFIX)%.clstr:%.cl
 	$(CLSCOMMAND) $< > $@
 
-$(LDOPTSFILE):
-	@echo Linking needs to be redone
-	echo -n "$(subst ",\",$(LDCMD))" > $@
-
-$(COPTSFILE):
-	@echo C sources need to be recompiled
-	echo -n "$(subst ",\",$(CCMD))" > $@
-
-$(FOPTSFILE):
-	@echo Fortran sources need to be recompiled
-	echo -n "$(subst ",\",$(FCMD))" > $@
-
-$(CPPOPTSFILE):
-	@echo C++ sources need to be recompiled
-	echo -n "$(subst ",\",$(CPPCMD))" > $@
-
 -include $(CDEPEND)
 
 install: $(PROG)
