--- libflame-5.1/CMakeLists.txt.orig	2025-05-12 17:41:42.395659119 +0000
+++ libflame-5.1/CMakeLists.txt	2025-05-12 17:43:16.556106723 +0000
@@ -978,6 +978,7 @@
                  ${CMAKE_SOURCE_DIR}/${BASE_INC_PATH}/FLAME.h
                  ${OBJECT_LIBRARIES}
                  )
+	set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME}${LIBNAME_SUFFIX} )
 	message(STATUS "Shared library PROJECT_NAME: ${PROJECT_NAME}")
 	target_compile_definitions("${PROJECT_NAME}" PUBLIC -DLIBFLAME_IS_BUILDING_LIBRARY)
 	target_compile_definitions("${PROJECT_NAME}" PUBLIC -DLIBFLAME_ENABLE_SHARED)
