diff -ru bcl2fastq.orig/src/cxx/include/io/GzipDecompressor.hh bcl2fastq/src/cxx/include/io/GzipDecompressor.hh
--- bcl2fastq.orig/src/cxx/include/io/GzipDecompressor.hh	2021-05-26 14:42:38.247311070 +0000
+++ bcl2fastq/src/cxx/include/io/GzipDecompressor.hh	2021-05-26 14:43:00.521444573 +0000
@@ -28,6 +28,9 @@
 #include <boost/iostreams/concepts.hpp>
 
 #include <zlib.h>
+#ifndef OF
+#define OF(x) x
+#endif
 
 #include "common/Exceptions.hh"
 #include "common/StaticMemPool.hh"
