Correct the pool_ member name in StaticMemPool copy operations.
Author: Emik Lin (HKUMed CPOS)
--- a/src/cxx/include/common/StaticMemPool.hpp
+++ b/src/cxx/include/common/StaticMemPool.hpp
@@ -46,6 +46,6 @@
 : pool_()
 , metadata_()
 {
-    std::copy(&that.pool_[0][0], &that.pool[PageCount-1][PageSize], &pool_[0][0]);
+    std::copy(&that.pool_[0][0], &that.pool_[PageCount-1][PageSize], &pool_[0][0]);
     std::copy(&that.metadata_[0], &that.metadata_[PageCount], &metadata_[0]);
 }
@@ -55,7 +55,7 @@
 {
     if (this != &rhs)
     {
-        std::copy(&rhs.pool_[0][0], &rhs.pool[PageCount-1][PageSize], &pool_[0][0]);
+        std::copy(&rhs.pool_[0][0], &rhs.pool_[PageCount-1][PageSize], &pool_[0][0]);
         std::copy(&rhs.metadata_[0], &rhs.metadata_[PageCount], &metadata_[0]);
     }
     return *this;
