Check for alternative hostnames when determining localhost
Author: Samuel Moors, Vrije Universiteit Brussel (VUB)
diff -ur scoop-0.7.1.1.orig/scoop/utils.py scoop-0.7.1.1/scoop/utils.py
--- scoop-0.7.1.1.orig/scoop/utils.py	2015-08-21 15:53:21.000000000 +0200
+++ scoop-0.7.1.1/scoop/utils.py	2019-10-16 09:07:43.867711000 +0200
@@ -39,6 +39,8 @@
     socket.getfqdn().split('.')[0],
 ]
 
+localHostnames.extend(os.popen('hostname -A').read().split())
+
 localHostnames.extend([
     ip for ip in socket.gethostbyname_ex(socket.gethostname())[2]
         if not ip.startswith("127.")][:1]
