diff -ruN q2cli-2025.4.0/q2cli/util.py q2cli-2025.4.0_new/q2cli/util.py
--- q2cli-2025.4.0/q2cli/util.py	2025-05-01 23:16:56.000000000 +0200
+++ q2cli-2025.4.0_new/q2cli/util.py	2025-09-15 17:20:00.801266434 +0200
@@ -13,7 +13,7 @@
 
 def get_app_dir():
     import os
-    conda_prefix = os.environ.get('CONDA_PREFIX')
+    conda_prefix = os.environ.get('CONDA_PREFIX') or ''
     environment_cache = os.path.join(conda_prefix, 'var', 'q2cli')
     if conda_prefix is not None and (os.access(conda_prefix, os.W_OK | os.X_OK)
                                      or os.path.exists(environment_cache)):
