Patch to make the make.inc file EasyBuild compatible
Author: J. Sassmannshausen (Imperial College London/UK)
diff --git a/slicot-5.0+20101122.orig/make.inc b/slicot-5.0+20101122/make.inc
index 3136485..c1458e9 100755
--- a/slicot-5.0+20101122.orig/make.inc
+++ b/slicot-5.0+20101122/make.inc
@@ -16,10 +16,10 @@
 #  selected.  Define LOADER and LOADOPTS to refer to the loader and 
 #  desired load options for your machine.
 #
-FORTRAN  = f77
-OPTS     = -O4 -native -u
-NOOPT    = -u
-LOADER   = f77
+FORTRAN  = gfortran
+OPTS     = -O2 -mtune=native -fPIC -fdefault-integer-8
+NOOPT    =
+LOADER   = gfortran
 LOADOPTS = $(SLICOTLIB) $(LPKAUXLIB) $(LAPACKLIB)
 #
 #  The archiver and the flag(s) to use when building archive (library)
@@ -35,6 +35,6 @@ ARCHFLAGS= r
 #  possible.)
 #
 #BLASLIB     = -L/software/lib
-LAPACKLIB    = -L/software/lib -llapack -lblas
+LAPACKLIB    = -L${EBROOTOPENBLAS}/lib -lflexiblas
 SLICOTLIB    = ../slicot.a
 LPKAUXLIB    = ../lpkaux.a
