--- Src/configure.orig	2017-04-03 18:44:23.972762590 +0000
+++ Src/configure	2017-04-03 18:24:41.170035637 +0000
@@ -6069,7 +6069,8 @@
   LIBS="$save_LIBS"
 fi
 if test $tw_netcdf_ok = yes; then
-  DEFS="$DEFS CDF";NETCDF_INTERFACE="libnetcdf_f90.a"
+  DEFS="$DEFS CDF";
+#  DEFS="$DEFS CDF";NETCDF_INTERFACE="libnetcdf_f90.a"
 else
   NETCDF_LIBS="";tw_netcdf_ok=no;if test ! -z "$with_netcdf"; then { { echo "$as_me:$LINENO: error: Could not find NetCDF library." >&5
 echo "$as_me: error: Could not find NetCDF library." >&2;}
