don't override EasyBuild's optarch
Author: Miguel Dias Costa (University of Coimbra)
--- Config/cmake/toolchains/gnu.cmake.orig	2025-04-08 15:30:44.064203615 +0100
+++ Config/cmake/toolchains/gnu.cmake	2025-04-08 15:31:44.774901121 +0100
@@ -24,10 +24,10 @@
 set(Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
   CACHE STRING "Build type independent Fortran compiler flags")
 
-set(Fortran_FLAGS_RELEASE "-O3 -march=native"
+set(Fortran_FLAGS_RELEASE ""
   CACHE STRING "Fortran compiler flags for Release build")
 
-set(Fortran_FLAGS_MINSIZEREL "-Os -march=native"
+set(Fortran_FLAGS_MINSIZEREL "-Os"
   CACHE STRING "Fortran compiler flags for minimum size build")
 
 set(Fortran_FLAGS_RELWITHDEBINFO "-g ${Fortran_FLAGS_RELEASE} -fbacktrace"
@@ -46,7 +46,7 @@
 set(C_FLAGS "${CMAKE_C_FLAGS}"
   CACHE STRING "Build type independent C compiler flags")
 
-set(C_FLAGS_RELEASE "-O3 -march=native"
+set(C_FLAGS_RELEASE ""
   CACHE STRING  "C compiler flags for Release build")
 
 set(C_FLAGS_RELWITDEBINFO "-g ${C_FLAGS_RELEASE}"
