diff -ru kneaddata_v0.5.1.original/kneaddata/config.py kneaddata_v0.5.1.patched/kneaddata/config.py
--- kneaddata_v0.5.1.original/kneaddata/config.py	2016-02-02 22:55:16.000000000 +0100
+++ kneaddata_v0.5.1.patched/kneaddata/config.py	2018-05-01 16:59:37.000000000 +0200
@@ -31,7 +31,8 @@
 
 bmtagger_exe="bmtagger.sh"
 
-trimmomatic_jar="trimmomatic-0.33.jar"
+#trimmomatic_jar="trimmomatic-0.33.jar"
+trimmomatic_jar="trimmomatic.jar"
 trimmomatic_memory="500m"
 trimmomatic_flag_start="-"
 trimmomatic_slidingwindow_option="SLIDINGWINDOW:4:20"
diff -ru kneaddata_v0.5.1.original/kneaddata/knead_data.py kneaddata_v0.5.1.patched/kneaddata/knead_data.py
--- kneaddata_v0.5.1.original/kneaddata/knead_data.py	2016-02-02 22:55:16.000000000 +0100
+++ kneaddata_v0.5.1.patched/kneaddata/knead_data.py	2018-05-01 16:59:05.000000000 +0200
@@ -289,7 +289,8 @@
         
     # find the location of trimmomatic, trimmomatic does not need to be executable
     if not args.bypass_trim:
-        args.trimmomatic_path=utilities.find_dependency(args.trimmomatic_path,config.trimmomatic_jar,"trimmomatic",
+        #args.trimmomatic_path=utilities.find_dependency(args.trimmomatic_path,config.trimmomatic_jar,"trimmomatic",
+        args.trimmomatic_path=utilities.find_dependency(os.environ["EBROOTTRIMMOMATIC"],config.trimmomatic_jar,"trimmomatic",
             "--trimmomatic", bypass_permissions_check=True)
     
     # find the location of bmtagger, if set to run
