diff -Naur BGW-2.0.0_orig/Makefile BGW-2.0.0/Makefile
--- BGW-2.0.0_orig/Makefile	2019-04-13 23:27:18.757685895 +0000
+++ BGW-2.0.0/Makefile	2019-04-13 23:28:08.243773969 +0000
@@ -124,7 +124,7 @@
 # install cannot work on a whole directory
 	cp -rf examples  $(INSTDIR)/share/BerkeleyGW/
 	cp -rf testsuite $(INSTDIR)/share/BerkeleyGW/
-	install manual.html $(INSTDIR)/share/BerkeleyGW/
+	#install manual.html $(INSTDIR)/share/BerkeleyGW/
 else
 	$(error Error: Please define installation prefix INSTDIR via 'make install INSTDIR='.)
 endif
