fix paths in pbs template
Author: Miguel Dias Costa
--- opt/lumerical/fdtd/templates/fdtd-pbs-template.sh.orig	2019-05-09 09:30:30.424131000 +0800
+++ opt/lumerical/fdtd/templates/fdtd-pbs-template.sh	2019-05-09 09:33:28.842542991 +0800
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/fdtd/bin/fdtd-engine-mpichp4"
+MY_PROG="$EBROOTFDTD_SOLUTIONS/bin/fdtd-engine-mpich2nem"
 INPUT="<filename>"
-/opt/lumerical/fdtd/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
+$EBROOTFDTD_SOLUTIONS/mpich2/nemesis/bin/mpiexec -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
 echo "Job finished at: `date`"
 exit
