# Fix hardcoded paths in python code
#
# Åke Sandgren, 2017-02-07
diff -ru opt.orig/lumerical/fdtd/api/python/lumapi.py opt/lumerical/fdtd/api/python/lumapi.py
--- opt.orig/lumerical/fdtd/api/python/lumapi.py	2016-09-10 02:54:25.000000000 +0200
+++ opt/lumerical/fdtd/api/python/lumapi.py	2017-02-02 12:54:37.413273068 +0100
@@ -1,7 +1,8 @@
 from ctypes import *
 from numpy import *
+import os
 
-INTEROPLIB = "/opt/lumerical/interconnect/api/c/libinterop-api.so.1"
+INTEROPLIB = os.environ['EBROOTFDTD_SOLUTIONS'] + "/api/c/libinterop-api.so.1"
 
 class Session(Structure):
     _fields_ = [("p", c_void_p)]
