# Fix hardcoded paths
#
# Åke Sandgren, 2017-02-07
diff -ru opt.orig/lumerical/fdtd/mpich/ch_gm/bin/mpirun opt/lumerical/fdtd/mpich/ch_gm/bin/mpirun
--- opt.orig/lumerical/fdtd/mpich/ch_gm/bin/mpirun	2016-09-10 02:54:25.000000000 +0200
+++ opt/lumerical/fdtd/mpich/ch_gm/bin/mpirun	2017-01-31 16:54:11.454196201 +0100
@@ -26,8 +26,8 @@
 COMM=
 GLOBUSDIR=@GLOBUSDIR@
 CLINKER="gcc"
-prefix=/opt/lumerical/fdtd/mpich/ch_gm
-bindir=/opt/lumerical/fdtd/mpich/ch_gm/bin
+prefix=$EBROOTFDTD_SOLUTIONS/mpich/ch_gm
+bindir=$EBROOTFDTD_SOLUTIONS/mpich/ch_gm/bin
 # Record the value of sbindir.  Some useful utilities, such as 
 # cleanipcs (used for cleaning up shared memory segments and semaphores)
 # are placed in this directory.
@@ -36,7 +36,7 @@
 sbindir=${exec_prefix}/sbin
 # This value for datadir is the default value setup by configure
 # This value for datadir is the default value setup by configure
-datadir=/opt/lumerical/fdtd/mpich/ch_gm/share
+datadir=$EBROOTFDTD_SOLUTIONS/mpich/ch_gm/share
 DEFAULT_MACHINE=ch_gm
 DEFAULT_ARCH=LINUX
 
diff -ru opt.orig/lumerical/fdtd/mpich/ch_p4/bin/mpirun opt/lumerical/fdtd/mpich/ch_p4/bin/mpirun
--- opt.orig/lumerical/fdtd/mpich/ch_p4/bin/mpirun	2016-09-10 02:54:25.000000000 +0200
+++ opt/lumerical/fdtd/mpich/ch_p4/bin/mpirun	2017-01-31 16:54:21.182102363 +0100
@@ -26,8 +26,8 @@
 COMM=
 GLOBUSDIR=@GLOBUSDIR@
 CLINKER="cc"
-prefix=/opt/lumerical/fdtd/mpich/ch_p4
-bindir=/opt/lumerical/fdtd/mpich/ch_p4/bin
+prefix=$EBROOTFDTD_SOLUTIONS/mpich/ch_p4
+bindir=$EBROOTFDTD_SOLUTIONS/mpich/ch_p4/bin
 # Record the value of sbindir.  Some useful utilities, such as 
 # cleanipcs (used for cleaning up shared memory segments and semaphores)
 # are placed in this directory.
@@ -36,7 +36,7 @@
 sbindir=${exec_prefix}/sbin
 # This value for datadir is the default value setup by configure
 # This value for datadir is the default value setup by configure
-datadir=/opt/lumerical/fdtd/mpich/ch_p4/share
+datadir=$EBROOTFDTD_SOLUTIONS/mpich/ch_p4/share
 DEFAULT_MACHINE=ch_p4
 DEFAULT_ARCH=LINUX
 
diff -ru opt.orig/lumerical/fdtd/mpich/ch_shmem/bin/mpirun opt/lumerical/fdtd/mpich/ch_shmem/bin/mpirun
--- opt.orig/lumerical/fdtd/mpich/ch_shmem/bin/mpirun	2016-09-10 02:54:25.000000000 +0200
+++ opt/lumerical/fdtd/mpich/ch_shmem/bin/mpirun	2017-01-31 16:54:28.526031524 +0100
@@ -26,8 +26,8 @@
 COMM=
 GLOBUSDIR=@GLOBUSDIR@
 CLINKER="cc"
-prefix=/opt/lumerical/fdtd/mpich/ch_shmem
-bindir=/opt/lumerical/fdtd/mpich/ch_shmem/bin
+prefix=$EBROOTFDTD_SOLUTIONS/mpich/ch_shmem
+bindir=$EBROOTFDTD_SOLUTIONS/mpich/ch_shmem/bin
 # Record the value of sbindir.  Some useful utilities, such as 
 # cleanipcs (used for cleaning up shared memory segments and semaphores)
 # are placed in this directory.
@@ -36,7 +36,7 @@
 sbindir=${exec_prefix}/sbin
 # This value for datadir is the default value setup by configure
 # This value for datadir is the default value setup by configure
-datadir=/opt/lumerical/fdtd/mpich/ch_shmem/share
+datadir=$EBROOTFDTD_SOLUTIONS/mpich/ch_shmem/share
 DEFAULT_MACHINE=smp
 DEFAULT_ARCH=LINUX
 
diff -ru opt.orig/lumerical/fdtd/templates/fdtd-pbs-template.sh opt/lumerical/fdtd/templates/fdtd-pbs-template.sh
--- opt.orig/lumerical/fdtd/templates/fdtd-pbs-template.sh	2016-09-10 02:54:24.000000000 +0200
+++ opt/lumerical/fdtd/templates/fdtd-pbs-template.sh	2017-01-31 16:55:33.917400758 +0100
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/fdtd/bin/fdtd-engine-mpichp4"
+MY_PROG="$EBROOTFDTD_SOLUTIONS/bin/fdtd-engine-mpichp4"
 INPUT="<filename>"
-/opt/lumerical/fdtd/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG ./${INPUT}
+$EBROOTFDTD_SOLUTIONS/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG ./${INPUT}
 echo "Job finished at: `date`"
 exit
