diff -Nru fmt-258194623141a3f4293a70b4388d52362cafc93a.orig/CMakeLists.txt fmt-258194623141a3f4293a70b4388d52362cafc93a/CMakeLists.txt
--- fmt-258194623141a3f4293a70b4388d52362cafc93a.orig/CMakeLists.txt	2021-05-17 17:22:22.506058000 +0100
+++ fmt-258194623141a3f4293a70b4388d52362cafc93a/CMakeLists.txt	2021-05-17 17:40:50.458887139 +0100
@@ -174,7 +174,9 @@
 if (WIN32)
   check_symbol_exists(_strtod_l "${strtod_l_headers}" HAVE_STRTOD_L)
 else ()
+  set(CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
   check_symbol_exists(strtod_l "${strtod_l_headers}" HAVE_STRTOD_L)
+  unset(CMAKE_REQUIRED_DEFINITIONS)
 endif ()
 
 function(add_headers VAR)
