diff -Nru openmc-0.13.2.orig/CMakeLists.txt openmc-0.13.2/CMakeLists.txt
--- openmc-0.13.2.orig/CMakeLists.txt	2022-12-19 10:18:51.950463066 -0500
+++ openmc-0.13.2/CMakeLists.txt	2022-12-19 10:19:31.456736753 -0500
@@ -210,12 +210,6 @@
   endif()
 endif()
 
-# Check to see if submodules exist (by checking one)
-if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/vendor/pugixml/CMakeLists.txt")
-  message(FATAL_ERROR "The git submodules were not downloaded! GIT_SUBMODULE was \
-    turned off or failed. Please update submodules and try again.")
-endif()
-
 #===============================================================================
 # pugixml library
 #===============================================================================
