diff -Nru openmc-0.12.1.orig/CMakeLists.txt openmc-0.12.1/CMakeLists.txt
--- openmc-0.12.1.orig/CMakeLists.txt	2021-05-17 13:33:38.405781000 +0100
+++ openmc-0.12.1/CMakeLists.txt	2021-05-18 10:56:34.335295741 +0100
@@ -173,7 +173,7 @@
 
 if (NOT pugixml_FOUND)
   add_subdirectory(vendor/pugixml)
-  set_target_properties(pugixml PROPERTIES CXX_STANDARD 14 CXX_EXTENSIONS OFF)
+  #  set_target_properties(pugixml PROPERTIES CXX_STANDARD 14 CXX_EXTENSIONS OFF)
 endif()
 
 #===============================================================================
