diff -Nru pugixml-5c4225b5c90c67307f77e829dcea7fbb1df5c97a.orig/CMakeLists.txt pugixml-5c4225b5c90c67307f77e829dcea7fbb1df5c97a/CMakeLists.txt
--- pugixml-5c4225b5c90c67307f77e829dcea7fbb1df5c97a.orig/CMakeLists.txt	2021-05-17 15:26:25.679931000 +0100
+++ pugixml-5c4225b5c90c67307f77e829dcea7fbb1df5c97a/CMakeLists.txt	2021-05-18 12:00:35.620592000 +0100
@@ -40,7 +40,7 @@
 endif()
 
 if (NOT DEFINED CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
 
 if (USE_POSTFIX)
