diff -ru kahypar-1.3.2.old/tests/end_to_end/kahypar_integration_tests.cc kahypar-1.3.2/tests/end_to_end/kahypar_integration_tests.cc
--- kahypar-1.3.2.old/tests/end_to_end/kahypar_integration_tests.cc	2023-06-19 17:11:26.235741348 +0000
+++ kahypar-1.3.2/tests/end_to_end/kahypar_integration_tests.cc	2023-06-19 17:14:48.837313554 +0000
@@ -32,7 +32,7 @@
 
 namespace kahypar {
 TEST_F(KaHyParK, ComputesDirectKwayCutPartitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_alenex17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_alenex17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
@@ -59,7 +59,7 @@
 }
 
 TEST_F(KaHyParK, ComputesDirectKwayKm1Partitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_alenex17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_alenex17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -87,7 +87,7 @@
 
 
 TEST_F(KaHyParR, ComputesRecursiveBisectionCutPartitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/cut_rb_alenex16.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/cut_rb_alenex16.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
@@ -114,7 +114,7 @@
 }
 
 TEST_F(KaHyParR, ComputesRecursiveBisectionKm1Partitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/cut_rb_alenex16.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/cut_rb_alenex16.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -141,7 +141,7 @@
 }
 
 TEST_F(KaHyParCA, ComputesDirectKwayKm1Partitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_sea17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_sea17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -168,7 +168,7 @@
 }
 
 TEST_F(KaHyParCA, HandlesIndividualBlockWeights) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_sea17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_sea17.ini");
   context.partition.k = 6;
 
   context.partition.epsilon = 0;
@@ -208,7 +208,7 @@
 }
 
 TEST_F(KaHyParBP, ComputesBalancedSolutionWithNodeWeights) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_sea17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_sea17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -237,7 +237,7 @@
 }
 
 TEST_F(KaHyParBP, ComputesBalancedSolutionForIndividualBlockWeights) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_sea17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_sea17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -287,7 +287,7 @@
   context.evolutionary.replace_strategy = EvoReplaceStrategy::diverse;
   context.partition.quiet_mode = true;
   context.partition_evolutionary = true;
-  context.partition.graph_filename = "../../../tests/partition/evolutionary/TestHypergraph";
+  context.partition.graph_filename = "../../../kahypar-1.3.2/tests/partition/evolutionary/TestHypergraph";
   Hypergraph hypergraph(
     kahypar::io::createHypergraphFromFile(context.partition.graph_filename,
                                           context.partition.k));
@@ -309,7 +309,7 @@
 
 TEST(KaHyPar, SupportsIndividualBlockWeightsViaInterface) {
   kahypar_context_t* context = kahypar_context_new();
-  kahypar_configure_context_from_file(context, "../../../config/old_reference_configs/km1_direct_kway_sea17.ini");
+  kahypar_configure_context_from_file(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_sea17.ini");
 
   reinterpret_cast<kahypar::Context*>(context)->preprocessing.enable_community_detection = false;
   reinterpret_cast<kahypar::Context*>(context)->initial_partitioning.bp_algo = BinPackingAlgorithm::worst_fit;
@@ -374,14 +374,14 @@
 
 TEST(KaHyPar, ComputesBalancedSolutionForHypergraphsWithFixedVerticesAndIndividualBlockWeights) {
   Context context;
-  parseIniToContext(context, "../../../config/km1_kKaHyPar_sea20.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/km1_kKaHyPar_sea20.ini");
   context.partition.k = 2;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
   context.partition.use_individual_part_weights = true;
   context.partition.max_part_weights = { 466, 1875 };
-  context.partition.graph_filename = "../../../tests/end_to_end/test_instances/example_hypergraph_fixed_vertex_partitioning_with_custom_block_weights.hgr";
-  context.partition.fixed_vertex_filename = "../../../tests/end_to_end/test_instances/fixed_vertices_for_example_hypergraph_fixed_vertex_partitioning_with_custom_block_weights.fix";
+  context.partition.graph_filename = "../../../kahypar-1.3.2/tests/end_to_end/test_instances/example_hypergraph_fixed_vertex_partitioning_with_custom_block_weights.hgr";
+  context.partition.fixed_vertex_filename = "../../../kahypar-1.3.2/tests/end_to_end/test_instances/fixed_vertices_for_example_hypergraph_fixed_vertex_partitioning_with_custom_block_weights.fix";
 
   kahypar::Randomize::instance().setSeed(context.partition.seed);
 
diff -ru kahypar-1.3.2.old/tests/end_to_end/kahypar_test_fixtures.h kahypar-1.3.2/tests/end_to_end/kahypar_test_fixtures.h
--- kahypar-1.3.2.old/tests/end_to_end/kahypar_test_fixtures.h	2023-06-19 17:11:26.236741356 +0000
+++ kahypar-1.3.2/tests/end_to_end/kahypar_test_fixtures.h	2023-06-19 17:12:44.311347221 +0000
@@ -195,7 +195,7 @@
     context.evolutionary.replace_strategy = EvoReplaceStrategy::diverse;
     context.partition.quiet_mode = false;
     context.partition_evolutionary = true;
-    context.partition.graph_filename = "../../../tests/partition/evolutionary/TestHypergraph";
+    context.partition.graph_filename = "../../../kahypar-1.3.2/tests/partition/evolutionary/TestHypergraph";
 
     kahypar::Randomize::instance().setSeed(context.partition.seed);
   }
diff -ru kahypar-1.3.2.old/tests/interface/interface_test.cc kahypar-1.3.2/tests/interface/interface_test.cc
--- kahypar-1.3.2.old/tests/interface/interface_test.cc	2023-06-19 17:11:26.254741495 +0000
+++ kahypar-1.3.2/tests/interface/interface_test.cc	2023-06-19 17:14:48.835313538 +0000
@@ -35,7 +35,7 @@
 TEST(KaHyPar, CanBeCalledViaInterface) {
   kahypar_context_t* context = kahypar_context_new();
 
-  kahypar_configure_context_from_file(context, "../../../config/km1_kKaHyPar_sea20.ini");
+  kahypar_configure_context_from_file(context, "../../../kahypar-1.3.2/config/km1_kKaHyPar_sea20.ini");
 
   const kahypar_hypernode_id_t num_vertices = 7;
   const kahypar_hyperedge_id_t num_hyperedges = 4;
@@ -162,7 +162,7 @@
   ASSERT_FALSE(hypergraph.isFixedVertex(6));
 
   kahypar_context_t* context = kahypar_context_new();
-  kahypar_configure_context_from_file(context, "../../../config/km1_kKaHyPar_sea20.ini");
+  kahypar_configure_context_from_file(context, "../../../kahypar-1.3.2/config/km1_kKaHyPar_sea20.ini");
 
   std::vector<kahypar_partition_id_t> partition(num_vertices, -1);
   kahypar_partition_hypergraph(kahypar_hypergraph, k, imbalance, &objective, context, partition.data());
@@ -217,7 +217,7 @@
 
 TEST(KaHyPar, CanImprovePartitionsViaInterface) {
   kahypar_context_t* context = kahypar_context_new();
-  kahypar_configure_context_from_file(context, "../../../config/km1_kKaHyPar_sea20.ini");
+  kahypar_configure_context_from_file(context, "../../../kahypar-1.3.2/config/km1_kKaHyPar_sea20.ini");
 
   // lower contraction limit to enforce contractions
   reinterpret_cast<kahypar::Context*>(context)->coarsening.contraction_limit_multiplier = 1;
@@ -371,7 +371,7 @@
 
 TEST(KaHyPar, SupportsIndividualBlockWeightsViaInterface) {
   kahypar_context_t* context = kahypar_context_new();
-  kahypar_configure_context_from_file(context, "../../../config/km1_kKaHyPar_sea20.ini");
+  kahypar_configure_context_from_file(context, "../../../kahypar-1.3.2/config/km1_kKaHyPar_sea20.ini");
 
   reinterpret_cast<kahypar::Context*>(context)->preprocessing.enable_community_detection = false;
   reinterpret_cast<kahypar::Context*>(context)->initial_partitioning.enable_early_restart = false;
@@ -634,7 +634,7 @@
 TEST(Seed, CanBeSetViaLibraryInterface) {
   kahypar_context_t* context = kahypar_context_new();
 
-  kahypar_configure_context_from_file(context, "../../../config/km1_kKaHyPar_sea20.ini");
+  kahypar_configure_context_from_file(context, "../../../kahypar-1.3.2/config/km1_kKaHyPar_sea20.ini");
 
   kahypar_set_seed(context, 42);
 
diff -ru kahypar-1.3.2.old/tests/partition/coarsening/full_vertex_pair_coarsener_test.cc kahypar-1.3.2/tests/partition/coarsening/full_vertex_pair_coarsener_test.cc
--- kahypar-1.3.2.old/tests/partition/coarsening/full_vertex_pair_coarsener_test.cc	2023-06-19 17:11:26.301741860 +0000
+++ kahypar-1.3.2/tests/partition/coarsening/full_vertex_pair_coarsener_test.cc	2023-06-19 17:15:06.117447651 +0000
@@ -128,7 +128,7 @@
   Context context;
   context.coarsening.max_allowed_node_weight = 3;
   kahypar::Randomize::instance().setSeed(context.partition.seed);
-  std::string graph_file("../../../../special_instances/bad_for_ec.hgr");
+  std::string graph_file("../../../../kahypar-1.3.2/special_instances/bad_for_ec.hgr");
   HypernodeID num_hypernodes;
   HyperedgeID num_hyperedges;
   io::readHypergraphFile(graph_file, num_hypernodes, num_hyperedges, index_vector, edge_vector);
diff -ru kahypar-1.3.2.old/tests/partition/evolutionary/combine_test.cc kahypar-1.3.2/tests/partition/evolutionary/combine_test.cc
--- kahypar-1.3.2.old/tests/partition/evolutionary/combine_test.cc	2023-06-19 17:11:26.311741938 +0000
+++ kahypar-1.3.2/tests/partition/evolutionary/combine_test.cc	2023-06-19 17:14:48.839313569 +0000
@@ -52,7 +52,7 @@
 std::vector<PartitionID> parent1 { 0, 1, 0, 1, 0, 1, 0, 1 };
 std::vector<PartitionID> parent2 { 0, 1, 1, 0, 0, 1, 1, 0 };
 TEST_F(ACombine, RespectsItsParents) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 2;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -91,7 +91,7 @@
   ASSERT_EQ(result.fitness(), 6);
 }
 TEST_F(ACombine, TakesTheBetterParent) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 2;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
diff -ru kahypar-1.3.2.old/tests/partition/evolutionary/evo_partitioner_test.cc kahypar-1.3.2/tests/partition/evolutionary/evo_partitioner_test.cc
--- kahypar-1.3.2.old/tests/partition/evolutionary/evo_partitioner_test.cc	2023-06-19 17:11:26.312741945 +0000
+++ kahypar-1.3.2/tests/partition/evolutionary/evo_partitioner_test.cc	2023-06-19 17:14:48.845313616 +0000
@@ -43,7 +43,7 @@
                HyperedgeVector { 0, 1, 2, 3, 4, 5 })
   {
     hypergraph.changeK(2);
-    parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+    parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
     context.partition.k = 2;
     context.partition.epsilon = 0.03;
     context.partition.objective = Objective::cut;
diff -ru kahypar-1.3.2.old/tests/partition/evolutionary/mutation_test.cc kahypar-1.3.2/tests/partition/evolutionary/mutation_test.cc
--- kahypar-1.3.2.old/tests/partition/evolutionary/mutation_test.cc	2023-06-19 17:11:26.314741961 +0000
+++ kahypar-1.3.2/tests/partition/evolutionary/mutation_test.cc	2023-06-19 17:14:48.841313585 +0000
@@ -49,7 +49,7 @@
 };
 
 TEST_F(AMutation, IsPerformingVcyclesCorrectly) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 2;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
@@ -75,7 +75,7 @@
   ASSERT_EQ(ind2.partition().at(3), ind2.partition().at(5));
 }
 TEST_F(AMutation, IsPerformingVcyclesNewIPCorrectly) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 2;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
diff -ru kahypar-1.3.2.old/tests/partition/evolutionary/population_test.cc kahypar-1.3.2/tests/partition/evolutionary/population_test.cc
--- kahypar-1.3.2.old/tests/partition/evolutionary/population_test.cc	2023-06-19 17:11:26.314741961 +0000
+++ kahypar-1.3.2/tests/partition/evolutionary/population_test.cc	2023-06-19 17:14:48.843313600 +0000
@@ -53,14 +53,14 @@
 };
 
 TEST_F(APopulation, IsCorrectlyGeneratingIndividuals) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
   context.partition.mode = Mode::direct_kway;
   context.initial_partitioning.bp_algo = BinPackingAlgorithm::worst_fit;
   context.local_search.algorithm = RefinementAlgorithm::kway_fm;
-  context.partition.graph_filename = "../../../../tests/partition/evolutionary/TestHypergraph";
+  context.partition.graph_filename = "../../../../kahypar-1.3.2/tests/partition/evolutionary/TestHypergraph";
   Hypergraph hypergraph(
     kahypar::io::createHypergraphFromFile(context.partition.graph_filename,
                                           context.partition.k));
@@ -71,7 +71,7 @@
   ASSERT_EQ(population.size(), 2);
 }
 TEST_F(APopulation, IsCorrectlyReplacingWithDiverseStrategy) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 4;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
@@ -141,7 +141,7 @@
   ASSERT_EQ(population.best(), 1);
 }
 TEST_F(APopulation, IsCorrectlyReplacingWithStrongDiverseStrategy) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 4;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -240,7 +240,7 @@
   ASSERT_EQ(population.difference(ind1, 0, true), 0);
 }
 TEST_F(APopulation, IsPerformingTournamentSelection) {
-  parseIniToContext(context, "../../../../config/old_reference_configs/km1_direct_kway_gecco18.ini");
+  parseIniToContext(context, "../../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_gecco18.ini");
   context.partition.k = 4;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
diff -ru kahypar-1.3.2.old/tests/regression/kahypar_regression_tests.cc kahypar-1.3.2/tests/regression/kahypar_regression_tests.cc
--- kahypar-1.3.2.old/tests/regression/kahypar_regression_tests.cc	2023-06-19 17:11:26.371742403 +0000
+++ kahypar-1.3.2/tests/regression/kahypar_regression_tests.cc	2023-06-19 17:14:48.847313631 +0000
@@ -56,7 +56,7 @@
 }
 
 TEST_F(KaHyParK, ComputesDirectKwayCutPartitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_alenex17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_alenex17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
@@ -73,7 +73,7 @@
 }
 
 TEST_F(KaHyParK, ComputesDirectKwayKm1Partitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_alenex17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_alenex17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -91,7 +91,7 @@
 
 
 TEST_F(KaHyParR, ComputesRecursiveBisectionCutPartitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/cut_rb_alenex16.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/cut_rb_alenex16.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::cut;
@@ -108,7 +108,7 @@
 }
 
 TEST_F(KaHyParR, ComputesRecursiveBisectionKm1Partitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/cut_rb_alenex16.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/cut_rb_alenex16.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -125,7 +125,7 @@
 }
 
 TEST_F(KaHyParCA, ComputesDirectKwayKm1Partitioning) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_sea17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_sea17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.objective = Objective::km1;
@@ -146,7 +146,7 @@
 }
 
 TEST_F(KaHyParCA, UsesSparsificationAndCommunityDetection) {
-  parseIniToContext(context, "../../../config/old_reference_configs/km1_direct_kway_sea17.ini");
+  parseIniToContext(context, "../../../kahypar-1.3.2/config/old_reference_configs/km1_direct_kway_sea17.ini");
   context.partition.k = 8;
   context.partition.epsilon = 0.03;
   context.partition.seed = 10;
