diff -ru kahypar-1.3.2.old/CMakeLists.txt kahypar-1.3.2/CMakeLists.txt
--- kahypar-1.3.2.old/CMakeLists.txt	2023-04-08 15:08:04.000000000 -0400
+++ kahypar-1.3.2/CMakeLists.txt	2023-12-04 09:29:26.257858310 -0500
@@ -17,9 +17,7 @@
 message(STATUS "Found Threads: ${CMAKE_THREAD_LIBS_INIT}")
 
 #ensure that gmock is built before tests are executed
-add_subdirectory(external_tools/googletest EXCLUDE_FROM_ALL)
-include_directories(SYSTEM ${gtest_SOURCE_DIR}/include ${gtest_SOURCE_DIR})
-include_directories(SYSTEM ${gtest_SOURCE_DIR}/../googlemock/include ${gtest_SOURCE_DIR}/../googlemock/)
+find_package(GTest REQUIRED)
 
 if(NOT CMAKE_BUILD_TYPE OR 
   ((NOT CMAKE_BUILD_TYPE STREQUAL "Debug") AND 
@@ -33,6 +31,7 @@
 
 # add third-party algorithms to project
 include_directories(external_tools)
+include_directories(..) #for WHFC
 
 option(KAHYPAR_PYTHON_INTERFACE
   "Enable building the python interface" OFF)
@@ -259,9 +258,8 @@
     # M1 doesn't seem to support -mtune=native -march=native
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g3 ")
   else()
-    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -mtune=native -march=native")
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
-    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -mtune=native -march=native -g3 ")
   endif()
 
 if(DEFINED ENV{TRAVIS_ENV})  
diff -ru kahypar-1.3.2.old/python/CMakeLists.txt kahypar-1.3.2/python/CMakeLists.txt
--- kahypar-1.3.2.old/python/CMakeLists.txt	2023-04-08 15:08:04.000000000 -0400
+++ kahypar-1.3.2/python/CMakeLists.txt	2023-12-04 09:27:26.318580579 -0500
@@ -10,7 +10,7 @@
 #file(GLOB_RECURSE KAHYPAR_SOURCES RELATIVE ${PROJECT_SOURCE_DIR} *.h)
 
 
-add_subdirectory(pybind11)
+find_package(pybind11 REQUIRED)
 include_directories(${PROJECT_SOURCE_DIR})
 pybind11_add_module(kahypar_python module.cpp)
 target_link_libraries(kahypar_python PRIVATE ${Boost_LIBRARIES})
