diff -ruN HiC-Pro-2.9.0/config-system.txt HiC-Pro-2.9.0-patched/config-system.txt
--- HiC-Pro-2.9.0/config-system.txt	1970-01-01 00:00:00.000000000 +0000
+++ HiC-Pro-2.9.0-patched/config-system.txt	2017-09-12 13:50:13.261855000 +0000
@@ -0,0 +1,10 @@
+#########################################################################
+## Paths and Settings  - Start editing here !
+#########################################################################
+
+BOWTIE2_PATH = $EBROOTBOWTIE2/bin 
+SAMTOOLS_PATH = $EBROOTSAMTOOLS/bin
+R_PATH = $EBROOTR/bin 
+PYTHON_PATH = $EBROOTPYTHON/bin
+INSTALL_PATH = $EBROOTHICMINPRO
+CLUSTER_SCRIPT = $$EBROOTHICMINPRO/scripts/make_torque_script.sh
diff -ruN HiC-Pro-2.9.0/Makefile HiC-Pro-2.9.0-patched/Makefile
--- HiC-Pro-2.9.0/Makefile	2017-06-20 09:40:24.000000000 +0000
+++ HiC-Pro-2.9.0-patched/Makefile	2017-09-04 15:23:50.962316000 +0000
@@ -15,7 +15,7 @@
 CONFIG_SYS=$(wildcard ./config-install.txt)
 
 
-install : config_check mapbuilder readstrimming iced cp
+install : mapbuilder readstrimming 
 
 ######################################
 ## Config file
diff -ruN HiC-Pro-2.9.0/scripts/install/Makefile HiC-Pro-2.9.0-patched/scripts/install/Makefile
--- HiC-Pro-2.9.0/scripts/install/Makefile	2017-06-20 09:40:24.000000000 +0000
+++ HiC-Pro-2.9.0-patched/scripts/install/Makefile	2017-09-04 15:24:51.644635000 +0000
@@ -34,5 +34,4 @@
 ##
 ######################################
 configure: config_check
-	./scripts/install/install_dependencies.sh -c $(CONFIG_SYS) -o  $(realpath $(PREFIX))/HiC-Pro_$(VNUM)
