Fix the paths to the EB prefix

Åke Sandgren, 20250310
diff -ru HiC-Pro-3.1.0.orig/bin/HiC-Pro HiC-Pro-3.1.0/bin/HiC-Pro
--- HiC-Pro-3.1.0.orig/bin/HiC-Pro	2021-08-27 20:04:54.000000000 +0200
+++ HiC-Pro-3.1.0/bin/HiC-Pro	2025-03-10 08:00:11.795996793 +0100
@@ -56,10 +56,10 @@
 }
 
 ## Set PATHS
-BIN_PATH=`dirname $0`
-ABS_BIN_PATH=`cd "$BIN_PATH"; pwd`
-SCRIPTS_PATH="$ABS_BIN_PATH/../scripts/"
-INSTALL_PATH="$ABS_BIN_PATH/../"
+BIN_PATH=$EBROOTHICMINPRO/bin
+ABS_BIN_PATH=$BIN_PATH
+SCRIPTS_PATH=$EBROOTHICMINPRO/scripts/
+INSTALL_PATH=$EBROOTHICMINPRO
 CUR_PATH=$PWD
 
 CLUSTER=0
diff -ru HiC-Pro-3.1.0.orig/scripts/install/Makefile HiC-Pro-3.1.0/scripts/install/Makefile
--- HiC-Pro-3.1.0.orig/scripts/install/Makefile	2021-08-27 20:04:54.000000000 +0200
+++ HiC-Pro-3.1.0/scripts/install/Makefile	2025-03-10 08:15:15.124324171 +0100
@@ -38,6 +38,6 @@
 ##
 ######################################
 configure: config_check
-	./scripts/install/install_dependencies.sh -c $(CONFIG_SYS) -p ${PREFIX} -o $(PREFIX)/HiC-Pro_$(VNUM) -q
+	./scripts/install/install_dependencies.sh -c $(CONFIG_SYS) -p ${PREFIX} -o $(PREFIX) -q
 
 
