Replace Intel MKL subroutines with FlexiBLAS wrappers
author: Alex Domingo (Vrije Universiteit Brussel)
--- a/executables/ICPT.cpp	2022-10-26 01:45:54.000000000 +0200
+++ b/executables/ICPT.cpp	2022-11-26 11:23:28.845637000 +0100
@@ -7,8 +7,7 @@
 #include "BlockContract.h"
 #include "CxDefs.h"
 #include "icpt.h"
-#include "mkl.h"
-#include "mkl_cblas.h"
+#include "flexiblas_api.h"
 using ct::TArray;
 using ct::FMemoryStack;
 using boost::format;
@@ -25,7 +24,7 @@
    //   - something to split up tensors (H) and combine them (RDM)
    //     regarding universal indices
 
-  mkl_set_num_threads(numthrds);
+  flexiblas_set_num_threads(numthrds);
 
    FJobContext
       Job;
