fix to use external PETSc
author: Kenneth Hoste (HPC-UGent)
diff -ru cmake/modules/FindPETSc.cmake cmake/modules/FindPETSc.cmake
--- cmake/modules/FindPETSc.cmake	2011-12-07 22:21:22.000000000 +0100
+++ cmake/modules/FindPETSc.cmake	2012-08-10 13:02:21.032955197 +0200
@@ -114,9 +114,10 @@
   petsc_get_variable(PETSC_INCLUDE PETSC_INCLUDE)          # 3.1
   petsc_get_variable(PETSC_CC_INCLUDES PETSC_CC_INCLUDES)  # dev
   set(PETSC_INCLUDE ${PETSC_INCLUDE} ${PETSC_CC_INCLUDES})
-  petsc_get_variable(PETSC_LIB_BASIC PETSC_LIB_BASIC)
+  petsc_get_variable(PETSC_WITH_EXTERNAL_LIB PETSC_WITH_EXTERNAL_LIB)
   petsc_get_variable(PETSC_LIB_DIR PETSC_LIB_DIR)
-  set(PETSC_LIB "-L${PETSC_LIB_DIR} ${PETSC_LIB_BASIC}")
+  set(PETSC_LIB "-L${PETSC_LIB_DIR} ${PETSC_WITH_EXTERNAL_LIB}")
+  message(STATUS "PETSC_LIB ${PETSC_LIB}")
 
   # Remove temporary Makefile
   file(REMOVE ${petsc_config_makefile})
