--- dolfin/CMakeLists.txt.orig	2012-03-27 16:40:12.595869000 +0200
+++ dolfin/CMakeLists.txt	2012-03-27 17:22:38.628180000 +0200
@@ -266,9 +266,14 @@
     "${_lib}"
     )
 
-  # Only add libraries that matches the form -L<libdir> -l<lib>
-  if ("${_linkflags}" MATCHES "-L.+ -l.+")
-    set(PKG_LINKFLAGS "${_linkflags} ${PKG_LINKFLAGS}")
+  #also add -Wl,x,x,x,x directives
+  if ("${_lib}" MATCHES "-Wl,[^ ]*")
+     set(PKG_LINKFLAGS "${_lib} ${PKG_LINKFLAGS}")
+  else()
+    # Only add libraries that matches the form -L<libdir> -l<lib>
+    if ("${_linkflags}" MATCHES "-L.+ -l.+")
+        set(PKG_LINKFLAGS "${_linkflags} ${PKG_LINKFLAGS}")
+    endif()
   endif()
 endforeach()
