--- rnammer.orig	2017-05-17 17:27:41.069294041 +0000
+++ rnammer	2017-05-17 18:56:39.907885733 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#! /usr/bin/env perl
 # VERSION 1.2
 my $version = '1.2';
 # 1.1b: config parameter 'multi' introduced to allow parallel execution
@@ -32,7 +32,7 @@
 ## PROGRAM CONFIGURATION BEGIN
 
 # the path of the program
-my $INSTALL_PATH = "/usr/cbs/bio/src/rnammer-1.2";
+my $INSTALL_PATH = "$ENV{'EBROOTRNAMMER'}";
 
 # The library in which HMMs can be found
 my $HMM_LIBRARY = "$INSTALL_PATH/lib";
@@ -47,8 +47,8 @@
 my $HMMSEARCH_BINARY;
 my $PERL;
 if ( $uname eq "Linux" ) {
-	$HMMSEARCH_BINARY = "/usr/cbs/bio/bin/linux64/hmmsearch";
-	$PERL = "/usr/bin/perl";
+	$HMMSEARCH_BINARY = "$ENV{'EBROOTHMMER'}/bin/hmmsearch";
+	$PERL = "$ENV{'EBROOTPERL'}/bin/perl";
 } elsif ( $uname eq "IRIX64" ) {
 	$HMMSEARCH_BINARY = "/usr/cbs/bio/bin/irix64/hmmsearch";
 	$PERL = "/usr/sbin/perl";
