# avoid hard-coding of c++ standard which breaks Boost compatibility
--- manta-1.6.0/src/cmake/cxxConfigure.cmake	2019-06-24 20:02:16.000000000 +0000
+++ manta-1.6.0.new/src/cmake/cxxConfigure.cmake	2025-01-13 17:22:16.298833605 +0000
@@ -432,11 +432,11 @@
 # other customizations
 #
 if (${GNU_COMPAT_COMPILER})
-    if ((NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel") OR (${COMPILER_VERSION} VERSION_LESS "15.0"))
-        append_args (CMAKE_CXX_FLAGS "-std=c++0x")
-    else ()
-        append_args (CMAKE_CXX_FLAGS "-std=c++11")
-    endif ()
+    #if ((NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel") OR (${COMPILER_VERSION} VERSION_LESS "15.0"))
+    #    append_args (CMAKE_CXX_FLAGS "-std=c++0x")
+    #else ()
+    #    append_args (CMAKE_CXX_FLAGS "-std=c++11")
+    #endif ()
 
     # The intel compilers make a LOT of remarks by default. Turn these down a little bit here so that more
     # critical info isn't drowned out:
