diff -ru manta-1.2.1.release_src.original/src/cmake/globals.cmake manta-1.2.1.release_src.patched/src/cmake/globals.cmake
--- manta-1.2.1.release_src.original/src/cmake/globals.cmake	2017-10-09 22:52:39.000000000 +0200
+++ manta-1.2.1.release_src.patched/src/cmake/globals.cmake	2018-07-06 22:07:45.000000000 +0200
@@ -27,6 +27,7 @@
 
 set(THIS_EXECUTABLE_PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ OWNER_EXECUTE GROUP_EXECUTE WORLD_EXECUTE)
 set(THIS_LIBRARY_PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ)
+set(THIS_DIRECTORY_PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE SETGID WORLD_READ WORLD_EXECUTE)
 
 install(CODE "
 
diff -ru manta-1.2.1.release_src.original/src/cmake/macros.cmake manta-1.2.1.release_src.patched/src/cmake/macros.cmake
--- manta-1.2.1.release_src.original/src/cmake/macros.cmake	2017-10-09 22:52:39.000000000 +0200
+++ manta-1.2.1.release_src.patched/src/cmake/macros.cmake	2018-07-06 22:09:59.000000000 +0200
@@ -134,6 +134,6 @@
 # TODO: generate py->pyc here as well
 #
 function(install_python_lib_dir fromdir todir)
-    install (DIRECTORY "${fromdir}/" DESTINATION "${todir}" FILE_PERMISSIONS ${THIS_LIBRARY_PERMISSIONS} FILES_MATCHING PATTERN "*.py")
-    install (DIRECTORY "${fromdir}/" DESTINATION "${todir}" FILE_PERMISSIONS ${THIS_EXECUTABLE_PERMISSIONS} FILES_MATCHING PATTERN "*.pyc")
+    install (DIRECTORY "${fromdir}/" DESTINATION "${todir}" FILE_PERMISSIONS ${THIS_LIBRARY_PERMISSIONS} DIRECTORY_PERMISSIONS ${THIS_DIRECTORY_PERMISSIONS} FILES_MATCHING PATTERN "*.py")
+    install (DIRECTORY "${fromdir}/" DESTINATION "${todir}" FILE_PERMISSIONS ${THIS_EXECUTABLE_PERMISSIONS} DIRECTORY_PERMISSIONS ${THIS_DIRECTORY_PERMISSIONS} FILES_MATCHING PATTERN "*.pyc")
 endfunction()
