Taken from Debian package for 3.2.6
Index: gconf-3.2.6/backends/markup-tree.c
===================================================================
--- gconf-3.2.6.orig/backends/markup-tree.c	2013-01-23 11:17:01.995020067 +0100
+++ gconf-3.2.6/backends/markup-tree.c	2013-01-23 11:17:01.991020067 +0100
@@ -71,6 +71,11 @@
 }
 #endif
 
+/* PATH_MAX is not defined in limits.h on some platforms */
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 typedef struct
 {
   char       *locale;
Index: gconf-3.2.6/backends/xml-dir.c
===================================================================
--- gconf-3.2.6.orig/backends/xml-dir.c	2013-01-23 11:17:01.995020067 +0100
+++ gconf-3.2.6/backends/xml-dir.c	2013-01-23 11:17:01.991020067 +0100
@@ -36,6 +36,11 @@
 
 #include "gconf/gconf-internals.h"
 
+/* PATH_MAX is not defined in limits.h on some platforms */
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 /* This makes hash table safer when debugging */
 #ifndef GCONF_ENABLE_DEBUG
 #define safe_g_hash_table_insert g_hash_table_insert
