Description: fix wrong return value
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=582865
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=532119

Index: gconf-3.2.6/gconf/gconf-client.c
===================================================================
--- gconf-3.2.6.orig/gconf/gconf-client.c	2013-01-23 11:17:01.879020062 +0100
+++ gconf-3.2.6/gconf/gconf-client.c	2013-01-23 11:17:01.875020063 +0100
@@ -2373,7 +2373,7 @@
       {
         g_free (dir);
         trace ("Negative cache hit on %s", key);
-        return TRUE;
+        return FALSE;
       }
     else 
       {
@@ -2389,7 +2389,7 @@
               {
                 g_free (dir);
                 trace ("Non-existing dir for %s", key);
-                return TRUE;
+                return FALSE;
               }
             not_cached = TRUE;
           }
