diff -ru freetype-2.6.orig/builds/unix/ftconfig.in freetype-2.6/builds/unix/ftconfig.in
--- freetype-2.6.orig/builds/unix/ftconfig.in	2015-07-21 00:56:16.977488898 +0200
+++ freetype-2.6/builds/unix/ftconfig.in	2015-07-21 00:59:00.216444582 +0200
@@ -363,11 +363,11 @@
 
   /* typeof condition taken from gnulib's `intprops.h' header file */
 #if ( __GNUC__ >= 2                         || \
-      defined( __IBM__TYPEOF__ )            || \
+      defined( __IBM__FT_TYPEOF__ )            || \
       ( __SUNPRO_C >= 0x5110 && !__STDC__ ) )
-#define TYPEOF( type )  (__typeof__ (type))
+#define FT_TYPEOF( type )  (__typeof__ (type))
 #else
-#define TYPEOF( type )  /* empty */
+#define FT_TYPEOF( type )  /* empty */
 #endif
 
 
diff -ru freetype-2.6.orig/builds/vms/ftconfig.h freetype-2.6/builds/vms/ftconfig.h
--- freetype-2.6.orig/builds/vms/ftconfig.h	2015-07-21 00:56:16.979488946 +0200
+++ freetype-2.6/builds/vms/ftconfig.h	2015-07-21 00:59:00.221444704 +0200
@@ -306,11 +306,11 @@
 
   /* typeof condition taken from gnulib's `intprops.h' header file */
 #if ( __GNUC__ >= 2                         || \
-      defined( __IBM__TYPEOF__ )            || \
+      defined( __IBM__FT_TYPEOF__ )            || \
       ( __SUNPRO_C >= 0x5110 && !__STDC__ ) )
-#define TYPEOF( type )  (__typeof__ (type))
+#define FT_TYPEOF( type )  (__typeof__ (type))
 #else
-#define TYPEOF( type )  /* empty */
+#define FT_TYPEOF( type )  /* empty */
 #endif
 
 
diff -ru freetype-2.6.orig/include/config/ftconfig.h freetype-2.6/include/config/ftconfig.h
--- freetype-2.6.orig/include/config/ftconfig.h	2015-07-21 00:56:16.987489141 +0200
+++ freetype-2.6/include/config/ftconfig.h	2015-07-21 00:59:00.222444729 +0200
@@ -333,11 +333,11 @@
 
   /* typeof condition taken from gnulib's `intprops.h' header file */
 #if ( __GNUC__ >= 2                         || \
-      defined( __IBM__TYPEOF__ )            || \
+      defined( __IBM__FT_TYPEOF__ )            || \
       ( __SUNPRO_C >= 0x5110 && !__STDC__ ) )
-#define TYPEOF( type )  (__typeof__ (type))
+#define FT_TYPEOF( type )  (__typeof__ (type))
 #else
-#define TYPEOF( type )  /* empty */
+#define FT_TYPEOF( type )  /* empty */
 #endif
 
 
diff -ru freetype-2.6.orig/include/internal/ftobjs.h freetype-2.6/include/internal/ftobjs.h
--- freetype-2.6.orig/include/internal/ftobjs.h	2015-07-21 00:56:16.981488994 +0200
+++ freetype-2.6/include/internal/ftobjs.h	2015-07-21 00:59:00.222444729 +0200
@@ -83,12 +83,12 @@
             x > y ? x + ( 3 * y >> 3 )   \
                   : y + ( 3 * x >> 3 ) )
 
-  /* we use the TYPEOF macro to suppress signedness compilation warnings */
-#define FT_PAD_FLOOR( x, n )  ( (x) & ~TYPEOF( x )( (n)-1 ) )
+  /* we use the FT_TYPEOF macro to suppress signedness compilation warnings */
+#define FT_PAD_FLOOR( x, n )  ( (x) & ~FT_TYPEOF( x )( (n)-1 ) )
 #define FT_PAD_ROUND( x, n )  FT_PAD_FLOOR( (x) + ((n)/2), n )
 #define FT_PAD_CEIL( x, n )   FT_PAD_FLOOR( (x) + ((n)-1), n )
 
-#define FT_PIX_FLOOR( x )     ( (x) & ~TYPEOF( x )63 )
+#define FT_PIX_FLOOR( x )     ( (x) & ~FT_TYPEOF( x )63 )
 #define FT_PIX_ROUND( x )     FT_PIX_FLOOR( (x) + 32 )
 #define FT_PIX_CEIL( x )      FT_PIX_FLOOR( (x) + 63 )
 
diff -ru freetype-2.6.orig/src/autofit/afwarp.h freetype-2.6/src/autofit/afwarp.h
--- freetype-2.6.orig/src/autofit/afwarp.h	2015-07-21 00:56:16.960488487 +0200
+++ freetype-2.6/src/autofit/afwarp.h	2015-07-21 00:59:00.216444582 +0200
@@ -25,7 +25,7 @@
 
 #define AF_WARPER_SCALE
 
-#define AF_WARPER_FLOOR( x )  ( (x) & ~TYPEOF( x )63 )
+#define AF_WARPER_FLOOR( x )  ( (x) & ~FT_TYPEOF( x )63 )
 #define AF_WARPER_CEIL( x )   AF_WARPER_FLOOR( (x) + 63 )
 
 
