Author: yqshao (Yunqi Shao), Update: Pavel Tománek (Inuits)
Update setup.py to use dependencies from EB
--- setup.py.orig	2024-11-04 12:15:44.508837000 +0100
+++ setup.py	2024-11-04 12:20:12.717679000 +0100
@@ -298,29 +298,33 @@
     CORE_C_FILES = filter(lambda x: "third_party/cares" not in x, CORE_C_FILES)
 
 if BUILD_WITH_SYSTEM_OPENSSL:
+    EBROOTOPENSSL = os.environ.get('EBROOTOPENSSL')
     CORE_C_FILES = filter(
         lambda x: "third_party/boringssl" not in x, CORE_C_FILES
     )
     CORE_C_FILES = filter(lambda x: "src/boringssl" not in x, CORE_C_FILES)
-    SSL_INCLUDE = (os.path.join("/usr", "include", "openssl"),)
+    SSL_INCLUDE = (os.path.join(EBROOTOPENSSL, "include", "openssl"),)
 
 if BUILD_WITH_SYSTEM_ZLIB:
+    EBROOTZLIB = os.environ.get('EBROOTZLIB')
     CORE_C_FILES = filter(lambda x: "third_party/zlib" not in x, CORE_C_FILES)
-    ZLIB_INCLUDE = (os.path.join("/usr", "include"),)
+    ZLIB_INCLUDE = (os.path.join(EBROOTZLIB, "include"),)
 
 if BUILD_WITH_SYSTEM_CARES:
     CORE_C_FILES = filter(lambda x: "third_party/cares" not in x, CORE_C_FILES)
     CARES_INCLUDE = (os.path.join("/usr", "include"),)
 
 if BUILD_WITH_SYSTEM_RE2:
+    EBROOTRE2 = os.environ.get('EBROOTRE2')
     CORE_C_FILES = filter(lambda x: "third_party/re2" not in x, CORE_C_FILES)
-    RE2_INCLUDE = (os.path.join("/usr", "include", "re2"),)
+    RE2_INCLUDE = (os.path.join(EBROOTRE2, "include", "re2"),)
 
 if BUILD_WITH_SYSTEM_ABSL:
+    EBROOTABSEIL = os.environ.get('EBROOTABSEIL')
     CORE_C_FILES = filter(
         lambda x: "third_party/abseil-cpp" not in x, CORE_C_FILES
     )
-    ABSL_INCLUDE = (os.path.join("/usr", "include"),)
+    ABSL_INCLUDE = (os.path.join(EBROOTABSEIL, "include"),)
 
 EXTENSION_INCLUDE_DIRECTORIES = (
     (PYTHON_STEM,)
@@ -363,8 +367,7 @@
     EXTENSION_LIBRARIES += ("re2",)
 if BUILD_WITH_SYSTEM_ABSL:
     EXTENSION_LIBRARIES += tuple(
-        lib.stem[3:]
-        for lib in sorted(pathlib.Path("/usr").glob("lib*/libabsl_*.so"))
+        lib.stem[3:] for lib in pathlib.Path(EBROOTABSEIL).glob("lib*/libabsl_*.so")
     )
 
 DEFINE_MACROS = (("_WIN32_WINNT", 0x600),)
