We only have protobuf and jansson as dynamic libraries.

Author: Petr Král (INUITS)
diff -u vg/Makefile.orig vg/Makefile
--- Makefile.orig       2023-09-13 14:23:25.431047504 +0200
+++ Makefile    2023-09-13 14:25:13.638368889 +0200
@@ -45,11 +45,11 @@
 # What pkg-config-controlled system dependencies should we use compile and link flags from?
 # Use PKG_CONFIG_PATH to point the build system at the right versions of these, if they aren't picked up automatically.
 # We can't do this for our bundled, pkg-config-supporting dependencies (like htslib) because they won't be built yet.
-PKG_CONFIG_DEPS := cairo libzstd
+PKG_CONFIG_DEPS := cairo libzstd protobuf jansson
 # These are like PKG_CONFIG_DEPS but we try to always link them statically, if possible.
 # Note that we then must *always* link anything *else* that uses them statically.
 # Jansson has to be in here because it has to come after libvgio, which is in the static deps.
-PKG_CONFIG_STATIC_DEPS := protobuf jansson
+PKG_CONFIG_STATIC_DEPS :=
 
 # We don't ask for -fopenmp here because how we get it can depend on the compiler.
 # We don't ask for automatic Make dependency file (*.d) generation here because
