Use MinPath from EasyBuild, rather than downloaded one
Author: J. Saßmannshausen (Imperial College London/UK)
diff --git a/humann-3.9.orig/setup.py b/humann-3.9/setup.py
index e31f477..fad2f1d 100644
--- a/humann-3.9.orig/setup.py
+++ b/humann-3.9/setup.py
@@ -277,8 +277,9 @@ def install_minpath(replace_install=None):
     minpath_install_folder="MinPath"
     
     # install minpath if not already installed
-    minpath_exe=os.path.join(fullpath_scripts,minpath_install_folder,"MinPath1.2.py")
-    if not os.path.isfile(minpath_exe) or replace_install:
+    # minpath_exe=os.path.join(fullpath_scripts,minpath_install_folder,"MinPath1.2.py")
+    minpath_exe=find_exe_in_path("MinPath.py")
+    if not minpath_exe or replace_install:
         download_file=os.path.join(fullpath_scripts, minpath_file)
         print("Installing minpath.")
         download_unpack_tar(minpath_url,download_file,fullpath_scripts,"minpath")
