From d82dd9a27e8ae056ddb399fdf42a3d2bc342e4a6 Mon Sep 17 00:00:00 2001
From: Jakob Schiotz <schiotz@fysik.dtu.dk>
Date: Thu, 14 Nov 2019 16:06:18 +0100
Subject: [PATCH] Use MKL instead of FFTW.

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index a884aec..095a1ee 100644
--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ This package requires that the FFTW library has already been installed.
 
 include_dirs = [
     np.get_include(),
-    os.path.join(os.getcwd(), 'fftw'),
+    #os.path.join(os.getcwd(), 'fftw'),
     os.path.join(os.getcwd(), 'source/'),
     ]
 
@@ -49,7 +49,7 @@ elif not is_conda and get_build_platform() == 'win-amd64':
 else:
     # Linux, MacOS and Anaconda on Windows.
     library_dirs = []
-    libraries = ['fftw3','fftw3f']
+    libraries = ['mkl_sequential','mkl_core', 'fftw3xc_intel_pic', 'mkl_rt']
 
 sources = ['memory_fftw3.cpp','data_containers.cpp','imagelib_fftw3.cpp',
           'stemlib.cpp','stemutil.cpp','fileio_fftw3.cpp','matrixlib.cpp','readparams.cpp']
-- 
1.8.3.1

