fix hardcoded statsmodel version
author: Pavel Grochal (INUITS)
--- NanoPlot-1.28.4/setup.py.orig	2020-02-18 08:28:53.000000000 +0100
+++ NanoPlot-1.28.4/setup.py	2020-03-22 17:34:05.929501866 +0100
@@ -41,7 +41,7 @@
                       'nanoget>=1.9.0',
                       'nanomath>=0.23.1',
                       "pauvre==0.1.86",
-                      "statsmodels==0.10.1",
+                      "statsmodels>=0.10.1",
                       'plotly>=4.1.0',
                       ],
     package_data={'NanoPlot': []},
