--- cp2k.orig/cp2k/makefiles/Makefile 2012-08-21 14:35:17.000000000 +0300
+++ cp2k/cp2k/makefiles/Makefile  2013-11-12 12:41:29.361582641 +0200
@@ -194,7 +194,7 @@
 #
 ifeq ($(CPPSHELL),)
 CPPSHELL := -D__COMPILE_ARCH="\"$(ARCH)\""\
-            -D__COMPILE_DATE="\"$(shell date)\""\
+            -D__COMPILE_DATE="\"$(shell date | sed 's/ /_/g')\""\
             -D__COMPILE_HOST="\"$(shell hostname)\""\
        -D__COMPILE_REVISION="\"$(strip $(REVISION))\""
 endif
