--- makefiles/Makefile.orig	2010-12-10 05:35:15.000000000 +0100
+++ makefiles/Makefile	2011-01-24 10:14:35.000000000 +0100
@@ -149,9 +149,9 @@
 #
 ifeq ($(CPPSHELL),)
 CPPSHELL := -D__COMPILE_ARCH="\"$(ARCH)\""\
-            -D__COMPILE_DATE="\"$(shell date)\""\
+            -D__COMPILE_DATE="\"$(shell date | sed 's/ /_/g')\""\
             -D__COMPILE_HOST="\"$(shell hostname)\""\
-            -D__COMPILE_LASTCVS="\"$(shell tail -n1 $(SRCDIR)/CVS/Entries)\""
+            -D__COMPILE_LASTCVS="\"$(shell tail -n1 $(SRCDIR)/CVS/Entries | sed 's/ /_/g')\""
 endif
 
 ifneq ($(CPP),)
