diff -ru cp2k.orig/src/dbcsr_lib/dbcsr_operations.F cp2k/src/dbcsr_lib/dbcsr_operations.F
--- cp2k.orig/src/dbcsr_lib/dbcsr_operations.F  2012-05-25 10:23:39.011852249 +0200
+++ cp2k/src/dbcsr_lib/dbcsr_operations.F   2012-05-25 10:23:50.297006614 +0200
@@ -4512,7 +4512,7 @@
        ngpus = dbcsr_cuda_get_n_devices (error)
        !$OMP PARALLEL default (none), &
        !$OMP          private (t_error) &
-       !$OMP          shared (mynode, ngpus, error)
+       !$OMP          shared (mynode, ngpus, error, has_ma)
        t_error = error
        IF (has_ma) THEN
           CALL dbcsr_cuda_init (card_num=ma_set_gpu_affinity(mynode), error=error)

