Update minimum required version to 3.5 to deal with CMake 4.0

Author: Jan Andre Reuter (JSC)

diff --color -Naur graphite-1.3.14.orig/CMakeLists.txt graphite-1.3.14/CMakeLists.txt
--- graphite-1.3.14.orig/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/CMakeLists.txt	2025-07-11 21:31:35.527318797 +0200
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
 cmake_policy(SET CMP0012 NEW)
 include(TestBigEndian)
 find_package(PythonInterp 3.6)
diff --color -Naur graphite-1.3.14.orig/gr2fonttest/CMakeLists.txt graphite-1.3.14/gr2fonttest/CMakeLists.txt
--- graphite-1.3.14.orig/gr2fonttest/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/gr2fonttest/CMakeLists.txt	2025-07-11 21:31:35.526462048 +0200
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
 
 project(gr2fonttest)
 
diff --color -Naur graphite-1.3.14.orig/src/CMakeLists.txt graphite-1.3.14/src/CMakeLists.txt
--- graphite-1.3.14.orig/src/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/src/CMakeLists.txt	2025-07-11 21:31:35.526531827 +0200
@@ -19,7 +19,7 @@
 #    Suite 500, Boston, MA 02110-1335, USA or visit their web page on the
 #    internet at http://www.fsf.org/licenses/lgpl.html.
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
 project(graphite2_core)
 cmake_policy(SET CMP0012 NEW)
 INCLUDE(CheckCXXSourceCompiles)
diff --color -Naur graphite-1.3.14.orig/tests/bittwiddling/CMakeLists.txt graphite-1.3.14/tests/bittwiddling/CMakeLists.txt
--- graphite-1.3.14.orig/tests/bittwiddling/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/tests/bittwiddling/CMakeLists.txt	2025-07-11 21:31:35.526664806 +0200
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
 project(bits)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
diff --color -Naur graphite-1.3.14.orig/tests/CMakeLists.txt graphite-1.3.14/tests/CMakeLists.txt
--- graphite-1.3.14.orig/tests/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/tests/CMakeLists.txt	2025-07-11 21:31:35.526602976 +0200
@@ -95,7 +95,7 @@
 fonttest(scher3 Scheherazadegr.ttf 0627 0031 0032 002D 0034 0035 0627 -rtl)
 fonttest(scher4 Scheherazadegr.ttf 0627 0653 06AF -rtl)
 fonttest(scher5 Scheherazadegr_noglyfs.ttf 0627 0653 06AF -rtl)
-fonttest(charis1 charis_r_gr.ttf 0069 02E6 02E8 02E5)
+fonttest(charis1 charis_r_gr.ttf 0069 02E6 03.52E5)
 fonttest(charis2 charis_r_gr.ttf 1D510 0041 1D513)
 fonttest(charis3 charis_r_gr.ttf 0054 0069 1ec3 0075 -feat lang=vie)
 fonttest(charis4 charis_r_gr.ttf 006b 0361 070)
diff --color -Naur graphite-1.3.14.orig/tests/json/CMakeLists.txt graphite-1.3.14/tests/json/CMakeLists.txt
--- graphite-1.3.14.orig/tests/json/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/tests/json/CMakeLists.txt	2025-07-11 21:31:35.527037241 +0200
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
 project(jsontest)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
diff --color -Naur graphite-1.3.14.orig/tests/sparsetest/CMakeLists.txt graphite-1.3.14/tests/sparsetest/CMakeLists.txt
--- graphite-1.3.14.orig/tests/sparsetest/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/tests/sparsetest/CMakeLists.txt	2025-07-11 21:31:35.527138889 +0200
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
 project(sparsetest)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
diff --color -Naur graphite-1.3.14.orig/tests/utftest/CMakeLists.txt graphite-1.3.14/tests/utftest/CMakeLists.txt
--- graphite-1.3.14.orig/tests/utftest/CMakeLists.txt	2020-04-01 04:53:13.000000000 +0200
+++ graphite-1.3.14/tests/utftest/CMakeLists.txt	2025-07-11 21:31:35.527189869 +0200
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5 FATAL_ERROR)
 project(utftest)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
