From b242c962f7281276ae022bbff0837ce2394a860c Mon Sep 17 00:00:00 2001
From: Simon Pinches <Simon.Pinches@iter.org>
Date: Tue, 17 Mar 2026 10:06:46 +0100
Subject: [PATCH] Remove explicit dependency upon VTK

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 9de85bf..fea3b34 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,7 +16,6 @@ dynamic = ["version"]
 requires-python = ">=3.8"
 dependencies = [
     "numpy >= 1.19.4",
-    "vtk >= 9.1.0",
     "imas-python[netcdf] >= 2.0.0",
 ]
 
-- 
2.39.3

