--- mumps.orig/examples/Makefile	2022-05-06 15:17:17.945992810 +0200
+++ mumps/examples/Makefile	2022-05-06 15:19:02.502917712 +0200
@@ -29,23 +29,23 @@
 LIBZMUMPS = -L$(libdir) -lzmumps$(PLAT) -lmumps_common$(PLAT)
 
 ssimpletest:   $$@.o
-	$(FL) -o $@ $(OPTL) ssimpletest.o  $(LIBSMUMPS) $(LORDERINGS) $(LIBS) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
+	$(FL) -o $@ $(OPTL) ssimpletest.o  $(LIBSMUMPS) $(LORDERINGS) $(LIBS) $(MPIFLIB) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
 
 dsimpletest:  $$@.o 
-	$(FL) -o $@ $(OPTL) dsimpletest.o  $(LIBDMUMPS) $(LORDERINGS) $(LIBS) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
+	$(FL) -o $@ $(OPTL) dsimpletest.o  $(LIBDMUMPS) $(LORDERINGS) $(LIBS) $(MPIFLIB) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
 
 csimpletest:  $$@.o
-	$(FL) -o $@ $(OPTL) csimpletest.o  $(LIBCMUMPS) $(LORDERINGS) $(LIBS) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
+	$(FL) -o $@ $(OPTL) csimpletest.o  $(LIBCMUMPS) $(LORDERINGS) $(LIBS) $(MPIFLIB) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
 
 zsimpletest:  $$@.o
-	$(FL) -o $@ $(OPTL) zsimpletest.o  $(LIBZMUMPS) $(LORDERINGS) $(LIBS) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
+	$(FL) -o $@ $(OPTL) zsimpletest.o  $(LIBZMUMPS) $(LORDERINGS) $(LIBS) $(MPIFLIB) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
 
 c_example: $$@.o
-	$(FL) -o $@ $(OPTL) $@.o $(LIBDMUMPS) $(LORDERINGS) $(LIBS) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
+	$(FL) -o $@ $(OPTL) $@.o $(LIBDMUMPS) $(LORDERINGS) $(LIBS) $(MPICLIB) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
 
 
 multiple_arithmetics_example: $$@.o
-	$(FL) -o $@ $(OPTL) $@.o $(LIBSMUMPS) $(LIBDMUMPS) $(LIBCMUMPS) $(LIBZMUMPS) $(LORDERINGS) $(LIBS) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
+	$(FL) -o $@ $(OPTL) $@.o $(LIBSMUMPS) $(LIBDMUMPS) $(LIBCMUMPS) $(LIBZMUMPS) $(LORDERINGS) $(LIBS) $(MPIFLIB) $(RPATH_OPT) $(LIBBLAS) $(LIBOTHERS)
 
 ssimpletest_save_restore:   $$@.o
 	$(FL) -o $@ $(OPTL) ssimpletest_save_restore.o  $(LIBSMUMPS) $(LORDERINGS) $(LIBS) $(LIBBLAS) $(LIBOTHERS)
