diff -ur openfoam-extend-foam-extend-3.1-094e842fc0b0285ca5e39aeb081fa9056dfefb94.orig/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options openfoam-extend-foam-extend-3.1-094e842fc0b0285ca5e39aeb081fa9056dfefb94/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options
--- openfoam-extend-foam-extend-3.1-094e842fc0b0285ca5e39aeb081fa9056dfefb94.orig/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options	2015-09-15 16:33:00.000000000 +0200
+++ openfoam-extend-foam-extend-3.1-094e842fc0b0285ca5e39aeb081fa9056dfefb94/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options	2015-11-19 11:18:11.654470435 +0100
@@ -22,11 +22,10 @@
 
 EXE_INC = \
     -I$(LIB_SRC)/finiteVolume/lnInclude \
-    -I$(PARMGRIDGEN_INCLUDE_DIR)/Lib    \
-    -I$(PARMGRIDGEN_INCLUDE_DIR)/IMlib  \
+    -I$(PARMGRIDGEN_INCLUDE_DIR)    \
     $(TYPE_REAL)
 
 LIB_LIBS = \
-    -L$(PARMGRIDGEN_LIB_DIR) -lMGridGen
+    -L$(PARMGRIDGEN_LIB_DIR) -lmgrid
 
 #endif
Only in openfoam-extend-foam-extend-3.1-094e842fc0b0285ca5e39aeb081fa9056dfefb94/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make: options~
