--- Allwmake.firstInstall.orig	2014-02-27 14:04:26.632583000 +0100
+++ Allwmake.firstInstall	2014-02-27 14:04:43.937744000 +0100
@@ -28,7 +28,8 @@
     echo "    Fedora:   \"export QT_BIN_DIR=/usr/lib64/qt4/bin\""
     echo "    openSuse: \"export QT_BIN_DIR=/usr/bin\""
     echo 
-    read -r -p "Proceed without compiling ParaView [Y/n] " response
+    echo "Proceed without compiling ParaView [Y/n] "
+    read -r response
     if [[ $response =~ ^([nN][oO]|[nN])$ ]]
     then
         exit 0
@@ -50,7 +50,8 @@
         echo "\$CUDA_ARCH is required by nvcc compiler but not set."
         echo "Check section '-gpu-architecture' in 'man nvcc' for details." 
         echo
-        read -r -p "Proceed without compiling cudaSolvers? [Y/n] " response
+        echo "Proceed without compiling cudaSolvers? [Y/n] "
+        read -r response
         if [[ $response =~ ^([nN][oO]|[nN])$ ]]
         then
             exit 0
