diff -ru OpenFOAM-1.6-ext.orig/wmake/rules/linux64Gcc/c OpenFOAM-1.6-ext/wmake/rules/linux64Gcc/c
--- OpenFOAM-1.6-ext.orig/wmake/rules/linux64Gcc/c	2013-09-13 11:44:12.000000000 +0200
+++ OpenFOAM-1.6-ext/wmake/rules/linux64Gcc/c	2014-04-30 15:28:49.313680000 +0200
@@ -2,7 +2,7 @@
 
 cWARN        = -Wall
 
-cc          = gcc -m64
+cc          = gcc -m64 -fpermissive
 
 include $(RULES)/c$(WM_COMPILE_OPTION)
 
diff -ru OpenFOAM-1.6-ext.orig/wmake/rules/linux64Gcc/c++ OpenFOAM-1.6-ext/wmake/rules/linux64Gcc/c++
--- OpenFOAM-1.6-ext.orig/wmake/rules/linux64Gcc/c++	2013-09-13 11:44:12.000000000 +0200
+++ OpenFOAM-1.6-ext/wmake/rules/linux64Gcc/c++	2014-04-30 15:28:38.265722000 +0200
@@ -2,7 +2,7 @@
 
 c++WARN     = -Wall -Wextra -Wno-unused-parameter -Wold-style-cast -Wnon-virtual-dtor
 
-CC          = g++ -m64
+CC          = g++ -m64 -fpermissive
 
 include $(RULES)/c++$(WM_COMPILE_OPTION)
 
