--- foam-extend-3.1/Allwmake.firstInstall.orig	2014-06-18 12:53:06.000000000 +0200
+++ foam-extend-3.1/Allwmake.firstInstall	2014-09-04 09:09:54.217856000 +0200
@@ -28,7 +28,8 @@
     echo "    Fedora:   \"export QT_BIN_DIR=/usr/lib64/qt4/bin\""
     echo "    openSuse: \"export QT_BIN_DIR=/usr/bin\""
     echo
-    read -r -p "Proceed without compiling ParaView [Y/n] " response
+    echo "Proceed without compiling ParaView [Y/n] "
+    read -r response
     if [[ $response =~ ^([nN][oO]|[nN])$ ]]
     then
         exit 0
@@ -50,7 +51,8 @@
         echo "\$CUDA_ARCH is required by nvcc compiler but not set."
         echo "Check section '-gpu-architecture' in 'man nvcc' for details."
         echo
-        read -r -p "Proceed without compiling cudaSolvers? [Y/n] " response
+        echo "Proceed without compiling cudaSolvers? [Y/n] "
+        read -r response
         if [[ $response =~ ^([nN][oO]|[nN])$ ]]
         then
             exit 0
