--- foam-extend-3.1/ThirdParty/AllMake.stage2.orig	2014-10-07 13:56:36.976406000 +0200
+++ foam-extend-3.1/ThirdParty/AllMake.stage2	2014-10-07 13:57:01.833010000 +0200
@@ -62,7 +62,7 @@
 echo ========================================
 echo
 
-if [ "$WM_MPLIB" != "SYSTEMOPENMPI" ]
+if [ "$WM_MPLIB" != "SYSTEMOPENMPI" ] && [ "$WM_MPLIB" != "EASYBUILDMPI" ]
 then
     # MPI
     #( rpm_make -p openmpi-1.4.1 -s openmpi-1.4.1.spec  -u http://www.open-mpi.org/software/ompi/v1.4/downloads/openmpi-1.4.1.tar.gz )
