--- OpenFOAM-1.6-ext/src/finiteVolume/fields/fvPatchFields/constraint/wedge/wedgeFvPatchScalarField.C.orig	2013-09-18 12:15:11.480831000 +0200
+++ OpenFOAM-1.6-ext/src/finiteVolume/fields/fvPatchFields/constraint/wedge/wedgeFvPatchScalarField.C	2013-09-18 12:15:49.570197000 +0200
@@ -49,7 +49,7 @@
         updateCoeffs();
     }
 
-    operator==(patchInternalField());
+    this->operator==(patchInternalField());
 }
 
 
--- OpenFOAM-1.6-ext/src/engine/derivedFvPatchFields/engineTimeVaryingUniformFixedValue/engineTimeVaryingUniformFixedValueFvPatchFields.C.orig	2013-09-18 14:14:40.558432000 +0200
+++ OpenFOAM-1.6-ext/src/engine/derivedFvPatchFields/engineTimeVaryingUniformFixedValue/engineTimeVaryingUniformFixedValueFvPatchFields.C	2013-09-18 14:15:01.318844000 +0200
@@ -48,7 +48,7 @@
 
     checkTable();
 
-    operator==
+    this->operator==
     (
         interpolateXY
         (
@@ -73,7 +73,7 @@
 
     checkTable();
 
-    operator==
+    this->operator==
     (
        -interpolateXY
         (
@@ -98,7 +98,7 @@
 
     checkTable();
 
-    operator==
+    this->operator==
     (
         interpolateXY
         (
@@ -123,7 +123,7 @@
 
     checkTable();
 
-    operator==
+    this->operator==
     (
         interpolateXY
         (
@@ -148,7 +148,7 @@
 
     checkTable();
 
-    operator==
+    this->operator==
     (
         interpolateXY
         (
--- OpenFOAM-1.6-ext/src/VectorN/finiteVolume/fields/fvPatchFields/wedgeFvPatchVectorNFields.C.orig	2013-09-18 14:13:33.186815000 +0200
+++ OpenFOAM-1.6-ext/src/VectorN/finiteVolume/fields/fvPatchFields/wedgeFvPatchVectorNFields.C	2013-09-18 14:13:46.312655000 +0200
@@ -54,7 +54,7 @@
         updateCoeffs();                                         \
     }                                                           \
                                                                 \
-    operator==(patchInternalField());                           \
+    this->operator==(patchInternalField());                     \
 }                                                               \
                                                                 \
 template<>                                                      \
--- OpenFOAM-1.6-ext/src/finiteArea/fields/faPatchFields/constraint/wedge/wedgeFaPatchScalarField.C.orig	2013-09-18 14:12:30.105165000 +0200
+++ OpenFOAM-1.6-ext/src/finiteArea/fields/faPatchFields/constraint/wedge/wedgeFaPatchScalarField.C	2013-09-18 14:12:40.006153000 +0200
@@ -53,7 +53,7 @@
         updateCoeffs();
     }
 
-    operator==(patchInternalField());
+    this->operator==(patchInternalField());
 }
 
 
