--- OpenFOAM-1.6-ext/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options.orig	2013-09-13 11:44:11.000000000 +0200
+++ OpenFOAM-1.6-ext/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options	2013-09-19 06:56:01.125747000 +0200
@@ -7,9 +7,8 @@
 
 EXE_INC = \
     -I$(LIB_SRC)/finiteVolume/lnInclude \
-    -I$(PARMGRIDGEN_INCLUDE_DIR)/Lib    \
-    -I$(PARMGRIDGEN_INCLUDE_DIR)/IMlib  \
+    -I$(PARMGRIDGEN_INCLUDE_DIR)    \
     $(TYPE_REAL)
 
 LIB_LIBS = \
-    -L$(PARMGRIDGEN_LIB_DIR) -lMGridGen
+    -L$(PARMGRIDGEN_LIB_DIR) -lmgrid
--- OpenFOAM-1.6-ext/src/decompositionMethods/parMetisDecomp/Make/options.orig	2013-09-13 11:44:11.000000000 +0200
+++ OpenFOAM-1.6-ext/src/decompositionMethods/parMetisDecomp/Make/options	2013-09-20 07:09:45.114891103 +0200
@@ -10,5 +10,4 @@
 
 LIB_LIBS = \
     -L$(FOAM_MPI_LIBBIN) \
-    -L$(PARMETIS_LIB_DIR) -lmetis-parmetis \
-    -L$(PARMETIS_LIB_DIR) -lparmetis
+    -L$(PARMETIS_LIB_DIR) -lparmetis -lmetis 
--- OpenFOAM-1.6-ext/src/decompositionMethods/metisDecomp/Make/options.orig	2013-09-13 11:44:11.000000000 +0200
+++ OpenFOAM-1.6-ext/src/decompositionMethods/metisDecomp/Make/options	2014-09-11 14:14:39.059743000 +0200
@@ -4,7 +4,7 @@
     -I../scotchDecomp/lnInclude
 
 LIB_LIBS = \
+    $(PLIBS) \
     -L$(FOAM_LIBBIN)/dummy \
-    -L$(METIS_LIB_DIR) -lmetis \
-    -L$(METIS_LIB_DIR) -lGKlib
+    -L$(METIS_LIB_DIR) -lmetis
 
