Fix MPI in run scripts
Author: Samuel Moors (Vrije Universiteit Brussel)

diff -ur utils23.orig/runcry23 utils23/runcry23
--- utils23.orig/runcry23	2026-01-08 15:19:03.718122000 +0100
+++ utils23/runcry23	2026-01-08 17:35:54.349524000 +0100
@@ -165,9 +165,6 @@
  set here = $PWD
  set TMPDIR = $CRY23_SCRDIR/tmp$$_`whoami`
 #
- if (! $?VERSION ) then
-  set VERSION = v1.0.1
- endif
  if (! $?CRY23_INP ) then
   set CRY23_INP = $here
  endif
@@ -175,7 +172,7 @@
   set OUTFILE = out
  endif
 #
- set XDIR    = $CRY23_EXEDIR/$VERSION
+ set XDIR    = $CRY23_EXEDIR
  set INPDIR  = $CRY23_INP
  set OUTDIR  = $here
 #
diff -ur utils23.orig/runMPPcry23 utils23/runMPPcry23
--- utils23.orig/runMPPcry23	2026-01-08 15:19:03.739398000 +0100
+++ utils23/runMPPcry23	2026-01-08 17:43:15.892592526 +0100
@@ -177,9 +177,6 @@
  set here = $PWD
  set TMPDIR = $CRY23_SCRDIR/tmp_p$$_`whoami`
 #
- if (! $?VERSION ) then
-  set VERSION = v1.0.1
- endif
  if (! $?CRY23_INP ) then
   set CRY23_INP = $here
  endif
@@ -190,11 +187,10 @@
   set OUTFILE = out
  endif
 #
- set XDIR    = $CRY23_EXEDIR/$VERSION
+ set XDIR    = $CRY23_EXEDIR
  set INPDIR  = $CRY23_INP
  set OUTDIR  = $here
 # Note that the job launching line, later on, is custom too.
- set MPIDIR   = /replace/this/line/with/your/own/mpibin/directory
  set MPIBIN   = mpirun
 #
  set NPROCS     = $1
@@ -211,14 +207,6 @@
  echo output data in $OUTDIR/$OUTFILE >>& $OUTDIR/$OUTFILE
  echo output data in $OUTDIR/$OUTFILE
 #
- if(-e $MPIDIR/$MPIBIN && ! -z $MPIDIR/$MPIBIN ) then
-  echo $MPIBIN executable in $MPIDIR >>& $OUTDIR/$OUTFILE
-  echo $MPIBIN executable in $MPIDIR
- else
-  echo $MPIBIN executable in $MPIDIR not found or empty >>& $OUTDIR/$OUTFILE
-  echo $MPIBIN executable in $MPIDIR not found or empty
-  exit
- endif
 #
  if(-e $XDIR/MPPcrystal && ! -z $XDIR/MPPcrystal ) then
   echo MPPcrystal executable in $XDIR >>& $OUTDIR/$OUTFILE
@@ -238,15 +226,6 @@
   exit
  endif
 #
- if (-e $CRY23P_MACH/machines.LINUX && ! -z $CRY23P_MACH/machines.LINUX) then
-  echo file machines.LINUX in $CRY23P_MACH >>& $OUTDIR/$OUTFILE
-  echo file machines.LINUX in $CRY23P_MACH
- else
-  echo file machines.LINUX in $CRY23P_MACH not found or empty >>& $OUTDIR/$OUTFILE
-  echo file machines.LINUX in $CRY23P_MACH not found or empty
-  exit
- endif
-#
  if (-e $CRY23P_MACH/nodes.par && ! -z $CRY23P_MACH/nodes.par) then
   echo list of nodes in $CRY23P_MACH/nodes.par >>& $OUTDIR/$OUTFILE
   echo list of nodes in $CRY23P_MACH/nodes.par
@@ -453,7 +432,7 @@
 #
  echo MPPcrystal job running in $TMPDIR on each node ...
 # Job Launching line:
- ${MPIDIR}/${MPIBIN} -machinefile $CRY23P_MACH/machines.LINUX -np $NPROCS $TMPDIR/MPPcrystal < $TMPDIR/INPUT >>& $OUTDIR/$OUTFILE
+ ${MPIBIN} -np $NPROCS $TMPDIR/MPPcrystal < $TMPDIR/INPUT >>& $OUTDIR/$OUTFILE
  date >>& $OUTDIR/$OUTFILE
  echo Contents of temporary directory in master node `hostname`: >>& $OUTDIR/$OUTFILE
  ls -l >> $OUTDIR/$OUTFILE
diff -ur utils23.orig/runPcry23 utils23/runPcry23
--- utils23.orig/runPcry23	2026-01-08 15:19:03.783881000 +0100
+++ utils23/runPcry23	2026-01-08 17:42:44.381464880 +0100
@@ -177,9 +177,6 @@
  set here = $PWD
  set TMPDIR = $CRY23_SCRDIR/tmp_p$$_`whoami`
 #
- if (! $?VERSION ) then
-  set VERSION = v1.0.1
- endif
  if (! $?CRY23_INP ) then
   set CRY23_INP = $here
  endif
@@ -190,12 +187,11 @@
   set OUTFILE = out
  endif
 #
- set XDIR    = $CRY23_EXEDIR/$VERSION
+ set XDIR    = $CRY23_EXEDIR
  set INPDIR  = $CRY23_INP
  set OUTDIR  = $here
 # Note that the job launching line, later on, is custom too.
 # The next line defines a varable for the path to the MPI bin directory
- set MPIDIR   = /replace/this/line/with/your/own/mpibin/directory
  set MPIBIN   = mpirun
 #
  set NPROCS     = $1
@@ -212,14 +208,6 @@
  echo output data in $OUTDIR/$OUTFILE >>& $OUTDIR/$OUTFILE
  echo output data in $OUTDIR/$OUTFILE
 #
- if(-e $MPIDIR/$MPIBIN && ! -z $MPIDIR/$MPIBIN ) then
-  echo $MPIBIN executable in $MPIDIR >>& $OUTDIR/$OUTFILE
-  echo $MPIBIN executable in $MPIDIR
- else
-  echo $MPIBIN executable in $MPIDIR not found or empty >>& $OUTDIR/$OUTFILE
-  echo $MPIBIN executable in $MPIDIR not found or empty
-  exit
- endif
 #
  if(-e $XDIR/Pcrystal && ! -z $XDIR/Pcrystal ) then
   echo Pcrystal executable in $XDIR >>& $OUTDIR/$OUTFILE
@@ -239,15 +227,6 @@
   exit
  endif
 #
- if (-e $CRY23P_MACH/machines.LINUX && ! -z $CRY23P_MACH/machines.LINUX) then
-  echo file machines.LINUX in $CRY23P_MACH >>& $OUTDIR/$OUTFILE
-  echo file machines.LINUX in $CRY23P_MACH
- else
-  echo file machines.LINUX in $CRY23P_MACH not found or empty >>& $OUTDIR/$OUTFILE
-  echo file machines.LINUX in $CRY23P_MACH not found or empty
-  exit
- endif
-#
  if (-e $CRY23P_MACH/nodes.par && ! -z $CRY23P_MACH/nodes.par) then
   echo list of nodes in $CRY23P_MACH/nodes.par >>& $OUTDIR/$OUTFILE
   echo list of nodes in $CRY23P_MACH/nodes.par
@@ -454,7 +433,7 @@
 #
  echo Pcrystal job running in $TMPDIR on each node ...
 # Job Launching line:
- ${MPIDIR}/${MPIBIN} -machinefile $CRY23P_MACH/machines.LINUX -np $NPROCS $TMPDIR/Pcrystal < $TMPDIR/INPUT >>& $OUTDIR/$OUTFILE
+ ${MPIBIN} -np $NPROCS $TMPDIR/Pcrystal < $TMPDIR/INPUT >>& $OUTDIR/$OUTFILE
  date >>& $OUTDIR/$OUTFILE
  echo Contents of temporary directory in master node `hostname`: >>& $OUTDIR/$OUTFILE
  ls -l >> $OUTDIR/$OUTFILE
diff -ur utils23.orig/runPprop23 utils23/runPprop23
--- utils23.orig/runPprop23	2026-01-08 15:19:03.812290000 +0100
+++ utils23/runPprop23	2026-01-08 17:55:55.089408294 +0100
@@ -113,7 +113,7 @@
   echo SEEBECK.DAT "  " inpfilename_wf-filename_SEEBECK.DAT "  " Seebeck coefficient "("BOLTZAO")"
   echo POWER.DAT "  " inpfilename_wf-filename_POWER.DAT "  " Power factor coefficient "("BOLTZAO")"
   echo KAPPA.DAT "    " inpfilename_wf-filename_KAPPA.DAT "    " electron thermal conductivity "("BOLTZAO")"
-  echo SIGMAS.DAT "   " inpfilename_wf-filename_SIGMAS.DAT "   " electron conductivity * Seebeck coefficient "("BOLTZAO")"
+  echo SIGMAS.DAT "   " inpfilename_wf-filename_SIGMAS.DAT "   " electron conductivity "*" Seebeck coefficient "("BOLTZAO")"
   echo TDF.DAT "      " inpfilename_wf-filename_TDF.DAT "      " transport distribution function "("BOLTZAO")"
 #
   echo definablename_numering_Kkindex_real.molden "     "definablename_numering_Kkindex_real.molden "    "data for Crystalline Orbitals Visualization "("ORBITALS")"
@@ -196,9 +196,6 @@
  set here = $PWD
  set TMPDIR = $CRY23_SCRDIR/tmp_p$$_`whoami`
 #
- if (! $?VERSION ) then
-  set VERSION = v1.0.1
- endif
  if (! $?CRY23_INP ) then
   set CRY23_INP = $here
  endif
@@ -209,11 +206,10 @@
   set OUTFILE = outp
  endif
 #
- set XDIR    = $CRY23_EXEDIR/$VERSION
+ set XDIR    = $CRY23_EXEDIR
  set INPDIR  = $CRY23_INP
  set OUTDIR  = $here
 # Note that the job launching line, later on, is custom too.
- set MPIDIR   = /replace/this/line/with/your/own/mpibin/directory
  set MPIBIN   = mpirun
 #
  set NPROCS     = $1
@@ -231,24 +227,6 @@
  echo output data in $OUTDIR/$OUTFILE >>& $OUTDIR/$OUTFILE
  echo output data in $OUTDIR/$OUTFILE
 #
- if(-e $MPIDIR/$MPIBIN && ! -z $MPIDIR/$MPIBIN ) then
-  echo $MPIBIN executable in $MPIDIR >>& $OUTDIR/$OUTFILE
-  echo $MPIBIN executable in $MPIDIR
- else
-  echo $MPIBIN executable in $MPIDIR not found or empty >>& $OUTDIR/$OUTFILE
-  echo $MPIBIN executable in $MPIDIR not found or empty
-  exit
- endif
-#
- if (-e $CRY23P_MACH/machines.LINUX && ! -z $CRY23P_MACH/machines.LINUX) then
-  echo file machines.LINUX in $CRY23P_MACH >>& $OUTDIR/$OUTFILE
-  echo file machines.LINUX in $CRY23P_MACH
- else
-  echo file machines.LINUX in $CRY23P_MACH not found or empty >>& $OUTDIR/$OUTFILE
-  echo file machines.LINUX in $CRY23P_MACH not found or empty
-  exit
- endif
-#
  if (-e $CRY23P_MACH/nodes.par && ! -z $CRY23P_MACH/nodes.par) then
   echo list of nodes in $CRY23P_MACH/nodes.par >>& $OUTDIR/$OUTFILE
   echo list of nodes in $CRY23P_MACH/nodes.par
@@ -340,7 +318,7 @@
    if (-e fort.9  && ! -z fort.9  ) then
    echo Pproperties job running in $TMPDIR ...
 # Job Launching line:
-   ${MPIDIR}/${MPIBIN} -machinefile $CRY23P_MACH/machines.LINUX -np $NPROCS $TMPDIR/Pproperties < $TMPDIR/INPUT >>& $OUTDIR/$OUTFILE
+   ${MPIBIN} -np $NPROCS $TMPDIR/Pproperties < $TMPDIR/INPUT >>& $OUTDIR/$OUTFILE
    date >>& $OUTDIR/$OUTFILE
     echo Contents of temporary directory: >>& $OUTDIR/$OUTFILE
     ls -l >>& $OUTDIR/$OUTFILE
diff -ur utils23.orig/runprop23 utils23/runprop23
--- utils23.orig/runprop23	2026-01-08 15:19:03.844898000 +0100
+++ utils23/runprop23	2026-01-08 17:40:25.599611000 +0100
@@ -89,7 +89,7 @@
   echo SEEBECK.DAT "  " inpfilename_wf-filename_SEEBECK.DAT "  " Seebeck coefficient "("BOLTZAO")"
   echo POWER.DAT "  " inpfilename_wf-filename_POWER.DAT "  " Power factor coefficient "("BOLTZAO")"
   echo KAPPA.DAT "    " inpfilename_wf-filename_KAPPA.DAT "    " electron thermal conductivity "("BOLTZAO")"
-  echo SIGMAS.DAT "   " inpfilename_wf-filename_SIGMAS.DAT "   " electron conductivity * Seebeck coefficient "("BOLTZAO")"
+  echo SIGMAS.DAT "   " inpfilename_wf-filename_SIGMAS.DAT "   " electron conductivity "*" Seebeck coefficient "("BOLTZAO")"
   echo TDF.DAT "      " inpfilename_wf-filename_TDF.DAT "      " transport distribution function "("BOLTZAO")"
 #
   echo definablename_numering_Kkindex_real.molden "     "definablename_numering_Kkindex_real.molden "    "data for Crystalline Orbitals Visualization "("ORBITALS")"
@@ -118,9 +118,6 @@
  set here = $PWD
  set TMPDIR = $CRY23_SCRDIR/tmp_p$$_`whoami`
 #
- if (! $?VERSION ) then
-  set VERSION = v1.0.1
- endif
  if (! $?CRY23_INP ) then
   set CRY23_INP = $here
  endif
@@ -130,7 +127,7 @@
   set OUTFILE = "$OUTFILE"p 
  endif
 #
- set XDIR    = $CRY23_EXEDIR/$VERSION
+ set XDIR    = $CRY23_EXEDIR
  set INPDIR  = $CRY23_INP
  set OUTDIR  = $here
 # Note that the job launching line, later on, is custom too.
