diff --git a/test/distributed/tensor/test_dtensor_ops.py b/test/distributed/tensor/test_dtensor_ops.py
index 8c650f6b0ce..04cfa7d4cc2 100644
--- a/test/distributed/tensor/test_dtensor_ops.py
+++ b/test/distributed/tensor/test_dtensor_ops.py
@@ -463,6 +463,7 @@ dtensor_fails = {
     skip("nn.functional.feature_alpha_dropout", "without_train"),
     skip("nn.functional.hinge_embedding_loss"),
     skip("nn.functional.cosine_embedding_loss"),
+    skip("nn.functional.multi_head_attention_forward"),  # randomness
     skip("fft.hfft"),
     skip("fft.hfft2"),
     skip("fft.hfft2"),
