diff --git a/test/test_cuda.py b/test/test_cuda.py
index 3726c377970..78b5e8c8af9 100644
--- a/test/test_cuda.py
+++ b/test/test_cuda.py
@@ -3633,6 +3633,7 @@ print(f"{{r1}}, {{r2}}")
         x = torch.cuda.device_count()
         self.assertEqual(f"{x}, 1", r)
 
+    @unittest.skip("Not applicable")
     def test_gds_fails_in_ci(self):
         if IS_WINDOWS or TEST_WITH_ROCM:
             error_msg = "is not supported on this platform"
