Fix a wrongly named variable in the tests

Author: Alexander Grund (TU Dresden)
diff --git a/test/run_test.py b/test/run_test.py
index 4309e65478..a0ffa51aad 100755
--- a/test/run_test.py
+++ b/test/run_test.py
@@ -738,7 +738,7 @@ def main():
 
     if options.continue_through_error and has_failed:
         for err in failure_messages:
-            print_to_stderr(message)
+            print_to_stderr(err)
         sys.exit(1)
 
 if __name__ == '__main__':
