Disable tests that requires external downloads

Åke Sandgren, 2024-05-14
diff --git a/test/test_hub.py b/test/test_hub.py
index a0c0c315c4e..6427d02045f 100644
--- a/test/test_hub.py
+++ b/test/test_hub.py
@@ -23,6 +23,7 @@ TORCHHUB_EXAMPLE_RELEASE_URL = 'https://github.com/ailzhang/torchhub_example/rel
 
 
 @unittest.skipIf(IS_SANDCASTLE, 'Sandcastle cannot ping external')
+@unittest.skipIf(True, 'EasyBuild: do not want tests to depend on downloading')
 class TestHub(TestCase):
 
     def setUp(self):
diff --git a/test/test_nn.py b/test/test_nn.py
index 66fc4cdc466..ed5d1ecc765 100644
--- a/test/test_nn.py
+++ b/test/test_nn.py
@@ -151,6 +151,7 @@ class TestNN(NNTestCase):
             for b in m.buffers():
                 self.assertFalse(b.requires_grad)
 
+    @unittest.skipIf(True, 'EasyBuild: do not want tests to depend on downloading')
     def test_module_backcompat(self):
         from torch.serialization import SourceChangeWarning
         path = download_file('https://download.pytorch.org/test_data/linear.pt')
