diff -h -Nru gnina-1.0.1.orig/gninasrc/gninaserver/Reorienter.h gnina-1.0.1/gninasrc/gninaserver/Reorienter.h
--- gnina-1.0.1.orig/gninasrc/gninaserver/Reorienter.h	2021-03-23 12:21:11.000000000 -0400
+++ gnina-1.0.1/gninasrc/gninaserver/Reorienter.h	2021-05-21 16:24:28.729509493 -0400
@@ -10,8 +10,8 @@
 #ifndef REORIENTER_H_
 #define REORIENTER_H_
 
-#include <eigen3/Eigen/Core>
-#include <eigen3/Eigen/Eigenvalues>
+#include "Eigen/Core"
+#include "Eigen/Eigenvalues"
 #include <iostream>
 #include "common.h"
 #include "parsing.h"
