diff -ru BioPerl-1.6.901.orig/t/SeqIO/SeqIO.t BioPerl-1.6.901/t/SeqIO/SeqIO.t
--- BioPerl-1.6.901.orig/t/SeqIO/SeqIO.t	2011-05-19 04:38:06.000000000 +0200
+++ BioPerl-1.6.901/t/SeqIO/SeqIO.t	2013-05-17 15:31:16.212464000 +0200
@@ -132,6 +132,6 @@
 
 throws_ok {
     Bio::SeqIO->new(-file => 'foo.bar');
-} qr/Can not open 'foo.bar' for reading: No such file or directory/,
+} qr/Can not open 'foo.bar' for reading:/,
     'Must pass a real file';
 
