--- Build.PL.orig	2016-11-23 16:01:53.000000000 +1300
+++ Build.PL	2016-11-23 16:11:19.000000000 +1300
@@ -78,2 +78,4 @@
 	    if -e "$samtools/include/$HeaderFile";
+    	$sam_include = "$samtools/include/bam"
+    	    if -e "$samtools/include/bam/$HeaderFile";
 	$sam_lib     = $samtools
@@ -166,3 +168,3 @@
 sub _samtools {
-    $ENV{SAMTOOLS} ||
+    $ENV{EBROOTSAMTOOLS} ||
     ( can_load(modules => {'Alien::SamTools' => undef, 'File::ShareDir' => undef}) &&
