#Use icc instead of gcc
#September 11 2014 by B. Hajgato
--- XML-Bare-0.53/Makefile.PL.orig	2013-04-30 04:00:00.000000000 +0200
+++ XML-Bare-0.53/Makefile.PL	2014-09-11 08:17:22.437966044 +0200
@@ -94,7 +94,7 @@
   my $div = (substr($ENV{'PATH'},0,1) eq '/') ? ':' : ';';
   my @path = split($div,$ENV{'PATH'});
   foreach my $dir ( @path ) {
-    return 'gcc' if( -e "$dir/gcc" || -e "$dir/gcc.exe" ); # prefer gcc
+    return 'icc' if( -e "$dir/icc" || -e "$dir/icc.exe" ); # prefer icc
     return 'cc'  if( -e "$dir/cc"  || -e "$dir/cc.exe"  );
   }               
   return 0;
