From: https://github.com/OpenMathLib/OpenBLAS/pull/4923
diff --git a/cpuid_x86.c b/cpuid_x86.c
index 9b2b7a51eb..4e13f1462c 100644
--- a/cpuid_x86.c
+++ b/cpuid_x86.c
@@ -1689,6 +1689,7 @@ int get_cpuname(void){
 	    return CPUTYPE_BARCELONA;
         }
       case 10: // Zen3/4
+      case 11: // Zen5
 #ifndef NO_AVX512
           if(support_avx512_bf16())
             return CPUTYPE_COOPERLAKE;
@@ -2479,7 +2480,7 @@ int get_coretype(void){
 	  }
 	  break;
 	}
-      } else if (exfamily == 8 || exfamily == 10) {
+      } else if (exfamily == 8 || exfamily == 10 || exfamily == 11) {
 	switch (model) {
 	case 1:
 	  // AMD Ryzen
