diff -ru OpenBLAS-0.2.18.orig/Makefile OpenBLAS-0.2.18/Makefile
--- OpenBLAS-0.2.18.orig/Makefile	2016-04-12 21:29:19.000000000 +0200
+++ OpenBLAS-0.2.18/Makefile	2016-10-22 13:42:21.502673471 +0200
@@ -224,6 +224,12 @@
 
 lapack_prebuild :
 ifndef NOFORTRAN
+	if [ -e lapack-$(LAPACK_VERSION).tgz ]; then \
+	    rm -rf $(NETLIB_LAPACK_DIR).old; \
+	    mv $(NETLIB_LAPACK_DIR) $(NETLIB_LAPACK_DIR).old; \
+	    $(TAR) zxf ./lapack-$(LAPACK_VERSION).tgz; \
+	    mv lapack-$(LAPACK_VERSION) $(NETLIB_LAPACK_DIR); \
+	fi
 	-@echo "FORTRAN     = $(FC)" > $(NETLIB_LAPACK_DIR)/make.inc
 	-@echo "OPTS        = $(LAPACK_FFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
 	-@echo "POPTS       = $(LAPACK_FPFLAGS)" >> $(NETLIB_LAPACK_DIR)/make.inc
