# Fix java paths when building with EB
# S.D.Pinches, 02-01-2024
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/jDispatcherIp.template mdsplus-java-7.131.6/mdsplus/java/jdispatcher/jDispatcherIp.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/jDispatcherIp.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jdispatcher/jDispatcherIp.template	2022-09-07 19:19:44.033416063 +0200
@@ -10,11 +10,11 @@
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
 CLASSPATH=\
-$MDSPLUS_DIR/java/classes/jDispatcher.jar:\
-$MDSPLUS_DIR/java/classes/jScope.jar:\
-$MDSPLUS_DIR/java/classes/mdsobjects.jar \
+$EBROOTMDSPLUS/java/classes/jDispatcher.jar:\
+$EBROOTMDSPLUS/java/classes/jScope.jar:\
+$EBROOTMDSPLUS/java/classes/mdsobjects.jar \
 java mds.jdispatcher.jDispatcherIp "$@"
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/jDispatchMonitor.template mdsplus-java-7.131.6/mdsplus/java/jdispatcher/jDispatchMonitor.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/jDispatchMonitor.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jdispatcher/jDispatchMonitor.template	2022-09-07 19:21:02.690783830 +0200
@@ -10,10 +10,10 @@
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
 CLASSPATH=\
-$MDSPLUS_DIR/java/classes/jDispatcher.jar:\
-$MDSPLUS_DIR/java/classes/mdsobjects.jar \
+$EBROOTMDSPLUS/java/classes/jDispatcher.jar:\
+$EBROOTMDSPLUS/java/classes/mdsobjects.jar \
 java mds.jdispatcher.jDispatchMonitor "$@"
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/jServer.template mdsplus-java-7.131.6/mdsplus/java/jdispatcher/jServer.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/jServer.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jdispatcher/jServer.template	2022-09-07 19:22:20.545675735 +0200
@@ -10,10 +10,10 @@
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
 CLASSPATH=\
-$MDSPLUS_DIR/java/classes/jDispatcher.jar:\
-$MDSPLUS_DIR/java/classes/mdsobjects.jar \
+$EBROOTMDSPLUS/java/classes/jDispatcher.jar:\
+$EBROOTMDSPLUS/java/classes/mdsobjects.jar \
 java -Xss10M mds.jdispatcher.jServer "$@"
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/src/main/java/mds/jdispatcher/MdsHelper.java mdsplus-java-7.131.6/mdsplus/java/jdispatcher/src/main/java/mds/jdispatcher/MdsHelper.java
--- mdsplus-java-7.131.6-orig/mdsplus/java/jdispatcher/src/main/java/mds/jdispatcher/MdsHelper.java	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jdispatcher/src/main/java/mds/jdispatcher/MdsHelper.java	2022-04-07 18:34:25.982218932 +0200
@@ -114,7 +114,7 @@
 		{
 			final Vector<String> paths = new Vector<>();
 			{// add mdsplus_dir
-				final String path = System.getenv("MDSPLUS_DIR");
+				final String path = System.getenv("EBROOTMDSPLUS");
 				if (path != null)
 					paths.add(new File(path, "local").getPath());
 			}
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jscope/jScope.template mdsplus-java-7.131.6/mdsplus/java/jscope/jScope.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jscope/jScope.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jscope/jScope.template	2022-09-08 10:17:42.884774294 +0200
@@ -10,7 +10,7 @@
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$MDSPLUS_DIR/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUS/${lib}:$MDSPLUS_DIR/${lib} \
 DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-java -Xmx2G -Djava.library.path=$MDSPLUS_DIR/${lib} -jar $MDSPLUS_DIR/java/classes/jScope.jar $*
+java -Xmx2G -Djava.library.path=$EBROOTMDSPLUS -jar $EBROOTMDSPLUS/java/classes/jScope.jar $*
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/CompileTree.template mdsplus-java-7.131.6/mdsplus/java/jtraverser/CompileTree.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/CompileTree.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jtraverser/CompileTree.template	2022-09-07 18:57:17.128387109 +0200
@@ -4,19 +4,19 @@
 #
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUS/lib64 ] ; then
     lib=/lib64
   fi
 else
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUS/lib32 ]; then
      lib=lib32
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
 CLASSPATH=\
-$MDSPLUS_DIR/java/classes/jTraverser.jar:\
-$MDSPLUS_DIR/java/classes/mdsobjects.jar \
-java -Xss5M -Djava.library.path=$MDSPLUS_DIR/${lib} mds.jtraverser.CompileTree $1 $2 
+$EBROOTMDSPLUS/java/classes/jTraverser.jar:\
+$EBROOTMDSPLUS/java/classes/mdsobjects.jar \
+java -Xss5M -Djava.library.path=$EBROOTMDSPLUS/${lib} mds.jtraverser.CompileTree $1 $2 
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/DecompileTree.template mdsplus-java-7.131.6/mdsplus/java/jtraverser/DecompileTree.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/DecompileTree.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jtraverser/DecompileTree.template	2022-09-07 18:58:23.561474832 +0200
@@ -4,19 +4,19 @@
 #
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUS/lib64 ] ; then
     lib=/lib64
   fi
 else
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUS/lib32 ]; then
      lib=lib32
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUS/${lib}$EBROOTMDSPLUS/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib} \
 CLASSPATH=\
-$MDSPLUS_DIR/java/classes/jTraverser.jar:\
-$MDSPLUS_DIR/java/classes/mdsobjects.jar \
-java -Xss5M -Djava.library.path=$MDSPLUS_DIR/${lib} mds.jtraverser.DecompileTree $1 $2 
+$EBROOTMDSPLUS/java/classes/jTraverser.jar:\
+$EBROOTMDSPLUS/java/classes/mdsobjects.jar \
+java -Xss5M -Djava.library.path=$EBROOTMDSPLUS/${lib} mds.jtraverser.DecompileTree $1 $2 
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/jTraverser.bat mdsplus-java-7.131.6/mdsplus/java/jtraverser/jTraverser.bat
--- mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/jTraverser.bat	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jtraverser/jTraverser.bat	2022-04-07 18:35:23.441894000 +0200
@@ -1 +1 @@
-@start "" javaw -Xss5M -jar "%MDSPLUS_DIR%\java\classes\jTraverser.jar" %*
+@start "" javaw -Xss5M -jar "%EBROOTMDSPLUS%\java\classes\jTraverser.jar" %*
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/jTraverser.template mdsplus-java-7.131.6/mdsplus/java/jtraverser/jTraverser.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/jTraverser.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jtraverser/jTraverser.template	2022-09-07 19:00:05.077859796 +0200
@@ -1,16 +1,16 @@
 #!/bin/sh
 lib=lib
 if java -d64 -version >  /dev/null 2>&1; then
-  if [ -r $MDSPLUS_DIR/lib64 ] ; then
+  if [ -r $EBROOTMDSPLUS/lib64 ] ; then
     lib=/lib64
   fi
 else
-  if [ -r $MDSPLUS_DIR/lib32 ]; then
+  if [ -r $EBROOTMDSPLUS/lib32 ]; then
      lib=lib32
   fi
 fi
 exec env \
-LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-SHLIB_PATH=$SHLIB_PATH:$MDSPLUS_DIR/${lib} \
-DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$MDSPLUS_DIR/${lib} \
-java -Xss5M -Djava.library.path=$MDSPLUS_DIR/${lib} -jar $MDSPLUS_DIR/java/classes/jTraverser.jar "$@"
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+SHLIB_PATH=$SHLIB_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$EBROOTMDSPLUS/${lib}:$EBROOTMDSPLUS/${lib} \
+java -Xss5M -Djava.library.path=$EBROOTMDSPLUS/${lib} -jar $EBROOTMDSPLUS/java/classes/jTraverser.jar "$@"
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/make_jTraverser.bat mdsplus-java-7.131.6/mdsplus/java/jtraverser/make_jTraverser.bat
--- mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser/make_jTraverser.bat	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jtraverser/make_jTraverser.bat	2022-04-07 18:35:23.445326708 +0200
@@ -21,7 +21,7 @@
 SET JAR="%JDK_HOME%\bin\jar.exe"
 SET JARDIR=%PWD%\classes
 SET JCFLAGS=-O -source 6 -g:none
-SET CLASSPATH=-classpath ".;%MDSPLUS_DIR%\java\classes\mdsobjects.jar;%MDSPLUS_DIR%\java\classes\jScope.jar"
+SET CLASSPATH=-classpath ".;%EBROOTMDSPLUS%\java\classes\mdsobjects.jar;%EBROOTMDSPLUS%\java\classes\jScope.jar"
 
 SET OUTFILE=%PWD%\jTraverser.jar
 SET SOURCES=*.java^
@@ -62,10 +62,10 @@
 CLS
 "%JDK_HOME%\bin\java.exe" -cp ^
 "%OUTFILE%^
-;%MDSPLUS_DIR%\java\classes\mdsobjects.jar^
-;%MDSPLUS_DIR%\java\classes\jScope.jar^
-;%MDSPLUS_DIR%\java\classes\DeviceBeans.jar^
-;%MDSPLUS_DIR%\java\classes\jDevices.jar^
+;%EBROOTMDSPLUS%\java\classes\mdsobjects.jar^
+;%EBROOTMDSPLUS%\java\classes\jScope.jar^
+;%EBROOTMDSPLUS%\java\classes\DeviceBeans.jar^
+;%EBROOTMDSPLUS%\java\classes\jDevices.jar^
 " -Xss5M jTraverser
 
 :end
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser2/jTraverser2.template mdsplus-java-7.131.6/mdsplus/java/jtraverser2/jTraverser2.template
--- mdsplus-java-7.131.6-orig/mdsplus/java/jtraverser2/jTraverser2.template	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/jtraverser2/jTraverser2.template	2022-09-08 10:19:44.701573830 +0200
@@ -1,2 +1,2 @@
 #!/bin/sh
-exec java -jar $MDSPLUS_DIR/java/classes/jTraverser2.jar "$@"
+exec java -jar $EBROOTMDSPLUS/java/classes/jTraverser2.jar "$@"
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/java/mdsobjects/src/test/java/MDSplus/MdsConnectionTest.java mdsplus-java-7.131.6/mdsplus/java/mdsobjects/src/test/java/MDSplus/MdsConnectionTest.java
--- mdsplus-java-7.131.6-orig/mdsplus/java/mdsobjects/src/test/java/MDSplus/MdsConnectionTest.java	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/java/mdsobjects/src/test/java/MDSplus/MdsConnectionTest.java	2022-04-07 18:34:03.293033301 +0200
@@ -23,7 +23,7 @@
 			try
 			{
 				new DatagramSocket(port).close();
-				java.lang.String hostspath = System.getenv("MDSPLUS_DIR") + "/testing/mdsip.hosts";
+				java.lang.String hostspath = System.getenv("EBROOTMDSPLUS") + "/testing/mdsip.hosts";
 				if (!new File(hostspath).exists())
 				{
 					hostspath = "/etc/mdsip.hosts";
diff -Nru mdsplus-java-7.131.6-orig/mdsplus/Makefile.in mdsplus-java-7.131.6/mdsplus/Makefile.in
--- mdsplus-java-7.131.6-orig/mdsplus/Makefile.in	2022-04-07 15:50:24.000000000 +0200
+++ mdsplus-java-7.131.6/mdsplus/Makefile.in	2022-09-08 13:34:32.010754311 +0200
@@ -42,8 +42,8 @@
 D3D_PACKAGE = d3dshr
 
 ifeq "@JAVA_ONLY@" "yes"
-NON_JAR =
-PARTS = $(JAVA_JAR)
+NON_JAR = @JAVA_APS@
+PARTS = $(NON_JAR) $(JAVA_JAR)
 else
 NON_JAR = \
 	mdsshr \
