--- hdf5-1.8.7.orig/configure	2011-05-10 16:20:59.000000000 +0200
+++ hdf5-1.8.7/configure	2012-06-19 10:39:14.925704379 +0200
@@ -18836,11 +18836,13 @@
 	 # linked, so don't bother handling this case.
 	 esac
        else
+     if [ "${prev}${p}" != "-l" ]; then
 	 if test -z "$postdeps_FC"; then
 	   postdeps_FC="${prev}${p}"
 	 else
 	   postdeps_FC="${postdeps_FC} ${prev}${p}"
 	 fi
+     fi
        fi
        prev=
        ;;
