diff -ru hdf5-1.8.7.orig/c++/src/H5Attribute.cpp hdf5-1.8.7/c++/src/H5Attribute.cpp
--- hdf5-1.8.7.orig/c++/src/H5Attribute.cpp	2012-02-02 11:12:36.935361641 +0100
+++ hdf5-1.8.7/c++/src/H5Attribute.cpp	2012-02-02 11:14:28.975575797 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/src/H5Exception.cpp hdf5-1.8.7/c++/src/H5Exception.cpp
--- hdf5-1.8.7.orig/c++/src/H5Exception.cpp	2012-02-02 11:12:36.935361641 +0100
+++ hdf5-1.8.7/c++/src/H5Exception.cpp	2012-02-02 11:14:00.075521742 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5FaccProp.cpp hdf5-1.8.7/c++/src/H5FaccProp.cpp
--- hdf5-1.8.7.orig/c++/src/H5FaccProp.cpp	2012-02-02 11:12:36.935361641 +0100
+++ hdf5-1.8.7/c++/src/H5FaccProp.cpp	2012-02-02 11:14:21.885563211 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5IdComponent.cpp hdf5-1.8.7/c++/src/H5IdComponent.cpp
--- hdf5-1.8.7.orig/c++/src/H5IdComponent.cpp	2012-02-02 11:12:36.935361641 +0100
+++ hdf5-1.8.7/c++/src/H5IdComponent.cpp	2012-02-02 11:14:12.115545800 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef H5_VMS
 #include <iostream>
 #endif /*H5_VMS*/
diff -ru hdf5-1.8.7.orig/c++/src/H5Library.cpp hdf5-1.8.7/c++/src/H5Library.cpp
--- hdf5-1.8.7.orig/c++/src/H5Library.cpp	2012-02-02 11:12:36.935361641 +0100
+++ hdf5-1.8.7/c++/src/H5Library.cpp	2012-02-02 11:14:08.795540708 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5CppDoc.h"	// included only for Doxygen to generate part of RM
diff -ru hdf5-1.8.7.orig/c++/src/H5Object.cpp hdf5-1.8.7/c++/src/H5Object.cpp
--- hdf5-1.8.7.orig/c++/src/H5Object.cpp	2012-02-02 11:12:36.935361641 +0100
+++ hdf5-1.8.7/c++/src/H5Object.cpp	2012-02-02 11:14:14.825550481 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5PropList.cpp hdf5-1.8.7/c++/src/H5PropList.cpp
--- hdf5-1.8.7.orig/c++/src/H5PropList.cpp	2012-02-02 11:12:36.935361641 +0100
+++ hdf5-1.8.7/c++/src/H5PropList.cpp	2012-02-02 11:14:25.215569149 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/src/H5DataType.cpp hdf5-1.8.7/c++/src/H5DataType.cpp
--- hdf5-1.8.7.orig/c++/src/H5DataType.cpp	2012-02-02 11:26:26.987018618 +0100
+++ hdf5-1.8.7/c++/src/H5DataType.cpp	2012-02-02 11:26:38.877034311 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/src/H5DcreatProp.cpp hdf5-1.8.7/c++/src/H5DcreatProp.cpp
--- hdf5-1.8.7.orig/c++/src/H5DcreatProp.cpp	2012-02-02 11:26:26.987018618 +0100
+++ hdf5-1.8.7/c++/src/H5DcreatProp.cpp	2012-02-02 11:26:35.617028101 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5DxferProp.cpp hdf5-1.8.7/c++/src/H5DxferProp.cpp
--- hdf5-1.8.7.orig/c++/src/H5DxferProp.cpp	2012-02-02 11:26:26.997018629 +0100
+++ hdf5-1.8.7/c++/src/H5DxferProp.cpp	2012-02-02 11:26:37.217030970 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5FcreatProp.cpp hdf5-1.8.7/c++/src/H5FcreatProp.cpp
--- hdf5-1.8.7.orig/c++/src/H5FcreatProp.cpp	2012-02-02 11:26:26.987018618 +0100
+++ hdf5-1.8.7/c++/src/H5FcreatProp.cpp	2012-02-02 11:26:33.937025687 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5AbstractDs.cpp hdf5-1.8.7/c++/src/H5AbstractDs.cpp
--- hdf5-1.8.7.orig/c++/src/H5AbstractDs.cpp	2012-02-02 13:59:23.414524577 +0100
+++ hdf5-1.8.7/c++/src/H5AbstractDs.cpp	2012-02-02 13:59:38.874553071 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5AtomType.cpp hdf5-1.8.7/c++/src/H5AtomType.cpp
--- hdf5-1.8.7.orig/c++/src/H5AtomType.cpp	2012-02-02 13:59:23.424524595 +0100
+++ hdf5-1.8.7/c++/src/H5AtomType.cpp	2012-02-02 13:59:41.814559578 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5DataSpace.cpp hdf5-1.8.7/c++/src/H5DataSpace.cpp
--- hdf5-1.8.7.orig/c++/src/H5DataSpace.cpp	2012-02-02 13:59:23.424524595 +0100
+++ hdf5-1.8.7/c++/src/H5DataSpace.cpp	2012-02-02 13:59:35.594546050 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/src/H5EnumType.cpp hdf5-1.8.7/c++/src/H5EnumType.cpp
--- hdf5-1.8.7.orig/c++/src/H5EnumType.cpp	2012-02-02 15:30:37.365011430 +0100
+++ hdf5-1.8.7/c++/src/H5EnumType.cpp	2012-02-02 15:30:53.295043192 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5PredType.cpp hdf5-1.8.7/c++/src/H5PredType.cpp
--- hdf5-1.8.7.orig/c++/src/H5PredType.cpp	2012-02-02 15:30:37.365011430 +0100
+++ hdf5-1.8.7/c++/src/H5PredType.cpp	2012-02-02 15:30:50.465037500 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5FloatType.cpp hdf5-1.8.7/c++/src/H5FloatType.cpp
--- hdf5-1.8.7.orig/c++/src/H5FloatType.cpp	2012-02-02 15:50:54.127461179 +0100
+++ hdf5-1.8.7/c++/src/H5FloatType.cpp	2012-02-02 15:55:58.008025222 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5IntType.cpp hdf5-1.8.7/c++/src/H5IntType.cpp
--- hdf5-1.8.7.orig/c++/src/H5IntType.cpp	2012-02-02 15:50:54.137461203 +0100
+++ hdf5-1.8.7/c++/src/H5IntType.cpp	2012-02-02 15:55:51.978014661 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5StrType.cpp hdf5-1.8.7/c++/src/H5StrType.cpp
--- hdf5-1.8.7.orig/c++/src/H5StrType.cpp	2012-02-02 15:50:54.137461203 +0100
+++ hdf5-1.8.7/c++/src/H5StrType.cpp	2012-02-02 15:55:54.648019633 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5ArrayType.cpp hdf5-1.8.7/c++/src/H5ArrayType.cpp
--- hdf5-1.8.7.orig/c++/src/H5ArrayType.cpp	2012-02-02 16:14:07.260165102 +0100
+++ hdf5-1.8.7/c++/src/H5ArrayType.cpp	2012-02-02 16:14:22.730208893 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5CompType.cpp hdf5-1.8.7/c++/src/H5CompType.cpp
--- hdf5-1.8.7.orig/c++/src/H5CompType.cpp	2012-02-02 16:14:07.260165102 +0100
+++ hdf5-1.8.7/c++/src/H5CompType.cpp	2012-02-02 16:14:28.500229768 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5VarLenType.cpp hdf5-1.8.7/c++/src/H5VarLenType.cpp
--- hdf5-1.8.7.orig/c++/src/H5VarLenType.cpp	2012-02-02 16:14:07.260165102 +0100
+++ hdf5-1.8.7/c++/src/H5VarLenType.cpp	2012-02-02 16:14:25.460218858 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5CommonFG.cpp hdf5-1.8.7/c++/src/H5CommonFG.cpp
--- hdf5-1.8.7.orig/c++/src/H5CommonFG.cpp	2012-02-02 16:52:44.484863179 +0100
+++ hdf5-1.8.7/c++/src/H5CommonFG.cpp	2012-02-02 16:53:17.974929963 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.7.orig/c++/src/H5DataSet.cpp hdf5-1.8.7/c++/src/H5DataSet.cpp
--- hdf5-1.8.7.orig/c++/src/H5DataSet.cpp	2012-02-02 16:52:44.494863194 +0100
+++ hdf5-1.8.7/c++/src/H5DataSet.cpp	2012-02-02 16:53:07.714909380 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/src/H5File.cpp hdf5-1.8.7/c++/src/H5File.cpp
--- hdf5-1.8.7.orig/c++/src/H5File.cpp	2012-02-02 16:52:44.494863194 +0100
+++ hdf5-1.8.7/c++/src/H5File.cpp	2012-02-02 16:53:23.984942010 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/src/H5Group.cpp hdf5-1.8.7/c++/src/H5Group.cpp
--- hdf5-1.8.7.orig/c++/src/H5Group.cpp	2012-02-02 16:52:44.494863194 +0100
+++ hdf5-1.8.7/c++/src/H5Group.cpp	2012-02-02 16:53:20.684935117 +0100
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/dsets.cpp hdf5-1.8.7/c++/test/dsets.cpp
--- hdf5-1.8.7.orig/c++/test/dsets.cpp	2012-02-02 21:11:23.624694720 +0100
+++ hdf5-1.8.7/c++/test/dsets.cpp	2012-02-02 21:11:41.504731006 +0100
@@ -25,6 +25,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/tattr.cpp hdf5-1.8.7/c++/test/tattr.cpp
--- hdf5-1.8.7.orig/c++/test/tattr.cpp	2012-02-02 21:11:23.624694720 +0100
+++ hdf5-1.8.7/c++/test/tattr.cpp	2012-02-02 21:11:35.284719238 +0100
@@ -20,6 +20,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/tcompound.cpp hdf5-1.8.7/c++/test/tcompound.cpp
--- hdf5-1.8.7.orig/c++/test/tcompound.cpp	2012-02-02 21:11:23.624694720 +0100
+++ hdf5-1.8.7/c++/test/tcompound.cpp	2012-02-02 21:11:32.164712986 +0100
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/testhdf5.cpp hdf5-1.8.7/c++/test/testhdf5.cpp
--- hdf5-1.8.7.orig/c++/test/testhdf5.cpp	2012-02-02 21:11:23.624694720 +0100
+++ hdf5-1.8.7/c++/test/testhdf5.cpp	2012-02-02 21:11:28.234703541 +0100
@@ -41,6 +41,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/tfile.cpp hdf5-1.8.7/c++/test/tfile.cpp
--- hdf5-1.8.7.orig/c++/test/tfile.cpp	2012-02-02 21:11:23.624694720 +0100
+++ hdf5-1.8.7/c++/test/tfile.cpp	2012-02-02 21:11:45.134736604 +0100
@@ -23,6 +23,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/tfilter.cpp hdf5-1.8.7/c++/test/tfilter.cpp
--- hdf5-1.8.7.orig/c++/test/tfilter.cpp	2012-02-02 21:11:23.624694720 +0100
+++ hdf5-1.8.7/c++/test/tfilter.cpp	2012-02-02 21:11:38.224724841 +0100
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/th5s.cpp hdf5-1.8.7/c++/test/th5s.cpp
--- hdf5-1.8.7.orig/c++/test/th5s.cpp	2012-02-02 21:11:23.624694720 +0100
+++ hdf5-1.8.7/c++/test/th5s.cpp	2012-02-02 21:11:48.604742444 +0100
@@ -22,6 +22,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/src/H5make_libsettings.c hdf5-1.8.7/src/H5make_libsettings.c
--- hdf5-1.8.7.orig/src/H5make_libsettings.c	2012-02-02 21:48:30.909192384 +0100
+++ hdf5-1.8.7/src/H5make_libsettings.c	2012-02-02 21:48:34.919202040 +0100
@@ -41,6 +41,7 @@
  *-------------------------------------------------------------------------
  */
 
+#include <mpi.h>
 #include <stdio.h>
 #include <time.h>
 #include "H5private.h"
diff -ru hdf5-1.8.7.orig/c++/test/h5cpputil.cpp hdf5-1.8.7/c++/test/h5cpputil.cpp
--- hdf5-1.8.7.orig/c++/test/h5cpputil.cpp	2012-02-02 21:57:45.880427519 +0100
+++ hdf5-1.8.7/c++/test/h5cpputil.cpp	2012-02-02 21:57:57.820457924 +0100
@@ -21,6 +21,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/tlinks.cpp hdf5-1.8.7/c++/test/tlinks.cpp
--- hdf5-1.8.7.orig/c++/test/tlinks.cpp	2012-02-02 21:57:45.880427519 +0100
+++ hdf5-1.8.7/c++/test/tlinks.cpp	2012-02-02 21:57:49.080434882 +0100
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/trefer.cpp hdf5-1.8.7/c++/test/trefer.cpp
--- hdf5-1.8.7.orig/c++/test/trefer.cpp	2012-02-02 21:57:45.880427519 +0100
+++ hdf5-1.8.7/c++/test/trefer.cpp	2012-02-02 21:57:51.680440980 +0100
@@ -20,6 +20,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/ttypes.cpp hdf5-1.8.7/c++/test/ttypes.cpp
--- hdf5-1.8.7.orig/c++/test/ttypes.cpp	2012-02-02 21:57:45.880427519 +0100
+++ hdf5-1.8.7/c++/test/ttypes.cpp	2012-02-02 21:57:55.990452298 +0100
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.7.orig/c++/test/tvlstr.cpp hdf5-1.8.7/c++/test/tvlstr.cpp
--- hdf5-1.8.7.orig/c++/test/tvlstr.cpp	2012-02-02 21:57:45.880427519 +0100
+++ hdf5-1.8.7/c++/test/tvlstr.cpp	2012-02-02 21:57:53.620445608 +0100
@@ -21,6 +21,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
