Create parent directories during install
Author: B. Hajgato
Updated: Orient Kružík (INUITS)
--- migrate-5.0.3_orig/src/configure	2021-11-08 22:48:21.000000000 +0100
+++ migrate-5.0.3/src/configure	2022-05-09 17:15:29.368752420 +0200
@@ -4104,7 +4104,7 @@
 	      test -s conftest.dir/conftest.one &&
 	      test -s conftest.dir/conftest.two
 	    then
-	      ac_cv_path_install="$as_dir$ac_prog$ac_exec_ext -c"
+	      ac_cv_path_install="$as_dir$ac_prog$ac_exec_ext -c -D"
 	      break 3
 	    fi
 	  fi