diff -ru migrate-4.4.4.orig/src/configure migrate-4.4.4/src/configure
--- migrate-4.4.4.orig/src/configure	2021-06-15 13:57:08.185880579 +0000
+++ migrate-4.4.4/src/configure	2021-06-15 13:57:56.112195739 +0000
@@ -3403,7 +3403,7 @@
 	      test -s conftest.dir/conftest.one &&
 	      test -s conftest.dir/conftest.two
 	    then
-	      ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c"
+	      ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c -D"
 	      break 3
 	    fi
 	  fi
diff -ru migrate-4.4.4.orig/src/Makefile.in migrate-4.4.4/src/Makefile.in
--- migrate-4.4.4.orig/src/Makefile.in	2021-06-15 13:57:08.185880579 +0000
+++ migrate-4.4.4/src/Makefile.in	2021-06-15 14:12:20.478803484 +0000
@@ -79,7 +79,7 @@
 MT_FILES          = SFMT.c
 #
 # to optimize Mersenne-Twister random number generator use ./configure --enable-sse2 on INTEL
-HARDWARECFLAGS    = @HARDWARECFLAGS@ @SYSTEMVERSION@
+HARDWARECFLAGS    = @HARDWARECFLAGS@ @SYSTEMVERSION@ $(OPTFLAGS)
 #
 #
 PAPERFORMAT       = @PAPERFORMAT@
@@ -177,7 +177,7 @@
 HARUDIR           = haru
 MTDIR             = SFMT-src-1.4.1
 OFILES            = $(CFILES:.c=.o) $(CPLUSFILES:.cpp=.o) $(MT_FILES:.c=.o)
-INSTALL_DEPENDS   = $(INSTALLDIR) all 
+INSTALL_DEPENDS   = $(INSTALLDIR) 
 PRODUCT_DEPENDS   = $(OFILES)
 
 .c.o:
@@ -327,7 +327,7 @@
 
 install:: $(INSTALL_DEPENDS)
 	$(INSTALL) $(IFLAGS) $(NAME) $(INSTALLDIR)
-	#$(INSTALL) $(IFLAGS) $(MPINAME) $(INSTALLDIR)
+	$(INSTALL) $(IFLAGS) $(MPINAME) $(INSTALLDIR)
 	$(INSTALL) $(IFLAGS) $(MANPAGE) $(MANPAGEDIR)
 
 kit:: haru $(ZLIBTARGET)
