#Create parent directories during install
#B. Hajgato January 18th, 2016
--- src/configure.orig	2014-12-28 03:46:09.000000000 +0100
+++ src/configure	2016-01-18 14:57:26.331047715 +0100
@@ -3411,7 +3411,7 @@
 	      test -s conftest.dir/conftest.one &&
 	      test -s conftest.dir/conftest.two
 	    then
-	      ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c"
+	      ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c -D"
 	      break 3
 	    fi
 	  fi
