#Create parent directories during install
#B. Hajgato 3rd September 2015
--- src/configure.old	2015-05-02 14:42:10.000000000 +0200
+++ src/configure	2015-09-03 10:00:54.320014335 +0200
@@ -3412,7 +3412,7 @@
 	      test -s conftest.dir/conftest.one &&
 	      test -s conftest.dir/conftest.two
 	    then
-	      ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c"
+	      ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c -D"
 	      break 3
 	    fi
 	  fi
