--- ViennaRNA-2.1.6/Kinfold/cache.c.orig	2014-03-17 14:07:33.308562799 +0100
+++ ViennaRNA-2.1.6/Kinfold/cache.c	2014-03-17 14:07:48.808965449 +0100
@@ -56,7 +56,6 @@
                 61,59,53,47,43,41,37,31,29,23,17,13,11,7,3,1};
 
 /* key must not be longer than 128 */
-#pragma inline (cache_f)
 unsigned cache_f(char *x) { 
   register char *s;
   register int i;
@@ -80,6 +79,7 @@
   int cacheval;
   cache_entry *c;
 
+  #pragma inline (cache_f)
   cacheval=cache_f(x);
   if ((c=cachetab[cacheval]))
     if (strcmp(c->structure,x)==0) return c;
@@ -92,6 +92,7 @@
   int cacheval;
   cache_entry *c;
   
+  #pragma inline (cache_f)
   cacheval=cache_f(x->structure);
   if ((c=cachetab[cacheval])) {
     free(c->structure);
