diff -ru ViennaRNA-2.0.7.orig/Kinfold/cache.c ViennaRNA-2.0.7/Kinfold/cache.c
--- ViennaRNA-2.0.7.orig/Kinfold/cache.c	2011-09-26 03:06:41.000000000 +0200
+++ ViennaRNA-2.0.7/Kinfold/cache.c	2012-10-09 16:28:02.033331859 +0200
@@ -56,7 +56,6 @@
                 61,59,53,47,43,41,37,31,29,23,17,13,11,7,3,1};
 
 /* key must not be longer than 128 */
-#pragma inline (cache_f)
 unsigned cache_f(char *x) { 
   register char *s;
   register int i;
@@ -80,6 +79,7 @@
   int cacheval;
   cache_entry *c;
 
+  #pragma inline (cache_f)
   cacheval=cache_f(x);
   if ((c=cachetab[cacheval]))
     if (strcmp(c->structure,x)==0) return c;
@@ -92,6 +92,7 @@
   int cacheval;
   cache_entry *c;
   
+  #pragma inline (cache_f)
   cacheval=cache_f(x->structure);
   if ((c=cachetab[cacheval])) {
     free(c->structure);
