# Fix linking error by hardcoding FlexiBLAS
--- ViennaRNA-2.7.0/src/RNAxplorer/configure.ac	2024-10-11 11:37:12.000000000 +0000
+++ ViennaRNA-2.7.0.new/src/RNAxplorer/configure.ac	2025-06-09 11:43:47.459093146 +0000
@@ -41,7 +41,7 @@
 AX_LAPACK([
       AC_CHECK_HEADERS([openblas/lapacke.h lapacke/lapacke.h lapacke.h],[rx_lapacke_header=yes;break;],)
       AS_IF([test "x$rx_lapacke_header" = "xyes"],[
-        LDFLAGS="$LDFLAGS -llapack"
+        LDFLAGS="$LDFLAGS -lflexiblas"
       ],[
         AC_MSG_ERROR([could not find lapacke.h])
       ])
