add missing include for C-style I/O, required because of printf
author: Kenneth Hoste (HPC-UGent)
--- babel-1.4.0/regression/enums/libCxx/enums_cartest_Impl.cxx.orig	2015-10-15 12:51:49.096618972 +0200
+++ babel-1.4.0/regression/enums/libCxx/enums_cartest_Impl.cxx	2015-10-15 12:52:06.326888680 +0200
@@ -8,6 +8,7 @@
 // WARNING: Automatically generated; only changes within splicers preserved
 // 
 // 
+#include <cstdio>
 #include "enums_cartest_Impl.hxx"
 
 // 
