From 844db0b9aaf1cbef3860eb7ec539bc0648918c82 Mon Sep 17 00:00:00 2001
From: Olivier Hoenen <Olivier.Hoenen@iter.org>
Date: Wed, 31 Jan 2024 16:19:27 +0100
Subject: [PATCH 1/2] Add missing C++ header for GCC 13 (#26)

---
 source/wrappers/c++/client.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source/wrappers/c++/client.hpp b/source/wrappers/c++/client.hpp
index 9a036d8c..9b881cdc 100755
--- a/source/wrappers/c++/client.hpp
+++ b/source/wrappers/c++/client.hpp
@@ -5,6 +5,7 @@
 #include <vector>
 #include <exception>
 #include <unordered_map>
+#include <cstdint>
 
 #include <clientserver/export.h>
 
-- 
2.18.4

From af7b114858d2831c0ab15c270f63e1fa993880f2 Mon Sep 17 00:00:00 2001
From: Olivier Hoenen <Olivier.Hoenen@iter.org>
Date: Wed, 31 Jan 2024 16:21:07 +0100
Subject: [PATCH 2/2] Update source and target for jdk, required for building
 with Java 21 (#26)

---
 source/wrappers/java/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/wrappers/java/CMakeLists.txt b/source/wrappers/java/CMakeLists.txt
index 4dc6ded8..d2903303 100755
--- a/source/wrappers/java/CMakeLists.txt
+++ b/source/wrappers/java/CMakeLists.txt
@@ -18,7 +18,7 @@ endif()
 
 include( UseJava )
 
-set( CMAKE_JAVA_COMPILE_FLAGS "-source" "1.6" "-target" "1.6" )
+set( CMAKE_JAVA_COMPILE_FLAGS "-source" "1.8" "-target" "1.8" )
 
 include_directories(
   ${CMAKE_SOURCE_DIR}/source
-- 
2.18.4

