--- test/python/base/tail_issues.py	2021-08-17 15:05:28.378668396 +0000
+++ test/python/base/tail_issues.py	2021-08-17 15:05:50.846691101 +0000
@@ -134,7 +134,7 @@
         dt = make_gf_from_fourier(d, tau_mesh, tail)
         max_im = np.max(np.abs(dt.data.imag))
         # print "Imag Delta", max_im
-        self.assertTrue(max_im < 1e-12)
+        self.assertTrue(max_im < 2e-12)
 
         # Check Delta(iw) extracted through triqs.gf.tools.delta
         D_exact = g.copy()
