see http://ubuntuforums.org/archive/index.php/t-1281327.html
--- want-2.3.0/ctools/memstat.c.orig	2012-12-04 10:44:18.732658162 +0100
+++ want-2.3.0/ctools/memstat.c	2012-12-04 10:44:32.664809437 +0100
@@ -7,6 +7,7 @@
 */
 
 #include "c_defs.h"
+#include <malloc.h>
 
 /* 
   This function return the numer of kilobytes allocated
@@ -17,7 +18,6 @@
 void F77_FUNC(memstat,MEMSTAT)(int *kilobytes)
 {
 #if defined(HAVE_MALLINFO)
-#include <malloc.h>
   struct mallinfo info;  
   info = mallinfo();
   *kilobytes = info.arena / 1024 ;
