diff -ru yambo-3.2.1-r.448.orig/interfaces/a2y/.objects yambo-3.2.1-r.448/interfaces/a2y/.objects
--- yambo-3.2.1-r.448.orig/interfaces/a2y/.objects	2009-07-07 13:58:56.000000000 +0200
+++ yambo-3.2.1-r.448/interfaces/a2y/.objects	2012-12-06 13:50:25.210969961 +0100
@@ -1,2 +1 @@
-objs = defs_datatypes.o hdr_io.o \
-       a2y_KSS_file_name.o a2y_i.o a2y_db1.o a2y_wf.o
+objs = defs_datatypes.o hdr_io.o a2y_KSS_file_name.o a2y_i.o a2y_db1.o a2y_wf.o
diff -ru yambo-3.2.1-r.448.orig/interfaces/p2y/.objects yambo-3.2.1-r.448/interfaces/p2y/.objects
--- yambo-3.2.1-r.448.orig/interfaces/p2y/.objects	2009-07-07 13:58:56.000000000 +0200
+++ yambo-3.2.1-r.448/interfaces/p2y/.objects	2012-12-06 13:50:58.556379123 +0100
@@ -1,26 +1,18 @@
 #if defined _P2Y_EXPORT
 
-objs = pw_errore.o mod_pw_data.o mod_pw_export.o \
-       mod_p2y.o \
-       p2y_db1.o p2y_wf.o p2y_i.o 
+objs = pw_errore.o mod_pw_data.o mod_pw_export.o mod_p2y.o p2y_db1.o p2y_wf.o p2y_i.o 
 
 #elif defined _P2Y_V31 || defined _P2Y_V311
 
-objs = pw_errore.o mod_pw_data.o qexml_v3.1.o  \
-       mod_p2y.o \
-       p2y_db1.o p2y_wf.o p2y_i.o
+objs = pw_errore.o mod_pw_data.o qexml_v3.1.o mod_p2y.o p2y_db1.o p2y_wf.o p2y_i.o
 
 #elif defined _P2Y_V32
 
-objs = pw_errore.o mod_pw_data.o qexml_v3.2.o \
-       mod_p2y.o \
-       p2y_db1.o p2y_wf.o p2y_i.o
+objs = pw_errore.o mod_pw_data.o qexml_v3.2.o mod_p2y.o p2y_db1.o p2y_wf.o p2y_i.o
 
 #elif defined _P2Y_V40
 
-objs = pw_errore.o mod_pw_data.o qexml_v4.0.o \
-       mod_p2y.o \
-       p2y_db1.o p2y_wf.o p2y_i.o
+objs = pw_errore.o mod_pw_data.o qexml_v4.0.o mod_p2y.o p2y_db1.o p2y_wf.o p2y_i.o
 
 #endif
 
diff -ru yambo-3.2.1-r.448.orig/lib/blas/.objects yambo-3.2.1-r.448/lib/blas/.objects
--- yambo-3.2.1-r.448.orig/lib/blas/.objects	2009-07-07 13:58:57.000000000 +0200
+++ yambo-3.2.1-r.448/lib/blas/.objects	2012-12-06 13:50:04.234712649 +0100
@@ -1,15 +1,5 @@
 #if defined _DOUBLE
-objs =  zaxpy.o  zdotc.o  zgemm.o  zgeru.o  zswap.o  ztrmv.o  \
-	izamax.o dznrm2.o zcopy.o  zdotu.o  zgemv.o  zscal.o  \
-	ztrmm.o  ztrsm.o  daxpy.o  dscal.o  zher2k.o zher2.o  \
-	zhemv.o  zsyr2k.o zgerc.o  zher.o   dcopy.o  idamax.o \
-	dzasum.o ztrsv.o  xerbla.o zsymm.o  zhemm.o  zsyrk.o  \
-        zherk.o  zdscal.o dcabs1.o ddot.o
+objs = zaxpy.o zdotc.o zgemm.o zgeru.o zswap.o ztrmv.o 	izamax.o dznrm2.o zcopy.o zdotu.o zgemv.o zscal.o 	ztrmm.o ztrsm.o daxpy.o dscal.o zher2k.o zher2.o 	zhemv.o zsyr2k.o zgerc.o zher.o dcopy.o idamax.o 	dzasum.o ztrsv.o xerbla.o zsymm.o zhemm.o zsyrk.o zherk.o zdscal.o dcabs1.o ddot.o
 #else
-objs =  caxpy.o  cdotc.o  cgemm.o  cgeru.o  cswap.o  ctrmv.o  \
-	icamax.o scnrm2.o ccopy.o  cdotu.o  cgemv.o  cscal.o  \
-	ctrmm.o  ctrsm.o  saxpy.o  sscal.o  cher2k.o cher2.o  \
-	chemv.o  csyr2k.o cgerc.o  cher.o   scopy.o  isamax.o \
-	scasum.o ctrsv.o  xerbla.o csymm.o  chemm.o  csyrk.o  \
-        cherk.o  csscal.o sdot.o
+objs = caxpy.o cdotc.o cgemm.o cgeru.o cswap.o ctrmv.o 	icamax.o scnrm2.o ccopy.o cdotu.o cgemv.o cscal.o 	ctrmm.o ctrsm.o saxpy.o sscal.o cher2k.o cher2.o 	chemv.o csyr2k.o cgerc.o cher.o scopy.o isamax.o 	scasum.o ctrsv.o xerbla.o csymm.o chemm.o csyrk.o cherk.o csscal.o sdot.o
 #endif
diff -ru yambo-3.2.1-r.448.orig/lib/lapack/.objects yambo-3.2.1-r.448/lib/lapack/.objects
--- yambo-3.2.1-r.448.orig/lib/lapack/.objects	2009-07-07 13:58:58.000000000 +0200
+++ yambo-3.2.1-r.448/lib/lapack/.objects	2012-12-06 13:50:15.482850656 +0100
@@ -1,51 +1,5 @@
 #if defined _DOUBLE
-objs = 	zgetf2.o  zhetrd.o  zlarft.o  zlaswp.o  \
-	zung2l.o  ilaenv.o  dlanst.o  dlassq.o  \
-	zgetrf.o  zlacgv.o  zlascl.o  zlatrd.o  \
-	zungql.o  lsame.o   dlapy2.o  dsterf.o  \
-	zgetri.o  zlanhe.o  zlaset.o  zsteqr.o  \
-	zungqr.o  dlae2.o   dlartg.o  zheev.o   \
-    zheevr.o \
-	zlarfb.o  zlasr.o   ztrti2.o  zungtr.o  \
-	dlaev2.o  dlascl.o  zhetd2.o  zlarfg.o  \
-	zlassq.o  ztrtri.o  ieeeck.o \
-	dlasrt.o  zhetri.o  zhetrf.o \
-	zladiv.o  zlarf.o   zung2r.o  dlapy3.o  \
-	dladiv.o  zlahef.o  zhetf2.o \
-	zgesvd.o  zlange.o  zgeqrf.o  zgebrd.o  \
-	zungbr.o  zbdsqr.o  zlacpy.o  zunmbr.o  \
-	zgelqf.o  zunglq.o  zgeqr2.o  zgebd2.o  \
-	dlasq1.o  dlasv2.o  zdrot.o   dlas2.o  \
-	zlabrd.o  zunmqr.o  zunmlq.o  zungl2.o \
-	zgelq2.o  dlasq2.o  zunm2r.o  zunml2.o \
-	dlasq3.o  dlasq4.o  dlasq5.o  dlasq6.o \
-	zgebal.o  zgebak.o  ztrevc.o  zhseqr.o \
-	zunghr.o  zgehrd.o  dlabad.o \
-	zgeev.o   zgehd2.o  zlahqr.o \
-	zlahrd.o  zlanhs.o  zlarfx.o  zlatrs.o dlamch.o dlaran.o
+objs = 	zgetf2.o zhetrd.o zlarft.o zlaswp.o 	zung2l.o ilaenv.o dlanst.o dlassq.o 	zgetrf.o zlacgv.o zlascl.o zlatrd.o 	zungql.o lsame.o dlapy2.o dsterf.o 	zgetri.o zlanhe.o zlaset.o zsteqr.o 	zungqr.o dlae2.o dlartg.o zheev.o zheevr.o 	zlarfb.o zlasr.o ztrti2.o zungtr.o 	dlaev2.o dlascl.o zhetd2.o zlarfg.o 	zlassq.o ztrtri.o ieeeck.o 	dlasrt.o zhetri.o zhetrf.o 	zladiv.o zlarf.o zung2r.o dlapy3.o 	dladiv.o zlahef.o zhetf2.o 	zgesvd.o zlange.o zgeqrf.o zgebrd.o 	zungbr.o zbdsqr.o zlacpy.o zunmbr.o 	zgelqf.o zunglq.o zgeqr2.o zgebd2.o 	dlasq1.o dlasv2.o zdrot.o dlas2.o 	zlabrd.o zunmqr.o zunmlq.o zungl2.o 	zgelq2.o dlasq2.o zunm2r.o zunml2.o 	dlasq3.o dlasq4.o dlasq5.o dlasq6.o 	zgebal.o zgebak.o ztrevc.o zhseqr.o 	zunghr.o zgehrd.o dlabad.o 	zgeev.o zgehd2.o zlahqr.o 	zlahrd.o zlanhs.o zlarfx.o zlatrs.o dlamch.o dlaran.o
 #else
-objs = 	cgetf2.o  chetrd.o  clarft.o  claswp.o  \
-	cung2l.o  ilaenv.o  slanst.o  slassq.o  \
-	cgetrf.o  clacgv.o  clascl.o  clatrd.o  \
-	cungql.o  lsame.o   slapy2.o  ssterf.o  \
-	cgetri.o  clanhe.o  claset.o  csteqr.o  \
-	cungqr.o  slae2.o   slartg.o  cheev.o   \
-    cheevr.o \
-	clarfb.o  clasr.o   ctrti2.o  cungtr.o  \
-	slaev2.o  slascl.o  chetd2.o  clarfg.o  \
-	classq.o  ctrtri.o  ieeeck.o \
-	slasrt.o  chetri.o  chetrf.o \
-	cladiv.o  clarf.o   cung2r.o  slapy3.o  \
-	sladiv.o  clahef.o  chetf2.o \
-	cgesvd.o  clange.o  cgeqrf.o  cgebrd.o  \
-	cungbr.o  cbdsqr.o  clacpy.o  cunmbr.o  \
-	cgelqf.o  cunglq.o  cgeqr2.o  cgebd2.o  \
-	slasq1.o  slasv2.o  csrot.o   slas2.o  \
-	clabrd.o  cunmqr.o  cunmlq.o  cungl2.o \
-	cgelq2.o  slasq2.o  cunm2r.o  cunml2.o \
-	slasq3.o  slasq4.o  slasq5.o  slasq6.o \
-	cgebal.o  cgebak.o  ctrevc.o  chseqr.o \
-	cunghr.o  cgehrd.o  slabad.o \
-	cgeev.o   cgehd2.o  clahqr.o \
-	clahrd.o  clanhs.o  clarfx.o  clatrs.o slamch.o dlaran.o
+objs = 	cgetf2.o chetrd.o clarft.o claswp.o 	cung2l.o ilaenv.o slanst.o slassq.o 	cgetrf.o clacgv.o clascl.o clatrd.o 	cungql.o lsame.o slapy2.o ssterf.o 	cgetri.o clanhe.o claset.o csteqr.o 	cungqr.o slae2.o slartg.o cheev.o cheevr.o 	clarfb.o clasr.o ctrti2.o cungtr.o 	slaev2.o slascl.o chetd2.o clarfg.o 	classq.o ctrtri.o ieeeck.o 	slasrt.o chetri.o chetrf.o 	cladiv.o clarf.o cung2r.o slapy3.o 	sladiv.o clahef.o chetf2.o 	cgesvd.o clange.o cgeqrf.o cgebrd.o 	cungbr.o cbdsqr.o clacpy.o cunmbr.o 	cgelqf.o cunglq.o cgeqr2.o cgebd2.o 	slasq1.o slasv2.o csrot.o slas2.o 	clabrd.o cunmqr.o cunmlq.o cungl2.o 	cgelq2.o slasq2.o cunm2r.o cunml2.o 	slasq3.o slasq4.o slasq5.o slasq6.o 	cgebal.o cgebak.o ctrevc.o chseqr.o 	cunghr.o cgehrd.o slabad.o 	cgeev.o cgehd2.o clahqr.o 	clahrd.o clanhs.o clarfx.o clatrs.o slamch.o dlaran.o
 #endif
diff -ru yambo-3.2.1-r.448.orig/lib/slatec/.objects yambo-3.2.1-r.448/lib/slatec/.objects
--- yambo-3.2.1-r.448.orig/lib/slatec/.objects	2009-07-07 13:58:57.000000000 +0200
+++ yambo-3.2.1-r.448/lib/slatec/.objects	2012-12-06 13:50:20.794915816 +0100
@@ -1,21 +1,5 @@
 #if defined _DOUBLE
-objs = 	davint.o  xermsg.o  fdump.o   j4save.o  xercnt.o  xerhlt.o \
-	    xerprn.o  xersve.o  i1mach.o  xgetua.o \
-        dpolft.o  dpcoef.o  dp1vlu.o \
-        dbesj0.o  dbesj1.o  dbesk0.o  dbesk1.o  dcsevl.o  initds.o  \
-        d1mach.o  dbesi0.o  dbsk0e.o  dbesi1.o  dbsk1e.o  dbsi0e.o \
-        dbsi1e.o  dqage.o   dqag.o    dqk15.o   dqk21.o   dqk31.o   \
-        dqk41.o   dqk51.o   dqk61.o   dqpsrt.o  dqawo.o   dqawoe.o \
-        dqc25f.o  dqelg.o   dqcheb.o  dqwgtf.o  dqk15w.o  dqwgtf.o \
-        dgtsl.o   d9b0mp.o  d9b1mp.o
+objs = 	davint.o xermsg.o fdump.o j4save.o xercnt.o xerhlt.o 	 xerprn.o xersve.o i1mach.o xgetua.o dpolft.o dpcoef.o dp1vlu.o dbesj0.o dbesj1.o dbesk0.o dbesk1.o dcsevl.o initds.o d1mach.o dbesi0.o dbsk0e.o dbesi1.o dbsk1e.o dbsi0e.o dbsi1e.o dqage.o dqag.o dqk15.o dqk21.o dqk31.o dqk41.o dqk51.o dqk61.o dqpsrt.o dqawo.o dqawoe.o dqc25f.o dqelg.o dqcheb.o dqwgtf.o dqk15w.o dqwgtf.o dgtsl.o d9b0mp.o d9b1mp.o
 #else
-objs = 	avint.o  xermsg.o fdump.o  j4save.o xercnt.o xerhlt.o \
-	    xerprn.o xersve.o i1mach.o xgetua.o \
-        polfit.o  pcoef.o  pvalue.o \
-        besj0.o  besj1.o  besk0.o  besk1.o  csevl.o  inits.o  \
-        r1mach.o besi0.o  besk0e.o besi1.o  besk1e.o besi0e.o \
-        besi1e.o qage.o   qag.o    qk15.o   qk21.o   qk31.o   \
-        qk41.o   qk51.o   qk61.o   qpsrt.o  qawo.o  qawoe.o \
-        qc25f.o  qelg.o   qcheb.o  qwgtf.o  qk15w.o  qwgtf.o \
-        sgtsl.o       
+objs = 	avint.o xermsg.o fdump.o j4save.o xercnt.o xerhlt.o 	 xerprn.o xersve.o i1mach.o xgetua.o polfit.o pcoef.o pvalue.o besj0.o besj1.o besk0.o besk1.o csevl.o inits.o r1mach.o besi0.o besk0e.o besi1.o besk1e.o besi0e.o besi1e.o qage.o qag.o qk15.o qk21.o qk31.o qk41.o qk51.o qk61.o qpsrt.o qawo.o qawoe.o qc25f.o qelg.o qcheb.o qwgtf.o qk15w.o qwgtf.o sgtsl.o 
 #endif
diff -ru yambo-3.2.1-r.448.orig/src/bse/.objects yambo-3.2.1-r.448/src/bse/.objects
--- yambo-3.2.1-r.448.orig/src/bse/.objects	2009-07-07 13:58:53.000000000 +0200
+++ yambo-3.2.1-r.448/src/bse/.objects	2012-12-06 13:49:43.307455892 +0100
@@ -1,5 +1 @@
-objs =	K_driver.o K_eh_setup.o \
-        K.o K_scattering.o K_exchange.o \
-        K_solvers.o K_Haydock.o K_diagonalization.o \
-        K_dump_to_o_file.o Kernel_by_V.o \
-        K_filling.o
+objs =	K_driver.o K_eh_setup.o K.o K_scattering.o K_exchange.o K_solvers.o K_Haydock.o K_diagonalization.o K_dump_to_o_file.o Kernel_by_V.o K_filling.o
diff -ru yambo-3.2.1-r.448.orig/src/bz_ops/.objects yambo-3.2.1-r.448/src/bz_ops/.objects
--- yambo-3.2.1-r.448.orig/src/bz_ops/.objects	2009-07-07 13:58:55.000000000 +0200
+++ yambo-3.2.1-r.448/src/bz_ops/.objects	2012-12-06 13:49:51.611557776 +0100
@@ -1,2 +1 @@
-objs = k_lattice.o k_expand.o k_ibz2bz.o k_reduce.o k_sym2sym.o \
-       bz_samp_indexes.o  k_the_nearest.o
+objs = k_lattice.o k_expand.o k_ibz2bz.o k_reduce.o k_sym2sym.o bz_samp_indexes.o k_the_nearest.o
diff -ru yambo-3.2.1-r.448.orig/src/common/.objects yambo-3.2.1-r.448/src/common/.objects
--- yambo-3.2.1-r.448.orig/src/common/.objects	2009-07-07 13:58:53.000000000 +0200
+++ yambo-3.2.1-r.448/src/common/.objects	2012-12-06 13:49:33.771338921 +0100
@@ -1,5 +1 @@
-objs = eval_G_minus_G.o damping.o \
-       fermi_level.o pol_fit.o \
-       coarse_grid.o kk.o freqs_setup.o \
-       extend_occupations.o QP_state_extract.o QP_state_table_setup.o \
-       crystal_lattice.o
+objs = eval_G_minus_G.o damping.o fermi_level.o pol_fit.o coarse_grid.o kk.o freqs_setup.o extend_occupations.o QP_state_extract.o QP_state_table_setup.o crystal_lattice.o
diff -ru yambo-3.2.1-r.448.orig/src/coulomb/.objects yambo-3.2.1-r.448/src/coulomb/.objects
--- yambo-3.2.1-r.448.orig/src/coulomb/.objects	2009-07-07 13:58:53.000000000 +0200
+++ yambo-3.2.1-r.448/src/coulomb/.objects	2012-12-06 13:49:37.787388198 +0100
@@ -1,4 +1 @@
-objs = col_driver.o rim.o rim_integrate.o \
-       bessel_F2.o bessel_F3.o bessel_F4.o bessel_F5.o bessel_F6.o bessel_J0.o \
-       cutoff_driver.o cutoff_sphere.o cutoff_box.o \
-       cutoff_test.o cutoff_cylinder.o
+objs = col_driver.o rim.o rim_integrate.o bessel_F2.o bessel_F3.o bessel_F4.o bessel_F5.o bessel_F6.o bessel_J0.o cutoff_driver.o cutoff_sphere.o cutoff_box.o cutoff_test.o cutoff_cylinder.o
diff -ru yambo-3.2.1-r.448.orig/src/interface/.objects yambo-3.2.1-r.448/src/interface/.objects
--- yambo-3.2.1-r.448.orig/src/interface/.objects	2009-07-07 13:58:55.000000000 +0200
+++ yambo-3.2.1-r.448/src/interface/.objects	2012-12-06 13:49:05.097987151 +0100
@@ -1,3 +1 @@
-objs = init_load.o init.o QP_init.o setup.o G_shells_finder.o \
-       parse_K_list.o \
-       init_q_pts.o  stop_now.o barriers.o check_periodic_directions.o
+objs = init_load.o init.o QP_init.o setup.o G_shells_finder.o parse_K_list.o init_q_pts.o stop_now.o barriers.o check_periodic_directions.o
diff -ru yambo-3.2.1-r.448.orig/src/io/.objects yambo-3.2.1-r.448/src/io/.objects
--- yambo-3.2.1-r.448.orig/src/io/.objects	2009-07-07 13:58:53.000000000 +0200
+++ yambo-3.2.1-r.448/src/io/.objects	2012-12-06 13:49:17.130134736 +0100
@@ -1,11 +1,3 @@
 RAS_objects =
 SC_objects =
-objs = ver_is_gt_or_eq.o IO_mute.o \
-       ioRIM.o        ioGROT.o     \
-       ioOSTNTS.o     ioQINDX.o    ioQP.o      ioX.o \
-       ioDB1.o        ioWF.o       ioKB_PP.o   ioXXVXC.o \
-       ioBS.o         \
-       ioBSS_diago.o  ioBSS_Haydock.o \
-       io_elemental.o io_bulk.o      io_header.o \
-       ioCOL_CUT.o    variables_BS.o  $(RAS_objects) \
-       $(SC_objects)
+objs = ver_is_gt_or_eq.o IO_mute.o ioRIM.o ioGROT.o ioOSTNTS.o ioQINDX.o ioQP.o ioX.o ioDB1.o ioWF.o ioKB_PP.o ioXXVXC.o ioBS.o ioBSS_diago.o ioBSS_Haydock.o io_elemental.o io_bulk.o io_header.o ioCOL_CUT.o variables_BS.o $(RAS_objects) $(SC_objects)
--- yambo-3.2.1-r.448.orig/src/modules/.objects	2009-07-07 13:58:53.000000000 +0200
+++ yambo-3.2.1-r.448/src/modules/.objects	2012-12-06 13:49:49.003525784 +0100
diff -ru yambo-3.2.1-r.448.orig/src/modules/.objects yambo-3.2.1-r.448/src/modules/.objects
@@ -1,13 +1,3 @@
 RT_SC_objects =
 ELPH_objects =
-objs = mod_pars.o mod_units.o  mod_stderr.o  mod_par_proc.o \
-       mod_collision.o mod_drivers.o  mod_FFT.o \
-       mod_timing.o  mod_logo.o \
-       mod_com.o mod_memory.o   mod_R_lattice.o  \
-       mod_par_indexes.o mod_wave_func.o \
-       mod_matrix_operate.o mod_D_lattice.o mod_frequency.o\
-       mod_wave_func.o mod_vec_operate.o mod_electrons.o mod_X.o \
-       mod_functions.o mod_zeros.o mod_pseudo.o \
-       mod_BS.o mod_QP.o mod_TDDFT.o mod_ACFDT.o \
-       mod_IO.o $(ELPH_objects) mod_xc_functionals.o $(RT_SC_objects)  \
-       mod_global_XC.o std_presets.o 
+objs = mod_pars.o mod_units.o mod_stderr.o mod_par_proc.o mod_collision.o mod_drivers.o mod_FFT.o mod_timing.o mod_logo.o mod_com.o mod_memory.o mod_R_lattice.o mod_par_indexes.o mod_wave_func.o mod_matrix_operate.o mod_D_lattice.o mod_frequency.o mod_wave_func.o mod_vec_operate.o mod_electrons.o mod_X.o mod_functions.o mod_zeros.o mod_pseudo.o mod_BS.o mod_QP.o mod_TDDFT.o mod_ACFDT.o mod_IO.o $(ELPH_objects) mod_xc_functionals.o $(RT_SC_objects) mod_global_XC.o std_presets.o 
diff -ru yambo-3.2.1-r.448.orig/src/parser/.objects yambo-3.2.1-r.448/src/parser/.objects
--- yambo-3.2.1-r.448.orig/src/parser/.objects	2009-07-07 13:58:55.000000000 +0200
+++ yambo-3.2.1-r.448/src/parser/.objects	2012-12-06 13:49:46.251492016 +0100
@@ -1,3 +1 @@
-objs =   math.o interface.o parser.o parser_exp.o symbols.o \
-         parser_lib.o mod_parser_m.o mod_itm.o Gclose.o \
-         init_convert.o ps_convert.o
+objs = math.o interface.o parser.o parser_exp.o symbols.o parser_lib.o mod_parser_m.o mod_itm.o Gclose.o init_convert.o ps_convert.o
diff -ru yambo-3.2.1-r.448.orig/src/pol_function/.objects yambo-3.2.1-r.448/src/pol_function/.objects
--- yambo-3.2.1-r.448.orig/src/pol_function/.objects	2009-07-07 13:58:54.000000000 +0200
+++ yambo-3.2.1-r.448/src/pol_function/.objects	2012-12-06 13:48:52.505832658 +0100
@@ -1,8 +1,3 @@
 SC_objects =
 RAS_objects = 
-objs = O_driver.o O_eels.o \
-       Dipole_driver.o \
-       Dipole_kb_pp_comp.o Dipole_kb_sum.o Dipole_kb_Ylm.o \
-       Dipole_transverse.o $(SC_objects)  \
-       X_drude.o  X_pre_setup.o X_Gf.o X_eh_setup.o X_em1.o X_os.o X_s.o \
-       X_delta_part.o X_bare_RIM_Gf.o $(RAS_objects)
+objs = O_driver.o O_eels.o Dipole_driver.o Dipole_kb_pp_comp.o Dipole_kb_sum.o Dipole_kb_Ylm.o Dipole_transverse.o $(SC_objects) X_drude.o X_pre_setup.o X_Gf.o X_eh_setup.o X_em1.o X_os.o X_s.o X_delta_part.o X_bare_RIM_Gf.o $(RAS_objects)
diff -ru yambo-3.2.1-r.448.orig/src/qp/.objects yambo-3.2.1-r.448/src/qp/.objects
--- yambo-3.2.1-r.448.orig/src/qp/.objects	2009-07-07 13:58:55.000000000 +0200
+++ yambo-3.2.1-r.448/src/qp/.objects	2012-12-06 13:49:09.066035796 +0100
@@ -1,4 +1,2 @@
 GW_REAL_AXIS_objects =
-objs = QP_driver.o QP_newton.o QP_ppa_chosex.o \
-       $(GW_REAL_AXIS_objects) \
-       QP_descriptions.o QP_XX_Vxc.o DFT_Vxc.o QP_of.o
+objs = QP_driver.o QP_newton.o QP_ppa_chosex.o $(GW_REAL_AXIS_objects) QP_descriptions.o QP_XX_Vxc.o DFT_Vxc.o QP_of.o
diff -ru yambo-3.2.1-r.448.orig/src/qp_ctl/.objects yambo-3.2.1-r.448/src/qp_ctl/.objects
--- yambo-3.2.1-r.448.orig/src/qp_ctl/.objects	2009-07-07 13:58:53.000000000 +0200
+++ yambo-3.2.1-r.448/src/qp_ctl/.objects	2012-12-06 13:49:25.371235856 +0100
@@ -1,3 +1,2 @@
-objs= mod_QP_CTL.o QP_apply_db.o QP_apply_done.o QP_apply_fit.o \
-      QP_apply_interpolate.o QP_apply_stretch.o
+objs= mod_QP_CTL.o QP_apply_db.o QP_apply_done.o QP_apply_fit.o QP_apply_interpolate.o QP_apply_stretch.o
 
diff -ru yambo-3.2.1-r.448.orig/src/tddft/.objects yambo-3.2.1-r.448/src/tddft/.objects
--- yambo-3.2.1-r.448.orig/src/tddft/.objects	2009-07-07 13:58:53.000000000 +0200
+++ yambo-3.2.1-r.448/src/tddft/.objects	2012-12-06 13:48:59.913923540 +0100
@@ -1,2 +1 @@
-objs =	tddft_do_X_W_typs.o \
-        tddft_alda_g_space.o tddft_alda_r_space.o
+objs =	tddft_do_X_W_typs.o tddft_alda_g_space.o tddft_alda_r_space.o
diff -ru yambo-3.2.1-r.448.orig/src/wf_and_fft/.objects yambo-3.2.1-r.448/src/wf_and_fft/.objects
--- yambo-3.2.1-r.448.orig/src/wf_and_fft/.objects	2009-07-07 13:58:54.000000000 +0200
+++ yambo-3.2.1-r.448/src/wf_and_fft/.objects	2012-12-06 13:49:30.347296906 +0100
@@ -3,5 +3,4 @@
 #else
 FFT_object = sgfft.o
 #endif
-objs = fft_setup.o fft_3d.o $(FFT_object)  \
-       scatter_Bamp.o scatter_Gamp.o space_inv_by_wf.o wf_load.o
+objs = fft_setup.o fft_3d.o $(FFT_object) scatter_Bamp.o scatter_Gamp.o space_inv_by_wf.o wf_load.o
diff -ru yambo-3.2.1-r.448.orig/src/xc_functionals/.objects yambo-3.2.1-r.448/src/xc_functionals/.objects
--- yambo-3.2.1-r.448.orig/src/xc_functionals/.objects	2009-07-07 13:58:55.000000000 +0200
+++ yambo-3.2.1-r.448/src/xc_functionals/.objects	2012-12-06 13:49:21.610189706 +0100
@@ -1,3 +1 @@
-objs =  mod_xc_costants.o xc_lda_driver.o xc_rpa_kp.o \
-        xcspol.o xcpzca.o xcwign.o xchelu.o xcxalp.o \
-        el_magnetization.o el_density.o 
+objs = mod_xc_costants.o xc_lda_driver.o xc_rpa_kp.o xcspol.o xcpzca.o xcwign.o xchelu.o xcxalp.o el_magnetization.o el_density.o 
diff -ru yambo-3.2.1-r.448.orig/ypp/.objects yambo-3.2.1-r.448/ypp/.objects
--- yambo-3.2.1-r.448.orig/ypp/.objects	2009-07-07 13:58:56.000000000 +0200
+++ yambo-3.2.1-r.448/ypp/.objects	2012-12-06 13:50:00.730669655 +0100
@@ -1,5 +1 @@
-objs = mod_YPP.o     ypp_i.o   ypp_init.o ypp_init_load.o \
-       k_grids.o \
-       excitons_driver.o electrons_driver.o plot_check_and_launch.o plot_gnuplot.o plot_xcrysden.o \
-       excitons_sort_and_report.o \
-       $(SPIN_objects) $(RAS_objects)
+objs = mod_YPP.o ypp_i.o ypp_init.o ypp_init_load.o k_grids.o excitons_driver.o electrons_driver.o plot_check_and_launch.o plot_gnuplot.o plot_xcrysden.o excitons_sort_and_report.o $(SPIN_objects) $(RAS_objects)
