--- fetchChromSizes.orig        2019-08-08 17:51:40.635598703 +0000
+++ fetchChromSizes     2019-08-08 17:51:53.745665396 +0000
@@ -65,7 +65,7 @@
 fi

 printf "INFO: trying MySQL ${MYSQL} for database ${DB}\n" 1>&2
-mysql -N --user=genome --host=genome-mysql.soe.ucsc.edu -A -e \
+mysql -N -A -e \
 "SELECT chrom,size FROM chromInfo ORDER BY size DESC;" ${DB}
 if [ $? -ne 0 ]; then
     printf "WARNING: mysql command failed\n" 1>&2

