# mcpdft has a rasscf dependency
# July 14th 2021 by B. Hajgato (UGent)
--- OpenMolcas-v21.06/CMakeLists.txt.orig	2021-06-09 11:56:51.000000000 +0200
+++ OpenMolcas-v21.06/CMakeLists.txt	2021-07-14 22:57:35.247625196 +0200
@@ -2652,6 +2652,9 @@
 
 # set program dependencies of super modules
 # (note dependencies are not recursive with object libraries, so child dependencies must be explicit)
+set (mcpdft_deplibs
+  rasscf
+)
 set (casvb_deplibs
   rasscf
 )
