fix for CMake error: unsupported C compiler: IntelLLVM
author: Kenneth Hoste (HPC-UGent)
--- OpenMolcas-v21.06/CMakeLists.txt.orig	2025-08-07 16:55:02.013768235 +0200
+++ OpenMolcas-v21.06/CMakeLists.txt	2025-08-07 16:57:00.260015113 +0200
@@ -615,7 +615,7 @@
 set (CXXFLAGS_Intel_RELEASE "-O2")
 set (CXXFLAGS_Intel_FAST "-O2")
 # C compiler
-list (APPEND SUPPORTED_C_COMPILERS "Intel")
+list (APPEND SUPPORTED_C_COMPILERS "IntelLLVM")
 set (CFLAGS_Intel_BASIC "-std=gnu99")
 set (CFLAGS_Intel_OPENMP "-qopenmp")
 # build targets
