ensuring compatibility with the newest hdf5 version
diff -ruN ncbi-vdb-3.0.0_orig/libs/hdf5/hdf5dir.c ncbi-vdb-3.0.0/libs/hdf5/hdf5dir.c
--- ncbi-vdb-3.0.0_orig/libs/hdf5/hdf5dir.c	2022-11-11 11:55:46.455701000 +0100
+++ ncbi-vdb-3.0.0/libs/hdf5/hdf5dir.c	2022-11-11 11:58:33.079945000 +0100
@@ -26,7 +26,7 @@
 
 typedef struct HDF5Dir HDF5Dir;
 #define KDIR_IMPL HDF5Dir
-
+#define H5_USE_18_API
 #define H5Gopen_vers 2
 #define H5Eset_auto_vers 2
 