--- ncbi-vdb-3.1.1/build/env.cmake.orig	2025-03-12 15:55:24.551481112 +0100
+++ ncbi-vdb-3.1.1/build/env.cmake	2025-03-12 15:57:04.742032520 +0100
@@ -109,6 +109,8 @@
     set(ARCH "x86_64")
 elseif ( ${_system_processor} STREQUAL "amd64")
     set(ARCH "x86_64")
+elseif ( ${_system_processor} STREQUAL "riscv64")
+    set(ARCH "riscv64")
 else ()
     message ( FATAL_ERROR "unknown architecture " ${_system_processor})
 endif ()
