Adding missing cstdlib
J. Sassmannshausen
diff --git a/ncbi-vdb-2.11.2.orig/vdb3/platform/memory/PrimordialMemoryMgr.cpp b/ncbi-vdb-2.11.2/vdb3/platform/memory/PrimordialMemoryMgr.cpp
index e53c791..05db314 100644
--- a/ncbi-vdb-2.11.2.orig/vdb3/platform/memory/PrimordialMemoryMgr.cpp
+++ b/ncbi-vdb-2.11.2/vdb3/platform/memory/PrimordialMemoryMgr.cpp
@@ -24,6 +24,7 @@
 *
 */
 
+#include <cstdlib>
 #include <memory/PrimordialMemoryMgr.hpp>
 
 #include <memory/MemoryBlockItf.hpp>
