fix the hardcoded compiler and use ${CXXFLAGS} 

diff -ru gfold.V1.1.4.orig/Makefile gfold.V1.1.4/Makefile
--- gfold.V1.1.4.orig/Makefile	2015-05-24 01:47:13.000000000 +0200
+++ gfold.V1.1.4/Makefile	2017-03-09 10:32:23.919031000 +0100
@@ -5,10 +5,10 @@
 all: program
 
 debug: DataProcessor.hpp GFOLD.hpp Utility.hpp GeneInfo.hpp main.cc
-	g++ -Wall -g main.cc -o gfold -lgsl -lgslcblas 
+	${CXX} ${CXXFLAGS} -Wall -g main.cc -o gfold -lgsl -lgslcblas 
 
 program: DataProcessor.hpp GFOLD.hpp Utility.hpp GeneInfo.hpp main.cc
-	g++ -O3 -Wall -g main.cc -o gfold -lgsl -lgslcblas 
+	${CXX} ${CXXFLAGS} -Wall -g main.cc -o gfold -lgsl -lgslcblas 
 
 docu: doc/gfold.pod
 	pod2man doc/gfold.pod > doc/gfold.man
