diff -ru rdkit-Release_2018_03_2.orig/CMakeLists.txt rdkit-Release_2018_03_2/CMakeLists.txt
--- rdkit-Release_2018_03_2.orig/CMakeLists.txt	2018-07-17 20:21:14.295051295 +0000
+++ rdkit-Release_2018_03_2/CMakeLists.txt	2018-07-17 20:21:53.390695329 +0000
@@ -187,7 +187,7 @@
   if(PYTHON_VERSION_MAJOR EQUAL 3)
     # Find boost-python3 using name specified as command line option then fall back to commonly used names
     set(RDK_BOOST_PYTHON3_NAME "python3" CACHE STRING "Name of the boost python3 library. If installed as libboost_python-xxx.so, use python-xxx.")
-    foreach(Boost_Python_Lib "${RDK_BOOST_PYTHON3_NAME}" "python-py3${PYTHON_VERSION_MINOR}" "python3")
+    foreach(Boost_Python_Lib "${RDK_BOOST_PYTHON3_NAME}" "python-py3${PYTHON_VERSION_MINOR}" "python3" "python3${PYTHON_VERSION_MINOR}")
       find_package(Boost 1.56.0 COMPONENTS "${Boost_Python_Lib}" QUIET)
       if(Boost_FOUND)
         break()
--- rdkit-Release_2018_03_2.orig/CMakeLists.txt	2018-07-17 18:35:15.887328831 +0000
+++ rdkit-Release_2018_03_2/CMakeLists.txt	2018-07-17 18:35:59.969930783 +0000
@@ -198,7 +198,7 @@
       find_package(Boost 1.56.0 COMPONENTS python REQUIRED)
     endif()
   else(PYTHON_VERSION_MAJOR EQUAL 3)
-     find_package(Boost 1.56.0 COMPONENTS python REQUIRED)
+     find_package(Boost 1.56.0 COMPONENTS python27 REQUIRED)
   endif(PYTHON_VERSION_MAJOR EQUAL 3)
 
 
