--- include.mk.orig	2018-08-14 22:11:10.003487846 +0000
+++ include.mk	2018-08-14 22:11:25.592357744 +0000
@@ -4,7 +4,7 @@
 
 #Modify this variable to set the location of sonLib
 #(sonlib is used only for cuTest at this potin)
-sonLibRootPath?=${rootPath}/../sonLib
+sonLibRootPath?=${rootPath}/sonLib
 sonLibPath=${sonLibRootPath}/lib
 
 include  ${sonLibRootPath}/include.mk
@@ -55,8 +55,8 @@
 
 #	Local Linux install (phast and clapack sister dirs to hal/)
 #	(note CLAPACKPATH not needed in Mac)
-	PHAST=../../phast
-	CLAPACKPATH=../../clapack
+	PHAST=../phast
+	CLAPACKPATH=../clapack
 
 ifeq ($(TARGETOS), Darwin)
 	phyloPcppflags += -DENABLE_PHYLOP -I${PHAST}/include -I${PHAST}/src/lib/pcre -DVECLIB
