Change Install script to run noninteractively using environment variables

Author: Samuel Moors (Vrije Universiteit Brussel)

--- Install.orig	2025-09-04 15:20:51.390000000 +0200
+++ Install	2025-09-17 13:59:29.814302000 +0200
@@ -71,11 +71,8 @@
 echo '   5) Linux on 64-bit ARM (e.g. for Raspberry Pi 3 or later).'
 echo
 printf "Select the type of your computer from the list above [$DEFexectype]: "
-read ans
-if [ x$ans != x ] ; then
-  exectype=$ans
-else
-  exectype=$DEFexectype
+if [[ -z $exectype ]]; then
+    exectype=$DEFexectype
 fi
 if [ $exectype = 1 ] ; then EXECUTABLE_TYPE=$exectype1 ; fi
 if [ $exectype = 2 ] ; then EXECUTABLE_TYPE=$exectype2 ; fi
@@ -88,10 +85,7 @@
 echo
 echo "Full directory name in which to install $MOD"
 printf "[$DEFmodinstall]: "
-read ans
-if [ x$ans != x ] ; then 
-   MODINSTALL=$ans
-else
+if [[ -z $MODINSTALL ]]; then
    MODINSTALL=$DEFmodinstall
 fi
 if [ ! -d $MODINSTALL ] ; then
@@ -103,7 +97,6 @@
 echo
 echo "License key, obtained from our academic license server at"
 printf "https://salilab.org/modeller/registration.html: "
-read KEY_MODELLER
 
 
 echo
@@ -118,7 +111,6 @@
 echo "If they are incorrect, abort installation (ctrl-c) and run Install again."
 echo
 printf "Press <Enter> to begin the installation: "
-read ans
 echo
 echo "------------------------------------------------------------------------"
 echo
@@ -153,7 +145,6 @@
 echo "license = '${KEY_MODELLER}'" >> ${config}
 echo "Installation complete."
 printf "Press <Enter> to continue: "
-read ans
 
 echo
 echo "------------------------------------------------------------------------"
@@ -180,7 +171,6 @@
 echo "------------------------------------------------------------------------"
 echo
 printf "Press <Enter> to continue: "
-read ans
 
 echo
 echo "------------------------------------------------------------------------"
