don't fix versions of h5py and matplotlib Python packages
author: Kenneth Hoste (HPC-UGent)
--- setup.py.orig	2016-03-29 08:34:20.000000000 +0200
+++ setup.py	2016-04-28 15:27:35.764121169 +0200
@@ -47,9 +47,9 @@
 TVB_TEAM += "Stuart Knock, Marmaduke Woodman, Paula Sansz Leon, "
 
 TVB_INSTALL_REQUIREMENTS = ["apscheduler", "beautifulsoup", "cherrypy", "genshi", "cfflib", "formencode==1.3.0a1",
-                            "h5py==2.3.0", "lxml", "minixsv", "mod_pywebsocket", "networkx", "nibabel", "numpy",
+                            "h5py", "lxml", "minixsv", "mod_pywebsocket", "networkx", "nibabel", "numpy",
                             "numexpr", "psutil", "scikit-learn", "scipy", "simplejson", "Pillow",
-                            "sqlalchemy==0.7.8", "sqlalchemy-migrate==0.7.2", "matplotlib==1.2.1"]
+                            "sqlalchemy==0.7.8", "sqlalchemy-migrate==0.7.2", "matplotlib"]
 
 EXCLUDE_INTROSPECT_FOLDERS = [folder for folder in os.listdir(".")
                               if os.path.isdir(os.path.join(".", folder)) and folder != "tvb"]
