diff -ru arrayfire-3.6.20181017.orig/arrayfire/library.py arrayfire-3.6.20181017/arrayfire/library.py
--- arrayfire-3.6.20181017.orig/arrayfire/library.py	2021-01-11 22:09:56.719064574 +0000
+++ arrayfire-3.6.20181017/arrayfire/library.py	2021-01-11 22:10:43.800281683 +0000
@@ -530,7 +530,7 @@
     def __libname(self, name, head='af', ver_major=AF_VER_MAJOR):
         post = self.__post.replace(_VER_MAJOR_PLACEHOLDER, ver_major)
         libname = self.__pre + head + name + post
-        libname_full = self.AF_PATH + '/lib/' + libname
+        libname_full = self.AF_PATH + '/lib64/' + libname
         return (libname, libname_full)
 
     def set_unsafe(self, name):
