fix compatibility with rpy2 v3.x
author: Kenneth Hoste (HPC-UGent)
diff -ru flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0.orig/bin/runDE.py flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0/bin/runDE.py
--- flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0.orig/bin/runDE.py	2020-06-30 18:58:44.000000000 +0200
+++ flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0/bin/runDE.py	2021-05-18 15:07:01.188990143 +0200
@@ -105,11 +105,11 @@
 
     # make the quant DF
     quantDF  = pd.read_table(matrix, header=0, sep='\t', index_col=0)
-    df = pandas2ri.py2ri(quantDF)
+    df = pandas2ri.py2rpy(quantDF)
 
     # import formula
     formulaDF     = pd.read_csv(formula,header=0, sep="\t",index_col=0)
-    sampleTable = pandas2ri.py2ri(formulaDF)
+    sampleTable = pandas2ri.py2rpy(formulaDF)
 
 
     if "batch" in list(formulaDF):
@@ -213,4 +213,4 @@
 
 
 if __name__ == "__main__":
-    main()
\ No newline at end of file
+    main()
diff -ru flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0.orig/bin/runDS.py flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0/bin/runDS.py
--- flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0.orig/bin/runDS.py	2020-06-30 18:58:44.000000000 +0200
+++ flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0/bin/runDS.py	2021-05-18 15:07:01.188990143 +0200
@@ -153,8 +153,8 @@
         quantDF[[col]] = quantDF[[col]] + 1
 
     # Convert pandas to R data frame.
-    samples      = pandas2ri.py2ri(formulaDF)
-    counts       = pandas2ri.py2ri(quantDF)
+    samples      = pandas2ri.py2rpy(formulaDF)
+    counts       = pandas2ri.py2rpy(quantDF)
 
     # DRIMSEQ part.
     if "batch" in list(formulaDF): R.assign('batch', samples.rx2('batch'))
diff -ru flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0.orig/bin/runDU.py flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0/bin/runDU.py
--- flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0.orig/bin/runDU.py	2020-06-30 18:58:44.000000000 +0200
+++ flair-0f71b5f7e37ff040eb3bf0e62fb62940e4a318f0/bin/runDU.py	2021-05-18 15:07:01.188990143 +0200
@@ -114,11 +114,11 @@
 
     # get quant table and formula table
     quantDF  = pd.read_table(matrix, header=0, sep='\t', index_col=0)
-    df       = pandas2ri.py2ri(quantDF)
+    df       = pandas2ri.py2rpy(quantDF)
 
     formulaDF = pd.read_csv(formula,header=0, sep="\t")
 
-    pydf      = pandas2ri.py2ri(formulaDF)
+    pydf      = pandas2ri.py2rpy(formulaDF)
 
     # Convert pandas to R data frame.
     samples = pydf
