--- configure.ac.orig	2025-01-23 06:58:19.935561088 +0000
+++ configure.ac	2025-01-23 06:46:35.599171724 +0000
@@ -755,7 +755,7 @@
 else
 
   dnl first check blas
-  AC_SEARCH_LIBS([dgemm],[flexiblas openblas satlas blas],[have_blas=yes],[have_blas=no])
+  AC_SEARCH_LIBS([dgemm],[flexiblas_intel openblas satlas blas],[have_blas=yes],[have_blas=no])
   AC_MSG_CHECKING([whether we can link a program with a blas lib])
   AC_MSG_RESULT([${have_blas}])
 
