diff -ru ELPA_2013.11.orig/test/test_complex2.F90 ELPA_2013.11/test/test_complex2.F90
--- ELPA_2013.11.orig/test/test_complex2.F90	2014-01-29 09:29:35.544248000 +0100
+++ ELPA_2013.11/test/test_complex2.F90	2014-02-21 13:48:09.662072163 +0100
@@ -79,6 +79,7 @@
    integer i, mpierr, my_blacs_ctxt, sc_desc(9), info, nprow, npcol
 
    integer, external :: numroc
+   integer           :: iargc
 
    real*8 err, errmax
    real*8, allocatable :: ev(:), xr(:,:)
diff -ru ELPA_2013.11.orig/test/test_complex.F90 ELPA_2013.11/test/test_complex.F90
--- ELPA_2013.11.orig/test/test_complex.F90	2014-01-29 09:29:35.534699000 +0100
+++ ELPA_2013.11/test/test_complex.F90	2014-02-21 13:48:33.118418351 +0100
@@ -78,6 +78,7 @@
    integer i, mpierr, my_blacs_ctxt, sc_desc(9), info, nprow, npcol
 
    integer, external :: numroc
+   integer           :: iargc
 
    real*8 err, errmax
    real*8, allocatable :: ev(:), xr(:,:)
diff -ru ELPA_2013.11.orig/test/test_real2.F90 ELPA_2013.11/test/test_real2.F90
--- ELPA_2013.11.orig/test/test_real2.F90	2014-01-29 09:29:35.573319000 +0100
+++ ELPA_2013.11/test/test_real2.F90	2014-02-21 13:48:51.240387595 +0100
@@ -80,6 +80,7 @@
    integer i, mpierr, my_blacs_ctxt, sc_desc(9), info, nprow, npcol
 
    integer, external :: numroc
+   integer           :: iargc
 
    real*8 err, errmax
    real*8, allocatable :: a(:,:), z(:,:), tmp1(:,:), tmp2(:,:), as(:,:), ev(:)
diff -ru ELPA_2013.11.orig/test/test_real.F90 ELPA_2013.11/test/test_real.F90
--- ELPA_2013.11.orig/test/test_real.F90	2014-01-29 09:29:35.563829000 +0100
+++ ELPA_2013.11/test/test_real.F90	2014-02-21 13:49:05.466094157 +0100
@@ -79,6 +79,7 @@
    integer i, mpierr, my_blacs_ctxt, sc_desc(9), info, nprow, npcol
 
    integer, external :: numroc
+   integer           :: iargc
 
    real*8 err, errmax
    real*8, allocatable :: a(:,:), z(:,:), tmp1(:,:), tmp2(:,:), as(:,:), ev(:)
