--- Makefile.am.orig	2017-02-16 10:32:55.789545672 +0100
+++ Makefile.am	2017-02-16 10:33:15.213234461 +0100
@@ -255,7 +255,7 @@
 build_lib = libelpatest@SUFFIX@.la libelpa@SUFFIX@.la
 
 # library with shared sources for the test files
-noinst_LTLIBRARIES += libelpatest@SUFFIX@.la
+lib_LTLIBRARIES += libelpatest@SUFFIX@.la
 libelpatest@SUFFIX@_la_FCFLAGS = $(AM_FCFLAGS) @FC_MODOUT@private_modules @FC_MODINC@private_modules
 libelpatest@SUFFIX@_la_SOURCES = \
 	test/shared/util.F90 \
