fix csvtohtml for use with Python 3
author: Kenneth Hoste (HPC-UGent)
--- IMOD/manpages/csvtohtml.orig	2021-09-03 13:27:28.865210827 +0200
+++ IMOD/manpages/csvtohtml	2021-09-03 13:27:34.320159119 +0200
@@ -15,7 +15,7 @@
    return last
 
 import csv, sys
-sys.path.append('../pysrc')
+sys.path.insert(0, '../pysrc')
 from imodpy import *
 
 insertTag = 'INSERT CSV FILE'
@@ -80,7 +80,7 @@
       inrows.append(row)
       maxlen = min(maxUse, max(maxlen, len(row)))
 
-print maxlen
+print(maxlen)
 
 for row in inrows:
    lastNonEmpty = lastNonEmptyIndex(row)
