# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2023/08
diff -ru IMOD/flib/beadtrack/beadtrack.cpp IMOD_tiltalign_include/flib/beadtrack/beadtrack.cpp
--- IMOD/flib/beadtrack/beadtrack.cpp	2023-08-25 12:38:55.864116440 +0200
+++ IMOD_tiltalign_include/flib/beadtrack/beadtrack.cpp	2023-08-25 17:13:50.646567630 +0200
@@ -21,7 +21,7 @@
 #include "btfuncs.h"
 #include "tafuncs.h"
 #include "arraymaxes.h"
-#include "mapsepgroups.h"
+#include "../tiltalign/mapsepgroups.h"
 #include "cgpixels.h"
 #include "tltcntrl.h"
 
