# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2023/08
diff -ru IMOD/pysrc/imodpy.py IMOD_pipImport/pysrc/imodpy.py
--- IMOD/pysrc/imodpy.py	2023-08-25 12:38:55.863116418 +0200
+++ IMOD_pipImport/pysrc/imodpy.py	2023-08-25 17:55:21.862674887 +0200
@@ -110,7 +110,7 @@
 
 # other modules needed by imodpy
 import sys, os, re, time, glob, signal, stat, shlex, copy, shutil
-from pip import exitError, PipGetInteger
+#from pip import exitError, PipGetInteger
 
 pyVersion = 100 * sys.version_info[0] + 10 * sys.version_info[1]
 
