Avoid scipy and numpy version requirement
Author: Cintia Willemyns (Vrije Universiteit Brussel)
--- a/pyproject.toml	2023-07-13 16:11:09.662444000 +0200
+++ b/pyproject.toml	2023-07-13 16:12:18.320284000 +0200
@@ -5,8 +5,8 @@
     "wheel",
     "Cython>=0.28.5",
     "setuptools_scm[toml]>=3.4",  # https://scikit-hep.org/developer/packaging#git-tags-official-pypa-method
-    "oldest-supported-numpy",  # https://github.com/scipy/oldest-supported-numpy
-    "scipy>=0.19.1",
+    "numpy",
+    "scipy",
 ]
 build-backend = "setuptools.build_meta" 
