diff -ruN flagser.orig/include/definitions.h flagser/include/definitions.h
--- flagser.orig/include/definitions.h	2025-12-01 17:02:24.987037362 +0100
+++ flagser/include/definitions.h	2026-02-16 13:21:58.496908211 +0100
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <algorithm>
 #include <iostream>
 #include <string>
diff -ruN flagser.orig/include/persistence.h flagser/include/persistence.h
--- flagser.orig/include/persistence.h	2025-12-01 17:02:25.001869000 +0100
+++ flagser/include/persistence.h	2025-12-01 17:11:14.312539204 +0100
@@ -8,6 +8,7 @@
 #include <deque>
 #include <iostream>
 #include <queue>
+#include <limits>
 
 #include "definitions.h"
 #include "output/base.h"
