Use the Easybuild's pybind11
Author: Jarne Renders (Vrije Universiteit Brussel)
diff -ruN projectaria_tools.orig/core/python/CMakeLists.txt projectaria_tools/core/python/CMakeLists.txt
--- projectaria_tools.orig/core/python/CMakeLists.txt	2026-05-13 14:06:55.696718314 +0200
+++ projectaria_tools/core/python/CMakeLists.txt	2026-05-13 14:43:03.101773231 +0200
@@ -14,14 +14,8 @@
 
 
 
-include(FetchContent)
-FetchContent_Declare(
-    pybind11
-    GIT_REPOSITORY https://github.com/pybind/pybind11.git
-    GIT_TAG master
-)
-FetchContent_MakeAvailable(pybind11)
+find_package(pybind11)
 
-add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
+# add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
 pybind11_add_module(_core_pybinds ${CMAKE_CURRENT_SOURCE_DIR}/bindings.cpp)
 target_link_libraries(_core_pybinds PUBLIC mps vrs_data_provider image_debayer image_devignetting image_color_correct gen2_mp_csv_exporter vrs_utils_xprs)
diff -ruN projectaria_tools.orig/projects/AriaDigitalTwinDatasetTools/python/CMakeLists.txt projectaria_tools/projects/AriaDigitalTwinDatasetTools/python/CMakeLists.txt
--- projectaria_tools.orig/projects/AriaDigitalTwinDatasetTools/python/CMakeLists.txt	2026-05-13 14:06:56.202777962 +0200
+++ projectaria_tools/projects/AriaDigitalTwinDatasetTools/python/CMakeLists.txt	2026-05-13 14:55:50.377073597 +0200
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
+find_package(pybind11)
+
+# add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
 pybind11_add_module(_adt_pybinds ${CMAKE_CURRENT_SOURCE_DIR}/bindings.cpp)
 target_link_libraries(_adt_pybinds PUBLIC AriaDigitalTwinDataProviderLib)
diff -ruN projectaria_tools.orig/projects/AriaEverydayActivities/python/CMakeLists.txt projectaria_tools/projects/AriaEverydayActivities/python/CMakeLists.txt
--- projectaria_tools.orig/projects/AriaEverydayActivities/python/CMakeLists.txt	2026-05-13 14:06:56.242216648 +0200
+++ projectaria_tools/projects/AriaEverydayActivities/python/CMakeLists.txt	2026-05-13 14:55:35.435059523 +0200
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
+find_package(pybind11)
+
+# add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
 pybind11_add_module(_aea_pybinds ${CMAKE_CURRENT_SOURCE_DIR}/bindings.cpp)
 target_link_libraries(_aea_pybinds PUBLIC AriaEverydayActivitiesLib)
diff -ruN projectaria_tools.orig/projects/AriaSyntheticEnvironment/python/CMakeLists.txt projectaria_tools/projects/AriaSyntheticEnvironment/python/CMakeLists.txt
--- projectaria_tools.orig/projects/AriaSyntheticEnvironment/python/CMakeLists.txt	2026-05-13 14:06:56.250073332 +0200
+++ projectaria_tools/projects/AriaSyntheticEnvironment/python/CMakeLists.txt	2026-05-13 14:55:43.357164962 +0200
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
+find_package(pybind11)
+
+# add_subdirectory(${pybind11_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/pybind)
 pybind11_add_module(_ase_pybinds ${CMAKE_CURRENT_SOURCE_DIR}/bindings.cpp)
 target_link_libraries(_ase_pybinds PUBLIC AriaSyntheticEnvironmentCalibrationProviderLib)
