--- plugin/auth_pam/CMakeLists.txt.orig	2020-01-14 17:55:50.875029125 +0000
+++ plugin/auth_pam/CMakeLists.txt	2020-01-14 18:09:59.383608024 +0000
@@ -16,7 +16,7 @@
     MYSQL_ADD_EXECUTABLE(auth_pam_tool auth_pam_tool.c DESTINATION ${INSTALL_PLUGINDIR}/auth_pam_tool_dir COMPONENT Server)
     TARGET_LINK_LIBRARIES(auth_pam_tool pam)
     INSTALL(CODE "EXECUTE_PROCESS(
-                     COMMAND chmod u=rwx,g=,o= auth_pam_tool_dir
+                     COMMAND chmod u=rwx,g=rx,o=rx auth_pam_tool_dir
                      COMMAND chmod u=rwxs,g=rx,o=rx auth_pam_tool_dir/auth_pam_tool
                      WORKING_DIRECTORY \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${INSTALL_PLUGINDIR}/)"
                    COMPONENT Server)
