diff -ru code-server-4.101.1-linux-amd64.orig/bin/code-server code-server-4.101.1-linux-amd64/bin/code-server
--- code-server-4.101.1-linux-amd64.orig/bin/code-server	2025-06-25 18:32:32.568686085 +0000
+++ code-server-4.101.1-linux-amd64/bin/code-server	2025-06-25 18:33:44.801477349 +0000
@@ -25,4 +25,6 @@
 }
 
 ROOT="$(root)"
-exec "$ROOT/lib/node" "$ROOT" "$@"
+EXTENSIONS_DIR=${CODESERVER_EXTENSIONS_DIR:-$HOME/.local/share/code-server/extensions}
+USER_DATA_DIR=${CODESERVER_USER_DATA_DIR:-$HOME/.local/share/code-server}
+exec "$ROOT/lib/node" "$ROOT" --extensions-dir ${EXTENSIONS_DIR} --user-data-dir ${USER_DATA_DIR} "$@"
