make CufflinksGTF_to_CodingQuarryGFF3.py script compatible with Python 3.x
author: Kenneth Hoste (HPC-UGent)
--- CufflinksGTF_to_CodingQuarryGFF3.py.orig	2016-03-09 01:35:28.000000000 +0100
+++ CufflinksGTF_to_CodingQuarryGFF3.py	2023-04-10 21:10:51.666438036 +0200
@@ -5,7 +5,7 @@
 try :
     myFile = open(sys.argv[1], 'rU')
 except :
-    print 'run: \n\npython CufflinksGTF_to_CodingQuarryGFF3.py transcripts.gtf > out.gff3\n'
+    print('run: \n\npython CufflinksGTF_to_CodingQuarryGFF3.py transcripts.gtf > out.gff3\n')
     exit()
 
 number = 0
@@ -18,8 +18,8 @@
             number += 1
         else :
             number = 1
-        print '%s\t%s\texon\t%s\t%s\t%s\t%s\t%s\tID=exon:%s.%i;Parent=%s;' %  \
-            (fields[0], fields[1], fields[3], fields[4], fields[5], fields[6], fields[7], details[3], number, details[3]) 
+        print('%s\t%s\texon\t%s\t%s\t%s\t%s\t%s\tID=exon:%s.%i;Parent=%s;' %  \
+            (fields[0], fields[1], fields[3], fields[4], fields[5], fields[6], fields[7], details[3], number, details[3]))
         IDprev = details[3]
 
 myFile.close()
--- QuarryFiles/scripts/fastaTranslate.py.orig	2016-03-09 01:35:28.000000000 +0100
+++ QuarryFiles/scripts/fastaTranslate.py	2023-04-10 21:30:47.355740964 +0200
@@ -33,7 +33,7 @@
 		else :
 			protein+='*'
 		i+=3
-        print '>%s' % (record.id)
-        print protein
+        print('>%s' % (record.id))
+        print(protein)
 
 myFASTA.close()
