diff -aur a/configure.ac b/configure.ac
As we don't have SVG support we disable the search for the converter.
Adopted from:
Author: Alexander Grund (TU Dresden)

--- a/configure.ac	2023-08-18 17:58:28.032333478 +0200
+++ b/configure.ac	2023-08-18 17:59:51.554807635 +0200
@@ -49,7 +49,7 @@
 AM_CONDITIONAL(ALLOW_RENDERING, test "x$allow_rendering" = "xyes")
 
 symbolic_encode_sizes="16x16 32x32"
-AC_PATH_PROG([GTK_ENCODE_SYMBOLIC_SVG], [gtk-encode-symbolic-svg], [false])
+GTK_ENCODE_SYMBOLIC_SVG="false"
 if test "x$GTK_ENCODE_SYMBOLIC_SVG" = "xfalse"; then
    symbolic_encode_sizes=""
 fi
