--- build-backend/configure.orig	2013-11-09 22:55:07.707702200 +0100
+++ build-backend/configure	2013-11-09 22:56:59.949861730 +0100
@@ -19889,7 +19889,7 @@
 else
   echo "$as_me:$LINENO: Running $QMAKE --version:" >&5
   $QMAKE --version >&5 2>&1
-  qmake_version_sed='/^.*\([0-9]\.[0-9]\.[0-9]\).*$/!d;s//\1/'
+  qmake_version_sed='/^.*Qt.version.\([0-9]\.[0-9]\.[0-9]\).in.*$/!d;s//\1/'
   at_cv_QT_VERSION=`$QMAKE --version 2>&1 | sed "$qmake_version_sed"`
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $at_cv_QT_VERSION" >&5
