diff -ur -x autom4te.cache -x '*~' -x defmsg.h -x errcode.c mvapich2-1.9b.orig/Makefile.am mvapich2-1.9b/Makefile.am
--- mvapich2-1.9b.orig/Makefile.am	2013-02-26 22:24:43.000000000 +0200
+++ mvapich2-1.9b/Makefile.am	2013-04-12 02:32:35.819030753 +0300
@@ -50,7 +50,9 @@
 external_subdirs = src/mpl src/openpa
 # FIXME how/where does external_libs/ldflags get used?
 external_ldflags = -L$(top_builddir)/src/mpl -L$(top_builddir)/src/openpa/src
-external_libs = -lmpl -lopa
+# _OSU_MVAPICH_
+external_libs = $(top_builddir)/src/mpl/libmpl.la $(top_builddir)/src/openpa/src/libopa.la
+# _OSU_MVAPICH_
 
 # NOTE on our semi-unconventional usage of DIST_SUBDIRS:
 # The automake manual recommends thinking of DIST_SUBDIRS as the list of all
diff -ur -x autom4te.cache -x '*~' -x defmsg.h -x errcode.c mvapich2-1.9b.orig/Makefile.in mvapich2-1.9b/Makefile.in
--- mvapich2-1.9b.orig/Makefile.in	2013-02-28 02:14:19.000000000 +0200
+++ mvapich2-1.9b/Makefile.in	2013-04-12 02:35:09.318964105 +0300
@@ -1409,7 +1409,7 @@
 @BUILD_CH3_TRUE@@BUILD_MRAIL_HWLOC_TRUE@@BUILD_MRAIL_TRUE@			  $(HWLOC_EMBEDDED_CPPFLAGS)
 
 @BUILD_CH3_TRUE@@BUILD_MRAIL_HWLOC_TRUE@@BUILD_MRAIL_TRUE@am__append_91 = -L$(top_builddir)/src/mpid/ch3/channels/mrail/src/hwloc/src
-@BUILD_CH3_TRUE@@BUILD_MRAIL_HWLOC_TRUE@@BUILD_MRAIL_TRUE@am__append_92 = -lhwloc_embedded
+@BUILD_CH3_TRUE@@BUILD_MRAIL_HWLOC_TRUE@@BUILD_MRAIL_TRUE@am__append_92 = $(top_builddir)/src/mpid/ch3/channels/mrail/src/hwloc/src/libhwloc_embedded.la
 @BUILD_CH3_TRUE@@BUILD_MRAIL_CUDA_KERNELS_TRUE@@BUILD_MRAIL_TRUE@am__append_93 = \
 @BUILD_CH3_TRUE@@BUILD_MRAIL_CUDA_KERNELS_TRUE@@BUILD_MRAIL_TRUE@    src/mpid/ch3/channels/mrail/src/cuda/pack_unpack.cu
 
@@ -2640,11 +2640,9 @@
 	$(lib_lib@MPICXXLIBNAME@_la_LDFLAGS) $(LDFLAGS) -o $@
 @BUILD_CXX_LIB_TRUE@am_lib_lib@MPICXXLIBNAME@_la_rpath = -rpath \
 @BUILD_CXX_LIB_TRUE@	$(libdir)
-am__DEPENDENCIES_1 = $(am__append_12) $(am__append_13)
-am__DEPENDENCIES_2 =
-lib_lib@MPILIBNAME@_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(am__append_4) $(am__append_17) $(am__append_82) \
-	$(am__DEPENDENCIES_2) $(am__append_98)
+lib_lib@MPILIBNAME@_la_DEPENDENCIES = $(external_libs) $(am__append_4) \
+	$(am__append_17) $(am__append_82) $(am__append_92) \
+	$(am__append_98)
 am__lib_lib@MPILIBNAME@_la_SOURCES_DIST = src/mpi/attr/attrutil.c \
 	src/mpi/attr/dup_fn.c src/mpi/coll/allred_group.c \
 	src/mpi/coll/barrier_group.c src/mpi/coll/helper_fns.c \
@@ -5996,7 +5994,7 @@
 @BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@	src/pm/smpd/mpiexec_rsh.$(OBJEXT)
 src_pm_smpd_mpiexec_OBJECTS = $(am_src_pm_smpd_mpiexec_OBJECTS)
 @BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@src_pm_smpd_mpiexec_DEPENDENCIES = lib/lib@MPILIBNAME@.la \
-@BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@	$(am__DEPENDENCIES_1)
+@BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@	$(external_libs)
 src_pm_smpd_mpiexec_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(AM_CFLAGS) $(CFLAGS) $(src_pm_smpd_mpiexec_LDFLAGS) \
@@ -6008,7 +6006,7 @@
 @BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@	src/pm/smpd/smpd_watchprocs.$(OBJEXT)
 src_pm_smpd_smpd_OBJECTS = $(am_src_pm_smpd_smpd_OBJECTS)
 @BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@src_pm_smpd_smpd_DEPENDENCIES = lib/lib@MPILIBNAME@.la \
-@BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@	$(am__DEPENDENCIES_1)
+@BUILD_PM_SMPD_TRUE@@PRIMARY_PM_SMPD_TRUE@	$(external_libs)
 src_pm_smpd_smpd_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(AM_CFLAGS) $(CFLAGS) $(src_pm_smpd_smpd_LDFLAGS) $(LDFLAGS) \
@@ -7019,7 +7017,11 @@
 external_subdirs = src/mpl src/openpa
 # FIXME how/where does external_libs/ldflags get used?
 external_ldflags = -L$(top_builddir)/src/mpl -L$(top_builddir)/src/openpa/src
-external_libs = -lmpl -lopa $(am__append_12) $(am__append_13)
+# _OSU_MVAPICH_
+external_libs = $(top_builddir)/src/mpl/libmpl.la \
+	$(top_builddir)/src/openpa/src/libopa.la $(am__append_12) \
+	$(am__append_13)
+# _OSU_MVAPICH_
 
 # NOTE on our semi-unconventional usage of DIST_SUBDIRS:
 # The automake manual recommends thinking of DIST_SUBDIRS as the list of all
diff -ur -x autom4te.cache -x '*~' -x defmsg.h -x errcode.c mvapich2-1.9b.orig/src/include/coll_shmem.h mvapich2-1.9b/src/include/coll_shmem.h
--- mvapich2-1.9b.orig/src/include/coll_shmem.h	2013-02-26 22:24:43.000000000 +0200
+++ mvapich2-1.9b/src/include/coll_shmem.h	2013-04-12 02:32:35.819030753 +0300
@@ -282,8 +282,8 @@
 void unlock_shmem_region(void);
 
 /* utils */
-inline int mv2_increment_shmem_coll_counter(MPID_Comm *comm_ptr); 
-inline int mv2_increment_allgather_coll_counter(MPID_Comm *comm_ptr); 
+int mv2_increment_shmem_coll_counter(MPID_Comm *comm_ptr); 
+int mv2_increment_allgather_coll_counter(MPID_Comm *comm_ptr); 
 void increment_mv2_shmem_comm_count(void);
 int get_mv2_shmem_comm_count(void);
 int MPIDI_CH3I_SHMEM_Coll_get_free_block(); 
diff -ur -x autom4te.cache -x '*~' -x defmsg.h -x errcode.c mvapich2-1.9b.orig/src/mpi/coll/ch3_shmem_coll.c mvapich2-1.9b/src/mpi/coll/ch3_shmem_coll.c
--- mvapich2-1.9b.orig/src/mpi/coll/ch3_shmem_coll.c	2013-02-26 22:24:43.000000000 +0200
+++ mvapich2-1.9b/src/mpi/coll/ch3_shmem_coll.c	2013-04-12 02:32:35.820099733 +0300
@@ -1106,7 +1106,7 @@
     return mv2_enable_shmem_collectives;
 }
 
-inline int mv2_increment_shmem_coll_counter(MPID_Comm *comm_ptr)
+int mv2_increment_shmem_coll_counter(MPID_Comm *comm_ptr)
 {
    int mpi_errno = MPI_SUCCESS, flag=0; 
    PMPI_Comm_test_inter(comm_ptr->handle, &flag);
@@ -1136,7 +1136,7 @@
   goto fn_exit; 
 } 
 
-inline int mv2_increment_allgather_coll_counter(MPID_Comm *comm_ptr)
+int mv2_increment_allgather_coll_counter(MPID_Comm *comm_ptr)
 {   
    int mpi_errno = MPI_SUCCESS, flag=0, errflag=0;
    PMPI_Comm_test_inter(comm_ptr->handle, &flag);
diff -ur -x autom4te.cache -x '*~' -x defmsg.h -x errcode.c mvapich2-1.9b.orig/src/mpid/ch3/channels/mrail/src/Makefile.mk mvapich2-1.9b/src/mpid/ch3/channels/mrail/src/Makefile.mk
--- mvapich2-1.9b.orig/src/mpid/ch3/channels/mrail/src/Makefile.mk	2013-02-26 22:24:43.000000000 +0200
+++ mvapich2-1.9b/src/mpid/ch3/channels/mrail/src/Makefile.mk	2013-04-12 02:32:35.820099733 +0300
@@ -121,7 +121,7 @@
 
 
 lib_lib@MPILIBNAME@_la_LDFLAGS += -L$(top_builddir)/src/mpid/ch3/channels/mrail/src/hwloc/src
-lib_lib@MPILIBNAME@_la_LIBADD += -lhwloc_embedded
+lib_lib@MPILIBNAME@_la_LIBADD += $(top_builddir)/src/mpid/ch3/channels/mrail/src/hwloc/src/libhwloc_embedded.la
 endif
 
 if BUILD_MRAIL_CUDA_KERNELS
