diff -Nru statsmodels-0.14.1-orig/setup.py statsmodels-0.14.1/setup.py
--- statsmodels-0.14.1-orig/setup.py	2023-12-07 14:22:55.000000000 +0100
+++ statsmodels-0.14.1/setup.py	2024-01-11 09:13:32.010072459 +0100
@@ -394,6 +394,7 @@
     ext_modules=extensions,
     maintainer_email=MAINTAINER_EMAIL,
     description=DESCRIPTION,
+    version="0.14.1",
     license=LICENSE,
     url=URL,
     download_url=DOWNLOAD_URL,
