fix path of binaries in config file
author: Pavel Tomanek (INUITS)
--- config/phyluce.conf.orig	2024-02-01 11:38:15.279943701 +0100
+++ config/phyluce.conf	2024-02-01 14:52:31.565662528 +0100
@@ -1,27 +1,27 @@
 [binaries]
-abyss:$CONDA/bin/ABYSS
-abyss-pe:$CONDA/bin/abyss-pe
-bcftools:$CONDA/bin/bcftools
-bedtools:$CONDA/bin/bedtools
-bwa:$CONDA/bin/bwa
-gblocks:$CONDA/bin/Gblocks
-lastz:$CONDA/bin/lastz
-mafft:$CONDA/bin/mafft
-muscle:$CONDA/bin/muscle
-pilon:$CONDA/bin/pilon
-raxml-ng:$CONDA/bin/raxml-ng
-samtools:$CONDA/bin/samtools
-seqtk:$CONDA/bin/seqtk
-spades:$CONDA/bin/spades.py
-trimal:$CONDA/bin/trimal
-velvetg:$CONDA/bin/velvetg
-velveth:$CONDA/bin/velveth
-snakemake:$CONDA/bin/Snakemake
+abyss:$EBROOTABYSS/bin/ABYSS
+abyss-pe:$EBROOTABYSS/bin/abyss-pe
+bcftools:$EBROOTBCFTOOLS/bin/bcftools
+bedtools:$EBROOTBEDTOOLS/bin/bedtools
+bwa:$EBROOTBWA/bin/bwa
+gblocks:$EBROOTGBLOCKS/Gblocks
+lastz:$EBROOTLASTZ/bin/lastz
+mafft:$EBROOTMAFFT/bin/mafft
+muscle:$EBROOTMUSCLE/bin/muscle
+pilon:$EBROOTPILON/bin/pilon
+raxml-ng:$EBROOTRAXMLMINNG/bin/raxml-ng
+samtools:$EBROOTSAMTOOLS/bin/samtools
+seqtk:$EBROOTSEQTK/bin/seqtk
+spades:$EBROOTSPADES/bin/spades.py
+trimal:$EBROOTTRIMAL/bin/trimal
+velvetg:$EBROOTVELVET/bin/velvetg
+velveth:$EBROOTVELVET/bin/velveth
+snakemake:$EBROOTSNAKEMAKE/bin/snakemake
 
 [workflows]
-mapping:$WORKFLOWS/mapping/Snakefile
-correction:$WORKFLOWS/contig-correction/Snakefile
-phasing:$WORKFLOWS/phasing/Snakefile
+mapping:$EBROOTPHYLUCE/phyluce/workflows/mapping/Snakefile
+correction:$EBROOTPHYLUCE/phyluce/workflows/contig-correction/Snakefile
+phasing:$EBROOTPHYLUCE/phyluce/workflows/phasing/Snakefile
 
 #----------------
 #    Advanced
