fix linking problems due to non-existing libraries
author: Bart Oldeman (Compute Canada)
diff -ur ncl_ncarg-6.4.0.orig/config/Project ncl_ncarg-6.4.0/config/Project
--- ncl_ncarg-6.4.0.orig/config/Project	2017-02-23 19:11:55.000000000 +0000
+++ ncl_ncarg-6.4.0/config/Project	2017-06-29 01:42:39.348187224 +0000
@@ -246,12 +246,12 @@
 #endif	/* PNGlib */
 
 #ifndef CAIROlib
-#define CAIROlib -lcairo -lXrender -lfontconfig -lpixman-1 -lfreetype -lexpat -lpng -lz -liconv -lbz2
+#define CAIROlib -lcairo -lXrender -lfontconfig -lpixman-1 -lfreetype -lexpat -lpng -lz -lbz2
 #endif
 
 /* This should be same as CAIROlib except with no explicit paths */
 #ifndef CAIROlibuser
-#define CAIROlibuser -lcairo -lXrender -lfontconfig -lpixman-1 -lfreetype -lexpat -lpng -lz -liconv -lbz2
+#define CAIROlibuser -lcairo -lXrender -lfontconfig -lpixman-1 -lfreetype -lexpat -lpng -lz -lbz2
 #endif
 
 #ifdef	BuildHDFEOS
diff -ur ncl_ncarg-6.4.0.orig/config/Template ncl_ncarg-6.4.0/config/Template
--- ncl_ncarg-6.4.0.orig/config/Template	2017-02-23 19:11:55.000000000 +0000
+++ ncl_ncarg-6.4.0/config/Template	2017-06-29 16:15:39.127448096 +0000
@@ -496,7 +496,7 @@
 #endif	/* LexCommand */
 
 #ifndef	LexLibrary
-#define	LexLibrary	-ll
+#define	LexLibrary	-lfl
 #endif	/* LexLibrary */
 
 #ifndef	YaccCommand

