fix compilation problems due to incorrect type
author: Kenneth Hoste (HPC-UGent)
--- ni/src/lib/nfp/SCRIP2KMLW.c.orig	2017-04-21 11:22:11.523861897 +0200
+++ ni/src/lib/nfp/SCRIP2KMLW.c	2017-04-21 11:22:17.773923222 +0200
@@ -8,17 +8,17 @@
 NhlErrorTypes SCRIP2KML_W(void) {
     /* Defining the Arguments */
     /* Argument # 0 */
-    string * scrip_filename;
+    NclQuark* scrip_filename;
     char * c_scrip_filename;
     int ncid;
 
     /* Argument # 1 */
-    string * kml_filename;
+    NclQuark* kml_filename;
     char * c_kml_filename;
     FILE * fid;
 
     /* Argument # 2 */
-    string * gridname;
+    NclQuark* gridname;
     char * c_gridname;
 
     /* End of Defining the Arguments */
@@ -48,7 +48,7 @@
 
     /* Getting Arguments values */
     /* Argument # 0 */
-    scrip_filename = (string *) NclGetArgValue(
+    scrip_filename = (NclQuark*) NclGetArgValue(
             0,
             3,
             NULL,
@@ -61,7 +61,7 @@
     c_scrip_filename = NrmQuarkToString(*scrip_filename);
 
     /* Argument # 1 */
-    kml_filename = (string *) NclGetArgValue(
+    kml_filename = (NclQuark*) NclGetArgValue(
             1,
             3,
             NULL,
@@ -74,7 +74,7 @@
     c_kml_filename = NrmQuarkToString(*kml_filename);
 
     /* Argument # 1 */
-    gridname = (string *) NclGetArgValue(
+    gridname = (NclQuark*) NclGetArgValue(
             2,
             3,
             NULL,
@@ -212,4 +212,4 @@
     free(grid_corner_lon);
     fclose(fid);
     return (NhlNOERROR);
-}
\ No newline at end of file
+}
--- ni/src/lib/nfp/Unstruct2KMLW.c.orig	2017-04-21 11:20:50.403067496 +0200
+++ ni/src/lib/nfp/Unstruct2KMLW.c	2017-04-21 11:21:34.883504667 +0200
@@ -7,12 +7,12 @@
 NhlErrorTypes Unstruct2KML_W(void) {
     /* Defining the Arguments */
     /* Argument # 0 */
-    string * filename;
+    NclQuark* filename;
     char * c_filename;
     FILE * fid;
 
     /* Argument # 1 */
-    string * gridname;
+    NclQuark* gridname;
     char * c_gridname;
 
     /* Argument # 2 */
@@ -50,7 +50,7 @@
 
     /* Getting Arguments values */
     /* Argument # 0 */
-    filename = (string *) NclGetArgValue(
+    filename = (NclQuark*) NclGetArgValue(
             0,
             6,
             NULL,
@@ -63,7 +63,7 @@
     c_filename = NrmQuarkToString(*filename);
 
     /* Argument # 1 */
-    gridname = (string *) NclGetArgValue(
+    gridname = (NclQuark*) NclGetArgValue(
             1,
             6,
             NULL,
