The Copyright file is missing in the sources.
Author: Pavel Tomanek (Inuits)
--- ncarview/src/bin/ictrans/yMakefile.orig	2025-12-03 12:40:02.230350910 +0100
+++ ncarview/src/bin/ictrans/yMakefile	2025-12-03 12:39:28.007059462 +0100
@@ -41,7 +41,7 @@
 DependTarget($(SRCS))
 
 all-local::
-	@$(CAT) Copyright
+	-@if test -f Copyright; then $(CAT) Copyright; fi
 
 lint: $(HDR) $(SRCS)
 	lint $(CDEFINES) $(LINTLIBS) $(SRCS)  1> lint.out 2>&1
