diff -Nru ncl_ncarg-6.4.0-orig/config/Template ncl_ncarg-6.4.0/config/Template
--- ncl_ncarg-6.4.0-orig/config/Template	2017-02-23 20:11:55.000000000 +0100
+++ ncl_ncarg-6.4.0/config/Template	2018-07-17 09:55:13.000000000 +0200
@@ -711,7 +711,7 @@
 /*
  *CCOPTIONS	= CcOptions $(EXTRA_CCOPTIONS) -framework OpenCL
  */
-CCOPTIONS	= CcOptions $(EXTRA_CCOPTIONS)
+CCOPTIONS	= CcOptions $(EXTRA_CCOPTIONS) -DH5Rdereference_vers=1
 
 BUILDINCDIR	= $(TOP)/include
 /*
