Make sure to link to libaec.

author: Oliver Stueker (ACENET)
date:   2024-01-18
diff -u --recursive ncl-6.6.2.orig/.build/Site.local.template ncl-6.6.2/.build/Site.local.template
--- ncl-6.6.2.orig/.build/Site.local.template	2019-02-27 20:14:39.000000000 -0330
+++ ncl-6.6.2/.build/Site.local.template	2024-01-18 11:43:56.287631825 -0330
@@ -13,9 +13,9 @@
 
 #define HDF5lib -lhdf5_hl -lhdf5 -lz
 #define NetCDF4lib  -lhdf5_hl -lhdf5
-#define GRIB2lib ${grib2_dir}/libgrib2c.a -ljasper -lpng -lz -ljpeg
+#define GRIB2lib ${grib2_dir}/libgrib2c.a -ljasper -lpng -lz -ljpeg -laec
 /* libgdal contains its own libgrib2c, so the proper libgrib2c should always be linked before libgdal. */
-#define GDALlib  ${grib2_dir}/libgrib2c.a -lgdal -lproj -ljpeg
+#define GDALlib  ${grib2_dir}/libgrib2c.a -lgdal -lproj -ljpeg -laec
 #define LexLibrary  ${PREFIX}/lib/libfl.a
 
 ${CAIROLIB}
diff -u --recursive ncl-6.6.2.orig/config/Project ncl-6.6.2/config/Project
--- ncl-6.6.2.orig/config/Project	2019-02-27 20:14:39.000000000 -0330
+++ ncl-6.6.2/config/Project	2024-01-18 11:45:01.575555673 -0330
@@ -302,7 +302,7 @@
 #endif
 
 #ifndef	GRIB2lib
-#define       GRIB2lib   -lgrib2c -ljasper -lpng -lz -ljpeg
+#define       GRIB2lib   -lgrib2c -ljasper -lpng -lz -ljpeg -laec
 #endif	/* GRIB2lib */
 
 #ifndef	EEMDlib

