Only use CXXFLAGS from EasyBuild.

Åke Sandgren, 2019-10-23
diff -ru xmippViz-3.19.04-Apollo.orig/SConstruct xmippViz-3.19.02/SConstruct
--- xmippViz-3.19.04-Apollo.orig/SConstruct	2019-01-30 14:46:50.000000000 +0100
+++ xmippViz-3.19.04-Apollo/SConstruct	2019-10-21 16:25:12.825364682 +0200
@@ -142,13 +142,7 @@
 env['LINKERFORPROGRAMS'] = os.environ.get('LINKERFORPROGRAMS')
 env['CCFLAGS'] = os.environ.get('CCFLAGS', '').split()
 cxxFlags = os.environ.get('CXXFLAGS', '') 
-if os.environ.get('DEBUG', '0') == 'True': #FIXME, use 1, true, yes...
-   cxxFlags += ' -g'
-else:
-    if cxxFlags.find("-O")==-1:
-        cxxFlags += (" -O3" if 'TRAVIS' not in os.environ else " -O0") #don't optimize on Travis, as it slows down the build
 env['CXXFLAGS'] = cxxFlags.split()
-os.environ['CXXFLAGS'] = cxxFlags # FIXME use only env or os.environ in the rest of the code
 env['LINKFLAGS'] = os.environ.get('LINKFLAGS', '').split()
 
 
