Don't try to download DL models.

Åke Sandgren, 20191023
diff -ru xmipp-3.19.02.orig/xmipp xmipp-3.19.02/xmipp
--- xmipp-3.19.02.orig/xmipp	2019-02-12 16:07:49.000000000 +0100
+++ xmipp-3.19.02/xmipp	2019-10-22 11:26:19.330738967 +0200
@@ -904,17 +904,6 @@
     ok = ok and runJob(cpCmd+" -Lr src/xmippViz/external/imagej "+dirname+"/bindings/java/")
     ok = ok and runJob(cpCmd+" src/xmippViz/bindings/python/xmippViz.py "+dirname+"/bindings/python/")
 
-    if 'TRAVIS' not in os.environ:
-        firstTime = True
-        global pDLdownload
-        while pDLdownload is not None and pDLdownload.poll() is None:
-            if firstTime:
-                print(blue("The download of some Deep Learning models is not "
-                           "finish yet. Continuing the download in the foregroung..."))
-                firstTime = False
-        createDir(dirname+"/models")
-        ok = ok and runJob("cat src/xmipp/models/xmipp_model_*.tgz | tar xzf - -i --directory="+dirname+"/models")
-
     if not ok:
         print(red("\nSome error occurred during the installation.\n"))
         sys.exit(1)
