Only use CXXFLAGS from EasyBuild.

Åke Sandgren, 2019-10-10
diff -ru xmipp-3.19.04-Apollo.orig/SConstruct xmipp-3.19.02/SConstruct
--- xmipp-3.19.04-Apollo.orig/SConstruct	2019-10-10 10:24:19.955942471 +0200
+++ xmipp-3.19.04-Apollo/SConstruct	2019-10-10 10:30:58.256076334 +0200
@@ -141,13 +141,7 @@
 env['LINKERFORPROGRAMS'] = os.environ.get('LINKERFORPROGRAMS')
 env['CCFLAGS'] = os.environ.get('CCFLAGS', '').split()
 cxxFlags = os.environ.get('CXXFLAGS', '')
-if os.environ.get('DEBUG', '0') == 'True': #FIXME, use 1, true, yes...
-   cxxFlags += ' -g'
-else:
-    if cxxFlags.find("-O")==-1:
-        cxxFlags += (" -O3" if 'TRAVIS' not in os.environ else " -O0") #don't optimize on Travis, as it slows down the build
 env['CXXFLAGS'] = cxxFlags.split()
-os.environ['CXXFLAGS'] = cxxFlags # FIXME use only env or os.environ in the rest of the code
 env['LINKFLAGS'] = os.environ.get('LINKFLAGS', '').split()
 
 
