diff -ru xmipp/libraries/data/rwIMAGIC.cpp xmipp_working/libraries/data/rwIMAGIC.cpp
--- xmipp/libraries/data/rwIMAGIC.cpp	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/rwIMAGIC.cpp	2015-03-12 17:35:53.292172326 +0100
@@ -40,6 +40,8 @@
 */

 #define IMAGICSIZE 1024 // Size of the IMAGIC header for each image
+#define SIZEOF_INT sizeof(int)
+

 ///@defgroup Imagic Imagic File format
 ///@ingroup ImageFormats
