commit 36ca78858e442f1df23f5cb6b87c877a864eff44
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Tue Oct 18 14:23:16 2022 +0200

    Only use CXXFLAGS from EasyBuild.

diff --git xmippCore-3.22.07-Helios/SConstruct xmippCore-3.22.07-Helios/SConstruct
index 40ed01f..277443d 100644
--- xmippCore-3.22.07-Helios/SConstruct
+++ xmippCore-3.22.07-Helios/SConstruct
@@ -136,13 +136,7 @@ env['CXX'] = os.environ.get('CXX')
 env['LINKERFORPROGRAMS'] = os.environ.get('LINKERFORPROGRAMS')
 env['CCFLAGS'] = os.environ.get('CCFLAGS', '').split()
 cxxFlags = os.environ.get('CXXFLAGS', '') 
-if os.environ.get('DEBUG', '0') == 'True': #FIXME, use 1, true, yes...
-   cxxFlags += ' -g'
-else:
-    if cxxFlags.find("-O")==-1:
-        cxxFlags += (" -O3" if 'TRAVIS' not in os.environ else " -O0") #don't optimize on Travis, as it slows down the build
 env['CXXFLAGS'] = cxxFlags.split()
-os.environ['CXXFLAGS'] = cxxFlags # FIXME use only env or os.environ in the rest of the code
 env['LINKFLAGS'] = os.environ.get('LINKFLAGS', '').split()
 
 
