commit 4c1e2d678c5a33eae2ddb9f0558523d6d697e2b6
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Thu Oct 27 08:04:04 2022 +0200

    Don't add useless libraries to XmippInterfaceCuda and XmippParallelCuda

diff --git xmipp-3.22.07-Helios/src/xmipp/SConstruct xmipp-3.22.07-Helios/src/xmipp/SConstruct
index e70ea2cd..077541e8 100644
--- xmipp-3.22.07-Helios/src/xmipp/SConstruct
+++ xmipp-3.22.07-Helios/src/xmipp/SConstruct
@@ -209,9 +209,7 @@ def addCppLibrary(env, name, dirs=[], tars=[], untarTargets=['configure'], patte
     """
     _libs = list(libs)
     _libpath = list(libpath)+external_libdirs
-    if name == "XmippInterfaceCuda":
-        _libs.append("XmippCuda")
-    if "Cuda" in name:
+    if name == "XmippCuda":
         _libs.append("cudart")
         #_libs.append("cuda")
         _libs.append("cufft")
