Fix detection of OpenCV >= 3

Åke Sandgren, 2022-11-14
diff --git a/src/xmipp/SConscript b/src/xmipp/SConscript
index a767fb99..89777947 100644
--- a/src/xmipp/SConscript
+++ b/src/xmipp/SConscript
@@ -46,7 +46,7 @@ debug = get('DEBUG')
 matlab = get('MATLAB')
 opencv = env.GetOption('opencv') and get('OPENCV')
 opencvsupportscuda = get('OPENCVSUPPORTSCUDA')
-opencv_3 = get('OPENCV3')
+opencv_ver = int(os.environ.get('OPENCV_VERSION', '0'))
 starpu = get('STARPU')
 
 if opencv:
@@ -55,7 +55,7 @@ if opencv:
                   'opencv_video',
                   'libopencv_calib3d']
     if opencvsupportscuda:
-        if opencv_3:
+        if opencv_ver >= 3:
             opencvLibs+=['libopencv_cudaoptflow', 'libopencv_cudaarithm']
         else:
             opencvLibs+=['libopencv_gpu']
