Add missing pthread lib to XmippParallel

Åke Sandgren, 20191023
diff -ru xmipp-3.19.04-Apollo.orig/SConscript xmipp-3.19.02/SConscript
--- xmipp-3.19.04-Apollo.orig/SConscript	2019-02-12 16:07:49.000000000 +0100
+++ xmipp-3.19.04-Apollo/SConscript	2019-10-22 09:50:57.978158329 +0200
@@ -159,7 +159,7 @@
 # MPI
 dirs = ['libraries']
 patterns=['parallel/*.cpp']
-addLib('XmippParallel',dirs=dirs,patterns=patterns, libs=['Xmipp'], mpi=True)
+addLib('XmippParallel',dirs=dirs,patterns=patterns, libs=['Xmipp', 'pthread'], mpi=True)
 
 
 #  ***********************************************************************
