Add missing pthread lib to XmippInterfaceCuda

Åke Sandgren, 20191023
diff -ru xmipp-3.19.04-Apollo.orig/SConscript xmipp-3.19.02/SConscript
--- xmipp-3.19.04-Apollo.orig/SConscript	2019-02-12 16:07:49.000000000 +0100
+++ xmipp-3.19.04-Apollo/SConscript	2019-10-23 18:45:10.051773978 +0200
@@ -151,7 +154,7 @@
 
 # CUDA
 if cuda:
-    addLib('XmippInterfaceCuda', dirs=['libraries'], patterns=['reconstruction_adapt_cuda/*.cpp'], libs=['Xmipp'])
+    addLib('XmippInterfaceCuda', dirs=['libraries'], patterns=['reconstruction_adapt_cuda/*.cpp'], libs=['Xmipp', 'pthread'])
     addLib('XmippCuda', dirs=['libraries'], patterns=['reconstruction_cuda/*.cpp'], nvcc=True, suffix=".a")
     addLib('XmippParallelCuda', dirs=['libraries'], patterns=['parallel_adapt_cuda/*.cpp'],
            libs=['Xmipp','XmippInterfaceCuda','XmippCuda'], mpi=True)
