--- cmake/./macros/macro_copy_target_properties.cmake.orig	2025-04-28 14:00:11.000000000 -0700
+++ cmake/./macros/macro_copy_target_properties.cmake	2025-04-28 14:00:20.000000000 -0700
@@ -133,7 +133,7 @@
   if(NOT "${_include_directories}" STREQUAL "")
     remove_duplicates(_include_directories)
     message(STATUS "    INCLUDE_DIRECTORIES: ${_include_directories}")
-    target_include_directories(${_destination_target} SYSTEM INTERFACE ${_include_directories})
+    target_include_directories(${_destination_target} INTERFACE ${_include_directories})
   endif()
 
   if(NOT "${_compile_definitions}" STREQUAL "")
