From 89b89043c3fc45883e9b11bba6276a23cfd9feb3 Mon Sep 17 00:00:00 2001
From: David Wells <drwells@email.unc.edu>
Date: Thu, 8 Dec 2022 10:09:03 -0500
Subject: [PATCH] step-70: remove a dependency on boost.

---
 examples/step-70/step-70.cc | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/examples/step-70/step-70.cc b/examples/step-70/step-70.cc
index 2ce4d97488..879e5c1641 100644
--- a/examples/step-70/step-70.cc
+++ b/examples/step-70/step-70.cc
@@ -837,8 +835,12 @@ namespace Step70
         const auto &manifold_id   = pair.first;
         const auto &cad_file_name = pair.second;
 
-        const auto extension = boost::algorithm::to_lower_copy(
-          cad_file_name.substr(cad_file_name.find_last_of('.') + 1));
+        std::string extension =
+          cad_file_name.substr(cad_file_name.find_last_of('.') + 1);
+        std::transform(extension.begin(),
+                       extension.end(),
+                       extension.begin(),
+                       [](const char c) -> char { return std::tolower(c); });
 
         TopoDS_Shape shape;
         if (extension == "iges" || extension == "igs")
-- 
2.50.1

