--- BipartiteMatchings/ApproxWeightPerfectMatching.h.orig	2019-03-18 22:31:43.553409785 +0000
+++ BipartiteMatchings/ApproxWeightPerfectMatching.h	2019-03-18 22:31:59.394432722 +0000
@@ -9,7 +9,7 @@
 #ifndef ApproxWeightPerfectMatching_h
 #define ApproxWeightPerfectMatching_h
 
-#include "../CombBLAS.h"
+#include "../CombBLAS/CombBLAS.h"
 #include "BPMaximalMatching.h"
 #include "BPMaximumMatching.h"
 #include <parallel/algorithm>
--- BipartiteMatchings/BPMaximalMatching.h.orig	2019-03-18 22:35:20.138723402 +0000
+++ BipartiteMatchings/BPMaximalMatching.h	2019-03-18 22:36:35.425832418 +0000
@@ -1,7 +1,7 @@
 #ifndef BP_MAXIMAL_MATCHING_H
 #define BP_MAXIMAL_MATCHING_H
 
-#include "../CombBLAS.h"
+#include "../CombBLAS/CombBLAS.h"
 #include <iostream>
 #include <functional>
 #include <algorithm>
--- BipartiteMatchings/BPMaximumMatching.h.orig	2019-03-18 22:38:11.435971442 +0000
+++ BipartiteMatchings/BPMaximumMatching.h	2019-03-18 22:38:25.678992066 +0000
@@ -1,7 +1,7 @@
 #ifndef BP_MAXIMUM_MATCHING_H
 #define BP_MAXIMUM_MATCHING_H
 
-#include "../CombBLAS.h"
+#include "../CombBLAS/CombBLAS.h"
 #include <mpi.h>
 #include <sys/time.h>
 #include <iostream>
--- BipartiteMatchings/MatchingDefs.h.orig	2019-03-18 22:37:42.795929971 +0000
+++ BipartiteMatchings/MatchingDefs.h	2019-03-18 22:37:56.265949475 +0000
@@ -9,7 +9,7 @@
 #ifndef MatchingDefs_h
 #define MatchingDefs_h
 
-#include "../CombBLAS.h"
+#include "../CombBLAS/CombBLAS.h"
 #include <iostream>
 
 namespace combblas {
--- BipartiteMatchings/Utility.h.orig	2019-03-18 22:36:50.706854545 +0000
+++ BipartiteMatchings/Utility.h	2019-03-18 22:37:26.625906557 +0000
@@ -1,7 +1,7 @@
 #ifndef BP_UTILITY_H
 #define BP_UTILITY_H
 
-#include "../CombBLAS.h"
+#include "../CombBLAS/CombBLAS.h"
 
 namespace combblas {
 
