diff -ru clr-rocm-5.6.0.orig/CMakeLists.txt clr-rocm-5.6.0/CMakeLists.txt
--- clr-rocm-5.6.0.orig/CMakeLists.txt	2023-06-13 21:19:07.000000000 +0200
+++ clr-rocm-5.6.0/CMakeLists.txt	2023-07-07 07:51:24.161497334 +0200
@@ -32,7 +32,7 @@
 #############
 if(CLR_BUILD_HIP)
     if(NOT EXISTS ${HIPCC_BIN_DIR}/hipconfig)
-        message(FATAL_ERROR "Please pass hipcc/build or hipcc/bin using -DHIPCC_BIN_DIR. Passed HIPCC_BIN_DIR: "${HIPCC_BIN_DIR})
+        message(FATAL_ERROR "Please pass hipcc/build or hipcc/bin using -DHIPCC_BIN_DIR. Passed HIPCC_BIN_DIR: ${HIPCC_BIN_DIR}")
     endif()
     if(NOT DEFINED HIP_COMMON_DIR)
         message(FATAL_ERROR "Please pass HIP using -DHIP_COMMON_DIR. HIP_COMMON_DIR is incorrect")
