From 96f924e18d70add1841fe5b11c60e8d8d2a872b2 Mon Sep 17 00:00:00 2001
From: snehring <snehring@iastate.edu>
Date: Wed, 27 Apr 2022 11:29:07 -0500
Subject: [PATCH] Makefile changes

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index c473df9..b0d754d 100644
--- a/Makefile
+++ b/Makefile
@@ -2,8 +2,7 @@
 # HAPCUT2 MAKEFILE
 default: all
 
-CC=gcc -g -O3 -Wall -D_GNU_SOURCE
-CFLAGS=-Wall
+CFLAGS=-Wall -g -O3 -Wall -D_GNU_SOURCE
 
 # DIRECTORIES
 B=build
