--- makemake.orig	2018-01-14 22:29:42.000000000 -0800
+++ makemake	2018-01-14 22:30:20.000000000 -0800
@@ -73,7 +73,7 @@
 
 # List the default make targets.
 
-DEFAULT_TARGETS="lib grfont.dat prog pgplot.doc"
+DEFAULT_TARGETS="lib grfont.dat pgplot.doc cpg pgxwin_server pgplot.doc pgplot-routines.tex"
 
 # Get the configuration variables.
 
@@ -806,7 +806,7 @@
 # libraries.
 #-----------------------------------------------------------------------
 
-lib : libpgplot.a $(SHARED_LIB)
+lib : libpgplot.a libcpgplot.a $(SHARED_LIB)
 
 libpgplot.a : $(PG_ROUTINES) $(PG_NON_STANDARD) $(GR_ROUTINES) \
 	   $(DISPATCH_ROUTINE) $(DRIVERS) $(SYSTEM_ROUTINES)
@@ -816,6 +816,13 @@
 		$(DRIVERS) $(SYSTEM_ROUTINES) | sort | uniq`
 	$(RANLIB) libpgplot.a
 
+libpgplot.so : $(PG_ROUTINES) $(PG_NON_STANDARD) $(GR_ROUTINES) \
+	$(DISPATCH_ROUTINE) $(DRIVERS) $(SYSTEM_ROUTINES)
+	f77 -shared -fPIC -o libpgplot.so \
+		`ls $(PG_ROUTINES) \
+		$(PG_NON_STANDARD) $(GR_ROUTINES) $(DISPATCH_ROUTINE) \
+		$(DRIVERS) $(SYSTEM_ROUTINES) | sort | uniq` $(SHARED_LIB_LIBS)
+
 EOD
 
 # Emit the shared library dependency if requested.
@@ -913,6 +920,7 @@
  echo "	$SRC/makehelp \$(PG_SOURCE) > pgplot.hlp"
  echo "pgplot-routines.tex: \$(PG_SOURCE)"
  echo "	$SRC/maketex \$(PG_SOURCE) > pgplot-routines.tex"
+ echo "	pdflatex pgplot-routines.tex" 
 ) >> makefile
 
 cat >> makefile << \EOD
@@ -1050,6 +1058,13 @@
 	$(RANLIB) libcpgplot.a
 	rm -f cpg*.o
 
+libcpgplot.so cpgplot.h: $(PG_SOURCE) pgbind
+	./pgbind $(PGBIND_FLAGS) -h -w $(PG_SOURCE)
+	$(CCOMPL) -c $(CFLAGC) cpg*.c
+	rm -f cpg*.c
+	f77 -shared -fPIC -o libcpgplot.so cpg*.o $(SHARED_LIB_LIBS)
+	rm -f cpg*.o
+
 cpgdemo: cpgplot.h $(SRC)/cpg/cpgdemo.c libcpgplot.a
 	$(CCOMPL) $(CFLAGD) -c -I. $(SRC)/cpg/cpgdemo.c
 	$(FCOMPL) -o cpgdemo cpgdemo.o $(CPGPLOT_LIB) $(LIBS)
--- maketex.orig	2018-01-14 22:30:07.000000000 -0800
+++ maketex	2018-01-14 22:30:27.000000000 -0800
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/env perl
 
 $, = ' ';		# set output field separator
 $\ = "\n";		# set output record separator
--- sys_linux/f77_gcc.conf.orig	2018-01-14 22:29:58.000000000 -0800
+++ sys_linux/f77_gcc.conf	2018-01-14 22:30:40.000000000 -0800
@@ -6,7 +6,7 @@
 #           X2DRIV (/xdisp and /figdisp).
 # The arguments needed by the C compiler to locate X-window include files.
  
-   XINCL="-I/usr/X11R6/include"
+   XINCL=
 
 # Optional: Needed by XMDRIV (/xmotif).
 # The arguments needed by the C compiler to locate Motif, Xt and
@@ -24,7 +24,7 @@
 # The arguments needed by the C compiler to locate Tcl, Tk and
 # X-window include files.
  
-   TK_INCL="-I/usr/include "
+   TK_INCL=
 
 # Optional: Needed by RVDRIV (/xrv).
 # The arguments needed by the C compiler to locate Rivet, Tcl, Tk and
@@ -41,7 +41,7 @@
 # The FORTRAN compiler flags to use when compiling the pgplot library.
 # (NB. makemake prepends -c to $FFLAGC where needed)
  
-   FFLAGC="-u"
+   FFLAGC="-fPIC"
 
 # Mandatory.
 # The FORTRAN compiler flags to use when compiling fortran demo programs.
@@ -58,7 +58,7 @@
 # Mandatory.
 # The C compiler flags to use when compiling the pgplot library.
  
-   CFLAGC="-DPG_PPU -O2"
+   CFLAGC="-DPG_PPU -fPIC -O2"
 
 # Mandatory.
 # The C compiler flags to use when compiling C demo programs.
@@ -75,7 +75,7 @@
 # The library-specification flags to use when linking normal pgplot
 # demo programs.
  
-   LIBS="-lf2c -L/usr/X11R6/lib -lX11 \`\$(SRC)/cpg/libgcc_path.sh\` -lgcc -lm -lc"
+   LIBS="-lX11 \`\$(SRC)/cpg/libgcc_path.sh\` -lgcc -lm -lc"
 
 # Optional: Needed by XMDRIV (/xmotif).
 # The library-specification flags to use when linking motif
@@ -104,7 +104,7 @@
 # Optional: Needed on systems that support shared libraries.
 # The name to give the shared pgplot library.
  
-   SHARED_LIB=""
+   SHARED_LIB="libpgplot.so libcpgplot.so"
 
 # Optional: Needed if SHARED_LIB is set.
 # How to create a shared library from a trailing list of object files.
@@ -118,7 +118,7 @@
 # library-specification flags used to specify these libraries to
 # $SHARED_LD
  
-   SHARED_LIB_LIBS=""
+   SHARED_LIB_LIBS="-lpng -lz -lX11"
 
 # Optional:
 # Compiler name used on Next systems to compile objective-C files.
