diff -ru opencv-2.4.13.orig/cmake/OpenCVDetectCUDA.cmake opencv-2.4.13/cmake/OpenCVDetectCUDA.cmake
--- opencv-2.4.13.orig/cmake/OpenCVDetectCUDA.cmake	2016-04-20 16:21:19.000000000 +0200
+++ opencv-2.4.13/cmake/OpenCVDetectCUDA.cmake	2018-04-25 20:45:51.624109176 +0200
@@ -103,6 +103,8 @@
     else()
       if(${CUDA_VERSION} VERSION_LESS "5.0")
         set(__cuda_arch_bin "1.1 1.2 1.3 2.0 2.1(2.0) 3.0")
+      elseif(${CUDA_VERSION} VERSION_GREATER_EQUAL "8.0")
+        set(__cuda_arch_bin "3.0 3.2 3.5 3.7 5.0 5.2 5.3 6.0 6.1")
       elseif(${CUDA_VERSION} VERSION_GREATER "6.5")
         set(__cuda_arch_bin "2.0 2.1(2.0) 3.0 3.5")
       else()
