diff -ru opencv-2.4.13.orig/cmake/templates/OpenCVConfig.cmake.in opencv-2.4.13/cmake/templates/OpenCVConfig.cmake.in
--- opencv-2.4.13.orig/cmake/templates/OpenCVConfig.cmake.in	2016-04-20 16:21:19.000000000 +0200
+++ opencv-2.4.13/cmake/templates/OpenCVConfig.cmake.in	2018-04-25 20:17:19.799142173 +0200
@@ -248,13 +248,8 @@
 
   set(OpenCV_CUDA_LIBS_ABSPATH ${CUDA_LIBRARIES})
 
-  if(${CUDA_VERSION} VERSION_LESS "5.5")
+  if(${CUDA_VERSION} VERSION_GREATER "4.0")
     list(APPEND OpenCV_CUDA_LIBS_ABSPATH ${CUDA_npp_LIBRARY})
-  else()
-    find_cuda_helper_libs(nppc)
-    find_cuda_helper_libs(nppi)
-    find_cuda_helper_libs(npps)
-    list(APPEND OpenCV_CUDA_LIBS_ABSPATH ${CUDA_nppc_LIBRARY} ${CUDA_nppi_LIBRARY} ${CUDA_npps_LIBRARY})
   endif()
 
   if(OpenCV_USE_CUBLAS)
