Fix CMake build to use Fortran-YAML from EasyBuild
author: Alex Domingo (Vrije Universiteit Brussel)
diff --git a/CMakeLists.txt.orig b/CMakeLists.txt
index 3756b30..41cb10e 100644
--- a/CMakeLists.txt.orig
+++ b/CMakeLists.txt
@@ -40,11 +40,7 @@ if(FLEXOUT_USE_NetCDF)
 endif()
 
 if (NOT TARGET yaml)
-  if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/extern/fortran-yaml/CMakeLists.txt")
-    message(FATAL_ERROR "Fortran-YAML not found at extern/fortran-yaml. Please retrieve this submodule first by running \"git submodule update --init\" within your flexout source directory.")
-  endif()
-  add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/extern/fortran-yaml" "${CMAKE_CURRENT_BINARY_DIR}/extern/yaml")
-  set_property(TARGET yaml PROPERTY FOLDER flexout)
+  include_directories(SYSTEM "$ENV{EBROOTFORTRANMINYAML}/include")
 endif()
 
 add_subdirectory(src)
