diff -Nru cell2location-0.05-alpha.orig/setup.py cell2location-0.05-alpha/setup.py
--- cell2location-0.05-alpha.orig/setup.py	2021-05-21 10:14:10.897660000 +0100
+++ cell2location-0.05-alpha/setup.py	2021-05-21 10:14:40.572975000 +0100
@@ -5,7 +5,7 @@
 
 
 def setup_package():
-    install_requires = ['pymc3', 'torch', 'theano', 'pygpu', 'numpy', 'pandas', 'scanpy', 'plotnine']
+    install_requires = ['pymc3', 'torch', 'theano', 'numpy', 'pandas', 'scanpy', 'plotnine']
     metadata = dict(
         name='cell2location',
         version='0.05',
