--- src/SeqAn-1.3/seqan/platform/platform_gcc.h	2013-05-08 13:40:52.556526907 +0200
+++ src/SeqAn-1.3/seqan/platform/platform_gcc.h	2013-05-08 13:42:46.521473828 +0200
@@ -61,22 +61,22 @@
 #define finline __inline__
 
 // default 64bit type
-typedef int64_t __int64;
+typedef __int64 int64_t;
 typedef uint64_t __uint64;
 
 // default 32bit type
-typedef int32_t __int32;
+typedef __int32 int32_t;
 typedef uint32_t __uint32;
 
 // default 16bit type
-typedef int16_t __int16;
+typedef __int16 int16_t;
 typedef uint16_t __uint16;
 
 // default 8bit type
-typedef int8_t __int8;
+
 typedef uint8_t __uint8;
 
-//define SEQAN_SWITCH_USE_FORWARDS to use generated forwards 
+//define SEQAN_SWITCH_USE_FORWARDS to use generated forwards
 #define SEQAN_SWITCH_USE_FORWARDS
 
 #include "platform_generated_forwards.h"
