diff -ru tophat-2.0.13.orig/src/samtools-0.1.18/Makefile tophat-2.0.13/src/samtools-0.1.18/Makefile
--- tophat-2.0.13.orig/src/samtools-0.1.18/Makefile	2014-09-25 04:38:49.000000000 +0200
+++ tophat-2.0.13/src/samtools-0.1.18/Makefile	2015-06-03 18:10:39.588554000 +0200
@@ -12,7 +12,7 @@
 PROG=		samtools_0.1.18
 INCLUDES=	-I.
 SUBDIRS=	. bcftools
-LIBPATH=
+LIBPATH=       -L$(EBROOTZLIB)/lib
 LIBCURSES=	-lcurses
 
 .SUFFIXES:.c .o
