fix for "error: AM_INIT_AUTOMAKE expanded multiple times"
author: Kenneth Hoste (HPC-UGent)
--- tophat-2.1.2/configure.ac.orig	2018-05-24 17:24:31.000000000 +0200
+++ tophat-2.1.2/configure.ac	2023-03-10 14:44:29.924070336 +0100
@@ -108,8 +108,6 @@
 CXXFLAGS="$CXXFLAGS $BAM_CPPFLAGS $BOOST_CPPFLAGS -I./SeqAn-1.4.2"
 LDFLAGS="$BAM_LDFLAGS $BOOST_LDFLAGS $user_LDFLAGS"
 
-AM_INIT_AUTOMAKE([-Wall foreign tar-pax foreign])
-
 # makefiles to configure
 AC_CONFIG_FILES([Makefile src/Makefile])
 
