# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/04
diff -ru pymol-open-source-2.4.0/contrib/vr/OpenVRAction.h pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRAction.h
--- pymol-open-source-2.4.0/contrib/vr/OpenVRAction.h	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRAction.h	2021-04-09 16:31:29.522314544 +0200
@@ -35,7 +35,7 @@
 #define _H_OpenVRAction
 
 // system headers
-#include "openvr.h"
+#include <openvr/openvr.h>
 
 struct OpenVRAction {
   char const* path;
diff -ru pymol-open-source-2.4.0/contrib/vr/OpenVRController.h pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRController.h
--- pymol-open-source-2.4.0/contrib/vr/OpenVRController.h	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRController.h	2021-04-09 16:31:29.522314544 +0200
@@ -35,7 +35,7 @@
 #define _H_OpenVRController
 
 // system headers
-#include "openvr.h"
+#include <openvr/openvr.h>
 
 // local headers
 #include "OpenVRLaserSource.h"
diff -ru pymol-open-source-2.4.0/contrib/vr/OpenVRControllerModel.h pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRControllerModel.h
--- pymol-open-source-2.4.0/contrib/vr/OpenVRControllerModel.h	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRControllerModel.h	2021-04-09 16:31:29.522314544 +0200
@@ -35,7 +35,7 @@
 #define _H_OpenVRControllerModel
 
 // system headers
-#include "openvr.h"
+#include <openvr/openvr.h>
 
 // pymol headers
 #include "PyMOLGlobals.h"
diff -ru pymol-open-source-2.4.0/contrib/vr/OpenVRMenu.h pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRMenu.h
--- pymol-open-source-2.4.0/contrib/vr/OpenVRMenu.h	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRMenu.h	2021-04-09 16:31:29.522314544 +0200
@@ -35,7 +35,7 @@
 #define _H_OpenVRMenu
 
 // system headers
-#include "openvr.h"
+#include <openvr/openvr.h>
 #include "os_gl.h"
 
 // pymol headers
diff -ru pymol-open-source-2.4.0/contrib/vr/OpenVRMode.cpp pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRMode.cpp
--- pymol-open-source-2.4.0/contrib/vr/OpenVRMode.cpp	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRMode.cpp	2021-04-09 16:36:47.604011461 +0200
@@ -40,7 +40,7 @@
 #include "os_python.h"
 #include <string>
 #include <vector>
-#include "openvr.h"
+#include <openvr/openvr.h>
 
 // pymol headers
 #include "PyMOLOptions.h"
diff -ru pymol-open-source-2.4.0/contrib/vr/OpenVRStubDevice.h pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRStubDevice.h
--- pymol-open-source-2.4.0/contrib/vr/OpenVRStubDevice.h	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRStubDevice.h	2021-04-09 16:31:29.523314559 +0200
@@ -35,7 +35,7 @@
 #define _H_OpenVRStubDevice
 
 // system headers
-#include "openvr.h"
+#include <openvr/openvr.h>
 
 namespace vr {
 namespace stub {
diff -ru pymol-open-source-2.4.0/contrib/vr/OpenVRStub.h pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRStub.h
--- pymol-open-source-2.4.0/contrib/vr/OpenVRStub.h	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_openvrpatch/contrib/vr/OpenVRStub.h	2021-04-09 16:31:29.523314559 +0200
@@ -35,7 +35,7 @@
 #define _H_OpenVRStub
 
 // system headers
-#include "openvr.h"
+#include <openvr/openvr.h>
 
 namespace vr {
 namespace stub {
