# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2021/04
diff -ru pymol-open-source-2.4.0/modules/pmg_tk/startup/apbs_tools.py pymol-open-source-2.4.0_APBS_tools/modules/pmg_tk/startup/apbs_tools.py
--- pymol-open-source-2.4.0/modules/pmg_tk/startup/apbs_tools.py	2020-05-20 10:22:48.000000000 +0200
+++ pymol-open-source-2.4.0_APBS_tools/modules/pmg_tk/startup/apbs_tools.py	2021-04-09 17:11:55.412756982 +0200
@@ -145,7 +145,7 @@
 
 import tempfile
 import os,math,re
-import string
+#import string
 import sys
 
 if sys.version_info[0] < 3:
@@ -269,7 +269,7 @@
     searchDirs.append(os.path.join("/sw", "share", "apbs-mpi-lammpi", "tools", "manip"))
     searchDirs.append(os.path.join("/usr", "local", "share", "tools", "manip"))
 
-    searchDirs.extend(string.split(os.environ["PATH"], ":"))
+    searchDirs.extend(os.environ["PATH"].split(":"))
     searchDirs.append(os.path.join("/usr", "local", "bin"))
     searchDirs.append(os.path.join("/opt", "local", "bin"))
     searchDirs.append(os.path.join("/sw", "bin"))
