--- ./CMakeLists.txt.orig	2019-03-18 23:35:20.955242818 +0000
+++ ./CMakeLists.txt	2019-03-19 18:53:58.847654681 +0000
@@ -84,7 +84,7 @@
   set(PROJECT_NAME_LIB_EXPORT libsuperlu_dist.a)
 endif()
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 
 if (XSDK_ENABLE_Fortran)
   enable_language (Fortran)
--- ./SRC/AWPM_CombBLAS.hpp.orig	2019-03-18 22:53:22.956332345 +0000
+++ ./SRC/AWPM_CombBLAS.hpp	2019-03-18 23:32:14.302958753 +0000
@@ -1,8 +1,8 @@
 #ifndef AWPM_CombBLAS_h
 #define AWPM_CombBLAS_h
 
-#include "CombBLAS.h"
-#include "ApproxWeightPerfectMatching.h"
+#include "CombBLAS/CombBLAS.h"
+#include "BipartiteMatchings/ApproxWeightPerfectMatching.h"
 #include "superlu_ddefs.h"
 
 
