Author: Jasper Grimm (UoY)
Cast size_t to int before calling abs()
diff -Nru fsom-a6ef318fbd347c53189384aef7f670c0e6ce89a3/fsom.c fsom-a6ef318fbd347c53189384aef7f670c0e6ce89a3.2/fsom.c
--- fsom-a6ef318fbd347c53189384aef7f670c0e6ce89a3/fsom.c	2014-11-19 11:03:18.000000000 +0000
+++ fsom-a6ef318fbd347c53189384aef7f670c0e6ce89a3.2/fsom.c	2021-05-05 15:55:11.000000000 +0100
@@ -591,10 +591,10 @@
 	som_neuron_t *neuron;
 	for ( i=0; i < net->output_layer->neurons_rows; ++i )
 	{
-		dist_i = abs( x - i );
+		dist_i = abs( int(x - i) );
 		for ( j=0; j < net->output_layer->neurons_cols; ++j )
 		{
-			dist     = pow( dist_i + abs ( y - j ), 4 );
+			dist     = pow( dist_i + abs ( int(y - j) ), 4 );
 			inv_dist = (1.0 / ((double) dist + 1)) * l_rate;
 			neuron   = net->output_layer->neurons[i][j];
 			for ( k=0; k < net->input_layer->neurons_count; ++k )
