remove memcpy wrapper to fix hang
author: John Dey (Fred Hutch)
diff -ruN FastANI-1.1.orig/Makefile.in FastANI-1.1/Makefile.in
--- FastANI-1.1.orig/Makefile.in	2018-08-01 09:29:53.000000000 -0700
+++ FastANI-1.1/Makefile.in	2018-12-07 16:00:29.231658000 -0800
@@ -6,7 +6,7 @@
 ifeq ($(UNAME_S),Darwin)
 	CXXFLAGS += -mmacosx-version-min=10.7 -stdlib=libc++
 else
-	CXXFLAGS += -include src/common/memcpyLink.h -Wl,--wrap=memcpy
+	CXXFLAGS += -include src/common/memcpyLink.h
 	CFLAGS += -include src/common/memcpyLink.h
 endif
 
