Removes dependency on opencv-python-headless which causes conflicts.

--- 3.0.1/foss-2023a-CUDA-12.1.1/Slideflow/slideflow-3.0.1/setup.py.orig	2024-10-02 14:50:03.699316470 +0200
+++ 3.0.1/foss-2023a-CUDA-12.1.1/Slideflow/slideflow-3.0.1/setup.py	2024-10-02 14:50:31.615652542 +0200
@@ -128,7 +128,6 @@
         'scikit-learn',
         'matplotlib>=3.2',
         'imageio',
-        opencv_pkg,
         'shapely',
         'umap-learn',
         'seaborn<0.14',
