Removes dependency on opencv-python-headless which causes conflicts.

--- 2.3.1/foss-2023a-PyTorch-2.1.2-CUDA-12.1.1/Slideflow/slideflow-2.3.1/setup.py.orig	2024-06-10 13:34:16.927695606 +0200
+++ 2.3.1/foss-2023a-PyTorch-2.1.2-CUDA-12.1.1/Slideflow/slideflow-2.3.1/setup.py	2024-06-10 13:35:12.698762839 +0200
@@ -104,7 +104,6 @@
         'scikit-learn',
         'matplotlib>=3.2',
         'imageio',
-        'opencv-python-headless',
         'shapely',
         'umap-learn',
         'seaborn<0.12',
