Removes dependency on ninja for running slideflow.

--- 2.3.1/foss-2023a-PyTorch-2.1.2-CUDA-12.1.1/Slideflow/slideflow-2.3.1/setup.py.orig	2024-06-11 08:37:58.380636699 +0200
+++ 2.3.1/foss-2023a-PyTorch-2.1.2-CUDA-12.1.1/Slideflow/slideflow-2.3.1/setup.py	2024-06-11 08:39:02.880974558 +0200
@@ -124,7 +124,6 @@
         'smac==1.4.0',
         'ConfigSpace',
         'pyarrow',
-        'ninja',
         'rich',
         'pillow>=6.0.0',
         'imgui>=2.0.0',
