From af79518add36401483d83429dfe45a2d2a5efcd8 Mon Sep 17 00:00:00 2001
From: Julian Morillo <julian.morillo@bsc.es>
Date: Thu, 17 Oct 2024 16:02:28 +0200
Subject: [PATCH] Add riscv64 to known architectures

---
 src/utk_platform_macros.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/utk_platform_macros.hpp b/src/utk_platform_macros.hpp
index 027b685..3662208 100644
--- a/src/utk_platform_macros.hpp
+++ b/src/utk_platform_macros.hpp
@@ -138,6 +138,8 @@
     #define UTK__Aarch_AARCH64
   #elif defined(__sh__)
     #define UTK__Arch_SuperH
+  #elif defined(__riscv)
+    #define UTK__Arch_RISCV64
   #else
     #error Architecture not supported.
   #endif
-- 
2.34.1
