diff -ruN tests.orig/quicktest tests/quicktest
--- tests.orig/quicktest        2021-05-31 16:00:15.905493851 +0200
+++ tests/quicktest     2021-05-31 16:03:02.301127053 +0200
@@ -10,7 +10,7 @@
     export LAUNCHER_DIR=../
 fi

-export LAUNCHER_JOB_FILE=commands
+export LAUNCHER_JOB_FILE=$LAUNCHER_DIR/extras/examples/helloworld_multi
 export LAUNCHER_SCHED=interleaved
 export LAUNCHER_WORKDIR=`pwd`

diff -ruN tests.orig/quicktest2 tests/quicktest2
--- tests.orig/quicktest2       2021-05-31 16:02:18.079348642 +0200
+++ tests/quicktest2    2021-05-31 16:02:48.965796851 +0200
@@ -10,7 +10,7 @@
     export LAUNCHER_DIR=../
 fi

-export LAUNCHER_JOB_FILE=$LAUNCHER_DIR/extras/examples/helloworld_short
+export LAUNCHER_JOB_FILE=$LAUNCHER_DIR/extras/examples/helloworld
 export LAUNCHER_SCHED=block
 export LAUNCHER_WORKDIR=`pwd`


