Dealing with changes of configparser in Python-3.12
Author: J. Saßmannshausen (Imperial College London/UK)
diff --git a/treetaggerwrapper-2.3.orig/treetaggerwrapper.py b/treetaggerwrapper-2.3/treetaggerwrapper.py
index 18236c3..7440f40 100644
--- a/treetaggerwrapper-2.3.orig/treetaggerwrapper.py
+++ b/treetaggerwrapper-2.3/treetaggerwrapper.py
@@ -541,7 +541,7 @@ ON_MACOSX = (platform.system() == "Darwin")
 ON_POSIX = (os.name == "posix")  # Care: true also under MACOSX.
 
 # Extra configuration storage within a config file.
-g_config = configparser.SafeConfigParser()
+g_config = configparser.RawConfigParser()
 
 # The config file is stored following XDG rules.
 CONFIG_FILENAME = "treetagger_wrapper.cfg"
