diff -Naur TauDEM-5.3.8.orig/Utilities/DEMVerifier/demverifier.c TauDEM-5.3.8/Utilities/DEMVerifier/demverifier.c
--- TauDEM-5.3.8.orig/Utilities/DEMVerifier/demverifier.c	2024-04-10 15:54:05.636830004 +0000
+++ TauDEM-5.3.8/Utilities/DEMVerifier/demverifier.c	2024-04-10 15:54:31.956956651 +0000
@@ -10,8 +10,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/time.h>
-#include <gdal/gdal.h>
-#include <gdal/cpl_conv.h>
+#include <gdal.h>
+#include <cpl_conv.h>
 #include <stdint.h>
 #include <stdlib.h>
 
