diff -ru meep-1.2.orig/libctl/Makefile.in meep-1.2/libctl/Makefile.in
--- meep-1.2.orig/libctl/Makefile.in	2012-07-20 19:45:13.000000000 +0200
+++ meep-1.2/libctl/Makefile.in	2012-10-31 18:48:30.455440437 +0100
@@ -342,7 +342,7 @@
 ##############################################################################
 
 # what is printed out when invoking your program with --version:
-VERSION_STRING = "Meep @VERSION@, Copyright (C) 2005-2012 Massachusetts Insitute of Technology."
+VERSION_STRING = "'Meep @VERSION@, Copyright (C) 2005-2012 Massachusetts Insitute of Technology.'"
 MY_DEFS = -DHAVE_CTL_HOOKS=1 -DHAVE_CTL_EXPORT_HOOK=1
 PROGRAM_NAME = meep
 SPECIFICATION_FILE = $(PROGRAM_NAME).scm
