--- meep-1.3/libctl/Makefile.in.orig	2015-03-31 17:47:53.000000000 +0200
+++ meep-1.3/libctl/Makefile.in	2015-04-08 13:23:54.206192661 +0200
@@ -369,7 +369,7 @@
 ##############################################################################
 
 # what is printed out when invoking your program with --version:
-VERSION_STRING = "Meep @VERSION@, Copyright (C) 2005-2015 Massachusetts Insitute of Technology."
+VERSION_STRING = "Meep_@VERSION@,_Copyright_(C)_2005-2015_Massachusetts_Insitute_of_Technology."
 MY_DEFS = -DHAVE_CTL_HOOKS=1 -DHAVE_CTL_EXPORT_HOOK=1
 PROGRAM_NAME = meep
 SPECIFICATION_FILE = $(PROGRAM_NAME).scm
