fix installation prefix for installing NGS Python bindings
author: Kenneth Hoste (HPC-UGent)
--- ngs-2.9.3/ngs-python/Makefile.python.orig	2019-02-12 09:33:55.145471383 +0100
+++ ngs-2.9.3/ngs-python/Makefile.python	2019-02-12 09:34:09.145729890 +0100
@@ -85,7 +85,7 @@
    
 else
 	@ echo "Installing ngs package..."
-	@ python setup.py -q install --user
+	@ python setup.py -q install --prefix $(INST_PYTHONDIR)
 endif
 
 ifneq (, $(NGS_LIBDIR))
