correctly specify prefix to install NGS-Python package to
author: Kenneth Hoste (HPC-UGent)
--- ngs-1.2.3/ngs-python/Makefile.python.orig	2016-03-31 12:34:54.124551782 +0200
+++ ngs-1.2.3/ngs-python/Makefile.python	2016-03-31 12:37:01.515699327 +0200
@@ -82,7 +82,7 @@
    
 else
 	@ echo "Installing ngs package..."
-	@ python setup.py -q install --user
+	@ python setup.py -q install --prefix $(INST_PYTHONDIR)
 endif
 
 ifneq (, $(NGS_LIBDIR))
