diff -Nru eclipse-orig/eclipse.ini eclipse/eclipse.ini
--- eclipse-orig/eclipse.ini	2023-12-01 22:22:06.710000000 +0100
+++ eclipse/eclipse.ini	2024-01-17 12:55:53.744683000 +0100
@@ -10,7 +10,7 @@
 openFile
 --launcher.appendVmargs
 -vm
-plugins/org.eclipse.justj.openjdk.hotspot.jre.full.linux.x86_64_17.0.9.v20231028-0858/jre/bin/java
+/work/imas/opt/EasyBuild/software/Java/17.0.6/bin/java
 -vmargs
 -Dorg.eclipse.ecf.provider.filetransfer.excludeContributors=org.eclipse.ecf.provider.filetransfer.httpclientjava
 -Dosgi.requiredJavaVersion=17
