pathlib is part of python standard since 3.4
--- setup.py.orig	2026-08-27 17:11:50.127131951 +0200
+++ setup.py	2026-08-27 17:11:57.457019474 +0200
@@ -45,7 +45,7 @@
     long_description=read("README.md"),
     long_description_content_type="text/markdown",
     packages=setuptools.find_packages(exclude=["tests"]),
-    install_requires=["ruamel.yaml", "pathlib"],
+    install_requires=["ruamel.yaml"],
     test_requires=["pytest", "flake8"],
     test_suite="tests",
     zip_safe=True,
