# Remove hard-coded optimisation and prefix
--- snp-dists-0.8.2/Makefile	2024-04-23 14:03:01.081536438 +0000
+++ snp-dists-0.8.2.new/Makefile	2024-04-23 14:02:58.806587939 +0000
@@ -1,10 +1,10 @@
 
 CC = gcc
-CFLAGS = -Wall -Wextra -Ofast -std=c99 -fopenmp
+CFLAGS = -Wall -Wextra -std=c99 -fopenmp
 LIBS = -lz -lm
 
 EXE = snp-dists
-PREFIX = /usr/local
+PREFIX = 
 TESTDIR = test
 
 .PHONY: all check clean format
@@ -16,7 +16,7 @@
 	$(CC) $(CFLAGS) -o $(EXE) $< $(LIBS)
 
 install: $(EXE)
-	install -v -t $(PREFIX)/bin $(EXE)
+	install -v -t $(PREFIX) $(EXE)
 
 clean:
 	$(RM) *~ *.o $(EXE)
