diff -ru sra-tools-3.0.0.orig/ngs/ngs-java/CMakeLists.txt sra-tools-3.0.0/ngs/ngs-java/CMakeLists.txt
--- sra-tools-3.0.0.orig/ngs/ngs-java/CMakeLists.txt	2022-05-27 19:55:41.525876987 +0000
+++ sra-tools-3.0.0/ngs/ngs-java/CMakeLists.txt	2022-05-27 19:56:11.210082556 +0000
@@ -133,7 +133,7 @@
                 ngs-doc-jar ALL COMMAND
                 bash -c "${Java_JAR_EXECUTABLE} -cfv ${CMAKE_JAR_OUTPUT_DIRECTORY}/ngs-doc.jar ."
                 DEPENDS ngs-doc_javadoc
-                WORKING_DIRECTORY "${TARGDIR}/obj/ngs/ngs-java/javadoc/ngs-doc"
+                WORKING_DIRECTORY "${TARGDIR}/ngs/ngs-java/javadoc/ngs-doc"
             )
             install( FILES ${CMAKE_JAR_OUTPUT_DIRECTORY}/ngs-doc.jar DESTINATION ${CMAKE_INSTALL_PREFIX}/jar/ )
 
