diff -ru afni_src.orig/thd_dset_to_vectim.c afni_src/thd_dset_to_vectim.c
--- afni_src.orig/thd_dset_to_vectim.c	2015-04-02 02:35:57.000000000 +0200
+++ afni_src/thd_dset_to_vectim.c	2015-04-03 17:08:33.340433024 +0200
@@ -74,8 +74,9 @@
 
    if( ignore > 0 ){  /* extract 1 at a time, save what we want */
 
+     float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
+     var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
      for( kk=iv=0 ; iv < nvox ; iv++ ){
        if( mmm[iv] == 0 ) continue ;
        (void)THD_extract_array( iv , dset , 0 , var ) ;
@@ -296,8 +297,9 @@
 
    if( nvals < DSET_NVALS(dset) ){ /* extract 1 at a time, save what we want */
 
+     float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(DSET_NVALS(dset))) ;
+     var = (float *)malloc(sizeof(float)*(DSET_NVALS(dset))) ;
      for( kk=iv=0 ; iv < nvox ; iv++ ){
        if( mmm[iv] == 0 ) continue ;
        (void)THD_extract_array( iv , dset , 0 , var ) ;
@@ -463,8 +465,9 @@
 
    if( ignore > 0 ){  /* extract 1 at a time, save what we want */
 
+    float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
+     var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
      mmmt = mmmv[0];
      for( kk=iv=0 ; iv < nvoxv[0] ; iv++ ){
        if( mmmt[iv] == 0 ) continue ;
