diff -ru afni_src.orig/thd_dset_to_vectim.c afni_src/thd_dset_to_vectim.c
--- afni_src.orig/thd_dset_to_vectim.c	2018-04-04 20:34:08.000000000 +0000
+++ afni_src/thd_dset_to_vectim.c	2018-04-10 18:52:19.237263493 +0000
@@ -79,8 +79,9 @@
 
    if( ignore > 0 ){  /* extract 1 at a time, save what we want */
 
+     float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
+     var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
 STATUS("ignore > 0 --> extracting one at a time") ;
      for( kk=iv=0 ; iv < nvox ; iv++ ){
        if( mmm[iv] == 0 ) continue ;
@@ -311,8 +312,9 @@
 
    if( nvals < DSET_NVALS(dset) ){ /* extract 1 at a time, save what we want */
 
+     float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(DSET_NVALS(dset))) ;
+     var = (float *)malloc(sizeof(float)*(DSET_NVALS(dset))) ;
      for( kk=iv=0 ; iv < nvox ; iv++ ){
        if( mmm[iv] == 0 ) continue ;
        (void)THD_extract_array( iv , dset , 0 , var ) ;
@@ -478,8 +480,9 @@
 
    if( ignore > 0 ){  /* extract 1 at a time, save what we want */
 
+    float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
+     var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
      mmmt = mmmv[0];
      for( kk=iv=0 ; iv < nvoxv[0] ; iv++ ){
        if( mmmt[iv] == 0 ) continue ;
Only in afni_src: thd_dset_to_vectim.c~
Only in afni_src: thd_dset_to_vectim.c.orig
Only in afni_src: thd_dset_to_vectim.c.rej
