fix for "error: this pragma must immediately precede a statement"
author: Kenneth Hoste (HPC-UGent)
--- src/thd_dset_to_vectim.c.orig	2018-05-01 05:07:33.000000000 +0200
+++ src/thd_dset_to_vectim.c	2018-05-02 16:50:20.962255151 +0200
@@ -79,8 +79,9 @@
 
    if( ignore > 0 ){  /* extract 1 at a time, save what we want */
 
+   float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
+     var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
 STATUS("ignore > 0 --> extracting one at a time") ;
      for( kk=iv=0 ; iv < nvox ; iv++ ){
        if( mmm[iv] == 0 ) continue ;
@@ -311,8 +312,9 @@
 
    if( nvals < DSET_NVALS(dset) ){ /* extract 1 at a time, save what we want */
 
+     float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(DSET_NVALS(dset))) ;
+     var = (float *)malloc(sizeof(float)*(DSET_NVALS(dset))) ;
      for( kk=iv=0 ; iv < nvox ; iv++ ){
        if( mmm[iv] == 0 ) continue ;
        (void)THD_extract_array( iv , dset , 0 , var ) ;
@@ -478,8 +480,9 @@
 
    if( ignore > 0 ){  /* extract 1 at a time, save what we want */
 
+    float *var;
 #pragma omp critical (MALLOC)
-     float *var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
+     var = (float *)malloc(sizeof(float)*(nvals+ignore)) ;
      mmmt = mmmv[0];
      for( kk=iv=0 ; iv < nvoxv[0] ; iv++ ){
        if( mmmt[iv] == 0 ) continue ;
