Add a setup.py

Author: Jakub Zárybnický <jakub@zarybnicky.com>

diff -ruN suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49/MANIFEST.in suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49-edit/MANIFEST.in
--- suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49/MANIFEST.in	1970-01-01 01:00:00.000000000 +0100
+++ suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49-edit/MANIFEST.in	2021-03-08 15:42:27.040090106 +0100
@@ -0,0 +1,2 @@
+graft static
+graft templates
diff -ruN suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49/setup.py suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49-edit/setup.py
--- suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49/setup.py	1970-01-01 01:00:00.000000000 +0100
+++ suave-45e4963d1a2c84cc0acdfaa603f29df10ff2ce49-edit/setup.py	2021-03-08 15:43:30.794938937 +0100
@@ -0,0 +1,20 @@
+from setuptools import setup, find_packages
+
+setup(
+    name='suave',
+    url='https://github.com/dellytools/suave',
+    packages=['.'],
+    install_requires=[
+        'Flask',
+        'click',
+        'pysam',
+        'h5py',
+    ],
+    include_package_data=True,
+    entry_points={
+        'console_scripts': [
+            'suave_server=suave_server:cli',
+            'suave_bam_to_h5=suave_bam_to_h5:cli',
+        ]
+    },
+)
