# Create position independent code
# S.D. Pinches
diff -Nru slatec4linux-orig/static/makefile slatec4linux/static/makefile
--- slatec4linux-orig/static/makefile	1999-12-22 00:32:54.000000000 +0100
+++ slatec4linux/static/makefile	2022-03-15 19:04:22.753929423 +0100
@@ -1,4 +1,4 @@
-FFLAGS= -O2
+FFLAGS= -O2 -fPIC
 OBJ = \
 aaaaaa.o  cnbdi.o   dcv.o     dqc25f.o  hpperm.o  qc25s.o       spoco.o \
 acosh.o   cnbfa.o   ddaini.o  dqc25s.o  hpsort.o  qcheb.o       spodi.o \
