--- deap-1.0.2.orig/setup.py	2015-05-12 04:06:49.000000000 +0200
+++ deap-1.0.2/setup.py	2015-05-21 19:06:03.792820042 +0200
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+import codecs
 import sys
 from distutils.core import setup
 try:
@@ -10,7 +11,7 @@
     from pypandoc import convert
 except ImportError:
     print("warning: pypandoc module not found, could not convert Markdown to RST")
-    read_md = lambda f: open(f, 'r').read()
+    read_md = lambda f: codecs.open(f, encoding='utf-8').read()
 else:
     read_md = lambda f: convert(f, 'rst')
 
