--- plugins/Make-arch.orig	2015-12-08 00:20:40.618420150 -0800
+++ plugins/Make-arch	2015-12-08 00:21:04.224451882 -0800
@@ -64,7 +64,7 @@
 	"CCFLAGS = -w -qinlglue -qarch=com -qtune=pwr3" \
 	"CXXFLAGS = -w -qstrict -Q -qarch=com -qtune=pwr3" \
 	"LDFLAGS = -qarch=com -qtune=pwr3" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf" \
 	"AR = ar" \
 	"NM = nm -B" \
@@ -82,7 +82,7 @@
 	"CCFLAGS = -w -qinlglue -qarch=com -qtune=pwr5" \
 	"CXXFLAGS = -w -qstrict -Q -qarch=com -qtune=pwr5" \
 	"LDFLAGS = -qarch=com -qtune=pwr5" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf" \
 	"AR = ar" \
 	"NM = nm -B" \
@@ -100,7 +100,7 @@
 	"CCFLAGS = -w -qinlglue -q64 -qarch=com -qtune=pwr5" \
 	"CXXFLAGS = -w -qstrict -Q -q64 -qarch=com -qtune=pwr5" \
 	"LDFLAGS = -q64 -qarch=com -qtune=pwr5" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf" \
 	"AR = ar -X64" \
 	"NM = nm -B" \
@@ -118,7 +118,7 @@
 	"CCFLAGS = -w -qinlglue -q64 -qarch=com -qtune=pwr5" \
 	"CXXFLAGS = -w -qstrict -Q -q64 -qarch=com -qtune=pwr5" \
 	"LDFLAGS = -q64 -qarch=com -qtune=pwr5" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf" \
 	"AR = ar -X64" \
 	"NM = nm -B" \
@@ -135,7 +135,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -O2 -fPIC -Wall -Wno-psabi -MMD -MP -fpic -ffunction-sections -funwind-tables -fstack-protector -D__ARM_ARCH_5__ -D__ARM_ARCH_5T__ -D__ARM_ARCH_5E__ -D__ARM_ARCH_5TE__ -march=armv7-a -msoft-float -mfpu=vfp -mthumb -Os -fomit-frame-pointer -fno-strict-aliasing -finline-limit=64 -Wa,--noexecstac $(ANDROIDINCS)" \
 	"CXXFLAGS = -O2 -fPIC -Wall -Wno-psabi -MMD -MP -fpic -ffunction-sections -funwind-tables -fstack-protector -D__ARM_ARCH_5__ -D__ARM_ARCH_5T__ -D__ARM_ARCH_5E__ -D__ARM_ARCH_5TE__ -march=armv7-a -msoft-float -mfpu=vfp -mthumb -Os -fomit-frame-pointer -fno-strict-aliasing -finline-limit=64 -Wa,--noexecstac $(ANDROIDINCS)" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"AR = ar" \
 	"NM = nm -p" \
 	"RANLIB = touch" \
@@ -169,7 +169,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -m64 -O2 -fPIC -Wall" \
 	"CXXFLAGS = -m64 -O2 -fPIC -Wall" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"AR = ar" \
 	"NM = nm -p" \
 	"RANLIB = touch" \
@@ -186,7 +186,7 @@
 	"CCFLAGS = -O2" \
 	"CXXFLAGS = -O2" \
 	"LDFLAGS = " \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf" \
 	"AR = ar" \
 	"NM = nm" \
@@ -204,7 +204,7 @@
 	"CCFLAGS = -O2 -fullwarn" \
 	"CXXFLAGS = -O2 -fullwarn" \
 	"LDFLAGS = -n32 -mips3" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf" \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -222,7 +222,7 @@
 	"CCFLAGS = -O2 -fullwarn" \
 	"CXXFLAGS = -O2 -fullwarn" \
 	"LDFLAGS = -64 -mips4" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf" \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -239,7 +239,7 @@
 	"DEF = -D" \
  	"CCFLAGS = -KPIC " \
  	"CXXFLAGS = -KPIC " \
-	"TCLLDFLAGS = -ltcl8.5 -lsocket -lnsl -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -lsocket -lnsl -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf -lsocket -lnsl -ldl" \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -256,7 +256,7 @@
 	"DEF = -D" \
  	"CCFLAGS = -KPIC " \
  	"CXXFLAGS = -KPIC " \
-	"TCLLDFLAGS = -ltcl8.5 -lsocket -lnsl -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -lsocket -lnsl -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf -lsocket -lnsl -ldl" \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -273,7 +273,7 @@
 	"DEF = -D" \
  	"CCFLAGS = -KPIC " \
  	"CXXFLAGS = -KPIC " \
-	"TCLLDFLAGS = -ltcl8.5 -lsocket -lnsl -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -lsocket -lnsl -ldl" \
 	"AR = ar" \
 	"NM = nm -p" \
 	"RANLIB = touch" \
@@ -289,7 +289,7 @@
 	"DEF = -D" \
  	"CCFLAGS = -KPIC " \
  	"CXXFLAGS = -KPIC " \
-	"TCLLDFLAGS = -ltcl8.5 -lsocket -lnsl -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -lsocket -lnsl -ldl" \
 	"AR = ar" \
 	"NM = nm -p" \
 	"RANLIB = touch" \
@@ -305,7 +305,7 @@
 	"DEF = -D" \
  	"CCFLAGS = -fast" \
  	"CXXFLAGS = -fast" \
-	"TCLLDFLAGS = -ltcl8.5 -lm" \
+	"TCLLDFLAGS = -ltcl8.6 -lm" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -B" \
@@ -320,7 +320,7 @@
 	"CC = gcc" \
 	"CXX = g++" \
 	"DEF = -D" \
-	"CCFLAGS = -O2 -Wall -I/usr/local/include/tcl8.5" \
+	"CCFLAGS = -O2 -Wall -I/usr/local/include/tcl8.6" \
 	"CXXFLAGS = -O2 -Wall" \
 	"TCLLDFLAGS = -L/usr/local/lib -ltcl85" \
 	"NETCDFLDFLAGS = -lnetcdf " \
@@ -337,7 +337,7 @@
 	"CC = gcc" \
 	"CXX = g++" \
 	"DEF = -D" \
-	"CCFLAGS = -m64 -O2 -Wall -I/usr/local/include/tcl8.5" \
+	"CCFLAGS = -m64 -O2 -Wall -I/usr/local/include/tcl8.6" \
 	"CXXFLAGS = -m64 -O2 -Wall" \
 	"TCLLDFLAGS = -L/usr/local/lib -ltcl85" \
 	"NETCDFLDFLAGS = -lnetcdf " \
@@ -356,7 +356,7 @@
 	"DEF = -D" \
 	"CCFLAGS = +O2 +DAportable -Ae +z" \
 	"CXXFLAGS = +O2 +DAportable +z" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -374,7 +374,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -m32 -O2 -fPIC -Wall" \
 	"CXXFLAGS = -m32 -O2 -fPIC -Wall" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -391,7 +391,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -O4 -arch host -tune host" \
 	"CXXFLAGS = -O4 -arch host -tune host" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -408,7 +408,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -m64 -O2 -fPIC -Wall" \
 	"CXXFLAGS = -m64 -O2 -fPIC -Wall" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -441,7 +441,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -O2 -fPIC -Wall" \
 	"CXXFLAGS = -O2 -fPIC -Wall" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -458,7 +458,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -m64 -O2 -fPIC -Wall" \
 	"CXXFLAGS = -m64 -O2 -fPIC -Wall" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"NETCDFLDFLAGS = " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -475,7 +475,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -O2 -fPIC -Wall" \
 	"CXXFLAGS = -O2 -fPIC -Wall" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -492,7 +492,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -O2 -fPIC" \
 	"CXXFLAGS = -O2 -fPIC" \
-	"TCLLDFLAGS = -ltcl8.5 -ldl" \
+	"TCLLDFLAGS = -ltcl8.6 -ldl" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
@@ -545,7 +545,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -m64 -Os -Wall -fPIC -dynamic" \
 	"CXXFLAGS = -m64 -Os -Wall -fPIC -dynamic" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"AR = ar" \
 	"NM = nm -p" \
 	"RANLIB = ranlib" \
@@ -562,7 +562,7 @@
 	"DEF = -D" \
 	"CCFLAGS = -O2 -Wall -ansi" \
 	"CXXFLAGS = -O2 -Wall" \
-	"TCLLDFLAGS = -ltcl8.5" \
+	"TCLLDFLAGS = -ltcl8.6" \
 	"NETCDFLDFLAGS = -lnetcdf " \
 	"AR = ar" \
 	"NM = nm -p" \
