diff -ru vmd-1.9.4a43.orig/configure vmd-1.9.4a43/configure
--- vmd-1.9.4a43.orig/configure	2020-10-26 18:48:05.001275411 -0000
+++ vmd-1.9.4a43/configure	2020-10-26 20:38:00.498580881 -0000
@@ -495,17 +495,15 @@
 $arch_ccpp        = "CC";
 $arch_nvcc        = "nvcc";
 $arch_nvccflags   = "-lineinfo --ptxas-options=-v " . 
-                    "-gencode arch=compute_30,code=compute_30 " .
-                    "-gencode arch=compute_30,code=sm_35 " .
-                    "-gencode arch=compute_30,code=sm_37 " .
-                    "-gencode arch=compute_50,code=compute_50 " .
+                    "-gencode arch=compute_35,code=sm_35 " .
+                    "-gencode arch=compute_37,code=sm_37 " .
                     "-gencode arch=compute_50,code=sm_50 " .
-                    "-gencode arch=compute_60,code=compute_60 " .
                     "-gencode arch=compute_60,code=sm_60 " .
-                    "-gencode arch=compute_70,code=compute_70 " .
                     "-gencode arch=compute_70,code=sm_70 " .
+                    "-gencode arch=compute_75,code=sm_75 " .
+                    "-gencode arch=compute_80,code=sm_80 " .
+                    "-gencode arch=compute_86,code=sm_86 " .
                     "--ftz=true ";
-#                    "-gencode arch=compute_75,code=sm_75 " .
 $arch_gcc         = "gcc";
 $arch_gccpp       = "g++";
 $arch_lex         = "lex";
@@ -2550,7 +2547,6 @@
 
   $arch_nvcc     = "/usr/local/cuda/bin/nvcc";
   $arch_nvccflags   = "--ptxas-options=-v " .
-                    "-gencode arch=compute_30,code=compute_30 " .
                     "-gencode arch=compute_70,code=compute_70 " .
                     "--ftz=true ";
   $cuda_library     = "-L/usr/local/cuda/lib64";
@@ -2574,11 +2570,8 @@
 
   $arch_nvcc     = "/usr/local/cuda-5.5/bin/nvcc";
   $arch_nvccflags   = "--ptxas-options=-v " . 
-                    "-gencode arch=compute_20,code=sm_20 " .
-                    "-gencode arch=compute_30,code=sm_30 " .
                     "-gencode arch=compute_35,code=sm_35 " .
                     "-gencode arch=compute_20,code=compute_20 " .
-                    "-gencode arch=compute_30,code=compute_30 " .
                     "--ftz=true ";
   $cuda_library     = "-L/usr/local/cuda/lib";
   $arch_cc       = "cc";
@@ -2912,7 +2905,6 @@
   
       # Only generate code for SM 7.0 on Summit
       $arch_nvccflags   = "--ptxas-options=-v " .
-                      "-gencode arch=compute_30,code=compute_30 " .
                       "-gencode arch=compute_70,code=sm_70 " .
                       "--ftz=true ";
       $arch_nvccflags   .= " --machine 64 -O3 $cuda_include";
@@ -3540,7 +3532,7 @@
 
 .cu.ptx:
 	\$(ECHO) "Compiling " \$< " --> " \$*.ptx " ..."; \\
-	\$(NVCC) \$(DEFINES) --use_fast_math $liboptix_include -gencode arch=compute_30,code=compute_30 -ptx \$< $arch_coptout$vmd_arch_dir/\$\@
+	\$(NVCC) \$(DEFINES) --use_fast_math $liboptix_include -gencode arch=compute_35,code=compute_35 -ptx \$< $arch_coptout$vmd_arch_dir/\$\@
 
 .y.o:
 
