Minor API changes when going to python 3.13
author: micketeer@gmail.com
diff -ru src.orig/PythonTextInterp.C src/PythonTextInterp.C
--- src.orig/PythonTextInterp.C	2026-06-09 15:12:54.300655905 +0200
+++ src/PythonTextInterp.C	2026-06-09 15:13:19.564270620 +0200
@@ -102,7 +102,7 @@
 
   for (int i=0; i<PyList_GET_SIZE(cblist); i++) {
     PyObject *obj = PyList_GET_ITEM(cblist, i);
-    PyObject *result = PyEval_CallObject(obj, arglist);
+    PyObject *result = PyObject_CallObject(obj, arglist);
     if (result == NULL) {
       PyErr_Print();
       PySequence_DelItem(cblist, i);
@@ -174,7 +174,6 @@
 
   // Do emit DeprecationWarnings
 #if PY_MAJOR_VERSION >= 3
-  PySys_AddWarnOption(L"default");
 #else
   PySys_AddWarnOption((char*) "default");
 #endif
