diff -ru vmd-1.9.4a43.orig/configure vmd-1.9.4a43/configure
--- vmd-1.9.4a43.orig/configure	2020-10-26 18:48:05.001275411 -0000
+++ vmd-1.9.4a43/configure	2020-10-26 20:16:05.497958595 -0000
@@ -495,17 +495,14 @@
 $arch_ccpp        = "CC";
 $arch_nvcc        = "nvcc";
 $arch_nvccflags   = "-lineinfo --ptxas-options=-v " . 
-                    "-gencode arch=compute_30,code=compute_30 " .
-                    "-gencode arch=compute_30,code=sm_35 " .
-                    "-gencode arch=compute_30,code=sm_37 " .
-                    "-gencode arch=compute_50,code=compute_50 " .
+                    "-gencode arch=compute_35,code=sm_35 " .
+                    "-gencode arch=compute_37,code=sm_37 " .
                     "-gencode arch=compute_50,code=sm_50 " .
-                    "-gencode arch=compute_60,code=compute_60 " .
                     "-gencode arch=compute_60,code=sm_60 " .
-                    "-gencode arch=compute_70,code=compute_70 " .
                     "-gencode arch=compute_70,code=sm_70 " .
+                    "-gencode arch=compute_75,code=sm_75 " .
+                    "-gencode arch=compute_80,code=sm_80 " .
                     "--ftz=true ";
-#                    "-gencode arch=compute_75,code=sm_75 " .
 $arch_gcc         = "gcc";
 $arch_gccpp       = "g++";
 $arch_lex         = "lex";
