Fix some bad printfs in surf.

Åke Sandgren, 2017-05-02
diff -ru vmd-1.9.3.orig/lib/surf/chull.c vmd-1.9.3/lib/surf/chull.c
--- vmd-1.9.3.orig/lib/surf/chull.c	1994-03-19 06:50:54.000000000 +0100
+++ vmd-1.9.3/lib/surf/chull.c	2017-05-02 13:44:07.046582827 +0200
@@ -378,7 +378,7 @@
 print_out( v )
 struct tvertex *v;
 {
-	fprintf( stderr, "\nAdding vertex %6x :\n", v );
+	fprintf( stderr, "\nAdding vertex %6p :\n", v );
 	print_verts();
 	print_edges();
 	print_fs();
@@ -398,11 +398,11 @@
 	temp = vertices;
 	fprintf (stderr, "Vertex List\n");
 	if (vertices) do {
-            fprintf(stderr,"  addr %6x\t", vertices );
+            fprintf(stderr,"  addr %6p\t", vertices );
             fprintf(stderr,"(%g,%g,%g)",vertices->v[X],
                     vertices->v[Y], vertices->v[Z] );
             fprintf(stderr,"   active:%3d", vertices->active );
-            fprintf(stderr,"   duplicate:%5x", vertices->duplicate );
+            fprintf(stderr,"   duplicate:%5p", vertices->duplicate );
             fprintf(stderr,"   mark:%2d\n", vertices->mark );
             vertices = vertices->next;
             } while ( vertices != temp );
@@ -424,13 +424,13 @@
 	temp = edges;
 	fprintf (stderr, "Edge List\n");
 	if (edges) do {
-            fprintf( stderr, "  addr: %6x\t", edges );
+            fprintf( stderr, "  addr: %6p\t", edges );
             fprintf( stderr, "adj: ");
             for (i=0; i<3; ++i) 
-                 fprintf( stderr, "%6x", edges->adjface[i] );
+                 fprintf( stderr, "%6p", edges->adjface[i] );
             fprintf( stderr, "  endpts:");
             for (i=0; i<2; ++i) 
-                 fprintf( stderr, "%8x", edges->endpts[i]);
+                 fprintf( stderr, "%8p", edges->endpts[i]);
             fprintf( stderr, "  del:%3d\n", edges->deleted );
             edges = edges->next; 
             } while (edges != temp );
@@ -452,13 +452,13 @@
 	temp = faces;
 	fprintf (stderr, "Face List\n");
 	if (faces) do {
-            fprintf(stderr, "  addr: %6x\t", faces );
+            fprintf(stderr, "  addr: %6p\t", faces );
             fprintf(stderr, "  edges:");
             for( i=0; i<3; ++i )
-                 fprintf(stderr, "%6x", faces->edg[i] );
+                 fprintf(stderr, "%6p", faces->edg[i] );
             fprintf(stderr, "  vert:");
             for ( i=0; i<3; ++i)
-                  fprintf(stderr, "%6x", faces->vert[i] );
+                  fprintf(stderr, "%6p", faces->vert[i] );
             fprintf(stderr, "  vis: %d\n", faces->visible );
             faces= faces->next;
             } while ( faces != temp );
@@ -552,8 +552,8 @@
                 temp_v = temp_v->next;
                 } while ( temp_v != vertices );
              do {                           
-                printf("3%5d%6d%6d\n", temp_f->vert[0]->vnum,
-                        temp_f->vert[1]->vnum, temp_f->vert[2]->vnum );
+                printf("3%5d%6d%6d\n", temp_f->vert[0]->vnum[0],
+                        temp_f->vert[1]->vnum[0], temp_f->vert[2]->vnum[0] );
                 temp_f = temp_f->next;
                 } while ( temp_f != faces );
 }
