diff -ru vmd-1.9.4a43.orig/configure vmd-1.9.4a43/configure
--- vmd-1.9.4a43.orig/configure	2020-10-26 16:14:26.085918685 -0000
+++ vmd-1.9.4a43/configure	2020-10-26 16:14:57.694229958 -0000
@@ -715,8 +715,8 @@
 ################ FLTK GUI
 $fltk_defines     = "-DVMDGUI -DVMDFLTK";
 $fltk_dir         = $ENV{'EBROOTFLTK'};
-$fltk_include     = "-I$fltk_dir/include";
-$fltk_library     = "-L$fltk_dir/$config_arch";
+$fltk_include     = "-I$fltk_dir/include/fltk";
+$fltk_library     = "-L$fltk_dir/lib/fltk";
 $fltk_libs        = "-lfltk -lX11 -lXfixes -lXcursor -lXinerama";
 #@fltk_cc          = ('forms_ui.c');
 @fltk_cu          = ();
