diff -ru vmd-1.9.3.orig/configure vmd-1.9.3/configure
--- vmd-1.9.3.orig/configure	2020-10-22 18:11:04.667535596 -0000
+++ vmd-1.9.3/configure	2020-10-22 18:11:33.977815552 -0000
@@ -667,8 +667,8 @@
 ################ FLTK GUI
 $fltk_defines     = "-DVMDGUI -DVMDFLTK";
 $fltk_dir         = $ENV{'EBROOTFLTK'};
-$fltk_include     = "-I$fltk_dir/include";
-$fltk_library     = "-L$fltk_dir/$config_arch";
+$fltk_include     = "-I$fltk_dir/include/fltk";
+$fltk_library     = "-L$fltk_dir/lib/fltk";
 $fltk_libs        = "-lfltk -lX11 -lXfixes -lXcursor -lXinerama";
 #@fltk_cc          = ('forms_ui.c');
 @fltk_cu          = ();
