Ensure compatability with newer version of GCC
diff -ruN vmd-1.9.4a57.orig/src/SelectionBuilder.C vmd-1.9.4a57/src/SelectionBuilder.C
--- vmd-1.9.4a57.orig/src/SelectionBuilder.C	2022-04-27 08:21:15.000000000 +0200
+++ vmd-1.9.4a57/src/SelectionBuilder.C	2025-10-08 14:15:01.312070096 +0200
@@ -182,7 +182,7 @@
   int line = self->macrobrowser->value();
   if (!line) {
-    self->macrooutput->value(NULL);
+    self->macrooutput->value(nullptr);
     return;
   }
   const char *name = self->macrobrowser->text(line);
