--- vmd-1.9.4a57/configure.orig	2024-01-09 14:34:21.934967638 +0000
+++ vmd-1.9.4a57/configure	2024-01-09 14:35:54.108764462 +0000
@@ -721,8 +721,8 @@
 ################ FLTK GUI
 $fltk_defines     = "-DVMDGUI -DVMDFLTK";
 $fltk_dir         = "$ENV{'EBROOTFLTK'}";
-$fltk_include     = "-I$fltk_dir/include";
-$fltk_library     = "-L$fltk_dir/$config_arch";
+$fltk_include     = "-I$fltk_dir/include/fltk";
+$fltk_library     = "-L$fltk_dir/lib64/fltk";
 $fltk_libs        = "-lfltk -lX11 -lXfixes -lXcursor -lXinerama";
 #@fltk_cc          = ('forms_ui.c');
 @fltk_cu          = ();
