Add riscv64 to known_arches

Julián Morillo, 2026-04-15

--- cmake/DyninstPlatform.cmake.ORIG	2026-04-15 10:00:32.506156521 +0200
+++ cmake/DyninstPlatform.cmake	2026-04-15 10:54:37.249591396 +0200
@@ -11,7 +11,7 @@
 # Equivalent to CMAKE_HOST_SYSTEM_PROCESSOR and `uname -m` on Unixes
 cmake_host_system_information(RESULT _host_arch QUERY OS_PLATFORM)
 
-set(_known_arches "x86_64" "ppc64le" "aarch64" "i386" "amd64")
+set(_known_arches "x86_64" "ppc64le" "aarch64" "i386" "amd64" "riscv64")
 if(NOT ${_host_arch} IN_LIST _known_arches)
   message(FATAL_ERROR "Unsupported architecture: '${_host_arch}'")
 endif()
