From 165402aa1640afb394b11a096b38c6bc3a5bf9d0 Mon Sep 17 00:00:00 2001
From: "Frank Ch. Eigler" <fche@elastic.org>
Date: Wed, 22 Jan 2025 15:35:55 -0500
Subject: [PATCH] Update sha1.C: #include <cstdint>

Subject header is needed on gcc15 to compile sha1.C, supplying declaration of types like uint32_t.
---
 common/src/sha1.C | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/src/sha1.C b/common/src/sha1.C
index 91e9ad2915..d6e76c5727 100644
--- a/common/src/sha1.C
+++ b/common/src/sha1.C
@@ -102,6 +102,7 @@ A million repetitions of "a"
 
 #include <stdio.h>
 #include <string.h>
+#include <cstdint>
 
 #include "dyntypes.h"
 #include "common/src/sha1.h"
