Define UC_PC(x) for riscv64

Julián Morillo, 2026-04-17

--- dyninstAPI_RT/src/RTlinux.c.ORIG	2026-04-17 10:58:28.452448100 +0200
+++ dyninstAPI_RT/src/RTlinux.c	2026-04-17 10:59:57.040426030 +0200
@@ -339,6 +339,8 @@
 	//#warning "UC_PC: in aarch64, pc is not directly accessable."
 	//aarch64 pc is not one of 31 GPRs, but an independent reg
 	#define UC_PC(x) x->uc_mcontext.pc
+#elif defined(arch_riscv64)
+  #define UC_PC(x) (x)->uc_mcontext.__pc
 #endif // UC_PC
 
 extern volatile unsigned long dyninstTrapTableUsed;
