diff -ru wannier_tools-2.5.1.orig/src/Makefile.gfortran-mpi wannier_tools-2.5.1/src/Makefile.gfortran-mpi
--- wannier_tools-2.5.1.orig/src/Makefile.gfortran-mpi	2020-03-10 19:15:21.000000000 +0000
+++ wannier_tools-2.5.1/src/Makefile.gfortran-mpi	2025-03-11 17:18:03.762735928 +0000
@@ -11,7 +11,7 @@
        main.o
 
 # compiler, here mpif90 should be comipled with gfortran
-F90  = mpif90 -cpp -DMPI -ffree-line-length-512                          
+F90  = mpif90 -cpp -DMPI -ffree-line-length-512 -fallow-argument-mismatch                         
 
 INCLUDE =
 WFLAG = #-Wall -Wextra -Warray-temporaries -Wconversion -fimplicit-none \
@@ -24,7 +24,7 @@
 
 
 # blas and lapack libraries
-LIBS = -llapack -lblas
+LIBS = -lflexiblas
 
 # Intel MKL also supports with gfortran comipler
 # dynamic linking
@@ -40,5 +40,9 @@
 .f90.o :
 	$(F90) $(FFLAG) $(INCLUDE) -c $*.f90
 
+install: wt.x
+	install -d $(PREFIX)/bin
+	install wt.x $(PREFIX)/bin
+
 clean :
 	rm -f *.o *.mod *~ wt.x
