source: https://github.com/apache/mxnet/pull/21093
diff -ruN mxnet-1.9.1_orig/cmake/ChooseBlas.cmake mxnet-1.9.1/cmake/ChooseBlas.cmake
--- mxnet-1.9.1_orig/cmake/ChooseBlas.cmake	2022-04-22 00:45:57.000000000 +0200
+++ mxnet-1.9.1/cmake/ChooseBlas.cmake	2023-07-10 12:13:14.448174000 +0200
@@ -16,10 +16,11 @@
 # under the License.
 
 set(BLAS "Open" CACHE STRING "Selected BLAS library")
-set_property(CACHE BLAS PROPERTY STRINGS "Atlas;Open;MKL")
+set_property(CACHE BLAS PROPERTY STRINGS "Atlas;Open;MKL;FlexiBLAS")
 
 if(DEFINED USE_BLAS)
   set(BLAS "${USE_BLAS}")
+  message(STATUS "BLAS is set to ${USE_BLAS}")
 else()
   if(USE_MKL_IF_AVAILABLE)
     if(NOT MKL_FOUND)
@@ -31,7 +32,14 @@
   endif()
 endif()
 
-if(BLAS STREQUAL "Atlas" OR BLAS STREQUAL "atlas")
+if(BLAS STREQUAL "FlexiBLAS" OR BLAS STREQUAL "flexiblas")
+  set(BLA_VENDOR FlexiBLAS)
+  find_package(BLAS REQUIRED)
+  include_directories(SYSTEM ${BLAS_INCLUDE_DIR})
+  add_definitions(-DMSHADOW_USE_CBLAS=1)
+  add_definitions(-DMSHADOW_USE_MKL=0)
+  add_definitions(-DMXNET_USE_LAPACK=1)
+elseif(BLAS STREQUAL "Atlas" OR BLAS STREQUAL "atlas")
   find_package(Atlas REQUIRED)
   include_directories(SYSTEM ${Atlas_INCLUDE_DIR})
   list(APPEND mshadow_LINKER_LIBS ${Atlas_LIBRARIES})
