diff -ru BeadArrayFiles-1.3.1.original/module/GenotypeCalls.py BeadArrayFiles-1.3.1.patched/module/GenotypeCalls.py
--- BeadArrayFiles-1.3.1.original/module/GenotypeCalls.py	2017-11-16 21:40:52.000000000 +0100
+++ BeadArrayFiles-1.3.1.patched/module/GenotypeCalls.py	2018-07-03 14:19:00.000000000 +0200
@@ -1,6 +1,6 @@
 import struct
 from numpy import cos, sin, pi, arctan2, float32, uint16, int32, seterr, frombuffer, dtype
-from .BeadArrayUtility import read_int, read_string, read_byte, read_float, complement
+from .BeadArrayUtility import read_int, read_string, read_byte, read_float, read_char, complement
 from .BeadPoolManifest import RefStrand, SourceStrand
 
 seterr(divide='ignore', invalid='ignore')
