diff -ru ensembl-tools-release-90.org/scripts/variant_effect_predictor/INSTALL.pl ensembl-tools-release-90.patched/scripts/variant_effect_predictor/INSTALL.pl
--- ensembl-tools-release-90.org/scripts/variant_effect_predictor/INSTALL.pl	2016-09-22 12:20:25.000000000 +0200
+++ ensembl-tools-release-90.patched/scripts/variant_effect_predictor/INSTALL.pl	2016-12-07 13:33:19.000000000 +0100
@@ -99,18 +99,21 @@
   if( $this_os eq 'darwin' && !$NO_HTSLIB) {
     print "3. added $DEST_DIR/htslib to your DYLD_LIBRARY_PATH environment variable?\n";
   }
-  print "(y/n)";
-
-  my $ok = <>;
-  if($ok !~ /^y/i) {
-    print "Exiting. Please \n";
-    print "1. add $DEST_DIR to your PERL5LIB environment variable\n";
-    print "2. add $DEST_DIR/htslib to your PATH environment variable\n";
-    if( $this_os eq 'darwin' && !$NO_HTSLIB) {
-      print "3. add $DEST_DIR/htslib to your DYLD_LIBRARY_PATH environment variable\n";
-    }
-    exit(0);
-  }
+#
+# Disable this check for non-interactive deploy with EasyBuild: we know what we are doing ;).
+# 
+#  print "(y/n)";
+#
+#  my $ok = <>;
+#  if($ok !~ /^y/i) {
+#    print "Exiting. Please \n";
+#    print "1. add $DEST_DIR to your PERL5LIB environment variable\n";
+#    print "2. add $DEST_DIR/htslib to your PATH environment variable\n";
+#    if( $this_os eq 'darwin' && !$NO_HTSLIB) {
+#      print "3. add $DEST_DIR/htslib to your DYLD_LIBRARY_PATH environment variable\n";
+#    }
+#    exit(0);
+#  }
   if( ! -d $DEST_DIR ) {
     mkdir $DEST_DIR || die "Could not make destination directory $DEST_DIR"
   }
