diff -ru ensembl-vep-release-90.5.org/INSTALL.pl ensembl-vep-release-90.5.patched/INSTALL.pl
--- ensembl-vep-release-90.5.org/INSTALL.pl	2017-09-14 15:23:25.000000000 +0200
+++ ensembl-vep-release-90.5.patched/INSTALL.pl	2017-09-17 18:24:49.000000000 +0200
@@ -378,18 +378,21 @@
     if( $this_os eq 'darwin' && !$NO_HTSLIB) {
       print "3. added $DEST_DIR/htslib to your DYLD_LIBRARY_PATH environment variable?\n";
     }
-    print "(y/n): ";
-
-    my $ok = <>;
-    if($ok !~ /^y/i) {
-      print "Exiting. Please \n";
-      print "1. add $DEST_DIR to your PERL5LIB environment variable\n";
-      print "2. add $DEST_DIR/htslib to your PATH environment variable\n";
-      if( $this_os eq 'darwin' && !$NO_HTSLIB) {
-        print "3. add $DEST_DIR/htslib to your DYLD_LIBRARY_PATH environment variable\n";
-      }
-      exit(0);
-    }
+    #
+    # Disable this check for non-interactive deploy with EasyBuild: we know what we are doing ;).
+    #print "(y/n): ";
+    #
+    #my $ok = <>;
+    #if($ok !~ /^y/i) {
+    #  print "Exiting. Please \n";
+    #  print "1. add $DEST_DIR to your PERL5LIB environment variable\n";
+    #  print "2. add $DEST_DIR/htslib to your PATH environment variable\n";
+    #  if( $this_os eq 'darwin' && !$NO_HTSLIB) {
+    #    print "3. add $DEST_DIR/htslib to your DYLD_LIBRARY_PATH environment variable\n";
+    #  }
+    #  exit(0);
+    #}
+    print "(y/n): Automagic deploy with EasyBuild said yes.";
     if( ! -d $DEST_DIR ) {
       mkdir $DEST_DIR || die "Could not make destination directory $DEST_DIR"
     }
