Fix linking of examples to PnetCDF
author: Alex Domingo (Vrije Universiteit Brussel)
--- examples/f03/CMakeLists.txt.orig	2026-08-13 15:42:38.535863097 +0200
+++ examples/f03/CMakeLists.txt	2026-08-13 15:43:35.059792858 +0200
@@ -29,8 +29,8 @@
 # Ensure NetCDF Fortran and C libraries are linked
 set(SRC examplePio.F90)
 add_executable(examplePio_f90 ${SRC})
-target_link_libraries(examplePio_f90 piof pioc ${TIMING_LINK_LIB})
+target_link_libraries(examplePio_f90 piof pioc pnetcdf ${TIMING_LINK_LIB})
 
 set(SRC exampleAsyncPio.F90)
 add_executable(exampleAsyncPio_f90 ${SRC})
-target_link_libraries(exampleAsyncPio_f90 piof pioc ${TIMING_LINK_LIB})
\ No newline at end of file
+target_link_libraries(exampleAsyncPio_f90 piof pioc pnetcdf ${TIMING_LINK_LIB})
--- examples/c/CMakeLists.txt.orig	2026-08-13 15:51:14.135383992 +0200
+++ examples/c/CMakeLists.txt	2026-08-13 15:50:44.758686579 +0200
@@ -29,19 +29,19 @@
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O0")
 
 add_executable(examplePio EXCLUDE_FROM_ALL examplePio.c)
-target_link_libraries(examplePio pioc)
+target_link_libraries(examplePio pioc pnetcdf)
 add_dependencies(tests examplePio)
 
 add_executable(piorcw piorcw.c)
-target_link_libraries(piorcw pioc)
+target_link_libraries(piorcw pioc pnetcdf)
 add_dependencies(tests piorcw)
 
 add_executable(example1 example1.c)
-target_link_libraries(example1 pioc)
+target_link_libraries(example1 pioc pnetcdf)
 add_dependencies(tests example1)
 
 add_executable(darray_no_async darray_no_async.c)
-target_link_libraries(darray_no_async pioc)
+target_link_libraries(darray_no_async pioc pnetcdf)
 add_dependencies(tests darray_no_async)
 
 # add_executable(darray_async darray_async.c)
