Do not start perl script in background
Author: Samuel Moors, Vrije Universiteit Brussel (VUB)
diff -ruN Annocript-84fd67b30b7edb0eedc4eafd0a3fad3a117055d9_orig/annocript.pl Annocript-84fd67b30b7edb0eedc4eafd0a3fad3a117055d9/annocript.pl
--- Annocript-84fd67b30b7edb0eedc4eafd0a3fad3a117055d9_orig/annocript.pl	2023-05-15 18:02:36.827824452 +0200
+++ Annocript-84fd67b30b7edb0eedc4eafd0a3fad3a117055d9/annocript.pl	2023-05-15 18:13:45.525029000 +0200
@@ -301,7 +301,7 @@
 		  ."\n\nPlease let us know if any problem occurs during the computation or you don't get from Annocript the expected result.\n"
 		  ."You may want to use the forum at: https://groups.google.com/forum/#!forum/annocript.\n",$log_file);#DEBUGCODE	
 
-	nice_printing("STARTING ANNOCRIPT IN BACKGROUND! BYE!",$log_file);
+	nice_printing("STARTING ANNOCRIPT IN FOREGROUND! BYE!",$log_file);
 
 	close ( $logFHandle );
 
@@ -325,7 +325,7 @@
 	chdir $program_folder;
 	
 	#This perl script will be run in background
-	my $runAnnocriptCmd = "nohup perl ".$configHash->{'ProgExecFolder'} ."/annocript_executor.pl $user_config $program_config $timesFile $configHashPath >> $log_file &";
+	my $runAnnocriptCmd = "perl ".$configHash->{'ProgExecFolder'} ."/annocript_executor.pl $user_config $program_config $timesFile $configHashPath >> $log_file";
 	( system($runAnnocriptCmd) ) == 0 
 	  or die("Unable to start annocript_executor!\n");
 	}
