Fix incorrect includes for gsl

Åke Sandgren, 20191029
diff -ru eman2-2.3.orig/libEM/emdata_transform.cpp eman2-2.3/libEM/emdata_transform.cpp
--- eman2-2.3.orig/libEM/emdata_transform.cpp	2019-04-25 21:03:56.000000000 +0200
+++ eman2-2.3/libEM/emdata_transform.cpp	2019-10-28 08:58:46.079788287 +0100
@@ -35,8 +35,8 @@
 #include <cstring>
 #include <cstdio>
 
-#include  "gsl/gsl_sf_result.h"
-#include  "gsl/gsl_sf_bessel.h"
+#include <gsl/gsl_sf_result.h>
+#include <gsl/gsl_sf_bessel.h>
 #include <iostream>
 #include <algorithm>
 #include <vector>
