diff -ru eman2-2.2.orig/libEM/emdata_transform.cpp eman2-2.2/libEM/emdata_transform.cpp
--- eman2-2.2.orig/libEM/emdata_transform.cpp	2017-06-01 00:08:09.000000000 +0200
+++ eman2-2.2/libEM/emdata_transform.cpp	2018-05-03 20:11:27.279288754 +0200
@@ -39,8 +39,8 @@
 #include <cstring>
 #include <cstdio>
 
-#include  "gsl_sf_result.h"
-#include  "gsl_sf_bessel.h"
+#include <gsl/gsl_sf_result.h>
+#include <gsl/gsl_sf_bessel.h>
 #include <iostream>
 #include <algorithm>
 #include <vector>
diff -ru eman2-2.2.orig/libEM/transform.cpp eman2-2.2/libEM/transform.cpp
--- eman2-2.2.orig/libEM/transform.cpp	2017-06-01 00:08:09.000000000 +0200
+++ eman2-2.2/libEM/transform.cpp	2018-05-03 20:07:18.152316248 +0200
@@ -49,9 +49,9 @@
 
 #include <algorithm> // for std::transform
 
-#include <gsl_matrix.h>
-#include <gsl_blas.h>
-#include <gsl_linalg.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_blas.h>
+#include <gsl/gsl_linalg.h>
 
 #include <ostream>
 using std::ostream_iterator;
