diff -ru eman2-2.2.orig/CMakeLists.txt eman2-2.2/CMakeLists.txt
--- eman2-2.2.orig/CMakeLists.txt	2017-06-01 00:08:09.000000000 +0200
+++ eman2-2.2/CMakeLists.txt	2018-05-04 15:01:18.693898481 +0200
@@ -50,9 +50,10 @@
 		file(WRITE ${EMAN_PY_INSTALL_PREFIX}/eman2dir_relative_path_to_sp_dir "../../Library/")
 	endif()
 else()
-	SET(EMAN_INSTALL_PREFIX $ENV{HOME}/EMAN2 CACHE PATH "installation prefix")
-	set(EMAN_PY_INSTALL_PREFIX     ${EMAN_INSTALL_PREFIX}/lib)
-	file(WRITE ${EMAN_PY_INSTALL_PREFIX}/eman2dir_relative_path_to_sp_dir "../")
+	SET(EMAN_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX} CACHE PATH "installation prefix")
+	set(EMAN_PY_INSTALL_PREFIX ${EMAN_INSTALL_PREFIX}/lib/python2.7/site-packages)
+	file(WRITE eman2dir_relative_path_to_sp_dir "../../../")
+	INSTALL(FILES eman2dir_relative_path_to_sp_dir DESTINATION ${EMAN_PY_INSTALL_PREFIX})
 endif()
 
 SET(CMAKE_INSTALL_PREFIX ${EMAN_INSTALL_PREFIX} CACHE INTERNAL "installation prefix")
