Fix issues with installation in custom directory
Author: Sam Moors, Vrije Universiteit Brussel (VUB)
diff -ur eman2-2.21a.orig/CMakeLists.txt eman2-2.21a/CMakeLists.txt
--- eman2-2.21a.orig/CMakeLists.txt	2018-02-14 15:51:44.000000000 +0100
+++ eman2-2.21a/CMakeLists.txt	2018-06-04 15:20:24.895107057 +0200
@@ -99,7 +99,10 @@
 endif()
 
 if(SP_DIR)
-	file(WRITE ${SP_DIR}/eman2dir_relative_path_to_sp_dir ${eman2dir_relative_path})
+	file(WRITE ${CMAKE_SOURCE_DIR}/eman2dir_relative_path_to_sp_dir ${eman2dir_relative_path})
+	install(FILES ${CMAKE_SOURCE_DIR}/eman2dir_relative_path_to_sp_dir
+			DESTINATION ${SP_DIR}
+			)
 endif()
 
 set(CMAKE_INSTALL_RPATH ${SP_DIR})
diff -ur eman2-2.21a.orig/programs/CMakeLists.txt eman2-2.21a/programs/CMakeLists.txt
--- eman2-2.21a.orig/programs/CMakeLists.txt	2018-02-14 15:51:44.000000000 +0100
+++ eman2-2.21a/programs/CMakeLists.txt	2018-06-04 15:19:59.211077890 +0200
@@ -17,7 +17,6 @@
 		)
 ENDIF(WIN32)
 
-find_program(GIT_EXECUTABLE git)
 if(GIT_EXECUTABLE)
 	execute_process(COMMAND ${GIT_EXECUTABLE} describe --always --dirty
 					WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
@@ -33,5 +32,9 @@
 
 string(TIMESTAMP EMAN_TIMESTAMP "%Y-%m-%d %H:%M")
 configure_file(${CMAKE_SOURCE_DIR}/libpyEM/EMAN2_meta.py.in
-				${SP_DIR}/EMAN2_meta.py
+				${CMAKE_SOURCE_DIR}/libpyEM/EMAN2_meta.py
 				)
+install(FILES ${CMAKE_SOURCE_DIR}/libpyEM/EMAN2_meta.py
+		DESTINATION ${SP_DIR}
+		COMPONENT PythonFiles
+		)
