diff --color -Naur hipSOLVER-rocm-6.4.1.orig/clients/CMakeLists.txt hipSOLVER-rocm-6.4.1/clients/CMakeLists.txt
--- hipSOLVER-rocm-6.4.1.orig/clients/CMakeLists.txt	2025-03-06 20:28:37.000000000 +0100
+++ hipSOLVER-rocm-6.4.1/clients/CMakeLists.txt	2025-06-22 20:54:35.165614276 +0200
@@ -51,9 +51,9 @@
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 if(HIPSOLVER_FIND_PACKAGE_LAPACK_CONFIG)
-  find_package(LAPACK 3.7 REQUIRED CONFIG)
+  find_package(LAPACK REQUIRED CONFIG $ENV{EBROOTFLEXIBLAS})
 else()
-  find_package(LAPACK 3.7 REQUIRED)
+  find_package(LAPACK REQUIRED PATHS $ENV{EBROOTFLEXIBLAS})
 endif()
 
 if(NOT LAPACK_LIBRARIES)
