--- qiime-1.8.0/scripts/print_qiime_config.py.orig	2013-12-12 03:53:03.000000000 +0100
+++ qiime-1.8.0/scripts/print_qiime_config.py	2014-08-27 21:18:30.659487219 +0200
@@ -349,11 +349,11 @@
         self.assertTrue(app_path('FastTree'),
          "FastTree not found. This may or may not be a problem depending on "+\
          "which components of QIIME you plan to use.")
-        command = "FastTree 2>&1 > %s | grep version" % devnull
+        command = "FastTree"
         proc = Popen(command,shell=True,universal_newlines=True,\
-                         stdout=PIPE,stderr=STDOUT)
-        stdout = proc.stdout.read()
-        version_string = stdout.strip().split(' ')[4].strip()
+                         stdout=PIPE,stderr=PIPE)
+        (stdout, stderr) = proc.communicate()
+        version_string = stderr.split('\n')[0].strip().split(' ')[2].strip()
         try:
             version = tuple(map(int,version_string.split('.')))
             pass_test = version == acceptable_version
