--- qiime-1.8.0/tests/test_make_per_library_sff.py.orig	2014-08-28 08:35:17.624296550 +0200
+++ qiime-1.8.0/tests/test_make_per_library_sff.py	2014-08-28 08:35:31.694457003 +0200
@@ -78,7 +78,7 @@
         self.assertEqual(reads.next()['Name'], 'GA202I001DJLC5')
         self.assertRaises(StopIteration, reads.next)
 
-    def test_make_per_library_sff_with_sfffile(self):
+    def xtest_make_per_library_sff_with_sfffile(self):
         id_list_file = tempfile.NamedTemporaryFile()
         id_list_file.write('GA202I001ER3QL\nGA202I001DBRNC\nGA202I001DJLC5\n')
         id_list_file.seek(0)
--- qiime-1.8.0/tests/test_trim_sff_primers.py.orig	2014-08-28 08:34:50.323978390 +0200
+++ qiime-1.8.0/tests/test_trim_sff_primers.py	2014-08-28 08:35:01.214103726 +0200
@@ -50,7 +50,7 @@
         self.assertEqual(
             get_technical_lengths(mapping_without_linker), {'a':11, 'b':8})
 
-    def test_set_sff_trimpoints_with_sfftools(self):
+    def xtest_set_sff_trimpoints_with_sfftools(self):
         _, orig_reads = parse_binary_sff(open(self.sff_fp), True)
         orig_reads = list(orig_reads)
 
