diff -ru qiime-1.9.1.orig/setup.py qiime-1.9.1/setup.py
--- qiime-1.9.1.orig/setup.py	2015-05-26 21:28:24.000000000 +0200
+++ qiime-1.9.1/setup.py	2016-05-17 17:39:03.699286509 +0200
@@ -362,11 +362,11 @@
 # invoked
 if all([e not in sys.argv for e in 'egg_info', 'sdist', 'register']):
     catch_install_errors(build_denoiser, 'denoiser')
-    catch_install_errors(download_UCLUST, 'UCLUST')
-    catch_install_errors(build_FastTree, 'FastTree')
-    catch_install_errors(build_SortMeRNA, 'SortMeRNA')
-    catch_install_errors(build_SUMACLUST, 'SUMACLUST')
-    catch_install_errors(build_swarm, 'swarm')
+    #catch_install_errors(download_UCLUST, 'UCLUST')
+    #catch_install_errors(build_FastTree, 'FastTree')
+    #catch_install_errors(build_SortMeRNA, 'SortMeRNA')
+    #catch_install_errors(build_SUMACLUST, 'SUMACLUST')
+    #catch_install_errors(build_swarm, 'swarm')
 
 classes = """Development Status :: 5 - Production/Stable
     Environment :: Console
