diff -Naur frealign_v9.11_orig/src/Makefile_linux_amd64_intel_mp frealign_v9.11/src/Makefile_linux_amd64_intel_mp
--- frealign_v9.11_orig/src/Makefile_linux_amd64_intel_mp	2018-04-28 16:32:17.450039910 +0000
+++ frealign_v9.11/src/Makefile_linux_amd64_intel_mp	2018-04-28 16:33:36.977283623 +0000
@@ -9,21 +9,21 @@
 SHELL  = /bin/sh
 
 ifeq ($(MKLROOT),"")
-  FFLAGS = -w -c -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -openmp -D_OPENMP
+  FFLAGS = -w -c -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -qopenmp -D_OPENMP
 else
-  FFLAGS = -w -c -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -openmp -D_OPENMP -mkl
+  FFLAGS = -w -c -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -qopenmp -D_OPENMP -mkl
 endif
 
-CFLAGS = -c -w -O3 -fpp -openmp -D_OPENMP
+CFLAGS = -c -w -O3 -fpp -qopenmp -D_OPENMP
 
 COMP   = ifort
 
 CC     = icc
 
 ifeq ($(MKLROOT),"")
-  LF     = -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -openmp
+  LF     = -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -qopenmp
 else
-  LF     = -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -openmp -mkl
+  LF     = -O3 -fpp -assume realloc_lhs -heap-arrays -gen-interfaces -qopenmp -mkl
 endif
 
 PGM    = frealign_v9
