diff -ru frealign_v9.07.orig/src/Makefile.inc frealign_v9.07/src/Makefile.inc
--- frealign_v9.07.orig/src/Makefile.inc	2014-02-20 17:15:10.000000000 +0100
+++ frealign_v9.07/src/Makefile.inc	2018-05-03 14:19:29.415455417 +0200
@@ -1,110 +1,131 @@
 # 
 # ------ Lists source files for use by Makefile  ------
 #  
-ELEMENTS =  \
- $(LIB)(a3d3.o) \
- $(LIB)(a3d3_s.o) \
- $(LIB)(ainterpo3ds.o) \
- $(LIB)(ainterpo3dbig.o) \
- $(LIB)(applycref.o) \
- $(LIB)(beautify.o) \
- $(LIB)(beautify_s.o) \
- $(LIB)(bfactor.o) \
- $(LIB)(boxft_lut.o) \
- $(LIB)(calcfx.o) \
- $(LIB)(calcsinc.o) \
- $(LIB)(card10.o) \
- $(LIB)(cards11and12.o) \
- $(LIB)(cards13and14.o) \
- $(LIB)(cards15to18.o) \
- $(LIB)(card1.o) \
- $(LIB)(card2.o) \
- $(LIB)(card3.o) \
- $(LIB)(card4.o) \
- $(LIB)(card5.o) \
- $(LIB)(card6.o) \
- $(LIB)(card7.o) \
- $(LIB)(cards8and9.o) \
- $(LIB)(ccoef.o) \
- $(LIB)(cc3.o) \
- $(LIB)(cc3m.o) \
- $(LIB)(cc3_c.o) \
- $(LIB)(cc3m_c.o) \
- $(LIB)(ccp.o) \
- $(LIB)(checksym.o) \
- $(LIB)(ctfapply.o) \
- $(LIB)(ctfapply_phase_only.o) \
- $(LIB)(ctf.o) \
- $(LIB)(ctfrefine.o) \
- $(LIB)(combine_arrays.o) \
- $(LIB)(correct3d.o) \
- $(LIB)(correct3d_c.o) \
- $(LIB)(d2mask.o) \
- $(LIB)(d3mask.o) \
- $(LIB)(erc.o) \
- $(LIB)(ewaldex.o) \
- $(LIB)(ewaldin.o) \
- $(LIB)(fd.o) \
- $(LIB)(fsh.o) \
- $(LIB)(fangle.o) \
- $(LIB)(find_fpart.o) \
- $(LIB)(find_rlim.o) \
- $(LIB)(fpart_cc.o) \
- $(LIB)(getsymmat.o) \
- $(LIB)(hextend.o) \
- $(LIB)(iof.o) \
- $(LIB)(limitsymm.o) \
- $(LIB)(magrefine.o) \
- $(LIB)(lmain.o) \
- $(LIB)(maskcos.o) \
- $(LIB)(maskcos2.o) \
- $(LIB)(maskcos_c.o) \
- $(LIB)(maskcos2_c.o) \
- $(LIB)(maskenv.o) \
- $(LIB)(mask3d.o) \
- $(LIB)(mask3d_c.o) \
- $(LIB)(match.o) \
- $(LIB)(matmul.o) \
- $(LIB)(matmul_t.o) \
- $(LIB)(opmaps.o) \
- $(LIB)(opmaps2.o) \
- $(LIB)(opresstathalf.o) \
- $(LIB)(opresstatmaps.o) \
- $(LIB)(pad.o) \
- $(LIB)(pda_fft.o) \
- $(LIB)(pdiff.o) \
- $(LIB)(pextract.o) \
- $(LIB)(pinc.o) \
- $(LIB)(pinsert.o) \
- $(LIB)(pinsert_c.o) \
- $(LIB)(pinsert_nearest.o) \
- $(LIB)(pinsert_trilin.o) \
- $(LIB)(pinsert_nearestb.o) \
- $(LIB)(pinsert_trilinb.o) \
- $(LIB)(pinsert_nearest_se.o) \
- $(LIB)(pinsert_trilin_se.o) \
- $(LIB)(pinsert_nearestb_se.o) \
- $(LIB)(pinsert_trilinb_se.o) \
- $(LIB)(prefine.o) \
- $(LIB)(presb.o) \
- $(LIB)(psearch.o) \
- $(LIB)(pweight.o) \
- $(LIB)(random.o) \
- $(LIB)(rlft3.o) \
- $(LIB)(rotate2d.o) \
- $(LIB)(rotmat.o) \
- $(LIB)(searchang.o) \
- $(LIB)(sheltest.o) \
- $(LIB)(shift2D.o) \
- $(LIB)(shift.o) \
- $(LIB)(shiftvol.o) \
- $(LIB)(sigma.o) \
- $(LIB)(slen2.o) \
- $(LIB)(stamp.o) \
- $(LIB)(storeshift.o) \
- $(LIB)(trilinmap.o) \
- $(LIB)(va04.o) \
- $(LIB)(variance.o) \
- $(LIB)(volmeasure.o) \
- $(LIB)(window.o)
+OBJS =  \
+ a3d3.o \
+ a3d3_s.o \
+ ainterpo3ds.o \
+ ainterpo3dbig.o \
+ applycref.o \
+ beautify.o \
+ beautify_s.o \
+ bfactor.o \
+ boxft_lut.o \
+ calcfx.o \
+ calcsinc.o \
+ card10.o \
+ cards11and12.o \
+ cards13and14.o \
+ cards15to18.o \
+ card1.o \
+ card2.o \
+ card3.o \
+ card4.o \
+ card5.o \
+ card6.o \
+ card7.o \
+ cards8and9.o \
+ ccoef.o \
+ cc3.o \
+ cc3m.o \
+ cc3_c.o \
+ cc3m_c.o \
+ ccp.o \
+ checksym.o \
+ ctfapply.o \
+ ctfapply_phase_only.o \
+ ctf.o \
+ ctfrefine.o \
+ combine_arrays.o \
+ correct3d.o \
+ correct3d_c.o \
+ d2mask.o \
+ d3mask.o \
+ erc.o \
+ ewaldex.o \
+ ewaldin.o \
+ fd.o \
+ fsh.o \
+ fangle.o \
+ find_fpart.o \
+ find_rlim.o \
+ fpart_cc.o \
+ getsymmat.o \
+ hextend.o \
+ iof.o \
+ limitsymm.o \
+ magrefine.o \
+ lmain.o \
+ maskcos.o \
+ maskcos2.o \
+ maskcos_c.o \
+ maskcos2_c.o \
+ maskenv.o \
+ mask3d.o \
+ mask3d_c.o \
+ match.o \
+ matmul.o \
+ matmul_t.o \
+ opmaps.o \
+ opmaps2.o \
+ opresstathalf.o \
+ opresstatmaps.o \
+ pad.o \
+ pda_fft.o \
+ pdiff.o \
+ pextract.o \
+ pinc.o \
+ pinsert.o \
+ pinsert_c.o \
+ pinsert_nearest.o \
+ pinsert_trilin.o \
+ pinsert_nearestb.o \
+ pinsert_trilinb.o \
+ pinsert_nearest_se.o \
+ pinsert_trilin_se.o \
+ pinsert_nearestb_se.o \
+ pinsert_trilinb_se.o \
+ prefine.o \
+ presb.o \
+ psearch.o \
+ pweight.o \
+ random.o \
+ rlft3.o \
+ rotate2d.o \
+ rotmat.o \
+ searchang.o \
+ sheltest.o \
+ shift2D.o \
+ shift.o \
+ shiftvol.o \
+ sigma.o \
+ slen2.o \
+ stamp.o \
+ storeshift.o \
+ trilinmap.o \
+ va04.o \
+ variance.o \
+ volmeasure.o \
+ window.o
 
+PGMS = $(PGM) set_polarity calc_occ rsample 
+
+all: $(PGMS)
+
+$(PGM) : $(PGM).o $(LIB)
+	$(FC) $(PGM).o $(LDFLAGS) $(FFLAGS) $(LIB) $(LIBS) -o ../bin/$(PGM).exe
+
+set_polarity:	set_polarity.o
+	$(FC) set_polarity.o $(LDFLAGS) $(FFLAGS) -o ../bin/set_polarity.exe
+
+calc_occ:	calc_occ.o
+	$(FC) calc_occ.o $(LDFLAGS) $(FFLAGS) -o ../bin/calc_occ.exe
+
+rsample:	rsample.o
+	$(FC) rsample.o $(LDFLAGS) $(FFLAGS) -o ../bin/rsample.exe
+
+$(LIB) : ioc.o $(OBJS)
+	$(AR) cr $(LIB) $(OBJS) ioc.o
+
+clean:
+	-rm -f $(OBJS) $(OBJS:.o=.mod)
diff -ru frealign_v9.07.orig/src/Makefile_linux_amd64_gnu frealign_v9.07/src/Makefile_linux_amd64_gnu
--- frealign_v9.07.orig/src/Makefile_linux_amd64_gnu	2014-01-16 19:49:18.000000000 +0100
+++ frealign_v9.07/src/Makefile_linux_amd64_gnu	2018-05-03 14:16:04.795561722 +0200
@@ -11,19 +11,7 @@
 
 SHELL  = /bin/sh
 
-FFLAGS = -m64 -O3 -s -fomit-frame-pointer -Wall -falign-functions=4 \
-	 -funroll-loops -fexpensive-optimizations -fschedule-insns2 \
-	 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
-	 -cpp -w -c
-
-CFLAGS = -m64 -O3 -DPROTOTYPE -D_FILE_OFFSET_BITS=64 \
-	 -D_LARGEFILE_SOURCE -cpp -w -c
-
-COMP   = gfortran
-
-CC     = gcc
-
-LF     = -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+LIBS     = $(LIBFFT)
 
 PGM    = frealign_v9
 
@@ -33,26 +21,8 @@
 
 include Makefile.inc
 
-$(PGM) : $(PGM).o $(LIB)
-	@echo linking $(PGM)
-	$(COMP) $(LF) $(PGM).o $(LIB) libfftw3f.a -o ../bin/$(PGM).exe
-	@\rm $(PGM).o
-
-$(PGM).o  : $(PGM).f
-	$(COMP) $(FFLAGS) -o $(PGM).o  $(PGM).f
-	$(COMP) -o ../bin/set_polarity.exe set_polarity.f
-	$(COMP) -o ../bin/calc_occ.exe calc_occ.f
-	$(COMP) -o ../bin/rsample.exe rsample.f
-
-ioc.o : ioc.c
-	$(CC) $(CFLAGS) -o ioc.o ioc.c
-	$(AR) r $(LIB) ioc.o
-	@\rm ioc.o
-
-$(LIB) : ioc.o $(ELEMENTS)
-	@echo all object files placed in library
-
-.f.a:
-	$(COMP) $(FFLAGS) $<
-	$(AR) r $(LIB) $*.o
-	@\rm $*.o
+%.o : %.f
+	$(FC) $(FFLAGS) -ffixed-line-length-none -cpp -c $<
+
+%.o : %.f90
+	$(FC) $(FFLAGS) -ffree-line-length-none -cpp -c $<
