Fix broken tests in configure.ac

Åke Sandgren, 20200205
diff -ru ntHits-ntHits-v0.0.1.orig/configure.ac ntHits-ntHits-v0.0.1/configure.ac
--- ntHits-ntHits-v0.0.1.orig/configure.ac	2019-02-22 00:41:34.000000000 +0100
+++ ntHits-ntHits-v0.0.1/configure.ac	2020-02-05 14:17:08.682168844 +0100
@@ -6,11 +6,11 @@
 AC_CONFIG_SRCDIR([nthits.cpp])
 AC_CONFIG_HEADER([config.h])
 
-if test -z $CXXFLAGS; then
+if test -z "$CXXFLAGS"; then
     CXXFLAGS='-O3'
 fi
 
-if test -z $CCFLAGS; then
+if test -z "$CCFLAGS"; then
     CCFLAGS='-O3'
 fi
 
@@ -70,7 +70,7 @@
 # Check for OpenMP.
 AC_LANG_PUSH([C++])
 AC_OPENMP
-if test -z $OPENMP_CXXFLAGS; then
+if test -z "$OPENMP_CXXFLAGS"; then
 	AC_MSG_ERROR([NTCARD must be compiled with a C++ compiler that supports OpenMP threading.])
 fi
 AC_LANG_POP([C++])
