--- Configure/arch/make.inc_afteramble.orig	2011-11-21 23:54:01.000000000 -0500
+++ Configure/arch/make.inc_afteramble	2018-06-04 14:14:39.575980219 -0400
@@ -9,7 +9,7 @@
 #--------------------------------------------------------------------------
 #  CAT Preprocessing Flags
 #--------------------------------------------------------------------------
-           CPPARGS =  $(CPPFLAGS) $(DEF_FLAGS) $(FLAG_1) $(FLAG_2) \
+           CPPARGS =  $(CPPFLAGS) $(FLAG_1) $(FLAG_2) \
    $(FLAG_3) $(FLAG_4) $(FLAG_5) $(FLAG_6)  \
    $(FLAG_7) $(FLAG_8) $(FLAG_9) $(FLAG_10) \
    $(FLAG_11) $(FLAG_12) $(FLAG_13) $(FLAG_14) \
