--- Configure/arch/Linux_x86_64_cc_gfortran.orig	2011-10-26 05:31:10.000000000 -0400
+++ Configure/arch/Linux_x86_64_cc_gfortran	2018-06-01 16:02:38.453867000 -0400
@@ -2,8 +2,9 @@
 #  gfortran defs
 #--------------------------------------------------------------------------
          CPP      = /usr/bin/cpp
-         COMPILER = -DGFORTRAN
-         FC       = gfortran  -O3
+         COMPILER = -DGFORTRAN -Df2cFortran
+         FC       = gfortran
+         CC       = cc
          DEBFLGS  =
          OPT      =
          CLIB     =

