--- Configure/arch/make.inc_preamble.orig	2011-11-21 23:53:45.000000000 -0500
+++ Configure/arch/make.inc_preamble	2018-06-04 14:11:22.327983752 -0400
@@ -58,7 +58,7 @@
             INSTALLDIR    = $(TOPDIR)/libs/install
 #===========================================================================
 # PREPROCESSOR OPTIONS FOR CPP
-            DEF_FLAGS     = -P -C -traditional 
+            DEF_FLAGS     = -P -traditional 
 #===========================================================================
 #
 ####################### MEDM ENVIRONMENT #####################################
@@ -93,7 +93,7 @@
 #                           SET PATH TO LIBRARIES WITH IOLIBS      
 #                           SET PATH TO INCLUDE FILES (netcdf.mod) WITH IOINCS
 #--------------------------------------------------------------------------
-             IOLIBS       =  -lnetcdf  #-L/hosts/mao/usr/medm/install/netcdf/3.6.3/em64t/lib -lnetcdf
+             IOLIBS       =  -L$(EBROOTNETCDF)/lib64 -lnetcdff  #-L/hosts/mao/usr/medm/install/netcdf/3.6.3/em64t/lib -lnetcdf
              IOINCS       =  #-I/hosts/mao/usr/medm/install/netcdf/3.6.3/em64t/include
 #             IOLIBS       =  -L/usr/local/install/netcdf/gcc_ifort/3.6.2/lib  -lnetcdf
 #             IOINCS       =  -I/usr/local/install/netcdf/gcc_ifort/3.6.2/include
@@ -147,7 +147,7 @@
 #                           UNCOMMENT TO INCLUDE MPI PARALLEL CAPABILITY
 #--------------------------------------------------------------------------
 
-             FLAG_4 = -DMULTIPROCESSOR
+#             FLAG_4 = -DMULTIPROCESSOR
              PARLIB = -lmetis  #-L/usr/local/lib -lmetis
 
 #--------------------------------------------------------------------------
