# unimportantly strict dependency version requirements
diff -ruN "CRISPResso2-2.1.2 copy/setup.py" CRISPResso2-2.1.2/setup.py
--- "CRISPResso2-2.1.2 copy/setup.py"	2021-06-28 15:30:43.197061000 +0200
+++ CRISPResso2-2.1.2/setup.py	2021-06-28 15:36:03.124070000 +0200
@@ -89,12 +89,12 @@
               'Programming Language :: Cython',
               ],
           install_requires=[
-              'pandas>=0.15,<=0.24',
-              'matplotlib>=1.3.1,<=2.2.3',
+              'pandas>=0.15',
+              'matplotlib>=1.3.1',
               'argparse>=1.3,<=1.4',
     	      'seaborn>0.7.1,<0.10',
-              'jinja2==2.10',
-    	      'scipy==1.1.0',
+              'jinja2>=2.10',
+    	      'scipy>=1.1.0',
               'numpy>=1.9,<=1.16.6',
               'kiwisolver<1.2',
               'pyparsing<3',
