Add install instructions

Åke Sandgren, 2026-01-27
diff -ru STARE-1.0.5.orig/Code/CMakeLists.txt STARE-1.0.5/Code/CMakeLists.txt
--- STARE-1.0.5.orig/Code/CMakeLists.txt	2025-03-31 16:21:03.000000000 +0200
+++ STARE-1.0.5/Code/CMakeLists.txt	2026-01-27 09:26:36.000000000 +0100
@@ -14,19 +14,25 @@
 
 add_executable( ReplaceInvalidChars ReplaceInvalidChars.cpp )
 target_link_libraries( ReplaceInvalidChars PUBLIC STARELib )
+install(TARGETS ReplaceInvalidChars DESTINATION bin)
 
 add_executable( Promoter_Windows Promoter_Windows.cpp )
 target_link_libraries( Promoter_Windows PUBLIC STARELib )
+install(TARGETS Promoter_Windows DESTINATION bin)
 
 add_executable( PSCM_to_PSEM PSCM_to_PSEM.cpp )
+install(TARGETS PSCM_to_PSEM DESTINATION bin)
 
 add_executable( TRAPmulti TRAPmulti.cpp )
+install(TARGETS TRAPmulti DESTINATION bin)
 
 add_executable( STARE_ABCpp STARE_ABCpp.cpp )
 target_link_libraries( STARE_ABCpp PUBLIC STARELib )
+install(TARGETS STARE_ABCpp DESTINATION bin)
 
 add_executable( TF_Gene_Scorer_Reshape TF_Gene_Scorer_Reshape.cpp )
 target_link_libraries( TF_Gene_Scorer_Reshape PUBLIC STARELib )
+install(TARGETS TF_Gene_Scorer_Reshape DESTINATION bin)
 
 find_package(Llvm)
 find_package(OpenMP REQUIRED)
@@ -37,7 +43,7 @@
 endif()
 find_package(Boost REQUIRED)
 
-
+install(PROGRAMS Juicebox_KR_normalization.sh Juicebox_normalization.sh STARE.sh DESTINATION bin)
 
 
 
