Use Flexiblas for foss/2021a toolchain

Åke Sandgren, 20211007
diff -ru dalton.orig/cmake/math/MathLibs.cmake dalton/cmake/math/MathLibs.cmake
--- dalton.orig/cmake/math/MathLibs.cmake	2021-10-05 14:33:14.000000000 +0200
+++ dalton/cmake/math/MathLibs.cmake	2021-10-07 19:19:10.571066231 +0200
@@ -74,19 +74,18 @@
 
 #-------------------------------------------------------------------------------
 # OPENBLAS
-# no lapack in OPENBLAS, set OPENBLAS_LAPACK_* to ATLAS and SYSTEM-NATIVE values
 
 set(OPENBLAS_BLAS_INCLUDE_PATH_SUFFIXES)
-set(OPENBLAS_LAPACK_INCLUDE_PATH_SUFFIXES atlas)
+set(OPENBLAS_LAPACK_INCLUDE_PATH_SUFFIXES)
 
-set(OPENBLAS_BLAS_HEADERS   cblas_openblas.h)
-set(OPENBLAS_LAPACK_HEADERS clapack.h)
+set(OPENBLAS_BLAS_HEADERS   cblas.h)
+set(OPENBLAS_LAPACK_HEADERS lapack.h)
 
 set(OPENBLAS_BLAS_LIBRARY_PATH_SUFFIXES)
-set(OPENBLAS_LAPACK_LIBRARY_PATH_SUFFIXES atlas atlas-base atlas-base/atlas atlas-sse3)
+set(OPENBLAS_LAPACK_LIBRARY_PATH_SUFFIXES)
 
-set(OPENBLAS_BLAS_LIBS   openblas)
-set(OPENBLAS_LAPACK_LIBS atlas lapack)
+set(OPENBLAS_BLAS_LIBS   flexiblas)
+set(OPENBLAS_LAPACK_LIBS flexiblas)
 
 #-------------------------------------------------------------------------------
 # MKL
