diff -ru dalton.orig/external/pelib/CMakeLists.txt dalton/external/pelib/CMakeLists.txt
--- dalton.orig/external/pelib/CMakeLists.txt	2021-03-23 18:56:25.683203719 +0000
+++ dalton/external/pelib/CMakeLists.txt	2021-03-23 18:56:56.652390761 +0000
@@ -82,7 +82,7 @@
     if(ENABLE_COVERAGE)
         message(FATAL_ERROR "Code coverage compilation requires GCC compiler.")
     endif()
-    set(CMAKE_Fortran_FLAGS_RELEASE "${CMAKE_Fortran_FLAGS_RELEASE} -O3 -ip -xHost")
+    set(CMAKE_Fortran_FLAGS_RELEASE "${CMAKE_Fortran_FLAGS_RELEASE} -O3 -ip ")
     set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} -traceback -check all -debug all -ftrapuv -warn all")
 elseif(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -ffree-line-length-none")
