From aff76dc42b9a939fb4b5eca9077d822fac046a7c Mon Sep 17 00:00:00 2001
From: Simon Pinches <Simon.Pinches@iter.org>
Date: Thu, 22 May 2025 13:37:56 +0200
Subject: [PATCH] Update license expression format in pyproject.toml

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index d584f0f..fcca57e 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,7 +8,7 @@ name = "imas-data-dictionary"
 description ="The Data Dictionary is the implementation of the Data Model of ITER's Integrated Modelling & Analysis Suite (IMAS)"
 authors = [{name = "ITER Organization", email = "imas-support@iter.org"}]
 readme = {file = "README.md", content-type = "text/markdown"}
-license = "GPL-3.0-or-later AND LGPL-3.0-or-later"
+license = {file = "LICENSE-LGPL"}
 keywords = ["IMAS", "Data Dictionary", "IDS"]
 classifiers = [
     "Development Status :: 4 - Beta",
-- 
2.39.3

