From fc8428e2ee6eab3dc8cd0d146e2c3a45d7b9f5cc Mon Sep 17 00:00:00 2001
From: prasad-sawantdesai <prasad.sawantdesai@gmail.com>
Date: Fri, 30 May 2025 18:50:55 +0200
Subject: [PATCH] fixed locating IDSdef.xml

---
 data_dictionary/idsinfo.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/data_dictionary/idsinfo.py b/data_dictionary/idsinfo.py
index 10290f0..05d3148 100644
--- a/data_dictionary/idsinfo.py
+++ b/data_dictionary/idsinfo.py
@@ -64,12 +64,19 @@ class IDSInfo:
         self.idsdef_path = ""
         self.legacy_doc_path = ""
         self.sphinx_doc_path = ""
+        current_fpath = os.path.dirname(os.path.realpath(__file__))
         # Check idsdef.xml is installed in the Python environment (system as well as local)
         if not self.idsdef_path:
+            _idsdef_path_newer = os.path.join(current_fpath, r"../../../../")
+            _idsdef_path_old = os.path.join(current_fpath, r"../../../")
             local_path = os.path.join(str(Path.home()), ".local")
             python_env_list = [sys.prefix]
             if os.path.exists(local_path):
                 python_env_list.append(local_path)
+            if os.path.isfile(_idsdef_path_newer):
+                python_env_list.append(_idsdef_path_newer)
+            if os.path.isfile(_idsdef_path_old):
+                python_env_list.append(_idsdef_path_old)
             reg_compile = re.compile("dd_*")
             version_list = None
             python_env_path = ""
@@ -93,7 +100,6 @@ class IDSInfo:
 
         # Search through higher level directories
         if not self.idsdef_path:
-            current_fpath = os.path.dirname(os.path.realpath(__file__))
             # Newer approach : IMAS/<VERSION>/lib/python3.8/site-packages/data_dictionary/idsinfo.py
             _idsdef_path = os.path.join(current_fpath, r"../../../../include/IDSDef.xml")
             if os.path.isfile(_idsdef_path):
-- 
2.39.3

