From ae97f5ef4d67284b42418750a95f581b2cac4133 Mon Sep 17 00:00:00 2001
From: prasad-sawantdesai <prasad.sawantdesai@gmail.com>
Date: Thu, 5 Jun 2025 16:34:56 +0200
Subject: [PATCH] removed conditions and provided absolute path

---
 data_dictionary/idsinfo.py | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/data_dictionary/idsinfo.py b/data_dictionary/idsinfo.py
index 05d3148..d491391 100644
--- a/data_dictionary/idsinfo.py
+++ b/data_dictionary/idsinfo.py
@@ -67,16 +67,14 @@ class IDSInfo:
         current_fpath = os.path.dirname(os.path.realpath(__file__))
         # Check idsdef.xml is installed in the Python environment (system as well as local)
         if not self.idsdef_path:
-            _idsdef_path_newer = os.path.join(current_fpath, r"../../../../")
-            _idsdef_path_old = os.path.join(current_fpath, r"../../../")
+            _idsdef_path_newer = os.path.abspath(os.path.normpath(os.path.join(current_fpath, r"../../../../")))
+            _idsdef_path_old = os.path.abspath(os.path.normpath(os.path.join(current_fpath, r"../../../")))
             local_path = os.path.join(str(Path.home()), ".local")
             python_env_list = [sys.prefix]
             if os.path.exists(local_path):
                 python_env_list.append(local_path)
-            if os.path.isfile(_idsdef_path_newer):
-                python_env_list.append(_idsdef_path_newer)
-            if os.path.isfile(_idsdef_path_old):
-                python_env_list.append(_idsdef_path_old)
+            python_env_list.append(_idsdef_path_newer)
+            python_env_list.append(_idsdef_path_old)
             reg_compile = re.compile("dd_*")
             version_list = None
             python_env_path = ""
-- 
2.39.3

