--- src/irfinder/ReadBlockProcessor_CoverageBlocks.cpp.orig	2019-06-13 21:21:00.911329206 +0000
+++ src/irfinder/ReadBlockProcessor_CoverageBlocks.cpp	2019-06-13 21:17:29.140471391 +0000
@@ -1,6 +1,6 @@
 #include "ReadBlockProcessor_CoverageBlocks.h"
 #include "includedefine.h"
-
+#include <cmath>
 void CoverageBlocks::loadRef(std::istream &IN) {
 	std::string myLine;
 	std::string myField;
@@ -231,7 +231,7 @@
 			}
 		}
 	}
-	return NAN;
+	return nan("");
 }
 
 double CoverageBlocks::trimmedMeanFromHist(const std::map<uint,uint> &hist, uint centerPercent) const {
