Custom settings for JupyterLab on VUB notebook platform
author: Alex Domingo (Vrije Universiteit Brussel)
diff -Nru etc/jupyter.orig/jupyter_lab_config.py etc/jupyter/jupyter_lab_config.py
--- a/etc/jupyter/jupyter_lab_config.py	1970-01-01 01:00:00.000000000 +0100
+++ b/etc/jupyter/jupyter_lab_config.py	2024-06-04 11:38:56.413751456 +0200
@@ -0,0 +1,11 @@
+# Configuration file for lab.
+
+c = get_config()  #noqa
+
+#--------------------------------------------------
+# VUB-HPC: notebook platform settings
+#--------------------------------------------------
+# use read-only extension manager in the Lab
+c.LabApp.extension_manager = "readonly"
+# never check for updates of the Lab
+c.LabApp.check_for_updates_class="jupyterlab.NeverCheckForUpdate"
