--- setup-mpi.py.orig	2010-05-18 11:06:15.918335000 +0200
+++ setup-mpi.py	2010-05-18 11:20:35.308879000 +0200
@@ -12,21 +12,21 @@
 #User can specify a directory for the libraries with -L : If no such parameter is specified, then /usr/lib is used
 optlist, a = getopt.getopt(sys.argv[2:], 'I:L:',['all', 'root=', 'record-rpm=', 'prefix=','help'])
 
-includeDir = "/usr/include"
-libDir = "/usr/lib"
+includeDir = ["/usr/include"]
+libDir = ["/usr/lib"]
 removeArg = -1
 removeArg2 = -1
 
 for opt in optlist:
     if (opt[0]=="-I"):
-        includeDir = opt[1]
+        includeDir = opt[1].split(',')
     c = 0
     for arg in sys.argv:
         if (arg[0:2]=="-I"):
             removeArg = c+2  # +2 because we omitted the 1st and 2nd argument in the beginning!
             break
     if (opt[0]=="-L"):
-        libDir = opt[1]
+        libDir = opt[1].split(',')
         c = 0
     for arg in sys.argv:
             if (arg[0:2]=="-L"):
@@ -42,9 +42,9 @@
 
 
 meep_swig = Extension ('_meep_mpi', ['meep_mpi.i'], 
-        swig_opts=['-c++','-w503,389,302,362,314,509,451','-I'+includeDir],
+        swig_opts=['-c++','-w503,389,302,362,314,509,451']+['-I%s'%x for x in includeDir],
         libraries = ['meep_mpi'],
-        library_dirs=[libDir],
+        library_dirs=libDir,
 )
 
 setup (name = 'python-meep',
@@ -56,5 +56,5 @@
        ext_modules = [meep_swig],
        data_files=[(sysconfig.get_python_lib(1,0,""),['custom.hpp'])],
        py_modules = ["meep_mpi"],
-       include_dirs=[includeDir])
+       include_dirs=includeDir)
 
