diff --git a/configure.ac b/configure.ac
index b010b3d..527c6ab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -62,16 +62,16 @@ AC_CHECK_LIB([arpack], [main],
              [], [AC_MSG_ERROR([could not find the arpack library])])
 
 AC_F77_FUNC(sgemm)
-AC_CHECK_LIB([openblas], [$sgemm],
-             [], [AC_MSG_ERROR([could not find the openblas library])])
+AC_CHECK_LIB([flexiblas], [$sgemm],
+             [], [AC_MSG_ERROR([could not find the flexiblas library])])
 
 # Check that the Armadillo runtime library is installed if we need it
 AS_IF([test "x$static_compile" = "xfalse" && test "x$arma_wrapper" = "xtrue"],
       [AC_CHECK_LIB([armadillo], [main],
                     [], AC_MSG_ERROR([could not find the library armadillo]))])
 
-# Add openblas and openmp libraries
-LIBS="-lopenblas $LIBS $OMP_LIBS"
+# Add flexiblas and openmp libraries
+LIBS="-lflexiblas $LIBS $OMP_LIBS"
 
 # Set config headers and makefiles
 AC_CONFIG_HEADERS([config.h])
