diff -ru bwa-mem2-2.2.1.orig/src/utils.h bwa-mem2-2.2.1/src/utils.h
--- bwa-mem2-2.2.1.orig/src/utils.h	2023-12-06 20:19:27.235896780 +0000
+++ bwa-mem2-2.2.1/src/utils.h	2023-12-06 20:22:12.451342264 +0000
@@ -48,7 +48,7 @@
 
 #define xassert(cond, msg) if ((cond) == 0) _err_fatal_simple_core(__func__, msg)
 
-#if defined(__GNUC__) && !defined(__clang__)
+#if defined(__GNUC__) && __GNUC__ < 11 && !defined(__clang__)
 #if defined(__i386__)
 static inline unsigned long long __rdtsc(void)
 {
