--- automake-1.15/bin/automake.in.orig	2015-01-06 03:25:55.000000000 +0800
+++ automake-1.15/bin/automake.in	2017-07-26 13:58:07.086205701 +0800
@@ -3878,7 +3878,7 @@
 sub substitute_ac_subst_variables
 {
   my ($text) = @_;
-  $text =~ s/\${([^ \t=:+{}]+)}/substitute_ac_subst_variables_worker ($1)/ge;
+  $text =~ s/\$\{([^ \t=:+{}]+)\}/substitute_ac_subst_variables_worker ($1)/ge;
   return $text;
 }
