Relax tolerance of tensor assertions in test_message_passing
author: Alex Domingo (Vrije Universiteit Brussel)
--- a/test/nn/conv/test_message_passing.py	2025-12-21 01:25:14.832351000 +0100
+++ b/test/nn/conv/test_message_passing.py	2025-12-21 01:25:44.626720767 +0100
@@ -300,7 +300,7 @@
     assert torch.allclose(conv((x1, x2), edge_index, value, (4, 2)), out1)
     if torch_geometric.typing.WITH_TORCH_SPARSE:
         assert torch.allclose(conv((x1, x2), adj.t()), out1)
-        assert torch.allclose(conv((x1, None), adj.t()), out2)
+        assert torch.allclose(conv((x1, None), adj.t()), out2, atol=1e-4)
 
 
 class MyMultipleAggrConv(MessagePassing):
