diff -ru Pasha-1.0.3.orig/google/dense_hash_map Pasha-1.0.3/google/dense_hash_map
--- Pasha-1.0.3.orig/google/dense_hash_map	2010-12-07 04:43:26.000000000 +0100
+++ Pasha-1.0.3/google/dense_hash_map	2012-08-02 10:15:22.758689520 +0200
@@ -103,7 +103,7 @@
 #include <functional>                // for equal_to
 #include <memory>                    // for alloc<>
 #include <utility>                   // for pair<>
-#include HASH_FUN_H                  // defined in config.h
+#include HASH_MAP_H                  // defined in config.h
 #include <google/sparsehash/libc_allocator_with_realloc.h>
 #include <google/sparsehash/densehashtable.h>
 
diff -ru Pasha-1.0.3.orig/google/dense_hash_set Pasha-1.0.3/google/dense_hash_set
--- Pasha-1.0.3.orig/google/dense_hash_set	2010-12-07 04:43:26.000000000 +0100
+++ Pasha-1.0.3/google/dense_hash_set	2012-08-02 10:15:12.680591773 +0200
@@ -107,7 +107,7 @@
 #include <functional>                // for equal_to
 #include <memory>                    // for alloc<>
 #include <utility>                   // for pair<>
-#include HASH_FUN_H                  // defined in config.h
+#include HASH_SET_H                  // defined in config.h
 #include <google/sparsehash/libc_allocator_with_realloc.h>
 #include <google/sparsehash/densehashtable.h>
 
diff -ru Pasha-1.0.3.orig/google/sparsehash/sparseconfig.h Pasha-1.0.3/google/sparsehash/sparseconfig.h
--- Pasha-1.0.3.orig/google/sparsehash/sparseconfig.h	2010-12-07 04:43:26.000000000 +0100
+++ Pasha-1.0.3/google/sparsehash/sparseconfig.h	2012-08-02 10:15:34.118799712 +0200
@@ -7,7 +7,8 @@
 #define GOOGLE_NAMESPACE ::google
 
 /* the location of the header defining hash functions */
-#define HASH_FUN_H <ext/hash_fun.h>
+#define HASH_MAP_H <ext/hash_map>
+#define HASH_SET_H <ext/hash_set>
 
 /* the namespace of the hash<> function */
 #define HASH_NAMESPACE __gnu_cxx
diff -ru Pasha-1.0.3.orig/google/sparse_hash_map Pasha-1.0.3/google/sparse_hash_map
--- Pasha-1.0.3.orig/google/sparse_hash_map	2010-12-07 04:43:26.000000000 +0100
+++ Pasha-1.0.3/google/sparse_hash_map	2012-08-02 10:15:26.615727009 +0200
@@ -91,7 +91,7 @@
 #include <functional>                // for equal_to
 #include <memory>                    // for alloc<>
 #include <utility>                   // for pair<>
-#include HASH_FUN_H                  // defined in config.h
+#include HASH_MAP_H                  // defined in config.h
 #include <google/sparsehash/libc_allocator_with_realloc.h>
 #include <google/sparsehash/sparsehashtable.h>
 
diff -ru Pasha-1.0.3.orig/google/sparse_hash_set Pasha-1.0.3/google/sparse_hash_set
--- Pasha-1.0.3.orig/google/sparse_hash_set	2010-12-07 04:43:26.000000000 +0100
+++ Pasha-1.0.3/google/sparse_hash_set	2012-08-02 10:15:17.334636915 +0200
@@ -95,7 +95,7 @@
 #include <functional>                // for equal_to
 #include <memory>                    // for alloc<>
 #include <utility>                   // for pair<>
-#include HASH_FUN_H                  // defined in config.h
+#include HASH_SET_H                  // defined in config.h
 #include <google/sparsehash/libc_allocator_with_realloc.h>
 #include <google/sparsehash/sparsehashtable.h>
 
