diff -ru Pasha-1.0.3.orig/Common/AlshaParams.cpp Pasha-1.0.3/Common/AlshaParams.cpp
--- Pasha-1.0.3.orig/Common/AlshaParams.cpp 2011-03-09 06:31:16.000000000 +0100
+++ Pasha-1.0.3/Common/AlshaParams.cpp  2013-03-28 17:01:17.868315244 +0100
@@ -11,6 +11,7 @@
 #include "AlshaUtils.h"
 #include "AlshaParams.h"
 #include <omp.h>
+#include <unistd.h>
 
 #define DEFAULT_KMERLENGTH     21          //kmer length must be odd number between 1 and 31
 //static member variables initialization
