commit 91184a42c74f3db7d588d64b606c60bdb5737c7e
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Fri Feb 21 12:12:02 2025 +0100

    Use external htslib and seqlib

diff --git a/STITCH/src/Makevars b/STITCH/src/Makevars
index 578060a..4e766f5 100644
--- a/STITCH/src/Makevars
+++ b/STITCH/src/Makevars
@@ -2,19 +2,19 @@ HTSLIB_DIR=./htslib-1.19.1/
 SEQLIB_DIR=./SeqLib/
 VCFPP_DIR=./vcfpp/
 
-libseq = ${SEQLIB_DIR}/libseq.a
-libhts = ${HTSLIB_DIR}/libhts.a
+libseq = -lseqlib
+libhts = -lhts
 
 DEP_LIBS = -lz -lm -lbz2 -llzma -lcurl
 
-PKG_CPPFLAGS = -I${SEQLIB_DIR} -I${HTSLIB_DIR} -fPIC
+PKG_CPPFLAGS = -fPIC
 PKG_LIBS = ${libhts} ${libseq} ${DEP_LIBS} $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) 
 
 .PHONY: all clean
 
 all : $(SHLIB)
 
-$(SHLIB): HTSLIB SEQLIB
+$(SHLIB): 
 
 CC=$(shell "R CMD config CC")
 CXX=$(shell "R CMD config CXX")
