diff --git a/utils/collections-management.in b/utils/collections-management.in
index 086c8976..eff4c142 100755
--- a/utils/collections-management.in
+++ b/utils/collections-management.in
@@ -333,7 +333,7 @@ get_build_install_item () {
 
     # download item (and possibly its driver)
     if test x"OpenKIM" = x"${item_type}" -o x"OpenKIM_with_history" = x"${item_type}"; then
-      local query='query={"type":{"$refex":"mo|sm"},"kim-api-version":{"$regex":"^'"${major_version}"'\\."}}'
+      local query='query={"type":{"$regex":"mo|sm"},"kim-api-version":{"$regex":"^'"${major_version}"'\\."}}'
       query="${query}"'&fields={"kimcode":1, "kim-api-version":1}'
       query="${query}"'&database=obj'
       if test x"OpenKIM_with_history" = x"${item_type}"; then
