fix Python include directory for Python 3.x
author: Damian Alvarez (Juelich Supercomputing Centre)
diff -ruN boost_1_65_1.orig/tools/build/src/tools/python.jam boost_1_65_1/tools/build/src/tools/python.jam
--- boost_1_65_1.orig/tools/build/src/tools/python.jam	2017-09-02 11:56:19.000000000 +0200
+++ boost_1_65_1/tools/build/src/tools/python.jam	2017-10-27 16:28:54.720484927 +0200
@@ -544,7 +544,7 @@
     }
     else
     {
-        includes ?= $(prefix)/include/python$(version) ;
+        includes ?= $(prefix)/include/python$(version)m ;
 
         local lib = $(exec-prefix)/lib ;
         libraries ?= $(lib)/python$(version)/config $(lib) ;
