diff -Nru pbbam-2.1.0-orig/src/meson.build pbbam-2.1.0/src/meson.build
--- pbbam-2.1.0-orig/src/meson.build	2023-03-02 11:14:27.989280000 +0100
+++ pbbam-2.1.0/src/meson.build	2023-03-02 11:32:01.493832000 +0100
@@ -13,30 +13,30 @@
 pbbam_config.set('PBBAM_VERSION_MINOR', pbbam_minor_version)
 pbbam_config.set('PBBAM_VERSION_PATCH', pbbam_patch_version)
 
-pbbam_version_sources = [
-
-  # old <-- to be removed
-  vcs_tag(
-    input : 'PbbamGitHash.cpp.in',
-    output : 'PbbamGitHash.cpp',
-    command : ['git', 'describe', '--always', '--dirty=*'],
-    replace_string : '@PBBAM_GIT_SHA1@'),
-  configure_file(
-    input : 'PbbamVersion.cpp.in',
-    output : 'PbbamVersion.cpp',
-    configuration : pbbam_config),
-
-  # new
-  vcs_tag(
-    input : 'LibraryGitHash.h.in',
-    output : 'LibraryGitHash.h',
-    command : ['git', 'describe', '--always', '--dirty=*'],
-    replace_string : '@PBBAM_GIT_SHA1@'),
-  configure_file(
-    input : 'LibraryVersion.h.in',
-    output : 'LibraryVersion.h',
-    configuration : pbbam_config)
-]
+#pbbam_version_sources = [
+#
+#  # old <-- to be removed
+#  vcs_tag(
+#    input : 'PbbamGitHash.cpp.in',
+#    output : 'PbbamGitHash.cpp',
+#    command : ['git', 'describe', '--always', '--dirty=*'],
+#    replace_string : '@PBBAM_GIT_SHA1@'),
+#  configure_file(
+#    input : 'PbbamVersion.cpp.in',
+#    output : 'PbbamVersion.cpp',
+#    configuration : pbbam_config),
+#
+#  # new
+#  vcs_tag(
+#    input : 'LibraryGitHash.h.in',
+#    output : 'LibraryGitHash.h',
+#    command : ['git', 'describe', '--always', '--dirty=*'],
+#    replace_string : '@PBBAM_GIT_SHA1@'),
+#  configure_file(
+#    input : 'LibraryVersion.h.in',
+#    output : 'LibraryVersion.h',
+#    configuration : pbbam_config)
+#]
 
 ###########
 # sources #
@@ -166,7 +166,7 @@
   'pugixml/pugixml.cpp'
 ])
 
-pbbam_cpp_sources += pbbam_version_sources
+#pbbam_cpp_sources += pbbam_version_sources
 
 pbbam_extra_flags = []
 if get_option('auto-validate')
