# Patch to fix MPI tests
# S.D. Pinches
diff -Nru fruit_3.4.3-orig/sample_mpi/rakefile fruit_3.4.3/sample_mpi/rakefile
--- fruit_3.4.3-orig/sample_mpi/rakefile	2018-07-20 17:34:13.507622832 +0200
+++ fruit_3.4.3/sample_mpi/rakefile	2018-07-23 11:38:34.669769977 +0200
@@ -46,7 +46,7 @@
 
 task :run => [ $goal, "mpi_machine_file" ] do
   sleep 3
-  sh "mpiexec -np 2 -machinefile mpi_machine_file #{$goal}"
+  sh "mpiexec -np 2 -machinefile mpi_machine_file "+Dir.pwd+"/#{$goal}"
 end
 
 
diff -Nru fruit_3.4.3-orig/self_test_mpi/rakefile fruit_3.4.3/self_test_mpi/rakefile
--- fruit_3.4.3-orig/self_test_mpi/rakefile	2018-07-20 17:34:13.506622790 +0200
+++ fruit_3.4.3/self_test_mpi/rakefile	2018-07-20 18:50:42.630311001 +0200
@@ -31,7 +31,7 @@
 
 task :run => [ $goal, "mpi_machine_file" ] do
   sleep 3
-  sh "mpiexec -np 3 -machinefile mpi_machine_file #{$goal} > stdout"
+  sh "mpiexec -np 3 -machinefile mpi_machine_file "+Dir.pwd+"/#{$goal} > stdout"
 end
 
 task :verify => [ :run ] do
