# Patch to build with Intel Fortran
# S.D. Pinches
diff -Nru fruit_3.4.3-orig/rake_base.rb fruit_3.4.3/rake_base.rb
--- fruit_3.4.3-orig/rake_base.rb	2018-07-20 17:34:13.506622790 +0200
+++ fruit_3.4.3/rake_base.rb	2018-07-20 18:04:32.166241482 +0200
@@ -9,11 +9,11 @@
   if RUBY_PLATFORM =~ /(darwin|linux)/i
     # Intel FORTRAN compiler tested on Linux
     $compiler = 'ifort'
-    $option = "-check all -warn all -fpp"
+    $option = "-check all -fpp -fPIC"
     $ext_obj = "o"
     $dosish_path = false
     $gcov = false
-		$prof_genx = "-prof-genx"
+    $prof_genx = "-prof-gen=srcpos"
     $mpiexec = nil
   else
     # Intel FORTRAN on Windows
