Fix makefile to use $CXX etc

Åke Sandgren, 2025-09-25
diff --git a/src/makefile b/src/makefile
index 2910e20..621f7a4 100644
--- a/src/makefile
+++ b/src/makefile
@@ -1,8 +1,6 @@
 ##Note: in some cases, static linking is neccessary. If it's needed, please add '-static' in line 3####
-OnTAD:	main.cpp step1.cpp step2.cpp step3.cpp step4.cpp common.cpp
-	g++ -c main.cpp step1.cpp step2.cpp step3.cpp step4.cpp common.cpp
-	g++ main.o step1.o step2.o step3.o step4.o common.o -lm -o OnTAD
-	rm main.o step1.o step2.o step3.o step4.o common.o 
+OnTAD:	main.o step1.o step2.o step3.o step4.o common.o
+	$(CXX) $(LDFLAGS) $? -lm -o OnTAD
 
 clean:
 	rm OnTAD
