--- deap-1.0.1.orig/setup.py	2014-04-08 15:49:52.000000000 +0200
+++ deap-1.0.1/setup.py	2015-05-21 22:25:37.345008000 +0200
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+import codecs
 import sys
 from distutils.core import setup
 try:
@@ -11,7 +12,7 @@
 setup(name='deap',
       version=deap.__revision__,
       description='Distributed Evolutionary Algorithms in Python',
-      long_description=open('README.txt').read(),
+      long_description=codecs.open('README.txt', encoding='utf-8').read(),
       author='deap Development Team',
       author_email='deap-users@googlegroups.com',
       url='http://deap.googlecode.com',
