diff -ru numpy-1.6.2.orig/numpy/distutils/intelccompiler.py numpy-1.6.2/numpy/distutils/intelccompiler.py
--- numpy-1.6.2.orig/numpy/distutils/intelccompiler.py	2012-03-25 16:05:46.000000000 +0200
+++ numpy-1.6.2/numpy/distutils/intelccompiler.py	2013-05-11 14:42:40.365737000 +0200
@@ -15,7 +15,7 @@
                              compiler_so=compiler,
                              compiler_cxx=compiler,
                              linker_exe=compiler,
-                             linker_so=compiler + ' -shared')
+                             linker_so=compiler + ' -shared -L$EBROOTICC/lib/intel64')
 
 class IntelItaniumCCompiler(IntelCCompiler):
     compiler_type = 'intele'
@@ -40,4 +40,4 @@
                              compiler_so=compiler,
                              compiler_cxx=compiler,
                              linker_exe=compiler,
-                             linker_so=compiler + ' -shared')
+                             linker_so=compiler + ' -shared  -L$EBROOTICC/lib/intel64')
