Use mathlib and threads in the LDFLAGS

Åke Sandgren, 2022-12-07
diff -ru grass-8.2.0.orig/configure grass-8.2.0/configure
--- grass-8.2.0.orig/configure	2022-06-03 15:27:06.000000000 +0200
+++ grass-8.2.0/configure	2022-12-07 14:23:37.234003038 +0100
@@ -4031,7 +4031,7 @@
             SHLIB_LD_FLAGS="-Wl,-soname,\$(notdir \$@)"
 	    SHLIB_SUFFIX=".so"
 	    SHLIB_LD="${CC} -shared"
-            LDFLAGS="-Wl,--export-dynamic"
+            LDFLAGS="-pthread -lm -Wl,--export-dynamic"
             LD_SEARCH_FLAGS='-Wl,-rpath-link,${LIB_RUNTIME_DIR} -Wl,-rpath,${INST_DIR}/lib'
             LD_LIBRARY_PATH_VAR="LD_LIBRARY_PATH"
             ;;
