diff -ur grass-7.0.4.orig/configure grass-7.0.4/configure
--- grass-7.0.4.orig/configure	2016-07-21 21:29:39.699203000 +1200
+++ grass-7.0.4/configure	2016-07-21 21:32:29.672539948 +1200
@@ -5925,7 +5925,7 @@
 ac_lib_var=`echo readline'_'readline | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-lreadline  $LIBS"
+LIBS="-lreadline -lncurses $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 5931 "configure"
 #include "confdefs.h"
@@ -5952,7 +5952,7 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  READLINELIB="$READLINELIB -lreadline "
+  READLINELIB="$READLINELIB -lreadline -lncurses "
 else
   echo "$ac_t""no" 1>&6
 
