diff -ur grass-7.0.2.orig/imagery/i.cca/Makefile grass-7.0.2/imagery/i.cca/Makefile
--- grass-7.0.2.orig/imagery/i.cca/Makefile	2009-08-29 21:36:02.000000000 +1200
+++ grass-7.0.2/imagery/i.cca/Makefile	2016-01-28 11:44:11.820588000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(IMAGERYLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(IMAGERYDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/imagery/i.fft/Makefile grass-7.0.2/imagery/i.fft/Makefile
--- grass-7.0.2.orig/imagery/i.fft/Makefile	2009-08-29 21:36:02.000000000 +1200
+++ grass-7.0.2/imagery/i.fft/Makefile	2016-01-28 11:51:26.914401000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/imagery/i.gensigset/Makefile grass-7.0.2/imagery/i.gensigset/Makefile
--- grass-7.0.2.orig/imagery/i.gensigset/Makefile	2009-08-29 21:36:03.000000000 +1200
+++ grass-7.0.2/imagery/i.gensigset/Makefile	2016-01-28 11:44:45.354763000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(IMAGERYLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(IMAGERYDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/imagery/i.ifft/Makefile grass-7.0.2/imagery/i.ifft/Makefile
--- grass-7.0.2.orig/imagery/i.ifft/Makefile	2009-08-29 21:36:02.000000000 +1200
+++ grass-7.0.2/imagery/i.ifft/Makefile	2016-01-28 11:51:56.509915000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/imagery/i.ortho.photo/i.ortho.camera/Makefile grass-7.0.2/imagery/i.ortho.photo/i.ortho.camera/Makefile
--- grass-7.0.2.orig/imagery/i.ortho.photo/i.ortho.camera/Makefile	2012-12-22 23:35:11.000000000 +1300
+++ grass-7.0.2/imagery/i.ortho.photo/i.ortho.camera/Makefile	2016-01-28 11:46:55.940817000 +1300
@@ -2,7 +2,8 @@
 
 PGM = i.ortho.camera
 
-EXTRA_CFLAGS = -I../lib
+EXTRA_CFLAGS = -I../lib $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 LIBES     = $(IORTHOLIB) $(IMAGERYLIB) $(GISLIB) $(GMATHLIB)
 DEPENDENCIES= $(IMAGERYDEP) $(IORTHODEP) $(GISDEP) $(GMATHDEP)
diff -ur grass-7.0.2.orig/imagery/i.ortho.photo/i.ortho.elev/Makefile grass-7.0.2/imagery/i.ortho.photo/i.ortho.elev/Makefile
--- grass-7.0.2.orig/imagery/i.ortho.photo/i.ortho.elev/Makefile	2012-12-15 23:35:06.000000000 +1300
+++ grass-7.0.2/imagery/i.ortho.photo/i.ortho.elev/Makefile	2016-01-28 11:45:50.324723000 +1300
@@ -5,7 +5,8 @@
 LIBES     = $(IMAGERYLIB) $(GISLIB) $(IORTHOLIB) $(GMATHLIB)
 DEPENDENCIES= $(IMAGERYDEP) $(IORTHODEP) $(GISDEP) $(GMATHDEP)
 
-EXTRA_CFLAGS = -I../lib
+EXTRA_CFLAGS = -I../lib $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/imagery/i.ortho.photo/i.ortho.rectify/Makefile grass-7.0.2/imagery/i.ortho.photo/i.ortho.rectify/Makefile
--- grass-7.0.2.orig/imagery/i.ortho.photo/i.ortho.rectify/Makefile	2013-02-23 23:35:28.000000000 +1300
+++ grass-7.0.2/imagery/i.ortho.photo/i.ortho.rectify/Makefile	2016-01-28 11:46:21.898790000 +1300
@@ -2,7 +2,8 @@
 
 PGM = i.ortho.rectify
 
-EXTRA_CFLAGS = -I../lib
+EXTRA_CFLAGS = -I../lib $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 LIBES     = $(IMAGERYLIB) $(RASTERLIB) $(GISLIB) $(IORTHOLIB) $(GMATHLIB)
 DEPENDENCIES= $(IMAGERYDEP) $(RASTERDEP) $(IORTHODEP) $(GISDEP) $(GMATHDEP)
diff -ur grass-7.0.2.orig/imagery/i.pca/Makefile grass-7.0.2/imagery/i.pca/Makefile
--- grass-7.0.2.orig/imagery/i.pca/Makefile	2013-06-01 21:22:17.000000000 +1200
+++ grass-7.0.2/imagery/i.pca/Makefile	2016-01-28 11:50:55.273866000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(IMAGERYLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(IMAGERYDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/imagery/i.smap/Makefile grass-7.0.2/imagery/i.smap/Makefile
--- grass-7.0.2.orig/imagery/i.smap/Makefile	2009-08-29 21:36:03.000000000 +1200
+++ grass-7.0.2/imagery/i.smap/Makefile	2016-01-28 11:50:06.236796000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(IMAGERYLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(IMAGERYDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/imagery/i.zc/Makefile grass-7.0.2/imagery/i.zc/Makefile
--- grass-7.0.2.orig/imagery/i.zc/Makefile	2009-08-29 21:36:02.000000000 +1200
+++ grass-7.0.2/imagery/i.zc/Makefile	2016-01-28 11:52:29.162941000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/lib/gmath/test/Makefile grass-7.0.2/lib/gmath/test/Makefile
--- grass-7.0.2.orig/lib/gmath/test/Makefile	2013-02-23 23:35:23.000000000 +1300
+++ grass-7.0.2/lib/gmath/test/Makefile	2016-01-27 17:06:18.499321000 +1300
@@ -5,6 +5,7 @@
 LIBES = $(GISLIB) $(GMATHLIB) $(OMPLIB)
 DEPENDENCIES = $(GISDEP) $(GMATHDEP) $(OMPDEP)
 EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/lib/gpde/test/Makefile grass-7.0.2/lib/gpde/test/Makefile
--- grass-7.0.2.orig/lib/gpde/test/Makefile	2011-12-17 23:35:05.000000000 +1300
+++ grass-7.0.2/lib/gpde/test/Makefile	2016-01-27 16:53:24.642159000 +1300
@@ -5,6 +5,7 @@
 LIBES = $(GISLIB) $(RASTER3DLIB) $(GPDELIB) $(OMPLIB) $(GMATHLIB)
 DEPENDENCIES = $(GISDEP) $(RASTER3DDEP) $(GPDEDEP)
 EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.gwflow/Makefile grass-7.0.2/raster/r.gwflow/Makefile
--- grass-7.0.2.orig/raster/r.gwflow/Makefile	2009-10-10 22:35:38.000000000 +1300
+++ grass-7.0.2/raster/r.gwflow/Makefile	2016-01-27 17:07:05.528064000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GPDELIB) $(RASTERLIB) $(GISLIB) $(GMATHLIB)
 DEPENDENCIES = $(GPDEDEP) $(RASTERDEP) $(GISDEP) $(GMATHDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.param.scale/Makefile grass-7.0.2/raster/r.param.scale/Makefile
--- grass-7.0.2.orig/raster/r.param.scale/Makefile	2009-08-29 21:35:38.000000000 +1200
+++ grass-7.0.2/raster/r.param.scale/Makefile	2016-01-27 17:07:50.080533000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.resamp.bspline/Makefile grass-7.0.2/raster/r.resamp.bspline/Makefile
--- grass-7.0.2.orig/raster/r.resamp.bspline/Makefile	2013-02-23 23:35:20.000000000 +1300
+++ grass-7.0.2/raster/r.resamp.bspline/Makefile	2016-01-27 17:08:39.473918000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(RASTERLIB) $(SEGMENTLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(RASTERDEP) $(SEGMENTDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.resamp.rst/Makefile grass-7.0.2/raster/r.resamp.rst/Makefile
--- grass-7.0.2.orig/raster/r.resamp.rst/Makefile	2012-12-29 23:35:10.000000000 +1300
+++ grass-7.0.2/raster/r.resamp.rst/Makefile	2016-01-27 17:09:20.374610000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(INTERPFLLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(INTERPFLDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.series.accumulate/Makefile grass-7.0.2/raster/r.series.accumulate/Makefile
--- grass-7.0.2.orig/raster/r.series.accumulate/Makefile	2014-03-28 04:54:58.000000000 +1300
+++ grass-7.0.2/raster/r.series.accumulate/Makefile	2016-01-27 17:09:51.998995000 +1300
@@ -5,7 +5,7 @@
 LIBES = $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(RASTERDEP) $(GISDEP)
 
-EXTRA_LIBS = $(OMPLIB)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 EXTRA_CFLAGS = $(OMPCFLAGS)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
diff -ur grass-7.0.2.orig/raster/r.sim/r.sim.sediment/Makefile grass-7.0.2/raster/r.sim/r.sim.sediment/Makefile
--- grass-7.0.2.orig/raster/r.sim/r.sim.sediment/Makefile	2014-11-29 06:50:34.000000000 +1300
+++ grass-7.0.2/raster/r.sim/r.sim.sediment/Makefile	2016-01-27 17:11:03.532489000 +1300
@@ -7,7 +7,8 @@
 LIBES     = $(SIMLIB) $(GMATHLIB) $(GISLIB)
 DEPENDENCIES = $(SIMDEP) $(GMATHDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) 
+EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.sim/r.sim.water/Makefile grass-7.0.2/raster/r.sim/r.sim.water/Makefile
--- grass-7.0.2.orig/raster/r.sim/r.sim.water/Makefile	2014-11-29 06:50:35.000000000 +1300
+++ grass-7.0.2/raster/r.sim/r.sim.water/Makefile	2016-01-27 17:10:39.359639000 +1300
@@ -7,7 +7,8 @@
 LIBES     = $(SIMLIB) $(GMATHLIB) $(GISLIB)
 DEPENDENCIES = $(SIMDEP) $(GMATHDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) 
+EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.solute.transport/Makefile grass-7.0.2/raster/r.solute.transport/Makefile
--- grass-7.0.2.orig/raster/r.solute.transport/Makefile	2013-02-23 23:35:21.000000000 +1300
+++ grass-7.0.2/raster/r.solute.transport/Makefile	2016-01-27 17:11:33.478411000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GISLIB) $(RASTERLIB) $(GPDELIB) $(GMATHLIB)
 DEPENDENCIES = $(GISDEP) $(RASTERDEP) $(GMATHDEP) $(GPDEDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.surf.fractal/Makefile grass-7.0.2/raster/r.surf.fractal/Makefile
--- grass-7.0.2.orig/raster/r.surf.fractal/Makefile	2009-08-29 21:35:39.000000000 +1200
+++ grass-7.0.2/raster/r.surf.fractal/Makefile	2016-01-27 17:12:01.472263000 +1300
@@ -5,6 +5,8 @@
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(FFTWINC)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.surf.gauss/Makefile grass-7.0.2/raster/r.surf.gauss/Makefile
--- grass-7.0.2.orig/raster/r.surf.gauss/Makefile	2009-08-29 21:35:45.000000000 +1200
+++ grass-7.0.2/raster/r.surf.gauss/Makefile	2016-01-27 17:12:29.099681000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster/r.surf.random/Makefile grass-7.0.2/raster/r.surf.random/Makefile
--- grass-7.0.2.orig/raster/r.surf.random/Makefile	2009-08-29 21:35:44.000000000 +1200
+++ grass-7.0.2/raster/r.surf.random/Makefile	2016-01-27 17:13:20.394477000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/raster3d/r3.gwflow/Makefile grass-7.0.2/raster3d/r3.gwflow/Makefile
--- grass-7.0.2.orig/raster3d/r3.gwflow/Makefile	2011-08-13 21:35:05.000000000 +1200
+++ grass-7.0.2/raster3d/r3.gwflow/Makefile	2016-01-28 11:36:49.263171000 +1300
@@ -4,6 +4,8 @@
 
 LIBES = $(GPDELIB) $(RASTER3DLIB) $(GISLIB) $(GMATHLIB)
 DEPENDENCIES = $(GPDEDEP) $(RASTER3DDEP) $(GISDEP) $(GMATHDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.extract/Makefile grass-7.0.2/vector/v.extract/Makefile
--- grass-7.0.2.orig/vector/v.extract/Makefile	2009-08-29 21:35:56.000000000 +1200
+++ grass-7.0.2/vector/v.extract/Makefile	2016-01-28 11:37:42.519279000 +1300
@@ -6,7 +6,8 @@
 LIBES = $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
  
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.kernel/Makefile grass-7.0.2/vector/v.kernel/Makefile
--- grass-7.0.2.orig/vector/v.kernel/Makefile	2009-08-29 21:35:52.000000000 +1200
+++ grass-7.0.2/vector/v.kernel/Makefile	2016-01-28 11:38:17.840815000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(GMATHLIB) $(VECTORLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(VECTORDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
  
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.lidar.correction/Makefile grass-7.0.2/vector/v.lidar.correction/Makefile
--- grass-7.0.2.orig/vector/v.lidar.correction/Makefile	2012-02-11 23:35:05.000000000 +1300
+++ grass-7.0.2/vector/v.lidar.correction/Makefile	2016-01-28 11:39:51.886027000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(RASTERLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP) $(RASTERDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.lidar.edgedetection/Makefile grass-7.0.2/vector/v.lidar.edgedetection/Makefile
--- grass-7.0.2.orig/vector/v.lidar.edgedetection/Makefile	2012-02-11 23:35:05.000000000 +1300
+++ grass-7.0.2/vector/v.lidar.edgedetection/Makefile	2016-01-28 11:40:16.843685000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS) 
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.lidar.growing/Makefile grass-7.0.2/vector/v.lidar.growing/Makefile
--- grass-7.0.2.orig/vector/v.lidar.growing/Makefile	2012-02-11 23:35:06.000000000 +1300
+++ grass-7.0.2/vector/v.lidar.growing/Makefile	2016-01-28 11:40:50.620191000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.outlier/Makefile grass-7.0.2/vector/v.outlier/Makefile
--- grass-7.0.2.orig/vector/v.outlier/Makefile	2012-02-11 23:35:05.000000000 +1300
+++ grass-7.0.2/vector/v.outlier/Makefile	2016-01-28 11:41:17.531036000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.rectify/Makefile grass-7.0.2/vector/v.rectify/Makefile
--- grass-7.0.2.orig/vector/v.rectify/Makefile	2012-04-28 21:35:05.000000000 +1200
+++ grass-7.0.2/vector/v.rectify/Makefile	2016-01-28 11:42:40.339675000 +1300
@@ -6,7 +6,8 @@
 DEPENDENCIES = $(GMATHDEP) $(IMAGERYDEP) $(VECTORDEP) $(GISDEP)
 
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.surf.bspline/Makefile grass-7.0.2/vector/v.surf.bspline/Makefile
--- grass-7.0.2.orig/vector/v.surf.bspline/Makefile	2012-01-07 23:35:09.000000000 +1300
+++ grass-7.0.2/vector/v.surf.bspline/Makefile	2016-01-28 11:43:08.598751000 +1300
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(RASTERLIB) $(GISLIB) $(SEGMENTLIB) $(MATHLIB) $(GPDELIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(RASTERDEP) $(SEGMENTDEP) $(GISDEP) $(GPDEDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.2.orig/vector/v.surf.rst/Makefile grass-7.0.2/vector/v.surf.rst/Makefile
--- grass-7.0.2.orig/vector/v.surf.rst/Makefile	2009-08-29 21:35:55.000000000 +1200
+++ grass-7.0.2/vector/v.surf.rst/Makefile	2016-01-28 11:43:37.061921000 +1300
@@ -7,7 +7,8 @@
 LIBES = $(INTERPFLLIB) $(QTREELIB) $(INTERPDATALIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(INTERPFLDEP) $(QTREEDEP) $(INTERPDATADEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
