diff -ur grass-7.0.4.orig/imagery/i.cca/Makefile grass-7.0.4/imagery/i.cca/Makefile
--- grass-7.0.4.orig/imagery/i.cca/Makefile	2016-07-21 21:29:42.447992000 +1200
+++ grass-7.0.4/imagery/i.cca/Makefile	2016-07-21 21:44:34.347981000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(IMAGERYLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(IMAGERYDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/imagery/i.fft/Makefile grass-7.0.4/imagery/i.fft/Makefile
--- grass-7.0.4.orig/imagery/i.fft/Makefile	2016-07-21 21:29:41.604488000 +1200
+++ grass-7.0.4/imagery/i.fft/Makefile	2016-07-21 21:44:34.357289000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/imagery/i.gensigset/Makefile grass-7.0.4/imagery/i.gensigset/Makefile
--- grass-7.0.4.orig/imagery/i.gensigset/Makefile	2016-07-21 21:29:41.442671000 +1200
+++ grass-7.0.4/imagery/i.gensigset/Makefile	2016-07-21 21:44:34.370444000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(IMAGERYLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(IMAGERYDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/imagery/i.ifft/Makefile grass-7.0.4/imagery/i.ifft/Makefile
--- grass-7.0.4.orig/imagery/i.ifft/Makefile	2016-07-21 21:29:41.793582000 +1200
+++ grass-7.0.4/imagery/i.ifft/Makefile	2016-07-21 21:44:34.381089000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/imagery/i.ortho.photo/i.ortho.camera/Makefile grass-7.0.4/imagery/i.ortho.photo/i.ortho.camera/Makefile
--- grass-7.0.4.orig/imagery/i.ortho.photo/i.ortho.camera/Makefile	2016-07-21 21:29:42.139841000 +1200
+++ grass-7.0.4/imagery/i.ortho.photo/i.ortho.camera/Makefile	2016-07-21 21:44:34.388934000 +1200
@@ -2,7 +2,8 @@
 
 PGM = i.ortho.camera
 
-EXTRA_CFLAGS = -I../lib
+EXTRA_CFLAGS = -I../lib $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 LIBES     = $(IORTHOLIB) $(IMAGERYLIB) $(GISLIB) $(GMATHLIB)
 DEPENDENCIES= $(IMAGERYDEP) $(IORTHODEP) $(GISDEP) $(GMATHDEP)
diff -ur grass-7.0.4.orig/imagery/i.ortho.photo/i.ortho.elev/Makefile grass-7.0.4/imagery/i.ortho.photo/i.ortho.elev/Makefile
--- grass-7.0.4.orig/imagery/i.ortho.photo/i.ortho.elev/Makefile	2016-07-21 21:29:42.146663000 +1200
+++ grass-7.0.4/imagery/i.ortho.photo/i.ortho.elev/Makefile	2016-07-21 21:44:34.392771000 +1200
@@ -5,7 +5,8 @@
 LIBES     = $(IMAGERYLIB) $(GISLIB) $(IORTHOLIB) $(GMATHLIB)
 DEPENDENCIES= $(IMAGERYDEP) $(IORTHODEP) $(GISDEP) $(GMATHDEP)
 
-EXTRA_CFLAGS = -I../lib
+EXTRA_CFLAGS = -I../lib $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/imagery/i.ortho.photo/i.ortho.rectify/Makefile grass-7.0.4/imagery/i.ortho.photo/i.ortho.rectify/Makefile
--- grass-7.0.4.orig/imagery/i.ortho.photo/i.ortho.rectify/Makefile	2016-07-21 21:29:42.046427000 +1200
+++ grass-7.0.4/imagery/i.ortho.photo/i.ortho.rectify/Makefile	2016-07-21 21:44:34.397336000 +1200
@@ -2,7 +2,8 @@
 
 PGM = i.ortho.rectify
 
-EXTRA_CFLAGS = -I../lib
+EXTRA_CFLAGS = -I../lib $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 LIBES     = $(IMAGERYLIB) $(RASTERLIB) $(GISLIB) $(IORTHOLIB) $(GMATHLIB)
 DEPENDENCIES= $(IMAGERYDEP) $(RASTERDEP) $(IORTHODEP) $(GISDEP) $(GMATHDEP)
diff -ur grass-7.0.4.orig/imagery/i.pca/Makefile grass-7.0.4/imagery/i.pca/Makefile
--- grass-7.0.4.orig/imagery/i.pca/Makefile	2016-07-21 21:29:41.427768000 +1200
+++ grass-7.0.4/imagery/i.pca/Makefile	2016-07-21 21:44:34.401107000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(IMAGERYLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(IMAGERYDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/imagery/i.smap/Makefile grass-7.0.4/imagery/i.smap/Makefile
--- grass-7.0.4.orig/imagery/i.smap/Makefile	2016-07-21 21:29:41.656965000 +1200
+++ grass-7.0.4/imagery/i.smap/Makefile	2016-07-21 21:44:34.405171000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(IMAGERYLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(IMAGERYDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/imagery/i.zc/Makefile grass-7.0.4/imagery/i.zc/Makefile
--- grass-7.0.4.orig/imagery/i.zc/Makefile	2016-07-21 21:29:41.407457000 +1200
+++ grass-7.0.4/imagery/i.zc/Makefile	2016-07-21 21:44:34.409179000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/lib/gmath/test/Makefile grass-7.0.4/lib/gmath/test/Makefile
--- grass-7.0.4.orig/lib/gmath/test/Makefile	2016-07-21 21:29:55.718124000 +1200
+++ grass-7.0.4/lib/gmath/test/Makefile	2016-07-21 21:44:34.413120000 +1200
@@ -5,6 +5,7 @@
 LIBES = $(GISLIB) $(GMATHLIB) $(OMPLIB)
 DEPENDENCIES = $(GISDEP) $(GMATHDEP) $(OMPDEP)
 EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/lib/gpde/test/Makefile grass-7.0.4/lib/gpde/test/Makefile
--- grass-7.0.4.orig/lib/gpde/test/Makefile	2016-07-21 21:29:48.875450000 +1200
+++ grass-7.0.4/lib/gpde/test/Makefile	2016-07-21 21:44:34.425858000 +1200
@@ -5,6 +5,7 @@
 LIBES = $(GISLIB) $(RASTER3DLIB) $(GPDELIB) $(OMPLIB) $(GMATHLIB)
 DEPENDENCIES = $(GISDEP) $(RASTER3DDEP) $(GPDEDEP)
 EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.gwflow/Makefile grass-7.0.4/raster/r.gwflow/Makefile
--- grass-7.0.4.orig/raster/r.gwflow/Makefile	2016-07-21 21:30:14.603813000 +1200
+++ grass-7.0.4/raster/r.gwflow/Makefile	2016-07-21 21:44:34.433059000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GPDELIB) $(RASTERLIB) $(GISLIB) $(GMATHLIB)
 DEPENDENCIES = $(GPDEDEP) $(RASTERDEP) $(GISDEP) $(GMATHDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.param.scale/Makefile grass-7.0.4/raster/r.param.scale/Makefile
--- grass-7.0.4.orig/raster/r.param.scale/Makefile	2016-07-21 21:30:04.451301000 +1200
+++ grass-7.0.4/raster/r.param.scale/Makefile	2016-07-21 21:44:34.441984000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.resamp.bspline/Makefile grass-7.0.4/raster/r.resamp.bspline/Makefile
--- grass-7.0.4.orig/raster/r.resamp.bspline/Makefile	2016-07-21 21:30:14.892471000 +1200
+++ grass-7.0.4/raster/r.resamp.bspline/Makefile	2016-07-21 21:44:34.449112000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(RASTERLIB) $(SEGMENTLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(RASTERDEP) $(SEGMENTDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.resamp.rst/Makefile grass-7.0.4/raster/r.resamp.rst/Makefile
--- grass-7.0.4.orig/raster/r.resamp.rst/Makefile	2016-07-21 21:30:15.995894000 +1200
+++ grass-7.0.4/raster/r.resamp.rst/Makefile	2016-07-21 21:44:34.452994000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(INTERPFLLIB) $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(INTERPFLDEP) $(GMATHDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.series.accumulate/Makefile grass-7.0.4/raster/r.series.accumulate/Makefile
--- grass-7.0.4.orig/raster/r.series.accumulate/Makefile	2016-07-21 21:30:05.443175000 +1200
+++ grass-7.0.4/raster/r.series.accumulate/Makefile	2016-07-21 21:44:34.457024000 +1200
@@ -5,7 +5,7 @@
 LIBES = $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(RASTERDEP) $(GISDEP)
 
-EXTRA_LIBS = $(OMPLIB)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 EXTRA_CFLAGS = $(OMPCFLAGS)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
diff -ur grass-7.0.4.orig/raster/r.sim/r.sim.sediment/Makefile grass-7.0.4/raster/r.sim/r.sim.sediment/Makefile
--- grass-7.0.4.orig/raster/r.sim/r.sim.sediment/Makefile	2016-07-21 21:30:13.976226000 +1200
+++ grass-7.0.4/raster/r.sim/r.sim.sediment/Makefile	2016-07-21 21:44:34.461195000 +1200
@@ -7,7 +7,8 @@
 LIBES     = $(SIMLIB) $(GMATHLIB) $(GISLIB)
 DEPENDENCIES = $(SIMDEP) $(GMATHDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) 
+EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.sim/r.sim.water/Makefile grass-7.0.4/raster/r.sim/r.sim.water/Makefile
--- grass-7.0.4.orig/raster/r.sim/r.sim.water/Makefile	2016-07-21 21:30:13.994334000 +1200
+++ grass-7.0.4/raster/r.sim/r.sim.water/Makefile	2016-07-21 21:44:34.465185000 +1200
@@ -7,7 +7,8 @@
 LIBES     = $(SIMLIB) $(GMATHLIB) $(GISLIB)
 DEPENDENCIES = $(SIMDEP) $(GMATHDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) 
+EXTRA_CFLAGS = -I ../simlib $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.solute.transport/Makefile grass-7.0.4/raster/r.solute.transport/Makefile
--- grass-7.0.4.orig/raster/r.solute.transport/Makefile	2016-07-21 21:30:12.338796000 +1200
+++ grass-7.0.4/raster/r.solute.transport/Makefile	2016-07-21 21:44:34.469092000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GISLIB) $(RASTERLIB) $(GPDELIB) $(GMATHLIB)
 DEPENDENCIES = $(GISDEP) $(RASTERDEP) $(GMATHDEP) $(GPDEDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.surf.fractal/Makefile grass-7.0.4/raster/r.surf.fractal/Makefile
--- grass-7.0.4.orig/raster/r.surf.fractal/Makefile	2016-07-21 21:30:14.094128000 +1200
+++ grass-7.0.4/raster/r.surf.fractal/Makefile	2016-07-21 21:44:34.472872000 +1200
@@ -5,6 +5,8 @@
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(FFTWINC)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.surf.gauss/Makefile grass-7.0.4/raster/r.surf.gauss/Makefile
--- grass-7.0.4.orig/raster/r.surf.gauss/Makefile	2016-07-21 21:30:08.582436000 +1200
+++ grass-7.0.4/raster/r.surf.gauss/Makefile	2016-07-21 21:44:34.477072000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster/r.surf.random/Makefile grass-7.0.4/raster/r.surf.random/Makefile
--- grass-7.0.4.orig/raster/r.surf.random/Makefile	2016-07-21 21:30:04.904779000 +1200
+++ grass-7.0.4/raster/r.surf.random/Makefile	2016-07-21 21:44:34.480982000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GMATHLIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(RASTERDEP) $(GISDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/raster3d/r3.gwflow/Makefile grass-7.0.4/raster3d/r3.gwflow/Makefile
--- grass-7.0.4.orig/raster3d/r3.gwflow/Makefile	2016-07-21 21:30:16.260812000 +1200
+++ grass-7.0.4/raster3d/r3.gwflow/Makefile	2016-07-21 21:44:34.485011000 +1200
@@ -4,6 +4,8 @@
 
 LIBES = $(GPDELIB) $(RASTER3DLIB) $(GISLIB) $(GMATHLIB)
 DEPENDENCIES = $(GPDEDEP) $(RASTER3DDEP) $(GISDEP) $(GMATHDEP)
+EXTRA_CFLAGS = $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.extract/Makefile grass-7.0.4/vector/v.extract/Makefile
--- grass-7.0.4.orig/vector/v.extract/Makefile	2016-07-21 21:30:21.366806000 +1200
+++ grass-7.0.4/vector/v.extract/Makefile	2016-07-21 21:44:34.489087000 +1200
@@ -6,7 +6,8 @@
 LIBES = $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
  
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.kernel/Makefile grass-7.0.4/vector/v.kernel/Makefile
--- grass-7.0.4.orig/vector/v.kernel/Makefile	2016-07-21 21:30:26.570487000 +1200
+++ grass-7.0.4/vector/v.kernel/Makefile	2016-07-21 21:44:34.493165000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(GMATHLIB) $(VECTORLIB) $(RASTERLIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(GMATHDEP) $(VECTORDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
  
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.lidar.correction/Makefile grass-7.0.4/vector/v.lidar.correction/Makefile
--- grass-7.0.4.orig/vector/v.lidar.correction/Makefile	2016-07-21 21:30:25.793151000 +1200
+++ grass-7.0.4/vector/v.lidar.correction/Makefile	2016-07-21 21:44:34.497018000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(RASTERLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP) $(RASTERDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.lidar.edgedetection/Makefile grass-7.0.4/vector/v.lidar.edgedetection/Makefile
--- grass-7.0.4.orig/vector/v.lidar.edgedetection/Makefile	2016-07-21 21:30:23.758978000 +1200
+++ grass-7.0.4/vector/v.lidar.edgedetection/Makefile	2016-07-21 21:44:34.501248000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS) 
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.lidar.growing/Makefile grass-7.0.4/vector/v.lidar.growing/Makefile
--- grass-7.0.4.orig/vector/v.lidar.growing/Makefile	2016-07-21 21:30:25.544410000 +1200
+++ grass-7.0.4/vector/v.lidar.growing/Makefile	2016-07-21 21:44:34.505156000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(RASTERLIB) $(GISLIB)
 DEPENDENCIES = $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(RASTERDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.outlier/Makefile grass-7.0.4/vector/v.outlier/Makefile
--- grass-7.0.4.orig/vector/v.outlier/Makefile	2016-07-21 21:30:24.225538000 +1200
+++ grass-7.0.4/vector/v.outlier/Makefile	2016-07-21 21:44:34.513858000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.rectify/Makefile grass-7.0.4/vector/v.rectify/Makefile
--- grass-7.0.4.orig/vector/v.rectify/Makefile	2016-07-21 21:30:25.983589000 +1200
+++ grass-7.0.4/vector/v.rectify/Makefile	2016-07-21 21:44:34.517553000 +1200
@@ -6,7 +6,8 @@
 DEPENDENCIES = $(GMATHDEP) $(IMAGERYDEP) $(VECTORDEP) $(GISDEP)
 
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.surf.bspline/Makefile grass-7.0.4/vector/v.surf.bspline/Makefile
--- grass-7.0.4.orig/vector/v.surf.bspline/Makefile	2016-07-21 21:30:23.579357000 +1200
+++ grass-7.0.4/vector/v.surf.bspline/Makefile	2016-07-21 21:44:34.525034000 +1200
@@ -5,7 +5,8 @@
 LIBES = $(LIDARLIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(RASTERLIB) $(GISLIB) $(SEGMENTLIB) $(MATHLIB) $(GPDELIB)
 DEPENDENCIES = $(LIDARDEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(RASTERDEP) $(SEGMENTDEP) $(GISDEP) $(GPDEDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
diff -ur grass-7.0.4.orig/vector/v.surf.rst/Makefile grass-7.0.4/vector/v.surf.rst/Makefile
--- grass-7.0.4.orig/vector/v.surf.rst/Makefile	2016-07-21 21:30:24.012506000 +1200
+++ grass-7.0.4/vector/v.surf.rst/Makefile	2016-07-21 21:44:34.533904000 +1200
@@ -7,7 +7,8 @@
 LIBES = $(INTERPFLLIB) $(QTREELIB) $(INTERPDATALIB) $(GMATHLIB) $(VECTORLIB) $(DBMILIB) $(GISLIB) $(MATHLIB)
 DEPENDENCIES = $(INTERPFLDEP) $(QTREEDEP) $(INTERPDATADEP) $(GMATHDEP) $(VECTORDEP) $(DBMIDEP) $(GISDEP)
 EXTRA_INC = $(VECT_INC)
-EXTRA_CFLAGS = $(VECT_CFLAGS)
+EXTRA_CFLAGS = $(VECT_CFLAGS) $(OMPCFLAGS)
+EXTRA_LIBS = $(OMPCFLAGS) $(OMPLIB)
 
 include $(MODULE_TOPDIR)/include/Make/Module.make
 
