diff -ru grass-7.8.4.orig/lib/python/ctypes/Makefile grass-7.8.4/lib/python/ctypes/Makefile
--- grass-7.8.4.orig/lib/python/ctypes/Makefile	2020-11-10 19:30:29.556692350 +0000
+++ grass-7.8.4/lib/python/ctypes/Makefile	2020-11-10 19:31:30.937299465 +0000
@@ -63,7 +63,7 @@
 
 SED = sed
 CTYPESGEN = ./ctypesgen.py
-CTYPESFLAGS = --cpp "$(CC) -E $(CPPFLAGS) $(LFS_CFLAGS) $(MAC_FLAGS) $(EXTRA_CFLAGS) $(NLS_CFLAGS) $(DEFS) $(EXTRA_INC) $(INC) -D__GLIBC_HAVE_LONG_LONG"
+CTYPESFLAGS = --cpp "$(CC) -E $(CPPFLAGS) $(LFS_CFLAGS) $(MAC_FLAGS) $(EXTRA_CFLAGS) -std=c90 $(NLS_CFLAGS) $(DEFS) $(EXTRA_INC) $(INC) -D__GLIBC_HAVE_LONG_LONG"
 EXTRA_CLEAN_FILES := $(wildcard ctypesgencore/*.pyc) $(wildcard ctypesgencore/*/*.pyc)
 
 ifneq ($(MINGW),)
