# Add explicit includes for building with GCC 13
#
diff -ur krakenuniq-1.0.4/src/report-cols.hpp krakenuniq-1.0.4.new/src/report-cols.hpp
--- krakenuniq-1.0.4/src/report-cols.hpp	2023-04-26 16:39:35.000000000 +0100
+++ krakenuniq-1.0.4.new/src/report-cols.hpp	2026-02-19 11:26:18.733169725 +0000
@@ -9,6 +9,7 @@
 
 #include<map>
 #include<string>
+#include<cstdint>
 
 enum class REPORTCOLS : uint8_t {
 	SPACED_NAME,
diff -ur krakenuniq-1.0.4/src/uid_mapping.hpp krakenuniq-1.0.4.new/src/uid_mapping.hpp
--- krakenuniq-1.0.4/src/uid_mapping.hpp	2023-04-26 16:39:35.000000000 +0100
+++ krakenuniq-1.0.4.new/src/uid_mapping.hpp	2026-02-19 11:25:18.639379105 +0000
@@ -20,6 +20,7 @@
 #ifndef UID_MAPPING_H
 #define UID_MAPPING_H
 
+#include<cstdint>
 #include<vector>
 #include<map>
 #include<unordered_map>
