fix for "error: expression must have a constant value"
author: Kenneth Hoste (HPC-UGent)
--- llvm-5.0.1.src/lib/IR/Attributes.cpp.orig	2017-06-23 01:22:36.000000000 +0200
+++ llvm-5.0.1.src/lib/IR/Attributes.cpp	2018-01-11 10:35:27.342109503 +0100
@@ -810,8 +810,8 @@
   static_assert(Attribute::EndAttrKinds <=
                     sizeof(AvailableFunctionAttrs) * CHAR_BIT,
                 "Too many attributes");
-  static_assert(attrIdxToArrayIdx(AttributeList::FunctionIndex) == 0U,
-                "function should be stored in slot 0");
+  //static_assert(attrIdxToArrayIdx(AttributeList::FunctionIndex) == 0U,
+  //              "function should be stored in slot 0");
   for (Attribute I : Sets[0]) {
     if (!I.isStringAttribute())
       AvailableFunctionAttrs |= 1ULL << I.getKindAsEnum();
