diff -Naur grib_api-1.28.0-Source_orig/src/grib_jasper_encoding.c grib_api-1.28.0-Source/src/grib_jasper_encoding.c
--- grib_api-1.28.0-Source_orig/src/grib_jasper_encoding.c	2019-01-22 22:37:33.793747123 +0000
+++ grib_api-1.28.0-Source/src/grib_jasper_encoding.c	2019-01-22 22:39:06.914932578 +0000
@@ -110,7 +110,7 @@
     image.maxcmpts_ = 1;
     image.clrspc_   = JAS_CLRSPC_SGRAY;
     image.cmprof_   = 0;
-    image.inmem_    = 1;
+    
 
     cmpt.tlx_       = 0;
     cmpt.tly_       = 0;
