Fix CMake check for Szip/SZ

Åke Sandgren, 20200310
diff -ru ADIOS-1.13.1.orig/CMakeLists.txt ADIOS-1.13.1/CMakeLists.txt
--- ADIOS-1.13.1.orig/CMakeLists.txt	2018-04-17 19:23:13.000000000 +0200
+++ ADIOS-1.13.1/CMakeLists.txt	2020-03-10 09:15:55.755501700 +0100
@@ -523,7 +523,7 @@
     set(SZIP_DIR $ENV{SZIP_DIR})
   endif()
 elseif(DEFINED ENV{SZIP})
-  if(NOT "$ENV{SZIP}" STREQUAL "")
+  if("$ENV{SZIP}" STREQUAL "")
     set(SZIP OFF CACHE BOOL "")
   else()
     set(SZIP_DIR $ENV{SZIP})
@@ -639,7 +639,7 @@
     set(SZ_DIR $ENV{SZ_DIR})
   endif()
 elseif(DEFINED ENV{SZ})
-  if(NOT "$ENV{SZ}" STREQUAL "")
+  if("$ENV{SZ}" STREQUAL "")
     set(SZ OFF CACHE BOOL "")
   else()
     set(SZ_DIR $ENV{SZ})
