diff -ru petsc-3.1-p8.orig/config/install.py petsc-3.1-p8/config/install.py
--- petsc-3.1-p8.orig/config/install.py	2018-03-22 17:42:46.667882869 +0000
+++ petsc-3.1-p8/config/install.py	2018-03-22 17:44:03.000000000 +0000
@@ -251,8 +251,8 @@
     fd = file(os.path.join('conf','petscvariables'),'w')
     fd.close()
     # if running as root then change file ownership back to user
-    if os.environ.has_key('SUDO_USER'):
-      os.chown(os.path.join(self.rootDir,'conf','petscvariables'),int(os.environ['SUDO_UID']),int(os.environ['SUDO_GID']))
+    #if os.environ.has_key('SUDO_USER'):
+    #  os.chown(os.path.join(self.rootDir,'conf','petscvariables'),int(os.environ['SUDO_UID']),int(os.environ['SUDO_GID']))
     self.createUninstaller()
     self.outputHelp()
     return
