diff -ru petsc-3.3-p2.orig/config/BuildSystem/config/setCompilers.py petsc-3.3-p2/config/BuildSystem/config/setCompilers.py
--- petsc-3.3-p2.orig/config/BuildSystem/config/setCompilers.py	2012-05-15 18:44:53.000000000 +0200
+++ petsc-3.3-p2/config/BuildSystem/config/setCompilers.py	2012-08-08 08:49:28.439031948 +0200
@@ -1060,7 +1060,7 @@
       yield ('win32fe tlib',self.getArchiverFlags('win32fe tlib'),envRanlib)
       yield ('win32fe lib',self.getArchiverFlags('win32fe lib'),envRanlib)
       raise RuntimeError('You set -RANLIB="'+envRanlib+'" (perhaps in your environment), but '+defaultRanlib+' cannot be used\n')
-    yield ('ar',self.getArchiverFlags('ar'),'ranlib -c')
+    #yield ('ar',self.getArchiverFlags('ar'),'ranlib -c')
     yield ('ar',self.getArchiverFlags('ar'),'ranlib')
     yield ('ar',self.getArchiverFlags('ar'),'true')
     # IBM with 64 bit pointers 
