diff -Nru petsc-3.19.4.orig/config/BuildSystem/config/setCompilers.py petsc-3.19.4/config/BuildSystem/config/setCompilers.py
--- petsc-3.19.4.orig/config/BuildSystem/config/setCompilers.py	2023-07-31 17:00:55.000000000 +0200
+++ petsc-3.19.4/config/BuildSystem/config/setCompilers.py	2023-08-16 12:24:03.847163699 +0200
@@ -2181,7 +2181,7 @@
       path = os.path.join(os.getcwd(),'lib','petsc','bin')
       war  = os.path.join(path,'win_lib')
       yield (war,self.getArchiverFlags(war),'true')
-    yield ('ar',self.getArchiverFlags('ar'),'ranlib -c')
+    #yield ('ar',self.getArchiverFlags('ar'),'ranlib -c')
     yield ('ar',self.getArchiverFlags('ar'),'ranlib')
     yield ('ar',self.getArchiverFlags('ar'),'true')
     # IBM with 64-bit pointers
