diff -ru petsc-3.3-p2.orig/config/PETSc/packages/SCALAPACK.py petsc-3.3-p2/config/PETSc/packages/SCALAPACK.py
--- petsc-3.3-p2.orig/config/PETSc/packages/SCALAPACK.py	2012-05-22 23:20:13.000000000 +0200
+++ petsc-3.3-p2/config/PETSc/packages/SCALAPACK.py	2013-03-28 12:05:49.073099369 +0100
@@ -20,7 +20,7 @@
     PETSc.package.NewPackage.setupDependencies(self, framework)
     self.blasLapack = framework.require('config.packages.BlasLapack',self)
     self.blacs      = framework.require('PETSc.packages.blacs',self)
-    self.deps       = [self.blacs, self.mpi, self.blasLapack]
+    self.deps       = [self.mpi, self.blasLapack]
     return
 
   def Install(self):
