--- config/BuildSystem/config/setCompilers.py.orig	2015-07-23 03:22:46.000000000 +1200
+++ config/BuildSystem/config/setCompilers.py	2015-08-13 11:26:45.043731899 +1200
@@ -1138,6 +1138,7 @@
     arcUnix    = os.path.join(self.tmpDir, 'libconf1.a')
     arcWindows = os.path.join(self.tmpDir, 'libconf1.lib')
     def checkArchive(command, status, output, error):
+      error = '\n'.join([errx for errx in error.split('\n') if "no version information available" not in errx])
       if error or status:
         self.logError('archiver', status, output, error)
         if os.path.isfile(objName):
