diff -urN libspatialite-4.1.1.orig/configure libspatialite-4.1.1/configure
--- libspatialite-4.1.1.orig/configure	2013-06-29 08:53:15.000000000 +0200
+++ libspatialite-4.1.1/configure	2014-04-02 11:37:56.484423000 +0200
@@ -263,7 +263,7 @@
   LIBS="$GEOS_LDFLAGS"
   AC_SEARCH_LIBS(GEOSTopologyPreserveSimplify,geos_c,,AC_MSG_ERROR([could not find libgeos_c - you may need to specify the directory of a geos-config file using --with-geosconfig]))
   LIBS="$LIBS_SAVE"
-  LIBS=$LIBS$GEOS_LDFLAGS' -lgeos_c'
+  LIBS="$LIBS $GEOS_LDFLAGS -lgeos_c"
 
   #-----------------------------------------------------------------------
   #   --enable-geosadvanced
