--- opal/etc/openmpi-mca-params.conf.orig	2009-12-08 21:36:08.000000000 +0100
+++ opal/etc/openmpi-mca-params.conf	2013-03-17 08:13:21.216969244 +0100
@@ -56,3 +56,8 @@
 
 # See "ompi_info --param all all" for a full listing of Open MPI MCA
 # parameters available and their default values.
+
+# FG, 20130317; customization for University of Luxembourg systems
+btl_openib_if_exclude=mlx4_1
+btl_openib_warn_nonexistent_if=0
+orte_rsh_agent=oarsh
