diff -ur openmpi-1.6.5.orig/ompi/mca/btl/openib/mca-btl-openib-device-params.ini openmpi-1.6.5/ompi/mca/btl/openib/mca-btl-openib-device-params.ini
--- openmpi-1.6.5.orig/ompi/mca/btl/openib/mca-btl-openib-device-params.ini	2013-06-14 15:22:51.000000000 +0000
+++ openmpi-1.6.5/ompi/mca/btl/openib/mca-btl-openib-device-params.ini	2018-03-21 13:04:56.026010407 +0000
@@ -2,7 +2,7 @@
 # Copyright (c) 2006-2013 Cisco Systems, Inc.  All rights reserved.
 # Copyright (c) 2006-2011 Mellanox Technologies. All rights reserved.
 # $COPYRIGHT$
-# 
+#
 # Additional copyrights may follow
 #
 
@@ -127,7 +127,7 @@
 ############################################################################
 
 [Mellanox Tavor Infinihost]
-vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba
+vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba,0x15b3
 vendor_part_id = 23108
 use_eager_rdma = 1
 mtu = 1024
@@ -136,7 +136,7 @@
 ############################################################################
 
 [Mellanox Arbel InfiniHost III MemFree/Tavor]
-vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba
+vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba,0x15b3
 vendor_part_id = 25208,25218
 use_eager_rdma = 1
 mtu = 1024
@@ -145,7 +145,7 @@
 ############################################################################
 
 [Mellanox Sinai Infinihost III]
-vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba
+vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba,0x15b3
 vendor_part_id = 25204,24204
 use_eager_rdma = 1
 mtu = 2048
@@ -156,7 +156,7 @@
 # A.k.a. ConnectX
 [Mellanox Hermon]
 vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba,0x15b3,0x119f
-vendor_part_id = 25408,25418,25428,25448,26418,26428,26438,26448,26468,26478,26488,4099
+vendor_part_id = 25408,25418,25428,25448,26418,26428,26438,26448,26468,26478,26488,4099,4103,4100
 use_eager_rdma = 1
 mtu = 2048
 max_inline_data = 128
@@ -172,6 +172,24 @@
 
 ############################################################################
 
+[Mellanox ConnectX4]
+vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba,0x15b3,0x119f
+vendor_part_id = 4115
+use_eager_rdma = 1
+mtu = 4096
+max_inline_data = 256
+
+############################################################################
+
+[Mellanox ConnectX5]
+vendor_id = 0x2c9,0x5ad,0x66a,0x8f1,0x1708,0x03ba,0x15b3,0x119f
+vendor_part_id = 4119,4121
+use_eager_rdma = 1
+mtu = 4096
+max_inline_data = 256
+
+############################################################################
+
 [IBM eHCA 4x and 12x]
 vendor_id = 0x5076
 vendor_part_id = 0
@@ -244,6 +262,14 @@
 receive_queues = P,65536,64
 max_inline_data = 280
 
+[Chelsio T6]
+vendor_id = 0x1425
+vendor_part_id = 0x6400,0x6401,0x6402,0x6403,0x6404,0x6405,0x6406,0x6407,0x6408,0x6409,0x640d,0x6410,0x6411,0x6414,0x6415
+use_eager_rdma = 1
+mtu = 2048
+receive_queues = P,65536,64
+max_inline_data = 280
+
 ############################################################################
 
 # I'm *assuming* that 0x4040 is the PCI ID...
@@ -269,6 +295,13 @@
 receive_queues = P,65536,256,192,128
 max_inline_data = 64
 
+[Intel HFI1]
+vendor_id = 0x1175
+vendor_part_id = 9456,9457
+use_eager_rdma = 1
+mtu = 4096
+max_inline_data = 0
+
 ############################################################################
 
 # Intel has several OUI's, including 0x8086.  Amusing.  :-) Intel has
@@ -279,3 +312,15 @@
 vendor_id = 0x8086
 vendor_part_id = 0
 ignore_device = 1
+
+############################################################################
+
+# IBM Soft iWARP device.
+
+[IBM Soft iWARP]
+vendor_id =  0x626d74
+vendor_part_id = 0
+use_eager_rdma = 1
+mtu = 2048
+receive_queues = P,65536,64
+max_inline_data = 72
