--- opal/etc/openmpi-mca-params.conf.orig 2015-06-05 16:21:25.535727000 +0200
+++ opal/etc/openmpi-mca-params.conf  2015-06-08 10:36:05.387902000 +0200
@@ -56,3 +56,6 @@
 
 # See "ompi_info --param all all" for a full listing of Open MPI MCA
 # parameters available and their default values.
+
+# Fix bug in the initialization of the value when using the infiniband module. (ref: http://www.open-mpi.org/community/lists/users/2015/05/26913.php)
+btl_openib_memalign_threshold=12288
