--- openmpi-1.7.3/ompi/contrib/vt/vt/vtlib/vt_cudart.c.orig	2014-09-11 13:02:17.089131966 -0700
+++ openmpi-1.7.3/ompi/contrib/vt/vt/vtlib/vt_cudart.c	2014-09-11 13:02:41.503307372 -0700
@@ -2722,13 +2722,13 @@
           /*  get an already created unused event */
           kernel->evt = vtDev->evtbuf_pos;
 
+#if defined(VT_CUPTI_EVENTS)
           if(!vt_cupti_events_enabled){
             /* increment buffers */
             vtDev->evtbuf_pos++;
             vtDev->buf_pos += sizeof(VTCUDAKernel);
           }
 
-#if defined(VT_CUPTI_EVENTS)
         
         /* zero CUPTI counter */
         if(vt_cupti_events_enabled){
