diff -ru loki-0.1.7.orig/Makefile.common loki-0.1.7/Makefile.common
--- loki-0.1.7.orig/Makefile.common	2007-05-28 22:26:26.000000000 -0400
+++ loki-0.1.7/Makefile.common	2016-04-21 18:20:54.280476472 -0400
@@ -4,4 +4,4 @@
 CXXWARNFLAGS := -Wall -Wold-style-cast -Wundef -Wsign-compare -Wconversion -Wpointer-arith -pedantic
 CXXFLAGS := $(CXXWARNFLAGS) -g -O2
 
-prefix := /usr
+prefix := $(EB_INSTALL_DIR)
