diff -ru fsl.orig/extras/src/cprob/const.c fsl/extras/src/cprob/const.c
--- fsl.orig/extras/src/cprob/const.c	2006-08-21 18:47:10.000000000 +0200
+++ fsl/extras/src/cprob/const.c	2012-08-02 15:58:31.812167292 +0200
@@ -89,12 +89,12 @@
 double THPIO4 =  2.35619449019234492885;       /* 3*pi/4 */
 double TWOOPI =  6.36619772367581343075535E-1; /* 2/pi */
 #ifdef INFINITIES
-double INFINITY = 1.0/0.0;  /* 99e999; */
+double INFINITY = __builtin_inff();
 #else
 double INFINITY =  1.79769313486231570815E308;    /* 2**1024*(1-MACHEP) */
 #endif
 #ifdef NANS
-double NAN = 1.0/0.0 - 1.0/0.0;
+double NAN = __builtin_nanf("");
 #else
 double NAN = 0.0;
 #endif
diff -ru fsl.orig/src/fslview/fsl/cprob/const.c fsl/src/fslview/fsl/cprob/const.c
--- fsl.orig/src/fslview/fsl/cprob/const.c	2006-08-21 18:47:10.000000000 +0200
+++ fsl/src/fslview/fsl/cprob/const.c	2012-08-02 15:18:48.404398489 +0200
@@ -89,12 +89,12 @@
 double THPIO4 =  2.35619449019234492885;       /* 3*pi/4 */
 double TWOOPI =  6.36619772367581343075535E-1; /* 2/pi */
 #ifdef INFINITIES
-double INFINITY = 1.0/0.0;  /* 99e999; */
+double INFINITY = __builtin_inff();
 #else
 double INFINITY =  1.79769313486231570815E308;    /* 2**1024*(1-MACHEP) */
 #endif
 #ifdef NANS
-double NAN = 1.0/0.0 - 1.0/0.0;
+double NAN = __builtin_nanf("");
 #else
 double NAN = 0.0;
 #endif
