--- OPERA-LG_v2.0.6/src/Makefile.orig	2019-01-10 18:57:13.775420038 +0100
+++ OPERA-LG_v2.0.6/src/Makefile	2019-01-10 18:57:55.155829905 +0100
@@ -5,12 +5,15 @@
 
 install : $(PROGRAM)
 
+CXX := g++
+CXXFLAGS := -O3
+
 $(PROGRAM) : $(OBJS)     
-	g++ -c -O3 -w $(CPP) $<
+	$(CXX) -c $(CXXFLAGS) -w $(CPP) $<
 #	g++ -c -g -fno-inline -w $(CPP) $<
 #	g++ -c -g -fno-inline -w $(CPP) $< -pg
 #	g++ -o ../bin/$(PROGRAM) *.o -pg
-	g++ -o ../bin/$(PROGRAM) *.o
+	$(CXX) $(CXXFLAGS) -o ../bin/$(PROGRAM) *.o
 #	rm *.o
 
 clean:
