Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Description: Fix version in setup.py
diff -ruN charade-elsa-d50cfb060eba.p1/setup.py charade-elsa-d50cfb060eba.p2/setup.py
--- charade-elsa-d50cfb060eba.p1/setup.py	2016-09-07 07:10:48.000000000 +0100
+++ charade-elsa-d50cfb060eba.p2/setup.py	2022-07-12 16:39:26.000000000 +0100
@@ -73,7 +73,7 @@
         ('build_scripts', build.build.has_scripts), ] 
 
 setup(name="lsa",
-    version="1.0.2",
+    version="1.0.6",
     description=doclines[0],
     long_description="\n".join(doclines[2:]),
     author="Li Charlie Xia",
