--- likwid-likwid-4.2.orig/config.mk	2016-12-22 13:22:23.000000000 +0100
+++ likwid-likwid-4.2/config.mk	2017-06-22 10:14:28.585696000 +0200
@@ -1,7 +1,7 @@
 # Please have a look in INSTALL and the WIKI for details on
 # configuration options setup steps.
 # supported: GCC, CLANG, ICC, MIC (ICC), GCCX86 (for 32bit systems)
-COMPILER = GCC#NO SPACE
+COMPILER = ICC#NO SPACE
 
 # Define the color of the likwid-pin output
 # Can be NONE, BLACK, RED, GREEN, YELLOW, BLUE,
@@ -9,7 +9,7 @@
 COLOR = BLUE#NO SPACE
 
 # Path were to install likwid
-PREFIX = /usr/local#NO SPACE
+# PREFIX = /usr/local#NO SPACE
 
 # uncomment to optionally set external lua@5.3:
 # default is use internally provide lua
@@ -40,7 +40,7 @@
 # chown installed tools to this user/group
 # if you change anything here, make sure that the user/group can access
 # the MSR devices and (on Intel) the PCI devices.
-INSTALL_CHOWN = -g root -o root
+INSTALL_CHOWN =
 
 # For the daemon based secure msr/pci access configure
 # the absolute path to the msr daemon executable.
@@ -51,7 +51,7 @@
 # Build the accessDaemon. Have a look in the WIKI for details.
 BUILDDAEMON = true#NO SPACE
 #Build the setFrequencies tool
-BUILDFREQ = true#NO SPACE
+BUILDFREQ = false#NO SPACE
 
 # Set the default mode for MSR access.
 # This can usually be overriden on the commandline.
@@ -82,8 +82,8 @@
 # a proper config file at CFG_FILE_PATH)
 MAX_NUM_THREADS = 263
 MAX_NUM_NODES = 64
-CFG_FILE_PATH = /etc/likwid.cfg
-TOPO_FILE_PATH = /etc/likwid_topo.cfg
+CFG_FILE_PATH = $(PREFIX)/etc/likwid.cfg
+TOPO_FILE_PATH = $(PREFIX)/etc/likwid_topo.cfg
 
 # Versioning Information
 VERSION = 4
