diff -ru netcdf-fortran-4.2.orig/configure netcdf-fortran-4.2/configure
--- netcdf-fortran-4.2.orig/configure   2011-10-03 16:10:43.000000000 +0200
+++ netcdf-fortran-4.2/configure    2012-06-12 16:48:55.061255254 +0200
@@ -8749,7 +8749,7 @@
 
 
 
-  lt_prog_compiler_wl=
+  lt_prog_compiler_wl='-Wl,'
 lt_prog_compiler_pic=
 lt_prog_compiler_static=
 
@@ -12198,7 +12198,7 @@
     ## There is no encapsulation within the following macros, do not change
     ## the running order or otherwise move them around unless you know exactly
     ## what you are doing...
-    lt_prog_compiler_wl_F77=
+    lt_prog_compiler_wl_F77='-Wl,'
 lt_prog_compiler_pic_F77=
 lt_prog_compiler_static_F77=
 
@@ -14901,7 +14901,7 @@
 
 
 
-    lt_prog_compiler_wl_FC=
+    lt_prog_compiler_wl_FC='-Wl,'
 lt_prog_compiler_pic_FC=
 lt_prog_compiler_static_FC=
