diff -Nru TrimGalore-39e1ad7c90108baa618e425ba1c83c78f03d6da2.orig/trim_galore TrimGalore-39e1ad7c90108baa618e425ba1c83c78f03d6da2/trim_galore
--- TrimGalore-39e1ad7c90108baa618e425ba1c83c78f03d6da2.orig/trim_galore	2019-03-01 13:55:31.000000000 +0000
+++ TrimGalore-39e1ad7c90108baa618e425ba1c83c78f03d6da2/trim_galore	2019-04-05 12:07:11.118812010 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 use strict;
 use warnings;
 use Getopt::Long;
