diff -ur /home/brob695/sixs-master/setup.py sixs-master/setup.py
--- /home/brob695/sixs-master/setup.py	2012-05-23 03:15:56.000000000 +1200
+++ sixs-master/setup.py	2016-12-19 16:38:51.529832000 +1300
@@ -29,7 +29,7 @@
         "sixs/IAPIBRDF.f", "sixs/IAPITOOLS.f", \
         "sixs/INTERP.f", "sixs/ISO.f", \
         "sixs/KERNEL.f", "sixs/KERNELPOL.f", \
-        "sixs/LAKEW.f", \
+        "sixs/LAKEW.f", "sixs/main.f", \
         "sixs/main_ssssss.f", "sixs/MAS.f", \
         "sixs/MERIS.f", "sixs/METEO.f", \
         "sixs/METH1.f", "sixs/METH2.f", \
