--- WRFV3/external/RSL_LITE/rsl_lite.h	2012-06-19 14:22:32.904311023 +0200
+++ WRFV3.orig/external/RSL_LITE/rsl_lite.h	2008-12-27 18:56:01.000000000 +0100
@@ -149,9 +149,9 @@
 #endif
 
 #ifndef MPI2_SUPPORT
-
-#  define MPI_Comm_c2f(comm) (int)(comm)
-#  define MPI_Comm_f2c(comm) (int)(comm)
+typedef int MPI_Fint;
+#  define MPI_Comm_c2f(comm) (MPI_Fint)(comm)
+#  define MPI_Comm_f2c(comm) (MPI_Comm)(comm)
 #endif
 
 typedef struct rsl_list {
