diff -ru WRF-4.2.1.orig-patched/configure WRF-4.2.1/configure
--- WRF-4.2.1.orig-patched/configure	2020-12-18 18:39:42.879404253 +0000
+++ WRF-4.2.1/configure	2020-12-18 18:51:50.174814854 +0000
@@ -989,11 +989,11 @@
 fi
 
 # testing for location of rpc/types.h file, used in landuse
-if [ -f /usr/include/rpc/types.h ] ; then
+if [ -f $EBROOTGENTOO/include/rpc/types.h ] ; then
   sed -e '/^ARCH_LOCAL/s/$/ -DRPC_TYPES=1/' configure.wrf > configure.wrf.edit
   mv configure.wrf.edit configure.wrf
 echo standard location of RPC
-elif [ -f /usr/include/tirpc/rpc/types.h ] ; then
+elif [ -f $EBROOTGENTOO/include/tirpc/rpc/types.h ] ; then
   sed -e '/^ARCH_LOCAL/s/$/ -DRPC_TYPES=2/' configure.wrf > configure.wrf.edit
   mv configure.wrf.edit configure.wrf
 echo newer location of RPC
