Fix a couple of smaller problems

Åke Sandgren, 2024-11-07
diff -ru gromacs-ls-2016.3.orig/cmake/gmxTestCXX11.cmake gromacs-ls-2016.3/cmake/gmxTestCXX11.cmake
--- gromacs-ls-2016.3.orig/cmake/gmxTestCXX11.cmake	2019-12-28 15:42:21.000000000 +0100
+++ gromacs-ls-2016.3/cmake/gmxTestCXX11.cmake	2024-11-07 10:45:40.060210373 +0100
@@ -49,7 +49,7 @@
     elseif(CYGWIN)
         set(CXX11_CXX_FLAG "-std=gnu++0x") #required for strdup
     else()
-        set(CXX11_CXX_FLAG "-std=c++0x")
+        set(CXX11_CXX_FLAG "-std=c++11")
     endif()
     CHECK_CXX_COMPILER_FLAG("${CXX11_CXX_FLAG}" CXXFLAG_STD_CXX0X)
     if(NOT CXXFLAG_STD_CXX0X)
diff -ru gromacs-ls-2016.3.orig/src/gromacs/mdlib/minimize.cpp gromacs-ls-2016.3/src/gromacs/mdlib/minimize.cpp
--- gromacs-ls-2016.3.orig/src/gromacs/mdlib/minimize.cpp	2019-12-28 15:42:28.000000000 +0100
+++ gromacs-ls-2016.3/src/gromacs/mdlib/minimize.cpp	2024-11-07 10:55:06.668206192 +0100
@@ -54,6 +54,7 @@
 
 #include <algorithm>
 #include <vector>
+#include <limits>
 
 #include "gromacs/commandline/filenm.h"
 #include "gromacs/domdec/domdec.h"
diff -ru gromacs-ls-2016.3.orig/src/programs/mdrun/runner.cpp gromacs-ls-2016.3/src/programs/mdrun/runner.cpp
--- gromacs-ls-2016.3.orig/src/programs/mdrun/runner.cpp	2019-12-28 15:42:30.000000000 +0100
+++ gromacs-ls-2016.3/src/programs/mdrun/runner.cpp	2024-11-07 10:33:22.588969615 +0100
@@ -175,7 +175,7 @@
     int                     localsspatialatom;
     int                     localsdispcor;
     int                     localspbc;
-    real                    localsmindihang;
+    real                    localsmindihangle;
     unsigned long           Flags;
 };
 
