diff -Nru extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh
--- extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh	2024-07-01 16:12:03.454963219 +0200
+++ extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_CYC.sh	2024-07-02 10:26:41.603395579 +0200
@@ -11,6 +11,20 @@
 ../../../src/merger/mpi2prv -f TRACE.mpits -o ${TRACE}.prv
 
 # Check
+if ! command -v papi_avail &> /dev/null
+then
+        echo "papi_avail could not be found"
+        exit 0
+fi
+
+PAPI_TOT_CYC_available=`papi_avail | grep PAPI_TOT_CYC | awk '{print $3}'`
+if [[ "$PAPI_TOT_CYC_available" == No ]]
+then
+        echo "PAPI_TOT_CYC is not available"
+        exit 0
+fi
+
+
 CheckEntryInPCF ${TRACE}.pcf PAPI_TOT_CYC
 
 rm -fr TRACE* set-0 ${TRACE}.???
diff -Nru extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS.sh extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS.sh
--- extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS.sh	2024-07-01 16:12:03.455963214 +0200
+++ extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS.sh	2024-07-02 10:28:14.723917087 +0200
@@ -11,6 +11,20 @@
 ../../../src/merger/mpi2prv -f TRACE.mpits -o ${TRACE}.prv
 
 # Check
+if ! command -v papi_avail &> /dev/null
+then
+        echo "papi_avail could not be found"
+        exit 0
+fi
+
+PAPI_TOT_INS_available=`papi_avail | grep PAPI_TOT_INS | awk '{print $3}'`
+if [[ "$PAPI_TOT_INS_available" == No ]]
+then
+        echo "PAPI_TOT_INS is not available"
+        exit 0
+fi
+
+
 CheckEntryInPCF ${TRACE}.pcf PAPI_TOT_INS
 
 rm -fr TRACE* set-0 ${TRACE}.???
diff -Nru extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS_CYC.sh extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS_CYC.sh
--- extrae-4.2.0.orig/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS_CYC.sh	2024-07-01 16:12:03.448963249 +0200
+++ extrae-4.2.0/tests/functional/hw-counters/check_Extrae_PAPI_TOT_INS_CYC.sh	2024-07-02 10:29:00.084684067 +0200
@@ -11,6 +11,20 @@
 ../../../src/merger/mpi2prv -f TRACE.mpits -o ${TRACE}.prv
 
 # Check
+if ! command -v papi_avail &> /dev/null
+then
+        echo "papi_avail could not be found"
+        exit 0
+fi
+
+PAPI_TOT_CYC_available=`papi_avail | grep PAPI_TOT_CYC | awk '{print $3}'`
+if [[ "$PAPI_TOT_CYC_available" == No ]]
+then
+        echo "PAPI_TOT_CYC is not available"
+        exit 0
+fi
+
+
 CheckEntryInPCF ${TRACE}.pcf PAPI_TOT_INS
 CheckEntryInPCF ${TRACE}.pcf PAPI_TOT_CYC
 
diff -Nru extrae-4.2.0.orig/tests/functional/xml/check_Extrae_xml_envvar_counters.sh extrae-4.2.0/tests/functional/xml/check_Extrae_xml_envvar_counters.sh
--- extrae-4.2.0.orig/tests/functional/xml/check_Extrae_xml_envvar_counters.sh	2024-07-01 16:12:03.484963068 +0200
+++ extrae-4.2.0/tests/functional/xml/check_Extrae_xml_envvar_counters.sh	2024-07-02 10:19:24.415644506 +0200
@@ -11,6 +11,20 @@
 ../../../src/merger/mpi2prv -f TRACE.mpits -o ${TRACE}.prv
 
 # Check
+if ! command -v papi_avail &> /dev/null
+then
+        echo "papi_avail could not be found"
+        exit 0
+fi
+
+PAPI_TOT_CYC_available=`papi_avail | grep PAPI_TOT_CYC | awk '{print $3}'`
+if [[ "$PAPI_TOT_CYC_available" == No ]]
+then
+        echo "PAPI_TOT_CYC is not available"
+        exit 0
+fi
+
+
 CheckEntryInPCF ${TRACE}.pcf PAPI_TOT_INS
 
 rm -fr TRACE* set-0 ${TRACE}.???
