fix configure tests for libbfd.a
author: Bernd Mohr (Juelich Supercomputing Centre, Germany)
--- configure.orig	2013-11-11 16:38:11.872367495 +0100
+++ configure	2013-11-11 16:38:32.872767289 +0100
@@ -27477,9 +27477,9 @@
             BFD_LIBSDIR="${binutils_home_dir}/lib${BITS}"
          elif test -r "${binutils_home_dir}/lib/libbfd.so" ; then
             BFD_LIBSDIR="${binutils_home_dir}/lib"
-         elif test -r "${binutils_home_dir}/lib${BITS}/libbfd.a" -a "${binutils_require_shared}" = "no" ; then
+         elif test -r "${binutils_home_dir}/lib${BITS}/libbfd.a"; then #-a "${binutils_require_shared}" = "no" ; then
             BFD_LIBSDIR="${binutils_home_dir}/lib${BITS}"
-         elif test -r "${binutils_home_dir}/lib/libbfd.a" -a "${binutils_require_shared}" = "no" ; then
+         elif test -r "${binutils_home_dir}/lib/libbfd.a"; then # -a "${binutils_require_shared}" = "no" ; then
             BFD_LIBSDIR="${binutils_home_dir}/lib"
          else
                                     if test -d ${binutils_home_dir}/lib${BITS} ; then
