diff -Nru extrae-4.2.0.orig/config/macros.m4 extrae-4.2.0/config/macros.m4
--- extrae-4.2.0.orig/config/macros.m4	2024-07-01 16:12:03.689962036 +0200
+++ extrae-4.2.0/config/macros.m4	2024-07-01 16:13:05.811649165 +0200
@@ -779,6 +779,8 @@
          elif test -r "${binutils_home_dir}/lib/libbfd.a" -a \
                       "${binutils_require_shared}" = "no" ; then
             BFD_LIBSDIR="${binutils_home_dir}/lib"
+	 elif test -r "${binutils_home_dir}/libbfd.so" ; then
+            BFD_LIBSDIR="${binutils_home_dir}"
          else
             dnl Try something more automatic using find command
             libbfd_lib=""
@@ -814,6 +816,8 @@
             LIBERTY_LIBSDIR="${binutils_home_dir}/lib"
          elif test -r "${binutils_home_dir}/lib/libiberty.a" ; then
             LIBERTY_LIBSDIR="${binutils_home_dir}/lib"
+	elif test -r "${binutils_home_dir}/libiberty.a" ; then
+            LIBERTY_LIBSDIR="${binutils_home_dir}"
          else
             dnl Try something more automatic using find command
             libiberty_lib=""
diff -Nru extrae-4.2.0.orig/configure extrae-4.2.0/configure
--- extrae-4.2.0.orig/configure	2024-07-01 16:12:03.308963954 +0200
+++ extrae-4.2.0/configure	2024-07-01 16:17:00.458465744 +0200
@@ -35074,6 +35074,8 @@
          elif test -r "${binutils_home_dir}/lib/libbfd.a" -a \
                       "${binutils_require_shared}" = "no" ; then
             BFD_LIBSDIR="${binutils_home_dir}/lib"
+	 elif test -r "${binutils_home_dir}/libbfd.so" ; then
+            BFD_LIBSDIR="${binutils_home_dir}"
          else
                         libbfd_lib=""
 
@@ -35108,6 +35110,8 @@
             LIBERTY_LIBSDIR="${binutils_home_dir}/lib"
          elif test -r "${binutils_home_dir}/lib/libiberty.a" ; then
             LIBERTY_LIBSDIR="${binutils_home_dir}/lib"
+	 elif test -r "${binutils_home_dir}/libiberty.a" ; then
+            LIBERTY_LIBSDIR="${binutils_home_dir}"
          else
                         libiberty_lib=""
 
