diff -Nru extrae-4.2.0.orig/src/tracer/hwc/papi_hwc.c extrae-4.2.0/src/tracer/hwc/papi_hwc.c
--- extrae-4.2.0.orig/src/tracer/hwc/papi_hwc.c	2024-07-01 16:12:04.562957639 +0200
+++ extrae-4.2.0/src/tracer/hwc/papi_hwc.c	2024-07-02 17:21:19.783586061 +0200
@@ -615,7 +615,7 @@
 						char EventName[PAPI_MAX_STR_LEN];
 
 						PAPI_event_code_to_name (HWC_sets[i].counters[j], EventName);
-						fprintf (stderr, PACKAGE_NAME": Error! Hardware counter %s (0x%08x) cannot be added in set %d (task %d, thread %d)\n", EventName, HWC_sets[i].counters[j], i+1, TASKID, threadid);
+						fprintf (stderr, PACKAGE_NAME": Error! Hardware counter %s (0x%08x) cannot be added in set %d (task %d, thread %d) because of %d\n", EventName, HWC_sets[i].counters[j], i+1, TASKID, threadid, rc);
 						HWC_sets[i].counters[j] = NO_COUNTER;
 						/* break; */
 					}
