fix wrong include statement
author: Kenneth Hoste (HPC-UGent)
--- ngs-tools-2.9.1/tools/skesa/readsgetter.hpp.orig	2018-07-04 14:19:19.740757775 +0200
+++ ngs-tools-2.9.1/tools/skesa/readsgetter.hpp	2018-07-04 14:19:40.110721460 +0200
@@ -27,7 +27,7 @@
 #ifndef _ReadsGetter_
 #define _ReadsGetter_
 
-#include <ngs/ncbi/NGS.hpp>
+#include <ncbi-vdb/NGS.hpp>
 #include <ngs/ErrorMsg.hpp>
 #include <ngs/ReadCollection.hpp>
 #include <ngs/ReadIterator.hpp>
