diff -Nru qualimap_v2.3.orig/qualimap qualimap_v2.3/qualimap
--- qualimap_v2.3.orig/qualimap	2024-12-13 12:47:50.335636162 -0500
+++ qualimap_v2.3/qualimap	2024-12-13 12:48:54.886990080 -0500
@@ -39,7 +39,7 @@
 	java_options=$JAVA_OPTS
 	echo "detected environment java options $java_options"
 else
-	java_options="-Xms32m -Xmx$JAVA_MEM_SIZE -XX:MaxPermSize=1024m"
+	java_options="-Xms32m -Xmx$JAVA_MEM_SIZE"
 fi
 
 
diff -Nru qualimap_v2.3.orig/qualimap_bash_fix qualimap_v2.3/qualimap_bash_fix
--- qualimap_v2.3.orig/qualimap_bash_fix	2024-12-13 12:47:50.290635915 -0500
+++ qualimap_v2.3/qualimap_bash_fix	2024-12-13 12:48:39.618906369 -0500
@@ -47,7 +47,7 @@
 	java_options=$JAVA_OPTS
 	echo "detected environment java options $java_options"
 else
-	java_options="-Xms32m -Xmx$JAVA_MEM_SIZE -XX:MaxPermSize=1024m"
+	java_options="-Xms32m -Xmx$JAVA_MEM_SIZE"
 fi
 
 
