Remove hard-coded hipcc paths to allow using rpath

Author: Jan Andre Reuter (j.reuter@fz-juelich.de)

diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/babel.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/babel.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/babel.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/babel.so/CMakeLists.txt	2025-06-25 10:31:06.899564996 +0200
@@ -64,8 +64,6 @@
 set(HIP_STREAM_BUILD_FLAGS "-DNONTEMPORAL=1 -DTBSIZE=1024 -O3 -std=c++17")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS} ${HIP_STREAM_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS} ${HIP_STREAM_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/edp.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/edp.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/edp.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/edp.so/CMakeLists.txt	2025-06-25 10:29:21.900362190 +0200
@@ -96,8 +96,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/gm.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/gm.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/gm.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/gm.so/CMakeLists.txt	2025-06-25 10:32:39.135304795 +0200
@@ -93,8 +93,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HSA_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/gpup.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/gpup.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/gpup.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/gpup.so/CMakeLists.txt	2025-06-25 10:30:41.920712083 +0200
@@ -93,8 +93,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/gst.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/gst.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/gst.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/gst.so/CMakeLists.txt	2025-06-25 10:30:09.450198087 +0200
@@ -95,8 +95,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HSA_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/iet.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/iet.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/iet.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/iet.so/CMakeLists.txt	2025-06-25 10:29:58.026719375 +0200
@@ -99,8 +99,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/mem.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/mem.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/mem.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/mem.so/CMakeLists.txt	2025-06-25 10:31:30.362484705 +0200
@@ -63,8 +63,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/pbqt.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/pbqt.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/pbqt.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/pbqt.so/CMakeLists.txt	2025-06-25 10:29:32.587877131 +0200
@@ -107,8 +107,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HSA_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/pebb.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/pebb.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/pebb.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/pebb.so/CMakeLists.txt	2025-06-25 10:32:14.368452079 +0200
@@ -108,8 +108,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HSA_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/peqt.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/peqt.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/peqt.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/peqt.so/CMakeLists.txt	2025-06-25 10:32:50.994754658 +0200
@@ -91,8 +91,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/perf.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/perf.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/perf.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/perf.so/CMakeLists.txt	2025-06-25 10:30:23.269566395 +0200
@@ -95,8 +95,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HSA_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/pesm.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/pesm.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/pesm.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/pesm.so/CMakeLists.txt	2025-06-25 10:30:52.416230494 +0200
@@ -91,8 +91,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HSA_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/rcqt.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/rcqt.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/rcqt.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/rcqt.so/CMakeLists.txt	2025-06-25 10:31:18.299040459 +0200
@@ -92,8 +92,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/rvs/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/rvs/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/rvs/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/rvs/CMakeLists.txt	2025-06-25 10:28:56.457513443 +0200
@@ -97,8 +97,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/rvslib/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/rvslib/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/rvslib/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/rvslib/CMakeLists.txt	2025-06-25 10:29:46.583240729 +0200
@@ -99,8 +99,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HSA_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 #Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/smqt.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/smqt.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/smqt.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/smqt.so/CMakeLists.txt	2025-06-25 10:31:54.897352426 +0200
@@ -90,8 +90,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HIP_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/testif.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/testif.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/testif.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/testif.so/CMakeLists.txt	2025-06-25 10:31:41.925951374 +0200
@@ -91,8 +91,6 @@
 set(HIP_HCC_BUILD_FLAGS "${HIP_HCC_BUILD_FLAGS} -fPIC ${HCC_CXX_FLAGS} -I${HIP_INC_DIR} ${ASAN_CXX_FLAGS}")
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
diff --color -rui ROCmValidationSuite-rocm-6.4.1.orig/tst.so/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/tst.so/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/tst.so/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/tst.so/CMakeLists.txt	2025-06-25 10:32:25.963915209 +0200
@@ -99,8 +99,6 @@
 
 
 # Set compiler and compiler flags
-set(CMAKE_CXX_COMPILER "${HIPCC_PATH}/bin/hipcc")
-set(CMAKE_C_COMPILER   "${HIPCC_PATH}/bin/hipcc")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HIP_HCC_BUILD_FLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ASAN_LD_FLAGS}")
