diff --color -Naur ROCmValidationSuite-rocm-6.4.1.orig/CMakeLists.txt ROCmValidationSuite-rocm-6.4.1/CMakeLists.txt
--- ROCmValidationSuite-rocm-6.4.1.orig/CMakeLists.txt	2025-04-28 22:05:34.000000000 +0200
+++ ROCmValidationSuite-rocm-6.4.1/CMakeLists.txt	2025-06-19 11:01:48.371711704 +0200
@@ -141,12 +141,11 @@
 # Making ROCM_PATH, CMAKE_INSTALL_PREFIX, CPACK_PACKAGING_INSTALL_PREFIX as CACHE
 # variables since we will pass them as cmake params appropriately, and 
 # all find_packages relevant to this build will be in ROCM path hence appending it to CMAKE_PREFIX_PATH 
-set(ROCM_PATH "/opt/rocm" CACHE PATH "ROCM install path")
-set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH "CMAKE installation directory")
-set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE PATH "Prefix used in built packages")
-list(APPEND CMAKE_PREFIX_PATH "${ROCM_PATH}")
+set(ROCM_PATH $ENV{EBROOTROCMMINLLVM} CACHE PATH "ROCM install path")
+# set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH "CMAKE installation directory")
+set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX} CACHE PATH "Prefix used in built packages")
 
-add_definitions(-DROCM_PATH="${ROCM_PATH}")
+add_definitions(-DROCM_PATH="${CMAKE_INSTALL_PREFIX}")
 if(FETCH_ROCMPATH_FROM_ROCMCORE)
   add_compile_options(-DFETCH_ROCMPATH_FROM_ROCMCORE=${FETCH_ROCMPATH_FROM_ROCMCORE})
   # rocm-core library will provide the ROCM_PATH during runtime
