diff -Nru OpenRefine-3.1.orig/refine.ini OpenRefine-3.1/refine.ini
--- OpenRefine-3.1.orig/refine.ini	2018-11-29 01:32:56.000000000 +0000
+++ OpenRefine-3.1/refine.ini	2019-05-15 12:52:34.648149497 +0000
@@ -21,3 +21,5 @@
 
 # Uncomment to increase autosave period to 60 mins (default: 5 minutes) for better performance of long-lasting transformations
 #REFINE_AUTOSAVE_PERIOD=60
+
+JAVA_OPTIONS="-Djava.security.egd=file:///dev/urandom -Drefine.headless=true"
