diff -Nru OpenRefine-2.8.orig/refine.ini OpenRefine-2.8/refine.ini
--- OpenRefine-2.8.orig/refine.ini	2017-11-19 04:02:06.000000000 +0000
+++ OpenRefine-2.8/refine.ini	2018-03-09 16:37:40.942909024 +0000
@@ -13,11 +13,7 @@
 # Set initial java heap space (default: 256M) for better performance with large datasets
 REFINE_MIN_MEMORY=1400M
 
-# Some sample configurations. These have no defaults.
-#ANT_HOME=C:\grefine\tools\apache-ant-1.8.1
-#JAVA_HOME=C:\Program Files\Java\jdk1.6.0_25
-#JAVA_OPTIONS=-XX:+UseParallelGC -verbose:gc -Drefine.headless=true
-#JAVA_OPTIONS=-Drefine.data_dir=C:\Users\user\AppData\Roaming\OpenRefine
+JAVA_OPTIONS="-Djava.security.egd=file:///dev/urandom -Drefine.headless=true"
 
 # Uncomment to increase autosave period to 60 mins (default: 5 minutes) for better performance of long-lasting transformations
 #REFINE_AUTOSAVE_PERIOD=60
