--- OpenRefine-3.4.1.orig/refine.ini	2021-03-05 20:05:30.611723165 +0000
+++ OpenRefine-3.4.1/refine.ini	2021-03-05 20:09:36.633992193 +0000
@@ -12,15 +12,12 @@
 
 # Memory and max form size allocations
 #REFINE_MAX_FORM_CONTENT_SIZE=1048576
-REFINE_MEMORY=1400M
+REFINE_MEMORY=$(free -m | grep -oP '\d+' | head -n 1)M
 
 # Set initial java heap space (default: 256M) for better performance with large datasets
-REFINE_MIN_MEMORY=1400M
+#REFINE_MIN_MEMORY=1400M
 
-# Some sample configurations. These have no defaults.
-#JAVA_HOME=C:\Program Files\Java\jdk1.8.0_151
-# Use a single JAVA_OPTIONS that includes any JVM options you need upon OpenRefine startup
-#JAVA_OPTIONS=-XX:+UseParallelGC -verbose:gc -Drefine.headless=true -Drefine.data_dir=C:\Users\user\AppData\Roaming\OpenRefine
+JAVA_OPTIONS="-Djava.security.egd=file:///dev/urandom"
 
 # Uncomment to increase autosave period to 60 mins (default: 5 minutes) for better performance of long-lasting transformations
 #REFINE_AUTOSAVE_PERIOD=60
