diff -Nru OpenRefine-3.3.orig/refine.ini OpenRefine-3.3/refine.ini
--- OpenRefine-3.3.orig/refine.ini	2020-01-31 17:22:18.000000000 +0000
+++ OpenRefine-3.3/refine.ini	2020-04-24 15:23:10.776571412 +0000
@@ -8,15 +8,16 @@

 # Memory and max form size allocations
 #REFINE_MAX_FORM_CONTENT_SIZE=1048576
-REFINE_MEMORY=1400M
+REFINE_MEMORY=$(free -m | grep -oP '\d+' | head -n 1)M

 # Set initial java heap space (default: 256M) for better performance with large datasets
-REFINE_MIN_MEMORY=1400M
+#REFINE_MIN_MEMORY=1400M

 # Some sample configurations. These have no defaults.
 #JAVA_HOME=C:\Program Files\Java\jdk1.8.0_151
 #JAVA_OPTIONS=-XX:+UseParallelGC -verbose:gc -Drefine.headless=true
 #JAVA_OPTIONS=-Drefine.data_dir=C:\Users\user\AppData\Roaming\OpenRefine
+JAVA_OPTIONS="-Djava.security.egd=file:///dev/urandom"

 # Uncomment to increase autosave period to 60 mins (default: 5 minutes) for better performance of long-lasting transformations
 #REFINE_AUTOSAVE_PERIOD=60
