diff -Nru nbclassic-1.0.0.orig/setup.cfg nbclassic-1.0.0/setup.cfg
--- nbclassic-1.0.0.orig/setup.cfg	2023-08-18 19:14:00.310355968 +0200
+++ nbclassic-1.0.0/setup.cfg	2023-08-18 19:14:37.547459988 +0200
@@ -1,6 +1,6 @@
 [metadata]
 name = nbclassic
-version = attr: nbclassic.__version__
+version = 1.0.0
 description = Jupyter Notebook as a Jupyter Server extension.
 long_description = file: README.md
 long_description_content_type = text/markdown
