diff -ru opt.orig/lumerical/v212/templates/fdtd-pbs-template.sh opt/lumerical/v212/templates/fdtd-pbs-template.sh
--- opt.orig/lumerical/v212/templates/fdtd-pbs-template.sh	2022-03-08 02:39:22.404552396 +0000
+++ opt/lumerical/v212/templates/fdtd-pbs-template.sh	2022-03-08 18:10:49.388662191 +0000
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/v212/bin/fdtd-engine-mpich2nem"
+MY_PROG="$EBROOTLUMERICAL/bin/fdtd-engine-mpich2nem"
 INPUT="<filename>"
-/opt/lumerical/v212/mpich2/nemesis/bin/mpiexec.hydra -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
+$EBROOTLUMERICAL/mpich2/nemesis/bin/mpiexec.hydra -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
 echo "Job finished at: `date`"
 exit
diff -ru opt.orig/lumerical/v212/templates/varfdtd-pbs-template.sh opt/lumerical/v212/templates/varfdtd-pbs-template.sh
--- opt.orig/lumerical/v212/templates/varfdtd-pbs-template.sh	2022-03-08 02:39:22.405552402 +0000
+++ opt/lumerical/v212/templates/varfdtd-pbs-template.sh	2022-03-08 18:12:47.251456026 +0000
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/v212/bin/varfdtd-engine-mpichp4"
+MY_PROG="$EBROOTLUMERICAL/bin/varfdtd-engine-mpichp4"
 INPUT="<filename>"
-/opt/lumerical/v212/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
+$EBROOTLUMERICAL/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
 echo "Job finished at: `date`"
 exit
