fix paths in pbs templates
Author: Miguel Dias Costa, Oliver Stueker
--- opt/lumerical/2019b/templates/fdtd-pbs-template.sh.orig	2019-10-29 16:32:16.099736120 -0230
+++ opt/lumerical/2019b/templates/fdtd-pbs-template.sh	2019-10-29 16:35:26.792972497 -0230
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/2019b/bin/fdtd-engine-mpich2nem"
+MY_PROG="$EBROOTLUMERICAL/bin/fdtd-engine-mpich2nem"
 INPUT="<filename>"
-/opt/lumerical/2019b/mpich2/nemesis/bin/mpiexec.hydra -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
+$EBROOTLUMERICAL/mpich2/nemesis/bin/mpiexec.hydra -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
 echo "Job finished at: `date`"
 exit
--- opt/lumerical/2019b/templates/varfdtd-pbs-template.sh.orig	2019-10-29 16:35:48.014805674 -0230
+++ opt/lumerical/2019b/templates/varfdtd-pbs-template.sh	2019-10-29 16:36:44.120621049 -0230
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/2019b/bin/varfdtd-engine-mpichp4"
+MY_PROG="$EBROOTLUMERICAL/bin/varfdtd-engine-mpichp4"
 INPUT="<filename>"
-/opt/lumerical/2019b/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
+$EBROOTLUMERICAL/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
 echo "Job finished at: `date`"
 exit
