diff -ru opt.orig/lumerical/v222/templates/fdtd-pbs-template.sh opt/lumerical/v222/templates/fdtd-pbs-template.sh
--- opt.orig/lumerical/v222/templates/fdtd-pbs-template.sh	2022-08-19 17:19:20.768726632 +0000
+++ opt/lumerical/v222/templates/fdtd-pbs-template.sh	2022-08-19 17:21:59.603935216 +0000
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/v222/bin/fdtd-engine-mpich2nem"
+MY_PROG="$EBROOTLUMERICAL/bin/fdtd-engine-mpich2nem"
 INPUT="<filename>"
-/opt/lumerical/v222/mpich2/nemesis/bin/mpiexec.hydra -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
+$EBROOTLUMERICAL/mpich2/nemesis/bin/mpiexec.hydra -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
 echo "Job finished at: `date`"
 exit
diff -ru opt.orig/lumerical/v222/templates/varfdtd-pbs-template.sh opt/lumerical/v222/templates/varfdtd-pbs-template.sh
--- opt.orig/lumerical/v222/templates/varfdtd-pbs-template.sh	2022-08-19 17:19:20.771726655 +0000
+++ opt/lumerical/v222/templates/varfdtd-pbs-template.sh	2022-08-19 17:23:02.855416499 +0000
@@ -10,8 +10,8 @@
 NUM_PROCS=`/bin/awk 'END {print NR}' $PBS_NODEFILE`
 echo "Starting run at: `date`"
 echo "Running on $NUM_PROCS processors."
-MY_PROG="/opt/lumerical/v222/bin/varfdtd-engine-mpichp4"
+MY_PROG="$EBROOTLUMERICAL/bin/varfdtd-engine-mpichp4"
 INPUT="<filename>"
-/opt/lumerical/v222/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
+$EBROOTLUMERICAL/mpich/ch_p4/bin/mpirun -n $NUM_PROCS $MY_PROG -t 1 ./${INPUT}
 echo "Job finished at: `date`"
 exit
