diff -Naur alamode-v.1.0.2_orig/alm/Makefile.linux alamode-v.1.0.2/alm/Makefile.linux
--- alamode-v.1.0.2_orig/alm/Makefile.linux	2018-03-28 17:48:58.742965034 +0000
+++ alamode-v.1.0.2/alm/Makefile.linux	2018-03-28 22:18:03.756391593 +0000
@@ -6,7 +6,7 @@
 #-----------------------------------------------
 
 CXX = icpc 
-CXXFLAGS = -O2 -xHOST -openmp -std=c++11
+CXXFLAGS = -O2 -xHOST -qopenmp -std=c++11
 INCLUDE = -I../include
 
 CXXL = ${CXX}
@@ -43,6 +43,6 @@
 .PHONY: clean
 
 depend:
-	gcc -MM *.cpp > .depend
+	icpc -MM *.cpp > .depend
 
 -include .depend
diff -Naur alamode-v.1.0.2_orig/anphon/Makefile.linux alamode-v.1.0.2/anphon/Makefile.linux
--- alamode-v.1.0.2_orig/anphon/Makefile.linux	2018-03-28 17:48:58.743965024 +0000
+++ alamode-v.1.0.2/anphon/Makefile.linux	2018-03-28 21:23:36.313921402 +0000
@@ -6,7 +6,7 @@
 #-----------------------------------------------
 
 MPICXX = mpicxx 
-CXXFLAGS = -O3 -xHOST -openmp -std=c++11
+CXXFLAGS = -O3 -xHOST -qopenmp -std=c++11
 INCLUDE = -I../include 
 
 LINKER = ${MPICXX}
diff -Naur alamode-v.1.0.2_orig/Makefile alamode-v.1.0.2/Makefile
--- alamode-v.1.0.2_orig/Makefile	1970-01-01 00:00:00.000000000 +0000
+++ alamode-v.1.0.2/Makefile	2018-03-28 18:35:46.615984480 +0000
@@ -0,0 +1,9 @@
+all:
+	cd alm && make -f Makefile.linux 
+	cd anphon && make -f Makefile.linux 
+	cd tools && make 
+
+clean:
+	cd alm && make -f Makefile.linux clean 
+	cd anphon && make -f Makefile.linux clean 
+	cd tools && make clean 
diff -Naur alamode-v.1.0.2_orig/tools/Makefile alamode-v.1.0.2/tools/Makefile
--- alamode-v.1.0.2_orig/tools/Makefile	2018-03-28 17:48:58.741965043 +0000
+++ alamode-v.1.0.2/tools/Makefile	2018-03-28 18:36:27.943581033 +0000
@@ -6,10 +6,10 @@
 #-----------------------------------------------
 
 #Intel compiler
-#CXX = icpc
+CXX = icpc
 
 # OSX
-CXX = g++-7
+#CXX = g++-7
 
 CXXFLAGS = -O2 
 
@@ -48,6 +48,6 @@
 .PHONY: clean
 
 depend:
-	gcc -MM *.cpp > .depend
+	icpc -MM *.cpp > .depend
 
 -include .depend
