Don't use CMAKE_LIBRARY_ARCHITECTURE.

Åke Sandgren, 2022-05-02
diff -ru ogdf-dogwood-202202.orig/cmake/coin.cmake ogdf-dogwood-202202/cmake/coin.cmake
--- ogdf-dogwood-202202.orig/cmake/coin.cmake	2022-02-02 22:09:01.000000000 +0100
+++ ogdf-dogwood-202202/cmake/coin.cmake	2022-05-02 14:22:06.017468966 +0200
@@ -55,10 +55,10 @@
 endif()
 
 # installation
-set(COIN_INSTALL_LIBRARY_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}" CACHE PATH "Installation path of COIN library")
+set(COIN_INSTALL_LIBRARY_DIR "lib" CACHE PATH "Installation path of COIN library")
 set(COIN_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of COIN runtime targets")
 set(COIN_INSTALL_INCLUDE_DIR "include" CACHE PATH "Installation path of COIN header files (creates subdirectory)")
-set(COIN_INSTALL_CMAKE_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake/OGDF/" CACHE PATH "Installation path of COIN files for CMake")
+set(COIN_INSTALL_CMAKE_DIR "lib/cmake/OGDF/" CACHE PATH "Installation path of COIN files for CMake")
 mark_as_advanced(COIN_INSTALL_LIBRARY_DIR COIN_INSTALL_BIN_DIR COIN_INSTALL_INCLUDE_DIR COIN_INSTALL_CMAKE_DIR)
 install(TARGETS COIN
   EXPORT CoinTargets
diff -ru ogdf-dogwood-202202.orig/cmake/ogdf.cmake ogdf-dogwood-202202/cmake/ogdf.cmake
--- ogdf-dogwood-202202.orig/cmake/ogdf.cmake	2022-02-02 22:09:01.000000000 +0100
+++ ogdf-dogwood-202202/cmake/ogdf.cmake	2022-05-02 14:22:16.929387664 +0200
@@ -193,10 +193,10 @@
 endif()
 
 # installation
-set(OGDF_INSTALL_LIBRARY_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}" CACHE PATH "Installation path of OGDF library")
+set(OGDF_INSTALL_LIBRARY_DIR "lib" CACHE PATH "Installation path of OGDF library")
 set(OGDF_INSTALL_BIN_DIR "bin" CACHE PATH "Installation path of OGDF runtime targets")
 set(OGDF_INSTALL_INCLUDE_DIR "include" CACHE PATH "Installation path of OGDF header files (creates subdirectory)")
-set(OGDF_INSTALL_CMAKE_DIR "lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake/OGDF/" CACHE PATH "Installation path of OGDF files for CMake")
+set(OGDF_INSTALL_CMAKE_DIR "lib/cmake/OGDF/" CACHE PATH "Installation path of OGDF files for CMake")
 mark_as_advanced(OGDF_INSTALL_LIBRARY_DIR OGDF_INSTALL_BIN_DIR OGDF_INSTALL_INCLUDE_DIR OGDF_INSTALL_CMAKE_DIR)
 configure_file(cmake/ogdf-config.cmake "${PROJECT_BINARY_DIR}/ogdf-config.cmake" @ONLY)
 install(TARGETS OGDF
