--- pset/install_fc.sh.orig	2009-07-27 13:35:06.238363000 +0200
+++ pset/install_fc.sh	2009-07-27 13:35:27.035870000 +0200
@@ -138,7 +138,7 @@
     EXTRACT_PATH=""
     FORMALIZE_ONLY_MODE=0		# changed by -f option
     IGNORE_OLD_DB_MODE=0 		# changed by -z option
-    INSTALL_PATH_SPECIFIED=0		# changed by -i option
+    INSTALL_PATH_SPECIFIED=1		# changed by -i option
     LICENSE_PATH_SPECIFIED=0		# changed by -l option
     LOG_PATH_SPECIFIED=0		# changed by -j option
     NO_EULA_MODE=0 			# changed by -e option 
