From 844b80f479dc2c92268506631e9603a028e69ada Mon Sep 17 00:00:00 2001
From: Benedikt Riedel <benedikt.riedel@gmail.com>
Date: Mon, 30 Oct 2023 15:05:10 -0500
Subject: [PATCH] Adding libnvidia-nvvm.so to nvliblist.conf to accomodate
 changes in CUDA12 and NVIDIA Drivers > 530

Signed-off-by: Benedikt Riedel <benedikt.riedel@gmail.com>
---
 etc/nvliblist.conf | 1 +

diff --git a/etc/nvliblist.conf b/etc/nvliblist.conf
index 92c680325..a47bcfae0 100644
--- a/etc/nvliblist.conf
+++ b/etc/nvliblist.conf
@@ -45,6 +45,7 @@ libnvidia-gtk2.so
 libnvidia-gtk3.so
 libnvidia-ifr.so
 libnvidia-ml.so
+libnvidia-nvvm.so
 libnvidia-opencl.so
 libnvidia-opticalflow.so
 libnvidia-ptxjitcompiler.so
