diff --git a/internal/pkg/buildcfg/confgen/gen.go b/internal/pkg/buildcfg/confgen/gen.go
index d302e3e97..fa44137f3 100644
--- a/internal/pkg/buildcfg/confgen/gen.go
+++ b/internal/pkg/buildcfg/confgen/gen.go
@@ -56,7 +56,7 @@ func (d Define) WriteLine() (s string) {
 		"APPTAINER_CONFDIR",
 		"PLUGIN_ROOTDIR":
 		varType = "var"
-		varStatement = d.Words[1] + " = relocatePath(" + s + ")"
+		varStatement = d.Words[1] + " = " + s
 	case "APPTAINER_SUID_INSTALL":
 		varType = "var"
 		varStatement = d.Words[1] + " = isSuidInstall()"
