--- bigmemory/MD5.orig	2016-03-28 18:59:53.000000000 +0200
+++ bigmemory/MD5	2016-07-09 11:10:14.293902207 +0200
@@ -7,7 +7,7 @@
 08d8ac748bb99e6cd73cbc851ba24e6f *R/zzz.R
 3d01d7e508b363c81587088af3221fa7 *build/vignette.rds
 cff4b0071fbcb3165aaf35c802b46b34 *cleanup
-eb219c5b9917fe3022b63e0296e61953 *configure
+04b3ce27c656a79eded3b3e21bdaa97b *configure
 fc5c8f75e424213609b3e3cd9b57f5bf *configure.win
 0cd9765e8adee62083d32a17a6b9aa07 *inst/CITATION
 f883e7fbbec7e35f5142805cd80548cf *inst/doc/Overview.R
