diff -ru bigmemory.orig/configure bigmemory/configure
--- bigmemory.orig/configure    2016-05-18 12:00:18.021806723 -0700
+++ bigmemory/configure 2016-05-18 12:01:36.757128058 -0700
@@ -38,3 +38,5 @@
 
 echo "${FLAGS}" > src/Makevars
 echo "${LIBS}" >> src/Makevars
+echo "# disable Intel C++ compiler (icpc) warning/error #308, to avoid 'member \"std::complex<double>::_M_value\" is inaccessible'" >> src/Makevars
+echo "PKG_CXXFLAGS = -wd308" >> src/Makevars
diff -ru bigmemory.orig/MD5 bigmemory/MD5
--- bigmemory.orig/MD5  2016-05-18 12:00:18.021806723 -0700
+++ bigmemory/MD5       2016-05-18 12:02:11.564828529 -0700
@@ -7,7 +7,7 @@
 08d8ac748bb99e6cd73cbc851ba24e6f *R/zzz.R
 3d01d7e508b363c81587088af3221fa7 *build/vignette.rds
 cff4b0071fbcb3165aaf35c802b46b34 *cleanup
-eb219c5b9917fe3022b63e0296e61953 *configure
+04b3ce27c656a79eded3b3e21bdaa97b *configure
 fc5c8f75e424213609b3e3cd9b57f5bf *configure.win
 0cd9765e8adee62083d32a17a6b9aa07 *inst/CITATION
 f883e7fbbec7e35f5142805cd80548cf *inst/doc/Overview.R

