--- forecast.orig/src/Makevars  2016-05-18 09:21:25.805883692 -0700
+++ forecast/src/Makevars       2016-05-18 09:22:18.541408635 -0700
@@ -1,2 +1,4 @@
 ## Use the R_HOME indirection to support installations of multiple R version
 PKG_LIBS=$(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
+# disable Intel C++ compiler (icpc) warning/error #308, to avoid 'member "std::complex<double>::_M_value" is inaccessible'
+PKG_CXXFLAGS = -wd308
