diff -ru R-2.15.1.orig/src/main/Makefile.in R-2.15.1/src/main/Makefile.in
--- R-2.15.1.orig/src/main/Makefile.in	2012-03-02 00:02:24.000000000 +0100
+++ R-2.15.1/src/main/Makefile.in	2012-09-03 13:51:41.118835149 +0200
@@ -181,7 +181,7 @@
 libR: $(libR_la)
 
 $(libR_la): $(libR_la_OBJECTS) $(libR_la_DEPENDENCIES)
-	$(DYLIB_LINK) $(LIBR_LDFLAGS) -o $@ $(libR_la_OBJECTS) $(libR_la_LIBADD)
+	$(DYLIB_LINK) $(LDFLAGS) $(LIBR_LDFLAGS) -o $@ $(libR_la_OBJECTS) $(libR_la_LIBADD)
 
 install-lib-local: $(libR_la)
 	@$(MAKE) rhome="$(abs_top_builddir)" install-lib
