diff -ru Rmpi.orig/configure Rmpi/configure
--- Rmpi.orig/configure	2014-04-08 18:12:40.000000000 +0200
+++ Rmpi/configure	2015-01-08 14:35:03.518018605 +0100
@@ -3780,7 +3780,7 @@
     PKG_LIBS="-L${MPI_LIBPATH} -lmpich -lmpl -lopa ${MPI_LIBS}"
     ;;
   INTELMPI)
-    PKG_LIBS="-Wl,-rpath,${MPI_LIBPATH} -L${MPI_LIBPATH} -lmpich -lmpigf -lmpigi ${MPI_LIBS}"
+    PKG_LIBS="-Wl,-rpath,${MPI_LIBPATH} -L${MPI_LIBPATH} -lmpi -lmpigf -lmpigi ${MPI_LIBS}"
     ;;
   CRAY)
     PKG_LIBS="-L${MPI_LIBPATH} -lmpich_cray_default64 -lmpl -lopa ${MPI_LIBS}"
