diff -ru SU2-7.5.1.orig/meson.build SU2-7.5.1/meson.build
--- SU2-7.5.1.orig/meson.build	2023-07-18 13:38:57.898138197 +0000
+++ SU2-7.5.1/meson.build	2023-07-18 14:21:32.422747149 +0000
@@ -199,6 +199,13 @@
   blas_dep = dependency(get_option('blas-name'))
   su2_deps += blas_dep
 
+elif get_option('blas-name') != ''
+
+  su2_cpp_args += '-DHAVE_BLAS'
+  su2_cpp_args += '-DHAVE_LAPACK'
+  blas_dep = dependency(get_option('blas-name'))
+  su2_deps += blas_dep
+
 endif
 
 if get_option('enable-librom')
