--- boost/config/stdlib/libstdcpp3.hpp.orig	2014-05-27 22:32:33.070983000 +0200
+++ boost/config/stdlib/libstdcpp3.hpp	2014-05-27 22:32:49.609032000 +0200
@@ -33,7 +33,7 @@
 
 #ifdef __GLIBCXX__ // gcc 3.4 and greater:
 #  if defined(_GLIBCXX_HAVE_GTHR_DEFAULT) \
-        || defined(_GLIBCXX__PTHREADS)
+        || defined(_GLIBCXX__PTHREADS) || defined(_GLIBCXX_HAS_GTHREADS)
       //
       // If the std lib has thread support turned on, then turn it on in Boost
       // as well.  We do this because some gcc-3.4 std lib headers define _REENTANT
