--- boost_1_44_0.orig/boost/config/stdlib/libstdcpp3.hpp	2011-04-25 14:10:48.000000000 +0200
+++ boost_1_44_0/boost/config/stdlib/libstdcpp3.hpp	2013-12-12 22:48:04.304826665 +0100
@@ -33,7 +33,8 @@
 
 #ifdef __GLIBCXX__ // gcc 3.4 and greater:
 #  if defined(_GLIBCXX_HAVE_GTHR_DEFAULT) \
-        || defined(_GLIBCXX__PTHREADS)
+	|| defined(_GLIBCXX__PTHREADS) \
+	|| defined(_GLIBCXX_HAS_GTHREADS)
       //
       // If the std lib has thread support turned on, then turn it on in Boost
       // as well.  We do this because some gcc-3.4 std lib headers define _REENTANT
