--- tools/build/v2/tools/intel-linux.jam.orig	2010-06-24 15:16:30.987009211 +0200
+++ tools/build/v2/tools/intel-linux.jam	2010-06-24 15:17:29.954033026 +0200
@@ -114,6 +114,8 @@
 {
     gcc.setup-threading $(targets) : $(sources) : $(properties) ;
     gcc.setup-fpic $(targets) : $(sources) : $(properties) ;
+    OPTIONS on $(targets) += -fPIC ;	
+    OPTIONS on $(targets) += -v ;
     gcc.setup-address-model $(targets) : $(sources) : $(properties) ;
     DEPENDS $(<) : [ on $(<) return $(PCH_FILE) ] ;
 }
@@ -127,6 +129,8 @@
 {
     gcc.setup-threading $(targets) : $(sources) : $(properties) ;
     gcc.setup-fpic $(targets) : $(sources) : $(properties) ;
+    OPTIONS on $(targets) += -fPIC ;
+    OPTIONS on $(targets) += -v ;
     gcc.setup-address-model $(targets) : $(sources) : $(properties) ;    
     DEPENDS $(<) : [ on $(<) return $(PCH_FILE) ] ;
 }

