diff -ru Clp-1.17.6.orig/Clp/src/ClpCholeskyMumps.cpp Clp-1.17.6/Clp/src/ClpCholeskyMumps.cpp
--- Clp-1.17.6.orig/Clp/src/ClpCholeskyMumps.cpp	2022-05-19 13:25:10.584211211 +0000
+++ Clp-1.17.6/Clp/src/ClpCholeskyMumps.cpp	2022-05-19 13:25:41.233420750 +0000
@@ -13,6 +13,7 @@
 #define USE_COMM_WORLD -987654
 extern "C" {
 #include "dmumps_c.h"
+}
 // In newer ThirdParty/Mumps, mpi.h is renamed to mumps_mpi.h.
 // We get informed about this by having COIN_USE_MUMPS_MPI_H defined.
 #ifdef COIN_USE_MUMPS_MPI_H
@@ -20,7 +21,6 @@
 #else
 #include "mpi.h"
 #endif
-}
 
 #include "ClpCholeskyMumps.hpp"
 #include "ClpMessage.hpp"
