Removed hardcoded flags

Author: Pavel Grochal (INUITS)
diff -ru tsne-0.1.8.orig/setup.py tsne-0.1.8/setup.py
--- tsne-0.1.8.orig/setup.py	2018-06-06 19:22:12.000000000 +0200
+++ tsne-0.1.8/setup.py	2020-09-15 12:42:01.908482715 +0200
@@ -50,11 +50,11 @@
                    language='c++')]
 else:
     extra_link_args = ['-lcblas']
-    dist = platform.linux_distribution(full_distribution_name=0)[0]
-    redhat_dists = set(["redhat", "fedora", "centos"])
-    if dist in redhat_dists:
-        extra_link_args = ['-lsatlas']
-    
+    # dist = platform.linux_distribution(full_distribution_name=0)[0]
+    # redhat_dists = set(["redhat", "fedora", "centos"])
+    # if dist in redhat_dists:
+    #     extra_link_args = ['-lsatlas']
+
     # LINUX
     ext_modules = [Extension(name='bh_sne',
                    sources=['tsne/bh_sne_src/quadtree.cpp', 'tsne/bh_sne_src/tsne.cpp', 'tsne/bh_sne.pyx'],

