diff -Naur mrcc.2023-08-28.orig/fock.f mrcc.2023-08-28/fock.f
--- mrcc.2023-08-28.orig/fock.f	2024-04-18 17:20:51.556975808 +0000
+++ mrcc.2023-08-28/fock.f	2024-04-18 17:14:43.511302380 +0000
@@ -61,6 +61,8 @@
       logical lbeta,lscale,lwdfn,llc,sclroute,linv,lmcl,lfdm,lpssp,lexc
       logical qscf_hess,lscrohfldf
       COMMON/MEMCOM/ IMEM1
+      integer imemscr
+      common/memscr/ imemscr
 C
       call getkey('pssp',4,cpssp,8)
       if(cpssp.eq.'off     ') then
@@ -973,6 +975,7 @@
 c      write(6,*) 'focka scf'
 c      write(6,"(850f12.8)") focka
 C
+       if(istore.gt.2) istore=max(istore,imemscr+2)
       RETURN
       END  
 ************************************************************************
