diff -Naur mrcc.2023-08-28.orig/teint.f mrcc.2023-08-28/teint.f
--- mrcc.2023-08-28.orig/teint.f	2024-04-18 17:21:48.339079700 +0000
+++ mrcc.2023-08-28/teint.f	2024-04-18 17:15:38.458402922 +0000
@@ -3760,7 +3760,7 @@
              do while(imo2.lt.nocc)!.and.imo2.le.imo1+0)
               mfitmax=max(mfitmax,
 c    $max(nmobasis(imo2+1),dfnmobasis(imo2+1))**2+
-     $nmobasis(imo2+1)**2, !???
+     $nmobasis(imo2+1)**2, ! small fock matrix (f) in dfexc
      $dfnmobasis(imo2+1)*(dfnmobasis(imo2+1)+1)/2,overlapmem,lfmem)
               if(loccri) occrimem=nocc*(imo2-imo1+2)*dfnbasis
               if(loccri.and.locfit.ge.2) occrimem=occrimem+
@@ -3783,7 +3783,7 @@
      $crsh*hailen+crsh*nstateo*ncd*nmobasis(imo2+1)*dfnmobasis(imo2+1)
      $.le.maxmem.and. ! Calculation of (mui|P)
      $crsh*hailen+crsh*nstateo*nmobasis(imo2+1)*dfnmobasis(imo2+1)
-     $+mfitmax+occrimem.le.maxcor-(imem-imem1).and. ! Fittimg of (mui|P)
+     $+mfitmax+domain_mem+occrimem.le.maxcor-(imem-imem1).and. ! Fittimg of (mui|P)
 C HF exchange derivative
      $(oroute.ne.2.or.(crsh*hailen+crsh*ncd*nstateo*nmobasis(imo2+1)*
      $dfnmobasis(imo2+1)+j*dfnbasis*nocc*(imo2-imo1+2)+4*dfnbasis**2+
