diff -Naur mrcc.2023-08-28.orig/integ.f mrcc.2023-08-28/integ.f
--- mrcc.2023-08-28.orig/integ.f	2024-04-18 17:21:04.188998919 +0000
+++ mrcc.2023-08-28/integ.f	2024-04-18 17:14:55.595324493 +0000
@@ -10231,7 +10231,7 @@
        if(.not.lexc.and.trim(dfbasis_scf).ne.'none'.and..not.lpssp) then
 c       call getkey('store',5,store,8)
 c        if(trim(store).eq.'on') istore=1
-         if(trim(scfalg).ne.'disk') istore=1
+         if(trim(scfalg).ne.'disk') istore=0
          if(trim(ccalc).eq.'scf') then
           call getkeym('grid',4,gridtype,8)
            if(gridtype.eq.'        '.and.mgga.eq.0) 
