diff -Naur mrcc.2023-08-28.orig/scf.f mrcc.2023-08-28/scf.f
--- mrcc.2023-08-28.orig/scf.f	2024-04-18 17:21:32.003049810 +0000
+++ mrcc.2023-08-28/scf.f	2024-04-18 17:15:26.253380589 +0000
@@ -3759,6 +3759,8 @@
       character*12 diisdtolval
       character*16 cscr16
       character*32 cscr32
+      integer imemscr
+      common/memscr/ imemscr
 !!! !HB
       real*8 temp
       logical llg
@@ -4591,6 +4593,7 @@
        else
         i=dblalloc(4*sqrsize) ! r8heap r8heap1n r8heap2n r8heap3n
        endif
+      imemscr=imem
       call dbldealloc(i)
 C
        if(lnos) SORIG_PTR=imem+2*nbasis**2
