diff -Naur mrcc.2023-08-28.orig/dmrcc.f mrcc.2023-08-28/dmrcc.f
--- mrcc.2023-08-28.orig/dmrcc.f	2024-04-18 17:20:13.571906305 +0000
+++ mrcc.2023-08-28/dmrcc.f	2024-04-18 17:13:51.800185120 +0000
@@ -628,7 +628,7 @@
       character*3 qro
       character*4 localcc,cmult,ccprog,ctol,cscr4,molden,scfdtol,scftol
       character*4 ccsdrest,cgrad
-      character*6 dhexc
+      character*10 dhexc
       character*5 scftype
       character*8 lccrest
       character*7 scfiguess,mcscfiguess
@@ -1157,7 +1157,7 @@
           call getvar('lseqc     ',lseqc)
           call getkey('scfalg',6,scfalg,8)
           call getkey('cialg',5,cialg,8)
-          call getkey('dhexc',5,dhexc,6)
+          call getkey('dhexc',5,dhexc,10)
            if(lf12) then
             call runit('mp2f12',intcpud,intwald)   
              if(trim(calctype).ne.'mp2-f12') then
@@ -1256,7 +1256,8 @@
      $           ((trim(cialg).eq.'direct'.or.trim(cialg).eq.'direct2'
      $           .or.trim(cialg).eq.'direct3').and.((trim(calctype).ne.
      $           'cis'.and.trim(calctype).ne.'scf').or.
-     $           trim(dhexc).eq.'cis(d)'))) then
+     $           (trim(dhexc).eq.'cis(d)'.or.trim(dhexc).eq.'sos-cis(d)'
+     $            .or.trim(dhexc).eq.'scs-cis(d)')))) then
               
               if(denscorr.eq.1) call ishell('cp VARS VARS.old')
               call runit('drpa',cput,walt)
