Adding a missing include needed for compilation.
Author: Stamen Miroslavov <stamen.miroslavov@bsc.es>
Barcelona Supercomputing Center 2026
diff -ruNBbw a/Source/ThirdParty/ANGLE/src/common/mathutil.h b/Source/ThirdParty/ANGLE/src/common/mathutil.h
--- a/Source/ThirdParty/ANGLE/src/common/mathutil.h	2020-03-10 03:14:41.000000000 +0100
+++ b/Source/ThirdParty/ANGLE/src/common/mathutil.h	2026-03-24 15:46:11.703979726 +0100
@@ -16,6 +16,7 @@
 #include <algorithm>
 #include <string.h>
 #include <stdlib.h>
+#include <cstdint>
 
 namespace gl
 {
