Don't use hardcoded path to python

Åke Sandgren, 2022-12-08
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/generate-bytecode-files qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/generate-bytecode-files
--- qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/generate-bytecode-files	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/generate-bytecode-files	2022-12-08 07:56:12.528708137 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/env python
 
 # Copyright (C) 2014 Apple Inc. All rights reserved.
 #
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/Scripts/cssmin.py qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/Scripts/cssmin.py
--- qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/Scripts/cssmin.py	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/Scripts/cssmin.py	2022-12-08 07:56:12.528708137 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 # Copyright (C) 2013 Apple Inc. All rights reserved.
 #
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/Scripts/generate-combined-inspector-json.py qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/Scripts/generate-combined-inspector-json.py
--- qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/Scripts/generate-combined-inspector-json.py	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/Scripts/generate-combined-inspector-json.py	2022-12-08 07:56:12.532708103 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 # Copyright (C) 2013 Apple Inc. All rights reserved.
 #
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/Scripts/UpdateContents.py qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/Scripts/UpdateContents.py
--- qtwebkit-5.212.0-alpha4.orig/Source/JavaScriptCore/Scripts/UpdateContents.py	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/JavaScriptCore/Scripts/UpdateContents.py	2022-12-08 07:56:12.532708103 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 # Copyright (C) 2015 Apple Inc.  All rights reserved.
 #
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/ThirdParty/ANGLE/src/libANGLE/renderer/generate_new_renderer.py qtwebkit-5.212.0-alpha4/Source/ThirdParty/ANGLE/src/libANGLE/renderer/generate_new_renderer.py
--- qtwebkit-5.212.0-alpha4.orig/Source/ThirdParty/ANGLE/src/libANGLE/renderer/generate_new_renderer.py	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/ThirdParty/ANGLE/src/libANGLE/renderer/generate_new_renderer.py	2022-12-08 07:56:12.524708171 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 #
 # Copyright (c) 2015 The ANGLE Project Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/WebCore/AVFoundationSupport.py qtwebkit-5.212.0-alpha4/Source/WebCore/AVFoundationSupport.py
--- qtwebkit-5.212.0-alpha4.orig/Source/WebCore/AVFoundationSupport.py	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/WebCore/AVFoundationSupport.py	2022-12-08 07:56:12.524708171 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 # Copyright (C) 2015 Apple Inc.  All rights reserved.
 #
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/WebInspectorUI/Scripts/cssmin.py qtwebkit-5.212.0-alpha4/Source/WebInspectorUI/Scripts/cssmin.py
--- qtwebkit-5.212.0-alpha4.orig/Source/WebInspectorUI/Scripts/cssmin.py	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/WebInspectorUI/Scripts/cssmin.py	2022-12-08 07:56:12.528708137 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 # Copyright (C) 2013 Apple Inc. All rights reserved.
 #
