Don't use multiarch lib dir

Åke Sandgren, 2022-12-08
diff -ru qtwebkit-5.212.0-alpha4.orig/Source/cmake/KDEInstallDirs.cmake qtwebkit-5.212.0-alpha4/Source/cmake/KDEInstallDirs.cmake
--- qtwebkit-5.212.0-alpha4.orig/Source/cmake/KDEInstallDirs.cmake	2020-03-04 18:16:37.000000000 +0100
+++ qtwebkit-5.212.0-alpha4/Source/cmake/KDEInstallDirs.cmake	2022-12-08 11:04:09.788027403 +0100
@@ -210,11 +210,6 @@
 # See http://wiki.debian.org/Multiarch
 if((CMAKE_SYSTEM_NAME MATCHES "Linux|kFreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "GNU")
    AND NOT CMAKE_CROSSCOMPILING)
-  if (EXISTS "/etc/debian_version") # is this a debian system ?
-    if(CMAKE_LIBRARY_ARCHITECTURE)
-      set(_LIBDIR_DEFAULT "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
-    endif()
-  else() # not debian, rely on CMAKE_SIZEOF_VOID_P:
     if(NOT DEFINED CMAKE_SIZEOF_VOID_P)
       message(AUTHOR_WARNING
         "Unable to determine default LIB_INSTALL_LIBDIR directory because no target architecture is known. "
@@ -224,7 +219,6 @@
         set(_LIBDIR_DEFAULT "lib64")
       endif()
     endif()
-  endif()
 endif()
 
 set(_gnu_install_dirs_vars
