--- MMseqs2-14-7e284/CMakeLists.txt.orig	2025-03-10 14:51:04.045108773 +0100
+++ MMseqs2-14-7e284/CMakeLists.txt	2025-03-10 14:55:00.494079598 +0100
@@ -90,6 +90,8 @@
         set(ZARCH 1)
     elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^sparc")
         set(SPARC 1)
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "riscv64")
+	set(RISCV64 1)
     else ()
         message(WARNING "CPU without native SIMD instructions. Performance will be bad.")
     endif ()
@@ -113,6 +115,8 @@
         set(MMSEQS_ARCH "-mcpu=native")
     elseif (ZARCH)
         set(MMSEQS_ARCH "-mcpu=native -mzvector")
+    elseif (RISCV64)
+        set(MMSEQS_ARCH "-march=rv64gc")
     else ()
         set(MMSEQS_ARCH "-march=native")
     endif ()
