diff -ru ./src/H5make_libsettings.c.orig ./src/H5make_libsettings.c
--- ./src/H5make_libsettings.c.orig	2014-05-06 04:13:01.000000000 +0200
+++ ./src/H5make_libsettings.c	2014-07-23 11:39:19.709674579 +0200
@@ -41,6 +41,7 @@
  *-------------------------------------------------------------------------
  */
 
+#include <mpi.h>
 #include <stdio.h>
 #include <time.h>
 #include "H5private.h"
diff -ru ./c++/src/H5IntType.cpp.orig ./c++/src/H5IntType.cpp
--- ./c++/src/H5IntType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5IntType.cpp	2014-07-23 11:39:19.703674592 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5FloatType.cpp.orig ./c++/src/H5FloatType.cpp
--- ./c++/src/H5FloatType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5FloatType.cpp	2014-07-23 11:39:19.702674596 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5CommonFG.cpp.orig ./c++/src/H5CommonFG.cpp
--- ./c++/src/H5CommonFG.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5CommonFG.cpp	2014-07-23 11:39:19.696674600 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5DataSpace.cpp.orig ./c++/src/H5DataSpace.cpp
--- ./c++/src/H5DataSpace.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5DataSpace.cpp	2014-07-23 11:39:19.697674596 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5PredType.cpp.orig ./c++/src/H5PredType.cpp
--- ./c++/src/H5PredType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5PredType.cpp	2014-07-23 11:39:19.704674589 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5IdComponent.cpp.orig ./c++/src/H5IdComponent.cpp
--- ./c++/src/H5IdComponent.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5IdComponent.cpp	2014-07-23 11:39:19.702674596 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5EnumType.cpp.orig ./c++/src/H5EnumType.cpp
--- ./c++/src/H5EnumType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5EnumType.cpp	2014-07-23 11:39:19.699674601 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5FcreatProp.cpp.orig ./c++/src/H5FcreatProp.cpp
--- ./c++/src/H5FcreatProp.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5FcreatProp.cpp	2014-07-23 11:39:19.701674600 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5File.cpp.orig ./c++/src/H5File.cpp
--- ./c++/src/H5File.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5File.cpp	2014-07-23 11:39:19.701674600 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5AtomType.cpp.orig ./c++/src/H5AtomType.cpp
--- ./c++/src/H5AtomType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5AtomType.cpp	2014-07-23 11:39:19.694674615 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5DxferProp.cpp.orig ./c++/src/H5DxferProp.cpp
--- ./c++/src/H5DxferProp.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5DxferProp.cpp	2014-07-23 11:39:19.699674601 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5PropList.cpp.orig ./c++/src/H5PropList.cpp
--- ./c++/src/H5PropList.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5PropList.cpp	2014-07-23 11:39:19.705674586 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5VarLenType.cpp.orig ./c++/src/H5VarLenType.cpp
--- ./c++/src/H5VarLenType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5VarLenType.cpp	2014-07-23 11:39:19.705674586 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5Attribute.cpp.orig ./c++/src/H5Attribute.cpp
--- ./c++/src/H5Attribute.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5Attribute.cpp	2014-07-23 11:39:19.695674606 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5Location.cpp.orig ./c++/src/H5Location.cpp
--- ./c++/src/H5Location.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5Location.cpp	2014-07-23 11:39:19.709674579 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5CompType.cpp.orig ./c++/src/H5CompType.cpp
--- ./c++/src/H5CompType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5CompType.cpp	2014-07-23 11:39:19.696674600 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5DcreatProp.cpp.orig ./c++/src/H5DcreatProp.cpp
--- ./c++/src/H5DcreatProp.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5DcreatProp.cpp	2014-07-23 11:39:19.698674597 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5ArrayType.cpp.orig ./c++/src/H5ArrayType.cpp
--- ./c++/src/H5ArrayType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5ArrayType.cpp	2014-07-23 11:39:19.694674615 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5Library.cpp.orig ./c++/src/H5Library.cpp
--- ./c++/src/H5Library.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5Library.cpp	2014-07-23 11:39:19.703674592 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5CppDoc.h"	// included only for Doxygen to generate part of RM
diff -ru ./c++/src/H5DataSet.cpp.orig ./c++/src/H5DataSet.cpp
--- ./c++/src/H5DataSet.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5DataSet.cpp	2014-07-23 11:39:19.697674596 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5FaccProp.cpp.orig ./c++/src/H5FaccProp.cpp
--- ./c++/src/H5FaccProp.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5FaccProp.cpp	2014-07-23 11:39:19.700674601 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5StrType.cpp.orig ./c++/src/H5StrType.cpp
--- ./c++/src/H5StrType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5StrType.cpp	2014-07-23 11:39:19.705674586 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5Object.cpp.orig ./c++/src/H5Object.cpp
--- ./c++/src/H5Object.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5Object.cpp	2014-07-23 11:39:19.704674589 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5DataType.cpp.orig ./c++/src/H5DataType.cpp
--- ./c++/src/H5DataType.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5DataType.cpp	2014-07-23 11:39:19.698674597 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5Group.cpp.orig ./c++/src/H5Group.cpp
--- ./c++/src/H5Group.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5Group.cpp	2014-07-23 11:39:19.702674596 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/src/H5AbstractDs.cpp.orig ./c++/src/H5AbstractDs.cpp
--- ./c++/src/H5AbstractDs.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5AbstractDs.cpp	2014-07-23 11:39:19.694674615 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/src/H5Exception.cpp.orig ./c++/src/H5Exception.cpp
--- ./c++/src/H5Exception.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/src/H5Exception.cpp	2014-07-23 11:39:19.700674601 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru ./c++/test/tfilter.cpp.orig ./c++/test/tfilter.cpp
--- ./c++/test/tfilter.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tfilter.cpp	2014-07-23 11:39:19.708674580 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/tfile.cpp.orig ./c++/test/tfile.cpp
--- ./c++/test/tfile.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tfile.cpp	2014-07-23 11:39:19.708674580 +0200
@@ -23,6 +23,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/tdspl.cpp.orig ./c++/test/tdspl.cpp
--- ./c++/test/tdspl.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tdspl.cpp	2014-07-23 11:47:37.014675961 +0200
@@ -20,6 +20,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/h5cpputil.cpp.orig ./c++/test/h5cpputil.cpp
--- ./c++/test/h5cpputil.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/h5cpputil.cpp	2014-07-23 11:39:19.706674583 +0200
@@ -21,6 +21,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/th5s.cpp.orig ./c++/test/th5s.cpp
--- ./c++/test/th5s.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/th5s.cpp	2014-07-23 11:39:19.708674580 +0200
@@ -22,6 +22,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/tattr.cpp.orig ./c++/test/tattr.cpp
--- ./c++/test/tattr.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tattr.cpp	2014-07-23 11:39:19.707674581 +0200
@@ -20,6 +20,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/trefer.cpp.orig ./c++/test/trefer.cpp
--- ./c++/test/trefer.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/trefer.cpp	2014-07-23 11:39:19.708674580 +0200
@@ -20,6 +20,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/ttypes.cpp.orig ./c++/test/ttypes.cpp
--- ./c++/test/ttypes.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/ttypes.cpp	2014-07-23 11:39:19.709674579 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/tvlstr.cpp.orig ./c++/test/tvlstr.cpp
--- ./c++/test/tvlstr.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tvlstr.cpp	2014-07-23 11:39:19.709674579 +0200
@@ -21,6 +21,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/tcompound.cpp.orig ./c++/test/tcompound.cpp
--- ./c++/test/tcompound.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tcompound.cpp	2014-07-23 11:39:19.707674581 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/testhdf5.cpp.orig ./c++/test/testhdf5.cpp
--- ./c++/test/testhdf5.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/testhdf5.cpp	2014-07-23 11:39:19.708674580 +0200
@@ -41,6 +41,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/dsets.cpp.orig ./c++/test/dsets.cpp
--- ./c++/test/dsets.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/dsets.cpp	2014-07-23 11:39:19.706674583 +0200
@@ -25,6 +25,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/tobject.cpp.orig ./c++/test/tobject.cpp
--- ./c++/test/tobject.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tobject.cpp	2014-07-23 11:46:37.657791929 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru ./c++/test/tlinks.cpp.orig ./c++/test/tlinks.cpp
--- ./c++/test/tlinks.cpp.orig	2014-05-06 04:13:16.000000000 +0200
+++ ./c++/test/tlinks.cpp	2014-07-23 11:39:19.708674580 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
