diff -ru hdf5-1.8.9.orig/c++/src/H5AbstractDs.cpp hdf5-1.8.9/c++/src/H5AbstractDs.cpp
--- hdf5-1.8.9.orig/c++/src/H5AbstractDs.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5AbstractDs.cpp	2012-06-19 20:26:22.516500000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5ArrayType.cpp hdf5-1.8.9/c++/src/H5ArrayType.cpp
--- hdf5-1.8.9.orig/c++/src/H5ArrayType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5ArrayType.cpp	2012-06-19 20:26:22.565402000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5AtomType.cpp hdf5-1.8.9/c++/src/H5AtomType.cpp
--- hdf5-1.8.9.orig/c++/src/H5AtomType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5AtomType.cpp	2012-06-19 20:26:22.522772000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5Attribute.cpp hdf5-1.8.9/c++/src/H5Attribute.cpp
--- hdf5-1.8.9.orig/c++/src/H5Attribute.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5Attribute.cpp	2012-06-19 20:26:22.450170000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5CommonFG.cpp hdf5-1.8.9/c++/src/H5CommonFG.cpp
--- hdf5-1.8.9.orig/c++/src/H5CommonFG.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5CommonFG.cpp	2012-06-19 20:26:22.584477000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5CompType.cpp hdf5-1.8.9/c++/src/H5CompType.cpp
--- hdf5-1.8.9.orig/c++/src/H5CompType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5CompType.cpp	2012-06-19 20:26:22.571885000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5DataSet.cpp hdf5-1.8.9/c++/src/H5DataSet.cpp
--- hdf5-1.8.9.orig/c++/src/H5DataSet.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5DataSet.cpp	2012-06-19 20:26:22.590819000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5DataSpace.cpp hdf5-1.8.9/c++/src/H5DataSpace.cpp
--- hdf5-1.8.9.orig/c++/src/H5DataSpace.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5DataSpace.cpp	2012-06-19 20:26:22.529072000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5DataType.cpp hdf5-1.8.9/c++/src/H5DataType.cpp
--- hdf5-1.8.9.orig/c++/src/H5DataType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5DataType.cpp	2012-06-19 20:26:22.491189000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5DcreatProp.cpp hdf5-1.8.9/c++/src/H5DcreatProp.cpp
--- hdf5-1.8.9.orig/c++/src/H5DcreatProp.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5DcreatProp.cpp	2012-06-19 20:26:22.497485000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5DxferProp.cpp hdf5-1.8.9/c++/src/H5DxferProp.cpp
--- hdf5-1.8.9.orig/c++/src/H5DxferProp.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5DxferProp.cpp	2012-06-19 20:26:22.503631000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5EnumType.cpp hdf5-1.8.9/c++/src/H5EnumType.cpp
--- hdf5-1.8.9.orig/c++/src/H5EnumType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5EnumType.cpp	2012-06-19 20:26:22.535146000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5Exception.cpp hdf5-1.8.9/c++/src/H5Exception.cpp
--- hdf5-1.8.9.orig/c++/src/H5Exception.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5Exception.cpp	2012-06-19 20:26:22.452097000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5FaccProp.cpp hdf5-1.8.9/c++/src/H5FaccProp.cpp
--- hdf5-1.8.9.orig/c++/src/H5FaccProp.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5FaccProp.cpp	2012-06-19 20:26:22.459968000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5FcreatProp.cpp hdf5-1.8.9/c++/src/H5FcreatProp.cpp
--- hdf5-1.8.9.orig/c++/src/H5FcreatProp.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5FcreatProp.cpp	2012-06-19 20:26:22.510375000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5File.cpp hdf5-1.8.9/c++/src/H5File.cpp
--- hdf5-1.8.9.orig/c++/src/H5File.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5File.cpp	2012-06-19 20:26:22.597224000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5FloatType.cpp hdf5-1.8.9/c++/src/H5FloatType.cpp
--- hdf5-1.8.9.orig/c++/src/H5FloatType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5FloatType.cpp	2012-06-19 20:26:22.547455000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5Group.cpp hdf5-1.8.9/c++/src/H5Group.cpp
--- hdf5-1.8.9.orig/c++/src/H5Group.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5Group.cpp	2012-06-19 20:26:22.603257000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5IdComponent.cpp hdf5-1.8.9/c++/src/H5IdComponent.cpp
--- hdf5-1.8.9.orig/c++/src/H5IdComponent.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5IdComponent.cpp	2012-06-19 20:26:22.466094000 +0200
@@ -13,6 +13,8 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
+
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5IntType.cpp hdf5-1.8.9/c++/src/H5IntType.cpp
--- hdf5-1.8.9.orig/c++/src/H5IntType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5IntType.cpp	2012-06-19 20:26:22.552991000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5Library.cpp hdf5-1.8.9/c++/src/H5Library.cpp
--- hdf5-1.8.9.orig/c++/src/H5Library.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5Library.cpp	2012-06-19 20:26:22.471699000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5CppDoc.h"	// included only for Doxygen to generate part of RM
diff -ru hdf5-1.8.9.orig/c++/src/H5Object.cpp hdf5-1.8.9/c++/src/H5Object.cpp
--- hdf5-1.8.9.orig/c++/src/H5Object.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5Object.cpp	2012-06-19 20:26:22.478435000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5PredType.cpp hdf5-1.8.9/c++/src/H5PredType.cpp
--- hdf5-1.8.9.orig/c++/src/H5PredType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5PredType.cpp	2012-06-19 20:26:22.541295000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5PropList.cpp hdf5-1.8.9/c++/src/H5PropList.cpp
--- hdf5-1.8.9.orig/c++/src/H5PropList.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5PropList.cpp	2012-06-19 20:26:22.484759000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/src/H5StrType.cpp hdf5-1.8.9/c++/src/H5StrType.cpp
--- hdf5-1.8.9.orig/c++/src/H5StrType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5StrType.cpp	2012-06-19 20:26:22.559556000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/src/H5VarLenType.cpp hdf5-1.8.9/c++/src/H5VarLenType.cpp
--- hdf5-1.8.9.orig/c++/src/H5VarLenType.cpp	2012-05-09 17:06:23.000000000 +0200
+++ hdf5-1.8.9/c++/src/H5VarLenType.cpp	2012-06-19 20:26:22.577670000 +0200
@@ -13,6 +13,7 @@
  * access to either file, you may request a copy from help@hdfgroup.org.     *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#include <mpi.h>
 #include <string>
 
 #include "H5Include.h"
diff -ru hdf5-1.8.9.orig/c++/test/dsets.cpp hdf5-1.8.9/c++/test/dsets.cpp
--- hdf5-1.8.9.orig/c++/test/dsets.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/dsets.cpp	2012-06-19 20:26:22.609694000 +0200
@@ -25,6 +25,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/h5cpputil.cpp hdf5-1.8.9/c++/test/h5cpputil.cpp
--- hdf5-1.8.9.orig/c++/test/h5cpputil.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/h5cpputil.cpp	2012-06-19 20:26:22.658746000 +0200
@@ -21,6 +21,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/tattr.cpp hdf5-1.8.9/c++/test/tattr.cpp
--- hdf5-1.8.9.orig/c++/test/tattr.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/tattr.cpp	2012-06-19 20:26:22.616193000 +0200
@@ -20,6 +20,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/tcompound.cpp hdf5-1.8.9/c++/test/tcompound.cpp
--- hdf5-1.8.9.orig/c++/test/tcompound.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/tcompound.cpp	2012-06-19 20:26:22.622451000 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/testhdf5.cpp hdf5-1.8.9/c++/test/testhdf5.cpp
--- hdf5-1.8.9.orig/c++/test/testhdf5.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/testhdf5.cpp	2012-06-19 20:26:22.627951000 +0200
@@ -41,6 +41,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/tfile.cpp hdf5-1.8.9/c++/test/tfile.cpp
--- hdf5-1.8.9.orig/c++/test/tfile.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/tfile.cpp	2012-06-19 20:26:22.634634000 +0200
@@ -23,6 +23,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/tfilter.cpp hdf5-1.8.9/c++/test/tfilter.cpp
--- hdf5-1.8.9.orig/c++/test/tfilter.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/tfilter.cpp	2012-06-19 20:26:22.640716000 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/th5s.cpp hdf5-1.8.9/c++/test/th5s.cpp
--- hdf5-1.8.9.orig/c++/test/th5s.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/th5s.cpp	2012-06-19 20:26:22.647004000 +0200
@@ -22,6 +22,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/tlinks.cpp hdf5-1.8.9/c++/test/tlinks.cpp
--- hdf5-1.8.9.orig/c++/test/tlinks.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/tlinks.cpp	2012-06-19 20:26:22.665527000 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/trefer.cpp hdf5-1.8.9/c++/test/trefer.cpp
--- hdf5-1.8.9.orig/c++/test/trefer.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/trefer.cpp	2012-06-19 20:26:22.671624000 +0200
@@ -20,6 +20,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/ttypes.cpp hdf5-1.8.9/c++/test/ttypes.cpp
--- hdf5-1.8.9.orig/c++/test/ttypes.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/ttypes.cpp	2012-06-19 20:26:22.677816000 +0200
@@ -19,6 +19,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/c++/test/tvlstr.cpp hdf5-1.8.9/c++/test/tvlstr.cpp
--- hdf5-1.8.9.orig/c++/test/tvlstr.cpp	2012-05-09 17:06:22.000000000 +0200
+++ hdf5-1.8.9/c++/test/tvlstr.cpp	2012-06-19 20:26:22.684971000 +0200
@@ -21,6 +21,7 @@
 
  ***************************************************************************/
 
+#include <mpi.h>
 #ifdef OLD_HEADER_FILENAME
 #include <iostream.h>
 #else
diff -ru hdf5-1.8.9.orig/src/H5make_libsettings.c hdf5-1.8.9/src/H5make_libsettings.c
--- hdf5-1.8.9.orig/src/H5make_libsettings.c	2012-05-09 17:05:58.000000000 +0200
+++ hdf5-1.8.9/src/H5make_libsettings.c	2012-06-19 20:26:22.653140000 +0200
@@ -41,6 +41,7 @@
  *-------------------------------------------------------------------------
  */
 
+#include <mpi.h>
 #include <stdio.h>
 #include <time.h>
 #include "H5private.h"
