Drop custom bool type as C23 already has bool
author: micketeer@gmail.com
--- src/minisat/minisat.h.orig	2026-05-28 22:16:28.006943392 +0200
+++ src/minisat/minisat.h	2026-05-28 22:16:57.923674432 +0200
@@ -34,11 +34,6 @@
 /*====================================================================*/
 /* Simple types: */
 
-typedef int bool;
-
-#define true  1
-#define false 0
-
 typedef int  lit;
 #if 0 /* by mao */
 typedef char lbool;
