--- src/stream_engine.hpp.orig	2013-01-04 20:28:53.549015595 +0100
+++ src/stream_engine.hpp	2013-01-04 20:29:15.384116143 +0100
@@ -94,7 +94,7 @@
 
         //  Size of the greeting message:
         //  Preamble (10 bytes) + version (1 byte) + socket type (1 byte).
-        const static size_t greeting_size = 12;
+        static const size_t greeting_size = 12;
 
         //  True iff we are registered with an I/O poller.
         bool io_enabled;
