diff -ru zeromq-3.2.2.orig/src/stream_engine.hpp zeromq-3.2.2/src/stream_engine.hpp
--- zeromq-3.2.2.orig/src/stream_engine.hpp	2012-11-14 02:09:14.000000000 +0100
+++ zeromq-3.2.2/src/stream_engine.hpp	2014-02-15 13:57:14.074526627 +0100
@@ -94,7 +94,7 @@
 
         //  Size of the greeting message:
         //  Preamble (10 bytes) + version (1 byte) + socket type (1 byte).
-        const static size_t greeting_size = 12;
+        static const size_t greeting_size = 12;
 
         //  True iff we are registered with an I/O poller.
         bool io_enabled;
