Removed debugging flag
J. Sassmannshausen
diff --git a/GEMMA-0.98.5-orig/Makefile b/GEMMA-0.98.5/Makefile
index 662584c..140ef21 100644
--- a/GEMMA-0.98.5-orig/Makefile
+++ b/GEMMA-0.98.5/Makefile
@@ -65,7 +65,7 @@ endif
 
 # Leave blank after "=" to disable; put "= 1" to enable
 DIST_NAME              = gemma-$(GEMMA_VERSION)
-DEBUG                  = 1                # DEBUG mode, set DEBUG=0 for a release
+DEBUG                  = 0                # DEBUG mode, set DEBUG=0 for a release
 PROFILING              =                  # Add profiling info
 SHOW_COMPILER_WARNINGS =
 WITH_OPENBLAS          = 1                # Without OpenBlas uses LAPACK
