commit b6082151e404525973c6a397fd91497417e9ae81
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Fri Mar 21 13:09:27 2025 +0100

    Use external Catch2

diff --git a/Makefile b/Makefile
index 662584c..ea9d07c 100644
--- a/Makefile
+++ b/Makefile
@@ -141,13 +141,13 @@ ifneq ($(CXX), clang++)
   debug check fast-check: CPPFLAGS += -Og -Wfatal-errors
 endif
 
-debug check fast-check: CPPFLAGS += -g $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Icontrib/catch-1.9.7 -Isrc
+debug check fast-check: CPPFLAGS += -g $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Isrc
 
-profile: CPPFLAGS += -g $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Icontrib/catch-1.9.7 -Isrc
+profile: CPPFLAGS += -g $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Isrc
 
-release: CPPFLAGS += -DNDEBUG -O3 $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Icontrib/catch-1.9.7 -Isrc
+release: CPPFLAGS += -DNDEBUG -O3 $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Isrc
 
-static: CPPFLAGS += -DNDEBUG -O3 $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Icontrib/catch-1.9.7 -Isrc
+static: CPPFLAGS += -DNDEBUG -O3 $(GCC_FLAGS) $(GSL_INCLUDE_PATH) -Isrc
 
 
 ifeq ($(SYS), WIN)
@@ -227,7 +227,7 @@ $(OBJS): $(HDR)
 
 .SUFFIXES : .cpp .c .o $(SUFFIXES)
 
-./bin/unittests-gemma: contrib/catch-1.9.7/catch.hpp $(TEST_SRC_DIR)/unittests-main.o $(TEST_SRC_DIR)/unittests-math.o $(OBJS)
+./bin/unittests-gemma: $(TEST_SRC_DIR)/unittests-main.o $(TEST_SRC_DIR)/unittests-math.o $(OBJS)
 	$(CPP) $(CPPFLAGS) $(TEST_SRC_DIR)/unittests-main.o  $(TEST_SRC_DIR)/unittests-math.o $(filter-out src/main.o, $(OBJS)) $(LIBS) -o ./bin/unittests-gemma
 
 unittests: all ./bin/unittests-gemma
diff --git a/test/src/unittests-main.cpp b/test/src/unittests-main.cpp
index 3c3c6f2..afeca98 100644
--- a/test/src/unittests-main.cpp
+++ b/test/src/unittests-main.cpp
@@ -1,2 +1,2 @@
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
-#include <catch.hpp>
+#include <catch2/catch.hpp>
diff --git a/test/src/unittests-math.cpp b/test/src/unittests-math.cpp
index 4872e9a..25ebec3 100644
--- a/test/src/unittests-math.cpp
+++ b/test/src/unittests-math.cpp
@@ -1,4 +1,4 @@
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 #include <iostream>
 #include <fenv.h>
 
