diff --git a/build_system/Makefile.SanityCheck b/build_system/Makefile.SanityCheck
index 7e71290..d5db06f 100644
--- a/build_system/Makefile.SanityCheck
+++ b/build_system/Makefile.SanityCheck
@@ -30,6 +30,11 @@
 # Release: This file is part of the kim-api.git repository.
 #
 
+ifeq ($(SANITY_CHECKED),TRUE) # include guard
+# do nothing
+else
+# set flag and perform sanity check
+export SANITY_CHECKED = TRUE
 
 # check version of make
 NEED_MAKE_VERSION  := 3.81
@@ -39,7 +44,6 @@ ifndef MAKE_VERSION_IS_OK
 endif
 
 # check for existence of compilers
-# check for existence of compilers
 ifeq ($(shell $(CC) --version 2>&1),)
   $(error C compiler "$(CC)" not found)
 endif
@@ -117,3 +121,5 @@ endif
 ifeq ($(flavor XLANGLDLIBS),undefined)
    $(error Make variable XLANGLDLIBS must be defined)
 endif
+
+endif # include guard
