make configure script aware of Intel compilers (icc)
author: Kenneth Hoste (HPC-UGent)
--- mira-4.9.6/configure.orig	2016-05-01 18:27:31.000000000 +0200
+++ mira-4.9.6/configure	2017-12-18 15:50:37.575413022 +0100
@@ -4462,6 +4462,9 @@
 *clang*)
   bachCCOMPILER=clang
 ;;
+*icc*)
+  bachCCOMPILER=icc
+;;
 *)
   bachCCOMPILER=unknown
 ;;
@@ -19921,14 +19924,14 @@
 
 # OK, we're using C++11/14, makes life a whole lot easier
 case "${bachCCOMPILER=unknown}" in
-*gcc*)
+*gcc*|*icc*)
   CXXFLAGS="${CXXFLAGS} -std=c++14"
 ;;
 *clang*)
   CXXFLAGS="${CXXFLAGS} -std=c++14 -stdlib=libc++"
 ;;
 *)
-  echo "The compiler is not GCC nor clang and I need a flag to enable at least C++11 standard, please contact the author!"
+  echo "The compiler ${bachCCOMPILER} is not GCC nor clang and I need a flag to enable at least C++11 standard, please contact the author!"
   exit 100;
 ;;
 esac
