add missing include statements where Intel compilers trip over
author: Kenneth Hoste (HPC-UGent)
--- /tmp/vsc40023/easybuild_build/MIRA/4.0.2/intel-2017b/mira-4.0.2/src/util/dptools.H.orig	2017-12-18 16:59:38.764576048 +0100
+++ /tmp/vsc40023/easybuild_build/MIRA/4.0.2/intel-2017b/mira-4.0.2/src/util/dptools.H	2017-12-18 16:59:54.904866947 +0100
@@ -32,6 +32,7 @@
 #include <iostream>
 
 #include <vector>
+#include <math.h>
 
 #include "stdinc/defines.H"
 
--- /tmp/vsc40023/easybuild_build/MIRA/4.0.2/intel-2017b/mira-4.0.2/src/progs/quirks.C.orig	2017-12-18 17:25:00.235149614 +0100
+++ /tmp/vsc40023/easybuild_build/MIRA/4.0.2/intel-2017b/mira-4.0.2/src/progs/quirks.C	2017-12-18 17:25:10.795167066 +0100
@@ -22,7 +22,7 @@
  *
  */
 
-
+#include <iostream>
 #include <boost/filesystem.hpp>
 
 // make the "tcmalloc: large alloc" messages from TCMallom disappear
