diff --git a/adm/templates/OpenCASCADEConfig.cmake.in.orig b/adm/templates/OpenCASCADEConfig.cmake.in
index e391ddfe8e..f66a3c115a 100644
--- a/adm/templates/OpenCASCADEConfig.cmake.in.orig
+++ b/adm/templates/OpenCASCADEConfig.cmake.in
@@ -30,6 +30,9 @@ endif()
 if (OpenCASCADE_INSTALL_PREFIX MATCHES "/lib$")
   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
 endif()
+if (OpenCASCADE_INSTALL_PREFIX MATCHES "/lib64$")
+  get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
+endif()
 if (OpenCASCADE_INSTALL_PREFIX MATCHES "/libs/${CMAKE_ANDROID_ARCH_ABI}$")
   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
