diff -Naur DaliLite.v5.orig/bin/Makefile DaliLite.v5/bin/Makefile
--- DaliLite.v5.orig/bin/Makefile	2019-06-26 11:23:07.000000000 +0000
+++ DaliLite.v5/bin/Makefile	2021-01-18 21:49:20.264526219 +0000
@@ -5,7 +5,7 @@
 # # start of configuration section                                                             #
 # ##############################################################################################
  SHELL = /bin/sh
- OPENMPI_PATH = /usr/lib64/openmpi/bin
+ OPENMPI_PATH = $($EBROOTOPENMPI)/bin
  SRC = ../src
  BIN = ./
  CMBI = $(SRC)/dsspcmbi
