Remove statement labels from those lines without any statement
author: Alex Domingo (Vrije Universiteit Brussel)
diff --git a/src/pipe.f.orig b/src/pipe.f
index d127f2e..e8096cb 100644
--- a/src/pipe.f.orig
+++ b/src/pipe.f
@@ -36,12 +36,12 @@ c
                 !filnam=constructfilnam(cd1,dalidatpath_1,'.dat')
                 !open(91,file=filnam,status='old',err=11)
                 !goto 12
-11              !filnam=constructfilnam(cd1,dalidatpath_2,'.dat')
+                !filnam=constructfilnam(cd1,dalidatpath_2,'.dat')
                 !open(11,file=filnam,status='old',err=10)
-12              !continue
+                !continue
                 ! try second dalidatpath
                  !       i=0
-200              !       read(91,530,end=219) line
+                 !       read(91,530,end=219) line
                  !       if(line(1:4).ne.'>>>>') goto 200
                  !       read(line,551) ndom
                  !        i=i+1
@@ -50,7 +50,7 @@ c
                 !         read(91,710,end=219) j,domns(j),
 !     $                    ((domseglist(i,k,j),k=1,2),i=1,domns(j))
                 !        end do
-219             !        close(91)
+                !        close(91)
                 !close(91)
 c
 c               output
