diff -Nru pastix-6.0.0-orig/cmake_modules/morse_cmake/modules/find/FindBLAS.cmake pastix-6.0.0/cmake_modules/morse_cmake/modules/find/FindBLAS.cmake
--- pastix-6.0.0-orig/cmake_modules/morse_cmake/modules/find/FindBLAS.cmake	2017-09-29 16:34:31.000000000 +0200
+++ pastix-6.0.0/cmake_modules/morse_cmake/modules/find/FindBLAS.cmake	2018-02-19 11:36:12.727582747 +0100
@@ -481,7 +481,7 @@
       set(BLAS_COMPILER_FLAGS "")
       if (NOT BLA_VENDOR STREQUAL "Intel10_64lp_seq")
 	if (CMAKE_C_COMPILER_ID STREQUAL "Intel")
-	  list(APPEND BLAS_COMPILER_FLAGS "-openmp")
+	  list(APPEND BLAS_COMPILER_FLAGS "-qopenmp")
 	endif()
 	if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
 	  list(APPEND BLAS_COMPILER_FLAGS "-fopenmp")
