Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Description:
 Ensure examples correctly link all required libraries
-----------------------------------------------------------------------------------------
diff -ruN pastix-v6.2.2_p2/example/CMakeLists.txt pastix-v6.2.2_p3.1/example/CMakeLists.txt
--- pastix-v6.2.2_p2/example/CMakeLists.txt	2022-09-09 17:31:34.000000000 +0100
+++ pastix-v6.2.2_p3.1/example/CMakeLists.txt	2022-09-26 15:54:16.000000000 +0100
@@ -55,6 +55,7 @@
   add_executable(${_name_we} ${_file})
 
   target_include_directories( ${_name_we} PRIVATE
+    $<BUILD_INTERFACE:${CBLAS_DIR}/include>
     $<BUILD_INTERFACE:${PASTIX_SOURCE_DIR}>
     $<BUILD_INTERFACE:${PASTIX_BINARY_DIR}> )
   target_link_libraries( ${_name_we} PRIVATE
