diff -Nru pastix_5.2.2.22-orig/src/config/LINUX-GNU.in pastix_5.2.2.22/src/config/LINUX-GNU.in
--- pastix_5.2.2.22-orig/src/config/LINUX-GNU.in	2015-09-01 09:54:45.000000000 +0200
+++ pastix_5.2.2.22/src/config/LINUX-GNU.in	2018-02-20 09:55:57.000000000 +0100
@@ -131,7 +131,7 @@
 ###################################################################
 
 # Show memory usage statistics
-#CCPASTIX   := $(CCPASTIX) -DMEMORY_USAGE
+CCPASTIX   := $(CCPASTIX) -DMEMORY_USAGE
 
 # Show memory usage statistics in solver
 #CCPASTIX   := $(CCPASTIX) -DSTATS_SOPALIN
