# Build with support for complex numbers
# S.D.Pinches, 2020
diff -Nru pastix_5.2.3-orig/scripts/templates/config.in pastix_5.2.3/scripts/templates/config.in
--- pastix_5.2.3-orig/scripts/templates/config.in	2016-09-23 14:25:52.000000000 +0200
+++ pastix_5.2.3/scripts/templates/config.in	2020-06-03 15:19:07.374178000 +0200
@@ -36,9 +36,9 @@
 MPCCPROG:=$(MPCCPROG) -DFORCE_DOUBLE
 
 # uncomment the following lines for float=complex support
-#VERSIONFLT=_complex
-#CCPROG:=$(CCPROG) -DFORCE_COMPLEX
-#MPCCPROG:=$(MPCCPROG) -DFORCE_COMPLEX
+VERSIONFLT=_complex
+CCPROG:=$(CCPROG) -DFORCE_COMPLEX
+MPCCPROG:=$(MPCCPROG) -DFORCE_COMPLEX
 
 # uncomment the following lines for sequential (NOMPI) version
 #VERSIONMPI=_nompi
diff -Nru pastix_5.2.3-orig/src/config/LINUX-GNU.in pastix_5.2.3/src/config/LINUX-GNU.in
--- pastix_5.2.3-orig/src/config/LINUX-GNU.in	2016-09-23 14:25:52.000000000 +0200
+++ pastix_5.2.3/src/config/LINUX-GNU.in	2020-06-03 15:15:28.853214494 +0200
@@ -77,8 +77,8 @@
 CCTYPESFLT := $(CCTYPESFLT) -DPREC_DOUBLE
 
 # Uncomment the following lines for float=complex support
-#VERSIONFLT  = _complex
-#CCTYPESFLT := $(CCTYPESFLT) -DTYPE_COMPLEX
+VERSIONFLT  = _complex
+CCTYPESFLT := $(CCTYPESFLT) -DTYPE_COMPLEX
 
 ###################################################################
 #                          FORTRAN MANGLING                       #
diff -Nru pastix_5.2.3-orig/src/config/LINUX-INTEL.in pastix_5.2.3/src/config/LINUX-INTEL.in
--- pastix_5.2.3-orig/src/config/LINUX-INTEL.in	2016-09-23 14:25:52.000000000 +0200
+++ pastix_5.2.3/src/config/LINUX-INTEL.in	2020-06-03 15:15:49.274254800 +0200
@@ -78,8 +78,8 @@
 CCTYPESFLT := $(CCTYPESFLT) -DPREC_DOUBLE
 
 # Uncomment the following lines for float=complex support
-#VERSIONFLT  = _complex
-#CCTYPESFLT := $(CCTYPESFLT) -DTYPE_COMPLEX
+VERSIONFLT  = _complex
+CCTYPESFLT := $(CCTYPESFLT) -DTYPE_COMPLEX
 
 
 ###################################################################
diff -Nru pastix_5.2.3-orig/src/config/PGI.in pastix_5.2.3/src/config/PGI.in
--- pastix_5.2.3-orig/src/config/PGI.in	2016-09-23 14:25:52.000000000 +0200
+++ pastix_5.2.3/src/config/PGI.in	2020-06-03 15:16:32.726595701 +0200
@@ -77,8 +77,8 @@
 CCTYPESFLT := $(CCTYPESFLT) -DPREC_DOUBLE
 
 # Uncomment the following lines for float=complex support
-#VERSIONFLT  = _complex
-#CCTYPESFLT := $(CCTYPESFLT) -DTYPE_COMPLEX
+VERSIONFLT  = _complex
+CCTYPESFLT := $(CCTYPESFLT) -DTYPE_COMPLEX
 
 
 ###################################################################
