diff -ru mrsfast-ultra-3.3.1.orig/Makefile mrsfast-ultra-3.3.1/Makefile
--- mrsfast-ultra-3.3.1.orig/Makefile	2014-03-31 10:36:25.000000000 +0200
+++ mrsfast-ultra-3.3.1/Makefile	2014-05-07 11:44:59.677675000 +0200
@@ -1,3 +1,4 @@
+CC=icc
 DEBUG := 0
 PROFILE := 0
 MRSFAST_VERSION := "3.3.1"
@@ -18,13 +19,13 @@
 
 mrsfast:
 ifeq ($(shell uname -s),Linux)
-	gcc -w $(objects) -o $@ ${LDFLAGS} ${LIBS}
+	$(CC) -w $(objects) -o $@ ${LDFLAGS} ${LIBS}
 else
-	gcc -Wl,-no_pie -fno-pic -w $(objects) -o $@ ${LDFLAGS} ${LIBS}
+	$(CC) -Wl,-no_pie -fno-pic -w $(objects) -o $@ ${LDFLAGS} ${LIBS}
 endif
 
 snp_indexer: SNPIndexer.o
-	gcc $^ -o $@ ${LDFLAGS} ${LIBS}
+	$(CC) $^ -o $@ ${LDFLAGS} ${LIBS}
 
 clean:
 	@rm -f $(objects)
@@ -41,7 +42,7 @@
 	@ld -r -b binary -o HELP.o HELP
 else
 	@touch HELPstub.c
-	gcc -o HELPstub.o -c HELPstub.c
+	$(CC) -o HELPstub.o -c HELPstub.c
 	ld -r -o HELP.o -sectcreate binary HELP HELP HELPstub.o
 endif
 
@@ -62,7 +63,7 @@
 		$(shell echo "-DSSE4=0")
 else
         	$(eval CFLAGS = $(CFLAGS) \
-        	$(shell gv=`gcc -dumpversion`; \
+        	$(shell gv=`$(CC) -dumpversion`; \
             	    sc=`grep -c "sse4" /proc/cpuinfo`; \
                 	echo $$sc.$$gv | awk -F. '{if($$1>0 && $$2>=4 && $$3>=4) print "-DSSE4=1 -msse4.2"; else print "-DSSE4=0"}'))
 endif
@@ -71,7 +72,7 @@
 		$(shell echo "-DSSE4=0")
 else
         $(eval CFLAGS = $(CFLAGS) \
-        $(shell gv=`gcc -dumpversion`; \
+        $(shell gv=`$(CC) -dumpversion`; \
                 sc=`sysctl -n machdep.cpu.features | grep -c "SSE4"` ;\
                 echo $$sc.$$gv | awk -F. '{if($$1>0 && $$2>=4 && $$3>=4) print "-DSSE4=1 -msse4.2"; else print "-DSSE4=0"}'))
 endif
