# Remove static flag.
# Author: Fokke Dijkstra - University of Groningen
--- Makefile	2011-03-25 12:16:03.000000000 +0100
+++ Makefile	2015-03-12 13:57:32.814999663 +0100
@@ -27,7 +27,7 @@
 #                                    on systems where gcc supports the long
 #                                    long data type and on Windows.
 # 
-CFLAGS=-O2 -I./libsrc -I./pdf  -D_FILE_OFFSET_BITS=64 -static
+CFLAGS=-O2 -I./libsrc -I./pdf  -D_FILE_OFFSET_BITS=64
 
 # executable file names and locations
 BINDIR = executables

