diff -ru ncbi-blast-2.2.27+-src.orig/c++/src/build-system/configure ncbi-blast-2.2.27+-src/c++/src/build-system/configure
--- ncbi-blast-2.2.27+-src.orig/c++/src/build-system/configure	2012-08-08 17:36:55.000000000 +0200
+++ ncbi-blast-2.2.27+-src/c++/src/build-system/configure	2013-01-18 10:41:24.546316596 +0100
@@ -4723,7 +4723,7 @@
    compiler_ver="$icc_ver"
    compiler="ICC"
    ncbi_compiler="ICC"
-   ncbi_compiler_ver="`echo $icc_ver | sed 's%.*Version \([0-9.]*\).*%\1%'`"
+   ncbi_compiler_ver="`echo $icc_ver | sed 's%.*Version \([0-9.]*\).*%\1%'` | sed 's/\.//g'"
              WithFeatures="$WithFeatures${WithFeaturesSep}ICC"; WithFeaturesSep=" "
 elif test "$VAC" = "yes" ; then
    compiler_ver="$vac_ver"
