diff -ru ncbi-blast-2.2.29+-src.orig/c++/src/build-system/configure ncbi-blast-2.2.29+-src/c++/src/build-system/configure
--- ncbi-blast-2.2.29+-src.orig/c++/src/build-system/configure	2013-11-26 08:58:10.000000000 +1300
+++ ncbi-blast-2.2.29+-src/c++/src/build-system/configure	2014-02-20 16:37:25.145550267 +1300
@@ -5381,6 +5381,10 @@
      ncbi_cv_prog_cxx_11=no
      ncbi_cv_prog_c_99=no
      ;;
+   ICC:13?? )
+     ncbi_cv_prog_cxx_11=-std=c++0x
+     ncbi_cv_prog_c_99=no
+     ;;
 esac
 
 echo "$as_me:$LINENO: checking how to enable C++ '11 features in $CXX" >&5
@@ -5873,6 +5877,8 @@
     mips-sgi-irix*:KCC )
       ARCH_CFLAGS="-64"
       ;;
+    *:ICC )
+      ;;
     * )
       { { echo "$as_me:$LINENO: error: Do not know how to compile 64-bit with compiler $CXX $compiler_ver $host:$compiler" >&5
 echo "$as_me: error: Do not know how to compile 64-bit with compiler $CXX $compiler_ver $host:$compiler" >&2;}
@@ -6098,7 +6104,7 @@
         ;;
       ICC:* )
         # Core 2; ICC 11 goes up to SSE4.2 (i7), ICC 12 up to CORE-AVX-I
-        DEF_FAST_FLAGS="-O3 -axSSSE3 -ip"
+        DEF_FAST_FLAGS="-O3 -xHOST"
         ;;
       MIPSpro* )
         # -Ofast would be preferable, but runs into resource limits when
