diff -Naur icon-model.orig/externals/ecrad/radiation/radiation_mcica_acc_sw.F90 icon-model/externals/ecrad/radiation/radiation_mcica_acc_sw.F90
--- icon-model.orig/externals/ecrad/radiation/radiation_mcica_acc_sw.F90        2024-02-14 10:14:44.000000000 +0100
+++ icon-model/externals/ecrad/radiation/radiation_mcica_acc_sw.F90     2024-02-17 01:32:38.621200392 +0100
@@ -364,7 +364,7 @@

         ! Do cloudy-sky calculation
         call cloud_generator_acc(ng, nlev, &
-             &  single_level%iseed(jcol), &
+             &  single_level%iseed(jcol) + 0, & ! HPC SDK inlining workaround
              &  config%cloud_fraction_threshold, &
              &  frac(:,jcol), overlap_param(:,jcol), &
              &  config%cloud_inhom_decorr_scaling, frac_std(:,jcol), &
