diff --git a/src/config_yaml.c b/src/config_yaml.c
index 6cfac2f3..068bbc1c 100644
--- icon-model.orig/externals/yac/src/config_yaml.c
+++ icon-model/externals/yac/src/config_yaml.c
@@ -210,7 +210,7 @@ DEF_NAME_TYPE_PAIRS(
   DEF_NAME_TYPE_PAIR(target, 0));
 
 typedef
-#if defined __NVCOMPILER && (__NVCOMPILER_MAJOR__ <= 22 || __NVCOMPILER_MAJOR__ == 23 && __NVCOMPILER_MINOR__ <= 11)
+#if defined __NVCOMPILER && (__NVCOMPILER_MAJOR__ <= 23 || __NVCOMPILER_MAJOR__ == 24 && __NVCOMPILER_MINOR__ <= 1)
 // Older versions of NVHPC have serious problems with unions that contain
 // pointers that are not the first members.
 struct
@@ -239,7 +239,7 @@ struct interp_method_parameter {
   char const * name;
   enum interp_method_parameter_value_type type;
 
-#if defined __NVCOMPILER && (__NVCOMPILER_MAJOR__ <= 22 || __NVCOMPILER_MAJOR__ == 23 && __NVCOMPILER_MINOR__ <= 11)
+#if defined __NVCOMPILER && (__NVCOMPILER_MAJOR__ <= 23 || __NVCOMPILER_MAJOR__ == 24 && __NVCOMPILER_MINOR__ <= 1)
   // Older versions of NVHPC generate invalid LLVM IR for unions that contain
   // structures with doubles: an attempt to initialize a member of type double
   // that is not the first member of the respective structure will lead to a
