build process does not find its own libraries

diff -ruN jbigkit-2.1.orig/Makefile jbigkit-2.1/Makefile
--- jbigkit-2.1.orig/Makefile	2014-03-27 11:47:15.000000000 -0700
+++ jbigkit-2.1/Makefile	2019-03-21 12:05:55.442478000 -0700
@@ -4,7 +4,7 @@
 CC = gcc
 
 # Options for the compiler: A high optimization level is suggested
-CFLAGS = -O2 -W -Wno-unused-result
+CFLAGS = -O2 -W -Wno-unused-result -I../libjbig
 # CFLAGS = -O -g -W -Wall -Wno-unused-result -ansi -pedantic # -DDEBUG
 
 export CC CFLAGS
diff -ruN jbigkit-2.1.orig/pbmtools/Makefile jbigkit-2.1/pbmtools/Makefile
--- jbigkit-2.1.orig/pbmtools/Makefile	2014-03-27 11:47:15.000000000 -0700
+++ jbigkit-2.1/pbmtools/Makefile	2019-03-21 11:51:23.338623000 -0700
@@ -4,7 +4,7 @@
 CC = gcc
 
 # Options for the compiler
-CFLAGS = -g -O -W -Wall -Wno-unused-result -ansi -pedantic # --coverage
+CFLAGS = -I../libjbig -g -O -W -Wall -Wno-unused-result -ansi -pedantic # --coverage
 CPPFLAGS = -I../libjbig 
 
 .SUFFIXES: .1 .5 .txt $(SUFFIXES)
