--- Makefile.orig	2016-01-11 13:27:20.537470900 +1030
+++ Makefile	2016-01-11 13:27:35.557588464 +1030
@@ -1,11 +1,10 @@
-CC=cc
-CFLAGS=-O3 -Wall -Wextra
+ZLIB=-L$(EBROOTZLIB)/lib
 LDFLAGS=-lz
 ZOPFLI=zopfli/src/zopfli/
 # use gcc and gmake on Solaris
 
 pigz: pigz.o yarn.o try.o ${ZOPFLI}deflate.o ${ZOPFLI}blocksplitter.o ${ZOPFLI}tree.o ${ZOPFLI}lz77.o ${ZOPFLI}cache.o ${ZOPFLI}hash.o ${ZOPFLI}util.o ${ZOPFLI}squeeze.o ${ZOPFLI}katajainen.o
-	$(CC) $(LDFLAGS) -o pigz $^ -lpthread -lm
+	$(CC) $(ZLIB) -o pigz $^ -lpthread -lm $(LDFLAGS)
 	ln -f pigz unpigz
 
 pigz.o: pigz.c yarn.h try.h ${ZOPFLI}deflate.h ${ZOPFLI}util.h
