make build system more flexible
author: Paul Jähne
diff -ru multichoose.orig/Makefile multichoose/Makefile
--- multichoose.orig/Makefile	2014-12-03 18:04:41.000000000 +0100
+++ multichoose/Makefile	2022-01-27 08:53:35.738914531 +0100
@@ -1,17 +1,21 @@
+CC ?=	gcc
+CXX ?=	g++
+
 all: multichoose multipermute
 
 #multichoose_recursive: multichoose_recursive.cpp
-#	g++ multichoose_recursive.cpp -o multichoose_recursive
+#	${CXX} $(LDFLAGS) $(CPPFLAGS) $(CXXFLAGS) multichoose_recursive.cpp -o multichoose_recursive
 
 multichoose: multichoose.cpp multichoose.h
-	g++ multichoose.cpp -o multichoose
+	${CXX} $(LDFLAGS) $(CPPFLAGS) $(CXXFLAGS) multichoose.cpp -o multichoose
 
 multipermute: multipermute.cpp multipermute.h
-	g++ multipermute.cpp -o multipermute
+	${CXX} $(LDFLAGS) $(CPPFLAGS) $(CXXFLAGS) multipermute.cpp -o multipermute
 
 cmultichoose: multichoose.c
-	gcc multichoose.c -o cmultichoose
+	${CC} $(LDFLAGS) $(CPPFLAGS) $(CXXFLAGS) multichoose.c -o cmultichoose
 
 clean:
-	rm cmultichoose
-	rm multichoose
+	rm -f cmultichoose
+	rm -f multichoose
+	rm -f multipermute
