--- ./tools/cpp/unix_cc_configure.bzl.orig    1980-01-01 00:00:00.000000000 +0000
+++ ./tools/cpp/unix_cc_configure.bzl         2017-10-24 14:13:21.618422285 +0000
@@ -189,8 +189,6 @@
           "-headerpad_max_install_names",
           ] if darwin else [
               "-B" + str(repository_ctx.path(cc).dirname),
-              # Always have -B/usr/bin, see https://github.com/bazelbuild/bazel/issues/760.
-              "-B/usr/bin",
               # Gold linker only? Can we enable this by default?
               # "-Wl,--warn-execstack",
               # "-Wl,--detect-odr-violations"
@@ -222,8 +220,6 @@
           # Enable a few more warnings that aren't part of -Wall.
       ] + (["-Wthread-safety", "-Wself-assign"] if darwin else [
           "-B" + escape_string(str(repository_ctx.path(cc).dirname)),
-          # Always have -B/usr/bin, see https://github.com/bazelbuild/bazel/issues/760.
-          "-B/usr/bin",
       ]) + (
           # Disable problematic warnings.
           _add_option_if_supported(repository_ctx, cc, "-Wunused-but-set-parameter") +
