diff -ru ntl-11.4.3.orig/src/DoConfig ntl-11.4.3/src/DoConfig
--- ntl-11.4.3.orig/src/DoConfig	2020-01-04 18:49:16.000000000 +0000
+++ ntl-11.4.3/src/DoConfig	2021-03-22 20:12:33.808309272 +0000
@@ -13,8 +13,8 @@
 
 %MakeVal = (
 
-'CXX'	      => 'g++',
-'CXXFLAGS'    => '-g -O2',
+'CXX'	      => '$(CXX)',
+'CXXFLAGS'    => '$(CXXFLAGS)',
 'CXXAUTOFLAGS'=> '',
 'NOCONTRACT'  => '',
 'AR'	      => 'ar',
diff -ru ntl-11.4.3.orig/src/mfile ntl-11.4.3/src/mfile
--- ntl-11.4.3.orig/src/mfile	2020-01-04 18:49:16.000000000 +0000
+++ ntl-11.4.3/src/mfile	2021-03-22 20:22:20.836184532 +0000
@@ -7,11 +7,11 @@
 
 
 
-CXX=@{CXX}
+CXX:=@{CXX}
 # A C++ compiler, e.g., g++, CC, xlC
 
 
-CXXFLAGS=@{CXXFLAGS}
+CXXFLAGS:=@{CXXFLAGS}
 # Flags for the C++ compiler
 
 CXXAUTOFLAGS=@{CXXAUTOFLAGS}
