diff --git a/gadopt-4.1.0.orig/gadopt/utility.py b/gadopt-4.1.0/gadopt/utility.py
index 51030c8..fa44aef 100644
--- a/gadopt-4.1.0.orig/gadopt/utility.py
+++ b/gadopt-4.1.0/gadopt/utility.py
@@ -8,7 +8,7 @@ from firedrake import outer, ds_v, ds_t, ds_b, CellDiameter, CellVolume, dot, Ja
 from firedrake import sqrt, Function, FiniteElement, TensorProductElement, FunctionSpace, VectorFunctionSpace
 from firedrake import as_vector, SpatialCoordinate, Constant, max_value, min_value, dx, assemble, tanh
 from firedrake import op2, VectorElement, DirichletBC, utils
-from firedrake.__future__ import Interpolator
+from firedrake.__future__ import interpolate
 from firedrake.ufl_expr import extract_unique_domain
 import ufl
 import time
@@ -71,7 +71,7 @@ class TimestepAdaptor:
         # J^-1 u is a discontinuous expression, using op2.MAX it takes the maximum value
         # in all adjacent elements when interpolating it to a continuous function space
         # We do need to ensure we reset ref_vel to zero, as it also takes the max with any previous values
-        self.ref_vel_interpolator = Interpolator(abs(dot(JacobianInverse(self.mesh), self.u)), V, access=op2.MAX)
+        self.ref_vel_interpolate = interpolate(abs(dot(JacobianInverse(self.mesh), self.u)), V, access=op2.MAX) 
 
     def compute_timestep(self):
         max_ts = float(self.dt_const)*self.increase_tolerance
diff --git a/gadopt-4.1.0.orig/pyproject.toml b/gadopt-4.1.0/pyproject.toml
index b73143a..6cd834c 100644
--- a/gadopt-4.1.0.orig/pyproject.toml
+++ b/gadopt-4.1.0/pyproject.toml
@@ -4,11 +4,11 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "gadopt"
+version = "4.1.0"
 authors = [
   { name = "G-ADOPT Contributors" }
 ]
 description = "A computational platform for inverse geodynamics"
-dynamic = ["version"]
 readme = "README.md"
 classifiers = [
   "Programming Language :: Python :: 3",
