--- cmake-3.16.3.orig/Modules/Platform/UnixPaths.cmake	2020-03-31 21:18:14.651896322 +0000
+++ cmake-3.16.3/Modules/Platform/UnixPaths.cmake	2020-04-01 00:08:52.180061633 +0000
@@ -23,7 +23,8 @@
 # search types.
 list(APPEND CMAKE_SYSTEM_PREFIX_PATH
   # Standard
-  /usr/local /usr /
+  $ENV{NIXUSER_PROFILE}
+  # /usr/local /usr /
 
   # CMake install location
   "${_CMAKE_INSTALL_DIR}"
@@ -43,24 +44,29 @@
 
 # Non "standard" but common install prefixes
 list(APPEND CMAKE_SYSTEM_PREFIX_PATH
-  /usr/X11R6
-  /usr/pkg
-  /opt
+  # /usr/X11R6
+  # /usr/pkg
+  # /opt
   )
 
 # List common include file locations not under the common prefixes.
 list(APPEND CMAKE_SYSTEM_INCLUDE_PATH
+  $ENV{NIXUSER_PROFILE}/include
   # X11
-  /usr/include/X11
+  # /usr/include/X11
   )
 
 list(APPEND CMAKE_SYSTEM_LIBRARY_PATH
+  $ENV{NIXUSER_PROFILE}/lib
+  $ENV{NIXUSER_PROFILE}/lib64
   # X11
-  /usr/lib/X11
+  # /usr/lib/X11
   )
 
 list(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
-  /lib /lib32 /lib64 /usr/lib /usr/lib32 /usr/lib64
+  $ENV{NIXUSER_PROFILE}/lib
+  $ENV{NIXUSER_PROFILE}/lib64
+  # /lib /lib32 /lib64 /usr/lib /usr/lib32 /usr/lib64
   )
 
 if(CMAKE_SYSROOT_COMPILE)
