--- cmake-3.16.3/Modules/Platform/UnixPaths.cmake.orig	2020-01-21 15:17:47.000000000 +0000
+++ cmake-3.16.3/Modules/Platform/UnixPaths.cmake	2020-04-01 15:33:56.549844004 +0000
@@ -22,9 +22,6 @@
 # List common installation prefixes.  These will be used for all
 # search types.
 list(APPEND CMAKE_SYSTEM_PREFIX_PATH
-  # Standard
-  /usr/local /usr /
-
   # CMake install location
   "${_CMAKE_INSTALL_DIR}"
   )
@@ -43,24 +40,19 @@
 
 # Non "standard" but common install prefixes
 list(APPEND CMAKE_SYSTEM_PREFIX_PATH
-  /usr/X11R6
-  /usr/pkg
-  /opt
   )
 
 # List common include file locations not under the common prefixes.
 list(APPEND CMAKE_SYSTEM_INCLUDE_PATH
-  # X11
-  /usr/include/X11
+  @libc_dev@/include
   )
 
 list(APPEND CMAKE_SYSTEM_LIBRARY_PATH
-  # X11
-  /usr/lib/X11
+  @libc_lib@/lib @libc_lib@/lib64
   )
 
 list(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
-  /lib /lib32 /lib64 /usr/lib /usr/lib32 /usr/lib64
+  @libc_lib@/lib @libc_lib@/lib64
   )
 
 if(CMAKE_SYSROOT_COMPILE)
@@ -73,15 +65,15 @@
 # parsing the implicit directory information from compiler output.
 set(_CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES_INIT
   ${CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES}
-  "${_cmake_sysroot_compile}/usr/include"
+  @libc_dev@/include
   )
 set(_CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES_INIT
   ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES}
-  "${_cmake_sysroot_compile}/usr/include"
+  @libc_dev@/include
   )
 set(_CMAKE_CUDA_IMPLICIT_INCLUDE_DIRECTORIES_INIT
   ${CMAKE_CUDA_IMPLICIT_INCLUDE_DIRECTORIES}
-  "${_cmake_sysroot_compile}/usr/include"
+  @libc_dev@/include
   )
 
 unset(_cmake_sysroot_compile)
--- cmake-3.16.3/Modules/Platform/WindowsPaths.cmake.orig	2020-01-21 15:17:47.000000000 +0000
+++ cmake-3.16.3/Modules/Platform/WindowsPaths.cmake	2020-04-01 15:23:16.013261616 +0000
@@ -66,7 +66,7 @@
 
 if(CMAKE_CROSSCOMPILING AND NOT CMAKE_HOST_SYSTEM_NAME MATCHES "Windows")
   # MinGW (useful when cross compiling from linux with CMAKE_FIND_ROOT_PATH set)
-  list(APPEND CMAKE_SYSTEM_PREFIX_PATH /)
+  # list(APPEND CMAKE_SYSTEM_PREFIX_PATH /)
 endif()
 
 list(APPEND CMAKE_SYSTEM_INCLUDE_PATH
