diff -ru cmake-3.7.2.original/CMakeLists.txt cmake-3.7.2.patched/CMakeLists.txt
--- cmake-3.7.2.original/CMakeLists.txt	2017-01-13 15:05:40.000000000 +0100
+++ cmake-3.7.2.patched/CMakeLists.txt	2018-07-06 16:06:11.000000000 +0200
@@ -778,7 +778,7 @@
     DESTINATION ${CMAKE_DATA_DIR}
     FILE_PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
     DIRECTORY_PERMISSIONS OWNER_READ OWNER_EXECUTE OWNER_WRITE
-                          GROUP_READ GROUP_EXECUTE
+                          GROUP_READ GROUP_EXECUTE SETGID
                           WORLD_READ WORLD_EXECUTE
     PATTERN "*.sh*" PERMISSIONS OWNER_READ OWNER_EXECUTE OWNER_WRITE
                                 GROUP_READ GROUP_EXECUTE
