VTK has deprecated the vtk* component names.

Åke Sandgren, 20211001
diff -ru avogadrolibs-1.95.1.orig/avogadro/vtk/CMakeLists.txt avogadrolibs-1.95.1/avogadro/vtk/CMakeLists.txt
--- avogadrolibs-1.95.1.orig/avogadro/vtk/CMakeLists.txt	2021-08-27 00:29:23.000000000 +0200
+++ avogadrolibs-1.95.1/avogadro/vtk/CMakeLists.txt	2021-10-01 12:59:32.504221836 +0200
@@ -13,9 +13,9 @@
 
 find_package(VTK
   COMPONENTS
-    vtkRenderingOpenGL2 vtkGUISupportQt vtkDomainsChemistry
-    vtkRenderingVolumeOpenGL2 vtkViewsCore vtkRenderingFreeType
-    vtkChartsCore vtkViewsContext2D vtkRenderingContextOpenGL2
+    RenderingOpenGL2 GUISupportQt DomainsChemistry
+    RenderingVolumeOpenGL2 ViewsCore RenderingFreeType
+    ChartsCore ViewsContext2D RenderingContextOpenGL2
   REQUIRED)
 include_directories(SYSTEM ${VTK_INCLUDE_DIRS})
 set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ${VTK_DEFINITIONS})
