diff -ru ThirdParty-2.2.2.orig/Allwmake ThirdParty-2.2.2/Allwmake
--- ThirdParty-2.2.2.orig/Allwmake	2013-03-05 19:09:55.000000000 +0100
+++ ThirdParty-2.2.2/Allwmake	2013-05-15 09:06:16.834069000 +0200
@@ -110,12 +110,12 @@
     ;;
 
 MPICH)
-    if [ -r $MPI_ARCH_PATH/lib/libmpich.so ]
+    if [ -r $MPI_ARCH_PATH/lib64/libmpich.so ]
     then
         echo "    have $WM_MPLIB shared library ($FOAM_MPI)"
         echo
         echo
-    elif [ -r $MPI_ARCH_PATH/lib/libmpich.a ]
+    elif [ -r $MPI_ARCH_PATH/lib64/libmpich.a ]
     then
         echo "    have $WM_MPLIB static library ($FOAM_MPI)"
         echo
