diff -ru OpenFOAM-v1706.orig/applications/utilities/mesh/manipulation/setSet/Allwmake OpenFOAM-v1706/applications/utilities/mesh/manipulation/setSet/Allwmake
--- OpenFOAM-v1706.orig/applications/utilities/mesh/manipulation/setSet/Allwmake	2019-02-01 20:43:08.582807600 +0000
+++ OpenFOAM-v1706/applications/utilities/mesh/manipulation/setSet/Allwmake	2019-02-01 20:47:04.555751223 +0000
@@ -9,11 +9,11 @@
 #
 # use readline if available
 #
-if [ -f $EBROOTLIBREADLINE/include/readline/readline.h ]
+if [ -f $NIXUSER_PROFILE/include/readline/readline.h ]
 then
     echo "    found <readline/readline.h>  --  enabling readline support."
     export COMP_FLAGS="-DHAS_READLINE"
-    export LINK_FLAGS="-L$EBROOTLIBREADLINE/lib -lreadline -L$EBROOTNCURSES -lncurses"
+    export LINK_FLAGS="-lreadline"
 fi
 
 wmake $targetType
diff -ru OpenFOAM-v1706.orig/etc/config.sh/gperftools OpenFOAM-v1706/etc/config.sh/gperftools
--- OpenFOAM-v1706.orig/etc/config.sh/gperftools	2019-02-01 20:43:07.715815156 +0000
+++ OpenFOAM-v1706/etc/config.sh/gperftools	2019-02-01 20:47:58.859277997 +0000
@@ -49,7 +49,7 @@
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
 GPERFTOOLS_VERSION=gperftools-$EBVERSIONGPERFTOOLS
-GPERFTOOLS_ARCH_PATH=$EBROOTGPERFTOOLS
+GPERFTOOLS_ARCH_PATH=$NIXUSER_PROFILE
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
