diff -ru OpenFOAM-v1812.patched1/etc/config.sh/gperftools OpenFOAM-v1812/etc/config.sh/gperftools
--- OpenFOAM-v1812.patched1/etc/config.sh/gperftools	2019-05-23 20:25:44.397419813 +0000
+++ OpenFOAM-v1812/etc/config.sh/gperftools	2019-05-23 20:37:06.134064663 +0000
@@ -36,6 +36,6 @@
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
 ##EasyBuild## settings
-GPERFTOOLS_ARCH_PATH=$EBROOTGPERFTOOLS
+GPERFTOOLS_ARCH_PATH=$NIXUSER_PROFILE
 
 #------------------------------------------------------------------------------
diff -ru OpenFOAM-v1812.patched1/wmake/scripts/have_readline OpenFOAM-v1812/wmake/scripts/have_readline
--- OpenFOAM-v1812.patched1/wmake/scripts/have_readline	2019-05-23 20:25:43.686417054 +0000
+++ OpenFOAM-v1812/wmake/scripts/have_readline	2019-05-23 20:33:19.096183852 +0000
@@ -61,7 +61,7 @@
     library="libreadline$extLibso"
 
     # System only?
-    header=$(findFirstFile $EBROOTLIBREADLINE/include/$header /usr/include/$header)
+    header=$(findFirstFile $EBROOTLIBREADLINE/include/$header $NIXUSER_PROFILE/include/$header /usr/include/$header)
 
     case "$header" in
     $EBROOTLIBREADLINE/*)
@@ -70,6 +70,12 @@
             $EBROOTLIBREADLINE/lib$WM_COMPILER_LIB_ARCH/$library \
         )
         ;;
+    $NIXUSER_PROFILE/*)
+        library=$(findFirstFile \
+            $NIXUSER_PROFILE/lib/$library \
+            $NIXUSER_PROFILE/lib$WM_COMPILER_LIB_ARCH/$library \
+        )
+        ;;
 
     *)
         library=$(findFirstFile \
