diff -ru WRFV3.ORIG/main/Makefile WRFV3/main/Makefile
--- WRFV3.ORIG/main/Makefile	2012-02-21 16:39:29.377412110 +0100
+++ WRFV3/main/Makefile	2012-02-21 16:41:24.077884305 +0100
@@ -47,6 +47,10 @@
 module_initialize : ../dyn_$(SOLVER)/module_initialize_$(IDEAL_CASE).o
 	( cd ../dyn_$(SOLVER) ;  $(MAKE) module_initialize_$(IDEAL_CASE).o )
 
+# add dependencies so parallel build works
+real_$(SOLVER).o ndown_$(SOLVER).o nup_$(SOLVER).o tc_$(SOLVER).o ideal_nmm.o ideal_em.o ideal.o: ../dyn_$(SOLVER)/module_initialize_$(IDEAL_CASE).o
+
+
 ## prevent real being compiled for OMP -- only for regtesting
 #$(SOLVER)_real : module_initialize real_$(SOLVER).o
 #	$(RANLIB) $(LIBWRFLIB)
