--- CMakeLists.txt.orig	2024-07-04 21:39:45.754107105 +1200
+++ CMakeLists.txt	2024-07-04 21:40:50.286823631 +1200
@@ -152,16 +152,18 @@
 	add_definitions( -DASAGI_NOMPI )
 else( NOMPI )
 	# Find MPI
-	find_package( MPI REQUIRED )
-	if( MPI_LIBRARY )
-		include_directories( SYSTEM ${MPI_C_INCLUDE_PATH} )
-	else()
-		#Workaround for stupid mpicc/mpicxx
-		get_filename_component( MPI_BASE ${MPI_C_COMPILER} PATH )
-		set( MPI_BASE ${MPI_BASE}/.. )
-		include_directories( SYSTEM ${MPI_BASE}/include )
-		link_directories( ${MPI_BASE}/lib )
-	endif()
+#	find_package( MPI REQUIRED )
+#	if( MPI_LIBRARY )
+#		include_directories( SYSTEM ${MPI_C_INCLUDE_PATH} )
+#	else()
+    #Workaround for stupid mpicc/mpicxx
+    get_filename_component( MPI_BASE ${MPI_C_COMPILER} PATH )
+    set( MPI_BASE ${MPI_BASE}/.. )
+    include_directories( SYSTEM ${MPI_BASE}/include )
+    link_directories( ${MPI_BASE}/lib )
+    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+#	endif()
 	set( CMAKE_C_COMPILER ${MPI_C_COMPILER} )
 	set( CMAKE_CXX_COMPILER ${MPI_CXX_COMPILER} )
 	set( CMAKE_Fortran_COMPILER ${MPI_Fortran_COMPILER} )
