Inject correct installation path
Author: Davide Vanzo (Vanderbilt University), ported to Qwt 6.2.0 by Kenneth Hoste (HPC-UGent)

--- qwt-6.2.0/qwtconfig.pri.orig	2021-07-18 13:19:33.656367878 +0200
+++ qwt-6.2.0/qwtconfig.pri	2022-06-22 15:27:47.919619835 +0200
@@ -16,10 +16,10 @@
 # Install paths
 ######################################################################
 
-QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
+QWT_INSTALL_PREFIX = $$(QWT_PREFIX)
 
 unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
+    QWT_INSTALL_PREFIX    = $$(QWT_PREFIX)
     # QWT_INSTALL_PREFIX = /usr/local/qwt-$$QWT_VERSION-qt-$$QT_VERSION
 }
 
