Inject correct installation path and enable pkgconfig file generation
Author: Davide Vanzo (Vanderbilt University)
diff -ru qwt-6.1.4.orig/qwtconfig.pri qwt-6.1.4/qwtconfig.pri
--- qwt-6.1.4.orig/qwtconfig.pri	2019-08-02 09:37:56.689744813 -0500
+++ qwt-6.1.4/qwtconfig.pri	2019-08-02 09:58:08.929778356 -0500
@@ -16,10 +16,10 @@
 # Install paths
 ######################################################################
 
-QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
+QWT_INSTALL_PREFIX = $$(QWT_PREFIX)
 
 unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
+    QWT_INSTALL_PREFIX    = $$(QWT_PREFIX)
     # QWT_INSTALL_PREFIX = /usr/local/qwt-$$QWT_VERSION-qt-$$QT_VERSION
 }
 
@@ -171,5 +171,5 @@
 
 unix {
 
-    #QWT_CONFIG     += QwtPkgConfig
+    QWT_CONFIG     += QwtPkgConfig
 }
