# Pin to older Yarn version for Node compatibility on 2023a
#
--- rstudio-2023.09.1-494/dependencies/common/install-yarn      2023-10-13 14:52:00.000000000 +0000
+++ rstudio-2023.09.1-494.new/dependencies/common/install-yarn  2025-11-27 12:43:37.514083717 +0000
@@ -34,5 +34,5 @@
   YARN_VERSION=$(PATH="${INSTALL_PATH}" yarn --version)
   echo "yarn ${YARN_VERSION} already installed at '$(which yarn)'"
 else
-  npm install -g yarn
+  npm install -g yarn@1.21.1
 fi
diff -ur rstudio-2023.09.1-494/dependencies/common/install-yarn-linux-aarch64 rstudio-2023.09.1-494.new/dependencies/common/install-yarn-linux-aarch64
--- rstudio-2023.09.1-494/dependencies/common/install-yarn-linux-aarch64        2023-10-13 14:52:00.000000000 +0000
+++ rstudio-2023.09.1-494.new/dependencies/common/install-yarn-linux-aarch64    2025-11-27 12:43:58.667908487 +0000
@@ -24,7 +24,7 @@
 PATH="$(pwd)/${NODE_SUBDIR}/bin:${PATH}"

 # install via node
-npm install -g yarn
+npm install -g yarn@1.21.1

 # update INSTALL_PATH
 INSTALL_PATH="${HOME}/.yarn/bin:${HOME}/.config/yarn/global/node_modules/.bin:${PATH}"
