diff -Nru rstudio-1.3.1093.orig/src/cpp/server/ServerOptions.cpp rstudio-1.3.1093/src/cpp/server/ServerOptions.cpp
--- rstudio-1.3.1093.orig/src/cpp/server/ServerOptions.cpp	2021-04-19 12:42:58.217742905 -0400
+++ rstudio-1.3.1093/src/cpp/server/ServerOptions.cpp	2021-04-19 13:00:06.787497937 -0400
@@ -214,7 +214,7 @@
                                       core::system::effectiveUserIsRoot()),
          "run program as daemon")
       ("server-pid-file",
-         value<std::string>(&serverPidFile_)->default_value("/var/run/rstudio-server.pid"),
+         value<std::string>(&serverPidFile_)->default_value("/tmp/rstudio-server.pid"),
          "location of pid file to write (only in daemon mode)")
       ("server-app-armor-enabled",
         value<bool>(&serverAppArmorEnabled_)->default_value(0),
@@ -226,7 +226,7 @@
         value<std::string>(&secureCookieKeyFile_)->default_value(""),
         "path override for secure cookie key")
       ("server-data-dir",
-         value<std::string>(&serverDataDir_)->default_value("/var/run/rstudio-server"),
+         value<std::string>(&serverDataDir_)->default_value("/tmp/rstudio-server"),
          "path to data directory where rstudio server will write run-time state")
       ("server-add-header",
        value<std::vector<std::string>>(&serverAddHeaders_)->default_value(std::vector<std::string>{})->multitoken(),
