diff -Nru rstudio-1.2.1335.orig/src/cpp/core/system/PosixSystem.cpp rstudio-1.2.1335/src/cpp/core/system/PosixSystem.cpp
--- rstudio-1.2.1335.orig/src/cpp/core/system/PosixSystem.cpp	2019-03-28 05:08:20.000000000 +0000
+++ rstudio-1.2.1335/src/cpp/core/system/PosixSystem.cpp	2019-05-04 19:23:19.014327797 +0000
@@ -2195,6 +2195,7 @@
 
    // setup environment
    core::system::Options env;
+   core::system::environment(&env);
    copyEnvironmentVar("PATH", &env);
    copyEnvironmentVar("MANPATH", &env);
    copyEnvironmentVar("LANG", &env);
