diff -ru --new-file minimap2-2.23.orig/minimap2.pc minimap2-2.23/minimap2.pc
--- minimap2-2.23.orig/minimap2.pc	1970-01-01 00:00:00.000000000 +0000
+++ minimap2-2.23/minimap2.pc	2021-11-22 15:53:20.685192223 +0000
@@ -0,0 +1,14 @@
+prefix=~~PREFIX~~
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: minimap2
+Description: minimap2
+URL: https://github.com/lh3/minimap2.git
+Version: ~~VERSION~~
+Requires:
+Cflags: -I${includedir}
+Libs: -L${libdir} -lminimap2
+Libs.private: -pthread
+Requires.private:
