diff -ru minimap2-2.18.orig/Makefile minimap2-2.18/Makefile
--- minimap2-2.18.orig/Makefile	2021-11-22 18:59:45.965125116 +0000
+++ minimap2-2.18/Makefile	2021-11-22 19:05:56.626714837 +0000
@@ -1,5 +1,5 @@
-CFLAGS=		-g -Wall -O2 -Wc++-compat #-Wextra
-CPPFLAGS=	-DHAVE_KALLOC
+CFLAGS :=		-g -Wall -O2 -Wc++-compat $(CFLAGS) #-Wextra
+CPPFLAGS:=	-DHAVE_KALLOC $(CPPFLAGS) 
 INCLUDES=
 OBJS=		kthread.o kalloc.o misc.o bseq.o sketch.o sdust.o options.o index.o chain.o align.o hit.o map.o format.o pe.o esterr.o splitidx.o ksw2_ll_sse.o
 PROG=		minimap2
