diff -ru mii-1.1.1.patch1/init/bash mii-1.1.1/init/bash
--- mii-1.1.1.patch1/init/bash	2021-09-30 13:50:05.504541753 +0000
+++ mii-1.1.1/init/bash	2021-09-30 13:50:18.523625221 +0000
@@ -34,5 +34,5 @@
     # check if being sourced
     [[ -n "${BASH_SOURCE[1]}" ]] && SOURCED=1 || SOURCED=0
 
-    source $THISDIR/common $@
+    source $THISDIR/common "$@"
 }
diff -ru mii-1.1.1.patch1/init/common mii-1.1.1/init/common
--- mii-1.1.1.patch1/init/common	2021-09-30 13:50:05.505541760 +0000
+++ mii-1.1.1/init/common	2021-09-30 13:50:22.478650578 +0000
@@ -20,7 +20,7 @@
 if [ $res = 0 ]; then
     echo "[mii] loading ${mods[@]} ..." 1>&2
     module load ${mods[@]}
-    $@
+    "$@"
 elif [ $res = 2 ]; then
     exec echo $0: $(printf "$(gettext bash '%s: command not found')" $1) 1>&2
 fi
diff -ru mii-1.1.1.patch1/init/zsh mii-1.1.1/init/zsh
--- mii-1.1.1.patch1/init/zsh	2021-09-30 13:50:05.504541753 +0000
+++ mii-1.1.1/init/zsh	2021-09-30 13:50:27.626683584 +0000
@@ -34,5 +34,5 @@
     # check if sourced
     [[ $ZSH_EVAL_CONTEXT =~ '.*file.*' ]] && SOURCED=1 || SOURCED=0
 
-    source $THISDIR/common $@
+    source $THISDIR/common "$@"
 }
