diff -ru mii-1.1.0.orig/init/common mii-1.1.0/init/common
--- mii-1.1.0.orig/init/common	2021-09-08 13:53:56.068454387 +0000
+++ mii-1.1.0/init/common	2021-09-08 13:55:55.360245682 +0000
@@ -2,7 +2,7 @@
 
 # return if not interactive or sourced
 if [[ $- != *i* ]] || [[ "$SOURCED" = "1" ]]; then
-    exec echo "$1: command not found!" 1>&2
+    exec echo $0: $(printf "$(gettext bash '%s: command not found')" $1) 1>&2
 fi
 
 # grab status, don't do anything if not enabled
@@ -10,7 +10,7 @@
 
 if [ "$state" = "disabled" ]; then
     # show a normal command not found prompt
-    exec echo "$1: command not found!" 1>&2
+    exec echo $0: $(printf "$(gettext bash '%s: command not found')" $1) 1>&2
 fi
 
 # run mii select on the command, see if it returns OK
@@ -22,5 +22,5 @@
     module load ${mods[@]}
     $@
 elif [ $res = 2 ]; then
-    echo "$1: command not found!" 1>&2
+    exec echo $0: $(printf "$(gettext bash '%s: command not found')" $1) 1>&2
 fi
