--- arch/configure.defaults.orig	2017-04-05 18:31:51.664114280 +0000
+++ arch/configure.defaults	2017-04-05 18:40:52.610834742 +0000
@@ -24,7 +24,7 @@
 CPP                 = /opt/ibmcmp/xlf/9.1/exe/cpp -P -traditional
 # on NYBlue
 CPP                 = /opt/ibmcmp/xlf/bg/10.1/exe/cpp -P -traditional
-CPPFLAGS            = -DAIX -DIBM4 -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -DAIX -DIBM4 -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 
 ########################################################################################################################
@@ -49,7 +49,7 @@
 LDFLAGS             =
 CFLAGS              = -O    
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -73,7 +73,7 @@
 LDFLAGS             =
 CFLAGS              = -w
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            =
 
@@ -97,7 +97,7 @@
 LDFLAGS             = 
 CFLAGS              = 
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_DOUBLEUNDERSCORE -DBYTESWAP -DLINUXG95 -DIO_NETCDF -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_DOUBLEUNDERSCORE -DBYTESWAP -DLINUXG95 -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -121,7 +121,7 @@
 LDFLAGS             =
 CFLAGS              =
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DBIT32 -DNO_SIGNAL CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 -DNO_SIGNAL CONFIGURE_MPI
 RANLIB              = ranlib
 
 ########################################################################################################################
@@ -144,7 +144,7 @@
 LDFLAGS             =
 CFLAGS              =
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DBIT32 -DNO_SIGNAL CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 -DNO_SIGNAL CONFIGURE_MPI
 RANLIB              = ranlib 
 
 ########################################################################################################################
@@ -169,7 +169,7 @@
 LDFLAGS             = 
 CFLAGS              = -O
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -195,7 +195,7 @@
 LDFLAGS             = -L$(MPI_ROOT)/lib -lmpi
 CFLAGS              = -w -O3
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -219,7 +219,7 @@
 LDFLAGS             = 
 CFLAGS              = -O
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -243,7 +243,7 @@
 LDFLAGS             = 
 CFLAGS              = -w
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -267,7 +267,7 @@
 LDFLAGS             = -L$(MPI_ROOT)/lib -lmpi
 CFLAGS              = -w
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -291,7 +291,7 @@
 LDFLAGS             = 
 CFLAGS              = -w
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -315,7 +315,7 @@
 LDFLAGS             = -g
 CFLAGS              = -O
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 -DMACOS CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 -DMACOS CONFIGURE_MPI
 ARFLAGS             =
 RANLIB              = ranlib
 CC_TOOLS            = 
@@ -340,7 +340,7 @@
 LDFLAGS             = -g
 CFLAGS              = -O
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 -DMACOS CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 -DMACOS CONFIGURE_MPI
 ARFLAGS             =
 RANLIB              = ranlib
 CC_TOOLS            = 
@@ -365,7 +365,7 @@
 LDFLAGS             = 
 CFLAGS              = -w
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 -DMACOS CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 -DMACOS CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -389,7 +389,7 @@
 LDFLAGS             = -g
 CFLAGS              = -g
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_DOUBLEUNDERSCORE -DBYTESWAP -DIO_NETCDF -DBIT32 -DMACOS  CONFIGURE_MPI
+CPPFLAGS            = -D_DOUBLEUNDERSCORE -DBYTESWAP -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 -DMACOS  CONFIGURE_MPI
 ARFLAGS             =
 RANLIB              = ranlib
 CC_TOOLS            = 
@@ -414,7 +414,7 @@
 LDFLAGS             =
 CFLAGS              =
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DIO_NETCDF -DBIT32 -DMACOS -DNO_SIGNAL CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 -DMACOS -DNO_SIGNAL CONFIGURE_MPI
 ARFLAGS             =
 RANLIB              = ranlib
 CC_TOOLS            =
@@ -440,7 +440,7 @@
 LDFLAGS             = # -lgomp 
 CFLAGS              =
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DIO_NETCDF -DBIT32 -DMACOS -DNO_SIGNAL CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 -DMACOS -DNO_SIGNAL CONFIGURE_MPI
 ARFLAGS             =
 RANLIB              = ranlib
 CC_TOOLS            =
@@ -469,7 +469,7 @@
 LDFLAGS             = -Wl,-stack_size,10000000,-stack_addr,0xc000000
 CFLAGS              = -O
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -DAIX -DMACOS -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -DAIX -DMACOS -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 RANLIB              = ranlib
 CC_TOOLS            = 
@@ -496,7 +496,7 @@
 CC                  = CONFIGURE_CC
 CFLAGS              = -O
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -DAIX -DMACOS -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -DAIX -DMACOS -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 RANLIB              = ranlib
 
 ########################################################################################################################
@@ -519,7 +519,7 @@
 LDFLAGS             = -g
 CFLAGS              = -g
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUXG95 -DIO_NETCDF -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUXG95 -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 RANLIB              = ranlib
 CC_TOOLS            =
@@ -548,7 +548,7 @@
 LDFLAGS             =
 CFLAGS              =
 CPP                 = /usr/lib/cpp -P -traditional
-CPPFLAGS            = -DAIX -DIBM4 -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -DAIX -DIBM4 -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 
 ########################################################################################################################
@@ -570,7 +570,7 @@
 LDFLAGS             =    
 CFLAGS              =    
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DALPHA -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32  CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DALPHA -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32  CONFIGURE_MPI
 ARFLAGS             =
 
 ########################################################################################################################
@@ -593,7 +593,7 @@
 LDFLAGS             = -g
 CFLAGS              = -g
 CPP                 = /usr/bin/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUXG95 -DIO_NETCDF -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUXG95 -DIO_NETCDF -DNO_LEAP_CALENDAR -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 
 ########################################################################################################################
@@ -623,7 +623,7 @@
 LDFLAGS             = -g
 CFLAGS              = -g
 CPP                 = cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DWIN32 -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DWIN32 -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 
 ########################################################################################################################
@@ -646,7 +646,7 @@
 LDFLAGS             = -64
 CFLAGS              = -64
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DSGI -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DSGI -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -676,7 +676,7 @@
 LDFLAGS             = 
 CFLAGS              = -w
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            = 
 
@@ -700,7 +700,7 @@
 LDFLAGS             = 
 CFLAGS              = -O
 CPP                 = /usr/ccs/lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 
 ########################################################################################################################
@@ -723,7 +723,7 @@
 LDFLAGS             =
 CFLAGS              = -O2
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            =
 
@@ -747,7 +747,7 @@
 LDFLAGS             =
 CFLAGS              =
 CPP                 = /lib/cpp -P -traditional
-CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DNO_LEAP_CALENDAR -DIO_BINARY -DIO_GRIB1 -DBIT32 CONFIGURE_MPI
 ARFLAGS             =
 CC_TOOLS            =
 
