diff -ur cutlass-4.1.0-orig/examples/CMakeLists.txt cutlass-4.1.0/examples/CMakeLists.txt
--- cutlass-4.1.0-orig/examples/CMakeLists.txt	2025-08-11 17:00:38.794407814 +0200
+++ cutlass-4.1.0/examples/CMakeLists.txt	2025-08-11 17:01:10.589518764 +0200
@@ -163,7 +163,6 @@
   82_blackwell_distributed_gemm
   83_blackwell_sparse_gemm
   84_blackwell_narrow_precision_sparse_gemm
-  88_hopper_fmha
   )
 
   add_subdirectory(${EXAMPLE})
