--- l_mpi_p_4.0.2.003/pset/install.sh.orig  2011-04-04 09:07:35.000000000 +0200
+++ l_mpi_p_4.0.2.003/pset/install.sh   2013-03-27 17:09:41.098297635 +0100
@@ -1936,7 +1936,7 @@
     [[ -z "$dst_dir" ]] && dst_dir=$("$RPM_EXTR_TOOL" -qp --qf %{PREFIXES} "$rpm_path")
     dst_dir=$(echo "$dst_dir" | sed -e"s/\/\{1,\}/\//g")
     dst_dir=$(echo "$dst_dir" | sed -e"s/\/\{1,\}$//g")
-    local db="$INTEL_SDP_PRODUCTS_DB"
+    local db="/tmp/$USER-$RANDOM/easybuild_impi/intel_sdp_products.db"
     local db_dir=$(dirname "$db") || DIE "Unexpected error."
     local cur_dir=$(pwd) || DIE "Unable to find current directory."
     local rpm_file=$(basename "$rpm_path") || DIE "Unexpected error."
@@ -2044,6 +2044,8 @@
         fi
     fi
 
+    rm -rf $(dirname "$db_dir")
+
     return ${ERR_OK}
 } # NONRPM_INSTALL_PACKAGE
 
