Removes test which lead to 'f-block' not available error
Author: J. Saßmannshausen (Imperial College London/UK)
diff --git a/tblite-0.4.0.orig/test/unit/test_ceh.f90 b/tblite-0.4.0/test/unit/test_ceh.f90
index 803a0f1..4f775aa 100644
--- a/tblite-0.4.0.orig/test/unit/test_ceh.f90
+++ b/tblite-0.4.0/test/unit/test_ceh.f90
@@ -64,7 +64,6 @@ contains
          new_unittest("scaled-selfenergy-LiH", test_scaled_selfenergy_lih), &
          new_unittest("scaled-selfenergy-S2", test_scaled_selfenergy_s2), &
          new_unittest("scaled-selfenergy-SiH4", test_scaled_selfenergy_sih4), &
-         new_unittest("scaled-selfenergy-AcCl6", test_scaled_selfenergy_accl6), &
          new_unittest("hamiltonian-H2", test_hamiltonian_h2), &
          new_unittest("hamiltonian-LiH", test_hamiltonian_lih), &
          new_unittest("hamiltonian-S2", test_hamiltonian_s2), &
@@ -76,9 +75,6 @@ contains
          new_unittest("q-mol-h2", test_q_h2), &
          new_unittest("q-mol-lih", test_q_lih), &
          new_unittest("q-mol-sih4", test_q_sih4), &
-         new_unittest("q-mol-cecl3", test_q_cecl3), &
-         new_unittest("q-mol-accl6", test_q_accl6), &
-         new_unittest("q-mol-panp", test_q_panp), &
          new_unittest("q-mol-mb01", test_q_mb01), &
          new_unittest("q-mol-mb02", test_q_mb02), &
          new_unittest("q-mol-mb03", test_q_mb03), &
diff --git a/tblite-0.4.0.orig/test/unit/test_integral_overlap.f90 b/tblite-0.4.0/test/unit/test_integral_overlap.f90
index 445451e..c4b267a 100644
--- a/tblite-0.4.0.orig/test/unit/test_integral_overlap.f90
+++ b/tblite-0.4.0/test/unit/test_integral_overlap.f90
@@ -53,7 +53,6 @@ subroutine collect_integral_overlap(testsuite)
       new_unittest("overlap-f2", test_overlap_f2), &
       new_unittest("overlap-h2", test_overlap_h2), &
       new_unittest("overlap-lih", test_overlap_lih), &
-      new_unittest("overlap-diat-cecl3", test_overlap_diat_cecl3), &
       new_unittest("overlap-grad-ss", test_overlap_grad_ss), &
       new_unittest("overlap-grad-pp", test_overlap_grad_pp), &
       new_unittest("overlap-grad-dd", test_overlap_grad_dd), &
