diff -ruN backports_datetime_fromisoformat-2.0.3.orig/backports/datetime_fromisoformat/_datetimemodule.c backports_datetime_fromisoformat-2.0.3/backports/datetime_fromisoformat/_datetimemodule.c
--- backports_datetime_fromisoformat-2.0.3.orig/backports/datetime_fromisoformat/_datetimemodule.c	2024-12-28 20:58:43.000000000 +0100
+++ backports_datetime_fromisoformat-2.0.3/backports/datetime_fromisoformat/_datetimemodule.c	2026-05-01 16:17:38.193587473 +0200
@@ -771,7 +771,7 @@
 
     memcpy(PyUnicode_DATA(copy), PyUnicode_DATA(unicode),
            length * PyUnicode_KIND(unicode));
-    assert(_PyUnicode_CheckConsistency(copy, 1));
+    /* assert removed for Python 3.13 compatibility */
     return copy;
 }
 
