# Use fltk-config from EB built FLTK
#
# Åke Sandgren, 20180702
diff -ru p4vasp-0.3.30.orig/src/fltk-config.py p4vasp-0.3.30/src/fltk-config.py
--- p4vasp-0.3.30.orig/src/fltk-config.py	2015-10-06 21:52:37.000000000 +0200
+++ p4vasp-0.3.30/src/fltk-config.py	2018-07-02 10:36:32.674048096 +0200
@@ -12,7 +12,5 @@
 if 0 and strip(run("fltk-config --version")) in ["1.1.5","1.1.6","1.1.7","1.1.8"]:
     print run("fltk-config "+" ".join(sys.argv[1:]))
 else:
-    if not os.path.exists("../ext/bin/fltk-config"):
-        os.system("cd ../ext; bash build-fltk.sh >build-fltk.log 2>build-fltk.err")
-    print run("../ext/bin/fltk-config "+" ".join(sys.argv[1:]))
+    print run("fltk-config "+" ".join(sys.argv[1:]))
     
