--- Makefile.orig	2018-11-08 20:53:49.586902426 +0000
+++ Makefile	2018-11-08 20:53:59.293813935 +0000
@@ -13,7 +13,7 @@
 # BINDIR        - where the executable will go (e.g. /usr/bin)       #
 ######################################################################
 
--include install/Configuration.mk
+#-include install/Configuration.mk
 
 P4VCONFIG     = lib/p4vasp/config.py
 VINFO         = vinfo.py
