Fix Issue #22383 - replace static libs by shared ones
--- pyproject.toml.orig	2025-03-05 13:42:52.584413801 +0100
+++ pyproject.toml	2025-03-05 14:10:11.543253729 +0100
@@ -16,6 +16,8 @@
 
 [tool.maturin]
 strip = true
+no-default-features = true
+features = ["extension-module", "snappy", "lz4", "bzip2", "brotli", "xz", "zstd", "gzip", "zlib-shared", "deflate", "blosc2", "use-system-isal-shared", "ideflate-shared", "izlib-shared"]
 
 [project.optional-dependencies]
 dev = [