--- ncurses-5.9-20130406/configure.orig	2013-04-07 00:09:21.000000000 +0200
+++ ncurses-5.9-20130406/configure	2015-06-03 12:20:33.765043843 +0200
@@ -5587,7 +5587,7 @@
 		chmod +x mk_shared_lib.sh
 		;;
 	darwin*) #(vi
-		cf_try_cflags="no-cpp-precomp"
+		#cf_try_cflags="no-cpp-precomp"
 		CC_SHARED_OPTS="-dynamic"
 		MK_SHARED_LIB='${CC} ${CFLAGS} -dynamiclib -install_name ${libdir}/`basename $@` -compatibility_version ${ABI_VERSION} -current_version ${ABI_VERSION} -o $@'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=abi
