diff -r -u freebayes-1.2.0-orig/scripts/freebayes-parallel freebayes-1.2.0-new/scripts/freebayes-parallel
--- freebayes-1.2.0-orig/scripts/freebayes-parallel	2018-04-09 15:09:32.000000000 +0000
+++ freebayes-1.2.0-new/scripts/freebayes-parallel	2019-04-11 10:41:06.518823479 +0000
@@ -36,5 +36,5 @@
 #$command | head -100 | grep "^#" # generate header
 # iterate over regions using gnu parallel to dispatch jobs
 cat "$regionsfile" | parallel -k -j "$ncpus" "${command[@]}" --region {}
-) | ../vcflib/scripts/vcffirstheader \
-    | ../vcflib/bin/vcfstreamsort -w 1000 | vcfuniq # remove duplicates at region edges
+) | vcffirstheader \
+    | vcfstreamsort -w 1000 | vcfuniq # remove duplicates at region edges
