Remove deprecated python libraries
Author: Lara Peeters (UGent)
diff -ruN funannotate-1.8.17.orig/funannotate/library.py funannotate-1.8.17/funannotate/library.py
--- funannotate-1.8.17.orig/funannotate/library.py	2024-03-01 05:02:32.000000000 +0100
+++ funannotate-1.8.17/funannotate/library.py	2025-03-27 16:01:42.053852188 +0100
@@ -1,7 +1,6 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import, division, print_function, unicode_literals
 from contextlib import contextmanager
-from importlib.util import module_for_loader
 from Bio import BiopythonWarning
 import os
 import uuid
