diff -ru CBLAS.orig/Makefile CBLAS/Makefile
--- CBLAS.orig/Makefile	2010-03-05 00:34:22.000000000 +0100
+++ CBLAS/Makefile	2013-05-11 21:43:29.571885362 +0200
@@ -193,3 +193,7 @@
 	( cd src && rm -f a.out core *.o $(CBLIB) )
 	( cd testing && rm -f *.out core *.o x[sdcz]cblat[123] )
 	( cd examples && rm -f *.o cblas_ex1 cblas_ex2 )
+
+shared: alllib
+	( mkdir tmp && cd tmp && cp $(CBLIB) . && ar x $(CBLIB) && $(CC) -shared -o libcblas.so.1.0.0 *.o -Wl,-soname=libcblas.so.1 && cp -p libcblas.so.1.0.0 ../lib && cd ../lib && ln -s libcblas.so.1.0.0 libcblas.so.1 && ln -s libcblas.so.1.0.0 libcblas.so )
+
