--- a/src/timing/TIMING_clock_write.F
+++ b/src/timing/TIMING_clock_write.F
@@ -85,7 +85,7 @@
      integer           :: i_c
      CHARACTER(lchlen) :: ch,time_str,chin
      ch=chin
-     if (total_time(i_c)<600.) then
+     if (total_time(i_c)<86400.) then
        WRITE (chin, '(a,1x,f11.4,"s ",a)') trim(ch),total_time(i_c),trim(cpu_str)
      else
        time_str=time_string(total_time(i_c))
--- a/src/modules/mod_timing.F
+++ b/src/modules/mod_timing.F
@@ -346,7 +346,8 @@
      !
      total_time = real(internal_list%clock(1)%total_time,SP)
      !
-     ch='            [Time-Profile]: '//trim(time_string(total_time))
+     !ch='            [Time-Profile]: '//trim(time_string(total_time))
+     write (ch,'(a,f11.4,a)') '            [Time-Profile]: ',total_time,'s'
      call msg('rn',ch(:100))
      if (log_as_a_file) call LIVE_message(message="[TIMING]"//ch(:100),lfmt="n")
      !
