--- lib/archive/package.list.orig	2020-12-09 21:13:31.274065486 +0000
+++ lib/archive/package.list	2020-12-09 21:20:05.541790539 +0000
@@ -53,7 +53,7 @@
 url_netcdff=https://github.com/yambo-code/yambo/files/855477/$(tarball_netcdff)
 url_hdf5=https://github.com/yambo-code/yambo/files/838459/$(tarball_hdf5)
 url_libxc=https://github.com/yambo-code/yambo/files/783150/$(tarball_libxc)
-url_iotk=https://github.com/yambo-code/yambo/files/783147/$(tarball_iotk)
+url_iotk=https://github.com/yambo-code/yambo/files/962173/$(tarball_iotk)
 url_etsf_io=https://github.com/yambo-code/yambo/files/845218/$(tarball_etsf)
 url_lapack=https://github.com/yambo-code/yambo/files/783151/$(tarball_lapack)
 url_blacs=https://github.com/yambo-code/yambo/files/783152/$(tarball_blacs)
--- configure.orig	2019-12-01 14:05:40.000000000 +0000
+++ configure	2020-12-10 19:26:00.117131668 +0000
@@ -1189,9 +1189,9 @@
     oldincludedir=$ac_optarg ;;
 
   -prefix | --prefix | --prefi | --pref | --pre | --pr | --p)
-    ac_prev=prefix ;;
+     ;;
   -prefix=* | --prefix=* | --prefi=* | --pref=* | --pre=* | --pr=* | --p=*)
-    prefix=$ac_optarg ;;
+     ;;
 
   -program-prefix | --program-prefix | --program-prefi | --program-pref \
   | --program-pre | --program-pr | --program-p)
