Fixes issue https://github.com/dftbplus/dftbplus/issues/1147
author: Alex Domingo (Vrije Universiteit Brussel)
--- cmake/Modules/Findelsi.cmake.orig	2022-11-24 00:37:30.290625000 +0100
+++ cmake/Modules/Findelsi.cmake	2022-11-24 00:38:20.269201000 +0100
@@ -15,6 +15,11 @@
       INTERFACE
       "${ELSI_INCLUDE_DIRS}"
     )
+    target_include_directories(
+      elsi::elsi
+      INTERFACE
+      "${ELSI_INCLUDEDIR}"
+    )
 
     foreach(_lib IN LISTS ELSI_LINK_LIBRARIES)
       if(_lib MATCHES "pexsi")
