Drop tests that fail with Intel

Åke Sandgren, 20211025
diff -ru dftbplus-21.1.orig/test/prog/dftb+/tests dftbplus-21.1/test/prog/dftb+/tests
--- dftbplus-21.1.orig/test/prog/dftb+/tests	2021-05-12 20:23:51.000000000 +0200
+++ dftbplus-21.1/test/prog/dftb+/tests	2021-10-25 16:50:14.035199535 +0200
@@ -109,7 +109,6 @@
 solvers/ice_NTPoly                     #? WITH_ELSI and MPI_PROCS <= 4 and OMP_THREADS == 1
 solvers/Si32C32_OMM                    #? WITH_ELSI and MPI_PROCS <= 4 and OMP_THREADS == 1
 solvers/Si32C32_OMM_sparse             #? WITH_ELSI and MPI_PROCS <= 4 and OMP_THREADS == 1
-solvers/Si384_OMM                      #? WITH_ELSI and MPI_PROCS <= 4 and OMP_THREADS == 1
 solvers/Si32C32_OMM_kpoints            #? WITH_ELSI and MPI_PROCS % 4 == 0 and MPI_PROCS <= 20 and OMP_THREADS == 1
 solvers/Si32C32_OMM_ks                 #? WITH_ELSI and MPI_PROCS % 8 == 0 and MPI_PROCS <= 40 and OMP_THREADS == 1
 solvers/Si32C32_OMM_k_sparse           #? WITH_ELSI and MPI_PROCS % 4 == 0 and MPI_PROCS <= 20 and OMP_THREADS == 1
