--- interproscan.sh.orig	2020-02-24 18:47:30.234667696 +0000
+++ interproscan.sh	2020-02-24 18:48:37.846034641 +0000
@@ -32,9 +32,9 @@
 JAVA_VERSION=$("$JAVA" -Xms32M -Xmx32M -version 2>&1 | head -n 1 | awk -F '"' '{print $2}' )
 JAVA_MAJOR_VERSION_FULL="$( cut -d ';' -f 1 <<< "$JAVA_VERSION" )"
 JAVA_MAJOR_VERSION="${JAVA_MAJOR_VERSION_FULL%%.*}"
-if [[ "${JAVA_MAJOR_VERSION}" -ne "11" ]];
+if [[ "${JAVA_MAJOR_VERSION}" -le "11" ]];
 then
-    printf 'Java version 11 is required to run InterProScan.\n'
+    printf 'Java version >11 is required to run InterProScan.\n'
     printf 'Detected version %s\n' "${JAVA_VERSION}"
     printf 'Please install the correct version.\n'
     exit 1
--- interproscan.properties.orig	2020-02-01 17:34:22.396486122 +0000
+++ interproscan.properties	2020-02-25 05:18:15.441039604 +0000
@@ -91,7 +91,7 @@
 gene3d.hmm.path=data/gene3d/4.2.0/gene3d_main.hmm
 gene3d.model2sf_map.path=data/gene3d/4.2.0/model_to_family_map.tsv
 gene3d.hmmsearch.force=true
-
+gene3d.discontinuous_regs.path=data/gene3d/4.2.0/discontinuous_regs.pkl.py3
 # HAMAP
 hamap.profile.models.path=data/hamap/2019_01/hamap.prf
 hamap.profile.models.dir=data/hamap/2019_01/profiles
