--- psi4.0b5/configure.orig	2013-12-23 15:48:15.547085479 +0100
+++ psi4.0b5/configure	2013-12-23 15:49:03.008030357 +0100
@@ -4977,7 +4977,7 @@
 if test "${am_cv_python_include+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  am_cv_python_include=`${PYTHON}-config --includes`
+  am_cv_python_include=`python ${PYTHON}-config --includes`
 fi
 { $as_echo "$as_me:$LINENO: result: $am_cv_python_include" >&5
 $as_echo "$am_cv_python_include" >&6; }
@@ -4989,7 +4989,7 @@
 if test "${am_cv_python_ldflags+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  am_cv_python_ldflags=`${PYTHON}-config --ldflags`
+  am_cv_python_ldflags=`python ${PYTHON}-config --ldflags`
 fi
 { $as_echo "$as_me:$LINENO: result: $am_cv_python_ldflags" >&5
 $as_echo "$am_cv_python_ldflags" >&6; }
