diff -ru SIFT4G_v2.0.0.orig/sift4g/src/database_search.cpp SIFT4G_v2.0.0/sift4g/src/database_search.cpp
--- SIFT4G_v2.0.0.orig/sift4g/src/database_search.cpp	2018-08-22 15:06:26.624022755 +0000
+++ SIFT4G_v2.0.0/sift4g/src/database_search.cpp	2018-08-22 15:07:40.416372229 +0000
@@ -8,6 +8,7 @@
 
 #include <algorithm>
 #include <limits>
+#include <cmath>
 
 #include "hash.hpp"
 #include "utils.hpp"
Only in SIFT4G_v2.0.0/sift4g/src: .database_search.cpp.swp
