From 343a96210009ae420f633c2cb2daf663f1147dee Mon Sep 17 00:00:00 2001
From: Andrey Alekseenko <al42and@gmail.com>
Date: Thu, 13 Nov 2025 01:36:48 +0100
Subject: [PATCH] Disable optimizations for reference data generation

---
 .../trajectoryanalysis/tests/surfacearea.cpp        | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/gromacs/trajectoryanalysis/tests/surfacearea.cpp b/src/gromacs/trajectoryanalysis/tests/surfacearea.cpp
index 0a823140004..43f2d502901 100644
--- a/src/gromacs/trajectoryanalysis/tests/surfacearea.cpp
+++ b/src/gromacs/trajectoryanalysis/tests/surfacearea.cpp
@@ -64,6 +64,11 @@
 #include "testutils/refdata.h"
 #include "testutils/testasserts.h"
 
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ >= 13 && GMX_SIMD_ARM_SVE \
+        && GMX_SIMD_ARM_SVE_LENGTH_VALUE == 512
+#    define WORKAROUND_GCC13_ARM_SVE512 1
+#endif
+
 namespace gmx
 {
 namespace test
@@ -123,6 +128,11 @@ public:
             addSphere(x[XX], x[YY], x[ZZ], radius);
         }
     }
+
+#ifdef WORKAROUND_GCC13_ARM_SVE512
+#    pragma GCC push_options
+#    pragma GCC optimize("O0")
+#endif
     void translatePoints(real x, real y, real z)
     {
         for (size_t i = 0; i < x_.size(); ++i)
@@ -132,6 +142,9 @@ public:
             x_[i][ZZ] += z;
         }
     }
+#ifdef WORKAROUND_GCC13_ARM_SVE512
+#    pragma GCC pop_options
+#endif
 
     void calculate(int ndots, int flags, bool bPBC)
     {
-- 
GitLab

