diff -Nru gromacs-2022.3.orig/cmake/gmxManageNvccConfig.cmake gromacs-2022.3/cmake/gmxManageNvccConfig.cmake
--- gromacs-2022.3.orig/cmake/gmxManageNvccConfig.cmake	2022-09-02 09:49:50.000000000 +0200
+++ gromacs-2022.3/cmake/gmxManageNvccConfig.cmake	2022-09-14 17:03:57.056501377 +0200
@@ -132,7 +132,7 @@
             message(STATUS "Checking if nvcc accepts flags ${ARGN} - Assuming success when using gcc 7")
         else()
             if(NOT(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11))
-              set(CCBIN "-ccbin ${CUDA_HOST_COMPILER}")
+              set(CCBIN -ccbin ${CUDA_HOST_COMPILER})
             endif()
             execute_process(
                 COMMAND ${CUDA_NVCC_EXECUTABLE} ${ARGN} ${CCBIN} "${CMAKE_SOURCE_DIR}/cmake/TestCUDA.cu"
