diff -bru ./gromacs-5.1.4.orig/share/top/oplsaa.ff/aminoacids.rtp ./gromacs-5.1.4/share/top/oplsaa.ff/aminoacids.rtp
--- ./gromacs-5.1.4.orig/share/top/oplsaa.ff/aminoacids.rtp	2016-09-07 13:00:41.000000000 -0230
+++ ./gromacs-5.1.4/share/top/oplsaa.ff/aminoacids.rtp	2016-10-25 07:11:23.000000000 -0230
@@ -728,7 +728,7 @@
    ND1    opls_511   -0.490     3
    CD2    opls_508    0.015     4
    HD2    opls_146    0.115     4 
-   CE1    opls_506    0.182     5
+   CE1    opls_506    0.295     5
    HE1    opls_146    0.115     5
    NE2    opls_503   -0.570     6
    HE2    opls_504    0.420     6
