Use more EasyBuild convetional installation location
diff -ruN gocryptfs_v2.5.4_src-deps.orig/Makefile gocryptfs_v2.5.4_src-deps/Makefile
--- gocryptfs_v2.5.4_src-deps.orig/Makefile	2025-04-13 11:58:31.000000000 +0200
+++ gocryptfs_v2.5.4_src-deps/Makefile	2025-05-17 21:08:49.138265238 +0200
@@ -23,19 +23,19 @@
 # Keep in sync with uninstall!
 .phony: install
 install:
-	install -Dm755 -t "$(DESTDIR)/usr/bin/" gocryptfs
-	install -Dm755 -t "$(DESTDIR)/usr/bin/" gocryptfs-xray/gocryptfs-xray
-	install -Dm644 -t "$(DESTDIR)/usr/share/man/man1/" Documentation/gocryptfs.1
-	install -Dm644 -t "$(DESTDIR)/usr/share/man/man1/" Documentation/gocryptfs-xray.1
-	install -Dm644 -t "$(DESTDIR)/usr/share/licenses/gocryptfs" LICENSE
+	install -Dm755 -t "$(DESTDIR)/bin/" gocryptfs
+	install -Dm755 -t "$(DESTDIR)/bin/" gocryptfs-xray/gocryptfs-xray
+	install -Dm644 -t "$(DESTDIR)/share/man/man1/" Documentation/gocryptfs.1
+	install -Dm644 -t "$(DESTDIR)/share/man/man1/" Documentation/gocryptfs-xray.1
+	install -Dm644 -t "$(DESTDIR)/share/licenses/gocryptfs" LICENSE
 
 .phony: uninstall
 uninstall:
-	rm -f "$(DESTDIR)/usr/bin/gocryptfs"
-	rm -f "$(DESTDIR)/usr/bin/gocryptfs-xray"
-	rm -f "$(DESTDIR)/usr/share/man/man1/gocryptfs.1"
-	rm -f "$(DESTDIR)/usr/share/man/man1/gocryptfs-xray.1"
-	rm -f "$(DESTDIR)/usr/share/licenses/gocryptfs/LICENSE"
+	rm -f "$(DESTDIR)/bin/gocryptfs"
+	rm -f "$(DESTDIR)/bin/gocryptfs-xray"
+	rm -f "$(DESTDIR)/share/man/man1/gocryptfs.1"
+	rm -f "$(DESTDIR)/share/man/man1/gocryptfs-xray.1"
+	rm -f "$(DESTDIR)/share/licenses/gocryptfs/LICENSE"
 
 .phony: ci
 ci:
