--- pyproject.toml.orig	2025-10-03 18:09:55.709869086 +0200
+++ pyproject.toml	2025-10-03 18:12:51.500613432 +0200
@@ -4,6 +4,7 @@
 
 [project]
 name = "emulators"
+version = "2.4.0"
 description = "Monorepo for emulation team products"
 readme = "README.md"
 authors = [
@@ -24,10 +25,10 @@
 dependencies = [
   "pulser-core==1.6.*",
   "torch>=2.8.0"]  # The version in .pre-commit-config.yaml must match
-dynamic = ["version"]
+# dynamic = ["version"]
 
 [tool.setuptools]
-packages = ["emu_base", "emu_mps", "emu_sv"]
+packages = ["emu_base", "emu_base.math", "emu_mps", "emu_mps.optimatrix", "emu_sv"]
 
 [tool.coverage.run]
 branch = true
