diff -Nru detonate-1.11.orig/ref-eval/sam/Makefile detonate-1.11/ref-eval/sam/Makefile
--- detonate-1.11.orig/ref-eval/sam/Makefile	2021-01-28 10:13:08.008241220 -0500
+++ detonate-1.11/ref-eval/sam/Makefile	2021-01-28 10:34:48.017323770 -0500
@@ -20,7 +20,7 @@
 INCLUDES=	-I.
 SUBDIRS=	. bcftools misc
 LIBPATH=
-LIBCURSES=	-lcurses # -lXCurses
+LIBCURSES=	-lcurses -ltinfo # -lXCurses
 
 .SUFFIXES:.c .o
 
diff -Nru detonate-1.11.orig/rsem-eval/sam/Makefile detonate-1.11/rsem-eval/sam/Makefile
--- detonate-1.11.orig/rsem-eval/sam/Makefile	2021-01-28 10:13:07.393238413 -0500
+++ detonate-1.11/rsem-eval/sam/Makefile	2021-01-28 10:34:29.799236206 -0500
@@ -19,7 +19,7 @@
 INCLUDES=	-I.
 SUBDIRS=	. bcftools misc
 LIBPATH=
-LIBCURSES=	-lcurses # -lXCurses
+LIBCURSES=	-lcurses -ltinfo # -lXCurses
 
 .SUFFIXES:.c .o
 .PHONY: all lib
