diff -ru detonate-1.11.orig/rsem-eval/buildReadIndex.cpp detonate-1.11/rsem-eval/buildReadIndex.cpp
--- detonate-1.11.orig/rsem-eval/buildReadIndex.cpp	2018-11-20 15:34:30.836197966 +0000
+++ detonate-1.11/rsem-eval/buildReadIndex.cpp	2018-11-20 15:38:49.241813520 +0000
@@ -35,15 +35,15 @@
 		streampos pos = fin.tellg();
 		success = true;
 
-		success = (getline(fin, line));
+		success = (getline(fin, line)).good();
 		if (!success) continue;
-		success = (getline(fin, line));
+		success = (getline(fin, line)).good();
 		if (!success) continue;
 
 		if (hasQ) {
-			success = (getline(fin, line));
+			success = (getline(fin, line)).good();
 			if (!success) continue;
-			success = (getline(fin, line));
+			success = (getline(fin, line)).good();
 			if (!success) continue;
 		}
 
diff -ru detonate-1.11.orig/rsem-eval/PairedEndHit.h detonate-1.11/rsem-eval/PairedEndHit.h
--- detonate-1.11.orig/rsem-eval/PairedEndHit.h	2018-11-20 15:34:30.835197975 +0000
+++ detonate-1.11/rsem-eval/PairedEndHit.h	2018-11-20 15:35:31.568637556 +0000
@@ -26,7 +26,7 @@
 
 bool PairedEndHit::read(std::istream& in) {
 	conprb = 0.0;
-    return (in>>sid>>pos>>insertL);
+    return (in>>sid>>pos>>insertL).good();
 }
 
 void PairedEndHit::write(std::ostream& out) {
diff -ru detonate-1.11.orig/rsem-eval/SingleHit.h detonate-1.11/rsem-eval/SingleHit.h
--- detonate-1.11.orig/rsem-eval/SingleHit.h	2018-11-20 15:34:30.836197966 +0000
+++ detonate-1.11/rsem-eval/SingleHit.h	2018-11-20 15:35:09.929837228 +0000
@@ -43,7 +43,7 @@
 
 bool SingleHit::read(std::istream& in) {
 	conprb = 0.0;
-	return (in>>sid>>pos);
+	return (in>>sid>>pos).good();
 }
 
 void SingleHit::write(std::ostream& out) {
