diff -ru Mothur.1.30.2.orig/makefile Mothur.1.30.2/makefile
--- Mothur.1.30.2.orig/makefile	2013-04-19 15:57:36.000000000 +0200
+++ Mothur.1.30.2/makefile	2013-05-17 16:33:15.000000000 +0200
@@ -70,7 +70,7 @@
 
 ifeq  ($(strip $(USEMPI)),yes)
     CXX = mpic++
-    CXXFLAGS += -DUSE_MPI
+    CXXFLAGS += -DUSE_MPI -DMPICH_IGNORE_CXX_SEEK
 endif
 
 # if you want to enable reading and writing of compressed files, set to yes.
