Don't destroy PYTHONPATH when launching subprocesses

Åke Sandgren, 2024-04-09
diff -ru ignite-0.4.13.orig/tests/ignite/distributed/test_launcher.py ignite-0.4.13/tests/ignite/distributed/test_launcher.py
--- ignite-0.4.13.orig/tests/ignite/distributed/test_launcher.py	2023-10-18 09:31:33.000000000 +0200
+++ ignite-0.4.13/tests/ignite/distributed/test_launcher.py	2024-04-09 14:24:37.233741947 +0200
@@ -45,7 +45,7 @@
     import ignite
 
     env = dict(os.environ) if env is None else env
-    env["PYTHONPATH"] = f"{os.path.dirname(ignite.__path__[0])}"
+    #env["PYTHONPATH"] = f"{os.path.dirname(ignite.__path__[0])}"
     process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=env)
     process.wait()
     if process.returncode != 0:
