--- fallbacks/build-abinit-fallbacks.sh.in.orig	2020-10-10 08:08:37.786784297 +0000
+++ fallbacks/build-abinit-fallbacks.sh.in	2020-10-13 17:02:02.971176075 +0000
@@ -1,30 +1,30 @@
 #!/bin/bash
 
 # Init
-fallbacks_prefix="@abinit_builddir@/fallbacks/install_fb/@abi_fc_vendor@/@abi_fc_version@"
+fallbacks_prefix="$(pwd)/install"
 
 # Find and Unpack tarball
-tarfile=`basename $(ls @abinit_srcdir@/fallbacks/*.tar.gz)`
+tarfile=`basename $(ls $(pwd)/*.tar.gz)`
 source=${tarfile%.tar.gz}
 
-mkdir -p $source && tar -xzf @abinit_srcdir@/fallbacks/$tarfile -C $source --strip-components=1
+#mkdir -p $source && tar -xzf $(pwd)/$tarfile -C $source --strip-components=1
 cd $source
 
 # Configure
+config/scripts/make-makefiles-fallbacks
 ./configure \
   --prefix="${fallbacks_prefix}" \
   --with-tardir="${HOME}/.abinit/tarballs" \
-  --with-linalg-incs="@sd_linalg_fcflags@" \
-  --with-linalg-libs="@sd_linalg_libs@" \
-  --with-fc-vendor="@abi_fc_vendor@" \
-  --with-fc-version="@abi_fc_version@" \
-  --disable-bigdft \
-  --disable-atompaw \
-  --disable-wannier90 \
+  --with-linalg-incs="-I${BLAS_INC_DIR}" \
+  --with-linalg-libs="${LIBBLAS}" \
+  --with-fc-vendor="intel" \
+  --with-fc-version="19" \
+  --with-netcdf4=$EBROOTNETCDF \
+  --with-hdf5=$EBROOTHDF5 \
   LIBS_NETCDF4_FORTRAN="-ldl -lm -lz" \
-  CC="@CC@" \
-  CXX="@CXX@" \
-  FC="@FC@"
+  CC="mpicc" \
+  CXX="mpicxx" \
+  FC="mpif90"
 
-make -j 4 install
+make -j 8 install PATCH=patch
 rc=`echo $?`
