--- gamess/comp	2022-10-01 04:59:43.000000000 +0000
+++ gamess/comp	2023-06-22 21:25:12.293497176 +0000
@@ -596,7 +596,7 @@
          endsw
 
          if (".$GMS_DEBUG_FLAGS" != .) set OPT="$GMS_DEBUG_FLAGS"
-         set EXTRAOPT=""
+         set EXTRAOPT="-axCORE-AVX512,CORE-AVX2 -mavx2"
 
          if (($MODULE == cchem) || ($MODULE == ga)) then
             mv $MODULE.f $MODULE.F
@@ -2483,7 +2483,7 @@
          if  ($MODULE == nebpath) set OPT='-O0' # added by authors of NEBPATH implementation
 
          if (".$GMS_DEBUG_FLAGS" != ".") set OPT="$GMS_DEBUG_FLAGS"
-         set EXTRAOPT=""
+         set EXTRAOPT=" -axCORE-AVX512,CORE-AVX2 "
 
          # the following is for Intel Mac and MKL
          if ((`uname` == Darwin) && ($GMS_MATHLIB == mkl)) then
