--- gamess/comp	2021-10-26 03:49:56.000000000 -0500
+++ gamess/comp	2021-12-17 12:59:14.576975597 -0600
@@ -2563,7 +2563,7 @@
          if  ($MODULE == nebpath) set OPT='-O0' # added by authors of NEBPATH implementation
 
          if (".$GMS_DEBUG_FLAGS" != .) set OPT="$GMS_DEBUG_FLAGS"
-         set EXTRAOPT=""
+         set EXTRAOPT="-axCORE-AVX512,CORE-AVX2,SSE4.2"
 
          if (($MODULE == cchem) || ($MODULE == ga)) then
             mv $MODULE.f $MODULE.F
