--- config.orig	2015-03-03 10:33:00.023695000 +0100
+++ config	2015-03-03 10:33:37.858993156 +0100
@@ -452,6 +452,8 @@
             breaksw
          case 4.6:
          case 4.7:
+         case 4.8:
+         case 4.9:
             echo "   Good, the newest gfortran can compile REAL*16 data type."
             breaksw
          default:
@@ -823,6 +825,8 @@
             breaksw
          case 4.6:
          case 4.7:
+         case 4.8:
+         case 4.9:
             echo "   Good, the newest gfortran can compile REAL*16 data type."
             breaksw
          default:
@@ -879,6 +883,8 @@
       breaksw
    case 4.6:
    case 4.7:
+   case 4.8:
+   case 4.9:
       echo "   Good, the newest gfortran can compile REAL*16 data type."
       breaksw
    default:
