--- gamess/gms-files.csh	2020-07-03 01:04:26.000000000 +0000
+++ gamess/gms-files.csh	2020-02-12 20:20:12.000000000 +0000
@@ -29,9 +29,13 @@
 #
 #---quiet---set echo
 setenv AUXDATA $GMSPATH/auxdata
-setenv  EXTBAS /dev/null
+if (! $?EXTBAS ) then
+  setenv  EXTBAS /dev/null
+endif
 setenv  NUCBAS /dev/null
-setenv  EXTCAB /dev/null
+if (! $?EXTCAB ) then
+  setenv  EXTCAB /dev/null
+endif
 setenv  POSBAS /dev/null
 #
 setenv DFTBPAR $AUXDATA/DFTB
