--- ddi/compddi.orig	2020-01-20 15:30:34.898177000 +0000
+++ ddi/compddi	2020-01-20 15:34:42.195534732 +0000
@@ -878,9 +878,9 @@
         case aocc
         case gfortran:
            set CFLAGS = "$CFLAGS -Dgetarg_=_gfortran_getarg_i4"
-           set CFLAGS = "$CFLAGS -Diargc_=_gfortran_iargc"
+           set CFLAGS = "$CFLAGS -Diargc_=_gfortran_iargc -mcmodel=medium"
            set NumUS=1
-           set FORT_FLAGS = "-O2 $ARCH -fdefault-integer-8 -std=legacy"
+           set FORT_FLAGS = "-O2 $ARCH -fdefault-integer-8 -std=legacy -mcmodel=medium"
            breaksw
         case pgfortran
         case pgf77:
--- lked.orig	2020-01-20 15:30:07.080655000 +0000
+++ lked	2020-01-20 15:31:18.282300535 +0000
@@ -562,7 +562,7 @@
          breaksw
       case gfortran:
          set LDR='gfortran'
-         set LDOPTS=' '
+         set LDOPTS=' -mcmodel=medium -fopenmp '
          set FORTLIBS=' '
          if ($GMS_OPENMP == true)    set LDOPTS="$LDOPTS -fopenmp"
          breaksw
--- comp.orig	2020-01-20 15:31:55.444654000 +0000
+++ comp	2020-01-20 16:01:53.294085542 +0000
@@ -2487,7 +2487,7 @@
       set OPT='-O2'
       if ($MODULE == zheev)  set OPT='-O0'  # defensive compiling
 
-      set EXTRAOPT=" "
+      set EXTRAOPT=" -mcmodel=medium "
       if (($MODULE_F:e == f90) || ($MODULE_F:e == F90)) then
          set EXTRAOPT="$EXTRAOPT -ffree-line-length-none"
       else
