Set mcmodel=medium when using intel compilers
--- lked.orig	2020-01-16 19:58:13.669467000 +0000
+++ lked	2020-01-17 15:04:56.867750526 +0000
@@ -568,7 +568,7 @@
          breaksw
       case ifort:
          set LDR='ifort'
-         set LDOPTS='-i8'  # -Wl,-M generates a load map
+         set LDOPTS='-i8 -mcmodel=medium'  # -Wl,-M generates a load map
 #            this auxiliary library seems to have disappeared in ifort 10
          set FORTLIBS=' '
          if ($GMS_IFORT_VERNO < 10) set FORTLIBS='-Vaxlib'
--- comp.orig	2020-01-17 15:05:15.659690000 +0000
+++ comp	2020-01-17 15:04:27.194664947 +0000
@@ -2849,7 +2849,7 @@
       if ($GMS_IFORT_VERNO >= 14) set EXTRAOPT="$EXTRAOPT -qopt-report=0"
       if ($GMS_IFORT_VERNO  < 12) set EXTRAOPT="$EXTRAOPT -w95 -cm"
       if ($GMS_IFORT_VERNO >= 12) then
-          set EXTRAOPT="$EXTRAOPT -warn nousage -inline-level=0"
+          set EXTRAOPT="$EXTRAOPT -warn nousage -inline-level=0 -mcmodel=medium"
           if  ($MODULE =~ "omp*") set EXTRAOPT="$EXTRAOPT -inline-level=2"
           if  ($MODULE =~ "mod_*") set EXTRAOPT="$EXTRAOPT -inline-level=2"
       endif
--- ddi/compddi.orig	2020-01-17 15:05:24.122243000 +0000
+++ ddi/compddi	2020-01-17 15:00:37.872450971 +0000
@@ -896,7 +896,8 @@
            breaksw
         case ifort:
            set NumUS=1
-           set FORT_FLAGS = "-O2 -i8"
+           set FORT_FLAGS = "-O2 -i8 -mcmodel=medium"
+           set CFLAGS = "$CFLAGS -mcmodel=medium"
            if ($GMS_PHI == knc) set FORT_FLAGS="$FORT_FLAGS -mmic"
            breaksw
         case pathf90:
