--- gamess/gms-files.csh	2023-06-22 18:59:54.787050068 +0000
+++ gamess/gms-files.csh	2023-06-22 18:53:30.109712882 +0000
@@ -30,10 +30,17 @@
 set echo
 setenv NO_STOP_MESSAGE 1
 setenv AUXDATA $GMSPATH/auxdata
-setenv  EXTBAS /dev/null
-setenv  NUCBAS $JOB_PATH/$JOB.nucbas
+#setenv  EXTBAS /dev/null
+if (! $?EXTBAS ) then
+  setenv  EXTBAS /dev/null
+endif
+if (! $?NUCBAS ) then
+    setenv  NUCBAS $JOB_PATH/$JOB.nucbas
+endif
 setenv  EXTCAB $AUXDATA/AUXBAS/auxdef21
-setenv  POSBAS $JOB_PATH/$JOB.posbas
+if (! $?POSBAS ) then
+    setenv  POSBAS $JOB_PATH/$JOB.posbas
+endif
 #
 setenv DFTBPAR $AUXDATA/DFTB
 setenv ERICFMT $AUXDATA/ericfmt.dat
