--- gamess/gms-files.csh	2021-10-26 03:49:56.000000000 -0500
+++ gamess/gms-files.csh	2021-12-17 12:48:24.354822823 -0600
@@ -30,10 +30,16 @@
 set echo
 setenv NO_STOP_MESSAGE 1
 setenv AUXDATA $GMSPATH/auxdata
-setenv  EXTBAS /dev/null
-setenv  NUCBAS /dev/null
+if (! $?EXTBAS ) then
+  setenv  EXTBAS /dev/null
+endif
+if (! $?NUCBAS ) then
+    setenv  NUCBAS /dev/null
+endif
+if (! $?POSBAS ) then
+    setenv  POSBAS /dev/null
+endif
 setenv  EXTCAB $AUXDATA/AUXBAS/auxdef21
-setenv  POSBAS /dev/null
 #
 setenv DFTBPAR $AUXDATA/DFTB
 setenv ERICFMT $AUXDATA/ericfmt.dat
