--- gamess/rungms-dev	2021-12-17 13:33:15.434996151 -0600
+++ gamess/rungms-dev	2021-12-17 18:51:26.987023902 -0600
@@ -62,7 +62,7 @@
 #
 # Make life simple and set GMSPATH as the current directory
 #
-set GMSPATH=`pwd`
+set GMSPATH=$EBROOTGAMESSMINUS
 #OVERRIDE GMSPATH
 #
 if ( -e $GMSPATH/install.info ) then
--- gamess/rungms-dev	2021-10-26 03:49:56.000000000 -0500
+++ gamess/rungms-dev	2021-12-17 12:04:36.752629075 -0600
@@ -214,8 +214,8 @@
   if ( -d $HOME/restart ) then
     set USERSCR=$HOME/restart
   else
-    mkdir -p $HOME/restart
-    set USERSCR=$HOME/restart
+    mkdir -p $PWD/restart
+    set USERSCR=$PWD/restart
   endif
 endif
 #
@@ -1053,7 +1053,7 @@
         if ($NNODES == 1) then
           orterun --mca orte_base_help_aggregate 0 \
             --mca orte_tmpdir_base /tmp \
-            --mca btl vader,self,tcp \
+            --mca pml ucx --mca osc ucx --oversubscribe \
           -np $NPROCS \
           --bind-to none \
           -map-by ppr:${PPN2}:socket \
@@ -1062,7 +1062,7 @@
         else
           orterun --mca orte_base_help_aggregate 0 \
             --mca orte_tmpdir_base /tmp \
-            --mca btl vader,self,tcp \
+            --mca pml ucx --mca osc ucx --oversubscribe \
           -np $NPROCS \
           --bind-to none \
           -map-by ppr:${PPN2}:node \
@@ -1083,7 +1083,7 @@
           orterun \
             --mca orte_base_help_aggregate 0 \
             --mca orte_tmpdir_base /tmp \
-            --mca btl vader,self,tcp \
+            --mca pml ucx --mca osc ucx --oversubscribe \
           -np $NPROCS \
           --map-by ppr:${PPN2}:node \
           --report-bindings \
@@ -1410,7 +1410,7 @@
        breaksw
      case orte:
        set echo
-       orterun --mca btl vader,self,tcp -np $NPROCS --bind-to none \
+       orterun --mca pml ucx --mca osc ucx --oversubscribe -np $NPROCS --bind-to none \
                --map-by ppr:${NPROCS}:node \
                --report-bindings \
                $GMSPATH/gamess.cchem.$VERNO.x < /dev/null
