--- gamess/rungms-dev	2021-10-26 08:49:56.000000000 +0000
+++ gamess/rungms-dev	2021-12-21 21:24:57.000000000 +0000
@@ -62,7 +62,7 @@
 #
 # Make life simple and set GMSPATH as the current directory
 #
-set GMSPATH=`pwd`
+set GMSPATH=$EBROOTGAMESSMINUS
 #OVERRIDE GMSPATH
 #
 if ( -e $GMSPATH/install.info ) then
@@ -153,27 +153,28 @@
       set SCR=/scratch/$PBS_JOBID
    endif
 else if ($SCHED == SLURM) then
-  if ($?TMPDIR) then
-     set SCR=$TMPDIR/$SLURM_JOBID
-  else
-    if ( $GMS_HPC_SYSTEM_TARGET == birch ) then
-       set SCR=`pwd`/scratch
-    else
+  if ( ! $?SCR) then
+    if ($?SLURM_TMPDIR) then
+     set SCR=$SLURM_TMDPIR
+    else  if ( -d /scratch/$USER )  then
        set SCR=/scratch/$USER/$SLURM_JOBID
+    else 
+       SCR=`pwd`/scratch
     endif
+    mkdir -p $SCR
+  else
+     echo "SCR defined as $SCR"
   endif
 else
   # Running locally
-  if ( -d /scratch ) then
-     set SCR=/scratch
-  else if ( -d /local ) then
-     if ( -d /local/$USER ) then
-       set SCR=/local/$USER
-     else
-       set SCR=/local
-     endif
-  else
+  if (! $?SCR) then
+   if ($?SCRATCH) then
+     set SCR=$SCRATCH
+   else 
      set SCR=`pwd`/scratch
+     mkdir -p $SCR
+   endif
+  echo "SCR is defined a $SCR" 
   endif
 endif
 #
@@ -214,8 +215,8 @@
   if ( -d $HOME/restart ) then
     set USERSCR=$HOME/restart
   else
-    mkdir -p $HOME/restart
-    set USERSCR=$HOME/restart
+    mkdir -p $PWD/restart
+    set USERSCR=$PWD/restart
   endif
 endif
 #
@@ -1053,7 +1054,7 @@
         if ($NNODES == 1) then
           orterun --mca orte_base_help_aggregate 0 \
             --mca orte_tmpdir_base /tmp \
-            --mca btl vader,self,tcp \
+            --mca pml ucx --mca osc ucx --oversubscribe \
           -np $NPROCS \
           --bind-to none \
           -map-by ppr:${PPN2}:socket \
@@ -1062,7 +1063,7 @@
         else
           orterun --mca orte_base_help_aggregate 0 \
             --mca orte_tmpdir_base /tmp \
-            --mca btl vader,self,tcp \
+            --mca pml ucx --mca osc ucx --oversubscribe \
           -np $NPROCS \
           --bind-to none \
           -map-by ppr:${PPN2}:node \
@@ -1083,7 +1084,7 @@
           orterun \
             --mca orte_base_help_aggregate 0 \
             --mca orte_tmpdir_base /tmp \
-            --mca btl vader,self,tcp \
+            --mca pml ucx --mca osc ucx --oversubscribe \
           -np $NPROCS \
           --map-by ppr:${PPN2}:node \
           --report-bindings \
@@ -1410,7 +1411,7 @@
        breaksw
      case orte:
        set echo
-       orterun --mca btl vader,self,tcp -np $NPROCS --bind-to none \
+       orterun --mca pml ucx --mca osc ucx --oversubscribe -np $NPROCS --bind-to none \
                --map-by ppr:${NPROCS}:node \
                --report-bindings \
                $GMSPATH/gamess.cchem.$VERNO.x < /dev/null
