# Use the settings from easybuild
# by wpoely86@gmail.com
diff -ur multitail-6.2.1.orig/Makefile multitail-6.2.1/Makefile
--- multitail-6.2.1.orig/Makefile	2014-02-16 15:06:54.000000000 +0100
+++ multitail-6.2.1/Makefile	2015-01-23 14:02:43.704383230 +0100
@@ -1,14 +1,14 @@
 include version
 
 UTF8_SUPPORT=yes
-DESTDIR=/
+DESTDIR:=$(DESTDIR)
 CONFIG_FILE=$(DESTDIR)/etc/multitail.conf
 
-CC=gcc
+CC:=$(CC)
 DEBUG=-g -D_FORTIFY_SOURCE=2 # -D_DEBUG # -pg #  -D_DEBUG  #-pg -W -pedantic # -pg #-fprofile-arcs
 ifeq ($(UTF8_SUPPORT),yes)
-LDFLAGS+=-lpanelw -lncursesw -lutil -lm $(DEBUG) -rdynamic
-CFLAGS+=-funsigned-char -D`uname` -O2 -Wall -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\" -DUTF8_SUPPORT
+LDFLAGS:= $(LDFLAGS) -lpanelw -lncursesw -lutil -lm $(DEBUG)
+CFLAGS:= -funsigned-char -D`uname` -Wall -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\" -DUTF8_SUPPORT $(CFLAGS)
 else
 LDFLAGS+=-lpanel -lncurses -lutil -lm $(DEBUG) -rdynamic
 CFLAGS+=-funsigned-char -D`uname` -O2 -Wall -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\"
