Python 3.10.4 for GCCcore-11.3.0 uses jsonschema 4.4.0
Author: Petr Král (INUITS)
diff -u pydicom-seg-0.4.1/pyproject.toml.orig pydicom-seg-0.4.1/pyproject.toml
--- pydicom-seg-0.4.1/pyproject.toml.orig       2023-02-16 11:32:28.351423500 +0100
+++ pydicom-seg-0.4.1/pyproject.toml    2023-06-23 12:08:00.072754917 +0200
@@ -18,7 +18,7 @@
 pydicom = ">=1.4.2"
 SimpleITK = ">1.2.4"
 numpy = "^1.18.0"
-jsonschema = "^3.2.0"
+jsonschema = ">=3.2.0"
 
 [tool.poetry.dev-dependencies]
 pytest = "^5.3.2"
diff -u pydicom-seg-0.4.1/setup.py.orig pydicom-seg-0.4.1/setup.py
--- pydicom-seg-0.4.1/setup.py.orig     2023-02-16 11:34:18.241670100 +0100
+++ pydicom-seg-0.4.1/setup.py  2023-06-23 12:08:22.012960453 +0200
@@ -9,7 +9,7 @@
 
 install_requires = \
 ['SimpleITK>1.2.4',
- 'jsonschema>=3.2.0,<4.0.0',
+ 'jsonschema>=3.2.0',
  'numpy>=1.18.0,<2.0.0',
  'pydicom>=1.4.2']
 
diff -u pydicom-seg-0.4.1/PKG-INFO.orig pydicom-seg-0.4.1/PKG-INFO
--- pydicom-seg-0.4.1/PKG-INFO.orig     2023-02-16 11:34:18.242097400 +0100
+++ pydicom-seg-0.4.1/PKG-INFO  2023-06-23 12:08:38.909784959 +0200
@@ -14,7 +14,7 @@
 Classifier: Programming Language :: Python :: 3.8
 Classifier: Programming Language :: Python :: 3.9
 Requires-Dist: SimpleITK (>1.2.4)
-Requires-Dist: jsonschema (>=3.2.0,<4.0.0)
+Requires-Dist: jsonschema (>=3.2.0)
 Requires-Dist: numpy (>=1.18.0,<2.0.0)
 Requires-Dist: pydicom (>=1.4.2)
 Description-Content-Type: text/markdown
