fix issue https://github.com/ndaniel/fusioncatcher/issues/156
author: Alex Domingo (Vrije Universiteit Brussel)
--- bin/fusioncatcher.py.orig	2020-07-13 15:48:23.133889000 +0200
+++ bin/fusioncatcher.py	2020-07-13 15:48:53.736536839 +0200
@@ -2322,7 +2322,7 @@
         "is in the corresponding PATH!"))
 
     # check version
-    os.system(_BP_+"bbmap.sh --version 2>&1 |head -2 |tail -1 > '%s'" % (outdir('bbmap_version.txt'),))
+    os.system(_BP_+"bbmap.sh --version 2>&1 |grep 'BBMap version' > '%s'" % (outdir('bbmap_version.txt'),))
     last_line = file(outdir('bbmap_version.txt'),'r').readline().lower().rstrip("\r\n")
     correct_version = ('bbmap version 38.44',)
     if last_line not in correct_version:
