diff -ru casper_v0.8.2.orig/util.c casper_v0.8.2/util.c
--- casper_v0.8.2.orig/util.c	2018-11-19 21:42:33.723455436 +0000
+++ casper_v0.8.2/util.c	2018-11-19 21:43:06.833149805 +0000
@@ -16,7 +16,7 @@
 -----------------------------------------------------*/
 void open_outputfile( ofstream &fp, const string & filename ) {
   fp.open( filename.c_str() );
-  if( fp==NULL ) {
+  if( !fp.good() ) {
     cout << "Error: Cannot open output file (" << filename << ")" << endl;
     exit(EXIT_FAILURE);
   }
