From cd35405a5f1874f529639f6cf8c115b2560d227e Mon Sep 17 00:00:00 2001
From: "Ryan S. Elliott" <relliott@umn.edu>
Date: Tue, 11 Dec 2018 12:55:12 -0600
Subject: [PATCH] Fix major version variable name in CM

---
 utils/collections-management.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/collections-management.in b/utils/collections-management.in
index f9a3f0ba..11b617a2 100755
--- a/utils/collections-management.in
+++ b/utils/collections-management.in
@@ -35,7 +35,7 @@
 #
 
 collections_info=@CMAKE_INSTALL_FULL_LIBEXECDIR@/@PROJECT_NAME@/@PROJECT_NAME@-collections-info
-major_version=@PROJECT_MAJOR_VERSION@
+major_version=@PROJECT_VERSION_MAJOR@
 
 make_command="make"
 cmake_command="cmake"
-- 
2.17.2 (Apple Git-113)

