--- build_system/Makefile.ModelDriver.orig	2017-09-14 08:25:08.963005439 +0200
+++ build_system/Makefile.ModelDriver	2017-09-14 08:25:12.284960024 +0200
@@ -98,7 +98,7 @@
 else # everything else (linux and freebsd)
 	$(QUELL)printf "{\n"                                                        > $@
 	$(QUELL)printf "   global:\n"                                              >> $@
-	$(QUELL)printf "           kim_item_type\n"                                >> $@
+	$(QUELL)printf "           kim_item_type;\n"                                >> $@
 	$(QUELL)printf "           $(MODEL_DRIVER_NAME)_init_pointer;\n"           >> $@
 	$(QUELL)printf "           $(MODEL_DRIVER_NAME)_compiled_with_version;\n"  >> $@
 	$(QUELL)printf "\n"                                                        >> $@
