diff -ru src/config/makefile.h.orig src/config/makefile.h
--- src/config/makefile.h.orig  2013-10-16 01:36:46.000000000 +0200
+++ src/config/makefile.h       2014-07-10 10:22:48.381962260 +0200
@@ -1348,7 +1348,7 @@
     CORE_SUBDIRS_EXTRA = blas lapack
          CC = gcc
      RANLIB = ranlib
-  MAKEFLAGS = -j 1 --no-print-directory
+  MAKEFLAGS = -j 8 --no-print-directory
     INSTALL = @echo $@ is built
         CPP = gcc -E -nostdinc -undef -P
    FCONVERT = (/bin/cp $< /tmp/$$$$.c; \
@@ -1714,7 +1714,7 @@
         endif
       endif
       DEFINES  += -DEXT_INT
-      MAKEFLAGS = -j 1 --no-print-directory
+      MAKEFLAGS = -j 8 --no-print-directory
      _CPU = $(shell uname -m  )
      ifeq ($(BLAS_LIB),)
        CORE_SUBDIRS_EXTRA += blas
@@ -1823,7 +1823,7 @@
 endif # end of ia32 bit
     ifeq ($(_CPU),x86_64)
 #
-      MAKEFLAGS = -j 2 --no-print-directory
+      MAKEFLAGS = -j 8 --no-print-directory
       COPTIMIZE = -O1
 ifeq ($(NWCHEM_TARGET),CATAMOUNT)
       FC=pgf90
