--- ./src/config/GNUmakefile.orig	2017-06-14 02:02:10.891142393 +0000
+++ ./src/config/GNUmakefile	2017-06-14 02:04:51.186616246 +0000
@@ -15,13 +15,13 @@
 # force serial compiler in case CC=mpicc used (even though it should not be)
 #
 ifeq ($(TARGET),LINUX)
-  DEPEND_CC = /usr/bin/gcc
+  DEPEND_CC = icc
 endif
 ifeq ($(TARGET),LINUX64)
-  DEPEND_CC = /usr/bin/gcc
+  DEPEND_CC = icc
 endif
 ifeq ($(TARGET),BGQ)
-  DEPEND_CC = /usr/bin/gcc
+  DEPEND_CC = icc
 endif
 
 
@@ -35,10 +35,10 @@
   DEPEND_CC = cc
 endif
 ifeq ($(TARGET),BGL)
- DEPEND_CC=gcc
+ DEPEND_CC=icc
 endif
 ifeq ($(TARGET),BGP)
- DEPEND_CC=gcc
+ DEPEND_CC=icc
 endif
 
 $(BINDIR)/depend.x:	depend.c
