Index: src/config/makefile.h
===================================================================
--- src/config/makefile.h	(revision 28485)
+++ src/config/makefile.h	(revision 28486)
@@ -257,6 +257,13 @@
     ifdef USE_ARUR
         ARFLAGS = rU
     endif
+# strip long paths when FC and/or CC are set from user
+  ifneq ($(FC),f77)
+   _FC = $(notdir $(FC))
+  endif
+  ifneq ($(CC),cc)
+   _CC = $(notdir $(CC))
+  endif
 #
 # Machine specific stuff
 #
