--- relion-3.1.1.orig/src/pipeline_jobs.cpp	2019-05-07 12:53:25.000000000 +0200
+++ relion-3.1.1/src/pipeline_jobs.cpp	2019-05-11 15:23:09.341930500 +0200
@@ -558,7 +558,9 @@
 		replacing["XXXnameXXX"] = outputname;
 		replacing["XXXerrfileXXX"] = outputname + "run.err";
 		replacing["XXXoutfileXXX"] = outputname + "run.out";
-		replacing["XXXqueueXXX"] = joboptions["queuename"].getString();
+       if (!joboptions["queuename"].getString().empty()) {
+           replacing["XXXqosXXX"] = "-q " + joboptions["queuename"].getString();
+       }
 		char * extra_count_text = getenv ("RELION_QSUB_EXTRA_COUNT");
 		const char extra_count_val = (extra_count_text ? atoi(extra_count_text) : 2);
 		for (int i=1; i<=extra_count_val; i++)
