--- configure.orig	2014-08-11 17:18:31.686813247 +0200
+++ configure	2014-08-11 17:18:39.390957858 +0200
@@ -15665,7 +15665,7 @@
 #Check for FLTK
 # as g++ seems needed and not gcc for compilation,
 #just check for presence of fltk-config program
-fltkfile=`pwd`/bin/fltk-config
+fltkfile=`which fltk-config`
 as_ac_File=`$as_echo "ac_cv_file_$fltkfile" | $as_tr_sh`
 { $as_echo "$as_me:$LINENO: checking for $fltkfile" >&5
 $as_echo_n "checking for $fltkfile... " >&6; }
