diff -ru relion-3.1.1.orig/CMakeLists.txt relion-3.1.1/CMakeLists.txt
--- relion-3.1.1.orig/CMakeLists.txt	2021-01-14 18:16:11.032466068 +0000
+++ relion-3.1.1/CMakeLists.txt	2021-01-14 18:16:52.757628329 +0000
@@ -215,10 +215,10 @@
 # -------------------------------------------------------------------------------MPI--
 find_package(MPI REQUIRED)
 
-if ("${MPI_CXX_INCLUDE_DIRS}" STREQUAL "")
-	include_directories("${MPI_CXX_INCLUDE_PATH}")
+if ("${MPI_INCLUDE_DIRS}" STREQUAL "")
+	include_directories("${MPI_INCLUDE_PATH}")
 else()
-	include_directories("${MPI_CXX_INCLUDE_DIRS}")
+	include_directories("${MPI_INCLUDE_DIRS}")
 endif()
 
 message(STATUS "MPI_INCLUDE_PATH : ${MPI_INCLUDE_PATH}")
