Fix version (autoresolve fails)
S.D.Pinches, 10.01.2024
diff -Nru matplotlib-3.8.2-orig/setup.py matplotlib-3.8.2/setup.py
--- matplotlib-3.8.2-orig/setup.py	2023-11-17 20:55:51.000000000 +0100
+++ matplotlib-3.8.2/setup.py	2024-01-10 18:48:34.370649021 +0100
@@ -279,6 +279,7 @@
 setup(  # Finally, pass this all along to setuptools to do the heavy lifting.
     name="matplotlib",
     description="Python plotting package",
+    version="3.8.2",
     author="John D. Hunter, Michael Droettboom",
     author_email="matplotlib-users@python.org",
     url="https://matplotlib.org",
