diff -ru matplotlib-1.4.0.orig/extern/qhull/qhull_a.h matplotlib-1.4.0/extern/qhull/qhull_a.h
--- matplotlib-1.4.0.orig/extern/qhull/qhull_a.h	2014-10-15 12:03:56.290623347 -0500
+++ matplotlib-1.4.0/extern/qhull/qhull_a.h	2014-10-15 12:07:15.553745107 -0500
@@ -102,7 +102,7 @@
 #elif defined(__MWERKS__) && defined(__INTEL__)
 #   define QHULL_OS_WIN
 #endif
-#if defined(__INTEL_COMPILER) && !defined(QHULL_OS_WIN)
+#if defined(__INTEL_COMPILER) && !defined(QHULL_OS_WIN) && !defined(__INTEL_COMPILER)
 template <typename T>
 inline void qhullUnused(T &x) { (void)x; }
 #  define QHULL_UNUSED(x) qhullUnused(x);
