diff --git a/project/Linux-x86_64-gnu.cmake b/project/Linux-x86_64-gnu.cmake
index 6dcf7cb..c189773 100644
--- a/project/Linux-x86_64-gnu.cmake
+++ b/project/Linux-x86_64-gnu.cmake
@@ -18,10 +18,10 @@ set(CMAKE_BUILD_TYPE ${TMP_BUILD_TYPE})
 # enabled or they will fail
 add_definitions(-DLittle_Endian)
 
-set(LAPACK_LIBRARIES "lapack")
+set(LAPACK_LIBRARIES "-lflexiblas")
 message(STATUS "LAPACK_LIBRARIES=${LAPACK_LIBRARIES}")
 
-set(BLAS_LIBRARIES "blas")
+set(BLAS_LIBRARIES "-lflexiblas")
 message(STATUS "BLAS_LIBRARIES=${BLAS_LIBRARIES}")
 
 set(CMAKE_C_FLAGS "-march=native -w -fpic")
