diff --git a/doc/platforms/Linux/EasyBuild/config_intel.py b/doc/platforms/Linux/EasyBuild/config_intel.py
index 125ca3bfc3..64b49fc8d6 100644
--- a/doc/platforms/Linux/EasyBuild/config_intel.py
+++ b/doc/platforms/Linux/EasyBuild/config_intel.py
@@ -1,4 +1,5 @@
-mpicompiler = 'mpiicc'
+mpi = True
+compiler = 'mpiicpx'
 
 # FFTW should be configured from environment variables, but they do
 # not report the correct names for a dynamically loaded library.
