diff --git a/gpaw/cli/sbatch.py b/gpaw/cli/sbatch.py
index 250652a..151cc78 100644
--- a/gpaw/cli/sbatch.py
+++ b/gpaw/cli/sbatch.py
@@ -41,7 +41,7 @@ class CLICommand:
         script += ('OMP_NUM_THREADS=1 '
                    'mpiexec `echo $GPAW_MPI_OPTIONS` gpaw-python {}\n'
                    .format(' '.join(args.arguments[i:])))
-        cmd = ['sbatch', '--export=NONE'] + args.arguments[:i]
+        cmd = ['sbatch'] + args.arguments[:i]
         if args.dry_run:
             print('sbatch command:')
             print(' '.join(cmd))
