--- gpaw-1.4.0/config.py.orig	2018-06-28 16:19:26.372282501 +0200
+++ gpaw-1.4.0/config.py	2018-06-28 16:22:23.366924242 +0200
@@ -154,7 +154,9 @@
                     openblas = True
                     libdir = dir
                     break
-            if openblas:  # prefer openblas
+            if 'MKLROOT' in os.environ:
+                libraries += ['mkl_intel_lp64', 'mkl_sequential', 'mkl_core', 'irc']
+            elif openblas:  # prefer openblas
                 libraries += ['openblas', 'lapack']
                 library_dirs += [libdir]
             else:
