--- customize.py	2017-02-24 15:10:13.635016617 +0100
+++ /home/camp/mikst/Desktop/customize.py	2017-02-24 15:19:02.267012659 +0100
@@ -32,15 +32,21 @@
 # mpicompiler = 'mpicc'  # use None if you don't want to build a gpaw-python
 # mpilinker = 'mpicc'
 # platform_id = ''
-# scalapack = False
+scalapack = True
+libraries = ['readline',
+             'gfortran',
+             'scalapack',
+             'openblas',
+             'xc'] 
 
+extra_compile_args = ['-O2', '-std=c99', '-fPIC', '-Wall']
 # Use ScaLAPACK:
 # Warning! At least scalapack 2.0.1 is required!
 # See https://trac.fysik.dtu.dk/projects/gpaw/ticket/230
 if scalapack:
-    libraries += ['scalapack-openmpi',
-                  'blacsCinit-openmpi',
-                  'blacs-openmpi']
+#    libraries += ['scalapack-openmpi',
+#                  'blacsCinit-openmpi',
+#                  'blacs-openmpi']
     define_macros += [('GPAW_NO_UNDERSCORE_CBLACS', '1')]
     define_macros += [('GPAW_NO_UNDERSCORE_CSCALAPACK', '1')]
 
