diff -ru gpaw-1.3.0/config.py.orig gpaw-1.3.0/config.py
--- gpaw-1.3.0/config.py.orig	2017-11-21 11:17:59.258035671 +0100
+++ gpaw-1.3.0/config.py	2017-11-21 11:28:43.606364292 +0100
@@ -147,7 +147,9 @@
                     openblas = True
                     libdir = dir
                     break
-            if openblas:  # prefer openblas
+            if 'MKLROOT' in os.environ:
+                libraries += ['mkl_intel_lp64', 'mkl_sequential', 'mkl_core', 'irc']
+            elif openblas:  # prefer openblas
                 libraries += ['openblas']
                 library_dirs += [libdir]
             else:
