From 025f680015484e3a127f1dd2c400217075630fbf Mon Sep 17 00:00:00 2001
From: Ask Hjorth Larsen <asklarsen@gmail.com>
Date: Fri, 11 Jan 2019 18:33:29 +0000
Subject: [PATCH 1/2] fix memory corruption in revtpss

---
 c/xc/revtpss.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/c/xc/revtpss.c b/c/xc/revtpss.c
index 28b7655..d60316d 100644
--- a/c/xc/revtpss.c
+++ b/c/xc/revtpss.c
@@ -231,15 +231,14 @@ static void c_revtpss_12(revtpss_params *p, const double *rho, const double *sig
   int nder = (p->common.nspin==XC_UNPOLARIZED) ? 1 : 3;
   for(i=0; i<nder; i++){
     if(p->common.nspin==XC_UNPOLARIZED) dauxdsigma[i] /= 2.;
-    double dCdsigma[i]; 
-    dCdsigma[i]=  dCdcsi*dcsidsigma[i];
-	
+    double dCdsigma =  dCdcsi*dcsidsigma[i];
+
     /* partial derivatives*/
-    de_PKZBdsigma[i] = de_PBEdsigma[i] * (1.0 + C * zsq) + dens * e_PBE * dCdsigma[i] * zsq
-      - zsq * (dens * dCdsigma[i] * aux + (1.0 + C) * dauxdsigma[i]);
+    de_PKZBdsigma[i] = de_PBEdsigma[i] * (1.0 + C * zsq) + dens * e_PBE * dCdsigma * zsq
+      - zsq * (dens * dCdsigma * aux + (1.0 + C) * dauxdsigma[i]);
 
   }
-} 
+}
 
 
 static void 
-- 
1.8.3.1

