diff -ru gpaw-1.3.0/config.py.orig gpaw-1.3.0/config.py
--- gpaw-1.3.0/config.py.orig   2017-11-21 18:35:37.844884537 +0100
+++ gpaw-1.3.0/config.py        2017-11-21 18:40:33.793391908 +0100
@@ -478,7 +478,7 @@
     include_dirs.append(cfgDict['CONFINCLUDEPY'])
     includes = ' '.join(['-I' + incdir for incdir in include_dirs])
 
-    library_dirs.append(cfgDict['LIBPL'])
+    #library_dirs.append(cfgDict['LIBPL'])
     lib_dirs = ' '.join(['-L' + lib for lib in library_dirs])
 
     libs = ' '.join(['-l' + lib for lib in libraries if lib.strip()])
