diff -r -w -u SOAPdenovo2-src-r240.orig/sparsePregraph/build_preArc.cpp SOAPdenovo2-src-r240/sparsePregraph/build_preArc.cpp
--- SOAPdenovo2-src-r240.orig/sparsePregraph/build_preArc.cpp	2013-11-11 11:03:42.236283000 +0100
+++ SOAPdenovo2-src-r240/sparsePregraph/build_preArc.cpp	2013-11-11 11:12:22.841393000 +0100
@@ -21,12 +21,14 @@
  */
 
 
+#include <unistd.h>
 #include "stdinc.h"
 #include "build_preArc.h"
 #include "seq_util.h"
 #include "global.h"
 #include "multi_threads.h"
 
+
 #include <math.h>
 #include "bam.h"
 #include "faidx.h"
diff -r -w -u SOAPdenovo2-src-r240.orig/sparsePregraph/multi_threads.cpp SOAPdenovo2-src-r240/sparsePregraph/multi_threads.cpp
--- SOAPdenovo2-src-r240.orig/sparsePregraph/multi_threads.cpp	2013-11-11 11:03:41.749247000 +0100
+++ SOAPdenovo2-src-r240/sparsePregraph/multi_threads.cpp	2013-11-11 11:10:04.695845000 +0100
@@ -20,6 +20,7 @@
  *
  */
 
+#include <unistd.h>
 #include "multi_threads.h"
 #include "global.h"
 #include "build_graph.h"
diff -r -w -u SOAPdenovo2-src-r240.orig/sparsePregraph/pregraph_sparse.cpp SOAPdenovo2-src-r240/sparsePregraph/pregraph_sparse.cpp
--- SOAPdenovo2-src-r240.orig/sparsePregraph/pregraph_sparse.cpp	2013-11-11 11:03:42.243266000 +0100
+++ SOAPdenovo2-src-r240/sparsePregraph/pregraph_sparse.cpp	2013-11-11 11:10:30.945711000 +0100
@@ -20,6 +20,7 @@
  *
  */
 
+#include <unistd.h>
 #include "global.h"
 #include "stdinc.h"
 #include "core.h"
