honor configured installation prefix for Vamp plugin
author: Kenneth Hoste (HPC-UGent)

--- essentia-2.1_beta5/src/examples/wscript.orig	2019-09-30 19:30:45.546467766 +0200
+++ essentia-2.1_beta5/src/examples/wscript	2019-09-30 19:30:56.046411027 +0200
@@ -163,7 +163,7 @@
         if sys.platform == 'darwin':
             install_path = os.environ['HOME'] + '/Library/Audio/Plug-Ins/Vamp'
         elif sys.platform.startswith('linux'):
-            install_path = '/usr/local/lib/vamp'
+            install_path = '${PREFIX}/lib/vamp'
         else:
             install_path = None

