take into account current $PYTHONPATH when running Python tests
author: Kenneth Hoste (HPC-UGent)

--- essentia-2.1_beta5/wscript.orig	2019-10-01 20:26:06.136279484 +0200
+++ essentia-2.1_beta5/wscript	2019-10-01 20:26:57.286223056 +0200
@@ -338,7 +338,7 @@
     os.system('cp -r src/python/essentia build/python/')
     os.system('cp build/src/python/_essentia*.so build/python/essentia')

-    ret = os.system('PYTHONPATH=build/python %s test/src/unittests/all_tests.py' % sys.executable)
+    ret = os.system('PYTHONPATH=build/python:$PYTHONPATH %s test/src/unittests/all_tests.py' % sys.executable)
     if ret:
         ctx.fatal('failed to run python tests. Check test output')

