--- src/nco++/Makefile.in	2014-01-30 08:10:11.000000000 +0100
+++ src/nco++/Makefile.in	2014-03-06 20:05:44.717417461 +0100
@@ -116,7 +116,7 @@
 am__v_lt_1 = 
 ncap2_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
-	$(CXXFLAGS) $(ncap2_LDFLAGS) $(LDFLAGS) -o $@
+	$(CXXFLAGS) $(LDFLAGS) $(ncap2_LDFLAGS) -o $@
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
