adapt to recent gfortran and easybuild variables
author: Miguel Dias Costa (National University of Singapore)
--- configure/LINUX-X86_64-GFORTRAN-MPI.orig	2023-12-11 13:44:09.206123000 +0800
+++ configure/LINUX-X86_64-GFORTRAN-MPI		2023-12-11 13:45:08.202378000 +0800
@@ -13,21 +13,21 @@
      CPPFLAGS='-D__Linux -D__GNU -D__PARALLEL -D__HAS_FFT_DEFAULT    '
      FFLAGS_GROMOS='-ffixed-form $(FFLAGS)' 
      FFLAGS_GROMOS_MODULES=' $(FFLAGS)' 
-     AR='/usr/bin/ar ruv'
-     RANLIB='/usr/bin/ranlib'
+     AR='ar ruv'
+     RANLIB='ranlib'
      if [ $debug ]; then
-       FFLAGS='-g -O0 -fcheck=all -Wuninitialized -Wall'
+       FFLAGS='-g -O0 -fcheck=all -Wuninitialized -Wall -fallow-argument-mismatch -ffree-line-length-none'
        CFLAGS='-g -O0 -Wall'
      else
-       FFLAGS='-O2 -Wall'
+       FFLAGS='-O2 -Wall -fallow-argument-mismatch -ffree-line-length-none'
        CFLAGS='-O2 -Wall'
      fi
      if [ $omp ]; then
        FFLAGS=${FFLAGS}' -fopenmp'
-       OMP3_DISABLED=`{ ${FC} -v; } 2>&1 | ${GREP} version | ${AWK} '{ print ( $3 < "4.7.0" ) ? "true" : "false" }'`
-       LIBS='/home/manish/lapack-3.5.0/liblapack.a /home/manish/OPENBLAS/0.2.18/lib/libopenblas.a'
+       OMP3_DISABLED="false"
+       LIBS=${LIBLAPACK_MT}
      else
-       LIBS='/home/manish/lapack-3.5.0/liblapack.a /home/manish/OPENBLAS/0.2.18/lib/libopenblas.a'
+       LIBS=${LIBLAPACK_MT}
      fi
      LFLAGS=' '${LIBS}
      if [ $coverage ]; then
