--- fourier/Makefile.orig    2018-07-16 14:29:24.061445092 -0700
+++ fourier/Makefile 2018-07-16 14:29:52.354475255 -0700
@@ -13,14 +13,14 @@
 #LDFLAGS=
 ######################################################
 # intel v8.x/v9.x/v10.x compiler. uncomment for OpenMP
-#OPENMP=-openmp 
+OPENMP=-openmp 
 ######################################################
-#F90=ifort $(OPENMP)
-#F77=ifort
-#LD=$(F90)
-#F90FLAGS=-O3  -pc64 
-#F77FLAGS=-O3  -pc64 
-#LDFLAGS=-i-static
+F90=ifort $(OPENMP)
+F77=ifort
+LD=$(F90)
+F90FLAGS=-O3   
+F77FLAGS=-O3  
+LDFLAGS=
 ######################################################
 # g95 compiler
 #F90=g95
  
