patch to fix FLAGS hardcoding on Makefile
author: Oriol Mula-Valls (HPCNow!)
diff -Naur /dev/shm/MUSCLE/3.8.1551/GCC-8.2.0-2.31.1.orig/Makefile /dev/shm/MUSCLE/3.8.1551/GCC-8.2.0-2.31.1/Makefile
--- /dev/shm/MUSCLE/3.8.1551/GCC-8.2.0-2.31.1.orig/Makefile	2014-07-12 18:42:52.000000000 +0200
+++ /dev/shm/MUSCLE/3.8.1551/GCC-8.2.0-2.31.1/Makefile	2019-10-07 17:10:34.261586204 +0200
@@ -9,7 +9,7 @@
 # On OSX, using -static gives the error "ld: can't locate file for: -lcrt0.o",
 # this is fixed by deleting "-static" from the LDLIBS line.
 
-CFLAGS = -O3 -funroll-loops -Winline -DNDEBUG=1
+CFLAGS = $(CXXFLAGS)
 LDLIBS = -lm -static
 # LDLIBS = -lm
 
