--- gdal-3.0.0/swig/python/setup.py.orig	2019-05-05 11:50:23.000000000 +0200
+++ gdal-3.0.0/swig/python/setup.py	2019-09-25 21:09:07.507467548 +0200
@@ -36,12 +42,18 @@
 if 'CXX' in os.environ and os.environ['CXX'].strip().find(' ') >= 0:
     if os.environ['CXX'].strip().startswith('ccache ') and os.environ['CXX'].strip()[len('ccache '):].find(' ') < 0:
         os.environ['CXX'] = os.environ['CXX'].strip()[len('ccache '):]
+    # if libtool, then the last word is the actual compiler
+    elif "libtool" in os.environ['CXX']:
+        os.environ['CXX'] = os.environ['CXX'].strip().split(' ')[-1]
     else:
         print('WARNING: "CXX=%s" was defined in the environment and contains more than one word. Unsetting it since that is incompatible of distutils' % os.environ['CXX'])
         del os.environ['CXX']
 if 'CC' in os.environ and os.environ['CC'].strip().find(' ') >= 0:
     if os.environ['CC'].strip().startswith('ccache ') and os.environ['CC'].strip()[len('ccache '):].find(' ') < 0:
         os.environ['CC'] = os.environ['CC'].strip()[len('ccache '):]
+    # if libtool, then the last word is the actual compiler
+    elif "libtool" in os.environ['CC']:
+        os.environ['CC'] = os.environ['CC'].strip().split(' ')[-1]
     else:
         print('WARNING: "CC=%s" was defined in the environment and contains more than one word. Unsetting it since that is incompatible of distutils' % os.environ['CC'])
         del os.environ['CC']
