--- dakota-6.15.0-public-src-cli-ORIG/cmake/DakotaFindSystemTPLs.cmake	2021-11-05 18:27:37.000000000 +0100
+++ dakota-6.15.0-public-src-cli/cmake/DakotaFindSystemTPLs.cmake	2022-04-06 09:37:20.000000000 +0200
@@ -47,7 +47,7 @@
 # BMA TODO: Update to use CMake BLAS/LAPACK probes and properly use their output
 macro(dakota_find_linalg)
   if(NOT DEFINED BLAS_LIBS OR NOT DEFINED LAPACK_LIBS)
-    # Historically on MSVC, tried to use CMake config file approach first.  
+    # Historically on MSVC, tried to use CMake config file approach first.
     # Could probably just use the Unix logic below instead...
     if(MSVC)
       find_package(LAPACK REQUIRED NO_MODULE)
@@ -64,9 +64,9 @@
       if(NOT BLAS_LIBS OR NOT LAPACK_LIBS)
 	# if not a system blas and lapack, then look for a cmake built LAPACK
 	# with find_package
-	find_package(LAPACK REQUIRED NO_MODULE)
-	set(BLAS_LIBS blas)
-	set(LAPACK_LIBS lapack)
+	find_package(LAPACK REQUIRED MODULE)
+	set(BLAS_LIBS ${BLAS_LIBRARIES})
+	set(LAPACK_LIBS ${LAPACK_LIBRARIES})
       endif()
     endif()
   endif()
