# Fix LAPACK detection with CMake
diff -ru dakota-6.11.0.src-ORIG/CMakeLists.txt dakota-6.11.0.src/CMakeLists.txt
--- dakota-6.11.0.src-ORIG/CMakeLists.txt	2020-03-27 15:13:32.698872450 +0100
+++ dakota-6.11.0.src/CMakeLists.txt	2020-03-27 16:49:44.825255901 +0100
@@ -296,9 +296,9 @@
     if(NOT BLAS_LIBS OR NOT LAPACK_LIBS)
       # if not a system blas and lapack, then look for a cmake built LAPACK
       # with find_package
-      find_package(LAPACK REQUIRED NO_MODULE)
-      set(BLAS_LIBS blas)
-      set(LAPACK_LIBS lapack)
+      find_package(LAPACK REQUIRED MODULE)
+      set(BLAS_LIBS ${BLAS_LIBRARIES})
+      set(LAPACK_LIBS ${LAPACK_LIBRARIES})
     endif()
   endif()
 endif()
