# author: Sam Moors
diff -ur dakota-6.11.0.src.orig/CMakeLists.txt dakota-6.11.0.src/CMakeLists.txt
--- dakota-6.11.0.src.orig/CMakeLists.txt       2019-11-08 01:19:53.000000000 +0100
+++ dakota-6.11.0.src/CMakeLists.txt    2022-09-01 16:38:30.349794048 +0200
@@ -360,7 +360,7 @@
   set(Boost_USE_STATIC_LIBS TRUE)
 endif()
 # Dakota requires Boost 1.49 or newer; enforce for all libs in the build
-find_package(Boost 1.49 REQUIRED COMPONENTS "${dakota_boost_components}")
+find_package(Boost 1.49 REQUIRED COMPONENTS ${dakota_boost_components})
 # This cache variable is used to package the Boost liraries on Darwin (see
 # cmake/InstallDarwinDylibs.cmake). The variable Boost_LIBRARY_DIRS is set
 # in this scope by the CMake Boost probe (at least for CMake 3.5.2), but