diff -Nru --no-dereference dakota-6.16.0-public-src-cli.org/test/dakota_test.perl dakota-6.16.0-public-src-cli/test/dakota_test.perl
--- dakota-6.16.0-public-src-cli.org/test/dakota_test.perl	2022-08-11 10:33:35.000000000 +0200
+++ dakota-6.16.0-public-src-cli/test/dakota_test.perl	2022-08-11 10:34:21.000000000 +0200
@@ -34,7 +34,7 @@
 my @test_inputs = ();        # input files to run or extract
 my $test_num = undef;        # undef since can be zero
 my $test_props_dir = "";     # write test properties to this directory
-my $using_srun = 0;          # use srun to launch parallel dakota (and serial on Cray)
+my $using_srun = 1;          # use srun to launch parallel dakota (and serial on Cray)
 my $using_aprun = 0;         # use aprun to launch both serial and parallel dakota
 my $on_cray = 0;             # Testing is being done on cray
 my $run_valgrind = 0;        # boolean for whether to run valgrind
