--- install.sh.orig	2014-04-03 10:46:46.000000000 +0300
+++ install.sh	2015-06-17 10:09:32.000000000 +0300
@@ -820,8 +820,8 @@
           DO_PYTHON=1
           DO_PYMOD=1
         elif [ "${WITH_PYTHON}" = "false" ]; then
-          DO_PYTHON=1
-          DO_PYMOD=1
+          DO_PYTHON=0
+          DO_PYMOD=0
         else
           echoRed "Parameter --python only accept true or false values. Ignored and assuming default value."
         fi
