diff -ru xmipp/libraries/data/metadata_sql.h xmipp_working/libraries/data/metadata_sql.h
--- xmipp/libraries/data/metadata_sql.h	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/metadata_sql.h	2015-03-12 17:31:49.701372176 +0100
@@ -29,7 +29,7 @@
 #include <iostream>
 #include <map>
 #include "xmipp_strings.h"
-#include <external/sqlite-3.6.23/sqlite3.h>
+#include <sqlite3.h>
 #include "metadata_label.h"
 #include <vector>
 class MDSqlStaticInit;
diff -ru xmipp/libraries/data/rwHDF5.h xmipp_working/libraries/data/rwHDF5.h
--- xmipp/libraries/data/rwHDF5.h	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/rwHDF5.h	2015-03-12 17:32:16.060809000 +0100
@@ -26,7 +26,7 @@
 #ifndef RWHDF5_H_
 #define RWHDF5_H_
 
-#include "../../external/hdf5-1.8.10/src/hdf5.h"
+#include "hdf5.h"
 
 
 
diff -ru xmipp/libraries/data/rwIMAGIC.cpp xmipp_working/libraries/data/rwIMAGIC.cpp
--- xmipp/libraries/data/rwIMAGIC.cpp	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/rwIMAGIC.cpp	2015-03-12 17:35:53.292172326 +0100
@@ -40,6 +40,8 @@
 */
 
 #define IMAGICSIZE 1024 // Size of the IMAGIC header for each image
+#define SIZEOF_INT sizeof(int)
+
 
 ///@defgroup Imagic Imagic File format
 ///@ingroup ImageFormats
diff -ru xmipp/libraries/data/rwJPEG.cpp xmipp_working/libraries/data/rwJPEG.cpp
--- xmipp/libraries/data/rwJPEG.cpp	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/rwJPEG.cpp	2015-03-12 17:35:06.781738000 +0100
@@ -24,7 +24,7 @@
  ***************************************************************************/
 
 #include "xmipp_image_base.h"
-#include "../../external/jpeg-8c/jpeglib.h"
+#include "jpeglib.h"
 
 
 //#include <jpeglib.h>
diff -ru xmipp/libraries/data/xmipp_fftw.h xmipp_working/libraries/data/xmipp_fftw.h
--- xmipp/libraries/data/xmipp_fftw.h	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/xmipp_fftw.h	2015-03-12 17:32:35.751658248 +0100
@@ -28,7 +28,7 @@
 #define __XmippFFTW_H
 
 #include <complex>
-#include "../../external/fftw-3.3.3/api/fftw3.h"
+#include "fftw3.h"
 #include "multidim_array.h"
 #include "multidim_array_generic.h"
 #include "xmipp_fft.h"
diff -ru xmipp/libraries/data/xmipp_hdf5.h xmipp_working/libraries/data/xmipp_hdf5.h
--- xmipp/libraries/data/xmipp_hdf5.h	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/xmipp_hdf5.h	2015-03-12 17:32:50.922130000 +0100
@@ -28,8 +28,8 @@
 
 #include <iostream>
 #include<map>
-#include "../../external/hdf5-1.8.10/src/hdf5.h"
-#include "../../external/hdf5-1.8.10/c++/src/H5Cpp.h"
+#include "hdf5.h"
+#include "H5Cpp.h"
 #include "matrix1d.h"
 
 
diff -ru xmipp/libraries/data/xmipp_image_base.h xmipp_working/libraries/data/xmipp_image_base.h
--- xmipp/libraries/data/xmipp_image_base.h	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/xmipp_image_base.h	2015-03-12 17:34:39.112206978 +0100
@@ -33,8 +33,8 @@
 #include "xmipp_datatype.h"
 //
 //// Includes for rwTIFF which cannot be inside it
-#include "../../external/tiff-3.9.4/libtiff/tiffio.h"
-#include "../../external/hdf5-1.8.10/src/hdf5.h"
+#include "tiffio.h"
+#include "hdf5.h"
 
 
 /* Minimum size of a TIFF file to be mapped to a tempfile in case of mapping from
diff -ru xmipp/libraries/data/xmipp_program_sql.h xmipp_working/libraries/data/xmipp_program_sql.h
--- xmipp/libraries/data/xmipp_program_sql.h	2014-04-03 09:46:48.000000000 +0200
+++ xmipp_working/libraries/data/xmipp_program_sql.h	2015-03-12 17:32:25.300863000 +0100
@@ -28,7 +28,7 @@
 
 #include <map>
 #include "xmipp_program.h"
-#include "external/sqlite-3.6.23/sqlite3.h"
+#include "sqlite3.h"
 
 typedef std::map<const char*, String> DictDB;
 
