diff -Nru qcint-4.4.6.orig/CMakeLists.txt qcint-4.4.6/CMakeLists.txt
--- qcint-4.4.6.orig/CMakeLists.txt	2021-11-24 20:49:33.598089998 -0500
+++ qcint-4.4.6/CMakeLists.txt	2021-11-24 20:50:19.865367290 -0500
@@ -18,8 +18,8 @@
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=native -O2")
 else()
   # This appears to be the lowest version that compiles
-  message("Build with -march=sse3")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse3 -O2")
+  #message("Build with -march=sse3")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
 endif()
 if ("${CMAKE_C_COMPILER_ID}" STREQUAL "Intel")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -unroll-aggressive -ipo ")
