# S.D.Pinches (Jan 2024): Patch to allow compilation with GCC v13
diff -Nru mdbm-4.13.0-orig/src/lib/mdbm.c mdbm-4.13.0/src/lib/mdbm.c
--- mdbm-4.13.0-orig/src/lib/mdbm.c	2018-11-12 17:20:21.000000000 +0100
+++ mdbm-4.13.0/src/lib/mdbm.c	2024-02-13 10:22:55.937708854 +0100
@@ -7,6 +7,7 @@
 #include <sys/vfs.h>
 #endif
 
+#include <stdint.h>
 #include <assert.h>
 #include <ctype.h>
 #include <dlfcn.h>
