Disable building Cling demos.

Author: Jens Henrik Goebbert (j.goebbert@fz-juelich.de)

diff -Naur cling-1d93ee8d5855a82bbff2aab31515760630e8c1a4.orig/tools/CMakeLists.txt cling-1d93ee8d5855a82bbff2aab31515760630e8c1a4/tools/CMakeLists.txt
--- cling-1d93ee8d5855a82bbff2aab31515760630e8c1a4.orig/tools/CMakeLists.txt	2023-02-05 16:02:08.108301585 +0100
+++ cling-1d93ee8d5855a82bbff2aab31515760630e8c1a4/tools/CMakeLists.txt	2023-02-05 15:59:53.711904401 +0100
@@ -11,7 +11,6 @@
   add_subdirectory(driver)
   add_subdirectory(Jupyter)
   add_subdirectory(libcling)
-  add_subdirectory(demo)
 endif()
 
 add_subdirectory(plugins)

