don't use wheel so that install_pkgconfig runs
Author: Miguel Dias Costa (National University of Singapore)
--- setup.py.orig	2019-11-14 18:31:19.945243256 +0800
+++ setup.py	2019-11-14 18:31:49.910877467 +0800
@@ -306,7 +306,7 @@
         # prefix. pip uses wheels as cache and when installing with --user
         # and then to a virtualenv, the wheel gets reused containing the
         # wrong paths. So in case bdist_wheel is used, just skip this command.
-        cmd = self.distribution.get_command_obj("bdist_wheel", create=False)
+        cmd = None  # self.distribution.get_command_obj("bdist_wheel", create=False)
         if cmd is not None:
             log.info(
                 "Skipping install_pkgconfig, not supported with bdist_wheel")
@@ -397,7 +397,7 @@
         (out, _) = self.copy_file(source, lib_header_path)
         self.outfiles.append(out)
 
-        cmd = self.distribution.get_command_obj("bdist_wheel", create=False)
+        cmd = None  # self.distribution.get_command_obj("bdist_wheel", create=False)
         if cmd is not None:
             return
         cmd = self.distribution.get_command_obj("bdist_egg", create=False)
