fix compilation errors with Intel compilers: 'error: expected a ">"'
author: Kenneth Hoste (HPC-UGent)
--- pybedtools-0.7.10/src/bedFile.h.orig	2017-05-16 14:10:33.438960918 +0200
+++ pybedtools-0.7.10/src/bedFile.h	2017-05-16 14:13:07.112514598 +0200
@@ -455,7 +455,7 @@
                 //cerr << "Error: malformed BED entry at line " << _lineNum << ". Start was greater than end. Exiting." << endl; 
                 return BED_MALFORMED;
             }
-            else if ( (bed.start < 0) || (bed.end < 0) ) {
+            else if ( !(bed.start >= 0) || !(bed.end >= 0) ) {
                 //cerr << "Error: malformed BED entry at line " << _lineNum << ". Coordinate detected that is < 0. Exiting." << endl;
                 return BED_MALFORMED;
             }
@@ -502,7 +502,7 @@
                 //cerr << "Error: malformed VCF entry at line " << _lineNum << ". Start was greater than end. Exiting." << endl;
                 return BED_MALFORMED;
             }
-            else if ( (bed.start < 0) || (bed.end < 0) ) {
+            else if ( !(bed.start >= 0) || !(bed.end >= 0) ) {
                 //cerr << "Error: malformed VCF entry at line " << _lineNum << ". Coordinate detected that is < 0. Exiting." << endl;
                 return BED_MALFORMED;
             }
@@ -550,7 +550,7 @@
                 //cerr << "Error: malformed GFF entry at line " << _lineNum << ". Start was greater than end. Exiting." << endl;
                 return BED_MALFORMED;
             }
-            else if ( (bed.start < 0) || (bed.end < 0) ) {
+            else if ( !(bed.start >= 0) || !(bed.end >= 0) ) {
                 //cerr << "Error: malformed GFF entry at line " << _lineNum << ". Coordinate detected that is < 1. Exiting." << endl;
                 return BED_MALFORMED;
             }
