--- shapeit4-4.2.2/makefile	2021-09-07 13:08:23.000000000 +0100
+++ shapeit4-4.2.2.new/makefile	2021-10-06 17:34:49.006112748 +0100
@@ -1,14 +1,14 @@
 #COMPILER MODE C++11
-CXX=g++ -std=c++11
+#CXX=g++ -std=c++11
 
 #HTSLIB LIBRARY [SPECIFY YOUR OWN PATHS]
-HTSLIB_INC=$(HOME)/Tools/htslib-1.11
-HTSLIB_LIB=$(HOME)/Tools/htslib-1.11/libhts.a
+#HTSLIB_INC=$(HOME)/Tools/htslib-1.11
+#HTSLIB_LIB=$(HOME)/Tools/htslib-1.11/libhts.a
 
 #BOOST IOSTREAM & PROGRAM_OPTION LIBRARIES [SPECIFY YOUR OWN PATHS]
-BOOST_INC=/usr/include
-BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
+#BOOST_INC=/usr/include
+#BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
+#BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
 
 #HTSLIB LIBRARY [SPECIFY YOUR OWN PATHS]
 #HTSLIB_INC=/software/UHTS/Analysis/samtools/1.4/include
@@ -25,12 +25,12 @@
 #CXXFLAG=-O3 -march=native
 
 #Good performance and portable on most CPUs
-CXXFLAG=-O3 -mavx2 -mfma 
+#CXXFLAG=-O3 -mavx2 -mfma 
 
 #Portable version without avx2 (much slower)
 #CXXFLAG=-O3
 
-LDFLAG=-O3
+#LDFLAG=-O3
 
 
 #DYNAMIC LIBRARIES
