Disable hard-coded library and flag settings

Adam Huffman

--- shapeit4-4.2.0/makefile	2021-01-21 08:05:26.000000000 +0000
+++ shapeit4-4.2.0.new/makefile	2021-02-19 17:32:01.321128179 +0000
@@ -1,14 +1,14 @@
 #COMPILER MODE C++11
-CXX=g++ -std=c++11
+#CXX=g++ -std=c++11
 
 #HTSLIB LIBRARY [SPECIFY YOUR OWN PATHS]
-HTSLIB_INC=$(HOME)/Tools/htslib-1.9
-HTSLIB_LIB=$(HOME)/Tools/htslib-1.9/libhts.a
+#HTSLIB_INC=$(HOME)/Tools/htslib-1.9
+#HTSLIB_LIB=$(HOME)/Tools/htslib-1.9/libhts.a
 
 #BOOST IOSTREAM & PROGRAM_OPTION LIBRARIES [SPECIFY YOUR OWN PATHS]
-BOOST_INC=/usr/include
-BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
+#BOOST_INC=/usr/include
+#BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
+#BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
 
 #HTSLIB LIBRARY [SPECIFY YOUR OWN PATHS]
 #HTSLIB_INC=/software/UHTS/Analysis/samtools/1.4/include
@@ -25,12 +25,14 @@
 #CXXFLAG=-O3 -march=native
 
 #Good performance and portable on most CPUs
-CXXFLAG=-O3 -mavx2 -mfma 
+#CXXFLAG=-O3 -mavx2 -mfma 
 
 #Portable version without avx2 (much slower)
 #CXXFLAG=-O3
 
-LDFLAG=-O3
+#CXXFLAG=${CXXFLAGS}
+
+#LDFLAG=${OPTFLAGS}
 
 
 #DYNAMIC LIBRARIES
