--- makefile.orig	2021-03-11 08:25:12.000000000 +0000
+++ makefile	2021-07-13 15:38:30.973337609 +0000
@@ -1,14 +1,14 @@
 #COMPILER MODE C++11
-CXX=g++ -std=c++11
+#CXX=g++ -std=c++11
 
 #HTSLIB LIBRARY [SPECIFY YOUR OWN PATHS]
-HTSLIB_INC=$(HOME)/Tools/htslib-1.11
-HTSLIB_LIB=$(HOME)/Tools/htslib-1.11/libhts.a
+#HTSLIB_INC=$(HOME)/Tools/htslib-1.11
+#HTSLIB_LIB=$(HOME)/Tools/htslib-1.11/libhts.a
 
 #BOOST IOSTREAM & PROGRAM_OPTION LIBRARIES [SPECIFY YOUR OWN PATHS]
-BOOST_INC=/usr/include
-BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
+#BOOST_INC=/usr/include
+#BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
+#BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
 
 #HTSLIB LIBRARY [SPECIFY YOUR OWN PATHS]
 #HTSLIB_INC=/software/UHTS/Analysis/samtools/1.4/include
@@ -25,19 +25,21 @@
 #CXXFLAG=-O3 -march=native
 
 #Good performance and portable on most CPUs
-CXXFLAG=-O3 -mavx2 -mfma 
+#CXXFLAG=-O3 -mavx2 -mfma 
 
 #Portable version without avx2 (much slower)
 #CXXFLAG=-O3
 
-LDFLAG=-O3
+#CXXFLAG=${CXXFLAGS}
+
+#LDFLAG=${OPTFLAGS}
 
 
 #DYNAMIC LIBRARIES
 DYN_LIBS=-lz -lbz2 -lm -lpthread -llzma -lcurl -lssl -lcrypto
 
 #SHAPEIT SOURCES & BINARY
-BFILE=bin/shapeit4.2
+BFILE=bin/shapeit4
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
 OFILE=$(shell for file in `find src -name *.cpp`; do echo obj/$$(basename $$file .cpp).o; done)
