--- n2p2-2.2.0/src/pynnp/setup.py.orig	2024-09-10 17:12:18.680477012 +0000
+++ n2p2-2.2.0/src/pynnp/setup.py	2024-09-10 17:10:55.095988783 +0000
@@ -17,6 +17,8 @@
                        language="c++")]
 
 setup(
+    name="pynnp",
+    version="2.2.0",
     ext_modules=cythonize(extension,
                           compiler_directives={'language_level' : "3",
                                                'c_string_type' : 'unicode',
