diff -ruN TALON-5.0/requirements.txt TALON-5.0_edit/requirements.txt
--- TALON-5.0/requirements.txt	2020-03-20 05:13:53.000000000 +0100
+++ TALON-5.0_edit/requirements.txt	2021-05-26 14:36:01.257975762 +0200
@@ -1,4 +1,4 @@
 pandas
 pyfaidx
-pysam==0.15.4
+pysam>=0.15.4
 pybedtools
diff -ruN TALON-5.0/setup.py TALON-5.0_edit/setup.py
--- TALON-5.0/setup.py	2020-03-20 05:13:53.000000000 +0100
+++ TALON-5.0_edit/setup.py	2021-05-26 14:36:06.873983709 +0200
@@ -49,7 +49,7 @@
     python_requires=">=3.6",
     install_requires=[
         "pandas",
-        "pysam==0.15.4",
+        "pysam>=0.15.4",
         "pybedtools",
         "pyfaidx"
     ],
