--- perfexpert-4.1.1/tools/sniffer/sniffer.c.orig	2013-10-20 18:27:58.000000000 -0500
+++ perfexpert-4.1.1/tools/sniffer/sniffer.c	2014-04-24 12:10:10.000000000 -0500
@@ -620,13 +620,13 @@
                 "GFLOPS_(%%_max).scalar = (%s / PAPI_TOT_CYC) / 8",
                 event_list[FP_COMP_OPS_EXE_SSE_FP_SCALAR_SINGLE_SSE_SCALAR_DOUBLE].PAPI_event_name);
         } else {
-            USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_SCALAR);
+            USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_SCALAR_SINGLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_SCALAR_DOUBLE);
             USE_EVENT(SIMD_FP_256_PACKED_SINGLE);
             USE_EVENT(SIMD_FP_256_PACKED_DOUBLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_PACKED_DOUBLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_PACKED_SINGLE);
-            
+
             sprintf(overall,
                 "GFLOPS_(%%_max).overall = ((%s*8 + (%s + %s)*4 + %s*2 + %s + %s) / PAPI_TOT_CYC) / 8",
                 event_list[SIMD_FP_256_PACKED_SINGLE].PAPI_event_name,
@@ -701,13 +701,13 @@
             USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_SCALAR_SINGLE_SSE_SCALAR_DOUBLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_PACKED_DOUBLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_PACKED_SINGLE);
-            
+
             sprintf(temp_str, "%s + %s + %s",
                 event_list[FP_COMP_OPS_EXE_SSE_PACKED_SINGLE].PAPI_event_name,
                 event_list[FP_COMP_OPS_EXE_SSE_FP_PACKED_DOUBLE].PAPI_event_name,
                 event_list[FP_COMP_OPS_EXE_SSE_FP_SCALAR_SINGLE_SSE_SCALAR_DOUBLE].PAPI_event_name);
         } else {
-            USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_SCALAR);
+            USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_SCALAR_SINGLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_SCALAR_DOUBLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_FP_PACKED_DOUBLE);
             USE_EVENT(FP_COMP_OPS_EXE_SSE_PACKED_SINGLE);
