--- CMakeLists.txt.orig	2021-04-08 23:29:55.000000000 -0700
+++ CMakeLists.txt	2021-04-08 23:30:04.000000000 -0700
@@ -587,7 +587,7 @@
             if (CUDA_VERSION_MAJOR LESS 6)
                 set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_11 )
             else()
-                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_20 )
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_70 )
             endif()
         endif()
     endif()
