From 4b98faf69bb2a677491c7caac81c12b707cd9b53 Mon Sep 17 00:00:00 2001
From: Kasparas Karlauskas <kasparas.karlauskas@gmail.com>
Date: Fri, 22 Aug 2025 22:20:54 +0900
Subject: [PATCH] Fix LasHeader default format singleton bug

---
 laspy/header.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/laspy/header.py b/laspy/header.py
index 65a1f30..0da460f 100644
--- a/laspy/header.py
+++ b/laspy/header.py
@@ -195,7 +195,7 @@ def __init__(
 
         if version is None and point_format is None:
             version = LasHeader.DEFAULT_VERSION
-            point_format = LasHeader.DEFAULT_POINT_FORMAT
+            point_format = deepcopy(LasHeader.DEFAULT_POINT_FORMAT)
         elif version is not None and point_format is None:
             point_format = PointFormat(dims.min_point_format_for_version(str(version)))
         elif version is None and point_format is not None: