--- src/conv/vdicom/Makefile.orig       2021-08-04 10:05:44.441938000 -0000
+++ src/conv/vdicom/Makefile    2021-08-04 10:06:42.881974000 -0000
@@ -13,11 +13,9 @@
 INCLUDE_PATH += ${LIPSIA_INCLUDE}
 LIB_PATH += ${LIPSIA_LIB}

-DBG_OPT=-O2
-DBG_OPT=-g

 #CXXFLAGS += ${DBG_OPT} $(addprefix -I,${INCLUDE_PATH}) -std=c++11
-CXXFLAGS += ${DBG_OPT} $(addprefix -I,${INCLUDE_PATH}) -std=c++0x
+CXXFLAGS += $(addprefix -I,${INCLUDE_PATH}) -std=c++0x -fkeep-inline-functions
 LDFLAGS +=  $(addprefix -L,${LIB_PATH})

 CC=${CXX}
