diff -ru beagle-lib-20120124.orig/libhmsbeagle/CPU/Precision.h beagle-lib-20120124/libhmsbeagle/CPU/Precision.h
--- beagle-lib-20120124.orig/libhmsbeagle/CPU/Precision.h	2012-01-24 16:12:06.000000000 +0100
+++ beagle-lib-20120124/libhmsbeagle/CPU/Precision.h	2013-03-27 14:03:22.546952068 +0100
@@ -8,6 +8,8 @@
 #ifndef PRECISION_H_
 #define PRECISION_H_
 
+#include <cstring>
+
 #define DOUBLE_PRECISION (sizeof(REALTYPE) == 8)
 
 template<typename T, typename F> 
diff -ru beagle-lib-20120124.orig/libhmsbeagle/GPU/Precision.h beagle-lib-20120124/libhmsbeagle/GPU/Precision.h
--- beagle-lib-20120124.orig/libhmsbeagle/GPU/Precision.h	2012-01-24 16:12:07.000000000 +0100
+++ beagle-lib-20120124/libhmsbeagle/GPU/Precision.h	2013-03-27 14:03:33.617003846 +0100
@@ -8,6 +8,8 @@
 #ifndef GPU_PRECISION_H_
 #define GPU_PRECISION_H_
 
+#include <cstring>
+
 template<typename T, typename F> 
 inline void beagleMemCpy( T* to, F* from, unsigned int length )
 {
