diff -Nru beagle-lib-4.0.1.orig/CMakeLists.txt beagle-lib-4.0.1/CMakeLists.txt
--- beagle-lib-4.0.1.orig/CMakeLists.txt	2024-06-21 08:40:15.193530410 -0400
+++ beagle-lib-4.0.1/CMakeLists.txt	2024-06-21 08:47:15.278390785 -0400
@@ -14,7 +14,7 @@
 message(STATUS "BEAGLE_VERSION = ${BEAGLE_VERSION}")
 message(STATUS "BEAGLE_PLUGIN_VERSION = ${BEAGLE_PLUGIN_VERSION}")
 
-set(BEAGLE_OPTIMIZE_FOR_NATIVE_ARCH true)
+set(BEAGLE_OPTIMIZE_FOR_NATIVE_ARCH false)
 
 option(BUILD_OPENCL "Build beagle with OpenCL library" ON)
 option(BUILD_CUDA "Build beagle with CUDA library" ON)
