Make it possible to configure Ludwig via environment variables

diff -rupN ludwig-ludwig-0.23.1.orig/config/unix-mpicc-default.mk ludwig-ludwig-0.23.1/config/unix-mpicc-default.mk
--- ludwig-ludwig-0.23.1.orig/config/unix-mpicc-default.mk	2025-05-07 15:17:41.000000000 +0200
+++ ludwig-ludwig-0.23.1/config/unix-mpicc-default.mk	2025-07-08 10:29:18.767714702 +0200
@@ -7,11 +7,11 @@
 #
 ##############################################################################
 
-BUILD   = parallel
-MODEL   = -D_D3Q19_
-TARGET  =
+BUILD   ?= parallel
+MODEL   ?= -D_D3Q19_
+TARGET  ?=
 
-CC      = mpicc -fopenmp
-CFLAGS  = -O2 -g -Wall
+CC      ?= mpicc -fopenmp
+CFLAGS  ?= -O2 -g -Wall
 
-LAUNCH_MPIRUN_CMD = mpirun -np 1
+LAUNCH_MPIRUN_CMD ?= mpirun -np 1
