diff -ru primme-3.2-old/Python/primme.c primme-3.2/Python/primme.c
--- primme-3.2-old/Python/primme.c	2023-11-02 20:02:49.995355406 +0000
+++ primme-3.2/Python/primme.c	2023-11-27 15:27:23.188608546 +0000
@@ -647,7 +647,7 @@
 
     /* NumPy API declarations from "numpy/__init__.pxd" */
     
-#include "../include/primme.h"
+#include "primme.h"
 #include "pythread.h"
 #include <stdlib.h>
 #include "pystate.h"
@@ -3079,7 +3079,7 @@
 static const char __pyx_k_Error_from_PRIMME_first_stage[] = "Error from PRIMME first stage: ";
 static const char __pyx_k_Invalid_option_s_with_value_s[] = "Invalid option '%s' with value '%s'";
 static const char __pyx_k_locking_and_initSize_numEvals[] = "'locking' and 'initSize' > 'numEvals'";
-static const char __pyx_k_scipy_sparse_linalg_interface[] = "scipy.sparse.linalg.interface";
+static const char __pyx_k_scipy_sparse_linalg_interface[] = "scipy.sparse.linalg";
 static const char __pyx_k_unable_to_allocate_array_data[] = "unable to allocate array data.";
 static const char __pyx_k_Error_from_PRIMME_second_stage[] = "Error from PRIMME second stage: ";
 static const char __pyx_k_locking_0_and_target_is_primme[] = "'locking' == 0 and 'target' is 'primme_closest_leq' or 'primme_closet_geq'";
@@ -51840,7 +51840,7 @@
  * 
  * import numpy as np             # <<<<<<<<<<<<<<
  * cimport numpy as np
- * from scipy.sparse.linalg.interface import aslinearoperator
+ * from scipy.sparse.linalg import aslinearoperator
  */
   __pyx_t_1 = __Pyx_Import(__pyx_n_s_numpy, 0, -1); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 3, __pyx_L1_error)
   __Pyx_GOTREF(__pyx_t_1);
@@ -51850,7 +51850,7 @@
   /* "primme.pyx":5
  * import numpy as np
  * cimport numpy as np
- * from scipy.sparse.linalg.interface import aslinearoperator             # <<<<<<<<<<<<<<
+ * from scipy.sparse.linalg import aslinearoperator             # <<<<<<<<<<<<<<
  * cimport cython
  * from cython cimport view
  */
diff -ru primme-3.2-old/Python/primme.pyx primme-3.2/Python/primme.pyx
--- primme-3.2-old/Python/primme.pyx	2023-11-02 20:02:49.997355424 +0000
+++ primme-3.2/Python/primme.pyx	2023-11-27 15:28:40.415282930 +0000
@@ -2,7 +2,7 @@
 
 import numpy as np
 cimport numpy as np
-from scipy.sparse.linalg.interface import aslinearoperator
+from scipy.sparse.linalg import aslinearoperator
 cimport cython
 from cython cimport view
 try:
@@ -52,7 +52,7 @@
         return np.dtype(np.float64)
 
 
-cdef extern from "../include/primme.h":
+cdef extern from "primme.h":
     struct primme_params:
         pass
     ctypedef int primme_preset_method
@@ -697,7 +697,7 @@
 
 
 
-cdef extern from "../include/primme.h":
+cdef extern from "primme.h":
     struct primme_svds_params:
         pass
     ctypedef int primme_svds_preset_method
diff -ru primme-3.2-old/Python/setup.py primme-3.2/Python/setup.py
--- primme-3.2-old/Python/setup.py	2023-11-02 20:02:49.998355433 +0000
+++ primme-3.2/Python/setup.py	2023-11-27 15:27:23.199608642 +0000
@@ -43,7 +43,7 @@
        blaslapack_libraries = ['lapack', 'blas']
 
    r = dict(
-                   include_dirs = [numpy_include, "primme/include", "primme/src/include"],
+                   include_dirs = [numpy_include, "../include", "primme/src/include"],
                    library_dirs = blaslapack_library_dirs,
                    libraries = blaslapack_libraries,
                    extra_link_args = blaslapack_extra_link_args
