diff --git a/chunk/makefile b/chunk/makefile
index 5517484..94601de 100644
--- a/chunk/makefile
+++ b/chunk/makefile
@@ -11,7 +11,7 @@ CXXFLAG=-O3
 LDFLAG=-O3
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -23,12 +23,12 @@ BFILE=bin/GLIMPSE_$(NAME)
 EXEFILE=bin/GLIMPSE_$(NAME)_static
 
 #CONDITIONAL PATH DEFINITON
-desktop: HTSSRC=../..
-desktop: HTSLIB_INC=$(HTSSRC)/htslib
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib/libhts.a
-desktop: BOOST_INC=/usr/include
-desktop: BOOST_LIB_IO=/usr/local/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/usr/local/lib/libboost_program_options.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/concordance/makefile b/concordance/makefile
index 5517484..94601de 100644
--- a/concordance/makefile
+++ b/concordance/makefile
@@ -11,7 +11,7 @@ CXXFLAG=-O3
 LDFLAG=-O3
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -23,12 +23,12 @@ BFILE=bin/GLIMPSE_$(NAME)
 EXEFILE=bin/GLIMPSE_$(NAME)_static
 
 #CONDITIONAL PATH DEFINITON
-desktop: HTSSRC=../..
-desktop: HTSLIB_INC=$(HTSSRC)/htslib
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib/libhts.a
-desktop: BOOST_INC=/usr/include
-desktop: BOOST_LIB_IO=/usr/local/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/usr/local/lib/libboost_program_options.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/ligate/makefile b/ligate/makefile
index 5517484..94601de 100644
--- a/ligate/makefile
+++ b/ligate/makefile
@@ -11,7 +11,7 @@ CXXFLAG=-O3
 LDFLAG=-O3
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -23,12 +23,12 @@ BFILE=bin/GLIMPSE_$(NAME)
 EXEFILE=bin/GLIMPSE_$(NAME)_static
 
 #CONDITIONAL PATH DEFINITON
-desktop: HTSSRC=../..
-desktop: HTSLIB_INC=$(HTSSRC)/htslib
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib/libhts.a
-desktop: BOOST_INC=/usr/include
-desktop: BOOST_LIB_IO=/usr/local/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/usr/local/lib/libboost_program_options.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/makefile b/makefile
index 34b21ed..ca1de54 100644
--- a/makefile
+++ b/makefile
@@ -1,4 +1,4 @@
-projects = chunk concordance ligate phase sample snparray
+projects = chunk concordance ligate phase sample snparray stats
 
 .PHONY: all $(projects)
 
diff --git a/phase/makefile b/phase/makefile
index 5517484..94601de 100644
--- a/phase/makefile
+++ b/phase/makefile
@@ -11,7 +11,7 @@ CXXFLAG=-O3
 LDFLAG=-O3
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -23,12 +23,12 @@ BFILE=bin/GLIMPSE_$(NAME)
 EXEFILE=bin/GLIMPSE_$(NAME)_static
 
 #CONDITIONAL PATH DEFINITON
-desktop: HTSSRC=../..
-desktop: HTSLIB_INC=$(HTSSRC)/htslib
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib/libhts.a
-desktop: BOOST_INC=/usr/include
-desktop: BOOST_LIB_IO=/usr/local/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/usr/local/lib/libboost_program_options.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/sample/makefile b/sample/makefile
index 5517484..94601de 100644
--- a/sample/makefile
+++ b/sample/makefile
@@ -11,7 +11,7 @@ CXXFLAG=-O3
 LDFLAG=-O3
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -23,12 +23,12 @@ BFILE=bin/GLIMPSE_$(NAME)
 EXEFILE=bin/GLIMPSE_$(NAME)_static
 
 #CONDITIONAL PATH DEFINITON
-desktop: HTSSRC=../..
-desktop: HTSLIB_INC=$(HTSSRC)/htslib
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib/libhts.a
-desktop: BOOST_INC=/usr/include
-desktop: BOOST_LIB_IO=/usr/local/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/usr/local/lib/libboost_program_options.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/snparray/makefile b/snparray/makefile
index 5517484..94601de 100644
--- a/snparray/makefile
+++ b/snparray/makefile
@@ -11,7 +11,7 @@ CXXFLAG=-O3
 LDFLAG=-O3
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -23,12 +23,12 @@ BFILE=bin/GLIMPSE_$(NAME)
 EXEFILE=bin/GLIMPSE_$(NAME)_static
 
 #CONDITIONAL PATH DEFINITON
-desktop: HTSSRC=../..
-desktop: HTSLIB_INC=$(HTSSRC)/htslib
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib/libhts.a
-desktop: BOOST_INC=/usr/include
-desktop: BOOST_LIB_IO=/usr/local/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/usr/local/lib/libboost_program_options.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/stats/makefile b/stats/makefile
index 5517484..94601de 100644
--- a/stats/makefile
+++ b/stats/makefile
@@ -11,7 +11,7 @@ CXXFLAG=-O3
 LDFLAG=-O3
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -23,12 +23,12 @@ BFILE=bin/GLIMPSE_$(NAME)
 EXEFILE=bin/GLIMPSE_$(NAME)_static
 
 #CONDITIONAL PATH DEFINITON
-desktop: HTSSRC=../..
-desktop: HTSLIB_INC=$(HTSSRC)/htslib
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib/libhts.a
-desktop: BOOST_INC=/usr/include
-desktop: BOOST_LIB_IO=/usr/local/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/usr/local/lib/libboost_program_options.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
