diff --git a/chunk/makefile b/chunk/makefile
index 0fdda08..b637e70 100644
--- a/chunk/makefile
+++ b/chunk/makefile
@@ -14,7 +14,7 @@ LDFLAG=-O3
 #LDFLAG=-O0
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -60,23 +60,22 @@ endif
 
 #CONDITIONAL PATH DEFINITON
 
-system: DYN_LIBS=-lz -lpthread -lbz2 -llzma
-system: HTSSRC=/usr/local
-system: HTSLIB_INC=$(HTSSRC)/include/htslib
+system: HTSSRC=$(EBROOTHTSLIB)
+system: HTSLIB_INC=$(HTSSRC)/include
 system: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
-system: BOOST_INC=/usr/include
-system: BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-system: BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
-system: BOOST_LIB_SE=/usr/lib/x86_64-linux-gnu/libboost_serialization.a
+system: BOOST_INC=$(BOOST_ROOT)
+system: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+system: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+system: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 system: $(BFILE)
 
-desktop: HTSSRC=/home/srubinac/git
-desktop: HTSLIB_INC=$(HTSSRC)/htslib_minimal
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib_minimal/libhts.a
-desktop: BOOST_INC=/home/srubinac/software/boost_v1.74.0/include
-desktop: BOOST_LIB_IO=/home/srubinac/software/boost_v1.74.0/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/home/srubinac/software/boost_v1.74.0/lib/libboost_program_options.a
-desktop: BOOST_LIB_SE=/home/srubinac/software/boost_v1.74.0/lib/libboost_serialization.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+desktop: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/concordance/makefile b/concordance/makefile
index 0fdda08..b637e70 100644
--- a/concordance/makefile
+++ b/concordance/makefile
@@ -14,7 +14,7 @@ LDFLAG=-O3
 #LDFLAG=-O0
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -60,23 +60,22 @@ endif
 
 #CONDITIONAL PATH DEFINITON
 
-system: DYN_LIBS=-lz -lpthread -lbz2 -llzma
-system: HTSSRC=/usr/local
-system: HTSLIB_INC=$(HTSSRC)/include/htslib
+system: HTSSRC=$(EBROOTHTSLIB)
+system: HTSLIB_INC=$(HTSSRC)/include
 system: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
-system: BOOST_INC=/usr/include
-system: BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-system: BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
-system: BOOST_LIB_SE=/usr/lib/x86_64-linux-gnu/libboost_serialization.a
+system: BOOST_INC=$(BOOST_ROOT)
+system: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+system: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+system: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 system: $(BFILE)
 
-desktop: HTSSRC=/home/srubinac/git
-desktop: HTSLIB_INC=$(HTSSRC)/htslib_minimal
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib_minimal/libhts.a
-desktop: BOOST_INC=/home/srubinac/software/boost_v1.74.0/include
-desktop: BOOST_LIB_IO=/home/srubinac/software/boost_v1.74.0/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/home/srubinac/software/boost_v1.74.0/lib/libboost_program_options.a
-desktop: BOOST_LIB_SE=/home/srubinac/software/boost_v1.74.0/lib/libboost_serialization.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+desktop: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/ligate/makefile b/ligate/makefile
index 0fdda08..b637e70 100644
--- a/ligate/makefile
+++ b/ligate/makefile
@@ -14,7 +14,7 @@ LDFLAG=-O3
 #LDFLAG=-O0
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -60,23 +60,22 @@ endif
 
 #CONDITIONAL PATH DEFINITON
 
-system: DYN_LIBS=-lz -lpthread -lbz2 -llzma
-system: HTSSRC=/usr/local
-system: HTSLIB_INC=$(HTSSRC)/include/htslib
+system: HTSSRC=$(EBROOTHTSLIB)
+system: HTSLIB_INC=$(HTSSRC)/include
 system: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
-system: BOOST_INC=/usr/include
-system: BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-system: BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
-system: BOOST_LIB_SE=/usr/lib/x86_64-linux-gnu/libboost_serialization.a
+system: BOOST_INC=$(BOOST_ROOT)
+system: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+system: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+system: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 system: $(BFILE)
 
-desktop: HTSSRC=/home/srubinac/git
-desktop: HTSLIB_INC=$(HTSSRC)/htslib_minimal
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib_minimal/libhts.a
-desktop: BOOST_INC=/home/srubinac/software/boost_v1.74.0/include
-desktop: BOOST_LIB_IO=/home/srubinac/software/boost_v1.74.0/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/home/srubinac/software/boost_v1.74.0/lib/libboost_program_options.a
-desktop: BOOST_LIB_SE=/home/srubinac/software/boost_v1.74.0/lib/libboost_serialization.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+desktop: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/phase/makefile b/phase/makefile
index 273c932..f8827d8 100644
--- a/phase/makefile
+++ b/phase/makefile
@@ -14,7 +14,7 @@ LDFLAG=-O3
 #LDFLAG=-O0
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -59,23 +59,22 @@ endif
 
 #CONDITIONAL PATH DEFINITON
 
-system: DYN_LIBS=-lz -lpthread -lbz2 -llzma
-system: HTSSRC=/usr/local
-system: HTSLIB_INC=$(HTSSRC)/include/htslib
+system: HTSSRC=$(EBROOTHTSLIB)
+system: HTSLIB_INC=$(HTSSRC)/include
 system: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
-system: BOOST_INC=/usr/include
-system: BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-system: BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
-system: BOOST_LIB_SE=/usr/lib/x86_64-linux-gnu/libboost_serialization.a
+system: BOOST_INC=$(BOOST_ROOT)
+system: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+system: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+system: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 system: $(BFILE)
 
-desktop: HTSSRC=/home/srubinac/git
-desktop: HTSLIB_INC=$(HTSSRC)/htslib_minimal
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib_minimal/libhts.a
-desktop: BOOST_INC=/home/srubinac/software/boost_v1.74.0/include
-desktop: BOOST_LIB_IO=/home/srubinac/software/boost_v1.74.0/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/home/srubinac/software/boost_v1.74.0/lib/libboost_program_options.a
-desktop: BOOST_LIB_SE=/home/srubinac/software/boost_v1.74.0/lib/libboost_serialization.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+desktop: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
diff --git a/split_reference/makefile b/split_reference/makefile
index 0fdda08..b637e70 100644
--- a/split_reference/makefile
+++ b/split_reference/makefile
@@ -14,7 +14,7 @@ LDFLAG=-O3
 #LDFLAG=-O0
 
 #DYNAMIC LIBRARIES
-DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto
+DYN_LIBS=-lz -lpthread -lbz2 -llzma -lcurl -lcrypto -ldeflate
 
 HFILE=$(shell find src -name *.h)
 CFILE=$(shell find src -name *.cpp)
@@ -60,23 +60,22 @@ endif
 
 #CONDITIONAL PATH DEFINITON
 
-system: DYN_LIBS=-lz -lpthread -lbz2 -llzma
-system: HTSSRC=/usr/local
-system: HTSLIB_INC=$(HTSSRC)/include/htslib
+system: HTSSRC=$(EBROOTHTSLIB)
+system: HTSLIB_INC=$(HTSSRC)/include
 system: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
-system: BOOST_INC=/usr/include
-system: BOOST_LIB_IO=/usr/lib/x86_64-linux-gnu/libboost_iostreams.a
-system: BOOST_LIB_PO=/usr/lib/x86_64-linux-gnu/libboost_program_options.a
-system: BOOST_LIB_SE=/usr/lib/x86_64-linux-gnu/libboost_serialization.a
+system: BOOST_INC=$(BOOST_ROOT)
+system: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+system: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+system: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 system: $(BFILE)
 
-desktop: HTSSRC=/home/srubinac/git
-desktop: HTSLIB_INC=$(HTSSRC)/htslib_minimal
-desktop: HTSLIB_LIB=$(HTSSRC)/htslib_minimal/libhts.a
-desktop: BOOST_INC=/home/srubinac/software/boost_v1.74.0/include
-desktop: BOOST_LIB_IO=/home/srubinac/software/boost_v1.74.0/lib/libboost_iostreams.a
-desktop: BOOST_LIB_PO=/home/srubinac/software/boost_v1.74.0/lib/libboost_program_options.a
-desktop: BOOST_LIB_SE=/home/srubinac/software/boost_v1.74.0/lib/libboost_serialization.a
+desktop: HTSSRC=$(EBROOTHTSLIB)
+desktop: HTSLIB_INC=$(HTSSRC)/include
+desktop: HTSLIB_LIB=$(HTSSRC)/lib/libhts.a
+desktop: BOOST_INC=$(BOOST_ROOT)
+desktop: BOOST_LIB_IO=$(BOOST_ROOT)/lib/libboost_iostreams.a
+desktop: BOOST_LIB_PO=$(BOOST_ROOT)/lib/libboost_program_options.a
+desktop: BOOST_LIB_SE=$(BOOST_ROOT)/lib/libboost_serialization.a
 desktop: $(BFILE)
 
 olivier: HTSSRC=$(HOME)/Tools
