--- qt-everywhere-opensource-src-4.8.4/mkspecs/linux-icc/qmake.conf.orig	2013-12-23 14:54:00.272609496 +0100
+++ qt-everywhere-opensource-src-4.8.4/mkspecs/linux-icc/qmake.conf	2013-12-23 14:54:09.392773091 +0100
@@ -25,7 +25,7 @@
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -w1 -Wcheck -wd654,1572,411,873,1125,2259
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -O2 -falign-functions=16 -ansi-alias -fstrict-aliasing
+QMAKE_CFLAGS_RELEASE	= -O1 -falign-functions=16 -ansi-alias -fstrict-aliasing
 QMAKE_CFLAGS_DEBUG	= -O0 -g
 QMAKE_CFLAGS_SHLIB	= -fPIC -fno-jump-tables
 QMAKE_CFLAGS_STATIC_LIB	= $$QMAKE_CFLAGS_SHLIB
