--- src/pmemd/src/cuda/gpu.cpp.orig	2023-03-24 19:28:18.598964693 +1100
+++ src/pmemd/src/cuda/gpu.cpp	2023-03-24 19:28:45.644046190 +1100
@@ -2849,7 +2849,7 @@
     }
     // torsions, resttype = 3
     else if (resttype[i] == 3) {
-      if (nmrat[i][0] >= 0 && nmrat[i][1] >= 0 && nmrat[i][2] >= 0 && nmrat[3] >= 0) {
+      if (nmrat[i][0] >= 0 && nmrat[i][1] >= 0 && nmrat[i][2] >= 0 && nmrat[i][3] >= 0) {
         torsions++;
       }
       else {
