--- libharu-RELEASE_2_2_0/demo/CMakeLists.txt.orig	2013-05-28 20:49:33.000000000 +0200
+++ libharu-RELEASE_2_2_0/demo/CMakeLists.txt	2013-05-28 20:49:52.000000000 +0200
@@ -40,7 +40,7 @@
 # create demos
 FOREACH( demo ${demos_NAMES} )
   ADD_EXECUTABLE( ${demo} ${demo}.c )
-  TARGET_LINK_LIBRARIES( ${demo} libharu )
+  TARGET_LINK_LIBRARIES( ${demo} haru )
   IF( DEMO_C_FLAGS )
     SET_TARGET_PROPERTIES( ${demo} PROPERTIES COMPILE_FLAGS ${DEMO_C_FLAGS} )
     MESSAGE( "DEMO_C_FLAGS=${DEMO_C_FLAGS}" )
@@ -50,7 +50,7 @@
 # some demos need grid_sheet.c compiled in
 FOREACH( demo ${demos_with_grid_NAMES} )
   ADD_EXECUTABLE( ${demo} ${demo}.c grid_sheet.c )
-  TARGET_LINK_LIBRARIES( ${demo} libharu )
+  TARGET_LINK_LIBRARIES( ${demo} haru )
   IF( DEMO_C_FLAGS )
     SET_TARGET_PROPERTIES( ${demo} PROPERTIES COMPILE_FLAGS ${DEMO_C_FLAGS} )
   ENDIF( DEMO_C_FLAGS )
@@ -58,7 +58,7 @@
   
 # the grid_sheet demo needs extra defines
 ADD_EXECUTABLE( grid_sheet grid_sheet.c )
-TARGET_LINK_LIBRARIES( grid_sheet libharu )
+TARGET_LINK_LIBRARIES( grid_sheet haru )
 SET_TARGET_PROPERTIES( grid_sheet PROPERTIES COMPILE_FLAGS "${DEMO_C_FLAGS} -DSTAND_ALONE" )
 
 # install libary and demos
