Use OSMesa and MesaGL etc from EasyBuild.

Åke Sandgren, 2022-03-08
diff -ru visit3.2.2.orig/src/CMake/FindMesaGL.cmake visit3.2.2/src/CMake/FindMesaGL.cmake
--- visit3.2.2.orig/src/CMake/FindMesaGL.cmake	2022-01-13 18:39:58.000000000 +0100
+++ visit3.2.2/src/CMake/FindMesaGL.cmake	2022-03-03 16:52:52.849032348 +0100
@@ -49,7 +49,7 @@
 
 if (VISIT_MESAGL_DIR)
 
-  find_library(MESAGL_LIBRARY GL  PATH ${VISIT_MESAGL_DIR}/lib NO_DEFAULT_PATH)
+  find_library(MESAGL_LIBRARY GL  PATH ${VISIT_MESAGL_DIR}/lib)
   if (MESAGL_LIBRARY)
       execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory
                       ${VISIT_BINARY_DIR}/lib/mesagl
@@ -65,11 +65,11 @@
                       RESULT_VARIABLE MESAGL_SONAME_RESULT
                       OUTPUT_VARIABLE MESAGL_SONAME
                       ERROR_VARIABLE MESAGL_SONAME_ERROR)
-      if(MESAGL_SONAME)
-          string(REPLACE "SONAME" "" MESAGL_SONAME ${MESAGL_SONAME})
-          string(STRIP ${MESAGL_SONAME} MESAGL_SONAME)
-          set(MESAGL_LIBRARY ${VISIT_MESAGL_DIR}/lib/${MESAGL_SONAME})
-      endif()
+      #if(MESAGL_SONAME)
+      #    string(REPLACE "SONAME" "" MESAGL_SONAME ${MESAGL_SONAME})
+      #    string(STRIP ${MESAGL_SONAME} MESAGL_SONAME)
+      #    set(MESAGL_LIBRARY ${VISIT_MESAGL_DIR}/lib/${MESAGL_SONAME})
+      #endif()
 
       execute_process(COMMAND ${CMAKE_COMMAND} -E copy
                               ${MESAGL_LIBRARY}
@@ -83,7 +83,7 @@
       message(FATAL_ERROR "VISIT_MESAGL_DIR provided, but it doesn't contain GL library")
   endif()
 
-  find_library(MESAGL_API_LIBRARY glapi  PATH ${VISIT_MESAGL_DIR}/lib NO_DEFAULT_PATH)
+  find_library(MESAGL_API_LIBRARY glapi  PATH ${VISIT_MESAGL_DIR}/lib)
   if (MESAGL_API_LIBRARY)
       get_filename_component(MESAGL_API_LIB ${MESAGL_API_LIBRARY} NAME)
       execute_process(COMMAND objdump -p ${MESAGL_API_LIBRARY}
@@ -91,11 +91,11 @@
                       RESULT_VARIABLE MESAGL_API_SONAME_RESULT
                       OUTPUT_VARIABLE MESAGL_API_SONAME
                       ERROR_VARIABLE MESAGL_API_SONAME_ERROR)
-      if(MESAGL_API_SONAME)
-          string(REPLACE "SONAME" "" MESAGL_API_SONAME ${MESAGL_API_SONAME})
-          string(STRIP ${MESAGL_API_SONAME} MESAGL_API_SONAME)
-          set(MESAGL_API_LIBRARY ${VISIT_MESAGL_DIR}/lib/${MESAGL_API_SONAME})
-      endif()
+      #if(MESAGL_API_SONAME)
+      #    string(REPLACE "SONAME" "" MESAGL_API_SONAME ${MESAGL_API_SONAME})
+      #    string(STRIP ${MESAGL_API_SONAME} MESAGL_API_SONAME)
+      #    set(MESAGL_API_LIBRARY ${VISIT_MESAGL_DIR}/lib/${MESAGL_API_SONAME})
+      #endif()
 
       execute_process(COMMAND ${CMAKE_COMMAND} -E copy
                               ${MESAGL_API_LIBRARY}
@@ -107,7 +107,7 @@
   endif()
 
 
-  find_library(MESAGLU_LIBRARY GLU  PATH ${VISIT_MESAGL_DIR}/lib NO_DEFAULT_PATH)
+  find_library(MESAGLU_LIBRARY GLU  PATH ${VISIT_MESAGL_DIR}/lib)
   if (MESAGLU_LIBRARY)
       get_filename_component(MESAGLU_LIB ${MESAGLU_LIBRARY} NAME)
       execute_process(COMMAND objdump -p ${MESAGLU_LIBRARY}
@@ -115,11 +115,11 @@
                       RESULT_VARIABLE MESAGLU_SONAME_RESULT
                       OUTPUT_VARIABLE MESAGLU_SONAME
                       ERROR_VARIABLE MESAGLU_SONAME_ERROR)
-      if(MESAGLU_SONAME)
-          string(REPLACE "SONAME" "" MESAGLU_SONAME ${MESAGLU_SONAME})
-          string(STRIP ${MESAGLU_SONAME} MESAGLU_SONAME)
-          set(MESAGLU_LIBRARY ${VISIT_MESAGL_DIR}/lib/${MESAGLU_SONAME})
-      endif()
+      #if(MESAGLU_SONAME)
+      #    string(REPLACE "SONAME" "" MESAGLU_SONAME ${MESAGLU_SONAME})
+      #    string(STRIP ${MESAGLU_SONAME} MESAGLU_SONAME)
+      #    set(MESAGLU_LIBRARY ${VISIT_MESAGL_DIR}/lib/${MESAGLU_SONAME})
+      #endif()
 
       execute_process(COMMAND ${CMAKE_COMMAND} -E copy
                               ${MESAGLU_LIBRARY}
@@ -130,7 +130,7 @@
   endif()
 
   if (VISIT_LLVM_DIR)
-    find_library(MESAGL_LLVM_LIBRARY LLVM  PATH ${VISIT_LLVM_DIR}/lib NO_DEFAULT_PATH)
+    find_library(MESAGL_LLVM_LIBRARY LLVM  PATH ${VISIT_LLVM_DIR}/lib)
     if (MESAGL_LLVM_LIBRARY)
         get_filename_component(MESAGL_LLVM_LIB ${MESAGL_LLVM_LIBRARY} NAME)
 
@@ -140,11 +140,11 @@
                         OUTPUT_VARIABLE MESAGL_LLVM_SONAME
                         ERROR_VARIABLE MESAGL_LLVM_SONAME_ERROR)
 
-        if(MESAGL_LLVM_SONAME)
-            string(REPLACE "SONAME" "" MESAGL_LLVM_SONAME ${MESAGL_LLVM_SONAME})
-            string(STRIP ${MESAGL_LLVM_SONAME} MESAGL_LLVM_SONAME)
-            set(MESAGL_LLVM_LIBRARY ${VISIT_LLVM_DIR}/lib/${MESAGL_LLVM_SONAME})
-        endif()
+        #if(MESAGL_LLVM_SONAME)
+        #    string(REPLACE "SONAME" "" MESAGL_LLVM_SONAME ${MESAGL_LLVM_SONAME})
+        #    string(STRIP ${MESAGL_LLVM_SONAME} MESAGL_LLVM_SONAME)
+        #    set(MESAGL_LLVM_LIBRARY ${VISIT_LLVM_DIR}/lib/${MESAGL_LLVM_SONAME})
+        #endif()
 
         execute_process(COMMAND ${CMAKE_COMMAND} -E copy
                               ${MESAGL_LLVM_LIBRARY}
diff -ru visit3.2.2.orig/src/CMake/FindOSMesa.cmake visit3.2.2/src/CMake/FindOSMesa.cmake
--- visit3.2.2.orig/src/CMake/FindOSMesa.cmake	2022-01-13 18:39:58.000000000 +0100
+++ visit3.2.2/src/CMake/FindOSMesa.cmake	2022-03-03 16:51:26.674279962 +0100
@@ -43,8 +43,7 @@
 
 if (VISIT_OSMESA_DIR)
     find_library(OSMESA_LIBRARY OSMesa
-                 PATH ${VISIT_OSMESA_DIR}/lib
-                 NO_DEFAULT_PATH)
+                 PATH ${VISIT_OSMESA_DIR}/lib)
     if (OSMESA_LIBRARY)
         set(OSMESA_FOUND true)
         if (MESAGL_LIBRARY)
@@ -67,11 +66,11 @@
                         OUTPUT_VARIABLE OSMESA_SONAME
                         ERROR_VARIABLE  OSMESA_SONAME_ERROR)
 
-        if(OSMESA_SONAME)
-                string(REPLACE "SONAME" "" OSMESA_SONAME ${OSMESA_SONAME})
-                string(STRIP ${OSMESA_SONAME} OSMESA_SONAME)
-                set(OSMESA_LIBRARY ${VISIT_OSMESA_DIR}/lib/${OSMESA_SONAME})
-        endif()
+        #if(OSMESA_SONAME)
+        #        string(REPLACE "SONAME" "" OSMESA_SONAME ${OSMESA_SONAME})
+        #        string(STRIP ${OSMESA_SONAME} OSMESA_SONAME)
+        #        set(OSMESA_LIBRARY ${VISIT_OSMESA_DIR}/lib/${OSMESA_SONAME})
+        #endif()
         set(OSMESA_LIBRARIES ${OSMESA_LIBRARY} CACHE STRING "OSMesa libraries")
         set(OSMESA_INCLUDE_DIR ${VISIT_OSMESA_DIR}/include)
 
@@ -90,8 +89,7 @@
 
     set(OSMESA_INCLUDE_DIR ${VISIT_OSMESA_DIR}/include CACHE PATH "OSMesa include path")
 
-    find_library(GLAPI_LIBRARY glapi PATH ${VISIT_OSMESA_DIR}/lib
-                 NO_DEFAULT_PATH)
+    find_library(GLAPI_LIBRARY glapi PATH ${VISIT_OSMESA_DIR}/lib)
     if (GLAPI_LIBRARY)
         get_filename_component(GLAPI_LIB ${GLAPI_LIBRARY} NAME)
         execute_process(COMMAND objdump -p ${GLAPI_LIBRARY}
@@ -100,11 +98,11 @@
                         OUTPUT_VARIABLE GLAPI_SONAME
                         ERROR_VARIABLE  GLAPI_SONAME_ERROR)
 
-        if(GLAPI_SONAME)
-            string(REPLACE "SONAME" "" GLAPI_SONAME ${GLAPI_SONAME})
-            string(STRIP ${GLAPI_SONAME} GLAPI_SONAME)
-            set(GLAPI_LIBRARY ${VISIT_OSMESA_DIR}/lib/${GLAPI_SONAME})
-        endif()
+        #if(GLAPI_SONAME)
+        #    string(REPLACE "SONAME" "" GLAPI_SONAME ${GLAPI_SONAME})
+        #    string(STRIP ${GLAPI_SONAME} GLAPI_SONAME)
+        #    set(GLAPI_LIBRARY ${VISIT_OSMESA_DIR}/lib/${GLAPI_SONAME})
+        #endif()
 
         execute_process(COMMAND ${CMAKE_COMMAND} -E copy
                                 ${GLAPI_LIBRARY}
@@ -113,8 +111,7 @@
         list(APPEND OSMESA_LIBRARIES ${GLAPI_LIBRARY})
     endif()
 
-    find_library(GLU_LIBRARY GLU PATH ${VISIT_OSMESA_DIR}/lib
-                 NO_DEFAULT_PATH)
+    find_library(GLU_LIBRARY GLU PATH ${VISIT_OSMESA_DIR}/lib)
     if (GLU_LIBRARY)
         get_filename_component(GLU_LIB ${GLU_LIBRARY} NAME)
         execute_process(COMMAND objdump -p ${GLU_LIBRARY}
@@ -123,11 +120,11 @@
                         OUTPUT_VARIABLE GLU_SONAME
                         ERROR_VARIABLE  GLU_SONAME_ERROR)
 
-        if(GLU_SONAME)
-            string(REPLACE "SONAME" "" GLU_SONAME ${GLU_SONAME})
-            string(STRIP ${GLU_SONAME} GLU_SONAME)
-            set(GLU_LIBRARY ${VISIT_OSMESA_DIR}/lib/${GLU_SONAME})
-        endif()
+        #if(GLU_SONAME)
+        #    string(REPLACE "SONAME" "" GLU_SONAME ${GLU_SONAME})
+        #    string(STRIP ${GLU_SONAME} GLU_SONAME)
+        #    set(GLU_LIBRARY ${VISIT_OSMESA_DIR}/lib/${GLU_SONAME})
+        #endif()
 
         execute_process(COMMAND ${CMAKE_COMMAND} -E copy
                                 ${GLU_LIBRARY}
@@ -181,8 +178,7 @@
 
     if (VISIT_LLVM_DIR)
         find_library(LLVM_LIBRARY LLVM
-                         PATH ${VISIT_LLVM_DIR}/lib
-                         NO_DEFAULT_PATH)
+                         PATH ${VISIT_LLVM_DIR}/lib)
         if (LLVM_LIBRARY)
             get_filename_component(LLVM_LIB ${LLVM_LIBRARY} NAME)
 
@@ -192,11 +188,11 @@
                             OUTPUT_VARIABLE LLVM_SONAME
                             ERROR_VARIABLE  LLVM_SONAME_ERROR)
 
-            if(LLVM_SONAME)
-                string(REPLACE "SONAME" "" LLVM_SONAME ${LLVM_SONAME})
-                string(STRIP ${LLVM_SONAME} LLVM_SONAME)
-                set(LLVM_LIBRARY ${VISIT_LLVM_DIR}/lib/${LLVM_SONAME})
-            endif()
+            #if(LLVM_SONAME)
+            #    string(REPLACE "SONAME" "" LLVM_SONAME ${LLVM_SONAME})
+            #    string(STRIP ${LLVM_SONAME} LLVM_SONAME)
+            #    set(LLVM_LIBRARY ${VISIT_LLVM_DIR}/lib/${LLVM_SONAME})
+            #endif()
             execute_process(COMMAND ${CMAKE_COMMAND} -E copy
                                   ${LLVM_LIBRARY}
                                   ${VISIT_BINARY_DIR}/lib/osmesa/)
