--- gompi-2020a/build_visit3_2_1.orig	2021-08-03 12:50:24.418763913 +0000
+++ gompi-2020a/build_visit3_2_1	2021-08-03 12:56:04.731586594 +0000
@@ -19570,6 +19570,7 @@
                     echo "VISIT_OPTION_DEFAULT(QT_QTUITOOLS_INCLUDE_DIR ${QT_QTUITOOLS_INCLUDE_DIR})" >> $HOSTCONF
                     echo "VISIT_OPTION_DEFAULT(VISIT_QT_BIN ${QT_BIN_DIR})" >> $HOSTCONF
                     echo "SET(VISIT_QT_SKIP_INSTALL ON)" >> $HOSTCONF
+		    echo "VISIT_OPTION_DEFAULT(VISIT_QT_DIR ${QT_INSTALL_DIR})" >> $HOSTCONF
                 else
                     echo "VISIT_OPTION_DEFAULT(VISIT_QT_DIR \${VISITHOME}/qt/\${QT_VERSION}/\${VISITARCH})" >> $HOSTCONF
                 fi
