--- gompi-2020a/build_visit3_2_1.orig	2021-08-03 11:22:43.237700579 +0000
+++ gompi-2020a/build_visit3_2_1	2021-08-03 11:23:10.573012888 +0000
@@ -10710,7 +10710,7 @@
         # In order to ensure $FORTRANARGS is expanded to build the arguments to
         # configure, we wrap the invokation in 'sh -c "..."' syntax
         issue_command sh -c "./configure CXX=\"$CXX_COMPILER\" CC=\"$C_COMPILER\" \
-        CFLAGS=\"$CFLAGS $C_OPT_FLAGS\" LIBS=\"-lm\" \
+        CFLAGS=\"$CFLAGS $C_OPT_FLAGS\" LIBS=\"-lm -ltirpc\" \
         CPPFLAGS=\"-I$VISITDIR/${VTK_INSTALL_DIR}/${VTK_VERSION}/$VISITARCH/include/vtk-${VTK_SHORT_VERSION} \
         -I$VISITDIR/${VTK_INSTALL_DIR}/${VTK_VERSION}/$VISITARCH/include/vtk-${VTK_SHORT_VERSION}/vtkjpeg\" \
         $FORTRANARGS $hdf4_build_mode \
