fix hardcoded 'gcc' + add optimization flags through $CFLAGS
author: Kenneth Hoste (HPC-UGent)
--- snape-pooled/trunk/Makefile.orig	2016-03-10 10:23:16.000000000 +0100
+++ snape-pooled/trunk/Makefile	2016-03-10 10:23:32.000000000 +0100
@@ -8,6 +8,6 @@
 	latex $<
 	dvips -o snape-pooled-manual.ps snape-pooled-manual.dvi
 snape-pooled-fast: snape-pooled.c
-	gcc -g -o $@ $< -lm
+	${CC} ${CFLAGS} -g -o $@ $< -lm
 clean:
 	rm *.cmx *.cmi *.cmo pooled.o snape-pooled *.log *.ps *.dvi *.aux
