--- rsem-1.2.11/makefile	2014-02-15 04:28:50.000000000 +1300
+++ rsem-1.2.11/makefile	2014-03-03 10:41:58.070308000 +1300
@@ -1,6 +1,6 @@
 CC = g++
-CFLAGS = -Wall -c -I.
-COFLAGS = -Wall -O3 -ffast-math -c -I.
+CFLAGS = -Wall -c -I. -I$(EBROOTZLIB)/include
+COFLAGS = -Wall -O3 -ffast-math -c -I. -I$(EBROOTZLIB)/include
 PROGRAMS = rsem-extract-reference-transcripts rsem-synthesis-reference-transcripts rsem-preref rsem-parse-alignments rsem-build-read-index rsem-run-em rsem-tbam2gbam rsem-run-gibbs rsem-calculate-credibility-intervals rsem-simulate-reads rsem-bam2wig rsem-get-unique rsem-bam2readdepth rsem-sam-validator rsem-scan-for-paired-end-reads
 
 all : $(PROGRAMS)
@@ -50,7 +50,7 @@
 
 
 rsem-parse-alignments : parseIt.o sam/libbam.a
-	$(CC) -o rsem-parse-alignments parseIt.o sam/libbam.a -lz -lpthread 
+	$(CC) -o rsem-parse-alignments parseIt.o sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread 
 
 parseIt.o : utils.h GroupInfo.h Read.h SingleRead.h SingleReadQ.h PairedEndRead.h PairedEndReadQ.h SingleHit.h PairedEndHit.h HitContainer.h SamParser.h Transcripts.h sam/sam.h sam/bam.h parseIt.cpp
 	$(CC) -Wall -O2 -c -I. parseIt.cpp
@@ -83,7 +83,7 @@
 sampling.h : boost/random.hpp
 
 rsem-run-em : EM.o sam/libbam.a
-	$(CC) -o rsem-run-em EM.o sam/libbam.a -lz -lpthread
+	$(CC) -o rsem-run-em EM.o sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread
 
 EM.o : utils.h my_assert.h Read.h SingleRead.h SingleReadQ.h PairedEndRead.h PairedEndReadQ.h SingleHit.h PairedEndHit.h Model.h SingleModel.h SingleQModel.h PairedEndModel.h PairedEndQModel.h Refs.h GroupInfo.h HitContainer.h ReadIndex.h ReadReader.h Orientation.h LenDist.h RSPD.h QualDist.h QProfile.h NoiseQProfile.h ModelParams.h RefSeq.h RefSeqPolicy.h PolyARules.h Profile.h NoiseProfile.h Transcript.h Transcripts.h HitWrapper.h BamWriter.h sam/bam.h sam/sam.h simul.h sam_rsem_aux.h sampling.h boost/random.hpp EM.cpp
 	$(CC) $(COFLAGS) EM.cpp
@@ -93,13 +93,13 @@
 BamConverter.h : utils.h my_assert.h sam/sam.h sam/bam.h sam_rsem_aux.h sam_rsem_cvt.h bc_aux.h Transcript.h Transcripts.h
 
 rsem-tbam2gbam : utils.h Transcripts.h Transcript.h bc_aux.h BamConverter.h sam/sam.h sam/bam.h sam/libbam.a sam_rsem_aux.h sam_rsem_cvt.h tbam2gbam.cpp sam/libbam.a
-	$(CC) -O3 -Wall tbam2gbam.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall tbam2gbam.cpp sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread -o $@
 
 rsem-bam2wig : utils.h my_assert.h wiggle.h wiggle.o sam/libbam.a bam2wig.cpp
-	$(CC) -O3 -Wall bam2wig.cpp wiggle.o sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall bam2wig.cpp wiggle.o sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread -o $@
 
 rsem-bam2readdepth : utils.h my_assert.h wiggle.h wiggle.o sam/libbam.a bam2readdepth.cpp
-	$(CC) -O3 -Wall bam2readdepth.cpp wiggle.o sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall bam2readdepth.cpp wiggle.o sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread -o $@
 
 wiggle.o: sam/bam.h sam/sam.h wiggle.cpp wiggle.h
 	$(CC) $(COFLAGS) wiggle.cpp
@@ -127,13 +127,13 @@
 	$(CC) $(COFLAGS) calcCI.cpp
 
 rsem-get-unique : sam/bam.h sam/sam.h getUnique.cpp sam/libbam.a
-	$(CC) -O3 -Wall getUnique.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall getUnique.cpp sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread -o $@
 
 rsem-sam-validator : sam/bam.h sam/sam.h my_assert.h samValidator.cpp sam/libbam.a
-	$(CC) -O3 -Wall samValidator.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall samValidator.cpp sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread -o $@
 
 rsem-scan-for-paired-end-reads : sam/bam.h sam/sam.h my_assert.h scanForPairedEndReads.cpp sam/libbam.a
-	$(CC) -O3 -Wall scanForPairedEndReads.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall scanForPairedEndReads.cpp sam/libbam.a -L$(EBROOTZLIB)/lib -lz -lpthread -o $@
 
 .PHONY: ebseq
 
--- rsem-1.2.11/sam/Makefile	2014-02-15 04:28:51.000000000 +1300
+++ rsem-1.2.11/sam/Makefile	2014-03-03 10:41:58.071514000 +1300
@@ -16,10 +16,10 @@
 			cut_target.o phase.o bam2depth.o padding.o bedcov.o bamshuf.o \
 			bam_tview_curses.o bam_tview_html.o
 PROG=		samtools
-INCLUDES=	-I.
+INCLUDES=	-I$(EBROOTNCURSES)/include -I$(EBROOTZLIB)/include -I.
 SUBDIRS=	. bcftools misc
-LIBPATH=
-LIBCURSES=	-lcurses # -lXCurses
+LIBPATH= -L$(EBROOTZLIB)/lib 
+LIBCURSES=	-L$(EBROOTNCURSES)/lib -lcurses -L$(EBROOTZLIB)/lib -lz # -lXCurses
 
 .SUFFIXES:.c .o
 .PHONY: all lib
@@ -48,13 +48,13 @@
 		$(AR) -csru $@ $(LOBJS)
 
 samtools:lib-recur $(AOBJS)
-		$(CC) $(CFLAGS) -o $@ $(AOBJS) $(LDFLAGS) libbam.a -Lbcftools -lbcf $(LIBPATH) $(LIBCURSES) -lm -lz -lpthread
+		$(CC) $(CFLAGS) -o $@ $(AOBJS) $(LDFLAGS) libbam.a -Lbcftools -lbcf $(LIBPATH) $(LIBCURSES) -lm -L$(EBROOTZLIB)/lib -lz -lpthread
 
 razip:razip.o razf.o $(KNETFILE_O)
-		$(CC) $(CFLAGS) -o $@ $^ -lz
+		$(CC) $(CFLAGS) -o $@ $^ -L$(EBROOTZLIB)/lib -lz
 
 bgzip:bgzip.o bgzf.o $(KNETFILE_O)
-		$(CC) $(CFLAGS) -o $@ $^ -lz -lpthread
+		$(CC) $(CFLAGS) -o $@ $^ -L$(EBROOTZLIB)/lib -lz -lpthread
 
 bgzf.o:bgzf.c bgzf.h
 		$(CC) -c $(CFLAGS) $(DFLAGS) -DBGZF_CACHE $(INCLUDES) bgzf.c -o $@
@@ -85,10 +85,10 @@
 
 
 libbam.1.dylib-local:$(LOBJS)
-		libtool -dynamic $(LOBJS) -o libbam.1.dylib -lc -lz
+		libtool -dynamic $(LOBJS) -o libbam.1.dylib -lc -L$(EBROOTZLIB)/lib  -lz
 
 libbam.so.1-local:$(LOBJS)
-		$(CC) -shared -Wl,-soname,libbam.so -o libbam.so.1 $(LOBJS) -lc -lz
+		$(CC) -shared -Wl,-soname,libbam.so -o libbam.so.1 $(LOBJS) -lc -L$(EBROOTZLIB)/lib  -lz
 
 dylib:
 		@$(MAKE) cleanlocal; \
