Add install target.

Åke Sandgren, 2022-05-02
diff --git a/tmap/CMakeLists.txt b/tmap/CMakeLists.txt
index 24ff444..e3ec61f 100644
--- a/tmap/CMakeLists.txt
+++ b/tmap/CMakeLists.txt
@@ -57,3 +57,6 @@ endif(MSVC)
 
 target_link_libraries(tmap PRIVATE Threads::Threads)
 target_link_libraries(tmap PRIVATE ${OpenMP_CXX_FLAGS})
+
+include(GNUInstallDirs)
+install(TARGETS tmap LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages")
