Having conflicts with OpenImageIO in both EasyBuild and Nix
author: Chris Want (University of Alberta)
--- blender-2.79b/intern/cycles/CMakeLists.txt.orig	2018-11-19 23:06:44.525367937 +0000
+++ blender-2.79b/intern/cycles/CMakeLists.txt	2018-11-19 23:15:14.733598330 +0000
@@ -210,13 +210,13 @@
 endif()
 
 include_directories(
-	SYSTEM
 	${BOOST_INCLUDE_DIR}
 	${OPENIMAGEIO_INCLUDE_DIRS}
 	${OPENIMAGEIO_INCLUDE_DIRS}/OpenImageIO
 	${OPENEXR_INCLUDE_DIR}
 	${OPENEXR_INCLUDE_DIRS}
 	${PUGIXML_INCLUDE_DIR}
+	SYSTEM
 )
 
 if(CYCLES_STANDALONE_REPOSITORY)
