diff -ruN colour-0.4.6.orig/colour/blindness/machado2009.py colour-0.4.6/colour/blindness/machado2009.py
--- colour-0.4.6.orig/colour/blindness/machado2009.py	2024-10-11 07:37:53.000000000 +0200
+++ colour-0.4.6/colour/blindness/machado2009.py	2025-05-28 15:32:51.896445370 +0200
@@ -120,17 +120,17 @@
 
     R, G, B = tsplit(primaries.values)
 
-    WS_R = np.trapezoid(R * WS, wavelengths)  # pyright: ignore
-    WS_G = np.trapezoid(G * WS, wavelengths)  # pyright: ignore
-    WS_B = np.trapezoid(B * WS, wavelengths)  # pyright: ignore
-
-    YB_R = np.trapezoid(R * YB, wavelengths)  # pyright: ignore
-    YB_G = np.trapezoid(G * YB, wavelengths)  # pyright: ignore
-    YB_B = np.trapezoid(B * YB, wavelengths)  # pyright: ignore
-
-    RG_R = np.trapezoid(R * RG, wavelengths)  # pyright: ignore
-    RG_G = np.trapezoid(G * RG, wavelengths)  # pyright: ignore
-    RG_B = np.trapezoid(B * RG, wavelengths)  # pyright: ignore
+    WS_R = np.trapz(R * WS, wavelengths)  # pyright: ignore
+    WS_G = np.trapz(G * WS, wavelengths)  # pyright: ignore
+    WS_B = np.trapz(B * WS, wavelengths)  # pyright: ignore
+
+    YB_R = np.trapz(R * YB, wavelengths)  # pyright: ignore
+    YB_G = np.trapz(G * YB, wavelengths)  # pyright: ignore
+    YB_B = np.trapz(B * YB, wavelengths)  # pyright: ignore
+
+    RG_R = np.trapz(R * RG, wavelengths)  # pyright: ignore
+    RG_G = np.trapz(G * RG, wavelengths)  # pyright: ignore
+    RG_B = np.trapz(B * RG, wavelengths)  # pyright: ignore
 
     M_G = as_float_array(
         [
@@ -219,8 +219,8 @@
             "deuteranomaly simulation."
         )
 
-    area_L = np.trapezoid(L, cmfs.wavelengths)  # pyright: ignore
-    area_M = np.trapezoid(M, cmfs.wavelengths)  # pyright: ignore
+    area_L = np.trapz(L, cmfs.wavelengths)  # pyright: ignore
+    area_M = np.trapz(M, cmfs.wavelengths)  # pyright: ignore
 
     def alpha(x: NDArrayFloat) -> NDArrayFloat:
         """Compute :math:`alpha` factor."""
diff -ruN colour-0.4.6.orig/colour/colorimetry/photometry.py colour-0.4.6/colour/colorimetry/photometry.py
--- colour-0.4.6.orig/colour/colorimetry/photometry.py	2024-10-11 07:37:53.000000000 +0200
+++ colour-0.4.6/colour/colorimetry/photometry.py	2025-05-28 15:33:54.754453000 +0200
@@ -84,7 +84,7 @@
         extrapolator_kwargs={"method": "Constant", "left": 0, "right": 0},
     )
 
-    flux = K_m * np.trapezoid(lef.values * sd.values, sd.wavelengths)  # pyright: ignore
+    flux = K_m * np.trapz(lef.values * sd.values, sd.wavelengths)  # pyright: ignore
 
     return as_float_scalar(flux)
 
@@ -130,9 +130,9 @@
         extrapolator_kwargs={"method": "Constant", "left": 0, "right": 0},
     )
 
-    efficiency = np.trapezoid(  # pyright: ignore
+    efficiency = np.trapz(  # pyright: ignore
         lef.values * sd.values, sd.wavelengths
-    ) / np.trapezoid(  # pyright: ignore
+    ) / np.trapz(  # pyright: ignore
         sd.values, sd.wavelengths
     )
 
