--- configure.ac.orig	2019-08-06 22:02:53.340429091 +0000
+++ configure.ac	2019-08-06 22:03:01.690472492 +0000
@@ -48,7 +48,7 @@
 dnl DEFAULT_VECTOR_FLAG=
 dnl DEFAULT_VECTOR_FLAG=-mavx
 dnl DEFAULT_VECTOR_FLAG=-msse3
-DEFAULT_VECTOR_FLAG=-march=native
+DEFAULT_VECTOR_FLAG=
 AC_ARG_ENABLE([native],
               [AS_HELP_STRING([--disable-native],
                               [Remove AVX or SSE compilation options.])])
