--- scripts/fastx_barcode_splitter.pl.orig	2017-12-13 21:37:20.342301303 +0000
+++ scripts/fastx_barcode_splitter.pl	2017-12-13 21:37:28.885223385 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 #    FASTX-toolkit - FASTA/FASTQ preprocessing tools.
 #    Copyright (C) 2009-2013  A. Gordon (assafgordon@gmail.com)
--- scripts/fasta_clipping_histogram.pl.orig	2017-12-13 21:36:45.582618345 +0000
+++ scripts/fasta_clipping_histogram.pl	2017-12-13 21:36:56.885515249 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 
 #    FASTX-toolkit - FASTA/FASTQ preprocessing tools.
 #    Copyright (C) 2009-2013  A. Gordon (assafgordon@gmail.com)
