Error when downloads would happen in setup.py

Author: Alexander Grund (TU Dresden)

diff --git a/python/setup.py b/python/setup.py
index cef967241..47976b1be 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -115,6 +115,7 @@ def get_thirdparty_packages(triton_cache_path):
         version_file_path = os.path.join(package_dir, "version.txt")
         if p.syspath_var_name not in os.environ and\
            (not os.path.exists(version_file_path) or Path(version_file_path).read_text() != p.url):
+            raise RuntimeError('Forbidden download')
             try:
                 shutil.rmtree(package_root_dir)
             except Exception:
@@ -139,6 +140,7 @@ def get_thirdparty_packages(triton_cache_path):
 def download_and_copy(src_path, variable, version, url_func):
     if variable in os.environ:
         return
+    raise RuntimeError('Forbidden download')
     base_dir = os.path.dirname(__file__)
     arch = platform.machine()
     if arch == "x86_64":
@@ -280,7 +282,7 @@ class CMakeBuild(build_ext):
             build_args += ["--", "/m"]
         else:
             cmake_args += ["-DCMAKE_BUILD_TYPE=" + cfg]
-            max_jobs = os.getenv("MAX_JOBS", str(2 * os.cpu_count()))
+            max_jobs = os.getenv("MAX_JOBS", str(len(os.sched_getaffinity(0))))
             build_args += ['-j' + max_jobs]
 
         if check_env_flag("TRITON_BUILD_WITH_CLANG_LLD"):
