diff -Nru Bracken-2.5.orig/src/Makefile Bracken-2.5/src/Makefile
--- Bracken-2.5.orig/src/Makefile	2019-09-04 15:57:45.613682516 +0000
+++ Bracken-2.5/src/Makefile	2019-09-04 16:01:53.239965212 +0000
@@ -1,10 +1,11 @@
-C++ = g++
+C++ = icpc
+
 CFLAGS = -c -g -O3 -pedantic -std=c++11 -fopenmp 
 
 all: kmer_counter
 
 kmer_counter: 	kmer2read_distr.o time.o taxonomy.o kraken_processing.o
-		$(C++) -o kmer2read_distr time.o kmer2read_distr.o taxonomy.o kraken_processing.o -lgomp
+		$(C++) -o kmer2read_distr time.o kmer2read_distr.o taxonomy.o kraken_processing.o -liomp5
 
 clean:
 		rm -f *.o
