diff --git a/bracken-build b/bracken-build
index 2924317..270e9b5 100755
--- a/bracken-build
+++ b/bracken-build
@@ -186,8 +186,8 @@ else
         #database.kraken not found, must create
         echo "      >> ${KINSTALL}kraken --db $DATABASE --threads ${THREADS} $filenames > $DATABASE/database.kraken.tmp"
         ${KINSTALL}kraken --db $DATABASE --threads ${THREADS} $filenames > $DATABASE/database.kraken.tmp
-
-    if [ $0 -eq 0 ]
+    fi 
+    if [ $? -eq 0 ]
     then
         mv $DATABASE/database.kraken.tmp $DATABASE/database.kraken
         echo "          Finished creating database.kraken [in DB folder]"
