Change Boltzmann's constant value to kJ/mol units

Author: Pavel Grochal (INUITS)
diff -ru --new-file wham.orig/wham/wham.h wham/wham/wham.h
--- wham.orig/wham/wham.h	2017-05-29 18:41:23.000000000 +0200
+++ wham/wham/wham.h	2020-05-27 11:48:14.455178702 +0200
@@ -20,8 +20,8 @@
 #define DEGREES   360.0
 #define RADIANS   6.28318530717959
 
-#define k_B 0.0019872041 // Boltzmann's constant in kcal/mol K
-//#define k_B  0.0083144621 // Boltzmann's constant kJ/mol-K
+//#define k_B 0.0019872041 // Boltzmann's constant in kcal/mol K
+#define k_B  0.0083144621 // Boltzmann's constant kJ/mol-K
 
 
 // global (untrimmed) histogram, global to prevent reallocation
diff -ru --new-file wham.orig/wham-2d/wham-2d.h wham/wham-2d/wham-2d.h
--- wham.orig/wham-2d/wham-2d.h	2017-05-29 18:41:23.000000000 +0200
+++ wham/wham-2d/wham-2d.h	2020-05-27 11:49:13.563450564 +0200
@@ -25,8 +25,8 @@
 #define DEGREES   360.0
 #define RADIANS   6.28318530717959
 
-#define k_B 0.0019872041 // Boltzmann's constant in kcal/mol K
-//#define k_B  0.0083144621 // Boltzmann's constant kJ/mol-K
+//#define k_B 0.0019872041 // Boltzmann's constant in kcal/mol K
+#define k_B  0.0083144621 // Boltzmann's constant kJ/mol-K
 
 
 // Value inserted for the free energy of masked values
