--- gate_v6.2/cluster_tools/filemerger/Makefile.orig	2014-12-02 12:00:20.331450693 +0100
+++ gate_v6.2/cluster_tools/filemerger/Makefile	2014-12-02 12:01:35.263325236 +0100
@@ -16,6 +16,8 @@
 INCLUDE := -I./include `geant4-config --cflags` `root-config --cflags`
 LDFLAGS := `geant4-config --libs` `root-config --glibs`
 
+PREFIX := /usr/local
+
 TARGET := gjm
 
 .PHONY: all clean directories cleanall install uninstall
@@ -49,8 +51,9 @@
 
 install:
 	@echo Installing ...
-	@$(CP) $(TARGET) /usr/local/bin
+	mkdir -p $(PREFIX)/bin
+	@$(CP) $(TARGET) $(PREFIX)/bin
 
 uninstall:
 	@echo Uninstalling...
-	@$(RM) /usr/local/bin/$(TARGET)
+	@$(RM) $(PREFIX)/bin/$(TARGET)
--- gate_v6.2/cluster_tools/jobsplitter/Makefile.orig	2014-12-02 12:01:48.033648372 +0100
+++ gate_v6.2/cluster_tools/jobsplitter/Makefile	2014-12-02 12:02:32.104712220 +0100
@@ -16,6 +16,8 @@
 INCLUDE := -I./include `geant4-config --cflags`
 LDFLAGS := `geant4-config --libs`
 
+PREFIX := /usr/local
+
 TARGET := gjs
 
 .PHONY: all clean directories cleanall install uninstall
@@ -57,8 +59,9 @@
 
 install:
 	@echo Installing ...
-	@$(CP) $(TARGET) /usr/local/bin
+	mkdir -p $(PREFIX)/bin
+	@$(CP) $(TARGET) $(PREFIX)/bin
 
 uninstall:
 	@echo Uninstalling...
-	@$(RM) /usr/local/bin/$(TARGET)
+	@$(RM) $(PREFIX)/bin/$(TARGET)
