diff -Naur Gate-9.3.orig/cluster_tools/filemerger/Makefile Gate-9.3/cluster_tools/filemerger/Makefile
--- Gate-9.3.orig/cluster_tools/filemerger/Makefile	2023-07-08 03:01:05.335087465 +0000
+++ Gate-9.3/cluster_tools/filemerger/Makefile	2023-07-08 03:07:43.833592531 +0000
@@ -12,6 +12,8 @@
 vpath %.hh ./include
 vpath %.cc ./src
 
+PREFIX := /usr/local
+
 CXXFLAGS :=
 INCLUDE := -I./include `geant4-config --cflags` `root-config --cflags`
 LDFLAGS := `geant4-config --libs` `root-config --glibs`
@@ -49,8 +51,9 @@
 
 install:
 	@echo Installing ...
-	@$(CP) $(TARGET) /usr/local/bin
+	mkdir -p $(PREFIX)/bin
+	@$(CP) $(TARGET) $(PREFIX)/bin
 
 uninstall:
 	@echo Uninstalling...
-	@$(RM) /usr/local/bin/$(TARGET)
+	@$(RM) $(PREFIX)/bin/$(TARGET)
diff -Naur Gate-9.3.orig/cluster_tools/jobsplitter/Makefile Gate-9.3/cluster_tools/jobsplitter/Makefile
--- Gate-9.3.orig/cluster_tools/jobsplitter/Makefile	2023-07-08 03:01:05.353087618 +0000
+++ Gate-9.3/cluster_tools/jobsplitter/Makefile	2023-07-08 03:08:05.680788533 +0000
@@ -12,6 +12,8 @@
 vpath %.hh ./include
 vpath %.cc ./src
 
+PREFIX := /usr/local
+
 CXXFLAGS := -DGC_DEFAULT_PLATFORM=\"condor\"
 INCLUDE := -I./include `geant4-config --cflags`
 LDFLAGS := `geant4-config --libs`
@@ -57,8 +59,8 @@
 
 install:
 	@echo Installing ...
-	@$(CP) $(TARGET) /usr/local/bin
+	@$(CP) $(TARGET) $(PREFIX)/bin/
 
 uninstall:
 	@echo Uninstalling...
-	@$(RM) /usr/local/bin/$(TARGET)
+	@$(RM) $(PREFIX)/bin/$(TARGET)
