# Add missing namespace specificator.
# Author: maxim-masterov (SURF)
#
diff -Nru Gate-8.2.orig/source/general/src/GateIAEAPhsp.cc Gate-8.2/source/general/src/GateIAEAPhsp.cc
--- Gate-8.2.orig/source/general/src/GateIAEAPhsp.cc	2022-01-03 10:55:29.472080000 +0100
+++ Gate-8.2/source/general/src/GateIAEAPhsp.cc	2022-01-04 10:58:47.913831000 +0100
@@ -343,7 +343,7 @@
       *Emax = 0.f;
       for(int i=0;i<MAX_NUM_PARTICLES;i++)
       {
-        *Emax = (IAEA_Float) max(*Emax,p_iaea_header[*id]->maximumKineticEnergy[i]);
+        *Emax = (IAEA_Float) std::max((double)(*Emax),p_iaea_header[*id]->maximumKineticEnergy[i]);
       }
       return;
 }
