--- ./bed2diffs/src/Makefile.orig	2017-10-27 19:59:26.361806426 +0000
+++ ./bed2diffs/src/Makefile	2017-10-27 20:59:12.388429690 +0000
@@ -1,5 +1,5 @@
 
-PLINKIO = /usr/local
+PLINKIO = ${EBROOTLIBPLINKIO}
 EXE = bed2diffs_v1
 OBJ = bed2diffs_v1.o data.o
 
--- ./runeems_sats/src/Makefile.orig	2017-10-27 21:08:54.380041331 +0000
+++ ./runeems_sats/src/Makefile	2017-10-27 21:09:27.128739303 +0000
@@ -1,7 +1,7 @@
 
-EIGEN_INC = /usr/local/opt/eigen/include/eigen3
-BOOST_LIB = /usr/local/lib
-BOOST_INC = /usr/local/include
+EIGEN_INC = ${EBROOTEIGEN}/include/Eigen
+BOOST_LIB = ${EBROOTBOOST}/lib
+BOOST_INC = ${EBROOTBOOST}/include
 
 EXE = runeems_sats
 OBJ = runeems_sats.o eems.o util.o mcmc.o draw.o habitat.o graph.o
--- ./runeems_snps/src/Makefile.orig	2017-10-27 21:04:11.172662894 +0000
+++ ./runeems_snps/src/Makefile	2017-10-27 21:06:57.502122902 +0000
@@ -1,7 +1,7 @@
 
-EIGEN_INC = /usr/local/opt/eigen/include/eigen3
-BOOST_LIB = /usr/local/lib
-BOOST_INC = /usr/local/include
+EIGEN_INC = ${EBROOTEIGEN}/include/Eigen
+BOOST_LIB = ${EBROOTBOOST}/lib
+BOOST_INC = ${EBROOTBOOST}/include
 
 EXE = runeems_snps
 OBJ = runeems_snps.o eems.o util.o mcmc.o draw.o habitat.o graph.o
