diff -ru dudes-dudes_v0.08.original/DUDesDB.py dudes-dudes_v0.08.patched/DUDesDB.py
--- dudes-dudes_v0.08.original/DUDesDB.py	2017-11-08 13:11:15.000000000 +0100
+++ dudes-dudes_v0.08.patched/DUDesDB.py	2018-02-01 12:54:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/env python
 # The MIT License (MIT)
 # 
 # Copyright (c) 2015 - Vitor C. Piro - PiroV@rki.de - vitorpiro@gmail.com
diff -ru dudes-dudes_v0.08.original/DUDes.py dudes-dudes_v0.08.patched/DUDes.py
--- dudes-dudes_v0.08.original/DUDes.py	2017-11-08 13:11:15.000000000 +0100
+++ dudes-dudes_v0.08.patched/DUDes.py	2018-02-01 12:54:37.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/env python
 # The MIT License (MIT)
 # 
 # Copyright (c) 2015 - Vitor C. Piro - PiroV@rki.de - vitorpiro@gmail.com
