One can't do += on find_library

Åke Sandgren, 2025-06-17
diff -ru util-linux-2.41.orig/meson.build util-linux-2.41/meson.build
--- util-linux-2.41.orig/meson.build	2025-03-18 13:50:51.821370364 +0100
+++ util-linux-2.41/meson.build	2025-06-17 09:21:40.730532032 +0200
@@ -1479,7 +1479,7 @@
 endif
 
 if not cc.has_function('sem_close') and conf.get('HAVE_SEMAPHORE_H').to_string() == '1'
-  posixipc_libs += cc.find_library('pthread', required : true)
+  posixipc_libs = [posixipc_libs, cc.find_library('pthread', required : true)]
 endif
 
 opt = get_option('build-ipcmk').require(has_seminfo_type).allowed()
