--- setup.py.ORIG	2017-01-03 15:39:57.138359435 +0100
+++ setup.py	2017-01-03 15:40:14.824909536 +0100
@@ -59,7 +59,7 @@
 
 include_dirs = _find_args('-i', 'INCLUDE_DIRS')
 library_dirs = _find_args('-l', 'LIBRARY_DIRS')
-szip_installed = 'SZIP' in os.environ
+szip_installed = 'EBROOTSZIP' in os.environ
 compress = 'NO_COMPRESS' not in os.environ
 extra_link_args = os.environ.get('LINK_ARGS', '')
 
