diff --git a/Code/CMakeLists.txt b/Code/CMakeLists.txt
index 7939ad5..dcfb373 100644
--- a/Code/CMakeLists.txt
+++ b/Code/CMakeLists.txt
@@ -73,14 +73,14 @@ include(SimVascularFunctionCheckCompilerFlags)
 #-----------------------------------------------------------------------------
 # RPATH handling
 # Skip the full RPATH for the build tree
-set(CMAKE_SKIP_BUILD_RPATH  TRUE)
+#set(CMAKE_SKIP_BUILD_RPATH  TRUE)
 # Use the install RPATH
-set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+#set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
 # The RPATH that will be used when installing
-set(CMAKE_INSTALL_RPATH "")
+#set(CMAKE_INSTALL_RPATH "")
 # Automatically add determined parts of the RPATH which point to
 # directories outside the build tree to the install tree
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+#set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 #-----------------------------------------------------------------------------
 
 #-----------------------------------------------------------------------------
