--- /home/myuser/.local/easybuild/build/dagmc_dev/3.2.2/foss-2022a/DAGMC/cmake/FindMOAB.cmake.orig	2024-04-11 11:00:09.538052758 +0000
+++ /home/myuser/.local/easybuild/build/dagmc_dev/3.2.2/foss-2022a/DAGMC/cmake/FindMOAB.cmake	2024-04-11 11:06:11.078058472 +0000
@@ -2,9 +2,15 @@
 
 # Find MOAB cmake config file
 # Only used to determine the location of the HDF5 with which MOAB was built
-set(MOAB_SEARCH_DIRS)
-file(GLOB MOAB_SEARCH_DIRS ${MOAB_SEARCH_DIRS} "${MOAB_DIR}/lib*/cmake/MOAB")
-string(REPLACE "\n" ";" MOAB_SEARCH_DIRS ${MOAB_SEARCH_DIRS})
+
+# Directly set MOAB_SEARCH_DIRS
+set(MOAB_SEARCH_DIRS
+    "${MOAB_DIR}/lib/cmake/MOAB"
+    "${MOAB_DIR}/lib64/cmake/MOAB"
+    "${MOAB_DIR}/cmake/MOAB"
+    "${MOAB_DIR}/MOAB"
+    "${MOAB_DIR}"
+    )
 find_path(MOAB_CMAKE_CONFIG
   NAMES MOABConfig.cmake
   PATHS ${MOAB_SEARCH_DIRS}
