Make sure easybuild libs are used for llvmlite
Author: Ariel Lozano (ULB)
--- a/ffi/Makefile.linux	2019-10-10 21:15:38.000000000 +0200
+++ b/ffi/Makefile.linux	2019-11-13 12:22:07.061890000 +0100
@@ -7,7 +7,7 @@
 
 CXXFLAGS := $(CPPFLAGS) $(CXXFLAGS) $(LLVM_CXXFLAGS) $(CXX_FLTO_FLAGS)
 LDFLAGS := $(LDFLAGS) $(LLVM_LDFLAGS) $(LD_FLTO_FLAGS)
-LIBS = $(LLVM_LIBS)
+LIBS := $(LIBS) $(LLVM_LIBS)
 INCLUDE = core.h
 SRC = assembly.cpp bitcode.cpp core.cpp initfini.cpp module.cpp value.cpp \
 	  executionengine.cpp transforms.cpp passmanagers.cpp targets.cpp dylib.cpp \
