--- ffi/value.cpp.orig	2022-12-07 16:25:09.106947789 +0000
+++ ffi/value.cpp	2022-12-07 16:25:58.887604268 +0000
@@ -154,7 +154,7 @@
     Argument *arg = unwrap<Argument>(A);
     unsigned argno = arg->getArgNo();
     AttributeSet attrs =
-        arg->getParent()->getAttributes().getParamAttributes(argno);
+        arg->getParent()->getAttributes().getParamAttrs(argno);
     return wrap(new AttributeSetIterator(attrs.begin(), attrs.end()));
 }
 
