Update CMakeLists.txt to use an external cxxopts
diff -Nru tsne-cuda-3.0.1.orig/CMakeLists.txt tsne-cuda-3.0.1/CMakeLists.txt
--- tsne-cuda-3.0.1.orig/CMakeLists.txt	2024-01-18 17:31:58.841767793 +0000
+++ tsne-cuda-3.0.1/CMakeLists.txt	2024-01-18 17:34:46.095207526 +0000
@@ -157,6 +157,14 @@
 endif()
 include_directories(${FAISS_INCLUDE_DIR})
 
+# CXXOPTS Configuration
+#-------------------------------------------------------------------------------
+find_package(CXXOPTS REQUIRED)
+if(NOT ${CXXOPTS_FOUND})
+    message("-- CXXOPTS not installed. PLease install CXXOPTS.")
+endif()
+include_directories(${CXXOPTS_INCLUDE_DIR})
+
 # Project Setup
 #-------------------------------------------------------------------------------
 include_directories(
@@ -164,7 +172,6 @@
     src/include
     ${CUDA_INCLUDE_DIRS}
     third_party/
-    third_party/cxxopts/include/
     ${ZMQ_INCLUDE_DIR}
 )
 link_directories(
