--- spider/src/copy1.f.orig	2015-06-16 18:11:37.000000000 +0300
+++ spider/src/copy1.f	2015-06-16 18:12:12.000000000 +0300
@@ -72,6 +72,8 @@
         INTEGER, PARAMETER    :: LUNXM1  = 83 
         INTEGER, PARAMETER    :: LUNXM2  = 84 
 
+        LOGICAL  :: VERBOSET
+
 C       FROM/'AS','MR','PD','RA','NI','TE','VA','EM','NT','XP'/
   
 C       TO/'AS','BR','MR','PO','RA','TI','XP','OP','JPG'/  
--- spider/src/pr3d.f.orig	2015-06-16 18:49:45.000000000 +0300
+++ spider/src/pr3d.f	2015-06-16 18:50:05.000000000 +0300
@@ -75,7 +75,7 @@
 
         CALL SET_MPI(ICOMM,MYPID,MPIERR)
 
-        WANTSQRTS = (FSCOP == .TRUE.)
+        WANTSQRTS = (FSCOP .eqv. .TRUE.)
 
 C       INPUT FIRST IMAGE
         MAXIM = 0
