diff -ru sprng2.0.orig/TESTS/metropolis.c sprng2.0/TESTS/metropolis.c
--- sprng2.0.orig/TESTS/metropolis.c	1999-06-29 18:42:11.000000000 +0200
+++ sprng2.0/TESTS/metropolis.c	2014-10-11 13:03:19.742813305 +0200
@@ -154,7 +154,7 @@
  
   #ifdef SYNC
     printf("\n\tStreams are synchronized!\n");
-  #elif !SYNC
+  #else
     printf("\n\tStreams are not synchronized!\n");
   #endif
 printf("         \tEnergy\t\tEnergy_error\tSigma_Energy\tCv\t\tCv_error\tSigma_Cv\n");
@@ -169,7 +169,7 @@
 	dE = energy_difference(k);
 #ifdef SYNC
 	if(expJ[dE]>sprng(genptr[k])){ 
-#elif !SYNC
+#else
 	if(dE<=0 || expJ[dE]>sprng(genptr[k])){ 
 #endif
 	  energy += dE;
@@ -321,7 +321,7 @@
 	dE = energy_difference(k);
 #ifdef SYNC
 	if(expJ[dE]>sprng(genptr[k])) 
-#elif !SYNC
+#else
 	if(dE<=0 || expJ[dE]>sprng(genptr[k]))
 #endif
 	{    
