--- setup.sh-orig	2022-03-18 08:51:22.299365270 +0100
+++ setup.sh	2022-03-18 08:47:49.044743450 +0100
@@ -1,10 +1,10 @@
 #!/bin/bash
-vaspkit_path=`pwd`
+vaspkit_path="$EBROOTVASPKIT"
 vaspkit_env=~
 cp ~/.bashrc ~/.bashrc.bk
    echo "+------------------------------------------------------------------------+"
 if [ ! -f "${vaspkit_env}/.vaspkit" ]; then
-   cp how_to_set_environment_variables ~/.vaspkit
+   cp ${vaspkit_path}/how_to_set_environment_variables ~/.vaspkit
    echo "| The ~/.vaspkit file for vaspkit code has been added.                   |"
    echo "| Please modify ~/.vaspkit based on your account settings!               |"
 else
