diff -ru ncl_ncarg-6.0.0.orig/ncarview/src/lib/libncarg_ras/hdf.c ncl_ncarg-6.0.0/ncarview/src/lib/libncarg_ras/hdf.c
--- ncl_ncarg-6.0.0.orig/ncarview/src/lib/libncarg_ras/hdf.c	2012-05-09 22:58:48.848167557 +0200
+++ ncl_ncarg-6.0.0/ncarview/src/lib/libncarg_ras/hdf.c	2012-05-09 22:59:34.458089634 +0200
@@ -51,9 +51,9 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ncarg/c.h>
-#include <hdf/hdf.h>
-#include <hdf/df.h>
-#include <hdf/dfgr.h>
+#include <hdf.h>
+#include <df.h>
+#include <dfgr.h>
 #include "ncarg_ras.h"
 #include "hdfP.h"
 #include "options.h"
diff -ru ncl_ncarg-6.0.0.orig/ni/src/ncl/NclHDF.c ncl_ncarg-6.0.0/ni/src/ncl/NclHDF.c
--- ncl_ncarg-6.0.0.orig/ni/src/ncl/NclHDF.c	2012-05-09 22:58:48.788167670 +0200
+++ ncl_ncarg-6.0.0/ni/src/ncl/NclHDF.c	2012-05-09 22:59:53.188056780 +0200
@@ -29,7 +29,7 @@
 #endif
 #include "defs.h"
 #define HAVE_NETCDF
-#include <hdf/mfhdf.h>
+#include <mfhdf.h>
 #include "NclData.h"
 #include "NclFileInterfaces.h"
 #include <math.h>
diff -ru ncl_ncarg-6.0.0.orig/ni/src/ncl/NclHDFEOS5.c ncl_ncarg-6.0.0/ni/src/ncl/NclHDFEOS5.c
--- ncl_ncarg-6.0.0.orig/ni/src/ncl/NclHDFEOS5.c	2012-05-09 22:58:48.808167632 +0200
+++ ncl_ncarg-6.0.0/ni/src/ncl/NclHDFEOS5.c	2012-05-09 22:59:53.338056502 +0200
@@ -28,7 +28,7 @@
 #endif
 #include "defs.h"
 #define HAVE_NETCDF
-#include <hdf/mfhdf.h>
+#include <mfhdf.h>
 #include "NclDataDefs.h"
 #include "NclData.h"
 #include "NclFileInterfaces.h"
diff -ru ncl_ncarg-6.0.0.orig/ni/src/ncl/NclHDFEOS.c ncl_ncarg-6.0.0/ni/src/ncl/NclHDFEOS.c
--- ncl_ncarg-6.0.0.orig/ni/src/ncl/NclHDFEOS.c	2012-05-09 22:58:48.798167651 +0200
+++ ncl_ncarg-6.0.0/ni/src/ncl/NclHDFEOS.c	2012-05-09 22:59:53.258056650 +0200
@@ -28,7 +28,7 @@
 #endif
 #include "defs.h"
 #define HAVE_NETCDF
-#include <hdf/mfhdf.h>
+#include <mfhdf.h>
 #include "NclDataDefs.h"
 #include "NclFileInterfaces.h"
 #include "NclData.h"
