Fix error in test
Author: Samuel Moors, Vrije Universiteit Brussel (VUB)
diff -ur lz4-1.9.0.orig/tests/Makefile lz4-1.9.0/tests/Makefile
--- lz4-1.9.0.orig/tests/Makefile	2019-04-16 22:55:28.000000000 +0200
+++ lz4-1.9.0/tests/Makefile	2019-04-19 15:08:53.616562619 +0200
@@ -315,7 +315,7 @@
 	! $(LZ4) -c --fast=-1 tmp-tlb-dg20K # lz4 should fail when fast=-1
 	# Test for #596
 	@echo "TEST" > tmp-tlb-test
-	$(LZ4) tmp-tlb-test
+	$(LZ4) tmp-tlb-test tmp-tlb-test.lz4
 	$(LZ4) tmp-tlb-test.lz4 tmp-tlb-test2
 	$(DIFF) -q tmp-tlb-test tmp-tlb-test2
 	@$(RM) tmp-tlb*
