declare type of loop index, otherwise some compilers complain
--- src/readwrite_fleur.f.orig	2019-09-27 10:00:52.664831000 +0800
+++ src/readwrite_fleur.f	2019-09-27 10:02:00.187676737 +0800
@@ -677,7 +677,7 @@
       complex_dp,  intent(out), optional :: cmtout(:,:,:,:,:)
       MCOMPLEX_dp, intent(out), optional :: cpwout(:,:,:,:)
       logical                            :: lmt,lpw,lbcnt
-      integer                            :: b1,b2
+      integer                            :: b1,b2,i
       if(str/=1 Mpi(.and.str/=Msize) )            Bug('Wrong stride (str) argument.')
       if(min(size(cmt,3),size(cpw,2))<size(band)) Bug('Band dimension too small.')
       if(min(size(cmt,4),size(cpw,3))<size(kpt1)) Bug('kpoint dimension too small.')
