fix for "undefined reference to `aligned_alloc'" on CentOS 6.x
author: Kenneth Hoste (HPC-UGent)
--- c-blosc-1.12.1/tests/test_common.h.orig	2017-12-08 18:25:53.622934199 +0100
+++ c-blosc-1.12.1/tests/test_common.h	2017-12-08 18:26:11.973155771 +0100
@@ -62,7 +62,7 @@
   void *block = NULL;
   int32_t res = 0;
 
-#if _ISOC11_SOURCE || (__STDC_VERSION__ >= 201112L && !defined(__APPLE__))
+#if _ISOC11_SOURCE && __STDC_VERSION__ >= 201112L && !defined(__APPLE__)
   /* C11 aligned allocation. 'size' must be a multiple of the alignment. */
   block = aligned_alloc(alignment, size);
 #elif defined(_WIN32)
