--- ioapi/Makeinclude.Linux2_x86_64ifort.orig	2023-08-17 17:23:50.454259601 +0000
+++ ioapi/Makeinclude.Linux2_x86_64ifort	2023-08-17 17:17:37.804980776 +0000
@@ -8,7 +8,7 @@
 AR   = ar
 CC   = icc
 CXX  = icpc
-FC   = ifort -auto -warn notruncated_source -Bstatic -static-intel
+FC   = ifort -auto -warn notruncated_source 
 M4   = m4
 LINK = $(CC)
 
@@ -20,10 +20,10 @@
 # MFLAGS  = -traceback -msse4.1 -xSSE4.1                        # penryn
 # MFLAGS  = -traceback -mavx2   -xAVX    -march=corei7-avx      # SandyBridge
 # MFLAGS  = -traceback -mavx    -xAVX2   -march=corei7-avx2     # Haswell
-MFLAGS    = -traceback -xHost                                   # this-machine
+MFLAGS    = -traceback                                    # this-machine
 
-OMPFLAGS  = -openmp      # for Intel compilers, version 15 or earlier
-OMPLIBS   = -openmp
+OMPFLAGS  = #-openmp      # for Intel compilers, version 15 or earlier
+OMPLIBS   = #-openmp
 # OMPFLAGS  = -openmp -qopenmp
 # OMPLIBS   = -openmp -qopenmp -qopenmp-link=static -shared_intel
 
@@ -39,7 +39,7 @@
  -DIOAPI_NO_STDOUT=1 \
  -DAVOID_FLUSH=1 -DBIT32=1
 PARFLAGS  =
-ARCHLIB   = -Bstatic
+ARCHLIB   = 
 
 # M4DEFFILE = $(PVM_ROOT)/conf/$(PVM_ARCH).m4
 # PVMLIBS   = -L$(PVM_ROOT)/lib/$(PVM_ARCH) -lpvm3
