--- OMCompiler/m4/lapack.m4.orig	2021-10-25 16:39:20.973477145 +0200
+++ OMCompiler/m4/lapack.m4	2021-10-25 16:39:35.173414588 +0200
@@ -32,7 +32,7 @@
       else
         PKGCONFIG="pkg-config --libs --silence-errors"
       fi
-      for flags in "-lopenblas" "`$PKGCONFIG lapack`" "`$PKGCONFIG lapack blas`" "-llapack -lblas" "-llapack -lblas -lm -latlas"; do
+      for flags in "-lflexiblas" "`$PKGCONFIG lapack`" "`$PKGCONFIG lapack blas`" "-llapack -lblas" "-llapack -lblas -lm -latlas"; do
         for extra in "" "-lgfortran" "-lgfortran -lquadmath"; do
           THESELIBS="$LD_LAPACK_STATIC_HEAD $flags $extra $LD_LAPACK_STATIC_TAIL"
           LIBS="-shared $THESELIBS $LAPACK_LINKER_FLAGS"
