diff -Nru iqtree2-2.1.2.orig/terraphast/CMakeLists.txt iqtree2-2.1.2/terraphast/CMakeLists.txt
--- iqtree2-2.1.2.orig/terraphast/CMakeLists.txt	2021-08-04 15:12:20.788183643 +0000
+++ iqtree2-2.1.2/terraphast/CMakeLists.txt	2021-08-04 15:13:19.200591205 +0000
@@ -14,7 +14,7 @@
 Option(TERRAPHAST_BUILD_CLIB "Build the C library" ON)
 Option(TERRAPHAST_BUILD_APPS "Build the tools" ON)
 Option(TERRAPHAST_BUILD_TESTS "Build the tests" ON)
-Option(TERRAPHAST_ARCH_NATIVE "Use -march=native compiler flag" ON)
+Option(TERRAPHAST_ARCH_NATIVE "Use  compiler flag" ON)
 
 #####################################################################
 # build library
@@ -205,13 +205,13 @@
     set(TERRAPHAST_PLATFORM_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/lib/intel")
     set(TERRAPHAST_COMPILE_FLAGS -Wall -Wextra -Wconversion -Wsign-conversion -Werror)
     if(TERRAPHAST_ARCH_NATIVE)
-        set(TERRAPHAST_COMPILE_FLAGS -march=native ${TERRAPHAST_COMPILE_FLAGS})
+        set(TERRAPHAST_COMPILE_FLAGS  ${TERRAPHAST_COMPILE_FLAGS})
     endif()
 else()
     set(TERRAPHAST_PLATFORM_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/lib/gcc_clang")
     set(TERRAPHAST_COMPILE_FLAGS -Wall -Wextra -Wpedantic -Wconversion -Wsign-conversion -Werror)
     if(TERRAPHAST_ARCH_NATIVE)
-        set(TERRAPHAST_COMPILE_FLAGS -march=native ${TERRAPHAST_COMPILE_FLAGS})
+        set(TERRAPHAST_COMPILE_FLAGS  ${TERRAPHAST_COMPILE_FLAGS})
     endif()
 endif()
 
