Patch to use EasyBuild installed LSD2

Åke Sandgren, 20201217
diff -ru iqtree2-2.1.2.orig/CMakeLists.txt iqtree2-2.1.2/CMakeLists.txt
--- iqtree2-2.1.2.orig/CMakeLists.txt	2020-10-22 08:41:46.000000000 +0200
+++ iqtree2-2.1.2/CMakeLists.txt	2020-12-17 19:11:53.000000000 +0100
@@ -581,10 +581,6 @@
    add_subdirectory(terrace)
 endif()
 
-if (USE_LSD2)
-    add_subdirectory(lsd2)
-endif()
-
 ##################################################################
 # the main executable
 ##################################################################
@@ -619,9 +615,6 @@
         if (USE_TERRAPHAST)
             set_target_properties(terrace terraphast PROPERTIES COMPILE_FLAGS "${SSE_FLAGS}")
         endif()
-        if (USE_LSD2)
-            set_target_properties(lsd2 PROPERTIES COMPILE_FLAGS "${SSE_FLAGS}")
-        endif()
         if (USE_BOOSTER)
             set_target_properties(booster PROPERTIES COMPILE_FLAGS "${SSE_FLAGS}")
         endif()
diff -ru iqtree2-2.1.2.orig/main/timetree.cpp iqtree2-2.1.2/main/timetree.cpp
--- iqtree2-2.1.2.orig/main/timetree.cpp	2020-10-22 08:41:46.000000000 +0200
+++ iqtree2-2.1.2/main/timetree.cpp	2020-12-17 19:07:50.000000000 +0100
@@ -8,7 +8,7 @@
 #include "timetree.h"
 
 #ifdef USE_LSD2
-#include "lsd2/src/lsd.h"
+#include "lsd.h"
 #endif
 
 /** map from taxon name to date */
