--- ./CMakeLists.txt.orig	2024-10-29 17:31:22.882678659 +0000
+++ ./CMakeLists.txt	2024-10-29 17:32:29.627385880 +0000
@@ -758,10 +758,6 @@
    add_subdirectory(terracetphast)
 endif()
 
-if (USE_LSD2)
-    add_subdirectory(lsd2)
-endif()
-
 add_library(kernelsse tree/phylokernelsse.cpp)
 
 if (NOT BINARY32 AND NOT IQTREE_FLAGS MATCHES "novx")
@@ -820,9 +816,6 @@
         if (USE_TERRAPHAST)
             set_target_properties(terracetphast terraphast PROPERTIES COMPILE_FLAGS "${SSE_FLAGS}")
         endif()
-        if (USE_LSD2)
-            set_target_properties(lsd2 PROPERTIES COMPILE_FLAGS "${SSE_FLAGS}")
-        endif()
         if (USE_BOOSTER)
             set_target_properties(booster PROPERTIES COMPILE_FLAGS "${SSE_FLAGS}")
         endif()
--- ./main/timetree.cpp.orig	2024-10-29 17:32:37.300467180 +0000
+++ ./main/timetree.cpp	2024-10-29 17:32:57.676683088 +0000
@@ -8,7 +8,7 @@
 #include "timetree.h"
 
 #ifdef USE_LSD2
-#include "lsd2/src/lsd.h"
+#include "lsd.h"
 #endif
 
 /** map from taxon name to date */
