--- ./pll/genericParallelization.h.orig	2024-10-29 17:33:13.740853302 +0000
+++ ./pll/genericParallelization.h	2024-10-29 17:34:00.688350750 +0000
@@ -63,7 +63,7 @@
 /******************/
 /* MPI SPECIFIC   */
 /******************/
-#if defined(_FINE_GRAIN_MPI) || (defined(_IQTREE_MPI) && defined(CLANG_UNDER_VS))
+#if defined(_FINE_GRAIN_MPI) || defined(_IQTREE_MPI)
 #include <mpi.h>
 #ifdef DEBUG_MPI_EACH_SEND
 #define DEBUG_PRINT(text, elem) printf(text, elem)
