--- iqtree2-2.3.6/CMakeLists.txt.orig	2025-02-17 14:50:38.770063068 +0100
+++ iqtree2-2.3.6/CMakeLists.txt	2025-02-17 14:55:24.387629770 +0100
@@ -541,6 +541,8 @@
 elseif (GCC OR CLANG)
     if (__ARM_NEON)
         set(SSE_FLAGS "${SSE_FLAGS} -march=armv8-a+fp+simd+crypto+crc")  # changed from -msse3
+    elseif(__riscv)
+        set(SSE_FLAGS "${SSE_FLAGS}")  # Don't want -msse3 option in RISCV
     else()
         set(SSE_FLAGS "${SSE_FLAGS} -msse3")
     endif()
