diff -ru IQ-TREE-2.0.7.orig/terraphast/CMakeLists.txt IQ-TREE-2.0.7/terraphast/CMakeLists.txt
--- IQ-TREE-2.0.7.orig/terraphast/CMakeLists.txt	2021-03-22 15:35:32.981901943 +0000
+++ IQ-TREE-2.0.7/terraphast/CMakeLists.txt	2021-03-22 15:36:52.577394681 +0000
@@ -205,13 +205,13 @@
     set(TERRAPHAST_PLATFORM_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/lib/intel")
     set(TERRAPHAST_COMPILE_FLAGS -Wall -Wextra -Wconversion -Wsign-conversion -Werror)
     if(TERRAPHAST_ARCH_NATIVE)
-        set(TERRAPHAST_COMPILE_FLAGS -march=native ${TERRAPHAST_COMPILE_FLAGS})
+        set(TERRAPHAST_COMPILE_FLAGS ${TERRAPHAST_COMPILE_FLAGS})
     endif()
 else()
     set(TERRAPHAST_PLATFORM_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/lib/gcc_clang")
     set(TERRAPHAST_COMPILE_FLAGS -Wall -Wextra -Wpedantic -Wconversion -Wsign-conversion -Werror)
     if(TERRAPHAST_ARCH_NATIVE)
-        set(TERRAPHAST_COMPILE_FLAGS -march=native ${TERRAPHAST_COMPILE_FLAGS})
+        set(TERRAPHAST_COMPILE_FLAGS ${TERRAPHAST_COMPILE_FLAGS})
     endif()
 endif()
 
