diff -ru cctbx_project/boost_adaptbx/meta_ext.cpp cctbx_project/boost_adaptbx/meta_ext.cpp
--- cctbx_project/boost_adaptbx/meta_ext.cpp	2023-06-30 22:32:31.000000000 +0200
+++ modules/cctbx_project/boost_adaptbx/meta_ext.cpp	2023-07-14 11:39:39.958095126 +0200
@@ -301,8 +301,8 @@
 #if defined(BOOST_PYTHON_HAVE_CXXABI_CXA_DEMANGLE_IS_BROKEN) \
     && !defined(USE_CONDA) \
     && !defined(__MINGW32__) // workaround for MinGW linking problem
-    result += "boost::python::cxxabi_cxa_demangle_is_broken(): "
-            + to_str(boost::python::cxxabi_cxa_demangle_is_broken()) + nl;
+//    result += "boost::python::cxxabi_cxa_demangle_is_broken(): "
+//            + to_str(boost::python::cxxabi_cxa_demangle_is_broken()) + nl;
 #endif
 #if defined(__GXX_WEAK__)
     result += "__GXX_WEAK__ = " + to_str(__GXX_WEAK__) + nl;
