diff -ru cctbx_project/libtbx/env_config.py cctbx_project-2023.6_configure_prefix/libtbx/env_config.py
--- modules/cctbx_project/libtbx/env_config.py	2023-06-30 22:32:31.000000000 +0200
+++ modules/cctbx_project_configure_prefix/libtbx/env_config.py	2023-07-18 18:04:12.355326353 +0200
@@ -2649,7 +2649,8 @@
         msvc_arch_flag=default_msvc_arch_flag,
         enable_cxx11=default_enable_cxx11,
         cxxstd=default_cxxstd,
-        skip_phenix_dispatchers=False):
+        skip_phenix_dispatchers=False,
+        prefix=''):
 
     adopt_init_args(self, locals())
     assert self.mode in build_options.supported_modes
@@ -2943,6 +2944,12 @@
       action="store_true",
       default=False,
       help="Skip all dispatchers with 'phenix' in the title")
+    parser.option("--prefix",
+      type="string",
+      action="store",
+      help="--prefix is ignored", 
+      default=None,
+      metavar="STRING")
     self.command_line = parser.process(args=args)
     if (len(self.command_line.args) == 0):
       raise RuntimeError(
Only in cctbx_project-2023.6_configure_prefix/libtbx: .env_config.py.swo
Only in cctbx_project-2023.6_configure_prefix/libtbx: .env_config.py.swp
