diff -ru cctbx_project-2023.6_ori/modules/cctbx_project/mmtbx/ions/svm/tst_classifier.py cctbx_project-2023.6_svm/modules/cctbx_project/mmtbx/ions/svm/tst_classifier.py
--- cctbx_project-2023.6_ori/modules/cctbx_project/mmtbx/ions/svm/tst_classifier.py	2023-06-30 22:32:31.000000000 +0200
+++ cctbx_project-2023.6_svm/modules/cctbx_project/mmtbx/ions/svm/tst_classifier.py	2023-08-01 11:22:35.908624694 +0200
@@ -109,7 +109,7 @@
     warnings.warn("chem_data not available, skipping this test")
   else :
     try :
-      import svm
+      from libsvm import svm
     except ImportError :
       warnings.warn("libsvm not available, skipping this test")
     else :
