Fixes for python 3.11
Author: micketeer@gmail.com

--- src/Util.cpp.orig	2023-11-30 15:37:41.303679907 +0000
+++ src/Util.cpp	2023-11-30 15:38:10.439525959 +0000
@@ -214,7 +214,7 @@
     //
     // Get name of current function.
     //
-    PyFrameObject *f = PyThreadState_GET()->frame;
+    PyFrameObject *f = PyEval_GetFrame();
     PyObjectHandle code = PyObject_GetAttrString(reinterpret_cast<PyObject*>(f), STRCAST("f_code"));
     assert(code.get());
     PyObjectHandle func = PyObject_GetAttrString(code.get(), STRCAST("co_name"));
--- src/Slice.cpp.orig	2023-11-30 15:37:31.530731550 +0000
+++ src/Slice.cpp	2019-07-31 19:49:38.000000000 +0000
@@ -21,7 +21,7 @@
 // Python headers needed for PyEval_EvalCode.
 //
 #include <compile.h>
-#include <eval.h>
+#include <ceval.h>
 
 using namespace std;
 using namespace IcePy;
