Author: Jasper Grimm <jasper.grimm@york.ac.uk>
Add <cstdint> include for gcc >= 13 support

diff -Nru NECAT-0.0.1_update20200803.orig/src/fsa/simple_align.hpp NECAT-0.0.1_update20200803/src/fsa/simple_align.hpp
--- NECAT-0.0.1_update20200803.orig/src/fsa/simple_align.hpp	2024-10-09 15:51:42.877779089 +0100
+++ NECAT-0.0.1_update20200803/src/fsa/simple_align.hpp	2024-10-09 15:52:17.356716279 +0100
@@ -2,6 +2,7 @@
 #define FSA_ALIGN_SIMPLE_ALIGN_HPP
 
 #include <array>
+#include <cstdint>
 #include <string>
 #include <unordered_map>
 #include <vector>
