diff -ur openspeedshop-aac3d20da34391e7a46cb3360568126b8be150a8.orig/libopenss-guidialogs/AttachProgramDialog.cxx openspeedshop-aac3d20da34391e7a46cb3360568126b8be150a8/libopenss-guidialogs/AttachProgramDialog.cxx
--- openspeedshop-aac3d20da34391e7a46cb3360568126b8be150a8.orig/libopenss-guidialogs/AttachProgramDialog.cxx	2017-11-10 20:15:30.000000000 +0000
+++ openspeedshop-aac3d20da34391e7a46cb3360568126b8be150a8/libopenss-guidialogs/AttachProgramDialog.cxx	2018-05-15 01:41:24.392207440 +0000
@@ -40,7 +40,7 @@
 #include <qobjectlist.h>
 #include <qdialog.h>
 
-
+#define MPP 1
 #ifdef MPP
 void
 AttachProgramDialog::parallelCBSelected()
