diff -ruN geant4-11.4.0.orig/cmake/Modules/G4ConfigureGNUMakeHelpers.cmake geant4-11.4.0/cmake/Modules/G4ConfigureGNUMakeHelpers.cmake
--- geant4-11.4.0.orig/cmake/Modules/G4ConfigureGNUMakeHelpers.cmake	2025-12-12 10:18:00.714791000 +0100
+++ geant4-11.4.0/cmake/Modules/G4ConfigureGNUMakeHelpers.cmake	2025-12-12 10:21:46.200302187 +0100
@@ -499,11 +499,11 @@
   endif()
 
   # - Resource file paths
-  set(GEANT4_TC_DATASETS )
-  foreach(_ds ${GEANT4_EXPORTED_DATASETS})
-    _g4tc_setenv_command(_dssetenvcmd ${SHELL_FAMILY} ${${_ds}_ENVVAR} ${${_ds}_PATH})
-    set(GEANT4_TC_DATASETS "${GEANT4_TC_DATASETS}${_dssetenvcmd}\n")
-  endforeach()
+  # set(GEANT4_TC_DATASETS )
+  # foreach(_ds ${GEANT4_EXPORTED_DATASETS})
+  # _g4tc_setenv_command(_dssetenvcmd ${SHELL_FAMILY} ${${_ds}_ENVVAR} ${${_ds}_PATH})
+  # set(GEANT4_TC_DATASETS "${GEANT4_TC_DATASETS}${_dssetenvcmd}\n")
+  # endforeach()
 
   set(GEANT4_TC_TOOLS_FONT_PATH "# FREETYPE SUPPORT NOT AVAILABLE")
   if(GEANT4_USE_FREETYPE)
@@ -1011,8 +1011,8 @@
 
     # - Set data paths
     set(GEANT4_ENV_DATASETS )
-    _g4tc_setenv_command(_dssetenvcmd ${_shell} GEANT4_DATA_DIR ${GEANT4_DATA_DIR})
-    set(GEANT4_ENV_DATASETS "${GEANT4_ENV_DATASETS}${_dssetenvcmd}\n")
+    # _g4tc_setenv_command(_dssetenvcmd ${_shell} GEANT4_DATA_DIR ${GEANT4_DATA_DIR})
+    # set(GEANT4_ENV_DATASETS "${GEANT4_ENV_DATASETS}${_dssetenvcmd}\n")
     set(_dssetenvcmd "
 # - Variables for individual datasets
 # Uncomment the line and edit the path to the dataset if installed in not standard location.")
@@ -1104,10 +1104,10 @@
 
   # - Set data paths
   set(GEANT4_ENV_DATASETS )
-  _g4tc_setenv_command(_dssetenvcmd ${_shell} GEANT4_DATA_DIR ${GEANT4_DATA_DIRW})
-  set(GEANT4_ENV_DATASETS "${GEANT4_ENV_DATASETS}${_dssetenvcmd}\n")
-  set(_dssetenvcmd "FOR /F %%i IN ( \"%GEANT4_DATA_DIR%\" ) DO set \"GEANT4_DATA_DIR=%%~fi\"")
-  set(GEANT4_ENV_DATASETS "${GEANT4_ENV_DATASETS}${_dssetenvcmd}\n\n")
+  #  _g4tc_setenv_command(_dssetenvcmd ${_shell} GEANT4_DATA_DIR ${GEANT4_DATA_DIRW})
+  #  set(GEANT4_ENV_DATASETS "${GEANT4_ENV_DATASETS}${_dssetenvcmd}\n")
+  #  set(_dssetenvcmd "FOR /F %%i IN ( \"%GEANT4_DATA_DIR%\" ) DO set \"GEANT4_DATA_DIR=%%~fi\"")
+  #  set(GEANT4_ENV_DATASETS "${GEANT4_ENV_DATASETS}${_dssetenvcmd}\n\n")
   set(_dssetenvcmd "
 rem - Variables for individual datasets
 rem Uncomment the line and edit the path to the dataset if installed in not standard location.
