--- src/Makefile.orig	2017-10-04 18:39:47.781646526 +0000
+++ src/Makefile	2017-10-04 18:42:00.040393832 +0000
@@ -55,7 +55,7 @@
 	@echo " "
 	@echo "See the README_ files in ./Portability for more information."
 
-install: checkroot LICENSE_44R2_ACCEPTED MAILHOME_44R2 makedirs cwpstuff plot sustuff tristuff tetrastuff compstuff reflstuff
+install:  makedirs cwpstuff plot sustuff tristuff tetrastuff compstuff reflstuff
 
 xtinstall: xtcwp_
 xminstall: xmcwp_
