diff -ur ispc-1.13.0.orig/cmake/GenerateBuiltins.cmake ispc-1.13.0/cmake/GenerateBuiltins.cmake
--- ispc-1.13.0.orig/cmake/GenerateBuiltins.cmake	2020-04-23 23:29:06.000000000 -0000
+++ ispc-1.13.0/cmake/GenerateBuiltins.cmake	2020-07-17 16:37:05.693931079 -0000
@@ -328,7 +328,7 @@
         list (APPEND supported_oses "ps4")
     endif()
     message (STATUS "ISPC will be built with support of ${supported_oses} for ${supported_archs}")
-    foreach (bit 32 64)
+    foreach (bit 64)
         foreach (os_name "windows" "linux" "freebsd" "macos" "android" "ios" "ps4" "web")
             foreach (arch "x86" "arm" "wasm32")
                 builtin_to_cpp(${bit} ${os_name} ${arch} "${supported_archs}" "${supported_oses}" res${bit}${os_name}${arch})
