fix for: undefined reference to `SCOTCH_errorPrint'
author: Kenneth Hoste (HPC-UGent)
--- porefoam-eb02ba3ede738befe02dc7c587d21f05fd40cb9f/pkgs/foamx4m/applications/utilities/parallelProcessing/decomposePar/Make/options.orig	2022-02-25 16:29:05.774243797 +0100
+++ porefoam-eb02ba3ede738befe02dc7c587d21f05fd40cb9f/pkgs/foamx4m/applications/utilities/parallelProcessing/decomposePar/Make/options	2022-02-25 16:29:28.484450588 +0100
@@ -10,7 +10,7 @@
     -l:libdecompositionMethods.$(SO) \
     -l:libdecomposeReconstruct.$(SO) \
     -l:libmeshTools.$(SO) \
-    -l:libscotchDecomp.$(SO) -lscotch 
+    -l:libscotchDecomp.$(SO) -lscotch -lscotcherrexit
 
 delsSkip =     -ltetFiniteElement
 delsSkip =   -lfiniteArea 
