bool is standard in new C standard and isn't allowed
to be typedefed.
author: micketeer@gmail.com
--- rmq.h.orig	2026-10-07 11:22:29.655083182 +0200
+++ rmq.h	2026-10-07 11:30:52.198503929 +0200
@@ -2,6 +2,7 @@
 #define _rmq_h_
 
 #include <math.h>
+#include <stdbool.h>
 
 #define false 0
 #define true 1
@@ -10,7 +11,6 @@
 typedef unsigned int DTidx;     // for indexing in arrays
 typedef unsigned char DTsucc;
 typedef unsigned short DTsucc2;
-typedef int bool;
 DTidx query(DTidx, DTidx);
 void RMQ_succinct(DT* a, DTidx n);
 void RMQ_succinct_destroy(void);
