diff -rau treeseg-0.2.2-orig/src/pcdPointTreeseg2txt.cpp treeseg-0.2.2/src/pcdPointTreeseg2txt.cpp
--- treeseg-0.2.2-orig/src/pcdPointTreeseg2txt.cpp	2022-04-06 19:34:28.754655748 +0100
+++ treeseg-0.2.2/src/pcdPointTreeseg2txt.cpp	2022-04-06 19:36:20.372888230 +0100
@@ -1,6 +1,7 @@
 #include "treeseg.h"
 
 #include <pcl/io/pcd_io.h>
+#include <boost/algorithm/string.hpp>
 
 int main (int argc, char **argv)
 {
diff -rau treeseg-0.2.2-orig/src/pcdPointXYZRGB2txt.cpp treeseg-0.2.2/src/pcdPointXYZRGB2txt.cpp
--- treeseg-0.2.2-orig/src/pcdPointXYZRGB2txt.cpp	2022-04-06 19:34:28.754655748 +0100
+++ treeseg-0.2.2/src/pcdPointXYZRGB2txt.cpp	2022-04-06 19:36:05.177720560 +0100
@@ -1,4 +1,5 @@
 #include <pcl/io/pcd_io.h>
+#include <boost/algorithm/string.hpp>
 
 int main (int argc, char **argv)
 {
diff -rau treeseg-0.2.2-orig/src/rxp2tree.cpp treeseg-0.2.2/src/rxp2tree.cpp
--- treeseg-0.2.2-orig/src/rxp2tree.cpp	2022-04-06 19:34:28.754655748 +0100
+++ treeseg-0.2.2/src/rxp2tree.cpp	2022-04-06 19:36:32.506022111 +0100
@@ -7,6 +7,7 @@
 #include <pcl/io/pcd_io.h>
 #include <pcl/common/common.h>
 #include <pcl/filters/passthrough.h>
+#include <boost/algorithm/string.hpp>
 
 struct pcloud 
 {
diff -rau treeseg-0.2.2-orig/src/treeseg.cpp treeseg-0.2.2/src/treeseg.cpp
--- treeseg-0.2.2-orig/src/treeseg.cpp	2022-04-06 19:34:28.755655759 +0100
+++ treeseg-0.2.2/src/treeseg.cpp	2022-04-06 19:36:45.349163827 +0100
@@ -39,6 +39,7 @@
 #include <pcl/segmentation/region_growing.h>
 #include <pcl/segmentation/sac_segmentation.h>
 #include <pcl/octree/octree.h>
+#include <boost/algorithm/string.hpp>
 
 //File IO
 
diff -rau treeseg-0.2.2-orig/src/txtPointTreeseg2pcd.cpp treeseg-0.2.2/src/txtPointTreeseg2pcd.cpp
--- treeseg-0.2.2-orig/src/txtPointTreeseg2pcd.cpp	2022-04-06 19:34:28.755655759 +0100
+++ treeseg-0.2.2/src/txtPointTreeseg2pcd.cpp	2022-04-06 19:36:56.489286752 +0100
@@ -1,6 +1,7 @@
 #include "treeseg.h"
 
 #include <pcl/io/pcd_io.h>
+#include <boost/algorithm/string.hpp>
 
 int main (int argc, char **argv)
 {
