--- projects/rocprofiler-sdk/external/CMakeLists.txt.orig	2026-09-29 14:27:29.303590461 +0530
+++ projects/rocprofiler-sdk/external/CMakeLists.txt	2026-09-29 14:27:37.769612899 +0530
@@ -21,12 +21,12 @@
 # filesystem library
 if(ROCPROFILER_BUILD_GHC_FS)
     # checkout submodule if not already checked out or clone repo if no .gitmodules file
-    rocprofiler_checkout_git_submodule(
-        RECURSIVE
-        RELATIVE_PATH external/filesystem
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        REPO_URL https://github.com/gulrak/filesystem.git
-        REPO_BRANCH "v1.5.14")
+    # rocprofiler_checkout_git_submodule(
+        # RECURSIVE
+        # RELATIVE_PATH external/filesystem
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # REPO_URL https://github.com/gulrak/filesystem.git
+        # REPO_BRANCH "v1.5.14")
 
     target_compile_definitions(
         rocprofiler-sdk-cxx-filesystem
@@ -40,12 +40,12 @@
     if(ROCPROFILER_BUILD_GTEST)
         # checkout submodule if not already checked out or clone repo if no .gitmodules
         # file
-        rocprofiler_checkout_git_submodule(
-            RECURSIVE
-            RELATIVE_PATH external/googletest
-            WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-            REPO_URL https://github.com/google/googletest.git
-            REPO_BRANCH "main")
+        # rocprofiler_checkout_git_submodule(
+            # RECURSIVE
+            # RELATIVE_PATH external/googletest
+            # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+            # REPO_URL https://github.com/google/googletest.git
+            # REPO_BRANCH "main")
 
         set(BUILD_GMOCK OFF)
         set(INSTALL_GTEST OFF)
@@ -71,12 +71,12 @@
 
 if(ROCPROFILER_BUILD_GLOG)
     # checkout submodule if not already checked out or clone repo if no .gitmodules file
-    rocprofiler_checkout_git_submodule(
-        RECURSIVE
-        RELATIVE_PATH external/glog
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        REPO_URL https://github.com/google/glog.git
-        REPO_BRANCH "master")
+    # rocprofiler_checkout_git_submodule(
+        # RECURSIVE
+        # RELATIVE_PATH external/glog
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # REPO_URL https://github.com/google/glog.git
+        # REPO_BRANCH "master")
 
     # May want to use GFLAGS in the future
     set(WITH_GFLAGS OFF)
@@ -96,12 +96,12 @@
 
 if(ROCPROFILER_BUILD_FMT)
     # checkout submodule if not already checked out or clone repo if no .gitmodules file
-    rocprofiler_checkout_git_submodule(
-        RECURSIVE
-        RELATIVE_PATH external/fmt
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        REPO_URL https://github.com/fmtlib/fmt.git
-        REPO_BRANCH "master")
+    # rocprofiler_checkout_git_submodule(
+        # RECURSIVE
+        # RELATIVE_PATH external/fmt
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # REPO_URL https://github.com/fmtlib/fmt.git
+        # REPO_BRANCH "master")
 
     set(FMT_TEST OFF)
     add_subdirectory(fmt EXCLUDE_FROM_ALL)
@@ -117,11 +117,11 @@
 
 if(NOT TARGET PTL::ptl-static)
     # checkout submodule if not already checked out or clone repo if no .gitmodules file
-    rocprofiler_checkout_git_submodule(
-        RELATIVE_PATH external/ptl
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        REPO_URL https://github.com/jrmadsen/PTL.git
-        REPO_BRANCH rocprofiler)
+    # rocprofiler_checkout_git_submodule(
+        # RELATIVE_PATH external/ptl
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # REPO_URL https://github.com/jrmadsen/PTL.git
+        # REPO_BRANCH rocprofiler)
 
     set(PTL_USE_TBB OFF)
     set(PTL_USE_GPU OFF)
@@ -131,12 +131,12 @@
     add_subdirectory(ptl EXCLUDE_FROM_ALL)
 endif()
 
-rocprofiler_checkout_git_submodule(
-    RECURSIVE
-    RELATIVE_PATH external/yaml-cpp
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/jbeder/yaml-cpp.git
-    REPO_BRANCH "master")
+# rocprofiler_checkout_git_submodule(
+    # RECURSIVE
+    # RELATIVE_PATH external/yaml-cpp
+    # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    # REPO_URL https://github.com/jbeder/yaml-cpp.git
+    # REPO_BRANCH "master")
 
 add_subdirectory(yaml-cpp EXCLUDE_FROM_ALL)
 
@@ -147,12 +147,12 @@
     INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/yaml-cpp/include>)
 
 # checkout submodule if not already checked out or clone repo if no .gitmodules file
-rocprofiler_checkout_git_submodule(
-    RECURSIVE
-    RELATIVE_PATH external/cereal
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/jrmadsen/cereal.git
-    REPO_BRANCH "rocprofiler")
+# rocprofiler_checkout_git_submodule(
+    # RECURSIVE
+    # RELATIVE_PATH external/cereal
+    # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    # REPO_URL https://github.com/jrmadsen/cereal.git
+    # REPO_BRANCH "rocprofiler")
 
 target_compile_definitions(rocprofiler-sdk-cereal
                            INTERFACE $<BUILD_INTERFACE:CEREAL_THREAD_SAFE=1>)
@@ -162,23 +162,23 @@
 
 # doxygen-awesome
 if(ROCPROFILER_BUILD_DOCS)
-    rocprofiler_checkout_git_submodule(
-        RECURSIVE
-        RELATIVE_PATH external/doxygen-awesome-css
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        TEST_FILE Makefile
-        REPO_URL https://github.com/jothepro/doxygen-awesome-css.git
-        REPO_BRANCH "main")
+    # rocprofiler_checkout_git_submodule(
+        # RECURSIVE
+        # RELATIVE_PATH external/doxygen-awesome-css
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # TEST_FILE Makefile
+        # REPO_URL https://github.com/jothepro/doxygen-awesome-css.git
+        # REPO_BRANCH "main")
 endif()
 
 # perfetto
-rocprofiler_checkout_git_submodule(
-    RECURSIVE
-    RELATIVE_PATH external/perfetto
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    TEST_FILE meson.build
-    REPO_URL https://github.com/google/perfetto
-    REPO_BRANCH "v44.0")
+# rocprofiler_checkout_git_submodule(
+    # RECURSIVE
+    # RELATIVE_PATH external/perfetto
+    # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    # TEST_FILE meson.build
+    # REPO_URL https://github.com/google/perfetto
+    # REPO_BRANCH "v44.0")
 
 add_library(rocprofiler-sdk-perfetto-static-library STATIC)
 target_sources(
@@ -196,12 +196,12 @@
     INTERFACE $<BUILD_INTERFACE:rocprofiler-sdk-perfetto-static-library>)
 
 # ELFIO
-rocprofiler_checkout_git_submodule(
-    RECURSIVE
-    RELATIVE_PATH external/elfio
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/serge1/ELFIO.git
-    REPO_BRANCH "Release_3.12")
+# rocprofiler_checkout_git_submodule(
+    # RECURSIVE
+    # RELATIVE_PATH external/elfio
+    # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    # REPO_URL https://github.com/serge1/ELFIO.git
+    # REPO_BRANCH "Release_3.12")
 
 set(ELFIO_BUILD_EXAMPLES OFF)
 set(ELFIO_BUILD_TESTS OFF)
@@ -217,12 +217,12 @@
 #
 add_subdirectory(otf2)
 
-rocprofiler_checkout_git_submodule(
-    RECURSIVE
-    RELATIVE_PATH external/json
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/nlohmann/json.git
-    REPO_BRANCH "develop")
+# rocprofiler_checkout_git_submodule(
+    # RECURSIVE
+    # RELATIVE_PATH external/json
+    # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    # REPO_URL https://github.com/nlohmann/json.git
+    # REPO_BRANCH "develop")
 
 #
 # JSON
@@ -237,13 +237,13 @@
 #
 if(ROCPROFILER_BUILD_SQLITE3)
     # checkout submodule if not already checked out or clone repo if no .gitmodules file
-    rocprofiler_checkout_git_submodule(
-        RECURSIVE
-        RELATIVE_PATH external/sqlite
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        TEST_FILE configure
-        REPO_URL https://github.com/sqlite/sqlite
-        REPO_BRANCH "version-3.47.0")
+    # rocprofiler_checkout_git_submodule(
+        # RECURSIVE
+        # RELATIVE_PATH external/sqlite
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # TEST_FILE configure
+        # REPO_URL https://github.com/sqlite/sqlite
+        # REPO_BRANCH "version-3.47.0")
 
     find_program(
         MAKE_COMMAND
@@ -283,12 +283,12 @@
 #
 if(ROCPROFILER_BUILD_PYBIND11)
     # checkout submodule if not already checked out or clone repo if no .gitmodules file
-    rocprofiler_checkout_git_submodule(
-        RECURSIVE
-        RELATIVE_PATH external/pybind11
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        REPO_URL https://github.com/pybind/pybind11.git
-        REPO_BRANCH "v2.9.2")
+    # rocprofiler_checkout_git_submodule(
+        # RECURSIVE
+        # RELATIVE_PATH external/pybind11
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # REPO_URL https://github.com/pybind/pybind11.git
+        # REPO_BRANCH "v2.9.2")
 
     target_include_directories(
         rocprofiler-sdk-pybind11 SYSTEM
@@ -303,12 +303,12 @@
 #
 if(ROCPROFILER_BUILD_GOTCHA)
     # checkout submodule if not already checked out or clone repo if no .gitmodules file
-    rocprofiler_checkout_git_submodule(
-        RECURSIVE
-        RELATIVE_PATH external/gotcha
-        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-        REPO_URL https://github.com/jrmadsen/GOTCHA.git
-        REPO_BRANCH "rocprofiler")
+    # rocprofiler_checkout_git_submodule(
+        # RECURSIVE
+        # RELATIVE_PATH external/gotcha
+        # WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        # REPO_URL https://github.com/jrmadsen/GOTCHA.git
+        # REPO_BRANCH "rocprofiler")
 
     function(add_gotcha_subdirectory)
         set(GOTCHA_ENABLE_TESTS OFF)
