netcdf.h needs to be imported before netcdf_par.h

Author: Jakub Zárybnický

diff -ru cdms-3.1.5/Src/Cdunifmodule.c cdms-3.1.5-edit/Src/Cdunifmodule.c
--- cdms-3.1.5/Src/Cdunifmodule.c	2020-07-31 03:01:33.000000000 +0200
+++ cdms-3.1.5-edit/Src/Cdunifmodule.c	2021-02-16 15:33:00.944155237 +0100
@@ -12,11 +12,11 @@
 /*#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION*/
 #include "Python.h"
 #include "numpy/ndarrayobject.h"
+#include "netcdf.h"
 #ifdef PARALLEL
 #include "mpi.h"
 #include "netcdf_par.h"
 #endif
-#include "netcdf.h"
 
 #define _CDUNIF_MODULE
 #include "Cdunifmodule.h"
