diff -ru pycf-1.6.9/setup.py pycf-1.6.9-edit/setup.py
--- pycf-1.6.9/setup.py	2021-03-02 11:22:22.131437470 +0100
+++ pycf-1.6.9-edit/setup.py	2021-03-02 11:22:31.139345625 +0100
@@ -128,13 +128,12 @@
 
 lapack_dir, lapack_lib = breakLibraryPath(lapack_libraries)
 libdirs = netcdf_libdirs + [lapack_dir]
-libs = netcdf_libs + [lapack_lib]
+libs = netcdf_libs + ['openblas']
 if USE_MKL:
   libs += ['mkl_sequential', 'mkl_rt']
 if blas_libraries:
   blas_dir, blas_lib = breakLibraryPath(blas_libraries)
   libdirs += [blas_dir]
-  libs += [blas_lib]
 
 # list all the directories that contain source file to be compiled 
 # into a shared library
@@ -167,6 +166,8 @@
                  "--netcdf_incdir={}".format(netcdf_incdir),
                  "--srcdir=./", "--builddir=./", 
                  "--config=py/libCFConfig.py"])
+print(subprocess.check_output(['pwd']).decode("utf-8"))
+print(subprocess.check_output(['cat', 'py/libCFConfig.py']).decode("utf-8"))
 
 def_macs = [('HAVE_LAPACK_UNDERSCORE', 1)]
 if logging:
