diff -ruN arp2012-guidescan_public-649b8282f299.orig/guidescan-source_code/setup.py arp2012-guidescan_public-649b8282f299/guidescan-source_code/setup.py
--- arp2012-guidescan_public-649b8282f299.orig/guidescan-source_code/setup.py	2018-05-15 15:58:21.000000000 -0700
+++ arp2012-guidescan_public-649b8282f299/guidescan-source_code/setup.py	2020-08-25 22:56:33.007065000 -0700
@@ -22,7 +22,7 @@
     long_description = long_description,
 
     install_requires = [
-        'biopython>=1.66','pysam==0.8.3','pyfaidx==0.4.7.1','bx-python==0.7.3'
+        'biopython>=1.66','pysam==0.8.4','pyfaidx==0.4.7.1','bx-python==0.7.3'
     ],
     dependency_links = ['https://github.com/bxlab/bx-python.git'
                         ],
