Remove building docs due to fetching things online

author: micketeer@gmail.com
--- doc/meson.build.orig	2025-12-15 15:22:00.441948086 +0100
+++ doc/meson.build	2025-12-15 15:22:07.824840049 +0100
@@ -1,8 +1,8 @@
-custom_target(
-  'xfce4-terminal.1',
-  input: 'xfce4-terminal.xml',
-  output: 'xfce4-terminal.1',
-  command: [xsltproc, '--nonet', '--output', '@OUTPUT@', 'http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl', '@INPUT@'],
-  install: true,
-  install_dir: get_option('prefix') / get_option('datadir') / 'man' / 'man1',
-)
+# custom_target(
+#   'xfce4-terminal.1',
+#   input: 'xfce4-terminal.xml',
+#   output: 'xfce4-terminal.1',
+#   command: [xsltproc, '--nonet', '--output', '@OUTPUT@', 'http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl', '@INPUT@'],
+#   install: true,
+#   install_dir: get_option('prefix') / get_option('datadir') / 'man' / 'man1',
+# )
