--- qe-7.3.1/external/mbd.make.orig	2024-11-06 21:20:18.101960792 +0000
+++ qe-7.3.1/external/mbd.make	2024-11-06 21:20:01.533793054 +0000
@@ -15,9 +15,9 @@
 
 
 mbd.o: mbd_constants.o mbd_damping.o mbd_formulas.o mbd_geom.o mbd_gradients.o mbd_methods.o mbd_ts.o mbd_utils.o mbd_vdw_param.o fallback_version
-mbd_c_api.o: mbd_constants.o mbd_coulomb.o mbd_damping.o mbd_dipole.o mbd_geom.o mbd_gradients.o mbd_matrix.o mbd_methods.o mbd_ts.o mbd_utils.o
+mbd_c_api.o: mbd_constants.o mbd_coulomb.o mbd_damping.o mbd_dipole.o mbd_geom.o mbd_gradients.o mbd_matrix.o mbd_methods.o mbd_ts.o mbd_utils.o f_c_string_correction
 mbd_constants.o: 
-mbd_coulomb.o: mbd_constants.o mbd_damping.o mbd_dipole.o mbd_geom.o mbd_lapack.o mbd_linalg.o mbd_matrix.o
+mbd_coulomb.o: mbd_constants.o mbd_damping.o mbd_dipole.o mbd_geom.o mbd_lapack.o mbd_linalg.o mbd_matrix.o 
 mbd_damping.o: mbd_constants.o mbd_gradients.o mbd_utils.o
 mbd_dipole.o: mbd_constants.o mbd_damping.o mbd_geom.o mbd_gradients.o mbd_lapack.o mbd_linalg.o mbd_matrix.o mbd_utils.o
 mbd_formulas.o: mbd_constants.o mbd_gradients.o mbd_utils.o
@@ -34,6 +34,9 @@
 mbd_utils.o: mbd_constants.o mbd_gradients.o
 mbd_vdw_param.o: mbd_constants.o mbd_utils.o
 
+f_c_string_correction:
+	sed -i 's/f_c_string/f_c_string1/g' mbd_c_api.F90
+	touch f_c_string_correction
 fallback_version:
 	echo '#define MBD_VERSION_MAJOR 0'  > version.h
 	echo '#define MBD_VERSION_MINOR 10'>> version.h
