--- espresso-5.0.2/install/plugins_makefile.orig	2013-01-16 12:16:32.313720000 +0100
+++ espresso-5.0.2/install/plugins_makefile	2013-01-16 12:16:42.534343000 +0100
@@ -289,6 +289,7 @@
 	cd ../YAMBO ; \
 	./configure --with-blas="$(BLAS_LIBS)" \
 	--with-lapack="$(LAPACK_LIBS)" \
+	--with-fftw=$(FFT_LIB_DIR) --with-fftw-lib="$(FFT_LIBS)" \
 	--with-iotk="$(TOPDIR)/iotk" \
 	PFC="$(MPIF90)" \
 	FC="$(LD)" \
