diff -ru espresso-5.0.2.orig/install/plugins_makefile espresso-5.0.2/install/plugins_makefile
--- espresso-5.0.2.orig/install/plugins_makefile	2012-12-01 17:56:20.790511260 +0100
+++ espresso-5.0.2/install/plugins_makefile	2012-12-01 17:58:49.096158108 +0100
@@ -149,8 +149,8 @@
 ##########################################################
 # GIPAW
 ###########################################################
-#gipaw:  uncompress-gipaw configure-gipaw
-gipaw:  download-gipaw configure-gipaw
+gipaw:  uncompress-gipaw configure-gipaw
+#gipaw:  download-gipaw configure-gipaw
 	if test -d ../GIPAW; then \
 	cd ../GIPAW; $(MAKE) all; cd ../bin; ln -fs ../GIPAW/bin/gipaw.x . ; fi
 	touch make-gipaw
