--- espresso-4.2/Makefile.orig	2012-12-04 10:25:10.267891983 +0100
+++ espresso-4.2/Makefile	2012-12-04 10:25:27.356082579 +0100
@@ -165,12 +165,10 @@
 	cd plugins ; $(MAKE) $(MFLAGS) w90
 
 want:
-	if test -e plugins/archive/want-2.3.0.tar.gz ; then \
-	( cd plugins ; $(MAKE) $(MFLAGS) $@) ; fi
+	cd plugins ; $(MAKE) $(MFLAGS) $@
 
 yambo:
-	if test -e plugins/archive/yambo-3.2.1-r.448.tar.gz ; then \
-	( cd plugins ; $(MAKE) $(MFLAGS) $@) ; fi
+	cd plugins ; $(MAKE) $(MFLAGS) $@
 
 #########################################################
 # Links and copies. "make links" is likely obsolete.
diff -ru espresso-4.2.orig/plugins/Makefile espresso-4.2/plugins/Makefile
--- espresso-4.2.orig/plugins/Makefile	2010-04-19 10:49:33.000000000 +0200
+++ espresso-4.2/plugins/Makefile	2013-01-16 16:03:48.586392758 +0100
@@ -88,6 +88,7 @@
 	cd ../$(YAMBO) ; \
 	./configure --with-blas="$(BLAS_LIBS)" \
 	--with-lapack="$(LAPACK_LIBS)" \
+	--with-fftw=$(FFT_LIB_DIR) --with-fftw-lib="$(FFT_LIBS)" \
 	--with-iotk="$(TOPDIR)/iotk" \
 	PFC="$(MPIF90)" \
 	FC="$(LD)" \
