--- qe-6.2.2/install/extlibs_makefile	2019-04-30 19:46:32.248575377 +0000
+++ qe-6.2.2/install/extlibs_makefile	2019-04-30 19:46:48.502502797 +0000
@@ -80,7 +80,7 @@
 	if test ! -d ../FoX; then \
         mkdir ../FoX; \
 	(gzip -dc ../archive/fox.tgz | (cd ../FoX; tar -xvf -)); \
-	cd ../FoX/fox/; export FC=$(F90); ./configure --prefix=$(TOPDIR)/FoX ;\
+	cd ../FoX/fox/; export FC=$(F90); LIBS="" ./configure --prefix=$(TOPDIR)/FoX ;\
 	$(MAKE) install; cd ../; rm -fr fox;fi
 # ELPA
 libelpa : libelpa_$(ELPA_LIBS_SWITCH)
