diff -ru qe-6.2.orig/Modules/environment.f90 qe-6.2/Modules/environment.f90
--- qe-6.2.orig/Modules/environment.f90	2017-12-11 17:53:17.000000000 +0100
+++ qe-6.2/Modules/environment.f90	2018-03-21 21:37:09.033802059 +0100
@@ -135,7 +135,7 @@
   SUBROUTINE environment_end( code )
 
     CHARACTER(LEN=*), INTENT(IN) :: code
-#if defined(_HDF5)
+#if defined(__HDF5)
     CALL finalize_hdf5()
 #endif
     IF ( meta_ionode ) WRITE( stdout, * )
