--- espresso-4.2/Makefile.orig	2012-12-04 10:25:10.267891983 +0100
+++ espresso-4.2/Makefile	2012-12-04 10:25:27.356082579 +0100
@@ -165,12 +165,10 @@
 	cd plugins ; $(MAKE) $(MFLAGS) w90
 
 want:
-	if test -e plugins/archive/want-2.3.0.tar.gz ; then \
-	( cd plugins ; $(MAKE) $(MFLAGS) $@) ; fi
+	cd plugins ; $(MAKE) $(MFLAGS) $@
 
 yambo:
-	if test -e plugins/archive/yambo-3.2.1-r.448.tar.gz ; then \
-	( cd plugins ; $(MAKE) $(MFLAGS) $@) ; fi
+	cd plugins ; $(MAKE) $(MFLAGS) $@
 
 #########################################################
 # Links and copies. "make links" is likely obsolete.
