diff -ru ../yambo-3.2.5.orig/interfaces/p2y/.objects ../yambo-3.2.5/interfaces/p2y/.objects
--- ../yambo-3.2.5.orig/interfaces/p2y/.objects	2012-01-12 11:05:09.000000000 +0100
+++ ../yambo-3.2.5/interfaces/p2y/.objects	2012-12-06 13:41:21.012454313 +0100
@@ -1,30 +1,18 @@
 #if defined _P2Y_EXPORT
 
-objs = pw_errore.o mod_pw_data.o mod_pw_export.o \
-       mod_p2y.o \
-       p2y_db1.o p2y_wf.o p2y_i.o 
+objs = pw_errore.o mod_pw_data.o mod_pw_export.o mod_p2y.o p2y_db1.o p2y_wf.o p2y_i.o 
 
 #elif defined _P2Y_V31 || defined _P2Y_V311
 
-objs = pw_errore.o mod_pw_data.o qexml_v3.1.o  \
-       mod_p2y.o \
-       p2y_db1.o p2y_wf.o p2y_i.o
+objs = pw_errore.o mod_pw_data.o qexml_v3.1.o mod_p2y.o p2y_db1.o p2y_wf.o p2y_i.o
 
 #elif defined _P2Y_V32
 
-objs = pw_errore.o mod_pw_data.o qexml_v3.2.o \
-       mod_p2y.o \
-       p2y_db1.o p2y_wf.o p2y_i.o
+objs = pw_errore.o mod_pw_data.o qexml_v3.2.o mod_p2y.o p2y_db1.o p2y_wf.o p2y_i.o
 
 #elif defined _P2Y_V40
 
-objs = pw_errore.o mod_pw_data.o qexml_v4.0.o \
-       mod_p2y.o numrec_kinds.o mod_numerical.o \
-       numrec_module.o numrec_locate.o numrec_polint.o numrec_polcof.o numrec_ddpoly.o num_interpolation_module.o \
-	   pw_struct_module.o pw_basis_module.o pw_wfc_module.o \
-	   pw_pseudo_module.o pw_atoms_module.o \
-	   pw_pseudovelocity_module.o \
-       Vnl_driver.o p2y_db1.o p2y_wf.o p2y_i.o
+objs = pw_errore.o mod_pw_data.o qexml_v4.0.o mod_p2y.o numrec_kinds.o mod_numerical.o numrec_module.o numrec_locate.o numrec_polint.o numrec_polcof.o numrec_ddpoly.o num_interpolation_module.o 	 pw_struct_module.o pw_basis_module.o pw_wfc_module.o 	 pw_pseudo_module.o pw_atoms_module.o 	 pw_pseudovelocity_module.o Vnl_driver.o p2y_db1.o p2y_wf.o p2y_i.o
 
 #endif
 
diff -ru ../yambo-3.2.5.orig/lib/blas/.objects ../yambo-3.2.5/lib/blas/.objects
--- ../yambo-3.2.5.orig/lib/blas/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/lib/blas/.objects	2012-12-06 13:41:10.164328326 +0100
@@ -1,16 +1,5 @@
 #if defined _DOUBLE
-objs =  zaxpy.o  zdotc.o  zgemm.o  zgeru.o  zswap.o  ztrmv.o  \
-	izamax.o dznrm2.o zcopy.o  zdotu.o  zgemv.o  zscal.o  \
-	ztrmm.o  ztrsm.o  daxpy.o  dscal.o  zher2k.o zher2.o  \
-	zhemv.o  zsyr2k.o zgerc.o  zher.o   dcopy.o  idamax.o \
-	dzasum.o ztrsv.o  xerbla.o zsymm.o  zhemm.o  zsyrk.o  \
-        zherk.o  zdscal.o dcabs1.o ddot.o dgemm.o dger.o dtrsm.o \
-	dswap.o
+objs = zaxpy.o zdotc.o zgemm.o zgeru.o zswap.o ztrmv.o 	izamax.o dznrm2.o zcopy.o zdotu.o zgemv.o zscal.o 	ztrmm.o ztrsm.o daxpy.o dscal.o zher2k.o zher2.o 	zhemv.o zsyr2k.o zgerc.o zher.o dcopy.o idamax.o 	dzasum.o ztrsv.o xerbla.o zsymm.o zhemm.o zsyrk.o zherk.o zdscal.o dcabs1.o ddot.o dgemm.o dger.o dtrsm.o 	dswap.o
 #else
-objs =  caxpy.o  cdotc.o  cgemm.o  cgeru.o  cswap.o  ctrmv.o  \
-	icamax.o scnrm2.o ccopy.o  cdotu.o  cgemv.o  cscal.o  \
-	ctrmm.o  ctrsm.o  saxpy.o  sscal.o  cher2k.o cher2.o  \
-	chemv.o  csyr2k.o cgerc.o  cher.o   scopy.o  isamax.o \
-	scasum.o ctrsv.o  xerbla.o csymm.o  chemm.o  csyrk.o  \
-        cherk.o  csscal.o sdot.o sgemm.o strsm.o sger.o sswap.o
+objs = caxpy.o cdotc.o cgemm.o cgeru.o cswap.o ctrmv.o 	icamax.o scnrm2.o ccopy.o cdotu.o cgemv.o cscal.o 	ctrmm.o ctrsm.o saxpy.o sscal.o cher2k.o cher2.o 	chemv.o csyr2k.o cgerc.o cher.o scopy.o isamax.o 	scasum.o ctrsv.o xerbla.o csymm.o chemm.o csyrk.o cherk.o csscal.o sdot.o sgemm.o strsm.o sger.o sswap.o
 #endif
diff -ru ../yambo-3.2.5.orig/lib/lapack/.objects ../yambo-3.2.5/lib/lapack/.objects
--- ../yambo-3.2.5.orig/lib/lapack/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/lib/lapack/.objects	2012-12-06 13:41:12.356353744 +0100
@@ -1,53 +1,5 @@
 #if defined _DOUBLE
-objs = 	zgetf2.o  zhetrd.o  zlarft.o  zlaswp.o  \
-	zung2l.o  ilaenv.o  dlanst.o  dlassq.o  \
-	zgetrf.o  zlacgv.o  zlascl.o  zlatrd.o  \
-	zungql.o  lsame.o   dlapy2.o  dsterf.o  \
-	zgetri.o  zlanhe.o  zlaset.o  zsteqr.o  \
-	zungqr.o  dlae2.o   dlartg.o  zheev.o   \
-    zheevr.o \
-	zlarfb.o  zlasr.o   ztrti2.o  zungtr.o  \
-	dlaev2.o  dlascl.o  zhetd2.o  zlarfg.o  \
-	zlassq.o  ztrtri.o  ieeeck.o \
-	dlasrt.o  zhetri.o  zhetrf.o \
-	zladiv.o  zlarf.o   zung2r.o  dlapy3.o  \
-	dladiv.o  zlahef.o  zhetf2.o \
-	zgesvd.o  zlange.o  zgeqrf.o  zgebrd.o  \
-	zungbr.o  zbdsqr.o  zlacpy.o  zunmbr.o  \
-	zgelqf.o  zunglq.o  zgeqr2.o  zgebd2.o  \
-	dlasq1.o  dlasv2.o  zdrot.o   dlas2.o  \
-	zlabrd.o  zunmqr.o  zunmlq.o  zungl2.o \
-	zgelq2.o  dlasq2.o  zunm2r.o  zunml2.o \
-	dlasq3.o  dlasq4.o  dlasq5.o  dlasq6.o \
-	zgebal.o  zgebak.o  ztrevc.o  zhseqr.o \
-	zunghr.o  zgehrd.o  dlabad.o \
-	zgeev.o   zgehd2.o  zlahqr.o \
-	zlahrd.o  zlanhs.o  zlarfx.o  zlatrs.o dlamch.o dlaran.o \
-	dlacpy.o  dgetrf.o  dgetrs.o  dgetf2.o dlaswp.o
+objs = 	zgetf2.o zhetrd.o zlarft.o zlaswp.o 	zung2l.o ilaenv.o dlanst.o dlassq.o 	zgetrf.o zlacgv.o zlascl.o zlatrd.o 	zungql.o lsame.o dlapy2.o dsterf.o 	zgetri.o zlanhe.o zlaset.o zsteqr.o 	zungqr.o dlae2.o dlartg.o zheev.o zheevr.o 	zlarfb.o zlasr.o ztrti2.o zungtr.o 	dlaev2.o dlascl.o zhetd2.o zlarfg.o 	zlassq.o ztrtri.o ieeeck.o 	dlasrt.o zhetri.o zhetrf.o 	zladiv.o zlarf.o zung2r.o dlapy3.o 	dladiv.o zlahef.o zhetf2.o 	zgesvd.o zlange.o zgeqrf.o zgebrd.o 	zungbr.o zbdsqr.o zlacpy.o zunmbr.o 	zgelqf.o zunglq.o zgeqr2.o zgebd2.o 	dlasq1.o dlasv2.o zdrot.o dlas2.o 	zlabrd.o zunmqr.o zunmlq.o zungl2.o 	zgelq2.o dlasq2.o zunm2r.o zunml2.o 	dlasq3.o dlasq4.o dlasq5.o dlasq6.o 	zgebal.o zgebak.o ztrevc.o zhseqr.o 	zunghr.o zgehrd.o dlabad.o 	zgeev.o zgehd2.o zlahqr.o 	zlahrd.o zlanhs.o zlarfx.o zlatrs.o dlamch.o dlaran.o 	dlacpy.o dgetrf.o dgetrs.o dgetf2.o dlaswp.o
 #else
-objs = 	cgetf2.o  chetrd.o  clarft.o  claswp.o  \
-	cung2l.o  ilaenv.o  slanst.o  slassq.o  \
-	cgetrf.o  clacgv.o  clascl.o  clatrd.o  \
-	cungql.o  lsame.o   slapy2.o  ssterf.o  \
-	cgetri.o  clanhe.o  claset.o  csteqr.o  \
-	cungqr.o  slae2.o   slartg.o  cheev.o   \
-    cheevr.o \
-	clarfb.o  clasr.o   ctrti2.o  cungtr.o  \
-	slaev2.o  slascl.o  chetd2.o  clarfg.o  \
-	classq.o  ctrtri.o  ieeeck.o \
-	slasrt.o  chetri.o  chetrf.o \
-	cladiv.o  clarf.o   cung2r.o  slapy3.o  \
-	sladiv.o  clahef.o  chetf2.o \
-	cgesvd.o  clange.o  cgeqrf.o  cgebrd.o  \
-	cungbr.o  cbdsqr.o  clacpy.o  cunmbr.o  \
-	cgelqf.o  cunglq.o  cgeqr2.o  cgebd2.o  \
-	slasq1.o  slasv2.o  csrot.o   slas2.o  \
-	clabrd.o  cunmqr.o  cunmlq.o  cungl2.o \
-	cgelq2.o  slasq2.o  cunm2r.o  cunml2.o \
-	slasq3.o  slasq4.o  slasq5.o  slasq6.o \
-	cgebal.o  cgebak.o  ctrevc.o  chseqr.o \
-	cunghr.o  cgehrd.o  slabad.o \
-	cgeev.o   cgehd2.o  clahqr.o \
-	clahrd.o  clanhs.o  clarfx.o  clatrs.o slamch.o dlaran.o \
-	slacpy.o  sgetrf.o sgetrs.o sgetf2.o slaswp.o
+objs = 	cgetf2.o chetrd.o clarft.o claswp.o 	cung2l.o ilaenv.o slanst.o slassq.o 	cgetrf.o clacgv.o clascl.o clatrd.o 	cungql.o lsame.o slapy2.o ssterf.o 	cgetri.o clanhe.o claset.o csteqr.o 	cungqr.o slae2.o slartg.o cheev.o cheevr.o 	clarfb.o clasr.o ctrti2.o cungtr.o 	slaev2.o slascl.o chetd2.o clarfg.o 	classq.o ctrtri.o ieeeck.o 	slasrt.o chetri.o chetrf.o 	cladiv.o clarf.o cung2r.o slapy3.o 	sladiv.o clahef.o chetf2.o 	cgesvd.o clange.o cgeqrf.o cgebrd.o 	cungbr.o cbdsqr.o clacpy.o cunmbr.o 	cgelqf.o cunglq.o cgeqr2.o cgebd2.o 	slasq1.o slasv2.o csrot.o slas2.o 	clabrd.o cunmqr.o cunmlq.o cungl2.o 	cgelq2.o slasq2.o cunm2r.o cunml2.o 	slasq3.o slasq4.o slasq5.o slasq6.o 	cgebal.o cgebak.o ctrevc.o chseqr.o 	cunghr.o cgehrd.o slabad.o 	cgeev.o cgehd2.o clahqr.o 	clahrd.o clanhs.o clarfx.o clatrs.o slamch.o dlaran.o 	slacpy.o sgetrf.o sgetrs.o sgetf2.o slaswp.o
 #endif
diff -ru ../yambo-3.2.5.orig/lib/slatec/.objects ../yambo-3.2.5/lib/slatec/.objects
--- ../yambo-3.2.5.orig/lib/slatec/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/lib/slatec/.objects	2012-12-06 13:41:15.829394100 +0100
@@ -1,21 +1,5 @@
 #if defined _DOUBLE
-objs = 	davint.o  xermsg.o  fdump.o   j4save.o  xercnt.o  xerhlt.o \
-	    xerprn.o  xersve.o  i1mach.o  xgetua.o \
-        dpolft.o  dpcoef.o  dp1vlu.o \
-        dbesj0.o  dbesj1.o  dbesk0.o  dbesk1.o  dcsevl.o  initds.o  \
-        d1mach.o  dbesi0.o  dbsk0e.o  dbesi1.o  dbsk1e.o  dbsi0e.o \
-        dbsi1e.o  dqage.o   dqag.o    dqk15.o   dqk21.o   dqk31.o   \
-        dqk41.o   dqk51.o   dqk61.o   dqpsrt.o  dqawo.o   dqawoe.o \
-        dqc25f.o  dqelg.o   dqcheb.o  dqwgtf.o  dqk15w.o  dqwgtf.o \
-        dgtsl.o   d9b0mp.o  d9b1mp.o
+objs = 	davint.o xermsg.o fdump.o j4save.o xercnt.o xerhlt.o 	 xerprn.o xersve.o i1mach.o xgetua.o dpolft.o dpcoef.o dp1vlu.o dbesj0.o dbesj1.o dbesk0.o dbesk1.o dcsevl.o initds.o d1mach.o dbesi0.o dbsk0e.o dbesi1.o dbsk1e.o dbsi0e.o dbsi1e.o dqage.o dqag.o dqk15.o dqk21.o dqk31.o dqk41.o dqk51.o dqk61.o dqpsrt.o dqawo.o dqawoe.o dqc25f.o dqelg.o dqcheb.o dqwgtf.o dqk15w.o dqwgtf.o dgtsl.o d9b0mp.o d9b1mp.o
 #else
-objs = 	avint.o  xermsg.o fdump.o  j4save.o xercnt.o xerhlt.o \
-	    xerprn.o xersve.o i1mach.o xgetua.o \
-        polfit.o  pcoef.o  pvalue.o \
-        besj0.o  besj1.o  besk0.o  besk1.o  csevl.o  inits.o  \
-        r1mach.o besi0.o  besk0e.o besi1.o  besk1e.o besi0e.o \
-        besi1e.o qage.o   qag.o    qk15.o   qk21.o   qk31.o   \
-        qk41.o   qk51.o   qk61.o   qpsrt.o  qawo.o  qawoe.o \
-        qc25f.o  qelg.o   qcheb.o  qwgtf.o  qk15w.o  qwgtf.o \
-        sgtsl.o       
+objs = 	avint.o xermsg.o fdump.o j4save.o xercnt.o xerhlt.o 	 xerprn.o xersve.o i1mach.o xgetua.o polfit.o pcoef.o pvalue.o besj0.o besj1.o besk0.o besk1.o csevl.o inits.o r1mach.o besi0.o besk0e.o besi1.o besk1e.o besi0e.o besi1e.o qage.o qag.o qk15.o qk21.o qk31.o qk41.o qk51.o qk61.o qpsrt.o qawo.o qawoe.o qc25f.o qelg.o qcheb.o qwgtf.o qk15w.o qwgtf.o sgtsl.o 
 #endif
diff -ru ../yambo-3.2.5.orig/src/bse/.objects ../yambo-3.2.5/src/bse/.objects
--- ../yambo-3.2.5.orig/src/bse/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/bse/.objects	2012-12-06 13:41:00.052210850 +0100
@@ -1,5 +1,2 @@
 DISTRIBUTED_objects = K.o
-objs =	K_driver.o K_eh_setup.o \
-        $(DISTRIBUTED_objects) K_scattering.o K_exchange.o \
-        K_solvers.o K_Haydock.o K_diagonalization.o K_eps_interpolate.o \
-        K_output_file.o K_BSmat_by_V.o K_filling.o K_inversion.o
+objs =	K_driver.o K_eh_setup.o $(DISTRIBUTED_objects) K_scattering.o K_exchange.o K_solvers.o K_Haydock.o K_diagonalization.o K_eps_interpolate.o K_output_file.o K_BSmat_by_V.o K_filling.o K_inversion.o
diff -ru ../yambo-3.2.5.orig/src/bz_ops/.objects ../yambo-3.2.5/src/bz_ops/.objects
--- ../yambo-3.2.5.orig/src/bz_ops/.objects	2012-01-12 11:05:07.000000000 +0100
+++ ../yambo-3.2.5/src/bz_ops/.objects	2012-12-06 13:41:05.892278688 +0100
@@ -1,2 +1 @@
-objs = k_lattice.o k_expand.o k_ibz2bz.o k_reduce.o k_sym2sym.o \
-       bz_samp_indexes.o  k_the_nearest.o
+objs = k_lattice.o k_expand.o k_ibz2bz.o k_reduce.o k_sym2sym.o bz_samp_indexes.o k_the_nearest.o
diff -ru ../yambo-3.2.5.orig/src/common/.objects ../yambo-3.2.5/src/common/.objects
--- ../yambo-3.2.5.orig/src/common/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/common/.objects	2012-12-06 13:40:56.661171495 +0100
@@ -1,6 +1 @@
-objs = eval_G_minus_G.o \
-       FREQUENCIES_damping.o FREQUENCIES_setup.o FREQUENCIES_coarse_grid.o \
-       FREQUENCIES_Green_Function.o FREQUENCIES_reset.o \
-       crystal_lattice.o Kramers_Kronig.o RIntegrate.o CIntegrate.o QPartilize.o \
-       OCCUPATIONS_Fermi.o OCCUPATIONS_Extend.o \
-       QP_state_extract.o pol_fit.o pert_inversion.o
+objs = eval_G_minus_G.o FREQUENCIES_damping.o FREQUENCIES_setup.o FREQUENCIES_coarse_grid.o FREQUENCIES_Green_Function.o FREQUENCIES_reset.o crystal_lattice.o Kramers_Kronig.o RIntegrate.o CIntegrate.o QPartilize.o OCCUPATIONS_Fermi.o OCCUPATIONS_Extend.o QP_state_extract.o pol_fit.o pert_inversion.o
diff -ru ../yambo-3.2.5.orig/src/coulomb/.objects ../yambo-3.2.5/src/coulomb/.objects
--- ../yambo-3.2.5.orig/src/coulomb/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/coulomb/.objects	2012-12-06 13:40:58.132188564 +0100
@@ -1,4 +1 @@
-objs = col_driver.o rim.o rim_integrate.o rim_spherical.o \
-       bessel_F2.o bessel_F3.o bessel_F4.o bessel_F5.o bessel_F6.o bessel_J0.o \
-       cutoff_driver.o cutoff_sphere.o cutoff_box.o \
-       cutoff_test.o cutoff_cylinder.o
+objs = col_driver.o rim.o rim_integrate.o rim_spherical.o bessel_F2.o bessel_F3.o bessel_F4.o bessel_F5.o bessel_F6.o bessel_J0.o cutoff_driver.o cutoff_sphere.o cutoff_box.o cutoff_test.o cutoff_cylinder.o
diff -ru ../yambo-3.2.5.orig/src/interface/.objects ../yambo-3.2.5/src/interface/.objects
--- ../yambo-3.2.5.orig/src/interface/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/interface/.objects	2012-12-06 13:40:45.652043841 +0100
@@ -1,3 +1,2 @@
-objs = init_load.o init.o QP_init.o parse_K_list.o \
-       init_q_pts.o  stop_now.o barriers.o QP_ctl_load.o QP_ctl_switch.o
+objs = init_load.o init.o QP_init.o parse_K_list.o init_q_pts.o stop_now.o barriers.o QP_ctl_load.o QP_ctl_switch.o
 
diff -ru ../yambo-3.2.5.orig/src/io/.objects ../yambo-3.2.5/src/io/.objects
--- ../yambo-3.2.5.orig/src/io/.objects	2012-01-12 11:05:07.000000000 +0100
+++ ../yambo-3.2.5/src/io/.objects	2012-12-06 13:40:48.612078185 +0100
@@ -1,13 +1,4 @@
 SC_objects = 
 ELPH_objects = 
 DISTRIBUTED_objects = ioBS.o
-objs = ver_is_gt_or_eq.o IO_and_Messaging_switch.o \
-       io_Vnl.o \
-       ioRIM.o        ioGROT.o     $(ELPH_objects) \
-       io_DIPOLES.o   ioQINDX.o    ioQP.o      ioX.o \
-       ioDB1.o        ioWF.o       ioKB_PP.o   io_HF_and_locXC.o \
-       $(DISTRIBUTED_objects)         \
-       ioBSS_diago.o  ioBSS_Haydock.o  ioBSS_invert.o \
-       io_elemental.o io_bulk.o      io_header.o ioE_RIM.o \
-       ioCOL_CUT.o    ioDB1_reload.o variables_BS.o variables_X.o variables_DIPOLES.o $(SC_objects)  \
-       IO_Setup.o Fragments_Restart.o Fragments_Synchronize.o
+objs = ver_is_gt_or_eq.o IO_and_Messaging_switch.o io_Vnl.o ioRIM.o ioGROT.o $(ELPH_objects) io_DIPOLES.o ioQINDX.o ioQP.o ioX.o ioDB1.o ioWF.o ioKB_PP.o io_HF_and_locXC.o $(DISTRIBUTED_objects) ioBSS_diago.o ioBSS_Haydock.o ioBSS_invert.o io_elemental.o io_bulk.o io_header.o ioE_RIM.o ioCOL_CUT.o ioDB1_reload.o variables_BS.o variables_X.o variables_DIPOLES.o $(SC_objects) IO_Setup.o Fragments_Restart.o Fragments_Synchronize.o
diff -ru ../yambo-3.2.5.orig/src/modules/.objects ../yambo-3.2.5/src/modules/.objects
--- ../yambo-3.2.5.orig/src/modules/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/modules/.objects	2012-12-06 13:41:04.404261397 +0100
@@ -1,6 +1,6 @@
 #if defined _SC
-SC_objects       = mod_SC.o
-RT_objects       = mod_collisions_IO.o mod_collisions_CACHE.o
+SC_objects = mod_SC.o
+RT_objects = mod_collisions_IO.o mod_collisions_CACHE.o
 #endif
 #if defined _ELPH 
 ELPH_objects = mod_ELPH.o
@@ -11,15 +11,4 @@
 #if defined _LIBXC
 LIBXC_objects = mod_libxc_int.o
 #endif
-objs = mod_pars.o mod_units.o  mod_stderr.o  mod_par_proc.o \
-       mod_wrapper.o mod_fields.o mod_collision.o mod_drivers.o  mod_FFT.o \
-       mod_timing.o  mod_logo.o \
-       mod_com.o mod_memory.o   mod_R_lattice.o  \
-       mod_par_indexes.o mod_wave_func.o \
-       mod_matrix_operate.o mod_D_lattice.o mod_frequency.o\
-       mod_wave_func.o mod_vec_operate.o mod_electrons.o mod_X.o \
-       mod_functions.o mod_zeros.o mod_pseudo.o \
-       mod_BS.o mod_QP.o mod_TDDFT.o mod_ACFDT.o $(DISTRIBUTED_objects)\
-       mod_IO.o mod_fragments.o $(ELPH_objects) mod_xc_functionals.o mod_global_XC.o \
-       $(SC_objects) $(RT_objects) $(MAGNETIC_objects) mod_X_output.o std_presets.o \
-       $(LIBXC_objects)
+objs = mod_pars.o mod_units.o mod_stderr.o mod_par_proc.o mod_wrapper.o mod_fields.o mod_collision.o mod_drivers.o mod_FFT.o mod_timing.o mod_logo.o mod_com.o mod_memory.o mod_R_lattice.o mod_par_indexes.o mod_wave_func.o mod_matrix_operate.o mod_D_lattice.o mod_frequency.o mod_wave_func.o mod_vec_operate.o mod_electrons.o mod_X.o mod_functions.o mod_zeros.o mod_pseudo.o mod_BS.o mod_QP.o mod_TDDFT.o mod_ACFDT.o $(DISTRIBUTED_objects) mod_IO.o mod_fragments.o $(ELPH_objects) mod_xc_functionals.o mod_global_XC.o $(SC_objects) $(RT_objects) $(MAGNETIC_objects) mod_X_output.o std_presets.o $(LIBXC_objects)
diff -ru ../yambo-3.2.5.orig/src/parser/.objects ../yambo-3.2.5/src/parser/.objects
--- ../yambo-3.2.5.orig/src/parser/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/parser/.objects	2012-12-06 13:41:01.541228133 +0100
@@ -1,3 +1 @@
-objs =   math.o interface.o parser.o parser_exp.o symbols.o \
-         parser_lib.o mod_parser_m.o mod_itm.o Gclose.o \
-         init_convert.o ps_convert.o
+objs = math.o interface.o parser.o parser_exp.o symbols.o parser_lib.o mod_parser_m.o mod_itm.o Gclose.o init_convert.o ps_convert.o
diff -ru ../yambo-3.2.5.orig/src/pol_function/.objects ../yambo-3.2.5/src/pol_function/.objects
--- ../yambo-3.2.5.orig/src/pol_function/.objects	2012-01-12 11:05:07.000000000 +0100
+++ ../yambo-3.2.5/src/pol_function/.objects	2012-12-06 13:40:39.411971570 +0100
@@ -1,7 +1,2 @@
 RAS_objects = 
-objs = O_driver.o O_eels.o \
-       Dipole_driver.o Dipole_dip_dot_field.o \
-       Dipole_kb_pp_comp.o Dipole_kb_sum.o Dipole_kb_Ylm.o \
-       Dipole_transverse.o  \
-       X_drude.o  X_pre_setup.o X_simple_GreenF.o X_eh_setup.o X_em1.o X_os.o X_s.o \
-       X_delta_part.o X_bare_RIM_GreenF.o $(RAS_objects) O_select_q_and_G.o 
+objs = O_driver.o O_eels.o Dipole_driver.o Dipole_dip_dot_field.o Dipole_kb_pp_comp.o Dipole_kb_sum.o Dipole_kb_Ylm.o Dipole_transverse.o X_drude.o X_pre_setup.o X_simple_GreenF.o X_eh_setup.o X_em1.o X_os.o X_s.o X_delta_part.o X_bare_RIM_GreenF.o $(RAS_objects) O_select_q_and_G.o 
diff -ru ../yambo-3.2.5.orig/src/qp/.objects ../yambo-3.2.5/src/qp/.objects
--- ../yambo-3.2.5.orig/src/qp/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/qp/.objects	2012-12-06 13:40:47.156061286 +0100
@@ -1,7 +1,2 @@
 ELPH_objects =
-objs = QP_driver.o QP_newton.o QP_ppa_cohsex.o \
-       QP_secant.o QP_real_axis.o QP_W2Sc.o QP_life_transitions.o secant.o bracket.o \
-       $(ELPH_objects) \
-       QP_descriptions.o QP_report_and_write.o  \
-       QP_Green_Function.o QP_of.o \
-       XCo_driver.o XCo_Hartree_Fock.o XCo_local.o $(SC_objects)
+objs = QP_driver.o QP_newton.o QP_ppa_cohsex.o QP_secant.o QP_real_axis.o QP_W2Sc.o QP_life_transitions.o secant.o bracket.o $(ELPH_objects) QP_descriptions.o QP_report_and_write.o QP_Green_Function.o QP_of.o XCo_driver.o XCo_Hartree_Fock.o XCo_local.o $(SC_objects)
diff -ru ../yambo-3.2.5.orig/src/qp_ctl/.objects ../yambo-3.2.5/src/qp_ctl/.objects
--- ../yambo-3.2.5.orig/src/qp_ctl/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/qp_ctl/.objects	2012-12-06 13:40:53.604136035 +0100
@@ -1,2 +1 @@
-objs= mod_QP_CTL.o QP_load_DB.o QP_check_if_corrected.o QP_fit_DB_values.o \
-      QP_apply_DB_interpolation.o QP_apply_global_stretch.o QP_merge_DBs.o
+objs= mod_QP_CTL.o QP_load_DB.o QP_check_if_corrected.o QP_fit_DB_values.o QP_apply_DB_interpolation.o QP_apply_global_stretch.o QP_merge_DBs.o
diff -ru ../yambo-3.2.5.orig/src/setup/.objects ../yambo-3.2.5/src/setup/.objects
--- ../yambo-3.2.5.orig/src/setup/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/setup/.objects	2012-12-06 13:41:02.980244863 +0100
@@ -1,3 +1 @@
-objs = setup.o  check_periodic_directions.o  \
-       G_shells_finder.o  eval_minus_G.o QP_state_table_setup.o  \
-       QP_external_corrections.o 
+objs = setup.o check_periodic_directions.o G_shells_finder.o eval_minus_G.o QP_state_table_setup.o QP_external_corrections.o 
diff -ru ../yambo-3.2.5.orig/src/tddft/.objects ../yambo-3.2.5/src/tddft/.objects
--- ../yambo-3.2.5.orig/src/tddft/.objects	2012-01-12 11:05:07.000000000 +0100
+++ ../yambo-3.2.5/src/tddft/.objects	2012-12-06 13:40:44.100025864 +0100
@@ -1,2 +1 @@
-objs =	tddft_do_X_W_typs.o \
-        tddft_alda_g_space.o tddft_alda_r_space.o
+objs =	tddft_do_X_W_typs.o tddft_alda_g_space.o tddft_alda_r_space.o
diff -ru ../yambo-3.2.5.orig/src/wf_and_fft/.objects ../yambo-3.2.5/src/wf_and_fft/.objects
--- ../yambo-3.2.5.orig/src/wf_and_fft/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/wf_and_fft/.objects	2012-12-06 13:40:55.156154055 +0100
@@ -4,6 +4,4 @@
 FFT_object = sgfft.o
 #endif
 WF_rot_object =
-objs = fft_setup.o fft_3d.o $(FFT_object)  \
-       scatter_Bamp.o scatter_Gamp.o WF_spatial_invertion.o WF_load.o \
-       $(WF_rot_object) $(DISTRIBUTED_objects)
+objs = fft_setup.o fft_3d.o $(FFT_object) scatter_Bamp.o scatter_Gamp.o WF_spatial_invertion.o WF_load.o $(WF_rot_object) $(DISTRIBUTED_objects)
diff -ru ../yambo-3.2.5.orig/src/xc_functionals/.objects ../yambo-3.2.5/src/xc_functionals/.objects
--- ../yambo-3.2.5.orig/src/xc_functionals/.objects	2012-01-12 11:05:08.000000000 +0100
+++ ../yambo-3.2.5/src/xc_functionals/.objects	2012-12-06 13:40:50.020094530 +0100
@@ -3,9 +3,6 @@
 BOLTZMANN_objects = el_current.o
 #endif
 #if defined _LIBXC
-LIBXC_objects =  XC_libxc_driver.o
+LIBXC_objects = XC_libxc_driver.o
 #endif
-objs =  mod_xc_costants.o  xc_rpa_kp.o \
-        xcspol.o xcpzca.o xcwign.o xchelu.o xcxalp.o \
-        spin_density.o el_density.o XC_potential_driver.o\
-        $(LIBXC_objects) $(BOLTZMANN_objects)
+objs = mod_xc_costants.o xc_rpa_kp.o xcspol.o xcpzca.o xcwign.o xchelu.o xcxalp.o spin_density.o el_density.o XC_potential_driver.o $(LIBXC_objects) $(BOLTZMANN_objects)
diff -ru ../yambo-3.2.5.orig/ypp/.objects ../yambo-3.2.5/ypp/.objects
--- ../yambo-3.2.5.orig/ypp/.objects	2012-01-12 11:05:07.000000000 +0100
+++ ../yambo-3.2.5/ypp/.objects	2012-12-06 13:41:08.756311947 +0100
@@ -1,9 +1,4 @@
 #if defined _SPIN
 SPIN_objects = excitons_collinear.o excitons_non_collinear.o
 #endif
-objs = mod_YPP.o     ypp_i.o   ypp_init.o ypp_init_load.o \
-       k_grids.o  qpdb_init.o qpdb_setup.o \
-       excitons_driver.o electrons_driver.o electrons_charge.o electrons_dos_and_charge.o \
-       plot_check_and_launch.o plot_gnuplot.o plot_xcrysden.o plot_cube.o \
-       excitons_sort_and_report.o $(SPIN_objects)   k_random.o \
-       $(ELPH_objects) $(RAS_objects) $(RT_objects)  $(magnetic_objects) $(boltz_objects)
+objs = mod_YPP.o ypp_i.o ypp_init.o ypp_init_load.o k_grids.o qpdb_init.o qpdb_setup.o excitons_driver.o electrons_driver.o electrons_charge.o electrons_dos_and_charge.o plot_check_and_launch.o plot_gnuplot.o plot_xcrysden.o plot_cube.o excitons_sort_and_report.o $(SPIN_objects) k_random.o $(ELPH_objects) $(RAS_objects) $(RT_objects) $(magnetic_objects) $(boltz_objects)
