--- yambo-3.2.1-r.448/configure.orig	2012-12-05 14:15:18.450791872 +0100
+++ yambo-3.2.1-r.448/configure	2012-12-05 14:17:59.435653191 +0100
@@ -11683,7 +11683,7 @@
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
   HAVE_FFTW="yes"; FFT_CPP="-D_FFTW";
- FFT_DESCRIPTION="FFTW Fast Fourier transform";LDFLAGS="$FFTW_PATH"
+ FFT_DESCRIPTION="FFTW Fast Fourier transform";LDFLAGS="${LDFLAGS} $FFTW_PATH"
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
