--- yambo-3.2.5/sbin/make_makefile.sh.in.orig	2012-01-12 11:05:08.000000000 +0100
+++ yambo-3.2.5/sbin/make_makefile.sh.in	2012-12-06 11:32:14.121395203 +0100
@@ -106,8 +106,10 @@
 echo "dmpi=$dmpi" >> $cdir/Makefile
 echo "dopts=$dopts -D_\$(os)" >> $cdir/Makefile
 echo "lock_file=$lock_file" >> $cdir/Makefile
+echo >> $cdir/Makefile
 cp $cdir/$ofile $cdir/$ofile.c
-$cpp $cppflags $dopts -D_$os $cdir/$ofile.c >> $cdir/Makefile
+echo "$cpp $cppflags $dopts -D_$os $cdir/$ofile.c >> $cdir/Makefile"
+$cpp $cppflags $dopts -D_$os $cdir/$ofile.c | sed ':a;N;$!ba;s/\n/\\\n/g' | sed 's/^\(objs[ ]*=[ ]*\)/\n\1/g' >> $cdir/Makefile
 rm -f $cdir/$ofile.c
 
 if [ "$mode" = "x" ] ; then 
