--- yambo-3.2.1-r.448/config/setup.in.orig	2012-12-05 09:31:38.290005555 +0100
+++ yambo-3.2.1-r.448/config/setup.in	2012-12-05 09:32:03.264302989 +0100
@@ -35,7 +35,7 @@
 exec_prefix = @exec_prefix@/bin
 srcdir      = @srcdir_path@
 libdir      = @srcdir_path@/lib
-libpath     = -L@srcdir_path@/lib @LDFLAGS@
+libpath     = @LDFLAGS@ -L@srcdir_path@/lib
 includedir  = @srcdir_path@/include
 lblas       = @BLAS_LIBS@
 llapack     = @LAPACK_LIBS@
