--- yambo-3.2.1-r.448/sbin/make_makefile.sh.in.orig	2012-12-05 08:51:44.516808759 +0100
+++ yambo-3.2.1-r.448/sbin/make_makefile.sh.in	2012-12-05 08:55:17.956341890 +0100
@@ -102,7 +102,7 @@
 echo "dopts=$dopts -D_\$(os)" >> $cdir/Makefile
 echo "lock_file=$lock_file" >> $cdir/Makefile
 cp $cdir/$ofile $cdir/$ofile.c
-$cpp $cppflags $dopts -D_$os $cdir/$ofile.c >> $cdir/Makefile
+$cpp $cppflags $dopts -D_$os $cdir/$ofile.c | sed ':a;N;$!ba;s/\n/\\\n/g' >> $cdir/Makefile
 rm -f $cdir/$ofile.c
 
 if [ "$mode" = "x" ] ; then 
