Requirement on pybind11 is unncessarily strict.
Author: micketeer@gmail.com
--- setup.py.orig	2021-08-31 11:25:48.110514794 +0200
+++ setup.py	2021-08-31 11:25:53.937501309 +0200
@@ -325,7 +325,7 @@
           ext_package='qd.cae',  # where to place c extensions
           ext_modules=[dyna_extension],
           install_requires=['numpy>=1.8',
-                            'diversipy', 'pybind11==2.1.1', 'h5py'],
+                            'diversipy', 'pybind11', 'h5py'],
           keywords=['cae',
                       'simulation',
                       'engineering',
