--- noarch/launch.sh	2012-02-15 14:59:33.000000000 +0100
+++ noarch/launch.sh	2012-07-04 12:12:01.599792424 +0200
@@ -50,7 +50,8 @@
 export QTDIR QT_PLUGIN_PATH
 
 ## Edit next line if you need to move the license file
-LM_LICENSE_FILE="${dirname}/license/ecore_floating.lic"
+LM_LICENSE_FILE2="${dirname}/license/ecore_floating.lic"
+LM_LICENSE_FILE=${LM_LICENSE_FILE-$LM_LICENSE_FILE2} #allow users to overwrite license file path by setting LM_LICENSE_FILE
 export LM_LICENSE_FILE
 
 exec "${dirname}/${archstr}/bin/${exename}" "$@"

