diff -Naur flashpca.orig/Makefile flashpca/Makefile
--- flashpca.orig/Makefile	2020-09-24 16:59:00.600934431 +0000
+++ flashpca/Makefile	2020-09-24 17:23:34.687119165 +0000
@@ -26,7 +26,7 @@
 	CXXFLAGS += -msse2 -DEIGEN_DONT_PARALLELIZE -std=c++11
 	BOOST = ${BOOST_LIB}/libboost_program_options.a
 else
-	CXXFLAGS += -march=native -std=c++0x
+	CXXFLAGS += -std=c++0x
 	BOOST = -L${BOOST_LIB} -lboost_program_options
 endif
 
