Adapt to newer numpy/matplotlib

Åke Sandgren, 2026-01-29
diff --git a/smcpp/__init__.py b/smcpp/__init__.py
index b1cc7ac..af7dbaa 100644
--- a/smcpp/__init__.py
+++ b/smcpp/__init__.py
@@ -2,7 +2,7 @@ import smcpp.ad as ad
 import numbers
 import numpy as np
 import warnings
-warnings.filterwarnings("error", category=np.VisibleDeprecationWarning)
+warnings.filterwarnings("error", category=np.exceptions.VisibleDeprecationWarning)
 
 
 np.seterr(divide='raise', invalid='raise', over='raise')
diff --git a/smcpp/commands/posterior.py b/smcpp/commands/posterior.py
index 13d22c4..bf5d4d6 100644
--- a/smcpp/commands/posterior.py
+++ b/smcpp/commands/posterior.py
@@ -1,6 +1,6 @@
 import matplotlib
 import matplotlib.style
-matplotlib.style.use('seaborn-ticks')
+matplotlib.style.use('seaborn-v0_8-ticks')
 matplotlib.use('Agg')
 import matplotlib.pyplot as plt
 from matplotlib.image import NonUniformImage
diff --git a/smcpp/plotting.py b/smcpp/plotting.py
index 7450541..52da095 100644
--- a/smcpp/plotting.py
+++ b/smcpp/plotting.py
@@ -3,7 +3,7 @@ import json
 import matplotlib, matplotlib.style, matplotlib.cm
 
 matplotlib.use("Agg")
-matplotlib.style.use("seaborn-ticks")
+matplotlib.style.use("seaborn-v0_8-ticks")
 import numpy as np
 from numpy import array
 from collections import defaultdict
