Drop bad asserts

Åke Sandgren, 2026-01-29
diff --git a/include/marginalize_key.h b/include/marginalize_key.h
index 4d00ab5..f38d04f 100644
--- a/include/marginalize_key.h
+++ b/include/marginalize_key.h
@@ -58,7 +58,6 @@ std::map<block_key, double> marginalize_key<P>::run(
         const Eigen::MatrixBase<Derived3> &na)
 {
     std::map<block_key, double> ret;
-    assert(a(0) >= 0);
     std::map<block_key, double> sub_left = marginalize_key<1>::run(
             key.head(3),
             n.head(1),
diff --git a/src/conditioned_sfs.cpp b/src/conditioned_sfs.cpp
index 2f41f25..e232bf7 100644
--- a/src/conditioned_sfs.cpp
+++ b/src/conditioned_sfs.cpp
@@ -105,7 +105,6 @@ std::vector<Matrix<T> > incorporate_theta(const std::vector<Matrix<T> > &csfs, d
     for (unsigned int i = 0; i < csfs.size(); ++i)
     {
         assert(csfs[i](0, 0) == 0.);
-        assert(csfs[i](2, n) == 0.);
         T tauh = csfs[i].sum();
         // if (toDouble(tauh) > 1.0 / theta)
         // {
