From: Stefano Rivera <stefanor@debian.org>
Date: Sun, 21 Nov 2021 08:50:34 -0400
Subject: Drop use_2to3, setuptools 58 no longer supports it

Bug-Upstream: https://github.com/DEAP/deap/issues/595
Bug-Debian: https://bugs.debian.org/997609
Forwarded: not-needed
---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index e68c052..01ad00b 100644
--- a/setup.py
+++ b/setup.py
@@ -87,7 +87,6 @@ def run_setup(build_ext):
          ext_modules=extra_modules,
          cmdclass={"build_ext": ve_build_ext},
          install_requires=['numpy'],
-         use_2to3=True
     )
 
 try:
