--- scipy/special/tests/test_nan_inputs.py.orig	2026-03-09 10:54:47.868343657 +0100
+++ scipy/special/tests/test_nan_inputs.py	2026-03-12 11:09:27.691330088 +0100
@@ -8,6 +8,7 @@
 import pytest
 import scipy.special as sc
 
+import platform
 
 KNOWNFAILURES: dict[str, Callable] = {}
 
@@ -34,7 +35,10 @@
 
 UFUNCS, UFUNC_NAMES = _get_ufuncs()
 
-
+@pytest.mark.skipif(
+    platform.machine()=='riscv64',
+    reason="Fails on HiFive Premier P550"
+)
 @pytest.mark.thread_unsafe
 @pytest.mark.parametrize("func", UFUNCS, ids=UFUNC_NAMES)
 def test_nan_inputs(func):
