Corrects return value type in gobj() method.
Author: Davide Vanzo (Vanderbilt University)
diff -ru glibmm-2.49.7.orig/glib/glibmm/threads.h glibmm-2.49.7/glib/glibmm/threads.h
--- glibmm-2.49.7.orig/glib/glibmm/threads.h	2019-08-06 16:30:02.024398871 -0500
+++ glibmm-2.49.7/glib/glibmm/threads.h	2019-08-06 16:48:55.020430222 -0500
@@ -658,7 +658,7 @@
    */
   inline void replace(T* data);
 
-  GPrivate* gobj() { return gobject_; }
+  GPrivate* gobj() { return &gobject_; }
 
 private:
   GPrivate gobject_;
