--- Makefile.orig	2013-07-16 14:55:29.971221036 +0200
+++ Makefile	2013-07-16 14:58:28.543495360 +0200
@@ -68,7 +68,7 @@
 	      4.3.1 4.3.2 4.3.3 \
 	      4.4.1       4.4.3 \
 	      4.6.1 \
-	      4.7.1
+	      4.7.1	4.7.2
 
 ALLOWED_GCC_VERSIONS=$(ALLOWED_GCC_3xx_VERSIONS) $(ALLOWED_GCC_4xx_VERSIONS)
 
@@ -322,7 +322,7 @@
 ifdef DARWIN
 	XMKMF := $(PREFIX)/bin/xmkmf
 else
-	XMKMF := /usr/bin/X11/xmkmf
+	XMKMF := xmkmf
 endif
 
 MAKEDEPEND_PLAIN = makedepend
