Add missing value for System Pybind11 Bazel config

Author: Alexander Grund (TU Dresden)

--- xla-orig/third_party/tsl/third_party/systemlibs/pybind11.BUILD
+++ xla-4ccfe33c71665ddcbca5b127fefe8baa3ed632d4/third_party/tsl/third_party/systemlibs/pybind11.BUILD
@@ -6,3 +6,10 @@
         "@tsl//third_party/python_runtime:headers",
     ],
 )
+
+# Needed by pybind11_bazel.
+config_setting(
+    name = "osx",
+    constraint_values = ["@platforms//os:osx"],
+)
+

