Fix jax version to not show .dev
diff -ru jax-jax-v0.7.0/jax/version.py jax-jax-v0.4.35_version/jax/version.py
--- jax-jax-v0.7.0/jax/version.py.orig	2024-10-22 21:00:23.000000000 +0200
+++ jax-jax-v0.7.0/jax/version.py	2024-11-28 13:10:52.508536023 +0100
@@ -33,6 +33,7 @@
 def _get_version_string() -> str:
   # The build/source distribution for jax & jaxlib overwrites _release_version.
   # In this case we return it directly.
+  return _version
   if _release_version is not None:
     return _release_version
   if os.getenv("WHEEL_VERSION_SUFFIX"):
@@ -82,6 +83,7 @@
   - if JAX_NIGHTLY or JAXLIB_NIGHTLY are set: version looks like "0.4.16.dev20230906"
   - if none are set: version looks like "0.4.16.dev20230906+ge58560fdc
   """
+  return _version
   if _release_version is not None:
     return _release_version
   if os.environ.get('JAX_NIGHTLY') or os.environ.get('JAXLIB_NIGHTLY'):