# using system environment variables
diff -ruN tensorflow-2.5.0_orig/third_party/mlir/tblgen.bzl tensorflow-2.5.0/third_party/mlir/tblgen.bzl
--- tensorflow-2.5.0_orig/third_party/mlir/tblgen.bzl	2021-05-12 15:26:41.000000000 +0200
+++ tensorflow-2.5.0/third_party/mlir/tblgen.bzl	2021-08-11 12:23:34.301593036 +0200
@@ -153,6 +153,7 @@
         inputs = trans_srcs,
         executable = ctx.executable.tblgen,
         arguments = [args],
+        use_default_shell_env = True,
     )
 
     return [DefaultInfo()]