diff -ru xla-12eee889e1f2ad41e27d7b0e970cb92d282d3ec5_old/xla/service/gpu/model/indexing_analysis.cc xla-12eee889e1f2ad41e27d7b0e970cb92d282d3ec5/xla/service/gpu/model/indexing_analysis.cc
--- xla-12eee889e1f2ad41e27d7b0e970cb92d282d3ec5_old/xla/service/gpu/model/indexing_analysis.cc	2024-02-05 19:41:29.000000000 +0100
+++ xla-12eee889e1f2ad41e27d7b0e970cb92d282d3ec5/xla/service/gpu/model/indexing_analysis.cc	2024-02-12 12:09:35.301680070 +0100
@@ -687,7 +687,7 @@
 llvm::SmallVector<AffineExpr, 4> DelinearizeInBoundsIndex(
     mlir::AffineExpr linear, absl::Span<const int64_t> sizes,
     absl::Span<const int64_t> strides) {
-  llvm::SmallVector<AffineExpr> result;
+  llvm::SmallVector<AffineExpr, 4> result;  // THEMBL; see commit c10075688d773c43c22e658c814a94ade3cbb372
   result.reserve(sizes.size());
   for (auto [size, stride] : llvm::zip(sizes, strides)) {
     result.push_back(linear.floorDiv(stride) % size);
