diff -Naur circos-0.69-6_orig/bin/list.modules circos-0.69-6/bin/list.modules
--- circos-0.69-6_orig/bin/list.modules	2018-02-07 15:49:25.130398451 +0000
+++ circos-0.69-6/bin/list.modules	2018-02-07 15:50:52.018553202 +0000
@@ -2,4 +2,5 @@
 
 # Contributed by Charles Howes
 # http://groups.google.com/group/circos-data-visualization/browse_thread/thread/96e74d863a53e405?hl=en_US
-awk '!/^[\t ]*use /{next};$2~/^(lib|Circos.*|base|strict|vars|warnings);?$/{next};{sub(";","",$2);print $2}' circos ../lib/Circos/*pm ../lib/Circos.pm ../lib/Circos/*/*pm | sort -u 
+
+awk '!/^[\t ]*use /{next};$2~/^(lib|Circos.*|base|strict|vars|warnings);?$/{next};{sub(";","",$2);print $2}' `which circos` $EBROOTCIRCOS/lib/Circos/*pm $EBROOTCIRCOS/lib/Circos.pm $EBROOTCIRCOS/lib/Circos/*/*pm | sort -u 
