Patch to use EasyBuild provided pybind11
Author: Alexander Grund (TU-Dresden/Germany)
--- oxDNA-3.6.0-orig/oxpy/CMakeLists.txt	2024-02-28 09:13:45.655751347 +0100
+++ oxDNA-3.6.0/oxpy/CMakeLists.txt	2024-02-28 09:15:27.791928120 +0100
@@ -6,13 +6,13 @@
 IF(Python)
 	# tell pybind11 to use python3
 	set(PYBIND11_PYTHON_VERSION 3.9 CACHE STRING "Python version to use for compiling modules")
-	ADD_SUBDIRECTORY(pybind11)
+	find_package(pybind11 CONFIG REQUIRED)
 	ADD_DEFINITIONS(-DPYTHON_BINDINGS)
 	
 	INCLUDE_DIRECTORIES( ${PYTHON_INCLUDE_DIRS} )
 	MESSAGE(STATUS "The detected Python's include dirs are: ${PYTHON_INCLUDE_DIRS}" )
 	
-	INCLUDE_DIRECTORIES( ${PROJECT_SOURCE_DIR}/oxpy/pybind11/include )
+	INCLUDE_DIRECTORIES( ${pybind11_INCLUDE_DIR} )
 	INCLUDE_DIRECTORIES( ${PROJECT_SOURCE_DIR}/src )
 	
 	MESSAGE(STATUS "Enabling Python bindings")
