diff -Nru hh-suite-3.3.0.orig/cmake/CheckSSEFeatures.cmake hh-suite-3.3.0/cmake/CheckSSEFeatures.cmake
--- hh-suite-3.3.0.orig/cmake/CheckSSEFeatures.cmake	2021-07-28 18:33:46.507081784 +0000
+++ hh-suite-3.3.0/cmake/CheckSSEFeatures.cmake	2021-07-28 18:35:56.679994633 +0000
@@ -1,6 +1,6 @@
 macro (PCL_CHECK_FOR_SSE)
     include(CheckCXXSourceRuns)
-    set(CMAKE_REQUIRED_FLAGS "-march=native")
+    #set(CMAKE_REQUIRED_FLAGS "-march=native")
     check_cxx_source_runs("
       #include <immintrin.h>
       int main()
diff -Nru hh-suite-3.3.0.orig/src/CMakeLists.txt hh-suite-3.3.0/src/CMakeLists.txt
--- hh-suite-3.3.0.orig/src/CMakeLists.txt	2021-07-28 18:33:46.525081911 +0000
+++ hh-suite-3.3.0/src/CMakeLists.txt	2021-07-28 18:35:32.645826091 +0000
@@ -53,17 +53,17 @@
             message(WARNING "At least SSE4.1 is needed for best performance")
         endif ()
         # clang has a problem with march=native on travis
-        if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.0.0")
-            set(ARCH_FLAGS "${SSE_FLAGS}")
-        else()
-            set(ARCH_FLAGS "-march=native")
-        endif()
-    else ()
-        if (PPC64)
-            set(ARCH_FLAGS "-mcpu=native")
-        else ()
-            set(ARCH_FLAGS "-march=native")
-        endif ()
+        #if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.0.0")
+        #    set(ARCH_FLAGS "${SSE_FLAGS}")
+        #else()
+        #    set(ARCH_FLAGS "-march=native")
+        #endif()
+    #else ()
+    #    if (PPC64) 
+    #     set(ARCH_FLAGS "-mcpu=native")
+    #    else ()
+    #        set(ARCH_FLAGS "-march=native")
+    #    endif ()
     endif ()
 endif ()
 
