--- charm-6.10.2/src/conv-core/memoryaffinity.C.orig	2020-08-11 10:32:08.024080784 +0100
+++ charm-6.10.2/src/conv-core/memoryaffinity.C	2020-08-11 10:32:28.356832962 +0100
@@ -182,7 +182,7 @@
         for (i=0; i<strlen((const char *)nodemap); i++) {
             if (nodemap[i]==',') nodemapArrSize++;
         }
-        nodemapArr = malloc(nodemapArrSize*sizeof(int));
+        nodemapArr = (int *) malloc(nodemapArrSize*sizeof(int));
         prevIntStart=j=0;
         for (i=0; i<strlen((const char *)nodemap); i++) {
             if (nodemap[i]==',') {
