--- NAMD_2.8_Source/charm-6.3.2/include/cc-icc.sh.orig	2014-09-10 07:34:18.701304616 +0200
+++ NAMD_2.8_Source/charm-6.3.2/include/cc-icc.sh	2014-09-10 07:34:36.201671197 +0200
@@ -1,7 +1,7 @@
 
 # test version
-ICC_ver=`icc -v 2>&1 | grep Version`
-ICC_ver=`echo $ICC_ver | awk '{ print $2; }' | awk 'BEGIN {FS="."}; { print $1; }'`
+ICC_ver=`icc -v 2>&1 | grep -i Version`
+ICC_ver=`echo $ICC_ver | awk '{ print $3; }' | awk 'BEGIN {FS="."}; { print $1; }'`
 test -z "$ICC_ver" && echo "ICC compiler not found!" && exit 1
 #echo version:$ICC_ver
 
