--- charm-6.10.2/src/arch/mpi-linux-x86_64/cc-mpicxx.sh	2021-11-08 12:08:11.000000000 +0100
+++ charm-6.10.2/src/arch/mpi-linux-x86_64/cc-mpicxx.sh	2021-11-08 12:08:19.000000000 +0100
@@ -45,8 +45,8 @@
 # fortran compiler
 # for Intel Fortran compiler 8.0 and higher which is renamed to ifort from ifc
 # does not work for ifc 7.0
-CMK_CF77="mpif77 -auto -fPIC "
-CMK_CF90="mpif90 -auto -fPIC "
+CMK_CF77="mpif77 -fPIC "
+CMK_CF90="mpif90 -fPIC "
 CMK_CF90_FIXED="$CMK_CF90 -132 -FI "
 F90DIR=`which ifort 2> /dev/null`
 if test -x "$F90DIR"
