diff -Naur networkx-2.5.1.orig/networkx.egg-info/requires.txt networkx-2.5.1/networkx.egg-info/requires.txt
--- networkx-2.5.1.orig/networkx.egg-info/requires.txt	2021-04-04 00:20:57.000000000 +0200
+++ networkx-2.5.1/networkx.egg-info/requires.txt	2021-06-27 10:48:45.453840000 +0200
@@ -1,4 +1,4 @@
-decorator<5,>=4.3
+decorator>=5.0.7,<6
 
 [all]
 numpy
diff -Naur networkx-2.5.1.orig/setup.py networkx-2.5.1/setup.py
--- networkx-2.5.1.orig/setup.py	2021-04-04 00:01:13.000000000 +0200
+++ networkx-2.5.1/setup.py	2021-06-27 11:02:45.537344000 +0200
@@ -124,7 +124,7 @@
     "networkx.utils": ["tests/*.py"],
 }
 
-install_requires = ["decorator>=4.3,<5"]
+install_requires = ["decorator>=5.0.7,<6"]
 extras_require = {
     "all": [
         "numpy",
