--- ./UFconfig/UFconfig.mk      2011-12-08 20:19:48.000000000 +0100
+++ ../SuiteSparse_patched/UFconfig/UFconfig.mk 2012-03-26 13:46:32.081278439 +0200
@@ -53,7 +53,7 @@
 # default make environment.

 # C and C++ compiler flags.  The first three are standard for *.c and *.cpp
-CF = $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -O3 -fexceptions -fPIC
+CF = $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -fPIC #-O3 -fexceptions 

 # ranlib, and ar, for generating libraries.  If you don't need ranlib,
 # just change it to RANLAB = echo
@@ -148,8 +148,9 @@
 # The path is relative to where it is used, in CHOLMOD/Lib, CHOLMOD/MATLAB, etc.
 # You may wish to use an absolute path.  METIS is optional.  Compile
 # CHOLMOD with -DNPARTITION if you do not wish to use METIS.
-METIS_PATH = ../../metis-4.0
-METIS = ../../metis-4.0/libmetis.a
+METIS_PATH = $(SOFTROOTPARMETIS)
+METIS = $(SOFTROOTPARMETIS)/lib/libparmetis.a $(SOFTROOTPARMETIS)/lib/libmetis.a
+

 #------------------------------------------------------------------------------
 # UMFPACK configuration:
@@ -251,9 +252,15 @@
 # Linux
 #------------------------------------------------------------------------------

-# Using default compilers:
-# CC = gcc
-# CFLAGS = -O3 -fexceptions
+# Using mpi compilers:
+CC = $(SOFTVARMPICC)
+CFLAGS = $(SOFTVARCFLAGS)
+CXX = $(SOFTVARMPICXX)
+F77 = $(SOFTVARMPIF77)
+F77FLAGS = $(SOFTVARFFLAGS)
+INTEL_LIB = $(MKLROOT)/lib/em64t
+BLAS = -L$(INTEL_LIB) -Bstatic -lmkl_em64t -Bdynamic -liomp5 -lguide -lpthread
+LAPACK = -L$(INTEL_LIB) -Bstatic -lmkl_lapack -lmkl_em64t -Bdynamic -liomp5 -lguide -lpthread

 # alternatives:
 # CFLAGS = -g -fexceptions \


