diff -ur rmats-turbo-4.1.2-orig/rMATS_C/Makefile rmats-turbo-4.1.2/rMATS_C/Makefile
--- rmats-turbo-4.1.2-orig/rMATS_C/Makefile	2023-09-11 22:06:31.552985769 +0000
+++ rmats-turbo-4.1.2/rMATS_C/Makefile	2023-09-11 22:08:45.710162837 +0000
@@ -9,7 +9,7 @@
 FC := gfortran
 endif
 
-LDFLAGS := -lm -lgfortran -lgsl -lgslcblas -lgomp -lblas -llapack
+LDFLAGS := -lm -lgfortran -lgsl -lgslcblas -lgomp -L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_intel_ilp64 -lmkl_gnu_thread -lmkl_core -lpthread
 CFLAGS := -Wall -O2 -msse2 -funroll-loops -fopenmp
 
 # Allow gsl flags to be set by environment variable
