--- greenlet-0.4.2/slp_platformselect.h.orig	2014-01-06 23:48:38.000000000 +0100
+++ greenlet-0.4.2/slp_platformselect.h	2014-01-17 17:44:34.793309559 +0100
@@ -6,9 +6,9 @@
 #include "platform/switch_x86_msvc.h" /* MS Visual Studio on X86 */
 #elif defined(MS_WIN64) && defined(_M_X64) && defined(_MSC_VER)
 #include "platform/switch_x64_msvc.h" /* MS Visual Studio on X64 */
-#elif defined(__GNUC__) && defined(__amd64__) && defined(__ILP32__)
+#elif defined(__GNUC__) && defined(__x86_64__) && defined(__ILP32__)
 #include "platform/switch_x32_unix.h" /* gcc on amd64 with x32 ABI */
-#elif defined(__GNUC__) && defined(__amd64__)
+#elif defined(__GNUC__) && defined(__x86_64__)
 #include "platform/switch_amd64_unix.h" /* gcc on amd64 */
 #elif defined(__GNUC__) && defined(__i386__)
 #include "platform/switch_x86_unix.h" /* gcc on X86 */
