--- slp_platformselect.h.orig	2013-01-04 22:22:14.349540291 +0100
+++ slp_platformselect.h	2013-01-04 22:22:24.479120125 +0100
@@ -6,7 +6,7 @@
 #include "platform/switch_x86_msvc.h" /* MS Visual Studio on X86 */
 #elif defined(MS_WIN64) && defined(_M_X64)
 #include "platform/switch_x64_msvc.h" /* MS Visual Studio on X64 */
-#elif defined(__GNUC__) && defined(__amd64__)
+#elif defined(__GNUC__) && defined(__x86_64__)
 #include "platform/switch_amd64_unix.h" /* gcc on amd64 */
 #elif defined(__GNUC__) && defined(__i386__)
 #include "platform/switch_x86_unix.h" /* gcc on X86 */
