--- slp_platformselect.h.orig	2014-01-06 23:48:38.000000000 +0100
+++ slp_platformselect.h	2014-01-31 18:05:36.125349420 +0100
@@ -8,7 +8,7 @@
 #include "platform/switch_x64_msvc.h" /* MS Visual Studio on X64 */
 #elif defined(__GNUC__) && defined(__amd64__) && defined(__ILP32__)
 #include "platform/switch_x32_unix.h" /* gcc on amd64 with x32 ABI */
-#elif defined(__GNUC__) && defined(__amd64__)
+#elif defined(__GNUC__) && defined(__x86_64__)
 #include "platform/switch_amd64_unix.h" /* gcc on amd64 */
 #elif defined(__GNUC__) && defined(__i386__)
 #include "platform/switch_x86_unix.h" /* gcc on X86 */
