--- arch/makefile.include.linux_intel.orig	2025-08-11 15:41:38.000000000 -0700
+++ arch/makefile.include.linux_intel	2025-08-13 12:04:34.000000000 -0700
@@ -7,16 +7,21 @@
              -Davoidalloc \
              -Duse_bse_te \
              -Dtbdyn \
-             -Duse_shmem
+             -Duse_shmem \
+	     -Dsol_compat \
+	     -Dlibbeef \
+	     -DVASP2WANNIER90
 
 CPP        = fpp -f_com=no -free -w0  $*$(FUFFIX) $*$(SUFFIX) $(CPP_OPTIONS)
 
 FC         = mpiifort
-FCL        = mpiifort -mkl=sequential -lstdc++
+FCL        = mpiifort 
 
 FREE       = -free -names lowercase
 
 FFLAGS     = -assume byterecl -w
+VASP_TARGET_CPU ?= -march=core-avx2 -axCore-AVX2
+FFLAGS     += $(VASP_TARGET_CPU)
 OFLAG      = -O2
 OFLAG_IN   = $(OFLAG)
 DEBUG      = -O0
@@ -31,7 +36,7 @@
 
 INCS       =-I$(MKLROOT)/include/fftw
 
-LLIBS      = $(SCALAPACK) $(LAPACK) $(BLAS)
+LLIBS      = $(SCALAPACK) $(LAPACK) $(BLAS) -lbeef -lwannier
 
 
 OBJECTS_O1 += fftw3d.o fftmpi.o fftmpiw.o
@@ -66,14 +71,13 @@
 
 CC         = icc
 CXX        = icpc
-CFLAGS     = -fPIC -DADD_ -Wall -openmp -DMAGMA_WITH_MKL -DMAGMA_SETAFFINITY -DGPUSHMEM=300 -DHAVE_CUBLAS
+CFLAGS     = -fPIC -DADD_ -Wall -qopenmp -DMAGMA_WITH_MKL -DMAGMA_SETAFFINITY -DGPUSHMEM=300 -DHAVE_CUBLAS
 
 CUDA_ROOT  ?= /usr/local/cuda/
-NVCC       := $(CUDA_ROOT)/bin/nvcc -ccbin=icc
+NVCC       := nvcc -ccbin=icc
 CUDA_LIB   := -L$(CUDA_ROOT)/lib64 -lnvToolsExt -lcudart -lcuda -lcufft -lcublas
 
-GENCODE_ARCH    := -gencode=arch=compute_30,code=\"sm_30,compute_30\" \
-                   -gencode=arch=compute_35,code=\"sm_35,compute_35\" \
-                   -gencode=arch=compute_60,code=\"sm_60,compute_60\"
+GENCODE_ARCH    := -gencode=arch=compute_35,code=\"sm_35,compute_35\" \
+                   -gencode=arch=compute_60,code=\"sm_60,compute_60\" 
 
 MPI_INC    = $(I_MPI_ROOT)/include64/
