--- src/main.F.orig	2024-10-02 10:10:43.000000000 -0700
+++ src/main.F	2024-10-02 11:26:44.000000000 -0700
@@ -3517,7 +3517,7 @@
       ENDIF
 
       CALL CHAIN_FORCE(T_INFO%NIONS,DYN%POSION,TOTEN,TIFOR, &
-           LATT_CUR%A,LATT_CUR%B,IO%IU6)
+           TSIF,LATT_CUR%A,LATT_CUR%B,IO%IU6)
 
       CALL PARALLEL_TEMPERING(NSTEP,T_INFO%NIONS,DYN%POSION,DYN%VEL,TOTEN,TIFOR,DYN%TEBEG,DYN%TEEND, &
            LATT_CUR%A,LATT_CUR%B,IO%IU6)
--- src/makefile.orig	2024-10-02 11:27:09.000000000 -0700
+++ src/makefile	2024-10-02 11:28:35.000000000 -0700
@@ -14,7 +14,7 @@
 OFLAG=$(OFLAG_2)
 OFLAG_IN=$(OFLAG)
 
-LIB=lib parser
+LIB=lib parser pyamff_fortran
 LLIB=-Llib -ldmy -Lparser -lparser
 
 SRCDIR=../../src
@@ -145,7 +145,7 @@
 	$(MAKE) -C $@ -j1
 #	$(MAKE) -C $@
 
-dependencies: sources
+dependencies: sources libs
 	$(MAKE) depend
 
 depend: $(F90SRC)
--- src/.objects.orig	2024-10-02 10:08:47.000000000 -0700
+++ src/.objects	2024-10-02 11:40:37.000000000 -0700
@@ -115,6 +115,21 @@
 	dos.o \
 	elf.o \
 	hamil_rot.o \
+	bfgs.o \
+	dynmat.o \
+	instanton.o \
+	lbfgs.o \
+	sd.o \
+	cg.o \
+	dimer.o \
+	bbm.o \
+        fire.o \
+	lanczos.o \
+	neb.o \
+	qm.o \
+	pyamff_fortran/*.o \
+	ml_pyamff.o \
+	opt.o \
 	chain.o \
 	dyna.o \
 	fileio.o \
