--- src/.objects.orig	2019-05-31 10:47:03.000000000 -0700
+++ src/.objects	2019-05-31 10:48:40.000000000 -0700
@@ -69,6 +69,19 @@
 	tet.o \
 	tetweight.o \
 	hamil_rot.o \
+        bfgs.o \
+        dynmat.o \
+        instanton.o \
+        lbfgs.o \
+        sd.o \
+        cg.o \
+        dimer.o \
+        bbm.o \
+        fire.o \
+        lanczos.o \
+        neb.o \
+        qm.o \
+        opt.o \
 	chain.o \
 	dyna.o \
 	k-proj.o \
--- src/main.F.orig	2019-06-03 22:39:52.527709533 +0000
+++ src/main.F	2019-06-03 22:40:42.174860559 +0000
@@ -3144,7 +3144,7 @@
       ENDIF
 
       CALL CHAIN_FORCE(T_INFO%NIONS,DYN%POSION,TOTEN,TIFOR, &
-           LATT_CUR%A,LATT_CUR%B,IO%IU6)
+           TSIF,LATT_CUR%A,LATT_CUR%B,IO%IU6)
 
       CALL PARALLEL_TEMPERING(NSTEP,T_INFO%NIONS,DYN%POSION,DYN%VEL,TOTEN,TIFOR,DYN%TEBEG,DYN%TEEND, &
            LATT_CUR%A,LATT_CUR%B,IO%IU6)
