--- arch/makefile.include.linux_intel.orig	2018-07-19 20:04:45.661341455 +0000
+++ arch/makefile.include.linux_intel	2018-07-19 20:12:37.936177534 +0000
@@ -2,13 +2,12 @@
 CPP_OPTIONS= -DMPI -DHOST=\"IFC91_ompi\" -DIFC \
              -DCACHE_SIZE=4000 -DPGF90 -Davoidalloc \
              -DMPI_BLOCK=8000 -DscaLAPACK -Duse_collective \
-             -DnoAugXCmeta -Duse_bse_te \
-             -Duse_shmem -Dtbdyn
+             -Duse_bse_te -Duse_shmem -Dtbdyn -Dlibbeef
 
 CPP        = fpp -f_com=no -free -w0  $*$(FUFFIX) $*$(SUFFIX) $(CPP_OPTIONS)
 
-FC         = mpiifort
-FCL        = mpiifort -mkl
+FC         = mpifort
+FCL        = mpifort -mkl -lstdc++
 
 FREE       = -free -names lowercase
 
@@ -20,14 +19,13 @@
 MKL_PATH   = $(MKLROOT)/lib/intel64
 BLAS       =
 LAPACK     =
-BLACS      = -lmkl_blacs_intelmpi_lp64
+BLACS      = -lmkl_blacs_openmpi_lp64
 SCALAPACK  = $(MKL_PATH)/libmkl_scalapack_lp64.a $(BLACS)
 
-OBJECTS    = fftmpiw.o fftmpi_map.o fftw3d.o fft3dlib.o \
-             $(MKLROOT)/interfaces/fftw3xf/libfftw3xf_intel.a
+OBJECTS    = fftmpiw.o fftmpi_map.o fftw3d.o fft3dlib.o 
 INCS       =-I$(MKLROOT)/include/fftw
 
-LLIBS      = $(SCALAPACK) $(LAPACK) $(BLAS)
+LLIBS      = $(SCALAPACK) $(LAPACK) $(BLAS) -lbeef
 
 OBJECTS_O1 += fft3dfurth.o fftw3d.o fftmpi.o fftmpiw.o
 OBJECTS_O2 += fft3dlib.o

--- src/.objects.orig	2018-07-20 17:10:13.104748655 +0000
+++ src/.objects	2018-07-20 17:20:19.153257130 +0000
@@ -64,6 +64,7 @@
 	tet.o \
 	tetweight.o \
 	hamil_rot.o \
+	bfgs.o dynmat.o  instanton.o  lbfgs.o sd.o   cg.o dimer.o bbm.o fire.o lanczos.o neb.o  qm.o opt.o \
 	chain.o \
 	dyna.o \
 	k-proj.o \

--- src/main.F.orig	2018-07-20 17:53:29.741144151 +0000
+++ src/main.F	2018-07-20 17:48:18.001983108 +0000
@@ -3220,7 +3220,7 @@
       CALL DYNMATFULL_ENERGY_FORCE(SCALEE, T_INFO%NIONS, DYN%POSION, TOTEN, TIFOR, LATT_CUR%A , IO%IU0)
 
       CALL CHAIN_FORCE(T_INFO%NIONS,DYN%POSION,TOTEN,TIFOR, &
-           LATT_CUR%A,LATT_CUR%B,IO%IU6)
+           TSIF,LATT_CUR%A,LATT_CUR%B,IO%IU6)
 
       CALL PARALLEL_TEMPERING(NSTEP,T_INFO%NIONS,DYN%POSION,DYN%VEL,TOTEN,TIFOR,DYN%TEBEG,DYN%TEEND, &
            LATT_CUR%A,LATT_CUR%B,IO%IU6)
