--- src/pot.F.orig	2024-09-24 09:53:04.460499398 -0700
+++ src/pot.F	2024-09-24 09:56:23.092612759 -0700
@@ -357,6 +357,12 @@
       IF (LBEXTERNAL()) CALL BEXT_ADDV(CVTOT,GRIDC,SIZE(CVTOT,2))
 ! bexternal__
       CALL POT_FLIP(CVTOT, GRIDC,WDES%NCDIJ )
+
+! solvation__
+! vaclevel
+      IF (LSOL) CALL GET_FERMISHIFT(LATT_CUR, T_INFO, GRIDC, WDES, CWORK, CHTOT)
+! vaclevel
+! solvation__
 !=======================================================================
 ! if overlap is used :
 ! copy CVTOT to SV and set contribution of unbalanced lattice-vectors
