diff -Nru vasp.5.4.4.pl2.orig/arch/makefile.include.linux_intel vasp.5.4.4.pl2/arch/makefile.include.linux_intel
--- vasp.5.4.4.pl2.orig/arch/makefile.include.linux_intel	2021-01-22 19:51:02.942571000 +0000
+++ vasp.5.4.4.pl2/arch/makefile.include.linux_intel	2021-05-27 12:51:24.198792000 +0100
@@ -66,13 +66,13 @@
 
 CC         = icc
 CXX        = icpc
-CFLAGS     = -fPIC -DADD_ -Wall -openmp -DMAGMA_WITH_MKL -DMAGMA_SETAFFINITY -DGPUSHMEM=300 -DHAVE_CUBLAS
+CFLAGS     = -fPIC -DADD_ -Wall -qopenmp -DMAGMA_WITH_MKL -DMAGMA_SETAFFINITY -DGPUSHMEM=300 -DHAVE_CUBLAS
 
 CUDA_ROOT  ?= /usr/local/cuda/
-NVCC       := $(CUDA_ROOT)/bin/nvcc -ccbin=icc
+NVCC       := $(CUDA_ROOT)/bin/nvcc -ccbin=icc -std=c++11
 CUDA_LIB   := -L$(CUDA_ROOT)/lib64 -lnvToolsExt -lcudart -lcuda -lcufft -lcublas
 
-GENCODE_ARCH    := -gencode=arch=compute_30,code=\"sm_30,compute_30\" \
+GENCODE_ARCH    ?= -gencode=arch=compute_30,code=\"sm_30,compute_30\" \
                    -gencode=arch=compute_35,code=\"sm_35,compute_35\" \
                    -gencode=arch=compute_60,code=\"sm_60,compute_60\"
 
