--- src/main.F.orig	2021-04-29 11:13:19.000000000 -0700
+++ src/main.F	2024-10-01 18:18:30.000000000 -0700
@@ -3403,7 +3403,7 @@
       ENDIF
 
       CALL CHAIN_FORCE(T_INFO%NIONS,DYN%POSION,TOTEN,TIFOR, &
-           LATT_CUR%A,LATT_CUR%B,IO%IU6)
+           TSIF,LATT_CUR%A,LATT_CUR%B,IO%IU6) 
 
       CALL PARALLEL_TEMPERING(NSTEP,T_INFO%NIONS,DYN%POSION,DYN%VEL,TOTEN,TIFOR,DYN%TEBEG,DYN%TEEND, &
            LATT_CUR%A,LATT_CUR%B,IO%IU6)
--- src/.objects.orig	2021-04-29 11:13:18.000000000 -0700
+++ src/.objects	2024-10-01 18:18:33.000000000 -0700
@@ -98,6 +98,19 @@
 	dos.o \
 	elf.o \
 	hamil_rot.o \
+        bfgs.o \
+        dynmat.o \
+        instanton.o \
+        lbfgs.o \
+        sd.o \
+        cg.o \
+        dimer.o \
+        bbm.o \
+        fire.o \
+        lanczos.o \
+        neb.o \
+        qm.o \
+        opt.o \
 	chain.o \
 	dyna.o \
 	fileio.o \
@@ -352,6 +365,19 @@
 	dos.o \
 	elf.o \
 	hamil_rot.o \
+        bfgs.o \
+        dynmat.o \
+        instanton.o \
+        lbfgs.o \
+        sd.o \
+        cg.o \
+        dimer.o \
+        bbm.o \
+        fire.o \
+        lanczos.o \
+        neb.o \
+        qm.o \
+        opt.o \
 	chain.o \
 	dyna.o \
 	us.o \
