--- arch/makefile.include.linux_intel.orig      2022-02-01 11:55:53.000000000 -0800
+++ arch/makefile.include.linux_intel   2022-02-01 11:57:42.000000000 -0800
@@ -30,9 +30,9 @@
 BLACS      = -lmkl_blacs_intelmpi_lp64
 SCALAPACK  = $(MKL_PATH)/libmkl_scalapack_lp64.a $(BLACS)
 
-OBJECTS    = fftmpiw.o fftmpi_map.o fft3dlib.o fftw3d.o
+OBJECTS    = fftmpiw.o  fftw3d.o
 
-INCS       =-I$(MKLROOT)/include/fftw
+INCS       = -I$(EBROOTFFTW)/include
 
 LLIBS      = $(SCALAPACK) $(LAPACK) $(BLAS) -lbeef -lwannier

