--- src/.objects.orig	2024-09-24 10:01:12.368777591 -0700
+++ src/.objects	2024-09-24 10:00:25.150750685 -0700
@@ -69,6 +69,19 @@
 	tet.o \
 	tetweight.o \
 	hamil_rot.o \
+        bfgs.o \
+        dynmat.o \
+        instanton.o \
+        lbfgs.o \
+        sd.o \
+        cg.o \
+        dimer.o \
+        bbm.o \
+        fire.o \
+        lanczos.o \
+        neb.o \
+        qm.o \
+        opt.o \
 	chain.o \
 	dyna.o \
 	k-proj.o \
@@ -268,6 +281,19 @@
 	tet.o \
 	tetweight.o \
 	hamil_rot.o \
+        bfgs.o \
+        dynmat.o \
+        instanton.o \
+        lbfgs.o \
+        sd.o \
+        cg.o \
+        dimer.o \
+        bbm.o \
+        fire.o \
+        lanczos.o \
+        neb.o \
+        qm.o \
+        opt.o \
 	chain.o \
 	dyna.o \
 	k-proj.o \
--- src/main.F.orig	2024-09-24 09:58:30.624685428 -0700
+++ src/main.F	2024-09-24 09:59:20.321713745 -0700
@@ -3144,7 +3144,7 @@
       ENDIF
 
       CALL CHAIN_FORCE(T_INFO%NIONS,DYN%POSION,TOTEN,TIFOR, &
-           LATT_CUR%A,LATT_CUR%B,IO%IU6)
+           TSIF,LATT_CUR%A,LATT_CUR%B,IO%IU6) 
 
       CALL PARALLEL_TEMPERING(NSTEP,T_INFO%NIONS,DYN%POSION,DYN%VEL,TOTEN,TIFOR,DYN%TEBEG,DYN%TEEND, &
            LATT_CUR%A,LATT_CUR%B,IO%IU6)
