--- arch/makefile.include.linux_intel.orig	2024-10-01 15:42:52.000000000 -0700
+++ arch/makefile.include.linux_intel	2024-10-01 15:54:03.000000000 -0700
@@ -7,7 +7,10 @@
              -Dvasp6 \
              -Duse_bse_te \
              -Dtbdyn \
-             -Dfock_dblbuf
+             -Dfock_dblbuf \
+	     -Dsol_compat \
+	     -Dlibbeef \
+	     -DVASP2WANNIER90
 
 CPP        = fpp -f_com=no -free -w0  $*$(FUFFIX) $*$(SUFFIX) $(CPP_OPTIONS)
 
@@ -16,7 +19,7 @@
 
 FREE       = -free -names lowercase
 
-FFLAGS     = -assume byterecl -w -xHOST
+FFLAGS     = -assume byterecl -w 
 OFLAG      = -O2
 OFLAG_IN   = $(OFLAG)
 DEBUG      = -O0
@@ -29,9 +32,9 @@
 
 OBJECTS    = fftmpiw.o fftmpi_map.o fft3dlib.o fftw3d.o
 
-INCS       =-I$(MKLROOT)/include/fftw
+INCS       =-I$(EBROOTFFTW)/include
 
-LLIBS      = $(SCALAPACK) $(LAPACK) $(BLAS)
+LLIBS      = -lflexiblas_intel -lscalapack -lfftw3 -lbeef -lwannier
 
 OBJECTS_O1 += fftw3d.o fftmpi.o fftmpiw.o
 OBJECTS_O2 += fft3dlib.o
@@ -67,7 +70,7 @@
 
 # Minimal requirement is CUDA >= 10.X. For "sm_80" you need CUDA >= 11.X.
 CUDA_ROOT  ?= /usr/local/cuda/
-NVCC       := $(CUDA_ROOT)/bin/nvcc -ccbin=icc -allow-unsupported-compiler
+NVCC       := nvcc -ccbin=icc -allow-unsupported-compiler
 CUDA_LIB   := -L$(CUDA_ROOT)/lib64 -lnvToolsExt -lcudart -lcuda -lcufft -lcublas
 
 GENCODE_ARCH    := -gencode=arch=compute_60,code=\"sm_60,compute_60\" \
