--- src/Makefile.vtst	2013-09-06 19:44:53.143005671 -0700
+++ src/Makefile	2013-09-06 19:43:46.093137453 -0700
@@ -248,6 +248,8 @@
          mix.o      charge.o   xcgrad.o   xcspin.o    potex1.o   potex2.o  \
          metagga.o  constrmag.o pot.o      cl_shift.o force.o    dos.o      elf.o      \
          tet.o      hamil.o    steep.o    \
+         dimer.o    dynmat.o   neb.o      lanczos.o   instanton.o \
+         sd.o       cg.o       qm.o       lbfgs.o     bfgs.o  fire.o   opt.o \
          chain.o    dyna.o     relativistic.o LDApU.o sphpro.o  bdrpro.o paw.o   us.o \
          ebs.o      wavpre.o   wavpre_noio.o broyden.o \
          dynbr.o    rmm-diis.o reader.o   writer.o   tutor.o xml_writer.o \
--- src/instanton.F.vtst	2013-09-05 16:27:31.799957812 -0700
+++ src/instanton.F	2013-09-05 16:35:47.352646705 -0700
@@ -1175,6 +1175,6 @@
 
 !**********************************************************************
  
- #endif
+#endif
 
   END MODULE instanton
--- src/dimer.F.vtst	2013-09-05 16:49:31.218318123 -0700
+++ src/dimer.F	2013-09-05 16:56:50.234945792 -0700
@@ -76,10 +76,10 @@
       IU5 = IO%IU5
       IU6 = IO%IU6
       Nions = T_info%nions
-     #if defined(MPI_CHAIN)
+#if defined(MPI_CHAIN)
         IF (IU6>=0) WRITE(IU0,*) 'The IMAGES tag should not be set for this implementation of the dimer method.'
         STOP
-     #endif
+#endif
       IF (IU6>=0) OPEN(UNIT=IUdim, FILE='DIMCAR')
       IF (IU6>=0) WRITE(IUdim,'(A5,5A14)') 'Step','Force','Torque','Energy','Curvature','Angle'
       Itr = 1
