Enable DFT-D4
Author: Ine Arts (Universiteit Antwerpen)
--- vasp.6.6.0_orig/arch/makefile.include.oneapi_omp	2026-03-10 16:24:02.156613120 +0100
+++ vasp.6.6.0/arch/makefile.include.oneapi_omp	2026-03-10 16:31:59.831507585 +0100
@@ -80,3 +80,5 @@
 ## This may be required for the C++17 filesystem library if the underlying
 ## system compiler is older than GNU 9.1. For newer versions this can be removed.
 #LLIBS       += -lstdc++fs
+
+# DFT-D4
+CPP_OPTIONS += -DDFTD4
+LLIBS       += $(shell pkg-config --libs dftd4,multicharge,mctc-lib,mstore)
+INCS        += $(shell pkg-config --cflags dftd4,multicharge,mctc-lib,mstore)
