Replaces import of scipy.random with numpy.random.

--- /dev/shm/yaff/1.6.0/foss-2023a/yaff/yaff-1.6.0/yaff/analysis/pca.py.orig	2023-12-11 16:29:17.807097965 +0000
+++ /dev/shm/yaff/1.6.0/foss-2023a/yaff/yaff-1.6.0/yaff/analysis/pca.py	2023-12-11 16:29:46.094953909 +0000
@@ -30,7 +30,7 @@
 import numpy as np
 import scipy.linalg as spla
 import matplotlib.pyplot as pt
-from scipy import random
+from numpy import random
 
 from molmod.units import *
 from molmod.constants import boltzmann

