From b4986e0d78aa9bd54dd48a4965bf5773cc4354e5 Mon Sep 17 00:00:00 2001
From: Toon Verstraelen <Toon.Verstraelen@UGent.be>
Date: Thu, 9 Nov 2017 06:37:36 +0100
Subject: [PATCH] Add missing include math.h

---
 yaff/pes/grid.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/yaff/pes/grid.c b/yaff/pes/grid.c
index 8cd4392..fe4d019 100644
--- a/yaff/pes/grid.c
+++ b/yaff/pes/grid.c
@@ -28,6 +28,8 @@
 #include <stdio.h>
 #endif
 
+#include <math.h>
+
 #include "grid.h"
 
 
-- 
1.8.3.1

