Use external FACE build

Åke Sandgren, 2022-01-25
diff -ru forbear.orig/fobos forbear/fobos
--- forbear.orig/fobos	2017-06-11 14:56:47.000000000 +0200
+++ forbear/fobos	2022-01-25 09:16:13.737587610 +0100
@@ -17,7 +17,7 @@
 $DEBUG_PGI   = -C -g -gopt -Mchkfpstk -Mchkptr -Mchkstk -Mcoff -traceback
 $OPTIMIZE    = -O3
 $EXDIRS      = PENF/src/tests/
-               FACE/src/tests/ FACE/src/third_party/
+#               FACE/src/tests/ FACE/src/third_party/
 
 # main modes
 # GNU
@@ -117,6 +117,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-static-gnu]
 compiler     = gnu
@@ -130,6 +132,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-shared-gnu-debug]
 compiler     = gnu
@@ -144,6 +148,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-static-gnu-debug]
 compiler     = gnu
@@ -158,6 +164,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-shared-intel]
 compiler     = intel
@@ -171,6 +179,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-static-intel]
 compiler     = intel
@@ -184,6 +194,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-shared-intel-debug]
 compiler     = intel
@@ -198,6 +210,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-static-intel-debug]
 compiler     = intel
@@ -212,6 +226,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-static-pgi]
 compiler     = pgi
@@ -225,6 +241,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 [template-static-pgi-debug]
 compiler     = pgi
@@ -239,6 +257,8 @@
 quiet        = False
 log          = True
 jobs         = 2
+ext_libs = face
+include = $EBROOTFACE/include
 
 # rules
 [rule-makedoc]
