Use external FACE build

Åke Sandgren, 2022-01-25
diff --git a/fobos b/fobos
index 4359fca..76da757 100644
--- a/fobos
+++ b/fobos
@@ -117,6 +117,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-static-gnu]
 compiler     = gnu
@@ -130,6 +132,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-shared-gnu-debug]
 compiler     = gnu
@@ -144,6 +148,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-static-gnu-debug]
 compiler     = gnu
@@ -158,6 +164,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-shared-intel]
 compiler     = intel
@@ -171,6 +179,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-static-intel]
 compiler     = intel
@@ -184,6 +194,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-shared-intel-debug]
 compiler     = intel
@@ -198,6 +210,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-static-intel-debug]
 compiler     = intel
@@ -212,6 +226,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-static-pgi]
 compiler     = pgi
@@ -225,6 +241,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 [template-static-pgi-debug]
 compiler     = pgi
@@ -239,6 +257,8 @@ colors       = True
 quiet        = False
 log          = True
 jobs         = 2
+include = $EBROOTFACE/include
+ext_libs  = face
 
 # rules
 [rule-makedoc]
