Use easy build installation roots
author: Christoph Siegert (Leipzig University)

diff -ruN RoseTTAFold.orig/input_prep/make_ss.sh RoseTTAFold/input_prep/make_ss.sh
--- RoseTTAFold.orig/input_prep/make_ss.sh	2021-08-16 12:54:05.769231102 +0200
+++ RoseTTAFold/input_prep/make_ss.sh	2021-08-16 13:03:22.003742639 +0200
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-DATADIR="$CONDA_PREFIX/share/psipred_4.01/data"
+DATADIR="$EBROOTPSIPRED/data"
 echo $DATADIR
 
 i_a3m="$1"
@@ -8,7 +8,7 @@
 
 ID=$(basename $i_a3m .a3m).tmp
 
-$PIPEDIR/csblast-2.2.3/bin/csbuild -i $i_a3m -I a3m -D $PIPEDIR/csblast-2.2.3/data/K4000.crf -o $ID.chk -O chk
+csbuild -i $i_a3m -I a3m -D $EBROOTCSBLAST/data/K4000.crf -o $ID.chk -O chk
 
 head -n 2 $i_a3m > $ID.fasta
 echo $ID.chk > $ID.pn
diff -ruN RoseTTAFold.orig/run_e2e_ver.sh RoseTTAFold/run_e2e_ver.sh
--- RoseTTAFold.orig/run_e2e_ver.sh	2021-08-16 12:54:04.081235616 +0200
+++ RoseTTAFold/run_e2e_ver.sh	2021-08-16 13:01:01.476118686 +0200
@@ -12,8 +12,7 @@
 # <<< conda initialize <<<
 ############################################################
 
-SCRIPT=`realpath -s $0`
-export PIPEDIR=`dirname $SCRIPT`
+export PIPEDIR=$EBROOTROSETTAFOLD
 
 CPU="8"  # number of CPUs to use
 MEM="64" # max memory (in GB)
diff -ruN RoseTTAFold.orig/run_pyrosetta_ver.sh RoseTTAFold/run_pyrosetta_ver.sh
--- RoseTTAFold.orig/run_pyrosetta_ver.sh	2021-08-16 12:54:04.081235616 +0200
+++ RoseTTAFold/run_pyrosetta_ver.sh	2021-08-16 13:00:45.388161737 +0200
@@ -12,8 +12,7 @@
 # <<< conda initialize <<<
 ############################################################
 
-SCRIPT=`realpath -s $0`
-export PIPEDIR=`dirname $SCRIPT`
+export PIPEDIR=$EBROOTROSETTAFOLD
 
 CPU="8"  # number of CPUs to use
 MEM="64" # max memory (in GB)
