Do not attempt to use the conda environment
author: Christoph Siegert (Leipzig University)

diff -ruN RoseTTAFold.orig/run_e2e_ver.sh RoseTTAFold/run_e2e_ver.sh
--- RoseTTAFold.orig/run_e2e_ver.sh	2021-08-16 12:54:04.081235616 +0200
+++ RoseTTAFold/run_e2e_ver.sh	2021-08-16 12:55:10.969056632 +0200
@@ -3,15 +3,6 @@
 # make the script stop when error (non-true exit code) is occured
 set -e
 
-############################################################
-# >>> conda initialize >>>
-# !! Contents within this block are managed by 'conda init' !!
-__conda_setup="$('conda' 'shell.bash' 'hook' 2> /dev/null)"
-eval "$__conda_setup"
-unset __conda_setup
-# <<< conda initialize <<<
-############################################################
-
 SCRIPT=`realpath -s $0`
 export PIPEDIR=`dirname $SCRIPT`
 
@@ -27,7 +18,6 @@
 
 mkdir -p $WDIR/log
 
-conda activate RoseTTAFold
 ############################################################
 # 1. generate MSAs
 ############################################################
diff -ruN RoseTTAFold.orig/run_pyrosetta_ver.sh RoseTTAFold/run_pyrosetta_ver.sh
--- RoseTTAFold.orig/run_pyrosetta_ver.sh	2021-08-16 12:54:04.081235616 +0200
+++ RoseTTAFold/run_pyrosetta_ver.sh	2021-08-16 12:55:43.860968614 +0200
@@ -3,15 +3,6 @@
 # make the script stop when error (non-true exit code) is occured
 set -e
 
-############################################################
-# >>> conda initialize >>>
-# !! Contents within this block are managed by 'conda init' !!
-__conda_setup="$('conda' 'shell.bash' 'hook' 2> /dev/null)"
-eval "$__conda_setup"
-unset __conda_setup
-# <<< conda initialize <<<
-############################################################
-
 SCRIPT=`realpath -s $0`
 export PIPEDIR=`dirname $SCRIPT`
 
@@ -27,7 +18,6 @@
 
 mkdir -p $WDIR/log
 
-conda activate RoseTTAFold
 ############################################################
 # 1. generate MSAs
 ############################################################
@@ -81,9 +71,6 @@
 ############################################################
 mkdir -p $WDIR/pdb-3track
 
-conda deactivate
-conda activate folding
-
 for m in 0 1 2
 do
     for p in 0.05 0.15 0.25 0.35 0.45
