diff -ru minc-toolkit-v2-1.9.16.orig/EBTKS/templates/Matrix.cc minc-toolkit-v2-1.9.16/EBTKS/templates/Matrix.cc
--- minc-toolkit-v2-1.9.16.orig/EBTKS/templates/Matrix.cc	2018-08-22 17:32:17.547496083 +0000
+++ minc-toolkit-v2-1.9.16/EBTKS/templates/Matrix.cc	2018-08-22 17:33:37.363794839 +0000
@@ -32,7 +32,7 @@
 #ifdef USE_COMPMAT
 #ifdef __GNUC__
 #pragma do_not_instantiate Mat<dcomplex>::array
-#pragma do_not_instantiate Mat<dcomplex>::applyIndexFunction
+//#pragma do_not_instantiate Mat<dcomplex>::applyIndexFunction
 #endif
 #endif
 
