diff -ru mofapy2-0.6.1/setup.py mofapy2-0.6.1diff/setup.py
--- mofapy2-0.6.1/setup.py	2021-01-29 17:52:55.000000000 +0000
+++ mofapy2-0.6.1diff/setup.py	2021-03-31 10:42:16.015823000 +0100
@@ -6,7 +6,7 @@
 exec(open(os.path.join(os.path.dirname(__file__), 'mofapy2', 'version.py')).read())
 
 def setup_package():
-  install_requires = ['pandas', 'scipy', 'numpy', 'sklearn', 'argparse', 'h5py']
+  install_requires = ['pandas', 'scipy', 'numpy', 'scikit-learn', 'h5py']
   metadata = dict(
       name = 'mofapy2',
       version = __version__,
