diff -ru mmcif_pdbx-2.0.0/MANIFEST.in mmcif_pdbx-2.0.0_patched/MANIFEST.in
--- mmcif_pdbx-2.0.0/MANIFEST.in	2020-08-02 02:48:02.000000000 +0200
+++ mmcif_pdbx-2.0.0_patched/MANIFEST.in	2021-04-12 15:53:02.900875747 +0200
@@ -1,5 +1,2 @@
 include README.md
 include setup.py setup.cfg
-
-include versioneer.py
-include pdbx/_version.py
diff -ru mmcif_pdbx-2.0.0/pdbx/__init__.py mmcif_pdbx-2.0.0_patched/pdbx/__init__.py
--- mmcif_pdbx-2.0.0/pdbx/__init__.py	2020-08-02 02:48:02.000000000 +0200
+++ mmcif_pdbx-2.0.0_patched/pdbx/__init__.py	2021-04-12 15:57:55.901133230 +0200
@@ -12,10 +12,8 @@
 # import pdbx.reader
 from .errors import PdbxSyntaxError, PdbxError  # noqa: F401
 from .containers import DataCategory, DataContainer  # noqa: F401
-from ._version import get_versions
 
-__version__ = get_versions()["version"]
-del get_versions
+__version__="2.0.0"
 
 
 def load(fp) -> list:
diff -ru mmcif_pdbx-2.0.0/setup.py mmcif_pdbx-2.0.0_patched/setup.py
--- mmcif_pdbx-2.0.0/setup.py	2020-08-02 02:48:02.000000000 +0200
+++ mmcif_pdbx-2.0.0_patched/setup.py	2021-04-12 15:55:20.479915813 +0200
@@ -1,13 +1,12 @@
 """Python packaging for PDBx/mmCIF utilities."""
 import setuptools
-import versioneer
 
 with open("README.md", "r") as fh:
     LONG_DESCRIPTION = fh.read()
 
 setuptools.setup(
     name="mmcif_pdbx",
-    version=versioneer.get_version(),
+    version="2.0.0",
     author="John Westbrook",
     author_email="jwest@rcsb.rutgers.edu",
     maintainer="Nathan Baker",
