Fix license definition for flit
author: micketeer@gmail.com
--- pyproject.toml.orig	2025-04-30 17:24:36.228375457 +0000
+++ pyproject.toml	2025-04-30 17:28:41.544623964 +0000
@@ -13,10 +13,7 @@
 urls.Download = "https://pypi.org/project/Sphinx/"
 urls.Homepage = "https://www.sphinx-doc.org/"
 urls."Issue tracker" = "https://github.com/sphinx-doc/sphinx/issues"
+license = { file = "LICENSE.rst" }
-license = "BSD-2-Clause"
-license-files = [
-    "LICENSE.rst",
-]
 requires-python = ">=3.11"
 
 # Classifiers list: https://pypi.org/classifiers/
