commit d3cf498a7d97708598e44a591c634ba6dfea5e1f
Author: Ake Sandgren <ake.sandgren@hpc2n.umu.se>
Date:   Tue Mar 18 08:50:30 2025 +0100

    Take CC/CFLAGS from outside if defined

diff --git a/src/Makefile b/src/Makefile
index 2ddaa03..7d4a402 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,5 +1,5 @@
-CC=			gcc
-CFLAGS=		-g -w -O2 -fPIC #-m64 #-arch ppc
+CC ?=		gcc
+CFLAGS ?=	-g -w -O2 -fPIC #-m64 #-arch ppc
 DFLAGS=		-D_FILE_OFFSET_BITS=64 -D_USE_KNETFILE -DBGZF_CACHE
 LOBJS=		bgzf.o kstring.o knetfile.o index.o bedidx.o
 AOBJS=		main.o
