--- magma-2.5.0/Makefile.bak	2019-04-09 01:29:46.298154414 +0000
+++ magma-2.5.0/Makefile	2019-04-09 01:30:11.620195774 +0000
@@ -69,7 +69,7 @@
 # NVCC options for the different cards
 # First, add smXX for architecture names
 ifneq ($(findstring Kepler, $(GPU_TARGET)),)
-    GPU_TARGET += sm_30 sm_35
+    GPU_TARGET += sm_35
 endif
 ifneq ($(findstring Maxwell, $(GPU_TARGET)),)
     GPU_TARGET += sm_50
