--- Makefile.orig	2018-03-22 18:54:44.526845057 +0000
+++ Makefile	2018-03-22 18:55:19.466512170 +0000
@@ -1,6 +1,6 @@
 # Set this variable to your static gsl libraries
 # GSL = /usr/local/lib/libgsl.a /usr/local/lib/libgslcblas.a
-GSL=/usr/local/lib/libgsl.a /usr/local/lib/libgslcblas.a
+GSL=${EBROOTGSL}/lib/libgsl.a ${EBROOTGSL}/lib/libgslcblas.a
 
 build/msmc : model/*.d powell.d brent.d maximization_step.d expectation_step.d msmc.d branchlength.d logger.d
 	dmd -O ${GSL} -odbuild -ofbuild/msmc $^
