Remove missing test for ASE
Author: Samuel Moors (Vrije Universiteit Brussel)
diff -ur molecularGSM-02ae717e8f3b3bdfb4fe778c04db61d39194f746.orig/TEST/CMakeLists.txt molecularGSM-02ae717e8f3b3bdfb4fe778c04db61d39194f746/TEST/CMakeLists.txt
--- molecularGSM-02ae717e8f3b3bdfb4fe778c04db61d39194f746.orig/TEST/CMakeLists.txt	2019-08-26 19:05:31.000000000 +0200
+++ molecularGSM-02ae717e8f3b3bdfb4fe778c04db61d39194f746/TEST/CMakeLists.txt	2021-03-30 12:11:41.946968000 +0200
@@ -78,7 +78,7 @@
     #GFSTRINGQ_TEST(methanolFormaldehydeHydTransfer/orca/de-gsm methanol)
 elseif(GSM_ENABLE_ASE)
 		MESSAGE(STATUS "ASE is enabled.")
-    GFSTRINGQ_TEST(dielsAlder/ase/de-gsm diels)
+    #GFSTRINGQ_TEST(dielsAlder/ase/de-gsm diels)
 elseif(GSM_ENABLE_TURBOMOLE)
 		MESSAGE(STATUS "TURBOMOLE is enabled.")
 		GFSTRINGQ_TEST(dielsAlder/turbomole/de-gsm-wrapper diels-w-wrapper)
