diff -Naur ASTRID-2.2.1.orig/src/CMakeLists.txt ASTRID-2.2.1/src/CMakeLists.txt
--- ASTRID-2.2.1.orig/src/CMakeLists.txt	2019-04-17 00:29:25.000000000 +0200
+++ ASTRID-2.2.1/src/CMakeLists.txt	2019-09-26 16:56:19.577170290 +0200
@@ -9,7 +9,7 @@
 file(GLOB TEST_SOURCES *.cpp *.c fastme/*.c)
 list(REMOVE_ITEM TEST_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/astrid.cpp)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}-std=c++14 -g -O4 -Wall")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -g -O4 -Wall")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -O4 -Wall")
 
 include_directories(${JNI_INCLUDE_DIRS} ${JNI_INCLUDE_DIRS2})
