diff -ru WRFV3.orig/phys/module_cu_g3.F WRFV3/phys/module_cu_g3.F
--- WRFV3.orig/phys/module_cu_g3.F	2011-08-26 18:22:51.000000000 +0200
+++ WRFV3/phys/module_cu_g3.F	2012-06-20 15:44:07.494077875 +0200
@@ -3119,7 +3119,7 @@
        pcrit,acrit,acritt
 
      integer :: nall2,ixxx,irandom
-     integer,  dimension (8) :: seed
+     integer,  dimension (12) :: seed
 
 
       DATA PCRIT/850.,800.,750.,700.,650.,600.,550.,500.,450.,400.,    &
