Relax requirements of `atomistics`.

--- 0.5.1/foss-2023a/atomistics/atomistics-atomistics-0.1.15/pyproject.toml.orig	2024-01-09 12:42:59.508409527 +0000
+++ 0.5.1/foss-2023a/atomistics/atomistics-atomistics-0.1.15/pyproject.toml 2024-01-09 12:43:44.589293464 +0000
@@ -26,9 +26,9 @@
 ]
 dependencies = [
     "ase==3.22.1",
-    "numpy==1.26.2",
-    "scipy==1.11.4",
-    "spglib==2.2.0",
+    "numpy>=1.25.1",
+    "scipy>=1.11.1",
+    "spglib>=2.1.0",
 ]
 dynamic = ["version"]
 
