Fix versioning issue and relax requirements.

--- 0.5.1/foss-2023a/pylammpsmpi/pylammpsmpi-pylammpsmpi-0.2.10/setup.py.orig	2023-12-21 17:27:54.405181324 +0000
+++ 0.5.1/foss-2023a/pylammpsmpi/pylammpsmpi-pylammpsmpi-0.2.10/setup.py	2023-12-21 17:31:38.136048756 +0000
@@ -1,8 +1,4 @@
 from setuptools import setup
 
-import versioneer
+setup(version='0.2.10')
 
-setup(
-    version=versioneer.get_version(),
-    cmdclass=versioneer.get_cmdclass(),
-)
\ Ingen nyrad vid filslut

--- 0.5.1/foss-2023a/pylammpsmpi/pylammpsmpi-pylammpsmpi-0.2.10/pyproject.toml.orig	2023-12-21 17:57:36.066162000 +0000
+++ 0.5.1/foss-2023a/pylammpsmpi/pylammpsmpi-pylammpsmpi-0.2.10/pyproject.toml	2023-12-21 17:58:37.373851635 +0000
@@ -24,11 +24,11 @@
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-    "mpi4py==3.1.5",
+    "mpi4py>=3.1.4",
     "pympipool==0.7.9",
-    "numpy==1.26.2",
+    "numpy>=1.25.1",
 ]
-dynamic = ["version"]
+version = '0.2.10'
 
 [project.urls]
 Homepage = "https://github.com/pyiron/pylammpsmpi"
@@ -43,13 +43,3 @@
 
 [tool.setuptools.packages.find]
 include = ["pylammpsmpi*"]
-
-[tool.setuptools.dynamic]
-version = {attr = "pylammpsmpi.__version__"}
-
-[tool.versioneer]
-VCS = "git"
-style = "pep440-pre"
-versionfile_source = "pylammpsmpi/_version.py"
-parentdir_prefix = "pylammpsmpi"
-tag_prefix = "pylammpsmpi-"
