Fixes attempt to open nonexistent README.md.

--- 0.5.1/foss-2023a/emmetcore/emmet-core-0.75.0/setup.py.orig	2023-12-21 18:58:49.428075405 +0000
+++ 0.5.1/foss-2023a/emmetcore/emmet-core-0.75.0/setup.py	2023-12-21 18:59:10.200024324 +0000
@@ -7,8 +7,6 @@
     description="Core Emmet Library",
     author="The Materials Project",
     author_email="feedback@materialsproject.org",
-    long_description=open("../README.md").read(),  # noqa: SIM115
-    long_description_content_type="text/markdown",
     url="https://github.com/materialsproject/emmet",
     packages=find_namespace_packages(include=["emmet.*"]),
     package_data={
