Fix versioning related issue in pyiron atomistics. 

--- 0.5.1/foss-2023a-Python-3.11.3/pyiron_atomistics/pyiron_atomistics-pyiron_atomistics-0.3.11/pyproject.toml.orig	2023-12-13 17:37:16.618007520 +0000
+++ 0.5.1/foss-2023a-Python-3.11.3/pyiron_atomistics/pyiron_atomistics-pyiron_atomistics-0.3.11/pyproject.toml	2023-12-13 17:38:08.416737658 +0000
@@ -43,7 +43,7 @@
     "spglib==2.1.0",
     "structuretoolkit==0.0.15",
 ]
-dynamic = ["version"]
+version = "0.3.11"
 
 [project.urls]
 Homepage = "https://pyiron.org"
@@ -55,13 +55,3 @@
 
 [tool.setuptools.package-data]
 "*" = ["data/*.csv"]
-
-[tool.setuptools.dynamic]
-version = {attr = "pyiron_atomistics.__version__"}
-
-[tool.versioneer]
-VCS = "git"
-style = "pep440-pre"
-versionfile_source = "pyiron_atomistics/_version.py"
-parentdir_prefix = "pyiron_atomistics"
-tag_prefix = "pyiron_atomistics-"
--- 0.5.1/foss-2023a-Python-3.11.3/pyiron_atomistics/pyiron_atomistics-pyiron_atomistics-0.3.11/setup.py.orig	2023-12-13 17:41:20.162738914 +0000
+++ 0.5.1/foss-2023a-Python-3.11.3/pyiron_atomistics/pyiron_atomistics-pyiron_atomistics-0.3.11/setup.py	2023-12-13 17:41:48.007593887 +0000
@@ -3,6 +3,6 @@
 import versioneer
 
 setup(
-    version=versioneer.get_version(),
-    cmdclass=versioneer.get_cmdclass(),
-)
\
+    version="0.3.11",
+)
+
--- 0.5.1/foss-2023a-Python-3.11.3/atomistics/atomistics-atomistics-0.1.15/setup.py.orig	2023-12-13 17:41:20.162738914 +0000
+++ 0.5.1/foss-2023a-Python-3.11.3/atomistics/atomistics-atomistics-0.1.15/setup.py	2023-12-13 17:41:48.007593887 +0000
@@ -3,6 +3,6 @@
 import versioneer
 
 setup(
-    version=versioneer.get_version(),
-    cmdclass=versioneer.get_cmdclass(),
-)
\
+    version="0.3.11",
+)
+
