# Patch makefile to use EB values
# wpoely86@gmail.com
--- Makefile.orig	2017-08-14 12:33:11.280163633 +0200
+++ Makefile	2017-08-14 12:36:46.085730155 +0200
@@ -1,6 +1,6 @@
-CXX = g++
-CXXFLAGS= -Wall -O3 -std=c++0x
-LINKFLAGS = -I . -lpthread -lz 
+CXX := $(CXX) 
+CXXFLAGS := $(CXXFLAGS)
+LINKFLAGS := -I. $(LDFLAGS) $(LIBS) -lz
 DEBUG=
 jellyfishPath = `which jellyfish`
 jellyfishVersion = `jellyfish --version | cut -f2 -d' ' | cut -f1 -d'.'`
