# Based on https://github.com/ignitionrobotics/ign-physics/pull/282
diff -ru ign-physics-ignition-physics2_2.4.0.orig/src/CMakeLists.txt ign-physics-ignition-physics2_2.4.0/src/CMakeLists.txt
--- ign-physics-ignition-physics2_2.4.0.orig/src/CMakeLists.txt	2021-08-03 19:42:16.003202212 +0000
+++ ign-physics-ignition-physics2_2.4.0/src/CMakeLists.txt	2021-08-03 19:42:55.701482371 +0000
@@ -5,11 +5,8 @@
 target_link_libraries(${PROJECT_LIBRARY_TARGET_NAME}
   PUBLIC
     ${ignition-math${IGN_MATH_VER}_LIBRARIES}
-    ignition-plugin${IGN_PLUGIN_VER}::register)
-
-target_include_directories(${PROJECT_LIBRARY_TARGET_NAME}
-  SYSTEM PUBLIC
-    ${EIGEN3_INCLUDE_DIRS})
+    ignition-plugin${IGN_PLUGIN_VER}::register
+    Eigen3::Eigen)
 
 ign_build_tests(
   TYPE UNIT
