Configuration to build on CPU with MPI support.

Author: Loris Ercole <loris.ercole@epfl.ch>

diff --git a/config.git.mk b/config.mk
similarity index 93%
rename from config.git.mk
rename to config.mk
index d354d8e..b6657b8 100644
--- a/config.git.mk
+++ b/config.mk
@@ -12,9 +12,9 @@
 #  - Start the simulation using `./poiseuille2d`
 
 # Compiler to use for C++ files, change to `mpic++` when using OpenMPI and GCC
-CXX             := g++
+CXX             := mpicxx
 # Compiler to use for C files (used for emebedded dependencies)
-CC              := gcc
+CC              := mpicc
 
 # Suggested optimized build flags for GCC, consult `config/` for further examples
 CXXFLAGS        := -O3 -Wall -march=native -mtune=native
@@ -34,7 +34,7 @@ LDFLAGS         :=
 # Parallelization mode, must be one of: OFF, MPI, OMP, HYBRID
 # Note that for MPI and HYBRID the compiler also needs to be adapted.
 # See e.g. `config/cpu_gcc_openmpi.mk`
-PARALLEL_MODE   := OFF
+PARALLEL_MODE   := MPI
 
 # optional MPI and OpenMP flags
 MPIFLAGS        :=
@@ -44,7 +44,7 @@ OMPFLAGS        := -fopenmp
 # Both CPU_SIMD and GPU_CUDA require system-specific adjustment of compiler flags.
 # See e.g. `config/cpu_simd_intel_mpi.mk` or `config/gpu_only.mk` for examples.
 # CPU_SISD must always be present.
-PLATFORMS       := CPU_SISD
+PLATFORMS       := CPU_SISD CPU_SIMD
 
 # Fundamental arithmetic data type
 # Common options are float or double
@@ -55,4 +55,4 @@ FLOATING_POINT_TYPE := double
 FEATURES        :=
 
 # Set to OFF if libz and tinyxml2 are provided by the system (optional)
-USE_EMBEDDED_DEPENDENCIES := ON
+USE_EMBEDDED_DEPENDENCIES := OFF
