--- OpenLB/olb-1.4r0/config.mk	2020-11-15 19:12:41.000000000 +0000
+++ config.mk	2022-01-26 16:49:30.263472000 +0000
@@ -23,9 +23,9 @@
 ###########################################################################
 ###########################################################################
 
-CXX             := g++
+#CXX             := g++
 #CXX             := icpc -D__aligned__=ignored
-#CXX             := mpiCC
+CXX             := mpiCC
 #CXX             := mpic++
 
 CC              := gcc                                          # necessary for zlib, for Intel use icc
@@ -50,8 +50,8 @@
 
 LDFLAGS         :=
 
-PARALLEL_MODE   := OFF
-#PARALLEL_MODE   := MPI
+#PARALLEL_MODE   := OFF
+PARALLEL_MODE   := MPI
 #PARALLEL_MODE   := OMP
 #PARALLEL_MODE   := HYBRID
 
