diff -ruN dgl.orig/CMakeLists.txt dgl/CMakeLists.txt
--- dgl.orig/CMakeLists.txt	2026-05-22 11:43:40.614277000 +0200
+++ dgl/CMakeLists.txt	2026-05-29 11:32:57.692994780 +0200
@@ -115,6 +115,12 @@
   endif(BUILD_CPP_TEST)
 endif()
 
+# Set these after the check for external libraries which would disable BUILD_CPP_TEST
+set(EXTERNAL_DLPACK_PATH "$ENV{EBROOTDLPACK}")
+set(EXTERNAL_DMLC_PATH "$ENV{EBROOTDMLCMINCORE}")
+set(EXTERNAL_DMLC_LIB_PATH "$ENV{EBROOTDMLCMINCORE}/lib")
+set(EXTERNAL_PHMAP_PATH "$ENV{EBROOTPARALLELMINHASHMAP}")
+
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 
 # Set optimization options for different build types.
@@ -445,14 +451,14 @@
 # Testing
 if(BUILD_CPP_TEST)
   message(STATUS "Build with unittest")
-  add_subdirectory(./third_party/googletest)
+  # add_subdirectory(./third_party/googletest)
   enable_testing()
   include_directories(${gtest_SOURCE_DIR}/include ${gtest_SOURCE_DIR})
   include_directories("include")
-  include_directories("third_party/dlpack/include")
-  include_directories("third_party/dmlc-core/include")
+  # include_directories("third_party/dlpack/include")
+  # include_directories("third_party/dmlc-core/include")
   include_directories("third_party/tsl_robin_map/include")
-  include_directories("third_party/libxsmm/include")
+  # include_directories("third_party/libxsmm/include")
   include_directories("third_party/pcg/include")
   file(GLOB_RECURSE TEST_SRC_FILES ${PROJECT_SOURCE_DIR}/tests/cpp/*.cc)
   add_executable(runUnitTests ${TEST_SRC_FILES})
