--- configure.orig	2022-06-14 20:23:46.344204063 -0400
+++ configure	2022-06-15 10:59:14.344338860 -0400
@@ -180,6 +180,10 @@
         HPX_CONF_DIR=${HPX_CONF_DIR_INPLACE}
         INTERACTIVE=0
 	;;
+      --prefix=*)
+        echo "Delibarately trick EB hehe. Stubbornly ignore given prefix."
+        shift
+    ;;
       *)
         echo "Unknown option: $i"
 	echoLn "\n${usage}\n\n${usage2}\n\n"
