From 518b6feebf34c2dcc54cbab872dfc71d03a53c87 Mon Sep 17 00:00:00 2001
From: Atsushi Togo <atz.togo@gmail.com>
Date: Tue, 17 Jun 2025 15:59:25 +0900
Subject: [PATCH] Update phonopy version dependency

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index fd4a1ff4..50cc085f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,7 +16,7 @@ dependencies = [
     "matplotlib",
     "h5py",
     "spglib",
-    "phonopy>=2.39,<2.40",
+    "phonopy>=2.40,<2.41",
 ]
 license = "BSD-3-Clause"
 license-files = ["LICENSE"]
