diff -ruN walberla-v6.1.orig/CMakeLists.txt walberla-v6.1/CMakeLists.txt
--- walberla-v6.1.orig/CMakeLists.txt	2023-10-17 13:06:02.619565000 +0200
+++ walberla-v6.1/CMakeLists.txt	2023-10-26 19:28:03.622447001 +0200
@@ -1313,8 +1313,9 @@
 configure_file ( src/waLBerlaDefinitions.in.h
                  src/waLBerlaDefinitions.h    )
 
-install( FILES ${walberla_BINARY_DIR}/src/waLBerlaDefinitions.h DESTINATION walberla/ )
-
+install( FILES ${walberla_BINARY_DIR}/src/waLBerlaDefinitions.h DESTINATION . )
+install( DIRECTORY ${walberla_BINARY_DIR}/apps/benchmarks/ DESTINATION benchmarks/ )
+install( DIRECTORY ${walberla_BINARY_DIR}/apps/tutorials/ DESTINATION tutorials/ )
 
 # test
 if ( WALBERLA_BUILD_TESTS )
