diff -ruN walberla-v6.1.orig/CMakeLists.txt walberla-v6.1/CMakeLists.txt
--- walberla-v6.1.orig/CMakeLists.txt	2023-10-17 13:06:02.619565000 +0200
+++ walberla-v6.1/CMakeLists.txt	2023-10-18 14:47:35.047985995 +0200
@@ -1314,7 +1314,7 @@
                  src/waLBerlaDefinitions.h    )
 
 install( FILES ${walberla_BINARY_DIR}/src/waLBerlaDefinitions.h DESTINATION walberla/ )
-
+install( DIRECTORY ${walberla_BINARY_DIR}/apps/ DESTINATION walberla/ )
 
 # test
 if ( WALBERLA_BUILD_TESTS )
