Author: Julia Morillo, 2026-03-09

Avoid the use of -march=native for RISC-V

--- tests/simd/CMakeLists.txt.orig	2026-03-09 12:21:57.316405449 +0100
+++ tests/simd/CMakeLists.txt	2026-03-09 12:25:59.500350061 +0100
@@ -12,6 +12,10 @@
   set( MarchNativeString "-march=native" )
 endif()
 
+if ( CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "riscv64" )
+  set( MarchNativeString "-march=rv64gc" )
+endif()
+
 waLBerla_compile_test( NAME   AVX2_AVX_Equivalence FILES SIMD_Equivalence.cpp  )
 set_property         ( TARGET AVX2_AVX_Equivalence PROPERTY COMPILE_FLAGS "-DIS0_AVX2 -DIS1_AVX" )
 waLBerla_execute_test( NAME   AVX2_AVX_Equivalence )
