# Fix link error by hard-coding openblas
--- caviar.orig/CAVIAR-C++/Makefile	2020-08-24 09:21:32.000000000 +0000
+++ caviar/CAVIAR-C++/Makefile	2020-08-24 09:36:11.868606878 +0000
@@ -1,7 +1,7 @@
 CC=g++
 DIC=$(PWD)
 CFLAGS=-c -Wall -g  -I $(DIC) 
-LDFLAGS= -I $(DIC)/armadillo/include/ -DARMA_DONT_USE_WRAPPER -llapack -lblas -lgslcblas  -lgsl
+LDFLAGS= -I $(DIC)/armadillo/include/ -DARMA_DONT_USE_WRAPPER -lopenblas -lgslcblas  -lgsl
 SOURCES1=caviar.cpp PostCal.cpp Util.cpp TopKSNP.cpp 
 SOURCES2=ecaviar.cpp PostCal.cpp Util.cpp
 SOURCES3=setcaviar.cpp PostCal.cpp Util.cpp
