diff -ruN bagel-1.2.2.org/src/util/serialization.h bagel-1.2.2/src/util/serialization.h
--- bagel-1.2.2.org/src/util/serialization.h	2023-06-29 07:53:18.377601798 +0000
+++ bagel-1.2.2/src/util/serialization.h	2023-06-29 07:56:21.190019649 +0000
@@ -33,6 +33,7 @@
 #include <type_traits>
 #include <boost/version.hpp>
 #include <boost/serialization/serialization.hpp>
+#include <boost/archive/basic_archive.hpp>
 #include <boost/serialization/complex.hpp>
 #include <boost/serialization/bitset.hpp>
 #include <boost/serialization/string.hpp>
