diff -ru RHPC_2.0.orig/src/repast_hpc/SharedNetwork.h RHPC_2.0/src/repast_hpc/SharedNetwork.h
--- RHPC_2.0.orig/src/repast_hpc/SharedNetwork.h	2018-11-19 20:47:06.278274245 +0000
+++ RHPC_2.0/src/repast_hpc/SharedNetwork.h	2018-11-19 20:47:47.127895659 +0000
@@ -241,10 +241,10 @@
 						removeMap.find(emIter->first);
 				if (iter == removeMap.end()) {
 					std::vector<std::pair<AgentId, AgentId> > vec;
-					vec.push_back(std::make_pair<AgentId, AgentId>(edge->source()->getId(), edge->target()->getId()));
+					vec.push_back(std::make_pair(edge->source()->getId(), edge->target()->getId()));
 					removeMap[emIter->first] = vec;
 				} else {
-					iter->second.push_back(std::make_pair<AgentId, AgentId>(edge->source()->getId(),
+					iter->second.push_back(std::make_pair(edge->source()->getId(),
 							edge->target()->getId()));
 				}
 				edgeIter = edges->erase(edgeIter);
