diff -ru RHPC_2.0.orig/INSTALLATION/Makefile RHPC_2.0/INSTALLATION/Makefile
--- RHPC_2.0.orig/INSTALLATION/Makefile	2018-01-19 21:21:30.852690862 +0000
+++ RHPC_2.0/INSTALLATION/Makefile	2018-01-19 21:29:46.931821434 +0000
@@ -2,13 +2,13 @@
 # VARIABLES (Supply values for these; for definitions and examples, see INSTALL)
 CXX=mpicxx
-CXXLD=mpicxx -static
-BOOST_INCLUDE_DIR=./
-BOOST_LIB_DIR=./
-BOOST_INFIX=-mt-s
-NETCDF_INCLUDE_DIR=./
-NETCDF_LIB_DIR=./
-CURL_INCLUDE_DIR=./
-CURL_LIB_DIR=./
+CXXLD=mpicxx 
+BOOST_INCLUDE_DIR=$(EBROOTBOOST)/include
+BOOST_LIB_DIR=$(EBROOTBOOST)/lib
+BOOST_INFIX=-mt
+NETCDF_INCLUDE_DIR=$(EBROOTNETCDF)/include
+NETCDF_LIB_DIR=$(EBROOTNETCDF)/lib
+CURL_INCLUDE_DIR=$(NIXUSER_PROFILE)/include
+CURL_LIB_DIR=$(NIXUSER_PROFILE)/lib
 
 # DIRECTORIES 
 SRC_DIR=../src
diff -ru RHPC_2.0.orig/src/repast_hpc/logger.cpp RHPC_2.0/src/repast_hpc/logger.cpp
--- RHPC_2.0.orig/src/repast_hpc/logger.cpp	2018-01-19 21:21:31.312686340 +0000
+++ RHPC_2.0/src/repast_hpc/logger.cpp	2018-01-19 21:22:21.711190956 +0000
@@ -679,6 +679,7 @@
 			create_appender_bidx(lexer.key(), lexer.value());
 			break;
 		case ERRORT:
+			error = lexer.error();
 			error_warn();
 			break;
 		default:
