diff -ru cufflinks-1.3.0/configure cufflinks-1.3.0_patched/configure
--- cufflinks-1.3.0/configure	2012-01-02 13:36:44.000000000 +0000
+++ cufflinks-1.3.0_patched/configure	2015-01-27 18:32:24.810282390 +0000
@@ -6386,7 +6386,8 @@
 CFLAGS="${generic_CFLAGS} ${ext_CFLAGS} ${user_CFLAGS} ${debug_CFLAGS} ${OPENMP_CFLAGS}"
 CXXFLAGS="$CFLAGS"
 CXXFLAGS="$CXXFLAGS $BOOST_CPPFLAGS $BAM_CPPFLAGS"
-LDFLAGS="$ext_LDFLAGS"
+user_LDFLAGS="$LDFLAGS"
+LDFLAGS="${ext_LDFLAGS} ${user_LDFLAGS}"
 
 # Checks for structures/functions that can be used to determine system memory
 echo "$as_me:$LINENO: checking for struct sysinfo.totalram" >&5
diff -ru cufflinks-1.3.0/src/Makefile.in cufflinks-1.3.0_patched/src/Makefile.in
--- cufflinks-1.3.0/src/Makefile.in	2012-01-02 13:36:43.000000000 +0000
+++ cufflinks-1.3.0_patched/src/Makefile.in	2015-01-27 18:20:02.944386541 +0000
@@ -438,24 +438,24 @@
 #	(echo '#!$(PYTHON)'; sed '/^#!/d' $<) > $@
 cufflinks_SOURCES = cufflinks.cpp  
 cufflinks_LDADD = libcufflinks.a libgc.a $(BOOST_THREAD_LIB) $(BAM_LIB) 
-cufflinks_LDFLAGS = $(BOOST_LDFLAGS) $(BAM_LDFLAGS) #$(ZLIB_LDFLAGS) 
+cufflinks_LDFLAGS = $(LDFLAGS) $(BOOST_LDFLAGS) $(BAM_LDFLAGS) #$(ZLIB_LDFLAGS) 
 cuffcompare_SOURCES = cuffcompare.cpp
 cuffcompare_LDADD = libgc.a
 gffread_SOURCES = gffread.cpp
 gffread_LDADD = libgc.a
 cuffdiff_SOURCES = cuffdiff.cpp
 cuffdiff_LDADD = libcufflinks.a libgc.a $(BOOST_THREAD_LIB) $(BAM_LIB)
-cuffdiff_LDFLAGS = $(BOOST_LDFLAGS) $(BAM_LDFLAGS) 
+cuffdiff_LDFLAGS = $(LDFLAGS) $(BOOST_LDFLAGS) $(BAM_LDFLAGS) 
 gtf_to_sam_SOURCES = gtf_to_sam.cpp
 gtf_to_sam_LDADD = libcufflinks.a libgc.a $(BOOST_THREAD_LIB) $(BAM_LIB)
-gtf_to_sam_LDFLAGS = $(BOOST_LDFLAGS) $(BAM_LDFLAGS) 
+gtf_to_sam_LDFLAGS = $(LDFLAGS) $(BOOST_LDFLAGS) $(BAM_LDFLAGS) 
 
 #cuffcluster_SOURCES = cuffcluster.cpp
 #cuffcluster_LDADD = libcufflinks.a libgc.a $(BOOST_THREAD_LIB) $(BAM_LIB)
 #cuffcluster_LDFLAGS =  $(BOOST_LDFLAGS) $(BAM_LDFLAGS) 
 compress_gtf_SOURCES = compress_gtf.cpp
 compress_gtf_LDADD = libcufflinks.a libgc.a $(BOOST_THREAD_LIB) $(BAM_LIB)
-compress_gtf_LDFLAGS = $(BOOST_LDFLAGS) $(BAM_LDFLAGS) 
+compress_gtf_LDFLAGS = $(LDFLAGS) $(BOOST_LDFLAGS) $(BAM_LDFLAGS) 
 all: all-am
 
 .SUFFIXES:
