diff -ru zlib-1.2.5.orig/configure zlib-1.2.5/configure
--- zlib-1.2.5.orig/configure	2010-04-20 06:15:19.000000000 +0200
+++ zlib-1.2.5/configure	2012-10-28 15:07:12.602164801 +0100
@@ -233,7 +233,7 @@
   echo Checking for shared library support...
   # we must test in two steps (cc then ld), required at least on SunOS 4.x
   if test "`($CC -w -c $SFLAGS $test.c) 2>&1`" = "" &&
-     test "`($LDSHARED $SFLAGS -o $test$shared_ext $test.o) 2>&1`" = ""; then
+     test "`($LDSHARED $SFLAGS -o $test$shared_ext $test.o) 2>&1 | grep -v 'icc: command line warning'`" = ""; then
     echo Building shared library $SHAREDLIBV with $CC.
   elif test -z "$old_cc" -a -z "$old_cflags"; then
     echo No shared library support.
