# Author: julianmorillo
# Description: Include cstdint
--- metaeuk-6-a5d39d9/lib/mmseqs/src/commons/Util.h.orig	2025-03-10 11:38:28.354675930 +0100
+++ metaeuk-6-a5d39d9/lib/mmseqs/src/commons/Util.h	2025-03-10 11:42:18.404538368 +0100
@@ -8,6 +8,7 @@
 #include <limits>
 #include <map>
 #include "MMseqsMPI.h"
+#include <cstdint>
 
 #ifndef EXIT
 #define EXIT(exitCode) do { int __status = (exitCode); std::cerr.flush(); std::cout.flush(); exit(__status); } while(0)
