diff --git a/src/openms/include/OpenMS/DATASTRUCTURES/LPWrapper.h b/src/openms/include/OpenMS/DATASTRUCTURES/LPWrapper.h
index 83c81d78a2..a0fad23bb1 100644
--- a/src/openms/include/OpenMS/DATASTRUCTURES/LPWrapper.h
+++ b/src/openms/include/OpenMS/DATASTRUCTURES/LPWrapper.h
@@ -51,15 +51,8 @@ class CoinModel;
 #define GLP_PROB_DEFINED
 // depending on the glpk version
 // define glp_prob as forward or struct
-#if OPENMS_GLPK_VERSION_MINOR < 48
-typedef struct
-{
-  double _opaque_prob[100];
-} glp_prob;
-#else
 class glp_prob;
 #endif
-#endif
 
 namespace OpenMS
 {
