--- contrafold.orig/src/Makefile	2008-08-14 02:08:30.000000000 +0200
+++ contrafold/src/Makefile	2014-02-04 13:51:29.041589171 +0100
@@ -1,6 +1,6 @@
 CXX = g++
 
-CXXFLAGS = -O3 -DNDEBUG -W -pipe -Wundef -Winline --param large-function-growth=100000 -Wall
+CXXFLAGS = -O3 -DNDEBUG -W -pipe -Wundef -Winline --param large-function-growth=100000 -Wall -fpermissive
 LINKFLAGS = -lm
 GDLINKFLAGS = -lgd -lpng
 
--- contrafold.orig/src/Utilities.cpp	2008-08-14 02:08:31.000000000 +0200
+++ contrafold/src/Utilities.cpp	2014-02-04 14:09:09.082650184 +0100
@@ -2,6 +2,7 @@
 // Utilities.cpp
 //////////////////////////////////////////////////////////////////////
 
+#include <limits.h>
 #include "Utilities.hpp"
 
 bool toggle_error = false;
