Adapt immutabledict-4.2.2 to older poetry version

Åke Sandgren, 2025-12-01
diff -ru immutabledict-4.2.2.orig/pyproject.toml immutabledict-4.2.2/pyproject.toml
--- immutabledict-4.2.2.orig/pyproject.toml	1970-01-01 01:00:00.000000000 +0100
+++ immutabledict-4.2.2/pyproject.toml	2025-12-01 09:03:40.944511510 +0100
@@ -29,6 +29,12 @@
 include = [
     { path = "tests", format = "sdist" }
 ]
+name = "immutabledict"
+version = "4.2.2"
+description = "Immutable wrapper around dictionaries (a fork of frozendict)"
+authors = [
+    "Corentin Garcia",
+]
 
 [project.urls]
 homepage = "https://github.com/corenting/immutabledict"
