diff -Nru spark-2.1.1.orig/conf/spark-env.sh spark-2.1.1/conf/spark-env.sh
--- spark-2.1.1.orig/conf/spark-env.sh	1970-01-01 00:00:00.000000000 +0000
+++ spark-2.1.1/conf/spark-env.sh	2017-06-29 18:37:12.084342798 +0000
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+export SPARK_LOG_DIR="$HOME/.spark/2.1.1/log"
+export SPARK_WORKER_DIR="$SCRATCH/spark/2.1.1/$(hostname)"
+if [[ -z "$SPARK_CONF_DIR" ]]; then
+    export SPARK_CONF_DIR="$HOME/.spark/2.1.1/conf"
+    if [[ -f $SPARK_CONF_DIR/spark-env.sh ]]; then
+        . $SPARK_CONF_DIR/spark-env.sh
+    fi
+fi
+
diff -Nru spark-2.1.1.orig/dev/make-distribution.sh spark-2.1.1/dev/make-distribution.sh
--- spark-2.1.1.orig/dev/make-distribution.sh	2017-06-29 14:52:08.681829448 +0000
+++ spark-2.1.1/dev/make-distribution.sh	2017-06-29 15:34:36.340057826 +0000
@@ -241,8 +241,7 @@
 fi
 
 # Copy other things
-mkdir "$DISTDIR"/conf
-cp "$SPARK_HOME"/conf/*.template "$DISTDIR"/conf
+cp -r "$SPARK_HOME"/conf "$DISTDIR"/conf
 cp "$SPARK_HOME/README.md" "$DISTDIR"
 cp -r "$SPARK_HOME/bin" "$DISTDIR"
 cp -r "$SPARK_HOME/python" "$DISTDIR"
