From e3bec5a44cbe20aa4dd2470855943c1572e5feac Mon Sep 17 00:00:00 2001
From: Viktor Rehnberg <viktor.rehnberg@gmail.com>
Date: Fri, 1 Nov 2024 15:01:51 +0000
Subject: [PATCH] Make Bar dataclass immutable

---
 tests/implicitron/test_config.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/implicitron/test_config.py b/tests/implicitron/test_config.py
index 471451b..2ae4f4a 100644
--- a/tests/implicitron/test_config.py
+++ b/tests/implicitron/test_config.py
@@ -246,7 +246,7 @@ class TestConfig(unittest.TestCase):
 
         enable_get_default_args(Foo)
 
-        @dataclass()
+        @dataclass(frozen=True)
         class Bar:
             aa: int = 9
             bb: int = 9
-- 
2.39.3

