--- express-1.5.1-src.orig/src/CMakeLists.txt	2013-12-08 20:09:18.000000000 +0100
+++ express-1.5.1-src/src/CMakeLists.txt	2014-03-27 23:09:17.014311307 +0100
@@ -16,7 +16,7 @@
 if(WIN32)
   set(LIBRARIES ${LIBRARIES} "${CMAKE_CURRENT_SOURCE_DIR}/../bamtools/lib/libbamtools.lib" "${CMAKE_CURRENT_SOURCE_DIR}/../win_build/zlibd.lib")
 else(WIN32)
-  set(LIBRARIES ${LIBRARIES} "${CMAKE_CURRENT_SOURCE_DIR}/../bamtools/lib/libbamtools.a" "pthread")
+  set(LIBRARIES ${LIBRARIES} "$ENV{EBROOTBAMTOOLS}/lib/libbamtools.a" "pthread")
 endif(WIN32)
 
 if (PROTOBUF_FOUND)
