--- generated_code/yateto/codegen/fused_gemms/factory.py.orig	2024-07-18 14:27:36.981043324 +1200
+++ generated_code/yateto/codegen/fused_gemms/factory.py	2024-07-18 14:27:56.580958672 +1200
@@ -5,7 +5,7 @@
   if gb_spec:
     gb = gb_spec.loader.load_module()
     version = pkg_resources.get_distribution('chainforge').version
-    if not version == '0.0.3':
+    if not version >= '0.0.3':
       raise RuntimeError(f'chainforge version 0.0.3 is requried. Found: {version}. Please, update using `pip3`')
 
     from .external_generator import FusedGemms
